{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "execution_event_id": "c062db36-c8b1-4c6b-8abc-0154a2423d7a",
    "last_executed_text": "%load_ext autoreload\n%autoreload 2\n\nfolder = '/home/olga/data_sm/kmer-hashing/classify_coding_vs_noncoding/'",
    "persistent_id": "71299be4-a00f-4638-92a2-ea174ddf208d"
   },
   "outputs": [],
   "source": [
    "%load_ext autoreload\n",
    "%autoreload 2\n",
    "\n",
    "folder = '/home/olga/data_sm/kmer-hashing/classify_coding_vs_noncoding/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "execution_event_id": "69499170-c142-47ce-b657-51928aa9ab6d",
    "last_executed_text": "cd $folder",
    "persistent_id": "1d547cf2-254f-4fe2-9bf0-5b7e39be4376"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/mnt/data_sm/olga/kmer-hashing/classify_coding_vs_noncoding\n"
     ]
    }
   ],
   "source": [
    "cd $folder"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "execution_event_id": "d5f5d31e-86f0-4784-9173-0a728e9cfb8b",
    "last_executed_text": "ls",
    "persistent_id": "a30b0ea1-cbfb-4de5-bce5-329cfeb8e50c"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;31mHomo_sapiens.GRCh38.cds.all.fa.gz\u001b[0m\n",
      "\u001b[01;31mHomo_sapiens.GRCh38.ncrna.fa.gz\u001b[0m\n",
      "\u001b[01;31mHomo_sapiens.GRCh38.pep.all.fa.gz\u001b[0m\n",
      "Homo_sapiens.GRCh38.pep.subset.fa\n",
      "\u001b[01;31mHomo_sapiens.GRCh38.pep.subset.fa.gz\u001b[0m\n",
      "human_cds_k15.nodegraph\n",
      "human_cds_k15.nodegraph.info\n",
      "human_cds_k17.nodegraph\n",
      "human_cds_k17.nodegraph.info\n",
      "human_cds_k19.nodegraph\n",
      "human_cds_k19.nodegraph.info\n",
      "human_cds_k21.nodegraph\n",
      "human_cds_k21.nodegraph.info\n",
      "human_cds_k23.nodegraph\n",
      "human_cds_k23.nodegraph.info\n",
      "human_cds_k25.nodegraph\n",
      "human_cds_k25.nodegraph.info\n",
      "human_cds_k27.nodegraph\n",
      "human_cds_k27.nodegraph.info\n",
      "human_cds_k31.nodegraph\n",
      "human_cds_k31.nodegraph.info\n",
      "human_ncrna_k15.nodegraph\n",
      "human_ncrna_k15.nodegraph.info\n",
      "human_ncrna_k15.nodegraph.tagset\n",
      "human_ncrna_k17.nodegraph\n",
      "human_ncrna_k17.nodegraph.info\n",
      "human_ncrna_k19.nodegraph\n",
      "human_ncrna_k19.nodegraph.info\n",
      "human_ncrna_k21.nodegraph\n",
      "human_ncrna_k21.nodegraph.info\n",
      "human_ncrna_k23.nodegraph\n",
      "human_ncrna_k23.nodegraph.info\n",
      "human_ncrna_k25.nodegraph\n",
      "human_ncrna_k25.nodegraph.info\n",
      "human_ncrna_k27.nodegraph\n",
      "human_ncrna_k27.nodegraph.info\n",
      "human_ncrna_k31.nodegraph\n",
      "human_ncrna_k31.nodegraph.info\n",
      "\u001b[01;31mSRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz\u001b[0m\n",
      "SRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.coding_peptides.fasta\n",
      "\u001b[01;31mSRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.fq.gz\u001b[0m\n",
      "SRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.low_complexity_nucleotides.fasta\n",
      "SRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.low_complexity_peptides.fasta\n",
      "SRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.noncoding_nucleotides.fasta\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "execution_event_id": "08771cb9-218b-4722-be2c-e2b8c0fe62d4",
    "last_executed_text": "ls -lha",
    "persistent_id": "9c72c9dd-4e3a-4661-ab4f-a9b7bf1d822a"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 239G\n",
      "drwxrwxr-x 2 olga olga 4.0K Oct 13 16:57 \u001b[0m\u001b[01;34m.\u001b[0m/\n",
      "drwxrwxr-x 5 olga olga 4.0K Sep 14 08:39 \u001b[01;34m..\u001b[0m/\n",
      "-rw-rw-r-- 1 olga olga  21M Sep 14 06:46 \u001b[01;31mHomo_sapiens.GRCh38.cds.all.fa.gz\u001b[0m\n",
      "-rw-rw-r-- 1 olga olga  16M Sep 14 06:46 \u001b[01;31mHomo_sapiens.GRCh38.ncrna.fa.gz\u001b[0m\n",
      "-rw-rw-r-- 1 olga olga  14M Sep 14 06:46 \u001b[01;31mHomo_sapiens.GRCh38.pep.all.fa.gz\u001b[0m\n",
      "-rw-rw-r-- 1 olga olga 2.6M Oct 13 17:26 Homo_sapiens.GRCh38.pep.subset.fa\n",
      "-rw-rw-r-- 1 olga olga 489K Oct 13 17:26 \u001b[01;31mHomo_sapiens.GRCh38.pep.subset.fa.gz\u001b[0m\n",
      "-rw-rw-r-- 1 olga olga  15G Sep 14 07:11 human_cds_k15.nodegraph\n",
      "-rw-rw-r-- 1 olga olga   65 Sep 14 07:11 human_cds_k15.nodegraph.info\n",
      "-rw-rw-r-- 1 olga olga  15G Sep 14 07:22 human_cds_k17.nodegraph\n",
      "-rw-rw-r-- 1 olga olga   65 Sep 14 07:22 human_cds_k17.nodegraph.info\n",
      "-rw-rw-r-- 1 olga olga  15G Sep 14 07:30 human_cds_k19.nodegraph\n",
      "-rw-rw-r-- 1 olga olga   65 Sep 14 07:30 human_cds_k19.nodegraph.info\n",
      "-rw-rw-r-- 1 olga olga  15G Sep 14 07:39 human_cds_k21.nodegraph\n",
      "-rw-rw-r-- 1 olga olga   65 Sep 14 07:39 human_cds_k21.nodegraph.info\n",
      "-rw-rw-r-- 1 olga olga  15G Sep 14 07:47 human_cds_k23.nodegraph\n",
      "-rw-rw-r-- 1 olga olga   65 Sep 14 07:47 human_cds_k23.nodegraph.info\n",
      "-rw-rw-r-- 1 olga olga  15G Sep 14 07:55 human_cds_k25.nodegraph\n",
      "-rw-rw-r-- 1 olga olga   65 Sep 14 07:55 human_cds_k25.nodegraph.info\n",
      "-rw-rw-r-- 1 olga olga  15G Sep 14 08:03 human_cds_k27.nodegraph\n",
      "-rw-rw-r-- 1 olga olga   65 Sep 14 08:03 human_cds_k27.nodegraph.info\n",
      "-rw-rw-r-- 1 olga olga  15G Sep 14 08:12 human_cds_k31.nodegraph\n",
      "-rw-rw-r-- 1 olga olga   65 Sep 14 08:12 human_cds_k31.nodegraph.info\n",
      "-rw-rw-r-- 1 olga olga  15G Sep 14 07:18 human_ncrna_k15.nodegraph\n",
      "-rw-rw-r-- 1 olga olga   65 Sep 14 07:18 human_ncrna_k15.nodegraph.info\n",
      "-rw-rw-r-- 1 olga olga 5.7M Sep 14 07:15 human_ncrna_k15.nodegraph.tagset\n",
      "-rw-rw-r-- 1 olga olga  15G Sep 14 07:26 human_ncrna_k17.nodegraph\n",
      "-rw-rw-r-- 1 olga olga   65 Sep 14 07:26 human_ncrna_k17.nodegraph.info\n",
      "-rw-rw-r-- 1 olga olga  15G Sep 14 07:34 human_ncrna_k19.nodegraph\n",
      "-rw-rw-r-- 1 olga olga   65 Sep 14 07:34 human_ncrna_k19.nodegraph.info\n",
      "-rw-rw-r-- 1 olga olga  15G Sep 14 07:42 human_ncrna_k21.nodegraph\n",
      "-rw-rw-r-- 1 olga olga   65 Sep 14 07:43 human_ncrna_k21.nodegraph.info\n",
      "-rw-rw-r-- 1 olga olga  15G Sep 14 07:51 human_ncrna_k23.nodegraph\n",
      "-rw-rw-r-- 1 olga olga   65 Sep 14 07:51 human_ncrna_k23.nodegraph.info\n",
      "-rw-rw-r-- 1 olga olga  15G Sep 14 07:59 human_ncrna_k25.nodegraph\n",
      "-rw-rw-r-- 1 olga olga   65 Sep 14 07:59 human_ncrna_k25.nodegraph.info\n",
      "-rw-rw-r-- 1 olga olga  15G Sep 14 08:07 human_ncrna_k27.nodegraph\n",
      "-rw-rw-r-- 1 olga olga   65 Sep 14 08:07 human_ncrna_k27.nodegraph.info\n",
      "-rw-rw-r-- 1 olga olga  15G Sep 14 08:16 human_ncrna_k31.nodegraph\n",
      "-rw-rw-r-- 1 olga olga   65 Sep 14 08:16 human_ncrna_k31.nodegraph.info\n",
      "-rw-r--r-- 1 olga olga  52M Sep 18 05:33 \u001b[01;31mSRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz\u001b[0m\n",
      "-rw-rw-r-- 1 olga olga  26K Oct 11 23:03 SRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.coding_peptides.fasta\n",
      "-rw-r--r-- 1 olga olga  72K Sep 18 05:33 \u001b[01;31mSRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.fq.gz\u001b[0m\n",
      "-rw-rw-r-- 1 olga olga  132 Oct 11 23:03 SRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.low_complexity_nucleotides.fasta\n",
      "-rw-rw-r-- 1 olga olga    0 Oct 11 22:41 SRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.low_complexity_peptides.fasta\n",
      "-rw-rw-r-- 1 olga olga  91K Oct 11 23:03 SRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.noncoding_nucleotides.fasta\n"
     ]
    }
   ],
   "source": [
    "ls -lha"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "execution_event_id": "f37b8d30-6d34-4007-bedd-9324c2d0ee5e",
    "last_executed_text": "cat human_ncrna_k17.nodegraph.info",
    "persistent_id": "39d48808-d467-4d42-a862-97485096088c"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "30178100 unique k-mers\n",
      "false positive rate estimated to be 0.000\n"
     ]
    }
   ],
   "source": [
    "cat human_ncrna_k17.nodegraph.info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "execution_event_id": "2577571b-fe1d-4b01-acbf-927a9a571f47",
    "last_executed_text": "import gzip\nfrom io import StringIO\nfrom pprint import pprint\n\nfrom khmer import Nodegraph\nimport matplotlib.pyplot as plt\nimport screed\nfrom sourmash._minhash import hash_murmur\nfrom khmer.khmer_args import calculate_graphsize\nimport pandas as pd\nfrom sourmash.logging import notify\nimport seaborn as sns\nfrom sklearn import metrics\nfrom tqdm import tqdm\nfrom Bio.Seq import Seq\nfrom Bio import SeqIO\n\n# make all warnings error to get a stack trace\nimport warnings\nwarnings.filterwarnings(\"default\")\n\npd.options.display.max_colwidth = 100\n\nfrom khtools.compare_kmer_content import kmerize, hpize, dayhoffize\n",
    "persistent_id": "56e48bd4-fcc6-4d93-b4e3-0984be1fc1d5"
   },
   "outputs": [],
   "source": [
    "import gzip\n",
    "from io import StringIO\n",
    "from pprint import pprint\n",
    "\n",
    "from khmer import Nodegraph\n",
    "import matplotlib.pyplot as plt\n",
    "import screed\n",
    "from sourmash._minhash import hash_murmur\n",
    "from khmer.khmer_args import calculate_graphsize\n",
    "import pandas as pd\n",
    "from sourmash.logging import notify\n",
    "import seaborn as sns\n",
    "from sklearn import metrics\n",
    "from tqdm import tqdm\n",
    "from Bio.Seq import Seq\n",
    "from Bio import SeqIO\n",
    "\n",
    "# make all warnings error to get a stack trace\n",
    "import warnings\n",
    "warnings.filterwarnings(\"default\")\n",
    "\n",
    "pd.options.display.max_colwidth = 100\n",
    "\n",
    "from khtools.compare_kmer_content import kmerize, hpize, dayhoffize\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "execution_event_id": "593de35b-b769-4a97-8e84-bdf33d45db5f",
    "last_executed_text": "from khtools.partition_reads import score_reads, six_frame_translation_no_stops, compute_low_complexity\nfrom khtools.bloom_filter import make_peptide_bloom_filter",
    "persistent_id": "1e54230e-ceec-42fc-afa6-7b32e219378b"
   },
   "outputs": [],
   "source": [
    "from khtools.partition_reads import score_reads, six_frame_translation_no_stops, compute_low_complexity\n",
    "from khtools.bloom_filter import make_peptide_bloom_filter"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "execution_event_id": "e2d214d9-529c-4f41-b87b-689aee4bb54f",
    "last_executed_text": "%%time\npeptide_ksize = 7\n\npeptide_graph = make_peptide_bloom_filter(\"Homo_sapiens.GRCh38.pep.all.fa.gz\", peptide_ksize)",
    "persistent_id": "be06ded1-d7d4-4f21-9961-3aed5ff30cbf"
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/screed/openscreed.py:39: ResourceWarning: unclosed file <_io.BufferedReader name='Homo_sapiens.GRCh38.pep.all.fa.gz'>\n",
      "  self.iter_fn = self.open_reader(filename, *args, **kwargs)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 37.9 s, sys: 2.71 s, total: 40.7 s\n",
      "Wall time: 40.7 s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "peptide_ksize = 7\n",
    "\n",
    "peptide_graph = make_peptide_bloom_filter(\"Homo_sapiens.GRCh38.pep.all.fa.gz\", peptide_ksize)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "execution_event_id": "44ba989a-c413-41fd-90ad-ac2e5840ef40",
    "last_executed_text": "peptide_graph.n_unique_kmers()",
    "persistent_id": "f6bb84e5-c5f5-4d4b-90fb-724792d5d78d"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "10841698"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "peptide_graph.n_unique_kmers()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "execution_event_id": "eae6ceba-3755-4689-ae6d-d2db9817c29f",
    "last_executed_text": "list(peptide_graph.tags())",
    "persistent_id": "b32b1264-231a-4b32-9229-c49de4f17595"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list(peptide_graph.tags())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "execution_event_id": "23d8bbd0-edf6-4a8d-81e0-166573ce7973",
    "last_executed_text": "peptide_graph.add_tag?",
    "persistent_id": "5e420857-1689-4657-983e-7e72891cf0ef"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\u001b[0;31mDocstring:\u001b[0m\n",
       "Hashgraph.add_tag(self, kmer)\n",
       "Add a k-mer to the tagset.\n",
       "\u001b[0;31mType:\u001b[0m      builtin_function_or_method\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "peptide_graph.add_tag?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "execution_event_id": "149575ae-ec74-4e52-8425-63edd24dd85d",
    "last_executed_text": "dir(peptide_graph)",
    "persistent_id": "d28dc029-70e7-4a3f-9455-6ed6882da2c9"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['__class__',\n",
       " '__delattr__',\n",
       " '__dir__',\n",
       " '__doc__',\n",
       " '__eq__',\n",
       " '__format__',\n",
       " '__ge__',\n",
       " '__getattribute__',\n",
       " '__gt__',\n",
       " '__hash__',\n",
       " '__init__',\n",
       " '__init_subclass__',\n",
       " '__le__',\n",
       " '__lt__',\n",
       " '__ne__',\n",
       " '__new__',\n",
       " '__pyx_vtable__',\n",
       " '__reduce__',\n",
       " '__reduce_ex__',\n",
       " '__repr__',\n",
       " '__setattr__',\n",
       " '__setstate__',\n",
       " '__sizeof__',\n",
       " '__str__',\n",
       " '__subclasshook__',\n",
       " '_kmer_type_error',\n",
       " '_validate_partitionmap',\n",
       " 'abundance_distribution',\n",
       " 'add',\n",
       " 'add_stop_tag',\n",
       " 'add_tag',\n",
       " 'assign_partition_id',\n",
       " 'calc_connected_graph_size',\n",
       " 'consume',\n",
       " 'consume_and_tag',\n",
       " 'consume_partitioned_fasta',\n",
       " 'consume_seqfile',\n",
       " 'consume_seqfile_and_tag',\n",
       " 'consume_seqfile_banding',\n",
       " 'consume_seqfile_banding_with_mask',\n",
       " 'consume_seqfile_with_mask',\n",
       " 'count',\n",
       " 'count_kmers_within_radius',\n",
       " 'count_partitions',\n",
       " 'divide_tags_into_subsets',\n",
       " 'do_subset_partition',\n",
       " 'extract_unique_paths',\n",
       " 'find_all_tags',\n",
       " 'find_all_tags_list',\n",
       " 'find_high_degree_nodes',\n",
       " 'find_spectral_error_positions',\n",
       " 'get',\n",
       " 'get_kmer_counts',\n",
       " 'get_kmer_hashes',\n",
       " 'get_kmer_hashes_as_hashset',\n",
       " 'get_kmers',\n",
       " 'get_max_count',\n",
       " 'get_median_count',\n",
       " 'get_min_count',\n",
       " 'get_partition_id',\n",
       " 'get_raw_tables',\n",
       " 'get_stop_tags',\n",
       " 'get_tags_and_positions',\n",
       " 'get_tags_for_sequence',\n",
       " 'get_tagset',\n",
       " 'get_use_bigcount',\n",
       " 'hash',\n",
       " 'hashsizes',\n",
       " 'iter_stop_tags',\n",
       " 'join_partitions',\n",
       " 'kmer_degree',\n",
       " 'ksize',\n",
       " 'load',\n",
       " 'load_partitionmap',\n",
       " 'load_stop_tags',\n",
       " 'load_tagset',\n",
       " 'median_at_least',\n",
       " 'merge_subset',\n",
       " 'merge_subset_from_disk',\n",
       " 'n_occupied',\n",
       " 'n_tables',\n",
       " 'n_tags',\n",
       " 'n_unique_kmers',\n",
       " 'neighbors',\n",
       " 'output_partitions',\n",
       " 'partition',\n",
       " 'print_stop_tags',\n",
       " 'print_tagset',\n",
       " 'repartition_largest_partition',\n",
       " 'reverse_hash',\n",
       " 'sanitize_kmer',\n",
       " 'sanitize_seq_kmer',\n",
       " 'save',\n",
       " 'save_partitionmap',\n",
       " 'save_stop_tags',\n",
       " 'save_tagset',\n",
       " 'set_partition_id',\n",
       " 'set_use_bigcount',\n",
       " 'tag_density',\n",
       " 'tags',\n",
       " 'traverse_linear_path',\n",
       " 'trim_below_abundance',\n",
       " 'trim_on_abundance',\n",
       " 'trim_on_stoptags',\n",
       " 'update']"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dir(peptide_graph)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "execution_event_id": "15a19f9b-3c70-4a46-8603-836193e9faa0",
    "last_executed_text": "%%time\n\nwarnings.filterwarnings(\"error\")\n\nreads = 'SRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz'\n\ncoding_scores = score_reads(reads, peptide_graph, peptide_ksize)",
    "persistent_id": "be06ded1-d7d4-4f21-9961-3aed5ff30cbf"
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Exception ignored in: <_io.FileIO name='SRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz' mode='rb' closefd=True>\n",
      "ResourceWarning: unclosed file <_io.BufferedReader name='SRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz'>\n",
      "795517it [07:23, 1794.13it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 7min 16s, sys: 10.9 s, total: 7min 26s\n",
      "Wall time: 7min 23s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "\n",
    "warnings.filterwarnings(\"error\")\n",
    "\n",
    "reads = 'SRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz'\n",
    "\n",
    "coding_scores = score_reads(reads, peptide_graph, peptide_ksize)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "execution_event_id": "04e328a1-1898-4ffd-9a91-993e7d4e53fb",
    "last_executed_text": "coding_scores.head()",
    "persistent_id": "b44eec0d-17d9-459a-8fac-49e78b25a50c"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>read_id</th>\n",
       "      <th>jaccard_in_peptide_db</th>\n",
       "      <th>n_kmers</th>\n",
       "      <th>classification</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SRR306838.1531 Ibis_Run100924_C3PO:6:1:15718:1062/1</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>2</td>\n",
       "      <td>low complexity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SRR306838.1617 Ibis_Run100924_C3PO:6:1:9036:1074/1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>non-coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SRR306838.1672 Ibis_Run100924_C3PO:6:1:15751:1077/1</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>low complexity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SRR306838.1938 Ibis_Run100924_C3PO:6:1:15775:1103/1</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>low complexity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SRR306838.2318 Ibis_Run100924_C3PO:6:1:15779:1141/1</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>7</td>\n",
       "      <td>non-coding</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                               read_id  jaccard_in_peptide_db  \\\n",
       "0  SRR306838.1531 Ibis_Run100924_C3PO:6:1:15718:1062/1                   -1.0   \n",
       "1   SRR306838.1617 Ibis_Run100924_C3PO:6:1:9036:1074/1                    0.0   \n",
       "2  SRR306838.1672 Ibis_Run100924_C3PO:6:1:15751:1077/1                   -1.0   \n",
       "3  SRR306838.1938 Ibis_Run100924_C3PO:6:1:15775:1103/1                   -1.0   \n",
       "4  SRR306838.2318 Ibis_Run100924_C3PO:6:1:15779:1141/1                   -1.0   \n",
       "\n",
       "   n_kmers  classification  \n",
       "0        2  low complexity  \n",
       "1        0      non-coding  \n",
       "2        1  low complexity  \n",
       "3        1  low complexity  \n",
       "4        7      non-coding  "
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coding_scores.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "execution_event_id": "923fc40f-808d-4f44-9d13-faefc5a4bec6",
    "last_executed_text": "coding_scores.shape",
    "persistent_id": "c381cc92-ba84-43f1-95f7-d19b70d547f5"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(795517, 4)"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coding_scores.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "execution_event_id": "7accd32b-900f-47af-b32a-0c4c30b49fe7",
    "last_executed_text": "coding_scores_not_low_complexity = coding_scores.query('jaccard_in_peptide_db >= 0')\nprint(coding_scores_not_low_complexity.shape)\ncoding_scores_not_low_complexity.head()",
    "persistent_id": "73b49c34-8379-4aa5-835e-57c54aced1fa"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(793969, 4)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>read_id</th>\n",
       "      <th>jaccard_in_peptide_db</th>\n",
       "      <th>n_kmers</th>\n",
       "      <th>classification</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SRR306838.1617 Ibis_Run100924_C3PO:6:1:9036:1074/1</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>non-coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>SRR306838.2723 Ibis_Run100924_C3PO:6:1:15754:1168/1</td>\n",
       "      <td>0.769231</td>\n",
       "      <td>13</td>\n",
       "      <td>non-coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>SRR306838.3707 Ibis_Run100924_C3PO:6:1:5241:1269/1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>16</td>\n",
       "      <td>coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>SRR306838.4251 Ibis_Run100924_C3PO:6:1:5141:1314/1</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>15</td>\n",
       "      <td>non-coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>SRR306838.4936 Ibis_Run100924_C3PO:6:1:5473:1375/1</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>14</td>\n",
       "      <td>non-coding</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                read_id  \\\n",
       "1    SRR306838.1617 Ibis_Run100924_C3PO:6:1:9036:1074/1   \n",
       "7   SRR306838.2723 Ibis_Run100924_C3PO:6:1:15754:1168/1   \n",
       "9    SRR306838.3707 Ibis_Run100924_C3PO:6:1:5241:1269/1   \n",
       "10   SRR306838.4251 Ibis_Run100924_C3PO:6:1:5141:1314/1   \n",
       "11   SRR306838.4936 Ibis_Run100924_C3PO:6:1:5473:1375/1   \n",
       "\n",
       "    jaccard_in_peptide_db  n_kmers classification  \n",
       "1                0.000000        0     non-coding  \n",
       "7                0.769231       13     non-coding  \n",
       "9                1.000000       16         coding  \n",
       "10               0.000000       15     non-coding  \n",
       "11               0.000000       14     non-coding  "
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coding_scores_not_low_complexity = coding_scores.query('jaccard_in_peptide_db >= 0')\n",
    "print(coding_scores_not_low_complexity.shape)\n",
    "coding_scores_not_low_complexity.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "execution_event_id": "838337ba-9b4d-4ac8-9641-9b0a7f474408",
    "last_executed_text": "folder = '/home/olga/data_sm/kmer-hashing/brawand2011/human/'\n! ls -lha $folder\n\ncoding_read_ids = pd.read_csv(f\"{folder}/SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.out_CDS_without_stop_codon_or_utr.read_ids.txt\", \n                                   header=None, squeeze=True)\n\n# Try just CDS minus stop codons .... but classification goes down from 92% to 90%\n# coding_read_ids = pd.read_csv(f\"{folder}/SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.CDS_without_stop_codon.read_ids.txt\", \n#                                    header=None, squeeze=True)\nprint(coding_read_ids.shape)\ncoding_read_ids.head()",
    "persistent_id": "75c01782-427c-4c63-bc29-ed308ae4a142"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 362M\n",
      "drwxrwxr-x 2 olga olga 4.0K Oct 11 21:11 .\n",
      "drwxrwxr-x 3 olga olga 4.0K Sep 10 15:07 ..\n",
      "-rw-rw-r-- 1 olga olga  52M Oct 11 18:07 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.bam\n",
      "-rw-rw-r-- 1 olga olga 2.9M Oct 11 18:08 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.bam.bai\n",
      "-rw-rw-r-- 1 olga olga 1.9M Oct 11 21:11 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.CDS_without_stop_codon.read_ids.txt\n",
      "-rw-rw-r-- 1 olga olga 7.0M Oct 11 20:58 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.out_CDS_without_stop_codon.bam\n",
      "-rw-rw-r-- 1 olga olga 6.1M Oct 11 18:08 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.out_CDS_without_stop_codon_or_utr.bam\n",
      "-rw-rw-r-- 1 olga olga 1.8M Oct 11 18:08 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.out_CDS_without_stop_codon_or_utr.bam.bai\n",
      "-rw-rw-r-- 1 olga olga 1.7M Oct 11 18:08 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.out_CDS_without_stop_codon_or_utr.read_ids.txt\n",
      "-rw-rw-r-- 1 olga olga  15M Oct 11 18:08 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.read_ids.txt\n",
      "-rw-rw-r-- 1 olga olga  15M Sep 10 15:55 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out_CDS_without_stop_codon.bam\n",
      "-rw-rw-r-- 1 olga olga 1.7M Oct 11 18:08 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out_CDS_without_stop_codon.bam.bai\n",
      "-rw-rw-r-- 1 olga olga  15M Sep 18 07:15 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out_CDS_without_stop_codon_or_utr.bam\n",
      "-rw-rw-r-- 1 olga olga 1.7M Oct 11 18:08 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out_CDS_without_stop_codon_or_utr.bam.bai\n",
      "-rw-rw-r-- 1 olga olga 2.8M Sep 18 10:14 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out_CDS_without_stop_codon_or_utr.read_ids.txt\n",
      "-rw-rw-r-- 1 olga olga 4.1M Sep 10 15:56 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out_CDS_without_stop_codon.read_ids.txt\n",
      "-rw-r--r-- 1 olga olga 235M Sep 10 15:10 SRR306838_GSM752691_hsa_br_F_1_trimmed_k15.tsv\n",
      "(91741,)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0    SRR306838.13598607\n",
       "1    SRR306838.23303318\n",
       "2    SRR306838.19917713\n",
       "3    SRR306838.19937719\n",
       "4    SRR306838.20862490\n",
       "Name: 0, dtype: object"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "folder = '/home/olga/data_sm/kmer-hashing/brawand2011/human/'\n",
    "! ls -lha $folder\n",
    "\n",
    "coding_read_ids = pd.read_csv(f\"{folder}/SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.out_CDS_without_stop_codon_or_utr.read_ids.txt\", \n",
    "                                   header=None, squeeze=True)\n",
    "\n",
    "# Try just CDS minus stop codons .... but classification goes down from 92% to 90%\n",
    "# coding_read_ids = pd.read_csv(f\"{folder}/SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.CDS_without_stop_codon.read_ids.txt\", \n",
    "#                                    header=None, squeeze=True)\n",
    "print(coding_read_ids.shape)\n",
    "coding_read_ids.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "execution_event_id": "86031fee-645a-47a2-8947-1baf2c347724",
    "last_executed_text": "true_coding_read_ids = set(coding_read_ids.values)\nlen(true_coding_read_ids)",
    "persistent_id": "75c01782-427c-4c63-bc29-ed308ae4a142"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "90307"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "true_coding_read_ids = set(coding_read_ids.values)\n",
    "len(true_coding_read_ids)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "execution_event_id": "02e1482f-4fd8-4c9f-8982-d8c73a935885",
    "last_executed_text": "unspliced_read_ids_series = pd.read_csv(f\"{folder}/SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.read_ids.txt\", \n                                   header=None, squeeze=True)\nprint(unspliced_read_ids_series.shape)\nunspliced_read_ids_series.head()\n\nunspliced_reads = set(unspliced_read_ids_series.values)",
    "persistent_id": "317bb484-c93f-4e1c-b831-50734c9e6614"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(807698,)\n"
     ]
    }
   ],
   "source": [
    "unspliced_read_ids_series = pd.read_csv(f\"{folder}/SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.read_ids.txt\", \n",
    "                                   header=None, squeeze=True)\n",
    "print(unspliced_read_ids_series.shape)\n",
    "unspliced_read_ids_series.head()\n",
    "\n",
    "unspliced_reads = set(unspliced_read_ids_series.values)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {
    "execution_event_id": "3d635fcc-aaf1-4f37-bea4-3bb537343e27",
    "last_executed_text": "# coding_scores_not_low_complexity['read_id'].str.split().str[0]",
    "persistent_id": "5ef9f333-5da4-493e-9f95-93efd87f9316"
   },
   "outputs": [],
   "source": [
    "# coding_scores_not_low_complexity['read_id'].str.split().str[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {
    "execution_event_id": "e530e624-170a-42c3-bf95-2b873f1ff402",
    "last_executed_text": "coding_scores_not_low_complexity.loc['short_read_id', :] = coding_scores_not_low_complexity['read_id'].str.split().str[0]\nprint(coding_scores_not_low_complexity.shape)\ncoding_scores_not_low_complexity.head()",
    "persistent_id": "49d36e97-c1f4-400a-a81a-a2605e537be5"
   },
   "outputs": [
    {
     "ename": "SettingWithCopyWarning",
     "evalue": "\nA value is trying to be set on a copy of a slice from a DataFrame\n\nSee the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mSettingWithCopyWarning\u001b[0m                    Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-38-a88882f1fe13>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mcoding_scores_not_low_complexity\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mloc\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'short_read_id'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m:\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcoding_scores_not_low_complexity\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'read_id'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msplit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      2\u001b[0m \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcoding_scores_not_low_complexity\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshape\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      3\u001b[0m \u001b[0mcoding_scores_not_low_complexity\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mhead\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/indexing.py\u001b[0m in \u001b[0;36m__setitem__\u001b[0;34m(self, key, value)\u001b[0m\n\u001b[1;32m    188\u001b[0m             \u001b[0mkey\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcom\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mapply_if_callable\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mobj\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    189\u001b[0m         \u001b[0mindexer\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_get_setitem_indexer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 190\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_setitem_with_indexer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mindexer\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    191\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    192\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_validate_key\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkey\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/indexing.py\u001b[0m in \u001b[0;36m_setitem_with_indexer\u001b[0;34m(self, indexer, value)\u001b[0m\n\u001b[1;32m    375\u001b[0m                     \u001b[0mlabels\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mindex\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0minsert\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mindex\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    376\u001b[0m                     \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mobj\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_data\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mobj\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreindex\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlabels\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_data\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 377\u001b[0;31m                     \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mobj\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_maybe_update_cacher\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mclear\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    378\u001b[0m                     \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mobj\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_is_copy\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    379\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/generic.py\u001b[0m in \u001b[0;36m_maybe_update_cacher\u001b[0;34m(self, clear, verify_is_copy)\u001b[0m\n\u001b[1;32m   3142\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3143\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mverify_is_copy\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3144\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_check_setitem_copy\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mstacklevel\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m5\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mt\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'referant'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3145\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3146\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mclear\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/generic.py\u001b[0m in \u001b[0;36m_check_setitem_copy\u001b[0;34m(self, stacklevel, t, force)\u001b[0m\n\u001b[1;32m   3285\u001b[0m             \u001b[0;32melif\u001b[0m \u001b[0mvalue\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m'warn'\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3286\u001b[0m                 warnings.warn(t, com.SettingWithCopyWarning,\n\u001b[0;32m-> 3287\u001b[0;31m                               stacklevel=stacklevel)\n\u001b[0m\u001b[1;32m   3288\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3289\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m__delitem__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mSettingWithCopyWarning\u001b[0m: \nA value is trying to be set on a copy of a slice from a DataFrame\n\nSee the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy"
     ]
    }
   ],
   "source": [
    "coding_scores_not_low_complexity.loc['short_read_id', :] = coding_scores_not_low_complexity['read_id'].str.split().str[0]\n",
    "print(coding_scores_not_low_complexity.shape)\n",
    "coding_scores_not_low_complexity.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {
    "execution_event_id": "780e8e35-6843-4d81-a036-f002c4b93605",
    "last_executed_text": "coding_scores_not_low_complexity_unspliced = coding_scores_not_low_complexity.query('short_read_id in @unspliced_reads')\ncoding_scores_not_low_complexity_unspliced['in_cds_no_stop_codon'] = coding_scores_not_low_complexity_unspliced.short_read_id.isin(true_coding_read_ids)\nprint(coding_scores_not_low_complexity_unspliced.shape)\ncoding_scores_not_low_complexity_unspliced.head()",
    "persistent_id": "b321c9fc-54af-40bc-96a9-40c59ea87994"
   },
   "outputs": [
    {
     "ename": "SettingWithCopyWarning",
     "evalue": "\nA value is trying to be set on a copy of a slice from a DataFrame.\nTry using .loc[row_indexer,col_indexer] = value instead\n\nSee the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mSettingWithCopyWarning\u001b[0m                    Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-39-75293f21b612>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0mcoding_scores_not_low_complexity_unspliced\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcoding_scores_not_low_complexity\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mquery\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'short_read_id in @unspliced_reads'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m \u001b[0mcoding_scores_not_low_complexity_unspliced\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'in_cds_no_stop_codon'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcoding_scores_not_low_complexity_unspliced\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshort_read_id\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0misin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtrue_coding_read_ids\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      3\u001b[0m \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcoding_scores_not_low_complexity_unspliced\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshape\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[0mcoding_scores_not_low_complexity_unspliced\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mhead\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m__setitem__\u001b[0;34m(self, key, value)\u001b[0m\n\u001b[1;32m   3368\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3369\u001b[0m             \u001b[0;31m# set column\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3370\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_set_item\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3371\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3372\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_setitem_slice\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkey\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m_set_item\u001b[0;34m(self, key, value)\u001b[0m\n\u001b[1;32m   3450\u001b[0m         \u001b[0;31m# value exception to occur first\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3451\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3452\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_check_setitem_copy\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3453\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3454\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0minsert\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mloc\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcolumn\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mallow_duplicates\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/generic.py\u001b[0m in \u001b[0;36m_check_setitem_copy\u001b[0;34m(self, stacklevel, t, force)\u001b[0m\n\u001b[1;32m   3285\u001b[0m             \u001b[0;32melif\u001b[0m \u001b[0mvalue\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m'warn'\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3286\u001b[0m                 warnings.warn(t, com.SettingWithCopyWarning,\n\u001b[0;32m-> 3287\u001b[0;31m                               stacklevel=stacklevel)\n\u001b[0m\u001b[1;32m   3288\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3289\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m__delitem__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mSettingWithCopyWarning\u001b[0m: \nA value is trying to be set on a copy of a slice from a DataFrame.\nTry using .loc[row_indexer,col_indexer] = value instead\n\nSee the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy"
     ]
    }
   ],
   "source": [
    "coding_scores_not_low_complexity_unspliced = coding_scores_not_low_complexity.query('short_read_id in @unspliced_reads')\n",
    "coding_scores_not_low_complexity_unspliced['in_cds_no_stop_codon'] = coding_scores_not_low_complexity_unspliced.short_read_id.isin(true_coding_read_ids)\n",
    "print(coding_scores_not_low_complexity_unspliced.shape)\n",
    "coding_scores_not_low_complexity_unspliced.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {
    "execution_event_id": "d10407ad-db67-4ae8-9121-2cf587fa2ee3",
    "last_executed_text": "y_true = coding_scores_not_low_complexity_unspliced['in_cds_no_stop_codon']\ny_score = coding_scores_not_low_complexity_unspliced['jaccard_in_peptide_db']\ny_pred = y_score > 0.9\n\nfpr, tpr, thresholds = metrics.roc_curve(y_true, y_score)\nauc = metrics.auc(fpr, tpr)\nprint(f\"AUC: {auc}\")\n\nfig, ax = plt.subplots()\nlw = 2\nax.plot(fpr, tpr, lw=2, color='darkorange')\nax.plot([0, 1], [0, 1], color='navy', lw=lw, linestyle='--')\nax.set(xlabel='False Positive Rate', ylabel='True Positive Rate')\n\nprint(\"accuracy score:\", metrics.accuracy_score(y_true, y_pred))\n\nprint(metrics.classification_report(y_true, y_pred))",
    "persistent_id": "2c810e70-a822-4d9a-a9b0-2622dfa247bf"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AUC: 0.9332425339782375\n",
      "accuracy score: 0.9019756727303703\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "       False       0.99      0.90      0.94    590637\n",
      "        True       0.58      0.94      0.72     90245\n",
      "\n",
      "   micro avg       0.90      0.90      0.90    680882\n",
      "   macro avg       0.79      0.92      0.83    680882\n",
      "weighted avg       0.94      0.90      0.91    680882\n",
      "\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAYUAAAEKCAYAAAD9xUlFAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDMuMC4zLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvnQurowAAIABJREFUeJzt3Xl8FFW2wPHfyQJh3xHZkUVFkIgBFBUXFlEkOOooiiCKICjKuI36XBCfz0FcUUSMwLgMKg6DGhwYZBRcEBAEEQGVsAhBkS1BhCRkOe+P6sQ2ZulAqquX8/188qGrutJ9KiF1+t576l5RVYwxxhiAGK8DMMYYEzosKRhjjCliScEYY0wRSwrGGGOKWFIwxhhTxJKCMcaYIpYUjDHGFLGkYIwxpoglBWOMMUXivA6goho2bKitW7f2OgxjjAkrX3755V5VbVTecWGXFFq3bs2qVau8DsMYY8KKiPwQyHHWfWSMMaaIJQVjjDFFLCkYY4wpYknBGGNMEUsKxhhjiriWFERkpojsFpFvSnleROQ5EUkTka9FpKtbsRhjjAmMmy2FV4D+ZTx/EdDe9zUKeNHFWIwxxgTAtfsUVPUTEWldxiGDgNfUWQ90uYjUFZHjVfUnt2IyxpgKU4WCXMg/8tu//o8LStlf+LiglP2F35vvd0wpz33xXRUSYo9w6tk9oOd4V0/Xy5vXmgE7/LbTffv+kBREZBROa4KWLVsGJThjjAu0wHehK+WiGMiFsyIX5jK/N9DXzfPux6Vwz7/78tTH3Tj1+J/5ov0G4l1+Ty+TgpSwT0s6UFVTgBSApKSkEo8xJuoUXWAD+7QZ/Iuu/zGFn5q9u8Aek5g4iKkCsVUgJt751/9xTBWIjS/lmLKeK2N/bBUkpgp8vws++Zl+yd3J794jopNCOtDCb7s58KNHsZhQoupc8Py/KHyspewvfnxJr6G/vUZJjwN6vnB/fukXvmBddDXfq9/QsSm6kAZyQa2Ei27hMSXtD+DCTEw8SEmfYd2RmZnNli0ZdO16PAATJucyePTeom23eZkUUoGxIvIW0AM4YOMJflQhOwMO74KsvZCdCfnZkJcFeX7/5mf/9gmsIM+5UBTkO/s0v9i+PN9+vwtqQb7fhbO0xyVt+18k8397D82nxAtySRfr3x3nd+E1gQvkwhesi25AF+a4oF5gw817733LmDH/JiZGWL/+ZurUSaBatfigJQRwMSmIyJvAeUBDEUkHxoPT8lHVacB84GIgDTgMXO9WLGEjPxd2rYRv34RNc+DQLq8j8pBATCxIjPNFjHMxKdwu2lfCF8WOE/E7Vvye93tc6r5ijwufi4nz/qJrF9iIsXv3IW67bQGzZ68H4IwzmpOZmU2dOglBj8XN6qOry3legVvcev+wcnAnrJwE377htAoKxdeEmk2hWiNIqAuxCRBfHWKrQlw1Zzuumu9iEQ8S67uQxvpdNIrviwWJ+/1FMyaWootm4YWYwv3+F2f/i3ThxZLfv4f/Mf7HFr5WaRfq372uXehMdFBVZs1ax7hx/2H//iyqV4/nsccuYOzY7sTGenNvcdhNnR1Rcg/Dskdg9bOQn+Psq9sO2lwMp1wHjRN/u/AaYyLOmDH/5qWXvgSgT58TSEm5hDZt6nkakyUFL6g6XUSf3gcHtzv72v0Jut8LTbrZJ2VjosSll57E7Nnreeqpflx/fSISAn/7lhSCTRU+vdfpLgJolAh9XoSmZ3gblzHGdZs27ePDD7cyenQSAP37t2PbtnGejB2UxpJCMGkBfDQOvpri9L33fgE63+jrbzfGRKq8vAKefnoZ48cvIScnj8TEJpxxRnOAkEoIYEkheAryIPUK2PyeMwA88F/QLtnrqIwxLlu7dhcjRqTy5ZdOxf2wYV1o376+x1GVzpJCMKjCopuchFC1LgycA616ex2VMcZFOTl5PProJ0ycuJS8vAJatqzDSy9dQv/+7bwOrUyWFIJh8zz4ZqZTPnrZfGh6ptcRGWNcdt99H/LMM8sBuOWWbvztb72pVauqx1GVz5KC21Thvzc5j8961BKCMVHir389i2XL0pk0qQ/nnNPK63ACZkXwbtu33rkzObYqnDbW62iMMS5ZtGgzl1/+Nnl5BQA0aVKTzz+/IawSAlhScN+er51/G3VxpicwxkSUjIwsRox4j379/sHcuRv5+9/XFD0XCvcdVJR1H7lt3wbn39b9vI3DGFPp3nlnIzffPJ9du36latVYxo8/l+HDE70O65hYUnDbPmeCK+p39DYOY0yl2bXrV269dQFz5jgf+nr2bMGMGcmcdFJDjyM7dpYU3FaYFBqe4m0cxphK89573zJnzgZq1Ihn4sQ+3HxzN2Jiwq+rqCSWFNyUlw2Zm51J7eqd6HU0xphjkJ2dR0KCc8kcOfJ0tmzJYMyYbrRuXdfjyCqXDTS7af93ztQWddtBXOjXJxtj/qigQJky5QvatJnMDz9kAhATIzz+eN+ISwhgScFdhV1HDazryJhw9N13e+nV6+/ceusCdu36lTff/MbrkFxn3UduKqw8svEEY8JKbm4+Tz75ORMmfExOTj7HHVeDqVMHcNllJ3sdmussKbjJKo+MCTvffLObYcPeYc0aZznc669P5Kmn+lGvXjWPIwsOSwpuspaCMWGnoEBZt243rVrVISVlIP36tfU6pKCypOCWvGzITPNVHnXwOhpjTBnWr99Nx46NEBFOPfU43ntvML16taJmzeibhcAGmt3yu8qj0FpEwxjjOHgwh7Fj59Op04v8618bi/ZffHH7qEwIYC0F9xR2HVnlkTEhaeHCNEaNep/t2w8QFxfDtm2ZXocUEiwpuKWoHNUGmY0JJfv3Z3H77Qt57bW1AHTtejwzZiSTmNjE48hCgyUFt9g9CsaEnK++2kX//v/g558PUbVqLBMmnMedd/YkLs560gtZUnBLUfeRtRSMCRUdOjSgZs0qdOjQgOnTk+nQoYHXIYUcSwpu8K88qm9zHhnjFVXljTfWMXDgidSuXZXq1eNZsmQ4TZvWipgJ7CqbtZnckPG9VR4Z47Ft2zK58MJ/cO2173Dvvf8t2t+8eW1LCGWwloIb9togszFeyc8vYOrUldx334ccOpRL/frV6NmzhddhhQ1LCm6wQWZjPLFx4x5GjEhl2bJ0AK688hSef/4iGjeu4XFk4cOSghvsHgVjgm7r1gwSE1/iyJF8jj++JlOnDuDSS0/yOqyw42pSEJH+wGQgFpiuqhOLPd8SeBWo6zvmXlWd72ZMQWH3KBgTdG3a1OPPf+5IQkIcTz7Zj7p1bTzvaLiWFEQkFngB6AukAytFJFVVN/gd9gDwtqq+KCIdgflAa7diCgqrPDImKLKycnnkkY/5059Opnv3ZgC8+uqlxMZa/cyxcPOn1x1IU9UtqnoEeAsYVOwYBWr7HtcBfnQxnuAoqjxqa5VHxrjk009/IDHxJSZOXMqoUfMoKFAASwiVwM3uo2bADr/tdKBHsWMeBj4QkVuBGkAfF+MJjr02yGyMW375JYf77vsvU6euAqBjx0ZMm3aJlZhWIjfTakm/JS22fTXwiqo2By4GXheRP8QkIqNEZJWIrNqzZ48LoVai/TbIbIwb5s/fRKdOU5k6dRVxcTE89FAvVq8exRlnNPc6tIjiZkshHfAvDm7OH7uHRgD9AVR1mYgkAA2B3f4HqWoKkAKQlJRUPLGEFrtHwZhKd+BANkOGzCUzM5ukpKbMmJHMqace53VYEcnNpLASaC8ibYCdwGDgmmLHbAd6A6+IyMlAAhDiTYFy2D0KxlQKVUUVYmKEOnUSeO65/vz88yH+8pczbAI7F7mWFFQ1T0TGAgtxyk1nqup6EXkEWKWqqcCdwMsicjtO19JwVQ3tlkBZ8nKs8siYSvDjjwe5+eZ/c845Lbnzzp4ADB3axeOoooOr9yn47jmYX2zfQ36PNwBnuRlDUGX4Vlur194qj4w5CqrKzJlruPPODzhwIIfly9O5+eZuVKsW73VoUcPuaK5MVnlkzFHbsiWDkSPn8dFHWwEYMKA906ZdYgkhyCwpVKb9toaCMRWVn1/Ac8+t4P77PyIrK4+GDavz3HP9GTy4EyJWahpslhQqk7UUjDkqc+ZsJCsrj6uv7sTkyf1p1MgmsPOKJYXKZBPhGROQI0fyOXgwhwYNqhMbG8OMGcls2rSPgQOtQMNrVtdVWazyyJiArFy5k6SkFIYOfYfCYsOTTmpoCSFEWEuhsmR8B5pvlUfGlOLw4VzGj1/M008vp6BAOXw4l927D3HccTW9Ds34saRQWazryJhSLVmyjZEj55GWtp+YGOGuu85kwoTzqV7dKotCjSWFymJrKBjzB6rKbbctYMqUlQB07tyYGTOS6datmceRmdJYUqgsVnlkzB+ICLVrVyU+PoYHHujFvfeeTZUqsV6HZcpgSaGy7LN7FIwB2Lv3MJs376dHD2f20gcfPJchQ06lY8dGHkdmAhFQ9ZGIVBGRdm4HE7Z+V3lka8Ka6KSqvPXWN5x88gtceulsMjKyAEhIiLOEEEbKTQoiMgBYByzybSeKyDtuBxZWCiuPbLU1E6XS039h0KC3uPrqf7F372E6dmzE4cO5XodljkIg3UeP4KyYthhAVb+yVkMxhV1H9a3ryESXggJl+vTV3H33In75JYfatavy1FP9GDHiNJuiIkwFkhRyVTWz2C84fKe3dkNh5VFDG2Q20WXEiFReeeUrAJKTT2Tq1Itp1qx2Od9lQlkgYwobReRKIEZE2ojIs8Byl+MKL3aPgolS117bmcaNa/DWW5fz7rtXWUKIAIEkhbHA6UABMBfIBsa5GVTYsSU4TZT45pvdTJ7822fC3r1PYMuW27jqKpvRNFIE0n10oareA9xTuENELsNJEMa/8qiezd1iIlNOTh5/+9tnPPbYp+TmFpCU1JSzzmoJQI0aVTyOzlSmQFoKD5Sw7/7KDiRsZXzvVB7VOQHiq3kdjTGVbsWKdE4/PYUJEz4mN7eAMWOS6Nz5OK/DMi4ptaUgIhcC/YFmIvK031O1cbqSDPhNb2HjCSayHDp0hAcfXMyzzy5HFdq3r8/06cn06tXK69CMi8rqPtoNfIMzhrDeb/9B4F43gworVnlkItT993/E5MkriIkR7r77TB5++DxbGjMKlJoUVHUNsEZEZqlqdhBjCi82vYWJUPfffw7r1u3m8cf7kJTU1OtwTJAEMqbQTETeEpGvReT7wi/XIwsXNhGeiRCpqd9x8cWzyM3NB6BRoxp8+OEwSwhRJpCk8Arwd0CAi4C3gbdcjCl8WOWRiQC7dx9i8OA5DBr0FgsWpPHqq2u9Dsl4KJCkUF1VFwKo6mZVfQA4392wwoRVHpkwpqr84x9fc/LJLzB79nqqV49n8uT+XH99otehGQ8Fcp9Cjjh3pWwWkdHATqCxu2GFCas8MmFq+/YDjB79PgsWpAHQp88JpKRcQps29TyOzHgtkKRwO1ATuA34P6AOcIObQYUNG2Q2YeqDDzazYEEadesm8PTT/Rg+PNHuSDZAAElBVVf4Hh4EhgKISHM3gwobVo5qwsihQ0eK7j4eMeI0du78hVGjTuf442t5HJkJJWWOKYhINxG5VEQa+rZPEZHXsAnxHFZ5ZMJAXl4BkyYtpVWrZ9myJQNwlskcP/48SwjmD0pNCiLyN2AWMAT4j4jcj7OmwlqgQ3DCC2FWeWTCwNq1u+jRYzr33PNf9u3L4t13v/U6JBPiyuo+GgR0UdUsEakP/Ojb/i44oYW4wsqjuu2s8siEnJycPB599BMmTlxKXl4BLVvWISXlEi680NbHMmUrKylkq2oWgKruF5FvLSH4sUFmE6LWrPmJIUPmsnHjXkRg7NhuPPZYb2rVqup1aCYMlJUUThCRwumxBWjtt42qXlbei4tIf2AyEAtMV9WJJRxzJfAwzmpua1X1msDD95CVo5oQVbVqHJs3Z3DiiQ2YPj2Zs89u6XVIJoyUlRQuL7Y9pSIvLCKxwAtAXyAdWCkiqaq6we+Y9sB9wFmqmiEi4XP/g1UemRCyevVPnHZaE0SEjh0bsWDBEHr2bEFCQiBV58b8pqwJ8T48xtfuDqSp6hYAEXkLZ5xig98xI4EXVDXD9567j/E9g6ew+6i+dR8Z72RkZHHXXR8wc+ZXvPnm5Qwe3AmACy5o43FkJly5+TGiGbDDbzsd6FHsmA4AIrIUp4vpYVX9T/EXEpFRwCiAli1DoCmclwMZm5zKo/oneR2NiVLvvLORm2+ez65dv1K1aiz79h32OiQTAdxMCiXdHqklvH974DygOfCpiHRS1czffZNqCpACkJSUVPw1gs8qj4yHdu36lVtvXcCcOU5r9ayzWjB9ejInndTQ48hMJAg4KYhIVVXNqcBrpwMt/Lab45S1Fj9muarmAltF5DucJLGyAu8TfFZ5ZDzy5Zc/0rfv62RkZFOjRjwTJ/bh5pu7ERNjU1SYylHuLKki0l1E1gGbfNtdROT5AF57JdBeRNqISBVgMJBa7Jh38c246rtrugOwpQLxe8Mqj4xHOnZsRKNGNbjwwrasX38zY8d2t4RgKlUgU2c/B1wC7ANQ1bUEMHW2quYBY4GFwEbgbVVdLyKPiEiy77CFwD4R2YBzt/Tdqrqv4qcRZNZSMEFSUKCkpHxJZqaz+GG1avF88slwFiwYQqtWdT2OzkSiQLqPYlT1h2IzKOYH8uKqOh+YX2zfQ36PFbjD9xU+rKVgguC77/Zy443z+Oyz7axcuZOXX3Y+Sx13XE2PIzORLJCksENEugPqu/fgViB6l+MsrDxCrPLIuCI3N5+nnlrGww8vIScnnyZNanLRRe29DstEiUCSwhicLqSWwM/Af337olPmJl/lUVurPDKVbs2anxgxIpU1a3YBcP31iTz1VD/q1bP/ayY4AkkKeao62PVIwoVNl21csnnzfrp3n05eXgGtW9clJeUS+vZt63VYJsoEkhRW+kpFZwNzVfWgyzGFNhtPMC5p27Y+Q4eeSq1aVfi//+tNzZpVvA7JRKFyq49UtS3wKHA6sE5E3hWR6G05WOWRqSS//nqE225bwLJlv934P2NGMpMnX2QJwXgmkJJUVPVzVb0N6Ar8grP4TnSyloKpBAsXpnHKKVN5/vkvGD363ziFeNg6ycZz5XYfiUhNnInsBgMnA+8BPV2OKzTlH/GrPLLV1kzF7d+fxe23L+S119YCcPrpxzNjRrIlAxMyAhlT+AaYB0xS1U9djie0Fc151Bbiq3sdjQkzc+Zs4JZb5rN79yESEuKYMOE87rjjTOLiAmqwGxMUgSSFE1S1wPVIwoFVHpmjlJmZzahR88jIyKZXr1a8/PJAOnRo4HVYxvxBqUlBRJ5S1TuBf4nIH2YmDWTltYhjg8ymAlSVggIlNjaGunUTmDp1ABkZWdx0U5LNV2RCVlkthdm+fyu04lpEs0FmE6Bt2zIZNWoeF1zQhnvvPRugaAEcY0JZqZ2ZqvqF7+HJqvqh/xfOgHP0saRgypGfX8Bzz62gU6epLFq0hSlTviA7O8/rsIwJWCAjXDeUsG9EZQcS8qzyyJRj48Y99Or1CuPG/YdDh3IZPLgTq1ffZOskm7BS1pjCVThlqG1EZK7fU7WAzJK/K4JZ5ZEpRV5eAY8//hmPPPIJR47k07RpLV58cQDJyfbhwYSfsj7CfIGzhkJz4AW//QeBNW4GFZIKK4/q2yCz+b2YGOGDD7Zw5Eg+I0d2ZdKkvtStm+B1WMYclVKTgqpuBbbizIpqCiuPGtp4goGsrFwOHjxC48Y1iIkRpk8fyI4dv3DBBW28Ds2YY1LqmIKIfOz7N0NE9vt9ZYjI/uCFGCJskNn4fPLJD3TpMo1rr51bND1F+/YNLCGYiFBW91HhkpsNgxFIyLN7FKLeL7/kcN99/2Xq1FUAxMfHsnfvYRo1quFxZMZUnrJKUgvvYm4BxKpqPnAmcBMQXX8F+UecxXVstbWotWDBJjp1msrUqauIi4th/PhzWb16lCUEE3ECqZV7F+gmIm2B14B/A28Al7gZWEjJ+B4K8qzyKAqpKiNHzmPGDKe2IimpKTNnJtO583EeR2aMOwK5T6FAVXOBy4BnVfVWoJm7YYWYwq4jqzyKOiJC8+a1SUiI48kn+7Js2QhLCCaiBbQcp4j8GRgKXOrbF+9eSCGosBzVKo+iwo8/HmTz5v2cc04rAP7nf85h6NBTadu2vseRGeO+QO9oPh9n6uwtItIGeNPdsEJMUeWRtRQimaoyY8ZqOnZ8gcsvf5t9+w4DUKVKrCUEEzXKbSmo6jcichvQTkROAtJU9f/cDy2EFFUeWUshUm3ZksHIkfP46KOtAFxySQdyc23GeBN9All57RzgdWAnIEATERmqqkvdDi4kWOVRRCucwO6BBxZz+HAuDRtW57nn+jN4cCdbDc1EpUDGFJ4BLlbVDQAicjJOkkhyM7CQkbHJqTyqc4JVHkWgYcPe5Y031gFwzTWdefbZC63M1ES1QMYUqhQmBABV3QhUcS+kEGN3Mke0kSO70rx5bVJTBzNr1mWWEEzUC6SlsFpEXsJpHQAMIZomxLPKo4iycuVOPvpoK/fc4yx8c955rUlLu5WqVW16a2MgsKQwGrgN+CvOmMInwPNuBhVS9tv0FpHg8OFcxo9fzNNPL6egQOnZs0VRyaklBGN+U+Zfg4h0BtoC76jqpOCEFGL2WvdRuFuyZBs33pjK5s0ZxMQId911Jqef3tTrsIwJSWXNkvo/OFNcDAEWiUhJK7CVSUT6i8h3IpImIveWcdwVIqIiElqD11Z5FNYOHMjmppvmcf75r7J5cwadOzdm+fIRPPFEP6pXj677L40JVFkthSHAqap6SEQaAfOBmYG+sIjE4izO0xdIB1aKSKr/oLXvuFo43VMrKhq866zyKKw9+OBiUlJWEx8fw4MP9uKee86mSpVYr8MyJqSVlRRyVPUQgKruEZFAKpX8dce50W0LgIi8BQwCNhQ77n+BScBdFXx991nlUdhR1aL7Cx566Fy2bs1k4sTenHJKY48jMyY8lHWhP0FE5vq+3gHa+m3PLeP7CjUDdvhtp1NsIj0ROQ1ooarvVzjyYLA1FMKGqvLGG+u44ILXOHIkH4CGDaszb97VlhCMqYCyWgqXF9ueUsHXLul2UC160ml5PAMML/eFREYBowBatmxZwTCOwT4rRw0H6em/MGbMv3n//e8BmDXra66//jSPozImPJW1RvOHx/ja6TgL9BRqDvzot10L6AQs8TX3mwCpIpKsqquKxZICpAAkJSUpwWKVRyGtoEB5+eUvufvuRRw8eIQ6dary1FP9GD480evQjAlbbhZorwTa+2ZV3QkMBq4pfFJVD+C31KeILAHuKp4QPGOVRyEtLW0/I0fOY8mSbQAMGnQiU6cOoGnTWt4GZkyYcy0pqGqeiIwFFgKxwExVXS8ijwCrVDXVrfeuFFZ5FNI+/fQHlizZRuPGNZgy5SKuuKKjTWBnTCUIOCmISFVVzanIi6vqfJxSVv99D5Vy7HkVeW3X2RoKISczM5u6dRMAGD48kT17DjNixGk0aGBJ25jKUm6ZqYh0F5F1wCbfdhcRifxpLmwNhZCRk5PH+PGLadXqWTZt2gc4y2T+9a9nWUIwppIF0lJ4DrgE5+5mVHWtiJzvalShwCqPQsLy5emMGJHKhg17AFi4cDPt2zfwOCpjIlcgSSFGVX8o1l+b71I8ocPuUfDUoUNHePDBxTz77HJUoX37+syYkVw0iZ0xxh2BJIUdItIdUN/UFbcC37sblsfyj0DG9ziVRyd7HU3UWbEinWuumcuWLRnExgp33dWT8ePPpVo1m6/IGLcFkhTG4HQhtQR+Bv7r2xe5iiqP2ljlkQfq1k1g585f6NLlOGbMSLYZTY0JonKTgqruxrnHIHrYIHPQffbZds46qwUiwoknNuSjj66jW7emxMfbBHbGBFO5SUFEXsZveopCqjrKlYhCgU2EFzS7dx/ittsWMHv2el599VKGDesCQM+eLcr5TmOMGwLpPvqv3+ME4E/8fqK7yGP3KLhOVZk1ax3jxv2H/fuzqF49vmgiO2OMdwLpPprtvy0irwOLXIsoFBR2H1k5qiu2bz/A6NHvs2BBGgB9+55ASspAWreu63FkxpijmeaiDRC5dYFWeeSqFSvS6dPndX799Qh16ybwzDMXct11XWyKCmNCRCBjChn8NqYQA+wHSl1aM+xlplnlkYsSE5vQokVtTjqpIS+8cDHHH28T2BkTSspMCuJ8fOuCM8spQIGqBm/qai/YdNmVKi+vgClTvmDYsC7Ur1+NqlXjWLr0BurVq+Z1aMaYEpQ595EvAbyjqvm+r8hOCGCDzJVo7dpd9OgxndtvX8gddyws2m8JwZjQFci6y1+ISFfXIwkVdo/CMcvOzuOBBz4iKellVq/+iZYt63D11Z28DssYE4BSu49EJE5V84CzgZEishk4hLPMpqpqZCYKmwjvmHz++Q5GjEjl22/3IgJjx3bjscd6U6tWVa9DM8YEoKwxhS+ArsClQYrFe0WVR9hqa0chLW0/55zzdwoKlBNPbMCMGcmcdVYQ19Q2xhyzspKCAKjq5iDF4r3fVR7V8DqasNOuXX1GjepK/frVePDBc0lIcHO1V2OMG8r6q20kIneU9qSqPu1CPN6yyqMKycjI4s47P+D66xOLprSeOnWA3XNgTBgrKynEAjXxtRiigq2hELC5czdyyy3z2bXrV7788ie++uomRMQSgjFhrqyk8JOqPhK0SEKBTYRXrl27fmXs2Pn8618bATj77JZMnz7QkoExEaLcMYWoYvcolEpVee21tdx++0IyMrKpWbMKjz/eh9Gjk4iJib7/KsZEqrKSQu+gRREK8nN/qzxqYHMeFZeZmc2dd35ARkY2/fu3Y9q0AbRqZRPYGRNpSk0Kqro/mIF4LnOTVR4VU1CgFBQocXEx1KtXjZdeuoTDh3O59tpTrbvImAgVyB3N0WGvdR35+/bbvfTq9XcmTvysaN/ll3dk6FCb0dSYSGZJoZBNbwFAbm4+jz32KV26TGPp0h3MmLGG7Ow8r8MyxgSJ3V1UyCqPWLPmJ264IZWvvtoFwIgRp/HEE33tJjRjooj9tRfb6oM5AAAQbUlEQVSK4sqj3Nx8xo9fwqRJS8nPV1q3rsvLLw+kT58TvA7NGBNklhQg6iuP4uJiWLFiJwUFyrhxPXj00QuoWbOK12EZYzxgSQF+qzyq3TpqKo8OHszh4MEjNG1aCxFh+vSB7Nr1K2ee2cLr0IwxHrKBZvhtkDlKpsteuDCNTp1eZMiQuRSum9SmTT1LCMYYSwpA1EyEt2/fYa677l3695/F9u0HOHgwh337srwOyxgTQlxNCiLSX0S+E5E0Ebm3hOfvEJENIvK1iHwoIq3cjKdUET7IrKrMmbOBjh2n8tpra0lIiGPSpD4sX34jDRtW9zo8Y0wIcW1MQURigReAvkA6sFJEUlV1g99ha4AkVT0sImOAScBVbsVUqgi+R0FVGTJkLm+++Q0AvXq14uWXB9KhQwOPIzPGhCI3WwrdgTRV3aKqR4C3gEH+B6jqYlU97NtcDjR3MZ6SRXjlkYjQsWMjatWqwosvDmDx4ussIRhjSuVm9VEzYIffdjrQo4zjRwALSnpCREYBowBatqzk5R0zN0FBbkRVHm3dmsGWLRn07u3cZ3DPPWcxfHgizZvX9jgyY0yoc7OlUNIEOVrigSLXAknAEyU9r6opqpqkqkmNGjWqxBCJqMqj/PwCJk9eTqdOL3LVVXPYvfsQAPHxsZYQjDEBcbOlkA741zg2B34sfpCI9AHuB85V1RwX4ylZYeVR/fAeZN6wYQ833pjKsmXpACQnn2jrHBhjKszNpLASaC8ibYCdwGDgGv8DROQ04CWgv6rudjGW0oV5SyE3N5/HH1/K//7vJxw5kk/TprV48cUBJCef6HVoxpgw5FpSUNU8ERkLLMRZ73mmqq4XkUeAVaqaitNdVBP4p2865u2qmuxWTCUK84nwrrlmLnPmOIlt5MiuPPFEX+rUSfA4KmNMuHJ1mgtVnQ/ML7bvIb/Hfdx8/3L5Vx7VP8nTUI7WuHE9+OqrXbz00iVccEEbr8MxxoS56L6jOTPtt8qjKjW9jiYgH3+8jQkTlhRtn312SzZuvMUSgjGmUkT3hHiFXUdhMJ7wyy853HPPIqZN+xKA889vQ69ezg3gcXHRnduNMZUnupNCmFQezZ+/iZtuep/09F+Ij4/h/vvP4Ywzgn+fnzEm8kV3UgjxyqO9ew/zl7/8h1mz1gHQvXszZsxIplOnxh5HZoyJVFGeFEJ7IrxHHvmYWbPWUa1aHI8+egHjxvUgNta6iowx7onepPC7yqPQmfNIVfGV5zJhwnn8/PMhHnvsAtq2re9xZMaYaBC9HztDrPJIVXn55S/p2XMm2dl5ANSrV43Zs6+whGCMCZroTQoh1HW0efN+evd+jVGj3mf58nTefnu91yEZY6JU9HYfhcAaCs4Edit44IGPyMrKo1Gj6jz//EVceWVoDnwbYyJf9CaFvd62FNav380NN6TyxRc7ARgypDPPPtvfVkIzxngqepOCxzeurVmziy++2EmzZrV46aVLGDCggydxGGOMv+hMCh5VHu3Zc4hGjZyFfIYM6UxmZjZDh55qE9gZY0JGdA40F1UetQpK5dHhw7ncddcHtG49mY0b9wDOMpljx3a3hGCMCSnR2VII4iDz4sVbGTlyHps3ZxATI3zyyQ+cfHIlrx5njDGVJEqTgvtrKBw4kM1f/7qIlJTVAHTu3JiZMweRlNTUtfc0xphjFZ1JweXKo88+287gwXPYufMg8fExPPhgL+6552yqVIl15f2MMaayRGdS2O/uRHhNmtRk374szjijOdOnD+SUU2wCO2NMeIi+pJCfC/u/cx5XUuWRqrJo0Rb69j0BEaFdu/p89tn1JCY2sQnsjDFhJfquWJVcebRjxwEGDnyTCy/8B3//+1dF+08/vaklBGNM2Im+lkIlVR4VFDgT2N199yIOHjxCnTpVqVrVxgyMMeEtCpPCsQ8yb9q0j5Ej5/Hxxz8AcOmlJ/HCCxfTtGmtyojQGGM8E4VJ4dhaCp9/voPevV8jOzuPxo1rMGXKRVxxRceiNRCMMSacRWFSOLY5j5KSmtK+fX1OO+14nn66Hw0a2AR2xpjIEV1J4Sgqj3Jy8njyyc+56aYkGjasTpUqsSxdegO1alV1MVBjjPFGdCWFzM0VqjxavjydESNS2bBhDxs37uUf/7gMwBKCMSZiRVdSCHCQ+dChIzzwwEdMnrwCVejQoQE33XR6EAI0xhhvRWlSKH084cMPtzBy5Dy2bs0kNla4++6ejB9/HgkJ0fWjMsZEp+i60pVTefT99/vo2/d1VCExsQkzZiTTtevxQQzQGGO8FWVJoezuow4dGjBuXA8aNarB3Xf3JD7ebkYzxkSX6EkK/pVHvqTw88+/cttt/2H06NM5//w2ADzzTH+vIjTGGM+5OjmPiPQXke9EJE1E7i3h+aoiMtv3/AoRae1aMH6VRxpfg9dfX0vHjlN5++313H77QlTVtbc2xphw4VpSEJFY4AXgIqAjcLWIFO+3GQFkqGo74BngcbfiKew62l6QyIABbzBs2Lvs359Fv35teffdwXZHsjHG4G73UXcgTVW3AIjIW8AgYIPfMYOAh32P5wBTRETUhY/tBXvWM21pN+75TyK/ZqVRr14CzzxzIcOGdbGEYIwxPm4mhWbADr/tdKBHaceoap6IHAAaAHsrO5gDO75lwqJz+TVLuPzyk5ky5WKaNDn2qbONMSaSuJkUSvr4XbwFEMgxiMgoYBRAy5YtjyqYejWV6YMXcqTHo1x+Y/JRvYYxxkQ6N5NCOtDCb7s58GMpx6SLSBxQB9hf/IVUNQVIAUhKSjq6rqVL3mTggIKj+lZjjIkWblYfrQTai0gbEakCDAZSix2TClzne3wF8JEb4wlFJMb5MsYYUyLXWgq+MYKxwEIgFpipqutF5BFglaqmAjOA10UkDaeFMNiteIwxxpTP1ZvXVHU+ML/Yvof8HmcDf3YzBmOMMYGzvhRjjDFFLCkYY4wpYknBGGNMEUsKxhhjilhSMMYYU0TCbXZQEdkD/HCU394QF6bQCHF2ztHBzjk6HMs5t1LVRuUdFHZJ4ViIyCpVTfI6jmCyc44Ods7RIRjnbN1HxhhjilhSMMYYUyTakkKK1wF4wM45Otg5RwfXzzmqxhSMMcaULdpaCsYYY8oQkUlBRPqLyHcikiYi95bwfFURme17foWItA5+lJUrgHO+Q0Q2iMjXIvKhiLTyIs7KVN45+x13hYioiIR9pUog5ywiV/p+1+tF5I1gx1jZAvi/3VJEFovIGt//74u9iLOyiMhMEdktIt+U8ryIyHO+n8fXItK1UgNQ1Yj6wpmmezNwAlAFWAt0LHbMzcA03+PBwGyv4w7COZ8PVPc9HhMN5+w7rhbwCbAcSPI67iD8ntsDa4B6vu3GXscdhHNOAcb4HncEtnkd9zGecy+gK/BNKc9fDCzAWbnyDGBFZb5/JLYUugNpqrpFVY8AbwGDih0zCHjV93gO0FtESloaNFyUe86qulhVD/s2l+OshBfOAvk9A/wvMAnIDmZwLgnknEcCL6hqBoCq7g5yjJUtkHNWoLbvcR3+uMJjWFHVTyhhBUo/g4DX1LEcqCsix1fW+0diUmgG7PDbTvftK/EYVc0DDgANghKdOwI5Z38jcD5phLNyz1lETgNaqOr7wQzMRYH8njsAHURkqYgsF5H+QYvOHYGc88PAtSKSjrN+y63BCc0zFf17rxBXF9nxSEmf+IuXWAVyTDgJ+HxE5FogCTjX1YjcV+Y5i0gM8AwwPFgBBUEgv+c4nC6k83Bag5+KSCdVzXQ5NrcEcs5XA6+o6lMicibOao6dVDVSF2V39foViS2FdKCF33Zz/ticLDpGROJwmpxlNddCXSDnjIj0Ae4HklU1J0ixuaW8c64FdAKWiMg2nL7X1DAfbA70//Z7qpqrqluB73CSRLgK5JxHAG8DqOoyIAFnjqBIFdDf+9GKxKSwEmgvIm1EpArOQHJqsWNSget8j68APlLfCE6YKvecfV0pL+EkhHDvZ4ZyzllVD6hqQ1VtraqtccZRklV1lTfhVopA/m+/i1NUgIg0xOlO2hLUKCtXIOe8HegNICIn4ySFPUGNMrhSgWG+KqQzgAOq+lNlvXjEdR+pap6IjAUW4lQuzFTV9SLyCLBKVVOBGThNzDScFsJg7yI+dgGe8xNATeCfvjH17aqa7FnQxyjAc44oAZ7zQqCfiGwA8oG7VXWfd1EfmwDP+U7gZRG5HacbZXg4f8gTkTdxuv8a+sZJxgPxAKo6DWfc5GIgDTgMXF+p7x/GPztjjDGVLBK7j4wxxhwlSwrGGGOKWFIwxhhTxJKCMcaYIpYUjDHGFLGkYEKOiOSLyFd+X63LOLZ1abNJVvA9l/hm4lzrmyLixKN4jdEiMsz3eLiINPV7brqIdKzkOFeKSGIA3/MXEal+rO9tooMlBROKslQ10e9rW5Ded4iqdsGZLPGJin6zqk5T1dd8m8OBpn7P3aiqGyolyt/inEpgcf4FsKRgAmJJwYQFX4vgUxFZ7fvqWcIxp4jIF77Wxdci0t63/1q//S+JSGw5b/cJ0M73vb198/Sv881zX9W3f6L8tj7Fk759D4vIXSJyBc78UrN871nN9wk/SUTGiMgkv5iHi8jzRxnnMvwmQhORF0VklTjrKEzw7bsNJzktFpHFvn39RGSZ7+f4TxGpWc77mChiScGEomp+XUfv+PbtBvqqalfgKuC5Er5vNDBZVRNxLsrpvmkPrgLO8u3PB4aU8/4DgXUikgC8Alylqp1xZgAYIyL1gT8Bp6jqqcCj/t+sqnOAVTif6BNVNcvv6TnAZX7bVwGzjzLO/jjTWhS6X1WTgFOBc0XkVFV9DmdenPNV9Xzf1BcPAH18P8tVwB3lvI+JIhE3zYWJCFm+C6O/eGCKrw89H2dOn+KWAfeLSHNgrqpuEpHewOnASt/0HtVwEkxJZolIFrANZ/rlE4Gtqvq97/lXgVuAKTjrM0wXkX8DAU/Nrap7RGSLb86aTb73WOp73YrEWQNn2gf/VbeuFJFROH/Xx+MsOPN1se89w7d/qe99quD83IwBLCmY8HE78DPQBaeF+4dFc1T1DRFZAQwAForIjTjTDL+qqvcF8B5D/CfME5ES19jwzcfTHWcStsHAWOCCCpzLbOBK4FvgHVVVca7QAceJswLZROAF4DIRaQPcBXRT1QwReQVnYrjiBFikqldXIF4TRaz7yISLOsBPvjnyh+J8Sv4dETkB2OLrMknF6Ub5ELhCRBr7jqkvga9P/S3QWkTa+baHAh/7+uDrqOp8nEHckiqADuJM312SucClOOsAzPbtq1CcqpqL0w10hq/rqTZwCDggIscBF5USy3LgrMJzEpHqIlJSq8tEKUsKJlxMBa4TkeU4XUeHSjjmKuAbEfkKOAlnycINOBfPD0Tka2ARTtdKuVQ1G2cGyn+KyDqgAJiGc4F93/d6H+O0Yop7BZhWONBc7HUzgA1AK1X9wrevwnH6xiqeAu5S1bU4azOvB2bidEkVSgEWiMhiVd2DUxn1pu99luP8rIwBbJZUY4wxfqylYIwxpoglBWOMMUUsKRhjjCliScEYY0wRSwrGGGOKWFIwxhhTxJKCMcaYIpYUjDHGFPl/zB2bZgYa6voAAAAASUVORK5CYII=\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "y_true = coding_scores_not_low_complexity_unspliced['in_cds_no_stop_codon']\n",
    "y_score = coding_scores_not_low_complexity_unspliced['jaccard_in_peptide_db']\n",
    "y_pred = y_score > 0.9\n",
    "\n",
    "fpr, tpr, thresholds = metrics.roc_curve(y_true, y_score)\n",
    "auc = metrics.auc(fpr, tpr)\n",
    "print(f\"AUC: {auc}\")\n",
    "\n",
    "fig, ax = plt.subplots()\n",
    "lw = 2\n",
    "ax.plot(fpr, tpr, lw=2, color='darkorange')\n",
    "ax.plot([0, 1], [0, 1], color='navy', lw=lw, linestyle='--')\n",
    "ax.set(xlabel='False Positive Rate', ylabel='True Positive Rate')\n",
    "\n",
    "print(\"accuracy score:\", metrics.accuracy_score(y_true, y_pred))\n",
    "\n",
    "print(metrics.classification_report(y_true, y_pred))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {
    "execution_event_id": "4110a9f9-c104-4005-b3fd-5ecb5e61d217",
    "last_executed_text": "\n\nsns.violinplot(x='in_cds_no_stop_codon', y='jaccard_in_peptide_db', data=coding_scores_not_low_complexity_unspliced, \n               bw=0.1, cut=True)",
    "persistent_id": "72511a16-18da-4358-93cf-c36a17c23f8e"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<matplotlib.axes._subplots.AxesSubplot at 0x7f3e49d8c0b8>"
      ]
     },
     "execution_count": 87,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAYYAAAELCAYAAADdriHjAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDMuMC4zLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvnQurowAAIABJREFUeJzt3Xl4XPV97/H3d0aStcubvGDjFcfGoVwIhkChBZI0hUChTbgFmvRpetvSJYQEaLMXCLT3ydKkFwgk4eayZGlYUkwMpkDSgNnBS8xuwLFlWxCwLNuytVmame/9Y87IM/JIOrY1c0aaz+t55plzzpxz5qvx+Hzmd5bfMXdHREQkIxZ1ASIiUloUDCIikkPBICIiORQMIiKSQ8EgIiI5FAwiIpJDwSAiIjkUDCIikkPBICIiOSqiLuBQTJ061efNmxd1GSIiY8ratWt3uHvzSPONyWCYN28ea9asiboMEZExxcy2hJlPu5JERCSHgkFERHIoGEREJIeCQUREcigYREQkh4JBRERyKBhERCSHgkFERHKMyQvcZHR8+9vf5tHHVgHQ2NjAzTfdRFNTU8RViUjU1GIoU4lEgocffoTdiQp2VkzhrdZW1q9fH3VZIlICChoMZnarmW03s5eHeN3M7AYz22hmL5rZ+wpZj+zX0tLCvn299M04ht75vwexOK+++mrUZYlICSj0rqTbge8APxzi9bOBRcHj/cB3g2cpsC1b0l2mpGonQyyO10ykpaUl2qJkVLg73d3d7Nmzh46ODvbs2cPevXsHhrMfiUQi6nLHvHg8Tn19PU1NTTQ2Ng48mpqaaGhoGBiuq6sjFhsbO2kKGgzu/riZzRtmlvOBH7q7A8+a2UQzm+nuvy1kXQKtra0ApCY0AJCoamDL1m1RliR59Pf352zg8z3Sr3WwJ3jeu7eLRDI55DprK436SqeuIkmlpYr41xycnoTRkzBqKpyaCo+6nCElMbYlKujsN7r6nKEqjZnRUF9HY1MjjY0TaQyCZHCgDA6XCRMmFPXvgegPPs8CsrdGrcE0BUOBdXZ2YhWVEEt/BbxiAt3duyKuShKJBF/+0pfYvPk37N3bSU/vviHnrYxDQxXUVaSojyeZVpliQa1T35QKNvxOfWV6OPNcV+HEx8CP1jc7Krhhw3T+8LxzePjBlXzyqHdZ1FT6rZuUQ1fC6Oo3OvtjdPZb8IjRmTA6+3vo7Gun87fG29sqeCORnmdfYujgm1BVSUN9PdNnzuTaa69jypQpBf87og4GyzMt7ydkZpcAlwDMmTOnkDWVhUQiAZa1hbAYicTQvzKleKZNn05nZycxi5FM7qavv/+AeczSodBUmUw/qlI0TUgxsSpFU5UzcUIqPa0qxYR4BH/EYdqwq4I//Mg5/MOnPo07bFj3wzERDDGDhkqnodKBFP0p6NgXo6Mvxu4+o6Mvtv+xL0FFLEbMKtiZgmSexltFPE5jYyOTp0xh2rTpVFQUZ5MddTC0Akdmjc8G3s43o7vfAtwCsGzZstJtV44RjY2NeP8+SKUgFsMS+2ior4+6rLJXUVHBlVdeOTDu7nR1dbFz586hH+072Nrezu539pDyA/9r1FQYDVVOTUWK2liS2opgONhFk/Mc9wNer4qlg6iYlkxKcMODK3GHR/5rJZctiSYU+pLQkzS6g91a3YlY8Gz7n5ODx+P0JOPsDXYt5dPUUM/kKVOYPGUqCyZPZvKgx5QpU5g8eTINDQ1YsT98og+GFcClZnYn6YPOHTq+UBzNzembOFl/Fz6hgVhfFzOOnBFxVTKYmVFfX099ff2ILeVkMklHR8dAYLS3tw8Md3R00NnZSVdXJ7v27mVbVydde7rp7unF84RJtrhBTaXtD4xYMtjvvz886kY5XBY1JbhsybtsWPdDLluSOKTWwuFs1LsTRne/kwhxCKamegJ1tbXU1ddTX99Ac309dXV1NDU15WzkM49JkyYV7Zf/oSpodWb2U+AMYKqZtQJXA5UA7v494EHgI8BGoBv4y0LWI/vNnz8fgHj3ThJV9VT07mL+/FMirkoORzweH9j4hJVKpejp6QlCo2vgOfPIP54Ol9aDDJeJ1TCzpo9ZdUmOqE1yRF2SWXVJ6ivzL7uoaeRA6E3A291x3uqq4O2ueHq4u4qdvRz8Rr2hgWn1DdTV1VFXV0d9sIEfbry2tpZ4fAzuqxtBoc9KuniE1x34VCFrkPwWLlxILBYj1tVGrLoJT/SxePHiqMuSIovFYgMbukOVCZfsIMkOlMxwW1sbLZs389jWLezr23/cpGmCcURNXxAUiYHAaKz0gVZGV78FAZB+vN0V5+2eKtp79tdREY8ze/Yslh6zgBkzZuRsyMtpoz4aSrs9IwVTXV3N/AULeGN7G6kJjQAcffTREVclY1F2uEybNm3E+VOpFO+++y5btmyhpaWFrVu30rJ5E8+0tND9Vu/AfO+fto+/f28nyzfX8POW2oHpE6oqOXLOHI6fN5958+Yxd+5c5s6dyxFHHFHyu2jGCn2KZex3jjmGTfevJFnVQG1dnc72kqKIxWLMnDmTmTNncvLJJw9Md3d27NjBli1beO6557jnnnvYvLeC7T1xzjzzTD784Q8zb948pk+fPmYuFBurFAxlbMmSJdx3331U7trMkuOO1X82iZSZ0dzcTHNzMyeccAI7duzg0UcfpbGxkS996UtUVlZGXWLZ0JagjB111FEAWLKfoxYujLgakf3MjNNOOw2AE044QaFQZAqGMjZ79uyBYe1GklI10hlPMvoUDGWsurp6YHj69OkRViJyoMyFXVFc4FXuFAwCcFDnvosUk1oMxadgEAAaGhqiLkEkh1oK0VEwCAA1NTVRlyAiJULBIABUVVVFXYKIlAgFgwCoawARGaBgEABd3CYiA7Q1EEAH+kRkPwWDiIjkUDAIoHPFRWQ/BYMACgYR2U/BIICCQUT2UzCIiEgOBYMAajGIyH4KBgEUDCKyn4JBRERyKBhERCSHgkFERHIoGEREJIeCQUREcigYREQkh4JBAPWuKiL7KRhERCRHwYPBzM4ys9fNbKOZfSHP63PM7FEz+7WZvWhmHyl0TSIiMrSCBoOZxYGbgLOBpcDFZrZ00GxfAe529+OBi4CbC1mTiIgMr9AthpOAje6+yd37gDuB8wfN40BjMNwEvF3gmkREZBgVBV7/LGBb1ngr8P5B81wDPGJmnwbqgA8VuCbJQ30liUhGoVsM+U51GbwFuhi43d1nAx8BfmRmB9RlZpeY2RozW9PW1laAUsubgkFEMgodDK3AkVnjszlwV9FfAXcDuPszQDUwdfCK3P0Wd1/m7suam5sLVG75SqVSUZcgIiUiVDCY2RQzu9HM1pnZWjO73symhFh0NbDIzOabWRXpg8srBs2zFfhg8D5Hkw4GNQmKTC0GEckI22K4E9gOfAy4gPSG+66RFnL3BHAp8DDwGumzj14xs2vN7LxgtiuBvzGzF4CfAp90baWKLplMRl2CiJSIsAefJ7v7dVnj/2JmfxxmQXd/EHhw0LSrsoZfBU4NWYcUiIJBRDLCthgeNbOLzCwWPP4UWFnIwqS4EolE1CWISIkYtsVgZntJn0VkwBXAj4KX4kAncHVBq5OiUYtBRDKGDQZ3byhWIRIttRhEJGOkFsP7hnvd3deNbjkSFZ2uKiIZIx18/lbwXA0sA14gvVvpWOA54LTClSbFpGAQkYxhDz67+5nufiawBXhfcIHZCcDxwMZiFCgiIsUV9qykJe7+UmbE3V8GjitMSSIiEqWw1zG8ZmY/AH5M+iylT5C+YE3GCd3BTUQywgbDXwJ/D3wmGH8c+G5BKpJIxGK6mZ+IpIUKBnfvBf49eBzAzP7T3T82moVJccXj8ahLEJESMVo/ExeM0nokIgoGEckYrWBQp3djnIJBRDK0Y1kABYOI7DdawaBTWsY4BYOIZIQOBjOrMbPFQ7z8+VGqRyKis5JEJCPsHdz+CFgPPBSMH2dmA3dic/dHClOeFIuuYxCRjLA/E68BTgJ2A7j7emBeYUqSKCgYRCQjbDAk3L2joJVIpBQMIpIR9srnl83sz4C4mS0CLgOeLlxZUmy6zbaIZIRtMXwaeC+wD/gpsAf4bKGKkuJTMIhIRtguMbqBLwcPGYcUDCKSMdId3O5nmKua3f28Ua9IIqFgEJGMkVoM/xY8fxSYQbrbbYCLgZYC1SQRUDCISMawweDuqwDM7Dp3//2sl+43s8cLWpmIiEQi7MHnZjMb6EHVzOYDzYUpSaKgFoOIZIQ9XfVy4DEz2xSMzwP+tiAVSSR0HYOIZIQ9K+mh4PqFJcGkDe6+r3BliYhIVEY6K+kD7v4rM/vooJcWmhnufm8Ba5MiUotBSo12b0ZnpBbD6cCvgD/K85oDCoZxQsEgpSYTDPpuFt9IZyVdHQxe6+6bs18LDkCPyMzOAq4H4sAP3P1reeb5U9Id9Tnwgrv/WZh1y+jRfz4RyQh7VtJ/5pn2s5EWMrM4cBNwNrAUuNjMlg6aZxHwReBUd38v6mojEgoGEckY6RjDEtJ9JDUNOs7QCFSHWP9JwEZ33xSs707gfODVrHn+BrjJ3XcBuPv28OXLaNGNeqTUpFIpAJLJZMSVlJ+RjjEsBs4FJpJ7nGEv6Q36SGYB27LGW4H3D5rnPQBm9hTp3U3XuPtDg1dkZpcAlwDMmTMnxFuLyFi2fXv6N6IOQhffSMcYfg783MxOcfdnDmH9+fZPDP5XrgAWAWcAs4EnzOwYd989qJZbgFsAli1bpm+KyDj31ltvAbCzvT3iSspP2P0H75rZ/WbWZmbbzezn2VdCD6MVODJrfDbwdp55fu7u/cEB7tdJB4UUkX6VSalpa2sLnt+NuJLyEzYY/gO4G5gJHAHcQ/q+DCNZDSwys/lmVgVcBKwYNM99wJkAZjaV9K6lTUhRZfbnipSK3t4eAPb16lraYgsbDObuP3L3RPD4McN0x53h7gngUuBh4DXgbnd/xcyuNbNMl90PA+1m9irwKPBP7q62Y5EpGKTUTJs2HYDm6dMjrqT8hO0r6VEz+wJwJ+lAuBBYaWaTAdx951ALuvuDwIODpl2VNezAFcFDIqJgkFIzc+ZMYH9ASPGEDYYLg+fBHef9L9JBEeZ4g5SwRCIRdQkiOWbMmAFARUXYzZSMlrCd6IW6ylnGLrUYpNTU1NQAusYmCqE+cTOrNbOvmNktwfgiMzu3sKVJManFIKVKV+UXX9govg3oA343GG8F/qUgFUkk1GKQUpM5hVpXPhdf2GBY6O7fAPoB3L2H/BevyRil/3xSajo6OqIuoWyFDYY+M6shOEXVzBYCOrl4HFGLQUpN5srnrq6uiCspP2EP918NPAQcaWY/AU4FPlmooqT41GKQUrNtW7qbtW1bt0RcSfkJe1bSL8xsHXAy6V1In3H3HQWtTETKWuaEiER/f8SVlJ+DOUH4dOA00ruTKoHlBalIIqG+kqTU1NXVAVAbPEvxhD1d9Wbg74CXgJeBvzWzmwpZmIiUt+lBVxjTZ8yMuJLyE7bFcDpwTNB9BWZ2B+mQEBEpiLlz5wJQW1sbcSXlJ+xZSa8D2XfHORJ4cfTLkajoIiIpNZldSfF4POJKyk/YFsMU4DUzez4YPxF4xsxWALj7eUMuKSIiY0rYYLhq5FlkLFN/NFKq1JotvrCnq64a7nUze8bdTxmdkiQK+s8npUpnzBXfaP1MrB6l9UhEFAxSajLfSX03i2+0gkGRPsbpP5+UKrUYik87lgVQMEjp0XcyOqMVDPoXHOP0n1BEMkYrGP58lNYjEdFZSSKSEbZLjI+a2Ztm1mFme8xsr5ntybzu7i8XrkQRESmmsNcxfAP4I3d/rZDFSHS0K0lEMsLuP3hXoSAiUh7CthjWmNldwH1k3bnN3e8tSFVSdDolUEQywgZDI9ANfDhrmgMKhnFCwSAiGWG7xPjLQhci0dI9n0UkY9hgMLPPufs3zOxG8lzd7O6XFawyKSrd81lEMkZqMWQOOK8pdCESLQWDiGQMGwzufn/wfMdw85nZje7+6SFeOwu4HogDP3D3rw0x3wXAPcCJ7q4gKjIFg4hkjNblrqfmm2hmceAm4GxgKXCxmS3NM18DcBnw3CjVIwdJwSAiGYXuB+EkYKO7b3L3PuBO4Pw8811H+iK63gLXI0NIJBJRlyAiJaLQwTAL2JY13hpMG2BmxwNHuvsDBa5FhqGzkkQko9C9q+abPnB2k5nFgH8HrhzxDcwuMbM1Zramra3t0KqUIWlXkohkjFYwXD/E9FbgyKzx2cDbWeMNwDHAY2bWApwMrDCzZYNX5O63uPsyd1/W3Nw8OlXLAF3gJiIZI13HcD/D3J3N3c8Lnm8fYpbVwCIzmw+8BVwE/FnW8h3A1Kz3ewz4R52VVHwKBhHJGOk6hn8Lnj8KzAB+HIxfDLSMtHJ3T5jZpcDDpE9XvdXdXzGza4E17r7ikKqWUaf7MYhIxkjXMawCMLPr3P33s16638weD/MG7v4g8OCgaVcNMe8ZYdYpIiKFE/ZnYrOZLciMBLuGtKN/HFGLQUQywvauejnpA8SbgvF5wCUFqUgioRv1iEjGiMEQnFK6B1gELAkmb3D3fUMvJWONWgwikjFiMLh7ysy+5e6nAC8UoSaJgFoMIpIR9mfiI2b2MdPWY9xSi0FEMsIeY7gCqAMSZtZL+opmd/fGglUmRaXMF5GMsHdwayh0IRIttRhEJCNsiwEzm0T6AHR1Zpq7h7qWQUqfWgwikhEqGMzsr4HPkO7raD3pPo2eAT5QuNKkmNRiEJGMsFuDzwAnAlvc/UzgeEBdnI4jajGISEbYYOh1914AM5vg7huAxYUrS4pNLQYRyQh7jKHVzCYC9wG/MLNd5HafLWOcWgwikhH2rKQ/CQavMbNHgSbgoYJVJUWnYBCRjFD7D8zsZDNrgIEeVx8lfZxBRETGmbA7lr8LdGaNdwXTRERknAkbDOZZt/hy9xQHcQ2ElD7dwU1EMsIGwyYzu8zMKoPHZ4BNIy4lIiJjTthg+Dvgd0nft7kVeD+6H8O4kkwmoy5BREpE2LOStgMXFbgWiZB2JYlIRtizku4IrmPIjE8ys1sLV5YUWyqViroEESkRYXclHevuuzMj7r4Lna46rmhXkohkhA2GWNC7KgBmNhmdlTSuqMUgIhlhN+7fAp42s58F4/8T+NfClCRRUItBRDLCHnz+oZmtBc4kffe2j7r7qwWtTIoqkUhEXYKIlIjQu4Pc/RUzayO4UY+ZzXH3rQWrTIqqv78/6hJEpESEPSvpPDN7E9gMrAJagP8qYF1SZH19fVGXICIlIuzB5+tI37XtDXefD3wQeKpgVUlRZF+70NvbG2ElIlJKwgZDv7u3kz47KebujwLHFbAuKYLsMOju7o6wEhEpJWGPMew2s3rgCeAnZrYd0NHKMa6zszPvsIiUt7AthvOAbtL3fn4I2AicG2ZBMzvLzF43s41m9oU8r19hZq+a2Ytm9t9mNjds8XJ49u7dm3dYRMrbsMFgZk8Gg+8Cu4FdwHeA/w1sNrPNZvYPwywfB24CzgaWAheb2dJBs/0aWObuxwI/A75xKH+IHLw9e/bkHRaR8jZsMLj7acFzg7s3Bo+G4NEELCPdihjKScBGd9/k7n3AncD5g97jUXfP7OB+Fph9qH+MHJyOjo6B4d27dw8zp4iUk7C7kvIKDkifMcwss4BtWeOtwbSh/BVDnAZrZpeY2RozW9PW1nawpUoemTBIVjfR0aEWg4ikHVYwALj7b4d5Od8d5vP272xmnyDdAvnmEO9zi7svc/dlzc3NB1+oHCDTYkhVN7Fz166IqxGRUlHojvBagSOzxmcDbw+eycw+BHwZON3d9xW4Jgl0dHSAxYjt28uepE4yE5G0w24xjGA1sMjM5ptZFemb/azInsHMjge+D5wX3BBIiiQdDIYl++jq7FRHeiICFDgY3D0BXAo8DLwG3B30uXStmZ0XzPZNoB64x8zWm9mKIVYno2z37t04MbAY7q4zk0QEKMI9Fdz9QeDBQdOuyhr+UKFrkPzad+5Mh4Klfx/s2rWLSZMmjbCUiIx3hd6VJCVsx452PBYDiwPQ3t4ecUUiUgoUDGWqr6+PvXs60qEQS38NduzYEXFVIlIKFAxlavv29HF+j8XxoMXwzjvvRFmSiJQIBUOZGggBi6fPTJpQz29/O9wlKSJSLhQMZaq1tTU9EEu3FhJVDfuniUhZUzCUqdbWVixeMbAbKVndwNZtCgYRUTCUrW3btpGqbhrotCRV3UTn3j26lkFEFAzlatPmFhITGgfGU9UTAdiyZUtUJYlIiVAwlKG9e/fStv1dUjWTB6alatIXtv3mN7+JqiwRKREKhjL0xhtvAJCs3R8MXlWHVVazYcOGqMoSkRKhYChDa9euBYuRbJi+f6IZffUzWL16De55e0YXkTKhYCgz7s4TTz5Fsn4axKtyXks2zaK9fQdvvvlmRNWJSClQMJSZl156iW1bt9A/5agDXuufNA+LV3D//fdHUJmIlAoFQ5m58847sYoJ9E9ecOCLFRPYN2k+Dz/8iDrUEyljCoYy8tJLL/H000/TO/29EM/f43rfzGPp6+/njjvuKHJ1IlIqFAxlIpVKcdPNN2NVtfRNfy8AE7Y+S7y7nXh3OzUbHmTC1mfx6ib6mhfzwAMP0NLSEm3RIhIJBUOZ+OUvf8mG116je9YJEK8EINa9E0v2Y8l+Kva+Q6x7JwB9RxyHxyq58Tvf0RlKImVIwVAG9u3bx/e+/31S9c0k8hx0Hswra+iZeRxr16zh+eefL0KFIlJKFAxlYOXKlexsb6d31glgFmqZ/mlLoLqBW2+9Ta0GkTKjYBjnEokEP/7JT0g2zCDZMDP8grE4vTOO5fXXN6QviBORsqFgGOeeeeYZdra3pw84h2wtZPRPWYhVVrNixYoCVScipUjBMM4tv+8+mFBHYuKRB79wrIJ9U47iySefHLgVqIiMfwqGcaylpYV1a9eyb+oSsEP7p+5rXkIqlVKrQaSMKBjGsdtuuw2LVdDf/J5DXodXN5KYNId7ly/X1dAiZULBME499dRTrFq1it6Zx+KVNYe1rt5ZJ9LT08sNN9wwStWJjExnw0VHwTAOtba28rWvfx2vnUzfjGMPe31e00TvzONYtWoVy5cvH4UKRcKzgzxpQg6fgmGc2blzJ1dc+Y909vTTtfBMiI3OP3HfzN8hMXEO199wA6tWrRqVdYqEoZZD8SkYxpHu7m4+97nP07ajnc6jPoRXN43eyi1Gz4IzSNU1c911/8ILL7wweusWyUOBEJ2CB4OZnWVmr5vZRjP7Qp7XJ5jZXcHrz5nZvELXNB719/fzlX/+Zzb+ZiNdC84gVd88+m8Sr6Br0YdIVNbxhS9+kU2bNo3+e4gEMsGQSqUirqT8FDQYzCwO3AScDSwFLjazpYNm+ytgl7sfBfw78PVC1jRe3XPPPaxbu5aeuaeSPJRrFsKqqKZz0YfpTcC1116nX3VSMDt27Ii6hLJV6BbDScBGd9/k7n3AncD5g+Y5H8h0/v8z4IOmo00Hpa+vj7vv+RnJxiNIHMapqWH5hHq6Z51AS8tmVq9eXfD3k/K0bds2AHbt3BlxJeWn0MEwC9iWNd4aTMs7j7sngA5gSoHrGlc2bNjA7l076Zu66OAWTPZRXV3NBRdcQHV1NST7Qi+amLwAYnGeeOKJg6xWJJzMdTM7duiq+2IrdDDk++U/eN9DmHkws0vMbI2ZrWlraxuV4saLhQsXUlFRSbzz4D4XS/Rx7rnncumll3LOOedgifDBEOveCakkS5cO3jMoMjo6O/cGz10RV1J+8t/fcfS0Atk7vGcDbw8xT6uZVQBNwAFtR3e/BbgFYNmyZdqxnaWuro5TTjmZJ59+lkTTrNDHGLyiigceeAB3Z+XKlXhFbajlrL+Hmq3PUFFZyWmnnXY4pYsMqbl5GvAKzc0FOJFChlXoFsNqYJGZzTezKuAiYHCnOyuAvwiGLwB+5TqiedA++9nPsmDeXGo3/pLKttdDLZNqmEFnrJa773+YzlgtqYYZIy5jvR3Ub1jJhP49XPvVr9LQ0HC4pYvktXjx4vTzkqMjrqT8WKG3wWb2EeD/AHHgVnf/VzO7Fljj7ivMrBr4EXA86ZbCRe4+7HmQy5Yt8zVr1hS07rGou7ubq66+mjWrV5OYNI/eOSfjVeFaASPyFJXvvkLN27+mvq6Ob3z9axx9tP7DSuG4Ox0dHTQ2NhIbpQs1y52ZrXX3ZSPONxZ/nCsYhpZIJLjrrru4/fbbSbjRc8TxpGoP81h+sp+at9dhXe2cfPLJXH755UyfPn10ChaRogkbDIU+xiBFVlFRwcc//nFOP/10vv3tb7Nu3XOjst6Jkybz2X+6htNPP11914iMc2oxjGPuzmuvvUZPT89hrcfMWLx4MXV1daNUmYhEQS0Gwcx0OqmIHDQd0RERkRwKBhERyaFgEBGRHAoGERHJoWAQEZEcCgYREcmhYBARkRxj8gI3M2sDtkRdxzgyFdDtsqQU6bs5uua6+4jd1Y7JYJDRZWZrwlwNKVJs+m5GQ7uSREQkh4JBRERyKBgEgjvjiZQgfTcjoGMMIiKSQy0GERHJoW63xyEzSwIvZU36Y3dvGWLeecAD7n5M4SsTATObAvx3MDoDSAJtwfhJ7t4XSWEyQMEwPvW4+3FRFyGSj7u3A8cBmNk1QKe7/1v2PJa+TaC5e6r4FYp2JZUJM5tnZk+Y2brg8bt55nmvmT1vZuvN7EUzWxRM/0TW9O+bWbz4f4GMd2Z2lJm9bGbfA9YBR5rZ7qzXLzKzHwTD083sXjNbE3w3T46q7vFIwTA+1QQb8fVmtjyYth34A3d/H3AhcEOe5f4OuD5obSwDWs3s6GD+U4PpSeDjhf8TpEwtBf6fux8PvDXMfDcA3wgufvtT4AfFKK5caFfS+JRvV1Il8B0zy2zc35NnuWeAL5vZbOBed3/TzD4InACsTrfuqSEdMiKF8Bt3Xx1ivg8Bi4PvJMAkM6tx98O7wbkACoZycjnwLvA/SLcUewfP4O7/YWbPAecAD5vZXwMG3OHuXyxmsVK2urKGU6S/fxnVWcOGDlQXjHYllY8m4Lf256AxAAAExUlEQVTBwbw/Bw44TmBmC4BN7n4DsAI4lvTZIxeY2bRgnslmNrd4ZUu5Cr6ru8xskZnFgD/JevmXwKcyI0FLWEaJgqF83Az8hZk9S3o3UleeeS4EXjaz9cAS4Ifu/irwFeARM3sR+AUws0g1i3weeIj0D5TWrOmfAk4NTpJ4FfibKIobr3Tls4iI5FCLQUREcigYREQkh4JBRERyKBhERCSHgkFERHIoGEREJIeCQSJhZk8XcN2PmVlJ3EDezCaa2T9EXUc2M2sxs6lR1yGlS8EgkXD3A3p3HacmAiUVDCIjUTBIJMysM3g+I/iF/zMz22BmP7GsntHyLHeimT1tZi8E3S03mFmNmd0ZXAV7F+mO/jCzuJndHnTl/JKZXT7Meh8zs68H63zDzH4vmF5tZrcFy//azM4cZh35ui3/GrAwmPZNS/tmVk0XZn0Oj5vZcjN71cy+F3QDMdR7nRV0n/6Cmf13MG2ymd0XvPezZnZsMH2KmT0S1P99svofMrMrglpeNrPPBtPmmdlrZvZ/zeyVYNmaoWqRccjd9dCj6A/SN2cBOAPoAGaT/qHyDHDaEMtUAZuAE4PxRtIdQV4B3BpMOxZIkO42/ATgF1nLTxymnseAbwXDHwF+GQxfCdwWDC8BtgLVQ6zjRuDjWbXWAPOAl7Pm+RjpbkXiwPRgfTODz6EXWBC89gvggiHepxnYBswPxidnvf/VwfAHgPXB8A3AVcHwOYADU4PP5yWgDqgHXgGOD2pOAMcFy9wNfCLq74wexXuoxSCl4Hl3b/V0p2nrSW+Y8llMuiPA1QDuvsfdE8DvAz8Opr0IvBjMvwlYYGY3mtlZwJ4R6rg3eF6bVcNpwI+CdW8AtpC/y3JIh9qXzOzzwFzP3wX0acBP3T3p7u8Cq4ATg9eed/dN7p4EfhrMm8/JwOPuvjmoa2eeWn8FTDGzJnI/n5XArqz5l7t7l7t3Bn//7wWvbXb39Xk+DykDCgYpBfuyhpMM3R28kf61m88B0919F+luxh8j3enaSDdzydSRXcOQu7XyvN9/AOcBPaS7Lf9AntmGW9/gv2Gov3WozyHfun3Qc9hawv6byDikYJCxZANwhJmdCBAcX6gAHie4q5yZHUN6dxLBmTcxd/9P4J+B9x3Ce2av+z3AHOD1fDMO0W35XqBh0PouDI5/NJP+Nf988NpJZjY/OLZwIfDkEDU9A5xuZvOD952cp9YzgB3uvmfQ9LOBSVnz/7GZ1ZpZHelurZ8I86HI+KZfATJmuHtfcLD2xuBgaA/pO3l9F7gt6BZ8Pfs3tLOC6ZkfQIdys6Gbge+Z2Uuk97t/0t33DTHvhcAnzKwfeAe41t13mtlTZvYy8F/A54BTgBdI/4r/nLu/Y2ZLSG/wvwb8DumN9vJ8b+LubWZ2CXBv8LdtB/4AuCbrc+gG/iJY5KvAT81sHeldV1uD9awzs9vZ/3n9wN1/bWbzDvIzknFG3W6LlIDgF/4/uvu5Udciol1JIiKSQy0GKUlmthyYP2jy59394cNc703AqYMmX+/utx3EOv4Q+PqgyZvd/U/yzX84LH0P7gmDJv+5u7802u8lkqFgEBGRHNqVJCIiORQMIiKSQ8EgIiI5FAwiIpJDwSAiIjn+P0FVBUEXKCEVAAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "\n",
    "\n",
    "sns.violinplot(x='in_cds_no_stop_codon', y='jaccard_in_peptide_db', data=coding_scores_not_low_complexity_unspliced, \n",
    "               bw=0.1, cut=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "persistent_id": "f6a4203e-98e0-4971-ba68-c0885953466f"
   },
   "source": [
    "## Where do they disagree?\n",
    "\n",
    "### False negative - Low jaccard but the read is in the coding dataset\n",
    "Truly a coding read but low \"coding score\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {
    "execution_event_id": "d5caa457-ae0c-48fa-b4dc-3f8801d0f2da",
    "last_executed_text": "coding_scores_not_low_complexity_unspliced.query(\"jaccard_in_peptide_db < 0.9 & in_cds_no_stop_codon == True\").shape",
    "persistent_id": "23edfe41-6d93-44d6-9324-ba4ac19d25e0"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(3, 6)"
      ]
     },
     "execution_count": 90,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coding_scores_not_low_complexity_unspliced.query(\"jaccard_in_peptide_db < 0.9 & in_cds_no_stop_codon == True\").shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {
    "execution_event_id": "3f4f1c9d-cf40-45f8-8c76-e00bb7d57491",
    "last_executed_text": "coding_scores_not_low_complexity_unspliced.query(\"jaccard_in_peptide_db < 0.9 & in_cds_no_stop_codon == True\").head()",
    "persistent_id": "1254868e-73a0-4c0a-82c6-bb1b85e78cb0"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>read_id</th>\n",
       "      <th>jaccard_in_peptide_db</th>\n",
       "      <th>n_kmers</th>\n",
       "      <th>classification</th>\n",
       "      <th>short_read_id</th>\n",
       "      <th>in_cds_no_stop_codon</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>289</th>\n",
       "      <td>SRR306838.8536423 Ibis_Run100924_C3PO:6:41:4477:5833/1</td>\n",
       "      <td>0.588235</td>\n",
       "      <td>17</td>\n",
       "      <td>non-coding</td>\n",
       "      <td>SRR306838.8536423</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>740</th>\n",
       "      <td>SRR306838.4158712 Ibis_Run100924_C3PO:6:19:4365:17337/1</td>\n",
       "      <td>0.533333</td>\n",
       "      <td>15</td>\n",
       "      <td>non-coding</td>\n",
       "      <td>SRR306838.4158712</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>991</th>\n",
       "      <td>SRR306838.20339512 Ibis_Run100924_C3PO:6:102:15178:5329/1</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>16</td>\n",
       "      <td>non-coding</td>\n",
       "      <td>SRR306838.20339512</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                       read_id  \\\n",
       "289     SRR306838.8536423 Ibis_Run100924_C3PO:6:41:4477:5833/1   \n",
       "740    SRR306838.4158712 Ibis_Run100924_C3PO:6:19:4365:17337/1   \n",
       "991  SRR306838.20339512 Ibis_Run100924_C3PO:6:102:15178:5329/1   \n",
       "\n",
       "     jaccard_in_peptide_db  n_kmers classification       short_read_id  \\\n",
       "289               0.588235       17     non-coding   SRR306838.8536423   \n",
       "740               0.533333       15     non-coding   SRR306838.4158712   \n",
       "991               0.000000       16     non-coding  SRR306838.20339512   \n",
       "\n",
       "     in_cds_no_stop_codon  \n",
       "289                  True  \n",
       "740                  True  \n",
       "991                  True  "
      ]
     },
     "execution_count": 89,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coding_scores_not_low_complexity_unspliced.query(\"jaccard_in_peptide_db < 0.9 & in_cds_no_stop_codon == True\").head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {
    "execution_event_id": "74205ae1-615d-4c09-953f-a0881183b10e",
    "last_executed_text": "df = coding_scores_not_low_complexity_unspliced.query(\"jaccard_in_peptide_db < 0.9 & in_cds_no_stop_codon == True\")\nfor read_id in df.short_read_id:\n    ! samtools view $folder/$unspliced_bam | grep $read_id",
    "persistent_id": "4acd19e5-3727-407f-adb5-b0676c493cc6"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SRR306838.8536423\t0\tchr1\t39448668\t255\t70M\t*\t0\t0\tAAAGACTTTGCTTCCCGAAGATACTCAGAAACTTGACAATTTCCTAGGAGAAGTCAGAGACAAATGGGAT\t-0454575:38569357656746757387639475375567787758567758596:6957665665585\tNH:i:1\tHI:i:1\tAS:i:66\tnM:i:1\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\n",
      "SRR306838.4158712\t16\tchrX\t129814671\t255\t49M14S\t*\t0\t0\tGACGATGTTGAAGGCGAAGACATAGATTGTGAGGAGGGAGAAAGAAAGGGAAGTAGAAGTAGG\t836857646958738766836846574664767626764766584546386474576222000\tNH:i:1\tHI:i:1\tAS:i:46\tnM:i:1\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\n",
      "SRR306838.20339512\t0\tchr2\t26192330\t255\t54M16S\t*\t0\t0\tCTTCGGACTCGCTAAAATACTATCCATGTCAGAATTCAAATCCTTCCTCTTCACTTATCATCTTCACAAT\t31/3331654544412543575556554166666648555667768766468567466675:68555765\tNH:i:1\tHI:i:1\tAS:i:51\tnM:i:1\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\n"
     ]
    }
   ],
   "source": [
    "df = coding_scores_not_low_complexity_unspliced.query(\"jaccard_in_peptide_db < 0.9 & in_cds_no_stop_codon == True\")\n",
    "for read_id in df.short_read_id:\n",
    "    ! samtools view $folder/$unspliced_bam | grep $read_id"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "persistent_id": "93e35499-4621-47d4-98fd-ec2d1751a603"
   },
   "source": [
    "Read maps to \"MACF1\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "persistent_id": "a05a891d-cc98-4a52-9c1f-6e18cb57c7d7"
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {
    "execution_event_id": "9e155fe4-76eb-49ac-8269-069f24ff1458",
    "last_executed_text": "score_single_sequence(\"AAAGACTTTGCTTCCCGAAGATACTCAGAAACTTGACAATTTCCTAGGAGAAGTCAGAGACAAATGGGAT\", peptide_graph, peptide_ksize)",
    "persistent_id": "50faf94d-9309-4931-8fdf-21f88ecfeff9"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\ttranslation: \tKTLLPEDTQKLDNFLGEVRDKWD\n",
      "\tkmers: FLGEVRD DTQKLDN LPEDTQK KLDNFLG LLPEDTQ EVRDKWD GEVRDKW PEDTQKL TQKLDNF LDNFLGE EDTQKLD NFLGEVR LGEVRDK KTLLPED TLLPEDT DNFLGEV QKLDNFL\n",
      "\tK-mers in peptide database:\n",
      "{('DNFLGEV', 7517166875925614125): 1,\n",
      " ('DTQKLDN', 5951274799996309732): 0,\n",
      " ('EDTQKLD', 14887061883264050655): 0,\n",
      " ('EVRDKWD', 829818351693786468): 1,\n",
      " ('FLGEVRD', 9516214906786795876): 1,\n",
      " ('GEVRDKW', 8385754174638241851): 1,\n",
      " ('KLDNFLG', 3011079111732742999): 1,\n",
      " ('KTLLPED', 14836733754548320237): 1,\n",
      " ('LDNFLGE', 14534966961722312621): 1,\n",
      " ('LGEVRDK', 79391250229811418): 1,\n",
      " ('LLPEDTQ', 14973099275059575838): 0,\n",
      " ('LPEDTQK', 7311319343166724544): 0,\n",
      " ('NFLGEVR', 966627286140107258): 1,\n",
      " ('PEDTQKL', 908198253004630471): 0,\n",
      " ('QKLDNFL', 4722373045142000694): 1,\n",
      " ('TLLPEDT', 7583591577639096127): 0,\n",
      " ('TQKLDNF', 1242088564466415818): 0}\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(0.5882352941176471, 17)"
      ]
     },
     "execution_count": 76,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "\n",
    "score_single_sequence(\"AAAGACTTTGCTTCCCGAAGATACTCAGAAACTTGACAATTTCCTAGGAGAAGTCAGAGACAAATGGGAT\", peptide_graph, peptide_ksize)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {
    "execution_event_id": "a383af3b-9046-4926-896c-fe97c4abd1b6",
    "last_executed_text": "unspliced_bam = 'SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.bam'\n\n# This reads aligns to the CBLN2 \nnoncoding_read_id = 'SRR306838.8536423'\n! samtools view $folder/$unspliced_bam | grep $noncoding_read_id",
    "persistent_id": "783e2fb0-f110-45d7-8d85-507d5ff6a343"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SRR306838.8536423\t0\tchr1\t39448668\t255\t70M\t*\t0\t0\tAAAGACTTTGCTTCCCGAAGATACTCAGAAACTTGACAATTTCCTAGGAGAAGTCAGAGACAAATGGGAT\t-0454575:38569357656746757387639475375567787758567758596:6957665665585\tNH:i:1\tHI:i:1\tAS:i:66\tnM:i:1\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\n"
     ]
    }
   ],
   "source": [
    "unspliced_bam = 'SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.bam'\n",
    "\n",
    "# This reads aligns to the CBLN2 \n",
    "noncoding_read_id = 'SRR306838.8536423'\n",
    "! samtools view $folder/$unspliced_bam | grep $noncoding_read_id"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {
    "execution_event_id": "cbbfe960-18da-4d43-b895-225060159f07",
    "last_executed_text": "bam = 'SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out_CDS_without_stop_codon_or_utr.bam'",
    "persistent_id": "e4c114ee-5198-42ee-97bb-d9974f296764"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SRR306838.7339725\t0\tchr3\t194639082\t255\t57M\t*\t0\t0\tTTGGGGCAAAACACTGGTCCTCTCTGCACTTGTGTGTTCCTTTCTGTAAAAGGAGGA\t-4566323463646545356576866534585655666655868:953234765867\tNH:i:1\tHI:i:1\tAS:i:56\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\n",
      "SRR306838.733972\t16\tchr18\t72538162\t255\t49M\t*\t0\t0\tAGGGAGCTCTGTGTTTATAGAGGAAACACCAAGAAGCCCGAGAATGTGG\t6666776687536257653736764476765533375546588363312\tNH:i:1\tHI:i:1\tAS:i:46\tnM:i:1\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\n"
     ]
    }
   ],
   "source": [
    "unspliced_bam = 'SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.bam'\n",
    "\n",
    "# This reads aligns to the CBLN2 \n",
    "noncoding_read_id = 'SRR306838.733972\t'\n",
    "! samtools view $folder/$unspliced_bam | grep $noncoding_read_id"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {
    "execution_event_id": "b425a531-29c6-47ed-ab32-dd1ce9130736",
    "last_executed_text": "unspliced_bam = 'SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.bam'\n\n# This reads aligns to the CBLN2 \nnoncoding_read_id = 'SRR306838.17561853\t'\n! samtools view $folder/$unspliced_bam | grep $noncoding_read_id",
    "persistent_id": "8b2488c1-3c99-4997-b2e9-eb0caae3979c"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SRR306838.17561853\t16\tchrM\t3811\t255\t67M\t*\t0\t0\tCAAGAACACCTCTGATTACTCCTGCCATCATGACCCTTGGCCATAATATGATTTATCTCCACACTAG\t78677564566847334362785755:66:5525674587556457767724344164757626035\tNH:i:1\tHI:i:1\tAS:i:65\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\n"
     ]
    }
   ],
   "source": [
    "unspliced_bam = 'SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.bam'\n",
    "\n",
    "# This reads aligns to the CBLN2 \n",
    "noncoding_read_id = 'SRR306838.17561853\t'\n",
    "! samtools view $folder/$unspliced_bam | grep $noncoding_read_id"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {
    "execution_event_id": "67ddabce-8f94-4b0b-819d-cfbbabc58699",
    "last_executed_text": "df = coding_scores_unspliced.query(\"jaccard_in_peptide_db < 0.5 & in_cds_no_stop_codon == True\")\nfor read_id in df.short_read_id:\n    ! samtools view $folder/$unspliced_bam | grep $read_id",
    "persistent_id": "eea06bec-d676-41c3-a79f-56bbc89f9e98"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SRR306838.17561853\t16\tchrM\t3811\t255\t67M\t*\t0\t0\tCAAGAACACCTCTGATTACTCCTGCCATCATGACCCTTGGCCATAATATGATTTATCTCCACACTAG\t78677564566847334362785755:66:5525674587556457767724344164757626035\tNH:i:1\tHI:i:1\tAS:i:65\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\n",
      "SRR306838.3607263\t16\tchrM\t8401\t255\t12S60M\t*\t0\t0\tGTCATGCAAAAGAATTACCCCCATACTCCTTACACTATTCCTTATCACCCAACTAAAAACATTAAACACAAA\t557:977:86<;8744366867844:78935686846668:434655896;58479533962386971613.\tNH:i:1\tHI:i:1\tAS:i:55\tnM:i:2\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\n",
      "SRR306838.23727244\t16\tchrM\t14157\t255\t66M\t*\t0\t0\tCCCGAGCAATCTCAATTACAATATATACACCAACAAACAATGTTCAACCAGTAACTACTACTAATC\t6678867778855785236875525246355726854665351366456341337226254200.1\tNH:i:1\tHI:i:1\tAS:i:64\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\n",
      "SRR306838.3497462\t16\tchrM\t8428\t255\t58M\t*\t0\t0\tCCTCATCACCCAACTAAAAACATTAAACACAAACTACCACCTACCTCCCTCACCAAAG\t7:78:573468:571665736743343457634611672683/472883267672035\tNH:i:1\tHI:i:1\tAS:i:55\tnM:i:1\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\n",
      "SRR306838.12840151\t16\tchrM\t4955\t255\t69M\t*\t0\t0\tCATAGCAGGCAGTTGAGGTGGATTAAACCAAACCCAGCTACGCAAAATCTTAGCATACTCCTCAATTAC\t67363788668536976554975565546:5266576643655645669336554327137354222/1\tNH:i:1\tHI:i:1\tAS:i:67\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\n",
      "SRR306838.11563592\t0\tchrM\t6434\t255\t67M\t*\t0\t0\tCCAATACCAAACGCCCCTCTTCGTCTGATCCGTCCTAATCACAGCAGTCCTACTTCTCCTATCTCTC\t514325394725555764655664754525673664554744458682956265487765559673:\tNH:i:1\tHI:i:1\tAS:i:65\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\n",
      "SRR306838.6391414\t0\tchr5\t53885588\t255\t64M\t*\t0\t0\tTGAATATATAGATGATTCTGTCTTTGTTATCTTTGTCAAATCCATGCCATCTTAAACCCTAACC\t-422347255616624476547566736456748639455766898775797565464574487\tNH:i:1\tHI:i:1\tAS:i:62\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\n",
      "SRR306838.639141\t16\tchrM\t6586\t255\t4S63M\t*\t0\t0\tGGAGCCATTCTATACCAACACCTATTCTGATTTTTCGGTCACCCTGAAGTTTATATTCTTATCCTAC\t6:65766659345367746368553685873244677547269967565366343232236477202\tNH:i:1\tHI:i:1\tAS:i:62\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\n",
      "SRR306838.13119715\t0\tchrM\t5349\t255\t67M\t*\t0\t0\tCTACGCCTAATCTACTCCACCTCAATCACACTACTCCCCATATCTAACAACGTAAAAATAAAATGAC\t301.31/213143/21465774755474666546547457666:55372375155437546675557\tNH:i:1\tHI:i:1\tAS:i:65\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\n",
      "SRR306838.9230309\t16\tchr1\t629343\t3\t65M\t*\t0\t0\tATGAAAAAACTTCCTACCACTCACCCTAGCATTACTTATATGATATGTCTCCATACCCATTACAA\t657656655846975366575724685658:65463355578767645634882.276;44/65-\tNH:i:2\tHI:i:1\tAS:i:63\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\n",
      "SRR306838.9230309\t272\tchrM\t4173\t3\t65M\t*\t0\t0\tATGAAAAAACTTCCTACCACTCACCCTAGCATTACTTATATGATATGTCTCCATACCCATTACAA\t657656655846975366575724685658:65463355578767645634882.276;44/65-\tNH:i:2\tHI:i:2\tAS:i:63\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\n",
      "SRR306838.6978270\t16\tchrM\t4931\t255\t61M\t*\t0\t0\tCCTCACTCTCTCAATCTTATCCATCATAGCAGGCAGTTGAGGTGGATTAAACCAAACCCAG\t7457474635546648315677757652676765554685765563355404668078641\tNH:i:1\tHI:i:1\tAS:i:60\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\n",
      "SRR306838.15709562\t16\tchrM\t8389\t255\t68M\t*\t0\t0\tATGGCCCACCATAATTACCCCCATACTCCTTACACTATTCCTCATCACCCAACTAAAAACATTAAACA\t5785766436:677544655578536669444647575468268364766837257544682531.4.\tNH:i:1\tHI:i:1\tAS:i:64\tnM:i:1\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\n",
      "SRR306838.16880644\t16\tchrM\t4938\t255\t63M\t*\t0\t0\tCTCTCAATCTTATCCATCATAGCAGGCAGTTGAGGTGGATTAAACCAAACCCAGCTACGCAAA\t96958957823655674794647843746367466466635663577315563362520702,\tNH:i:1\tHI:i:1\tAS:i:62\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\n",
      "SRR306838.14545842\t16\tchrM\t3738\t255\t10S56M\t*\t0\t0\tAAAAAATCAACACCCTAGCCATCATTCTACTATCAACATTACTAATAAGTGGCTCCTTTAACCTCT\t6684753788615853644775::6275486466;576414626446464633277422605604)\tNH:i:1\tHI:i:1\tAS:i:55\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\n",
      "SRR306838.11004483\t16\tchrM\t12894\t255\t69M\t*\t0\t0\tAGCATGATTTATCCTACACTCCAACTCATGAGACCCACAACAAATAGCCCTTCTAAACGCTAATCCAAG\t9576485553576:44778376:6756:4776565686:588655677775675545673458578243\tNH:i:1\tHI:i:1\tAS:i:67\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\n",
      "SRR306838.2963056\t0\tchrM\t4975\t255\t67M\t*\t0\t0\tGATTAAACCAAACCCAGCTACGCAAAATCTTAGCATACTCCTCAATTACCCACATAGGATGAATAAT\t30114636884264667645696655246544:374337877;7576285738775:6976975666\tNH:i:1\tHI:i:1\tAS:i:65\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\n",
      "SRR306838.13753456\t16\tchr1\t629331\t3\t9S63M\t*\t0\t0\tAAGAACGGCCATACACCTCCTATGAAAAAACTTCCTACCACTCACCCTAGCATTACTTATATGATATGTCTC\t647735748:754665865847494877345357:41988648265937877444633843476376304/2\tNH:i:2\tHI:i:1\tAS:i:62\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\n",
      "SRR306838.13753456\t272\tchrM\t4161\t3\t9S63M\t*\t0\t0\tAAGAACGGCCATACACCTCCTATGAAAAAACTTCCTACCACTCACCCTAGCATTACTTATATGATATGTCTC\t647735748:754665865847494877345357:41988648265937877444633843476376304/2\tNH:i:2\tHI:i:2\tAS:i:62\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\n",
      "SRR306838.12460240\t16\tchrM\t5395\t255\t13S54M\t*\t0\t0\tCAGACACTCAGACACAACGTAAAAATAAAATGACAGTTTGAACATACAAAACCCACCCCATTCCTCC\t766674779:83766:454275837767564434852344356744753515261563750367/52\tNH:i:1\tHI:i:1\tAS:i:53\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\n",
      "SRR306838.15383614\t256\tchr1\t631921\t3\t72M\t*\t0\t0\tCTAGGGTTTATCGTGTGAGCACACCATATATTTACAGTAGGAATAGACGTAGACACACGAGCATATTTCACC\t631875778516655478875766887845976577856978577866645658595678979447359555\tNH:i:2\tHI:i:2\tAS:i:70\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\n",
      "SRR306838.15383614\t0\tchrM\t6750\t3\t72M\t*\t0\t0\tCTAGGGTTTATCGTGTGAGCACACCATATATTTACAGTAGGAATAGACGTAGACACACGAGCATATTTCACC\t631875778516655478875766887845976577856978577866645658595678979447359555\tNH:i:2\tHI:i:1\tAS:i:70\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\n",
      "SRR306838.2587809\t16\tchrM\t14800\t255\t47M\t*\t0\t0\tCATCGACCTCCCCACCCCATCCAACATCTCCGCATGATGAAACTTCG\t76366368365654765863577577563773464726658161145\tNH:i:1\tHI:i:1\tAS:i:46\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\n",
      "SRR306838.16843581\t16\tchrM\t5367\t255\t7S65M\t*\t0\t0\tATTCTTAACCTCAATCACACTACTCCCCATATCTAACAACGTAAAAATAAAATGACAGTTTGAACATACAAA\t85:964847>98:777586:356667658475=6657937533435347676653654333:546:41724.\tNH:i:1\tHI:i:1\tAS:i:63\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\n",
      "SRR306838.21182939\t256\tchr1\t632905\t3\t5S62M\t*\t0\t0\tTTTATTAACAAATACAAACATCTCAGACGCTCAGGAAATAGAAACCGTCTGAACTATCCTGCCCGCC\t,403341368324478644666774824544788643455886453636676486654468772675\tNH:i:2\tHI:i:2\tAS:i:57\tnM:i:2\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\n",
      "SRR306838.21182939\t0\tchrM\t7734\t3\t5S62M\t*\t0\t0\tTTTATTAACAAATACAAACATCTCAGACGCTCAGGAAATAGAAACCGTCTGAACTATCCTGCCCGCC\t,403341368324478644666774824544788643455886453636676486654468772675\tNH:i:2\tHI:i:1\tAS:i:57\tnM:i:2\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\n",
      "SRR306838.1759208\t16\tchrM\t15817\t255\t66M\t*\t0\t0\tATACTTCACAACAATCCTAATCCTAATACCAACTATCTCCCTAATTGAAAACAAAATACTCAAATG\t955<47656638<44865858982:753679378668386<18866:56627464334917412.3\tNH:i:1\tHI:i:1\tAS:i:64\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\n",
      "SRR306838.11331605\t16\tchrM\t8425\t255\t41M\t*\t0\t0\tATTCCTCATCACCCAACTAAAAACATTAAACACAAACTACC\t84567467674566747367653764255466668170,63\tNH:i:1\tHI:i:1\tAS:i:38\tnM:i:1\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\n",
      "SRR306838.23942767\t0\tchr1\t629659\t3\t70M\t*\t0\t0\tCCGTCATCTACTCTACCATCTTTGCAGGCACACTCATCACAGCGCTAAGCTCGCACTGATTTTTTACCTG\t4/3/267854836536563756467565756473:776654763734284667647987445454465:5\tNH:i:2\tHI:i:1\tAS:i:68\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\n",
      "SRR306838.23942767\t256\tchrM\t4489\t3\t70M\t*\t0\t0\tCCGTCATCTACTCTACCATCTTTGCAGGCACACTCATCACAGCGCTAAGCTCGCACTGATTTTTTACCTG\t4/3/267854836536563756467565756473:776654763734284667647987445454465:5\tNH:i:2\tHI:i:2\tAS:i:68\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\n",
      "SRR306838.15264190\t256\tchr1\t629315\t3\t70M\t*\t0\t0\tTCCGCTACGACCAACTCATACACCTCCTATGAAAAAACTTCCTACCACTCACCCTAGCATTACTTATATG\t,034012654852332564573654564475665432476757475465666876594755467573665\tNH:i:2\tHI:i:2\tAS:i:68\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\n",
      "SRR306838.15264190\t0\tchrM\t4145\t3\t70M\t*\t0\t0\tTCCGCTACGACCAACTCATACACCTCCTATGAAAAAACTTCCTACCACTCACCCTAGCATTACTTATATG\t,034012654852332564573654564475665432476757475465666876594755467573665\tNH:i:2\tHI:i:1\tAS:i:68\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\n",
      "SRR306838.4040046\t0\tchrM\t4887\t255\t17S45M1S\t*\t0\t0\tCACACATCCACCTCGGGTCCATCATATACCAAATCTCTCCCTCACTAAACGTAAGCCTTCTCA\t511214665784471441667566455465466485956665765444226554965568597\tNH:i:1\tHI:i:1\tAS:i:42\tnM:i:1\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\n",
      "SRR306838.9874975\t16\tchrM\t8479\t255\t6S64M\t*\t0\t0\tAATACTACCAAAGCCCATAAAAATAAAAAATTATAACAAACCCTGAGAACCAAAATGAACGAAAATCTGT\t8742834777672557;677565439566544644357514572436546565530742446351/71/(\tNH:i:1\tHI:i:1\tAS:i:62\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\n",
      "SRR306838.1128147\t16\tchr1\t631932\t3\t48M\t*\t0\t0\tCGTGTGAGCACACCATATATTTACAGTAGGAATAGACGTAGACACACG\t555668657555546566334447:63674566295565588626034\tNH:i:2\tHI:i:1\tAS:i:47\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\n",
      "SRR306838.1128147\t272\tchrM\t6761\t3\t48M\t*\t0\t0\tCGTGTGAGCACACCATATATTTACAGTAGGAATAGACGTAGACACACG\t555668657555546566334447:63674566295565588626034\tNH:i:2\tHI:i:2\tAS:i:47\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\n",
      "SRR306838.22091604\t0\tchrM\t6588\t255\t70M\t*\t0\t0\tATTCTATACCAACACCTATTCTGATTTTTCGGTCACCCTGAAGTTTATATTCTTATCCTACCAGGCTTCG\t)30335737552662643369:745554335554356597336373576667665484738768675377\tNH:i:1\tHI:i:1\tAS:i:68\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\n"
     ]
    }
   ],
   "source": [
    "df = coding_scores_unspliced.query(\"jaccard_in_peptide_db < 0.5 & in_cds_no_stop_codon == True\")\n",
    "for read_id in df.short_read_id:\n",
    "    ! samtools view $folder/$unspliced_bam | grep $read_id"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "A lot of these are chromosome M (mitochondria) so let's ignore all reads mapping to the mitochondria"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {
    "execution_event_id": "9d335cd6-f39c-4da2-a7b6-f18a13907be6",
    "last_executed_text": "unspliced_bam = 'SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.bam'\n\n# This reads aligns to the CBLN2 \nnoncoding_read_id = 'SRR306838.3607263'\n! samtools view $folder/$unspliced_bam | grep $noncoding_read_id",
    "persistent_id": "e75ae01f-e817-4e22-93d9-62740bd49be3"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SRR306838.3607263\t16\tchrM\t8401\t255\t12S60M\t*\t0\t0\tGTCATGCAAAAGAATTACCCCCATACTCCTTACACTATTCCTTATCACCCAACTAAAAACATTAAACACAAA\t557:977:86<;8744366867844:78935686846668:434655896;58479533962386971613.\tNH:i:1\tHI:i:1\tAS:i:55\tnM:i:2\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\n"
     ]
    }
   ],
   "source": [
    "unspliced_bam = 'SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.bam'\n",
    "\n",
    "# This reads aligns to the CBLN2 \n",
    "noncoding_read_id = 'SRR306838.3607263'\n",
    "! samtools view $folder/$unspliced_bam | grep $noncoding_read_id"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {
    "execution_event_id": "f3fe66ae-bf1a-435b-a340-580f3b4a5956",
    "last_executed_text": "unspliced_bam = 'SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.bam'\n\n# This reads aligns to the CBLN2 \nnoncoding_read_id = 'SRR306838.23727244'\n! samtools view $folder/$unspliced_bam | grep $noncoding_read_id",
    "persistent_id": "beff2498-4747-40c1-acfd-e30bc468e81a"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SRR306838.23727244\t16\tchrM\t14157\t255\t66M\t*\t0\t0\tCCCGAGCAATCTCAATTACAATATATACACCAACAAACAATGTTCAACCAGTAACTACTACTAATC\t6678867778855785236875525246355726854665351366456341337226254200.1\tNH:i:1\tHI:i:1\tAS:i:64\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\n"
     ]
    }
   ],
   "source": [
    "unspliced_bam = 'SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.bam'\n",
    "\n",
    "# This reads aligns to the CBLN2 \n",
    "noncoding_read_id = 'SRR306838.23727244'\n",
    "! samtools view $folder/$unspliced_bam | grep $noncoding_read_id"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "persistent_id": "755d4144-565d-4c60-868d-43bf6eebdf73"
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "execution_event_id": "60a404cc-6533-4522-888d-efd179f975be",
    "last_executed_text": "from khtools.partition_reads import score_single_sequence",
    "persistent_id": "a31a1fe9-c86c-4d99-b575-14f954026bc9"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\ttranslation: \tRTPLITPAIMTLGHNMIYLHT\n",
      "\tkmers: NMIYLHT PLITPAI TPLITPA LITPAIM AIMTLGH HNMIYLH GHNMIYL ITPAIMT RTPLITP IMTLGHN MTLGHNM TLGHNMI TPAIMTL PAIMTLG LGHNMIY\n",
      "\tK-mers in peptide database:\n",
      "{('AIMTLGH', 4934959604245948732): 0,\n",
      " ('GHNMIYL', 8679214931415330990): 0,\n",
      " ('HNMIYLH', 9860068405428524224): 0,\n",
      " ('IMTLGHN', 5007379133261696389): 0,\n",
      " ('ITPAIMT', 1845456223218030233): 0,\n",
      " ('LGHNMIY', 14440126941365645435): 0,\n",
      " ('LITPAIM', 12273716599056031256): 0,\n",
      " ('MTLGHNM', 8490348177331113914): 0,\n",
      " ('NMIYLHT', 5523607257588686990): 0,\n",
      " ('PAIMTLG', 2938204897806652236): 0,\n",
      " ('PLITPAI', 1294566589529317893): 0,\n",
      " ('RTPLITP', 14525035971622691714): 0,\n",
      " ('TLGHNMI', 15987995585519721535): 0,\n",
      " ('TPAIMTL', 2070009544365237801): 0,\n",
      " ('TPLITPA', 14027991606434210618): 0}\n",
      "\ttranslation: \tSVEINHIMAKGHDGRSNQRCS\n",
      "\tkmers: RSNQRCS EINHIMA GHDGRSN DGRSNQR GRSNQRC AKGHDGR INHIMAK HDGRSNQ SVEINHI VEINHIM KGHDGRS NHIMAKG IMAKGHD HIMAKGH MAKGHDG\n",
      "\tK-mers in peptide database:\n",
      "{('AKGHDGR', 11854149275921946754): 0,\n",
      " ('DGRSNQR', 12662774000725594405): 0,\n",
      " ('EINHIMA', 4141968266383781556): 0,\n",
      " ('GHDGRSN', 1538347941759508716): 0,\n",
      " ('GRSNQRC', 18345884528669914456): 0,\n",
      " ('HDGRSNQ', 11735731444601073774): 0,\n",
      " ('HIMAKGH', 11132219095374069408): 0,\n",
      " ('IMAKGHD', 16251833971127667720): 0,\n",
      " ('INHIMAK', 5902322659715217329): 0,\n",
      " ('KGHDGRS', 969856777363063149): 0,\n",
      " ('MAKGHDG', 2437442772461895986): 0,\n",
      " ('NHIMAKG', 9193714866121813898): 0,\n",
      " ('RSNQRCS', 11818278895836394990): 0,\n",
      " ('SVEINHI', 1758683071434437859): 0,\n",
      " ('VEINHIM', 11381693998765891514): 0}\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(0, 15)"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from khtools.partition_reads import score_single_sequence\n",
    "s = \"CAAGAACACCTCTGATTACTCCTGCCATCATGACCCTTGGCCATAATATGATTTATCTCCACACTAG\"\n",
    "seq = Seq(s)\n",
    "\n",
    "score_single_sequence(s, peptide_graph, peptide_ksize, 'protein', verbose=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "execution_event_id": "c8d223de-6e63-4a28-9f46-751d43300fa1",
    "last_executed_text": "hash_murmur(\"QSSSPEF\")",
    "persistent_id": "92a5366a-574a-449f-8014-2b931a91ee77"
   },
   "outputs": [],
   "source": [
    "hash_murmur(\"QSSSPEF\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "execution_event_id": "55fe9c6e-a3f8-4e8e-98eb-0fc5caa8f1cf",
    "last_executed_text": "peptide_graph.get(9478686570769398934)",
    "persistent_id": "bd39f49d-b475-49c9-81f0-ef32c88fa033"
   },
   "outputs": [],
   "source": [
    "peptide_graph.get(9478686570769398934)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {
    "execution_event_id": "6e534b29-89d6-45fd-aa4d-c58390afa0fc",
    "last_executed_text": "for record in tqdm(screed.open('Homo_sapiens.GRCh38.pep.all.fa.gz')):\n    if 'QSSSPEFRVQSFSERTNARKKNNH' in record['sequence']:\n        print(\"---- QSSSPEFRVQSFSERTNARKKNNH ----\")\n        print(record)\n    if \"TOPORS\" in record['name']:\n        print(\"---- TOPORS ----\")\n        print(record)",
    "persistent_id": "39826558-68ec-4086-b31f-de7f8d208545"
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/screed/openscreed.py:39: ResourceWarning: unclosed file <_io.BufferedReader name='Homo_sapiens.GRCh38.pep.all.fa.gz'>\n",
      "  self.iter_fn = self.open_reader(filename, *args, **kwargs)\n",
      "46151it [00:01, 36395.48it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "---- QSSSPEFRVQSFSERTNARKKNNH ----\n",
      "{'name': 'ENSP00000353735.2 pep chromosome:GRCh38:9:32540544:32552586:-1 gene:ENSG00000197579.8 transcript:ENST00000360538.7 gene_biotype:protein_coding transcript_biotype:protein_coding gene_symbol:TOPORS description:TOP1 binding arginine/serine rich protein [Source:HGNC Symbol;Acc:HGNC:21653]', 'sequence': 'MGSQPPLGSPLSREEGEAPPPAPASEGRRRSRRVRLRGSCRHRPSFLGCRELAASAPARPAPASSEIMASAAKEFKMDNFSPKAGTSKLQQTVPADASPDSKCPICLDRFDNVSYLDRCLHKFCFRCVQEWSKNKAECPLCKQPFDSIFHSVRAEDDFKEYVLRPSYNGSFVTPDRRFRYRTTLTRERNASVYSPSGPVNRRTTTPPDSGVLFEGLGISTRPRDVEIPQFMRQIAVRRPTTADERSLRKIQEQDIINFRRTLYRAGARVRNIEDGGRYRDISAEFFRRNPACLHRLVPWLKRELTVLFGAHGSLVNIVQHIIMSNVTRYDLESQAFVSDLRPFLLNRTEHFIHEFISFARSPFNMAAFDQHANYDCPAPSYEEGSHSDSSVITISPDEAETQELDINVATVSQAPWDDETPGPSYSSSEQVHVTMSSLLNTSDSSDEELVTGGATSQIQGVQTNDDLNNDSDDSSDNCVIVGFVKPLAERTPELVELSSDSEDLGSYEKMETVKTQEQEQSYSSGDSDVSRCSSPHSVLGKDEQINKGHCDSSTRIKSKKEEKRSTSLSSPRNLNSSVRGDRVYSPYNHRHRKRGRSRSSDSRSQSRSGHDQKNHRKHHGKKRMKSKRSRSRESSRPRGRRDKKRSRTRDSSWSRRSQTLSLSSESTSRSRSRSSDHGKRRSRSRNRDRYYLRNNYGSRYKWEYTYYSRNKDRDGYESSYRRRTLSRAHYSRQSSSPEFRVQSFSERTNARKKNNHSERKYYYYERHRSRSLSSNRSRTASTGTDRVRNEKPGGKRKYKTRHLEGTNEVAQPSREFASKAKDSHYQKSSSKLDGNYKNESDTFSDSRSSDRETKHKRRKRKTRSLSVEIVYEGKATDTTKHHKKKKKKHKKKHKKHHGDNASRSPVVITIDSDSDKDSEVKEDTECDNSGPQDPLQNEFLAPSLEPFETKDVVTIEAEFGVLDKECDIATLSNNLNNANKTVDNIPPLAASVEQTLDVREESTFVSDLENQPSNIVSLQTEPSRQLPSPRTSLMSVCLGRDCDMS', 'description': ''}\n",
      "---- TOPORS ----\n",
      "{'name': 'ENSP00000353735.2 pep chromosome:GRCh38:9:32540544:32552586:-1 gene:ENSG00000197579.8 transcript:ENST00000360538.7 gene_biotype:protein_coding transcript_biotype:protein_coding gene_symbol:TOPORS description:TOP1 binding arginine/serine rich protein [Source:HGNC Symbol;Acc:HGNC:21653]', 'sequence': 'MGSQPPLGSPLSREEGEAPPPAPASEGRRRSRRVRLRGSCRHRPSFLGCRELAASAPARPAPASSEIMASAAKEFKMDNFSPKAGTSKLQQTVPADASPDSKCPICLDRFDNVSYLDRCLHKFCFRCVQEWSKNKAECPLCKQPFDSIFHSVRAEDDFKEYVLRPSYNGSFVTPDRRFRYRTTLTRERNASVYSPSGPVNRRTTTPPDSGVLFEGLGISTRPRDVEIPQFMRQIAVRRPTTADERSLRKIQEQDIINFRRTLYRAGARVRNIEDGGRYRDISAEFFRRNPACLHRLVPWLKRELTVLFGAHGSLVNIVQHIIMSNVTRYDLESQAFVSDLRPFLLNRTEHFIHEFISFARSPFNMAAFDQHANYDCPAPSYEEGSHSDSSVITISPDEAETQELDINVATVSQAPWDDETPGPSYSSSEQVHVTMSSLLNTSDSSDEELVTGGATSQIQGVQTNDDLNNDSDDSSDNCVIVGFVKPLAERTPELVELSSDSEDLGSYEKMETVKTQEQEQSYSSGDSDVSRCSSPHSVLGKDEQINKGHCDSSTRIKSKKEEKRSTSLSSPRNLNSSVRGDRVYSPYNHRHRKRGRSRSSDSRSQSRSGHDQKNHRKHHGKKRMKSKRSRSRESSRPRGRRDKKRSRTRDSSWSRRSQTLSLSSESTSRSRSRSSDHGKRRSRSRNRDRYYLRNNYGSRYKWEYTYYSRNKDRDGYESSYRRRTLSRAHYSRQSSSPEFRVQSFSERTNARKKNNHSERKYYYYERHRSRSLSSNRSRTASTGTDRVRNEKPGGKRKYKTRHLEGTNEVAQPSREFASKAKDSHYQKSSSKLDGNYKNESDTFSDSRSSDRETKHKRRKRKTRSLSVEIVYEGKATDTTKHHKKKKKKHKKKHKKHHGDNASRSPVVITIDSDSDKDSEVKEDTECDNSGPQDPLQNEFLAPSLEPFETKDVVTIEAEFGVLDKECDIATLSNNLNNANKTVDNIPPLAASVEQTLDVREESTFVSDLENQPSNIVSLQTEPSRQLPSPRTSLMSVCLGRDCDMS', 'description': ''}\n",
      "---- QSSSPEFRVQSFSERTNARKKNNH ----\n",
      "{'name': 'ENSP00000369187.1 pep chromosome:GRCh38:9:32540826:32552553:-1 gene:ENSG00000197579.8 transcript:ENST00000379858.1 gene_biotype:protein_coding transcript_biotype:protein_coding gene_symbol:TOPORS description:TOP1 binding arginine/serine rich protein [Source:HGNC Symbol;Acc:HGNC:21653]', 'sequence': 'MIMASAAKEFKMDNFSPKAGTSKLQQTVPADASPDSKCPICLDRFDNVSYLDRCLHKFCFRCVQEWSKNKAECPLCKQPFDSIFHSVRAEDDFKEYVLRPSYNGSFVTPDRRFRYRTTLTRERNASVYSPSGPVNRRTTTPPDSGVLFEGLGISTRPRDVEIPQFMRQIAVRRPTTADERSLRKIQEQDIINFRRTLYRAGARVRNIEDGGRYRDISAEFFRRNPACLHRLVPWLKRELTVLFGAHGSLVNIVQHIIMSNVTRYDLESQAFVSDLRPFLLNRTEHFIHEFISFARSPFNMAAFDQHANYDCPAPSYEEGSHSDSSVITISPDEAETQELDINVATVSQAPWDDETPGPSYSSSEQVHVTMSSLLNTSDSSDEELVTGGATSQIQGVQTNDDLNNDSDDSSDNCVIVGFVKPLAERTPELVELSSDSEDLGSYEKMETVKTQEQEQSYSSGDSDVSRCSSPHSVLGKDEQINKGHCDSSTRIKSKKEEKRSTSLSSPRNLNSSVRGDRVYSPYNHRHRKRGRSRSSDSRSQSRSGHDQKNHRKHHGKKRMKSKRSRSRESSRPRGRRDKKRSRTRDSSWSRRSQTLSLSSESTSRSRSRSSDHGKRRSRSRNRDRYYLRNNYGSRYKWEYTYYSRNKDRDGYESSYRRRTLSRAHYSRQSSSPEFRVQSFSERTNARKKNNHSERKYYYYERHRSRSLSSNRSRTASTGTDRVRNEKPGGKRKYKTRHLEGTNEVAQPSREFASKAKDSHYQKSSSKLDGNYKNESDTFSDSRSSDRETKHKRRKRKTRSLSVEIVYEGKATDTTKHHKKKKKKHKKKHKKHHGDNASRSPVVITIDSDSDKDSEVKEDTECDNSGPQDPLQNEFLAPSLEPFETKDVVTIEAEFGVLDKECDIATLSNNLNNANKTVDNIPPLAASVEQTLDVREESTFVSDLENQPSNIVSLQTEPSRQLPSPRTSLMSVCLGRDCDMS', 'description': ''}\n",
      "---- TOPORS ----\n",
      "{'name': 'ENSP00000369187.1 pep chromosome:GRCh38:9:32540826:32552553:-1 gene:ENSG00000197579.8 transcript:ENST00000379858.1 gene_biotype:protein_coding transcript_biotype:protein_coding gene_symbol:TOPORS description:TOP1 binding arginine/serine rich protein [Source:HGNC Symbol;Acc:HGNC:21653]', 'sequence': 'MIMASAAKEFKMDNFSPKAGTSKLQQTVPADASPDSKCPICLDRFDNVSYLDRCLHKFCFRCVQEWSKNKAECPLCKQPFDSIFHSVRAEDDFKEYVLRPSYNGSFVTPDRRFRYRTTLTRERNASVYSPSGPVNRRTTTPPDSGVLFEGLGISTRPRDVEIPQFMRQIAVRRPTTADERSLRKIQEQDIINFRRTLYRAGARVRNIEDGGRYRDISAEFFRRNPACLHRLVPWLKRELTVLFGAHGSLVNIVQHIIMSNVTRYDLESQAFVSDLRPFLLNRTEHFIHEFISFARSPFNMAAFDQHANYDCPAPSYEEGSHSDSSVITISPDEAETQELDINVATVSQAPWDDETPGPSYSSSEQVHVTMSSLLNTSDSSDEELVTGGATSQIQGVQTNDDLNNDSDDSSDNCVIVGFVKPLAERTPELVELSSDSEDLGSYEKMETVKTQEQEQSYSSGDSDVSRCSSPHSVLGKDEQINKGHCDSSTRIKSKKEEKRSTSLSSPRNLNSSVRGDRVYSPYNHRHRKRGRSRSSDSRSQSRSGHDQKNHRKHHGKKRMKSKRSRSRESSRPRGRRDKKRSRTRDSSWSRRSQTLSLSSESTSRSRSRSSDHGKRRSRSRNRDRYYLRNNYGSRYKWEYTYYSRNKDRDGYESSYRRRTLSRAHYSRQSSSPEFRVQSFSERTNARKKNNHSERKYYYYERHRSRSLSSNRSRTASTGTDRVRNEKPGGKRKYKTRHLEGTNEVAQPSREFASKAKDSHYQKSSSKLDGNYKNESDTFSDSRSSDRETKHKRRKRKTRSLSVEIVYEGKATDTTKHHKKKKKKHKKKHKKHHGDNASRSPVVITIDSDSDKDSEVKEDTECDNSGPQDPLQNEFLAPSLEPFETKDVVTIEAEFGVLDKECDIATLSNNLNNANKTVDNIPPLAASVEQTLDVREESTFVSDLENQPSNIVSLQTEPSRQLPSPRTSLMSVCLGRDCDMS', 'description': ''}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "110048it [00:03, 36127.01it/s]\n"
     ]
    }
   ],
   "source": [
    "for record in tqdm(screed.open('Homo_sapiens.GRCh38.pep.all.fa.gz')):\n",
    "    if 'QSSSPEFRVQSFSERTNARKKNNH' in record['sequence']:\n",
    "        print(\"---- QSSSPEFRVQSFSERTNARKKNNH ----\")\n",
    "        print(record)\n",
    "    if \"TOPORS\" in record['name']:\n",
    "        print(\"---- TOPORS ----\")\n",
    "        print(record)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {
    "execution_event_id": "7e137be4-048f-4578-aa65-623f59becd4c",
    "last_executed_text": "is_low_complexity(s, 21)",
    "persistent_id": "239be055-6b08-4181-9716-7d8401f8166d"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(False, 51)"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "is_low_complexity(s, 21)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {
    "execution_event_id": "9aae5cdd-9d41-4da2-95ff-9f88d8a8a418",
    "last_executed_text": "six_frame_translation_no_stops(seq)",
    "persistent_id": "0949482e-76ad-4702-9d85-71f276cacf59"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[Seq('LECEITYFMCLLTKNTNDKGKSH', ExtendedIUPACProtein()),\n",
       " Seq('WLFPLSLVFLVRRHMKYVISHSR', ExtendedIUPACProtein())]"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "six_frame_translation_no_stops(seq)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "persistent_id": "59a89914-d079-492e-8c12-72d449f3b808"
   },
   "source": [
    "## False positive - high jaccard but in noncoding set"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {
    "persistent_id": "fe1f3c00-8d86-4a0f-813e-08936163f4f0"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(67, 6)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>read_id</th>\n",
       "      <th>jaccard_in_peptide_db</th>\n",
       "      <th>n_kmers</th>\n",
       "      <th>classification</th>\n",
       "      <th>short_read_id</th>\n",
       "      <th>in_cds_no_stop_codon</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>SRR306838.7944356 Ibis_Run100924_C3PO:6:38:5591:5659/1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>7</td>\n",
       "      <td>coding</td>\n",
       "      <td>SRR306838.7944356</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>SRR306838.6939297 Ibis_Run100924_C3PO:6:33:5394:5316/1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>15</td>\n",
       "      <td>coding</td>\n",
       "      <td>SRR306838.6939297</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>SRR306838.23317726 Ibis_Run100924_C3PO:6:115:10861:16547/1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>17</td>\n",
       "      <td>coding</td>\n",
       "      <td>SRR306838.23317726</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82</th>\n",
       "      <td>SRR306838.18085420 Ibis_Run100924_C3PO:6:91:3302:10193/1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>16</td>\n",
       "      <td>coding</td>\n",
       "      <td>SRR306838.18085420</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>106</th>\n",
       "      <td>SRR306838.16938253 Ibis_Run100924_C3PO:6:85:7026:15703/1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>17</td>\n",
       "      <td>coding</td>\n",
       "      <td>SRR306838.16938253</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                        read_id  \\\n",
       "20       SRR306838.7944356 Ibis_Run100924_C3PO:6:38:5591:5659/1   \n",
       "64       SRR306838.6939297 Ibis_Run100924_C3PO:6:33:5394:5316/1   \n",
       "66   SRR306838.23317726 Ibis_Run100924_C3PO:6:115:10861:16547/1   \n",
       "82     SRR306838.18085420 Ibis_Run100924_C3PO:6:91:3302:10193/1   \n",
       "106    SRR306838.16938253 Ibis_Run100924_C3PO:6:85:7026:15703/1   \n",
       "\n",
       "     jaccard_in_peptide_db  n_kmers classification       short_read_id  \\\n",
       "20                     1.0        7         coding   SRR306838.7944356   \n",
       "64                     1.0       15         coding   SRR306838.6939297   \n",
       "66                     1.0       17         coding  SRR306838.23317726   \n",
       "82                     1.0       16         coding  SRR306838.18085420   \n",
       "106                    1.0       17         coding  SRR306838.16938253   \n",
       "\n",
       "     in_cds_no_stop_codon  \n",
       "20                  False  \n",
       "64                  False  \n",
       "66                  False  \n",
       "82                  False  \n",
       "106                 False  "
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "false_positives = coding_scores_not_low_complexity_unspliced.query(\"jaccard_in_peptide_db > 0.9 & in_cds_no_stop_codon == False\")\n",
    "print(false_positives.shape)\n",
    "false_positives.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {
    "execution_event_id": "6eff2520-3253-42fb-98f5-fffa2b234027",
    "last_executed_text": "\nnoncoding_read_id = 'SRR306838.7944356\t'\n! samtools view $folder/$unspliced_bam | grep $noncoding_read_id",
    "persistent_id": "3e3b6f8c-ae7e-45c3-820d-f15dbfe313d3"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SRR306838.7944356\t16\tchr19\t6501568\t255\t42M\t*\t0\t0\tCCATGGTATGTGCCTGTGGGGTCGATGCCATGTTCGTCACTG\t569626336635575648568542566376495452670712\tNH:i:1\tHI:i:1\tAS:i:41\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\n"
     ]
    }
   ],
   "source": [
    "\n",
    "noncoding_read_id = 'SRR306838.7944356\t'\n",
    "! samtools view $folder/$unspliced_bam | grep $noncoding_read_id"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {
    "execution_event_id": "7fa0ea33-5018-4927-8472-45b95614cccf",
    "last_executed_text": "\nnoncoding_read_id = 'SRR306838.6939297\t'\n! samtools view $folder/$unspliced_bam | grep $noncoding_read_id",
    "persistent_id": "fc1bcb0f-a943-4ecd-bf4f-7889239e3c6a"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SRR306838.6939297\t16\tchr11\t63758272\t255\t65M\t*\t0\t0\tATGAAGGAAAATACTCAGTGTCAGCTTGAGCCTGCATTCCAAGCTTTTTTTTTAATTTGGTGTTT\t57:6669965;82667:64748:4:588668966785767898958667558635755376530.\tNH:i:1\tHI:i:1\tAS:i:63\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\n"
     ]
    }
   ],
   "source": [
    "\n",
    "noncoding_read_id = 'SRR306838.6939297\t'\n",
    "! samtools view $folder/$unspliced_bam | grep $noncoding_read_id"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "persistent_id": "35253b87-6e00-4c6b-91f9-577a9866e60f"
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "history": [
   {
    "cell": {
     "executionCount": 1,
     "executionEventId": "d0e20505-b67f-45c3-88c9-b614b7d8fa34",
     "hasError": false,
     "id": "5e35ad66-c7bb-43f2-86d3-6da92b22e4a6",
     "outputs": [],
     "persistentId": "ecc17310-b85a-49bd-8d60-8de454040c04",
     "text": "folder = '/home/olga/data_sm/kmer-hashing/classify_coding_vs_noncoding/'"
    },
    "executionTime": "2019-10-11T15:46:28.945Z"
   },
   {
    "cell": {
     "executionCount": 2,
     "executionEventId": "dee535bd-f50f-4279-a494-632e131d6b8c",
     "hasError": true,
     "id": "e50808b6-6666-4543-9299-8075c789791c",
     "outputs": [
      {
       "ename": "SyntaxError",
       "evalue": "invalid syntax (partition_reads.py, line 109)",
       "output_type": "error",
       "traceback": [
        "Traceback \u001b[0;36m(most recent call last)\u001b[0m:\n",
        "  File \u001b[1;32m\"/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/IPython/core/interactiveshell.py\"\u001b[0m, line \u001b[1;32m2961\u001b[0m, in \u001b[1;35mrun_code\u001b[0m\n    exec(code_obj, self.user_global_ns, self.user_ns)\n",
        "\u001b[0;36m  File \u001b[0;32m\"<ipython-input-2-1e22708b2aee>\"\u001b[0;36m, line \u001b[0;32m1\u001b[0;36m, in \u001b[0;35m<module>\u001b[0;36m\u001b[0m\n\u001b[0;31m    from khtools.partition_reads import score_reads\u001b[0m\n",
        "\u001b[0;36m  File \u001b[0;32m\"/home/olga/code/kmer-hashing/kh-tools/khtools/partition_reads.py\"\u001b[0;36m, line \u001b[0;32m109\u001b[0m\n\u001b[0;31m    \"\\t{translation}\")\u001b[0m\n\u001b[0m                    ^\u001b[0m\n\u001b[0;31mSyntaxError\u001b[0m\u001b[0;31m:\u001b[0m invalid syntax\n"
       ]
      }
     ],
     "persistentId": "1e54230e-ceec-42fc-afa6-7b32e219378b",
     "text": "from khtools.partition_reads import score_reads"
    },
    "executionTime": "2019-10-11T15:47:13.217Z"
   },
   {
    "cell": {
     "executionCount": 3,
     "executionEventId": "1362cf3f-c25c-40bf-bafd-cdd8831e177c",
     "hasError": true,
     "id": "81fe1d1b-9150-4e3a-a0e2-f2470c0bb85c",
     "outputs": [
      {
       "ename": "ModuleNotFoundError",
       "evalue": "No module named 'khtools.compare_peptide'",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mModuleNotFoundError\u001b[0m                       Traceback (most recent call last)",
        "\u001b[0;32m<ipython-input-3-0c7717232dd3>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m     14\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     15\u001b[0m \u001b[0;32mfrom\u001b[0m \u001b[0msklearn\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0mmetrics\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 16\u001b[0;31m \u001b[0;32mfrom\u001b[0m \u001b[0mkhtools\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcompare_peptide\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0mkmerize\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mhpize\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdayhoffize\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     17\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     18\u001b[0m \u001b[0;32mfrom\u001b[0m \u001b[0mBio\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mSeq\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0mSeq\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;31mModuleNotFoundError\u001b[0m: No module named 'khtools.compare_peptide'"
       ]
      }
     ],
     "persistentId": "56e48bd4-fcc6-4d93-b4e3-0984be1fc1d5",
     "text": "import gzip\nfrom io import StringIO\nfrom pprint import pprint\n\nfrom khmer import Nodegraph\nimport matplotlib.pyplot as plt\nimport screed\nfrom sourmash._minhash import hash_murmur\nfrom khmer.khmer_args import calculate_graphsize\nimport pandas as pd\nfrom sourmash.logging import notify\n\nimport seaborn as sns\n\nfrom sklearn import metrics\nfrom khtools.compare_peptide import kmerize, hpize, dayhoffize\n\nfrom Bio.Seq import Seq\nfrom Bio import SeqIO"
    },
    "executionTime": "2019-10-11T15:47:20.354Z"
   },
   {
    "cell": {
     "executionCount": 4,
     "executionEventId": "df541bfc-35d5-4fea-bc6a-6028b6af56f9",
     "hasError": false,
     "id": "81fe1d1b-9150-4e3a-a0e2-f2470c0bb85c",
     "outputs": [],
     "persistentId": "56e48bd4-fcc6-4d93-b4e3-0984be1fc1d5",
     "text": "import gzip\nfrom io import StringIO\nfrom pprint import pprint\n\nfrom khmer import Nodegraph\nimport matplotlib.pyplot as plt\nimport screed\nfrom sourmash._minhash import hash_murmur\nfrom khmer.khmer_args import calculate_graphsize\nimport pandas as pd\nfrom sourmash.logging import notify\n\nimport seaborn as sns\n\nfrom sklearn import metrics\nfrom khtools.compare_kmer_content import kmerize, hpize, dayhoffize\n\nfrom Bio.Seq import Seq\nfrom Bio import SeqIO"
    },
    "executionTime": "2019-10-11T15:47:28.229Z"
   },
   {
    "cell": {
     "executionCount": 5,
     "executionEventId": "e2ee2ba9-e331-4ea4-a280-284de6cecb5f",
     "hasError": true,
     "id": "e50808b6-6666-4543-9299-8075c789791c",
     "outputs": [
      {
       "ename": "SyntaxError",
       "evalue": "invalid syntax (partition_reads.py, line 118)",
       "output_type": "error",
       "traceback": [
        "Traceback \u001b[0;36m(most recent call last)\u001b[0m:\n",
        "  File \u001b[1;32m\"/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/IPython/core/interactiveshell.py\"\u001b[0m, line \u001b[1;32m2961\u001b[0m, in \u001b[1;35mrun_code\u001b[0m\n    exec(code_obj, self.user_global_ns, self.user_ns)\n",
        "\u001b[0;36m  File \u001b[0;32m\"<ipython-input-5-1e22708b2aee>\"\u001b[0;36m, line \u001b[0;32m1\u001b[0;36m, in \u001b[0;35m<module>\u001b[0;36m\u001b[0m\n\u001b[0;31m    from khtools.partition_reads import score_reads\u001b[0m\n",
        "\u001b[0;36m  File \u001b[0;32m\"/home/olga/code/kmer-hashing/kh-tools/khtools/partition_reads.py\"\u001b[0;36m, line \u001b[0;32m118\u001b[0m\n\u001b[0;31m    'Low complexity sequence!!! n_kmers < ' \\\u001b[0m\n\u001b[0m                                          ^\u001b[0m\n\u001b[0;31mSyntaxError\u001b[0m\u001b[0;31m:\u001b[0m invalid syntax\n"
       ]
      }
     ],
     "persistentId": "1e54230e-ceec-42fc-afa6-7b32e219378b",
     "text": "from khtools.partition_reads import score_reads"
    },
    "executionTime": "2019-10-11T15:48:18.837Z"
   },
   {
    "cell": {
     "executionCount": 6,
     "executionEventId": "46f63861-bd0b-4f89-b980-3f687cbae9ac",
     "hasError": false,
     "id": "34402055-2d16-4b1d-bda2-6c89265dd6e3",
     "outputs": [],
     "persistentId": "c4087c95-84ba-48cb-b755-345fa94b8ef8",
     "text": "reads = 'SRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz'"
    },
    "executionTime": "2019-10-11T15:49:08.044Z"
   },
   {
    "cell": {
     "executionCount": 7,
     "executionEventId": "a7490dcf-a2c8-46bf-91ab-2fe4d6a1e7e7",
     "hasError": true,
     "id": "c43c763c-d708-4f43-a304-9fae85683297",
     "outputs": [
      {
       "ename": "NameError",
       "evalue": "name 'make_peptide_bloom_filter' is not defined",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
        "\u001b[0;32m<ipython-input-7-ac07a43642cd>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mpeptide_graph\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmake_peptide_bloom_filter\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"Homo_sapiens.GRCh38.pep.all.fa.gz\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;36m7\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
        "\u001b[0;31mNameError\u001b[0m: name 'make_peptide_bloom_filter' is not defined"
       ]
      }
     ],
     "persistentId": "be06ded1-d7d4-4f21-9961-3aed5ff30cbf",
     "text": "peptide_graph = make_peptide_bloom_filter(\"Homo_sapiens.GRCh38.pep.all.fa.gz\", 7)"
    },
    "executionTime": "2019-10-11T15:49:21.265Z"
   },
   {
    "cell": {
     "executionCount": 8,
     "executionEventId": "a52104b8-ca0f-4a18-b5d8-31fea6206571",
     "hasError": true,
     "id": "e50808b6-6666-4543-9299-8075c789791c",
     "outputs": [
      {
       "ename": "IndentationError",
       "evalue": "expected an indented block (partition_reads.py, line 184)",
       "output_type": "error",
       "traceback": [
        "Traceback \u001b[0;36m(most recent call last)\u001b[0m:\n",
        "  File \u001b[1;32m\"/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/IPython/core/interactiveshell.py\"\u001b[0m, line \u001b[1;32m2961\u001b[0m, in \u001b[1;35mrun_code\u001b[0m\n    exec(code_obj, self.user_global_ns, self.user_ns)\n",
        "\u001b[0;36m  File \u001b[0;32m\"<ipython-input-8-8a3a5562eacd>\"\u001b[0;36m, line \u001b[0;32m1\u001b[0;36m, in \u001b[0;35m<module>\u001b[0;36m\u001b[0m\n\u001b[0;31m    from khtools.partition_reads import score_reads, make_peptide_bloom_filter\u001b[0m\n",
        "\u001b[0;36m  File \u001b[0;32m\"/home/olga/code/kmer-hashing/kh-tools/khtools/partition_reads.py\"\u001b[0;36m, line \u001b[0;32m184\u001b[0m\n\u001b[0;31m    scoring_lines.append(\u001b[0m\n\u001b[0m                ^\u001b[0m\n\u001b[0;31mIndentationError\u001b[0m\u001b[0;31m:\u001b[0m expected an indented block\n"
       ]
      }
     ],
     "persistentId": "1e54230e-ceec-42fc-afa6-7b32e219378b",
     "text": "from khtools.partition_reads import score_reads, make_peptide_bloom_filter"
    },
    "executionTime": "2019-10-11T15:49:25.216Z"
   },
   {
    "cell": {
     "executionCount": 9,
     "executionEventId": "6c1c6805-6ff3-49c1-aa74-b98e3316315e",
     "hasError": true,
     "id": "e50808b6-6666-4543-9299-8075c789791c",
     "outputs": [
      {
       "ename": "IndentationError",
       "evalue": "expected an indented block (partition_reads.py, line 184)",
       "output_type": "error",
       "traceback": [
        "Traceback \u001b[0;36m(most recent call last)\u001b[0m:\n",
        "  File \u001b[1;32m\"/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/IPython/core/interactiveshell.py\"\u001b[0m, line \u001b[1;32m2961\u001b[0m, in \u001b[1;35mrun_code\u001b[0m\n    exec(code_obj, self.user_global_ns, self.user_ns)\n",
        "\u001b[0;36m  File \u001b[0;32m\"<ipython-input-9-8a3a5562eacd>\"\u001b[0;36m, line \u001b[0;32m1\u001b[0;36m, in \u001b[0;35m<module>\u001b[0;36m\u001b[0m\n\u001b[0;31m    from khtools.partition_reads import score_reads, make_peptide_bloom_filter\u001b[0m\n",
        "\u001b[0;36m  File \u001b[0;32m\"/home/olga/code/kmer-hashing/kh-tools/khtools/partition_reads.py\"\u001b[0;36m, line \u001b[0;32m184\u001b[0m\n\u001b[0;31m    scoring_lines.append(\u001b[0m\n\u001b[0m                ^\u001b[0m\n\u001b[0;31mIndentationError\u001b[0m\u001b[0;31m:\u001b[0m expected an indented block\n"
       ]
      }
     ],
     "persistentId": "1e54230e-ceec-42fc-afa6-7b32e219378b",
     "text": "from khtools.partition_reads import score_reads, make_peptide_bloom_filter"
    },
    "executionTime": "2019-10-11T15:53:15.389Z"
   },
   {
    "cell": {
     "executionCount": 10,
     "executionEventId": "793a50d7-bfc9-4bf7-8240-32a2eb7a6645",
     "hasError": false,
     "id": "0f1ea26a-ecfa-4359-8ba4-c0f0d6feb077",
     "outputs": [],
     "persistentId": "71299be4-a00f-4638-92a2-ea174ddf208d",
     "text": "%load_ext autoreload"
    },
    "executionTime": "2019-10-11T15:53:30.319Z"
   },
   {
    "cell": {
     "executionCount": 1,
     "executionEventId": "a7915bc9-5dca-443b-8772-be5522588278",
     "hasError": false,
     "id": "0f1ea26a-ecfa-4359-8ba4-c0f0d6feb077",
     "outputs": [],
     "persistentId": "71299be4-a00f-4638-92a2-ea174ddf208d",
     "text": "%load_ext autoreload\n%autoreload 2"
    },
    "executionTime": "2019-10-11T15:53:36.775Z"
   },
   {
    "cell": {
     "executionCount": 2,
     "executionEventId": "486d01dd-d258-44df-b3d3-1bf16c2d4338",
     "hasError": false,
     "id": "5e35ad66-c7bb-43f2-86d3-6da92b22e4a6",
     "outputs": [],
     "persistentId": "ecc17310-b85a-49bd-8d60-8de454040c04",
     "text": "folder = '/home/olga/data_sm/kmer-hashing/classify_coding_vs_noncoding/'"
    },
    "executionTime": "2019-10-11T15:53:36.785Z"
   },
   {
    "cell": {
     "executionCount": 3,
     "executionEventId": "ab5525e2-1eab-47e9-b7be-232b5d06e44b",
     "hasError": false,
     "id": "81fe1d1b-9150-4e3a-a0e2-f2470c0bb85c",
     "outputs": [],
     "persistentId": "56e48bd4-fcc6-4d93-b4e3-0984be1fc1d5",
     "text": "import gzip\nfrom io import StringIO\nfrom pprint import pprint\n\nfrom khmer import Nodegraph\nimport matplotlib.pyplot as plt\nimport screed\nfrom sourmash._minhash import hash_murmur\nfrom khmer.khmer_args import calculate_graphsize\nimport pandas as pd\nfrom sourmash.logging import notify\n\nimport seaborn as sns\n\nfrom sklearn import metrics\nfrom khtools.compare_kmer_content import kmerize, hpize, dayhoffize\n\nfrom Bio.Seq import Seq\nfrom Bio import SeqIO"
    },
    "executionTime": "2019-10-11T15:53:38.458Z"
   },
   {
    "cell": {
     "executionCount": 4,
     "executionEventId": "c5bb89af-3117-4e3d-8204-bcfce8570626",
     "hasError": true,
     "id": "e50808b6-6666-4543-9299-8075c789791c",
     "outputs": [
      {
       "ename": "IndentationError",
       "evalue": "expected an indented block (partition_reads.py, line 184)",
       "output_type": "error",
       "traceback": [
        "Traceback \u001b[0;36m(most recent call last)\u001b[0m:\n",
        "  File \u001b[1;32m\"/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/IPython/core/interactiveshell.py\"\u001b[0m, line \u001b[1;32m2961\u001b[0m, in \u001b[1;35mrun_code\u001b[0m\n    exec(code_obj, self.user_global_ns, self.user_ns)\n",
        "\u001b[0;36m  File \u001b[0;32m\"<ipython-input-4-8a3a5562eacd>\"\u001b[0;36m, line \u001b[0;32m1\u001b[0;36m, in \u001b[0;35m<module>\u001b[0;36m\u001b[0m\n\u001b[0;31m    from khtools.partition_reads import score_reads, make_peptide_bloom_filter\u001b[0m\n",
        "\u001b[0;36m  File \u001b[0;32m\"/home/olga/code/kmer-hashing/kh-tools/khtools/partition_reads.py\"\u001b[0;36m, line \u001b[0;32m184\u001b[0m\n\u001b[0;31m    scoring_lines.append(\u001b[0m\n\u001b[0m                ^\u001b[0m\n\u001b[0;31mIndentationError\u001b[0m\u001b[0;31m:\u001b[0m expected an indented block\n"
       ]
      }
     ],
     "persistentId": "1e54230e-ceec-42fc-afa6-7b32e219378b",
     "text": "from khtools.partition_reads import score_reads, make_peptide_bloom_filter"
    },
    "executionTime": "2019-10-11T15:53:38.548Z"
   },
   {
    "cell": {
     "executionCount": 5,
     "executionEventId": "453f0fc4-7c9f-48ab-9307-a000a7a4c479",
     "hasError": true,
     "id": "e50808b6-6666-4543-9299-8075c789791c",
     "outputs": [
      {
       "ename": "SyntaxError",
       "evalue": "'continue' not properly in loop (partition_reads.py, line 106)",
       "output_type": "error",
       "traceback": [
        "Traceback \u001b[0;36m(most recent call last)\u001b[0m:\n",
        "  File \u001b[1;32m\"/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/IPython/core/interactiveshell.py\"\u001b[0m, line \u001b[1;32m2961\u001b[0m, in \u001b[1;35mrun_code\u001b[0m\n    exec(code_obj, self.user_global_ns, self.user_ns)\n",
        "\u001b[0;36m  File \u001b[0;32m\"<ipython-input-5-8a3a5562eacd>\"\u001b[0;36m, line \u001b[0;32m1\u001b[0;36m, in \u001b[0;35m<module>\u001b[0;36m\u001b[0m\n\u001b[0;31m    from khtools.partition_reads import score_reads, make_peptide_bloom_filter\u001b[0m\n",
        "\u001b[0;36m  File \u001b[0;32m\"/home/olga/code/kmer-hashing/kh-tools/khtools/partition_reads.py\"\u001b[0;36m, line \u001b[0;32m106\u001b[0m\n\u001b[0;31m    continue\u001b[0m\n\u001b[0m    ^\u001b[0m\n\u001b[0;31mSyntaxError\u001b[0m\u001b[0;31m:\u001b[0m 'continue' not properly in loop\n"
       ]
      }
     ],
     "persistentId": "1e54230e-ceec-42fc-afa6-7b32e219378b",
     "text": "from khtools.partition_reads import score_reads, make_peptide_bloom_filter"
    },
    "executionTime": "2019-10-11T15:54:18.509Z"
   },
   {
    "cell": {
     "executionCount": 6,
     "executionEventId": "df735561-51a2-4047-b891-0777d0c68074",
     "hasError": true,
     "id": "e50808b6-6666-4543-9299-8075c789791c",
     "outputs": [
      {
       "ename": "SyntaxError",
       "evalue": "'continue' not properly in loop (partition_reads.py, line 106)",
       "output_type": "error",
       "traceback": [
        "Traceback \u001b[0;36m(most recent call last)\u001b[0m:\n",
        "  File \u001b[1;32m\"/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/IPython/core/interactiveshell.py\"\u001b[0m, line \u001b[1;32m2961\u001b[0m, in \u001b[1;35mrun_code\u001b[0m\n    exec(code_obj, self.user_global_ns, self.user_ns)\n",
        "\u001b[0;36m  File \u001b[0;32m\"<ipython-input-6-8a3a5562eacd>\"\u001b[0;36m, line \u001b[0;32m1\u001b[0;36m, in \u001b[0;35m<module>\u001b[0;36m\u001b[0m\n\u001b[0;31m    from khtools.partition_reads import score_reads, make_peptide_bloom_filter\u001b[0m\n",
        "\u001b[0;36m  File \u001b[0;32m\"/home/olga/code/kmer-hashing/kh-tools/khtools/partition_reads.py\"\u001b[0;36m, line \u001b[0;32m106\u001b[0m\n\u001b[0;31m    continue\u001b[0m\n\u001b[0m    ^\u001b[0m\n\u001b[0;31mSyntaxError\u001b[0m\u001b[0;31m:\u001b[0m 'continue' not properly in loop\n"
       ]
      }
     ],
     "persistentId": "1e54230e-ceec-42fc-afa6-7b32e219378b",
     "text": "from khtools.partition_reads import score_reads, make_peptide_bloom_filter"
    },
    "executionTime": "2019-10-11T15:55:30.216Z"
   },
   {
    "cell": {
     "executionCount": 7,
     "executionEventId": "4dbc9d04-de33-42bc-91c5-6d9cdb35653a",
     "hasError": true,
     "id": "e50808b6-6666-4543-9299-8075c789791c",
     "outputs": [
      {
       "ename": "SyntaxError",
       "evalue": "'continue' not properly in loop (partition_reads.py, line 121)",
       "output_type": "error",
       "traceback": [
        "Traceback \u001b[0;36m(most recent call last)\u001b[0m:\n",
        "  File \u001b[1;32m\"/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/IPython/core/interactiveshell.py\"\u001b[0m, line \u001b[1;32m2961\u001b[0m, in \u001b[1;35mrun_code\u001b[0m\n    exec(code_obj, self.user_global_ns, self.user_ns)\n",
        "\u001b[0;36m  File \u001b[0;32m\"<ipython-input-7-8a3a5562eacd>\"\u001b[0;36m, line \u001b[0;32m1\u001b[0;36m, in \u001b[0;35m<module>\u001b[0;36m\u001b[0m\n\u001b[0;31m    from khtools.partition_reads import score_reads, make_peptide_bloom_filter\u001b[0m\n",
        "\u001b[0;36m  File \u001b[0;32m\"/home/olga/code/kmer-hashing/kh-tools/khtools/partition_reads.py\"\u001b[0;36m, line \u001b[0;32m121\u001b[0m\n\u001b[0;31m    continue\u001b[0m\n\u001b[0m    ^\u001b[0m\n\u001b[0;31mSyntaxError\u001b[0m\u001b[0;31m:\u001b[0m 'continue' not properly in loop\n"
       ]
      }
     ],
     "persistentId": "1e54230e-ceec-42fc-afa6-7b32e219378b",
     "text": "from khtools.partition_reads import score_reads, make_peptide_bloom_filter"
    },
    "executionTime": "2019-10-11T15:55:33.641Z"
   },
   {
    "cell": {
     "executionCount": 8,
     "executionEventId": "33d99000-9cfa-4d02-a9c6-659e13954844",
     "hasError": true,
     "id": "e50808b6-6666-4543-9299-8075c789791c",
     "outputs": [
      {
       "ename": "SyntaxError",
       "evalue": "'continue' not properly in loop (partition_reads.py, line 121)",
       "output_type": "error",
       "traceback": [
        "Traceback \u001b[0;36m(most recent call last)\u001b[0m:\n",
        "  File \u001b[1;32m\"/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/IPython/core/interactiveshell.py\"\u001b[0m, line \u001b[1;32m2961\u001b[0m, in \u001b[1;35mrun_code\u001b[0m\n    exec(code_obj, self.user_global_ns, self.user_ns)\n",
        "\u001b[0;36m  File \u001b[0;32m\"<ipython-input-8-8a3a5562eacd>\"\u001b[0;36m, line \u001b[0;32m1\u001b[0;36m, in \u001b[0;35m<module>\u001b[0;36m\u001b[0m\n\u001b[0;31m    from khtools.partition_reads import score_reads, make_peptide_bloom_filter\u001b[0m\n",
        "\u001b[0;36m  File \u001b[0;32m\"/home/olga/code/kmer-hashing/kh-tools/khtools/partition_reads.py\"\u001b[0;36m, line \u001b[0;32m121\u001b[0m\n\u001b[0;31m    continue\u001b[0m\n\u001b[0m    ^\u001b[0m\n\u001b[0;31mSyntaxError\u001b[0m\u001b[0;31m:\u001b[0m 'continue' not properly in loop\n"
       ]
      }
     ],
     "persistentId": "1e54230e-ceec-42fc-afa6-7b32e219378b",
     "text": "from khtools.partition_reads import score_reads, make_peptide_bloom_filter"
    },
    "executionTime": "2019-10-11T15:55:38.782Z"
   },
   {
    "cell": {
     "executionCount": 9,
     "executionEventId": "8eab9c10-ca4c-40ae-bb32-5f9d9f8eccff",
     "hasError": true,
     "id": "c43c763c-d708-4f43-a304-9fae85683297",
     "outputs": [
      {
       "ename": "NameError",
       "evalue": "name 'make_peptide_bloom_filter' is not defined",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
        "\u001b[0;32m<ipython-input-9-ac07a43642cd>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mpeptide_graph\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmake_peptide_bloom_filter\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"Homo_sapiens.GRCh38.pep.all.fa.gz\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;36m7\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
        "\u001b[0;31mNameError\u001b[0m: name 'make_peptide_bloom_filter' is not defined"
       ]
      }
     ],
     "persistentId": "be06ded1-d7d4-4f21-9961-3aed5ff30cbf",
     "text": "peptide_graph = make_peptide_bloom_filter(\"Homo_sapiens.GRCh38.pep.all.fa.gz\", 7)"
    },
    "executionTime": "2019-10-11T15:55:59.672Z"
   },
   {
    "cell": {
     "executionCount": 10,
     "executionEventId": "d1278380-f27f-4332-8cac-0aa20aa4fdb0",
     "hasError": false,
     "id": "e50808b6-6666-4543-9299-8075c789791c",
     "outputs": [],
     "persistentId": "1e54230e-ceec-42fc-afa6-7b32e219378b",
     "text": "from khtools.partition_reads import score_reads, make_peptide_bloom_filter"
    },
    "executionTime": "2019-10-11T15:56:01.660Z"
   },
   {
    "cell": {
     "executionCount": 11,
     "executionEventId": "ef8afcb6-f87e-4a9f-80e6-34042d8cb64e",
     "hasError": true,
     "id": "c43c763c-d708-4f43-a304-9fae85683297",
     "outputs": [
      {
       "ename": "TypeError",
       "evalue": "__cinit__() got an unexpected keyword argument 'seed'",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
        "\u001b[0;32m<ipython-input-11-ac07a43642cd>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mpeptide_graph\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmake_peptide_bloom_filter\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"Homo_sapiens.GRCh38.pep.all.fa.gz\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;36m7\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
        "\u001b[0;32m~/code/kmer-hashing/kh-tools/khtools/partition_reads.py\u001b[0m in \u001b[0;36mmake_peptide_bloom_filter\u001b[0;34m(peptide_fasta, peptide_ksize, n_tables, molecule, tablesize, seed)\u001b[0m\n\u001b[1;32m     58\u001b[0m     \u001b[0;34m\"\"\"Create a bloom filter out of peptide seuqences\"\"\"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     59\u001b[0m     peptide_graph = Nodegraph(peptide_ksize, tablesize, n_tables=n_tables,\n\u001b[0;32m---> 60\u001b[0;31m                               seed=seed)\n\u001b[0m\u001b[1;32m     61\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     62\u001b[0m     \u001b[0;32mfor\u001b[0m \u001b[0mrecord\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mscreed\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mopen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mpeptide_fasta\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32mkhmer/_oxli/graphs.pyx\u001b[0m in \u001b[0;36mkhmer._oxli.graphs.Nodegraph.__cinit__\u001b[0;34m()\u001b[0m\n",
        "\u001b[0;31mTypeError\u001b[0m: __cinit__() got an unexpected keyword argument 'seed'"
       ]
      }
     ],
     "persistentId": "be06ded1-d7d4-4f21-9961-3aed5ff30cbf",
     "text": "peptide_graph = make_peptide_bloom_filter(\"Homo_sapiens.GRCh38.pep.all.fa.gz\", 7)"
    },
    "executionTime": "2019-10-11T15:56:01.974Z"
   },
   {
    "cell": {
     "executionCount": 12,
     "executionEventId": "01a94e0a-5a09-4cc5-bc1d-f24b4e927c0d",
     "hasError": false,
     "id": "e50808b6-6666-4543-9299-8075c789791c",
     "outputs": [],
     "persistentId": "1e54230e-ceec-42fc-afa6-7b32e219378b",
     "text": "from khtools.partition_reads import score_reads, make_peptide_bloom_filter"
    },
    "executionTime": "2019-10-11T15:56:38.814Z"
   },
   {
    "cell": {
     "executionCount": 13,
     "executionEventId": "49377a1a-66fc-4778-96e5-dd8144ce96b5",
     "hasError": true,
     "id": "c43c763c-d708-4f43-a304-9fae85683297",
     "outputs": [
      {
       "ename": "FileNotFoundError",
       "evalue": "[Errno 2] No such file or directory: 'Homo_sapiens.GRCh38.pep.all.fa.gz'",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
        "\u001b[0;32m<ipython-input-13-ac07a43642cd>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mpeptide_graph\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmake_peptide_bloom_filter\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"Homo_sapiens.GRCh38.pep.all.fa.gz\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;36m7\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
        "\u001b[0;32m~/code/kmer-hashing/kh-tools/khtools/partition_reads.py\u001b[0m in \u001b[0;36mmake_peptide_bloom_filter\u001b[0;34m(peptide_fasta, peptide_ksize, n_tables, molecule, tablesize)\u001b[0m\n\u001b[1;32m     58\u001b[0m     \u001b[0mpeptide_graph\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mNodegraph\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mpeptide_ksize\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtablesize\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mn_tables\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mn_tables\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     59\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 60\u001b[0;31m     \u001b[0;32mfor\u001b[0m \u001b[0mrecord\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mscreed\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mopen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mpeptide_fasta\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     61\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0;34m'*'\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mrecord\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'sequence'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     62\u001b[0m             \u001b[0;32mcontinue\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/screed/openscreed.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, filename, *args, **kwargs)\u001b[0m\n\u001b[1;32m     37\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m__init__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfilename\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     38\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msequencefile\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 39\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0miter_fn\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mopen_reader\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfilename\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     40\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0miter_fn\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     41\u001b[0m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__name__\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0miter_fn\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__name__\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/screed/openscreed.py\u001b[0m in \u001b[0;36mopen_reader\u001b[0;34m(self, filename, *args, **kwargs)\u001b[0m\n\u001b[1;32m     54\u001b[0m         }  # Inspired by http://stackoverflow.com/a/13044946/1585509\n\u001b[1;32m     55\u001b[0m         \u001b[0mfilename\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0m_normalize_filename\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfilename\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 56\u001b[0;31m         \u001b[0mbufferedfile\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mio\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mopen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfile\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mfilename\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmode\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'rb'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mbuffering\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m8192\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     57\u001b[0m         \u001b[0mnum_bytes_to_peek\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmax\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mx\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mx\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mmagic_dict\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     58\u001b[0m         \u001b[0mfile_start\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mbufferedfile\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpeek\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mnum_bytes_to_peek\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;31mFileNotFoundError\u001b[0m: [Errno 2] No such file or directory: 'Homo_sapiens.GRCh38.pep.all.fa.gz'"
       ]
      }
     ],
     "persistentId": "be06ded1-d7d4-4f21-9961-3aed5ff30cbf",
     "text": "peptide_graph = make_peptide_bloom_filter(\"Homo_sapiens.GRCh38.pep.all.fa.gz\", 7)"
    },
    "executionTime": "2019-10-11T15:56:40.838Z"
   },
   {
    "cell": {
     "executionCount": 14,
     "executionEventId": "65043f97-ca65-4674-90da-a361efbb2e88",
     "hasError": false,
     "id": "31c3fe01-1037-44d7-b566-b186d79dff80",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "/mnt/data_sm/olga/kmer-hashing/classify_coding_vs_noncoding\n"
      }
     ],
     "persistentId": "1d547cf2-254f-4fe2-9bf0-5b7e39be4376",
     "text": "cd $folder"
    },
    "executionTime": "2019-10-11T15:56:48.012Z"
   },
   {
    "cell": {
     "executionCount": 15,
     "executionEventId": "80552913-827e-47c6-bc1a-b6909f59a197",
     "hasError": false,
     "id": "c43c763c-d708-4f43-a304-9fae85683297",
     "outputs": [],
     "persistentId": "be06ded1-d7d4-4f21-9961-3aed5ff30cbf",
     "text": "peptide_graph = make_peptide_bloom_filter(\"Homo_sapiens.GRCh38.pep.all.fa.gz\", 7)"
    },
    "executionTime": "2019-10-11T15:56:53.938Z"
   },
   {
    "cell": {
     "executionCount": 17,
     "executionEventId": "e5b976fb-b605-494a-b87d-6df89d03c89d",
     "hasError": true,
     "id": "b242d89f-8d25-4127-b875-8a85fa69d914",
     "outputs": [
      {
       "ename": "KeyError",
       "evalue": "'description'",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
        "\u001b[0;32m<timed exec>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n",
        "\u001b[0;32m~/code/kmer-hashing/kh-tools/khtools/partition_reads.py\u001b[0m in \u001b[0;36mscore_reads\u001b[0;34m(reads, peptide_graph, peptide_ksize, jaccard_threshold, molecule, verbose)\u001b[0m\n\u001b[1;32m    139\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    140\u001b[0m     \u001b[0;32mfor\u001b[0m \u001b[0mrecord\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mscreed\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mopen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mreads\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 141\u001b[0;31m         \u001b[0mdescription\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mrecord\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'description'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    142\u001b[0m         \u001b[0mseq\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mSeq\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrecord\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'sequence'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    143\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/screed/screedRecord.py\u001b[0m in \u001b[0;36m__getitem__\u001b[0;34m(self, idx)\u001b[0m\n\u001b[1;32m     56\u001b[0m                 \u001b[0mtrimmed\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'quality'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mtrimmed\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'quality'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0midx\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     57\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mRecord\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m**\u001b[0m\u001b[0mtrimmed\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 58\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0md\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0midx\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     59\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     60\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m__delitem__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;31mKeyError\u001b[0m: 'description'"
       ]
      }
     ],
     "persistentId": "be06ded1-d7d4-4f21-9961-3aed5ff30cbf",
     "text": "%%time\n\nreads = 'SRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz'\n\ncoding_scores = score_reads(reads, peptide_graph, peptide_ksize)"
    },
    "executionTime": "2019-10-11T15:57:50.063Z"
   },
   {
    "cell": {
     "executionCount": 18,
     "executionEventId": "604c79a5-e63d-4619-a852-0571bbee4eec",
     "hasError": false,
     "id": "3281c496-16a4-496e-81d4-fa57e8d1a556",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "> \u001b[0;32m/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/screed/screedRecord.py\u001b[0m(58)\u001b[0;36m__getitem__\u001b[0;34m()\u001b[0m\n\u001b[0;32m     56 \u001b[0;31m                \u001b[0mtrimmed\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'quality'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mtrimmed\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'quality'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0midx\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     57 \u001b[0;31m            \u001b[0;32mreturn\u001b[0m \u001b[0mRecord\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m**\u001b[0m\u001b[0mtrimmed\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m---> 58 \u001b[0;31m        \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0md\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0midx\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     59 \u001b[0;31m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     60 \u001b[0;31m    \u001b[0;32mdef\u001b[0m \u001b[0m__delitem__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\n"
      },
      {
       "name": "stdin",
       "output_type": "stream",
       "text": "ipdb>  u\n"
      },
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "> \u001b[0;32m/home/olga/code/kmer-hashing/kh-tools/khtools/partition_reads.py\u001b[0m(141)\u001b[0;36mscore_reads\u001b[0;34m()\u001b[0m\n\u001b[0;32m    139 \u001b[0;31m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    140 \u001b[0;31m    \u001b[0;32mfor\u001b[0m \u001b[0mrecord\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mscreed\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mopen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mreads\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m--> 141 \u001b[0;31m        \u001b[0mdescription\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mrecord\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'description'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    142 \u001b[0;31m        \u001b[0mseq\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mSeq\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrecord\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'sequence'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    143 \u001b[0;31m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\n"
      },
      {
       "name": "stdin",
       "output_type": "stream",
       "text": "ipdb>  record\n"
      },
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "{'name': 'SRR306838.1531 Ibis_Run100924_C3PO:6:1:15718:1062/1', 'sequence': 'ACCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCC', 'annotations': '', 'quality': '(04147:;:9<<:7;88<<B::;<=:<:=<;>>=@>>8<;;<=;C;>;:5:;9<<::6@;E;?:C@=:9:67'}\n"
      },
      {
       "name": "stdin",
       "output_type": "stream",
       "text": "ipdb>  q\n"
      }
     ],
     "persistentId": "ede6e1ba-f524-4ceb-b35b-6c20f91d437c",
     "text": "%debug"
    },
    "executionTime": "2019-10-11T15:58:10.827Z"
   },
   {
    "cell": {
     "executionCount": 19,
     "executionEventId": "77f24723-8c13-4f62-9e66-6e9dc130a38c",
     "hasError": true,
     "id": "b242d89f-8d25-4127-b875-8a85fa69d914",
     "outputs": [
      {
       "name": "stderr",
       "output_type": "stream",
       "text": "/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n"
      },
      {
       "ename": "AttributeError",
       "evalue": "description",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/screed/screedRecord.py\u001b[0m in \u001b[0;36m__getattr__\u001b[0;34m(self, name)\u001b[0m\n\u001b[1;32m     40\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 41\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0md\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mname\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     42\u001b[0m         \u001b[0;32mexcept\u001b[0m \u001b[0mKeyError\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;31mKeyError\u001b[0m: 'description'",
        "\nDuring handling of the above exception, another exception occurred:\n",
        "\u001b[0;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
        "\u001b[0;32m<timed exec>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n",
        "\u001b[0;32m~/code/kmer-hashing/kh-tools/khtools/partition_reads.py\u001b[0m in \u001b[0;36mscore_reads\u001b[0;34m(reads, peptide_graph, peptide_ksize, jaccard_threshold, molecule, verbose)\u001b[0m\n\u001b[1;32m    151\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mlow_complexity\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    152\u001b[0m             scoring_lines.append(\n\u001b[0;32m--> 153\u001b[0;31m                 [record.description, -1, n_kmers, 'low complexity'])\n\u001b[0m\u001b[1;32m    154\u001b[0m             \u001b[0;32mcontinue\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    155\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/screed/screedRecord.py\u001b[0m in \u001b[0;36m__getattr__\u001b[0;34m(self, name)\u001b[0m\n\u001b[1;32m     41\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0md\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mname\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     42\u001b[0m         \u001b[0;32mexcept\u001b[0m \u001b[0mKeyError\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 43\u001b[0;31m             \u001b[0;32mraise\u001b[0m \u001b[0mAttributeError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mname\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     44\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     45\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m__len__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;31mAttributeError\u001b[0m: description"
       ]
      }
     ],
     "persistentId": "be06ded1-d7d4-4f21-9961-3aed5ff30cbf",
     "text": "%%time\n\nreads = 'SRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz'\n\ncoding_scores = score_reads(reads, peptide_graph, peptide_ksize)"
    },
    "executionTime": "2019-10-11T15:58:49.441Z"
   },
   {
    "cell": {
     "executionCount": 20,
     "executionEventId": "a11fb69b-fc36-4374-8cd8-76f6ee17adbf",
     "hasError": true,
     "id": "b242d89f-8d25-4127-b875-8a85fa69d914",
     "outputs": [
      {
       "name": "stderr",
       "output_type": "stream",
       "text": "0it [00:00, ?it/s]/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n\n"
      },
      {
       "ename": "AttributeError",
       "evalue": "description",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/screed/screedRecord.py\u001b[0m in \u001b[0;36m__getattr__\u001b[0;34m(self, name)\u001b[0m\n\u001b[1;32m     40\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 41\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0md\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mname\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     42\u001b[0m         \u001b[0;32mexcept\u001b[0m \u001b[0mKeyError\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;31mKeyError\u001b[0m: 'description'",
        "\nDuring handling of the above exception, another exception occurred:\n",
        "\u001b[0;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
        "\u001b[0;32m<timed exec>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n",
        "\u001b[0;32m~/code/kmer-hashing/kh-tools/khtools/partition_reads.py\u001b[0m in \u001b[0;36mscore_reads\u001b[0;34m(reads, peptide_graph, peptide_ksize, jaccard_threshold, molecule, verbose)\u001b[0m\n\u001b[1;32m    151\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mlow_complexity\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    152\u001b[0m             scoring_lines.append(\n\u001b[0;32m--> 153\u001b[0;31m                 [record.description, -1, n_kmers, 'low complexity'])\n\u001b[0m\u001b[1;32m    154\u001b[0m             \u001b[0;32mcontinue\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    155\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/screed/screedRecord.py\u001b[0m in \u001b[0;36m__getattr__\u001b[0;34m(self, name)\u001b[0m\n\u001b[1;32m     41\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0md\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mname\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     42\u001b[0m         \u001b[0;32mexcept\u001b[0m \u001b[0mKeyError\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 43\u001b[0;31m             \u001b[0;32mraise\u001b[0m \u001b[0mAttributeError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mname\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     44\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     45\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m__len__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;31mAttributeError\u001b[0m: description"
       ]
      }
     ],
     "persistentId": "be06ded1-d7d4-4f21-9961-3aed5ff30cbf",
     "text": "%%time\n\nreads = 'SRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz'\n\ncoding_scores = score_reads(reads, peptide_graph, peptide_ksize)"
    },
    "executionTime": "2019-10-11T15:59:28.027Z"
   },
   {
    "cell": {
     "executionCount": 21,
     "executionEventId": "9fe82ab8-fe5b-437b-9a7e-3cad2c104105",
     "hasError": true,
     "id": "b242d89f-8d25-4127-b875-8a85fa69d914",
     "outputs": [
      {
       "name": "stderr",
       "output_type": "stream",
       "text": "0it [00:00, ?it/s]/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n\n"
      },
      {
       "ename": "AttributeError",
       "evalue": "description",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/screed/screedRecord.py\u001b[0m in \u001b[0;36m__getattr__\u001b[0;34m(self, name)\u001b[0m\n\u001b[1;32m     40\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 41\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0md\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mname\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     42\u001b[0m         \u001b[0;32mexcept\u001b[0m \u001b[0mKeyError\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;31mKeyError\u001b[0m: 'description'",
        "\nDuring handling of the above exception, another exception occurred:\n",
        "\u001b[0;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
        "\u001b[0;32m<timed exec>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n",
        "\u001b[0;32m~/code/kmer-hashing/kh-tools/khtools/partition_reads.py\u001b[0m in \u001b[0;36mscore_reads\u001b[0;34m(reads, peptide_graph, peptide_ksize, jaccard_threshold, molecule, verbose)\u001b[0m\n\u001b[1;32m    151\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mlow_complexity\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    152\u001b[0m             scoring_lines.append(\n\u001b[0;32m--> 153\u001b[0;31m                 [record.description, -1, n_kmers, 'low complexity'])\n\u001b[0m\u001b[1;32m    154\u001b[0m             \u001b[0;32mcontinue\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    155\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/screed/screedRecord.py\u001b[0m in \u001b[0;36m__getattr__\u001b[0;34m(self, name)\u001b[0m\n\u001b[1;32m     41\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0md\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mname\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     42\u001b[0m         \u001b[0;32mexcept\u001b[0m \u001b[0mKeyError\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 43\u001b[0;31m             \u001b[0;32mraise\u001b[0m \u001b[0mAttributeError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mname\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     44\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     45\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m__len__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;31mAttributeError\u001b[0m: description"
       ]
      }
     ],
     "persistentId": "be06ded1-d7d4-4f21-9961-3aed5ff30cbf",
     "text": "%%time\n\nreads = 'SRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz'\n\ncoding_scores = score_reads(reads, peptide_graph, peptide_ksize)"
    },
    "executionTime": "2019-10-11T15:59:32.148Z"
   },
   {
    "cell": {
     "executionCount": 22,
     "executionEventId": "62c61d51-3d71-4273-bcbe-7e7f0728ffcd",
     "hasError": true,
     "id": "b242d89f-8d25-4127-b875-8a85fa69d914",
     "outputs": [
      {
       "name": "stderr",
       "output_type": "stream",
       "text": "0it [00:00, ?it/s]/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n\n"
      },
      {
       "ename": "AttributeError",
       "evalue": "description",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/screed/screedRecord.py\u001b[0m in \u001b[0;36m__getattr__\u001b[0;34m(self, name)\u001b[0m\n\u001b[1;32m     40\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 41\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0md\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mname\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     42\u001b[0m         \u001b[0;32mexcept\u001b[0m \u001b[0mKeyError\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;31mKeyError\u001b[0m: 'description'",
        "\nDuring handling of the above exception, another exception occurred:\n",
        "\u001b[0;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
        "\u001b[0;32m<timed exec>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n",
        "\u001b[0;32m~/code/kmer-hashing/kh-tools/khtools/partition_reads.py\u001b[0m in \u001b[0;36mscore_reads\u001b[0;34m(reads, peptide_graph, peptide_ksize, jaccard_threshold, molecule, verbose)\u001b[0m\n\u001b[1;32m    151\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mlow_complexity\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    152\u001b[0m             scoring_lines.append(\n\u001b[0;32m--> 153\u001b[0;31m                 [record.description, -1, n_kmers, 'low complexity'])\n\u001b[0m\u001b[1;32m    154\u001b[0m             \u001b[0;32mcontinue\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    155\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/screed/screedRecord.py\u001b[0m in \u001b[0;36m__getattr__\u001b[0;34m(self, name)\u001b[0m\n\u001b[1;32m     41\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0md\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mname\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     42\u001b[0m         \u001b[0;32mexcept\u001b[0m \u001b[0mKeyError\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 43\u001b[0;31m             \u001b[0;32mraise\u001b[0m \u001b[0mAttributeError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mname\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     44\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     45\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m__len__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;31mAttributeError\u001b[0m: description"
       ]
      }
     ],
     "persistentId": "be06ded1-d7d4-4f21-9961-3aed5ff30cbf",
     "text": "%%time\n\nreads = 'SRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz'\n\ncoding_scores = score_reads(reads, peptide_graph, peptide_ksize)"
    },
    "executionTime": "2019-10-11T15:59:35.521Z"
   },
   {
    "cell": {
     "executionCount": 23,
     "executionEventId": "78c75d49-d679-457a-b9e7-3099e34e749d",
     "hasError": true,
     "id": "b242d89f-8d25-4127-b875-8a85fa69d914",
     "outputs": [
      {
       "name": "stderr",
       "output_type": "stream",
       "text": "0it [00:00, ?it/s]/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n\n"
      },
      {
       "ename": "AttributeError",
       "evalue": "description",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/screed/screedRecord.py\u001b[0m in \u001b[0;36m__getattr__\u001b[0;34m(self, name)\u001b[0m\n\u001b[1;32m     40\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 41\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0md\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mname\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     42\u001b[0m         \u001b[0;32mexcept\u001b[0m \u001b[0mKeyError\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;31mKeyError\u001b[0m: 'description'",
        "\nDuring handling of the above exception, another exception occurred:\n",
        "\u001b[0;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
        "\u001b[0;32m<timed exec>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n",
        "\u001b[0;32m~/code/kmer-hashing/kh-tools/khtools/partition_reads.py\u001b[0m in \u001b[0;36mscore_reads\u001b[0;34m(reads, peptide_graph, peptide_ksize, jaccard_threshold, molecule, verbose)\u001b[0m\n\u001b[1;32m    151\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mlow_complexity\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    152\u001b[0m             scoring_lines.append(\n\u001b[0;32m--> 153\u001b[0;31m                 [record.description, -1, n_kmers, 'low complexity'])\n\u001b[0m\u001b[1;32m    154\u001b[0m             \u001b[0;32mcontinue\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    155\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/screed/screedRecord.py\u001b[0m in \u001b[0;36m__getattr__\u001b[0;34m(self, name)\u001b[0m\n\u001b[1;32m     41\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0md\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mname\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     42\u001b[0m         \u001b[0;32mexcept\u001b[0m \u001b[0mKeyError\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 43\u001b[0;31m             \u001b[0;32mraise\u001b[0m \u001b[0mAttributeError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mname\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     44\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     45\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m__len__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;31mAttributeError\u001b[0m: description"
       ]
      }
     ],
     "persistentId": "be06ded1-d7d4-4f21-9961-3aed5ff30cbf",
     "text": "%%time\n\nreads = 'SRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz'\n\ncoding_scores = score_reads(reads, peptide_graph, peptide_ksize)"
    },
    "executionTime": "2019-10-11T15:59:56.073Z"
   },
   {
    "cell": {
     "executionCount": 1,
     "executionEventId": "8181e65d-b7c8-4034-8c92-3f780000662b",
     "hasError": false,
     "id": "0f1ea26a-ecfa-4359-8ba4-c0f0d6feb077",
     "outputs": [],
     "persistentId": "71299be4-a00f-4638-92a2-ea174ddf208d",
     "text": "%load_ext autoreload\n%autoreload 2"
    },
    "executionTime": "2019-10-11T16:03:12.910Z"
   },
   {
    "cell": {
     "executionCount": 2,
     "executionEventId": "7a2e54cd-ad2e-46ee-9a73-a358968c9a46",
     "hasError": false,
     "id": "5e35ad66-c7bb-43f2-86d3-6da92b22e4a6",
     "outputs": [],
     "persistentId": "ecc17310-b85a-49bd-8d60-8de454040c04",
     "text": "folder = '/home/olga/data_sm/kmer-hashing/classify_coding_vs_noncoding/'"
    },
    "executionTime": "2019-10-11T16:03:13.059Z"
   },
   {
    "cell": {
     "executionCount": 3,
     "executionEventId": "44e4a9c5-14b7-4540-955c-a3e974387263",
     "hasError": false,
     "id": "31c3fe01-1037-44d7-b566-b186d79dff80",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "/mnt/data_sm/olga/kmer-hashing/classify_coding_vs_noncoding\n"
      }
     ],
     "persistentId": "1d547cf2-254f-4fe2-9bf0-5b7e39be4376",
     "text": "cd $folder"
    },
    "executionTime": "2019-10-11T16:03:13.251Z"
   },
   {
    "cell": {
     "executionCount": 4,
     "executionEventId": "5c4a72d6-812f-42d4-adda-d8b0aa32dc27",
     "hasError": false,
     "id": "81fe1d1b-9150-4e3a-a0e2-f2470c0bb85c",
     "outputs": [],
     "persistentId": "56e48bd4-fcc6-4d93-b4e3-0984be1fc1d5",
     "text": "import gzip\nfrom io import StringIO\nfrom pprint import pprint\n\nfrom khmer import Nodegraph\nimport matplotlib.pyplot as plt\nimport screed\nfrom sourmash._minhash import hash_murmur\nfrom khmer.khmer_args import calculate_graphsize\nimport pandas as pd\nfrom sourmash.logging import notify\n\nimport seaborn as sns\n\nfrom sklearn import metrics\nfrom khtools.compare_kmer_content import kmerize, hpize, dayhoffize\n\nfrom Bio.Seq import Seq\nfrom Bio import SeqIO\n\n# make all warnings error to get a stack trace\nimport warnings\nwarnings.filterwarnings(\"error\")\n"
    },
    "executionTime": "2019-10-11T16:03:15.405Z"
   },
   {
    "cell": {
     "executionCount": 5,
     "executionEventId": "d465413b-fc7e-4a69-b42f-db95bf7360c6",
     "hasError": false,
     "id": "e50808b6-6666-4543-9299-8075c789791c",
     "outputs": [],
     "persistentId": "1e54230e-ceec-42fc-afa6-7b32e219378b",
     "text": "from khtools.partition_reads import score_reads, make_peptide_bloom_filter"
    },
    "executionTime": "2019-10-11T16:03:15.440Z"
   },
   {
    "cell": {
     "executionCount": 7,
     "executionEventId": "05a3dcc1-455c-453d-9cea-3665c37b54cb",
     "hasError": true,
     "id": "b242d89f-8d25-4127-b875-8a85fa69d914",
     "outputs": [
      {
       "name": "stderr",
       "output_type": "stream",
       "text": "Exception ignored in: <_io.FileIO name='SRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.fq.gz' mode='rb' closefd=True>\nResourceWarning: unclosed file <_io.BufferedReader name='SRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.fq.gz'>\n0it [00:00, ?it/s]\n"
      },
      {
       "ename": "BiopythonWarning",
       "evalue": "Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mBiopythonWarning\u001b[0m                          Traceback (most recent call last)",
        "\u001b[0;32m<timed exec>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n",
        "\u001b[0;32m~/code/kmer-hashing/kh-tools/khtools/partition_reads.py\u001b[0m in \u001b[0;36mscore_reads\u001b[0;34m(reads, peptide_graph, peptide_ksize, jaccard_threshold, molecule, verbose)\u001b[0m\n\u001b[1;32m    148\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    149\u001b[0m         \u001b[0;31m# Check if nucleotide sequence is low complexity\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 150\u001b[0;31m         \u001b[0mlow_complexity\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mn_kmers\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mis_low_complexity\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mseq\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnucleotide_ksize\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    151\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mlow_complexity\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    152\u001b[0m             scoring_lines.append(\n",
        "\u001b[0;32m~/code/kmer-hashing/kh-tools/khtools/partition_reads.py\u001b[0m in \u001b[0;36mis_low_complexity\u001b[0;34m(sequence, ksize)\u001b[0m\n\u001b[1;32m    125\u001b[0m \u001b[0;32mdef\u001b[0m \u001b[0mis_low_complexity\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msequence\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mksize\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    126\u001b[0m     \u001b[0;34m\"\"\"CHeck if seqauence is low complexity, i.e. low entropy, mostly repetitive\"\"\"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 127\u001b[0;31m     \u001b[0mkmers\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mkmerize\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msequence\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mksize\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    128\u001b[0m     \u001b[0mn_kmers\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkmers\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    129\u001b[0m     \u001b[0mn_possible_kmers_on_sequence\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msequence\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m-\u001b[0m \u001b[0mksize\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0;36m1\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/code/kmer-hashing/kh-tools/khtools/compare_kmer_content.py\u001b[0m in \u001b[0;36mkmerize\u001b[0;34m(seq, ksize)\u001b[0m\n\u001b[1;32m     75\u001b[0m \u001b[0;32mdef\u001b[0m \u001b[0mkmerize\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mseq\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mksize\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     76\u001b[0m     \u001b[0;34m\"\"\"Return the set of unique k-mers from the sequence\"\"\"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 77\u001b[0;31m     \u001b[0;32mreturn\u001b[0m \u001b[0mset\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mseq\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0mksize\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mi\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mrange\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mseq\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m-\u001b[0m\u001b[0mksize\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     78\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     79\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py\u001b[0m in \u001b[0;36m__hash__\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    173\u001b[0m                       \u001b[0;34m\"or my_dict[id(my_seq)] if you want the old behaviour, \"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    174\u001b[0m                       \u001b[0;34m\"or use hash(str(my_seq)) or my_dict[str(my_seq)] for \"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 175\u001b[0;31m                       \"the new string hashing behaviour.\", BiopythonWarning)\n\u001b[0m\u001b[1;32m    176\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mhash\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    177\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;31mBiopythonWarning\u001b[0m: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour."
       ]
      }
     ],
     "persistentId": "be06ded1-d7d4-4f21-9961-3aed5ff30cbf",
     "text": "%%time\n\nreads = 'SRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz'\nsubsampled_reads = 'SRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.fq.gz'\n\ncoding_scores = score_reads(subsampled_reads, peptide_graph, peptide_ksize)"
    },
    "executionTime": "2019-10-11T16:03:20.666Z"
   },
   {
    "cell": {
     "executionCount": 8,
     "executionEventId": "5c0a9a9f-cd1b-413c-915e-13f7f7a13586",
     "hasError": true,
     "id": "b242d89f-8d25-4127-b875-8a85fa69d914",
     "outputs": [
      {
       "name": "stderr",
       "output_type": "stream",
       "text": "Exception ignored in: <_io.FileIO name='SRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.fq.gz' mode='rb' closefd=True>\nResourceWarning: unclosed file <_io.BufferedReader name='SRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.fq.gz'>\n0it [00:00, ?it/s]\n"
      },
      {
       "ename": "BiopythonWarning",
       "evalue": "Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mBiopythonWarning\u001b[0m                          Traceback (most recent call last)",
        "\u001b[0;32m<timed exec>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n",
        "\u001b[0;32m~/code/kmer-hashing/kh-tools/khtools/partition_reads.py\u001b[0m in \u001b[0;36mscore_reads\u001b[0;34m(reads, peptide_graph, peptide_ksize, jaccard_threshold, molecule, verbose)\u001b[0m\n\u001b[1;32m    148\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    149\u001b[0m         \u001b[0;31m# Check if nucleotide sequence is low complexity\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 150\u001b[0;31m         \u001b[0mlow_complexity\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mn_kmers\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mis_low_complexity\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mseq\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnucleotide_ksize\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    151\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mlow_complexity\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    152\u001b[0m             scoring_lines.append(\n",
        "\u001b[0;32m~/code/kmer-hashing/kh-tools/khtools/partition_reads.py\u001b[0m in \u001b[0;36mis_low_complexity\u001b[0;34m(sequence, ksize)\u001b[0m\n\u001b[1;32m    125\u001b[0m \u001b[0;32mdef\u001b[0m \u001b[0mis_low_complexity\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msequence\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mksize\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    126\u001b[0m     \u001b[0;34m\"\"\"CHeck if seqauence is low complexity, i.e. low entropy, mostly repetitive\"\"\"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 127\u001b[0;31m     \u001b[0mkmers\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mkmerize\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msequence\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mksize\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    128\u001b[0m     \u001b[0mn_kmers\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkmers\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    129\u001b[0m     \u001b[0mn_possible_kmers_on_sequence\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msequence\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m-\u001b[0m \u001b[0mksize\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0;36m1\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/code/kmer-hashing/kh-tools/khtools/compare_kmer_content.py\u001b[0m in \u001b[0;36mkmerize\u001b[0;34m(seq, ksize)\u001b[0m\n\u001b[1;32m     75\u001b[0m \u001b[0;32mdef\u001b[0m \u001b[0mkmerize\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mseq\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mksize\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     76\u001b[0m     \u001b[0;34m\"\"\"Return the set of unique k-mers from the sequence\"\"\"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 77\u001b[0;31m     \u001b[0;32mreturn\u001b[0m \u001b[0mset\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mseq\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0mksize\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mi\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mrange\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mseq\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m-\u001b[0m\u001b[0mksize\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     78\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     79\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py\u001b[0m in \u001b[0;36m__hash__\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    173\u001b[0m                       \u001b[0;34m\"or my_dict[id(my_seq)] if you want the old behaviour, \"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    174\u001b[0m                       \u001b[0;34m\"or use hash(str(my_seq)) or my_dict[str(my_seq)] for \"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 175\u001b[0;31m                       \"the new string hashing behaviour.\", BiopythonWarning)\n\u001b[0m\u001b[1;32m    176\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mhash\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    177\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;31mBiopythonWarning\u001b[0m: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour."
       ]
      }
     ],
     "persistentId": "be06ded1-d7d4-4f21-9961-3aed5ff30cbf",
     "text": "%%time\n\nreads = 'SRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz'\nsubsampled_reads = 'SRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.fq.gz'\n\ncoding_scores = score_reads(subsampled_reads, peptide_graph, peptide_ksize)"
    },
    "executionTime": "2019-10-11T16:04:56.900Z"
   },
   {
    "cell": {
     "executionCount": 9,
     "executionEventId": "a2358899-0cba-40f2-badf-3a40b56eded0",
     "hasError": false,
     "id": "81fe1d1b-9150-4e3a-a0e2-f2470c0bb85c",
     "outputs": [],
     "persistentId": "56e48bd4-fcc6-4d93-b4e3-0984be1fc1d5",
     "text": "import gzip\nfrom io import StringIO\nfrom pprint import pprint\n\nfrom khmer import Nodegraph\nimport matplotlib.pyplot as plt\nimport screed\nfrom sourmash._minhash import hash_murmur\nfrom khmer.khmer_args import calculate_graphsize\nimport pandas as pd\nfrom sourmash.logging import notify\n\nimport seaborn as sns\n\nfrom sklearn import metrics\nfrom khtools.compare_kmer_content import kmerize, hpize, dayhoffize\n\nfrom Bio.Seq import Seq\nfrom Bio import SeqIO\n\n# make all warnings error to get a stack trace\nimport warnings\nwarnings.filterwarnings(\"error\")\n"
    },
    "executionTime": "2019-10-11T16:05:09.349Z"
   },
   {
    "cell": {
     "executionCount": 10,
     "executionEventId": "40fe8b69-2d7d-4273-b632-d08a42a9f31c",
     "hasError": true,
     "id": "b242d89f-8d25-4127-b875-8a85fa69d914",
     "outputs": [
      {
       "name": "stderr",
       "output_type": "stream",
       "text": "Exception ignored in: <_io.FileIO name='SRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.fq.gz' mode='rb' closefd=True>\nResourceWarning: unclosed file <_io.BufferedReader name='SRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.fq.gz'>\n1000it [00:00, 2914.96it/s]\n"
      },
      {
       "ename": "AssertionError",
       "evalue": "2 columns passed, passed data had 4 columns",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mAssertionError\u001b[0m                            Traceback (most recent call last)",
        "\u001b[0;32m<timed exec>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n",
        "\u001b[0;32m~/code/kmer-hashing/kh-tools/khtools/partition_reads.py\u001b[0m in \u001b[0;36mscore_reads\u001b[0;34m(reads, peptide_graph, peptide_ksize, jaccard_threshold, molecule, verbose)\u001b[0m\n\u001b[1;32m    209\u001b[0m     scoring_df = pd.DataFrame(scoring_lines,\n\u001b[1;32m    210\u001b[0m                                  columns=['read_id',\n\u001b[0;32m--> 211\u001b[0;31m                                           \u001b[0;34m'jaccard_in_peptide_db'\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    212\u001b[0m                                           'classification'])\n\u001b[1;32m    213\u001b[0m     \u001b[0;32mreturn\u001b[0m \u001b[0mscoring_df\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, data, index, columns, dtype, copy)\u001b[0m\n\u001b[1;32m    433\u001b[0m                     \u001b[0;32mif\u001b[0m \u001b[0mis_named_tuple\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0mcolumns\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    434\u001b[0m                         \u001b[0mcolumns\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdata\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_fields\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 435\u001b[0;31m                     \u001b[0marrays\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcolumns\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mto_arrays\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcolumns\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdtype\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mdtype\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    436\u001b[0m                     \u001b[0mcolumns\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mensure_index\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    437\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/internals/construction.py\u001b[0m in \u001b[0;36mto_arrays\u001b[0;34m(data, columns, coerce_float, dtype)\u001b[0m\n\u001b[1;32m    402\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0misinstance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mlist\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtuple\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    403\u001b[0m         return _list_to_arrays(data, columns, coerce_float=coerce_float,\n\u001b[0;32m--> 404\u001b[0;31m                                dtype=dtype)\n\u001b[0m\u001b[1;32m    405\u001b[0m     \u001b[0;32melif\u001b[0m \u001b[0misinstance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcompat\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mMapping\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    406\u001b[0m         return _list_of_dict_to_arrays(data, columns,\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/internals/construction.py\u001b[0m in \u001b[0;36m_list_to_arrays\u001b[0;34m(data, columns, coerce_float, dtype)\u001b[0m\n\u001b[1;32m    434\u001b[0m         \u001b[0mcontent\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mlist\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlib\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mto_object_array\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mT\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    435\u001b[0m     return _convert_object_array(content, columns, dtype=dtype,\n\u001b[0;32m--> 436\u001b[0;31m                                  coerce_float=coerce_float)\n\u001b[0m\u001b[1;32m    437\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    438\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/internals/construction.py\u001b[0m in \u001b[0;36m_convert_object_array\u001b[0;34m(content, columns, coerce_float, dtype)\u001b[0m\n\u001b[1;32m    490\u001b[0m             raise AssertionError('{col:d} columns passed, passed data had '\n\u001b[1;32m    491\u001b[0m                                  '{con} columns'.format(col=len(columns),\n\u001b[0;32m--> 492\u001b[0;31m                                                         con=len(content)))\n\u001b[0m\u001b[1;32m    493\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    494\u001b[0m     \u001b[0;31m# provide soft conversion of object dtypes\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;31mAssertionError\u001b[0m: 2 columns passed, passed data had 4 columns"
       ]
      }
     ],
     "persistentId": "be06ded1-d7d4-4f21-9961-3aed5ff30cbf",
     "text": "%%time\n\nreads = 'SRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz'\nsubsampled_reads = 'SRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.fq.gz'\n\ncoding_scores = score_reads(subsampled_reads, peptide_graph, peptide_ksize)"
    },
    "executionTime": "2019-10-11T16:05:11.061Z"
   },
   {
    "cell": {
     "executionCount": 11,
     "executionEventId": "082afb20-7cd3-4f4f-8277-06158422eba5",
     "hasError": false,
     "id": "81fe1d1b-9150-4e3a-a0e2-f2470c0bb85c",
     "outputs": [],
     "persistentId": "56e48bd4-fcc6-4d93-b4e3-0984be1fc1d5",
     "text": "import gzip\nfrom io import StringIO\nfrom pprint import pprint\n\nfrom khmer import Nodegraph\nimport matplotlib.pyplot as plt\nimport screed\nfrom sourmash._minhash import hash_murmur\nfrom khmer.khmer_args import calculate_graphsize\nimport pandas as pd\nfrom sourmash.logging import notify\n\nimport seaborn as sns\n\nfrom sklearn import metrics\nfrom khtools.compare_kmer_content import kmerize, hpize, dayhoffize\n\nfrom Bio.Seq import Seq\nfrom Bio import SeqIO\n\n# make all warnings error to get a stack trace\nimport warnings\nwarnings.filterwarnings(\"default\")\n"
    },
    "executionTime": "2019-10-11T16:05:59.386Z"
   },
   {
    "cell": {
     "executionCount": 12,
     "executionEventId": "fa2a266c-9fe1-400d-b0ea-fb75347b14be",
     "hasError": true,
     "id": "b242d89f-8d25-4127-b875-8a85fa69d914",
     "outputs": [
      {
       "name": "stderr",
       "output_type": "stream",
       "text": "/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/screed/openscreed.py:39: ResourceWarning: unclosed file <_io.BufferedReader name='SRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.fq.gz'>\n  self.iter_fn = self.open_reader(filename, *args, **kwargs)\n1000it [00:00, 2926.89it/s]\n"
      },
      {
       "ename": "AssertionError",
       "evalue": "2 columns passed, passed data had 4 columns",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mAssertionError\u001b[0m                            Traceback (most recent call last)",
        "\u001b[0;32m<timed exec>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n",
        "\u001b[0;32m~/code/kmer-hashing/kh-tools/khtools/partition_reads.py\u001b[0m in \u001b[0;36mscore_reads\u001b[0;34m(reads, peptide_graph, peptide_ksize, jaccard_threshold, molecule, verbose)\u001b[0m\n\u001b[1;32m    209\u001b[0m     scoring_df = pd.DataFrame(scoring_lines,\n\u001b[1;32m    210\u001b[0m                                  columns=['read_id',\n\u001b[0;32m--> 211\u001b[0;31m                                           \u001b[0;34m'jaccard_in_peptide_db'\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    212\u001b[0m                                           'classification'])\n\u001b[1;32m    213\u001b[0m     \u001b[0;32mreturn\u001b[0m \u001b[0mscoring_df\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, data, index, columns, dtype, copy)\u001b[0m\n\u001b[1;32m    433\u001b[0m                     \u001b[0;32mif\u001b[0m \u001b[0mis_named_tuple\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0mcolumns\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    434\u001b[0m                         \u001b[0mcolumns\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdata\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_fields\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 435\u001b[0;31m                     \u001b[0marrays\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcolumns\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mto_arrays\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcolumns\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdtype\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mdtype\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    436\u001b[0m                     \u001b[0mcolumns\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mensure_index\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    437\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/internals/construction.py\u001b[0m in \u001b[0;36mto_arrays\u001b[0;34m(data, columns, coerce_float, dtype)\u001b[0m\n\u001b[1;32m    402\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0misinstance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mlist\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtuple\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    403\u001b[0m         return _list_to_arrays(data, columns, coerce_float=coerce_float,\n\u001b[0;32m--> 404\u001b[0;31m                                dtype=dtype)\n\u001b[0m\u001b[1;32m    405\u001b[0m     \u001b[0;32melif\u001b[0m \u001b[0misinstance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcompat\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mMapping\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    406\u001b[0m         return _list_of_dict_to_arrays(data, columns,\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/internals/construction.py\u001b[0m in \u001b[0;36m_list_to_arrays\u001b[0;34m(data, columns, coerce_float, dtype)\u001b[0m\n\u001b[1;32m    434\u001b[0m         \u001b[0mcontent\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mlist\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlib\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mto_object_array\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mT\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    435\u001b[0m     return _convert_object_array(content, columns, dtype=dtype,\n\u001b[0;32m--> 436\u001b[0;31m                                  coerce_float=coerce_float)\n\u001b[0m\u001b[1;32m    437\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    438\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/internals/construction.py\u001b[0m in \u001b[0;36m_convert_object_array\u001b[0;34m(content, columns, coerce_float, dtype)\u001b[0m\n\u001b[1;32m    490\u001b[0m             raise AssertionError('{col:d} columns passed, passed data had '\n\u001b[1;32m    491\u001b[0m                                  '{con} columns'.format(col=len(columns),\n\u001b[0;32m--> 492\u001b[0;31m                                                         con=len(content)))\n\u001b[0m\u001b[1;32m    493\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    494\u001b[0m     \u001b[0;31m# provide soft conversion of object dtypes\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;31mAssertionError\u001b[0m: 2 columns passed, passed data had 4 columns"
       ]
      }
     ],
     "persistentId": "be06ded1-d7d4-4f21-9961-3aed5ff30cbf",
     "text": "%%time\n\nreads = 'SRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz'\nsubsampled_reads = 'SRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.fq.gz'\n\ncoding_scores = score_reads(subsampled_reads, peptide_graph, peptide_ksize)"
    },
    "executionTime": "2019-10-11T16:06:01.452Z"
   },
   {
    "cell": {
     "executionCount": 13,
     "executionEventId": "99f4d730-fae6-40a0-97e7-ad3d5a29f3c7",
     "hasError": true,
     "id": "b242d89f-8d25-4127-b875-8a85fa69d914",
     "outputs": [
      {
       "name": "stderr",
       "output_type": "stream",
       "text": "/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/screed/openscreed.py:39: ResourceWarning: unclosed file <_io.BufferedReader name='SRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.fq.gz'>\n  self.iter_fn = self.open_reader(filename, *args, **kwargs)\n1000it [00:00, 2896.37it/s]\n"
      },
      {
       "ename": "AssertionError",
       "evalue": "3 columns passed, passed data had 4 columns",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mAssertionError\u001b[0m                            Traceback (most recent call last)",
        "\u001b[0;32m<timed exec>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n",
        "\u001b[0;32m~/code/kmer-hashing/kh-tools/khtools/partition_reads.py\u001b[0m in \u001b[0;36mscore_reads\u001b[0;34m(reads, peptide_graph, peptide_ksize, jaccard_threshold, molecule, verbose)\u001b[0m\n\u001b[1;32m    209\u001b[0m                                           \u001b[0;34m'jaccard_in_peptide_db'\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    210\u001b[0m                                           \u001b[0;34m'n_kmers'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 211\u001b[0;31m                                           'classification'])\n\u001b[0m\u001b[1;32m    212\u001b[0m     \u001b[0;32mreturn\u001b[0m \u001b[0mscoring_df\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    213\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, data, index, columns, dtype, copy)\u001b[0m\n\u001b[1;32m    433\u001b[0m                     \u001b[0;32mif\u001b[0m \u001b[0mis_named_tuple\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0mcolumns\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    434\u001b[0m                         \u001b[0mcolumns\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdata\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_fields\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 435\u001b[0;31m                     \u001b[0marrays\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcolumns\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mto_arrays\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcolumns\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdtype\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mdtype\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    436\u001b[0m                     \u001b[0mcolumns\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mensure_index\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    437\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/internals/construction.py\u001b[0m in \u001b[0;36mto_arrays\u001b[0;34m(data, columns, coerce_float, dtype)\u001b[0m\n\u001b[1;32m    402\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0misinstance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mlist\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtuple\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    403\u001b[0m         return _list_to_arrays(data, columns, coerce_float=coerce_float,\n\u001b[0;32m--> 404\u001b[0;31m                                dtype=dtype)\n\u001b[0m\u001b[1;32m    405\u001b[0m     \u001b[0;32melif\u001b[0m \u001b[0misinstance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcompat\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mMapping\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    406\u001b[0m         return _list_of_dict_to_arrays(data, columns,\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/internals/construction.py\u001b[0m in \u001b[0;36m_list_to_arrays\u001b[0;34m(data, columns, coerce_float, dtype)\u001b[0m\n\u001b[1;32m    434\u001b[0m         \u001b[0mcontent\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mlist\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlib\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mto_object_array\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mT\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    435\u001b[0m     return _convert_object_array(content, columns, dtype=dtype,\n\u001b[0;32m--> 436\u001b[0;31m                                  coerce_float=coerce_float)\n\u001b[0m\u001b[1;32m    437\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    438\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/internals/construction.py\u001b[0m in \u001b[0;36m_convert_object_array\u001b[0;34m(content, columns, coerce_float, dtype)\u001b[0m\n\u001b[1;32m    490\u001b[0m             raise AssertionError('{col:d} columns passed, passed data had '\n\u001b[1;32m    491\u001b[0m                                  '{con} columns'.format(col=len(columns),\n\u001b[0;32m--> 492\u001b[0;31m                                                         con=len(content)))\n\u001b[0m\u001b[1;32m    493\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    494\u001b[0m     \u001b[0;31m# provide soft conversion of object dtypes\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;31mAssertionError\u001b[0m: 3 columns passed, passed data had 4 columns"
       ]
      }
     ],
     "persistentId": "be06ded1-d7d4-4f21-9961-3aed5ff30cbf",
     "text": "%%time\n\nreads = 'SRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz'\nsubsampled_reads = 'SRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.fq.gz'\n\ncoding_scores = score_reads(subsampled_reads, peptide_graph, peptide_ksize)"
    },
    "executionTime": "2019-10-11T16:06:17.392Z"
   },
   {
    "cell": {
     "executionCount": 14,
     "executionEventId": "157df6e1-5f38-4dd7-94ed-0a503ca7b0b7",
     "hasError": true,
     "id": "b242d89f-8d25-4127-b875-8a85fa69d914",
     "outputs": [
      {
       "name": "stderr",
       "output_type": "stream",
       "text": "/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/screed/openscreed.py:39: ResourceWarning: unclosed file <_io.BufferedReader name='SRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.fq.gz'>\n  self.iter_fn = self.open_reader(filename, *args, **kwargs)\n1000it [00:00, 2919.20it/s]\n"
      },
      {
       "ename": "AssertionError",
       "evalue": "3 columns passed, passed data had 4 columns",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mAssertionError\u001b[0m                            Traceback (most recent call last)",
        "\u001b[0;32m<timed exec>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n",
        "\u001b[0;32m~/code/kmer-hashing/kh-tools/khtools/partition_reads.py\u001b[0m in \u001b[0;36mscore_reads\u001b[0;34m(reads, peptide_graph, peptide_ksize, jaccard_threshold, molecule, verbose)\u001b[0m\n\u001b[1;32m    209\u001b[0m                                           \u001b[0;34m'jaccard_in_peptide_db'\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    210\u001b[0m                                           \u001b[0;34m'n_kmers'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 211\u001b[0;31m                                           'classification'])\n\u001b[0m\u001b[1;32m    212\u001b[0m     \u001b[0;32mreturn\u001b[0m \u001b[0mscoring_df\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    213\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, data, index, columns, dtype, copy)\u001b[0m\n\u001b[1;32m    433\u001b[0m                     \u001b[0;32mif\u001b[0m \u001b[0mis_named_tuple\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0mcolumns\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    434\u001b[0m                         \u001b[0mcolumns\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdata\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_fields\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 435\u001b[0;31m                     \u001b[0marrays\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcolumns\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mto_arrays\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcolumns\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdtype\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mdtype\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    436\u001b[0m                     \u001b[0mcolumns\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mensure_index\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    437\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/internals/construction.py\u001b[0m in \u001b[0;36mto_arrays\u001b[0;34m(data, columns, coerce_float, dtype)\u001b[0m\n\u001b[1;32m    402\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0misinstance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mlist\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtuple\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    403\u001b[0m         return _list_to_arrays(data, columns, coerce_float=coerce_float,\n\u001b[0;32m--> 404\u001b[0;31m                                dtype=dtype)\n\u001b[0m\u001b[1;32m    405\u001b[0m     \u001b[0;32melif\u001b[0m \u001b[0misinstance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcompat\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mMapping\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    406\u001b[0m         return _list_of_dict_to_arrays(data, columns,\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/internals/construction.py\u001b[0m in \u001b[0;36m_list_to_arrays\u001b[0;34m(data, columns, coerce_float, dtype)\u001b[0m\n\u001b[1;32m    434\u001b[0m         \u001b[0mcontent\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mlist\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlib\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mto_object_array\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mT\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    435\u001b[0m     return _convert_object_array(content, columns, dtype=dtype,\n\u001b[0;32m--> 436\u001b[0;31m                                  coerce_float=coerce_float)\n\u001b[0m\u001b[1;32m    437\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    438\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/internals/construction.py\u001b[0m in \u001b[0;36m_convert_object_array\u001b[0;34m(content, columns, coerce_float, dtype)\u001b[0m\n\u001b[1;32m    490\u001b[0m             raise AssertionError('{col:d} columns passed, passed data had '\n\u001b[1;32m    491\u001b[0m                                  '{con} columns'.format(col=len(columns),\n\u001b[0;32m--> 492\u001b[0;31m                                                         con=len(content)))\n\u001b[0m\u001b[1;32m    493\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    494\u001b[0m     \u001b[0;31m# provide soft conversion of object dtypes\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;31mAssertionError\u001b[0m: 3 columns passed, passed data had 4 columns"
       ]
      }
     ],
     "persistentId": "be06ded1-d7d4-4f21-9961-3aed5ff30cbf",
     "text": "%%time\n\nreads = 'SRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz'\nsubsampled_reads = 'SRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.fq.gz'\n\ncoding_scores = score_reads(subsampled_reads, peptide_graph, peptide_ksize)"
    },
    "executionTime": "2019-10-11T16:06:21.676Z"
   },
   {
    "cell": {
     "executionCount": 15,
     "executionEventId": "6855e2cb-ad70-4332-887b-528239a3c096",
     "hasError": true,
     "id": "b242d89f-8d25-4127-b875-8a85fa69d914",
     "outputs": [
      {
       "name": "stderr",
       "output_type": "stream",
       "text": "/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/screed/openscreed.py:39: ResourceWarning: unclosed file <_io.BufferedReader name='SRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.fq.gz'>\n  self.iter_fn = self.open_reader(filename, *args, **kwargs)\n1000it [00:00, 2981.49it/s]\n"
      },
      {
       "ename": "AssertionError",
       "evalue": "3 columns passed, passed data had 4 columns",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mAssertionError\u001b[0m                            Traceback (most recent call last)",
        "\u001b[0;32m<timed exec>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n",
        "\u001b[0;32m~/code/kmer-hashing/kh-tools/khtools/partition_reads.py\u001b[0m in \u001b[0;36mscore_reads\u001b[0;34m(reads, peptide_graph, peptide_ksize, jaccard_threshold, molecule, verbose)\u001b[0m\n\u001b[1;32m    209\u001b[0m                                           \u001b[0;34m'jaccard_in_peptide_db'\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    210\u001b[0m                                           \u001b[0;34m'n_kmers'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 211\u001b[0;31m                                           'classification'])\n\u001b[0m\u001b[1;32m    212\u001b[0m     \u001b[0;32mreturn\u001b[0m \u001b[0mscoring_df\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    213\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, data, index, columns, dtype, copy)\u001b[0m\n\u001b[1;32m    433\u001b[0m                     \u001b[0;32mif\u001b[0m \u001b[0mis_named_tuple\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0mcolumns\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    434\u001b[0m                         \u001b[0mcolumns\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdata\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_fields\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 435\u001b[0;31m                     \u001b[0marrays\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcolumns\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mto_arrays\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcolumns\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdtype\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mdtype\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    436\u001b[0m                     \u001b[0mcolumns\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mensure_index\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    437\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/internals/construction.py\u001b[0m in \u001b[0;36mto_arrays\u001b[0;34m(data, columns, coerce_float, dtype)\u001b[0m\n\u001b[1;32m    402\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0misinstance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mlist\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtuple\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    403\u001b[0m         return _list_to_arrays(data, columns, coerce_float=coerce_float,\n\u001b[0;32m--> 404\u001b[0;31m                                dtype=dtype)\n\u001b[0m\u001b[1;32m    405\u001b[0m     \u001b[0;32melif\u001b[0m \u001b[0misinstance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcompat\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mMapping\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    406\u001b[0m         return _list_of_dict_to_arrays(data, columns,\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/internals/construction.py\u001b[0m in \u001b[0;36m_list_to_arrays\u001b[0;34m(data, columns, coerce_float, dtype)\u001b[0m\n\u001b[1;32m    434\u001b[0m         \u001b[0mcontent\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mlist\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlib\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mto_object_array\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mT\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    435\u001b[0m     return _convert_object_array(content, columns, dtype=dtype,\n\u001b[0;32m--> 436\u001b[0;31m                                  coerce_float=coerce_float)\n\u001b[0m\u001b[1;32m    437\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    438\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/internals/construction.py\u001b[0m in \u001b[0;36m_convert_object_array\u001b[0;34m(content, columns, coerce_float, dtype)\u001b[0m\n\u001b[1;32m    490\u001b[0m             raise AssertionError('{col:d} columns passed, passed data had '\n\u001b[1;32m    491\u001b[0m                                  '{con} columns'.format(col=len(columns),\n\u001b[0;32m--> 492\u001b[0;31m                                                         con=len(content)))\n\u001b[0m\u001b[1;32m    493\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    494\u001b[0m     \u001b[0;31m# provide soft conversion of object dtypes\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;31mAssertionError\u001b[0m: 3 columns passed, passed data had 4 columns"
       ]
      }
     ],
     "persistentId": "be06ded1-d7d4-4f21-9961-3aed5ff30cbf",
     "text": "%%time\n\nreads = 'SRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz'\nsubsampled_reads = 'SRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.fq.gz'\n\ncoding_scores = score_reads(subsampled_reads, peptide_graph, peptide_ksize)"
    },
    "executionTime": "2019-10-11T16:06:28.177Z"
   },
   {
    "cell": {
     "executionCount": 1,
     "executionEventId": "fdd6c74f-0b09-44f6-bded-52025ee7ce7d",
     "hasError": false,
     "id": "0f1ea26a-ecfa-4359-8ba4-c0f0d6feb077",
     "outputs": [],
     "persistentId": "71299be4-a00f-4638-92a2-ea174ddf208d",
     "text": "%load_ext autoreload\n%autoreload 2"
    },
    "executionTime": "2019-10-11T16:06:37.804Z"
   },
   {
    "cell": {
     "executionCount": 2,
     "executionEventId": "b0196c2d-6482-4bc9-bfc9-03dad3e127a1",
     "hasError": false,
     "id": "5e35ad66-c7bb-43f2-86d3-6da92b22e4a6",
     "outputs": [],
     "persistentId": "ecc17310-b85a-49bd-8d60-8de454040c04",
     "text": "folder = '/home/olga/data_sm/kmer-hashing/classify_coding_vs_noncoding/'"
    },
    "executionTime": "2019-10-11T16:06:37.951Z"
   },
   {
    "cell": {
     "executionCount": 3,
     "executionEventId": "9b6d2702-58b3-4498-be62-563587a39956",
     "hasError": false,
     "id": "31c3fe01-1037-44d7-b566-b186d79dff80",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "/mnt/data_sm/olga/kmer-hashing/classify_coding_vs_noncoding\n"
      }
     ],
     "persistentId": "1d547cf2-254f-4fe2-9bf0-5b7e39be4376",
     "text": "cd $folder"
    },
    "executionTime": "2019-10-11T16:06:38.109Z"
   },
   {
    "cell": {
     "executionCount": 4,
     "executionEventId": "563bd754-3da3-4877-895c-a403f4a1652c",
     "hasError": false,
     "id": "81fe1d1b-9150-4e3a-a0e2-f2470c0bb85c",
     "outputs": [],
     "persistentId": "56e48bd4-fcc6-4d93-b4e3-0984be1fc1d5",
     "text": "import gzip\nfrom io import StringIO\nfrom pprint import pprint\n\nfrom khmer import Nodegraph\nimport matplotlib.pyplot as plt\nimport screed\nfrom sourmash._minhash import hash_murmur\nfrom khmer.khmer_args import calculate_graphsize\nimport pandas as pd\nfrom sourmash.logging import notify\n\nimport seaborn as sns\n\nfrom sklearn import metrics\nfrom khtools.compare_kmer_content import kmerize, hpize, dayhoffize\n\nfrom Bio.Seq import Seq\nfrom Bio import SeqIO\n\n# make all warnings error to get a stack trace\nimport warnings\nwarnings.filterwarnings(\"default\")\n"
    },
    "executionTime": "2019-10-11T16:06:39.783Z"
   },
   {
    "cell": {
     "executionCount": 5,
     "executionEventId": "267489c7-a1bb-423b-a856-545bdf674962",
     "hasError": false,
     "id": "e50808b6-6666-4543-9299-8075c789791c",
     "outputs": [],
     "persistentId": "1e54230e-ceec-42fc-afa6-7b32e219378b",
     "text": "from khtools.partition_reads import score_reads, make_peptide_bloom_filter"
    },
    "executionTime": "2019-10-11T16:06:39.790Z"
   },
   {
    "cell": {
     "executionCount": 7,
     "executionEventId": "5341c8e8-0dd0-45be-acdd-927ac1d43c7b",
     "hasError": true,
     "id": "b242d89f-8d25-4127-b875-8a85fa69d914",
     "outputs": [
      {
       "name": "stderr",
       "output_type": "stream",
       "text": "/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/screed/openscreed.py:39: ResourceWarning: unclosed file <_io.BufferedReader name='SRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.fq.gz'>\n  self.iter_fn = self.open_reader(filename, *args, **kwargs)\n1000it [00:00, 2974.27it/s]\n"
      },
      {
       "ename": "AssertionError",
       "evalue": "3 columns passed, passed data had 4 columns",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mAssertionError\u001b[0m                            Traceback (most recent call last)",
        "\u001b[0;32m<timed exec>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n",
        "\u001b[0;32m~/code/kmer-hashing/kh-tools/khtools/partition_reads.py\u001b[0m in \u001b[0;36mscore_reads\u001b[0;34m(reads, peptide_graph, peptide_ksize, jaccard_threshold, molecule, verbose)\u001b[0m\n\u001b[1;32m    209\u001b[0m                                           \u001b[0;34m'jaccard_in_peptide_db'\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    210\u001b[0m                                           \u001b[0;34m'n_kmers'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 211\u001b[0;31m                                           'classification'])\n\u001b[0m\u001b[1;32m    212\u001b[0m     \u001b[0;32mreturn\u001b[0m \u001b[0mscoring_df\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    213\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, data, index, columns, dtype, copy)\u001b[0m\n\u001b[1;32m    433\u001b[0m                     \u001b[0;32mif\u001b[0m \u001b[0mis_named_tuple\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0mcolumns\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    434\u001b[0m                         \u001b[0mcolumns\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdata\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_fields\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 435\u001b[0;31m                     \u001b[0marrays\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcolumns\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mto_arrays\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcolumns\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdtype\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mdtype\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    436\u001b[0m                     \u001b[0mcolumns\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mensure_index\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    437\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/internals/construction.py\u001b[0m in \u001b[0;36mto_arrays\u001b[0;34m(data, columns, coerce_float, dtype)\u001b[0m\n\u001b[1;32m    402\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0misinstance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mlist\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtuple\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    403\u001b[0m         return _list_to_arrays(data, columns, coerce_float=coerce_float,\n\u001b[0;32m--> 404\u001b[0;31m                                dtype=dtype)\n\u001b[0m\u001b[1;32m    405\u001b[0m     \u001b[0;32melif\u001b[0m \u001b[0misinstance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcompat\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mMapping\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    406\u001b[0m         return _list_of_dict_to_arrays(data, columns,\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/internals/construction.py\u001b[0m in \u001b[0;36m_list_to_arrays\u001b[0;34m(data, columns, coerce_float, dtype)\u001b[0m\n\u001b[1;32m    434\u001b[0m         \u001b[0mcontent\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mlist\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlib\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mto_object_array\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mT\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    435\u001b[0m     return _convert_object_array(content, columns, dtype=dtype,\n\u001b[0;32m--> 436\u001b[0;31m                                  coerce_float=coerce_float)\n\u001b[0m\u001b[1;32m    437\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    438\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/internals/construction.py\u001b[0m in \u001b[0;36m_convert_object_array\u001b[0;34m(content, columns, coerce_float, dtype)\u001b[0m\n\u001b[1;32m    490\u001b[0m             raise AssertionError('{col:d} columns passed, passed data had '\n\u001b[1;32m    491\u001b[0m                                  '{con} columns'.format(col=len(columns),\n\u001b[0;32m--> 492\u001b[0;31m                                                         con=len(content)))\n\u001b[0m\u001b[1;32m    493\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    494\u001b[0m     \u001b[0;31m# provide soft conversion of object dtypes\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;31mAssertionError\u001b[0m: 3 columns passed, passed data had 4 columns"
       ]
      }
     ],
     "persistentId": "be06ded1-d7d4-4f21-9961-3aed5ff30cbf",
     "text": "%%time\n\nreads = 'SRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz'\nsubsampled_reads = 'SRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.fq.gz'\n\ncoding_scores = score_reads(subsampled_reads, peptide_graph, peptide_ksize)"
    },
    "executionTime": "2019-10-11T16:06:45.257Z"
   },
   {
    "cell": {
     "executionCount": 8,
     "executionEventId": "434ee348-89e9-4600-bcf0-d3297829f9bf",
     "hasError": false,
     "id": "b242d89f-8d25-4127-b875-8a85fa69d914",
     "outputs": [
      {
       "name": "stderr",
       "output_type": "stream",
       "text": "/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/screed/openscreed.py:39: ResourceWarning: unclosed file <_io.BufferedReader name='SRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.fq.gz'>\n  self.iter_fn = self.open_reader(filename, *args, **kwargs)\n1000it [00:00, 2943.78it/s]"
      },
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "CPU times: user 352 ms, sys: 0 ns, total: 352 ms\nWall time: 348 ms\n"
      },
      {
       "name": "stderr",
       "output_type": "stream",
       "text": "\n"
      }
     ],
     "persistentId": "be06ded1-d7d4-4f21-9961-3aed5ff30cbf",
     "text": "%%time\n\nreads = 'SRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz'\nsubsampled_reads = 'SRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.fq.gz'\n\ncoding_scores = score_reads(subsampled_reads, peptide_graph, peptide_ksize)"
    },
    "executionTime": "2019-10-11T16:07:06.578Z"
   },
   {
    "cell": {
     "executionCount": 9,
     "executionEventId": "25bb963a-52d7-42f8-83df-1d4c2eda8415",
     "hasError": false,
     "id": "3281c496-16a4-496e-81d4-fa57e8d1a556",
     "outputs": [
      {
       "data": {
        "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>read_id</th>\n      <th>jaccard_in_peptide_db</th>\n      <th>n_kmers</th>\n      <th>classification</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>SRR306838.10559374 Ibis_Run100924_C3PO:6:51:17...</td>\n      <td>1.0</td>\n      <td>1</td>\n      <td>coding</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>SRR306838.6196593 Ibis_Run100924_C3PO:6:29:167...</td>\n      <td>1.0</td>\n      <td>1</td>\n      <td>coding</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>SRR306838.20767303 Ibis_Run100924_C3PO:6:104:6...</td>\n      <td>1.0</td>\n      <td>1</td>\n      <td>coding</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>SRR306838.12582274 Ibis_Run100924_C3PO:6:62:11...</td>\n      <td>1.0</td>\n      <td>1</td>\n      <td>coding</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>SRR306838.13334230 Ibis_Run100924_C3PO:6:66:16...</td>\n      <td>1.0</td>\n      <td>1</td>\n      <td>coding</td>\n    </tr>\n  </tbody>\n</table>\n</div>",
        "text/plain": "                                             read_id  jaccard_in_peptide_db  \\\n0  SRR306838.10559374 Ibis_Run100924_C3PO:6:51:17...                    1.0   \n1  SRR306838.6196593 Ibis_Run100924_C3PO:6:29:167...                    1.0   \n2  SRR306838.20767303 Ibis_Run100924_C3PO:6:104:6...                    1.0   \n3  SRR306838.12582274 Ibis_Run100924_C3PO:6:62:11...                    1.0   \n4  SRR306838.13334230 Ibis_Run100924_C3PO:6:66:16...                    1.0   \n\n   n_kmers classification  \n0        1         coding  \n1        1         coding  \n2        1         coding  \n3        1         coding  \n4        1         coding  "
       },
       "execution_count": 9,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "ede6e1ba-f524-4ceb-b35b-6c20f91d437c",
     "text": "coding_scores.head()"
    },
    "executionTime": "2019-10-11T16:07:15.070Z"
   },
   {
    "cell": {
     "executionCount": 10,
     "executionEventId": "e1fd0c7d-79fa-460d-8307-1e1040658597",
     "hasError": false,
     "id": "cfea0c3d-642a-421a-a4ad-07fa69bea946",
     "outputs": [
      {
       "data": {
        "text/plain": "(1000, 4)"
       },
       "execution_count": 10,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "09393220-d1f9-4ddd-9c7c-bc15160f3680",
     "text": "coding_scores.shape"
    },
    "executionTime": "2019-10-11T16:07:17.075Z"
   },
   {
    "cell": {
     "executionCount": 11,
     "executionEventId": "9e6131d6-8c56-40ea-a5f5-8192f34251c7",
     "hasError": false,
     "id": "1f72a44d-8a09-4821-a744-b78674b825a6",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "total 358M\ndrwxrwxr-x 2 olga olga 4.0K Sep 18 10:15 .\ndrwxrwxr-x 3 olga olga 4.0K Sep 10 15:07 ..\n-rw-rw-r-- 1 olga olga  52M Sep 18 10:05 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.bam\n-rw-rw-r-- 1 olga olga 2.9M Sep 18 10:12 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.bam.bai\n-rw-rw-r-- 1 olga olga 9.3M Sep 18 10:12 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.out_CDS_without_stop_codon_or_utr.bam\n-rw-rw-r-- 1 olga olga 1.9M Sep 18 10:12 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.out_CDS_without_stop_codon_or_utr.bam.bai\n-rw-rw-r-- 1 olga olga 2.8M Sep 18 10:14 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.out_CDS_without_stop_codon_or_utr.read_ids.txt\n-rw-rw-r-- 1 olga olga  15M Sep 18 10:15 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.read_ids.txt\n-rw-rw-r-- 1 olga olga  15M Sep 10 15:55 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out_CDS_without_stop_codon.bam\n-rw-rw-r-- 1 olga olga 1.7M Sep 18 07:05 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out_CDS_without_stop_codon.bam.bai\n-rw-rw-r-- 1 olga olga  15M Sep 18 07:15 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out_CDS_without_stop_codon_or_utr.bam\n-rw-rw-r-- 1 olga olga 1.7M Sep 18 07:06 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out_CDS_without_stop_codon_or_utr.bam.bai\n-rw-rw-r-- 1 olga olga 2.8M Sep 18 10:14 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out_CDS_without_stop_codon_or_utr.read_ids.txt\n-rw-rw-r-- 1 olga olga 4.1M Sep 10 15:56 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out_CDS_without_stop_codon.read_ids.txt\n-rw-r--r-- 1 olga olga 235M Sep 10 15:10 SRR306838_GSM752691_hsa_br_F_1_trimmed_k15.tsv\n(151869,)\n(807698,)\n"
      }
     ],
     "persistentId": "75c01782-427c-4c63-bc29-ed308ae4a142",
     "text": "folder = '/home/olga/data_sm/kmer-hashing/brawand2011/human/'\n! ls -lha $folder\n\ncoding_read_ids = pd.read_csv(f\"{folder}/SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.out_CDS_without_stop_codon_or_utr.read_ids.txt\", \n                                   header=None, squeeze=True)\nprint(coding_read_ids.shape)\ncoding_read_ids.head()\n\ntrue_coding_read_ids = set(coding_read_ids.values)\nlen(true_coding_read_ids)\n\nunspliced_read_ids_series = pd.read_csv(f\"{folder}/SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.read_ids.txt\", \n                                   header=None, squeeze=True)\nprint(unspliced_read_ids_series.shape)\nunspliced_read_ids_series.head()\n\nunspliced_reads = set(unspliced_read_ids_series.values)"
    },
    "executionTime": "2019-10-11T16:07:31.424Z"
   },
   {
    "cell": {
     "executionCount": 12,
     "executionEventId": "81ff1c4d-f2fa-4eab-88f9-777b97a435cc",
     "hasError": false,
     "id": "15642de9-ed8a-4308-ac2b-2b166277d6b0",
     "outputs": [],
     "persistentId": "f1d6ee85-0548-4e2f-bdc2-2e0cbccbf430",
     "text": "coding_scores['in_cds_no_stop_codon'] = coding_scores.read_id.isin(true_coding_read_ids)"
    },
    "executionTime": "2019-10-11T16:08:22.501Z"
   },
   {
    "cell": {
     "executionCount": 13,
     "executionEventId": "806e676b-2166-4008-a96c-c4d727cde4d7",
     "hasError": false,
     "id": "ff4aac65-c57f-4622-a574-e7fde5a41cea",
     "outputs": [
      {
       "data": {
        "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>read_id</th>\n      <th>jaccard_in_peptide_db</th>\n      <th>n_kmers</th>\n      <th>classification</th>\n      <th>in_cds_no_stop_codon</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>SRR306838.10559374 Ibis_Run100924_C3PO:6:51:17...</td>\n      <td>1.0</td>\n      <td>1</td>\n      <td>coding</td>\n      <td>False</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>SRR306838.6196593 Ibis_Run100924_C3PO:6:29:167...</td>\n      <td>1.0</td>\n      <td>1</td>\n      <td>coding</td>\n      <td>False</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>SRR306838.20767303 Ibis_Run100924_C3PO:6:104:6...</td>\n      <td>1.0</td>\n      <td>1</td>\n      <td>coding</td>\n      <td>False</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>SRR306838.12582274 Ibis_Run100924_C3PO:6:62:11...</td>\n      <td>1.0</td>\n      <td>1</td>\n      <td>coding</td>\n      <td>False</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>SRR306838.13334230 Ibis_Run100924_C3PO:6:66:16...</td>\n      <td>1.0</td>\n      <td>1</td>\n      <td>coding</td>\n      <td>False</td>\n    </tr>\n  </tbody>\n</table>\n</div>",
        "text/plain": "                                             read_id  jaccard_in_peptide_db  \\\n0  SRR306838.10559374 Ibis_Run100924_C3PO:6:51:17...                    1.0   \n1  SRR306838.6196593 Ibis_Run100924_C3PO:6:29:167...                    1.0   \n2  SRR306838.20767303 Ibis_Run100924_C3PO:6:104:6...                    1.0   \n3  SRR306838.12582274 Ibis_Run100924_C3PO:6:62:11...                    1.0   \n4  SRR306838.13334230 Ibis_Run100924_C3PO:6:66:16...                    1.0   \n\n   n_kmers classification  in_cds_no_stop_codon  \n0        1         coding                 False  \n1        1         coding                 False  \n2        1         coding                 False  \n3        1         coding                 False  \n4        1         coding                 False  "
       },
       "execution_count": 13,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "3c8608eb-5dc0-47f2-b161-2a608e215755",
     "text": "coding_scores.head()"
    },
    "executionTime": "2019-10-11T16:08:26.025Z"
   },
   {
    "cell": {
     "executionCount": 14,
     "executionEventId": "f2794748-8146-480a-b8f0-1ff743834972",
     "hasError": false,
     "id": "29321b62-9931-4092-891b-7d4dca34c4da",
     "outputs": [],
     "persistentId": "b321c9fc-54af-40bc-96a9-40c59ea87994",
     "text": "coding_scores['short_read_id'] = coding_scores['read_id'].str.split().str[0]"
    },
    "executionTime": "2019-10-11T16:08:52.404Z"
   },
   {
    "cell": {
     "executionCount": 17,
     "executionEventId": "65618971-d30f-4f42-b44b-6cee0a0970ba",
     "hasError": false,
     "id": "4c484434-ea07-4b72-8073-567bf902bedb",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "(807698,)\n"
      }
     ],
     "persistentId": "75c01782-427c-4c63-bc29-ed308ae4a142",
     "text": "true_coding_read_ids = set(coding_read_ids.values)\nlen(true_coding_read_ids)\n\nunspliced_read_ids_series = pd.read_csv(f\"{folder}/SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.read_ids.txt\", \n                                   header=None, squeeze=True)\nprint(unspliced_read_ids_series.shape)\nunspliced_read_ids_series.head()\n\nunspliced_reads = set(unspliced_read_ids_series.values)"
    },
    "executionTime": "2019-10-11T16:09:10.088Z"
   },
   {
    "cell": {
     "executionCount": 19,
     "executionEventId": "8e82743b-7c28-4eab-bdf4-cf76374e83ab",
     "hasError": false,
     "id": "4c484434-ea07-4b72-8073-567bf902bedb",
     "outputs": [
      {
       "data": {
        "text/plain": "150289"
       },
       "execution_count": 19,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "75c01782-427c-4c63-bc29-ed308ae4a142",
     "text": "true_coding_read_ids = set(coding_read_ids.values)\nlen(true_coding_read_ids)"
    },
    "executionTime": "2019-10-11T16:09:29.011Z"
   },
   {
    "cell": {
     "executionCount": 20,
     "executionEventId": "b3a51201-a02c-4ea9-ac56-8dca01504804",
     "hasError": false,
     "id": "81fe1d1b-9150-4e3a-a0e2-f2470c0bb85c",
     "outputs": [],
     "persistentId": "56e48bd4-fcc6-4d93-b4e3-0984be1fc1d5",
     "text": "import gzip\nfrom io import StringIO\nfrom pprint import pprint\n\nfrom khmer import Nodegraph\nimport matplotlib.pyplot as plt\nimport screed\nfrom sourmash._minhash import hash_murmur\nfrom khmer.khmer_args import calculate_graphsize\nimport pandas as pd\nfrom sourmash.logging import notify\n\nimport seaborn as sns\n\nfrom sklearn import metrics\nfrom khtools.compare_kmer_content import kmerize, hpize, dayhoffize\n\nfrom Bio.Seq import Seq\nfrom Bio import SeqIO\n\n# make all warnings error to get a stack trace\nimport warnings\nwarnings.filterwarnings(\"default\")\n\npd.options.display.max_colwidth = 100"
    },
    "executionTime": "2019-10-11T16:09:57.728Z"
   },
   {
    "cell": {
     "executionCount": 22,
     "executionEventId": "cbbfe960-18da-4d43-b895-225060159f07",
     "hasError": false,
     "id": "832f29e2-c227-473d-9680-65ef79329349",
     "outputs": [],
     "persistentId": "e4c114ee-5198-42ee-97bb-d9974f296764",
     "text": "bam = 'SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out_CDS_without_stop_codon_or_utr.bam'"
    },
    "executionTime": "2019-10-11T16:10:30.109Z"
   },
   {
    "cell": {
     "executionCount": 23,
     "executionEventId": "50c9220f-8330-4329-860a-1510cf8f029f",
     "hasError": true,
     "id": "553028fa-f7ef-4001-8042-6657c6c34d0c",
     "outputs": [
      {
       "ename": "SyntaxError",
       "evalue": "EOL while scanning string literal (<ipython-input-23-a466b342e1b9>, line 2)",
       "output_type": "error",
       "traceback": [
        "\u001b[0;36m  File \u001b[0;32m\"<ipython-input-23-a466b342e1b9>\"\u001b[0;36m, line \u001b[0;32m2\u001b[0m\n\u001b[0;31m    '\u001b[0m\n\u001b[0m     ^\u001b[0m\n\u001b[0;31mSyntaxError\u001b[0m\u001b[0;31m:\u001b[0m EOL while scanning string literal\n"
       ]
      }
     ],
     "persistentId": "40aafb32-c709-4cab-9f22-be2225b0e525",
     "text": "! samtools view $folder/$bam | grep 'SRR306838.10559374\n'"
    },
    "executionTime": "2019-10-11T16:10:39.104Z"
   },
   {
    "cell": {
     "executionCount": 24,
     "executionEventId": "ac46a483-e04e-43d9-a033-0506a92bc3fe",
     "hasError": false,
     "id": "553028fa-f7ef-4001-8042-6657c6c34d0c",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "SRR306838.10559374\t16\tchr9\t5054875\t255\t10M784N56M1S\t*\t0\t0\tGACAGAACAGGATTTACAGTTATATTGCGATTTTCCTAATATTATTGATGTCAGTATTAAGCAAGCG\t835;:5459394634568734466464657444677535764562576775557332565142/223\tNH:i:1\tHI:i:1\tAS:i:66\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\n"
      }
     ],
     "persistentId": "40aafb32-c709-4cab-9f22-be2225b0e525",
     "text": "! samtools view $folder/$bam | grep 'SRR306838.10559374'"
    },
    "executionTime": "2019-10-11T16:10:42.331Z"
   },
   {
    "cell": {
     "executionCount": 25,
     "executionEventId": "9d653f0e-361b-415a-9c42-a9d35e1f1359",
     "hasError": false,
     "id": "9babb580-ed7e-4a80-bcb3-3d080fca4a29",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "(807698,)\n"
      }
     ],
     "persistentId": "317bb484-c93f-4e1c-b831-50734c9e6614",
     "text": "unspliced_read_ids_series = pd.read_csv(f\"{folder}/SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.read_ids.txt\", \n                                   header=None, squeeze=True)\nprint(unspliced_read_ids_series.shape)\nunspliced_read_ids_series.head()\n\nunspliced_reads = set(unspliced_read_ids_series.values)"
    },
    "executionTime": "2019-10-11T16:13:09.494Z"
   },
   {
    "cell": {
     "executionCount": 30,
     "executionEventId": "32e7173a-9b75-401e-ad25-5136668b4e16",
     "hasError": false,
     "id": "553028fa-f7ef-4001-8042-6657c6c34d0c",
     "outputs": [],
     "persistentId": "40aafb32-c709-4cab-9f22-be2225b0e525",
     "text": "! samtools view $folder/$bam | grep 'SRR306838.6196593'"
    },
    "executionTime": "2019-10-11T16:14:07.690Z"
   },
   {
    "cell": {
     "executionCount": 31,
     "executionEventId": "68b94524-193d-4c41-9165-47ea96de7ca3",
     "hasError": false,
     "id": "342df663-948c-4de9-b395-be7b619e90de",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "\u001b[0m\u001b[01;31mHomo_sapiens.GRCh38.cds.all.fa.gz\u001b[0m\n\u001b[01;31mHomo_sapiens.GRCh38.ncrna.fa.gz\u001b[0m\n\u001b[01;31mHomo_sapiens.GRCh38.pep.all.fa.gz\u001b[0m\nhuman_cds_k15.nodegraph\nhuman_cds_k15.nodegraph.info\nhuman_cds_k17.nodegraph\nhuman_cds_k17.nodegraph.info\nhuman_cds_k19.nodegraph\nhuman_cds_k19.nodegraph.info\nhuman_cds_k21.nodegraph\nhuman_cds_k21.nodegraph.info\nhuman_cds_k23.nodegraph\nhuman_cds_k23.nodegraph.info\nhuman_cds_k25.nodegraph\nhuman_cds_k25.nodegraph.info\nhuman_cds_k27.nodegraph\nhuman_cds_k27.nodegraph.info\nhuman_cds_k31.nodegraph\nhuman_cds_k31.nodegraph.info\nhuman_ncrna_k15.nodegraph\nhuman_ncrna_k15.nodegraph.info\nhuman_ncrna_k15.nodegraph.tagset\nhuman_ncrna_k17.nodegraph\nhuman_ncrna_k17.nodegraph.info\nhuman_ncrna_k19.nodegraph\nhuman_ncrna_k19.nodegraph.info\nhuman_ncrna_k21.nodegraph\nhuman_ncrna_k21.nodegraph.info\nhuman_ncrna_k23.nodegraph\nhuman_ncrna_k23.nodegraph.info\nhuman_ncrna_k25.nodegraph\nhuman_ncrna_k25.nodegraph.info\nhuman_ncrna_k27.nodegraph\nhuman_ncrna_k27.nodegraph.info\nhuman_ncrna_k31.nodegraph\nhuman_ncrna_k31.nodegraph.info\n\u001b[01;31mSRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz\u001b[0m\n\u001b[01;31mSRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.fq.gz\u001b[0m\n"
      }
     ],
     "persistentId": "a30b0ea1-cbfb-4de5-bce5-329cfeb8e50c",
     "text": "ls"
    },
    "executionTime": "2019-10-11T16:14:29.458Z"
   },
   {
    "cell": {
     "executionCount": 33,
     "executionEventId": "d0c2f642-e307-43a5-9dc3-ce8e7f3d4489",
     "hasError": true,
     "id": "14d3072d-40ce-4d7b-97e5-0cf1e1ad909f",
     "outputs": [
      {
       "name": "stderr",
       "output_type": "stream",
       "text": "[autoreload of khtools.partition_reads failed: Traceback (most recent call last):\n  File \"/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/IPython/extensions/autoreload.py\", line 245, in check\n    superreload(m, reload, self.old_objects)\n  File \"/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/IPython/extensions/autoreload.py\", line 368, in superreload\n    module = reload(module)\n  File \"/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/imp.py\", line 315, in reload\n    return importlib.reload(module)\n  File \"/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/importlib/__init__.py\", line 166, in reload\n    _bootstrap._exec(spec, module)\n  File \"<frozen importlib._bootstrap>\", line 618, in _exec\n  File \"<frozen importlib._bootstrap_external>\", line 674, in exec_module\n  File \"<frozen importlib._bootstrap_external>\", line 781, in get_code\n  File \"<frozen importlib._bootstrap_external>\", line 741, in source_to_code\n  File \"<frozen importlib._bootstrap>\", line 219, in _call_with_frames_removed\n  File \"/home/olga/code/kmer-hashing/kh-tools/khtools/partition_reads.py\", line 107\n    \"\\t{translation}\")\n                    ^\nSyntaxError: invalid syntax\n]\n"
      },
      {
       "ename": "ImportError",
       "evalue": "cannot import name 'score_single_sequence'",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mImportError\u001b[0m                               Traceback (most recent call last)",
        "\u001b[0;32m<ipython-input-33-8580f8559cb4>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0;32mfrom\u001b[0m \u001b[0mkhtools\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpartition_reads\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0mscore_single_sequence\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
        "\u001b[0;31mImportError\u001b[0m: cannot import name 'score_single_sequence'"
       ]
      }
     ],
     "persistentId": "a31a1fe9-c86c-4d99-b575-14f954026bc9",
     "text": "from khtools.partition_reads import score_single_sequence"
    },
    "executionTime": "2019-10-11T16:35:18.590Z"
   },
   {
    "cell": {
     "executionCount": 34,
     "executionEventId": "60a404cc-6533-4522-888d-efd179f975be",
     "hasError": false,
     "id": "14d3072d-40ce-4d7b-97e5-0cf1e1ad909f",
     "outputs": [],
     "persistentId": "a31a1fe9-c86c-4d99-b575-14f954026bc9",
     "text": "from khtools.partition_reads import score_single_sequence"
    },
    "executionTime": "2019-10-11T16:36:06.201Z"
   },
   {
    "cell": {
     "executionCount": 35,
     "executionEventId": "db558b0e-b002-4a06-916a-eb58f778e114",
     "hasError": true,
     "id": "37605b63-f85f-4dba-8bcc-fb65726ba4c4",
     "outputs": [
      {
       "ename": "TypeError",
       "evalue": "score_single_sequence() missing 4 required positional arguments: 'peptide_graph', 'peptide_ksize', 'molecule', and 'verbose'",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
        "\u001b[0;32m<ipython-input-35-ba304d13cfa6>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mscore_single_sequence\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"TCTAGAATGTGAAATAACGTACTTCATGTGTCTTCTTACCAAAAATACCAACGATAAGGGGAAAAGCCATC\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
        "\u001b[0;31mTypeError\u001b[0m: score_single_sequence() missing 4 required positional arguments: 'peptide_graph', 'peptide_ksize', 'molecule', and 'verbose'"
       ]
      }
     ],
     "persistentId": "943d8d95-2023-48f9-acad-1136aad810d7",
     "text": "score_single_sequence(\"TCTAGAATGTGAAATAACGTACTTCATGTGTCTTCTTACCAAAAATACCAACGATAAGGGGAAAAGCCATC\")"
    },
    "executionTime": "2019-10-11T16:36:12.085Z"
   },
   {
    "cell": {
     "executionCount": 36,
     "executionEventId": "15afcbb9-177c-4e9b-88b5-ac1df42c198f",
     "hasError": true,
     "id": "37605b63-f85f-4dba-8bcc-fb65726ba4c4",
     "outputs": [
      {
       "ename": "NameError",
       "evalue": "name 'peptid_ksize' is not defined",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
        "\u001b[0;32m<ipython-input-36-6cf086b3dac3>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mscore_single_sequence\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"TCTAGAATGTGAAATAACGTACTTCATGTGTCTTCTTACCAAAAATACCAACGATAAGGGGAAAAGCCATC\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpeptide_graph\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpeptid_ksize\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'protein'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mverbose\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
        "\u001b[0;31mNameError\u001b[0m: name 'peptid_ksize' is not defined"
       ]
      }
     ],
     "persistentId": "943d8d95-2023-48f9-acad-1136aad810d7",
     "text": "score_single_sequence(\"TCTAGAATGTGAAATAACGTACTTCATGTGTCTTCTTACCAAAAATACCAACGATAAGGGGAAAAGCCATC\", peptide_graph, peptid_ksize, 'protein', verbose=True)"
    },
    "executionTime": "2019-10-11T16:36:28.627Z"
   },
   {
    "cell": {
     "executionCount": 37,
     "executionEventId": "1184ed92-ded0-4fd8-9abc-b1f76c6a875e",
     "hasError": true,
     "id": "37605b63-f85f-4dba-8bcc-fb65726ba4c4",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "translation: \tprotein\n"
      },
      {
       "ename": "NameError",
       "evalue": "name 'fraction_in_peptide_db' is not defined",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
        "\u001b[0;32m<ipython-input-37-c59b9893ad0b>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mscore_single_sequence\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"TCTAGAATGTGAAATAACGTACTTCATGTGTCTTCTTACCAAAAATACCAACGATAAGGGGAAAAGCCATC\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpeptide_graph\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpeptide_ksize\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'protein'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mverbose\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
        "\u001b[0;32m~/code/kmer-hashing/kh-tools/khtools/partition_reads.py\u001b[0m in \u001b[0;36mscore_single_sequence\u001b[0;34m(sequence, peptide_graph, peptide_ksize, molecule, verbose)\u001b[0m\n\u001b[1;32m    158\u001b[0m                                                   verbose)\n\u001b[1;32m    159\u001b[0m         \u001b[0;31m# Update n_kmers if this is the best translation frame\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 160\u001b[0;31m         \u001b[0;32mif\u001b[0m \u001b[0mmax_fraction_in_peptide_db\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0mfraction_in_peptide_db\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    161\u001b[0m             \u001b[0mmax_n_kmers\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mn_kmers\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    162\u001b[0m             \u001b[0mmax_kmers_in_peptide_db\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mkmers_in_peptide_db\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;31mNameError\u001b[0m: name 'fraction_in_peptide_db' is not defined"
       ]
      }
     ],
     "persistentId": "943d8d95-2023-48f9-acad-1136aad810d7",
     "text": "score_single_sequence(\"TCTAGAATGTGAAATAACGTACTTCATGTGTCTTCTTACCAAAAATACCAACGATAAGGGGAAAAGCCATC\", peptide_graph, peptide_ksize, 'protein', verbose=True)"
    },
    "executionTime": "2019-10-11T16:36:31.020Z"
   },
   {
    "cell": {
     "executionCount": 41,
     "executionEventId": "755f7e23-2909-4c96-a751-448b437e9d02",
     "hasError": true,
     "id": "81fe1d1b-9150-4e3a-a0e2-f2470c0bb85c",
     "outputs": [
      {
       "ename": "ImportError",
       "evalue": "cannot import name 'six_frame_translation_no_stops'",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mImportError\u001b[0m                               Traceback (most recent call last)",
        "\u001b[0;32m<ipython-input-41-6e4adad1826c>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m     14\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     15\u001b[0m \u001b[0;32mfrom\u001b[0m \u001b[0msklearn\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0mmetrics\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 16\u001b[0;31m \u001b[0;32mfrom\u001b[0m \u001b[0mkhtools\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcompare_kmer_content\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0mkmerize\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mhpize\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdayhoffize\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msix_frame_translation_no_stops\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     17\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     18\u001b[0m \u001b[0;32mfrom\u001b[0m \u001b[0mBio\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mSeq\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0mSeq\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;31mImportError\u001b[0m: cannot import name 'six_frame_translation_no_stops'"
       ]
      }
     ],
     "persistentId": "56e48bd4-fcc6-4d93-b4e3-0984be1fc1d5",
     "text": "import gzip\nfrom io import StringIO\nfrom pprint import pprint\n\nfrom khmer import Nodegraph\nimport matplotlib.pyplot as plt\nimport screed\nfrom sourmash._minhash import hash_murmur\nfrom khmer.khmer_args import calculate_graphsize\nimport pandas as pd\nfrom sourmash.logging import notify\n\nimport seaborn as sns\n\nfrom sklearn import metrics\nfrom khtools.compare_kmer_content import kmerize, hpize, dayhoffize, six_frame_translation_no_stops\n\nfrom Bio.Seq import Seq\nfrom Bio import SeqIO\n\n# make all warnings error to get a stack trace\nimport warnings\nwarnings.filterwarnings(\"default\")\n\npd.options.display.max_colwidth = 100"
    },
    "executionTime": "2019-10-11T16:43:24.509Z"
   },
   {
    "cell": {
     "executionCount": 42,
     "executionEventId": "d5a9b01c-ec17-430a-90e7-d6279d64aef7",
     "hasError": false,
     "id": "81fe1d1b-9150-4e3a-a0e2-f2470c0bb85c",
     "outputs": [],
     "persistentId": "56e48bd4-fcc6-4d93-b4e3-0984be1fc1d5",
     "text": "import gzip\nfrom io import StringIO\nfrom pprint import pprint\n\nfrom khmer import Nodegraph\nimport matplotlib.pyplot as plt\nimport screed\nfrom sourmash._minhash import hash_murmur\nfrom khmer.khmer_args import calculate_graphsize\nimport pandas as pd\nfrom sourmash.logging import notify\n\nimport seaborn as sns\n\nfrom sklearn import metrics\nfrom khtools.compare_kmer_content import kmerize, hpize, dayhoffize\n\nfrom Bio.Seq import Seq\nfrom Bio import SeqIO\n\n# make all warnings error to get a stack trace\nimport warnings\nwarnings.filterwarnings(\"default\")\n\npd.options.display.max_colwidth = 100"
    },
    "executionTime": "2019-10-11T16:43:36.848Z"
   },
   {
    "cell": {
     "executionCount": 43,
     "executionEventId": "817aa788-e6ca-4b94-8431-b565bfce038d",
     "hasError": false,
     "id": "e50808b6-6666-4543-9299-8075c789791c",
     "outputs": [],
     "persistentId": "1e54230e-ceec-42fc-afa6-7b32e219378b",
     "text": "from khtools.partition_reads import score_reads, make_peptide_bloom_filter, six_frame_translation_no_stops"
    },
    "executionTime": "2019-10-11T16:43:36.948Z"
   },
   {
    "cell": {
     "executionCount": 45,
     "executionEventId": "5915cc76-d4da-44ce-845a-587acf36ede5",
     "hasError": true,
     "id": "c2bb43e2-8438-4a2b-8073-bada1130618d",
     "outputs": [
      {
       "ename": "TypeError",
       "evalue": "translate() takes exactly one argument (0 given)",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
        "\u001b[0;32m<ipython-input-45-320968ea1ab3>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0msix_frame_translation_no_stops\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mseq\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
        "\u001b[0;32m~/code/kmer-hashing/kh-tools/khtools/partition_reads.py\u001b[0m in \u001b[0;36msix_frame_translation_no_stops\u001b[0;34m(seq, debug)\u001b[0m\n\u001b[1;32m     92\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     93\u001b[0m \u001b[0;32mdef\u001b[0m \u001b[0msix_frame_translation_no_stops\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mseq\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdebug\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 94\u001b[0;31m     \u001b[0mforward_translations\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mthree_frame_translation_no_stops\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mseq\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdebug\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     95\u001b[0m     reverse_translations = three_frame_translation_no_stops(\n\u001b[1;32m     96\u001b[0m         seq.reverse_complement(), debug)\n",
        "\u001b[0;32m~/code/kmer-hashing/kh-tools/khtools/partition_reads.py\u001b[0m in \u001b[0;36mthree_frame_translation_no_stops\u001b[0;34m(seq, debug)\u001b[0m\n\u001b[1;32m     87\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     88\u001b[0m \u001b[0;32mdef\u001b[0m \u001b[0mthree_frame_translation_no_stops\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mseq\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdebug\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 89\u001b[0;31m     return [t for t in three_frame_translation(seq, debug)\n\u001b[0m\u001b[1;32m     90\u001b[0m             if '*' not in t]\n\u001b[1;32m     91\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/code/kmer-hashing/kh-tools/khtools/partition_reads.py\u001b[0m in \u001b[0;36m<listcomp>\u001b[0;34m(.0)\u001b[0m\n\u001b[1;32m     87\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     88\u001b[0m \u001b[0;32mdef\u001b[0m \u001b[0mthree_frame_translation_no_stops\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mseq\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdebug\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 89\u001b[0;31m     return [t for t in three_frame_translation(seq, debug)\n\u001b[0m\u001b[1;32m     90\u001b[0m             if '*' not in t]\n\u001b[1;32m     91\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/code/kmer-hashing/kh-tools/khtools/partition_reads.py\u001b[0m in \u001b[0;36mthree_frame_translation\u001b[0;34m(seq, debug)\u001b[0m\n\u001b[1;32m     82\u001b[0m         \u001b[0mwarnings\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msimplefilter\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mwarning_filter\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     83\u001b[0m         \u001b[0;32mfor\u001b[0m \u001b[0mframe\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mrange\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;36m3\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 84\u001b[0;31m             \u001b[0mtranslation\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mseq\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mframe\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtranslate\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     85\u001b[0m             \u001b[0;32myield\u001b[0m \u001b[0mtranslation\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     86\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;31mTypeError\u001b[0m: translate() takes exactly one argument (0 given)"
       ]
      }
     ],
     "persistentId": "0949482e-76ad-4702-9d85-71f276cacf59",
     "text": "six_frame_translation_no_stops(seq)"
    },
    "executionTime": "2019-10-11T16:43:54.641Z"
   },
   {
    "cell": {
     "executionCount": 47,
     "executionEventId": "9aae5cdd-9d41-4da2-95ff-9f88d8a8a418",
     "hasError": false,
     "id": "c2bb43e2-8438-4a2b-8073-bada1130618d",
     "outputs": [
      {
       "data": {
        "text/plain": "[Seq('LECEITYFMCLLTKNTNDKGKSH', ExtendedIUPACProtein()),\n Seq('WLFPLSLVFLVRRHMKYVISHSR', ExtendedIUPACProtein())]"
       },
       "execution_count": 47,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "0949482e-76ad-4702-9d85-71f276cacf59",
     "text": "six_frame_translation_no_stops(seq)"
    },
    "executionTime": "2019-10-11T16:44:15.969Z"
   },
   {
    "cell": {
     "executionCount": 48,
     "executionEventId": "632529a7-9a51-4ce2-83df-f823a9cb41ee",
     "hasError": false,
     "id": "e50808b6-6666-4543-9299-8075c789791c",
     "outputs": [],
     "persistentId": "1e54230e-ceec-42fc-afa6-7b32e219378b",
     "text": "from khtools.partition_reads import score_reads, make_peptide_bloom_filter, six_frame_translation_no_stops, is_low_complexity"
    },
    "executionTime": "2019-10-11T16:47:45.915Z"
   },
   {
    "cell": {
     "executionCount": 50,
     "executionEventId": "beafb3e7-0432-4e24-9492-75b64531212b",
     "hasError": true,
     "id": "663c7202-08c7-4579-b1b6-b40e7f55ce76",
     "outputs": [
      {
       "ename": "TypeError",
       "evalue": "is_low_complexity() missing 1 required positional argument: 'ksize'",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
        "\u001b[0;32m<ipython-input-50-12300b8ab47f>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mis_low_complexity\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0ms\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
        "\u001b[0;31mTypeError\u001b[0m: is_low_complexity() missing 1 required positional argument: 'ksize'"
       ]
      }
     ],
     "persistentId": "239be055-6b08-4181-9716-7d8401f8166d",
     "text": "is_low_complexity(s)"
    },
    "executionTime": "2019-10-11T16:48:05.726Z"
   },
   {
    "cell": {
     "executionCount": 51,
     "executionEventId": "7e137be4-048f-4578-aa65-623f59becd4c",
     "hasError": false,
     "id": "663c7202-08c7-4579-b1b6-b40e7f55ce76",
     "outputs": [
      {
       "data": {
        "text/plain": "(False, 51)"
       },
       "execution_count": 51,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "239be055-6b08-4181-9716-7d8401f8166d",
     "text": "is_low_complexity(s, 21)"
    },
    "executionTime": "2019-10-11T16:48:10.774Z"
   },
   {
    "cell": {
     "executionCount": 69,
     "executionEventId": "96419006-d8ef-491f-bbaa-3efff68af4ac",
     "hasError": false,
     "id": "b242d89f-8d25-4127-b875-8a85fa69d914",
     "outputs": [
      {
       "name": "stderr",
       "output_type": "stream",
       "text": "/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/screed/openscreed.py:39: ResourceWarning: unclosed file <_io.BufferedReader name='SRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.fq.gz'>\n  self.iter_fn = self.open_reader(filename, *args, **kwargs)\n302it [00:00, 1558.23it/s]"
      },
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "translation: ACLILTSIILGKSQYNCKSCSV\ntranslation: ACLILTSIILGKSQYNCKSCSV\ntranslation: TEQDLQLYCDFPNIIDVSIKQA\ntranslation: TEQDLQLYCDFPNIIDVSIKQA\ntranslation: QNRIYSYIAIFLILLMSVLSK\ntranslation: QNRIYSYIAIFLILLMSVLSK\ntranslation: LECEITYFMCLLTKNTNDKGKSH\ntranslation: LECEITYFMCLLTKNTNDKGKSH\ntranslation: WLFPLSLVFLVRRHMKYVISHSR\ntranslation: WLFPLSLVFLVRRHMKYVISHSR\ntranslation: QSSSLSNSRIFSSPQKEILNPL\ntranslation: QSSSLSNSRIFSSPQKEILNPL\ntranslation: ANGFRISFWGDENILELESDDD\ntranslation: ANGFRISFWGDENILELESDDD\ntranslation: KSEDLPGVFKDSLCSPCSQWCS\ntranslation: KSEDLPGVFKDSLCSPCSQWCS\ntranslation: KVKTSLGSSKTAFALHVANGA\ntranslation: KVKTSLGSSKTAFALHVANGA\ntranslation: RAPLATWRAKAVFEDPREVFTF\ntranslation: RAPLATWRAKAVFEDPREVFTF\ntranslation: EHHWLHGEQRLSLKTPGRSSLL\ntranslation: EHHWLHGEQRLSLKTPGRSSLL\ntranslation: VTLNSSEKIDNIVEWDGG\ntranslation: VTLNSSEKIDNIVEWDGG\ntranslation: VPPSHSTMLSIFSEELRV\ntranslation: VPPSHSTMLSIFSEELRV\ntranslation: QSSSPEFRVQSFSERTNARKKNNH\ntranslation: QSSSPEFRVQSFSERTNARKKNNH\ntranslation: SLQVQNLEFSPFLKEQMLGKKIIT\ntranslation: SLQVQNLEFSPFLKEQMLGKKIIT\ntranslation: LILDNFFWNKSFIYIRLVSLSSE\ntranslation: LILDNFFWNKSFIYIRLVSLSSE\ntranslation: SELNDTSLMYMKDLFQKKLSRIK\ntranslation: SELNDTSLMYMKDLFQKKLSRIK\ntranslation: TRLTYTICNTRWIAHTYDTI\ntranslation: TRLTYTICNTRWIAHTYDTI\ntranslation: EIVSYVCAIHLVLQIVYVSLV\ntranslation: EIVSYVCAIHLVLQIVYVSLV\ntranslation: KLCHMCVLSISCYRLCMLAL\ntranslation: KLCHMCVLSISCYRLCMLAL\ntranslation: CSVVHSQTATCKNKFTIHSALL\ntranslation: CSVVHSQTATCKNKFTIHSALL\ntranslation: VVWSIPRQQHARISLQYTQPF\ntranslation: VVWSIPRQQHARISLQYTQPF\ntranslation: RRAECIVNLFLHVAVWEWTTLQ\ntranslation: RRAECIVNLFLHVAVWEWTTLQ\ntranslation: KFNIPCNFSDLSFISCKYYYFP\ntranslation: KFNIPCNFSDLSFISCKYYYFP\ntranslation: DRLKLVLLFLTFSSLHSAVLSVL\ntranslation: DRLKLVLLFLTFSSLHSAVLSVL\ntranslation: STDSTAECRLENVKNNKTSFSLS\ntranslation: STDSTAECRLENVKNNKTSFSLS\ntranslation: AQTAQQSADWRTLRITKPVSVYQ\ntranslation: AQTAQQSADWRTLRITKPVSVYQ\ntranslation: NKVTFNNNLIHINRERAIDKVN\ntranslation: NKVTFNNNLIHINRERAIDKVN\ntranslation: IYFINCSFSIYVYQIIVKCYLI\ntranslation: IYFINCSFSIYVYQIIVKCYLI\ntranslation: FTLSIALSLFMCIRLLLNVTLF\ntranslation: FTLSIALSLFMCIRLLLNVTLF\ntranslation: LDPPYSRVITQRETENNQMTSE\ntranslation: LDPPYSRVITQRETENNQMTSE\ntranslation: SDVIWLFSVSLCVITLEYGGSK\ntranslation: SDVIWLFSVSLCVITLEYGGSK\ntranslation: NTSNLYFLKYVFLRYLRIHERA\ntranslation: NTSNLYFLKYVFLRYLRIHERA\ntranslation: ARSCILRYLRNTYLRKYKLDVL\ntranslation: ARSCILRYLRNTYLRKYKLDVL\ntranslation: QCRASGDCSRM\ntranslation: QCRASGDCSRM\ntranslation: NAVPVETVLVC\ntranslation: NAVPVETVLVC\ntranslation: MPCQWRLFSYA\ntranslation: MPCQWRLFSYA\ntranslation: GIREQSPLARH\ntranslation: GIREQSPLARH\ntranslation: AYENSLHWHGI\ntranslation: AYENSLHWHGI\ntranslation: HTRTVSTGTAL\ntranslation: HTRTVSTGTAL\ntranslation: LYLLHNELTRNNFARRAKAKTPE\ntranslation: LYLLHNELTRNNFARRAKAKTPE\ntranslation: VSGVLALALLAKLFLVNSLCRRYR\ntranslation: VSGVLALALLAKLFLVNSLCRRYR\ntranslation: GMRPRLKFAAKAQNLVSINSSDM\ntranslation: GMRPRLKFAAKAQNLVSINSSDM\ntranslation: MSEEFMLTRFWAFAANFSRGLIP\ntranslation: MSEEFMLTRFWAFAANFSRGLIP\ntranslation: SDEHGIDPTGTYH\ntranslation: SDEHGIDPTGTYH\ntranslation: VTNMASTPQAHTM\ntranslation: VTNMASTPQAHTM\ntranslation: PWYVPVGSMPCSSL\ntranslation: PWYVPVGSMPCSSL\ntranslation: HGMCLWGRCHVRH\ntranslation: HGMCLWGRCHVRH\ntranslation: MVCACGVDAMFVT\ntranslation: MVCACGVDAMFVT\ntranslation: MIKIMHRHGYLIFSNRKSHPYFL\ntranslation: MIKIMHRHGYLIFSNRKSHPYFL\ntranslation: QQPLCLISTPGPCRGNRNSQLS\ntranslation: QQPLCLISTPGPCRGNRNSQLS\ntranslation: AASMLDFHTGSVQRKQKLPTV\ntranslation: AASMLDFHTGSVQRKQKLPTV\ntranslation: RTVGSFCFLCTDPVWKSSIEAA\ntranslation: RTVGSFCFLCTDPVWKSSIEAA\ntranslation: DSWEFLFPLHGPGVEIKHRGC\ntranslation: DSWEFLFPLHGPGVEIKHRGC\ntranslation: LNSLRNKPGECVTVLGEDPAHS\ntranslation: LNSLRNKPGECVTVLGEDPAHS\ntranslation: SHHIYVLQIVYVSLVYVCLI\ntranslation: SHHIYVLQIVYVSLVYVCLI\ntranslation: YQQRSMMLHSCTMYALVVNGCE\ntranslation: YQQRSMMLHSCTMYALVVNGCE\ntranslation: RKASLRCSLLRIILFFSRCLRKC\ntranslation: RKASLRCSLLRIILFFSRCLRKC\ntranslation: ESKPTLLTITYNPFLFKMPEEV\ntranslation: ESKPTLLTITYNPFLFKMPEEV\ntranslation: HFLRHLEKKRIIRNSEQRRLAF\ntranslation: HFLRHLEKKRIIRNSEQRRLAF\ntranslation: TSSGILKRKGLYVIVSNVGLLS\ntranslation: TSSGILKRKGLYVIVSNVGLLS\ntranslation: LEGCLITEPSFL\ntranslation: LEGCLITEPSFL\ntranslation: SRNEGSVMRQPS\ntranslation: SRNEGSVMRQPS\ntranslation: HICRLVPEG\ntranslation: HICRLVPEG\ntranslation: TYAGWCPR\ntranslation: TYAGWCPR\ntranslation: HMQAGARG\ntranslation: HMQAGARG\ntranslation: PLGHQPAYV\ntranslation: PLGHQPAYV\ntranslation: PSGTSLHM\ntranslation: PSGTSLHM\ntranslation: PRAPACIC\ntranslation: PRAPACIC\ntranslation: LISLFIRLKCFSTEWYRDDICL\ntranslation: LISLFIRLKCFSTEWYRDDICL\ntranslation: HKHISSLYHSVLKHLSLINKEI\ntranslation: HKHISSLYHSVLKHLSLINKEI\ntranslation: ANSQSHHEFSLRKSAFVLTTWPG\ntranslation: ANSQSHHEFSLRKSAFVLTTWPG\ntranslation: SLAKWSKQRLIFGGKIHDGFENL\ntranslation: SLAKWSKQRLIFGGKIHDGFENL\ntranslation: KIHCFVFPQREILLEPVNRMYY\ntranslation: KIHCFVFPQREILLEPVNRMYY\ntranslation: NTFYLQVLVRSLFEGRQNSVF\ntranslation: NTFYLQVLVRSLFEGRQNSVF\ntranslation: LLIILDCAEMFLILRGLESSAHL\ntranslation: LLIILDCAEMFLILRGLESSAHL\ntranslation: PDNLRLCRNVSYFEGLRKLSAFD\ntranslation: PDNLRLCRNVSYFEGLRKLSAFD\ntranslation: IKCAELSKPLKIRNISAQSKIIR\ntranslation: IKCAELSKPLKIRNISAQSKIIR\ntranslation: TNHHPTMTNQTNLKTNDNHTQH\ntranslation: TNHHPTMTNQTNLKTNDNHTQH\ntranslation: VLCMVIICFEVSLISHCWVVIS\ntranslation: VLCMVIICFEVSLISHCWVVIS\ntranslation: NDYHLTNHTVCIRLLLKLQCA\ntranslation: NDYHLTNHTVCIRLLLKLQCA\ntranslation: MIIISQIIQCALDCFSNFSVH\ntranslation: MIIISQIIQCALDCFSNFSVH\ntranslation: CTLKFEKQSNAHCMICEMIII\ntranslation: CTLKFEKQSNAHCMICEMIII\ntranslation: SLSTQHRHAHKERLKKVKGTRQI\ntranslation: SLSTQHRHAHKERLKKVKGTRQI\ntranslation: FAEFLLLFLTFPYEHACVGLTVR\ntranslation: FAEFLLLFLTFPYEHACVGLTVR\ntranslation: NEMLVPRKLPKTKFHQMIKSKME\ntranslation: NEMLVPRKLPKTKFHQMIKSKME\ntranslation: LLFCSTLKLLKIKLAHVDTMW\ntranslation: LLFCSTLKLLKIKLAHVDTMW\ntranslation: LIYLFLIQLKRNFYMNLGFYSVPL\ntranslation: LIYLFLIQLKRNFYMNLGFYSVPL\ntranslation: LEETSLQTQNEIFINIKIVKQRS\ntranslation: LEETSLQTQNEIFINIKIVKQRS\ntranslation: RFFAHPYFPHTKAKIFVNANNM\ntranslation: RFFAHPYFPHTKAKIFVNANNM\ntranslation: HIVSIYKYFCFSMRKVRMGKEA\ntranslation: HIVSIYKYFCFSMRKVRMGKEA\ntranslation: LYYSVFRFVSSYVNFRARASKL\ntranslation: LYYSVFRFVSSYVNFRARASKL\ntranslation: LSPVSRRVIFDQENVVTSTESS\ntranslation: LSPVSRRVIFDQENVVTSTESS\ntranslation: RTLCASNHVLLIKYHSPTYRTQ\ntranslation: RTLCASNHVLLIKYHSPTYRTQ\ntranslation: IMECSSLTIAHCSFNLPGTSDP\ntranslation: IMECSSLTIAHCSFNLPGTSDP\ntranslation: RIACAWEVEAAVSYGQATALHY\ntranslation: RIACAWEVEAAVSYGQATALHY\ntranslation: GTELSMCSRTSWAMTDTTDCS\ntranslation: GTELSMCSRTSWAMTDTTDCS\ntranslation: AEQSVVSVIAQLVLEHMDNSVP\ntranslation: AEQSVVSVIAQLVLEHMDNSVP\ntranslation: RAICGVCHSPTSPGAHGQFCT\ntranslation: RAICGVCHSPTSPGAHGQFCT\ntranslation: EVTVYHIKMVHGKYMAHFLFVL\ntranslation: EVTVYHIKMVHGKYMAHFLFVL\ntranslation: QNKKKMCHIFSMYHLYMIYSDF\ntranslation: QNKKKMCHIFSMYHLYMIYSDF\ntranslation: SVEINHIMAKGHDGRSNQRCS\ntranslation: SVEINHIMAKGHDGRSNQRCS\ntranslation: RTPLITPAIMTLGHNMIYLHT\ntranslation: RTPLITPAIMTLGHNMIYLHT\ntranslation: SVFLFCKMGVPPTL\ntranslation: SVFLFCKMGVPPTL\ntranslation: QSSFSVRWGSRLPC\ntranslation: QSSFSVRWGSRLPC\ntranslation: GRVGGTPILQKRKTE\ntranslation: GRVGGTPILQKRKTE\ntranslation: WEVKGKKTLLSFFPVWEELRNW\ntranslation: WEVKGKKTLLSFFPVWEELRNW\ntranslation: GKLKGRKHCYHFFLFGKSLET\ntranslation: GKLKGRKHCYHFFLFGKSLET\ntranslation: QFLSSSQTGKNDNSVFFPLTSQ\ntranslation: QFLSSSQTGKNDNSVFFPLTSQ\ntranslation: PHSRASWCFLYKHRAP\ntranslation: PHSRASWCFLYKHRAP\ntranslation: HILGLLGVSSINTELP\ntranslation: HILGLLGVSSINTELP\ntranslation: RELCVYRGNTKKPENV\ntranslation: RELCVYRGNTKKPENV\ntranslation: GSSVFIEETPRSPRMW\ntranslation: GSSVFIEETPRSPRMW\ntranslation: QIDVNVQDHFWDGKGCEMIC\ntranslation: QIDVNVQDHFWDGKGCEMIC\ntranslation: TDHLTSLPIPEMILDIDINL\ntranslation: TDHLTSLPIPEMILDIDINL\ntranslation: YHAPSVLFYVHQLPILHPSRL\ntranslation: YHAPSVLFYVHQLPILHPSRL\ntranslation: IMHLLSFFMSINFLFYTLVG\ntranslation: IMHLLSFFMSINFLFYTLVG\ntranslation: LLYPSPSSLSSIQTSSKITIHWR\ntranslation: LLYPSPSSLSSIQTSSKITIHWR\ntranslation: SPVNGYFGGSLYARQGGRRWIKQ\ntranslation: SPVNGYFGGSLYARQGGRRWIKQ\ntranslation: QICMSSWKELLTLYLRPAREY\ntranslation: QICMSSWKELLTLYLRPAREY\ntranslation: NLHELLERVADFVSKTCQGIP\ntranslation: NLHELLERVADFVSKTCQGIP\ntranslation: WYSLAGLRYKVSNSFQELMQI\ntranslation: WYSLAGLRYKVSNSFQELMQI\ntranslation: GIPWQVLDTKSATLSRSSCKF\ntranslation: GIPWQVLDTKSATLSRSSCKF\ntranslation: PVFFSFKNVKLRVVHVFEVPMIF\ntranslation: PVFFSFKNVKLRVVHVFEVPMIF\ntranslation: NIIGTSKTCTTLNLTFLKEKKT\ntranslation: NIIGTSKTCTTLNLTFLKEKKT\ntranslation: FVFNVFSWVIRNSVRSMGVILLHD\ntranslation: FVFNVFSWVIRNSVRSMGVILLHD\ntranslation: VMQKNYPHTPYTIPYHPTKNIKHK\ntranslation: VMQKNYPHTPYTIPYHPTKNIKHK\ntranslation: SCKRITPILLTLFLITQLKTLNT\ntranslation: SCKRITPILLTLFLITQLKTLNT\ntranslation: IAFFCSSSFNNSAFSFSFTF\ntranslation: IAFFCSSSFNNSAFSFSFTF\ntranslation: SRFSAPVLSTTQHFRSLSL\ntranslation: SRFSAPVLSTTQHFRSLSL\ntranslation: RVFLLQFFQQLSIFVLFHF\ntranslation: RVFLLQFFQQLSIFVLFHF\ntranslation: KVKENENAELLKELEQKNA\ntranslation: KVKENENAELLKELEQKNA\ntranslation: LGLQLKFHFHDSNFYLIKGRLKY\ntranslation: LGLQLKFHFHDSNFYLIKGRLKY\ntranslation: NTKLKKKLGMQAQADTEYFPS\ntranslation: NTKLKKKLGMQAQADTEYFPS\ntranslation: MKENTQCQLEPAFQAFFLIWC\ntranslation: MKENTQCQLEPAFQAFFLIWC\ntranslation: EELITFHDHALIIIFLICFLVL\ntranslation: EELITFHDHALIIIFLICFLVL\ntranslation: GRFGTSLKIGIVGLPNVGKSTFF\ntranslation: GRFGTSLKIGIVGLPNVGKSTFF\ntranslation: KIWNLTENWYCWIAKCWEIYFL\ntranslation: KIWNLTENWYCWIAKCWEIYFL\ntranslation: KKVDFPTFGNPTIPIFSEVPNL\ntranslation: KKVDFPTFGNPTIPIFSEVPNL\ntranslation: RSCKLLTDQVCRTKSHYILLYL\ntranslation: RSCKLLTDQVCRTKSHYILLYL\ntranslation: ASKVICNDFLSYKLGQSAAYMI\ntranslation: ASKVICNDFLSYKLGQSAAYMI\ntranslation: SYVMQGICLMKYTNTPDQKLCL\ntranslation: SYVMQGICLMKYTNTPDQKLCL\ntranslation: ETKFLVRCICILHQAYALHYIA\ntranslation: ETKFLVRCICILHQAYALHYIA\ntranslation: DKVSGQVYLYTSSGICLALHS\ntranslation: DKVSGQVYLYTSSGICLALHS\ntranslation: LSTRFSNSIVKCSPLEHVLKYLD\ntranslation: LSTRFSNSIVKCSPLEHVLKYLD\ntranslation: SKYFRTCSRGEHLTIEFENLVE\ntranslation: SKYFRTCSRGEHLTIEFENLVE\ntranslation: HGKAKSQLNQGQRTLGPTDVP\ntranslation: HGKAKSQLNQGQRTLGPTDVP\ntranslation: GTSVGPRVLCPWFNCDLAFP\ntranslation: GTSVGPRVLCPWFNCDLAFP\ntranslation: GHQWGQGFSVPGSTVIWLSR\ntranslation: GHQWGQGFSVPGSTVIWLSR\ntranslation: PVKMNFLAEMKHELTEMKHEL\ntranslation: PVKMNFLAEMKHELTEMKHEL\ntranslation: VSSCFISVSSCFISARKFIFTG\ntranslation: VSSCFISVSSCFISARKFIFTG\ntranslation: KFMLHLCKFMLHLCKKIHFHG\ntranslation: KFMLHLCKFMLHLCKKIHFHG\ntranslation: SYSGRLSFNISTEVPKMYTQKA\ntranslation: SYSGRLSFNISTEVPKMYTQKA\ntranslation: MHLYSLFLIESYGY\ntranslation: MHLYSLFLIESYGY\ntranslation: CIFTHSFSLRAMA\ntranslation: CIFTHSFSLRAMA\ntranslation: GHPLHSALILLLPATMGRRICL\ntranslation: GHPLHSALILLLPATMGRRICL\ntranslation: GIHFTQLSFFSYLPPWVEGYV\ntranslation: GIHFTQLSFFSYLPPWVEGYV\ntranslation: LNFNKAHPAVSSNRQPFMITES\ntranslation: LNFNKAHPAVSSNRQPFMITES\ntranslation: QFQQGTSCSFQQQTTFHDNRI\ntranslation: QFQQGTSCSFQQQTTFHDNRI\ntranslation: RFCYHERLSVAGNCRMCLVEIE\ntranslation: RFCYHERLSVAGNCRMCLVEIE\ntranslation: DSVIMKGCLLLETAGCALLKL\ntranslation: DSVIMKGCLLLETAGCALLKL\ntranslation: FPEFLKSGILLGKHDMLMLKATL\ntranslation: FPEFLKSGILLGKHDMLMLKATL\ntranslation: RVAFNINISCFPSKIPLLRNSGK\ntranslation: RVAFNINISCFPSKIPLLRNSGK\ntranslation: CISHKFCSFFYHSSSDICQTSPF\ntranslation: CISHKFCSFFYHSSSDICQTSPF\ntranslation: ASVINSAHSSITAPLTFARPHHF\ntranslation: ASVINSAHSSITAPLTFARPHHF\ntranslation: INVHSILWLNNIPLLNIVCFVYL\ntranslation: INVHSILWLNNIPLLNIVCFVYL\ntranslation: MRVFSRVNEGFMLLMWFFVFHC\ntranslation: MRVFSRVNEGFMLLMWFFVFHC\ntranslation: RHCSISQSRNILLSLMTLTLFS\ntranslation: RHCSISQSRNILLSLMTLTLFS\ntranslation: DIKNIYKRMTGSEFDFEEMKRK\ntranslation: DIKNIYKRMTGSEFDFEEMKRK\ntranslation: GVVPFIHNIVNIHGLYYMCFSL\ntranslation: GVVPFIHNIVNIHGLYYMCFSL\ntranslation: SCPIYTQHCKHTRSILHVLQS\ntranslation: SCPIYTQHCKHTRSILHVLQS\ntranslation: ISLLSQHKYLWSLPLSSSAWSFS\ntranslation: ISLLSQHKYLWSLPLSSSAWSFS\ntranslation: YLCYPNTNIFGLSLSPPQPGLSL\ntranslation: YLCYPNTNIFGLSLSPPQPGLSL\ntranslation: IFVIPTQISLVSPSLLLSLVFLF\ntranslation: IFVIPTQISLVSPSLLLSLVFLF\ntranslation: KEKDQAEEERGRDQRYLCWDNKD\ntranslation: KEKDQAEEERGRDQRYLCWDNKD\ntranslation: KRKTRLRRREGETKDICVGITKI\ntranslation: KRKTRLRRREGETKDICVGITKI\ntranslation: VQSYFKSALKISFCSYSKYME\ntranslation: VQSYFKSALKISFCSYSKYME\ntranslation: YKATLSLLLRSVFAHIQNTWN\ntranslation: YKATLSLLLRSVFAHIQNTWN\ntranslation: AAKYFDFRLLSNLSFQKHFIK\ntranslation: AAKYFDFRLLSNLSFQKHFIK\ntranslation: AIKAKTKNKQTKTQPNNSTPST\ntranslation: AIKAKTKNKQTKTQPNNSTPST\ntranslation: RVEGVLLFGWVFVCLFFVFAFIA\ntranslation: RVEGVLLFGWVFVCLFFVFAFIA\ntranslation: EWKEYYCLVGFLFVCFLFLLLLP\ntranslation: EWKEYYCLVGFLFVCFLFLLLLP\ntranslation: SGRSTIVWLGFCLFVFCFCFYC\ntranslation: SGRSTIVWLGFCLFVFCFCFYC\ntranslation: FCYAVLVDLLHMFFYRLNTTTP\ntranslation: FCYAVLVDLLHMFFYRLNTTTP\ntranslation: VWWYLVYRRTCGAGQPKLHSR\ntranslation: VWWYLVYRRTCGAGQPKLHSR\ntranslation: KVASSSTPRTLNFLHFLSRQS\ntranslation: KVASSSTPRTLNFLHFLSRQS\ntranslation: ALSRQEMQEVQSSRSGRGGNF\ntranslation: ALSRQEMQEVQSSRSGRGGNF\ntranslation: LCLDKKCRKFRVLGVEEEATL\ntranslation: LCLDKKCRKFRVLGVEEEATL\ntranslation: QVSTPKTTSSKPQTYHTQTP\ntranslation: QVSTPKTTSSKPQTYHTQTP\ntranslation: RFLLQRPHHRNRKHIIHKRL\ntranslation: RFLLQRPHHRNRKHIIHKRL\ntranslation: GFYSKDHIIETANISYTNA\ntranslation: GFYSKDHIIETANISYTNA\ntranslation: QENDSVYYSQFDPMYPTVGSNT\ntranslation: QENDSVYYSQFDPMYPTVGSNT\ntranslation: AFKLNTHYLKNPKHITELLTPNW\ntranslation: AFKLNTHYLKNPKHITELLTPNW\ntranslation: SNLFETSNGFLSDKTFTTLSFKS\ntranslation: SNLFETSNGFLSDKTFTTLSFKS\ntranslation: ADLKDKVVNVLSLKNPLEVSNKLL\ntranslation: ADLKDKVVNVLSLKNPLEVSNKLL\ntranslation: NVVHQLSVTLEDLYNGATRK\ntranslation: NVVHQLSVTLEDLYNGATRK\ntranslation: VFLLHHYISLLGLLRADVQHF\ntranslation: VFLLHHYISLLGLLRADVQHF\ntranslation: EGFLAKSFSRTISRDLPVSHHD\ntranslation: EGFLAKSFSRTISRDLPVSHHD\ntranslation: APVPSLFKGTCTGIGNVPHLPSC\ntranslation: APVPSLFKGTCTGIGNVPHLPSC\ntranslation: QEGRWGTFPIPVHVPLNKLGTGA\ntranslation: QEGRWGTFPIPVHVPLNKLGTGA\ntranslation: AWNRMVCFNVYLCMFIFMVKGVN\ntranslation: AWNRMVCFNVYLCMFIFMVKGVN\ntranslation: HGTEWSASMFIYVCSSSWLKEL\ntranslation: HGTEWSASMFIYVCSSSWLKEL\ntranslation: LILCCYNEDKVHTETTVRLRNFH\ntranslation: LILCCYNEDKVHTETTVRLRNFH\ntranslation: CNLELCIIIQHIVPSIFKWFLQR\ntranslation: CNLELCIIIQHIVPSIFKWFLQR\ntranslation: PLQEPFKNAWNYMLNNYTKFQIA\ntranslation: PLQEPFKNAWNYMLNNYTKFQIA\ntranslation: RIGATHFFSPWFYAGSRSQVKS\ntranslation: RIGATHFFSPWFYAGSRSQVKS\ntranslation: DHIAILSCKGDWNSENLDL\ntranslation: DHIAILSCKGDWNSENLDL\ntranslation: IHLSLWLFVYYLCIFSCIDIIDI\ntranslation: IHLSLWLFVYYLCIFSCIDIIDI\ntranslation: IFAADLLFTMDSIMRFFFFVG\ntranslation: IFAADLLFTMDSIMRFFFFVG\ntranslation: FCCGSPIYNGLHYEILLLCWC\ntranslation: FCCGSPIYNGLHYEILLLCWC\ntranslation: TPTKKKNLIMESIVNRRSAAK\ntranslation: TPTKKKNLIMESIVNRRSAAK\ntranslation: LNDVQDRFEVNISELPDEIDISS\ntranslation: LNDVQDRFEVNISELPDEIDISS\ntranslation: GGDVYLIRQLTNIDLKAILHIIE\ntranslation: GGDVYLIRQLTNIDLKAILHIIE\ntranslation: EEMSISSGSSLILTSKRSCTSLR\ntranslation: EEMSISSGSSLILTSKRSCTSLR\ntranslation: QCTRDFKLLYLCQKFKTKQPPKS\ntranslation: QCTRDFKLLYLCQKFKTKQPPKS\ntranslation: SVREILNYFISVRNSKLNNLQSL\ntranslation: SVREILNYFISVRNSKLNNLQSL\ntranslation: ISFKVCLTDTSFISYDMLRLIFLF\ntranslation: ISFKVCLTDTSFISYDMLRLIFLF\ntranslation: KRKINLNISYEINEVSVKHTLKL\ntranslation: KRKINLNISYEINEVSVKHTLKL\ntranslation: SLLFTFSYFQKNSSEKYVMQ\ntranslation: SLLFTFSYFQKNSSEKYVMQ\ntranslation: LKVSHCLGFSIRILSSTFSVMPL\ntranslation: LKVSHCLGFSIRILSSTFSVMPL\ntranslation: RGITENVLDKILIENPKQWLTFK\ntranslation: RGITENVLDKILIENPKQWLTFK\ntranslation: RAISITIYTPTNNVQPVTTTN\ntranslation: RAISITIYTPTNNVQPVTTTN\ntranslation: VSWHCDHSIGNSLPKVGFCNF\ntranslation: VSWHCDHSIGNSLPKVGFCNF\ntranslation: GNCRSLPWEDCYQCCGHSASLL\ntranslation: GNCRSLPWEDCYQCCGHSASLL\ntranslation: EIAEAYLGKTVTNAVVTVPAY\ntranslation: EIAEAYLGKTVTNAVVTVPAY\ntranslation: KLQKPTLGRLLPMLWSQCQLT\ntranslation: KLQKPTLGRLLPMLWSQCQLT\ntranslation: NILMNIYGWTYVHVQWRK\ntranslation: NILMNIYGWTYVHVQWRK\ntranslation: PSSIAHEHMSIHIYSSEC\ntranslation: PSSIAHEHMSIHIYSSEC\ntranslation: LPPLHMNICPSIYIHQNV\ntranslation: LPPLHMNICPSIYIHQNV\ntranslation: GYEFFLLGLMRVVRMGGIREGG\ntranslation: GYEFFLLGLMRVVRMGGIREGG\ntranslation: PPSLIPPILTTLINPNKKNSYP\ntranslation: PPSLIPPILTTLINPNKKNSYP\ntranslation: IQHANDVTPLFIITTVIVKFI\ntranslation: IQHANDVTPLFIITTVIVKFI\ntranslation: NELNYHRGNDKERSYIISMLN\ntranslation: NELNYHRGNDKERSYIISMLN\ntranslation: RKCLLEKIDVITGEETEHNVL\ntranslation: RKCLLEKIDVITGEETEHNVL\ntranslation: NTLCSVSSPVITSIFSSKHF\ntranslation: NTLCSVSSPVITSIFSSKHF\ntranslation: QTQKQHLNSRPHPLNQTLL\ntranslation: QTQKQHLNSRPHPLNQTLL\ntranslation: WFKKSKLLVLNEAILDYDGFFSLL\ntranslation: WFKKSKLLVLNEAILDYDGFFSLL\ntranslation: VGKFKVELRFYLGQPAITRLG\ntranslation: VGKFKVELRFYLGQPAITRLG\ntranslation: LISLVCNLSWLISKTYYSFQL\ntranslation: LISLVCNLSWLISKTYYSFQL\ntranslation: SFLLFVIYHGSFQRHITAFS\ntranslation: SFLLFVIYHGSFQRHITAFS\ntranslation: QISHQQPTNHHPTMTNQTNLKTN\ntranslation: QISHQQPTNHHPTMTNQTNLKTN\ntranslation: ICFEVSLISHCWVVISRLLMRYL\ntranslation: ICFEVSLISHCWVVISRLLMRYL\ntranslation: EKWPRKP\ntranslation: EKWPRKP\ntranslation: RNGPESQ\ntranslation: RNGPESQ\ntranslation: EMAQKAR\ntranslation: EMAQKAR\ntranslation: PGFLGHF\ntranslation: PGFLGHF\ntranslation: LAFWAIS\ntranslation: LAFWAIS\ntranslation: WLSGPFL\ntranslation: WLSGPFL\ntranslation: LFWQIVLNKIKVAYISNITQHP\ntranslation: LFWQIVLNKIKVAYISNITQHP\ntranslation: CVCCSHHIYVLQIVYVSLVYVCL\ntranslation: CVCCSHHIYVLQIVYVSLVYVCL\ntranslation: VCVAVIISMCYRLCMLALCMCA\ntranslation: VCVAVIISMCYRLCMLALCMCA\ntranslation: QVRPLVLCMVIICFEVSLISHCW\ntranslation: QVRPLVLCMVIICFEVSLISHCW\ntranslation: IVPILIAMAFLMLTERKILGYIQ\ntranslation: IVPILIAMAFLMLTERKILGYIQ\ntranslation: CTHSNRNGIPNAYRTKNSRLYTT\ntranslation: CTHSNRNGIPNAYRTKNSRLYTT\ntranslation: MISNTTNTQEKKYNTANSVLQLCI\ntranslation: MISNTTNTQEKKYNTANSVLQLCI\ntranslation: ASKSQLLAHQFIWNMKTNIYLDE\ntranslation: ASKSQLLAHQFIWNMKTNIYLDE\ntranslation: GPPDLNKPCSLYTSLNWTYHLY\ntranslation: GPPDLNKPCSLYTSLNWTYHLY\ntranslation: MSQKFSHPETTKEKKSLIKCHAF\ntranslation: MSQKFSHPETTKEKKSLIKCHAF\ntranslation: SEVSQVEDVIELLVVVLFNVGG\ntranslation: SEVSQVEDVIELLVVVLFNVGG\ntranslation: RPPTLNRTTTKSSMTSSTWETSL\ntranslation: RPPTLNRTTTKSSMTSSTWETSL\ntranslation: TTYIKQNHYQELNDILNLGNFT\ntranslation: TTYIKQNHYQELNDILNLGNFT\ntranslation: LGSYQIKFQVKDNLLWQFKKNTE\ntranslation: LGSYQIKFQVKDNLLWQFKKNTE\ntranslation: WVHTRLNFKLKIIYYGSLKKTQN\ntranslation: WVHTRLNFKLKIIYYGSLKKTQN\ntranslation: PGFGEWGWGCNSVLGRNGGGGCA\ntranslation: PGFGEWGWGCNSVLGRNGGGGCA\ntranslation: RVWRMGMGVQLCVRKEWWGRLC\ntranslation: RVWRMGMGVQLCVRKEWWGRLC\ntranslation: QHNLPHHSFLTQSCTPIPILQTL\ntranslation: QHNLPHHSFLTQSCTPIPILQTL\ntranslation: AQPPPPFLPNTELHPHPHSPNP\ntranslation: AQPPPPFLPNTELHPHPHSPNP\ntranslation: SQSSALGIYLCSQR\ntranslation: SQSSALGIYLCSQR\ntranslation: LSPLHWEFIYVLNA\ntranslation: LSPLHWEFIYVLNA\ntranslation: SVLCTGNLFMFST\ntranslation: SVLCTGNLFMFST\ntranslation: CVENINKFPVQRTE\ntranslation: CVENINKFPVQRTE\ntranslation: KFFTLSTRFFPSVQRAVPLWT\ntranslation: KFFTLSTRFFPSVQRAVPLWT\ntranslation: NFLLSLQGFFLVSKELFLFG\ntranslation: NFLLSLQGFFLVSKELFLFG\ntranslation: SPKRNSSLDTRKKPCRESKKF\ntranslation: SPKRNSSLDTRKKPCRESKKF\ntranslation: VQRGTALWTLGKNLVERVKN\ntranslation: VQRGTALWTLGKNLVERVKN\ntranslation: LGYNFSSFPCGTISIAPGFNFYR\ntranslation: LGYNFSSFPCGTISIAPGFNFYR\ntranslation: LVIIFHLSLAVLYLLRQVSISIA\ntranslation: LVIIFHLSLAVLYLLRQVSISIA\ntranslation: DIQNKCKEKVRLFFCSDLSIKK\ntranslation: DIQNKCKEKVRLFFCSDLSIKK\ntranslation: TFKTNAKKRSGYSSAVILASRK\ntranslation: TFKTNAKKRSGYSSAVILASRK\ntranslation: SASFPSDIGLHQSQHHPLCAEG\ntranslation: SASFPSDIGLHQSQHHPLCAEG\ntranslation: PHHFHLILDFIKVSIILCVLRE\ntranslation: PHHFHLILDFIKVSIILCVLRE\ntranslation: FPQHTEDDADFDEVQYQMEMMR\ntranslation: FPQHTEDDADFDEVQYQMEMMR\ntranslation: GLRDKLHVSITIRLHMYVCVYL\ntranslation: GLRDKLHVSITIRLHMYVCVYL\ntranslation: PHHPTKNIKHKLPPTSLTK\ntranslation: PHHPTKNIKHKLPPTSLTK\ntranslation: LITQLKTLNTNYHLPPSPK\ntranslation: LITQLKTLNTNYHLPPSPK\ntranslation: LHFLALICSSLHVWFLIFCCCCC\ntranslation: LHFLALICSSLHVWFLIFCCCCC\ntranslation: PFPRTYLQQSACVVFDFLLLLL\ntranslation: PFPRTYLQQSACVVFDFLLLLL\ntranslation: TTTTTKNQKPHMQTAADKCEEME\ntranslation: TTTTTKNQKPHMQTAADKCEEME\ntranslation: QQQQQKIKNHTCRLLQISARKW\ntranslation: QQQQQKIKNHTCRLLQISARKW\ntranslation: KPNIINFDTSLPTSHTYLGADM\ntranslation: KPNIINFDTSLPTSHTYLGADM\ntranslation: ISAPRYVCDVGRLVSKFMMFGF\ntranslation: ISAPRYVCDVGRLVSKFMMFGF\ntranslation: FIEKSSLSVHPSNLMEKKPSA\ntranslation: FIEKSSLSVHPSNLMEKKPSA\ntranslation: AEGFFSIRLLGWTLRLDFSMK\ntranslation: AEGFFSIRLLGWTLRLDFSMK\ntranslation: RQVDVVITCTGNKNVVTREHLDR\ntranslation: RQVDVVITCTGNKNVVTREHLDR\ntranslation: MRSKCSRVTTFLFPVQVITTSTC\ntranslation: MRSKCSRVTTFLFPVQVITTSTC\ntranslation: CDPSAPVSLHSYFLCKLLRHRLA\ntranslation: CDPSAPVSLHSYFLCKLLRHRLA\ntranslation: AIQVLPCHYILISCASYYDIDLP\ntranslation: AIQVLPCHYILISCASYYDIDLP\ntranslation: AYTAYNSWRICLFCGMHYSFWNI\ntranslation: AYTAYNSWRICLFCGMHYSFWNI\ntranslation: HIQLIIPGESACFAVCIILFGIF\ntranslation: HIQLIIPGESACFAVCIILFGIF\ntranslation: EYSKKNNAYRKTSRFSRNYKLYM\ntranslation: EYSKKNNAYRKTSRFSRNYKLYM\ntranslation: NIPKRIMHTAKQADSPGIISCIC\ntranslation: NIPKRIMHTAKQADSPGIISCIC\ntranslation: SLYRLLLFCPFLSCHFACLFL\ntranslation: SLYRLLLFCPFLSCHFACLFL\ntranslation: LCIGFYCFALFCLVILLAFF\ntranslation: LCIGFYCFALFCLVILLAFF\ntranslation: EKGKQNDKTEKGKTIKAYTE\ntranslation: EKGKQNDKTEKGKTIKAYTE\ntranslation: SLSVSPSVLCLSISVSMCLCP\ntranslation: SLSVSPSVLCLSISVSMCLCP\ntranslation: LFLSPPLFSVSQSLSPCVSAP\ntranslation: LFLSPPLFSVSQSLSPCVSAP\ntranslation: SFCLPLCSLSLNLCLHVSLPL\ntranslation: SFCLPLCSLSLNLCLHVSLPL\ntranslation: QGQRHMETEIERQRTEGETER\ntranslation: QGQRHMETEIERQRTEGETER\ntranslation: RGRDTWRQRLRDREQRGRQKE\ntranslation: RGRDTWRQRLRDREQRGRQKE\ntranslation: TPNWTNLSPYRRTNVSISNMK\ntranslation: TPNWTNLSPYRRTNVSISNMK\ntranslation: KIMPIFLVQIVFGTLDLLVLMM\ntranslation: KIMPIFLVQIVFGTLDLLVLMM\ntranslation: IIRTNKSSVPNTICTKKIGIIF\ntranslation: IIRTNKSSVPNTICTKKIGIIF\ntranslation: SSELINQVFQIQFALKRLALF\ntranslation: SSELINQVFQIQFALKRLALF\ntranslation: LYWGGKAETERGLSRKHIIEGL\ntranslation: LYWGGKAETERGLSRKHIIEGL\ntranslation: LQFKCSDSKGNRQTSRKEGSK\ntranslation: LQFKCSDSKGNRQTSRKEGSK\ntranslation: CSLSVRIPRETDKPHERKEAS\ntranslation: CSLSVRIPRETDKPHERKEAS\ntranslation: LLPSFREVCLFPLESEHLNC\ntranslation: LLPSFREVCLFPLESEHLNC\ntranslation: KYVLTNNHTNTSHYHSLCPHSF\ntranslation: KYVLTNNHTNTSHYHSLCPHSF\ntranslation: RSPRRLVVAIKMIKDTSIRDQ\ntranslation: RSPRRLVVAIKMIKDTSIRDQ\ntranslation: PSQKRDTVDSQAGGLNLSKTFI\ntranslation: PSQKRDTVDSQAGGLNLSKTFI\ntranslation: QKYKSNKLQITLSLFLINTATVL\ntranslation: QKYKSNKLQITLSLFLINTATVL\ntranslation: RTVAVLIRKRDRVICSLFDLYFC\ntranslation: RTVAVLIRKRDRVICSLFDLYFC\ntranslation: EGSVGKFKVELRFYLGQPAITRL\ntranslation: EGSVGKFKVELRFYLGQPAITRL\ntranslation: PQVQTFGVCAWLRSQWGEATI\ntranslation: PQVQTFGVCAWLRSQWGEATI\ntranslation: AGSDIWCMCLAEEPMGRSYHL\ntranslation: AGSDIWCMCLAEEPMGRSYHL\ntranslation: DGSFAPLAPQPSTYTKCLNLR\ntranslation: DGSFAPLAPQPSTYTKCLNLR\ntranslation: AMFTFITNSLTETAIYSFLQQE\ntranslation: AMFTFITNSLTETAIYSFLQQE\ntranslation: SDLVSPALGPFLLRPRSLSPV\ntranslation: SDLVSPALGPFLLRPRSLSPV\ntranslation: QLGTKTWASGGRVPAQGTQGQ\ntranslation: QLGTKTWASGGRVPAQGTQGQ\ntranslation: NWGQRPGPQEEGSQRRGHKVR\ntranslation: NWGQRPGPQEEGSQRRGHKVR\ntranslation: TGDKDLGLRRKGPSAGDTRSE\ntranslation: TGDKDLGLRRKGPSAGDTRSE\ntranslation: VIEEYAKILRSWVWFNPPQLPAM\ntranslation: VIEEYAKILRSWVWFNPPQLPAM\ntranslation: HSRQLRWIKPNPATQNLSILLNY\ntranslation: HSRQLRWIKPNPATQNLSILLNY\ntranslation: FIFIILKRLCPFSNEKTKAYRGY\ntranslation: FIFIILKRLCPFSNEKTKAYRGY\ntranslation: ITFRGLIFHGELFQHIACCSDP\ntranslation: ITFRGLIFHGELFQHIACCSDP\ntranslation: AMLQSTLSLWSSWFFYLHCHLYP\ntranslation: AMLQSTLSLWSSWFFYLHCHLYP\ntranslation: DVTEHIESVVIVVLLSSLSPVS\ntranslation: DVTEHIESVVIVVLLSSLSPVS\ntranslation: QDTGDSEDRRTTMTTDSMCSVTS\ntranslation: QDTGDSEDRRTTMTTDSMCSVTS\ntranslation: ATITAQLLQRRRSVCASCSQSQR\ntranslation: ATITAQLLQRRRSVCASCSQSQR\ntranslation: PQSLLSFSKDVVLFALHVLKVKG\ntranslation: PQSLLSFSKDVVLFALHVLKVKG\ntranslation: HNHCSASPKTSFCLRFMFSKSK\ntranslation: HNHCSASPKTSFCLRFMFSKSK\ntranslation: RHSIVHRLFSTSINQ\ntranslation: RHSIVHRLFSTSINQ\ntranslation: AKLATSMFLAGESLVCIA\ntranslation: AKLATSMFLAGESLVCIA\ntranslation: MAMHTRLSPARNILVANF\ntranslation: MAMHTRLSPARNILVANF\ntranslation: WQCIPGFHLLETYLLPTL\ntranslation: WQCIPGFHLLETYLLPTL\ntranslation: HSVTCRTMQPWM\ntranslation: HSVTCRTMQPWM\ntranslation: IPSPAGRCSPG\ntranslation: IPSPAGRCSPG\ntranslation: FRHLQDDAALD\ntranslation: FRHLQDDAALD\ntranslation: HPGLHRPAGDGM\ntranslation: HPGLHRPAGDGM\ntranslation: IQGCIVLQVTE\ntranslation: IQGCIVLQVTE\ntranslation: LECSGMIIAHCSLNLPGSSDPP\ntranslation: LECSGMIIAHCSLNLPGSSDPP\ntranslation: GVQWHDHSSLQPQPPRLKRSSP\ntranslation: GVQWHDHSSLQPQPPRLKRSSP\ntranslation: DDLEGIMLNSHIDLMEHFKGKN\ntranslation: DDLEGIMLNSHIDLMEHFKGKN\ntranslation: ASSVDTSYLGGVERV\ntranslation: ASSVDTSYLGGVERV\ntranslation: KLLSIINHPFDFFRCQAILVIGD\ntranslation: KLLSIINHPFDFFRCQAILVIGD\ntranslation: NFSASLTILSISSGVRRFWSLVI\ntranslation: NFSASLTILSISSGVRRFWSLVI\ntranslation: ITNDQNRLTPEEIERMVNDAEKF\ntranslation: ITNDQNRLTPEEIERMVNDAEKF\ntranslation: HLVPCFKSSWMFCMSWGLKLTQ\ntranslation: HLVPCFKSSWMFCMSWGLKLTQ\ntranslation: WVSFSPHDMQNIHEDLKHGTRW\ntranslation: WVSFSPHDMQNIHEDLKHGTRW\ntranslation: FNSRQNIVQYEKHYMRQRQFSKI\ntranslation: FNSRQNIVQYEKHYMRQRQFSKI\ntranslation: STYNSLTNCWELSQKHLEHWW\ntranslation: STYNSLTNCWELSQKHLEHWW\ntranslation: PPVFQMLLTKLPTISQRIVSA\ntranslation: PPVFQMLLTKLPTISQRIVSA\ntranslation: FQNLFKNCCYYCLFQNSILVFSS\ntranslation: FQNLFKNCCYYCLFQNSILVFSS\ntranslation: SKTSLRIVAIIASSRTPFLCSP\ntranslation: SKTSLRIVAIIASSRTPFLCSP\ntranslation: RGEHKNGVLEEAIIATILKEVLE\ntranslation: RGEHKNGVLEEAIIATILKEVLE\ntranslation: VNENQIVKLSEKLINYRLLEQL\ntranslation: VNENQIVKLSEKLINYRLLEQL\ntranslation: PLLPSPYYISILGLVLISGLIL\ntranslation: PLLPSPYYISILGLVLISGLIL\ntranslation: FITISLLHFYSRVSVNLRSYS\ntranslation: FITISLLHFYSRVSVNLRSYS\ntranslation: KNFAALLEEHDIFVPEDDDDDD\ntranslation: KNFAALLEEHDIFVPEDDDDDD\ntranslation: RTLLLYLRNMTFLFQRMMMMMT\ntranslation: RTLLLYLRNMTFLFQRMMMMMT\ntranslation: SHHHHHPLEQKCHVPQVEQQSS\ntranslation: SHHHHHPLEQKCHVPQVEQQSS\ntranslation: SSSSSSSGTKMSCSSSRAAKFL\ntranslation: SSSSSSSGTKMSCSSSRAAKFL\ntranslation: PIPNAPLRLIRPNHSSPTSPIS\ntranslation: PIPNAPLRLIRPNHSSPTSPIS\ntranslation: RDRRSRTAVIRTDQTKRGVWYW\ntranslation: RDRRSRTAVIRTDQTKRGVWYW\ntranslation: GWLHTVVNIQMIHWANRNHDIRK\ntranslation: GWLHTVVNIQMIHWANRNHDIRK\ntranslation: KKKQVLKFTLDCTHPVEDGIMD\ntranslation: KKKQVLKFTLDCTHPVEDGIMD\ntranslation: IHDSIFYRVGAIKSELQNLLLF\ntranslation: IHDSIFYRVGAIKSELQNLLLF\ntranslation: SMIPSSTGWVQSRVNFRTCFFF\ntranslation: SMIPSSTGWVQSRVNFRTCFFF\ntranslation: VDPAESLNYSTSTLISLYFLP\ntranslation: VDPAESLNYSTSTLISLYFLP\ntranslation: QEENTEKLKLKYYNLMIQLDQ\ntranslation: QEENTEKLKLKYYNLMIQLDQ\ntranslation: SDVTKKSNSFRASFVSRSSPL\ntranslation: SDVTKKSNSFRASFVSRSSPL\ntranslation: DSGEDRDTKEALKEFDFLVTSE\ntranslation: DSGEDRDTKEALKEFDFLVTSE\ntranslation: SELSALERQKDEKITQQEEKYEA\ntranslation: SELSALERQKDEKITQQEEKYEA\ntranslation: IASYFSSCWVIFSSFCLSRALNS\ntranslation: IASYFSSCWVIFSSFCLSRALNS\ntranslation: QLYKMTKIDCRI\ntranslation: QLYKMTKIDCRI\ntranslation: DPTIYFSHFVQL\ntranslation: DPTIYFSHFVQL\ntranslation: ILQSILVILYS\ntranslation: ILQSILVILYS\ntranslation: SVLPPDYDSNPTQL\ntranslation: SVLPPDYDSNPTQL\ntranslation: QFCLLTMTVIPPS\ntranslation: QFCLLTMTVIPPS\ntranslation: AGWDYCHSQEAEL\ntranslation: AGWDYCHSQEAEL\ntranslation: SRFWSSMWSSLLSTLLTLKVKQ\ntranslation: SRFWSSMWSSLLSTLLTLKVKQ\ntranslation: HCFTLRVNNVERREDHIDDQNLL\ntranslation: HCFTLRVNNVERREDHIDDQNLL\ntranslation: LRSRAVRSPRKVTARAVSF\ntranslation: LRSRAVRSPRKVTARAVSF\ntranslation: KTHSPRCHFPRTSNSPRPQ\ntranslation: KTHSPRCHFPRTSNSPRPQ\ntranslation: KLTALAVTFLGLLTALDLN\ntranslation: KLTALAVTFLGLLTALDLN\ntranslation: PMLSERSQNTTVRITLFHLYETLE\ntranslation: PMLSERSQNTTVRITLFHLYETLE\ntranslation: EFRTSISVTPSLMCFCYCFEEGSS\ntranslation: EFRTSISVTPSLMCFCYCFEEGSS\ntranslation: FHHLLHLTEINLLEFLPVTLSFL\ntranslation: FHHLLHLTEINLLEFLPVTLSFL\ntranslation: QFYWVSILPSYKPQSTSSLPSPF\ntranslation: QFYWVSILPSYKPQSTSSLPSPF\ntranslation: NFTGSLFYPPTSLRVLRVSLHHF\ntranslation: NFTGSLFYPPTSLRVLRVSLHHF\ntranslation: ILLGLYFTLLQASEYFESPFTI\ntranslation: ILLGLYFTLLQASEYFESPFTI\ntranslation: PVSKAFLDRWCVVALVCAFSCYI\ntranslation: PVSKAFLDRWCVVALVCAFSCYI\ntranslation: PYRRPFWTGGVWWPWYVLSRVTS\ntranslation: PYRRPFWTGGVWWPWYVLSRVTS\ntranslation: RIEGLFGQVVCGGLGMCFLVLH\ntranslation: RIEGLFGQVVCGGLGMCFLVLH\ntranslation: RCNTRKHIPRPPHTTCPKRPSIR\ntranslation: RCNTRKHIPRPPHTTCPKRPSIR\ntranslation: DVTRESTYQGHHTPPVQKGLRYG\ntranslation: DVTRESTYQGHHTPPVQKGLRYG\ntranslation: EPFYTNTYSDFSVTLKFIFLSY\ntranslation: EPFYTNTYSDFSVTLKFIFLSY\ntranslation: HSALENQNERKSVRFIHCPHNP\ntranslation: HSALENQNERKSVRFIHCPHNP\ntranslation: LEFPFISCQLASQFLAYVFSWCV\ntranslation: LEFPFISCQLASQFLAYVFSWCV\ntranslation: WNFPSYLASWPPSFLLMFSPGVF\ntranslation: WNFPSYLASWPPSFLLMFSPGVF\ntranslation: GISLHILPVGLPVSCLCFLLVC\ntranslation: GISLHILPVGLPVSCLCFLLVC\ntranslation: NTPGENISKKLGGQLARYEGKFQ\ntranslation: NTPGENISKKLGGQLARYEGKFQ\ntranslation: FISFALLKQEAGRPASPLHI\ntranslation: FISFALLKQEAGRPASPLHI\ntranslation: SYLLHSSSKKPADLQALCTS\ntranslation: SYLLHSSSKKPADLQALCTS\ntranslation: HIFCTPQARSRQTCKPFAHL\ntranslation: HIFCTPQARSRQTCKPFAHL\ntranslation: KMCKGLAGLPASCLRSAKDM\ntranslation: KMCKGLAGLPASCLRSAKDM\ntranslation: DVQRACRSAGFLLEECKRYE\ntranslation: DVQRACRSAGFLLEECKRYE\ntranslation: HEQLSTQSLIELFESFKSFEGLF\ntranslation: HEQLSTQSLIELFESFKSFEGLF\ntranslation: LRCCKIITLSTMNTGTLLKTLL\ntranslation: LRCCKIITLSTMNTGTLLKTLL\ntranslation: KSVFSKVPVFIVDKVIILQHL\ntranslation: KSVFSKVPVFIVDKVIILQHL\ntranslation: NSEGLYRVSGFSDLIEDVKMAF\ntranslation: NSEGLYRVSGFSDLIEDVKMAF\ntranslation: KAILTSSIRSLNPDTRYSPSEL\ntranslation: KAILTSSIRSLNPDTRYSPSEL\ntranslation: RILEVRNVFTMTAKEGRKKSIR\ntranslation: RILEVRNVFTMTAKEGRKKSIR\ntranslation: TDRFLSSLFRSHSENVSYLKYP\ntranslation: TDRFLSSLFRSHSENVSYLKYP\ntranslation: RIDFFLPSFAVIVKTFLTSSIL\ntranslation: RIDFFLPSFAVIVKTFLTSSIL\ntranslation: QEKNIQKVIFGLAMSKSQLYKDL\ntranslation: QEKNIQKVIFGLAMSKSQLYKDL\ntranslation: KKKIFRKLFLALLCQKVNCTKIY\ntranslation: KKKIFRKLFLALLCQKVNCTKIY\ntranslation: RKKYSESYFWPCYVKKSTVQRST\ntranslation: RKKYSESYFWPCYVKKSTVQRST\ntranslation: FKQLSSLFRCHSLLLNSHSELPS\ntranslation: FKQLSSLFRCHSLLLNSHSELPS\ntranslation: DGSSLWEFKSKLWHRNKLDNCLN\ntranslation: DGSSLWEFKSKLWHRNKLDNCLN\ntranslation: HDKEIFFHYSEFSGDVDSLEL\ntranslation: HDKEIFFHYSEFSGDVDSLEL\ntranslation: MIRKSFSITVSSLVMLIAWNW\ntranslation: MIRKSFSITVSSLVMLIAWNW\ntranslation: PVPGYQHHQRTHCNGKRFPYH\ntranslation: PVPGYQHHQRTHCNGKRFPYH\ntranslation: QFQAINITRELTVMEKDFLIM\ntranslation: QFQAINITRELTVMEKDFLIM\ntranslation: NHIPNLSLTKRKPSPHSLNLIHH\ntranslation: NHIPNLSLTKRKPSPHSLNLIHH\ntranslation: IIYQISPSLNVSLLLTLSILSI\ntranslation: IIYQISPSLNVSLLLTLSILSI\ntranslation: MMDKIERVRRRLTFSEGEIWYMI\ntranslation: MMDKIERVRRRLTFSEGEIWYMI\ntranslation: CKIVNWYLGDFCSLLKKIAFVLK\ntranslation: CKIVNWYLGDFCSLLKKIAFVLK\ntranslation: QNSELVLGRLLFFVEKNCLCFE\ntranslation: QNSELVLGRLLFFVEKNCLCFE\ntranslation: ISKQRQFFSTKNRSLPSTNSLFC\ntranslation: ISKQRQFFSTKNRSLPSTNSLFC\ntranslation: FQNKGNFFQQRTEVSQVPIHYFA\ntranslation: FQNKGNFFQQRTEVSQVPIHYFA\ntranslation: FKTKAIFFNKEQKSPKYQFTIL\ntranslation: FKTKAIFFNKEQKSPKYQFTIL\ntranslation: HIVYCFHYIKHFLFSDVAIFIYV\ntranslation: HIVYCFHYIKHFLFSDVAIFIYV\ntranslation: DINKDGYITKEEMLDIMKAIYDM\ntranslation: DINKDGYITKEEMLDIMKAIYDM\ntranslation: LNMNCHFRFLTFVSKEKLRKSS\ntranslation: LNMNCHFRFLTFVSKEKLRKSS\ntranslation: LIEEYAMSDFRSVCRR\ntranslation: LIEEYAMSDFRSVCRR\ntranslation: PTTNRPKIAHCILFNQ\ntranslation: PTTNRPKIAHCILFNQ\ntranslation: LRQTDLKSLIAYSSI\ntranslation: LRQTDLKSLIAYSSI\ntranslation: SPVPGTCYYQHHALPISSTD\ntranslation: SPVPGTCYYQHHALPISSTD\ntranslation: LLCQELATTSTMPYQYPALT\ntranslation: LLCQELATTSTMPYQYPALT\ntranslation: GQCWILVGHGAGSSKFLAQE\ntranslation: GQCWILVGHGAGSSKFLAQE\ntranslation: CRATFQKLGEVPSLTVSTQPGP\ntranslation: CRATFQKLGEVPSLTVSTQPGP\ntranslation: QSYISEIGRSAKSYCEHTART\ntranslation: QSYISEIGRSAKSYCEHTART\ntranslation: GSWLCAHSKTWHFSQFLKCSSA\ntranslation: GSWLCAHSKTWHFSQFLKCSSA\ntranslation: NTHVVKRMIVRHTAFCFRNRILP\ntranslation: NTHVVKRMIVRHTAFCFRNRILP\ntranslation: GSILFLKQNAVCLTIILLTTCV\ntranslation: GSILFLKQNAVCLTIILLTTCV\ntranslation: LTRVERRAHFKILPCLLNLSP\ntranslation: LTRVERRAHFKILPCLLNLSP\ntranslation: LDHSEGFSVLQEELCTAQHKQK\ntranslation: LDHSEGFSVLQEELCTAQHKQK\ntranslation: WIIVKGSRYFKRSYALHSTSRK\ntranslation: WIIVKGSRYFKRSYALHSTSRK\ntranslation: LSACAVQCITPLEVPRTLHYDP\ntranslation: LSACAVQCITPLEVPRTLHYDP\ntranslation: SCDYKDTCIRMFIAALFTIAK\ntranslation: SCDYKDTCIRMFIAALFTIAK\ntranslation: TFCHATNTRLNYMHKIPNPGS\ntranslation: TFCHATNTRLNYMHKIPNPGS\ntranslation: FNPTCPCYLIRADDPSP\ntranslation: FNPTCPCYLIRADDPSP\ntranslation: KNQQGEKLTTSQLKLVKQNSLYS\ntranslation: KNQQGEKLTTSQLKLVKQNSLYS\ntranslation: KPTRRKIDYITTEISKTEFSLF\ntranslation: KPTRRKIDYITTEISKTEFSLF\ntranslation: AFSSQTEVWAEKASRGSHSLYHM\ntranslation: AFSSQTEVWAEKASRGSHSLYHM\ntranslation: PFPPRQKSGQRKPLEVPIPSIT\ntranslation: PFPPRQKSGQRKPLEVPIPSIT\ntranslation: LFLPDRSLGRESLSRFPFPLSH\ntranslation: LFLPDRSLGRESLSRFPFPLSH\ntranslation: HVIEGMGTSRGFLCPDFCLGGKG\ntranslation: HVIEGMGTSRGFLCPDFCLGGKG\ntranslation: CDRGNGNLERLSLPRLLSGRKR\ntranslation: CDRGNGNLERLSLPRLLSGRKR\ntranslation: SESTQQTRILHSSAILSVVSIN\ntranslation: SESTQQTRILHSSAILSVVSIN\ntranslation: LNPLNKLEFFTVVLFSVLLAST\ntranslation: LNPLNKLEFFTVVLFSVLLAST\ntranslation: VDANNTENSTTVKNSSLLSGFR\ntranslation: VDANNTENSTTVKNSSLLSGFR\ntranslation: HNSILIFFF\ntranslation: HNSILIFFF\ntranslation: QLYSNFFF\ntranslation: QLYSNFFF\ntranslation: KKKIRIEL\ntranslation: KKKIRIEL\ntranslation: LLSTLVSKPGFKDWRRASGT\ntranslation: LLSTLVSKPGFKDWRRASGT\ntranslation: VPEALLQSLKPGLLTKVLSN\ntranslation: VPEALLQSLKPGLLTKVLSN\ntranslation: CNFLEEEIRSS\ntranslation: CNFLEEEIRSS\ntranslation: ATFLKKKSEA\ntranslation: ATFLKKKSEA\ntranslation: GASDFFFKKVA\ntranslation: GASDFFFKKVA\ntranslation: ELLISSSRKL\ntranslation: ELLISSSRKL\ntranslation: CCCFVCVCVVVVVVVVAVVLCVC\ntranslation: CCCFVCVCVVVVVVVVAVVLCVC\ntranslation: AVVLCVCVLLLLLLLLLLFCVCV\ntranslation: AVVLCVCVLLLLLLLLLLFCVCV\ntranslation: LLFCVCVCCCCCCCCCCCFVCV\ntranslation: LLFCVCVCCCCCCCCCCCFVCV\ntranslation: HTHTKQQQQQQQQQQHTHTQNNS\ntranslation: HTHTKQQQQQQQQQQHTHTQNNS\ntranslation: THTQNNSNNNNNNNNTHTHKTTA\ntranslation: THTQNNSNNNNNNNNTHTHKTTA\ntranslation: HTHKTTATTTTTTTTHTHTKQQ\ntranslation: HTHKTTATTTTTTTTHTHTKQQ\ntranslation: GTCDCHWKLGRGLGLNSQPVESD\ntranslation: GTCDCHWKLGRGLGLNSQPVESD\ntranslation: EVSPLLGSRWTTRSAWPTPSS\ntranslation: EVSPLLGSRWTTRSAWPTPSS\ntranslation: RYHLFWAQDGQQEAPGLRHH\ntranslation: RYHLFWAQDGQQEAPGLRHH\ntranslation: GITSSGLKMDNKKRLAYAII\ntranslation: GITSSGLKMDNKKRLAYAII\ntranslation: DDGVGQALLVVHLEPRRGDT\ntranslation: DDGVGQALLVVHLEPRRGDT\ntranslation: RNDPLIFTTIHVLLVC\ntranslation: RNDPLIFTTIHVLLVC\ntranslation: AYEENMYGSKYQWIIP\ntranslation: AYEENMYGSKYQWIIP\ntranslation: HTRRTCMVVNISGSFR\ntranslation: HTRRTCMVVNISGSFR\ntranslation: LGWVFSWVFVLFISNTEKVLWAL\ntranslation: LGWVFSWVFVLFISNTEKVLWAL\ntranslation: WVGFFHGFLFCLFRILKRSFGL\ntranslation: WVGFFHGFLFCLFRILKRSFGL\ntranslation: RAQRTFSVFEINKTKTHEKTQP\ntranslation: RAQRTFSVFEINKTKTHEKTQP\ntranslation: VSVMVSVCKFSYCSCAPRCTHLS\ntranslation: VSVMVSVCKFSYCSCAPRCTHLS\ntranslation: SQSWLVFASLVTVAAHPGVHTF\ntranslation: SQSWLVFASLVTVAAHPGVHTF\ntranslation: FEVIAVPRRWRLSKFWRTTFR\ntranslation: FEVIAVPRRWRLSKFWRTTFR\ntranslation: PKGRPPKLTQPPPPRNSNDLK\ntranslation: PKGRPPKLTQPPPPRNSNDLK\ntranslation: LKVVLQNLLNLHLLGTAMTSK\ntranslation: LKVVLQNLLNLHLLGTAMTSK\ntranslation: KIGETLRSRQSVEAQYILFAL\ntranslation: KIGETLRSRQSVEAQYILFAL\ntranslation: KQIRYTEPQQIACSSESLLFL\ntranslation: KQIRYTEPQQIACSSESLLFL\ntranslation: LAAAPDRGLASLPFLLVWSSL\ntranslation: LAAAPDRGLASLPFLLVWSSL\ntranslation: SQQHQTEDSLLFPFCLFGVLW\ntranslation: SQQHQTEDSLLFPFCLFGVLW\ntranslation: RSSTRPRTRFSSLSACLEFF\ntranslation: RSSTRPRTRFSSLSACLEFF\ntranslation: PKNSKQAEREEKRVLGLVLLR\ntranslation: PKNSKQAEREEKRVLGLVLLR\ntranslation: QRTPNKQKGKRSESSVWCCCE\ntranslation: QRTPNKQKGKRSESSVWCCCE\ntranslation: KELQTSRKGREASPRSGAAA\ntranslation: KELQTSRKGREASPRSGAAA\ntranslation: CVLYDVNNEWMLLMPLVLFWTSP\ntranslation: CVLYDVNNEWMLLMPLVLFWTSP\ntranslation: GEVQKSTNGISNIHSLFTSYNTH\ntranslation: GEVQKSTNGISNIHSLFTSYNTH\ntranslation: VKSKKALTASVTSIHCLHRTIHT\ntranslation: VKSKKALTASVTSIHCLHRTIHT\ntranslation: GKLKSSLKHLVF\ntranslation: GKLKSSLKHLVF\ntranslation: KTRCFKLDLSF\ntranslation: KTRCFKLDLSF\ntranslation: TILGLMCCECCFFFSFVKIFK\ntranslation: TILGLMCCECCFFFSFVKIFK\ntranslation: FKYFNKGKKKATFTAHQAQNS\ntranslation: FKYFNKGKKKATFTAHQAQNS\ntranslation: LNILTKEKKKQHSQHIKPKIV\ntranslation: LNILTKEKKKQHSQHIKPKIV\ntranslation: QCITIFSFHLVPILSSFLSIL\ntranslation: QCITIFSFHLVPILSSFLSIL\ntranslation: VYNHLLFSSCSNSLLFPFHS\ntranslation: VYNHLLFSSCSNSLLFPFHS\ntranslation: QNGKERGENWNKMKREDGYTL\ntranslation: QNGKERGENWNKMKREDGYTL\ntranslation: EWKGKRRELEQDEKRRWLYT\ntranslation: EWKGKRRELEQDEKRRWLYT\ntranslation: AEKKRYLQTVAVIVQVLHQIS\ntranslation: AEKKRYLQTVAVIVQVLHQIS\ntranslation: GEEKIFTNSCRDCASVAPNFK\ntranslation: GEEKIFTNSCRDCASVAPNFK\ntranslation: FEIWCNTCTITATVCKYLFFS\ntranslation: FEIWCNTCTITATVCKYLFFS\ntranslation: LKFGATLAQSRQLFVNIFSSP\ntranslation: LKFGATLAQSRQLFVNIFSSP\ntranslation: SSNFNNWDSTTEELLTSVCTFF\ntranslation: SSNFNNWDSTTEELLTSVCTFF\ntranslation: KECTHRSQELFCGGIPIVKVTA\ntranslation: KECTHRSQELFCGGIPIVKVTA\ntranslation: KNVHTEVKSSSVVESQLLKLLL\ntranslation: KNVHTEVKSSSVVESQLLKLLL\ntranslation: KQVSPGAAGLLSSL\ntranslation: KQVSPGAAGLLSSL\ntranslation: SKSHLVLQDFSAPC\ntranslation: SKSHLVLQDFSAPC\ntranslation: GKELRSPAAPGETCL\ntranslation: GKELRSPAAPGETCL\ntranslation: GPLSKNWEAAKHNVSGCECENQ\ntranslation: GPLSKNWEAAKHNVSGCECENQ\ntranslation: SDFRIHSLKHYALQLPSSYSVDP\ntranslation: SDFRIHSLKHYALQLPSSYSVDP\ntranslation: SIRDQVRPLVLCMVIICFEVSL\ntranslation: SIRDQVRPLVLCMVIICFEVSL\ntranslation: CDPAWATKQDPVSNKKVSTKL\ntranslation: CDPAWATKQDPVSNKKVSTKL\ntranslation: QFRTHFFVRDRVLLCCPGWIT\ntranslation: QFRTHFFVRDRVLLCCPGWIT\ntranslation: SFVLTFLLETGSCFVAQAGSQ\ntranslation: SFVLTFLLETGSCFVAQAGSQ\ntranslation: VMGMETYHISNARVSGRKFFH\ntranslation: VMGMETYHISNARVSGRKFFH\ntranslation: MKKLPTTHPSITYMICLHTHY\ntranslation: MKKLPTTHPSITYMICLHTHY\ntranslation: SRVLTHSSINPKVQVQSLIQDK\ntranslation: SRVLTHSSINPKVQVQSLIQDK\ntranslation: RFKLEWLGNCSGLNDETYGYKE\ntranslation: RFKLEWLGNCSGLNDETYGYKE\ntranslation: HQFEKSPYQKNMNSAKIEAMLPE\ntranslation: HQFEKSPYQKNMNSAKIEAMLPE\ntranslation: INCNKSVLISSLALILRLCYSKF\ntranslation: INCNKSVLISSLALILRLCYSKF\ntranslation: PILITMQSNGFKAYIFHIIFSA\ntranslation: PILITMQSNGFKAYIFHIIFSA\ntranslation: NCMFFFHLDNLGSVESYLLTCY\ntranslation: NCMFFFHLDNLGSVESYLLTCY\ntranslation: FQIYLYLALRQQIEDSEFKW\ntranslation: FQIYLYLALRQQIEDSEFKW\ntranslation: NEELFSGWFSYQLFTQQSSKP\ntranslation: NEELFSGWFSYQLFTQQSSKP\ntranslation: TKSFFLDGFLISFSHNSQVNL\ntranslation: TKSFFLDGFLISFSHNSQVNL\ntranslation: LHRLSHFYHNNPTVKLSLRT\ntranslation: LHRLSHFYHNNPTVKLSLRT\ntranslation: SSQRQLHSRIIVIKMTQAMQ\ntranslation: SSQRQLHSRIIVIKMTQAMQ\ntranslation: RLFSPDIDHITECTSHQLFS\ntranslation: RLFSPDIDHITECTSHQLFS\ntranslation: HENAMLQTENNNLRVRIKAMQE\ntranslation: HENAMLQTENNNLRVRIKAMQE\ntranslation: VWMPFISCLIAVSRTSSTLLNY\ntranslation: VWMPFISCLIAVSRTSSTLLNY\ntranslation: VIQQSTRSPRYSNQTRNKGHPN\ntranslation: VIQQSTRSPRYSNQTRNKGHPN\ntranslation: NNDISSLPYSLGNLHLKFLALEGNP\ntranslation: NNDISSLPYSLGNLHLKFLALEGNP\ntranslation: LPCTIPMRHNYNKLHLPTTNR\ntranslation: LPCTIPMRHNYNKLHLPTTNR\ntranslation: SVCRRQMELVIIMPHRDSTRK\ntranslation: SVCRRQMELVIIMPHRDSTRK\ntranslation: FKITALHLVLQICFKARQVCIKP\ntranslation: FKITALHLVLQICFKARQVCIKP\ntranslation: SKSLPSILSFRSASKRARSASSL\ntranslation: SKSLPSILSFRSASKRARSASSL\ntranslation: QNHCPPSCPSDLLQSAPGLHQAF\ntranslation: QNHCPPSCPSDLLQSAPGLHQAF\ntranslation: RLDADLARFEADLKDKMEGSDFE\ntranslation: RLDADLARFEADLKDKMEGSDFE\ntranslation: RLCCCASYCCLMGLYLKCIVLMF\ntranslation: RLCCCASYCCLMGLYLKCIVLMF\ntranslation: IVLLCFILLPYGIILEMHCADV\ntranslation: IVLLCFILLPYGIILEMHCADV\ntranslation: NISTMHFKYNPIRQQYEAQQHN\ntranslation: NISTMHFKYNPIRQQYEAQQHN\ntranslation: VFVPQPPQCAHHKIQTQKTQQFF\ntranslation: VFVPQPPQCAHHKIQTQKTQQFF\ntranslation: SLFRNHHNALIIKYRHRKHSNS\ntranslation: SLFRNHHNALIIKYRHRKHSNS\ntranslation: EELLCFLCLYFMMSALWWLRNKD\ntranslation: EELLCFLCLYFMMSALWWLRNKD\ntranslation: RIAVFSVSVFYDERIVVVAEQR\ntranslation: RIAVFSVSVFYDERIVVVAEQR\ntranslation: ECVYVCVWNDSVCVWNACVC\ntranslation: ECVYVCVWNDSVCVWNACVC\ntranslation: NVSMCVYGMILCVYGMRVSV\ntranslation: NVSMCVYGMILCVYGMRVSV\ntranslation: HRHTHSIHTQNHSIHTHRHI\ntranslation: HRHTHSIHTQNHSIHTHRHI\ntranslation: TDTRIPYTHRIIPYTHIDTF\ntranslation: TDTRIPYTHRIIPYTHIDTF\ntranslation: EDGNHQFWQCFTGRSKARNFH\ntranslation: EDGNHQFWQCFTGRSKARNFH\ntranslation: KTETISFGSVSPGGVKLEIST\ntranslation: KTETISFGSVSPGGVKLEIST\ntranslation: VEISSFTPPGETLPKLMVSVF\ntranslation: VEISSFTPPGETLPKLMVSVF\ntranslation: GDLYCDHAELIAVCCATTTSCS\ntranslation: GDLYCDHAELIAVCCATTTSCS\ntranslation: NMRWWWRSRLRSILHGHSRDP\ntranslation: NMRWWWRSRLRSILHGHSRDP\ntranslation: LISVYLASSSSSSSSSVFHFL\ntranslation: LISVYLASSSSSSSSSVFHFL\ntranslation: DFCLLSIIFIIIIIICISFSL\ntranslation: DFCLLSIIFIIIIIICISFSL\ntranslation: KENEIQMMMMMMKMMLSKQKS\ntranslation: KENEIQMMMMMMKMMLSKQKS\ntranslation: HLQISHQQPTNHHPTMTNQTNL\ntranslation: HLQISHQQPTNHHPTMTNQTNL\ntranslation: EVSLISHCWVVISRLLMRYLEV\ntranslation: EVSLISHCWVVISRLLMRYLEV\ntranslation: LALKGEWEFRGQKIDPREYVVPT\ntranslation: LALKGEWEFRGQKIDPREYVVPT\ntranslation: VGTTYSLGSIFCPRNSHSPFKAK\ntranslation: VGTTYSLGSIFCPRNSHSPFKAK\ntranslation: LEPHTLWGLFFAHGIPIHPLRL\ntranslation: LEPHTLWGLFFAHGIPIHPLRL\ntranslation: KMLVFTDQEGNEVRRQTF\ntranslation: KMLVFTDQEGNEVRRQTF\ntranslation: NQANRFFSLIDLFSFDQRLKF\ntranslation: NQANRFFSLIDLFSFDQRLKF\ntranslation: KLYLKSKWTDICKVLINKCSTNRY\ntranslation: KLYLKSKWTDICKVLINKCSTNRY\ntranslation: YLFVEHLLIKTLHMSVHFDLRYNF\ntranslation: YLFVEHLLIKTLHMSVHFDLRYNF\ntranslation: ATTTEQQQSMLSIETLSRLSYT\ntranslation: ATTTEQQQSMLSIETLSRLSYT\ntranslation: SVSWYHMLLSLQTVKRFHFHLE\ntranslation: SVSWYHMLLSLQTVKRFHFHLE\ntranslation: QDENGIVSQSEVIRAYDTTKQ\ntranslation: QDENGIVSQSEVIRAYDTTKQ\ntranslation: KTLLPEDTQKLDNFLGEVRDKWD\ntranslation: KTLLPEDTQKLDNFLGEVRDKWD\ntranslation: GVFTYVGALFNGLTLLILALISL\ntranslation: GVFTYVGALFNGLTLLILALISL\ntranslation: VMCYLQILQNTHIHTHTHLPLCS\ntranslation: VMCYLQILQNTHIHTHTHLPLCS\ntranslation: SCAIYRFYKTHTYTHIHTCPYVL\ntranslation: SCAIYRFYKTHTYTHIHTCPYVL\ntranslation: HVLFTDFTKHTHTHTYTLALMF\ntranslation: HVLFTDFTKHTHTHTYTLALMF\ntranslation: VMSFYCALLSNHLCDFNKH\ntranslation: VMSFYCALLSNHLCDFNKH\ntranslation: ENECQSRKGHLRICSMKEKK\ntranslation: ENECQSRKGHLRICSMKEKK\ntranslation: HNCSSPVCQNHSGFQVNGQR\ntranslation: HNCSSPVCQNHSGFQVNGQR\ntranslation: QLQLSCMSKSFWFPGKWTK\ntranslation: QLQLSCMSKSFWFPGKWTK\ntranslation: PLSIYLETRMILTYRRAAVV\ntranslation: PLSIYLETRMILTYRRAAVV\ntranslation: FVHLPGNQNDFDIQESCSC\ntranslation: FVHLPGNQNDFDIQESCSC\ntranslation: WVWFNPPQLPAMMDKIERVR\ntranslation: WVWFNPPQLPAMMDKIERVR\ntranslation: PHSLNLIHHSRQLRWIKPNP\ntranslation: PHSLNLIHHSRQLRWIKPNP\ntranslation: QELENHDSFHPQIKPTQKYTTET\ntranslation: QELENHDSFHPQIKPTQKYTTET\ntranslation: KSLKIMIHFIHRSNQHRNIRQKL\ntranslation: KSLKIMIHFIHRSNQHRNIRQKL\ntranslation: EREEALKQHKTLSQELVNLRGEL\ntranslation: EREEALKQHKTLSQELVNLRGEL\ntranslation: FNVFSWVMRNSVRSMGVIMVGH\ntranslation: FNVFSWVMRNSVRSMGVIMVGH\ntranslation: MAHHNYPHTPYTIPHHPTKNIK\ntranslation: MAHHNYPHTPYTIPHHPTKNIK\ntranslation: WPTIITPILLTLFLITQLKTLN\ntranslation: WPTIITPILLTLFLITQLKTLN\ntranslation: RLFRRKLDPFPDDPHELSHSDI\ntranslation: RLFRRKLDPFPDDPHELSHSDI\ntranslation: DITVAQFMWIIRKRIQLPSEKA\ntranslation: DITVAQFMWIIRKRIQLPSEKA\ntranslation: ISLWLSSCGSSGKGSSFLLKR\ntranslation: ISLWLSSCGSSGKGSSFLLKR\ntranslation: HVQLLFTWNPSPLRPSKFSFEYL\ntranslation: HVQLLFTWNPSPLRPSKFSFEYL\ntranslation: MFNCCSHGTLLHFGLQSSRLNI\ntranslation: MFNCCSHGTLLHFGLQSSRLNI\ntranslation: EETHLSFKNKNYQKHKNKREM\ntranslation: EETHLSFKNKNYQKHKNKREM\ntranslation: EIQGKLEEKKPPEADMNIFED\ntranslation: EIQGKLEEKKPPEADMNIFED\ntranslation: VFKNIHVSLRRFLLFQLPLNL\ntranslation: VFKNIHVSLRRFLLFQLPLNL\ntranslation: LQKYSCQPQEVSSLPASPESQ\ntranslation: LQKYSCQPQEVSSLPASPESQ\ntranslation: LCVIILSCGFESAVLPRSKLEV\ntranslation: LCVIILSCGFESAVLPRSKLEV\ntranslation: SVSSFSVVGLKVLFFHGPNLKS\ntranslation: SVSSFSVVGLKVLFFHGPNLKS\ntranslation: VTSSLDRGRTALSNPQLRMMTQR\ntranslation: VTSSLDRGRTALSNPQLRMMTQR\ntranslation: DFKFGPWKNSTFKPTTENDDTE\ntranslation: DFKFGPWKNSTFKPTTENDDTE\ntranslation: MGVPLIGVIIRYSGKRGDFRDP"
      },
      {
       "name": "stderr",
       "output_type": "stream",
       "text": "491it [00:00, 1053.95it/s]"
      },
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "\ntranslation: MGVPLIGVIIRYSGKRGDFRDP\ntranslation: QGSLKSPLFPEYLMITPIKGTPI\ntranslation: QGSLKSPLFPEYLMITPIKGTPI\ntranslation: LRVHHRTERVGENRPSPKEEQV\ntranslation: LRVHHRTERVGENRPSPKEEQV\ntranslation: TCSSLGEGRFSPTLSVLWWTLN\ntranslation: TCSSLGEGRFSPTLSVLWWTLN\ntranslation: ISHCLVNNINSTFSQHVEYCFL\ntranslation: ISHCLVNNINSTFSQHVEYCFL\ntranslation: RLSSRYIIPLAIWTHQLSSSCD\ntranslation: RLSSRYIIPLAIWTHQLSSSCD\ntranslation: SQLLLNWCVQIAKGMMYLEERR\ntranslation: SQLLLNWCVQIAKGMMYLEERR\ntranslation: LSQGTKTCSGKNIFTLSYMLS\ntranslation: LSQGTKTCSGKNIFTLSYMLS\ntranslation: FTRYQDMFWQKHFHFKLYAV\ntranslation: FTRYQDMFWQKHFHFKLYAV\ntranslation: IKDTSIRDQVRPLVLCMVIICF\ntranslation: IKDTSIRDQVRPLVLCMVIICF\ntranslation: SALNFGKQLLCSGSPTWSNCYF\ntranslation: SALNFGKQLLCSGSPTWSNCYF\ntranslation: NSNLTKLDFHYKAAVFQSSML\ntranslation: NSNLTKLDFHYKAAVFQSSML\ntranslation: VFCIVLVAQLRMRILSPVTF\ntranslation: VFCIVLVAQLRMRILSPVTF\ntranslation: NVTGERILILSCATNTMQK\ntranslation: NVTGERILILSCATNTMQK\ntranslation: HFKSTRTHNSYNRHQPTTPSI\ntranslation: HFKSTRTHNSYNRHQPTTPSI\ntranslation: TSSQLGLIIVTIGINQPHLAF\ntranslation: TSSQLGLIIVTIGINQPHLAF\ntranslation: SCTSASSSFTFSAIFTCSFHFSS\ntranslation: SCTSASSSFTFSAIFTCSFHFSS\ntranslation: PVLQLVQVLHFQQFSLVLSISAP\ntranslation: PVLQLVQVLHFQQFSLVLSISAP\ntranslation: MELKWKEQVKIAENVKLELAEVQ\ntranslation: MELKWKEQVKIAENVKLELAEVQ\ntranslation: TCPKTSSLTVKVRSKMSVMLLSF\ntranslation: TCPKTSSLTVKVRSKMSVMLLSF\ntranslation: KDNNITDILDLTFTVNEEVFGQV\ntranslation: KDNNITDILDLTFTVNEEVFGQV\ntranslation: QLSPKQCSGAKCSKEKSSSARRW\ntranslation: QLSPKQCSGAKCSKEKSSSARRW\ntranslation: SYPQNSVVGQNVQKKNHPVHVD\ntranslation: SYPQNSVVGQNVQKKNHPVHVD\ntranslation: HLRALDDFSFEHFAPLHCFGDS\ntranslation: HLRALDDFSFEHFAPLHCFGDS\ntranslation: IYVHWMIFLLNILPHYTVLGIA\ntranslation: IYVHWMIFLLNILPHYTVLGIA\ntranslation: LSPSCTFFSNTKLKTTSNTLFS\ntranslation: LSPSCTFFSNTKLKTTSNTLFS\ntranslation: EKSVLDVVFNFVLLKKVHDGE\ntranslation: EKSVLDVVFNFVLLKKVHDGE\ntranslation: ATRRGIDNKFQQPNPFLQRFQ\ntranslation: ATRRGIDNKFQQPNPFLQRFQ\ntranslation: LREEALITSFNSQILFYRDF\ntranslation: LREEALITSFNSQILFYRDF\ntranslation: RIYIFFVCVFVNYPTDIECTL\ntranslation: RIYIFFVCVFVNYPTDIECTL\ntranslation: EYIFFSCVFLLTTLQILNAP\ntranslation: EYIFFSCVFLLTTLQILNAP\ntranslation: QGAFNICRVVNKNTHEKNIYS\ntranslation: QGAFNICRVVNKNTHEKNIYS\ntranslation: TSLFRIAVKRSWQRL\ntranslation: TSLFRIAVKRSWQRL\ntranslation: HNLCQLLFTAILNKLV\ntranslation: HNLCQLLFTAILNKLV\ntranslation: QPLPTSLHCNSKQTC\ntranslation: QPLPTSLHCNSKQTC\ntranslation: VFLGYFELSFNCKWKVTTESMI\ntranslation: VFLGYFELSFNCKWKVTTESMI\ntranslation: HKRPRYIAQNHLKISDFKSMIP\ntranslation: HKRPRYIAQNHLKISDFKSMIP\ntranslation: NKPILKILSTYVFYNILTMSLEK\ntranslation: NKPILKILSTYVFYNILTMSLEK\ntranslation: ENALQSISNSYTENFTYFYIL\ntranslation: ENALQSISNSYTENFTYFYIL\ntranslation: KMPCKVSAIHTQKISRTSIYC\ntranslation: KMPCKVSAIHTQKISRTSIYC\ntranslation: KCLAKYQQFIHRKFHVLLYI\ntranslation: KCLAKYQQFIHRKFHVLLYI\ntranslation: HLTNYSKTTDHTSVYFLLVPSEL\ntranslation: HLTNYSKTTDHTSVYFLLVPSEL\ntranslation: LRSWVWFNPPQLPAMMDKIE\ntranslation: LRSWVWFNPPQLPAMMDKIE\ntranslation: LNLIHHSRQLRWIKPNPATQ\ntranslation: LNLIHHSRQLRWIKPNPATQ\ntranslation: LMKDNAEITGWRVTSDREVLHY\ntranslation: LMKDNAEITGWRVTSDREVLHY\ntranslation: FHDADEVLQAKHVASVWPIMH\ntranslation: FHDADEVLQAKHVASVWPIMH\ntranslation: VHDGPDRGYVFGLENLISIME\ntranslation: VHDGPDRGYVFGLENLISIME\ntranslation: CMMGQTEATCLAWRTSSASWN\ntranslation: CMMGQTEATCLAWRTSSASWN\ntranslation: KPVCRQFSLLRHGSSFTLAPLK\ntranslation: KPVCRQFSLLRHGSSFTLAPLK\ntranslation: GFKGASVKLLPCLSKENCLHTG\ntranslation: GFKGASVKLLPCLSKENCLHTG\ntranslation: LQRGQCKTASMSKQRKLPTYWF\ntranslation: LQRGQCKTASMSKQRKLPTYWF\ntranslation: SCSRFEQLLCTVQI\ntranslation: SCSRFEQLLCTVQI\ntranslation: LVPDLSNFYAQYK\ntranslation: LVPDLSNFYAQYK\ntranslation: ICTVHRSCSNREQ\ntranslation: ICTVHRSCSNREQ\ntranslation: FVLCIEVAQIGNK\ntranslation: FVLCIEVAQIGNK\ntranslation: RIHSEIVKQSVFKDFYELEPE\ntranslation: RIHSEIVKQSVFKDFYELEPE\ntranslation: LWLQFIKILKDRLFHDLRVDP\ntranslation: LWLQFIKILKDRLFHDLRVDP\ntranslation: VTYPEVLFTVWCSTYLHFSKRKQ\ntranslation: VTYPEVLFTVWCSTYLHFSKRKQ\ntranslation: SHIRKFCLQYGAALIYTSVKENK\ntranslation: SHIRKFCLQYGAALIYTSVKENK\ntranslation: FNFVWAVFYQTKLLAFEMIYM\ntranslation: FNFVWAVFYQTKLLAFEMIYM\ntranslation: CVCICVFQCFIRSFHLWE\ntranslation: CVCICVFQCFIRSFHLWE\ntranslation: VCVFVCFSVSSGPSISGS\ntranslation: VCVFVCFSVSSGPSISGS\ntranslation: CVYLCVSVFHQVLPSLGV\ntranslation: CVYLCVSVFHQVLPSLGV\ntranslation: LPEMEGPDETLKHTNTHT\ntranslation: LPEMEGPDETLKHTNTHT\ntranslation: QRVSQEEVKKWAESLENLISHEC\ntranslation: QRVSQEEVKKWAESLENLISHEC\ntranslation: RSVSFSCNLIYIFCFCCLCFWGH\ntranslation: RSVSFSCNLIYIFCFCCLCFWGH\ntranslation: EVFHFHVISSTSFAFVAYVFGV\ntranslation: EVFHFHVISSTSFAFVAYVFGV\ntranslation: NGKFKLPTKNEFIPTNFEILPL\ntranslation: NGKFKLPTKNEFIPTNFEILPL\ntranslation: SNGKISKFVGINSFFVGSLNFPF\ntranslation: SNGKISKFVGINSFFVGSLNFPF\ntranslation: HPCTSSSQHEVCWKSTG\ntranslation: HPCTSSSQHEVCWKSTG\ntranslation: THALHLVSMRYVGNLLD\ntranslation: THALHLVSMRYVGNLLD\ntranslation: HRGPDAFRFENVNGYTNCCFGFH\ntranslation: HRGPDAFRFENVNGYTNCCFGFH\ntranslation: TEVQMHSVLRMSMDTPTAALDF\ntranslation: TEVQMHSVLRMSMDTPTAALDF\ntranslation: VKSKAAVGVSIDILKTECIWTSV\ntranslation: VKSKAAVGVSIDILKTECIWTSV\ntranslation: NHVVFTLSGQTRRFFKQVETNYL\ntranslation: NHVVFTLSGQTRRFFKQVETNYL\ntranslation: CLFKFCLFTLLYMIVCTEGAP\ntranslation: CLFKFCLFTLLYMIVCTEGAP\ntranslation: GAPSVQTIIYRSVNKQNLKRQ\ntranslation: GAPSVQTIIYRSVNKQNLKRQ\ntranslation: CMCGVYVCYRLMPSWNCVNVLL\ntranslation: CMCGVYVCYRLMPSWNCVNVLL\ntranslation: FSGFPLCGETLCCCCFGCLCCS\ntranslation: FSGFPLCGETLCCCCFGCLCCS\ntranslation: FLVFLCAVKLCVAVALAVCAVV\ntranslation: FLVFLCAVKLCVAVALAVCAVV\ntranslation: PLQHRQPKQQQHKVSPHKGKPEK\ntranslation: PLQHRQPKQQQHKVSPHKGKPEK\ntranslation: HYSTDSQSNSNTKFHRTKENQK\ntranslation: HYSTDSQSNSNTKFHRTKENQK\ntranslation: TTAQTAKATATQSFTAQRKTRK\ntranslation: TTAQTAKATATQSFTAQRKTRK\ntranslation: DEYLTCIVILCYIVTARWPER\ntranslation: DEYLTCIVILCYIVTARWPER\ntranslation: ITFTKGPFHNAIWLLKTLNTNYH\ntranslation: ITFTKGPFHNAIWLLKTLNTNYH\ntranslation: ESLNSLQYVSNISVGGYQNKTNA\ntranslation: ESLNSLQYVSNISVGGYQNKTNA\ntranslation: PHKAIPSLSELAEPSAENTNLG\ntranslation: PHKAIPSLSELAEPSAENTNLG\ntranslation: LIFIFLLVLFPILFALFFLYITLF\ntranslation: LIFIFLLVLFPILFALFFLYITLF\ntranslation: LSLSFSLSSFLSFLLSSFFTSLF\ntranslation: LSLSFSLSSFLSFLLSSFFTSLF\ntranslation: YLYLSPCPLSYPFCSLLSLHHSF\ntranslation: YLYLSPCPLSYPFCSLLSLHHSF\ntranslation: EKSDVKKEESKKDRKEDKEKDKDK\ntranslation: EKSDVKKEESKKDRKEDKEKDKDK\ntranslation: EENLRKFDIYFGSVIKQLRLLKF\ntranslation: EENLRKFDIYFGSVIKQLRLLKF\ntranslation: YRKKKYMYNSKLNFVCIGVC\ntranslation: YRKKKYMYNSKLNFVCIGVC\ntranslation: SILLSFFLVDRFELTEWVDILF\ntranslation: SILLSFFLVDRFELTEWVDILF\ntranslation: PFCCRFSLLIDLNLLNGLISFS\ntranslation: PFCCRFSLLIDLNLLNGLISFS\ntranslation: QKRISTHSVSSNLSTRKNDSRME\ntranslation: QKRISTHSVSSNLSTRKNDSRME\ntranslation: EKDINPFSKFKSINKEKRQQNG\ntranslation: EKDINPFSKFKSINKEKRQQNG\ntranslation: HCTRFVQQSIIKGFLSKPYVDS\ntranslation: HCTRFVQQSIIKGFLSKPYVDS\ntranslation: LHSVCPTEYNQGFSFKALRRQL\ntranslation: LHSVCPTEYNQGFSFKALRRQL\ntranslation: LTLTTKLTNTNISDAQEIETV\ntranslation: LTLTTKLTNTNISDAQEIETV\ntranslation: HHLIISSKIYVWLKLCQMD\ntranslation: HHLIISSKIYVWLKLCQMD\ntranslation: PFNHFQQNICLAEIMSNG\ntranslation: PFNHFQQNICLAEIMSNG\ntranslation: HHPMRQPARTPERRHILPILHP\ntranslation: HHPMRQPARTPERRHILPILHP\ntranslation: AYTILRSVPNKLGGVLALLL\ntranslation: AYTILRSVPNKLGGVLALLL\ntranslation: VKKGRFGWGWGVLGRKIRIK\ntranslation: VKKGRFGWGWGVLGRKIRIK\ntranslation: YFILIFLPKTPHPQPNLPFLT\ntranslation: YFILIFLPKTPHPQPNLPFLT\ntranslation: FYSNLSSQNTPPPTKSSLLN\ntranslation: FYSNLSSQNTPPPTKSSLLN\ntranslation: SCTEALYNIGLTYEKLNRLD\ntranslation: SCTEALYNIGLTYEKLNRLD\ntranslation: SWGFSSLLRNLILIPEANFFSLS\ntranslation: SWGFSSLLRNLILIPEANFFSLS\ntranslation: MLREKKFASGIKIRFLRREENPQL\ntranslation: MLREKKFASGIKIRFLRREENPQL\ntranslation: SPGEASSETLHSHSEHGSSQ\ntranslation: SPGEASSETLHSHSEHGSSQ\ntranslation: RQERPPVKPFILTRNMAQA\ntranslation: RQERPPVKPFILTRNMAQA\ntranslation: GLSHVPSENEGFHWRPLLA\ntranslation: GLSHVPSENEGFHWRPLLA\ntranslation: ILIIAHGLTSSLLFCLANSNYER\ntranslation: ILIIAHGLTSSLLFCLANSNYER\ntranslation: SHNRPRTYILITILPSKLKLRT\ntranslation: SHNRPRTYILITILPSKLKLRT\ntranslation: SCSTMSSDEFANTMPVRPPTV\ntranslation: SCSTMSSDEFANTMPVRPPTV\ntranslation: RVVRCLVMSLLIQCQLGHLR\ntranslation: RVVRCLVMSLLIQCQLGHLR\ntranslation: HRRWPNWHCISKLITRHRTTR\ntranslation: HRRWPNWHCISKLITRHRTTR\ntranslation: TVGGLTGIVLANSSLDIVLHD\ntranslation: TVGGLTGIVLANSSLDIVLHD\ntranslation: ENRLFPKSCLGSDSLCLCL\ntranslation: ENRLFPKSCLGSDSLCLCL\ntranslation: RRQRHRLSDPKQDLGNRRF\ntranslation: RRQRHRLSDPKQDLGNRRF\ntranslation: CDFALHHRCSVLLPAPVLLL\ntranslation: CDFALHHRCSVLLPAPVLLL\ntranslation: VILLCIIAVLCYCRLQYYCC\ntranslation: VILLCIIAVLCYCRLQYYCC\ntranslation: AAVVLEPAVAQNSDDAEQNH\ntranslation: AAVVLEPAVAQNSDDAEQNH\ntranslation: RLKEPLISNVDSRMMARVLIF\ntranslation: RLKEPLISNVDSRMMARVLIF\ntranslation: KINTLAIILLSTLLISGSFNL\ntranslation: KINTLAIILLSTLLISGSFNL\ntranslation: DLSRVMTMFYILGTSYMGICMYQ\ntranslation: DLSRVMTMFYILGTSYMGICMYQ\ntranslation: LVHAYTHVTSTQDIKHRHYSRKV\ntranslation: LVHAYTHVTSTQDIKHRHYSRKV\ntranslation: CATSSKSDVMGTWEMSEKCPKGF\ntranslation: CATSSKSDVMGTWEMSEKCPKGF\ntranslation: QNPFGHFSDISQVPITSDLELVA\ntranslation: QNPFGHFSDISQVPITSDLELVA\ntranslation: KSLRTFLRHLPSSHHVRFGAGSA\ntranslation: KSLRTFLRHLPSSHHVRFGAGSA\ntranslation: LWETFFALHTATIVHLCLQVN\ntranslation: LWETFFALHTATIVHLCLQVN\ntranslation: LGDLLCFAYSHHSTPVPPSEP\ntranslation: LGDLLCFAYSHHSTPVPPSEP\ntranslation: RFTWRHRCTMVAVCKAKKVSQ\ntranslation: RFTWRHRCTMVAVCKAKKVSQ\ntranslation: GSLGGTGVLWWLYAKQRRSPK\ntranslation: GSLGGTGVLWWLYAKQRRSPK\ntranslation: VHLEAQVYYGGCMQSKEGLPK\ntranslation: VHLEAQVYYGGCMQSKEGLPK\ntranslation: SYLRTCGVSCCFNHQCRSHPILK\ntranslation: SYLRTCGVSCCFNHQCRSHPILK\ntranslation: ATLGHVGFLAVSIISADPILSLR\ntranslation: ATLGHVGFLAVSIISADPILSLR\ntranslation: LKDRMGSALMIETARNPTCPKVA\ntranslation: LKDRMGSALMIETARNPTCPKVA\ntranslation: SKCMKICHVLKIKLTSVS\ntranslation: SKCMKICHVLKIKLTSVS\ntranslation: SAFLRQKARANCLSQHPKFTSFF\ntranslation: SAFLRQKARANCLSQHPKFTSFF\ntranslation: KGKRIMKRVIREL\ntranslation: KGKRIMKRVIREL\ntranslation: GKEDYEESHQRAV\ntranslation: GKEDYEESHQRAV\ntranslation: HSSLMTLFIILFP\ntranslation: HSSLMTLFIILFP\ntranslation: QLSDDSLHNPLSL\ntranslation: QLSDDSLHNPLSL\ntranslation: RRMTYSTVSPSLSQSLHPM\ntranslation: RRMTYSTVSPSLSQSLHPM\ntranslation: ENDVLHCVAFAVPKSSSN\ntranslation: ENDVLHCVAFAVPKSSSN\ntranslation: HWMKTLGQRRRHSGVRHSP\ntranslation: HWMKTLGQRRRHSGVRHSP\ntranslation: LDEDFGTAKATQWSTSFS\ntranslation: LDEDFGTAKATQWSTSFS\ntranslation: SNFFFTVCVFHFSCHHCQKFREVN\ntranslation: SNFFFTVCVFHFSCHHCQKFREVN\ntranslation: LISSSLSVSFIFLAIIVRNSGKS\ntranslation: LISSSLSVSFIFLAIIVRNSGKS\ntranslation: IDFPEFLTMMARKMKDTDSEEEIR\ntranslation: IDFPEFLTMMARKMKDTDSEEEIR\ntranslation: LPSTPFLNTHLIFTICFGHHYFP\ntranslation: LPSTPFLNTHLIFTICFGHHYFP\ntranslation: WKIMMSKANCEDQMCVKKRCRGE\ntranslation: WKIMMSKANCEDQMCVKKRCRGE\ntranslation: RKDRRAESVQTHSINPFYKVLIT\ntranslation: RKDRRAESVQTHSINPFYKVLIT\ntranslation: LYAPFNSSLSMCEVMSSSVA\ntranslation: LYAPFNSSLSMCEVMSSSVA\ntranslation: DATDEDITSHMESEELNGAYK\ntranslation: DATDEDITSHMESEELNGAYK\ntranslation: CYRRGHHLTHGKRGVEWCIQ\ntranslation: CYRRGHHLTHGKRGVEWCIQ\ntranslation: FMQFNEPSKEFTHLYTEFTSTS\ntranslation: FMQFNEPSKEFTHLYTEFTSTS\ntranslation: SCNLMNPQKNLHIFIQNSHLHQ\ntranslation: SCNLMNPQKNLHIFIQNSHLHQ\ntranslation: DDSPFTKAFNDLATVYCSSLISL\ntranslation: DDSPFTKAFNDLATVYCSSLISL\ntranslation: MTHHLQRPSMIWPLFTAQVSSL\ntranslation: MTHHLQRPSMIWPLFTAQVSSL\ntranslation: RYPHLQDLTKPDHKMLLYEEFP\ntranslation: RYPHLQDLTKPDHKMLLYEEFP\ntranslation: QEIPHKGAFCDLVWSDPEDVDT\ntranslation: QEIPHKGAFCDLVWSDPEDVDT\ntranslation: RKFLIKEHFVIWFGQILKMWIP\ntranslation: RKFLIKEHFVIWFGQILKMWIP\ntranslation: AFHNAWCHPLNNMFKDKSRHD\ntranslation: AFHNAWCHPLNNMFKDKSRHD\ntranslation: LFIMLGAIRLIICSKIKAGMI\ntranslation: LFIMLGAIRLIICSKIKAGMI\ntranslation: IMPAFIFEHIIKRMAPSIMKS\ntranslation: IMPAFIFEHIIKRMAPSIMKS\ntranslation: GNYHVNLNSSKIVLCLLDHDP\ntranslation: GNYHVNLNSSKIVLCLLDHDP\ntranslation: GLRVGHILGIMSLREII\ntranslation: GLRVGHILGIMSLREII\ntranslation: AACWAHPWHYEPERNH\ntranslation: AACWAHPWHYEPERNH\ntranslation: YDFSQAHNAKDVPNTQP\ntranslation: YDFSQAHNAKDVPNTQP\ntranslation: MISLRLIMPRMCPTRS\ntranslation: MISLRLIMPRMCPTRS\ntranslation: KTAQACLLQK\ntranslation: KTAQACLLQK\ntranslation: KQLRHVCCR\ntranslation: KQLRHVCCR\ntranslation: NSSGMFAAE\ntranslation: NSSGMFAAE\ntranslation: LLQQTCLSCF\ntranslation: LLQQTCLSCF\ntranslation: SAANMPELF\ntranslation: SAANMPELF\ntranslation: PPSVASSLNSSSVPNLRISSSS\ntranslation: PPSVASSLNSSSVPNLRISSSS\ntranslation: QELDDILKFGTEELFKDEATDGG\ntranslation: QELDDILKFGTEELFKDEATDGG\ntranslation: RSLMISSNLALRNYSRMKPLMEE\ntranslation: RSLMISSNLALRNYSRMKPLMEE\ntranslation: LVNSFLVRLFSLGDHVFSMCTCN\ntranslation: LVNSFLVRLFSLGDHVFSMCTCN\ntranslation: LRFQLSESQRLVINYYKITGD\ntranslation: LRFQLSESQRLVINYYKITGD\ntranslation: LKVYCKAVTLSNCCSATAYSEIY\ntranslation: LKVYCKAVTLSNCCSATAYSEIY\ntranslation: HIVFVLIAYQLSNCIDLPNYIF\ntranslation: HIVFVLIAYQLSNCIDLPNYIF\ntranslation: HSFENNFLFVLSLSRLTHTHT\ntranslation: HSFENNFLFVLSLSRLTHTHT\ntranslation: VCVYVSVCSEKGQKGNYFQNC\ntranslation: VCVYVSVCSEKGQKGNYFQNC\ntranslation: CVCMCQSAQRKDKKEIIFKTV\ntranslation: CVCMCQSAQRKDKKEIIFKTV\ntranslation: VCVCVSLLRERTKRKLFSKL\ntranslation: VCVCVSLLRERTKRKLFSKL\ntranslation: KDVHEDSLTVVFENNWQPER\ntranslation: KDVHEDSLTVVFENNWQPER\ntranslation: KMFMKTPLQLFLKIIGNQNA\ntranslation: KMFMKTPLQLFLKIIGNQNA\ntranslation: AFWLPIIFKNNCKGVFMNIF\ntranslation: AFWLPIIFKNNCKGVFMNIF\ntranslation: GSFLAIAMKISFTFIAVFADVSM\ntranslation: GSFLAIAMKISFTFIAVFADVSM\ntranslation: METSAKTAMNVNEIFMAIAKKLP\ntranslation: METSAKTAMNVNEIFMAIAKKLP\ntranslation: PAHDYFKDTGGDGQDSLYKIC\ntranslation: PAHDYFKDTGGDGQDSLYKIC\ntranslation: QPMTTLRTQEEMDKIPYIKYA\ntranslation: QPMTTLRTQEEMDKIPYIKYA\ntranslation: RSTNTFFMLHTYWEYFYSKYATV\ntranslation: RSTNTFFMLHTYWEYFYSKYATV\ntranslation: EALILSSCCTHTGNTFTVNMQLF\ntranslation: EALILSSCCTHTGNTFTVNMQLF\ntranslation: NSCIFTVKVFPVCVQHEESISAS\ntranslation: NSCIFTVKVFPVCVQHEESISAS\ntranslation: QVEKQMFQKIRQLKISLRML\ntranslation: QVEKQMFQKIRQLKISLRML\ntranslation: HSIRNEIFSWRIFWNICFSTC\ntranslation: HSIRNEIFSWRIFWNICFSTC\ntranslation: TAFAMRSLVGGFSGTFVFQLV\ntranslation: TAFAMRSLVGGFSGTFVFQLV\ntranslation: CVCVCVCV\ntranslation: CVCVCVCV\ntranslation: VCVCVCV\ntranslation: VCVCVCV\ntranslation: CVCVCVC\ntranslation: CVCVCVC\ntranslation: HTHTHTHT\ntranslation: HTHTHTHT\ntranslation: THTHTHT\ntranslation: THTHTHT\ntranslation: HTHTHTH\ntranslation: HTHTHTH\ntranslation: ARSPKETQSTSCRWACSEFV\ntranslation: ARSPKETQSTSCRWACSEFV\ntranslation: PGAQRKLSQPAVDGLVPSLF\ntranslation: PGAQRKLSQPAVDGLVPSLF\ntranslation: EQTRNKPIYSWLTEFPLGSW\ntranslation: EQTRNKPIYSWLTEFPLGSW\ntranslation: RSWLDHRISITVSDLRLLNRDL\ntranslation: RSWLDHRISITVSDLRLLNRDL\ntranslation: SSVSIISNNPRRSSTRCRAR\ntranslation: SSVSIISNNPRRSSTRCRAR\ntranslation: HLFPSSLTIHAGLPQGAEL\ntranslation: HLFPSSLTIHAGLPQGAEL\ntranslation: RALHLVEDLRGLLEMMETDE\ntranslation: RALHLVEDLRGLLEMMETDE\ntranslation: LDMSIIKIHPFANLCLKKECRIM\ntranslation: LDMSIIKIHPFANLCLKKECRIM\ntranslation: HYSTFFFQTKIRKWMYFDDAHVK\ntranslation: HYSTFFFQTKIRKWMYFDDAHVK\ntranslation: IILHSFFKQRFANGCILMMLMSR\ntranslation: IILHSFFKQRFANGCILMMLMSR\ntranslation: TCSSVDGHLTCFYFLVILNNATM\ntranslation: TCSSVDGHLTCFYFLVILNNATM\ntranslation: HKCSYSVFILELFFCLFFSFLDL\ntranslation: HKCSYSVFILELFFCLFFSFLDL\ntranslation: EIEIPPTRADIIHACDIVEDAA\ntranslation: EIEIPPTRADIIHACDIVEDAA\ntranslation: IAASSTISHAWIMSALVGGISIS\ntranslation: IAASSTISHAWIMSALVGGISIS\ntranslation: SCIFYNITCMDNVSSGWRDFNL\ntranslation: SCIFYNITCMDNVSSGWRDFNL\ntranslation: DSISKFLQFQLTVNVVAVIVAF\ntranslation: DSISKFLQFQLTVNVVAVIVAF\ntranslation: LKEANFDINQLYDCNWVVVNCST\ntranslation: LKEANFDINQLYDCNWVVVNCST\ntranslation: LKKPTSTSISYMTAIGLLSTAPL\ntranslation: LKKPTSTSISYMTAIGLLSTAPL\ntranslation: LLQVLCLQNDNSQLTSSRYVA\ntranslation: LLQVLCLQNDNSQLTSSRYVA\ntranslation: RYVPTAGKLTVVILEAKNLKK\ntranslation: RYVPTAGKLTVVILEAKNLKK\ntranslation: ISQVYTRMAQIYSDYSIITILYH\ntranslation: ISQVYTRMAQIYSDYSIITILYH\ntranslation: SARYIPGWHRFTVTILLLPSFII\ntranslation: SARYIPGWHRFTVTILLLPSFII\ntranslation: MIKDGNNRIVTVNLCHPGIYLAD\ntranslation: MIKDGNNRIVTVNLCHPGIYLAD\ntranslation: LEALANSRKGQFRGYNGLNAA\ntranslation: LEALANSRKGQFRGYNGLNAA\ntranslation: STKITFSTCLNIKAVVIWKSLVN\ntranslation: STKITFSTCLNIKAVVIWKSLVN\ntranslation: FTSDFQMTTALIFKHVLKVILVE\ntranslation: FTSDFQMTTALIFKHVLKVILVE\ntranslation: FNTSLWGTALLFIKFKIKIYFQ\ntranslation: FNTSLWGTALLFIKFKIKIYFQ\ntranslation: NPIVVVSCMSYLCFLDINPLL\ntranslation: NPIVVVSCMSYLCFLDINPLL\ntranslation: QPILTYKQGSTYFLESNVLYF\ntranslation: QPILTYKQGSTYFLESNVLYF\ntranslation: TIPHHPTKNIKHKLPPTSLTK\ntranslation: TIPHHPTKNIKHKLPPTSLTK\ntranslation: LFLITQLKTLNTNYHLPPSPK\ntranslation: LFLITQLKTLNTNYHLPPSPK\ntranslation: HAFYTHTTYTCTTYNTHYTHYKH\ntranslation: HAFYTHTTYTCTTYNTHYTHYKH\ntranslation: MHSTHTQHTHALHITHIIHITNT\ntranslation: MHSTHTQHTHALHITHIIHITNT\ntranslation: VFVMCIMCVICSACVCCVCVECM\ntranslation: VFVMCIMCVICSACVCCVCVECM\ntranslation: IVDLQINSAIFISLPESQGR\ntranslation: IVDLQINSAIFISLPESQGR\ntranslation: KGKKAGSTFALIKYFYVMLKTWK\ntranslation: KGKKAGSTFALIKYFYVMLKTWK\ntranslation: GVKPPRGILLYGPPGTGKTLV\ntranslation: GVKPPRGILLYGPPGTGKTLV\ntranslation: NQGLSCSRRSVKQDSSRRFHT\ntranslation: NQGLSCSRRSVKQDSSRRFHT\ntranslation: QQHTCVHC\ntranslation: QQHTCVHC\ntranslation: NNTLVFIV\ntranslation: NNTLVFIV\ntranslation: TTHLCSLW\ntranslation: TTHLCSLW\ntranslation: HNEHKCVV\ntranslation: HNEHKCVV\ntranslation: TMNTSVLL\ntranslation: TMNTSVLL\ntranslation: LMYPLLSSLQ\ntranslation: LMYPLLSSLQ\ntranslation: SCILYSVPSN\ntranslation: SCILYSVPSN\ntranslation: HVSFTQFPPM\ntranslation: HVSFTQFPPM\ntranslation: HWRELSKGYM\ntranslation: HWRELSKGYM\ntranslation: HIVVLIWISQVVMLSIFSCSLVI\ntranslation: HIVVLIWISQVVMLSIFSCSLVI\ntranslation: HCGFDMDFPGSDVEYFLMFIGHL\ntranslation: HCGFDMDFPGSDVEYFLMFIGHL\ntranslation: QMTNEHEKILNITTWEIHIKTTM\ntranslation: QMTNEHEKILNITTWEIHIKTTM\ntranslation: STIGSFDPSFRYAAFPSSSS\ntranslation: STIGSFDPSFRYAAFPSSSS\ntranslation: PLLVLLIHLSGMQLFLLRVL\ntranslation: PLLVLLIHLSGMQLFLLRVL\ntranslation: KELEEGKAAYLKDGSKEPIVE\ntranslation: KELEEGKAAYLKDGSKEPIVE\ntranslation: RTRRRKSCIPERWIKRTNSG\ntranslation: RTRRRKSCIPERWIKRTNSG\ntranslation: DLDSETRSMVEKMMYDQRQ\ntranslation: DLDSETRSMVEKMMYDQRQ\ntranslation: SVAGHTSSFPPCCESHCPG\ntranslation: SVAGHTSSFPPCCESHCPG\ntranslation: LSLVIHHLFHHAASLTVQV\ntranslation: LSLVIHHLFHHAASLTVQV\ntranslation: CRWSYIIFSTMLRVSLSR\ntranslation: CRWSYIIFSTMLRVSLSR\ntranslation: YAEGIGVSPCYIMLGYNFSSFPC\ntranslation: YAEGIGVSPCYIMLGYNFSSFPC\ntranslation: QKSWTLLYIKKSSNNFFPEPT\ntranslation: QKSWTLLYIKKSSNNFFPEPT\ntranslation: KNHGLYYTLKNLLTTFFLNLL\ntranslation: KNHGLYYTLKNLLTTFFLNLL\ntranslation: VGSGKKLLEDFLMYNKVHDF\ntranslation: VGSGKKLLEDFLMYNKVHDF\ntranslation: QDYPSLVLFLPLNNNVNLVKSFL\ntranslation: QDYPSLVLFLPLNNNVNLVKSFL\ntranslation: GLSIFSAFPAVKQQCELGQVIS\ntranslation: GLSIFSAFPAVKQQCELGQVIS\ntranslation: KPGFPEPTNNTPKGILHLSPD\ntranslation: KPGFPEPTNNTPKGILHLSPD\ntranslation: SLGFLNQQIILQKEFFICLLM\ntranslation: SLGFLNQQIILQKEFFICLLM\ntranslation: HQETNEEFLLEYYLLVQETQA\ntranslation: HQETNEEFLLEYYLLVQETQA\ntranslation: IRRQMKNSFWSIICWFRKPRL\ntranslation: IRRQMKNSFWSIICWFRKPRL\ntranslation: LGLAFRRAICCGSHELECRINHA\ntranslation: LGLAFRRAICCGSHELECRINHA\ntranslation: SSSSSFLISFFKQFIVSRLAFNLS\ntranslation: SSSSSFLISFFKQFIVSRLAFNLS\ntranslation: ERLKASRETINCLKNEIRKLEEDD\ntranslation: ERLKASRETINCLKNEIRKLEEDD\ntranslation: EDVQMERVRTANALNSTNFDEK\ntranslation: EDVQMERVRTANALNSTNFDEK\ntranslation: LLIKISRIQGICCSHSFHLNIF\ntranslation: LLIKISRIQGICCSHSFHLNIF\ntranslation: SKLGLSCLLIINPSISRFVVQGH\ntranslation: SKLGLSCLLIINPSISRFVVQGH\ntranslation: CPWTTNLDILGLIIRRQLRPSF\ntranslation: CPWTTNLDILGLIIRRQLRPSF\ntranslation: CEETFSVYWISHYS\ntranslation: CEETFSVYWISHYS\ntranslation: VKKHSQFIGYPITL\ntranslation: VKKHSQFIGYPITL\ntranslation: KSNGISNKLRMFLH\ntranslation: KSNGISNKLRMFLH\ntranslation: QYPPNTSKVYSYFECREKKTENS\ntranslation: QYPPNTSKVYSYFECREKKTENS\ntranslation: NIHPTQAKFIPTLNAVKRRQKTP\ntranslation: NIHPTQAKFIPTLNAVKRRQKTP\ntranslation: WSFLSSFHGIQSRNKLCLCWVDI\ntranslation: WSFLSSFHGIQSRNKLCLCWVDI\ntranslation: GVFCLLFTAFKVGINFACVGWIL\ntranslation: GVFCLLFTAFKVGINFACVGWIL\ntranslation: IHLLLVQFSLIEVQKANCAAN\ntranslation: IHLLLVQFSLIEVQKANCAAN\ntranslation: IGCTVSLLNFYKTELNKEEMY\ntranslation: IGCTVSLLNFYKTELNKEEMY\ntranslation: ANAQEYSHSKVSLLSYFLSTNQ\ntranslation: ANAQEYSHSKVSLLSYFLSTNQ\ntranslation: LMHKNTVTAKFPYSHISSVLI\ntranslation: LMHKNTVTAKFPYSHISSVLI\ntranslation: LTAPLLHTLFLNLSWACDLFS\ntranslation: LTAPLLHTLFLNLSWACDLFS\ntranslation: DLLPRLTPILKNRHEKVQEN\ntranslation: DLLPRLTPILKNRHEKVQEN\ntranslation: QFSCTFSCLFFKMGVSLGSRS\ntranslation: QFSCTFSCLFFKMGVSLGSRS\ntranslation: KILPGREQLSFNSSGLLFSPLT\ntranslation: KILPGREQLSFNSSGLLFSPLT\ntranslation: AVNGENKSPLLLNDSCSLPGRIF\ntranslation: AVNGENKSPLLLNDSCSLPGRIF\ntranslation: TVLPPFLNPIIYSLRNKEIKTAM\ntranslation: TVLPPFLNPIIYSLRNKEIKTAM\ntranslation: QFCLPSSIPSYIVLGIRRLKQP\ntranslation: QFCLPSSIPSYIVLGIRRLKQP\ntranslation: HGCFNLLIPKTIYDGIEEGRQNC\ntranslation: HGCFNLLIPKTIYDGIEEGRQNC\ntranslation: MAVLISLFLRLYMMGLRKGGKTV\ntranslation: MAVLISLFLRLYMMGLRKGGKTV\ntranslation: PRPNLRQNPFHYHIHRRKSNFL\ntranslation: PRPNLRQNPFHYHIHRRKSNFL\ntranslation: LDQTYAKIHFTIIFIGVNLTF\ntranslation: LDQTYAKIHFTIIFIGVNLTF\ntranslation: LNQRNPLRLCQRRRSKRSRLKQ\ntranslation: LNQRNPLRLCQRRRSKRSRLKQ\ntranslation: ESEKSTEVVPKKKIKKEQVET\ntranslation: ESEKSTEVVPKKKIKKEQVET\ntranslation: VSTCSFLIFFFGTTSVDFSDS\ntranslation: VSTCSFLIFFFGTTSVDFSDS\ntranslation: LALRWILLTWMVTRYIFPGIRS\ntranslation: LALRWILLTWMVTRYIFPGIRS\ntranslation: GFEMDITHLDGHKVHISRDKIT\ntranslation: GFEMDITHLDGHKVHISRDKIT\ntranslation: GDLIPGNMYLVTIQVSNIHLKA\ntranslation: GDLIPGNMYLVTIQVSNIHLKA\ntranslation: LRLKIDLRKVFRFTFSTIIHFIAS\ntranslation: LRLKIDLRKVFRFTFSTIIHFIAS\ntranslation: AGYEVNNGTESKPEDLSQIYFKSE\ntranslation: AGYEVNNGTESKPEDLSQIYFKSE\ntranslation: LGHAITFFTSTLPIRSSSQVS\ntranslation: LGHAITFFTSTLPIRSSSQVS\ntranslation: WGMPLPFSLLHYQLGVHPRY\ntranslation: WGMPLPFSLLHYQLGVHPRY\ntranslation: DTWDELLIGSVEVKKVMACP\ntranslation: DTWDELLIGSVEVKKVMACP\ntranslation: HTNLFSTSKGVMKGFCLFQAKN\ntranslation: HTNLFSTSKGVMKGFCLFQAKN\ntranslation: FFAWKRQKPFITPLLVENRFVC\ntranslation: FFAWKRQKPFITPLLVENRFVC\ntranslation: LQFFFCPLR\ntranslation: LQFFFCPLR\ntranslation: PIFFLSLE\ntranslation: PIFFLSLE\ntranslation: PQGTEKKLE\ntranslation: PQGTEKKLE\ntranslation: LKGQKKNW\ntranslation: LKGQKKNW\ntranslation: SRDRKKIG\ntranslation: SRDRKKIG\ntranslation: PGLLTSYRNCLALGYVQPGRGV\ntranslation: PGLLTSYRNCLALGYVQPGRGV\ntranslation: WASHQLQELPSTWVCTTWKRC\ntranslation: WASHQLQELPSTWVCTTWKRC\ntranslation: QFSVKETNFSEGNLKLKIGLQ\ntranslation: QFSVKETNFSEGNLKLKIGLQ\ntranslation: VFSERNKLFRGKFKIENWPP\ntranslation: VFSERNKLFRGKFKIENWPP\ntranslation: PGGQFSILNFPLKSLFLSLKT\ntranslation: PGGQFSILNFPLKSLFLSLKT\ntranslation: WRPIFNFKFPSEKFVSFTEN\ntranslation: WRPIFNFKFPSEKFVSFTEN\ntranslation: DNLTILLHTASVMQHLHHSIQVW\ntranslation: DNLTILLHTASVMQHLHHSIQVW\ntranslation: RPVLNDVSAALQKLCGEVWLSY\ntranslation: RPVLNDVSAALQKLCGEVWLSY\ntranslation: KQTGSPEQFDLIVLTMPVPEIL\ntranslation: KQTGSPEQFDLIVLTMPVPEIL\ntranslation: NKQAPLSSLILLFSQCQFLRFC\ntranslation: NKQAPLSSLILLFSQCQFLRFC\ntranslation: QNLRNWHCENNKIKLLRGACLF\ntranslation: QNLRNWHCENNKIKLLRGACLF\ntranslation: RISGTGIVRTIRSNCSGEPVC\ntranslation: RISGTGIVRTIRSNCSGEPVC\ntranslation: QYLLCNINSHITDKLIVVAAS\ntranslation: QYLLCNINSHITDKLIVVAAS\ntranslation: HIYRVCTNRKWLQITTRLGRIP\ntranslation: HIYRVCTNRKWLQITTRLGRIP\ntranslation: IFTEFAQIENGYKLQPGLAGYQ\ntranslation: IFTEFAQIENGYKLQPGLAGYQ\ntranslation: FDETFEFQVNLPELAMIRFVVLD\ntranslation: FDETFEFQVNLPELAMIRFVVLD\ntranslation: SRTTKRIMASSGRFTWNSKVSSK\ntranslation: SRTTKRIMASSGRFTWNSKVSSK\ntranslation: PEQQNGSWPAQVGLLGTQKFHQ\ntranslation: PEQQNGSWPAQVGLLGTQKFHQ\ntranslation: TVVSLTFLRMLILSGSAPEIL\ntranslation: TVVSLTFLRMLILSGSAPEIL\ntranslation: RISGADPDKMSILRKVRDTT\ntranslation: RISGADPDKMSILRKVRDTT\ntranslation: LESYEEDRVLRFQNKIRYFCLLN\ntranslation: LESYEEDRVLRFQNKIRYFCLLN\ntranslation: HLVKKVNSSFVSQLQKAMCLVG\ntranslation: HLVKKVNSSFVSQLQKAMCLVG\ntranslation: LGEESKFLLCLPAPEGNVSRWL\ntranslation: LGEESKFLLCLPAPEGNVSRWL\ntranslation: QPTRHIAFWSWETKEEFTFFTK\ntranslation: QPTRHIAFWSWETKEEFTFFTK\ntranslation: SQRDTLPSGAGRQRRNLLSSPS\ntranslation: SQRDTLPSGAGRQRRNLLSSPS\ntranslation: ANETHCLLELGDKGGIYFLHQV\ntranslation: ANETHCLLELGDKGGIYFLHQV\ntranslation: EVILPSFTHLVIDSVIQLITKYF\ntranslation: EVILPSFTHLVIDSVIQLITKYF\ntranslation: CTIPMRHNYNKLHLPTTNRPKIA\ntranslation: CTIPMRHNYNKLHLPTTNRPKIA\ntranslation: SDFRSVCRRQMELVIIMPHRDST\ntranslation: SDFRSVCRRQMELVIIMPHRDST\ntranslation: IKPNPATQNLSILLNYPHRMNN\ntranslation: IKPNPATQNLSILLNYPHRMNN\ntranslation: IIHPMWVIEEYAKILRSWVWFN\ntranslation: IIHPMWVIEEYAKILRSWVWFN\ntranslation: LLRIIEMSVFLEYFTLRTNLLS\ntranslation: LLRIIEMSVFLEYFTLRTNLLS\ntranslation: EDSRLVLSVKYSKNTDISMILNK\ntranslation: EDSRLVLSVKYSKNTDISMILNK\ntranslation: LGSAMSRPI\ntranslation: LGSAMSRPI\ntranslation: GKCHEQAH\ntranslation: GKCHEQAH\ntranslation: DGPAHGTSQ\ntranslation: DGPAHGTSQ\ntranslation: MGLLMALP\ntranslation: MGLLMALP\ntranslation: WACSWHFP\ntranslation: WACSWHFP\ntranslation: SLSVRVKIFYFLILKIVTIIDLL\ntranslation: SLSVRVKIFYFLILKIVTIIDLL\ntranslation: GKVLFQVRLKVLIWFFTREIGNN\ntranslation: GKVLFQVRLKVLIWFFTREIGNN\ntranslation: IIANLTCKKPDQHFKPYLKQHLP\ntranslation: IIANLTCKKPDQHFKPYLKQHLP\ntranslation: NKHNIDGSRIELVAMSCHCNYDI\ntranslation: NKHNIDGSRIELVAMSCHCNYDI\ntranslation: ECHNCSDRTLPPTLSYYHQCCVY\ntranslation: ECHNCSDRTLPPTLSYYHQCCVY\ntranslation: NVIIAVTGHCHQLYPTTINVVFI\ntranslation: NVIIAVTGHCHQLYPTTINVVFI\ntranslation: ALSPRLKYSGTIIAHCNLEFL\ntranslation: ALSPRLKYSGTIIAHCNLEFL\ntranslation: RKLLNETSVTRLPLPNPGDPF\ntranslation: RKLLNETSVTRLPLPNPGDPF\ntranslation: EGVSRIREWESCYRCFIQEFS\ntranslation: EGVSRIREWESCYRCFIQEFS\ntranslation: KGSPGLGSGSLVTDVSFKSF\ntranslation: KGSPGLGSGSLVTDVSFKSF\ntranslation: FFIVSVFFTMIFGERLFQSCKHQ\ntranslation: FFIVSVFFTMIFGERLFQSCKHQ\ntranslation: ERRRERAVLFLMTTPEKYQETSL\ntranslation: ERRRERAVLFLMTTPEKYQETSL\ntranslation: QGESVGAFSPAHPAE\ntranslation: QGESVGAFSPAHPAE\ntranslation: KEKVLEHSHRPIQL\ntranslation: KEKVLEHSHRPIQL\ntranslation: SAGWAGENAPTLSP\ntranslation: SAGWAGENAPTLSP\ntranslation: QLDGPVRMLQHFLL\ntranslation: QLDGPVRMLQHFLL\ntranslation: LNLGILIFGIFKFQSGPCISTS\ntranslation: LNLGILIFGIFKFQSGPCISTS\ntranslation: KFGHFNLWHLQVPVRTLHFNI\ntranslation: KFGHFNLWHLQVPVRTLHFNI\ntranslation: DVEMQGPDWNLKMPKIKMPKF\ntranslation: DVEMQGPDWNLKMPKIKMPKF\ntranslation: LTLQYNIYNVGSSAALDVELSDD\ntranslation: LTLQYNIYNVGSSAALDVELSDD\ntranslation: NHQIVPRLMQHLSQHCRCCTARS\ntranslation: NHQIVPRLMQHLSQHCRCCTARS\ntranslation: VIEAEDGLVISKPKGFLTCLIC\ntranslation: VIEAEDGLVISKPKGFLTCLIC\ntranslation: QMRQVKNPFGLEITNPSSASIT\ntranslation: QMRQVKNPFGLEITNPSSASIT\ntranslation: QHSASIFFREALGVFSRSRFLC\ntranslation: QHSASIFFREALGVFSRSRFLC\ntranslation: SILHQFSLGRPSEYSQGADSYV\ntranslation: SILHQFSLGRPSEYSQGADSYV\ntranslation: RHKNRLLENTPRASLKKIDAEC\ntranslation: RHKNRLLENTPRASLKKIDAEC\ntranslation: KTMQSDGILLHREGPNGDHITL\ntranslation: KTMQSDGILLHREGPNGDHITL\ntranslation: KPCRVMGFYSTGKGQMEITSHC\ntranslation: KPCRVMGFYSTGKGQMEITSHC\ntranslation: RTWHVKSKTDLFIPKKKCSFYDN\ntranslation: RTWHVKSKTDLFIPKKKCSFYDN\ntranslation: HLSPHLRYTLTFPYVFLKEPYCS\ntranslation: HLSPHLRYTLTFPYVFLKEPYCS\ntranslation: RKLRGTFSCQRHEFLCLIFKLF\ntranslation: RKLRGTFSCQRHEFLCLIFKLF\ntranslation: SNDDLLWCLRKLTRPPLTVS\ntranslation: SNDDLLWCLRKLTRPPLTVS\ntranslation: ETVSGGLVSFLKHHRRSSLL\ntranslation: ETVSGGLVSFLKHHRRSSLL\ntranslation: FFCYKLYSMMVPTMYEIPQKLDL\ntranslation: FFCYKLYSMMVPTMYEIPQKLDL\ntranslation: TCPNENQKMERQIHFLLGWTVTT\ntranslation: TCPNENQKMERQIHFLLGWTVTT\ntranslation: HAQMKTKRWKGRYTSYLGGQLQH\ntranslation: HAQMKTKRWKGRYTSYLGGQLQH\ntranslation: DVVTVHPSKKCICLSIFWFSFGHV\ntranslation: DVVTVHPSKKCICLSIFWFSFGHV\ntranslation: QVLKNLMASSA\ntranslation: QVLKNLMASSA\ntranslation: SPQEFDGFFCS\ntranslation: SPQEFDGFFCS\ntranslation: AAEEAIKFLRT\ntranslation: AAEEAIKFLRT\ntranslation: CRRSHQILEDL\ntranslation: CRRSHQILEDL\ntranslation: IFQTLELQGSKLPQDMLDLSTA\ntranslation: IFQTLELQGSKLPQDMLDLSTA\ntranslation: LKHVSAFYFKKTQTWSQNIFREQ\ntranslation: LKHVSAFYFKKTQTWSQNIFREQ\ntranslation: PKPNNSSTAYYNFTGSLFYPP\ntranslation: PKPNNSSTAYYNFTGSLFYPP\ntranslation: RNQIIQALLITILLGLYFTLL\ntranslation: RNQIIQALLITILLGLYFTLL\ntranslation: SFLLLVYLFCRHLPDEPLLTL\ntranslation: SFLLLVYLFCRHLPDEPLLTL\ntranslation: PESTEARQEDVYRKGRPKGGS\ntranslation: PESTEARQEDVYRKGRPKGGS\ntranslation: QSQQRLVRKMSTEKVDQKEEA\ntranslation: QSQQRLVRKMSTEKVDQKEEA\ntranslation: KSIVSQSSLIKRKNRNLHRR\ntranslation: KSIVSQSSLIKRKNRNLHRR\ntranslation: SSVQISVLSLNQGALRNNAF\ntranslation: SSVQISVLSLNQGALRNNAF\ntranslation: QGFFLVSKELFLFGLTVKFTR\ntranslation: QGFFLVSKELFLFGLTVKFTR\ntranslation: PLVNLTVSPKRNSSLDTRKKP\ntranslation: PLVNLTVSPKRNSSLDTRKKP\ntranslation: AEFACKSPAQLWYPVLLSVSNKD\ntranslation: AEFACKSPAQLWYPVLLSVSNKD\ntranslation: PSLLVSLQLNFGILFCYLYQTR\ntranslation: PSLLVSLQLNFGILFCYLYQTR\ntranslation: PCLIQITKQDTKVELETYKQTR\ntranslation: PCLIQITKQDTKVELETYKQTR\ntranslation: QSTDCYGLNVFPPDSCVEILTP\ntranslation: QSTDCYGLNVFPPDSCVEILTP\ntranslation: GLGFQHMNLEGTHSDHSSRCF\ntranslation: GLGFQHMNLEGTHSDHSSRCF\ntranslation: ETYHISNARVSGRKFFHRRCMAVL\ntranslation: ETYHISNARVSGRKFFHRRCMAVL\ntranslation: KNGHTPPMKKLPTTHPSITYMICL\ntranslation: KNGHTPPMKKLPTTHPSITYMICL\ntranslation: IKSVYFILYFINVFNRFLLFFF\ntranslation: IKSVYFILYFINVFNRFLLFFF\ntranslation: LSLSISFCTSSMCSIASCCSFS\ntranslation: LSLSISFCTSSMCSIASCCSFS\ntranslation: EKEQQEAIEHIDEVQNEIDRLN\ntranslation: EKEQQEAIEHIDEVQNEIDRLN\ntranslation: KFCLILHIPVNKIQKHCVFKIVS\ntranslation: KFCLILHIPVNKIQKHCVFKIVS\ntranslation: RHNLKNTMLLYFIHWNMENQTEF\ntranslation: RHNLKNTMLLYFIHWNMENQTEF\ntranslation: DTILKTQCFCILFTGIWRIKQN\ntranslation: DTILKTQCFCILFTGIWRIKQN\ntranslation: RYFSTEASLSIIESYFLRCSRK\ntranslation: RYFSTEASLSIIESYFLRCSRK\ntranslation: GTLARKLPLVLLRATSCDALGS\ntranslation: GTLARKLPLVLLRATSCDALGS\ntranslation: LPRASQEVALNNTKGSFRAKVP\ntranslation: LPRASQEVALNNTKGSFRAKVP\ntranslation: AAKFITHAPPGEFNEVFNDVRLL\ntranslation: AAKFITHAPPGEFNEVFNDVRLL\ntranslation: LLNSSLMHPQGNLMKYSMTFGY\ntranslation: LLNSSLMHPQGNLMKYSMTFGY\ntranslation: AFSYKLVRSKADWERWMESFGT\ntranslation: AFSYKLVRSKADWERWMESFGT\ntranslation: LFQMIPSTFPNRPYFSLICRKK\ntranslation: LFQMIPSTFPNRPYFSLICRKK\ntranslation: AAALMLSDPLQGHFPAQATS\ntranslation: AAALMLSDPLQGHFPAQATS\ntranslation: GRWPGQGSVPGGDHSASRLQQ\ntranslation: GRWPGQGSVPGGDHSASRLQQ\ntranslation: GGGLGREVSLEGITQHQGCS\ntranslation: GGGLGREVSLEGITQHQGCS\ntranslation: EVAWAGKCPWRGSLSIKAAA\ntranslation: EVAWAGKCPWRGSLSIKAAA\ntranslation: LVSGEST\ntranslation: LVSGEST\ntranslation: WFLVSPL\ntranslation: WFLVSPL\ntranslation: PVDSPET\ntranslation: PVDSPET\ntranslation: QWTHQKP\ntranslation: QWTHQKP\ntranslation: SGLTRNQ\ntranslation: SGLTRNQ\ntranslation: PFLTLYYLKSQWKFSVSCLCNS\ntranslation: PFLTLYYLKSQWKFSVSCLCNS\ntranslation: HECTLSLAEEYGNSSPTQQVS\ntranslation: HECTLSLAEEYGNSSPTQQVS\ntranslation: GLLVELENYCRIPLLSSGCIR\ntranslation: GLLVELENYCRIPLLSSGCIR\ntranslation: CTVNLMNFYKSEINKEEMYIRYIH\ntranslation: CTVNLMNFYKSEINKEEMYIRYIH\ntranslation: MDVADIHFFLVNLRFVKIHQVNSA\ntranslation: MDVADIHFFLVNLRFVKIHQVNSA\ntranslation: QKRYHWGYNEISFLTIESNKELIF\ntranslation: QKRYHWGYNEISFLTIESNKELIF\ntranslation: RKLTLYYFQLLRKKSHYIPNDNVS\ntranslation: RKLTLYYFQLLRKKSHYIPNDNVS\ntranslation: ETSLDEDSKAMIRVLRCLWIIS\ntranslation: ETSLDEDSKAMIRVLRCLWIIS\ntranslation: RDNPEASKNANHSLAVFIKRCF\ntranslation: RDNPEASKNANHSLAVFIKRCF\ntranslation: EIIQRHLRTRIIALLSSSRDV\ntranslation: EIIQRHLRTRIIALLSSSRDV\ntranslation: CIARYCPTCTLLHFFSTICDSPT\ntranslation: CIARYCPTCTLLHFFSTICDSPT\ntranslation: SGRITDCAEKMQEGAGGTVPCYA\ntranslation: SGRITDCAEKMQEGAGGTVPCYA\ntranslation: VGESQIVLKKCRRVQVGQYRAMQ\ntranslation: VGESQIVLKKCRRVQVGQYRAMQ\ntranslation: PSVLIVQFPPMSENMWCLVFCP\ntranslation: PSVLIVQFPPMSENMWCLVFCP\ntranslation: RTENQTPHVLTHRWELNNENTW\ntranslation: RTENQTPHVLTHRWELNNENTW\ntranslation: DFVFTSAFLHVLDRDNLRPAL\ntranslation: DFVFTSAFLHVLDRDNLRPAL\ntranslation: LCVYICISTCFRQRQFKACTL\ntranslation: LCVYICISTCFRQRQFKACTL\ntranslation: KSAGLKLSLSKTCRNADVNTK\ntranslation: KSAGLKLSLSKTCRNADVNTK\ntranslation: ISGREVFEFRPELVNDDDGEAD\ntranslation: ISGREVFEFRPELVNDDDGEAD\ntranslation: SVVVKCLNFVLNWSMMMMGKQM\ntranslation: SVVVKCLNFVLNWSMMMMGKQM\ntranslation: ICFPIIIIDQFRTKFKHFTTTD\ntranslation: ICFPIIIIDQFRTKFKHFTTTD\ntranslation: SASPSSSLTSSGRNSNTSRPLI\ntranslation: SASPSSSLTSSGRNSNTSRPLI\ntranslation: PPFPLSLLATLPLLHLCLGFKTF\ntranslation: PPFPLSLLATLPLLHLCLGFKTF\ntranslation: KVLNPKHRCSKGSVAKRERGNG\ntranslation: KVLNPKHRCSKGSVAKRERGNG\ntranslation: QQNINKYLLCSTFYCRLVLNF\ntranslation: QQNINKYLLCSTFYCRLVLNF\ntranslation: FPFKYITRYFLVLYMLFYSYFRH\ntranslation: FPFKYITRYFLVLYMLFYSYFRH\ntranslation: FPLNILLDIFWCCICCSILILGT\ntranslation: FPLNILLDIFWCCICCSILILGT\ntranslation: VPKIRIEQHIQHQKISSNIFKGK\ntranslation: VPKIRIEQHIQHQKISSNIFKGK\ntranslation: AARWQCIIADHRIFPSRNHSSSI\ntranslation: AARWQCIIADHRIFPSRNHSSSI\ntranslation: SKMAMHHSRPQNISFKESQQQY\ntranslation: SKMAMHHSRPQNISFKESQQQY\ntranslation: LDTCLPEKLGCSWEITGDYLRNIC\ntranslation: LDTCLPEKLGCSWEITGDYLRNIC\ntranslation: KYFLNSHLLFPKNIPTFPVGKCQ\ntranslation: KYFLNSHLLFPKNIPTFPVGKCQ\ntranslation: ESESYRKIADSIGCSTNNILFL\ntranslation: ESESYRKIADSIGCSTNNILFL\ntranslation: SETKCCWLSTQCCLQSFGNFHSP\ntranslation: SETKCCWLSTQCCLQSFGNFHSP\ntranslation: RREGGDQRVKSVINLLFAAYTGD\ntranslation: RREGGDQRVKSVINLLFAAYTGD\ntranslation: HLQYMQQTKDLSLTLPFDHHLLF\ntranslation: HLQYMQQTKDLSLTLPFDHHLLF\ntranslation: MHLSCLTAEPYLSLLAQFKTYTG\ntranslation: MHLSCLTAEPYLSLLAQFKTYTG\ntranslation: KGNIILIPERSNSPENTRNIFYK\ntranslation: KGNIILIPERSNSPENTRNIFYK\ntranslation: ACTTTERKIRDSEKSGIILLHIC\ntranslation: ACTTTERKIRDSEKSGIILLHIC\ntranslation: LAQLQKEKSEILKNLALYYFTFV\ntranslation: LAQLQKEKSEILKNLALYYFTFV"
      },
      {
       "name": "stderr",
       "output_type": "stream",
       "text": "757it [00:00, 1167.52it/s]"
      },
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "\ntranslation: DFLKSSSIGQKLRNKKKKRWN\ntranslation: DFLKSSSIGQKLRNKKKKRWN\ntranslation: FHLFFFLFLSFCPIEDDFRK\ntranslation: FHLFFFLFLSFCPIEDDFRK\ntranslation: WTQSENNSIHFHSFPYWLLIIC\ntranslation: WTQSENNSIHFHSFPYWLLIIC\ntranslation: KALNQNVYICLKMSLIIQYALDL\ntranslation: KALNQNVYICLKMSLIIQYALDL\ntranslation: SPKSECLYLPKDVLNNTVCFRS\ntranslation: SPKSECLYLPKDVLNNTVCFRS\ntranslation: QDLKHTVLLRTSLGKYKHSDLGL\ntranslation: QDLKHTVLLRTSLGKYKHSDLGL\ntranslation: PNISINSDFLITILFPLKVLTRN\ntranslation: PNISINSDFLITILFPLKVLTRN\ntranslation: FRVKTFKGNRIVIKKSEFIDILG\ntranslation: FRVKTFKGNRIVIKKSEFIDILG\ntranslation: LGSLAHLKHPA\ntranslation: LGSLAHLKHPA\ntranslation: GFSCSSEAPC\ntranslation: GFSCSSEAPC\ntranslation: RRVLQMSKRTQ\ntranslation: RRVLQMSKRTQ\ntranslation: QGASDEQENP\ntranslation: QGASDEQENP\ntranslation: EEWGGFCMFKLSFYFYVVSECL\ntranslation: EEWGGFCMFKLSFYFYVVSECL\ntranslation: RNGVGFVCSNCHFIFTLCLSV\ntranslation: RNGVGFVCSNCHFIFTLCLSV\ntranslation: RKEAFCILNQFICLLSASVIF\ntranslation: RKEAFCILNQFICLLSASVIF\ntranslation: EYDRCTEKADKLIQYTKGLFP\ntranslation: EYDRCTEKADKLIQYTKGLFP\ntranslation: KMDLQKLAAVVCSCFKISFLV\ntranslation: KMDLQKLAAVVCSCFKISFLV\ntranslation: TKNEILKQEQTTAASFCKSIF\ntranslation: TKNEILKQEQTTAASFCKSIF\ntranslation: SACTSLFYFSVSSILCIFVNSGL\ntranslation: SACTSLFYFSVSSILCIFVNSGL\ntranslation: CLYFSLLFQRLLYPLYFCELWT\ntranslation: CLYFSLLFQRLLYPLYFCELWT\ntranslation: ESRVHKNTEDRGDAEIKERSTSR\ntranslation: ESRVHKNTEDRGDAEIKERSTSR\ntranslation: RVYRVSTPYIYSRNRRRHTSIFH\ntranslation: RVYRVSTPYIYSRNRRRHTSIFH\ntranslation: VKYARVSTSIPTVNIWCAHTINP\ntranslation: VKYARVSTSIPTVNIWCAHTINP\ntranslation: VCLPCWTVSIYIRLLATSDWL\ntranslation: VCLPCWTVSIYIRLLATSDWL\ntranslation: RSFIMRRCWMGWGGR\ntranslation: RSFIMRRCWMGWGGR\ntranslation: EVSSCGDVGWGGEVD\ntranslation: EVSSCGDVGWGGEVD\ntranslation: KFHHAEMLDGVGRSM\ntranslation: KFHHAEMLDGVGRSM\ntranslation: HRPPHPIQHLRMMKL\ntranslation: HRPPHPIQHLRMMKL\ntranslation: STSPPHPTSPHDETS\ntranslation: STSPPHPTSPHDETS\ntranslation: HTLHLTSFGLTTSIIIVLSSVGI\ntranslation: HTLHLTSFGLTTSIIIVLSSVGI\ntranslation: YFTSDILWPNNIYYYSAQQCGH\ntranslation: YFTSDILWPNNIYYYSAQQCGH\ntranslation: MPTLLSTIIIDVVRPKDVRCKV\ntranslation: MPTLLSTIIIDVVRPKDVRCKV\ntranslation: EKFTAFCKAQG\ntranslation: EKFTAFCKAQG\ntranslation: RNSPPSARPR\ntranslation: RNSPPSARPR\ntranslation: EIHRLLQGPG\ntranslation: EIHRLLQGPG\ntranslation: ALGLAEGGEFL\ntranslation: ALGLAEGGEFL\ntranslation: PWALQKAVNF\ntranslation: PWALQKAVNF\ntranslation: PLVTVLSYR\ntranslation: PLVTVLSYR\ntranslation: WDWGCDGIKTNSPLYLCKCKILI\ntranslation: WDWGCDGIKTNSPLYLCKCKILI\ntranslation: DENLALAQVERRVSFYTIASPIP\ntranslation: DENLALAQVERRVSFYTIASPIP\ntranslation: CGVKPFQFELISLKKITLLQVL\ntranslation: CGVKPFQFELISLKKITLLQVL\ntranslation: RNEFPVDIQICLWILYSVPLV\ntranslation: RNEFPVDIQICLWILYSVPLV\ntranslation: NQWNRIENPETNLYIYRELIS\ntranslation: NQWNRIENPETNLYIYRELIS\ntranslation: WLYMSPIYSSRSMKPSAFLSISS\ntranslation: WLYMSPIYSSRSMKPSAFLSISS\ntranslation: EDIDKNADGFIDLEEYIGDMYS\ntranslation: EDIDKNADGFIDLEEYIGDMYS\ntranslation: HVQLLFTWNPSPLRPSKFSFEY\ntranslation: HVQLLFTWNPSPLRPSKFSFEY\ntranslation: MFNCCSHGTLLHFGLQSSRLNI\ntranslation: MFNCCSHGTLLHFGLQSSRLNI\ntranslation: RRVIFDQENVVTSTESSPSRL\ntranslation: RRVIFDQENVVTSTESSPSRL\ntranslation: KKIPLPALSFLTINPSHPSFFRM\ntranslation: KKIPLPALSFLTINPSHPSFFRM\ntranslation: VILKNEGWEGFIVRNDSAGRGIFL\ntranslation: VILKNEGWEGFIVRNDSAGRGIFL\ntranslation: LFSMCNLKLICTTGYLTGA\ntranslation: LFSMCNLKLICTTGYLTGA\ntranslation: VAPVRYPVVQISFKLHILNK\ntranslation: VAPVRYPVVQISFKLHILNK\ntranslation: GFGCFSCFSICIKKLVHVNTFH\ntranslation: GFGCFSCFSICIKKLVHVNTFH\ntranslation: WKVFTWTSFFIQMLKHEKHPKP\ntranslation: WKVFTWTSFFIQMLKHEKHPKP\ntranslation: GLLIRSEFIINVSLELSFRIL\ntranslation: GLLIRSEFIINVSLELSFRIL\ntranslation: KILKLNSSDTLMINSDLIKSP\ntranslation: KILKLNSSDTLMINSDLIKSP\ntranslation: FDQVPPPATM\ntranslation: FDQVPPPATM\ntranslation: LTRCHHLLPC\ntranslation: LTRCHHLLPC\ntranslation: HMVAGGGTWSK\ntranslation: HMVAGGGTWSK\ntranslation: HGSRWWHLVK\ntranslation: HGSRWWHLVK\ntranslation: QRPTNHHPTMTNQTNLKTNDNHT\ntranslation: QRPTNHHPTMTNQTNLKTNDNHT\ntranslation: CMVIICFEVSLISHCWVVISRSL\ntranslation: CMVIICFEVSLISHCWVVISRSL\ntranslation: PCYIMLGYNFSSFPCGTISIAPG\ntranslation: PCYIMLGYNFSSFPCGTISIAPG\ntranslation: LAILCLVIIFHLSLAVLYLLRQV\ntranslation: LAILCLVIIFHLSLAVLYLLRQV\ntranslation: FYCLKIIIEASKNIQAKTSQF\ntranslation: FYCLKIIIEASKNIQAKTSQF\ntranslation: LLFKNHNRSFKKHTSQDIPV\ntranslation: LLFKNHNRSFKKHTSQDIPV\ntranslation: QIKHILANFKNYQFFIGENMN\ntranslation: QIKHILANFKNYQFFIGENMN\ntranslation: IHVFTNKELVVFEISKDVLDL\ntranslation: IHVFTNKELVVFEISKDVLDL\ntranslation: QQQNNNKACSVLRHCQD\ntranslation: QQQNNNKACSVLRHCQD\ntranslation: TTEQQQSMLSIETLSRL\ntranslation: TTEQQQSMLSIETLSRL\ntranslation: NLDSVSILSMLCCCSVV\ntranslation: NLDSVSILSMLCCCSVV\ntranslation: ILPSKLKLRTHSQSHHNPLSRT\ntranslation: ILPSKLKLRTHSQSHHNPLSRT\ntranslation: FCLANSNYERTHSRIIILSQGL\ntranslation: FCLANSNYERTHSRIIILSQGL\ntranslation: PVLHTILLAPLLPFLSTLLLASE\ntranslation: PVLHTILLAPLLPFLSTLLLASE\ntranslation: CLAYHSSSSFIAFSFHSASCFR\ntranslation: CLAYHSSSSFIAFSFHSASCFR\ntranslation: SRKQEAEWKEKAIKELEEWYARQ\ntranslation: SRKQEAEWKEKAIKELEEWYARQ\ntranslation: SEARSRVERKGNKGARRMVCKT\ntranslation: SEARSRVERKGNKGARRMVCKT\ntranslation: FVSGVLFFVTLSSSRLYINRH\ntranslation: FVSGVLFFVTLSSSRLYINRH\ntranslation: PCLLIYSLDDERVTKNKTPDT\ntranslation: PCLLIYSLDDERVTKNKTPDT\ntranslation: VDDSVSVNDIDLSLYIPRDVD\ntranslation: VDDSVSVNDIDLSLYIPRDVD\ntranslation: HLHLLGCTGLNLCHLHSLNHQ\ntranslation: HLHLLGCTGLNLCHLHSLNHQ\ntranslation: STSLGMYRLKSMSFTLTESS\ntranslation: STSLGMYRLKSMSFTLTESS\ntranslation: NLQRVSYLVFDEADRMFDMGF\ntranslation: NLQRVSYLVFDEADRMFDMGF\ntranslation: IFKESLTLCLMKQIECLTWDL\ntranslation: IFKESLTLCLMKQIECLTWDL\ntranslation: QIPCQTFDLLHQTQGKRLFED\ntranslation: QIPCQTFDLLHQTQGKRLFED\ntranslation: KSHVKHSICFIKHKVRDSLKI\ntranslation: KSHVKHSICFIKHKVRDSLKI\ntranslation: RHLHFLPPHSFLPHRCPHLAGCL\ntranslation: RHLHFLPPHSFLPHRCPHLAGCL\ntranslation: DIFTSFLLIVSFLIDVLILLDAF\ntranslation: DIFTSFLLIVSFLIDVLILLDAF\ntranslation: EGIQQDEDIYEEGNYEEEGSEDV\ntranslation: EGIQQDEDIYEEGNYEEEGSEDV\ntranslation: KASSKMRTSMRKETMRRKEVKMS\ntranslation: KASSKMRTSMRKETMRRKEVKMS\ntranslation: RRLLESLRGPALDVIRVLKINN\ntranslation: RRLLESLRGPALDVIRVLKINN\ntranslation: GLFILRTRITSSAGPRRLSSNRL\ntranslation: GLFILRTRITSSAGPRRLSSNRL\ntranslation: QFSSMSTLLQCLLGTSSKRNGPI\ntranslation: QFSSMSTLLQCLLGTSSKRNGPI\ntranslation: MGPFLFEEVPKRHCKSVDMELNW\ntranslation: MGPFLFEEVPKRHCKSVDMELNW\ntranslation: ILLPVFISWNFRVYVNMSSMTQV\ntranslation: ILLPVFISWNFRVYVNMSSMTQV\ntranslation: FVCSNCHFIFTLLDMGSSVIEVKN\ntranslation: FVCSNCHFIFTLLDMGSSVIEVKN\ntranslation: AFPLGLSVTSIHSTFSLHFQCPV\ntranslation: AFPLGLSVTSIHSTFSLHFQCPV\ntranslation: FPSRSLSHLYTLYIFSALSMPC\ntranslation: FPSRSLSHLYTLYIFSALSMPC\ntranslation: ENMSFKGIGAFEWNKCFHILX\ntranslation: ENMSFKGIGAFEWNKCFHILX\ntranslation: NCEVSLKDKLKI\ntranslation: NCEVSLKDKLKI\ntranslation: LIFNLSFSDTSQ\ntranslation: LIFNLSFSDTSQ\ntranslation: SYHYHTNVEELTIPETRNNLYI\ntranslation: SYHYHTNVEELTIPETRNNLYI\ntranslation: QQILHNQKGQASSTQQCSSKR\ntranslation: QQILHNQKGQASSTQQCSSKR\ntranslation: SKFFTIKRDKHQARSNAAQNA\ntranslation: SKFFTIKRDKHQARSNAAQNA\ntranslation: ANSSQSKGTSIKHAAMQLKT\ntranslation: ANSSQSKGTSIKHAAMQLKT\ntranslation: SVLSCIAACLMLVPFDCEEFA\ntranslation: SVLSCIAACLMLVPFDCEEFA\ntranslation: LKHRCNQIAFNKQKSFYAWISFI\ntranslation: LKHRCNQIAFNKQKSFYAWISFI\ntranslation: NERDPGIKTLLLIESNLITPMFQ\ntranslation: NERDPGIKTLLLIESNLITPMFQ\ntranslation: KKREILARKKSKRSALENSEEHS\ntranslation: KKREILARKKSKRSALENSEEHS\ntranslation: AECSSLFSRADRFDFFRAKISLF\ntranslation: AECSSLFSRADRFDFFRAKISLF\ntranslation: LNALHYFQEQIVLTSFELKSLFF\ntranslation: LNALHYFQEQIVLTSFELKSLFF\ntranslation: ICLMLGEKCCFWVNRLGKVQDH\ntranslation: ICLMLGEKCCFWVNRLGKVQDH\ntranslation: WSWTFPNLLTQKQHFSPSIRQI\ntranslation: WSWTFPNLLTQKQHFSPSIRQI\ntranslation: FLETQGTGPSETSSVFSSVSF\ntranslation: FLETQGTGPSETSSVFSSVSF\ntranslation: FWKPRELVLLKLVLCSLQYHL\ntranslation: FWKPRELVLLKLVLCSLQYHL\ntranslation: PNDTEENTELVSEGPVPWVSKN\ntranslation: PNDTEENTELVSEGPVPWVSKN\ntranslation: KDEAFSFQVSREAS\ntranslation: KDEAFSFQVSREAS\ntranslation: RMKPSPSRFPEKP\ntranslation: RMKPSPSRFPEKP\ntranslation: GGFSGNLEGEGFIL\ntranslation: GGFSGNLEGEGFIL\ntranslation: EASLETWKEKASSL\ntranslation: EASLETWKEKASSL\ntranslation: RLLWKPGRRRLHP\ntranslation: RLLWKPGRRRLHP\ntranslation: LLVYNNAQGLDELHNQNQMTAF\ntranslation: LLVYNNAQGLDELHNQNQMTAF\ntranslation: RKLSFDFDYVVHPALGHCYTPVK\ntranslation: RKLSFDFDYVVHPALGHCYTPVK\ntranslation: AASSCLHSAETQRAQRNLCSGL\ntranslation: AASSCLHSAETQRAQRNLCSGL\ntranslation: LLLAVCTVQKHREHRGICALGY\ntranslation: LLLAVCTVQKHREHRGICALGY\ntranslation: HSPEHKFLCALCVSALCRQLEA\ntranslation: HSPEHKFLCALCVSALCRQLEA\ntranslation: VDSEGNNDLLVDCRYWAVNCQF\ntranslation: VDSEGNNDLLVDCRYWAVNCQF\ntranslation: LTVRVIMTCWLIVDIGLLIVS\ntranslation: LTVRVIMTCWLIVDIGLLIVS\ntranslation: ELTINSPISTINQQVIITLTVN\ntranslation: ELTINSPISTINQQVIITLTVN\ntranslation: PNWHCISKLITRHRTTR\ntranslation: PNWHCISKLITRHRTTR\ntranslation: LTGIVLANSSLDIVLHD\ntranslation: LTGIVLANSSLDIVLHD\ntranslation: SCSTMSSDEFANTMPVR\ntranslation: SCSTMSSDEFANTMPVR\ntranslation: RVVRCLVMSLLIQCQL\ntranslation: RVVRCLVMSLLIQCQL\ntranslation: TLPDFKLYYKAIVTKTEWHWHK\ntranslation: TLPDFKLYYKAIVTKTEWHWHK\ntranslation: QESCADFAIAQ\ntranslation: QESCADFAIAQ\ntranslation: RNPVQTLLLPR\ntranslation: RNPVQTLLLPR\ntranslation: GILCRLCYCP\ntranslation: GILCRLCYCP\ntranslation: LGNSKVCTGFL\ntranslation: LGNSKVCTGFL\ntranslation: WAIAKSAQDS\ntranslation: WAIAKSAQDS\ntranslation: SALLFLCLDMFRYTNTYYCVTTA\ntranslation: SALLFLCLDMFRYTNTYYCVTTA\ntranslation: FINKYKHLRRSGNRNRLNYPAR\ntranslation: FINKYKHLRRSGNRNRLNYPAR\ntranslation: GLLITTVLAKNLLLLSYMQMRK\ntranslation: GLLITTVLAKNLLLLSYMQMRK\ntranslation: EYVSWHKSSAQHRNFPAEVN\ntranslation: EYVSWHKSSAQHRNFPAEVN\ntranslation: FTSAGKFLCCADDLCQDTYS\ntranslation: FTSAGKFLCCADDLCQDTYS\ntranslation: SPLPGSSCVVLMICARIHIQ\ntranslation: SPLPGSSCVVLMICARIHIQ\ntranslation: LVQLSSGKHCIRPGILLLFNSR\ntranslation: LVQLSSGKHCIRPGILLLFNSR\ntranslation: LKGFIKEIKKNFCIFYPGSTA\ntranslation: LKGFIKEIKKNFCIFYPGSTA\ntranslation: RSAPRIKDTKVLFYFFNKALQ\ntranslation: RSAPRIKDTKVLFYFFNKALQ\ntranslation: VNGPRKQVSGPERTIPIT\ntranslation: VNGPRKQVSGPERTIPIT\ntranslation: GDGNGALRARDLFPWSIH\ntranslation: GDGNGALRARDLFPWSIH\ntranslation: VMGMVRSGPETCFLGPFT\ntranslation: VMGMVRSGPETCFLGPFT\ntranslation: PRGNRTASGYRNNNVKETLISVM\ntranslation: PRGNRTASGYRNNNVKETLISVM\ntranslation: PEVTEQLQVIEITMLRKLLSQS\ntranslation: PEVTEQLQVIEITMLRKLLSQS\ntranslation: ANALYPFSSLICSFFMILSLQ\ntranslation: ANALYPFSSLICSFFMILSLQ\ntranslation: LQTKYHEERADQRAEGIKSIC\ntranslation: LQTKYHEERADQRAEGIKSIC\ntranslation: THSFLLLALSSTIFLNFDTPSI\ntranslation: THSFLLLALSSTIFLNFDTPSI\ntranslation: IGVGLGYVRSLGHMCWRLR\ntranslation: IGVGLGYVRSLGHMCWRLR\ntranslation: VSISNTYGLDYVHNLNLLQC\ntranslation: VSISNTYGLDYVHNLNLLQC\ntranslation: GTVMRSLGQNPTEAELQDMIN\ntranslation: GTVMRSLGQNPTEAELQDMIN\ntranslation: SLIMSCNSASVGFCPRDLITVP\ntranslation: SLIMSCNSASVGFCPRDLITVP\ntranslation: VKSIVASTFIISLFPTTIFMCLD\ntranslation: VKSIVASTFIISLFPTTIFMCLD\ntranslation: KIHCRIHLYYQSLPHNNIHVPR\ntranslation: KIHCRIHLYYQSLPHNNIHVPR\ntranslation: SRHMNIVVGKRLIIKVDATMDF\ntranslation: SRHMNIVVGKRLIIKVDATMDF\ntranslation: GLILETFKCSLIVFQLFGSCIGS\ntranslation: GLILETFKCSLIVFQLFGSCIGS\ntranslation: YNTIHHLFLQVSTYSSQLGKR\ntranslation: YNTIHHLFLQVSTYSSQLGKR\ntranslation: ITLYIIFSCKYLHILASLGRG\ntranslation: ITLYIIFSCKYLHILASLGRG\ntranslation: PLPKLARICRYLQEKMMYSVI\ntranslation: PLPKLARICRYLQEKMMYSVI\ntranslation: QDLKTGTVKVTAVGWSEHPP\ntranslation: QDLKTGTVKVTAVGWSEHPP\ntranslation: GFKNWNGEGDSSRLERASP\ntranslation: GFKNWNGEGDSSRLERASP\ntranslation: LGDARSNRLLSPSPFQFLNP\ntranslation: LGDARSNRLLSPSPFQFLNP\ntranslation: GGCSLQPTAVTFTVPVFKS\ntranslation: GGCSLQPTAVTFTVPVFKS\ntranslation: SGADHTPSAWIPDHCFLDKNHL\ntranslation: SGADHTPSAWIPDHCFLDKNHL\ntranslation: KWFLSKKQWSGIQAEGVWSAP\ntranslation: KWFLSKKQWSGIQAEGVWSAP\ntranslation: NGSYLRSNGLESRQKACGQHQ\ntranslation: NGSYLRSNGLESRQKACGQHQ\ntranslation: SAWKAIVVQVTFDLLTFDLEHR\ntranslation: SAWKAIVVQVTFDLLTFDLEHR\ntranslation: ILCSKSKVKRSKVTCTTIAFHAE\ntranslation: ILCSKSKVKRSKVTCTTIAFHAE\ntranslation: FCVLNQKSKDQKSLAQLLLSMQN\ntranslation: FCVLNQKSKDQKSLAQLLLSMQN\ntranslation: MSDFRSVCRRQMELVIIMPHR\ntranslation: MSDFRSVCRRQMELVIIMPHR\ntranslation: PMRHNYNKLHLPTTNRPKIAH\ntranslation: PMRHNYNKLHLPTTNRPKIAH\ntranslation: QKPFSIECDKISFKFCLLVSY\ntranslation: QKPFSIECDKISFKFCLLVSY\ntranslation: VTTLQLFNRGAAGSVAEW\ntranslation: VTTLQLFNRGAAGSVAEW\ntranslation: PLCNTACCTSIKELQGGH\ntranslation: PLCNTACCTSIKELQGGH\ntranslation: HSATLPAAPLLKSCKVVT\ntranslation: HSATLPAAPLLKSCKVVT\ntranslation: EGETVPLLRDTMTLMPVSMKG\ntranslation: EGETVPLLRDTMTLMPVSMKG\ntranslation: VPFIETGISVMVSRSNGTVSPS\ntranslation: VPFIETGISVMVSRSNGTVSPS\ntranslation: ALHRDRHQCHGVTQQWDCLTF\ntranslation: ALHRDRHQCHGVTQQWDCLTF\ntranslation: QELAPNTTWLSLPLPVRTKECDV\ntranslation: QELAPNTTWLSLPLPVRTKECDV\ntranslation: RNLLQILLGYPFLYQLELKSVMY\ntranslation: RNLLQILLGYPFLYQLELKSVMY\ntranslation: HTSHSLVLTGKGRDNQVVFGASS\ntranslation: HTSHSLVLTGKGRDNQVVFGASS\ntranslation: KAVTFKLQKYRRINDCSPSSMLY\ntranslation: KAVTFKLQKYRRINDCSPSSMLY\ntranslation: KQLHSSYRNIEELMIVHQAACCT\ntranslation: KQLHSSYRNIEELMIVHQAACCT\ntranslation: MYSMLLGEQSLILRYFCSLNVTA\ntranslation: MYSMLLGEQSLILRYFCSLNVTA\ntranslation: QGHLLGSSETSISKSRQNDLI\ntranslation: QGHLLGSSETSISKSRQNDLI\ntranslation: QIRSFCLDFDMDVSEDPSRCP\ntranslation: QIRSFCLDFDMDVSEDPSRCP\ntranslation: KLGHSVLILIWMFQKTLADVP\ntranslation: KLGHSVLILIWMFQKTLADVP\ntranslation: EKGLKNVKNEYEQLNYAKQLKE\ntranslation: EKGLKNVKNEYEQLNYAKQLKE\ntranslation: FQNMKRFTISRCIHSEGHTGV\ntranslation: FQNMKRFTISRCIHSEGHTGV\ntranslation: PKYEKIHNFKVHTFRGPHWC\ntranslation: PKYEKIHNFKVHTFRGPHWC\ntranslation: TPVWPSECMHLEIVNLFIFW\ntranslation: TPVWPSECMHLEIVNLFIFW\ntranslation: NPILESISTIGFTTSMLDQDIPM\ntranslation: NPILESISTIGFTTSMLDQDIPM\ntranslation: SYSRVHINNRVYDLDVGSGHPD\ntranslation: SYSRVHINNRVYDLDVGSGHPD\ntranslation: KILRSDITSLPLSSMGYFPEFRI\ntranslation: KILRSDITSLPLSSMGYFPEFRI\ntranslation: ILNSGKYPIEESGSEVISLLRI\ntranslation: ILNSGKYPIEESGSEVISLLRI\ntranslation: LVCYIDIALHVSTGKYLDFNDSL\ntranslation: LVCYIDIALHVSTGKYLDFNDSL\ntranslation: QAIIEIQIFPSRDMQSNVNVTYK\ntranslation: QAIIEIQIFPSRDMQSNVNVTYK\ntranslation: QKFLLLLPGLETVEMMEE\ntranslation: QKFLLLLPGLETVEMMEE\ntranslation: DSSIISTVSKPGNRSRNF\ntranslation: DSSIISTVSKPGNRSRNF\ntranslation: IPPSFQQFPNQATGAEIF\ntranslation: IPPSFQQFPNQATGAEIF\ntranslation: FLHHFNSFQTRQQEQKFL\ntranslation: FLHHFNSFQTRQQEQKFL\ntranslation: SPFHKSGAFWEAETFRSQSNF\ntranslation: SPFHKSGAFWEAETFRSQSNF\ntranslation: PHFISQVHSGKQKHLEVKAIL\ntranslation: PHFISQVHSGKQKHLEVKAIL\ntranslation: SECLCPPKIHMLKPAYQC\ntranslation: SECLCPPKIHMLKPAYQC\ntranslation: HIGRQALTYEFWGDTSIQT\ntranslation: HIGRQALTYEFWGDTSIQT\ntranslation: SQTLIKPSDVMRTHLLSQEQHR\ntranslation: SQTLIKPSDVMRTHLLSQEQHR\ntranslation: VRFAEFLLLFLTFPYEHACVGL\ntranslation: VRFAEFLLLFLTFPYEHACVGL\ntranslation: STQHRHAHKERLKKVKGTRQIL\ntranslation: STQHRHAHKERLKKVKGTRQIL\ntranslation: SDFFMWVSSSSSRWSPKIDCSFN\ntranslation: SDFFMWVSSSSSRWSPKIDCSFN\ntranslation: LKEQSIFGDHRDEEEETHMKKS\ntranslation: LKEQSIFGDHRDEEEETHMKKS\ntranslation: DILRDPCSATDPLCRSCSAIDSL\ntranslation: DILRDPCSATDPLCRSCSAIDSL\ntranslation: QRVYCRTRSTQRVSSRTRITQNV\ntranslation: QRVYCRTRSTQRVSSRTRITQNV\ntranslation: RESIAEQDRHKGSVAEQGSRRMS\ntranslation: RESIAEQDRHKGSVAEQGSRRMS\ntranslation: CNMKVICLIEISSCLVKHT\ntranslation: CNMKVICLIEISSCLVKHT\ntranslation: SVCLTRQLDISIKQMTFILQ\ntranslation: SVCLTRQLDISIKQMTFILQ\ntranslation: LDDQFNQAYIWRQILVSRERRS\ntranslation: LDDQFNQAYIWRQILVSRERRS\ntranslation: TQMSHSWGMDTQTVVILGWCKS\ntranslation: TQMSHSWGMDTQTVVILGWCKS\ntranslation: KARIVEYEKEMEKMKNLIPFD\ntranslation: KARIVEYEKEMEKMKNLIPFD\ntranslation: DQMELSSSSSPSLSHILQSWPL\ntranslation: DQMELSSSSSPSLSHILQSWPL\ntranslation: SNGIKFFIFSISFSYSTILAF\ntranslation: SNGIKFFIFSISFSYSTILAF\ntranslation: AAQRLRSYTVNLQMLTQCYTVNF\ntranslation: AAQRLRSYTVNLQMLTQCYTVNF\ntranslation: CTKIKVIYSQSANVDTMLHCKF\ntranslation: CTKIKVIYSQSANVDTMLHCKF\ntranslation: LNGVLVTVAFLVSMYLFCLFFGD\ntranslation: LNGVLVTVAFLVSMYLFCLFFGD\ntranslation: KWCFSNCCISGFHVLILPFFWRF\ntranslation: KWCFSNCCISGFHVLILPFFWRF\ntranslation: ESPKKRQNKYMETRNATVTKTPF\ntranslation: ESPKKRQNKYMETRNATVTKTPF\ntranslation: NRQKKGRISTWKPEMQQLLKHHL\ntranslation: NRQKKGRISTWKPEMQQLLKHHL\ntranslation: RGYLQNCHLTMTNQTNLKT\ntranslation: RGYLQNCHLTMTNQTNLKT\ntranslation: AFGTSLFVPPSHIRFVFPEP\ntranslation: AFGTSLFVPPSHIRFVFPEP\ntranslation: PSAPASSCPQAISALCFLSQ\ntranslation: PSAPASSCPQAISALCFLSQ\ntranslation: VGSGNTKRIWLGGTKRLVPK\ntranslation: VGSGNTKRIWLGGTKRLVPK\ntranslation: LAQETQSGYGLGARRGWCRR\ntranslation: LAQETQSGYGLGARRGWCRR\ntranslation: WLRKHKADMAWGHEEAGAEG\ntranslation: WLRKHKADMAWGHEEAGAEG\ntranslation: SALHYSYIHSFFHLGSFSLSNVRG\ntranslation: SALHYSYIHSFFHLGSFSLSNVRG\ntranslation: CTSLLIHPFFLSFRFFFIVKCQRH\ntranslation: CTSLLIHPFFLSFRFFFIVKCQRH\ntranslation: FQFILHYHLTLCLNIHVKKVFP\ntranslation: FQFILHYHLTLCLNIHVKKVFP\ntranslation: SNLSCIIILPSVSIYMLRRSFP\ntranslation: SNLSCIIILPSVSIYMLRRSFP\ntranslation: GKDLLNMYIETEGKMIMQDKLE\ntranslation: GKDLLNMYIETEGKMIMQDKLE\ntranslation: HTYRTSLYIHGMKCHSFIYTI\ntranslation: HTYRTSLYIHGMKCHSFIYTI\ntranslation: LEISSFSLLHSSAGNWSLNSSIL\ntranslation: LEISSFSLLHSSAGNWSLNSSIL\ntranslation: KMEEFKDQLPADECNKLKEEISK\ntranslation: KMEEFKDQLPADECNKLKEEISK\ntranslation: IFGLLKLMFEDRLLSLLEGFA\ntranslation: IFGLLKLMFEDRLLSLLEGFA\ntranslation: MANPSSRDKSRSSNINLSRPKI\ntranslation: MANPSSRDKSRSSNINLSRPKI\ntranslation: QNVGCFFHKIISIGGYCSMTA\ntranslation: QNVGCFFHKIISIGGYCSMTA\ntranslation: REELKALLNNVFANDFQDFALL\ntranslation: REELKALLNNVFANDFQDFALL\ntranslation: EKSSRHFLTMFLRMTFKILLCW\ntranslation: EKSSRHFLTMFLRMTFKILLCW\ntranslation: QQSKILKVIRKNIVKKCLELFS\ntranslation: QQSKILKVIRKNIVKKCLELFS\ntranslation: LGSFLMVIFFASSRFFSLRKLSS\ntranslation: LGSFLMVIFFASSRFFSLRKLSS\ntranslation: QKIVYEEKRTWKKQRRLPLKMIQ\ntranslation: QKIVYEEKRTWKKQRRLPLKMIQ\ntranslation: EDSLRREKNLEEAKKITIKNDP\ntranslation: EDSLRREKNLEEAKKITIKNDP\ntranslation: SVQCFNLTQAYAEENVF\ntranslation: SVQCFNLTQAYAEENVF\ntranslation: RKYVPAFRRSGWLPHRVM\ntranslation: RKYVPAFRRSGWLPHRVM\ntranslation: HHPMRQPARTPERRHILP\ntranslation: HHPMRQPARTPERRHILP\ntranslation: VNCAFSFNFLSLLKLKICPTFHI\ntranslation: VNCAFSFNFLSLLKLKICPTFHI\ntranslation: DDYDKKKLVKAFKKKFACNGTVIE\ntranslation: DDYDKKKLVKAFKKKFACNGTVIE\ntranslation: IGVYQCLKNVSRNIAEGSSEVG\ntranslation: IGVYQCLKNVSRNIAEGSSEVG\ntranslation: LVSINVLKMFQETLLRVAVKL\ntranslation: LVSINVLKMFQETLLRVAVKL\ntranslation: NPPLGINILCHNKHPFSFTEIS\ntranslation: NPPLGINILCHNKHPFSFTEIS\ntranslation: IIKVSIHQGITVFNVYAPNDRAT\ntranslation: IIKVSIHQGITVFNVYAPNDRAT\ntranslation: ATSTYRQHLHSNKHQFKIIRQIF\ntranslation: ATSTYRQHLHSNKHQFKIIRQIF\ntranslation: LGFPAQSPASQ\ntranslation: LGFPAQSPASQ\ntranslation: WGFLLNHLHLR\ntranslation: WGFLLNHLHLR\ntranslation: GVSCSITCISD\ntranslation: GVSCSITCISD\ntranslation: SEMQVIEQETP\ntranslation: SEMQVIEQETP\ntranslation: LSSVSIFSLYSRAICCFFSAP\ntranslation: LSSVSIFSLYSRAICCFFSAP\ntranslation: LAPSQFSLCILEPSAVFSQHL\ntranslation: LAPSQFSLCILEPSAVFSQHL\ntranslation: GAEKKQQMAREYREKIETEL\ntranslation: GAEKKQQMAREYREKIETEL\ntranslation: HLSILFSIREIVGIRIRIVVKY\ntranslation: HLSILFSIREIVGIRIRIVVKY\ntranslation: YFTTILILIPTISLIENKILK\ntranslation: YFTTILILIPTISLIENKILK\ntranslation: QNEPIPERSPQNGESGKDMDDDK\ntranslation: QNEPIPERSPQNGESGKDMDDDK\ntranslation: RMNLFQKDLLRMVNQERIWMMTK\ntranslation: RMNLFQKDLLRMVNQERIWMMTK\ntranslation: GVIEDEQVLVLVFPKGKDKGVQD\ntranslation: GVIEDEQVLVLVFPKGKDKGVQD\ntranslation: ILDTLVFAFGKYQHKNLLILYDA\ntranslation: ILDTLVFAFGKYQHKNLLILYDA\ntranslation: SWTPLSLPLGNTSTRTCSSSMTP\ntranslation: SWTPLSLPLGNTSTRTCSSSMTP\ntranslation: LTSHRSFRSFWMSHRVSALFLS\ntranslation: LTSHRSFRSFWMSHRVSALFLS\ntranslation: AQEKCRYPVGHPEGPKRPVTSE\ntranslation: AQEKCRYPVGHPEGPKRPVTSE\ntranslation: LLINSIVSNYFCKYIIFPNWFL\ntranslation: LLINSIVSNYFCKYIIFPNWFL\ntranslation: KGSFHQCVTVRNKLMSS\ntranslation: KGSFHQCVTVRNKLMSS\ntranslation: KVHFTNVSQSETNSCL\ntranslation: KVHFTNVSQSETNSCL\ntranslation: RFISPMCHSQKQTHVF\ntranslation: RFISPMCHSQKQTHVF\ntranslation: SHVILLILMLCGTGSSSISAMPL\ntranslation: SHVILLILMLCGTGSSSISAMPL\ntranslation: TRYIAYFDVMWYRKFFHLCYAT\ntranslation: TRYIAYFDVMWYRKFFHLCYAT\ntranslation: SGIAEMEELPVPHNIKISNITC\ntranslation: SGIAEMEELPVPHNIKISNITC\ntranslation: STFNPPKPTKTRSPSQASLPIC\ntranslation: STFNPPKPTKTRSPSQASLPIC\ntranslation: PHSIHPNPQRPGLQARPLFPS\ntranslation: PHSIHPNPQRPGLQARPLFPS\ntranslation: HIQSTQTHKDQVSKPGLSSHL\ntranslation: HIQSTQTHKDQVSKPGLSSHL\ntranslation: ADGKRGLAWRPGLCGFGWIECG\ntranslation: ADGKRGLAWRPGLCGFGWIECG\ntranslation: QMGREAWLGDLVFVGLGGLNV\ntranslation: QMGREAWLGDLVFVGLGGLNV\ntranslation: QGRHCGLFSLNMLKWKDFPSVT\ntranslation: QGRHCGLFSLNMLKWKDFPSVT\ntranslation: VTLGKSFHFSMFKENSPQCLP\ntranslation: VTLGKSFHFSMFKENSPQCLP\ntranslation: FIGLTCIKIPCECAHSHISPTTI\ntranslation: FIGLTCIKIPCECAHSHISPTTI\ntranslation: AVKRLLDSSQCQQMSQAPTKYNH\ntranslation: AVKRLLDSSQCQQMSQAPTKYNH\ntranslation: QLRDFWTVHSVNKCHRLRPSITT\ntranslation: QLRDFWTVHSVNKCHRLRPSITT\ntranslation: EESHGIVANSMYDAVTKKHFSDS\ntranslation: EESHGIVANSMYDAVTKKHFSDS\ntranslation: KKAMALWLIPCMMQSQRNTFLT\ntranslation: KKAMALWLIPCMMQSQRNTFLT\ntranslation: RVRKVFLCDCIIHGISHNAMAFF\ntranslation: RVRKVFLCDCIIHGISHNAMAFF\ntranslation: ESEKCFFVTASYMELATMPWLSS\ntranslation: ESEKCFFVTASYMELATMPWLSS\ntranslation: LKCWLYVVNVRVIQSLVLSSLFW\ntranslation: LKCWLYVVNVRVIQSLVLSSLFW\ntranslation: EMLVVCGKCKSNTVSCTFLTVL\ntranslation: EMLVVCGKCKSNTVSCTFLTVL\ntranslation: PKTVRKVQETVLLLHLPHTTNIS\ntranslation: PKTVRKVQETVLLLHLPHTTNIS\ntranslation: QNSEESTRDCITLTFTTYNQHF\ntranslation: QNSEESTRDCITLTFTTYNQHF\ntranslation: IPHHPTKNIKHKL\ntranslation: IPHHPTKNIKHKL\ntranslation: FLITQLKTLNTNY\ntranslation: FLITQLKTLNTNY\ntranslation: HILPILHPSRLPSPTHRTNLHS\ntranslation: HILPILHPSRLPSPTHRTNLHS\ntranslation: TYFLFYTLVGSLPLLIALIYTH\ntranslation: TYFLFYTLVGSLPLLIALIYTH\ntranslation: FLSSHLDLLNLYFYLRLIYSTSI\ntranslation: FLSSHLDLLNLYFYLRLIYSTSI\ntranslation: ISAAELRHVMTNLGEKLTDEEVD\ntranslation: ISAAELRHVMTNLGEKLTDEEVD\ntranslation: PSLVSSTS\ntranslation: PSLVSSTS\ntranslation: IFSFLYLL\ntranslation: IFSFLYLL\ntranslation: QEVEETKD\ntranslation: QEVEETKD\ntranslation: GSIFCALITMLGHIRIGHGNRMH\ntranslation: GSIFCALITMLGHIRIGHGNRMH\ntranslation: RHLLYHLCRHTHHSAKLALIFYL\ntranslation: RHLLYHLCRHTHHSAKLALIFYL\ntranslation: RNQISTSTNGKLTTYEPSSNS\ntranslation: RNQISTSTNGKLTTYEPSSNS\ntranslation: PTSNRNSDSVLVCLVLCINGML\ntranslation: PTSNRNSDSVLVCLVLCINGML\ntranslation: HNIPFIHRTKQTSTESLLRLEV\ntranslation: HNIPFIHRTKQTSTESLLRLEV\ntranslation: FYQRAELRTLWLKYNFSP\ntranslation: FYQRAELRTLWLKYNFSP\ntranslation: GEKNYISATVFSTQPFDK\ntranslation: GEKNYISATVFSTQPFDK\ntranslation: GRKIIFQPQCSQLSPLIK\ntranslation: GRKIIFQPQCSQLSPLIK\ntranslation: LNYNAYFRMRNISFLPLKCHIEND\ntranslation: LNYNAYFRMRNISFLPLKCHIEND\ntranslation: DNIIFIIPLWFWFSCWKISEC\ntranslation: DNIIFIIPLWFWFSCWKISEC\ntranslation: TISFSLSHSGSGSPAGKSLN\ntranslation: TISFSLSHSGSGSPAGKSLN\ntranslation: AFRDFPAGEPEPEWDNENDIV\ntranslation: AFRDFPAGEPEPEWDNENDIV\ntranslation: HSEIFQQENQNQSGIMKMILS\ntranslation: HSEIFQQENQNQSGIMKMILS\ntranslation: LLHRLPGVLTRVHRCVLHRHLVI\ntranslation: LLHRLPGVLTRVHRCVLHRHLVI\ntranslation: CYIASLVFLHEFIDVYFIATLSF\ntranslation: CYIASLVFLHEFIDVYFIATLSF\ntranslation: ATSPPWCSYTSSSMCTSSPPCHS\ntranslation: ATSPPWCSYTSSSMCTSSPPCHS\ntranslation: CLITDGFCYKHQCDTCRTLQR\ntranslation: CLITDGFCYKHQCDTCRTLQR\ntranslation: PGLTMCLLQESL\ntranslation: PGLTMCLLQESL\ntranslation: PGSLCACSRSP\ntranslation: PGSLCACSRSP\ntranslation: RAHYVPAPGVP\ntranslation: RAHYVPAPGVP\ntranslation: QGLLEQAHSEPG\ntranslation: QGLLEQAHSEPG\ntranslation: RDSWSRHIVSP\ntranslation: RDSWSRHIVSP\ntranslation: QRPVPGPMLVDFHVLISLMFTQP\ntranslation: QRPVPGPMLVDFHVLISLMFTQP\ntranslation: GCVNIKEINTWKSTNMGPGTGR\ntranslation: GCVNIKEINTWKSTNMGPGTGR\ntranslation: EKTSVCLFGLYWESLSISSSCES\ntranslation: EKTSVCLFGLYWESLSISSSCES\ntranslation: ELSQLLEMLNDSQYNPKRQTLVF\ntranslation: ELSQLLEMLNDSQYNPKRQTLVF\ntranslation: CMDILKVCCRISVCLSFSLPT\ntranslation: CMDILKVCCRISVCLSFSLPT\ntranslation: GVGRENDRQTDILQQTFRMSIH\ntranslation: GVGRENDRQTDILQQTFRMSIH\ntranslation: GVQRIVLTVGLNDFLSILQGSF\ntranslation: GVQRIVLTVGLNDFLSILQGSF\ntranslation: EGALQYGQKIIKPYSKHYPLYS\ntranslation: EGALQYGQKIIKPYSKHYPLYS\ntranslation: KEPCNMDRKSLSPTVSTILCT\ntranslation: KEPCNMDRKSLSPTVSTILCT\ntranslation: RVLRVSLHHFRRHLRLNIFCSH\ntranslation: RVLRVSLHHFRRHLRLNIFCSH\ntranslation: EYFESPFTISDGIYGSTFFVA\ntranslation: EYFESPFTISDGIYGSTFFVA\ntranslation: SCFSKRFPLLPLLPLSSSMSF\ntranslation: SCFSKRFPLLPLLPLSSSMSF\ntranslation: KLIEDERGRRGRRGNLLEKH\ntranslation: KLIEDERGRRGRRGNLLEKH\ntranslation: YVINWSPACQKDCSPLQLAHFQ\ntranslation: YVINWSPACQKDCSPLQLAHFQ\ntranslation: TENGPTEVDYSPSDMQDSSLSHK\ntranslation: TENGPTEVDYSPSDMQDSSLSHK\ntranslation: LKMGQLKWTTVLLTCRTPVYHI\ntranslation: LKMGQLKWTTVLLTCRTPVYHI\ntranslation: HGYGEISPRLSFHYHCLPGCAC\ntranslation: HGYGEISPRLSFHYHCLPGCAC\ntranslation: TAMGKLARGLAFIITVSQGVLV\ntranslation: TAMGKLARGLAFIITVSQGVLV\ntranslation: TSTPWETVIMKAKPRANFPIAV\ntranslation: TSTPWETVIMKAKPRANFPIAV\ntranslation: AHYANLFSTILSLKTLTMKGKG\ntranslation: AHYANLFSTILSLKTLTMKGKG\ntranslation: RDFIFGESATNQ\ntranslation: RDFIFGESATNQ\ntranslation: VISYLENQQPTS\ntranslation: VISYLENQQPTS\ntranslation: GWLVADSPNMKS\ntranslation: GWLVADSPNMKS\ntranslation: FEVSLISHCWVVISRLLMRYL\ntranslation: FEVSLISHCWVVISRLLMRYL\ntranslation: QISHQQPTNHHPTMTNQTNLK\ntranslation: QISHQQPTNHHPTMTNQTNLK\ntranslation: TKSCKSRGSNLRVHFKNTRETAQ\ntranslation: TKSCKSRGSNLRVHFKNTRETAQ\ntranslation: RNHANQEVPIFVFTLRTLVKLL\ntranslation: RNHANQEVPIFVFTLRTLVKLL\ntranslation: ANVSENITNKRFFFFFIQQI\ntranslation: ANVSENITNKRFFFFFIQQI\ntranslation: QMSLKISQIKDFFFSSFSK\ntranslation: QMSLKISQIKDFFFSSFSK\ntranslation: NLLNEEKKKSFICDIFRDIC\ntranslation: NLLNEEKKKSFICDIFRDIC\ntranslation: CASSKLWSVMVIICCKMSPMSC\ntranslation: CASSKLWSVMVIICCKMSPMSC\ntranslation: QDIGDILQQIMTITDQSLDEA\ntranslation: QDIGDILQQIMTITDQSLDEA\ntranslation: DSVSILSMLCCCSVVVAVVLC\ntranslation: DSVSILSMLCCCSVVVAVVLC\ntranslation: HTKQQQQQQNNNKACSVLRHC\ntranslation: HTKQQQQQQNNNKACSVLRHC\ntranslation: HKTTATTTEQQQSMLSIETLS\ntranslation: HKTTATTTEQQQSMLSIETLS\ntranslation: KSITNKTTYNSYFNTKVTASPFP\ntranslation: KSITNKTTYNSYFNTKVTASPFP\ntranslation: RERGSSDLSIKITVISCFVCYRF\ntranslation: RERGSSDLSIKITVISCFVCYRF\ntranslation: LIEEYAMSDFRSVCR\ntranslation: LIEEYAMSDFRSVCR\ntranslation: TTNRPKIAHCILFNQ\ntranslation: TTNRPKIAHCILFNQ\ntranslation: RQTDLKSLIAYSSIS\ntranslation: RQTDLKSLIAYSSIS\ntranslation: IMLGIFFNVHSAVLIEDVPF\ntranslation: IMLGIFFNVHSAVLIEDVPF\ntranslation: EGNVLNQHSGMDIEKYSEHY\ntranslation: EGNVLNQHSGMDIEKYSEHY\ntranslation: KGTSSINTAEWTLKNIPSII\ntranslation: KGTSSINTAEWTLKNIPSII\ntranslation: QFLVLHKLTPLTVQWSPIQTDR\ntranslation: QFLVLHKLTPLTVQWSPIQTDR\ntranslation: SDQFEWVTIEQSGELVYEAPETV\ntranslation: SDQFEWVTIEQSGELVYEAPETV\ntranslation: KDGKLQIGDKLLAVNNVCLE\ntranslation: KDGKLQIGDKLLAVNNVCLE\ntranslation: LLNIRYSLLKVYLQSEVCHP\ntranslation: LLNIRYSLLKVYLQSEVCHP\ntranslation: EEHQCLVQSHLLYQFLDHFPHL\ntranslation: EEHQCLVQSHLLYQFLDHFPHL\ntranslation: SEESDLEIDKEGVIEPDTDAPQ\ntranslation: SEESDLEIDKEGVIEPDTDAPQ\ntranslation: EEFKENAYKDLAWKHGNIHLS\ntranslation: EEFKENAYKDLAWKHGNIHLS\ntranslation: KNLKKMLIKTWHGSMETFTS\ntranslation: KNLKKMLIKTWHGSMETFTS\ntranslation: IEENVNSLSQISADLPAFVSVVA\ntranslation: IEENVNSLSQISADLPAFVSVVA\ntranslation: SFNGNKVTITLLPDNLSQTIYL\ntranslation: SFNGNKVTITLLPDNLSQTIYL\ntranslation: VSCTRQAW\ntranslation: VSCTRQAW\ntranslation: FPAPGKP\ntranslation: FPAPGKP\ntranslation: FLHQASL\ntranslation: FLHQASL\ntranslation: PGLPGAGN\ntranslation: PGLPGAGN\ntranslation: QACLVQE\ntranslation: QACLVQE\ntranslation: RLAWCRK\ntranslation: RLAWCRK\ntranslation: ILSVSQTPLCFSFVALFPGPHEK\ntranslation: ILSVSQTPLCFSFVALFPGPHEK\ntranslation: FSHEVQEIGLQRKNIKESGKLTE\ntranslation: FSHEVQEIGLQRKNIKESGKLTE\ntranslation: CGRHLAYFVD\ntranslation: CGRHLAYFVD\ntranslation: VVGIWHILWT\ntranslation: VVGIWHILWT\ntranslation: ESTKYAKCLPQ\ntranslation: ESTKYAKCLPQ\ntranslation: SPQNMPNAYH\ntranslation: SPQNMPNAYH\ntranslation: VHKICQMPTT\ntranslation: VHKICQMPTT\ntranslation: ITGILPVFQCNNCCHHFVFFML\ntranslation: ITGILPVFQCNNCCHHFVFFML\ntranslation: SQEYYQYFNAIIVVITLCFLCW\ntranslation: SQEYYQYFNAIIVVITLCFLCW\ntranslation: LRTKIILMYMYHTLICNLHTDA\ntranslation: LRTKIILMYMYHTLICNLHTDA\ntranslation: MHQCVDCISVYGTCTSKLFLSL\ntranslation: MHQCVDCISVYGTCTSKLFLSL\ntranslation: ASVCRLHISVWYMYIKIIFVL\ntranslation: ASVCRLHISVWYMYIKIIFVL\ntranslation: LAHLSGPTDPCSTAVHMEPFS\ntranslation: LAHLSGPTDPCSTAVHMEPFS\ntranslation: SPISQDRLTHVQLLFTWNPS\ntranslation: SPISQDRLTHVQLLFTWNPS\ntranslation: RRVPCEQQLNMGQSVLRDGR\ntranslation: RRVPCEQQLNMGQSVLRDGR\ntranslation: VVCSSKKGHGAILCKAMTQLHP\ntranslation: VVCSSKKGHGAILCKAMTQLHP\ntranslation: CMFFKERSWGNSVQGNDPAPS\ntranslation: CMFFKERSWGNSVQGNDPAPS\ntranslation: DGAGSLPCTELPHDLSLKNIQ\ntranslation: DGAGSLPCTELPHDLSLKNIQ\ntranslation: RDKTFKTSVCPHTASNLSTPPKV\ntranslation: RDKTFKTSVCPHTASNLSTPPKV\ntranslation: ETKLLKQVFVLILPQTLAHHQRY\ntranslation: ETKLLKQVFVLILPQTLAHHQRY\ntranslation: PTSTSLSFSLLTIYVFAFNIV\ntranslation: PTSTSLSFSLLTIYVFAFNIV\ntranslation: LLLLPFLSPSSQSMSSPSTS\ntranslation: LLLLPFLSPSSQSMSSPSTS\ntranslation: YFYFPFFLPPHNLCLRLQHR\ntranslation: YFYFPFFLPPHNLCLRLQHR\ntranslation: DDVEGEDIDCEEGERKGSRSR\ntranslation: DDVEGEDIDCEEGERKGSRSR\ntranslation: PAQQQVSSKAQITPPPVHIIDNS\ntranslation: PAQQQVSSKAQITPPPVHIIDNS\ntranslation: GIINDMHWWRCYLCFRRYLLLSW\ntranslation: GIINDMHWWRCYLCFRRYLLLSW\ntranslation: LNGQIQNKPQEKGVSTILNLPS\ntranslation: LNGQIQNKPQEKGVSTILNLPS\ntranslation: FEGRFKMVDTPFSWGLFCICPL\ntranslation: FEGRFKMVDTPFSWGLFCICPL\ntranslation: QVFFLVQSNSYKRLLYKDFLKLL\ntranslation: QVFFLVQSNSYKRLLYKDFLKLL\ntranslation: LRPISLPGVSSTEDLQDLFRKT\ntranslation: LRPISLPGVSSTEDLQDLFRKT\ntranslation: LVLCMVIICFEVSLISHCW\ntranslation: LVLCMVIICFEVSLISHCW\ntranslation: PETTKTPGIKGFKFNSHSLQIQ\ntranslation: PETTKTPGIKGFKFNSHSLQIQ\ntranslation: LYLERVTIELETFYSGRLGSFW\ntranslation: LYLERVTIELETFYSGRLGSFW\ntranslation: PLRPTHTPPMKKLPTTHPSITYM\ntranslation: PLRPTHTPPMKKLPTTHPSITYM\ntranslation: GATNGAESKTLIYQMAPAVSK\ntranslation: GATNGAESKTLIYQMAPAVSK\ntranslation: HSRPSRKHNVFVEFRSGINVSCL\ntranslation: HSRPSRKHNVFVEFRSGINVSCL\ntranslation: QAADIDTRSEFYKHIVLSGGSTM\ntranslation: QAADIDTRSEFYKHIVLSGGSTM\ntranslation: RQLTLIPDLNSTNTLCFLEGLLC\ntranslation: RQLTLIPDLNSTNTLCFLEGLLC\ntranslation: FIQHLYKNHQDSSWQVKMFLNS\ntranslation: FIQHLYKNHQDSSWQVKMFLNS\ntranslation: KYFTDLFDYLPLTALVDGQIF\ntranslation: KYFTDLFDYLPLTALVDGQIF\ntranslation: NILQIFLTIFLSLPWWMGRSS\ntranslation: NILQIFLTIFLSLPWWMGRSS\ntranslation: EDLPIHQGSERKIVKKICKIF\ntranslation: EDLPIHQGSERKIVKKICKIF\ntranslation: GVSLQELNPEMGTDNDSENWK\ntranslation: GVSLQELNPEMGTDNDSENWK\ntranslation: PSNFHYHCQFPFLDSIPGEKHL\ntranslation: PSNFHYHCQFPFLDSIPGEKHL\ntranslation: LPIFTIIVSSHFWIQFLERNT\ntranslation: LPIFTIIVSSHFWIQFLERNT\ntranslation: FQFSLSLSVPISGFNSWRETP\ntranslation: FQFSLSLSVPISGFNSWRETP\ntranslation: GDQKEECTSSGWAGFPLISL\ntranslation: GDQKEECTSSGWAGFPLISL\ntranslation: EGLSEGIQPNQMKYILLFGHP\ntranslation: EGLSEGIQPNQMKYILLFGHP\ntranslation: RLIRGNPAQPDEVHSSFWSP\ntranslation: RLIRGNPAQPDEVHSSFWSP\ntranslation: RFHISQMRLVVQFILQVYSEHF\ntranslation: RFHISQMRLVVQFILQVYSEHF\ntranslation: LVHIYTVLHSMICQGRDCGV\ntranslation: LVHIYTVLHSMICQGRDCGV\ntranslation: SPHNPSPDKSYYEVRCRCEQV\ntranslation: SPHNPSPDKSYYEVRCRCEQV\ntranslation: HPTIPPLTNHTMKYGVDVNK\ntranslation: HPTIPPLTNHTMKYGVDVNK\ntranslation: LHKLFVMVLKYLCIKVRINR\ntranslation: LHKLFVMVLKYLCIKVRINR\ntranslation: PIDSNLDAEILQDHHKQLMQ\ntranslation: PIDSNLDAEILQDHHKQLMQ\ntranslation: LLILTLMQRYFRTITNSLCK\ntranslation: LLILTLMQRYFRTITNSLCK\ntranslation: GRIHGLSPVTDESSIKLGLVL\ntranslation: GRIHGLSPVTDESSIKLGLVL\ntranslation: FSTRPNFMLLSSVTGLRPCIRP\ntranslation: FSTRPNFMLLSSVTGLRPCIRP\ntranslation: QLNKSGYIHYCESIHHSSVEIPE\ntranslation: QLNKSGYIHYCESIHHSSVEIPE\ntranslation: SSTSLGIFTIVSQYTIALLKFLK\ntranslation: SSTSLGIFTIVSQYTIALLKFLK\ntranslation: LTKRLQLGNYESVEDVSTFLSKQ\ntranslation: LTKRLQLGNYESVEDVSTFLSKQ\ntranslation: LSVMHFNSLKYHYTCSKDYQQM\ntranslation: LSVMHFNSLKYHYTCSKDYQQM\ntranslation: CLSCILIHSNIITLAVKIINKC\ntranslation: CLSCILIHSNIITLAVKIINKC\ntranslation: ESIYSYNLILLDQSPLHLQFGSN\ntranslation: ESIYSYNLILLDQSPLHLQFGSN\ntranslation: KKLTAMEFACTSILVNALIHLLN\ntranslation: KKLTAMEFACTSILVNALIHLLN\ntranslation: FNKWMRALTSMEVHANSIAVSF\ntranslation: FNKWMRALTSMEVHANSIAVSF\ntranslation: MACVCVRACVCSESTS\ntranslation: MACVCVRACVCSESTS\ntranslation: WHVCVCVRVYVLSPLL\ntranslation: WHVCVCVRVYVLSPLL\ntranslation: KKWTQNIHTHAHTHMP\ntranslation: KKWTQNIHTHAHTHMP\ntranslation: RSGLRTYTRTHTHTCH\ntranslation: RSGLRTYTRTHTHTCH\ntranslation: EVDSEHTHARTHTHA\ntranslation: EVDSEHTHARTHTHA\ntranslation: DKKQMLSNQASWRKANLTCKIAI\ntranslation: DKKQMLSNQASWRKANLTCKIAI\ntranslation: TKSRCSAIRPHGGKLISPAKLQS\ntranslation: TKSRCSAIRPHGGKLISPAKLQS\ntranslation: KVSSKNSLESYAFNMKATVED\ntranslation: KVSSKNSLESYAFNMKATVED\ntranslation: LCCQAIKFSFRSIRGQISSQSVSP\ntranslation: LCCQAIKFSFRSIRGQISSQSVSP\ntranslation: SAVRLSSSLLGAFEVKSVPRVSHL\ntranslation: SAVRLSSSLLGAFEVKSVPRVSHL\ntranslation: MCVSPCQSRWVIMKCQTSKINV\ntranslation: MCVSPCQSRWVIMKCQTSKINV\ntranslation: QRTMQNSIANWIYGWDSLLLLKV\ntranslation: QRTMQNSIANWIYGWDSLLLLKV\ntranslation: REPCKTQLQIGYMDGILYYYLKF\ntranslation: REPCKTQLQIGYMDGILYYYLKF\ntranslation: LLFCLPESAPCMVSCLSWKYYAC\ntranslation: LLFCLPESAPCMVSCLSWKYYAC\ntranslation: IVLFARKCTLYGLLSKLEILCM\ntranslation: IVLFARKCTLYGLLSKLEILCM\ntranslation: HMHNISNLDRRPYRVHFLANKTI\ntranslation: HMHNISNLDRRPYRVHFLANKTI\ntranslation: ILIVSKLFQSDESDTIEMKSHSS\ntranslation: ILIVSKLFQSDESDTIEMKSHSS\ntranslation: VTCFDLEGRPLCPVT\ntranslation: VTCFDLEGRPLCPVT\ntranslation: LLALIWKVVHCALSH\ntranslation: LLALIWKVVHCALSH\ntranslation: DVTGHSGRPSKSKQV\ntranslation: DVTGHSGRPSKSKQV\ntranslation: CDRAQWTTFQIKASN\ntranslation: CDRAQWTTFQIKASN\ntranslation: LSLRFSLMFSISSSQKHLFLKNCD\ntranslation: LSLRFSLMFSISSSQKHLFLKNCD\ntranslation: HSSLRKDVSEKSLLKTLMKISMK\ntranslation: HSSLRKDVSEKSLLKTLMKISMK\ntranslation: PDFFRTKFYPVIFRRVSFVTLW\ntranslation: PDFFRTKFYPVIFRRVSFVTLW\ntranslation: RISSELNSIQSSSGECPLSLFG\ntranslation: RISSELNSIQSSSGECPLSLFG\ntranslation: PKSDKGHSPEDDWIEFSSEEIR\ntranslation: PKSDKGHSPEDDWIEFSSEEIR\ntranslation: WGCGGNHHPCIFQVLNGGTTPC\ntranslation: WGCGGNHHPCIFQVLNGGTTPC\ntranslation: GDVVGITTPASFKSLMVALLPA\ntranslation: GDVVGITTPASFKSLMVALLPA\ntranslation: AGSSATIKDLKDAGVVIPTTSP\ntranslation: AGSSATIKDLKDAGVVIPTTSP\ntranslation: RGKCLLRLKWNTLLFYELTAV\ntranslation: RGKCLLRLKWNTLLFYELTAV\ntranslation: LLLTHKREECSISVSINTSL\ntranslation: LLLTHKREECSISVSINTSL\ntranslation: LPYCQHRRIFSNQCPWGRRMT\ntranslation: LPYCQHRRIFSNQCPWGRRMT\ntranslation: GHPPAPGTLIRENPSVLAIRQ\ntranslation: GHPPAPGTLIRENPSVLAIRQ\ntranslation: ESNGSIMKIMSQKMWVLCVALIY\ntranslation: ESNGSIMKIMSQKMWVLCVALIY\ntranslation: IKRLYNEDNVSENVGSVCGTDL\ntranslation: IKRLYNEDNVSENVGSVCGTDL\ntranslation: SGLRDRR\ntranslation: SGLRDRR\ntranslation: PPIPQPR\ntranslation: PPIPQPR\ntranslation: LLSLNP\ntranslation: LLSLNP\ntranslation: SYPSTP\ntranslation: SYPSTP\ntranslation: LTSLHNDELLLGRRSSKDDL\ntranslation: LTSLHNDELLLGRRSSKDDL\ntranslation: QRSSLLERLPSRSSSLWRDVR\ntranslation: QRSSLLERLPSRSSSLWRDVR\ntranslation: RDRLCSNVSPAEAHHCGGMSE\ntranslation: RDRLCSNVSPAEAHHCGGMSE\ntranslation: EIVFARTSPQQKLIIVEGCQ\ntranslation: EIVFARTSPQQKLIIVEGCQ\ntranslation: AVDSLKSIYVAFTAGVFLDLNL\ntranslation: AVDSLKSIYVAFTAGVFLDLNL\ntranslation: SGHFSSHVLLYLFSTLCEQGLL\ntranslation: SGHFSSHVLLYLFSTLCEQGLL\ntranslation: LVIFQAMFYCICFLLYVSKVCC\ntranslation: LVIFQAMFYCICFLLYVSKVCC\ntranslation: HSLNAQFKMKHTATEYSMFLREF\ntranslation: HSLNAQFKMKHTATEYSMFLREF\ntranslation: PPPPPPPPPPHPPPPTHTTPPPPPP\ntranslation: PPPPPPPPPPHPPPPTHTTPPPPPP\ntranslation: PHPPPPPPPPTHPPPHTQHPPPPHP\ntranslation: PHPPPPPPPPTHPPPHTQHPPPPHP\ntranslation: PTPPPPPPPPPTPPHTHNTPPPPT\ntranslation: PTPPPPPPPPPTPPHTHNTPPPPT\ntranslation: GVGGGGVLCVWGGVGGGGGGGGGVG\ntranslation: GVGGGGVLCVWGGVGGGGGGGGGVG\ntranslation: GWGGGGCCVCGGGWVGGGGGGGGWG\ntranslation: GWGGGGCCVCGGGWVGGGGGGGGWG\ntranslation: GGGGGGVVCVGGGGWGGGGGGGGG\ntranslation: GGGGGGVVCVGGGGWGGGGGGGGG\ntranslation: EALAQRTYSDDHVKFFCFQVLEH\ntranslation: EALAQRTYSDDHVKFFCFQVLEH\ntranslation: DVPVLESRKTSHDHHCMSSGLEL\ntranslation: DVPVLESRKTSHDHHCMSSGLEL\ntranslation: MKSMKAAVMREMNLPMNLHSQLS\ntranslation: MKSMKAAVMREMNLPMNLHSQLS\n"
      },
      {
       "name": "stderr",
       "output_type": "stream",
       "text": "1000it [00:00, 1221.12it/s]"
      },
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "translation: EVNESSSDEGNEPANEFAQPAF\ntranslation: EVNESSSDEGNEPANEFAQPAF\ntranslation: MKAGCANSLAGSFPSSLLLSLTS\ntranslation: MKAGCANSLAGSFPSSLLLSLTS\ntranslation: ESWLCKFIGRFISLITAAFIDF\ntranslation: ESWLCKFIGRFISLITAAFIDF\ntranslation: VMNTWFYITYEKDPVLYMYQLL\ntranslation: VMNTWFYITYEKDPVLYMYQLL\ntranslation: HCGTFEYSFNFATYSADSK\ntranslation: HCGTFEYSFNFATYSADSK\ntranslation: IVVLLNTPLILLPILLIPN\ntranslation: IVVLLNTPLILLPILLIPN\ntranslation: FGISRIGSKIKGVFKSTTM\ntranslation: FGISRIGSKIKGVFKSTTM\ntranslation: LSHLTPTSFVLFCFCPLCTTFLFSN\ntranslation: LSHLTPTSFVLFCFCPLCTTFLFSN\ntranslation: FPTSPLPLLFCFVFALYVLHSYFLT\ntranslation: FPTSPLPLLFCFVFALYVLHSYFLT\ntranslation: LENKNVVHKGQKQNKTKEVGVRWE\ntranslation: LENKNVVHKGQKQNKTKEVGVRWE\ntranslation: FVYLSKLKELWLRNNPIESIP\ntranslation: FVYLSKLKELWLRNNPIESIP\ntranslation: EGMLSMGLFRNQSSFSLDKYTK\ntranslation: EGMLSMGLFRNQSSFSLDKYTK\ntranslation: RDAFNGVVSQPELLQFRQVYK\ntranslation: RDAFNGVVSQPELLQFRQVYK\ntranslation: SIQCIFRYYQHYKIESTFIL\ntranslation: SIQCIFRYYQHYKIESTFIL\ntranslation: EYESRFYFVMLIIPKDALNA\ntranslation: EYESRFYFVMLIIPKDALNA\ntranslation: FSKFCGSSKDFYVHTLSSLVL\ntranslation: FSKFCGSSKDFYVHTLSSLVL\ntranslation: SVSSVVPVRIFMYIRSRLSFW\ntranslation: SVSSVVPVRIFMYIRSRLSFW\ntranslation: QNERRERMYIKILTGTTELTE\ntranslation: QNERRERMYIKILTGTTELTE\ntranslation: FYLLNCLFRSLAGCFLCLPLHVA\ntranslation: FYLLNCLFRSLAGCFLCLPLHVA\ntranslation: LPAKLSFQKSCRMFSLFASPCCC\ntranslation: LPAKLSFQKSCRMFSLFASPCCC\ntranslation: LYRAVHMTKSRITVLLNQLLRYS\ntranslation: LYRAVHMTKSRITVLLNQLLRYS\ntranslation: AISEELVQKYSNSALGHVNCTIK\ntranslation: AISEELVQKYSNSALGHVNCTIK\ntranslation: THPPRVHHIPNLSLTKRKPS\ntranslation: THPPRVHHIPNLSLTKRKPS\ntranslation: HIHLGSIIYQISPSLNVSLL\ntranslation: HIHLGSIIYQISPSLNVSLL\ntranslation: RRLTFSEGEIWYMMDPRWMC\ntranslation: RRLTFSEGEIWYMMDPRWMC\ntranslation: LNLNNSCKNLIKRHQLGTQFL\ntranslation: LNLNNSCKNLIKRHQLGTQFL\ntranslation: EELCPKLMSLNEIFTGIVQIQ\ntranslation: EELCPKLMSLNEIFTGIVQIQ\ntranslation: STFTHIMSFEPHNVLGMESVVQ\ntranslation: STFTHIMSFEPHNVLGMESVVQ\ntranslation: MHLLRIRTLSYVTQLSLLEK\ntranslation: MHLLRIRTLSYVTQLSLLEK\ntranslation: HSHRNILSSLFIGKNNFRSTLKK\ntranslation: HSHRNILSSLFIGKNNFRSTLKK\ntranslation: NGACYSLTFLLSACVELLVIQA\ntranslation: NGACYSLTFLLSACVELLVIQA\ntranslation: IDKVKENPIENISLDYHQLS\ntranslation: IDKVKENPIENISLDYHQLS\ntranslation: QIADGNQGRYSQWDSPSLCQ\ntranslation: QIADGNQGRYSQWDSPSLCQ\ntranslation: GLIPCLLIYSLDDERVTKNKT\ntranslation: GLIPCLLIYSLDDERVTKNKT\ntranslation: GVLFFVTLSSSRLYINRHGIKP\ntranslation: GVLFFVTLSSSRLYINRHGIKP\ntranslation: KNGTKVNSTVNAHAYFIHIFNNK\ntranslation: KNGTKVNSTVNAHAYFIHIFNNK\ntranslation: KNIFSKIEEVISVLAIGATYLL\ntranslation: KNIFSKIEEVISVLAIGATYLL\ntranslation: NHMCFLQLFLLFWPNNPKNPYH\ntranslation: NHMCFLQLFLLFWPNNPKNPYH\ntranslation: TICVFCSYFSCSGQTTLKILT\ntranslation: TICVFCSYFSCSGQTTLKILT\ntranslation: LYFYLRLIYSTSITLLPISNN\ntranslation: LYFYLRLIYSTSITLLPISNN\ntranslation: CRYSPKTSRHFQSSPISQIKNT\ntranslation: CRYSPKTSRHFQSSPISQIKNT\ntranslation: VDILQKQADIFRVVQFPKSKTQ\ntranslation: VDILQKQADIFRVVQFPKSKTQ\ntranslation: CVFDLGNWTTLKMSACFWRIST\ntranslation: CVFDLGNWTTLKMSACFWRIST\ntranslation: ILFEMRSHFVTQTGVQWYNLGSL\ntranslation: ILFEMRSHFVTQTGVQWYNLGSL\ntranslation: ARFPKFLSWEEEIVALDHPW\ntranslation: ARFPKFLSWEEEIVALDHPW\ntranslation: TLSKIFKLGGRDSRSGSPMA\ntranslation: TLSKIFKLGGRDSRSGSPMA\ntranslation: AMGDPERLSLPPSLKILESV\ntranslation: AMGDPERLSLPPSLKILESV\ntranslation: KIVLYAKKVGYTPDWIFLLRN\ntranslation: KIVLYAKKVGYTPDWIFLLRN\ntranslation: FLSRNIQSGVYPTFLAYKTIF\ntranslation: FLSRNIQSGVYPTFLAYKTIF\ntranslation: LKRFIRQAEMNHYALFKCYMFL\ntranslation: LKRFIRQAEMNHYALFKCYMFL\ntranslation: QSEAKKGSSHNRKNKSQHTLEII\ntranslation: QSEAKKGSSHNRKNKSQHTLEII\ntranslation: DYESGFNISEHTIAKTQTGIR\ntranslation: DYESGFNISEHTIAKTQTGIR\ntranslation: SNPSLGLSYCVFRYVKATFVV\ntranslation: SNPSLGLSYCVFRYVKATFVV\ntranslation: FRFHLVSFLPFLSSIFDSLT\ntranslation: FRFHLVSFLPFLSSIFDSLT\ntranslation: DKVSSKNSLESYAFNMKATVEDE\ntranslation: DKVSSKNSLESYAFNMKATVEDE\ntranslation: DFGALSFACTESVVILVASKCQ\ntranslation: DFGALSFACTESVVILVASKCQ\ntranslation: GWHLLATKITTDSVQAKDKAPK\ntranslation: GWHLLATKITTDSVQAKDKAPK\ntranslation: AGIYWPPKSLQILYKQKTRHQN\ntranslation: AGIYWPPKSLQILYKQKTRHQN\ntranslation: LAFTGHQNHYRFCTSKRQGTKI\ntranslation: LAFTGHQNHYRFCTSKRQGTKI\ntranslation: LIWLKNSRGMMYWIVFALYTVI\ntranslation: LIWLKNSRGMMYWIVFALYTVI\ntranslation: QSQYREQKQSSTSSLYYFLTKS\ntranslation: QSQYREQKQSSTSSLYYFLTKS\ntranslation: QKFKDCTCQ\ntranslation: QKFKDCTCQ\ntranslation: KNSKTAHA\ntranslation: KNSKTAHA\ntranslation: KIQRLHMP\ntranslation: KIQRLHMP\ntranslation: LACAVFEFL\ntranslation: LACAVFEFL\ntranslation: WHVQSLNF\ntranslation: WHVQSLNF\ntranslation: LIPALVLLFRVCFFVFVLLLLL\ntranslation: LIPALVLLFRVCFFVFVLLLLL\ntranslation: SSLLLSCSSESVFLSLSFFFCS\ntranslation: SSLLLSCSSESVFLSLSFFFCS\ntranslation: HPCSCLALPSLFFCLCPSSFAL\ntranslation: HPCSCLALPSLFFCLCPSSFAL\ntranslation: QSKRRRTKTKKQTRKSKTRAGM\ntranslation: QSKRRRTKTKKQTRKSKTRAGM\ntranslation: EQKKKDKDKKTDSEEQDKSRDE\ntranslation: EQKKKDKDKKTDSEEQDKSRDE\ntranslation: SVLLDVDGAVKSILSSARLLTM\ntranslation: SVLLDVDGAVKSILSSARLLTM\ntranslation: IVSSLAEERMDFTAPSTSSKTL\ntranslation: IVSSLAEERMDFTAPSTSSKTL\ntranslation: IIGSETHNLLILKSLKFEIPGA\ntranslation: IIGSETHNLLILKSLKFEIPGA\ntranslation: FAPGISNLRDFKISKLCVSLPII\ntranslation: FAPGISNLRDFKISKLCVSLPII\ntranslation: LFCISVALCCVHFLLLPPKCWD\ntranslation: LFCISVALCCVHFLLLPPKCWD\ntranslation: YFVSQLLFAVSISSSCRLNAGI\ntranslation: YFVSQLLFAVSISSSCRLNAGI\ntranslation: ESQHLGGRRRKWTQQRATEIQNK\ntranslation: ESQHLGGRRRKWTQQRATEIQNK\ntranslation: KHLLCFFVSGCRSHFLMIFKKTL\ntranslation: KHLLCFFVSGCRSHFLMIFKKTL\ntranslation: QRFLENHEKMTSTTRYKKAEQMF\ntranslation: QRFLENHEKMTSTTRYKKAEQMF\ntranslation: RRVGDQHQRGRQ\ntranslation: RRVGDQHQRGRQ\ntranslation: GEWVTNISEGGS\ntranslation: GEWVTNISEGGS\ntranslation: AASLADVGHPLS\ntranslation: AASLADVGHPLS\ntranslation: LPPSLMLVTHSP\ntranslation: LPPSLMLVTHSP\ntranslation: GVSPCYIMLGYNFSSFPCGTISI\ntranslation: GVSPCYIMLGYNFSSFPCGTISI\ntranslation: GLVLAILCLVIIFHLSLAVLYLL\ntranslation: GLVLAILCLVIIFHLSLAVLYLL\ntranslation: VNLNSSLFLKQTSLCLFHIFSNW\ntranslation: VNLNSSLFLKQTSLCLFHIFSNW\ntranslation: PVTEYMKKAEARLLEEQRRVQVY\ntranslation: PVTEYMKKAEARLLEEQRRVQVY\ntranslation: GSSLEKGQPSSKMVSQEVCVQ\ntranslation: GSSLEKGQPSSKMVSQEVCVQ\ntranslation: FTGVLVYSQGFSTITIIFFQYIF\ntranslation: FTGVLVYSQGFSTITIIFFQYIF\ntranslation: HWCFSIFTRFFNHYHYLLPVHF\ntranslation: HWCFSIFTRFFNHYHYLLPVHF\ntranslation: KMYWKKIMVMVEKPCEYTKTPV\ntranslation: KMYWKKIMVMVEKPCEYTKTPV\ntranslation: RFSFILVLRVCYNFLFLWALVV\ntranslation: RFSFILVLRVCYNFLFLWALVV\ntranslation: QHCISCVWHCSIQMFVG\ntranslation: QHCISCVWHCSIQMFVG\ntranslation: NTVSLASGIVQSRCLLG\ntranslation: NTVSLASGIVQSRCLLG\ntranslation: TLYLLRLALFNPDVCWD\ntranslation: TLYLLRLALFNPDVCWD\ntranslation: VPTNIWIEQCQTQEIQC\ntranslation: VPTNIWIEQCQTQEIQC\ntranslation: SQQTSGLNNARRKRYSV\ntranslation: SQQTSGLNNARRKRYSV\ntranslation: IQSTSPKIKILSYQHSLILDR\ntranslation: IQSTSPKIKILSYQHSLILDR\ntranslation: LLYPTFTATFIRFVSHLPACLF\ntranslation: LLYPTFTATFIRFVSHLPACLF\ntranslation: FFTPLSLPLSSGLCLISLRVFS\ntranslation: FFTPLSLPLSSGLCLISLRVFS\ntranslation: SLPHFHCHFHQVCVSSPCVSF\ntranslation: SLPHFHCHFHQVCVSSPCVSF\ntranslation: EKTRREMRHKPDESGSESGVKK\ntranslation: EKTRREMRHKPDESGSESGVKK\ntranslation: PPFTIDFGSRFL\ntranslation: PPFTIDFGSRFL\ntranslation: PIYYRLWQQVF\ntranslation: PIYYRLWQQVF\ntranslation: QKPAAKVYSKWG\ntranslation: QKPAAKVYSKWG\ntranslation: KNLLPKSIVNG\ntranslation: KNLLPKSIVNG\ntranslation: SCTKLFICLFFVCNLSVKGFSLL\ntranslation: SCTKLFICLFFVCNLSVKGFSLL\ntranslation: SREKPFTDKLQTKKRQINNFVQE\ntranslation: SREKPFTDKLQTKKRQINNFVQE\ntranslation: VKLYFSYPIWFDIYTQTYISTSI\ntranslation: VKLYFSYPIWFDIYTQTYISTSI\ntranslation: VFALFYYYFFQCVGPTLCNLP\ntranslation: VFALFYYYFFQCVGPTLCNLP\ntranslation: FLLCFIIIFSNVWGLHFAIYL\ntranslation: FLLCFIIIFSNVWGLHFAIYL\ntranslation: FCFVLLLFFPMCGAYTLQST\ntranslation: FCFVLLLFFPMCGAYTLQST\ntranslation: MHPPSLAFTPEFHTSLSPLA\ntranslation: MHPPSLAFTPEFHTSLSPLA\ntranslation: LVFRCKPLITLSYLIVFSGVLL\ntranslation: LVFRCKPLITLSYLIVFSGVLL\ntranslation: SESFSLLSKVRLASLTFLSSSW\ntranslation: SESFSLLSKVRLASLTFLSSSW\ntranslation: PGGAQEGQGRQPHLGKQTKGFR\ntranslation: PGGAQEGQGRQPHLGKQTKGFR\ntranslation: QEELRKVKDANLTLESKLKDSE\ntranslation: QEELRKVKDANLTLESKLKDSE\ntranslation: LWCILTIIALIVVFCQDRHQT\ntranslation: LWCILTIIALIVVFCQDRHQT\ntranslation: LVYPHNYRSYCSFLPRQAPD\ntranslation: LVYPHNYRSYCSFLPRQAPD\ntranslation: GLVPVLAENYNKSDNCEDTPE\ntranslation: GLVPVLAENYNKSDNCEDTPE\ntranslation: VWCLSWQKTTIRAIIVRIHQ\ntranslation: VWCLSWQKTTIRAIIVRIHQ\ntranslation: LLISLMAAVQLIWKLLFYCEH\ntranslation: LLISLMAAVQLIWKLLFYCEH\ntranslation: DAHNKKAVSRSAEQQPSEKST\ntranslation: DAHNKKAVSRSAEQQPSEKST\ntranslation: MLTIKKQFPDQLNSSHQRNQQ\ntranslation: MLTIKKQFPDQLNSSHQRNQQ\ntranslation: LRCFLLNALWISYFLPYFLCSS\ntranslation: LRCFLLNALWISYFLPYFLCSS\ntranslation: QVFPIKCSVDFLLSSLFPLFQ\ntranslation: QVFPIKCSVDFLLSSLFPLFQ\ntranslation: TCKFYEILVFADASQKTTHLSFP\ntranslation: TCKFYEILVFADASQKTTHLSFP\ntranslation: LANFMKFLSLQMPVKKLHTYPSL\ntranslation: LANFMKFLSLQMPVKKLHTYPSL\ntranslation: RSLRLSTDSMYFFSFSSSSLYQD\ntranslation: RSLRLSTDSMYFFSFSSSSLYQD\ntranslation: QLEVVHRQHVFLLFLLIILVPG\ntranslation: QLEVVHRQHVFLLFLLIILVPG\ntranslation: SWYKDDEEKEKKYMLSVDNLKL\ntranslation: SWYKDDEEKEKKYMLSVDNLKL\ntranslation: PGTRMMRRKRRNTCCLWTTSSC\ntranslation: PGTRMMRRKRRNTCCLWTTSSC\ntranslation: YLLLNPANQRVHNRFREENTSK\ntranslation: YLLLNPANQRVHNRFREENTSK\ntranslation: SLKISSVNKEVLSRFFEYTYNIL\ntranslation: SLKISSVNKEVLSRFFEYTYNIL\ntranslation: LFLLYCIYSSDHLVTQDIRPVEV\ntranslation: LFLLYCIYSSDHLVTQDIRPVEV\ntranslation: LDLIFKAPYPDTTHLLSSGM\ntranslation: LDLIFKAPYPDTTHLLSSGM\ntranslation: HSRRQKVCSIRIRGFKYQIQ\ntranslation: HSRRQKVCSIRIRGFKYQIQ\ntranslation: RATLCVLLQCVLSVHPSLPSV\ntranslation: RATLCVLLQCVLSVHPSLPSV\ntranslation: VLHFVSCCNVFCLSIHLCPLY\ntranslation: VLHFVSCCNVFCLSIHLCPLY\ntranslation: CYTLCPAAMCSVCPSISALC\ntranslation: CYTLCPAAMCSVCPSISALC\ntranslation: YRGQRWMDRQNTLQQDTKCST\ntranslation: YRGQRWMDRQNTLQQDTKCST\ntranslation: TEGRDGWTDRTHCSRTQSVA\ntranslation: TEGRDGWTDRTHCSRTQSVA\ntranslation: PVDQCLAHGKCMSVLAS\ntranslation: PVDQCLAHGKCMSVLAS\ntranslation: FEMFFVCLFVFVCL\ntranslation: FEMFFVCLFVFVCL\ntranslation: LKCFLFVCLFLYVS\ntranslation: LKCFLFVCLFLYVS\ntranslation: ETYKNKQTNKKHFK\ntranslation: ETYKNKQTNKKHFK\ntranslation: RHTKTNKQTKNISK\ntranslation: RHTKTNKQTKNISK\ntranslation: LSVKRDFIHRPLTLACYSGMPI\ntranslation: LSVKRDFIHRPLTLACYSGMPI\ntranslation: VSEKRLYSQTSHFSMLFWDANL\ntranslation: VSEKRLYSQTSHFSMLFWDANL\ntranslation: RPEEEFHVPLDSYRCQLFIQPA\ntranslation: RPEEEFHVPLDSYRCQLFIQPA\ntranslation: YFQAIYSVFIICPWWPNNHACPP\ntranslation: YFQAIYSVFIICPWWPNNHACPP\ntranslation: RWTGMIIGPPRTNYENRIYSLKV\ntranslation: RWTGMIIGPPRTNYENRIYSLKV\ntranslation: DPWLSKFLSRYRKTHHIEPRTI\ntranslation: DPWLSKFLSRYRKTHHIEPRTI\ntranslation: IPGFPNSYHGIEKPITLNHVPL\ntranslation: IPGFPNSYHGIEKPITLNHVPL\ntranslation: FCKKPIKIKDSEVIRTLCFVLI\ntranslation: FCKKPIKIKDSEVIRTLCFVLI\ntranslation: NQNKTQGTDYFTVLYLDWFLTK\ntranslation: NQNKTQGTDYFTVLYLDWFLTK\ntranslation: IKTKHKVRITSLSFILIGFLQN\ntranslation: IKTKHKVRITSLSFILIGFLQN\ntranslation: TRGFISLSICSSDSHSVPTVCP\ntranslation: TRGFISLSICSSDSHSVPTVCP\ntranslation: HVDSFLYLSVHQIATQFLQFAH\ntranslation: HVDSFLYLSVHQIATQFLQFAH\ntranslation: HGQTVGTEWLSDEQIDKEMNPRV\ntranslation: HGQTVGTEWLSDEQIDKEMNPRV\ntranslation: KNRSLLHFKSTRTHNSYNRHQPTT\ntranslation: KNRSLLHFKSTRTHNSYNRHQPTT\ntranslation: KIVAFSTSSQLGLIIVTIGINQPH\ntranslation: KIVAFSTSSQLGLIIVTIGINQPH\ntranslation: LSVYCSSQLYSINLHFKGCIG\ntranslation: LSVYCSSQLYSINLHFKGCIG\ntranslation: HNDHIWASKVEGKPVFKNISRTS\ntranslation: HNDHIWASKVEGKPVFKNISRTS\ntranslation: EVREIFLKTGLPSTLLAHIWSLC\ntranslation: EVREIFLKTGLPSTLLAHIWSLC\ntranslation: GTLIRIFDTSSGHLIQELRRGSQ\ntranslation: GTLIRIFDTSSGHLIQELRRGSQ\ntranslation: ELRDHKTSVMNNFSTDVVVTA\ntranslation: ELRDHKTSVMNNFSTDVVVTA\ntranslation: TTSLCLTSLPSDLSRLKSAYVTL\ntranslation: TTSLCLTSLPSDLSRLKSAYVTL\ntranslation: DMIMAFSVNKANCNEIVLASTHD\ntranslation: DMIMAFSVNKANCNEIVLASTHD\ntranslation: SCVEAKTISLQFALLTENAMIIS\ntranslation: SCVEAKTISLQFALLTENAMIIS\ntranslation: QIKTFKSPMMTEWIIPHEMST\ntranslation: QIKTFKSPMMTEWIIPHEMST\ntranslation: MVLISWGIIHSVIIGDLKVLIC\ntranslation: MVLISWGIIHSVIIGDLKVLIC\ntranslation: KNYTKYEFGYNFLLKSPFLFLSF\ntranslation: KNYTKYEFGYNFLLKSPFLFLSF\ntranslation: MIGKETGISEENCNQIHTLYNF\ntranslation: MIGKETGISEENCNQIHTLYNF\ntranslation: RNCKISSLGQKNNSILTHKCLP\ntranslation: RNCKISSLGQKNNSILTHKCLP\ntranslation: GTARYPVSVKRTTQFLHINVCQ\ntranslation: GTARYPVSVKRTTQFLHINVCQ\ntranslation: FSNVCFSSAVISGFSVISSTSPV\ntranslation: FSNVCFSSAVISGFSVISSTSPV\ntranslation: LVMFASPLLSFLVSLSFPLPPQW\ntranslation: LVMFASPLLSFLVSLSFPLPPQW\ntranslation: HWGGRGNDRETRNDSRGEANITK\ntranslation: HWGGRGNDRETRNDSRGEANITK\ntranslation: TGEVEEMTEKPEMTAEEKQTLLK\ntranslation: TGEVEEMTEKPEMTAEEKQTLLK\ntranslation: AILCLVIIFHLSLAVLYLLRQ\ntranslation: AILCLVIIFHLSLAVLYLLRQ\ntranslation: YIMLGYNFSSFPCGTISIAP\ntranslation: YIMLGYNFSSFPCGTISIAP\ntranslation: FWTIFITTDQVAICIIILTTIII\ntranslation: FWTIFITTDQVAICIIILTTIII\ntranslation: SGPYSSPQIKLQFASSSSPPSS\ntranslation: SGPYSSPQIKLQFASSSSPPSS\ntranslation: LDHIHHHRSSCNLHHHPHHHHH\ntranslation: LDHIHHHRSSCNLHHHPHHHHH\ntranslation: DDDGGEDDDANCNLICGDEYGPE\ntranslation: DDDGGEDDDANCNLICGDEYGPE\ntranslation: FKDLEDKKVNQQPNTSDKKSSP\ntranslation: FKDLEDKKVNQQPNTSDKKSSP\ntranslation: QVVSIRRG\ntranslation: QVVSIRRG\ntranslation: PPPNTNYL\ntranslation: PPPNTNYL\ntranslation: PLLILTT\ntranslation: PLLILTT\ntranslation: EYLFIHISSALVTMATVQKILSN\ntranslation: EYLFIHISSALVTMATVQKILSN\ntranslation: ISFHSHLLCLSHNGNSTENSIK\ntranslation: ISFHSHLLCLSHNGNSTENSIK\ntranslation: WNMLNPPAEVTTKSQIPLSKIK\ntranslation: WNMLNPPAEVTTKSQIPLSKIK\ntranslation: LYFRERNLTFRCHLSGRIQHIP\ntranslation: LYFRERNLTFRCHLSGRIQHIP\ntranslation: SGIIVTLAGVYLSDGAEIKEKGF\ntranslation: SGIIVTLAGVYLSDGAEIKEKGF\ntranslation: RTSRHYWLNFPHYLLHPPTNISL\ntranslation: RTSRHYWLNFPHYLLHPPTNISL\ntranslation: GLHVIIGSTFLTICFIRQLIFHF\ntranslation: GLHVIIGSTFLTICFIRQLIFHF\ntranslation: LVLCMVIICFEVSLISHCWVVI\ntranslation: LVLCMVIICFEVSLISHCWVVI\ntranslation: PRNYSHHQLNRSSKHSQGLRDQ\ntranslation: PRNYSHHQLNRSSKHSQGLRDQ\ntranslation: SAVWLKVSSSSAVQIVPSLSD\ntranslation: SAVWLKVSSSSAVQIVPSLSD\ntranslation: HQIAKELFEQLKMKKPSAKQQ\ntranslation: HQIAKELFEQLKMKKPSAKQQ\ntranslation: HAQTEIIEGTNNPIFLSSIAFFQ\ntranslation: HAQTEIIEGTNNPIFLSSIAFFQ\ntranslation: YTEITLGYFCIILSYCLHSI\ntranslation: YTEITLGYFCIILSYCLHSI\ntranslation: HPKSTQYTFFSAPHHTYSKID\ntranslation: HPKSTQYTFFSAPHHTYSKID\ntranslation: TPSQHNIHSSQHHITLILKL\ntranslation: TPSQHNIHSSQHHITLILKL\ntranslation: MAKGKSNLILNKGEVQQPACR\ntranslation: MAKGKSNLILNKGEVQQPACR\ntranslation: GLQAGCCTSPLFNIRFDLPFAI\ntranslation: GLQAGCCTSPLFNIRFDLPFAI\ntranslation: VCKQVAALLPYSILGLICPSPL\ntranslation: VCKQVAALLPYSILGLICPSPL\ntranslation: KNYVSLVKCPFFFLPLIKSGSSA\ntranslation: KNYVSLVKCPFFFLPLIKSGSSA\ntranslation: HKCSNKQVEGHLELSCFADG\ntranslation: HKCSNKQVEGHLELSCFADG\ntranslation: TNAAINRWKVIWNSAVLQMD\ntranslation: TNAAINRWKVIWNSAVLQMD\ntranslation: SICKTAEFQMTFHLFIAAFV\ntranslation: SICKTAEFQMTFHLFIAAFV\ntranslation: TFLLCSYFLNGFKGTEINAIYM\ntranslation: TFLLCSYFLNGFKGTEINAIYM\ntranslation: FCSQLLIFFKNSFHGNKSWFSQ\ntranslation: FCSQLLIFFKNSFHGNKSWFSQ\ntranslation: PKSILTIPLNHLCSPVIVV\ntranslation: PKSILTIPLNHLCSPVIVV\ntranslation: TTITGEHRWFRGMVKMLL\ntranslation: TTITGEHRWFRGMVKMLL\ntranslation: GTRSILKTHFEKEDLEGHRTL\ntranslation: GTRSILKTHFEKEDLEGHRTL\ntranslation: WYHREESLVLGVVRCQSIFYWAC\ntranslation: WYHREESLVLGVVRCQSIFYWAC\ntranslation: GTTGRNHWFWVLLDVKAFSIGH\ntranslation: GTTGRNHWFWVLLDVKAFSIGH\ntranslation: KNQIKVDLVDENFTELRGEI\ntranslation: KNQIKVDLVDENFTELRGEI\ntranslation: KKALFKGSGSQTLKVTEKSPQ\ntranslation: KKALFKGSGSQTLKVTEKSPQ\ntranslation: RKVCRFKLEWLGNCSGLNDETY\ntranslation: RKVCRFKLEWLGNCSGLNDETY\ntranslation: NLQVEMIKQFHMQLNEMHSLLE\ntranslation: NLQVEMIKQFHMQLNEMHSLLE\ntranslation: FQQRMHFIQLHMKLFNHLHLQI\ntranslation: FQQRMHFIQLHMKLFNHLHLQI\ntranslation: KKNGSGWDADSNRSGSGWNDTT\ntranslation: KKNGSGWDADSNRSGSGWNDTT\ntranslation: KKMDLDGMLTVIGQGLVGMTP\ntranslation: KKMDLDGMLTVIGQGLVGMTP\ntranslation: VVSFQPDPDLLLSASHPDPFF\ntranslation: VVSFQPDPDLLLSASHPDPFF\ntranslation: WCHSNQTLTYYCQHPIQIHFF\ntranslation: WCHSNQTLTYYCQHPIQIHFF\ntranslation: MCCPSLWESVFVLDFLLEMWK\ntranslation: MCCPSLWESVFVLDFLLEMWK\ntranslation: CAAPACGNPFLCLIFCWRCGR\ntranslation: CAAPACGNPFLCLIFCWRCGR\ntranslation: PSTSPAKNQAQKRIPTGWGST\ntranslation: PSTSPAKNQAQKRIPTGWGST\ntranslation: LPHLQQKIKHKNGFPQAGAAH\ntranslation: LPHLQQKIKHKNGFPQAGAAH\ntranslation: FHISSKKSSTKTDSHRLGQHI\ntranslation: FHISSKKSSTKTDSHRLGQHI\ntranslation: LDRHVVICCCG\ntranslation: LDRHVVICCCG\ntranslation: WTDTLSFAAV\ntranslation: WTDTLSFAAV\ntranslation: GQTRCHLLLW\ntranslation: GQTRCHLLLW\ntranslation: PTAANDNVSVQ\ntranslation: PTAANDNVSVQ\ntranslation: PQQQMTTCLS\ntranslation: PQQQMTTCLS\ntranslation: DSQKLNFREHAKARVDHGAEI\ntranslation: DSQKLNFREHAKARVDHGAEI\ntranslation: MISAPWSTRALACSLKFNFWL\ntranslation: MISAPWSTRALACSLKFNFWL\ntranslation: DLSPMVHTGFSMLSEVQLLAV\ntranslation: DLSPMVHTGFSMLSEVQLLAV\ntranslation: AAMLSGHFRPPIISYFRTPE\ntranslation: AAMLSGHFRPPIISYFRTPE\ntranslation: PQGYGNMKLWEDENAQKAWL\ntranslation: PQGYGNMKLWEDENAQKAWL\ntranslation: SGVRKYEIMGGRKCPESMAA\ntranslation: SGVRKYEIMGGRKCPESMAA\ntranslation: SSQLQICESNCQPDFASQIP\ntranslation: SSQLQICESNCQPDFASQIP\ntranslation: PLSSRSVNLTANLTLPLRFP\ntranslation: PLSSRSVNLTANLTLPLRFP\ntranslation: GNLRGKVRLAVRFTDLELRG\ntranslation: GNLRGKVRLAVRFTDLELRG\ntranslation: ECPDGFAPLEETMECVGCEVGHW\ntranslation: ECPDGFAPLEETMECVGCEVGHW\ntranslation: NDQLHIPHIPWFLLMVQNHLDI\ntranslation: NDQLHIPHIPWFLLMVQNHLDI\ntranslation: LRVCYNFLFLWFTVHYFLLAIP\ntranslation: LRVCYNFLFLWFTVHYFLLAIP\ntranslation: SGFVIIFYFYGLPYIISCSPSP\ntranslation: SGFVIIFYFYGLPYIISCSPSP\ntranslation: RVSTSIPTVNIWCAHT\ntranslation: RVSTSIPTVNIWCAHT\ntranslation: CVYVYSYCKYMVCSH\ntranslation: CVYVYSYCKYMVCSH\ntranslation: RVSTPYIYSRNRRRHT\ntranslation: RVSTPYIYSRNRRRHT\ntranslation: KFHGIKLNRLIECHSNKQLFI\ntranslation: KFHGIKLNRLIECHSNKQLFI\ntranslation: DEQLLITVAFNQPVKLYSMKF\ntranslation: DEQLLITVAFNQPVKLYSMKF\ntranslation: RVSKLVSVLLSSASLSLSLPL\ntranslation: RVSKLVSVLLSSASLSLSLPL\ntranslation: GCPSWFQSYCHQHHCPYHCHC\ntranslation: GCPSWFQSYCHQHHCPYHCHC\ntranslation: GVQAGFSLTVISITVLITAT\ntranslation: GVQAGFSLTVISITVLITAT\ntranslation: TVAVIRTVMLMTVRLKPAWTP\ntranslation: TVAVIRTVMLMTVRLKPAWTP\ntranslation: SGSDKDSDADDSKTETSLDT\ntranslation: SGSDKDSDADDSKTETSLDT\ntranslation: LDSSNWTSEFSSAAQSVSLSSHN\ntranslation: LDSSNWTSEFSSAAQSVSLSSHN\ntranslation: LIHQIGHLNFLLLLNLFLFLLI\ntranslation: LIHQIGHLNFLLLLNLFLFLLI\ntranslation: YEKKEKQIEQQKKIQMSNLMNQ\ntranslation: YEKKEKQIEQQKKIQMSNLMNQ\ntranslation: PDKTHSLPLPCTIPMRHNYNKLH\ntranslation: PDKTHSLPLPCTIPMRHNYNKLH\ntranslation: RRGSLSLPSVNLHFRASGLFLS\ntranslation: RRGSLSLPSVNLHFRASGLFLS\ntranslation: QLKKSPEALKCRLTEGSERLPR\ntranslation: QLKKSPEALKCRLTEGSERLPR\ntranslation: PLHFMDDSPQRGEGTCPKLVNS\ntranslation: PLHFMDDSPQRGEGTCPKLVNS\ntranslation: SYECFKIINIWFPLNIAHHK\ntranslation: SYECFKIINIWFPLNIAHHK\ntranslation: GKLAKEIAEKLGFKHFPEAGI\ntranslation: GKLAKEIAEKLGFKHFPEAGI\ntranslation: SFLQCMFVAPLSKNEFTVDVWT\ntranslation: SFLQCMFVAPLSKNEFTVDVWT\ntranslation: LQSHFSNISVCIYVREYTCVCMYY\ntranslation: LQSHFSNISVCIYVREYTCVCMYY\ntranslation: YNLTLATYLCVFMYVNILVYVCIM\ntranslation: YNLTLATYLCVFMYVNILVYVCIM\ntranslation: VTLRSTPRRLTLRIPSCQTWTVR\ntranslation: VTLRSTPRRLTLRIPSCQTWTVR\ntranslation: DPEINTKKINPENSKLSDLDSE\ntranslation: DPEINTKKINPENSKLSDLDSE\ntranslation: VSLSRSDSLEFSGLIFLVLISGS\ntranslation: VSLSRSDSLEFSGLIFLVLISGS\ntranslation: ASRIQRNGFKLNQCDLVNG\ntranslation: ASRIQRNGFKLNQCDLVNG\ntranslation: ERFVLQFFCCKELCCSFPPKWFQ\ntranslation: ERFVLQFFCCKELCCSFPPKWFQ\ntranslation: KDLYYSSFVVKSCVVLFPQSGFS\ntranslation: KDLYYSSFVVKSCVVLFPQSGFS\ntranslation: AETTLGEKNNTTLYNKRTVVQIF\ntranslation: AETTLGEKNNTTLYNKRTVVQIF\ntranslation: KDAAKANLNYIGLDGNIGC\ntranslation: KDAAKANLNYIGLDGNIGC\ntranslation: AAYISIEANVVEICLSSIF\ntranslation: AAYISIEANVVEICLSSIF\ntranslation: FSLMRKTEHLPLLTVMIFTPPQ\ntranslation: FSLMRKTEHLPLLTVMIFTPPQ\ntranslation: WGLNAHSI\ntranslation: WGLNAHSI\ntranslation: GGSMLTV\ntranslation: GGSMLTV\ntranslation: GAQCSQY\ntranslation: GAQCSQY\ntranslation: NTVSIEPP\ntranslation: NTVSIEPP\ntranslation: LKQMVQISNFQIINDYVC\ntranslation: LKQMVQISNFQIINDYVC\ntranslation: ADIVIYDLKVGNLYHLFE\ntranslation: ADIVIYDLKVGNLYHLFE\ntranslation: TSVVLFTFYSSAFSKMLYIVCSFF\ntranslation: TSVVLFTFYSSAFSKMLYIVCSFF\ntranslation: NNNMLTVPKENYFKKRHNPDVPS\ntranslation: NNNMLTVPKENYFKKRHNPDVPS\ntranslation: IFHCEEGAEKDSSVLQMTIPV\ntranslation: IFHCEEGAEKDSSVLQMTIPV\ntranslation: DPHLQISHQQPTNHHPTMTN\ntranslation: DPHLQISHQQPTNHHPTMTN\ntranslation: ISHCWVVISRLLMRYLEVGI\ntranslation: ISHCWVVISRLLMRYLEVGI\ntranslation: ATSEQNPTSE\ntranslation: ATSEQNPTSE\ntranslation: RPRSRTQPPS\ntranslation: RPRSRTQPPS\ntranslation: DLGAEPNLR\ntranslation: DLGAEPNLR\ntranslation: ARRLGSAPRS\ntranslation: ARRLGSAPRS\ntranslation: LGGWVLLRGR\ntranslation: LGGWVLLRGR\ntranslation: SEVGFCSEV\ntranslation: SEVGFCSEV\ntranslation: AISRIVSRIRIVKMISVEGRLMVD\ntranslation: AISRIVSRIRIVKMISVEGRLMVD\ntranslation: STINLPSTLIIFTILILLTILEI\ntranslation: STINLPSTLIIFTILILLTILEI\ntranslation: RDQVRPLVLCMVIICFEVSLI\ntranslation: RDQVRPLVLCMVIICFEVSLI\ntranslation: FGDANQKKLNLETNLRDPLVQLL\ntranslation: FGDANQKKLNLETNLRDPLVQLL\ntranslation: RSWTRGSLRLVSRLSFFWLASPN\ntranslation: RSWTRGSLRLVSRLSFFWLASPN\ntranslation: RKYGKELLKKGHRQFKSNWSVF\ntranslation: RKYGKELLKKGHRQFKSNWSVF\ntranslation: QICRAAESP\ntranslation: QICRAAESP\ntranslation: KFVEQLSP\ntranslation: KFVEQLSP\ntranslation: WGLSCSTNL\ntranslation: WGLSCSTNL\ntranslation: GDSAALQI\ntranslation: GDSAALQI\ntranslation: GTQLLYKF\ntranslation: GTQLLYKF\ntranslation: LFIFFPEYLLKHQKLFFIHHLI\ntranslation: LFIFFPEYLLKHQKLFFIHHLI\ntranslation: AMRWWIKKSFWCFKRYSGKKMKR\ntranslation: AMRWWIKKSFWCFKRYSGKKMKR\ntranslation: NEMVDKKEFLVLQEIFRKKNEK\ntranslation: NEMVDKKEFLVLQEIFRKKNEK\ntranslation: HFPSFNYLLSLGLKFSLFKCIS\ntranslation: HFPSFNYLLSLGLKFSLFKCIS\ntranslation: TSLLSTICSAWASNSHFSSASH\ntranslation: TSLLSTICSAWASNSHFSSASH\ntranslation: LPFFQLFAQLGPQILTFQVHLM\ntranslation: LPFFQLFAQLGPQILTFQVHLM\ntranslation: SWDTASELVWYFNLRNTKLIRM\ntranslation: SWDTASELVWYFNLRNTKLIRM\ntranslation: LGHCLRTGMVLQFEKHKTDTN\ntranslation: LGHCLRTGMVLQFEKHKTDTN\ntranslation: IRISFVFLKLKYHTSSEAVSQ\ntranslation: IRISFVFLKLKYHTSSEAVSQ\ntranslation: QADMLHSKHDEEQKVWDEEIDDD\ntranslation: QADMLHSKHDEEQKVWDEEIDDD\ntranslation: KLTCYILNMMRSRRFGMKKLMM\ntranslation: KLTCYILNMMRSRRFGMKKLMM\ntranslation: SSSISSSQTFCSSSCLECNMSAW\ntranslation: SSSISSSQTFCSSSCLECNMSAW\ntranslation: AKHIHQMSEPAVPLVLSRITMA\ntranslation: AKHIHQMSEPAVPLVLSRITMA\ntranslation: VAMVILLSTRGTAGSDIWCMCLA\ntranslation: VAMVILLSTRGTAGSDIWCMCLA\ntranslation: CHGNPAQYERNRRFRHLVYVLG\ntranslation: CHGNPAQYERNRRFRHLVYVLG\ntranslation: CNQQLFYIYFNNANKICQKFS\ntranslation: CNQQLFYIYFNNANKICQKFS\ntranslation: AISNFSIYISIMPTKSAKSSL\ntranslation: AISNFSIYISIMPTKSAKSSL\ntranslation: RELLADFVGIIEIYIEKLLIA\ntranslation: RELLADFVGIIEIYIEKLLIA\ntranslation: RTVSLSRRRNSFIFCPPESF\ntranslation: RTVSLSRRRNSFIFCPPESF\ntranslation: ELSLCPGEGIHSFSVPQSPS\ntranslation: ELSLCPGEGIHSFSVPQSPS\ntranslation: KDSGGQKMNEFLLLDRETVL\ntranslation: KDSGGQKMNEFLLLDRETVL\ntranslation: IYLVHDRYKINIKVYLIKDYVYL\ntranslation: IYLVHDRYKINIKVYLIKDYVYL\ntranslation: QPSHHPQGKTLIHTRKHPHV\ntranslation: QPSHHPQGKTLIHTRKHPHV\ntranslation: SLLIILKVKPSFTRENTLMF\ntranslation: SLLIILKVKPSFTRENTLMF\ntranslation: NMRVFSRVNEGFTLRMMRRL\ntranslation: NMRVFSRVNEGFTLRMMRRL\ntranslation: LVLYTLYITIHLSIHVLMIIWV\ntranslation: LVLYTLYITIHLSIHVLMIIWV\ntranslation: CFVYLIYHNSFVHSCLDDYLG\ntranslation: CFVYLIYHNSFVHSCLDDYLG\ntranslation: ELDSDAPVGAKATSTSHSSPFFN\ntranslation: ELDSDAPVGAKATSTSHSSPFFN\ntranslation: RDQVRPLVLCMVIICFEVSLI\ntranslation: RDQVRPLVLCMVIICFEVSLI\ntranslation: CCLTELPSLMEIFISVLST\ntranslation: CCLTELPSLMEIFISVLST\ntranslation: LPHRTSFSDGNIYICAVYQ\ntranslation: LPHRTSFSDGNIYICAVYQ\ntranslation: LVDSTDINISIREGSSVRQ\ntranslation: LVDSTDINISIREGSSVRQ\ntranslation: GRQHRYKYFHQRRKFCEAA\ntranslation: GRQHRYKYFHQRRKFCEAA\ntranslation: RVNEGFMLLMWWVSEPHCVV\ntranslation: RVNEGFMLLMWWVSEPHCVV\ntranslation: TTQWGSLTHHINNIKPSFTR\ntranslation: TTQWGSLTHHINNIKPSFTR\ntranslation: HLWLPDPFGFSSWLTAILEA\ntranslation: HLWLPDPFGFSSWLTAILEA\ntranslation: SFQDSSQPAGKAEGIREPKV\ntranslation: SFQDSSQPAGKAEGIREPKV\ntranslation: NCKAKHFYEIVLPSRKYTQLQP\ntranslation: NCKAKHFYEIVLPSRKYTQLQP\ntranslation: GVEVVCIFLMVTLFHRNALPCNS\ntranslation: GVEVVCIFLMVTLFHRNALPCNS\ntranslation: KDSTVRKTQLQSFSQYIENRPEM\ntranslation: KDSTVRKTQLQSFSQYIENRPEM\ntranslation: KILQSEKLNFSLSANILRIDQR\ntranslation: KILQSEKLNFSLSANILRIDQR\ntranslation: HLWSILNILAERLKLSFSDCRIF\ntranslation: HLWSILNILAERLKLSFSDCRIF\ntranslation: LISLQISLIITFTATELIIFYIFF\ntranslation: LISLQISLIITFTATELIIFYIFF\ntranslation: NLPTNLLNYNIHSHRTNHILYLL\ntranslation: NLPTNLLNYNIHSHRTNHILYLL\ntranslation: FYPDRDLYSLLLQMYHDEEREIL\ntranslation: FYPDRDLYSLLLQMYHDEEREIL\ntranslation: FIQIEICILYYFKCTMMRKERS\ntranslation: FIQIEICILYYFKCTMMRKERS\ntranslation: GSLFPHHGTFEVIKNTDLDLDK\ntranslation: GSLFPHHGTFEVIKNTDLDLDK\ntranslation: ICITLQRISVELSVCTFFSNVYK\ntranslation: ICITLQRISVELSVCTFFSNVYK\ntranslation: LYHIAKNFSRIVSLHFFLKCVQ\ntranslation: LYHIAKNFSRIVSLHFFLKCVQ\ntranslation: LYTFEKKVQTDNSTEILCNVIQ\ntranslation: LYTFEKKVQTDNSTEILCNVIQ\ntranslation: PLEVVLMKEFPPYFFSLFFLLLA\ntranslation: PLEVVLMKEFPPYFFSLFFLLLA\ntranslation: HYLILSGIMFDLFNGALSDSSSS\ntranslation: HYLILSGIMFDLFNGALSDSSSS\ntranslation: GLELCLHRVVVRRELLVNVGFH\ntranslation: GLELCLHRVVVRRELLVNVGFH\ntranslation: MKSNIDKKFSAHYDAVEAELKS\ntranslation: MKSNIDKKFSAHYDAVEAELKS\ntranslation: PASLLHVGWPHQFWFCWNQPSP\ntranslation: PASLLHVGWPHQFWFCWNQPSP\ntranslation: QHLCCTLVGLINSGFVGISLLH\ntranslation: QHLCCTLVGLINSGFVGISLLH\ntranslation: SIFVARWLASSILVLLESAFSM\ntranslation: SIFVARWLASSILVLLESAFSM\ntranslation: MEKADSNKTRIDEANQRATKML\ntranslation: MEKADSNKTRIDEANQRATKML\ntranslation: WRRLIPTKPELMRPTNVQQRCW\ntranslation: WRRLIPTKPELMRPTNVQQRCW\ntranslation: VLYPVLCLLIEGKVLKALRPTS\ntranslation: VLYPVLCLLIEGKVLKALRPTS\ntranslation: GLLVSELSEPYLLSEDTVLGTAQ\ntranslation: GLLVSELSEPYLLSEDTVLGTAQ\ntranslation: ACWSQSFQNLTFYQKTQYWVQH\ntranslation: ACWSQSFQNLTFYQKTQYWVQH\ntranslation: LVGLRAFRTLPSIRRHSTGYST\ntranslation: LVGLRAFRTLPSIRRHSTGYST\ntranslation: SESHQQVV\ntranslation: SESHQQVV\ntranslation: VNLTNKLW\ntranslation: VNLTNKLW\ntranslation: PQLVGEIH\ntranslation: PQLVGEIH\ntranslation: HNLLVRFT\ntranslation: HNLLVRFT\ntranslation: NTVRCNLNCPQYLIKLLHLTQ\ntranslation: NTVRCNLNCPQYLIKLLHLTQ\ntranslation: MYVCVLCMAVYACECECEHVCG\ntranslation: MYVCVLCMAVYACECECEHVCG\ntranslation: CMYVCCVWQCMHVSVSVSMSV\ntranslation: CMYVCCVWQCMHVSVSVSMSV\ntranslation: PTDMLTLTLTCIHCHTQHTYIH\ntranslation: PTDMLTLTLTCIHCHTQHTYIH\ntranslation: PQTCSHSHSHAYTAIHNTHTYI\ntranslation: PQTCSHSHSHAYTAIHNTHTYI\ntranslation: HRHAHTHTHMHTLPYTTHIHT\ntranslation: HRHAHTHTHMHTLPYTTHIHT\ntranslation: QRCSCVV\ntranslation: QRCSCVV\ntranslation: EVFLCCD\ntranslation: EVFLCCD\ntranslation: ITTQEHL\ntranslation: ITTQEHL\ntranslation: SQHKNTS\ntranslation: SQHKNTS\ntranslation: HNTRTPL\ntranslation: HNTRTPL\ntranslation: PQSIYVRHS\ntranslation: PQSIYVRHS\ntranslation: SIYLREALL\ntranslation: SIYLREALL\ntranslation: GVPHVDRLR\ntranslation: GVPHVDRLR\ntranslation: ETRRQNVNSRSEIEINTAEFL\ntranslation: ETRRQNVNSRSEIEINTAEFL\ntranslation: KEFSCINFYFRSRIDILPSCF\ntranslation: KEFSCINFYFRSRIDILPSCF\ntranslation: RNSAVLISISDLELTFCLLV\ntranslation: RNSAVLISISDLELTFCLLV\ntranslation: SSGGLRHLSHRSGGPSGIWAGDC\ntranslation: SSGGLRHLSHRSGGPSGIWAGDC\ntranslation: GSHLPRSQKDLLSDDSSVSVHLS\ntranslation: GSHLPRSQKDLLSDDSSVSVHLS\ntranslation: IFQRCFVFLIKFLDLVHLFKLFL\ntranslation: IFQRCFVFLIKFLDLVHLFKLFL\ntranslation: KKKLKKVDKIEELDQENEAALEN\ntranslation: KKKLKKVDKIEELDQENEAALEN\ntranslation: RKSLKRWTRSRNLIKKTKQRWKM\ntranslation: RKSLKRWTRSRNLIKKTKQRWKM\ntranslation: LGATSLRFYCFTALNIIWLNLFSI\ntranslation: LGATSLRFYCFTALNIIWLNLFSI\ntranslation: SVNHVRFSVVFQK\ntranslation: SVNHVRFSVVFQK\ntranslation: CFWKTTENLTWLTL\ntranslation: CFWKTTENLTWLTL\ntranslation: FLEDYRKSNMVDT\ntranslation: FLEDYRKSNMVDT\ntranslation: LDFFLITVSLHCIHVGEQTCRCL\ntranslation: LDFFLITVSLHCIHVGEQTCRCL\ntranslation: WIFFLLQCHYTVFMWGNKHVGA\ntranslation: WIFFLLQCHYTVFMWGNKHVGA\ntranslation: VALLPSLLVHLSTHFNFPLIKMC\ntranslation: VALLPSLLVHLSTHFNFPLIKMC\ntranslation: HILIRGKLKCVLKCTNKLGSRAT\ntranslation: HILIRGKLKCVLKCTNKLGSRAT\ntranslation: ELEDKDLDFKNKDTDISIKP\ntranslation: ELEDKDLDFKNKDTDISIKP\ntranslation: SWKIKTWTLRIKTLTSQLNL\ntranslation: SWKIKTWTLRIKTLTSQLNL\ntranslation: SGLIEMSVSLFLKSKSLSSSS\ntranslation: SGLIEMSVSLFLKSKSLSSSS\ntranslation: VGYLIPVWVLAIVCSDMYVCVCV\ntranslation: VGYLIPVWVLAIVCSDMYVCVCV\ntranslation: GVSNPSLGLSYCVFRYVCVCVC\ntranslation: GVSNPSLGLSYCVFRYVCVCVC\ntranslation: THTHTYISEHTIAKTQTGIRYP\ntranslation: THTHTYISEHTIAKTQTGIRYP\ntranslation: LFIKFRTADLIPHLSHSDFS\ntranslation: LFIKFRTADLIPHLSHSDFS\ntranslation: LMQNGKRLKKKKTTIKKNTLNPY\ntranslation: LMQNGKRLKKKKTTIKKNTLNPY\ntranslation: VGVKCVLFNCCLFLLQPLTILHQ\ntranslation: VGVKCVLFNCCLFLLQPLTILHQ\ntranslation: HKVGIVCVTRVYRLFYLHALLIA\ntranslation: HKVGIVCVTRVYRLFYLHALLIA\ntranslation: FGLAKILSMSEFKSFLFTYHLHN\ntranslation: FGLAKILSMSEFKSFLFTYHLHN\ntranslation: SDSLKYYPCQNSNPSSSLIIFT\ntranslation: SDSLKYYPCQNSNPSSSLIIFT\ntranslation: FYTNTYSDFSVTLKFIFLSYQAS\ntranslation: FYTNTYSDFSVTLKFIFLSYQAS\ntranslation: CHLDLLMSQMYLSSSYTNVINYND\ntranslation: CHLDLLMSQMYLSSSYTNVINYND\ntranslation: GGGPAELVNLW\ntranslation: GGGPAELVNLW\ntranslation: PEIHQLSRTSS\ntranslation: PEIHQLSRTSS\ntranslation: QRFTNSAGPP\ntranslation: QRFTNSAGPP\ntranslation: RDSPTQQDLL\ntranslation: RDSPTQQDLL\ntranslation: IFCGTLASILHQESLCLHEM\ntranslation: IFCGTLASILHQESLCLHEM\ntranslation: SFHVNTVIPGEELRPKYHKK\ntranslation: SFHVNTVIPGEELRPKYHKK\ntranslation: KGLGRGGLPLIEDCIVDDTLQ\ntranslation: KGLGRGGLPLIEDCIVDDTLQ\ntranslation: VAMCHQQCSLQSMVIPLSPIL\ntranslation: VAMCHQQCSLQSMVIPLSPIL\ntranslation: CNVSSTMQSSINGNPPLPNP\ntranslation: CNVSSTMQSSINGNPPLPNP\ntranslation: LIKKLPPVLAIQLKRFDYDWE\ntranslation: LIKKLPPVLAIQLKRFDYDWE\nCPU times: user 792 ms, sys: 88 ms, total: 880 ms\nWall time: 828 ms\n"
      },
      {
       "name": "stderr",
       "output_type": "stream",
       "text": "\n"
      }
     ],
     "persistentId": "be06ded1-d7d4-4f21-9961-3aed5ff30cbf",
     "text": "%%time\n\nreads = 'SRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz'\nsubsampled_reads = 'SRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.fq.gz'\n\ncoding_scores = score_reads(subsampled_reads, peptide_graph, peptide_ksize)"
    },
    "executionTime": "2019-10-11T17:17:17.525Z"
   },
   {
    "cell": {
     "executionCount": 70,
     "executionEventId": "0ab94be1-8679-4e0c-a383-1eb7d65e9de1",
     "hasError": false,
     "id": "cfea0c3d-642a-421a-a4ad-07fa69bea946",
     "outputs": [
      {
       "data": {
        "text/plain": "(1000, 4)"
       },
       "execution_count": 70,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "09393220-d1f9-4ddd-9c7c-bc15160f3680",
     "text": "coding_scores.shape"
    },
    "executionTime": "2019-10-11T17:17:17.579Z"
   },
   {
    "cell": {
     "executionCount": 71,
     "executionEventId": "d502bef6-e492-4c96-94bb-987a74b7f5df",
     "hasError": false,
     "id": "3281c496-16a4-496e-81d4-fa57e8d1a556",
     "outputs": [
      {
       "data": {
        "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>read_id</th>\n      <th>jaccard_in_peptide_db</th>\n      <th>n_kmers</th>\n      <th>classification</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>SRR306838.10559374 Ibis_Run100924_C3PO:6:51:17601:17119/1</td>\n      <td>0</td>\n      <td>15</td>\n      <td>non-coding</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>SRR306838.6196593 Ibis_Run100924_C3PO:6:29:16733:12435/1</td>\n      <td>0</td>\n      <td>17</td>\n      <td>non-coding</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>SRR306838.20767303 Ibis_Run100924_C3PO:6:104:6864:5062/1</td>\n      <td>0</td>\n      <td>16</td>\n      <td>non-coding</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>SRR306838.12582274 Ibis_Run100924_C3PO:6:62:11779:17975/1</td>\n      <td>0</td>\n      <td>16</td>\n      <td>non-coding</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>SRR306838.13334230 Ibis_Run100924_C3PO:6:66:16579:20350/1</td>\n      <td>0</td>\n      <td>12</td>\n      <td>non-coding</td>\n    </tr>\n  </tbody>\n</table>\n</div>",
        "text/plain": "                                                     read_id  \\\n0  SRR306838.10559374 Ibis_Run100924_C3PO:6:51:17601:17119/1   \n1   SRR306838.6196593 Ibis_Run100924_C3PO:6:29:16733:12435/1   \n2   SRR306838.20767303 Ibis_Run100924_C3PO:6:104:6864:5062/1   \n3  SRR306838.12582274 Ibis_Run100924_C3PO:6:62:11779:17975/1   \n4  SRR306838.13334230 Ibis_Run100924_C3PO:6:66:16579:20350/1   \n\n   jaccard_in_peptide_db  n_kmers classification  \n0                      0       15     non-coding  \n1                      0       17     non-coding  \n2                      0       16     non-coding  \n3                      0       16     non-coding  \n4                      0       12     non-coding  "
       },
       "execution_count": 71,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "ede6e1ba-f524-4ceb-b35b-6c20f91d437c",
     "text": "coding_scores.head()"
    },
    "executionTime": "2019-10-11T17:17:17.659Z"
   },
   {
    "cell": {
     "executionCount": 73,
     "executionEventId": "48d05633-f2af-41c4-ad0a-7897b56bce40",
     "hasError": false,
     "id": "4c484434-ea07-4b72-8073-567bf902bedb",
     "outputs": [
      {
       "data": {
        "text/plain": "150289"
       },
       "execution_count": 73,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "75c01782-427c-4c63-bc29-ed308ae4a142",
     "text": "true_coding_read_ids = set(coding_read_ids.values)\nlen(true_coding_read_ids)"
    },
    "executionTime": "2019-10-11T17:17:18.500Z"
   },
   {
    "cell": {
     "executionCount": 74,
     "executionEventId": "92a8bddd-c29c-43cf-87e4-51ebfaa0e5cb",
     "hasError": false,
     "id": "b242d89f-8d25-4127-b875-8a85fa69d914",
     "outputs": [
      {
       "name": "stderr",
       "output_type": "stream",
       "text": "/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/screed/openscreed.py:39: ResourceWarning: unclosed file <_io.BufferedReader name='SRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.fq.gz'>\n  self.iter_fn = self.open_reader(filename, *args, **kwargs)\n266it [00:00, 1380.24it/s]"
      },
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "translation: ACLILTSIILGKSQYNCKSCSV\ntranslation: ACLILTSIILGKSQYNCKSCSV\ntranslation: TEQDLQLYCDFPNIIDVSIKQA\ntranslation: TEQDLQLYCDFPNIIDVSIKQA\ntranslation: QNRIYSYIAIFLILLMSVLSK\ntranslation: QNRIYSYIAIFLILLMSVLSK\ntranslation: LECEITYFMCLLTKNTNDKGKSH\ntranslation: LECEITYFMCLLTKNTNDKGKSH\ntranslation: WLFPLSLVFLVRRHMKYVISHSR\ntranslation: WLFPLSLVFLVRRHMKYVISHSR\ntranslation: QSSSLSNSRIFSSPQKEILNPL\ntranslation: QSSSLSNSRIFSSPQKEILNPL\ntranslation: ANGFRISFWGDENILELESDDD\ntranslation: ANGFRISFWGDENILELESDDD\ntranslation: KSEDLPGVFKDSLCSPCSQWCS\ntranslation: KSEDLPGVFKDSLCSPCSQWCS\ntranslation: KVKTSLGSSKTAFALHVANGA\ntranslation: KVKTSLGSSKTAFALHVANGA\ntranslation: RAPLATWRAKAVFEDPREVFTF\ntranslation: RAPLATWRAKAVFEDPREVFTF\ntranslation: EHHWLHGEQRLSLKTPGRSSLL\ntranslation: EHHWLHGEQRLSLKTPGRSSLL\ntranslation: VTLNSSEKIDNIVEWDGG\ntranslation: VTLNSSEKIDNIVEWDGG\ntranslation: VPPSHSTMLSIFSEELRV\ntranslation: VPPSHSTMLSIFSEELRV\ntranslation: QSSSPEFRVQSFSERTNARKKNNH\ntranslation: QSSSPEFRVQSFSERTNARKKNNH\ntranslation: SLQVQNLEFSPFLKEQMLGKKIIT\ntranslation: SLQVQNLEFSPFLKEQMLGKKIIT\ntranslation: LILDNFFWNKSFIYIRLVSLSSE\ntranslation: LILDNFFWNKSFIYIRLVSLSSE\ntranslation: SELNDTSLMYMKDLFQKKLSRIK\ntranslation: SELNDTSLMYMKDLFQKKLSRIK\ntranslation: TRLTYTICNTRWIAHTYDTI\ntranslation: TRLTYTICNTRWIAHTYDTI\ntranslation: EIVSYVCAIHLVLQIVYVSLV\ntranslation: EIVSYVCAIHLVLQIVYVSLV\ntranslation: KLCHMCVLSISCYRLCMLAL\ntranslation: KLCHMCVLSISCYRLCMLAL\ntranslation: CSVVHSQTATCKNKFTIHSALL\ntranslation: CSVVHSQTATCKNKFTIHSALL\ntranslation: VVWSIPRQQHARISLQYTQPF\ntranslation: VVWSIPRQQHARISLQYTQPF\ntranslation: RRAECIVNLFLHVAVWEWTTLQ\ntranslation: RRAECIVNLFLHVAVWEWTTLQ\ntranslation: KFNIPCNFSDLSFISCKYYYFP\ntranslation: KFNIPCNFSDLSFISCKYYYFP\ntranslation: DRLKLVLLFLTFSSLHSAVLSVL\ntranslation: DRLKLVLLFLTFSSLHSAVLSVL\ntranslation: STDSTAECRLENVKNNKTSFSLS\ntranslation: STDSTAECRLENVKNNKTSFSLS\ntranslation: AQTAQQSADWRTLRITKPVSVYQ\ntranslation: AQTAQQSADWRTLRITKPVSVYQ\ntranslation: NKVTFNNNLIHINRERAIDKVN\ntranslation: NKVTFNNNLIHINRERAIDKVN\ntranslation: IYFINCSFSIYVYQIIVKCYLI\ntranslation: IYFINCSFSIYVYQIIVKCYLI\ntranslation: FTLSIALSLFMCIRLLLNVTLF\ntranslation: FTLSIALSLFMCIRLLLNVTLF\ntranslation: LDPPYSRVITQRETENNQMTSE\ntranslation: LDPPYSRVITQRETENNQMTSE\ntranslation: SDVIWLFSVSLCVITLEYGGSK\ntranslation: SDVIWLFSVSLCVITLEYGGSK\ntranslation: NTSNLYFLKYVFLRYLRIHERA\ntranslation: NTSNLYFLKYVFLRYLRIHERA\ntranslation: ARSCILRYLRNTYLRKYKLDVL\ntranslation: ARSCILRYLRNTYLRKYKLDVL\ntranslation: QCRASGDCSRM\ntranslation: QCRASGDCSRM\ntranslation: NAVPVETVLVC\ntranslation: NAVPVETVLVC\ntranslation: MPCQWRLFSYA\ntranslation: MPCQWRLFSYA\ntranslation: GIREQSPLARH\ntranslation: GIREQSPLARH\ntranslation: AYENSLHWHGI\ntranslation: AYENSLHWHGI\ntranslation: HTRTVSTGTAL\ntranslation: HTRTVSTGTAL\ntranslation: LYLLHNELTRNNFARRAKAKTPE\ntranslation: LYLLHNELTRNNFARRAKAKTPE\ntranslation: VSGVLALALLAKLFLVNSLCRRYR\ntranslation: VSGVLALALLAKLFLVNSLCRRYR\ntranslation: GMRPRLKFAAKAQNLVSINSSDM\ntranslation: GMRPRLKFAAKAQNLVSINSSDM\ntranslation: MSEEFMLTRFWAFAANFSRGLIP\ntranslation: MSEEFMLTRFWAFAANFSRGLIP\ntranslation: SDEHGIDPTGTYH\ntranslation: SDEHGIDPTGTYH\ntranslation: VTNMASTPQAHTM\ntranslation: VTNMASTPQAHTM\ntranslation: PWYVPVGSMPCSSL\ntranslation: PWYVPVGSMPCSSL\ntranslation: HGMCLWGRCHVRH\ntranslation: HGMCLWGRCHVRH\ntranslation: MVCACGVDAMFVT\ntranslation: MVCACGVDAMFVT\ntranslation: MIKIMHRHGYLIFSNRKSHPYFL\ntranslation: MIKIMHRHGYLIFSNRKSHPYFL\ntranslation: QQPLCLISTPGPCRGNRNSQLS\ntranslation: QQPLCLISTPGPCRGNRNSQLS\ntranslation: AASMLDFHTGSVQRKQKLPTV\ntranslation: AASMLDFHTGSVQRKQKLPTV\ntranslation: RTVGSFCFLCTDPVWKSSIEAA\ntranslation: RTVGSFCFLCTDPVWKSSIEAA\ntranslation: DSWEFLFPLHGPGVEIKHRGC\ntranslation: DSWEFLFPLHGPGVEIKHRGC\ntranslation: LNSLRNKPGECVTVLGEDPAHS\ntranslation: LNSLRNKPGECVTVLGEDPAHS\ntranslation: SHHIYVLQIVYVSLVYVCLI\ntranslation: SHHIYVLQIVYVSLVYVCLI\ntranslation: YQQRSMMLHSCTMYALVVNGCE\ntranslation: YQQRSMMLHSCTMYALVVNGCE\ntranslation: RKASLRCSLLRIILFFSRCLRKC\ntranslation: RKASLRCSLLRIILFFSRCLRKC\ntranslation: ESKPTLLTITYNPFLFKMPEEV\ntranslation: ESKPTLLTITYNPFLFKMPEEV\ntranslation: HFLRHLEKKRIIRNSEQRRLAF\ntranslation: HFLRHLEKKRIIRNSEQRRLAF\ntranslation: TSSGILKRKGLYVIVSNVGLLS\ntranslation: TSSGILKRKGLYVIVSNVGLLS\ntranslation: LEGCLITEPSFL\ntranslation: LEGCLITEPSFL\ntranslation: SRNEGSVMRQPS\ntranslation: SRNEGSVMRQPS\ntranslation: HICRLVPEG\ntranslation: HICRLVPEG\ntranslation: TYAGWCPR\ntranslation: TYAGWCPR\ntranslation: HMQAGARG\ntranslation: HMQAGARG\ntranslation: PLGHQPAYV\ntranslation: PLGHQPAYV\ntranslation: PSGTSLHM\ntranslation: PSGTSLHM\ntranslation: PRAPACIC\ntranslation: PRAPACIC\ntranslation: LISLFIRLKCFSTEWYRDDICL\ntranslation: LISLFIRLKCFSTEWYRDDICL\ntranslation: HKHISSLYHSVLKHLSLINKEI\ntranslation: HKHISSLYHSVLKHLSLINKEI\ntranslation: ANSQSHHEFSLRKSAFVLTTWPG\ntranslation: ANSQSHHEFSLRKSAFVLTTWPG\ntranslation: SLAKWSKQRLIFGGKIHDGFENL\ntranslation: SLAKWSKQRLIFGGKIHDGFENL\ntranslation: KIHCFVFPQREILLEPVNRMYY\ntranslation: KIHCFVFPQREILLEPVNRMYY\ntranslation: NTFYLQVLVRSLFEGRQNSVF\ntranslation: NTFYLQVLVRSLFEGRQNSVF\ntranslation: LLIILDCAEMFLILRGLESSAHL\ntranslation: LLIILDCAEMFLILRGLESSAHL\ntranslation: PDNLRLCRNVSYFEGLRKLSAFD\ntranslation: PDNLRLCRNVSYFEGLRKLSAFD\ntranslation: IKCAELSKPLKIRNISAQSKIIR\ntranslation: IKCAELSKPLKIRNISAQSKIIR\ntranslation: TNHHPTMTNQTNLKTNDNHTQH\ntranslation: TNHHPTMTNQTNLKTNDNHTQH\ntranslation: VLCMVIICFEVSLISHCWVVIS\ntranslation: VLCMVIICFEVSLISHCWVVIS\ntranslation: NDYHLTNHTVCIRLLLKLQCA\ntranslation: NDYHLTNHTVCIRLLLKLQCA\ntranslation: MIIISQIIQCALDCFSNFSVH\ntranslation: MIIISQIIQCALDCFSNFSVH\ntranslation: CTLKFEKQSNAHCMICEMIII\ntranslation: CTLKFEKQSNAHCMICEMIII\ntranslation: SLSTQHRHAHKERLKKVKGTRQI\ntranslation: SLSTQHRHAHKERLKKVKGTRQI\ntranslation: FAEFLLLFLTFPYEHACVGLTVR\ntranslation: FAEFLLLFLTFPYEHACVGLTVR\ntranslation: NEMLVPRKLPKTKFHQMIKSKME\ntranslation: NEMLVPRKLPKTKFHQMIKSKME\ntranslation: LLFCSTLKLLKIKLAHVDTMW\ntranslation: LLFCSTLKLLKIKLAHVDTMW\ntranslation: LIYLFLIQLKRNFYMNLGFYSVPL\ntranslation: LIYLFLIQLKRNFYMNLGFYSVPL\ntranslation: LEETSLQTQNEIFINIKIVKQRS\ntranslation: LEETSLQTQNEIFINIKIVKQRS\ntranslation: RFFAHPYFPHTKAKIFVNANNM\ntranslation: RFFAHPYFPHTKAKIFVNANNM\ntranslation: HIVSIYKYFCFSMRKVRMGKEA\ntranslation: HIVSIYKYFCFSMRKVRMGKEA\ntranslation: LYYSVFRFVSSYVNFRARASKL\ntranslation: LYYSVFRFVSSYVNFRARASKL\ntranslation: LSPVSRRVIFDQENVVTSTESS\ntranslation: LSPVSRRVIFDQENVVTSTESS\ntranslation: RTLCASNHVLLIKYHSPTYRTQ\ntranslation: RTLCASNHVLLIKYHSPTYRTQ\ntranslation: IMECSSLTIAHCSFNLPGTSDP\ntranslation: IMECSSLTIAHCSFNLPGTSDP\ntranslation: RIACAWEVEAAVSYGQATALHY\ntranslation: RIACAWEVEAAVSYGQATALHY\ntranslation: GTELSMCSRTSWAMTDTTDCS\ntranslation: GTELSMCSRTSWAMTDTTDCS\ntranslation: AEQSVVSVIAQLVLEHMDNSVP\ntranslation: AEQSVVSVIAQLVLEHMDNSVP\ntranslation: RAICGVCHSPTSPGAHGQFCT\ntranslation: RAICGVCHSPTSPGAHGQFCT\ntranslation: EVTVYHIKMVHGKYMAHFLFVL\ntranslation: EVTVYHIKMVHGKYMAHFLFVL\ntranslation: QNKKKMCHIFSMYHLYMIYSDF\ntranslation: QNKKKMCHIFSMYHLYMIYSDF\ntranslation: SVEINHIMAKGHDGRSNQRCS\ntranslation: SVEINHIMAKGHDGRSNQRCS\ntranslation: RTPLITPAIMTLGHNMIYLHT\ntranslation: RTPLITPAIMTLGHNMIYLHT\ntranslation: SVFLFCKMGVPPTL\ntranslation: SVFLFCKMGVPPTL\ntranslation: QSSFSVRWGSRLPC\ntranslation: QSSFSVRWGSRLPC\ntranslation: GRVGGTPILQKRKTE\ntranslation: GRVGGTPILQKRKTE\ntranslation: WEVKGKKTLLSFFPVWEELRNW\ntranslation: WEVKGKKTLLSFFPVWEELRNW\ntranslation: GKLKGRKHCYHFFLFGKSLET\ntranslation: GKLKGRKHCYHFFLFGKSLET\ntranslation: QFLSSSQTGKNDNSVFFPLTSQ\ntranslation: QFLSSSQTGKNDNSVFFPLTSQ\ntranslation: PHSRASWCFLYKHRAP\ntranslation: PHSRASWCFLYKHRAP\ntranslation: HILGLLGVSSINTELP\ntranslation: HILGLLGVSSINTELP\ntranslation: RELCVYRGNTKKPENV\ntranslation: RELCVYRGNTKKPENV\ntranslation: GSSVFIEETPRSPRMW\ntranslation: GSSVFIEETPRSPRMW\ntranslation: QIDVNVQDHFWDGKGCEMIC\ntranslation: QIDVNVQDHFWDGKGCEMIC\ntranslation: TDHLTSLPIPEMILDIDINL\ntranslation: TDHLTSLPIPEMILDIDINL\ntranslation: YHAPSVLFYVHQLPILHPSRL\ntranslation: YHAPSVLFYVHQLPILHPSRL\ntranslation: IMHLLSFFMSINFLFYTLVG\ntranslation: IMHLLSFFMSINFLFYTLVG\ntranslation: LLYPSPSSLSSIQTSSKITIHWR\ntranslation: LLYPSPSSLSSIQTSSKITIHWR\ntranslation: SPVNGYFGGSLYARQGGRRWIKQ\ntranslation: SPVNGYFGGSLYARQGGRRWIKQ\ntranslation: QICMSSWKELLTLYLRPAREY\ntranslation: QICMSSWKELLTLYLRPAREY\ntranslation: NLHELLERVADFVSKTCQGIP\ntranslation: NLHELLERVADFVSKTCQGIP\ntranslation: WYSLAGLRYKVSNSFQELMQI\ntranslation: WYSLAGLRYKVSNSFQELMQI\ntranslation: GIPWQVLDTKSATLSRSSCKF\ntranslation: GIPWQVLDTKSATLSRSSCKF\ntranslation: PVFFSFKNVKLRVVHVFEVPMIF\ntranslation: PVFFSFKNVKLRVVHVFEVPMIF\ntranslation: NIIGTSKTCTTLNLTFLKEKKT\ntranslation: NIIGTSKTCTTLNLTFLKEKKT\ntranslation: FVFNVFSWVIRNSVRSMGVILLHD\ntranslation: FVFNVFSWVIRNSVRSMGVILLHD\ntranslation: VMQKNYPHTPYTIPYHPTKNIKHK\ntranslation: VMQKNYPHTPYTIPYHPTKNIKHK\ntranslation: SCKRITPILLTLFLITQLKTLNT\ntranslation: SCKRITPILLTLFLITQLKTLNT\ntranslation: IAFFCSSSFNNSAFSFSFTF\ntranslation: IAFFCSSSFNNSAFSFSFTF\ntranslation: SRFSAPVLSTTQHFRSLSL\ntranslation: SRFSAPVLSTTQHFRSLSL\ntranslation: RVFLLQFFQQLSIFVLFHF\ntranslation: RVFLLQFFQQLSIFVLFHF\ntranslation: KVKENENAELLKELEQKNA\ntranslation: KVKENENAELLKELEQKNA\ntranslation: LGLQLKFHFHDSNFYLIKGRLKY\ntranslation: LGLQLKFHFHDSNFYLIKGRLKY\ntranslation: NTKLKKKLGMQAQADTEYFPS\ntranslation: NTKLKKKLGMQAQADTEYFPS\ntranslation: MKENTQCQLEPAFQAFFLIWC\ntranslation: MKENTQCQLEPAFQAFFLIWC\ntranslation: EELITFHDHALIIIFLICFLVL\ntranslation: EELITFHDHALIIIFLICFLVL\ntranslation: GRFGTSLKIGIVGLPNVGKSTFF\ntranslation: GRFGTSLKIGIVGLPNVGKSTFF\ntranslation: KIWNLTENWYCWIAKCWEIYFL\ntranslation: KIWNLTENWYCWIAKCWEIYFL\ntranslation: KKVDFPTFGNPTIPIFSEVPNL\ntranslation: KKVDFPTFGNPTIPIFSEVPNL\ntranslation: RSCKLLTDQVCRTKSHYILLYL\ntranslation: RSCKLLTDQVCRTKSHYILLYL\ntranslation: ASKVICNDFLSYKLGQSAAYMI\ntranslation: ASKVICNDFLSYKLGQSAAYMI\ntranslation: SYVMQGICLMKYTNTPDQKLCL\ntranslation: SYVMQGICLMKYTNTPDQKLCL\ntranslation: ETKFLVRCICILHQAYALHYIA\ntranslation: ETKFLVRCICILHQAYALHYIA\ntranslation: DKVSGQVYLYTSSGICLALHS\ntranslation: DKVSGQVYLYTSSGICLALHS\ntranslation: LSTRFSNSIVKCSPLEHVLKYLD\ntranslation: LSTRFSNSIVKCSPLEHVLKYLD\ntranslation: SKYFRTCSRGEHLTIEFENLVE\ntranslation: SKYFRTCSRGEHLTIEFENLVE\ntranslation: HGKAKSQLNQGQRTLGPTDVP\ntranslation: HGKAKSQLNQGQRTLGPTDVP\ntranslation: GTSVGPRVLCPWFNCDLAFP\ntranslation: GTSVGPRVLCPWFNCDLAFP\ntranslation: GHQWGQGFSVPGSTVIWLSR\ntranslation: GHQWGQGFSVPGSTVIWLSR\ntranslation: PVKMNFLAEMKHELTEMKHEL\ntranslation: PVKMNFLAEMKHELTEMKHEL\ntranslation: VSSCFISVSSCFISARKFIFTG\ntranslation: VSSCFISVSSCFISARKFIFTG\ntranslation: KFMLHLCKFMLHLCKKIHFHG\ntranslation: KFMLHLCKFMLHLCKKIHFHG\ntranslation: SYSGRLSFNISTEVPKMYTQKA\ntranslation: SYSGRLSFNISTEVPKMYTQKA\ntranslation: MHLYSLFLIESYGY\ntranslation: MHLYSLFLIESYGY\ntranslation: CIFTHSFSLRAMA\ntranslation: CIFTHSFSLRAMA\ntranslation: GHPLHSALILLLPATMGRRICL\ntranslation: GHPLHSALILLLPATMGRRICL\ntranslation: GIHFTQLSFFSYLPPWVEGYV\ntranslation: GIHFTQLSFFSYLPPWVEGYV\ntranslation: LNFNKAHPAVSSNRQPFMITES\ntranslation: LNFNKAHPAVSSNRQPFMITES\ntranslation: QFQQGTSCSFQQQTTFHDNRI\ntranslation: QFQQGTSCSFQQQTTFHDNRI\ntranslation: RFCYHERLSVAGNCRMCLVEIE\ntranslation: RFCYHERLSVAGNCRMCLVEIE\ntranslation: DSVIMKGCLLLETAGCALLKL\ntranslation: DSVIMKGCLLLETAGCALLKL\ntranslation: FPEFLKSGILLGKHDMLMLKATL\ntranslation: FPEFLKSGILLGKHDMLMLKATL\ntranslation: RVAFNINISCFPSKIPLLRNSGK\ntranslation: RVAFNINISCFPSKIPLLRNSGK\ntranslation: CISHKFCSFFYHSSSDICQTSPF\ntranslation: CISHKFCSFFYHSSSDICQTSPF\ntranslation: ASVINSAHSSITAPLTFARPHHF\ntranslation: ASVINSAHSSITAPLTFARPHHF\ntranslation: INVHSILWLNNIPLLNIVCFVYL\ntranslation: INVHSILWLNNIPLLNIVCFVYL\ntranslation: MRVFSRVNEGFMLLMWFFVFHC\ntranslation: MRVFSRVNEGFMLLMWFFVFHC\ntranslation: RHCSISQSRNILLSLMTLTLFS\ntranslation: RHCSISQSRNILLSLMTLTLFS\ntranslation: DIKNIYKRMTGSEFDFEEMKRK\ntranslation: DIKNIYKRMTGSEFDFEEMKRK\ntranslation: GVVPFIHNIVNIHGLYYMCFSL\ntranslation: GVVPFIHNIVNIHGLYYMCFSL\ntranslation: SCPIYTQHCKHTRSILHVLQS\ntranslation: SCPIYTQHCKHTRSILHVLQS\ntranslation: ISLLSQHKYLWSLPLSSSAWSFS\ntranslation: ISLLSQHKYLWSLPLSSSAWSFS\ntranslation: YLCYPNTNIFGLSLSPPQPGLSL\ntranslation: YLCYPNTNIFGLSLSPPQPGLSL\ntranslation: IFVIPTQISLVSPSLLLSLVFLF\ntranslation: IFVIPTQISLVSPSLLLSLVFLF\ntranslation: KEKDQAEEERGRDQRYLCWDNKD\ntranslation: KEKDQAEEERGRDQRYLCWDNKD\ntranslation: KRKTRLRRREGETKDICVGITKI\ntranslation: KRKTRLRRREGETKDICVGITKI\ntranslation: VQSYFKSALKISFCSYSKYME\ntranslation: VQSYFKSALKISFCSYSKYME\ntranslation: YKATLSLLLRSVFAHIQNTWN\ntranslation: YKATLSLLLRSVFAHIQNTWN\ntranslation: AAKYFDFRLLSNLSFQKHFIK\ntranslation: AAKYFDFRLLSNLSFQKHFIK\ntranslation: AIKAKTKNKQTKTQPNNSTPST\ntranslation: AIKAKTKNKQTKTQPNNSTPST\ntranslation: RVEGVLLFGWVFVCLFFVFAFIA\ntranslation: RVEGVLLFGWVFVCLFFVFAFIA\ntranslation: EWKEYYCLVGFLFVCFLFLLLLP\ntranslation: EWKEYYCLVGFLFVCFLFLLLLP\ntranslation: SGRSTIVWLGFCLFVFCFCFYC\ntranslation: SGRSTIVWLGFCLFVFCFCFYC\ntranslation: FCYAVLVDLLHMFFYRLNTTTP\ntranslation: FCYAVLVDLLHMFFYRLNTTTP\ntranslation: VWWYLVYRRTCGAGQPKLHSR\ntranslation: VWWYLVYRRTCGAGQPKLHSR\ntranslation: KVASSSTPRTLNFLHFLSRQS\ntranslation: KVASSSTPRTLNFLHFLSRQS\ntranslation: ALSRQEMQEVQSSRSGRGGNF\ntranslation: ALSRQEMQEVQSSRSGRGGNF\ntranslation: LCLDKKCRKFRVLGVEEEATL\ntranslation: LCLDKKCRKFRVLGVEEEATL\ntranslation: QVSTPKTTSSKPQTYHTQTP\ntranslation: QVSTPKTTSSKPQTYHTQTP\ntranslation: RFLLQRPHHRNRKHIIHKRL\ntranslation: RFLLQRPHHRNRKHIIHKRL\ntranslation: GFYSKDHIIETANISYTNA\ntranslation: GFYSKDHIIETANISYTNA\ntranslation: QENDSVYYSQFDPMYPTVGSNT\ntranslation: QENDSVYYSQFDPMYPTVGSNT\ntranslation: AFKLNTHYLKNPKHITELLTPNW\ntranslation: AFKLNTHYLKNPKHITELLTPNW\ntranslation: SNLFETSNGFLSDKTFTTLSFKS\ntranslation: SNLFETSNGFLSDKTFTTLSFKS\ntranslation: ADLKDKVVNVLSLKNPLEVSNKLL\ntranslation: ADLKDKVVNVLSLKNPLEVSNKLL\ntranslation: NVVHQLSVTLEDLYNGATRK\ntranslation: NVVHQLSVTLEDLYNGATRK\ntranslation: VFLLHHYISLLGLLRADVQHF\ntranslation: VFLLHHYISLLGLLRADVQHF\ntranslation: EGFLAKSFSRTISRDLPVSHHD\ntranslation: EGFLAKSFSRTISRDLPVSHHD\ntranslation: APVPSLFKGTCTGIGNVPHLPSC\ntranslation: APVPSLFKGTCTGIGNVPHLPSC\ntranslation: QEGRWGTFPIPVHVPLNKLGTGA\ntranslation: QEGRWGTFPIPVHVPLNKLGTGA\ntranslation: AWNRMVCFNVYLCMFIFMVKGVN\ntranslation: AWNRMVCFNVYLCMFIFMVKGVN\ntranslation: HGTEWSASMFIYVCSSSWLKEL\ntranslation: HGTEWSASMFIYVCSSSWLKEL\ntranslation: LILCCYNEDKVHTETTVRLRNFH\ntranslation: LILCCYNEDKVHTETTVRLRNFH\ntranslation: CNLELCIIIQHIVPSIFKWFLQR\ntranslation: CNLELCIIIQHIVPSIFKWFLQR\ntranslation: PLQEPFKNAWNYMLNNYTKFQIA\ntranslation: PLQEPFKNAWNYMLNNYTKFQIA\ntranslation: RIGATHFFSPWFYAGSRSQVKS\ntranslation: RIGATHFFSPWFYAGSRSQVKS\ntranslation: DHIAILSCKGDWNSENLDL\ntranslation: DHIAILSCKGDWNSENLDL\ntranslation: IHLSLWLFVYYLCIFSCIDIIDI\ntranslation: IHLSLWLFVYYLCIFSCIDIIDI\ntranslation: IFAADLLFTMDSIMRFFFFVG\ntranslation: IFAADLLFTMDSIMRFFFFVG\ntranslation: FCCGSPIYNGLHYEILLLCWC\ntranslation: FCCGSPIYNGLHYEILLLCWC\ntranslation: TPTKKKNLIMESIVNRRSAAK\ntranslation: TPTKKKNLIMESIVNRRSAAK\ntranslation: LNDVQDRFEVNISELPDEIDISS\ntranslation: LNDVQDRFEVNISELPDEIDISS\ntranslation: GGDVYLIRQLTNIDLKAILHIIE\ntranslation: GGDVYLIRQLTNIDLKAILHIIE\ntranslation: EEMSISSGSSLILTSKRSCTSLR\ntranslation: EEMSISSGSSLILTSKRSCTSLR\ntranslation: QCTRDFKLLYLCQKFKTKQPPKS\ntranslation: QCTRDFKLLYLCQKFKTKQPPKS\ntranslation: SVREILNYFISVRNSKLNNLQSL\ntranslation: SVREILNYFISVRNSKLNNLQSL\ntranslation: ISFKVCLTDTSFISYDMLRLIFLF\ntranslation: ISFKVCLTDTSFISYDMLRLIFLF\ntranslation: KRKINLNISYEINEVSVKHTLKL\ntranslation: KRKINLNISYEINEVSVKHTLKL\ntranslation: SLLFTFSYFQKNSSEKYVMQ\ntranslation: SLLFTFSYFQKNSSEKYVMQ\ntranslation: LKVSHCLGFSIRILSSTFSVMPL\ntranslation: LKVSHCLGFSIRILSSTFSVMPL\ntranslation: RGITENVLDKILIENPKQWLTFK\ntranslation: RGITENVLDKILIENPKQWLTFK\ntranslation: RAISITIYTPTNNVQPVTTTN\ntranslation: RAISITIYTPTNNVQPVTTTN\ntranslation: VSWHCDHSIGNSLPKVGFCNF\ntranslation: VSWHCDHSIGNSLPKVGFCNF\ntranslation: GNCRSLPWEDCYQCCGHSASLL\ntranslation: GNCRSLPWEDCYQCCGHSASLL\ntranslation: EIAEAYLGKTVTNAVVTVPAY\ntranslation: EIAEAYLGKTVTNAVVTVPAY\ntranslation: KLQKPTLGRLLPMLWSQCQLT\ntranslation: KLQKPTLGRLLPMLWSQCQLT\ntranslation: NILMNIYGWTYVHVQWRK\ntranslation: NILMNIYGWTYVHVQWRK\ntranslation: PSSIAHEHMSIHIYSSEC\ntranslation: PSSIAHEHMSIHIYSSEC\ntranslation: LPPLHMNICPSIYIHQNV\ntranslation: LPPLHMNICPSIYIHQNV\ntranslation: GYEFFLLGLMRVVRMGGIREGG\ntranslation: GYEFFLLGLMRVVRMGGIREGG\ntranslation: PPSLIPPILTTLINPNKKNSYP\ntranslation: PPSLIPPILTTLINPNKKNSYP\ntranslation: IQHANDVTPLFIITTVIVKFI\ntranslation: IQHANDVTPLFIITTVIVKFI\ntranslation: NELNYHRGNDKERSYIISMLN\ntranslation: NELNYHRGNDKERSYIISMLN\ntranslation: RKCLLEKIDVITGEETEHNVL\ntranslation: RKCLLEKIDVITGEETEHNVL\ntranslation: NTLCSVSSPVITSIFSSKHF\ntranslation: NTLCSVSSPVITSIFSSKHF\ntranslation: QTQKQHLNSRPHPLNQTLL\ntranslation: QTQKQHLNSRPHPLNQTLL\ntranslation: WFKKSKLLVLNEAILDYDGFFSLL\ntranslation: WFKKSKLLVLNEAILDYDGFFSLL\ntranslation: VGKFKVELRFYLGQPAITRLG\ntranslation: VGKFKVELRFYLGQPAITRLG\ntranslation: LISLVCNLSWLISKTYYSFQL\ntranslation: LISLVCNLSWLISKTYYSFQL\ntranslation: SFLLFVIYHGSFQRHITAFS\ntranslation: SFLLFVIYHGSFQRHITAFS\ntranslation: QISHQQPTNHHPTMTNQTNLKTN\ntranslation: QISHQQPTNHHPTMTNQTNLKTN\ntranslation: ICFEVSLISHCWVVISRLLMRYL\ntranslation: ICFEVSLISHCWVVISRLLMRYL\ntranslation: EKWPRKP\ntranslation: EKWPRKP\ntranslation: RNGPESQ\ntranslation: RNGPESQ\ntranslation: EMAQKAR\ntranslation: EMAQKAR\ntranslation: PGFLGHF\ntranslation: PGFLGHF\ntranslation: LAFWAIS\ntranslation: LAFWAIS\ntranslation: WLSGPFL\ntranslation: WLSGPFL\ntranslation: LFWQIVLNKIKVAYISNITQHP\ntranslation: LFWQIVLNKIKVAYISNITQHP\ntranslation: CVCCSHHIYVLQIVYVSLVYVCL\ntranslation: CVCCSHHIYVLQIVYVSLVYVCL\ntranslation: VCVAVIISMCYRLCMLALCMCA\ntranslation: VCVAVIISMCYRLCMLALCMCA\ntranslation: QVRPLVLCMVIICFEVSLISHCW\ntranslation: QVRPLVLCMVIICFEVSLISHCW\ntranslation: IVPILIAMAFLMLTERKILGYIQ\ntranslation: IVPILIAMAFLMLTERKILGYIQ\ntranslation: CTHSNRNGIPNAYRTKNSRLYTT\ntranslation: CTHSNRNGIPNAYRTKNSRLYTT\ntranslation: MISNTTNTQEKKYNTANSVLQLCI\ntranslation: MISNTTNTQEKKYNTANSVLQLCI\ntranslation: ASKSQLLAHQFIWNMKTNIYLDE\ntranslation: ASKSQLLAHQFIWNMKTNIYLDE\ntranslation: GPPDLNKPCSLYTSLNWTYHLY\ntranslation: GPPDLNKPCSLYTSLNWTYHLY\ntranslation: MSQKFSHPETTKEKKSLIKCHAF\ntranslation: MSQKFSHPETTKEKKSLIKCHAF\ntranslation: SEVSQVEDVIELLVVVLFNVGG\ntranslation: SEVSQVEDVIELLVVVLFNVGG\ntranslation: RPPTLNRTTTKSSMTSSTWETSL\ntranslation: RPPTLNRTTTKSSMTSSTWETSL\ntranslation: TTYIKQNHYQELNDILNLGNFT\ntranslation: TTYIKQNHYQELNDILNLGNFT\ntranslation: LGSYQIKFQVKDNLLWQFKKNTE\ntranslation: LGSYQIKFQVKDNLLWQFKKNTE\ntranslation: WVHTRLNFKLKIIYYGSLKKTQN\ntranslation: WVHTRLNFKLKIIYYGSLKKTQN\ntranslation: PGFGEWGWGCNSVLGRNGGGGCA\ntranslation: PGFGEWGWGCNSVLGRNGGGGCA\ntranslation: RVWRMGMGVQLCVRKEWWGRLC\ntranslation: RVWRMGMGVQLCVRKEWWGRLC\ntranslation: QHNLPHHSFLTQSCTPIPILQTL\ntranslation: QHNLPHHSFLTQSCTPIPILQTL\ntranslation: AQPPPPFLPNTELHPHPHSPNP\ntranslation: AQPPPPFLPNTELHPHPHSPNP\ntranslation: SQSSALGIYLCSQR\ntranslation: SQSSALGIYLCSQR\ntranslation: LSPLHWEFIYVLNA\ntranslation: LSPLHWEFIYVLNA\ntranslation: SVLCTGNLFMFST\ntranslation: SVLCTGNLFMFST\ntranslation: CVENINKFPVQRTE\ntranslation: CVENINKFPVQRTE\ntranslation: KFFTLSTRFFPSVQRAVPLWT\ntranslation: KFFTLSTRFFPSVQRAVPLWT\ntranslation: NFLLSLQGFFLVSKELFLFG\ntranslation: NFLLSLQGFFLVSKELFLFG\ntranslation: SPKRNSSLDTRKKPCRESKKF\ntranslation: SPKRNSSLDTRKKPCRESKKF\ntranslation: VQRGTALWTLGKNLVERVKN\ntranslation: VQRGTALWTLGKNLVERVKN\ntranslation: LGYNFSSFPCGTISIAPGFNFYR\ntranslation: LGYNFSSFPCGTISIAPGFNFYR\ntranslation: LVIIFHLSLAVLYLLRQVSISIA\ntranslation: LVIIFHLSLAVLYLLRQVSISIA\ntranslation: DIQNKCKEKVRLFFCSDLSIKK\ntranslation: DIQNKCKEKVRLFFCSDLSIKK\ntranslation: TFKTNAKKRSGYSSAVILASRK\ntranslation: TFKTNAKKRSGYSSAVILASRK\ntranslation: SASFPSDIGLHQSQHHPLCAEG\ntranslation: SASFPSDIGLHQSQHHPLCAEG\ntranslation: PHHFHLILDFIKVSIILCVLRE\ntranslation: PHHFHLILDFIKVSIILCVLRE\ntranslation: FPQHTEDDADFDEVQYQMEMMR\ntranslation: FPQHTEDDADFDEVQYQMEMMR\ntranslation: GLRDKLHVSITIRLHMYVCVYL\ntranslation: GLRDKLHVSITIRLHMYVCVYL\ntranslation: PHHPTKNIKHKLPPTSLTK\ntranslation: PHHPTKNIKHKLPPTSLTK\ntranslation: LITQLKTLNTNYHLPPSPK\ntranslation: LITQLKTLNTNYHLPPSPK\ntranslation: LHFLALICSSLHVWFLIFCCCCC\ntranslation: LHFLALICSSLHVWFLIFCCCCC\ntranslation: PFPRTYLQQSACVVFDFLLLLL\ntranslation: PFPRTYLQQSACVVFDFLLLLL\ntranslation: TTTTTKNQKPHMQTAADKCEEME\ntranslation: TTTTTKNQKPHMQTAADKCEEME\ntranslation: QQQQQKIKNHTCRLLQISARKW\ntranslation: QQQQQKIKNHTCRLLQISARKW\ntranslation: KPNIINFDTSLPTSHTYLGADM\ntranslation: KPNIINFDTSLPTSHTYLGADM\ntranslation: ISAPRYVCDVGRLVSKFMMFGF\ntranslation: ISAPRYVCDVGRLVSKFMMFGF\ntranslation: FIEKSSLSVHPSNLMEKKPSA\ntranslation: FIEKSSLSVHPSNLMEKKPSA\ntranslation: AEGFFSIRLLGWTLRLDFSMK\ntranslation: AEGFFSIRLLGWTLRLDFSMK\ntranslation: RQVDVVITCTGNKNVVTREHLDR\ntranslation: RQVDVVITCTGNKNVVTREHLDR\ntranslation: MRSKCSRVTTFLFPVQVITTSTC\ntranslation: MRSKCSRVTTFLFPVQVITTSTC\ntranslation: CDPSAPVSLHSYFLCKLLRHRLA\ntranslation: CDPSAPVSLHSYFLCKLLRHRLA\ntranslation: AIQVLPCHYILISCASYYDIDLP\ntranslation: AIQVLPCHYILISCASYYDIDLP\ntranslation: AYTAYNSWRICLFCGMHYSFWNI\ntranslation: AYTAYNSWRICLFCGMHYSFWNI\ntranslation: HIQLIIPGESACFAVCIILFGIF\ntranslation: HIQLIIPGESACFAVCIILFGIF\ntranslation: EYSKKNNAYRKTSRFSRNYKLYM\ntranslation: EYSKKNNAYRKTSRFSRNYKLYM\ntranslation: NIPKRIMHTAKQADSPGIISCIC\ntranslation: NIPKRIMHTAKQADSPGIISCIC\ntranslation: SLYRLLLFCPFLSCHFACLFL\ntranslation: SLYRLLLFCPFLSCHFACLFL\ntranslation: LCIGFYCFALFCLVILLAFF\ntranslation: LCIGFYCFALFCLVILLAFF\ntranslation: EKGKQNDKTEKGKTIKAYTE\ntranslation: EKGKQNDKTEKGKTIKAYTE\ntranslation: SLSVSPSVLCLSISVSMCLCP\ntranslation: SLSVSPSVLCLSISVSMCLCP\ntranslation: LFLSPPLFSVSQSLSPCVSAP\ntranslation: LFLSPPLFSVSQSLSPCVSAP\ntranslation: SFCLPLCSLSLNLCLHVSLPL\ntranslation: SFCLPLCSLSLNLCLHVSLPL\ntranslation: QGQRHMETEIERQRTEGETER\ntranslation: QGQRHMETEIERQRTEGETER\ntranslation: RGRDTWRQRLRDREQRGRQKE\ntranslation: RGRDTWRQRLRDREQRGRQKE\ntranslation: TPNWTNLSPYRRTNVSISNMK\ntranslation: TPNWTNLSPYRRTNVSISNMK\ntranslation: KIMPIFLVQIVFGTLDLLVLMM\ntranslation: KIMPIFLVQIVFGTLDLLVLMM\ntranslation: IIRTNKSSVPNTICTKKIGIIF\ntranslation: IIRTNKSSVPNTICTKKIGIIF\ntranslation: SSELINQVFQIQFALKRLALF\ntranslation: SSELINQVFQIQFALKRLALF\ntranslation: LYWGGKAETERGLSRKHIIEGL\ntranslation: LYWGGKAETERGLSRKHIIEGL\ntranslation: LQFKCSDSKGNRQTSRKEGSK\ntranslation: LQFKCSDSKGNRQTSRKEGSK\ntranslation: CSLSVRIPRETDKPHERKEAS\ntranslation: CSLSVRIPRETDKPHERKEAS\ntranslation: LLPSFREVCLFPLESEHLNC\ntranslation: LLPSFREVCLFPLESEHLNC\ntranslation: KYVLTNNHTNTSHYHSLCPHSF\ntranslation: KYVLTNNHTNTSHYHSLCPHSF\ntranslation: RSPRRLVVAIKMIKDTSIRDQ\ntranslation: RSPRRLVVAIKMIKDTSIRDQ\ntranslation: PSQKRDTVDSQAGGLNLSKTFI\ntranslation: PSQKRDTVDSQAGGLNLSKTFI\ntranslation: QKYKSNKLQITLSLFLINTATVL\ntranslation: QKYKSNKLQITLSLFLINTATVL\ntranslation: RTVAVLIRKRDRVICSLFDLYFC\ntranslation: RTVAVLIRKRDRVICSLFDLYFC\ntranslation: EGSVGKFKVELRFYLGQPAITRL\ntranslation: EGSVGKFKVELRFYLGQPAITRL\ntranslation: PQVQTFGVCAWLRSQWGEATI\ntranslation: PQVQTFGVCAWLRSQWGEATI\ntranslation: AGSDIWCMCLAEEPMGRSYHL\ntranslation: AGSDIWCMCLAEEPMGRSYHL\ntranslation: DGSFAPLAPQPSTYTKCLNLR\ntranslation: DGSFAPLAPQPSTYTKCLNLR\ntranslation: AMFTFITNSLTETAIYSFLQQE\ntranslation: AMFTFITNSLTETAIYSFLQQE\ntranslation: SDLVSPALGPFLLRPRSLSPV\ntranslation: SDLVSPALGPFLLRPRSLSPV\ntranslation: QLGTKTWASGGRVPAQGTQGQ\ntranslation: QLGTKTWASGGRVPAQGTQGQ\ntranslation: NWGQRPGPQEEGSQRRGHKVR\ntranslation: NWGQRPGPQEEGSQRRGHKVR\ntranslation: TGDKDLGLRRKGPSAGDTRSE\ntranslation: TGDKDLGLRRKGPSAGDTRSE\ntranslation: VIEEYAKILRSWVWFNPPQLPAM\ntranslation: VIEEYAKILRSWVWFNPPQLPAM\ntranslation: HSRQLRWIKPNPATQNLSILLNY\ntranslation: HSRQLRWIKPNPATQNLSILLNY\ntranslation: FIFIILKRLCPFSNEKTKAYRGY\ntranslation: FIFIILKRLCPFSNEKTKAYRGY\ntranslation: ITFRGLIFHGELFQHIACCSDP\ntranslation: ITFRGLIFHGELFQHIACCSDP\ntranslation: AMLQSTLSLWSSWFFYLHCHLYP\ntranslation: AMLQSTLSLWSSWFFYLHCHLYP\ntranslation: DVTEHIESVVIVVLLSSLSPVS\ntranslation: DVTEHIESVVIVVLLSSLSPVS\ntranslation: QDTGDSEDRRTTMTTDSMCSVTS\ntranslation: QDTGDSEDRRTTMTTDSMCSVTS\ntranslation: ATITAQLLQRRRSVCASCSQSQR\ntranslation: ATITAQLLQRRRSVCASCSQSQR\ntranslation: PQSLLSFSKDVVLFALHVLKVKG\ntranslation: PQSLLSFSKDVVLFALHVLKVKG\ntranslation: HNHCSASPKTSFCLRFMFSKSK\ntranslation: HNHCSASPKTSFCLRFMFSKSK\ntranslation: RHSIVHRLFSTSINQ\ntranslation: RHSIVHRLFSTSINQ\ntranslation: AKLATSMFLAGESLVCIA\ntranslation: AKLATSMFLAGESLVCIA\ntranslation: MAMHTRLSPARNILVANF\ntranslation: MAMHTRLSPARNILVANF\ntranslation: WQCIPGFHLLETYLLPTL\ntranslation: WQCIPGFHLLETYLLPTL\ntranslation: HSVTCRTMQPWM\ntranslation: HSVTCRTMQPWM\ntranslation: IPSPAGRCSPG\ntranslation: IPSPAGRCSPG\ntranslation: FRHLQDDAALD\ntranslation: FRHLQDDAALD\ntranslation: HPGLHRPAGDGM\ntranslation: HPGLHRPAGDGM\ntranslation: IQGCIVLQVTE\ntranslation: IQGCIVLQVTE\ntranslation: LECSGMIIAHCSLNLPGSSDPP\ntranslation: LECSGMIIAHCSLNLPGSSDPP\ntranslation: GVQWHDHSSLQPQPPRLKRSSP\ntranslation: GVQWHDHSSLQPQPPRLKRSSP\ntranslation: DDLEGIMLNSHIDLMEHFKGKN\ntranslation: DDLEGIMLNSHIDLMEHFKGKN\ntranslation: ASSVDTSYLGGVERV\ntranslation: ASSVDTSYLGGVERV\ntranslation: KLLSIINHPFDFFRCQAILVIGD\ntranslation: KLLSIINHPFDFFRCQAILVIGD\ntranslation: NFSASLTILSISSGVRRFWSLVI\ntranslation: NFSASLTILSISSGVRRFWSLVI\ntranslation: ITNDQNRLTPEEIERMVNDAEKF\ntranslation: ITNDQNRLTPEEIERMVNDAEKF\ntranslation: HLVPCFKSSWMFCMSWGLKLTQ\ntranslation: HLVPCFKSSWMFCMSWGLKLTQ\ntranslation: WVSFSPHDMQNIHEDLKHGTRW\ntranslation: WVSFSPHDMQNIHEDLKHGTRW\ntranslation: FNSRQNIVQYEKHYMRQRQFSKI\ntranslation: FNSRQNIVQYEKHYMRQRQFSKI\ntranslation: STYNSLTNCWELSQKHLEHWW\ntranslation: STYNSLTNCWELSQKHLEHWW\ntranslation: PPVFQMLLTKLPTISQRIVSA\ntranslation: PPVFQMLLTKLPTISQRIVSA\ntranslation: FQNLFKNCCYYCLFQNSILVFSS\ntranslation: FQNLFKNCCYYCLFQNSILVFSS\ntranslation: SKTSLRIVAIIASSRTPFLCSP\ntranslation: SKTSLRIVAIIASSRTPFLCSP\ntranslation: RGEHKNGVLEEAIIATILKEVLE\ntranslation: RGEHKNGVLEEAIIATILKEVLE\ntranslation: VNENQIVKLSEKLINYRLLEQL\ntranslation: VNENQIVKLSEKLINYRLLEQL\ntranslation: PLLPSPYYISILGLVLISGLIL\ntranslation: PLLPSPYYISILGLVLISGLIL\ntranslation: FITISLLHFYSRVSVNLRSYS\ntranslation: FITISLLHFYSRVSVNLRSYS\ntranslation: KNFAALLEEHDIFVPEDDDDDD\ntranslation: KNFAALLEEHDIFVPEDDDDDD\ntranslation: RTLLLYLRNMTFLFQRMMMMMT\ntranslation: RTLLLYLRNMTFLFQRMMMMMT\ntranslation: SHHHHHPLEQKCHVPQVEQQSS\ntranslation: SHHHHHPLEQKCHVPQVEQQSS\ntranslation: SSSSSSSGTKMSCSSSRAAKFL\ntranslation: SSSSSSSGTKMSCSSSRAAKFL\ntranslation: PIPNAPLRLIRPNHSSPTSPIS\ntranslation: PIPNAPLRLIRPNHSSPTSPIS\ntranslation: RDRRSRTAVIRTDQTKRGVWYW\ntranslation: RDRRSRTAVIRTDQTKRGVWYW\ntranslation: GWLHTVVNIQMIHWANRNHDIRK\ntranslation: GWLHTVVNIQMIHWANRNHDIRK\ntranslation: KKKQVLKFTLDCTHPVEDGIMD\ntranslation: KKKQVLKFTLDCTHPVEDGIMD\ntranslation: IHDSIFYRVGAIKSELQNLLLF\ntranslation: IHDSIFYRVGAIKSELQNLLLF\ntranslation: SMIPSSTGWVQSRVNFRTCFFF\ntranslation: SMIPSSTGWVQSRVNFRTCFFF\ntranslation: VDPAESLNYSTSTLISLYFLP\ntranslation: VDPAESLNYSTSTLISLYFLP\ntranslation: QEENTEKLKLKYYNLMIQLDQ\ntranslation: QEENTEKLKLKYYNLMIQLDQ\ntranslation: SDVTKKSNSFRASFVSRSSPL\ntranslation: SDVTKKSNSFRASFVSRSSPL\ntranslation: DSGEDRDTKEALKEFDFLVTSE\ntranslation: DSGEDRDTKEALKEFDFLVTSE\ntranslation: SELSALERQKDEKITQQEEKYEA\ntranslation: SELSALERQKDEKITQQEEKYEA\ntranslation: IASYFSSCWVIFSSFCLSRALNS\ntranslation: IASYFSSCWVIFSSFCLSRALNS\ntranslation: QLYKMTKIDCRI\ntranslation: QLYKMTKIDCRI\ntranslation: DPTIYFSHFVQL\ntranslation: DPTIYFSHFVQL\ntranslation: ILQSILVILYS\ntranslation: ILQSILVILYS\ntranslation: SVLPPDYDSNPTQL\ntranslation: SVLPPDYDSNPTQL\ntranslation: QFCLLTMTVIPPS\ntranslation: QFCLLTMTVIPPS\ntranslation: AGWDYCHSQEAEL\ntranslation: AGWDYCHSQEAEL\ntranslation: SRFWSSMWSSLLSTLLTLKVKQ\ntranslation: SRFWSSMWSSLLSTLLTLKVKQ\ntranslation: HCFTLRVNNVERREDHIDDQNLL\ntranslation: HCFTLRVNNVERREDHIDDQNLL\ntranslation: LRSRAVRSPRKVTARAVSF\ntranslation: LRSRAVRSPRKVTARAVSF\ntranslation: KTHSPRCHFPRTSNSPRPQ\ntranslation: KTHSPRCHFPRTSNSPRPQ\ntranslation: KLTALAVTFLGLLTALDLN\ntranslation: KLTALAVTFLGLLTALDLN\ntranslation: PMLSERSQNTTVRITLFHLYETLE\ntranslation: PMLSERSQNTTVRITLFHLYETLE\ntranslation: EFRTSISVTPSLMCFCYCFEEGSS\ntranslation: EFRTSISVTPSLMCFCYCFEEGSS\ntranslation: FHHLLHLTEINLLEFLPVTLSFL\ntranslation: FHHLLHLTEINLLEFLPVTLSFL\ntranslation: QFYWVSILPSYKPQSTSSLPSPF\ntranslation: QFYWVSILPSYKPQSTSSLPSPF\ntranslation: NFTGSLFYPPTSLRVLRVSLHHF\ntranslation: NFTGSLFYPPTSLRVLRVSLHHF\ntranslation: ILLGLYFTLLQASEYFESPFTI\ntranslation: ILLGLYFTLLQASEYFESPFTI\ntranslation: PVSKAFLDRWCVVALVCAFSCYI\ntranslation: PVSKAFLDRWCVVALVCAFSCYI\ntranslation: PYRRPFWTGGVWWPWYVLSRVTS\ntranslation: PYRRPFWTGGVWWPWYVLSRVTS\ntranslation: RIEGLFGQVVCGGLGMCFLVLH\ntranslation: RIEGLFGQVVCGGLGMCFLVLH\ntranslation: RCNTRKHIPRPPHTTCPKRPSIR\ntranslation: RCNTRKHIPRPPHTTCPKRPSIR\ntranslation: DVTRESTYQGHHTPPVQKGLRYG\ntranslation: DVTRESTYQGHHTPPVQKGLRYG\ntranslation: EPFYTNTYSDFSVTLKFIFLSY\ntranslation: EPFYTNTYSDFSVTLKFIFLSY\ntranslation: HSALENQNERKSVRFIHCPHNP\ntranslation: HSALENQNERKSVRFIHCPHNP\ntranslation: LEFPFISCQLASQFLAYVFSWCV\ntranslation: LEFPFISCQLASQFLAYVFSWCV\ntranslation: WNFPSYLASWPPSFLLMFSPGVF\ntranslation: WNFPSYLASWPPSFLLMFSPGVF\ntranslation: GISLHILPVGLPVSCLCFLLVC\ntranslation: GISLHILPVGLPVSCLCFLLVC\ntranslation: NTPGENISKKLGGQLARYEGKFQ\ntranslation: NTPGENISKKLGGQLARYEGKFQ\ntranslation: FISFALLKQEAGRPASPLHI\ntranslation: FISFALLKQEAGRPASPLHI\ntranslation: SYLLHSSSKKPADLQALCTS\ntranslation: SYLLHSSSKKPADLQALCTS\ntranslation: HIFCTPQARSRQTCKPFAHL\ntranslation: HIFCTPQARSRQTCKPFAHL\ntranslation: KMCKGLAGLPASCLRSAKDM\ntranslation: KMCKGLAGLPASCLRSAKDM\ntranslation: DVQRACRSAGFLLEECKRYE\ntranslation: DVQRACRSAGFLLEECKRYE\ntranslation: HEQLSTQSLIELFESFKSFEGLF\ntranslation: HEQLSTQSLIELFESFKSFEGLF\ntranslation: LRCCKIITLSTMNTGTLLKTLL\ntranslation: LRCCKIITLSTMNTGTLLKTLL\ntranslation: KSVFSKVPVFIVDKVIILQHL\ntranslation: KSVFSKVPVFIVDKVIILQHL\ntranslation: NSEGLYRVSGFSDLIEDVKMAF\ntranslation: NSEGLYRVSGFSDLIEDVKMAF\ntranslation: KAILTSSIRSLNPDTRYSPSEL\ntranslation: KAILTSSIRSLNPDTRYSPSEL\ntranslation: RILEVRNVFTMTAKEGRKKSIR\ntranslation: RILEVRNVFTMTAKEGRKKSIR\ntranslation: TDRFLSSLFRSHSENVSYLKYP\ntranslation: TDRFLSSLFRSHSENVSYLKYP\ntranslation: RIDFFLPSFAVIVKTFLTSSIL\ntranslation: RIDFFLPSFAVIVKTFLTSSIL\ntranslation: QEKNIQKVIFGLAMSKSQLYKDL\ntranslation: QEKNIQKVIFGLAMSKSQLYKDL\ntranslation: KKKIFRKLFLALLCQKVNCTKIY\ntranslation: KKKIFRKLFLALLCQKVNCTKIY\ntranslation: RKKYSESYFWPCYVKKSTVQRST\ntranslation: RKKYSESYFWPCYVKKSTVQRST\ntranslation: FKQLSSLFRCHSLLLNSHSELPS\ntranslation: FKQLSSLFRCHSLLLNSHSELPS\ntranslation: DGSSLWEFKSKLWHRNKLDNCLN\ntranslation: DGSSLWEFKSKLWHRNKLDNCLN\ntranslation: HDKEIFFHYSEFSGDVDSLEL\ntranslation: HDKEIFFHYSEFSGDVDSLEL\ntranslation: MIRKSFSITVSSLVMLIAWNW\ntranslation: MIRKSFSITVSSLVMLIAWNW\ntranslation: PVPGYQHHQRTHCNGKRFPYH\ntranslation: PVPGYQHHQRTHCNGKRFPYH\ntranslation: QFQAINITRELTVMEKDFLIM\ntranslation: QFQAINITRELTVMEKDFLIM\ntranslation: NHIPNLSLTKRKPSPHSLNLIHH\ntranslation: NHIPNLSLTKRKPSPHSLNLIHH\ntranslation: IIYQISPSLNVSLLLTLSILSI\ntranslation: IIYQISPSLNVSLLLTLSILSI\ntranslation: MMDKIERVRRRLTFSEGEIWYMI\ntranslation: MMDKIERVRRRLTFSEGEIWYMI\ntranslation: CKIVNWYLGDFCSLLKKIAFVLK\ntranslation: CKIVNWYLGDFCSLLKKIAFVLK\ntranslation: QNSELVLGRLLFFVEKNCLCFE\ntranslation: QNSELVLGRLLFFVEKNCLCFE\ntranslation: ISKQRQFFSTKNRSLPSTNSLFC\ntranslation: ISKQRQFFSTKNRSLPSTNSLFC\ntranslation: FQNKGNFFQQRTEVSQVPIHYFA\ntranslation: FQNKGNFFQQRTEVSQVPIHYFA\ntranslation: FKTKAIFFNKEQKSPKYQFTIL\ntranslation: FKTKAIFFNKEQKSPKYQFTIL\ntranslation: HIVYCFHYIKHFLFSDVAIFIYV\ntranslation: HIVYCFHYIKHFLFSDVAIFIYV\ntranslation: DINKDGYITKEEMLDIMKAIYDM\ntranslation: DINKDGYITKEEMLDIMKAIYDM\ntranslation: LNMNCHFRFLTFVSKEKLRKSS\ntranslation: LNMNCHFRFLTFVSKEKLRKSS\ntranslation: LIEEYAMSDFRSVCRR\ntranslation: LIEEYAMSDFRSVCRR\ntranslation: PTTNRPKIAHCILFNQ\ntranslation: PTTNRPKIAHCILFNQ\ntranslation: LRQTDLKSLIAYSSI\ntranslation: LRQTDLKSLIAYSSI\ntranslation: SPVPGTCYYQHHALPISSTD\ntranslation: SPVPGTCYYQHHALPISSTD\ntranslation: LLCQELATTSTMPYQYPALT\ntranslation: LLCQELATTSTMPYQYPALT\ntranslation: GQCWILVGHGAGSSKFLAQE\ntranslation: GQCWILVGHGAGSSKFLAQE\ntranslation: CRATFQKLGEVPSLTVSTQPGP\ntranslation: CRATFQKLGEVPSLTVSTQPGP\ntranslation: QSYISEIGRSAKSYCEHTART\ntranslation: QSYISEIGRSAKSYCEHTART\ntranslation: GSWLCAHSKTWHFSQFLKCSSA\ntranslation: GSWLCAHSKTWHFSQFLKCSSA\ntranslation: NTHVVKRMIVRHTAFCFRNRILP\ntranslation: NTHVVKRMIVRHTAFCFRNRILP\ntranslation: GSILFLKQNAVCLTIILLTTCV\ntranslation: GSILFLKQNAVCLTIILLTTCV\ntranslation: LTRVERRAHFKILPCLLNLSP\ntranslation: LTRVERRAHFKILPCLLNLSP\ntranslation: LDHSEGFSVLQEELCTAQHKQK\ntranslation: LDHSEGFSVLQEELCTAQHKQK\ntranslation: WIIVKGSRYFKRSYALHSTSRK\ntranslation: WIIVKGSRYFKRSYALHSTSRK\ntranslation: LSACAVQCITPLEVPRTLHYDP\ntranslation: LSACAVQCITPLEVPRTLHYDP\ntranslation: SCDYKDTCIRMFIAALFTIAK\ntranslation: SCDYKDTCIRMFIAALFTIAK\ntranslation: TFCHATNTRLNYMHKIPNPGS\ntranslation: TFCHATNTRLNYMHKIPNPGS\ntranslation: FNPTCPCYLIRADDPSP\ntranslation: FNPTCPCYLIRADDPSP\ntranslation: KNQQGEKLTTSQLKLVKQNSLYS\ntranslation: KNQQGEKLTTSQLKLVKQNSLYS\ntranslation: KPTRRKIDYITTEISKTEFSLF\ntranslation: KPTRRKIDYITTEISKTEFSLF\ntranslation: AFSSQTEVWAEKASRGSHSLYHM\ntranslation: AFSSQTEVWAEKASRGSHSLYHM\ntranslation: PFPPRQKSGQRKPLEVPIPSIT\ntranslation: PFPPRQKSGQRKPLEVPIPSIT\ntranslation: LFLPDRSLGRESLSRFPFPLSH\ntranslation: LFLPDRSLGRESLSRFPFPLSH\ntranslation: HVIEGMGTSRGFLCPDFCLGGKG\ntranslation: HVIEGMGTSRGFLCPDFCLGGKG\ntranslation: CDRGNGNLERLSLPRLLSGRKR\ntranslation: CDRGNGNLERLSLPRLLSGRKR\ntranslation: SESTQQTRILHSSAILSVVSIN\ntranslation: SESTQQTRILHSSAILSVVSIN\ntranslation: LNPLNKLEFFTVVLFSVLLAST\ntranslation: LNPLNKLEFFTVVLFSVLLAST\ntranslation: VDANNTENSTTVKNSSLLSGFR\ntranslation: VDANNTENSTTVKNSSLLSGFR\ntranslation: HNSILIFFF\ntranslation: HNSILIFFF\ntranslation: QLYSNFFF\ntranslation: QLYSNFFF\ntranslation: KKKIRIEL\ntranslation: KKKIRIEL\ntranslation: LLSTLVSKPGFKDWRRASGT\ntranslation: LLSTLVSKPGFKDWRRASGT\ntranslation: VPEALLQSLKPGLLTKVLSN\ntranslation: VPEALLQSLKPGLLTKVLSN\ntranslation: CNFLEEEIRSS\ntranslation: CNFLEEEIRSS\ntranslation: ATFLKKKSEA\ntranslation: ATFLKKKSEA\ntranslation: GASDFFFKKVA\ntranslation: GASDFFFKKVA\ntranslation: ELLISSSRKL\ntranslation: ELLISSSRKL\ntranslation: CCCFVCVCVVVVVVVVAVVLCVC\ntranslation: CCCFVCVCVVVVVVVVAVVLCVC\ntranslation: AVVLCVCVLLLLLLLLLLFCVCV\ntranslation: AVVLCVCVLLLLLLLLLLFCVCV\ntranslation: LLFCVCVCCCCCCCCCCCFVCV\ntranslation: LLFCVCVCCCCCCCCCCCFVCV\ntranslation: HTHTKQQQQQQQQQQHTHTQNNS\ntranslation: HTHTKQQQQQQQQQQHTHTQNNS\ntranslation: THTQNNSNNNNNNNNTHTHKTTA\ntranslation: THTQNNSNNNNNNNNTHTHKTTA\ntranslation: HTHKTTATTTTTTTTHTHTKQQ\ntranslation: HTHKTTATTTTTTTTHTHTKQQ\ntranslation: GTCDCHWKLGRGLGLNSQPVESD\ntranslation: GTCDCHWKLGRGLGLNSQPVESD\ntranslation: EVSPLLGSRWTTRSAWPTPSS\ntranslation: EVSPLLGSRWTTRSAWPTPSS\ntranslation: RYHLFWAQDGQQEAPGLRHH\ntranslation: RYHLFWAQDGQQEAPGLRHH\ntranslation: GITSSGLKMDNKKRLAYAII\ntranslation: GITSSGLKMDNKKRLAYAII\ntranslation: DDGVGQALLVVHLEPRRGDT\ntranslation: DDGVGQALLVVHLEPRRGDT\ntranslation: RNDPLIFTTIHVLLVC\ntranslation: RNDPLIFTTIHVLLVC\ntranslation: AYEENMYGSKYQWIIP\ntranslation: AYEENMYGSKYQWIIP\ntranslation: HTRRTCMVVNISGSFR\ntranslation: HTRRTCMVVNISGSFR\ntranslation: LGWVFSWVFVLFISNTEKVLWAL\ntranslation: LGWVFSWVFVLFISNTEKVLWAL\ntranslation: WVGFFHGFLFCLFRILKRSFGL\ntranslation: WVGFFHGFLFCLFRILKRSFGL\ntranslation: RAQRTFSVFEINKTKTHEKTQP\ntranslation: RAQRTFSVFEINKTKTHEKTQP\ntranslation: VSVMVSVCKFSYCSCAPRCTHLS\ntranslation: VSVMVSVCKFSYCSCAPRCTHLS\ntranslation: SQSWLVFASLVTVAAHPGVHTF\ntranslation: SQSWLVFASLVTVAAHPGVHTF\ntranslation: FEVIAVPRRWRLSKFWRTTFR\ntranslation: FEVIAVPRRWRLSKFWRTTFR\ntranslation: PKGRPPKLTQPPPPRNSNDLK\ntranslation: PKGRPPKLTQPPPPRNSNDLK\ntranslation: LKVVLQNLLNLHLLGTAMTSK\ntranslation: LKVVLQNLLNLHLLGTAMTSK\ntranslation: KIGETLRSRQSVEAQYILFAL\ntranslation: KIGETLRSRQSVEAQYILFAL\ntranslation: KQIRYTEPQQIACSSESLLFL\ntranslation: KQIRYTEPQQIACSSESLLFL\ntranslation: LAAAPDRGLASLPFLLVWSSL\ntranslation: LAAAPDRGLASLPFLLVWSSL\ntranslation: SQQHQTEDSLLFPFCLFGVLW\ntranslation: SQQHQTEDSLLFPFCLFGVLW\ntranslation: RSSTRPRTRFSSLSACLEFF\ntranslation: RSSTRPRTRFSSLSACLEFF\ntranslation: PKNSKQAEREEKRVLGLVLLR\ntranslation: PKNSKQAEREEKRVLGLVLLR\ntranslation: QRTPNKQKGKRSESSVWCCCE\ntranslation: QRTPNKQKGKRSESSVWCCCE\ntranslation: KELQTSRKGREASPRSGAAA\ntranslation: KELQTSRKGREASPRSGAAA\ntranslation: CVLYDVNNEWMLLMPLVLFWTSP\ntranslation: CVLYDVNNEWMLLMPLVLFWTSP\ntranslation: GEVQKSTNGISNIHSLFTSYNTH\ntranslation: GEVQKSTNGISNIHSLFTSYNTH\ntranslation: VKSKKALTASVTSIHCLHRTIHT\ntranslation: VKSKKALTASVTSIHCLHRTIHT\ntranslation: GKLKSSLKHLVF\ntranslation: GKLKSSLKHLVF\ntranslation: KTRCFKLDLSF\ntranslation: KTRCFKLDLSF\ntranslation: TILGLMCCECCFFFSFVKIFK\ntranslation: TILGLMCCECCFFFSFVKIFK\ntranslation: FKYFNKGKKKATFTAHQAQNS\ntranslation: FKYFNKGKKKATFTAHQAQNS\ntranslation: LNILTKEKKKQHSQHIKPKIV\ntranslation: LNILTKEKKKQHSQHIKPKIV\ntranslation: QCITIFSFHLVPILSSFLSIL\ntranslation: QCITIFSFHLVPILSSFLSIL\ntranslation: VYNHLLFSSCSNSLLFPFHS\ntranslation: VYNHLLFSSCSNSLLFPFHS\ntranslation: QNGKERGENWNKMKREDGYTL\ntranslation: QNGKERGENWNKMKREDGYTL\ntranslation: EWKGKRRELEQDEKRRWLYT\ntranslation: EWKGKRRELEQDEKRRWLYT\ntranslation: AEKKRYLQTVAVIVQVLHQIS\ntranslation: AEKKRYLQTVAVIVQVLHQIS\ntranslation: GEEKIFTNSCRDCASVAPNFK\ntranslation: GEEKIFTNSCRDCASVAPNFK\ntranslation: FEIWCNTCTITATVCKYLFFS\ntranslation: FEIWCNTCTITATVCKYLFFS\ntranslation: LKFGATLAQSRQLFVNIFSSP\ntranslation: LKFGATLAQSRQLFVNIFSSP\ntranslation: SSNFNNWDSTTEELLTSVCTFF\ntranslation: SSNFNNWDSTTEELLTSVCTFF\ntranslation: KECTHRSQELFCGGIPIVKVTA\ntranslation: KECTHRSQELFCGGIPIVKVTA\ntranslation: KNVHTEVKSSSVVESQLLKLLL\ntranslation: KNVHTEVKSSSVVESQLLKLLL\ntranslation: KQVSPGAAGLLSSL\ntranslation: KQVSPGAAGLLSSL\ntranslation: SKSHLVLQDFSAPC\ntranslation: SKSHLVLQDFSAPC\ntranslation: GKELRSPAAPGETCL\ntranslation: GKELRSPAAPGETCL\ntranslation: GPLSKNWEAAKHNVSGCECENQ\ntranslation: GPLSKNWEAAKHNVSGCECENQ\ntranslation: SDFRIHSLKHYALQLPSSYSVDP\ntranslation: SDFRIHSLKHYALQLPSSYSVDP\ntranslation: SIRDQVRPLVLCMVIICFEVSL\ntranslation: SIRDQVRPLVLCMVIICFEVSL\ntranslation: CDPAWATKQDPVSNKKVSTKL\ntranslation: CDPAWATKQDPVSNKKVSTKL\ntranslation: QFRTHFFVRDRVLLCCPGWIT\ntranslation: QFRTHFFVRDRVLLCCPGWIT\ntranslation: SFVLTFLLETGSCFVAQAGSQ\ntranslation: SFVLTFLLETGSCFVAQAGSQ\ntranslation: VMGMETYHISNARVSGRKFFH\ntranslation: VMGMETYHISNARVSGRKFFH\ntranslation: MKKLPTTHPSITYMICLHTHY\ntranslation: MKKLPTTHPSITYMICLHTHY\ntranslation: SRVLTHSSINPKVQVQSLIQDK\ntranslation: SRVLTHSSINPKVQVQSLIQDK\ntranslation: RFKLEWLGNCSGLNDETYGYKE\ntranslation: RFKLEWLGNCSGLNDETYGYKE\ntranslation: HQFEKSPYQKNMNSAKIEAMLPE\ntranslation: HQFEKSPYQKNMNSAKIEAMLPE\ntranslation: INCNKSVLISSLALILRLCYSKF\ntranslation: INCNKSVLISSLALILRLCYSKF\ntranslation: PILITMQSNGFKAYIFHIIFSA\ntranslation: PILITMQSNGFKAYIFHIIFSA\ntranslation: NCMFFFHLDNLGSVESYLLTCY\ntranslation: NCMFFFHLDNLGSVESYLLTCY\ntranslation: FQIYLYLALRQQIEDSEFKW\ntranslation: FQIYLYLALRQQIEDSEFKW\ntranslation: NEELFSGWFSYQLFTQQSSKP\ntranslation: NEELFSGWFSYQLFTQQSSKP\ntranslation: TKSFFLDGFLISFSHNSQVNL\ntranslation: TKSFFLDGFLISFSHNSQVNL\ntranslation: LHRLSHFYHNNPTVKLSLRT\ntranslation: LHRLSHFYHNNPTVKLSLRT\ntranslation: SSQRQLHSRIIVIKMTQAMQ\ntranslation: SSQRQLHSRIIVIKMTQAMQ\ntranslation: RLFSPDIDHITECTSHQLFS\ntranslation: RLFSPDIDHITECTSHQLFS\ntranslation: HENAMLQTENNNLRVRIKAMQE\ntranslation: HENAMLQTENNNLRVRIKAMQE\ntranslation: VWMPFISCLIAVSRTSSTLLNY\ntranslation: VWMPFISCLIAVSRTSSTLLNY\ntranslation: VIQQSTRSPRYSNQTRNKGHPN\ntranslation: VIQQSTRSPRYSNQTRNKGHPN\ntranslation: NNDISSLPYSLGNLHLKFLALEGNP\ntranslation: NNDISSLPYSLGNLHLKFLALEGNP\ntranslation: LPCTIPMRHNYNKLHLPTTNR\ntranslation: LPCTIPMRHNYNKLHLPTTNR\ntranslation: SVCRRQMELVIIMPHRDSTRK\ntranslation: SVCRRQMELVIIMPHRDSTRK\ntranslation: FKITALHLVLQICFKARQVCIKP\ntranslation: FKITALHLVLQICFKARQVCIKP\ntranslation: SKSLPSILSFRSASKRARSASSL\ntranslation: SKSLPSILSFRSASKRARSASSL\ntranslation: QNHCPPSCPSDLLQSAPGLHQAF\ntranslation: QNHCPPSCPSDLLQSAPGLHQAF\ntranslation: RLDADLARFEADLKDKMEGSDFE\ntranslation: RLDADLARFEADLKDKMEGSDFE\ntranslation: RLCCCASYCCLMGLYLKCIVLMF\ntranslation: RLCCCASYCCLMGLYLKCIVLMF\ntranslation: IVLLCFILLPYGIILEMHCADV\ntranslation: IVLLCFILLPYGIILEMHCADV\ntranslation: NISTMHFKYNPIRQQYEAQQHN\ntranslation: NISTMHFKYNPIRQQYEAQQHN\ntranslation: VFVPQPPQCAHHKIQTQKTQQFF\ntranslation: VFVPQPPQCAHHKIQTQKTQQFF\ntranslation: SLFRNHHNALIIKYRHRKHSNS\ntranslation: SLFRNHHNALIIKYRHRKHSNS\ntranslation: EELLCFLCLYFMMSALWWLRNKD\ntranslation: EELLCFLCLYFMMSALWWLRNKD\ntranslation: RIAVFSVSVFYDERIVVVAEQR\ntranslation: RIAVFSVSVFYDERIVVVAEQR\ntranslation: ECVYVCVWNDSVCVWNACVC"
      },
      {
       "name": "stderr",
       "output_type": "stream",
       "text": "501it [00:00, 1190.51it/s]"
      },
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "\ntranslation: ECVYVCVWNDSVCVWNACVC\ntranslation: NVSMCVYGMILCVYGMRVSV\ntranslation: NVSMCVYGMILCVYGMRVSV\ntranslation: HRHTHSIHTQNHSIHTHRHI\ntranslation: HRHTHSIHTQNHSIHTHRHI\ntranslation: TDTRIPYTHRIIPYTHIDTF\ntranslation: TDTRIPYTHRIIPYTHIDTF\ntranslation: EDGNHQFWQCFTGRSKARNFH\ntranslation: EDGNHQFWQCFTGRSKARNFH\ntranslation: KTETISFGSVSPGGVKLEIST\ntranslation: KTETISFGSVSPGGVKLEIST\ntranslation: VEISSFTPPGETLPKLMVSVF\ntranslation: VEISSFTPPGETLPKLMVSVF\ntranslation: GDLYCDHAELIAVCCATTTSCS\ntranslation: GDLYCDHAELIAVCCATTTSCS\ntranslation: NMRWWWRSRLRSILHGHSRDP\ntranslation: NMRWWWRSRLRSILHGHSRDP\ntranslation: LISVYLASSSSSSSSSVFHFL\ntranslation: LISVYLASSSSSSSSSVFHFL\ntranslation: DFCLLSIIFIIIIIICISFSL\ntranslation: DFCLLSIIFIIIIIICISFSL\ntranslation: KENEIQMMMMMMKMMLSKQKS\ntranslation: KENEIQMMMMMMKMMLSKQKS\ntranslation: HLQISHQQPTNHHPTMTNQTNL\ntranslation: HLQISHQQPTNHHPTMTNQTNL\ntranslation: EVSLISHCWVVISRLLMRYLEV\ntranslation: EVSLISHCWVVISRLLMRYLEV\ntranslation: LALKGEWEFRGQKIDPREYVVPT\ntranslation: LALKGEWEFRGQKIDPREYVVPT\ntranslation: VGTTYSLGSIFCPRNSHSPFKAK\ntranslation: VGTTYSLGSIFCPRNSHSPFKAK\ntranslation: LEPHTLWGLFFAHGIPIHPLRL\ntranslation: LEPHTLWGLFFAHGIPIHPLRL\ntranslation: KMLVFTDQEGNEVRRQTF\ntranslation: KMLVFTDQEGNEVRRQTF\ntranslation: NQANRFFSLIDLFSFDQRLKF\ntranslation: NQANRFFSLIDLFSFDQRLKF\ntranslation: KLYLKSKWTDICKVLINKCSTNRY\ntranslation: KLYLKSKWTDICKVLINKCSTNRY\ntranslation: YLFVEHLLIKTLHMSVHFDLRYNF\ntranslation: YLFVEHLLIKTLHMSVHFDLRYNF\ntranslation: ATTTEQQQSMLSIETLSRLSYT\ntranslation: ATTTEQQQSMLSIETLSRLSYT\ntranslation: SVSWYHMLLSLQTVKRFHFHLE\ntranslation: SVSWYHMLLSLQTVKRFHFHLE\ntranslation: QDENGIVSQSEVIRAYDTTKQ\ntranslation: QDENGIVSQSEVIRAYDTTKQ\ntranslation: KTLLPEDTQKLDNFLGEVRDKWD\ntranslation: KTLLPEDTQKLDNFLGEVRDKWD\ntranslation: GVFTYVGALFNGLTLLILALISL\ntranslation: GVFTYVGALFNGLTLLILALISL\ntranslation: VMCYLQILQNTHIHTHTHLPLCS\ntranslation: VMCYLQILQNTHIHTHTHLPLCS\ntranslation: SCAIYRFYKTHTYTHIHTCPYVL\ntranslation: SCAIYRFYKTHTYTHIHTCPYVL\ntranslation: HVLFTDFTKHTHTHTYTLALMF\ntranslation: HVLFTDFTKHTHTHTYTLALMF\ntranslation: VMSFYCALLSNHLCDFNKH\ntranslation: VMSFYCALLSNHLCDFNKH\ntranslation: ENECQSRKGHLRICSMKEKK\ntranslation: ENECQSRKGHLRICSMKEKK\ntranslation: HNCSSPVCQNHSGFQVNGQR\ntranslation: HNCSSPVCQNHSGFQVNGQR\ntranslation: QLQLSCMSKSFWFPGKWTK\ntranslation: QLQLSCMSKSFWFPGKWTK\ntranslation: PLSIYLETRMILTYRRAAVV\ntranslation: PLSIYLETRMILTYRRAAVV\ntranslation: FVHLPGNQNDFDIQESCSC\ntranslation: FVHLPGNQNDFDIQESCSC\ntranslation: WVWFNPPQLPAMMDKIERVR\ntranslation: WVWFNPPQLPAMMDKIERVR\ntranslation: PHSLNLIHHSRQLRWIKPNP\ntranslation: PHSLNLIHHSRQLRWIKPNP\ntranslation: QELENHDSFHPQIKPTQKYTTET\ntranslation: QELENHDSFHPQIKPTQKYTTET\ntranslation: KSLKIMIHFIHRSNQHRNIRQKL\ntranslation: KSLKIMIHFIHRSNQHRNIRQKL\ntranslation: EREEALKQHKTLSQELVNLRGEL\ntranslation: EREEALKQHKTLSQELVNLRGEL\ntranslation: FNVFSWVMRNSVRSMGVIMVGH\ntranslation: FNVFSWVMRNSVRSMGVIMVGH\ntranslation: MAHHNYPHTPYTIPHHPTKNIK\ntranslation: MAHHNYPHTPYTIPHHPTKNIK\ntranslation: WPTIITPILLTLFLITQLKTLN\ntranslation: WPTIITPILLTLFLITQLKTLN\ntranslation: RLFRRKLDPFPDDPHELSHSDI\ntranslation: RLFRRKLDPFPDDPHELSHSDI\ntranslation: DITVAQFMWIIRKRIQLPSEKA\ntranslation: DITVAQFMWIIRKRIQLPSEKA\ntranslation: ISLWLSSCGSSGKGSSFLLKR\ntranslation: ISLWLSSCGSSGKGSSFLLKR\ntranslation: HVQLLFTWNPSPLRPSKFSFEYL\ntranslation: HVQLLFTWNPSPLRPSKFSFEYL\ntranslation: MFNCCSHGTLLHFGLQSSRLNI\ntranslation: MFNCCSHGTLLHFGLQSSRLNI\ntranslation: EETHLSFKNKNYQKHKNKREM\ntranslation: EETHLSFKNKNYQKHKNKREM\ntranslation: EIQGKLEEKKPPEADMNIFED\ntranslation: EIQGKLEEKKPPEADMNIFED\ntranslation: VFKNIHVSLRRFLLFQLPLNL\ntranslation: VFKNIHVSLRRFLLFQLPLNL\ntranslation: LQKYSCQPQEVSSLPASPESQ\ntranslation: LQKYSCQPQEVSSLPASPESQ\ntranslation: LCVIILSCGFESAVLPRSKLEV\ntranslation: LCVIILSCGFESAVLPRSKLEV\ntranslation: SVSSFSVVGLKVLFFHGPNLKS\ntranslation: SVSSFSVVGLKVLFFHGPNLKS\ntranslation: VTSSLDRGRTALSNPQLRMMTQR\ntranslation: VTSSLDRGRTALSNPQLRMMTQR\ntranslation: DFKFGPWKNSTFKPTTENDDTE\ntranslation: DFKFGPWKNSTFKPTTENDDTE\ntranslation: MGVPLIGVIIRYSGKRGDFRDP\ntranslation: MGVPLIGVIIRYSGKRGDFRDP\ntranslation: QGSLKSPLFPEYLMITPIKGTPI\ntranslation: QGSLKSPLFPEYLMITPIKGTPI\ntranslation: LRVHHRTERVGENRPSPKEEQV\ntranslation: LRVHHRTERVGENRPSPKEEQV\ntranslation: TCSSLGEGRFSPTLSVLWWTLN\ntranslation: TCSSLGEGRFSPTLSVLWWTLN\ntranslation: ISHCLVNNINSTFSQHVEYCFL\ntranslation: ISHCLVNNINSTFSQHVEYCFL\ntranslation: RLSSRYIIPLAIWTHQLSSSCD\ntranslation: RLSSRYIIPLAIWTHQLSSSCD\ntranslation: SQLLLNWCVQIAKGMMYLEERR\ntranslation: SQLLLNWCVQIAKGMMYLEERR\ntranslation: LSQGTKTCSGKNIFTLSYMLS\ntranslation: LSQGTKTCSGKNIFTLSYMLS\ntranslation: FTRYQDMFWQKHFHFKLYAV\ntranslation: FTRYQDMFWQKHFHFKLYAV\ntranslation: IKDTSIRDQVRPLVLCMVIICF\ntranslation: IKDTSIRDQVRPLVLCMVIICF\ntranslation: SALNFGKQLLCSGSPTWSNCYF\ntranslation: SALNFGKQLLCSGSPTWSNCYF\ntranslation: NSNLTKLDFHYKAAVFQSSML\ntranslation: NSNLTKLDFHYKAAVFQSSML\ntranslation: VFCIVLVAQLRMRILSPVTF\ntranslation: VFCIVLVAQLRMRILSPVTF\ntranslation: NVTGERILILSCATNTMQK\ntranslation: NVTGERILILSCATNTMQK\ntranslation: HFKSTRTHNSYNRHQPTTPSI\ntranslation: HFKSTRTHNSYNRHQPTTPSI\ntranslation: TSSQLGLIIVTIGINQPHLAF\ntranslation: TSSQLGLIIVTIGINQPHLAF\ntranslation: SCTSASSSFTFSAIFTCSFHFSS\ntranslation: SCTSASSSFTFSAIFTCSFHFSS\ntranslation: PVLQLVQVLHFQQFSLVLSISAP\ntranslation: PVLQLVQVLHFQQFSLVLSISAP\ntranslation: MELKWKEQVKIAENVKLELAEVQ\ntranslation: MELKWKEQVKIAENVKLELAEVQ\ntranslation: TCPKTSSLTVKVRSKMSVMLLSF\ntranslation: TCPKTSSLTVKVRSKMSVMLLSF\ntranslation: KDNNITDILDLTFTVNEEVFGQV\ntranslation: KDNNITDILDLTFTVNEEVFGQV\ntranslation: QLSPKQCSGAKCSKEKSSSARRW\ntranslation: QLSPKQCSGAKCSKEKSSSARRW\ntranslation: SYPQNSVVGQNVQKKNHPVHVD\ntranslation: SYPQNSVVGQNVQKKNHPVHVD\ntranslation: HLRALDDFSFEHFAPLHCFGDS\ntranslation: HLRALDDFSFEHFAPLHCFGDS\ntranslation: IYVHWMIFLLNILPHYTVLGIA\ntranslation: IYVHWMIFLLNILPHYTVLGIA\ntranslation: LSPSCTFFSNTKLKTTSNTLFS\ntranslation: LSPSCTFFSNTKLKTTSNTLFS\ntranslation: EKSVLDVVFNFVLLKKVHDGE\ntranslation: EKSVLDVVFNFVLLKKVHDGE\ntranslation: ATRRGIDNKFQQPNPFLQRFQ\ntranslation: ATRRGIDNKFQQPNPFLQRFQ\ntranslation: LREEALITSFNSQILFYRDF\ntranslation: LREEALITSFNSQILFYRDF\ntranslation: RIYIFFVCVFVNYPTDIECTL\ntranslation: RIYIFFVCVFVNYPTDIECTL\ntranslation: EYIFFSCVFLLTTLQILNAP\ntranslation: EYIFFSCVFLLTTLQILNAP\ntranslation: QGAFNICRVVNKNTHEKNIYS\ntranslation: QGAFNICRVVNKNTHEKNIYS\ntranslation: TSLFRIAVKRSWQRL\ntranslation: TSLFRIAVKRSWQRL\ntranslation: HNLCQLLFTAILNKLV\ntranslation: HNLCQLLFTAILNKLV\ntranslation: QPLPTSLHCNSKQTC\ntranslation: QPLPTSLHCNSKQTC\ntranslation: VFLGYFELSFNCKWKVTTESMI\ntranslation: VFLGYFELSFNCKWKVTTESMI\ntranslation: HKRPRYIAQNHLKISDFKSMIP\ntranslation: HKRPRYIAQNHLKISDFKSMIP\ntranslation: NKPILKILSTYVFYNILTMSLEK\ntranslation: NKPILKILSTYVFYNILTMSLEK\ntranslation: ENALQSISNSYTENFTYFYIL\ntranslation: ENALQSISNSYTENFTYFYIL\ntranslation: KMPCKVSAIHTQKISRTSIYC\ntranslation: KMPCKVSAIHTQKISRTSIYC\ntranslation: KCLAKYQQFIHRKFHVLLYI\ntranslation: KCLAKYQQFIHRKFHVLLYI\ntranslation: HLTNYSKTTDHTSVYFLLVPSEL\ntranslation: HLTNYSKTTDHTSVYFLLVPSEL\ntranslation: LRSWVWFNPPQLPAMMDKIE\ntranslation: LRSWVWFNPPQLPAMMDKIE\ntranslation: LNLIHHSRQLRWIKPNPATQ\ntranslation: LNLIHHSRQLRWIKPNPATQ\ntranslation: LMKDNAEITGWRVTSDREVLHY\ntranslation: LMKDNAEITGWRVTSDREVLHY\ntranslation: FHDADEVLQAKHVASVWPIMH\ntranslation: FHDADEVLQAKHVASVWPIMH\ntranslation: VHDGPDRGYVFGLENLISIME\ntranslation: VHDGPDRGYVFGLENLISIME\ntranslation: CMMGQTEATCLAWRTSSASWN\ntranslation: CMMGQTEATCLAWRTSSASWN\ntranslation: KPVCRQFSLLRHGSSFTLAPLK\ntranslation: KPVCRQFSLLRHGSSFTLAPLK\ntranslation: GFKGASVKLLPCLSKENCLHTG\ntranslation: GFKGASVKLLPCLSKENCLHTG\ntranslation: LQRGQCKTASMSKQRKLPTYWF\ntranslation: LQRGQCKTASMSKQRKLPTYWF\ntranslation: SCSRFEQLLCTVQI\ntranslation: SCSRFEQLLCTVQI\ntranslation: LVPDLSNFYAQYK\ntranslation: LVPDLSNFYAQYK\ntranslation: ICTVHRSCSNREQ\ntranslation: ICTVHRSCSNREQ\ntranslation: FVLCIEVAQIGNK\ntranslation: FVLCIEVAQIGNK\ntranslation: RIHSEIVKQSVFKDFYELEPE\ntranslation: RIHSEIVKQSVFKDFYELEPE\ntranslation: LWLQFIKILKDRLFHDLRVDP\ntranslation: LWLQFIKILKDRLFHDLRVDP\ntranslation: VTYPEVLFTVWCSTYLHFSKRKQ\ntranslation: VTYPEVLFTVWCSTYLHFSKRKQ\ntranslation: SHIRKFCLQYGAALIYTSVKENK\ntranslation: SHIRKFCLQYGAALIYTSVKENK\ntranslation: FNFVWAVFYQTKLLAFEMIYM\ntranslation: FNFVWAVFYQTKLLAFEMIYM\ntranslation: CVCICVFQCFIRSFHLWE\ntranslation: CVCICVFQCFIRSFHLWE\ntranslation: VCVFVCFSVSSGPSISGS\ntranslation: VCVFVCFSVSSGPSISGS\ntranslation: CVYLCVSVFHQVLPSLGV\ntranslation: CVYLCVSVFHQVLPSLGV\ntranslation: LPEMEGPDETLKHTNTHT\ntranslation: LPEMEGPDETLKHTNTHT\ntranslation: QRVSQEEVKKWAESLENLISHEC\ntranslation: QRVSQEEVKKWAESLENLISHEC\ntranslation: RSVSFSCNLIYIFCFCCLCFWGH\ntranslation: RSVSFSCNLIYIFCFCCLCFWGH\ntranslation: EVFHFHVISSTSFAFVAYVFGV\ntranslation: EVFHFHVISSTSFAFVAYVFGV\ntranslation: NGKFKLPTKNEFIPTNFEILPL\ntranslation: NGKFKLPTKNEFIPTNFEILPL\ntranslation: SNGKISKFVGINSFFVGSLNFPF\ntranslation: SNGKISKFVGINSFFVGSLNFPF\ntranslation: HPCTSSSQHEVCWKSTG\ntranslation: HPCTSSSQHEVCWKSTG\ntranslation: THALHLVSMRYVGNLLD\ntranslation: THALHLVSMRYVGNLLD\ntranslation: HRGPDAFRFENVNGYTNCCFGFH\ntranslation: HRGPDAFRFENVNGYTNCCFGFH\ntranslation: TEVQMHSVLRMSMDTPTAALDF\ntranslation: TEVQMHSVLRMSMDTPTAALDF\ntranslation: VKSKAAVGVSIDILKTECIWTSV\ntranslation: VKSKAAVGVSIDILKTECIWTSV\ntranslation: NHVVFTLSGQTRRFFKQVETNYL\ntranslation: NHVVFTLSGQTRRFFKQVETNYL\ntranslation: CLFKFCLFTLLYMIVCTEGAP\ntranslation: CLFKFCLFTLLYMIVCTEGAP\ntranslation: GAPSVQTIIYRSVNKQNLKRQ\ntranslation: GAPSVQTIIYRSVNKQNLKRQ\ntranslation: CMCGVYVCYRLMPSWNCVNVLL\ntranslation: CMCGVYVCYRLMPSWNCVNVLL\ntranslation: FSGFPLCGETLCCCCFGCLCCS\ntranslation: FSGFPLCGETLCCCCFGCLCCS\ntranslation: FLVFLCAVKLCVAVALAVCAVV\ntranslation: FLVFLCAVKLCVAVALAVCAVV\ntranslation: PLQHRQPKQQQHKVSPHKGKPEK\ntranslation: PLQHRQPKQQQHKVSPHKGKPEK\ntranslation: HYSTDSQSNSNTKFHRTKENQK\ntranslation: HYSTDSQSNSNTKFHRTKENQK\ntranslation: TTAQTAKATATQSFTAQRKTRK\ntranslation: TTAQTAKATATQSFTAQRKTRK\ntranslation: DEYLTCIVILCYIVTARWPER\ntranslation: DEYLTCIVILCYIVTARWPER\ntranslation: ITFTKGPFHNAIWLLKTLNTNYH\ntranslation: ITFTKGPFHNAIWLLKTLNTNYH\ntranslation: ESLNSLQYVSNISVGGYQNKTNA\ntranslation: ESLNSLQYVSNISVGGYQNKTNA\ntranslation: PHKAIPSLSELAEPSAENTNLG\ntranslation: PHKAIPSLSELAEPSAENTNLG\ntranslation: LIFIFLLVLFPILFALFFLYITLF\ntranslation: LIFIFLLVLFPILFALFFLYITLF\ntranslation: LSLSFSLSSFLSFLLSSFFTSLF\ntranslation: LSLSFSLSSFLSFLLSSFFTSLF\ntranslation: YLYLSPCPLSYPFCSLLSLHHSF\ntranslation: YLYLSPCPLSYPFCSLLSLHHSF\ntranslation: EKSDVKKEESKKDRKEDKEKDKDK\ntranslation: EKSDVKKEESKKDRKEDKEKDKDK\ntranslation: EENLRKFDIYFGSVIKQLRLLKF\ntranslation: EENLRKFDIYFGSVIKQLRLLKF\ntranslation: YRKKKYMYNSKLNFVCIGVC\ntranslation: YRKKKYMYNSKLNFVCIGVC\ntranslation: SILLSFFLVDRFELTEWVDILF\ntranslation: SILLSFFLVDRFELTEWVDILF\ntranslation: PFCCRFSLLIDLNLLNGLISFS\ntranslation: PFCCRFSLLIDLNLLNGLISFS\ntranslation: QKRISTHSVSSNLSTRKNDSRME\ntranslation: QKRISTHSVSSNLSTRKNDSRME\ntranslation: EKDINPFSKFKSINKEKRQQNG\ntranslation: EKDINPFSKFKSINKEKRQQNG\ntranslation: HCTRFVQQSIIKGFLSKPYVDS\ntranslation: HCTRFVQQSIIKGFLSKPYVDS\ntranslation: LHSVCPTEYNQGFSFKALRRQL\ntranslation: LHSVCPTEYNQGFSFKALRRQL\ntranslation: LTLTTKLTNTNISDAQEIETV\ntranslation: LTLTTKLTNTNISDAQEIETV\ntranslation: HHLIISSKIYVWLKLCQMD\ntranslation: HHLIISSKIYVWLKLCQMD\ntranslation: PFNHFQQNICLAEIMSNG\ntranslation: PFNHFQQNICLAEIMSNG\ntranslation: HHPMRQPARTPERRHILPILHP\ntranslation: HHPMRQPARTPERRHILPILHP\ntranslation: AYTILRSVPNKLGGVLALLL\ntranslation: AYTILRSVPNKLGGVLALLL\ntranslation: VKKGRFGWGWGVLGRKIRIK\ntranslation: VKKGRFGWGWGVLGRKIRIK\ntranslation: YFILIFLPKTPHPQPNLPFLT\ntranslation: YFILIFLPKTPHPQPNLPFLT\ntranslation: FYSNLSSQNTPPPTKSSLLN\ntranslation: FYSNLSSQNTPPPTKSSLLN\ntranslation: SCTEALYNIGLTYEKLNRLD\ntranslation: SCTEALYNIGLTYEKLNRLD\ntranslation: SWGFSSLLRNLILIPEANFFSLS\ntranslation: SWGFSSLLRNLILIPEANFFSLS\ntranslation: MLREKKFASGIKIRFLRREENPQL\ntranslation: MLREKKFASGIKIRFLRREENPQL\ntranslation: SPGEASSETLHSHSEHGSSQ\ntranslation: SPGEASSETLHSHSEHGSSQ\ntranslation: RQERPPVKPFILTRNMAQA\ntranslation: RQERPPVKPFILTRNMAQA\ntranslation: GLSHVPSENEGFHWRPLLA\ntranslation: GLSHVPSENEGFHWRPLLA\ntranslation: ILIIAHGLTSSLLFCLANSNYER\ntranslation: ILIIAHGLTSSLLFCLANSNYER\ntranslation: SHNRPRTYILITILPSKLKLRT\ntranslation: SHNRPRTYILITILPSKLKLRT\ntranslation: SCSTMSSDEFANTMPVRPPTV\ntranslation: SCSTMSSDEFANTMPVRPPTV\ntranslation: RVVRCLVMSLLIQCQLGHLR\ntranslation: RVVRCLVMSLLIQCQLGHLR\ntranslation: HRRWPNWHCISKLITRHRTTR\ntranslation: HRRWPNWHCISKLITRHRTTR\ntranslation: TVGGLTGIVLANSSLDIVLHD\ntranslation: TVGGLTGIVLANSSLDIVLHD\ntranslation: ENRLFPKSCLGSDSLCLCL\ntranslation: ENRLFPKSCLGSDSLCLCL\ntranslation: RRQRHRLSDPKQDLGNRRF\ntranslation: RRQRHRLSDPKQDLGNRRF\ntranslation: CDFALHHRCSVLLPAPVLLL\ntranslation: CDFALHHRCSVLLPAPVLLL\ntranslation: VILLCIIAVLCYCRLQYYCC\ntranslation: VILLCIIAVLCYCRLQYYCC\ntranslation: AAVVLEPAVAQNSDDAEQNH\ntranslation: AAVVLEPAVAQNSDDAEQNH\ntranslation: RLKEPLISNVDSRMMARVLIF\ntranslation: RLKEPLISNVDSRMMARVLIF\ntranslation: KINTLAIILLSTLLISGSFNL\ntranslation: KINTLAIILLSTLLISGSFNL\ntranslation: DLSRVMTMFYILGTSYMGICMYQ\ntranslation: DLSRVMTMFYILGTSYMGICMYQ\ntranslation: LVHAYTHVTSTQDIKHRHYSRKV\ntranslation: LVHAYTHVTSTQDIKHRHYSRKV\ntranslation: CATSSKSDVMGTWEMSEKCPKGF\ntranslation: CATSSKSDVMGTWEMSEKCPKGF\ntranslation: QNPFGHFSDISQVPITSDLELVA\ntranslation: QNPFGHFSDISQVPITSDLELVA\ntranslation: KSLRTFLRHLPSSHHVRFGAGSA\ntranslation: KSLRTFLRHLPSSHHVRFGAGSA\ntranslation: LWETFFALHTATIVHLCLQVN\ntranslation: LWETFFALHTATIVHLCLQVN\ntranslation: LGDLLCFAYSHHSTPVPPSEP\ntranslation: LGDLLCFAYSHHSTPVPPSEP\ntranslation: RFTWRHRCTMVAVCKAKKVSQ\ntranslation: RFTWRHRCTMVAVCKAKKVSQ\ntranslation: GSLGGTGVLWWLYAKQRRSPK\ntranslation: GSLGGTGVLWWLYAKQRRSPK\ntranslation: VHLEAQVYYGGCMQSKEGLPK\ntranslation: VHLEAQVYYGGCMQSKEGLPK\ntranslation: SYLRTCGVSCCFNHQCRSHPILK\ntranslation: SYLRTCGVSCCFNHQCRSHPILK\ntranslation: ATLGHVGFLAVSIISADPILSLR\ntranslation: ATLGHVGFLAVSIISADPILSLR\ntranslation: LKDRMGSALMIETARNPTCPKVA\ntranslation: LKDRMGSALMIETARNPTCPKVA\ntranslation: SKCMKICHVLKIKLTSVS\ntranslation: SKCMKICHVLKIKLTSVS\ntranslation: SAFLRQKARANCLSQHPKFTSFF\ntranslation: SAFLRQKARANCLSQHPKFTSFF\ntranslation: KGKRIMKRVIREL\ntranslation: KGKRIMKRVIREL\ntranslation: GKEDYEESHQRAV\ntranslation: GKEDYEESHQRAV\ntranslation: HSSLMTLFIILFP\ntranslation: HSSLMTLFIILFP\ntranslation: QLSDDSLHNPLSL\ntranslation: QLSDDSLHNPLSL\ntranslation: RRMTYSTVSPSLSQSLHPM\ntranslation: RRMTYSTVSPSLSQSLHPM\ntranslation: ENDVLHCVAFAVPKSSSN\ntranslation: ENDVLHCVAFAVPKSSSN\ntranslation: HWMKTLGQRRRHSGVRHSP\ntranslation: HWMKTLGQRRRHSGVRHSP\ntranslation: LDEDFGTAKATQWSTSFS\ntranslation: LDEDFGTAKATQWSTSFS\ntranslation: SNFFFTVCVFHFSCHHCQKFREVN\ntranslation: SNFFFTVCVFHFSCHHCQKFREVN\ntranslation: LISSSLSVSFIFLAIIVRNSGKS\ntranslation: LISSSLSVSFIFLAIIVRNSGKS\ntranslation: IDFPEFLTMMARKMKDTDSEEEIR\ntranslation: IDFPEFLTMMARKMKDTDSEEEIR\ntranslation: LPSTPFLNTHLIFTICFGHHYFP\ntranslation: LPSTPFLNTHLIFTICFGHHYFP\ntranslation: WKIMMSKANCEDQMCVKKRCRGE\ntranslation: WKIMMSKANCEDQMCVKKRCRGE\ntranslation: RKDRRAESVQTHSINPFYKVLIT\ntranslation: RKDRRAESVQTHSINPFYKVLIT\ntranslation: LYAPFNSSLSMCEVMSSSVA\ntranslation: LYAPFNSSLSMCEVMSSSVA\ntranslation: DATDEDITSHMESEELNGAYK\ntranslation: DATDEDITSHMESEELNGAYK\ntranslation: CYRRGHHLTHGKRGVEWCIQ\ntranslation: CYRRGHHLTHGKRGVEWCIQ\ntranslation: FMQFNEPSKEFTHLYTEFTSTS\ntranslation: FMQFNEPSKEFTHLYTEFTSTS\ntranslation: SCNLMNPQKNLHIFIQNSHLHQ\ntranslation: SCNLMNPQKNLHIFIQNSHLHQ\ntranslation: DDSPFTKAFNDLATVYCSSLISL\ntranslation: DDSPFTKAFNDLATVYCSSLISL\ntranslation: MTHHLQRPSMIWPLFTAQVSSL\ntranslation: MTHHLQRPSMIWPLFTAQVSSL\ntranslation: RYPHLQDLTKPDHKMLLYEEFP\ntranslation: RYPHLQDLTKPDHKMLLYEEFP\ntranslation: QEIPHKGAFCDLVWSDPEDVDT\ntranslation: QEIPHKGAFCDLVWSDPEDVDT\ntranslation: RKFLIKEHFVIWFGQILKMWIP\ntranslation: RKFLIKEHFVIWFGQILKMWIP\ntranslation: AFHNAWCHPLNNMFKDKSRHD\ntranslation: AFHNAWCHPLNNMFKDKSRHD\ntranslation: LFIMLGAIRLIICSKIKAGMI\ntranslation: LFIMLGAIRLIICSKIKAGMI\ntranslation: IMPAFIFEHIIKRMAPSIMKS\ntranslation: IMPAFIFEHIIKRMAPSIMKS\ntranslation: GNYHVNLNSSKIVLCLLDHDP\ntranslation: GNYHVNLNSSKIVLCLLDHDP\ntranslation: GLRVGHILGIMSLREII\ntranslation: GLRVGHILGIMSLREII\ntranslation: AACWAHPWHYEPERNH\ntranslation: AACWAHPWHYEPERNH\ntranslation: YDFSQAHNAKDVPNTQP\ntranslation: YDFSQAHNAKDVPNTQP\ntranslation: MISLRLIMPRMCPTRS\ntranslation: MISLRLIMPRMCPTRS\ntranslation: KTAQACLLQK\ntranslation: KTAQACLLQK\ntranslation: KQLRHVCCR\ntranslation: KQLRHVCCR\ntranslation: NSSGMFAAE\ntranslation: NSSGMFAAE\ntranslation: LLQQTCLSCF\ntranslation: LLQQTCLSCF\ntranslation: SAANMPELF\ntranslation: SAANMPELF\ntranslation: PPSVASSLNSSSVPNLRISSSS\ntranslation: PPSVASSLNSSSVPNLRISSSS\ntranslation: QELDDILKFGTEELFKDEATDGG\ntranslation: QELDDILKFGTEELFKDEATDGG\ntranslation: RSLMISSNLALRNYSRMKPLMEE\ntranslation: RSLMISSNLALRNYSRMKPLMEE\ntranslation: LVNSFLVRLFSLGDHVFSMCTCN\ntranslation: LVNSFLVRLFSLGDHVFSMCTCN\ntranslation: LRFQLSESQRLVINYYKITGD\ntranslation: LRFQLSESQRLVINYYKITGD\ntranslation: LKVYCKAVTLSNCCSATAYSEIY\ntranslation: LKVYCKAVTLSNCCSATAYSEIY\ntranslation: HIVFVLIAYQLSNCIDLPNYIF\ntranslation: HIVFVLIAYQLSNCIDLPNYIF\ntranslation: HSFENNFLFVLSLSRLTHTHT\ntranslation: HSFENNFLFVLSLSRLTHTHT\ntranslation: VCVYVSVCSEKGQKGNYFQNC\ntranslation: VCVYVSVCSEKGQKGNYFQNC\ntranslation: CVCMCQSAQRKDKKEIIFKTV\ntranslation: CVCMCQSAQRKDKKEIIFKTV\ntranslation: VCVCVSLLRERTKRKLFSKL\ntranslation: VCVCVSLLRERTKRKLFSKL\ntranslation: KDVHEDSLTVVFENNWQPER\ntranslation: KDVHEDSLTVVFENNWQPER\ntranslation: KMFMKTPLQLFLKIIGNQNA\ntranslation: KMFMKTPLQLFLKIIGNQNA\ntranslation: AFWLPIIFKNNCKGVFMNIF\ntranslation: AFWLPIIFKNNCKGVFMNIF\ntranslation: GSFLAIAMKISFTFIAVFADVSM\ntranslation: GSFLAIAMKISFTFIAVFADVSM\ntranslation: METSAKTAMNVNEIFMAIAKKLP\ntranslation: METSAKTAMNVNEIFMAIAKKLP\ntranslation: PAHDYFKDTGGDGQDSLYKIC\ntranslation: PAHDYFKDTGGDGQDSLYKIC\ntranslation: QPMTTLRTQEEMDKIPYIKYA\ntranslation: QPMTTLRTQEEMDKIPYIKYA\ntranslation: RSTNTFFMLHTYWEYFYSKYATV\ntranslation: RSTNTFFMLHTYWEYFYSKYATV\ntranslation: EALILSSCCTHTGNTFTVNMQLF\ntranslation: EALILSSCCTHTGNTFTVNMQLF\ntranslation: NSCIFTVKVFPVCVQHEESISAS\ntranslation: NSCIFTVKVFPVCVQHEESISAS\ntranslation: QVEKQMFQKIRQLKISLRML\ntranslation: QVEKQMFQKIRQLKISLRML\ntranslation: HSIRNEIFSWRIFWNICFSTC\ntranslation: HSIRNEIFSWRIFWNICFSTC\ntranslation: TAFAMRSLVGGFSGTFVFQLV\ntranslation: TAFAMRSLVGGFSGTFVFQLV\ntranslation: CVCVCVCV\ntranslation: CVCVCVCV\ntranslation: VCVCVCV\ntranslation: VCVCVCV\ntranslation: CVCVCVC\ntranslation: CVCVCVC\ntranslation: HTHTHTHT\ntranslation: HTHTHTHT\ntranslation: THTHTHT\ntranslation: THTHTHT\ntranslation: HTHTHTH\ntranslation: HTHTHTH\ntranslation: ARSPKETQSTSCRWACSEFV\ntranslation: ARSPKETQSTSCRWACSEFV\ntranslation: PGAQRKLSQPAVDGLVPSLF\ntranslation: PGAQRKLSQPAVDGLVPSLF\ntranslation: EQTRNKPIYSWLTEFPLGSW\ntranslation: EQTRNKPIYSWLTEFPLGSW\ntranslation: RSWLDHRISITVSDLRLLNRDL\ntranslation: RSWLDHRISITVSDLRLLNRDL\ntranslation: SSVSIISNNPRRSSTRCRAR\ntranslation: SSVSIISNNPRRSSTRCRAR\ntranslation: HLFPSSLTIHAGLPQGAEL\ntranslation: HLFPSSLTIHAGLPQGAEL\ntranslation: RALHLVEDLRGLLEMMETDE\ntranslation: RALHLVEDLRGLLEMMETDE\ntranslation: LDMSIIKIHPFANLCLKKECRIM\ntranslation: LDMSIIKIHPFANLCLKKECRIM\ntranslation: HYSTFFFQTKIRKWMYFDDAHVK\ntranslation: HYSTFFFQTKIRKWMYFDDAHVK\ntranslation: IILHSFFKQRFANGCILMMLMSR\ntranslation: IILHSFFKQRFANGCILMMLMSR\ntranslation: TCSSVDGHLTCFYFLVILNNATM\ntranslation: TCSSVDGHLTCFYFLVILNNATM\ntranslation: HKCSYSVFILELFFCLFFSFLDL\ntranslation: HKCSYSVFILELFFCLFFSFLDL\ntranslation: EIEIPPTRADIIHACDIVEDAA\ntranslation: EIEIPPTRADIIHACDIVEDAA\ntranslation: IAASSTISHAWIMSALVGGISIS\ntranslation: IAASSTISHAWIMSALVGGISIS\ntranslation: SCIFYNITCMDNVSSGWRDFNL\ntranslation: SCIFYNITCMDNVSSGWRDFNL\ntranslation: DSISKFLQFQLTVNVVAVIVAF\ntranslation: DSISKFLQFQLTVNVVAVIVAF\ntranslation: LKEANFDINQLYDCNWVVVNCST\ntranslation: LKEANFDINQLYDCNWVVVNCST\ntranslation: LKKPTSTSISYMTAIGLLSTAPL\ntranslation: LKKPTSTSISYMTAIGLLSTAPL\ntranslation: LLQVLCLQNDNSQLTSSRYVA\ntranslation: LLQVLCLQNDNSQLTSSRYVA\ntranslation: RYVPTAGKLTVVILEAKNLKK\ntranslation: RYVPTAGKLTVVILEAKNLKK\ntranslation: ISQVYTRMAQIYSDYSIITILYH\ntranslation: ISQVYTRMAQIYSDYSIITILYH\ntranslation: SARYIPGWHRFTVTILLLPSFII\ntranslation: SARYIPGWHRFTVTILLLPSFII\ntranslation: MIKDGNNRIVTVNLCHPGIYLAD\ntranslation: MIKDGNNRIVTVNLCHPGIYLAD\ntranslation: LEALANSRKGQFRGYNGLNAA\ntranslation: LEALANSRKGQFRGYNGLNAA\ntranslation: STKITFSTCLNIKAVVIWKSLVN\ntranslation: STKITFSTCLNIKAVVIWKSLVN\ntranslation: FTSDFQMTTALIFKHVLKVILVE\ntranslation: FTSDFQMTTALIFKHVLKVILVE\ntranslation: FNTSLWGTALLFIKFKIKIYFQ\ntranslation: FNTSLWGTALLFIKFKIKIYFQ\ntranslation: NPIVVVSCMSYLCFLDINPLL\ntranslation: NPIVVVSCMSYLCFLDINPLL\ntranslation: QPILTYKQGSTYFLESNVLYF\ntranslation: QPILTYKQGSTYFLESNVLYF\ntranslation: TIPHHPTKNIKHKLPPTSLTK\ntranslation: TIPHHPTKNIKHKLPPTSLTK\ntranslation: LFLITQLKTLNTNYHLPPSPK\ntranslation: LFLITQLKTLNTNYHLPPSPK\ntranslation: HAFYTHTTYTCTTYNTHYTHYKH\ntranslation: HAFYTHTTYTCTTYNTHYTHYKH\ntranslation: MHSTHTQHTHALHITHIIHITNT\ntranslation: MHSTHTQHTHALHITHIIHITNT\ntranslation: VFVMCIMCVICSACVCCVCVECM\ntranslation: VFVMCIMCVICSACVCCVCVECM\ntranslation: IVDLQINSAIFISLPESQGR\ntranslation: IVDLQINSAIFISLPESQGR\ntranslation: KGKKAGSTFALIKYFYVMLKTWK\ntranslation: KGKKAGSTFALIKYFYVMLKTWK\ntranslation: GVKPPRGILLYGPPGTGKTLV\ntranslation: GVKPPRGILLYGPPGTGKTLV\ntranslation: NQGLSCSRRSVKQDSSRRFHT\ntranslation: NQGLSCSRRSVKQDSSRRFHT\ntranslation: QQHTCVHC\ntranslation: QQHTCVHC\ntranslation: NNTLVFIV\ntranslation: NNTLVFIV\ntranslation: TTHLCSLW\ntranslation: TTHLCSLW\ntranslation: HNEHKCVV\ntranslation: HNEHKCVV\ntranslation: TMNTSVLL\ntranslation: TMNTSVLL\ntranslation: LMYPLLSSLQ\ntranslation: LMYPLLSSLQ\ntranslation: SCILYSVPSN\ntranslation: SCILYSVPSN\ntranslation: HVSFTQFPPM\ntranslation: HVSFTQFPPM\ntranslation: HWRELSKGYM\ntranslation: HWRELSKGYM\ntranslation: HIVVLIWISQVVMLSIFSCSLVI\ntranslation: HIVVLIWISQVVMLSIFSCSLVI\ntranslation: HCGFDMDFPGSDVEYFLMFIGHL\ntranslation: HCGFDMDFPGSDVEYFLMFIGHL\ntranslation: QMTNEHEKILNITTWEIHIKTTM\ntranslation: QMTNEHEKILNITTWEIHIKTTM\ntranslation: STIGSFDPSFRYAAFPSSSS\ntranslation: STIGSFDPSFRYAAFPSSSS\ntranslation: PLLVLLIHLSGMQLFLLRVL\ntranslation: PLLVLLIHLSGMQLFLLRVL\ntranslation: KELEEGKAAYLKDGSKEPIVE\ntranslation: KELEEGKAAYLKDGSKEPIVE\ntranslation: RTRRRKSCIPERWIKRTNSG\ntranslation: RTRRRKSCIPERWIKRTNSG\ntranslation: DLDSETRSMVEKMMYDQRQ\ntranslation: DLDSETRSMVEKMMYDQRQ\ntranslation: SVAGHTSSFPPCCESHCPG\ntranslation: SVAGHTSSFPPCCESHCPG\ntranslation: LSLVIHHLFHHAASLTVQV\ntranslation: LSLVIHHLFHHAASLTVQV\ntranslation: CRWSYIIFSTMLRVSLSR\ntranslation: CRWSYIIFSTMLRVSLSR\ntranslation: YAEGIGVSPCYIMLGYNFSSFPC\ntranslation: YAEGIGVSPCYIMLGYNFSSFPC\ntranslation: QKSWTLLYIKKSSNNFFPEPT\ntranslation: QKSWTLLYIKKSSNNFFPEPT\ntranslation: KNHGLYYTLKNLLTTFFLNLL\ntranslation: KNHGLYYTLKNLLTTFFLNLL\ntranslation: VGSGKKLLEDFLMYNKVHDF\ntranslation: VGSGKKLLEDFLMYNKVHDF\ntranslation: QDYPSLVLFLPLNNNVNLVKSFL\ntranslation: QDYPSLVLFLPLNNNVNLVKSFL\ntranslation: GLSIFSAFPAVKQQCELGQVIS\ntranslation: GLSIFSAFPAVKQQCELGQVIS\ntranslation: KPGFPEPTNNTPKGILHLSPD\ntranslation: KPGFPEPTNNTPKGILHLSPD\ntranslation: SLGFLNQQIILQKEFFICLLM\ntranslation: SLGFLNQQIILQKEFFICLLM\ntranslation: HQETNEEFLLEYYLLVQETQA\ntranslation: HQETNEEFLLEYYLLVQETQA\ntranslation: IRRQMKNSFWSIICWFRKPRL\ntranslation: IRRQMKNSFWSIICWFRKPRL\ntranslation: LGLAFRRAICCGSHELECRINHA\ntranslation: LGLAFRRAICCGSHELECRINHA\ntranslation: SSSSSFLISFFKQFIVSRLAFNLS\ntranslation: SSSSSFLISFFKQFIVSRLAFNLS\ntranslation: ERLKASRETINCLKNEIRKLEEDD\ntranslation: ERLKASRETINCLKNEIRKLEEDD\ntranslation: EDVQMERVRTANALNSTNFDEK\ntranslation: EDVQMERVRTANALNSTNFDEK\ntranslation: LLIKISRIQGICCSHSFHLNIF\ntranslation: LLIKISRIQGICCSHSFHLNIF\ntranslation: SKLGLSCLLIINPSISRFVVQGH\ntranslation: SKLGLSCLLIINPSISRFVVQGH\ntranslation: CPWTTNLDILGLIIRRQLRPSF\ntranslation: CPWTTNLDILGLIIRRQLRPSF\ntranslation: CEETFSVYWISHYS\ntranslation: CEETFSVYWISHYS\ntranslation: VKKHSQFIGYPITL\ntranslation: VKKHSQFIGYPITL\ntranslation: KSNGISNKLRMFLH\ntranslation: KSNGISNKLRMFLH\ntranslation: QYPPNTSKVYSYFECREKKTENS\ntranslation: QYPPNTSKVYSYFECREKKTENS\ntranslation: NIHPTQAKFIPTLNAVKRRQKTP\ntranslation: NIHPTQAKFIPTLNAVKRRQKTP\ntranslation: WSFLSSFHGIQSRNKLCLCWVDI\ntranslation: WSFLSSFHGIQSRNKLCLCWVDI\ntranslation: GVFCLLFTAFKVGINFACVGWIL\ntranslation: GVFCLLFTAFKVGINFACVGWIL\ntranslation: IHLLLVQFSLIEVQKANCAAN\ntranslation: IHLLLVQFSLIEVQKANCAAN\ntranslation: IGCTVSLLNFYKTELNKEEMY\ntranslation: IGCTVSLLNFYKTELNKEEMY\ntranslation: ANAQEYSHSKVSLLSYFLSTNQ\ntranslation: ANAQEYSHSKVSLLSYFLSTNQ\ntranslation: LMHKNTVTAKFPYSHISSVLI\ntranslation: LMHKNTVTAKFPYSHISSVLI\ntranslation: LTAPLLHTLFLNLSWACDLFS\ntranslation: LTAPLLHTLFLNLSWACDLFS\ntranslation: DLLPRLTPILKNRHEKVQEN\ntranslation: DLLPRLTPILKNRHEKVQEN\ntranslation: QFSCTFSCLFFKMGVSLGSRS\ntranslation: QFSCTFSCLFFKMGVSLGSRS\ntranslation: KILPGREQLSFNSSGLLFSPLT\ntranslation: KILPGREQLSFNSSGLLFSPLT\ntranslation: AVNGENKSPLLLNDSCSLPGRIF\ntranslation: AVNGENKSPLLLNDSCSLPGRIF\ntranslation: TVLPPFLNPIIYSLRNKEIKTAM\ntranslation: TVLPPFLNPIIYSLRNKEIKTAM\ntranslation: QFCLPSSIPSYIVLGIRRLKQP\ntranslation: QFCLPSSIPSYIVLGIRRLKQP\ntranslation: HGCFNLLIPKTIYDGIEEGRQNC\ntranslation: HGCFNLLIPKTIYDGIEEGRQNC\ntranslation: MAVLISLFLRLYMMGLRKGGKTV\ntranslation: MAVLISLFLRLYMMGLRKGGKTV\ntranslation: PRPNLRQNPFHYHIHRRKSNFL\ntranslation: PRPNLRQNPFHYHIHRRKSNFL\ntranslation: LDQTYAKIHFTIIFIGVNLTF\ntranslation: LDQTYAKIHFTIIFIGVNLTF\ntranslation: LNQRNPLRLCQRRRSKRSRLKQ\ntranslation: LNQRNPLRLCQRRRSKRSRLKQ\ntranslation: ESEKSTEVVPKKKIKKEQVET\ntranslation: ESEKSTEVVPKKKIKKEQVET\ntranslation: VSTCSFLIFFFGTTSVDFSDS\ntranslation: VSTCSFLIFFFGTTSVDFSDS\ntranslation: LALRWILLTWMVTRYIFPGIRS\ntranslation: LALRWILLTWMVTRYIFPGIRS\ntranslation: GFEMDITHLDGHKVHISRDKIT\ntranslation: GFEMDITHLDGHKVHISRDKIT\ntranslation: GDLIPGNMYLVTIQVSNIHLKA\ntranslation: GDLIPGNMYLVTIQVSNIHLKA\ntranslation: LRLKIDLRKVFRFTFSTIIHFIAS\ntranslation: LRLKIDLRKVFRFTFSTIIHFIAS\ntranslation: AGYEVNNGTESKPEDLSQIYFKSE\ntranslation: AGYEVNNGTESKPEDLSQIYFKSE\ntranslation: LGHAITFFTSTLPIRSSSQVS\ntranslation: LGHAITFFTSTLPIRSSSQVS\ntranslation: WGMPLPFSLLHYQLGVHPRY\ntranslation: WGMPLPFSLLHYQLGVHPRY\ntranslation: DTWDELLIGSVEVKKVMACP\ntranslation: DTWDELLIGSVEVKKVMACP\ntranslation: HTNLFSTSKGVMKGFCLFQAKN\ntranslation: HTNLFSTSKGVMKGFCLFQAKN\ntranslation: FFAWKRQKPFITPLLVENRFVC\ntranslation: FFAWKRQKPFITPLLVENRFVC\ntranslation: LQFFFCPLR\ntranslation: LQFFFCPLR\ntranslation: PIFFLSLE\ntranslation: PIFFLSLE\ntranslation: PQGTEKKLE\ntranslation: PQGTEKKLE\ntranslation: LKGQKKNW\ntranslation: LKGQKKNW\ntranslation: SRDRKKIG\ntranslation: SRDRKKIG\ntranslation: PGLLTSYRNCLALGYVQPGRGV\ntranslation: PGLLTSYRNCLALGYVQPGRGV\ntranslation: WASHQLQELPSTWVCTTWKRC\ntranslation: WASHQLQELPSTWVCTTWKRC\ntranslation: QFSVKETNFSEGNLKLKIGLQ\ntranslation: QFSVKETNFSEGNLKLKIGLQ\ntranslation: VFSERNKLFRGKFKIENWPP\ntranslation: VFSERNKLFRGKFKIENWPP\ntranslation: PGGQFSILNFPLKSLFLSLKT\ntranslation: PGGQFSILNFPLKSLFLSLKT\ntranslation: WRPIFNFKFPSEKFVSFTEN\ntranslation: WRPIFNFKFPSEKFVSFTEN\ntranslation: DNLTILLHTASVMQHLHHSIQVW\ntranslation: DNLTILLHTASVMQHLHHSIQVW\ntranslation: RPVLNDVSAALQKLCGEVWLSY\ntranslation: RPVLNDVSAALQKLCGEVWLSY\ntranslation: KQTGSPEQFDLIVLTMPVPEIL\ntranslation: KQTGSPEQFDLIVLTMPVPEIL\ntranslation: NKQAPLSSLILLFSQCQFLRFC\ntranslation: NKQAPLSSLILLFSQCQFLRFC\ntranslation: QNLRNWHCENNKIKLLRGACLF\ntranslation: QNLRNWHCENNKIKLLRGACLF\ntranslation: RISGTGIVRTIRSNCSGEPVC\ntranslation: RISGTGIVRTIRSNCSGEPVC\ntranslation: QYLLCNINSHITDKLIVVAAS\ntranslation: QYLLCNINSHITDKLIVVAAS\ntranslation: HIYRVCTNRKWLQITTRLGRIP\ntranslation: HIYRVCTNRKWLQITTRLGRIP\ntranslation: IFTEFAQIENGYKLQPGLAGYQ\ntranslation: IFTEFAQIENGYKLQPGLAGYQ\ntranslation: FDETFEFQVNLPELAMIRFVVLD\ntranslation: FDETFEFQVNLPELAMIRFVVLD\ntranslation: SRTTKRIMASSGRFTWNSKVSSK\ntranslation: SRTTKRIMASSGRFTWNSKVSSK\ntranslation: PEQQNGSWPAQVGLLGTQKFHQ\ntranslation: PEQQNGSWPAQVGLLGTQKFHQ\ntranslation: TVVSLTFLRMLILSGSAPEIL\ntranslation: TVVSLTFLRMLILSGSAPEIL\ntranslation: RISGADPDKMSILRKVRDTT\ntranslation: RISGADPDKMSILRKVRDTT\ntranslation: LESYEEDRVLRFQNKIRYFCLLN\ntranslation: LESYEEDRVLRFQNKIRYFCLLN\ntranslation: HLVKKVNSSFVSQLQKAMCLVG\ntranslation: HLVKKVNSSFVSQLQKAMCLVG\ntranslation: LGEESKFLLCLPAPEGNVSRWL\ntranslation: LGEESKFLLCLPAPEGNVSRWL\ntranslation: QPTRHIAFWSWETKEEFTFFTK\ntranslation: QPTRHIAFWSWETKEEFTFFTK\ntranslation: SQRDTLPSGAGRQRRNLLSSPS\ntranslation: SQRDTLPSGAGRQRRNLLSSPS\ntranslation: ANETHCLLELGDKGGIYFLHQV\ntranslation: ANETHCLLELGDKGGIYFLHQV\ntranslation: EVILPSFTHLVIDSVIQLITKYF\ntranslation: EVILPSFTHLVIDSVIQLITKYF\ntranslation: CTIPMRHNYNKLHLPTTNRPKIA\ntranslation: CTIPMRHNYNKLHLPTTNRPKIA\ntranslation: SDFRSVCRRQMELVIIMPHRDST\ntranslation: SDFRSVCRRQMELVIIMPHRDST\ntranslation: IKPNPATQNLSILLNYPHRMNN\ntranslation: IKPNPATQNLSILLNYPHRMNN\ntranslation: IIHPMWVIEEYAKILRSWVWFN\ntranslation: IIHPMWVIEEYAKILRSWVWFN\ntranslation: LLRIIEMSVFLEYFTLRTNLLS\ntranslation: LLRIIEMSVFLEYFTLRTNLLS\ntranslation: EDSRLVLSVKYSKNTDISMILNK\ntranslation: EDSRLVLSVKYSKNTDISMILNK\ntranslation: LGSAMSRPI\ntranslation: LGSAMSRPI\ntranslation: GKCHEQAH\ntranslation: GKCHEQAH\ntranslation: DGPAHGTSQ\ntranslation: DGPAHGTSQ\ntranslation: MGLLMALP\ntranslation: MGLLMALP\ntranslation: WACSWHFP\ntranslation: WACSWHFP\ntranslation: SLSVRVKIFYFLILKIVTIIDLL\ntranslation: SLSVRVKIFYFLILKIVTIIDLL\ntranslation: GKVLFQVRLKVLIWFFTREIGNN\ntranslation: GKVLFQVRLKVLIWFFTREIGNN\ntranslation: IIANLTCKKPDQHFKPYLKQHLP\ntranslation: IIANLTCKKPDQHFKPYLKQHLP\ntranslation: NKHNIDGSRIELVAMSCHCNYDI\ntranslation: NKHNIDGSRIELVAMSCHCNYDI\ntranslation: ECHNCSDRTLPPTLSYYHQCCVY\ntranslation: ECHNCSDRTLPPTLSYYHQCCVY\ntranslation: NVIIAVTGHCHQLYPTTINVVFI\ntranslation: NVIIAVTGHCHQLYPTTINVVFI\ntranslation: ALSPRLKYSGTIIAHCNLEFL\ntranslation: ALSPRLKYSGTIIAHCNLEFL\ntranslation: RKLLNETSVTRLPLPNPGDPF\ntranslation: RKLLNETSVTRLPLPNPGDPF\ntranslation: EGVSRIREWESCYRCFIQEFS\ntranslation: EGVSRIREWESCYRCFIQEFS\ntranslation: KGSPGLGSGSLVTDVSFKSF\ntranslation: KGSPGLGSGSLVTDVSFKSF\ntranslation: FFIVSVFFTMIFGERLFQSCKHQ\ntranslation: FFIVSVFFTMIFGERLFQSCKHQ\ntranslation: ERRRERAVLFLMTTPEKYQETSL\ntranslation: ERRRERAVLFLMTTPEKYQETSL\ntranslation: QGESVGAFSPAHPAE\ntranslation: QGESVGAFSPAHPAE\ntranslation: KEKVLEHSHRPIQL\ntranslation: KEKVLEHSHRPIQL\ntranslation: SAGWAGENAPTLSP\ntranslation: SAGWAGENAPTLSP\ntranslation: QLDGPVRMLQHFLL\ntranslation: QLDGPVRMLQHFLL\ntranslation: LNLGILIFGIFKFQSGPCISTS\ntranslation: LNLGILIFGIFKFQSGPCISTS\ntranslation: KFGHFNLWHLQVPVRTLHFNI\ntranslation: KFGHFNLWHLQVPVRTLHFNI\ntranslation: DVEMQGPDWNLKMPKIKMPKF\ntranslation: DVEMQGPDWNLKMPKIKMPKF\ntranslation: LTLQYNIYNVGSSAALDVELSDD\ntranslation: LTLQYNIYNVGSSAALDVELSDD\ntranslation: NHQIVPRLMQHLSQHCRCCTARS\ntranslation: NHQIVPRLMQHLSQHCRCCTARS\ntranslation: VIEAEDGLVISKPKGFLTCLIC\ntranslation: VIEAEDGLVISKPKGFLTCLIC\ntranslation: QMRQVKNPFGLEITNPSSASIT\ntranslation: QMRQVKNPFGLEITNPSSASIT\ntranslation: QHSASIFFREALGVFSRSRFLC\ntranslation: QHSASIFFREALGVFSRSRFLC\ntranslation: SILHQFSLGRPSEYSQGADSYV\ntranslation: SILHQFSLGRPSEYSQGADSYV\ntranslation: RHKNRLLENTPRASLKKIDAEC\ntranslation: RHKNRLLENTPRASLKKIDAEC\ntranslation: KTMQSDGILLHREGPNGDHITL\ntranslation: KTMQSDGILLHREGPNGDHITL\ntranslation: KPCRVMGFYSTGKGQMEITSHC\ntranslation: KPCRVMGFYSTGKGQMEITSHC\ntranslation: RTWHVKSKTDLFIPKKKCSFYDN\ntranslation: RTWHVKSKTDLFIPKKKCSFYDN\ntranslation: HLSPHLRYTLTFPYVFLKEPYCS\ntranslation: HLSPHLRYTLTFPYVFLKEPYCS\ntranslation: RKLRGTFSCQRHEFLCLIFKLF\ntranslation: RKLRGTFSCQRHEFLCLIFKLF\ntranslation: SNDDLLWCLRKLTRPPLTVS\ntranslation: SNDDLLWCLRKLTRPPLTVS\ntranslation: ETVSGGLVSFLKHHRRSSLL\ntranslation: ETVSGGLVSFLKHHRRSSLL\ntranslation: FFCYKLYSMMVPTMYEIPQKLDL\ntranslation: FFCYKLYSMMVPTMYEIPQKLDL\ntranslation: TCPNENQKMERQIHFLLGWTVTT\ntranslation: TCPNENQKMERQIHFLLGWTVTT\ntranslation: HAQMKTKRWKGRYTSYLGGQLQH\ntranslation: HAQMKTKRWKGRYTSYLGGQLQH\ntranslation: DVVTVHPSKKCICLSIFWFSFGHV\ntranslation: DVVTVHPSKKCICLSIFWFSFGHV\ntranslation: QVLKNLMASSA\ntranslation: QVLKNLMASSA\ntranslation: SPQEFDGFFCS\ntranslation: SPQEFDGFFCS\ntranslation: AAEEAIKFLRT\ntranslation: AAEEAIKFLRT\ntranslation: CRRSHQILEDL\ntranslation: CRRSHQILEDL\ntranslation: IFQTLELQGSKLPQDMLDLSTA\ntranslation: IFQTLELQGSKLPQDMLDLSTA\ntranslation: LKHVSAFYFKKTQTWSQNIFREQ\ntranslation: LKHVSAFYFKKTQTWSQNIFREQ\ntranslation: PKPNNSSTAYYNFTGSLFYPP\ntranslation: PKPNNSSTAYYNFTGSLFYPP\ntranslation: RNQIIQALLITILLGLYFTLL\ntranslation: RNQIIQALLITILLGLYFTLL\ntranslation: SFLLLVYLFCRHLPDEPLLTL\ntranslation: SFLLLVYLFCRHLPDEPLLTL\ntranslation: PESTEARQEDVYRKGRPKGGS\ntranslation: PESTEARQEDVYRKGRPKGGS\ntranslation: QSQQRLVRKMSTEKVDQKEEA\ntranslation: QSQQRLVRKMSTEKVDQKEEA\ntranslation: KSIVSQSSLIKRKNRNLHRR\ntranslation: KSIVSQSSLIKRKNRNLHRR\ntranslation: SSVQISVLSLNQGALRNNAF\ntranslation: SSVQISVLSLNQGALRNNAF\ntranslation: QGFFLVSKELFLFGLTVKFTR\ntranslation: QGFFLVSKELFLFGLTVKFTR\ntranslation: PLVNLTVSPKRNSSLDTRKKP\ntranslation: PLVNLTVSPKRNSSLDTRKKP\ntranslation: AEFACKSPAQLWYPVLLSVSNKD\ntranslation: AEFACKSPAQLWYPVLLSVSNKD\ntranslation: PSLLVSLQLNFGILFCYLYQTR\ntranslation: PSLLVSLQLNFGILFCYLYQTR\ntranslation: PCLIQITKQDTKVELETYKQTR\ntranslation: PCLIQITKQDTKVELETYKQTR\ntranslation: QSTDCYGLNVFPPDSCVEILTP\ntranslation: QSTDCYGLNVFPPDSCVEILTP\ntranslation: GLGFQHMNLEGTHSDHSSRCF\ntranslation: GLGFQHMNLEGTHSDHSSRCF\ntranslation: ETYHISNARVSGRKFFHRRCMAVL\ntranslation: ETYHISNARVSGRKFFHRRCMAVL\ntranslation: KNGHTPPMKKLPTTHPSITYMICL\ntranslation: KNGHTPPMKKLPTTHPSITYMICL\ntranslation: IKSVYFILYFINVFNRFLLFFF\ntranslation: IKSVYFILYFINVFNRFLLFFF\ntranslation: LSLSISFCTSSMCSIASCCSFS\ntranslation: LSLSISFCTSSMCSIASCCSFS\ntranslation: EKEQQEAIEHIDEVQNEIDRLN\ntranslation: EKEQQEAIEHIDEVQNEIDRLN\ntranslation: KFCLILHIPVNKIQKHCVFKIVS\ntranslation: KFCLILHIPVNKIQKHCVFKIVS\ntranslation: RHNLKNTMLLYFIHWNMENQTEF\ntranslation: RHNLKNTMLLYFIHWNMENQTEF\ntranslation: DTILKTQCFCILFTGIWRIKQN\ntranslation: DTILKTQCFCILFTGIWRIKQN\ntranslation: RYFSTEASLSIIESYFLRCSRK\ntranslation: RYFSTEASLSIIESYFLRCSRK\ntranslation: GTLARKLPLVLLRATSCDALGS\ntranslation: GTLARKLPLVLLRATSCDALGS\ntranslation: LPRASQEVALNNTKGSFRAKVP\ntranslation: LPRASQEVALNNTKGSFRAKVP\ntranslation: AAKFITHAPPGEFNEVFNDVRLL\ntranslation: AAKFITHAPPGEFNEVFNDVRLL\ntranslation: LLNSSLMHPQGNLMKYSMTFGY\ntranslation: LLNSSLMHPQGNLMKYSMTFGY\ntranslation: AFSYKLVRSKADWERWMESFGT\ntranslation: AFSYKLVRSKADWERWMESFGT\ntranslation: LFQMIPSTFPNRPYFSLICRKK\ntranslation: LFQMIPSTFPNRPYFSLICRKK\ntranslation: AAALMLSDPLQGHFPAQATS\ntranslation: AAALMLSDPLQGHFPAQATS\ntranslation: GRWPGQGSVPGGDHSASRLQQ\ntranslation: GRWPGQGSVPGGDHSASRLQQ\ntranslation: GGGLGREVSLEGITQHQGCS\ntranslation: GGGLGREVSLEGITQHQGCS"
      },
      {
       "name": "stderr",
       "output_type": "stream",
       "text": "771it [00:00, 1269.02it/s]"
      },
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "\ntranslation: EVAWAGKCPWRGSLSIKAAA\ntranslation: EVAWAGKCPWRGSLSIKAAA\ntranslation: LVSGEST\ntranslation: LVSGEST\ntranslation: WFLVSPL\ntranslation: WFLVSPL\ntranslation: PVDSPET\ntranslation: PVDSPET\ntranslation: QWTHQKP\ntranslation: QWTHQKP\ntranslation: SGLTRNQ\ntranslation: SGLTRNQ\ntranslation: PFLTLYYLKSQWKFSVSCLCNS\ntranslation: PFLTLYYLKSQWKFSVSCLCNS\ntranslation: HECTLSLAEEYGNSSPTQQVS\ntranslation: HECTLSLAEEYGNSSPTQQVS\ntranslation: GLLVELENYCRIPLLSSGCIR\ntranslation: GLLVELENYCRIPLLSSGCIR\ntranslation: CTVNLMNFYKSEINKEEMYIRYIH\ntranslation: CTVNLMNFYKSEINKEEMYIRYIH\ntranslation: MDVADIHFFLVNLRFVKIHQVNSA\ntranslation: MDVADIHFFLVNLRFVKIHQVNSA\ntranslation: QKRYHWGYNEISFLTIESNKELIF\ntranslation: QKRYHWGYNEISFLTIESNKELIF\ntranslation: RKLTLYYFQLLRKKSHYIPNDNVS\ntranslation: RKLTLYYFQLLRKKSHYIPNDNVS\ntranslation: ETSLDEDSKAMIRVLRCLWIIS\ntranslation: ETSLDEDSKAMIRVLRCLWIIS\ntranslation: RDNPEASKNANHSLAVFIKRCF\ntranslation: RDNPEASKNANHSLAVFIKRCF\ntranslation: EIIQRHLRTRIIALLSSSRDV\ntranslation: EIIQRHLRTRIIALLSSSRDV\ntranslation: CIARYCPTCTLLHFFSTICDSPT\ntranslation: CIARYCPTCTLLHFFSTICDSPT\ntranslation: SGRITDCAEKMQEGAGGTVPCYA\ntranslation: SGRITDCAEKMQEGAGGTVPCYA\ntranslation: VGESQIVLKKCRRVQVGQYRAMQ\ntranslation: VGESQIVLKKCRRVQVGQYRAMQ\ntranslation: PSVLIVQFPPMSENMWCLVFCP\ntranslation: PSVLIVQFPPMSENMWCLVFCP\ntranslation: RTENQTPHVLTHRWELNNENTW\ntranslation: RTENQTPHVLTHRWELNNENTW\ntranslation: DFVFTSAFLHVLDRDNLRPAL\ntranslation: DFVFTSAFLHVLDRDNLRPAL\ntranslation: LCVYICISTCFRQRQFKACTL\ntranslation: LCVYICISTCFRQRQFKACTL\ntranslation: KSAGLKLSLSKTCRNADVNTK\ntranslation: KSAGLKLSLSKTCRNADVNTK\ntranslation: ISGREVFEFRPELVNDDDGEAD\ntranslation: ISGREVFEFRPELVNDDDGEAD\ntranslation: SVVVKCLNFVLNWSMMMMGKQM\ntranslation: SVVVKCLNFVLNWSMMMMGKQM\ntranslation: ICFPIIIIDQFRTKFKHFTTTD\ntranslation: ICFPIIIIDQFRTKFKHFTTTD\ntranslation: SASPSSSLTSSGRNSNTSRPLI\ntranslation: SASPSSSLTSSGRNSNTSRPLI\ntranslation: PPFPLSLLATLPLLHLCLGFKTF\ntranslation: PPFPLSLLATLPLLHLCLGFKTF\ntranslation: KVLNPKHRCSKGSVAKRERGNG\ntranslation: KVLNPKHRCSKGSVAKRERGNG\ntranslation: QQNINKYLLCSTFYCRLVLNF\ntranslation: QQNINKYLLCSTFYCRLVLNF\ntranslation: FPFKYITRYFLVLYMLFYSYFRH\ntranslation: FPFKYITRYFLVLYMLFYSYFRH\ntranslation: FPLNILLDIFWCCICCSILILGT\ntranslation: FPLNILLDIFWCCICCSILILGT\ntranslation: VPKIRIEQHIQHQKISSNIFKGK\ntranslation: VPKIRIEQHIQHQKISSNIFKGK\ntranslation: AARWQCIIADHRIFPSRNHSSSI\ntranslation: AARWQCIIADHRIFPSRNHSSSI\ntranslation: SKMAMHHSRPQNISFKESQQQY\ntranslation: SKMAMHHSRPQNISFKESQQQY\ntranslation: LDTCLPEKLGCSWEITGDYLRNIC\ntranslation: LDTCLPEKLGCSWEITGDYLRNIC\ntranslation: KYFLNSHLLFPKNIPTFPVGKCQ\ntranslation: KYFLNSHLLFPKNIPTFPVGKCQ\ntranslation: ESESYRKIADSIGCSTNNILFL\ntranslation: ESESYRKIADSIGCSTNNILFL\ntranslation: SETKCCWLSTQCCLQSFGNFHSP\ntranslation: SETKCCWLSTQCCLQSFGNFHSP\ntranslation: RREGGDQRVKSVINLLFAAYTGD\ntranslation: RREGGDQRVKSVINLLFAAYTGD\ntranslation: HLQYMQQTKDLSLTLPFDHHLLF\ntranslation: HLQYMQQTKDLSLTLPFDHHLLF\ntranslation: MHLSCLTAEPYLSLLAQFKTYTG\ntranslation: MHLSCLTAEPYLSLLAQFKTYTG\ntranslation: KGNIILIPERSNSPENTRNIFYK\ntranslation: KGNIILIPERSNSPENTRNIFYK\ntranslation: ACTTTERKIRDSEKSGIILLHIC\ntranslation: ACTTTERKIRDSEKSGIILLHIC\ntranslation: LAQLQKEKSEILKNLALYYFTFV\ntranslation: LAQLQKEKSEILKNLALYYFTFV\ntranslation: DFLKSSSIGQKLRNKKKKRWN\ntranslation: DFLKSSSIGQKLRNKKKKRWN\ntranslation: FHLFFFLFLSFCPIEDDFRK\ntranslation: FHLFFFLFLSFCPIEDDFRK\ntranslation: WTQSENNSIHFHSFPYWLLIIC\ntranslation: WTQSENNSIHFHSFPYWLLIIC\ntranslation: KALNQNVYICLKMSLIIQYALDL\ntranslation: KALNQNVYICLKMSLIIQYALDL\ntranslation: SPKSECLYLPKDVLNNTVCFRS\ntranslation: SPKSECLYLPKDVLNNTVCFRS\ntranslation: QDLKHTVLLRTSLGKYKHSDLGL\ntranslation: QDLKHTVLLRTSLGKYKHSDLGL\ntranslation: PNISINSDFLITILFPLKVLTRN\ntranslation: PNISINSDFLITILFPLKVLTRN\ntranslation: FRVKTFKGNRIVIKKSEFIDILG\ntranslation: FRVKTFKGNRIVIKKSEFIDILG\ntranslation: LGSLAHLKHPA\ntranslation: LGSLAHLKHPA\ntranslation: GFSCSSEAPC\ntranslation: GFSCSSEAPC\ntranslation: RRVLQMSKRTQ\ntranslation: RRVLQMSKRTQ\ntranslation: QGASDEQENP\ntranslation: QGASDEQENP\ntranslation: EEWGGFCMFKLSFYFYVVSECL\ntranslation: EEWGGFCMFKLSFYFYVVSECL\ntranslation: RNGVGFVCSNCHFIFTLCLSV\ntranslation: RNGVGFVCSNCHFIFTLCLSV\ntranslation: RKEAFCILNQFICLLSASVIF\ntranslation: RKEAFCILNQFICLLSASVIF\ntranslation: EYDRCTEKADKLIQYTKGLFP\ntranslation: EYDRCTEKADKLIQYTKGLFP\ntranslation: KMDLQKLAAVVCSCFKISFLV\ntranslation: KMDLQKLAAVVCSCFKISFLV\ntranslation: TKNEILKQEQTTAASFCKSIF\ntranslation: TKNEILKQEQTTAASFCKSIF\ntranslation: SACTSLFYFSVSSILCIFVNSGL\ntranslation: SACTSLFYFSVSSILCIFVNSGL\ntranslation: CLYFSLLFQRLLYPLYFCELWT\ntranslation: CLYFSLLFQRLLYPLYFCELWT\ntranslation: ESRVHKNTEDRGDAEIKERSTSR\ntranslation: ESRVHKNTEDRGDAEIKERSTSR\ntranslation: RVYRVSTPYIYSRNRRRHTSIFH\ntranslation: RVYRVSTPYIYSRNRRRHTSIFH\ntranslation: VKYARVSTSIPTVNIWCAHTINP\ntranslation: VKYARVSTSIPTVNIWCAHTINP\ntranslation: VCLPCWTVSIYIRLLATSDWL\ntranslation: VCLPCWTVSIYIRLLATSDWL\ntranslation: RSFIMRRCWMGWGGR\ntranslation: RSFIMRRCWMGWGGR\ntranslation: EVSSCGDVGWGGEVD\ntranslation: EVSSCGDVGWGGEVD\ntranslation: KFHHAEMLDGVGRSM\ntranslation: KFHHAEMLDGVGRSM\ntranslation: HRPPHPIQHLRMMKL\ntranslation: HRPPHPIQHLRMMKL\ntranslation: STSPPHPTSPHDETS\ntranslation: STSPPHPTSPHDETS\ntranslation: HTLHLTSFGLTTSIIIVLSSVGI\ntranslation: HTLHLTSFGLTTSIIIVLSSVGI\ntranslation: YFTSDILWPNNIYYYSAQQCGH\ntranslation: YFTSDILWPNNIYYYSAQQCGH\ntranslation: MPTLLSTIIIDVVRPKDVRCKV\ntranslation: MPTLLSTIIIDVVRPKDVRCKV\ntranslation: EKFTAFCKAQG\ntranslation: EKFTAFCKAQG\ntranslation: RNSPPSARPR\ntranslation: RNSPPSARPR\ntranslation: EIHRLLQGPG\ntranslation: EIHRLLQGPG\ntranslation: ALGLAEGGEFL\ntranslation: ALGLAEGGEFL\ntranslation: PWALQKAVNF\ntranslation: PWALQKAVNF\ntranslation: PLVTVLSYR\ntranslation: PLVTVLSYR\ntranslation: WDWGCDGIKTNSPLYLCKCKILI\ntranslation: WDWGCDGIKTNSPLYLCKCKILI\ntranslation: DENLALAQVERRVSFYTIASPIP\ntranslation: DENLALAQVERRVSFYTIASPIP\ntranslation: CGVKPFQFELISLKKITLLQVL\ntranslation: CGVKPFQFELISLKKITLLQVL\ntranslation: RNEFPVDIQICLWILYSVPLV\ntranslation: RNEFPVDIQICLWILYSVPLV\ntranslation: NQWNRIENPETNLYIYRELIS\ntranslation: NQWNRIENPETNLYIYRELIS\ntranslation: WLYMSPIYSSRSMKPSAFLSISS\ntranslation: WLYMSPIYSSRSMKPSAFLSISS\ntranslation: EDIDKNADGFIDLEEYIGDMYS\ntranslation: EDIDKNADGFIDLEEYIGDMYS\ntranslation: HVQLLFTWNPSPLRPSKFSFEY\ntranslation: HVQLLFTWNPSPLRPSKFSFEY\ntranslation: MFNCCSHGTLLHFGLQSSRLNI\ntranslation: MFNCCSHGTLLHFGLQSSRLNI\ntranslation: RRVIFDQENVVTSTESSPSRL\ntranslation: RRVIFDQENVVTSTESSPSRL\ntranslation: KKIPLPALSFLTINPSHPSFFRM\ntranslation: KKIPLPALSFLTINPSHPSFFRM\ntranslation: VILKNEGWEGFIVRNDSAGRGIFL\ntranslation: VILKNEGWEGFIVRNDSAGRGIFL\ntranslation: LFSMCNLKLICTTGYLTGA\ntranslation: LFSMCNLKLICTTGYLTGA\ntranslation: VAPVRYPVVQISFKLHILNK\ntranslation: VAPVRYPVVQISFKLHILNK\ntranslation: GFGCFSCFSICIKKLVHVNTFH\ntranslation: GFGCFSCFSICIKKLVHVNTFH\ntranslation: WKVFTWTSFFIQMLKHEKHPKP\ntranslation: WKVFTWTSFFIQMLKHEKHPKP\ntranslation: GLLIRSEFIINVSLELSFRIL\ntranslation: GLLIRSEFIINVSLELSFRIL\ntranslation: KILKLNSSDTLMINSDLIKSP\ntranslation: KILKLNSSDTLMINSDLIKSP\ntranslation: FDQVPPPATM\ntranslation: FDQVPPPATM\ntranslation: LTRCHHLLPC\ntranslation: LTRCHHLLPC\ntranslation: HMVAGGGTWSK\ntranslation: HMVAGGGTWSK\ntranslation: HGSRWWHLVK\ntranslation: HGSRWWHLVK\ntranslation: QRPTNHHPTMTNQTNLKTNDNHT\ntranslation: QRPTNHHPTMTNQTNLKTNDNHT\ntranslation: CMVIICFEVSLISHCWVVISRSL\ntranslation: CMVIICFEVSLISHCWVVISRSL\ntranslation: PCYIMLGYNFSSFPCGTISIAPG\ntranslation: PCYIMLGYNFSSFPCGTISIAPG\ntranslation: LAILCLVIIFHLSLAVLYLLRQV\ntranslation: LAILCLVIIFHLSLAVLYLLRQV\ntranslation: FYCLKIIIEASKNIQAKTSQF\ntranslation: FYCLKIIIEASKNIQAKTSQF\ntranslation: LLFKNHNRSFKKHTSQDIPV\ntranslation: LLFKNHNRSFKKHTSQDIPV\ntranslation: QIKHILANFKNYQFFIGENMN\ntranslation: QIKHILANFKNYQFFIGENMN\ntranslation: IHVFTNKELVVFEISKDVLDL\ntranslation: IHVFTNKELVVFEISKDVLDL\ntranslation: QQQNNNKACSVLRHCQD\ntranslation: QQQNNNKACSVLRHCQD\ntranslation: TTEQQQSMLSIETLSRL\ntranslation: TTEQQQSMLSIETLSRL\ntranslation: NLDSVSILSMLCCCSVV\ntranslation: NLDSVSILSMLCCCSVV\ntranslation: ILPSKLKLRTHSQSHHNPLSRT\ntranslation: ILPSKLKLRTHSQSHHNPLSRT\ntranslation: FCLANSNYERTHSRIIILSQGL\ntranslation: FCLANSNYERTHSRIIILSQGL\ntranslation: PVLHTILLAPLLPFLSTLLLASE\ntranslation: PVLHTILLAPLLPFLSTLLLASE\ntranslation: CLAYHSSSSFIAFSFHSASCFR\ntranslation: CLAYHSSSSFIAFSFHSASCFR\ntranslation: SRKQEAEWKEKAIKELEEWYARQ\ntranslation: SRKQEAEWKEKAIKELEEWYARQ\ntranslation: SEARSRVERKGNKGARRMVCKT\ntranslation: SEARSRVERKGNKGARRMVCKT\ntranslation: FVSGVLFFVTLSSSRLYINRH\ntranslation: FVSGVLFFVTLSSSRLYINRH\ntranslation: PCLLIYSLDDERVTKNKTPDT\ntranslation: PCLLIYSLDDERVTKNKTPDT\ntranslation: VDDSVSVNDIDLSLYIPRDVD\ntranslation: VDDSVSVNDIDLSLYIPRDVD\ntranslation: HLHLLGCTGLNLCHLHSLNHQ\ntranslation: HLHLLGCTGLNLCHLHSLNHQ\ntranslation: STSLGMYRLKSMSFTLTESS\ntranslation: STSLGMYRLKSMSFTLTESS\ntranslation: NLQRVSYLVFDEADRMFDMGF\ntranslation: NLQRVSYLVFDEADRMFDMGF\ntranslation: IFKESLTLCLMKQIECLTWDL\ntranslation: IFKESLTLCLMKQIECLTWDL\ntranslation: QIPCQTFDLLHQTQGKRLFED\ntranslation: QIPCQTFDLLHQTQGKRLFED\ntranslation: KSHVKHSICFIKHKVRDSLKI\ntranslation: KSHVKHSICFIKHKVRDSLKI\ntranslation: RHLHFLPPHSFLPHRCPHLAGCL\ntranslation: RHLHFLPPHSFLPHRCPHLAGCL\ntranslation: DIFTSFLLIVSFLIDVLILLDAF\ntranslation: DIFTSFLLIVSFLIDVLILLDAF\ntranslation: EGIQQDEDIYEEGNYEEEGSEDV\ntranslation: EGIQQDEDIYEEGNYEEEGSEDV\ntranslation: KASSKMRTSMRKETMRRKEVKMS\ntranslation: KASSKMRTSMRKETMRRKEVKMS\ntranslation: RRLLESLRGPALDVIRVLKINN\ntranslation: RRLLESLRGPALDVIRVLKINN\ntranslation: GLFILRTRITSSAGPRRLSSNRL\ntranslation: GLFILRTRITSSAGPRRLSSNRL\ntranslation: QFSSMSTLLQCLLGTSSKRNGPI\ntranslation: QFSSMSTLLQCLLGTSSKRNGPI\ntranslation: MGPFLFEEVPKRHCKSVDMELNW\ntranslation: MGPFLFEEVPKRHCKSVDMELNW\ntranslation: ILLPVFISWNFRVYVNMSSMTQV\ntranslation: ILLPVFISWNFRVYVNMSSMTQV\ntranslation: FVCSNCHFIFTLLDMGSSVIEVKN\ntranslation: FVCSNCHFIFTLLDMGSSVIEVKN\ntranslation: AFPLGLSVTSIHSTFSLHFQCPV\ntranslation: AFPLGLSVTSIHSTFSLHFQCPV\ntranslation: FPSRSLSHLYTLYIFSALSMPC\ntranslation: FPSRSLSHLYTLYIFSALSMPC\ntranslation: ENMSFKGIGAFEWNKCFHILX\ntranslation: ENMSFKGIGAFEWNKCFHILX\ntranslation: NCEVSLKDKLKI\ntranslation: NCEVSLKDKLKI\ntranslation: LIFNLSFSDTSQ\ntranslation: LIFNLSFSDTSQ\ntranslation: SYHYHTNVEELTIPETRNNLYI\ntranslation: SYHYHTNVEELTIPETRNNLYI\ntranslation: QQILHNQKGQASSTQQCSSKR\ntranslation: QQILHNQKGQASSTQQCSSKR\ntranslation: SKFFTIKRDKHQARSNAAQNA\ntranslation: SKFFTIKRDKHQARSNAAQNA\ntranslation: ANSSQSKGTSIKHAAMQLKT\ntranslation: ANSSQSKGTSIKHAAMQLKT\ntranslation: SVLSCIAACLMLVPFDCEEFA\ntranslation: SVLSCIAACLMLVPFDCEEFA\ntranslation: LKHRCNQIAFNKQKSFYAWISFI\ntranslation: LKHRCNQIAFNKQKSFYAWISFI\ntranslation: NERDPGIKTLLLIESNLITPMFQ\ntranslation: NERDPGIKTLLLIESNLITPMFQ\ntranslation: KKREILARKKSKRSALENSEEHS\ntranslation: KKREILARKKSKRSALENSEEHS\ntranslation: AECSSLFSRADRFDFFRAKISLF\ntranslation: AECSSLFSRADRFDFFRAKISLF\ntranslation: LNALHYFQEQIVLTSFELKSLFF\ntranslation: LNALHYFQEQIVLTSFELKSLFF\ntranslation: ICLMLGEKCCFWVNRLGKVQDH\ntranslation: ICLMLGEKCCFWVNRLGKVQDH\ntranslation: WSWTFPNLLTQKQHFSPSIRQI\ntranslation: WSWTFPNLLTQKQHFSPSIRQI\ntranslation: FLETQGTGPSETSSVFSSVSF\ntranslation: FLETQGTGPSETSSVFSSVSF\ntranslation: FWKPRELVLLKLVLCSLQYHL\ntranslation: FWKPRELVLLKLVLCSLQYHL\ntranslation: PNDTEENTELVSEGPVPWVSKN\ntranslation: PNDTEENTELVSEGPVPWVSKN\ntranslation: KDEAFSFQVSREAS\ntranslation: KDEAFSFQVSREAS\ntranslation: RMKPSPSRFPEKP\ntranslation: RMKPSPSRFPEKP\ntranslation: GGFSGNLEGEGFIL\ntranslation: GGFSGNLEGEGFIL\ntranslation: EASLETWKEKASSL\ntranslation: EASLETWKEKASSL\ntranslation: RLLWKPGRRRLHP\ntranslation: RLLWKPGRRRLHP\ntranslation: LLVYNNAQGLDELHNQNQMTAF\ntranslation: LLVYNNAQGLDELHNQNQMTAF\ntranslation: RKLSFDFDYVVHPALGHCYTPVK\ntranslation: RKLSFDFDYVVHPALGHCYTPVK\ntranslation: AASSCLHSAETQRAQRNLCSGL\ntranslation: AASSCLHSAETQRAQRNLCSGL\ntranslation: LLLAVCTVQKHREHRGICALGY\ntranslation: LLLAVCTVQKHREHRGICALGY\ntranslation: HSPEHKFLCALCVSALCRQLEA\ntranslation: HSPEHKFLCALCVSALCRQLEA\ntranslation: VDSEGNNDLLVDCRYWAVNCQF\ntranslation: VDSEGNNDLLVDCRYWAVNCQF\ntranslation: LTVRVIMTCWLIVDIGLLIVS\ntranslation: LTVRVIMTCWLIVDIGLLIVS\ntranslation: ELTINSPISTINQQVIITLTVN\ntranslation: ELTINSPISTINQQVIITLTVN\ntranslation: PNWHCISKLITRHRTTR\ntranslation: PNWHCISKLITRHRTTR\ntranslation: LTGIVLANSSLDIVLHD\ntranslation: LTGIVLANSSLDIVLHD\ntranslation: SCSTMSSDEFANTMPVR\ntranslation: SCSTMSSDEFANTMPVR\ntranslation: RVVRCLVMSLLIQCQL\ntranslation: RVVRCLVMSLLIQCQL\ntranslation: TLPDFKLYYKAIVTKTEWHWHK\ntranslation: TLPDFKLYYKAIVTKTEWHWHK\ntranslation: QESCADFAIAQ\ntranslation: QESCADFAIAQ\ntranslation: RNPVQTLLLPR\ntranslation: RNPVQTLLLPR\ntranslation: GILCRLCYCP\ntranslation: GILCRLCYCP\ntranslation: LGNSKVCTGFL\ntranslation: LGNSKVCTGFL\ntranslation: WAIAKSAQDS\ntranslation: WAIAKSAQDS\ntranslation: SALLFLCLDMFRYTNTYYCVTTA\ntranslation: SALLFLCLDMFRYTNTYYCVTTA\ntranslation: FINKYKHLRRSGNRNRLNYPAR\ntranslation: FINKYKHLRRSGNRNRLNYPAR\ntranslation: GLLITTVLAKNLLLLSYMQMRK\ntranslation: GLLITTVLAKNLLLLSYMQMRK\ntranslation: EYVSWHKSSAQHRNFPAEVN\ntranslation: EYVSWHKSSAQHRNFPAEVN\ntranslation: FTSAGKFLCCADDLCQDTYS\ntranslation: FTSAGKFLCCADDLCQDTYS\ntranslation: SPLPGSSCVVLMICARIHIQ\ntranslation: SPLPGSSCVVLMICARIHIQ\ntranslation: LVQLSSGKHCIRPGILLLFNSR\ntranslation: LVQLSSGKHCIRPGILLLFNSR\ntranslation: LKGFIKEIKKNFCIFYPGSTA\ntranslation: LKGFIKEIKKNFCIFYPGSTA\ntranslation: RSAPRIKDTKVLFYFFNKALQ\ntranslation: RSAPRIKDTKVLFYFFNKALQ\ntranslation: VNGPRKQVSGPERTIPIT\ntranslation: VNGPRKQVSGPERTIPIT\ntranslation: GDGNGALRARDLFPWSIH\ntranslation: GDGNGALRARDLFPWSIH\ntranslation: VMGMVRSGPETCFLGPFT\ntranslation: VMGMVRSGPETCFLGPFT\ntranslation: PRGNRTASGYRNNNVKETLISVM\ntranslation: PRGNRTASGYRNNNVKETLISVM\ntranslation: PEVTEQLQVIEITMLRKLLSQS\ntranslation: PEVTEQLQVIEITMLRKLLSQS\ntranslation: ANALYPFSSLICSFFMILSLQ\ntranslation: ANALYPFSSLICSFFMILSLQ\ntranslation: LQTKYHEERADQRAEGIKSIC\ntranslation: LQTKYHEERADQRAEGIKSIC\ntranslation: THSFLLLALSSTIFLNFDTPSI\ntranslation: THSFLLLALSSTIFLNFDTPSI\ntranslation: IGVGLGYVRSLGHMCWRLR\ntranslation: IGVGLGYVRSLGHMCWRLR\ntranslation: VSISNTYGLDYVHNLNLLQC\ntranslation: VSISNTYGLDYVHNLNLLQC\ntranslation: GTVMRSLGQNPTEAELQDMIN\ntranslation: GTVMRSLGQNPTEAELQDMIN\ntranslation: SLIMSCNSASVGFCPRDLITVP\ntranslation: SLIMSCNSASVGFCPRDLITVP\ntranslation: VKSIVASTFIISLFPTTIFMCLD\ntranslation: VKSIVASTFIISLFPTTIFMCLD\ntranslation: KIHCRIHLYYQSLPHNNIHVPR\ntranslation: KIHCRIHLYYQSLPHNNIHVPR\ntranslation: SRHMNIVVGKRLIIKVDATMDF\ntranslation: SRHMNIVVGKRLIIKVDATMDF\ntranslation: GLILETFKCSLIVFQLFGSCIGS\ntranslation: GLILETFKCSLIVFQLFGSCIGS\ntranslation: YNTIHHLFLQVSTYSSQLGKR\ntranslation: YNTIHHLFLQVSTYSSQLGKR\ntranslation: ITLYIIFSCKYLHILASLGRG\ntranslation: ITLYIIFSCKYLHILASLGRG\ntranslation: PLPKLARICRYLQEKMMYSVI\ntranslation: PLPKLARICRYLQEKMMYSVI\ntranslation: QDLKTGTVKVTAVGWSEHPP\ntranslation: QDLKTGTVKVTAVGWSEHPP\ntranslation: GFKNWNGEGDSSRLERASP\ntranslation: GFKNWNGEGDSSRLERASP\ntranslation: LGDARSNRLLSPSPFQFLNP\ntranslation: LGDARSNRLLSPSPFQFLNP\ntranslation: GGCSLQPTAVTFTVPVFKS\ntranslation: GGCSLQPTAVTFTVPVFKS\ntranslation: SGADHTPSAWIPDHCFLDKNHL\ntranslation: SGADHTPSAWIPDHCFLDKNHL\ntranslation: KWFLSKKQWSGIQAEGVWSAP\ntranslation: KWFLSKKQWSGIQAEGVWSAP\ntranslation: NGSYLRSNGLESRQKACGQHQ\ntranslation: NGSYLRSNGLESRQKACGQHQ\ntranslation: SAWKAIVVQVTFDLLTFDLEHR\ntranslation: SAWKAIVVQVTFDLLTFDLEHR\ntranslation: ILCSKSKVKRSKVTCTTIAFHAE\ntranslation: ILCSKSKVKRSKVTCTTIAFHAE\ntranslation: FCVLNQKSKDQKSLAQLLLSMQN\ntranslation: FCVLNQKSKDQKSLAQLLLSMQN\ntranslation: MSDFRSVCRRQMELVIIMPHR\ntranslation: MSDFRSVCRRQMELVIIMPHR\ntranslation: PMRHNYNKLHLPTTNRPKIAH\ntranslation: PMRHNYNKLHLPTTNRPKIAH\ntranslation: QKPFSIECDKISFKFCLLVSY\ntranslation: QKPFSIECDKISFKFCLLVSY\ntranslation: VTTLQLFNRGAAGSVAEW\ntranslation: VTTLQLFNRGAAGSVAEW\ntranslation: PLCNTACCTSIKELQGGH\ntranslation: PLCNTACCTSIKELQGGH\ntranslation: HSATLPAAPLLKSCKVVT\ntranslation: HSATLPAAPLLKSCKVVT\ntranslation: EGETVPLLRDTMTLMPVSMKG\ntranslation: EGETVPLLRDTMTLMPVSMKG\ntranslation: VPFIETGISVMVSRSNGTVSPS\ntranslation: VPFIETGISVMVSRSNGTVSPS\ntranslation: ALHRDRHQCHGVTQQWDCLTF\ntranslation: ALHRDRHQCHGVTQQWDCLTF\ntranslation: QELAPNTTWLSLPLPVRTKECDV\ntranslation: QELAPNTTWLSLPLPVRTKECDV\ntranslation: RNLLQILLGYPFLYQLELKSVMY\ntranslation: RNLLQILLGYPFLYQLELKSVMY\ntranslation: HTSHSLVLTGKGRDNQVVFGASS\ntranslation: HTSHSLVLTGKGRDNQVVFGASS\ntranslation: KAVTFKLQKYRRINDCSPSSMLY\ntranslation: KAVTFKLQKYRRINDCSPSSMLY\ntranslation: KQLHSSYRNIEELMIVHQAACCT\ntranslation: KQLHSSYRNIEELMIVHQAACCT\ntranslation: MYSMLLGEQSLILRYFCSLNVTA\ntranslation: MYSMLLGEQSLILRYFCSLNVTA\ntranslation: QGHLLGSSETSISKSRQNDLI\ntranslation: QGHLLGSSETSISKSRQNDLI\ntranslation: QIRSFCLDFDMDVSEDPSRCP\ntranslation: QIRSFCLDFDMDVSEDPSRCP\ntranslation: KLGHSVLILIWMFQKTLADVP\ntranslation: KLGHSVLILIWMFQKTLADVP\ntranslation: EKGLKNVKNEYEQLNYAKQLKE\ntranslation: EKGLKNVKNEYEQLNYAKQLKE\ntranslation: FQNMKRFTISRCIHSEGHTGV\ntranslation: FQNMKRFTISRCIHSEGHTGV\ntranslation: PKYEKIHNFKVHTFRGPHWC\ntranslation: PKYEKIHNFKVHTFRGPHWC\ntranslation: TPVWPSECMHLEIVNLFIFW\ntranslation: TPVWPSECMHLEIVNLFIFW\ntranslation: NPILESISTIGFTTSMLDQDIPM\ntranslation: NPILESISTIGFTTSMLDQDIPM\ntranslation: SYSRVHINNRVYDLDVGSGHPD\ntranslation: SYSRVHINNRVYDLDVGSGHPD\ntranslation: KILRSDITSLPLSSMGYFPEFRI\ntranslation: KILRSDITSLPLSSMGYFPEFRI\ntranslation: ILNSGKYPIEESGSEVISLLRI\ntranslation: ILNSGKYPIEESGSEVISLLRI\ntranslation: LVCYIDIALHVSTGKYLDFNDSL\ntranslation: LVCYIDIALHVSTGKYLDFNDSL\ntranslation: QAIIEIQIFPSRDMQSNVNVTYK\ntranslation: QAIIEIQIFPSRDMQSNVNVTYK\ntranslation: QKFLLLLPGLETVEMMEE\ntranslation: QKFLLLLPGLETVEMMEE\ntranslation: DSSIISTVSKPGNRSRNF\ntranslation: DSSIISTVSKPGNRSRNF\ntranslation: IPPSFQQFPNQATGAEIF\ntranslation: IPPSFQQFPNQATGAEIF\ntranslation: FLHHFNSFQTRQQEQKFL\ntranslation: FLHHFNSFQTRQQEQKFL\ntranslation: SPFHKSGAFWEAETFRSQSNF\ntranslation: SPFHKSGAFWEAETFRSQSNF\ntranslation: PHFISQVHSGKQKHLEVKAIL\ntranslation: PHFISQVHSGKQKHLEVKAIL\ntranslation: SECLCPPKIHMLKPAYQC\ntranslation: SECLCPPKIHMLKPAYQC\ntranslation: HIGRQALTYEFWGDTSIQT\ntranslation: HIGRQALTYEFWGDTSIQT\ntranslation: SQTLIKPSDVMRTHLLSQEQHR\ntranslation: SQTLIKPSDVMRTHLLSQEQHR\ntranslation: VRFAEFLLLFLTFPYEHACVGL\ntranslation: VRFAEFLLLFLTFPYEHACVGL\ntranslation: STQHRHAHKERLKKVKGTRQIL\ntranslation: STQHRHAHKERLKKVKGTRQIL\ntranslation: SDFFMWVSSSSSRWSPKIDCSFN\ntranslation: SDFFMWVSSSSSRWSPKIDCSFN\ntranslation: LKEQSIFGDHRDEEEETHMKKS\ntranslation: LKEQSIFGDHRDEEEETHMKKS\ntranslation: DILRDPCSATDPLCRSCSAIDSL\ntranslation: DILRDPCSATDPLCRSCSAIDSL\ntranslation: QRVYCRTRSTQRVSSRTRITQNV\ntranslation: QRVYCRTRSTQRVSSRTRITQNV\ntranslation: RESIAEQDRHKGSVAEQGSRRMS\ntranslation: RESIAEQDRHKGSVAEQGSRRMS\ntranslation: CNMKVICLIEISSCLVKHT\ntranslation: CNMKVICLIEISSCLVKHT\ntranslation: SVCLTRQLDISIKQMTFILQ\ntranslation: SVCLTRQLDISIKQMTFILQ\ntranslation: LDDQFNQAYIWRQILVSRERRS\ntranslation: LDDQFNQAYIWRQILVSRERRS\ntranslation: TQMSHSWGMDTQTVVILGWCKS\ntranslation: TQMSHSWGMDTQTVVILGWCKS\ntranslation: KARIVEYEKEMEKMKNLIPFD\ntranslation: KARIVEYEKEMEKMKNLIPFD\ntranslation: DQMELSSSSSPSLSHILQSWPL\ntranslation: DQMELSSSSSPSLSHILQSWPL\ntranslation: SNGIKFFIFSISFSYSTILAF\ntranslation: SNGIKFFIFSISFSYSTILAF\ntranslation: AAQRLRSYTVNLQMLTQCYTVNF\ntranslation: AAQRLRSYTVNLQMLTQCYTVNF\ntranslation: CTKIKVIYSQSANVDTMLHCKF\ntranslation: CTKIKVIYSQSANVDTMLHCKF\ntranslation: LNGVLVTVAFLVSMYLFCLFFGD\ntranslation: LNGVLVTVAFLVSMYLFCLFFGD\ntranslation: KWCFSNCCISGFHVLILPFFWRF\ntranslation: KWCFSNCCISGFHVLILPFFWRF\ntranslation: ESPKKRQNKYMETRNATVTKTPF\ntranslation: ESPKKRQNKYMETRNATVTKTPF\ntranslation: NRQKKGRISTWKPEMQQLLKHHL\ntranslation: NRQKKGRISTWKPEMQQLLKHHL\ntranslation: RGYLQNCHLTMTNQTNLKT\ntranslation: RGYLQNCHLTMTNQTNLKT\ntranslation: AFGTSLFVPPSHIRFVFPEP\ntranslation: AFGTSLFVPPSHIRFVFPEP\ntranslation: PSAPASSCPQAISALCFLSQ\ntranslation: PSAPASSCPQAISALCFLSQ\ntranslation: VGSGNTKRIWLGGTKRLVPK\ntranslation: VGSGNTKRIWLGGTKRLVPK\ntranslation: LAQETQSGYGLGARRGWCRR\ntranslation: LAQETQSGYGLGARRGWCRR\ntranslation: WLRKHKADMAWGHEEAGAEG\ntranslation: WLRKHKADMAWGHEEAGAEG\ntranslation: SALHYSYIHSFFHLGSFSLSNVRG\ntranslation: SALHYSYIHSFFHLGSFSLSNVRG\ntranslation: CTSLLIHPFFLSFRFFFIVKCQRH\ntranslation: CTSLLIHPFFLSFRFFFIVKCQRH\ntranslation: FQFILHYHLTLCLNIHVKKVFP\ntranslation: FQFILHYHLTLCLNIHVKKVFP\ntranslation: SNLSCIIILPSVSIYMLRRSFP\ntranslation: SNLSCIIILPSVSIYMLRRSFP\ntranslation: GKDLLNMYIETEGKMIMQDKLE\ntranslation: GKDLLNMYIETEGKMIMQDKLE\ntranslation: HTYRTSLYIHGMKCHSFIYTI\ntranslation: HTYRTSLYIHGMKCHSFIYTI\ntranslation: LEISSFSLLHSSAGNWSLNSSIL\ntranslation: LEISSFSLLHSSAGNWSLNSSIL\ntranslation: KMEEFKDQLPADECNKLKEEISK\ntranslation: KMEEFKDQLPADECNKLKEEISK\ntranslation: IFGLLKLMFEDRLLSLLEGFA\ntranslation: IFGLLKLMFEDRLLSLLEGFA\ntranslation: MANPSSRDKSRSSNINLSRPKI\ntranslation: MANPSSRDKSRSSNINLSRPKI\ntranslation: QNVGCFFHKIISIGGYCSMTA\ntranslation: QNVGCFFHKIISIGGYCSMTA\ntranslation: REELKALLNNVFANDFQDFALL\ntranslation: REELKALLNNVFANDFQDFALL\ntranslation: EKSSRHFLTMFLRMTFKILLCW\ntranslation: EKSSRHFLTMFLRMTFKILLCW\ntranslation: QQSKILKVIRKNIVKKCLELFS\ntranslation: QQSKILKVIRKNIVKKCLELFS\ntranslation: LGSFLMVIFFASSRFFSLRKLSS\ntranslation: LGSFLMVIFFASSRFFSLRKLSS\ntranslation: QKIVYEEKRTWKKQRRLPLKMIQ\ntranslation: QKIVYEEKRTWKKQRRLPLKMIQ\ntranslation: EDSLRREKNLEEAKKITIKNDP\ntranslation: EDSLRREKNLEEAKKITIKNDP\ntranslation: SVQCFNLTQAYAEENVF\ntranslation: SVQCFNLTQAYAEENVF\ntranslation: RKYVPAFRRSGWLPHRVM\ntranslation: RKYVPAFRRSGWLPHRVM\ntranslation: HHPMRQPARTPERRHILP\ntranslation: HHPMRQPARTPERRHILP\ntranslation: VNCAFSFNFLSLLKLKICPTFHI\ntranslation: VNCAFSFNFLSLLKLKICPTFHI\ntranslation: DDYDKKKLVKAFKKKFACNGTVIE\ntranslation: DDYDKKKLVKAFKKKFACNGTVIE\ntranslation: IGVYQCLKNVSRNIAEGSSEVG\ntranslation: IGVYQCLKNVSRNIAEGSSEVG\ntranslation: LVSINVLKMFQETLLRVAVKL\ntranslation: LVSINVLKMFQETLLRVAVKL\ntranslation: NPPLGINILCHNKHPFSFTEIS\ntranslation: NPPLGINILCHNKHPFSFTEIS\ntranslation: IIKVSIHQGITVFNVYAPNDRAT\ntranslation: IIKVSIHQGITVFNVYAPNDRAT\ntranslation: ATSTYRQHLHSNKHQFKIIRQIF\ntranslation: ATSTYRQHLHSNKHQFKIIRQIF\ntranslation: LGFPAQSPASQ\ntranslation: LGFPAQSPASQ\ntranslation: WGFLLNHLHLR\ntranslation: WGFLLNHLHLR\ntranslation: GVSCSITCISD\ntranslation: GVSCSITCISD\ntranslation: SEMQVIEQETP\ntranslation: SEMQVIEQETP\ntranslation: LSSVSIFSLYSRAICCFFSAP\ntranslation: LSSVSIFSLYSRAICCFFSAP\ntranslation: LAPSQFSLCILEPSAVFSQHL\ntranslation: LAPSQFSLCILEPSAVFSQHL\ntranslation: GAEKKQQMAREYREKIETEL\ntranslation: GAEKKQQMAREYREKIETEL\ntranslation: HLSILFSIREIVGIRIRIVVKY\ntranslation: HLSILFSIREIVGIRIRIVVKY\ntranslation: YFTTILILIPTISLIENKILK\ntranslation: YFTTILILIPTISLIENKILK\ntranslation: QNEPIPERSPQNGESGKDMDDDK\ntranslation: QNEPIPERSPQNGESGKDMDDDK\ntranslation: RMNLFQKDLLRMVNQERIWMMTK\ntranslation: RMNLFQKDLLRMVNQERIWMMTK\ntranslation: GVIEDEQVLVLVFPKGKDKGVQD\ntranslation: GVIEDEQVLVLVFPKGKDKGVQD\ntranslation: ILDTLVFAFGKYQHKNLLILYDA\ntranslation: ILDTLVFAFGKYQHKNLLILYDA\ntranslation: SWTPLSLPLGNTSTRTCSSSMTP\ntranslation: SWTPLSLPLGNTSTRTCSSSMTP\ntranslation: LTSHRSFRSFWMSHRVSALFLS\ntranslation: LTSHRSFRSFWMSHRVSALFLS\ntranslation: AQEKCRYPVGHPEGPKRPVTSE\ntranslation: AQEKCRYPVGHPEGPKRPVTSE\ntranslation: LLINSIVSNYFCKYIIFPNWFL\ntranslation: LLINSIVSNYFCKYIIFPNWFL\ntranslation: KGSFHQCVTVRNKLMSS\ntranslation: KGSFHQCVTVRNKLMSS\ntranslation: KVHFTNVSQSETNSCL\ntranslation: KVHFTNVSQSETNSCL\ntranslation: RFISPMCHSQKQTHVF\ntranslation: RFISPMCHSQKQTHVF\ntranslation: SHVILLILMLCGTGSSSISAMPL\ntranslation: SHVILLILMLCGTGSSSISAMPL\ntranslation: TRYIAYFDVMWYRKFFHLCYAT\ntranslation: TRYIAYFDVMWYRKFFHLCYAT\ntranslation: SGIAEMEELPVPHNIKISNITC\ntranslation: SGIAEMEELPVPHNIKISNITC\ntranslation: STFNPPKPTKTRSPSQASLPIC\ntranslation: STFNPPKPTKTRSPSQASLPIC\ntranslation: PHSIHPNPQRPGLQARPLFPS\ntranslation: PHSIHPNPQRPGLQARPLFPS\ntranslation: HIQSTQTHKDQVSKPGLSSHL\ntranslation: HIQSTQTHKDQVSKPGLSSHL\ntranslation: ADGKRGLAWRPGLCGFGWIECG\ntranslation: ADGKRGLAWRPGLCGFGWIECG\ntranslation: QMGREAWLGDLVFVGLGGLNV\ntranslation: QMGREAWLGDLVFVGLGGLNV\ntranslation: QGRHCGLFSLNMLKWKDFPSVT\ntranslation: QGRHCGLFSLNMLKWKDFPSVT\ntranslation: VTLGKSFHFSMFKENSPQCLP\ntranslation: VTLGKSFHFSMFKENSPQCLP\ntranslation: FIGLTCIKIPCECAHSHISPTTI\ntranslation: FIGLTCIKIPCECAHSHISPTTI\ntranslation: AVKRLLDSSQCQQMSQAPTKYNH\ntranslation: AVKRLLDSSQCQQMSQAPTKYNH\ntranslation: QLRDFWTVHSVNKCHRLRPSITT\ntranslation: QLRDFWTVHSVNKCHRLRPSITT\ntranslation: EESHGIVANSMYDAVTKKHFSDS\ntranslation: EESHGIVANSMYDAVTKKHFSDS\ntranslation: KKAMALWLIPCMMQSQRNTFLT\ntranslation: KKAMALWLIPCMMQSQRNTFLT\ntranslation: RVRKVFLCDCIIHGISHNAMAFF\ntranslation: RVRKVFLCDCIIHGISHNAMAFF\ntranslation: ESEKCFFVTASYMELATMPWLSS\ntranslation: ESEKCFFVTASYMELATMPWLSS\ntranslation: LKCWLYVVNVRVIQSLVLSSLFW\ntranslation: LKCWLYVVNVRVIQSLVLSSLFW\ntranslation: EMLVVCGKCKSNTVSCTFLTVL\ntranslation: EMLVVCGKCKSNTVSCTFLTVL\ntranslation: PKTVRKVQETVLLLHLPHTTNIS\ntranslation: PKTVRKVQETVLLLHLPHTTNIS\ntranslation: QNSEESTRDCITLTFTTYNQHF\ntranslation: QNSEESTRDCITLTFTTYNQHF\ntranslation: IPHHPTKNIKHKL\ntranslation: IPHHPTKNIKHKL\ntranslation: FLITQLKTLNTNY\ntranslation: FLITQLKTLNTNY\ntranslation: HILPILHPSRLPSPTHRTNLHS\ntranslation: HILPILHPSRLPSPTHRTNLHS\ntranslation: TYFLFYTLVGSLPLLIALIYTH\ntranslation: TYFLFYTLVGSLPLLIALIYTH\ntranslation: FLSSHLDLLNLYFYLRLIYSTSI\ntranslation: FLSSHLDLLNLYFYLRLIYSTSI\ntranslation: ISAAELRHVMTNLGEKLTDEEVD\ntranslation: ISAAELRHVMTNLGEKLTDEEVD\ntranslation: PSLVSSTS\ntranslation: PSLVSSTS\ntranslation: IFSFLYLL\ntranslation: IFSFLYLL\ntranslation: QEVEETKD\ntranslation: QEVEETKD\ntranslation: GSIFCALITMLGHIRIGHGNRMH\ntranslation: GSIFCALITMLGHIRIGHGNRMH\ntranslation: RHLLYHLCRHTHHSAKLALIFYL\ntranslation: RHLLYHLCRHTHHSAKLALIFYL\ntranslation: RNQISTSTNGKLTTYEPSSNS\ntranslation: RNQISTSTNGKLTTYEPSSNS\ntranslation: PTSNRNSDSVLVCLVLCINGML\ntranslation: PTSNRNSDSVLVCLVLCINGML\ntranslation: HNIPFIHRTKQTSTESLLRLEV\ntranslation: HNIPFIHRTKQTSTESLLRLEV\ntranslation: FYQRAELRTLWLKYNFSP\ntranslation: FYQRAELRTLWLKYNFSP\ntranslation: GEKNYISATVFSTQPFDK\ntranslation: GEKNYISATVFSTQPFDK\ntranslation: GRKIIFQPQCSQLSPLIK\ntranslation: GRKIIFQPQCSQLSPLIK\ntranslation: LNYNAYFRMRNISFLPLKCHIEND\ntranslation: LNYNAYFRMRNISFLPLKCHIEND\ntranslation: DNIIFIIPLWFWFSCWKISEC\ntranslation: DNIIFIIPLWFWFSCWKISEC\ntranslation: TISFSLSHSGSGSPAGKSLN\ntranslation: TISFSLSHSGSGSPAGKSLN\ntranslation: AFRDFPAGEPEPEWDNENDIV\ntranslation: AFRDFPAGEPEPEWDNENDIV\ntranslation: HSEIFQQENQNQSGIMKMILS\ntranslation: HSEIFQQENQNQSGIMKMILS\ntranslation: LLHRLPGVLTRVHRCVLHRHLVI\ntranslation: LLHRLPGVLTRVHRCVLHRHLVI\ntranslation: CYIASLVFLHEFIDVYFIATLSF\ntranslation: CYIASLVFLHEFIDVYFIATLSF\ntranslation: ATSPPWCSYTSSSMCTSSPPCHS\ntranslation: ATSPPWCSYTSSSMCTSSPPCHS\ntranslation: CLITDGFCYKHQCDTCRTLQR\ntranslation: CLITDGFCYKHQCDTCRTLQR\ntranslation: PGLTMCLLQESL\ntranslation: PGLTMCLLQESL\ntranslation: PGSLCACSRSP\ntranslation: PGSLCACSRSP\ntranslation: RAHYVPAPGVP\ntranslation: RAHYVPAPGVP\ntranslation: QGLLEQAHSEPG\ntranslation: QGLLEQAHSEPG\ntranslation: RDSWSRHIVSP\ntranslation: RDSWSRHIVSP\ntranslation: QRPVPGPMLVDFHVLISLMFTQP\ntranslation: QRPVPGPMLVDFHVLISLMFTQP\ntranslation: GCVNIKEINTWKSTNMGPGTGR\ntranslation: GCVNIKEINTWKSTNMGPGTGR\ntranslation: EKTSVCLFGLYWESLSISSSCES\ntranslation: EKTSVCLFGLYWESLSISSSCES\ntranslation: ELSQLLEMLNDSQYNPKRQTLVF\ntranslation: ELSQLLEMLNDSQYNPKRQTLVF\ntranslation: CMDILKVCCRISVCLSFSLPT\ntranslation: CMDILKVCCRISVCLSFSLPT\ntranslation: GVGRENDRQTDILQQTFRMSIH\ntranslation: GVGRENDRQTDILQQTFRMSIH\ntranslation: GVQRIVLTVGLNDFLSILQGSF\ntranslation: GVQRIVLTVGLNDFLSILQGSF\ntranslation: EGALQYGQKIIKPYSKHYPLYS\ntranslation: EGALQYGQKIIKPYSKHYPLYS\ntranslation: KEPCNMDRKSLSPTVSTILCT\ntranslation: KEPCNMDRKSLSPTVSTILCT\ntranslation: RVLRVSLHHFRRHLRLNIFCSH\ntranslation: RVLRVSLHHFRRHLRLNIFCSH\ntranslation: EYFESPFTISDGIYGSTFFVA\ntranslation: EYFESPFTISDGIYGSTFFVA\ntranslation: SCFSKRFPLLPLLPLSSSMSF\ntranslation: SCFSKRFPLLPLLPLSSSMSF\ntranslation: KLIEDERGRRGRRGNLLEKH\ntranslation: KLIEDERGRRGRRGNLLEKH\ntranslation: YVINWSPACQKDCSPLQLAHFQ\ntranslation: YVINWSPACQKDCSPLQLAHFQ\ntranslation: TENGPTEVDYSPSDMQDSSLSHK\ntranslation: TENGPTEVDYSPSDMQDSSLSHK\ntranslation: LKMGQLKWTTVLLTCRTPVYHI\ntranslation: LKMGQLKWTTVLLTCRTPVYHI\ntranslation: HGYGEISPRLSFHYHCLPGCAC\ntranslation: HGYGEISPRLSFHYHCLPGCAC\ntranslation: TAMGKLARGLAFIITVSQGVLV\ntranslation: TAMGKLARGLAFIITVSQGVLV\ntranslation: TSTPWETVIMKAKPRANFPIAV\ntranslation: TSTPWETVIMKAKPRANFPIAV\ntranslation: AHYANLFSTILSLKTLTMKGKG\ntranslation: AHYANLFSTILSLKTLTMKGKG\ntranslation: RDFIFGESATNQ\ntranslation: RDFIFGESATNQ\ntranslation: VISYLENQQPTS\ntranslation: VISYLENQQPTS\ntranslation: GWLVADSPNMKS\ntranslation: GWLVADSPNMKS\ntranslation: FEVSLISHCWVVISRLLMRYL\ntranslation: FEVSLISHCWVVISRLLMRYL\ntranslation: QISHQQPTNHHPTMTNQTNLK\ntranslation: QISHQQPTNHHPTMTNQTNLK\ntranslation: TKSCKSRGSNLRVHFKNTRETAQ\ntranslation: TKSCKSRGSNLRVHFKNTRETAQ\ntranslation: RNHANQEVPIFVFTLRTLVKLL\ntranslation: RNHANQEVPIFVFTLRTLVKLL\ntranslation: ANVSENITNKRFFFFFIQQI\ntranslation: ANVSENITNKRFFFFFIQQI\ntranslation: QMSLKISQIKDFFFSSFSK\ntranslation: QMSLKISQIKDFFFSSFSK\ntranslation: NLLNEEKKKSFICDIFRDIC\ntranslation: NLLNEEKKKSFICDIFRDIC\ntranslation: CASSKLWSVMVIICCKMSPMSC\ntranslation: CASSKLWSVMVIICCKMSPMSC\ntranslation: QDIGDILQQIMTITDQSLDEA\ntranslation: QDIGDILQQIMTITDQSLDEA\ntranslation: DSVSILSMLCCCSVVVAVVLC\ntranslation: DSVSILSMLCCCSVVVAVVLC\ntranslation: HTKQQQQQQNNNKACSVLRHC\ntranslation: HTKQQQQQQNNNKACSVLRHC\ntranslation: HKTTATTTEQQQSMLSIETLS\ntranslation: HKTTATTTEQQQSMLSIETLS\ntranslation: KSITNKTTYNSYFNTKVTASPFP\ntranslation: KSITNKTTYNSYFNTKVTASPFP\ntranslation: RERGSSDLSIKITVISCFVCYRF\ntranslation: RERGSSDLSIKITVISCFVCYRF\ntranslation: LIEEYAMSDFRSVCR\ntranslation: LIEEYAMSDFRSVCR\ntranslation: TTNRPKIAHCILFNQ\ntranslation: TTNRPKIAHCILFNQ\ntranslation: RQTDLKSLIAYSSIS\ntranslation: RQTDLKSLIAYSSIS\ntranslation: IMLGIFFNVHSAVLIEDVPF\ntranslation: IMLGIFFNVHSAVLIEDVPF\ntranslation: EGNVLNQHSGMDIEKYSEHY\ntranslation: EGNVLNQHSGMDIEKYSEHY\ntranslation: KGTSSINTAEWTLKNIPSII\ntranslation: KGTSSINTAEWTLKNIPSII\ntranslation: QFLVLHKLTPLTVQWSPIQTDR\ntranslation: QFLVLHKLTPLTVQWSPIQTDR\ntranslation: SDQFEWVTIEQSGELVYEAPETV\ntranslation: SDQFEWVTIEQSGELVYEAPETV\ntranslation: KDGKLQIGDKLLAVNNVCLE\ntranslation: KDGKLQIGDKLLAVNNVCLE\ntranslation: LLNIRYSLLKVYLQSEVCHP\ntranslation: LLNIRYSLLKVYLQSEVCHP\ntranslation: EEHQCLVQSHLLYQFLDHFPHL\ntranslation: EEHQCLVQSHLLYQFLDHFPHL\ntranslation: SEESDLEIDKEGVIEPDTDAPQ\ntranslation: SEESDLEIDKEGVIEPDTDAPQ\ntranslation: EEFKENAYKDLAWKHGNIHLS\ntranslation: EEFKENAYKDLAWKHGNIHLS\ntranslation: KNLKKMLIKTWHGSMETFTS\ntranslation: KNLKKMLIKTWHGSMETFTS\ntranslation: IEENVNSLSQISADLPAFVSVVA\ntranslation: IEENVNSLSQISADLPAFVSVVA\ntranslation: SFNGNKVTITLLPDNLSQTIYL\ntranslation: SFNGNKVTITLLPDNLSQTIYL\ntranslation: VSCTRQAW\ntranslation: VSCTRQAW\ntranslation: FPAPGKP\ntranslation: FPAPGKP\ntranslation: FLHQASL\ntranslation: FLHQASL\ntranslation: PGLPGAGN\ntranslation: PGLPGAGN\ntranslation: QACLVQE\ntranslation: QACLVQE\ntranslation: RLAWCRK\ntranslation: RLAWCRK\ntranslation: ILSVSQTPLCFSFVALFPGPHEK\ntranslation: ILSVSQTPLCFSFVALFPGPHEK\ntranslation: FSHEVQEIGLQRKNIKESGKLTE\ntranslation: FSHEVQEIGLQRKNIKESGKLTE\ntranslation: CGRHLAYFVD\ntranslation: CGRHLAYFVD\ntranslation: VVGIWHILWT\ntranslation: VVGIWHILWT\ntranslation: ESTKYAKCLPQ\ntranslation: ESTKYAKCLPQ\ntranslation: SPQNMPNAYH\ntranslation: SPQNMPNAYH\ntranslation: VHKICQMPTT\ntranslation: VHKICQMPTT\ntranslation: ITGILPVFQCNNCCHHFVFFML\ntranslation: ITGILPVFQCNNCCHHFVFFML\ntranslation: SQEYYQYFNAIIVVITLCFLCW\ntranslation: SQEYYQYFNAIIVVITLCFLCW\ntranslation: LRTKIILMYMYHTLICNLHTDA\ntranslation: LRTKIILMYMYHTLICNLHTDA\ntranslation: MHQCVDCISVYGTCTSKLFLSL\ntranslation: MHQCVDCISVYGTCTSKLFLSL\ntranslation: ASVCRLHISVWYMYIKIIFVL\ntranslation: ASVCRLHISVWYMYIKIIFVL\ntranslation: LAHLSGPTDPCSTAVHMEPFS\ntranslation: LAHLSGPTDPCSTAVHMEPFS\ntranslation: SPISQDRLTHVQLLFTWNPS\ntranslation: SPISQDRLTHVQLLFTWNPS\ntranslation: RRVPCEQQLNMGQSVLRDGR\ntranslation: RRVPCEQQLNMGQSVLRDGR\ntranslation: VVCSSKKGHGAILCKAMTQLHP\ntranslation: VVCSSKKGHGAILCKAMTQLHP\ntranslation: CMFFKERSWGNSVQGNDPAPS\ntranslation: CMFFKERSWGNSVQGNDPAPS\ntranslation: DGAGSLPCTELPHDLSLKNIQ\ntranslation: DGAGSLPCTELPHDLSLKNIQ\ntranslation: RDKTFKTSVCPHTASNLSTPPKV\ntranslation: RDKTFKTSVCPHTASNLSTPPKV\ntranslation: ETKLLKQVFVLILPQTLAHHQRY\ntranslation: ETKLLKQVFVLILPQTLAHHQRY\ntranslation: PTSTSLSFSLLTIYVFAFNIV\ntranslation: PTSTSLSFSLLTIYVFAFNIV\ntranslation: LLLLPFLSPSSQSMSSPSTS\ntranslation: LLLLPFLSPSSQSMSSPSTS\ntranslation: YFYFPFFLPPHNLCLRLQHR\ntranslation: YFYFPFFLPPHNLCLRLQHR\ntranslation: DDVEGEDIDCEEGERKGSRSR\ntranslation: DDVEGEDIDCEEGERKGSRSR\ntranslation: PAQQQVSSKAQITPPPVHIIDNS\ntranslation: PAQQQVSSKAQITPPPVHIIDNS\ntranslation: GIINDMHWWRCYLCFRRYLLLSW\ntranslation: GIINDMHWWRCYLCFRRYLLLSW\ntranslation: LNGQIQNKPQEKGVSTILNLPS\ntranslation: LNGQIQNKPQEKGVSTILNLPS\ntranslation: FEGRFKMVDTPFSWGLFCICPL\ntranslation: FEGRFKMVDTPFSWGLFCICPL\ntranslation: QVFFLVQSNSYKRLLYKDFLKLL\ntranslation: QVFFLVQSNSYKRLLYKDFLKLL\ntranslation: LRPISLPGVSSTEDLQDLFRKT\ntranslation: LRPISLPGVSSTEDLQDLFRKT\ntranslation: LVLCMVIICFEVSLISHCW\ntranslation: LVLCMVIICFEVSLISHCW\ntranslation: PETTKTPGIKGFKFNSHSLQIQ\ntranslation: PETTKTPGIKGFKFNSHSLQIQ\ntranslation: LYLERVTIELETFYSGRLGSFW\ntranslation: LYLERVTIELETFYSGRLGSFW\ntranslation: PLRPTHTPPMKKLPTTHPSITYM\ntranslation: PLRPTHTPPMKKLPTTHPSITYM\ntranslation: GATNGAESKTLIYQMAPAVSK\ntranslation: GATNGAESKTLIYQMAPAVSK\ntranslation: HSRPSRKHNVFVEFRSGINVSCL\ntranslation: HSRPSRKHNVFVEFRSGINVSCL\ntranslation: QAADIDTRSEFYKHIVLSGGSTM\ntranslation: QAADIDTRSEFYKHIVLSGGSTM\ntranslation: RQLTLIPDLNSTNTLCFLEGLLC\ntranslation: RQLTLIPDLNSTNTLCFLEGLLC\ntranslation: FIQHLYKNHQDSSWQVKMFLNS\ntranslation: FIQHLYKNHQDSSWQVKMFLNS\ntranslation: KYFTDLFDYLPLTALVDGQIF\ntranslation: KYFTDLFDYLPLTALVDGQIF\ntranslation: NILQIFLTIFLSLPWWMGRSS\ntranslation: NILQIFLTIFLSLPWWMGRSS\ntranslation: EDLPIHQGSERKIVKKICKIF\ntranslation: EDLPIHQGSERKIVKKICKIF\ntranslation: GVSLQELNPEMGTDNDSENWK\ntranslation: GVSLQELNPEMGTDNDSENWK\ntranslation: PSNFHYHCQFPFLDSIPGEKHL\ntranslation: PSNFHYHCQFPFLDSIPGEKHL\ntranslation: LPIFTIIVSSHFWIQFLERNT\ntranslation: LPIFTIIVSSHFWIQFLERNT\ntranslation: FQFSLSLSVPISGFNSWRETP\ntranslation: FQFSLSLSVPISGFNSWRETP\ntranslation: GDQKEECTSSGWAGFPLISL\ntranslation: GDQKEECTSSGWAGFPLISL\ntranslation: EGLSEGIQPNQMKYILLFGHP\ntranslation: EGLSEGIQPNQMKYILLFGHP\ntranslation: RLIRGNPAQPDEVHSSFWSP\ntranslation: RLIRGNPAQPDEVHSSFWSP\ntranslation: RFHISQMRLVVQFILQVYSEHF\ntranslation: RFHISQMRLVVQFILQVYSEHF\ntranslation: LVHIYTVLHSMICQGRDCGV\ntranslation: LVHIYTVLHSMICQGRDCGV\ntranslation: SPHNPSPDKSYYEVRCRCEQV\ntranslation: SPHNPSPDKSYYEVRCRCEQV\ntranslation: HPTIPPLTNHTMKYGVDVNK\ntranslation: HPTIPPLTNHTMKYGVDVNK\ntranslation: LHKLFVMVLKYLCIKVRINR\ntranslation: LHKLFVMVLKYLCIKVRINR\ntranslation: PIDSNLDAEILQDHHKQLMQ\ntranslation: PIDSNLDAEILQDHHKQLMQ\ntranslation: LLILTLMQRYFRTITNSLCK\ntranslation: LLILTLMQRYFRTITNSLCK\ntranslation: GRIHGLSPVTDESSIKLGLVL\ntranslation: GRIHGLSPVTDESSIKLGLVL\ntranslation: FSTRPNFMLLSSVTGLRPCIRP\ntranslation: FSTRPNFMLLSSVTGLRPCIRP\ntranslation: QLNKSGYIHYCESIHHSSVEIPE\ntranslation: QLNKSGYIHYCESIHHSSVEIPE\ntranslation: SSTSLGIFTIVSQYTIALLKFLK\ntranslation: SSTSLGIFTIVSQYTIALLKFLK\ntranslation: LTKRLQLGNYESVEDVSTFLSKQ\ntranslation: LTKRLQLGNYESVEDVSTFLSKQ\ntranslation: LSVMHFNSLKYHYTCSKDYQQM\ntranslation: LSVMHFNSLKYHYTCSKDYQQM\ntranslation: CLSCILIHSNIITLAVKIINKC\ntranslation: CLSCILIHSNIITLAVKIINKC\ntranslation: ESIYSYNLILLDQSPLHLQFGSN\ntranslation: ESIYSYNLILLDQSPLHLQFGSN\ntranslation: KKLTAMEFACTSILVNALIHLLN\ntranslation: KKLTAMEFACTSILVNALIHLLN\ntranslation: FNKWMRALTSMEVHANSIAVSF\ntranslation: FNKWMRALTSMEVHANSIAVSF\ntranslation: MACVCVRACVCSESTS\ntranslation: MACVCVRACVCSESTS\ntranslation: WHVCVCVRVYVLSPLL\ntranslation: WHVCVCVRVYVLSPLL\ntranslation: KKWTQNIHTHAHTHMP\ntranslation: KKWTQNIHTHAHTHMP\ntranslation: RSGLRTYTRTHTHTCH\ntranslation: RSGLRTYTRTHTHTCH\ntranslation: EVDSEHTHARTHTHA\ntranslation: EVDSEHTHARTHTHA\ntranslation: DKKQMLSNQASWRKANLTCKIAI\ntranslation: DKKQMLSNQASWRKANLTCKIAI\ntranslation: TKSRCSAIRPHGGKLISPAKLQS\ntranslation: TKSRCSAIRPHGGKLISPAKLQS\ntranslation: KVSSKNSLESYAFNMKATVED\ntranslation: KVSSKNSLESYAFNMKATVED\ntranslation: LCCQAIKFSFRSIRGQISSQSVSP\ntranslation: LCCQAIKFSFRSIRGQISSQSVSP\ntranslation: SAVRLSSSLLGAFEVKSVPRVSHL\ntranslation: SAVRLSSSLLGAFEVKSVPRVSHL\ntranslation: MCVSPCQSRWVIMKCQTSKINV\ntranslation: MCVSPCQSRWVIMKCQTSKINV\ntranslation: QRTMQNSIANWIYGWDSLLLLKV\ntranslation: QRTMQNSIANWIYGWDSLLLLKV\ntranslation: REPCKTQLQIGYMDGILYYYLKF\ntranslation: REPCKTQLQIGYMDGILYYYLKF\ntranslation: LLFCLPESAPCMVSCLSWKYYAC\ntranslation: LLFCLPESAPCMVSCLSWKYYAC\ntranslation: IVLFARKCTLYGLLSKLEILCM\ntranslation: IVLFARKCTLYGLLSKLEILCM\ntranslation: HMHNISNLDRRPYRVHFLANKTI\ntranslation: HMHNISNLDRRPYRVHFLANKTI\ntranslation: ILIVSKLFQSDESDTIEMKSHSS\ntranslation: ILIVSKLFQSDESDTIEMKSHSS\ntranslation: VTCFDLEGRPLCPVT\ntranslation: VTCFDLEGRPLCPVT\ntranslation: LLALIWKVVHCALSH\ntranslation: LLALIWKVVHCALSH\ntranslation: DVTGHSGRPSKSKQV\ntranslation: DVTGHSGRPSKSKQV\ntranslation: CDRAQWTTFQIKASN\ntranslation: CDRAQWTTFQIKASN\ntranslation: LSLRFSLMFSISSSQKHLFLKNCD\ntranslation: LSLRFSLMFSISSSQKHLFLKNCD\ntranslation: HSSLRKDVSEKSLLKTLMKISMK\ntranslation: HSSLRKDVSEKSLLKTLMKISMK\ntranslation: PDFFRTKFYPVIFRRVSFVTLW\ntranslation: PDFFRTKFYPVIFRRVSFVTLW\ntranslation: RISSELNSIQSSSGECPLSLFG\ntranslation: RISSELNSIQSSSGECPLSLFG\ntranslation: PKSDKGHSPEDDWIEFSSEEIR\ntranslation: PKSDKGHSPEDDWIEFSSEEIR\ntranslation: WGCGGNHHPCIFQVLNGGTTPC\ntranslation: WGCGGNHHPCIFQVLNGGTTPC\ntranslation: GDVVGITTPASFKSLMVALLPA\ntranslation: GDVVGITTPASFKSLMVALLPA\ntranslation: AGSSATIKDLKDAGVVIPTTSP\ntranslation: AGSSATIKDLKDAGVVIPTTSP\ntranslation: RGKCLLRLKWNTLLFYELTAV\ntranslation: RGKCLLRLKWNTLLFYELTAV\ntranslation: LLLTHKREECSISVSINTSL\ntranslation: LLLTHKREECSISVSINTSL\ntranslation: LPYCQHRRIFSNQCPWGRRMT\ntranslation: LPYCQHRRIFSNQCPWGRRMT\ntranslation: GHPPAPGTLIRENPSVLAIRQ\ntranslation: GHPPAPGTLIRENPSVLAIRQ\ntranslation: ESNGSIMKIMSQKMWVLCVALIY\ntranslation: ESNGSIMKIMSQKMWVLCVALIY\ntranslation: IKRLYNEDNVSENVGSVCGTDL\ntranslation: IKRLYNEDNVSENVGSVCGTDL\ntranslation: SGLRDRR\ntranslation: SGLRDRR\ntranslation: PPIPQPR\ntranslation: PPIPQPR\ntranslation: LLSLNP\ntranslation: LLSLNP\ntranslation: SYPSTP\ntranslation: SYPSTP\ntranslation: LTSLHNDELLLGRRSSKDDL\ntranslation: LTSLHNDELLLGRRSSKDDL\ntranslation: QRSSLLERLPSRSSSLWRDVR\ntranslation: QRSSLLERLPSRSSSLWRDVR\ntranslation: RDRLCSNVSPAEAHHCGGMSE\ntranslation: RDRLCSNVSPAEAHHCGGMSE\ntranslation: EIVFARTSPQQKLIIVEGCQ\ntranslation: EIVFARTSPQQKLIIVEGCQ\ntranslation: AVDSLKSIYVAFTAGVFLDLNL\ntranslation: AVDSLKSIYVAFTAGVFLDLNL\ntranslation: SGHFSSHVLLYLFSTLCEQGLL\ntranslation: SGHFSSHVLLYLFSTLCEQGLL\ntranslation: LVIFQAMFYCICFLLYVSKVCC\ntranslation: LVIFQAMFYCICFLLYVSKVCC\ntranslation: HSLNAQFKMKHTATEYSMFLREF\ntranslation: HSLNAQFKMKHTATEYSMFLREF\ntranslation: PPPPPPPPPPHPPPPTHTTPPPPPP\ntranslation: PPPPPPPPPPHPPPPTHTTPPPPPP\ntranslation: PHPPPPPPPPTHPPPHTQHPPPPHP\ntranslation: PHPPPPPPPPTHPPPHTQHPPPPHP\ntranslation: PTPPPPPPPPPTPPHTHNTPPPPT\ntranslation: PTPPPPPPPPPTPPHTHNTPPPPT\ntranslation: GVGGGGVLCVWGGVGGGGGGGGGVG\ntranslation: GVGGGGVLCVWGGVGGGGGGGGGVG\ntranslation: GWGGGGCCVCGGGWVGGGGGGGGWG\ntranslation: GWGGGGCCVCGGGWVGGGGGGGGWG\ntranslation: GGGGGGVVCVGGGGWGGGGGGGGG\ntranslation: GGGGGGVVCVGGGGWGGGGGGGGG\ntranslation: EALAQRTYSDDHVKFFCFQVLEH\ntranslation: EALAQRTYSDDHVKFFCFQVLEH\ntranslation: DVPVLESRKTSHDHHCMSSGLEL\ntranslation: DVPVLESRKTSHDHHCMSSGLEL\ntranslation: MKSMKAAVMREMNLPMNLHSQLS\ntranslation: MKSMKAAVMREMNLPMNLHSQLS\ntranslation: EVNESSSDEGNEPANEFAQPAF\ntranslation: EVNESSSDEGNEPANEFAQPAF\ntranslation: MKAGCANSLAGSFPSSLLLSLTS\ntranslation: MKAGCANSLAGSFPSSLLLSLTS\ntranslation: ESWLCKFIGRFISLITAAFIDF\ntranslation: ESWLCKFIGRFISLITAAFIDF\ntranslation: VMNTWFYITYEKDPVLYMYQLL\ntranslation: VMNTWFYITYEKDPVLYMYQLL\ntranslation: HCGTFEYSFNFATYSADSK\ntranslation: HCGTFEYSFNFATYSADSK\ntranslation: IVVLLNTPLILLPILLIPN\ntranslation: IVVLLNTPLILLPILLIPN\ntranslation: FGISRIGSKIKGVFKSTTM\ntranslation: FGISRIGSKIKGVFKSTTM\ntranslation: LSHLTPTSFVLFCFCPLCTTFLFSN\ntranslation: LSHLTPTSFVLFCFCPLCTTFLFSN\ntranslation: FPTSPLPLLFCFVFALYVLHSYFLT\ntranslation: FPTSPLPLLFCFVFALYVLHSYFLT\ntranslation: LENKNVVHKGQKQNKTKEVGVRWE\ntranslation: LENKNVVHKGQKQNKTKEVGVRWE\ntranslation: FVYLSKLKELWLRNNPIESIP\ntranslation: FVYLSKLKELWLRNNPIESIP\ntranslation: EGMLSMGLFRNQSSFSLDKYTK\ntranslation: EGMLSMGLFRNQSSFSLDKYTK\ntranslation: RDAFNGVVSQPELLQFRQVYK\ntranslation: RDAFNGVVSQPELLQFRQVYK\ntranslation: SIQCIFRYYQHYKIESTFIL\ntranslation: SIQCIFRYYQHYKIESTFIL\ntranslation: EYESRFYFVMLIIPKDALNA\ntranslation: EYESRFYFVMLIIPKDALNA\ntranslation: FSKFCGSSKDFYVHTLSSLVL\ntranslation: FSKFCGSSKDFYVHTLSSLVL\ntranslation: SVSSVVPVRIFMYIRSRLSFW\ntranslation: SVSSVVPVRIFMYIRSRLSFW\ntranslation: QNERRERMYIKILTGTTELTE\ntranslation: QNERRERMYIKILTGTTELTE\ntranslation: FYLLNCLFRSLAGCFLCLPLHVA\ntranslation: FYLLNCLFRSLAGCFLCLPLHVA\ntranslation: LPAKLSFQKSCRMFSLFASPCCC\ntranslation: LPAKLSFQKSCRMFSLFASPCCC\ntranslation: LYRAVHMTKSRITVLLNQLLRYS\ntranslation: LYRAVHMTKSRITVLLNQLLRYS\ntranslation: AISEELVQKYSNSALGHVNCTIK\ntranslation: AISEELVQKYSNSALGHVNCTIK\ntranslation: THPPRVHHIPNLSLTKRKPS\ntranslation: THPPRVHHIPNLSLTKRKPS\ntranslation: HIHLGSIIYQISPSLNVSLL\ntranslation: HIHLGSIIYQISPSLNVSLL\ntranslation: RRLTFSEGEIWYMMDPRWMC\ntranslation: RRLTFSEGEIWYMMDPRWMC\ntranslation: LNLNNSCKNLIKRHQLGTQFL\ntranslation: LNLNNSCKNLIKRHQLGTQFL\ntranslation: EELCPKLMSLNEIFTGIVQIQ\ntranslation: EELCPKLMSLNEIFTGIVQIQ\ntranslation: STFTHIMSFEPHNVLGMESVVQ\ntranslation: STFTHIMSFEPHNVLGMESVVQ\ntranslation: MHLLRIRTLSYVTQLSLLEK\ntranslation: MHLLRIRTLSYVTQLSLLEK\ntranslation: HSHRNILSSLFIGKNNFRSTLKK\ntranslation: HSHRNILSSLFIGKNNFRSTLKK\ntranslation: NGACYSLTFLLSACVELLVIQA"
      },
      {
       "name": "stderr",
       "output_type": "stream",
       "text": "1000it [00:00, 1221.36it/s]"
      },
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "\ntranslation: NGACYSLTFLLSACVELLVIQA\ntranslation: IDKVKENPIENISLDYHQLS\ntranslation: IDKVKENPIENISLDYHQLS\ntranslation: QIADGNQGRYSQWDSPSLCQ\ntranslation: QIADGNQGRYSQWDSPSLCQ\ntranslation: GLIPCLLIYSLDDERVTKNKT\ntranslation: GLIPCLLIYSLDDERVTKNKT\ntranslation: GVLFFVTLSSSRLYINRHGIKP\ntranslation: GVLFFVTLSSSRLYINRHGIKP\ntranslation: KNGTKVNSTVNAHAYFIHIFNNK\ntranslation: KNGTKVNSTVNAHAYFIHIFNNK\ntranslation: KNIFSKIEEVISVLAIGATYLL\ntranslation: KNIFSKIEEVISVLAIGATYLL\ntranslation: NHMCFLQLFLLFWPNNPKNPYH\ntranslation: NHMCFLQLFLLFWPNNPKNPYH\ntranslation: TICVFCSYFSCSGQTTLKILT\ntranslation: TICVFCSYFSCSGQTTLKILT\ntranslation: LYFYLRLIYSTSITLLPISNN\ntranslation: LYFYLRLIYSTSITLLPISNN\ntranslation: CRYSPKTSRHFQSSPISQIKNT\ntranslation: CRYSPKTSRHFQSSPISQIKNT\ntranslation: VDILQKQADIFRVVQFPKSKTQ\ntranslation: VDILQKQADIFRVVQFPKSKTQ\ntranslation: CVFDLGNWTTLKMSACFWRIST\ntranslation: CVFDLGNWTTLKMSACFWRIST\ntranslation: ILFEMRSHFVTQTGVQWYNLGSL\ntranslation: ILFEMRSHFVTQTGVQWYNLGSL\ntranslation: ARFPKFLSWEEEIVALDHPW\ntranslation: ARFPKFLSWEEEIVALDHPW\ntranslation: TLSKIFKLGGRDSRSGSPMA\ntranslation: TLSKIFKLGGRDSRSGSPMA\ntranslation: AMGDPERLSLPPSLKILESV\ntranslation: AMGDPERLSLPPSLKILESV\ntranslation: KIVLYAKKVGYTPDWIFLLRN\ntranslation: KIVLYAKKVGYTPDWIFLLRN\ntranslation: FLSRNIQSGVYPTFLAYKTIF\ntranslation: FLSRNIQSGVYPTFLAYKTIF\ntranslation: LKRFIRQAEMNHYALFKCYMFL\ntranslation: LKRFIRQAEMNHYALFKCYMFL\ntranslation: QSEAKKGSSHNRKNKSQHTLEII\ntranslation: QSEAKKGSSHNRKNKSQHTLEII\ntranslation: DYESGFNISEHTIAKTQTGIR\ntranslation: DYESGFNISEHTIAKTQTGIR\ntranslation: SNPSLGLSYCVFRYVKATFVV\ntranslation: SNPSLGLSYCVFRYVKATFVV\ntranslation: FRFHLVSFLPFLSSIFDSLT\ntranslation: FRFHLVSFLPFLSSIFDSLT\ntranslation: DKVSSKNSLESYAFNMKATVEDE\ntranslation: DKVSSKNSLESYAFNMKATVEDE\ntranslation: DFGALSFACTESVVILVASKCQ\ntranslation: DFGALSFACTESVVILVASKCQ\ntranslation: GWHLLATKITTDSVQAKDKAPK\ntranslation: GWHLLATKITTDSVQAKDKAPK\ntranslation: AGIYWPPKSLQILYKQKTRHQN\ntranslation: AGIYWPPKSLQILYKQKTRHQN\ntranslation: LAFTGHQNHYRFCTSKRQGTKI\ntranslation: LAFTGHQNHYRFCTSKRQGTKI\ntranslation: LIWLKNSRGMMYWIVFALYTVI\ntranslation: LIWLKNSRGMMYWIVFALYTVI\ntranslation: QSQYREQKQSSTSSLYYFLTKS\ntranslation: QSQYREQKQSSTSSLYYFLTKS\ntranslation: QKFKDCTCQ\ntranslation: QKFKDCTCQ\ntranslation: KNSKTAHA\ntranslation: KNSKTAHA\ntranslation: KIQRLHMP\ntranslation: KIQRLHMP\ntranslation: LACAVFEFL\ntranslation: LACAVFEFL\ntranslation: WHVQSLNF\ntranslation: WHVQSLNF\ntranslation: LIPALVLLFRVCFFVFVLLLLL\ntranslation: LIPALVLLFRVCFFVFVLLLLL\ntranslation: SSLLLSCSSESVFLSLSFFFCS\ntranslation: SSLLLSCSSESVFLSLSFFFCS\ntranslation: HPCSCLALPSLFFCLCPSSFAL\ntranslation: HPCSCLALPSLFFCLCPSSFAL\ntranslation: QSKRRRTKTKKQTRKSKTRAGM\ntranslation: QSKRRRTKTKKQTRKSKTRAGM\ntranslation: EQKKKDKDKKTDSEEQDKSRDE\ntranslation: EQKKKDKDKKTDSEEQDKSRDE\ntranslation: SVLLDVDGAVKSILSSARLLTM\ntranslation: SVLLDVDGAVKSILSSARLLTM\ntranslation: IVSSLAEERMDFTAPSTSSKTL\ntranslation: IVSSLAEERMDFTAPSTSSKTL\ntranslation: IIGSETHNLLILKSLKFEIPGA\ntranslation: IIGSETHNLLILKSLKFEIPGA\ntranslation: FAPGISNLRDFKISKLCVSLPII\ntranslation: FAPGISNLRDFKISKLCVSLPII\ntranslation: LFCISVALCCVHFLLLPPKCWD\ntranslation: LFCISVALCCVHFLLLPPKCWD\ntranslation: YFVSQLLFAVSISSSCRLNAGI\ntranslation: YFVSQLLFAVSISSSCRLNAGI\ntranslation: ESQHLGGRRRKWTQQRATEIQNK\ntranslation: ESQHLGGRRRKWTQQRATEIQNK\ntranslation: KHLLCFFVSGCRSHFLMIFKKTL\ntranslation: KHLLCFFVSGCRSHFLMIFKKTL\ntranslation: QRFLENHEKMTSTTRYKKAEQMF\ntranslation: QRFLENHEKMTSTTRYKKAEQMF\ntranslation: RRVGDQHQRGRQ\ntranslation: RRVGDQHQRGRQ\ntranslation: GEWVTNISEGGS\ntranslation: GEWVTNISEGGS\ntranslation: AASLADVGHPLS\ntranslation: AASLADVGHPLS\ntranslation: LPPSLMLVTHSP\ntranslation: LPPSLMLVTHSP\ntranslation: GVSPCYIMLGYNFSSFPCGTISI\ntranslation: GVSPCYIMLGYNFSSFPCGTISI\ntranslation: GLVLAILCLVIIFHLSLAVLYLL\ntranslation: GLVLAILCLVIIFHLSLAVLYLL\ntranslation: VNLNSSLFLKQTSLCLFHIFSNW\ntranslation: VNLNSSLFLKQTSLCLFHIFSNW\ntranslation: PVTEYMKKAEARLLEEQRRVQVY\ntranslation: PVTEYMKKAEARLLEEQRRVQVY\ntranslation: GSSLEKGQPSSKMVSQEVCVQ\ntranslation: GSSLEKGQPSSKMVSQEVCVQ\ntranslation: FTGVLVYSQGFSTITIIFFQYIF\ntranslation: FTGVLVYSQGFSTITIIFFQYIF\ntranslation: HWCFSIFTRFFNHYHYLLPVHF\ntranslation: HWCFSIFTRFFNHYHYLLPVHF\ntranslation: KMYWKKIMVMVEKPCEYTKTPV\ntranslation: KMYWKKIMVMVEKPCEYTKTPV\ntranslation: RFSFILVLRVCYNFLFLWALVV\ntranslation: RFSFILVLRVCYNFLFLWALVV\ntranslation: QHCISCVWHCSIQMFVG\ntranslation: QHCISCVWHCSIQMFVG\ntranslation: NTVSLASGIVQSRCLLG\ntranslation: NTVSLASGIVQSRCLLG\ntranslation: TLYLLRLALFNPDVCWD\ntranslation: TLYLLRLALFNPDVCWD\ntranslation: VPTNIWIEQCQTQEIQC\ntranslation: VPTNIWIEQCQTQEIQC\ntranslation: SQQTSGLNNARRKRYSV\ntranslation: SQQTSGLNNARRKRYSV\ntranslation: IQSTSPKIKILSYQHSLILDR\ntranslation: IQSTSPKIKILSYQHSLILDR\ntranslation: LLYPTFTATFIRFVSHLPACLF\ntranslation: LLYPTFTATFIRFVSHLPACLF\ntranslation: FFTPLSLPLSSGLCLISLRVFS\ntranslation: FFTPLSLPLSSGLCLISLRVFS\ntranslation: SLPHFHCHFHQVCVSSPCVSF\ntranslation: SLPHFHCHFHQVCVSSPCVSF\ntranslation: EKTRREMRHKPDESGSESGVKK\ntranslation: EKTRREMRHKPDESGSESGVKK\ntranslation: PPFTIDFGSRFL\ntranslation: PPFTIDFGSRFL\ntranslation: PIYYRLWQQVF\ntranslation: PIYYRLWQQVF\ntranslation: QKPAAKVYSKWG\ntranslation: QKPAAKVYSKWG\ntranslation: KNLLPKSIVNG\ntranslation: KNLLPKSIVNG\ntranslation: SCTKLFICLFFVCNLSVKGFSLL\ntranslation: SCTKLFICLFFVCNLSVKGFSLL\ntranslation: SREKPFTDKLQTKKRQINNFVQE\ntranslation: SREKPFTDKLQTKKRQINNFVQE\ntranslation: VKLYFSYPIWFDIYTQTYISTSI\ntranslation: VKLYFSYPIWFDIYTQTYISTSI\ntranslation: VFALFYYYFFQCVGPTLCNLP\ntranslation: VFALFYYYFFQCVGPTLCNLP\ntranslation: FLLCFIIIFSNVWGLHFAIYL\ntranslation: FLLCFIIIFSNVWGLHFAIYL\ntranslation: FCFVLLLFFPMCGAYTLQST\ntranslation: FCFVLLLFFPMCGAYTLQST\ntranslation: MHPPSLAFTPEFHTSLSPLA\ntranslation: MHPPSLAFTPEFHTSLSPLA\ntranslation: LVFRCKPLITLSYLIVFSGVLL\ntranslation: LVFRCKPLITLSYLIVFSGVLL\ntranslation: SESFSLLSKVRLASLTFLSSSW\ntranslation: SESFSLLSKVRLASLTFLSSSW\ntranslation: PGGAQEGQGRQPHLGKQTKGFR\ntranslation: PGGAQEGQGRQPHLGKQTKGFR\ntranslation: QEELRKVKDANLTLESKLKDSE\ntranslation: QEELRKVKDANLTLESKLKDSE\ntranslation: LWCILTIIALIVVFCQDRHQT\ntranslation: LWCILTIIALIVVFCQDRHQT\ntranslation: LVYPHNYRSYCSFLPRQAPD\ntranslation: LVYPHNYRSYCSFLPRQAPD\ntranslation: GLVPVLAENYNKSDNCEDTPE\ntranslation: GLVPVLAENYNKSDNCEDTPE\ntranslation: VWCLSWQKTTIRAIIVRIHQ\ntranslation: VWCLSWQKTTIRAIIVRIHQ\ntranslation: LLISLMAAVQLIWKLLFYCEH\ntranslation: LLISLMAAVQLIWKLLFYCEH\ntranslation: DAHNKKAVSRSAEQQPSEKST\ntranslation: DAHNKKAVSRSAEQQPSEKST\ntranslation: MLTIKKQFPDQLNSSHQRNQQ\ntranslation: MLTIKKQFPDQLNSSHQRNQQ\ntranslation: LRCFLLNALWISYFLPYFLCSS\ntranslation: LRCFLLNALWISYFLPYFLCSS\ntranslation: QVFPIKCSVDFLLSSLFPLFQ\ntranslation: QVFPIKCSVDFLLSSLFPLFQ\ntranslation: TCKFYEILVFADASQKTTHLSFP\ntranslation: TCKFYEILVFADASQKTTHLSFP\ntranslation: LANFMKFLSLQMPVKKLHTYPSL\ntranslation: LANFMKFLSLQMPVKKLHTYPSL\ntranslation: RSLRLSTDSMYFFSFSSSSLYQD\ntranslation: RSLRLSTDSMYFFSFSSSSLYQD\ntranslation: QLEVVHRQHVFLLFLLIILVPG\ntranslation: QLEVVHRQHVFLLFLLIILVPG\ntranslation: SWYKDDEEKEKKYMLSVDNLKL\ntranslation: SWYKDDEEKEKKYMLSVDNLKL\ntranslation: PGTRMMRRKRRNTCCLWTTSSC\ntranslation: PGTRMMRRKRRNTCCLWTTSSC\ntranslation: YLLLNPANQRVHNRFREENTSK\ntranslation: YLLLNPANQRVHNRFREENTSK\ntranslation: SLKISSVNKEVLSRFFEYTYNIL\ntranslation: SLKISSVNKEVLSRFFEYTYNIL\ntranslation: LFLLYCIYSSDHLVTQDIRPVEV\ntranslation: LFLLYCIYSSDHLVTQDIRPVEV\ntranslation: LDLIFKAPYPDTTHLLSSGM\ntranslation: LDLIFKAPYPDTTHLLSSGM\ntranslation: HSRRQKVCSIRIRGFKYQIQ\ntranslation: HSRRQKVCSIRIRGFKYQIQ\ntranslation: RATLCVLLQCVLSVHPSLPSV\ntranslation: RATLCVLLQCVLSVHPSLPSV\ntranslation: VLHFVSCCNVFCLSIHLCPLY\ntranslation: VLHFVSCCNVFCLSIHLCPLY\ntranslation: CYTLCPAAMCSVCPSISALC\ntranslation: CYTLCPAAMCSVCPSISALC\ntranslation: YRGQRWMDRQNTLQQDTKCST\ntranslation: YRGQRWMDRQNTLQQDTKCST\ntranslation: TEGRDGWTDRTHCSRTQSVA\ntranslation: TEGRDGWTDRTHCSRTQSVA\ntranslation: PVDQCLAHGKCMSVLAS\ntranslation: PVDQCLAHGKCMSVLAS\ntranslation: FEMFFVCLFVFVCL\ntranslation: FEMFFVCLFVFVCL\ntranslation: LKCFLFVCLFLYVS\ntranslation: LKCFLFVCLFLYVS\ntranslation: ETYKNKQTNKKHFK\ntranslation: ETYKNKQTNKKHFK\ntranslation: RHTKTNKQTKNISK\ntranslation: RHTKTNKQTKNISK\ntranslation: LSVKRDFIHRPLTLACYSGMPI\ntranslation: LSVKRDFIHRPLTLACYSGMPI\ntranslation: VSEKRLYSQTSHFSMLFWDANL\ntranslation: VSEKRLYSQTSHFSMLFWDANL\ntranslation: RPEEEFHVPLDSYRCQLFIQPA\ntranslation: RPEEEFHVPLDSYRCQLFIQPA\ntranslation: YFQAIYSVFIICPWWPNNHACPP\ntranslation: YFQAIYSVFIICPWWPNNHACPP\ntranslation: RWTGMIIGPPRTNYENRIYSLKV\ntranslation: RWTGMIIGPPRTNYENRIYSLKV\ntranslation: DPWLSKFLSRYRKTHHIEPRTI\ntranslation: DPWLSKFLSRYRKTHHIEPRTI\ntranslation: IPGFPNSYHGIEKPITLNHVPL\ntranslation: IPGFPNSYHGIEKPITLNHVPL\ntranslation: FCKKPIKIKDSEVIRTLCFVLI\ntranslation: FCKKPIKIKDSEVIRTLCFVLI\ntranslation: NQNKTQGTDYFTVLYLDWFLTK\ntranslation: NQNKTQGTDYFTVLYLDWFLTK\ntranslation: IKTKHKVRITSLSFILIGFLQN\ntranslation: IKTKHKVRITSLSFILIGFLQN\ntranslation: TRGFISLSICSSDSHSVPTVCP\ntranslation: TRGFISLSICSSDSHSVPTVCP\ntranslation: HVDSFLYLSVHQIATQFLQFAH\ntranslation: HVDSFLYLSVHQIATQFLQFAH\ntranslation: HGQTVGTEWLSDEQIDKEMNPRV\ntranslation: HGQTVGTEWLSDEQIDKEMNPRV\ntranslation: KNRSLLHFKSTRTHNSYNRHQPTT\ntranslation: KNRSLLHFKSTRTHNSYNRHQPTT\ntranslation: KIVAFSTSSQLGLIIVTIGINQPH\ntranslation: KIVAFSTSSQLGLIIVTIGINQPH\ntranslation: LSVYCSSQLYSINLHFKGCIG\ntranslation: LSVYCSSQLYSINLHFKGCIG\ntranslation: HNDHIWASKVEGKPVFKNISRTS\ntranslation: HNDHIWASKVEGKPVFKNISRTS\ntranslation: EVREIFLKTGLPSTLLAHIWSLC\ntranslation: EVREIFLKTGLPSTLLAHIWSLC\ntranslation: GTLIRIFDTSSGHLIQELRRGSQ\ntranslation: GTLIRIFDTSSGHLIQELRRGSQ\ntranslation: ELRDHKTSVMNNFSTDVVVTA\ntranslation: ELRDHKTSVMNNFSTDVVVTA\ntranslation: TTSLCLTSLPSDLSRLKSAYVTL\ntranslation: TTSLCLTSLPSDLSRLKSAYVTL\ntranslation: DMIMAFSVNKANCNEIVLASTHD\ntranslation: DMIMAFSVNKANCNEIVLASTHD\ntranslation: SCVEAKTISLQFALLTENAMIIS\ntranslation: SCVEAKTISLQFALLTENAMIIS\ntranslation: QIKTFKSPMMTEWIIPHEMST\ntranslation: QIKTFKSPMMTEWIIPHEMST\ntranslation: MVLISWGIIHSVIIGDLKVLIC\ntranslation: MVLISWGIIHSVIIGDLKVLIC\ntranslation: KNYTKYEFGYNFLLKSPFLFLSF\ntranslation: KNYTKYEFGYNFLLKSPFLFLSF\ntranslation: MIGKETGISEENCNQIHTLYNF\ntranslation: MIGKETGISEENCNQIHTLYNF\ntranslation: RNCKISSLGQKNNSILTHKCLP\ntranslation: RNCKISSLGQKNNSILTHKCLP\ntranslation: GTARYPVSVKRTTQFLHINVCQ\ntranslation: GTARYPVSVKRTTQFLHINVCQ\ntranslation: FSNVCFSSAVISGFSVISSTSPV\ntranslation: FSNVCFSSAVISGFSVISSTSPV\ntranslation: LVMFASPLLSFLVSLSFPLPPQW\ntranslation: LVMFASPLLSFLVSLSFPLPPQW\ntranslation: HWGGRGNDRETRNDSRGEANITK\ntranslation: HWGGRGNDRETRNDSRGEANITK\ntranslation: TGEVEEMTEKPEMTAEEKQTLLK\ntranslation: TGEVEEMTEKPEMTAEEKQTLLK\ntranslation: AILCLVIIFHLSLAVLYLLRQ\ntranslation: AILCLVIIFHLSLAVLYLLRQ\ntranslation: YIMLGYNFSSFPCGTISIAP\ntranslation: YIMLGYNFSSFPCGTISIAP\ntranslation: FWTIFITTDQVAICIIILTTIII\ntranslation: FWTIFITTDQVAICIIILTTIII\ntranslation: SGPYSSPQIKLQFASSSSPPSS\ntranslation: SGPYSSPQIKLQFASSSSPPSS\ntranslation: LDHIHHHRSSCNLHHHPHHHHH\ntranslation: LDHIHHHRSSCNLHHHPHHHHH\ntranslation: DDDGGEDDDANCNLICGDEYGPE\ntranslation: DDDGGEDDDANCNLICGDEYGPE\ntranslation: FKDLEDKKVNQQPNTSDKKSSP\ntranslation: FKDLEDKKVNQQPNTSDKKSSP\ntranslation: QVVSIRRG\ntranslation: QVVSIRRG\ntranslation: PPPNTNYL\ntranslation: PPPNTNYL\ntranslation: PLLILTT\ntranslation: PLLILTT\ntranslation: EYLFIHISSALVTMATVQKILSN\ntranslation: EYLFIHISSALVTMATVQKILSN\ntranslation: ISFHSHLLCLSHNGNSTENSIK\ntranslation: ISFHSHLLCLSHNGNSTENSIK\ntranslation: WNMLNPPAEVTTKSQIPLSKIK\ntranslation: WNMLNPPAEVTTKSQIPLSKIK\ntranslation: LYFRERNLTFRCHLSGRIQHIP\ntranslation: LYFRERNLTFRCHLSGRIQHIP\ntranslation: SGIIVTLAGVYLSDGAEIKEKGF\ntranslation: SGIIVTLAGVYLSDGAEIKEKGF\ntranslation: RTSRHYWLNFPHYLLHPPTNISL\ntranslation: RTSRHYWLNFPHYLLHPPTNISL\ntranslation: GLHVIIGSTFLTICFIRQLIFHF\ntranslation: GLHVIIGSTFLTICFIRQLIFHF\ntranslation: LVLCMVIICFEVSLISHCWVVI\ntranslation: LVLCMVIICFEVSLISHCWVVI\ntranslation: PRNYSHHQLNRSSKHSQGLRDQ\ntranslation: PRNYSHHQLNRSSKHSQGLRDQ\ntranslation: SAVWLKVSSSSAVQIVPSLSD\ntranslation: SAVWLKVSSSSAVQIVPSLSD\ntranslation: HQIAKELFEQLKMKKPSAKQQ\ntranslation: HQIAKELFEQLKMKKPSAKQQ\ntranslation: HAQTEIIEGTNNPIFLSSIAFFQ\ntranslation: HAQTEIIEGTNNPIFLSSIAFFQ\ntranslation: YTEITLGYFCIILSYCLHSI\ntranslation: YTEITLGYFCIILSYCLHSI\ntranslation: HPKSTQYTFFSAPHHTYSKID\ntranslation: HPKSTQYTFFSAPHHTYSKID\ntranslation: TPSQHNIHSSQHHITLILKL\ntranslation: TPSQHNIHSSQHHITLILKL\ntranslation: MAKGKSNLILNKGEVQQPACR\ntranslation: MAKGKSNLILNKGEVQQPACR\ntranslation: GLQAGCCTSPLFNIRFDLPFAI\ntranslation: GLQAGCCTSPLFNIRFDLPFAI\ntranslation: VCKQVAALLPYSILGLICPSPL\ntranslation: VCKQVAALLPYSILGLICPSPL\ntranslation: KNYVSLVKCPFFFLPLIKSGSSA\ntranslation: KNYVSLVKCPFFFLPLIKSGSSA\ntranslation: HKCSNKQVEGHLELSCFADG\ntranslation: HKCSNKQVEGHLELSCFADG\ntranslation: TNAAINRWKVIWNSAVLQMD\ntranslation: TNAAINRWKVIWNSAVLQMD\ntranslation: SICKTAEFQMTFHLFIAAFV\ntranslation: SICKTAEFQMTFHLFIAAFV\ntranslation: TFLLCSYFLNGFKGTEINAIYM\ntranslation: TFLLCSYFLNGFKGTEINAIYM\ntranslation: FCSQLLIFFKNSFHGNKSWFSQ\ntranslation: FCSQLLIFFKNSFHGNKSWFSQ\ntranslation: PKSILTIPLNHLCSPVIVV\ntranslation: PKSILTIPLNHLCSPVIVV\ntranslation: TTITGEHRWFRGMVKMLL\ntranslation: TTITGEHRWFRGMVKMLL\ntranslation: GTRSILKTHFEKEDLEGHRTL\ntranslation: GTRSILKTHFEKEDLEGHRTL\ntranslation: WYHREESLVLGVVRCQSIFYWAC\ntranslation: WYHREESLVLGVVRCQSIFYWAC\ntranslation: GTTGRNHWFWVLLDVKAFSIGH\ntranslation: GTTGRNHWFWVLLDVKAFSIGH\ntranslation: KNQIKVDLVDENFTELRGEI\ntranslation: KNQIKVDLVDENFTELRGEI\ntranslation: KKALFKGSGSQTLKVTEKSPQ\ntranslation: KKALFKGSGSQTLKVTEKSPQ\ntranslation: RKVCRFKLEWLGNCSGLNDETY\ntranslation: RKVCRFKLEWLGNCSGLNDETY\ntranslation: NLQVEMIKQFHMQLNEMHSLLE\ntranslation: NLQVEMIKQFHMQLNEMHSLLE\ntranslation: FQQRMHFIQLHMKLFNHLHLQI\ntranslation: FQQRMHFIQLHMKLFNHLHLQI\ntranslation: KKNGSGWDADSNRSGSGWNDTT\ntranslation: KKNGSGWDADSNRSGSGWNDTT\ntranslation: KKMDLDGMLTVIGQGLVGMTP\ntranslation: KKMDLDGMLTVIGQGLVGMTP\ntranslation: VVSFQPDPDLLLSASHPDPFF\ntranslation: VVSFQPDPDLLLSASHPDPFF\ntranslation: WCHSNQTLTYYCQHPIQIHFF\ntranslation: WCHSNQTLTYYCQHPIQIHFF\ntranslation: MCCPSLWESVFVLDFLLEMWK\ntranslation: MCCPSLWESVFVLDFLLEMWK\ntranslation: CAAPACGNPFLCLIFCWRCGR\ntranslation: CAAPACGNPFLCLIFCWRCGR\ntranslation: PSTSPAKNQAQKRIPTGWGST\ntranslation: PSTSPAKNQAQKRIPTGWGST\ntranslation: LPHLQQKIKHKNGFPQAGAAH\ntranslation: LPHLQQKIKHKNGFPQAGAAH\ntranslation: FHISSKKSSTKTDSHRLGQHI\ntranslation: FHISSKKSSTKTDSHRLGQHI\ntranslation: LDRHVVICCCG\ntranslation: LDRHVVICCCG\ntranslation: WTDTLSFAAV\ntranslation: WTDTLSFAAV\ntranslation: GQTRCHLLLW\ntranslation: GQTRCHLLLW\ntranslation: PTAANDNVSVQ\ntranslation: PTAANDNVSVQ\ntranslation: PQQQMTTCLS\ntranslation: PQQQMTTCLS\ntranslation: DSQKLNFREHAKARVDHGAEI\ntranslation: DSQKLNFREHAKARVDHGAEI\ntranslation: MISAPWSTRALACSLKFNFWL\ntranslation: MISAPWSTRALACSLKFNFWL\ntranslation: DLSPMVHTGFSMLSEVQLLAV\ntranslation: DLSPMVHTGFSMLSEVQLLAV\ntranslation: AAMLSGHFRPPIISYFRTPE\ntranslation: AAMLSGHFRPPIISYFRTPE\ntranslation: PQGYGNMKLWEDENAQKAWL\ntranslation: PQGYGNMKLWEDENAQKAWL\ntranslation: SGVRKYEIMGGRKCPESMAA\ntranslation: SGVRKYEIMGGRKCPESMAA\ntranslation: SSQLQICESNCQPDFASQIP\ntranslation: SSQLQICESNCQPDFASQIP\ntranslation: PLSSRSVNLTANLTLPLRFP\ntranslation: PLSSRSVNLTANLTLPLRFP\ntranslation: GNLRGKVRLAVRFTDLELRG\ntranslation: GNLRGKVRLAVRFTDLELRG\ntranslation: ECPDGFAPLEETMECVGCEVGHW\ntranslation: ECPDGFAPLEETMECVGCEVGHW\ntranslation: NDQLHIPHIPWFLLMVQNHLDI\ntranslation: NDQLHIPHIPWFLLMVQNHLDI\ntranslation: LRVCYNFLFLWFTVHYFLLAIP\ntranslation: LRVCYNFLFLWFTVHYFLLAIP\ntranslation: SGFVIIFYFYGLPYIISCSPSP\ntranslation: SGFVIIFYFYGLPYIISCSPSP\ntranslation: RVSTSIPTVNIWCAHT\ntranslation: RVSTSIPTVNIWCAHT\ntranslation: CVYVYSYCKYMVCSH\ntranslation: CVYVYSYCKYMVCSH\ntranslation: RVSTPYIYSRNRRRHT\ntranslation: RVSTPYIYSRNRRRHT\ntranslation: KFHGIKLNRLIECHSNKQLFI\ntranslation: KFHGIKLNRLIECHSNKQLFI\ntranslation: DEQLLITVAFNQPVKLYSMKF\ntranslation: DEQLLITVAFNQPVKLYSMKF\ntranslation: RVSKLVSVLLSSASLSLSLPL\ntranslation: RVSKLVSVLLSSASLSLSLPL\ntranslation: GCPSWFQSYCHQHHCPYHCHC\ntranslation: GCPSWFQSYCHQHHCPYHCHC\ntranslation: GVQAGFSLTVISITVLITAT\ntranslation: GVQAGFSLTVISITVLITAT\ntranslation: TVAVIRTVMLMTVRLKPAWTP\ntranslation: TVAVIRTVMLMTVRLKPAWTP\ntranslation: SGSDKDSDADDSKTETSLDT\ntranslation: SGSDKDSDADDSKTETSLDT\ntranslation: LDSSNWTSEFSSAAQSVSLSSHN\ntranslation: LDSSNWTSEFSSAAQSVSLSSHN\ntranslation: LIHQIGHLNFLLLLNLFLFLLI\ntranslation: LIHQIGHLNFLLLLNLFLFLLI\ntranslation: YEKKEKQIEQQKKIQMSNLMNQ\ntranslation: YEKKEKQIEQQKKIQMSNLMNQ\ntranslation: PDKTHSLPLPCTIPMRHNYNKLH\ntranslation: PDKTHSLPLPCTIPMRHNYNKLH\ntranslation: RRGSLSLPSVNLHFRASGLFLS\ntranslation: RRGSLSLPSVNLHFRASGLFLS\ntranslation: QLKKSPEALKCRLTEGSERLPR\ntranslation: QLKKSPEALKCRLTEGSERLPR\ntranslation: PLHFMDDSPQRGEGTCPKLVNS\ntranslation: PLHFMDDSPQRGEGTCPKLVNS\ntranslation: SYECFKIINIWFPLNIAHHK\ntranslation: SYECFKIINIWFPLNIAHHK\ntranslation: GKLAKEIAEKLGFKHFPEAGI\ntranslation: GKLAKEIAEKLGFKHFPEAGI\ntranslation: SFLQCMFVAPLSKNEFTVDVWT\ntranslation: SFLQCMFVAPLSKNEFTVDVWT\ntranslation: LQSHFSNISVCIYVREYTCVCMYY\ntranslation: LQSHFSNISVCIYVREYTCVCMYY\ntranslation: YNLTLATYLCVFMYVNILVYVCIM\ntranslation: YNLTLATYLCVFMYVNILVYVCIM\ntranslation: VTLRSTPRRLTLRIPSCQTWTVR\ntranslation: VTLRSTPRRLTLRIPSCQTWTVR\ntranslation: DPEINTKKINPENSKLSDLDSE\ntranslation: DPEINTKKINPENSKLSDLDSE\ntranslation: VSLSRSDSLEFSGLIFLVLISGS\ntranslation: VSLSRSDSLEFSGLIFLVLISGS\ntranslation: ASRIQRNGFKLNQCDLVNG\ntranslation: ASRIQRNGFKLNQCDLVNG\ntranslation: ERFVLQFFCCKELCCSFPPKWFQ\ntranslation: ERFVLQFFCCKELCCSFPPKWFQ\ntranslation: KDLYYSSFVVKSCVVLFPQSGFS\ntranslation: KDLYYSSFVVKSCVVLFPQSGFS\ntranslation: AETTLGEKNNTTLYNKRTVVQIF\ntranslation: AETTLGEKNNTTLYNKRTVVQIF\ntranslation: KDAAKANLNYIGLDGNIGC\ntranslation: KDAAKANLNYIGLDGNIGC\ntranslation: AAYISIEANVVEICLSSIF\ntranslation: AAYISIEANVVEICLSSIF\ntranslation: FSLMRKTEHLPLLTVMIFTPPQ\ntranslation: FSLMRKTEHLPLLTVMIFTPPQ\ntranslation: WGLNAHSI\ntranslation: WGLNAHSI\ntranslation: GGSMLTV\ntranslation: GGSMLTV\ntranslation: GAQCSQY\ntranslation: GAQCSQY\ntranslation: NTVSIEPP\ntranslation: NTVSIEPP\ntranslation: LKQMVQISNFQIINDYVC\ntranslation: LKQMVQISNFQIINDYVC\ntranslation: ADIVIYDLKVGNLYHLFE\ntranslation: ADIVIYDLKVGNLYHLFE\ntranslation: TSVVLFTFYSSAFSKMLYIVCSFF\ntranslation: TSVVLFTFYSSAFSKMLYIVCSFF\ntranslation: NNNMLTVPKENYFKKRHNPDVPS\ntranslation: NNNMLTVPKENYFKKRHNPDVPS\ntranslation: IFHCEEGAEKDSSVLQMTIPV\ntranslation: IFHCEEGAEKDSSVLQMTIPV\ntranslation: DPHLQISHQQPTNHHPTMTN\ntranslation: DPHLQISHQQPTNHHPTMTN\ntranslation: ISHCWVVISRLLMRYLEVGI\ntranslation: ISHCWVVISRLLMRYLEVGI\ntranslation: ATSEQNPTSE\ntranslation: ATSEQNPTSE\ntranslation: RPRSRTQPPS\ntranslation: RPRSRTQPPS\ntranslation: DLGAEPNLR\ntranslation: DLGAEPNLR\ntranslation: ARRLGSAPRS\ntranslation: ARRLGSAPRS\ntranslation: LGGWVLLRGR\ntranslation: LGGWVLLRGR\ntranslation: SEVGFCSEV\ntranslation: SEVGFCSEV\ntranslation: AISRIVSRIRIVKMISVEGRLMVD\ntranslation: AISRIVSRIRIVKMISVEGRLMVD\ntranslation: STINLPSTLIIFTILILLTILEI\ntranslation: STINLPSTLIIFTILILLTILEI\ntranslation: RDQVRPLVLCMVIICFEVSLI\ntranslation: RDQVRPLVLCMVIICFEVSLI\ntranslation: FGDANQKKLNLETNLRDPLVQLL\ntranslation: FGDANQKKLNLETNLRDPLVQLL\ntranslation: RSWTRGSLRLVSRLSFFWLASPN\ntranslation: RSWTRGSLRLVSRLSFFWLASPN\ntranslation: RKYGKELLKKGHRQFKSNWSVF\ntranslation: RKYGKELLKKGHRQFKSNWSVF\ntranslation: QICRAAESP\ntranslation: QICRAAESP\ntranslation: KFVEQLSP\ntranslation: KFVEQLSP\ntranslation: WGLSCSTNL\ntranslation: WGLSCSTNL\ntranslation: GDSAALQI\ntranslation: GDSAALQI\ntranslation: GTQLLYKF\ntranslation: GTQLLYKF\ntranslation: LFIFFPEYLLKHQKLFFIHHLI\ntranslation: LFIFFPEYLLKHQKLFFIHHLI\ntranslation: AMRWWIKKSFWCFKRYSGKKMKR\ntranslation: AMRWWIKKSFWCFKRYSGKKMKR\ntranslation: NEMVDKKEFLVLQEIFRKKNEK\ntranslation: NEMVDKKEFLVLQEIFRKKNEK\ntranslation: HFPSFNYLLSLGLKFSLFKCIS\ntranslation: HFPSFNYLLSLGLKFSLFKCIS\ntranslation: TSLLSTICSAWASNSHFSSASH\ntranslation: TSLLSTICSAWASNSHFSSASH\ntranslation: LPFFQLFAQLGPQILTFQVHLM\ntranslation: LPFFQLFAQLGPQILTFQVHLM\ntranslation: SWDTASELVWYFNLRNTKLIRM\ntranslation: SWDTASELVWYFNLRNTKLIRM\ntranslation: LGHCLRTGMVLQFEKHKTDTN\ntranslation: LGHCLRTGMVLQFEKHKTDTN\ntranslation: IRISFVFLKLKYHTSSEAVSQ\ntranslation: IRISFVFLKLKYHTSSEAVSQ\ntranslation: QADMLHSKHDEEQKVWDEEIDDD\ntranslation: QADMLHSKHDEEQKVWDEEIDDD\ntranslation: KLTCYILNMMRSRRFGMKKLMM\ntranslation: KLTCYILNMMRSRRFGMKKLMM\ntranslation: SSSISSSQTFCSSSCLECNMSAW\ntranslation: SSSISSSQTFCSSSCLECNMSAW\ntranslation: AKHIHQMSEPAVPLVLSRITMA\ntranslation: AKHIHQMSEPAVPLVLSRITMA\ntranslation: VAMVILLSTRGTAGSDIWCMCLA\ntranslation: VAMVILLSTRGTAGSDIWCMCLA\ntranslation: CHGNPAQYERNRRFRHLVYVLG\ntranslation: CHGNPAQYERNRRFRHLVYVLG\ntranslation: CNQQLFYIYFNNANKICQKFS\ntranslation: CNQQLFYIYFNNANKICQKFS\ntranslation: AISNFSIYISIMPTKSAKSSL\ntranslation: AISNFSIYISIMPTKSAKSSL\ntranslation: RELLADFVGIIEIYIEKLLIA\ntranslation: RELLADFVGIIEIYIEKLLIA\ntranslation: RTVSLSRRRNSFIFCPPESF\ntranslation: RTVSLSRRRNSFIFCPPESF\ntranslation: ELSLCPGEGIHSFSVPQSPS\ntranslation: ELSLCPGEGIHSFSVPQSPS\ntranslation: KDSGGQKMNEFLLLDRETVL\ntranslation: KDSGGQKMNEFLLLDRETVL\ntranslation: IYLVHDRYKINIKVYLIKDYVYL\ntranslation: IYLVHDRYKINIKVYLIKDYVYL\ntranslation: QPSHHPQGKTLIHTRKHPHV\ntranslation: QPSHHPQGKTLIHTRKHPHV\ntranslation: SLLIILKVKPSFTRENTLMF\ntranslation: SLLIILKVKPSFTRENTLMF\ntranslation: NMRVFSRVNEGFTLRMMRRL\ntranslation: NMRVFSRVNEGFTLRMMRRL\ntranslation: LVLYTLYITIHLSIHVLMIIWV\ntranslation: LVLYTLYITIHLSIHVLMIIWV\ntranslation: CFVYLIYHNSFVHSCLDDYLG\ntranslation: CFVYLIYHNSFVHSCLDDYLG\ntranslation: ELDSDAPVGAKATSTSHSSPFFN\ntranslation: ELDSDAPVGAKATSTSHSSPFFN\ntranslation: RDQVRPLVLCMVIICFEVSLI\ntranslation: RDQVRPLVLCMVIICFEVSLI\ntranslation: CCLTELPSLMEIFISVLST\ntranslation: CCLTELPSLMEIFISVLST\ntranslation: LPHRTSFSDGNIYICAVYQ\ntranslation: LPHRTSFSDGNIYICAVYQ\ntranslation: LVDSTDINISIREGSSVRQ\ntranslation: LVDSTDINISIREGSSVRQ\ntranslation: GRQHRYKYFHQRRKFCEAA\ntranslation: GRQHRYKYFHQRRKFCEAA\ntranslation: RVNEGFMLLMWWVSEPHCVV\ntranslation: RVNEGFMLLMWWVSEPHCVV\ntranslation: TTQWGSLTHHINNIKPSFTR\ntranslation: TTQWGSLTHHINNIKPSFTR\ntranslation: HLWLPDPFGFSSWLTAILEA\ntranslation: HLWLPDPFGFSSWLTAILEA\ntranslation: SFQDSSQPAGKAEGIREPKV\ntranslation: SFQDSSQPAGKAEGIREPKV\ntranslation: NCKAKHFYEIVLPSRKYTQLQP\ntranslation: NCKAKHFYEIVLPSRKYTQLQP\ntranslation: GVEVVCIFLMVTLFHRNALPCNS\ntranslation: GVEVVCIFLMVTLFHRNALPCNS\ntranslation: KDSTVRKTQLQSFSQYIENRPEM\ntranslation: KDSTVRKTQLQSFSQYIENRPEM\ntranslation: KILQSEKLNFSLSANILRIDQR\ntranslation: KILQSEKLNFSLSANILRIDQR\ntranslation: HLWSILNILAERLKLSFSDCRIF\ntranslation: HLWSILNILAERLKLSFSDCRIF\ntranslation: LISLQISLIITFTATELIIFYIFF\ntranslation: LISLQISLIITFTATELIIFYIFF\ntranslation: NLPTNLLNYNIHSHRTNHILYLL\ntranslation: NLPTNLLNYNIHSHRTNHILYLL\ntranslation: FYPDRDLYSLLLQMYHDEEREIL\ntranslation: FYPDRDLYSLLLQMYHDEEREIL\ntranslation: FIQIEICILYYFKCTMMRKERS\ntranslation: FIQIEICILYYFKCTMMRKERS\ntranslation: GSLFPHHGTFEVIKNTDLDLDK\ntranslation: GSLFPHHGTFEVIKNTDLDLDK\ntranslation: ICITLQRISVELSVCTFFSNVYK\ntranslation: ICITLQRISVELSVCTFFSNVYK\ntranslation: LYHIAKNFSRIVSLHFFLKCVQ\ntranslation: LYHIAKNFSRIVSLHFFLKCVQ\ntranslation: LYTFEKKVQTDNSTEILCNVIQ\ntranslation: LYTFEKKVQTDNSTEILCNVIQ\ntranslation: PLEVVLMKEFPPYFFSLFFLLLA\ntranslation: PLEVVLMKEFPPYFFSLFFLLLA\ntranslation: HYLILSGIMFDLFNGALSDSSSS\ntranslation: HYLILSGIMFDLFNGALSDSSSS\ntranslation: GLELCLHRVVVRRELLVNVGFH\ntranslation: GLELCLHRVVVRRELLVNVGFH\ntranslation: MKSNIDKKFSAHYDAVEAELKS\ntranslation: MKSNIDKKFSAHYDAVEAELKS\ntranslation: PASLLHVGWPHQFWFCWNQPSP\ntranslation: PASLLHVGWPHQFWFCWNQPSP\ntranslation: QHLCCTLVGLINSGFVGISLLH\ntranslation: QHLCCTLVGLINSGFVGISLLH\ntranslation: SIFVARWLASSILVLLESAFSM\ntranslation: SIFVARWLASSILVLLESAFSM\ntranslation: MEKADSNKTRIDEANQRATKML\ntranslation: MEKADSNKTRIDEANQRATKML\ntranslation: WRRLIPTKPELMRPTNVQQRCW\ntranslation: WRRLIPTKPELMRPTNVQQRCW\ntranslation: VLYPVLCLLIEGKVLKALRPTS\ntranslation: VLYPVLCLLIEGKVLKALRPTS\ntranslation: GLLVSELSEPYLLSEDTVLGTAQ\ntranslation: GLLVSELSEPYLLSEDTVLGTAQ\ntranslation: ACWSQSFQNLTFYQKTQYWVQH\ntranslation: ACWSQSFQNLTFYQKTQYWVQH\ntranslation: LVGLRAFRTLPSIRRHSTGYST\ntranslation: LVGLRAFRTLPSIRRHSTGYST\ntranslation: SESHQQVV\ntranslation: SESHQQVV\ntranslation: VNLTNKLW\ntranslation: VNLTNKLW\ntranslation: PQLVGEIH\ntranslation: PQLVGEIH\ntranslation: HNLLVRFT\ntranslation: HNLLVRFT\ntranslation: NTVRCNLNCPQYLIKLLHLTQ\ntranslation: NTVRCNLNCPQYLIKLLHLTQ\ntranslation: MYVCVLCMAVYACECECEHVCG\ntranslation: MYVCVLCMAVYACECECEHVCG\ntranslation: CMYVCCVWQCMHVSVSVSMSV\ntranslation: CMYVCCVWQCMHVSVSVSMSV\ntranslation: PTDMLTLTLTCIHCHTQHTYIH\ntranslation: PTDMLTLTLTCIHCHTQHTYIH\ntranslation: PQTCSHSHSHAYTAIHNTHTYI\ntranslation: PQTCSHSHSHAYTAIHNTHTYI\ntranslation: HRHAHTHTHMHTLPYTTHIHT\ntranslation: HRHAHTHTHMHTLPYTTHIHT\ntranslation: QRCSCVV\ntranslation: QRCSCVV\ntranslation: EVFLCCD\ntranslation: EVFLCCD\ntranslation: ITTQEHL\ntranslation: ITTQEHL\ntranslation: SQHKNTS\ntranslation: SQHKNTS\ntranslation: HNTRTPL\ntranslation: HNTRTPL\ntranslation: PQSIYVRHS\ntranslation: PQSIYVRHS\ntranslation: SIYLREALL\ntranslation: SIYLREALL\ntranslation: GVPHVDRLR\ntranslation: GVPHVDRLR\ntranslation: ETRRQNVNSRSEIEINTAEFL\ntranslation: ETRRQNVNSRSEIEINTAEFL\ntranslation: KEFSCINFYFRSRIDILPSCF\ntranslation: KEFSCINFYFRSRIDILPSCF\ntranslation: RNSAVLISISDLELTFCLLV\ntranslation: RNSAVLISISDLELTFCLLV\ntranslation: SSGGLRHLSHRSGGPSGIWAGDC\ntranslation: SSGGLRHLSHRSGGPSGIWAGDC\ntranslation: GSHLPRSQKDLLSDDSSVSVHLS\ntranslation: GSHLPRSQKDLLSDDSSVSVHLS\ntranslation: IFQRCFVFLIKFLDLVHLFKLFL\ntranslation: IFQRCFVFLIKFLDLVHLFKLFL\ntranslation: KKKLKKVDKIEELDQENEAALEN\ntranslation: KKKLKKVDKIEELDQENEAALEN\ntranslation: RKSLKRWTRSRNLIKKTKQRWKM\ntranslation: RKSLKRWTRSRNLIKKTKQRWKM\ntranslation: LGATSLRFYCFTALNIIWLNLFSI\ntranslation: LGATSLRFYCFTALNIIWLNLFSI\ntranslation: SVNHVRFSVVFQK\ntranslation: SVNHVRFSVVFQK\ntranslation: CFWKTTENLTWLTL\ntranslation: CFWKTTENLTWLTL\ntranslation: FLEDYRKSNMVDT\ntranslation: FLEDYRKSNMVDT\ntranslation: LDFFLITVSLHCIHVGEQTCRCL\ntranslation: LDFFLITVSLHCIHVGEQTCRCL\ntranslation: WIFFLLQCHYTVFMWGNKHVGA\ntranslation: WIFFLLQCHYTVFMWGNKHVGA\ntranslation: VALLPSLLVHLSTHFNFPLIKMC\ntranslation: VALLPSLLVHLSTHFNFPLIKMC\ntranslation: HILIRGKLKCVLKCTNKLGSRAT\ntranslation: HILIRGKLKCVLKCTNKLGSRAT\ntranslation: ELEDKDLDFKNKDTDISIKP\ntranslation: ELEDKDLDFKNKDTDISIKP\ntranslation: SWKIKTWTLRIKTLTSQLNL\ntranslation: SWKIKTWTLRIKTLTSQLNL\ntranslation: SGLIEMSVSLFLKSKSLSSSS\ntranslation: SGLIEMSVSLFLKSKSLSSSS\ntranslation: VGYLIPVWVLAIVCSDMYVCVCV\ntranslation: VGYLIPVWVLAIVCSDMYVCVCV\ntranslation: GVSNPSLGLSYCVFRYVCVCVC\ntranslation: GVSNPSLGLSYCVFRYVCVCVC\ntranslation: THTHTYISEHTIAKTQTGIRYP\ntranslation: THTHTYISEHTIAKTQTGIRYP\ntranslation: LFIKFRTADLIPHLSHSDFS\ntranslation: LFIKFRTADLIPHLSHSDFS\ntranslation: LMQNGKRLKKKKTTIKKNTLNPY\ntranslation: LMQNGKRLKKKKTTIKKNTLNPY\ntranslation: VGVKCVLFNCCLFLLQPLTILHQ\ntranslation: VGVKCVLFNCCLFLLQPLTILHQ\ntranslation: HKVGIVCVTRVYRLFYLHALLIA\ntranslation: HKVGIVCVTRVYRLFYLHALLIA\ntranslation: FGLAKILSMSEFKSFLFTYHLHN\ntranslation: FGLAKILSMSEFKSFLFTYHLHN\ntranslation: SDSLKYYPCQNSNPSSSLIIFT\ntranslation: SDSLKYYPCQNSNPSSSLIIFT\ntranslation: FYTNTYSDFSVTLKFIFLSYQAS\ntranslation: FYTNTYSDFSVTLKFIFLSYQAS\ntranslation: CHLDLLMSQMYLSSSYTNVINYND\ntranslation: CHLDLLMSQMYLSSSYTNVINYND\ntranslation: GGGPAELVNLW\ntranslation: GGGPAELVNLW\ntranslation: PEIHQLSRTSS\ntranslation: PEIHQLSRTSS\ntranslation: QRFTNSAGPP\ntranslation: QRFTNSAGPP\ntranslation: RDSPTQQDLL\ntranslation: RDSPTQQDLL\ntranslation: IFCGTLASILHQESLCLHEM\ntranslation: IFCGTLASILHQESLCLHEM\ntranslation: SFHVNTVIPGEELRPKYHKK\ntranslation: SFHVNTVIPGEELRPKYHKK\ntranslation: KGLGRGGLPLIEDCIVDDTLQ\ntranslation: KGLGRGGLPLIEDCIVDDTLQ\ntranslation: VAMCHQQCSLQSMVIPLSPIL\ntranslation: VAMCHQQCSLQSMVIPLSPIL\ntranslation: CNVSSTMQSSINGNPPLPNP\ntranslation: CNVSSTMQSSINGNPPLPNP\ntranslation: LIKKLPPVLAIQLKRFDYDWE\ntranslation: LIKKLPPVLAIQLKRFDYDWE\nCPU times: user 816 ms, sys: 72 ms, total: 888 ms\nWall time: 830 ms\n"
      },
      {
       "name": "stderr",
       "output_type": "stream",
       "text": "\n"
      }
     ],
     "persistentId": "be06ded1-d7d4-4f21-9961-3aed5ff30cbf",
     "text": "%%time\n\nreads = 'SRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz'\nsubsampled_reads = 'SRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.fq.gz'\n\ncoding_scores = score_reads(subsampled_reads, peptide_graph, peptide_ksize)"
    },
    "executionTime": "2019-10-11T17:17:49.816Z"
   },
   {
    "cell": {
     "executionCount": 75,
     "executionEventId": "c2055ce1-a661-45ae-a9e2-c31bcb8f7958",
     "hasError": false,
     "id": "b242d89f-8d25-4127-b875-8a85fa69d914",
     "outputs": [
      {
       "name": "stderr",
       "output_type": "stream",
       "text": "/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/screed/openscreed.py:39: ResourceWarning: unclosed file <_io.BufferedReader name='SRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.fq.gz'>\n  self.iter_fn = self.open_reader(filename, *args, **kwargs)\n1000it [00:00, 2699.43it/s]"
      },
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "CPU times: user 384 ms, sys: 0 ns, total: 384 ms\nWall time: 379 ms\n"
      },
      {
       "name": "stderr",
       "output_type": "stream",
       "text": "\n"
      }
     ],
     "persistentId": "be06ded1-d7d4-4f21-9961-3aed5ff30cbf",
     "text": "%%time\n\nreads = 'SRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz'\nsubsampled_reads = 'SRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.fq.gz'\n\ncoding_scores = score_reads(subsampled_reads, peptide_graph, peptide_ksize)"
    },
    "executionTime": "2019-10-11T17:17:54.988Z"
   },
   {
    "cell": {
     "executionCount": 76,
     "executionEventId": "3a516df5-8801-40b4-a225-179fc58bf3ca",
     "hasError": false,
     "id": "cfea0c3d-642a-421a-a4ad-07fa69bea946",
     "outputs": [
      {
       "data": {
        "text/plain": "(1000, 4)"
       },
       "execution_count": 76,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "09393220-d1f9-4ddd-9c7c-bc15160f3680",
     "text": "coding_scores.shape"
    },
    "executionTime": "2019-10-11T17:18:02.943Z"
   },
   {
    "cell": {
     "executionCount": 77,
     "executionEventId": "ce320839-674d-4e59-9981-a3df2d32ad7c",
     "hasError": false,
     "id": "3281c496-16a4-496e-81d4-fa57e8d1a556",
     "outputs": [
      {
       "data": {
        "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>read_id</th>\n      <th>jaccard_in_peptide_db</th>\n      <th>n_kmers</th>\n      <th>classification</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>SRR306838.10559374 Ibis_Run100924_C3PO:6:51:17601:17119/1</td>\n      <td>0</td>\n      <td>15</td>\n      <td>non-coding</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>SRR306838.6196593 Ibis_Run100924_C3PO:6:29:16733:12435/1</td>\n      <td>0</td>\n      <td>17</td>\n      <td>non-coding</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>SRR306838.20767303 Ibis_Run100924_C3PO:6:104:6864:5062/1</td>\n      <td>0</td>\n      <td>16</td>\n      <td>non-coding</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>SRR306838.12582274 Ibis_Run100924_C3PO:6:62:11779:17975/1</td>\n      <td>0</td>\n      <td>16</td>\n      <td>non-coding</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>SRR306838.13334230 Ibis_Run100924_C3PO:6:66:16579:20350/1</td>\n      <td>0</td>\n      <td>12</td>\n      <td>non-coding</td>\n    </tr>\n  </tbody>\n</table>\n</div>",
        "text/plain": "                                                     read_id  \\\n0  SRR306838.10559374 Ibis_Run100924_C3PO:6:51:17601:17119/1   \n1   SRR306838.6196593 Ibis_Run100924_C3PO:6:29:16733:12435/1   \n2   SRR306838.20767303 Ibis_Run100924_C3PO:6:104:6864:5062/1   \n3  SRR306838.12582274 Ibis_Run100924_C3PO:6:62:11779:17975/1   \n4  SRR306838.13334230 Ibis_Run100924_C3PO:6:66:16579:20350/1   \n\n   jaccard_in_peptide_db  n_kmers classification  \n0                      0       15     non-coding  \n1                      0       17     non-coding  \n2                      0       16     non-coding  \n3                      0       16     non-coding  \n4                      0       12     non-coding  "
       },
       "execution_count": 77,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "ede6e1ba-f524-4ceb-b35b-6c20f91d437c",
     "text": "coding_scores.head()"
    },
    "executionTime": "2019-10-11T17:18:03.370Z"
   },
   {
    "cell": {
     "executionCount": 79,
     "executionEventId": "7bd9dc50-ab23-4f4a-9381-9e808c65e142",
     "hasError": false,
     "id": "4c484434-ea07-4b72-8073-567bf902bedb",
     "outputs": [
      {
       "data": {
        "text/plain": "150289"
       },
       "execution_count": 79,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "75c01782-427c-4c63-bc29-ed308ae4a142",
     "text": "true_coding_read_ids = set(coding_read_ids.values)\nlen(true_coding_read_ids)"
    },
    "executionTime": "2019-10-11T17:18:07.651Z"
   },
   {
    "cell": {
     "executionCount": 80,
     "executionEventId": "efb9c3ab-d531-4e00-bbfa-ef2cf9cfdae9",
     "hasError": false,
     "id": "9babb580-ed7e-4a80-bcb3-3d080fca4a29",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "(807698,)\n"
      }
     ],
     "persistentId": "317bb484-c93f-4e1c-b831-50734c9e6614",
     "text": "unspliced_read_ids_series = pd.read_csv(f\"{folder}/SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.read_ids.txt\", \n                                   header=None, squeeze=True)\nprint(unspliced_read_ids_series.shape)\nunspliced_read_ids_series.head()\n\nunspliced_reads = set(unspliced_read_ids_series.values)"
    },
    "executionTime": "2019-10-11T17:18:09.184Z"
   },
   {
    "cell": {
     "executionCount": 83,
     "executionEventId": "ad245673-5c0f-4fad-a012-c1013630556b",
     "hasError": false,
     "id": "efba9e80-c61d-4adb-9d6c-af7db00a9356",
     "outputs": [
      {
       "data": {
        "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>read_id</th>\n      <th>jaccard_in_peptide_db</th>\n      <th>n_kmers</th>\n      <th>classification</th>\n      <th>short_read_id</th>\n      <th>in_cds_no_stop_codon</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>5</th>\n      <td>SRR306838.2740879 Ibis_Run100924_C3PO:6:13:11155:5248/1</td>\n      <td>0</td>\n      <td>18</td>\n      <td>non-coding</td>\n      <td>SRR306838.2740879</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>13</th>\n      <td>SRR306838.4880582 Ibis_Run100924_C3PO:6:23:17413:5436/1</td>\n      <td>0</td>\n      <td>16</td>\n      <td>non-coding</td>\n      <td>SRR306838.4880582</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>52</th>\n      <td>SRR306838.17561853 Ibis_Run100924_C3PO:6:88:9412:18824/1</td>\n      <td>0</td>\n      <td>15</td>\n      <td>non-coding</td>\n      <td>SRR306838.17561853</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>55</th>\n      <td>SRR306838.733972 Ibis_Run100924_C3PO:6:4:4934:4948/1</td>\n      <td>0</td>\n      <td>10</td>\n      <td>non-coding</td>\n      <td>SRR306838.733972</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>56</th>\n      <td>SRR306838.12543451 Ibis_Run100924_C3PO:6:62:9024:13538/1</td>\n      <td>0</td>\n      <td>14</td>\n      <td>non-coding</td>\n      <td>SRR306838.12543451</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>61</th>\n      <td>SRR306838.3607263 Ibis_Run100924_C3PO:6:17:17893:5364/1</td>\n      <td>0</td>\n      <td>17</td>\n      <td>non-coding</td>\n      <td>SRR306838.3607263</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>62</th>\n      <td>SRR306838.13206565 Ibis_Run100924_C3PO:6:66:18235:6169/1</td>\n      <td>0</td>\n      <td>13</td>\n      <td>non-coding</td>\n      <td>SRR306838.13206565</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>65</th>\n      <td>SRR306838.22009706 Ibis_Run100924_C3PO:6:109:13259:19261/1</td>\n      <td>0</td>\n      <td>16</td>\n      <td>non-coding</td>\n      <td>SRR306838.22009706</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>76</th>\n      <td>SRR306838.15648532 Ibis_Run100924_C3PO:6:79:11606:4680/1</td>\n      <td>0</td>\n      <td>15</td>\n      <td>non-coding</td>\n      <td>SRR306838.15648532</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>81</th>\n      <td>SRR306838.15455056 Ibis_Run100924_C3PO:6:78:4010:4797/1</td>\n      <td>0</td>\n      <td>16</td>\n      <td>non-coding</td>\n      <td>SRR306838.15455056</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>82</th>\n      <td>SRR306838.18085420 Ibis_Run100924_C3PO:6:91:3302:10193/1</td>\n      <td>0</td>\n      <td>16</td>\n      <td>non-coding</td>\n      <td>SRR306838.18085420</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>89</th>\n      <td>SRR306838.14450220 Ibis_Run100924_C3PO:6:72:5878:19983/1</td>\n      <td>0</td>\n      <td>15</td>\n      <td>non-coding</td>\n      <td>SRR306838.14450220</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>91</th>\n      <td>SRR306838.22306128 Ibis_Run100924_C3PO:6:111:10517:5752/1</td>\n      <td>0</td>\n      <td>13</td>\n      <td>non-coding</td>\n      <td>SRR306838.22306128</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>95</th>\n      <td>SRR306838.21426981 Ibis_Run100924_C3PO:6:107:10499:5867/1</td>\n      <td>0</td>\n      <td>15</td>\n      <td>non-coding</td>\n      <td>SRR306838.21426981</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>100</th>\n      <td>SRR306838.4578147 Ibis_Run100924_C3PO:6:21:5524:16924/1</td>\n      <td>0</td>\n      <td>17</td>\n      <td>non-coding</td>\n      <td>SRR306838.4578147</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>101</th>\n      <td>SRR306838.21872584 Ibis_Run100924_C3PO:6:109:18205:6467/1</td>\n      <td>0</td>\n      <td>17</td>\n      <td>non-coding</td>\n      <td>SRR306838.21872584</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>111</th>\n      <td>SRR306838.16944188 Ibis_Run100924_C3PO:6:85:6220:16301/1</td>\n      <td>0</td>\n      <td>17</td>\n      <td>non-coding</td>\n      <td>SRR306838.16944188</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>112</th>\n      <td>SRR306838.23727244 Ibis_Run100924_C3PO:6:117:4784:12719/1</td>\n      <td>0</td>\n      <td>15</td>\n      <td>non-coding</td>\n      <td>SRR306838.23727244</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>115</th>\n      <td>SRR306838.14205894 Ibis_Run100924_C3PO:6:71:12647:13906/1</td>\n      <td>0</td>\n      <td>16</td>\n      <td>non-coding</td>\n      <td>SRR306838.14205894</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>117</th>\n      <td>SRR306838.23195980 Ibis_Run100924_C3PO:6:115:10389:5531/1</td>\n      <td>0</td>\n      <td>14</td>\n      <td>non-coding</td>\n      <td>SRR306838.23195980</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>128</th>\n      <td>SRR306838.16050088 Ibis_Run100924_C3PO:6:81:9253:5844/1</td>\n      <td>0</td>\n      <td>17</td>\n      <td>non-coding</td>\n      <td>SRR306838.16050088</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>130</th>\n      <td>SRR306838.11305329 Ibis_Run100924_C3PO:6:55:6820:17324/1</td>\n      <td>0</td>\n      <td>17</td>\n      <td>non-coding</td>\n      <td>SRR306838.11305329</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>133</th>\n      <td>SRR306838.21512796 Ibis_Run100924_C3PO:6:107:4518:13862/1</td>\n      <td>0</td>\n      <td>16</td>\n      <td>non-coding</td>\n      <td>SRR306838.21512796</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>144</th>\n      <td>SRR306838.3497462 Ibis_Run100924_C3PO:6:16:17526:15365/1</td>\n      <td>0</td>\n      <td>13</td>\n      <td>non-coding</td>\n      <td>SRR306838.3497462</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>147</th>\n      <td>SRR306838.12017478 Ibis_Run100924_C3PO:6:59:12959:15291/1</td>\n      <td>0</td>\n      <td>15</td>\n      <td>non-coding</td>\n      <td>SRR306838.12017478</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>149</th>\n      <td>SRR306838.15283701 Ibis_Run100924_C3PO:6:77:8512:7087/1</td>\n      <td>0</td>\n      <td>17</td>\n      <td>non-coding</td>\n      <td>SRR306838.15283701</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>166</th>\n      <td>SRR306838.12840151 Ibis_Run100924_C3PO:6:64:3090:6141/1</td>\n      <td>0</td>\n      <td>17</td>\n      <td>non-coding</td>\n      <td>SRR306838.12840151</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>170</th>\n      <td>SRR306838.20420782 Ibis_Run100924_C3PO:6:102:6980:13066/1</td>\n      <td>0</td>\n      <td>16</td>\n      <td>non-coding</td>\n      <td>SRR306838.20420782</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>177</th>\n      <td>SRR306838.2860378 Ibis_Run100924_C3PO:6:13:5714:16449/1</td>\n      <td>0</td>\n      <td>17</td>\n      <td>non-coding</td>\n      <td>SRR306838.2860378</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>182</th>\n      <td>SRR306838.16160561 Ibis_Run100924_C3PO:6:81:16949:17382/1</td>\n      <td>0</td>\n      <td>17</td>\n      <td>non-coding</td>\n      <td>SRR306838.16160561</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>...</th>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n    </tr>\n    <tr>\n      <th>797</th>\n      <td>SRR306838.13680341 Ibis_Run100924_C3PO:6:68:9310:17756/1</td>\n      <td>0</td>\n      <td>17</td>\n      <td>non-coding</td>\n      <td>SRR306838.13680341</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>798</th>\n      <td>SRR306838.4040046 Ibis_Run100924_C3PO:6:19:14960:5905/1</td>\n      <td>0</td>\n      <td>14</td>\n      <td>non-coding</td>\n      <td>SRR306838.4040046</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>803</th>\n      <td>SRR306838.13279517 Ibis_Run100924_C3PO:6:66:7643:14277/1</td>\n      <td>0</td>\n      <td>16</td>\n      <td>non-coding</td>\n      <td>SRR306838.13279517</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>811</th>\n      <td>SRR306838.18774600 Ibis_Run100924_C3PO:6:94:12239:17186/1</td>\n      <td>0</td>\n      <td>15</td>\n      <td>non-coding</td>\n      <td>SRR306838.18774600</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>816</th>\n      <td>SRR306838.14377476 Ibis_Run100924_C3PO:6:72:4654:12100/1</td>\n      <td>0</td>\n      <td>16</td>\n      <td>non-coding</td>\n      <td>SRR306838.14377476</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>821</th>\n      <td>SRR306838.21725737 Ibis_Run100924_C3PO:6:108:5879:13271/1</td>\n      <td>0</td>\n      <td>17</td>\n      <td>non-coding</td>\n      <td>SRR306838.21725737</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>825</th>\n      <td>SRR306838.15181345 Ibis_Run100924_C3PO:6:76:10030:16624/1</td>\n      <td>0</td>\n      <td>16</td>\n      <td>non-coding</td>\n      <td>SRR306838.15181345</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>826</th>\n      <td>SRR306838.16522761 Ibis_Run100924_C3PO:6:83:7550:14052/1</td>\n      <td>0</td>\n      <td>16</td>\n      <td>non-coding</td>\n      <td>SRR306838.16522761</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>830</th>\n      <td>SRR306838.12277341 Ibis_Run100924_C3PO:6:61:5712:3799/1</td>\n      <td>0</td>\n      <td>6</td>\n      <td>non-coding</td>\n      <td>SRR306838.12277341</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>833</th>\n      <td>SRR306838.21540966 Ibis_Run100924_C3PO:6:107:16617:16502/1</td>\n      <td>0</td>\n      <td>15</td>\n      <td>non-coding</td>\n      <td>SRR306838.21540966</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>836</th>\n      <td>SRR306838.9874975 Ibis_Run100924_C3PO:6:48:16430:4354/1</td>\n      <td>0</td>\n      <td>16</td>\n      <td>non-coding</td>\n      <td>SRR306838.9874975</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>863</th>\n      <td>SRR306838.14594886 Ibis_Run100924_C3PO:6:73:16781:15144/1</td>\n      <td>0</td>\n      <td>16</td>\n      <td>non-coding</td>\n      <td>SRR306838.14594886</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>866</th>\n      <td>SRR306838.12470892 Ibis_Run100924_C3PO:6:62:16712:5320/1</td>\n      <td>0</td>\n      <td>18</td>\n      <td>non-coding</td>\n      <td>SRR306838.12470892</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>883</th>\n      <td>SRR306838.12834148 Ibis_Run100924_C3PO:6:64:7605:5463/1</td>\n      <td>0</td>\n      <td>16</td>\n      <td>non-coding</td>\n      <td>SRR306838.12834148</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>884</th>\n      <td>SRR306838.19495546 Ibis_Run100924_C3PO:6:98:8225:5942/1</td>\n      <td>0</td>\n      <td>17</td>\n      <td>non-coding</td>\n      <td>SRR306838.19495546</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>887</th>\n      <td>SRR306838.12732563 Ibis_Run100924_C3PO:6:63:13899:14463/1</td>\n      <td>0</td>\n      <td>16</td>\n      <td>non-coding</td>\n      <td>SRR306838.12732563</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>900</th>\n      <td>SRR306838.8989824 Ibis_Run100924_C3PO:6:43:18075:12540/1</td>\n      <td>0</td>\n      <td>14</td>\n      <td>non-coding</td>\n      <td>SRR306838.8989824</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>902</th>\n      <td>SRR306838.11922626 Ibis_Run100924_C3PO:6:59:13438:4580/1</td>\n      <td>0</td>\n      <td>16</td>\n      <td>non-coding</td>\n      <td>SRR306838.11922626</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>904</th>\n      <td>SRR306838.20560054 Ibis_Run100924_C3PO:6:103:3833:5865/1</td>\n      <td>0</td>\n      <td>15</td>\n      <td>non-coding</td>\n      <td>SRR306838.20560054</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>907</th>\n      <td>SRR306838.22300311 Ibis_Run100924_C3PO:6:111:5860:5209/1</td>\n      <td>0</td>\n      <td>15</td>\n      <td>non-coding</td>\n      <td>SRR306838.22300311</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>912</th>\n      <td>SRR306838.1128147 Ibis_Run100924_C3PO:6:5:6210:20132/1</td>\n      <td>0</td>\n      <td>10</td>\n      <td>non-coding</td>\n      <td>SRR306838.1128147</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>916</th>\n      <td>SRR306838.22614986 Ibis_Run100924_C3PO:6:112:4987:13792/1</td>\n      <td>0</td>\n      <td>14</td>\n      <td>non-coding</td>\n      <td>SRR306838.22614986</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>946</th>\n      <td>SRR306838.23035540 Ibis_Run100924_C3PO:6:114:11432:11381/1</td>\n      <td>0</td>\n      <td>2</td>\n      <td>non-coding</td>\n      <td>SRR306838.23035540</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>958</th>\n      <td>SRR306838.2963305 Ibis_Run100924_C3PO:6:14:5053:5691/1</td>\n      <td>0</td>\n      <td>17</td>\n      <td>non-coding</td>\n      <td>SRR306838.2963305</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>970</th>\n      <td>SRR306838.13852206 Ibis_Run100924_C3PO:6:69:16741:16193/1</td>\n      <td>0</td>\n      <td>16</td>\n      <td>non-coding</td>\n      <td>SRR306838.13852206</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>976</th>\n      <td>SRR306838.3461741 Ibis_Run100924_C3PO:6:16:9292:11951/1</td>\n      <td>0</td>\n      <td>1</td>\n      <td>non-coding</td>\n      <td>SRR306838.3461741</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>986</th>\n      <td>SRR306838.19000486 Ibis_Run100924_C3PO:6:95:13556:19024/1</td>\n      <td>0</td>\n      <td>15</td>\n      <td>non-coding</td>\n      <td>SRR306838.19000486</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>989</th>\n      <td>SRR306838.2523598 Ibis_Run100924_C3PO:6:12:11847:5371/1</td>\n      <td>0</td>\n      <td>17</td>\n      <td>non-coding</td>\n      <td>SRR306838.2523598</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>993</th>\n      <td>SRR306838.22091604 Ibis_Run100924_C3PO:6:110:2523:6402/1</td>\n      <td>0</td>\n      <td>17</td>\n      <td>non-coding</td>\n      <td>SRR306838.22091604</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>999</th>\n      <td>SRR306838.3389047 Ibis_Run100924_C3PO:6:16:6019:4984/1</td>\n      <td>0</td>\n      <td>15</td>\n      <td>non-coding</td>\n      <td>SRR306838.3389047</td>\n      <td>True</td>\n    </tr>\n  </tbody>\n</table>\n<p>179 rows  6 columns</p>\n</div>",
        "text/plain": "                                                        read_id  \\\n5       SRR306838.2740879 Ibis_Run100924_C3PO:6:13:11155:5248/1   \n13      SRR306838.4880582 Ibis_Run100924_C3PO:6:23:17413:5436/1   \n52     SRR306838.17561853 Ibis_Run100924_C3PO:6:88:9412:18824/1   \n55         SRR306838.733972 Ibis_Run100924_C3PO:6:4:4934:4948/1   \n56     SRR306838.12543451 Ibis_Run100924_C3PO:6:62:9024:13538/1   \n61      SRR306838.3607263 Ibis_Run100924_C3PO:6:17:17893:5364/1   \n62     SRR306838.13206565 Ibis_Run100924_C3PO:6:66:18235:6169/1   \n65   SRR306838.22009706 Ibis_Run100924_C3PO:6:109:13259:19261/1   \n76     SRR306838.15648532 Ibis_Run100924_C3PO:6:79:11606:4680/1   \n81      SRR306838.15455056 Ibis_Run100924_C3PO:6:78:4010:4797/1   \n82     SRR306838.18085420 Ibis_Run100924_C3PO:6:91:3302:10193/1   \n89     SRR306838.14450220 Ibis_Run100924_C3PO:6:72:5878:19983/1   \n91    SRR306838.22306128 Ibis_Run100924_C3PO:6:111:10517:5752/1   \n95    SRR306838.21426981 Ibis_Run100924_C3PO:6:107:10499:5867/1   \n100     SRR306838.4578147 Ibis_Run100924_C3PO:6:21:5524:16924/1   \n101   SRR306838.21872584 Ibis_Run100924_C3PO:6:109:18205:6467/1   \n111    SRR306838.16944188 Ibis_Run100924_C3PO:6:85:6220:16301/1   \n112   SRR306838.23727244 Ibis_Run100924_C3PO:6:117:4784:12719/1   \n115   SRR306838.14205894 Ibis_Run100924_C3PO:6:71:12647:13906/1   \n117   SRR306838.23195980 Ibis_Run100924_C3PO:6:115:10389:5531/1   \n128     SRR306838.16050088 Ibis_Run100924_C3PO:6:81:9253:5844/1   \n130    SRR306838.11305329 Ibis_Run100924_C3PO:6:55:6820:17324/1   \n133   SRR306838.21512796 Ibis_Run100924_C3PO:6:107:4518:13862/1   \n144    SRR306838.3497462 Ibis_Run100924_C3PO:6:16:17526:15365/1   \n147   SRR306838.12017478 Ibis_Run100924_C3PO:6:59:12959:15291/1   \n149     SRR306838.15283701 Ibis_Run100924_C3PO:6:77:8512:7087/1   \n166     SRR306838.12840151 Ibis_Run100924_C3PO:6:64:3090:6141/1   \n170   SRR306838.20420782 Ibis_Run100924_C3PO:6:102:6980:13066/1   \n177     SRR306838.2860378 Ibis_Run100924_C3PO:6:13:5714:16449/1   \n182   SRR306838.16160561 Ibis_Run100924_C3PO:6:81:16949:17382/1   \n..                                                          ...   \n797    SRR306838.13680341 Ibis_Run100924_C3PO:6:68:9310:17756/1   \n798     SRR306838.4040046 Ibis_Run100924_C3PO:6:19:14960:5905/1   \n803    SRR306838.13279517 Ibis_Run100924_C3PO:6:66:7643:14277/1   \n811   SRR306838.18774600 Ibis_Run100924_C3PO:6:94:12239:17186/1   \n816    SRR306838.14377476 Ibis_Run100924_C3PO:6:72:4654:12100/1   \n821   SRR306838.21725737 Ibis_Run100924_C3PO:6:108:5879:13271/1   \n825   SRR306838.15181345 Ibis_Run100924_C3PO:6:76:10030:16624/1   \n826    SRR306838.16522761 Ibis_Run100924_C3PO:6:83:7550:14052/1   \n830     SRR306838.12277341 Ibis_Run100924_C3PO:6:61:5712:3799/1   \n833  SRR306838.21540966 Ibis_Run100924_C3PO:6:107:16617:16502/1   \n836     SRR306838.9874975 Ibis_Run100924_C3PO:6:48:16430:4354/1   \n863   SRR306838.14594886 Ibis_Run100924_C3PO:6:73:16781:15144/1   \n866    SRR306838.12470892 Ibis_Run100924_C3PO:6:62:16712:5320/1   \n883     SRR306838.12834148 Ibis_Run100924_C3PO:6:64:7605:5463/1   \n884     SRR306838.19495546 Ibis_Run100924_C3PO:6:98:8225:5942/1   \n887   SRR306838.12732563 Ibis_Run100924_C3PO:6:63:13899:14463/1   \n900    SRR306838.8989824 Ibis_Run100924_C3PO:6:43:18075:12540/1   \n902    SRR306838.11922626 Ibis_Run100924_C3PO:6:59:13438:4580/1   \n904    SRR306838.20560054 Ibis_Run100924_C3PO:6:103:3833:5865/1   \n907    SRR306838.22300311 Ibis_Run100924_C3PO:6:111:5860:5209/1   \n912      SRR306838.1128147 Ibis_Run100924_C3PO:6:5:6210:20132/1   \n916   SRR306838.22614986 Ibis_Run100924_C3PO:6:112:4987:13792/1   \n946  SRR306838.23035540 Ibis_Run100924_C3PO:6:114:11432:11381/1   \n958      SRR306838.2963305 Ibis_Run100924_C3PO:6:14:5053:5691/1   \n970   SRR306838.13852206 Ibis_Run100924_C3PO:6:69:16741:16193/1   \n976     SRR306838.3461741 Ibis_Run100924_C3PO:6:16:9292:11951/1   \n986   SRR306838.19000486 Ibis_Run100924_C3PO:6:95:13556:19024/1   \n989     SRR306838.2523598 Ibis_Run100924_C3PO:6:12:11847:5371/1   \n993    SRR306838.22091604 Ibis_Run100924_C3PO:6:110:2523:6402/1   \n999      SRR306838.3389047 Ibis_Run100924_C3PO:6:16:6019:4984/1   \n\n     jaccard_in_peptide_db  n_kmers classification       short_read_id  \\\n5                        0       18     non-coding   SRR306838.2740879   \n13                       0       16     non-coding   SRR306838.4880582   \n52                       0       15     non-coding  SRR306838.17561853   \n55                       0       10     non-coding    SRR306838.733972   \n56                       0       14     non-coding  SRR306838.12543451   \n61                       0       17     non-coding   SRR306838.3607263   \n62                       0       13     non-coding  SRR306838.13206565   \n65                       0       16     non-coding  SRR306838.22009706   \n76                       0       15     non-coding  SRR306838.15648532   \n81                       0       16     non-coding  SRR306838.15455056   \n82                       0       16     non-coding  SRR306838.18085420   \n89                       0       15     non-coding  SRR306838.14450220   \n91                       0       13     non-coding  SRR306838.22306128   \n95                       0       15     non-coding  SRR306838.21426981   \n100                      0       17     non-coding   SRR306838.4578147   \n101                      0       17     non-coding  SRR306838.21872584   \n111                      0       17     non-coding  SRR306838.16944188   \n112                      0       15     non-coding  SRR306838.23727244   \n115                      0       16     non-coding  SRR306838.14205894   \n117                      0       14     non-coding  SRR306838.23195980   \n128                      0       17     non-coding  SRR306838.16050088   \n130                      0       17     non-coding  SRR306838.11305329   \n133                      0       16     non-coding  SRR306838.21512796   \n144                      0       13     non-coding   SRR306838.3497462   \n147                      0       15     non-coding  SRR306838.12017478   \n149                      0       17     non-coding  SRR306838.15283701   \n166                      0       17     non-coding  SRR306838.12840151   \n170                      0       16     non-coding  SRR306838.20420782   \n177                      0       17     non-coding   SRR306838.2860378   \n182                      0       17     non-coding  SRR306838.16160561   \n..                     ...      ...            ...                 ...   \n797                      0       17     non-coding  SRR306838.13680341   \n798                      0       14     non-coding   SRR306838.4040046   \n803                      0       16     non-coding  SRR306838.13279517   \n811                      0       15     non-coding  SRR306838.18774600   \n816                      0       16     non-coding  SRR306838.14377476   \n821                      0       17     non-coding  SRR306838.21725737   \n825                      0       16     non-coding  SRR306838.15181345   \n826                      0       16     non-coding  SRR306838.16522761   \n830                      0        6     non-coding  SRR306838.12277341   \n833                      0       15     non-coding  SRR306838.21540966   \n836                      0       16     non-coding   SRR306838.9874975   \n863                      0       16     non-coding  SRR306838.14594886   \n866                      0       18     non-coding  SRR306838.12470892   \n883                      0       16     non-coding  SRR306838.12834148   \n884                      0       17     non-coding  SRR306838.19495546   \n887                      0       16     non-coding  SRR306838.12732563   \n900                      0       14     non-coding   SRR306838.8989824   \n902                      0       16     non-coding  SRR306838.11922626   \n904                      0       15     non-coding  SRR306838.20560054   \n907                      0       15     non-coding  SRR306838.22300311   \n912                      0       10     non-coding   SRR306838.1128147   \n916                      0       14     non-coding  SRR306838.22614986   \n946                      0        2     non-coding  SRR306838.23035540   \n958                      0       17     non-coding   SRR306838.2963305   \n970                      0       16     non-coding  SRR306838.13852206   \n976                      0        1     non-coding   SRR306838.3461741   \n986                      0       15     non-coding  SRR306838.19000486   \n989                      0       17     non-coding   SRR306838.2523598   \n993                      0       17     non-coding  SRR306838.22091604   \n999                      0       15     non-coding   SRR306838.3389047   \n\n     in_cds_no_stop_codon  \n5                    True  \n13                   True  \n52                   True  \n55                   True  \n56                   True  \n61                   True  \n62                   True  \n65                   True  \n76                   True  \n81                   True  \n82                   True  \n89                   True  \n91                   True  \n95                   True  \n100                  True  \n101                  True  \n111                  True  \n112                  True  \n115                  True  \n117                  True  \n128                  True  \n130                  True  \n133                  True  \n144                  True  \n147                  True  \n149                  True  \n166                  True  \n170                  True  \n177                  True  \n182                  True  \n..                    ...  \n797                  True  \n798                  True  \n803                  True  \n811                  True  \n816                  True  \n821                  True  \n825                  True  \n826                  True  \n830                  True  \n833                  True  \n836                  True  \n863                  True  \n866                  True  \n883                  True  \n884                  True  \n887                  True  \n900                  True  \n902                  True  \n904                  True  \n907                  True  \n912                  True  \n916                  True  \n946                  True  \n958                  True  \n970                  True  \n976                  True  \n986                  True  \n989                  True  \n993                  True  \n999                  True  \n\n[179 rows x 6 columns]"
       },
       "execution_count": 83,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "1254868e-73a0-4c0a-82c6-bb1b85e78cb0",
     "text": "coding_scores_unspliced.query(\"jaccard_in_peptide_db < 0.5 & in_cds_no_stop_codon == True\")"
    },
    "executionTime": "2019-10-11T17:18:37.136Z"
   },
   {
    "cell": {
     "executionCount": 84,
     "executionEventId": "3f024af3-e1eb-4a57-9966-c53155fc246c",
     "hasError": false,
     "id": "efba9e80-c61d-4adb-9d6c-af7db00a9356",
     "outputs": [
      {
       "data": {
        "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>read_id</th>\n      <th>jaccard_in_peptide_db</th>\n      <th>n_kmers</th>\n      <th>classification</th>\n      <th>short_read_id</th>\n      <th>in_cds_no_stop_codon</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>5</th>\n      <td>SRR306838.2740879 Ibis_Run100924_C3PO:6:13:11155:5248/1</td>\n      <td>0</td>\n      <td>18</td>\n      <td>non-coding</td>\n      <td>SRR306838.2740879</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>13</th>\n      <td>SRR306838.4880582 Ibis_Run100924_C3PO:6:23:17413:5436/1</td>\n      <td>0</td>\n      <td>16</td>\n      <td>non-coding</td>\n      <td>SRR306838.4880582</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>52</th>\n      <td>SRR306838.17561853 Ibis_Run100924_C3PO:6:88:9412:18824/1</td>\n      <td>0</td>\n      <td>15</td>\n      <td>non-coding</td>\n      <td>SRR306838.17561853</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>55</th>\n      <td>SRR306838.733972 Ibis_Run100924_C3PO:6:4:4934:4948/1</td>\n      <td>0</td>\n      <td>10</td>\n      <td>non-coding</td>\n      <td>SRR306838.733972</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>56</th>\n      <td>SRR306838.12543451 Ibis_Run100924_C3PO:6:62:9024:13538/1</td>\n      <td>0</td>\n      <td>14</td>\n      <td>non-coding</td>\n      <td>SRR306838.12543451</td>\n      <td>True</td>\n    </tr>\n  </tbody>\n</table>\n</div>",
        "text/plain": "                                                     read_id  \\\n5    SRR306838.2740879 Ibis_Run100924_C3PO:6:13:11155:5248/1   \n13   SRR306838.4880582 Ibis_Run100924_C3PO:6:23:17413:5436/1   \n52  SRR306838.17561853 Ibis_Run100924_C3PO:6:88:9412:18824/1   \n55      SRR306838.733972 Ibis_Run100924_C3PO:6:4:4934:4948/1   \n56  SRR306838.12543451 Ibis_Run100924_C3PO:6:62:9024:13538/1   \n\n    jaccard_in_peptide_db  n_kmers classification       short_read_id  \\\n5                       0       18     non-coding   SRR306838.2740879   \n13                      0       16     non-coding   SRR306838.4880582   \n52                      0       15     non-coding  SRR306838.17561853   \n55                      0       10     non-coding    SRR306838.733972   \n56                      0       14     non-coding  SRR306838.12543451   \n\n    in_cds_no_stop_codon  \n5                   True  \n13                  True  \n52                  True  \n55                  True  \n56                  True  "
       },
       "execution_count": 84,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "1254868e-73a0-4c0a-82c6-bb1b85e78cb0",
     "text": "coding_scores_unspliced.query(\"jaccard_in_peptide_db < 0.5 & in_cds_no_stop_codon == True\").head()"
    },
    "executionTime": "2019-10-11T17:18:43.170Z"
   },
   {
    "cell": {
     "executionCount": 87,
     "executionEventId": "c8d223de-6e63-4a28-9f46-751d43300fa1",
     "hasError": false,
     "id": "393b23a7-f1e1-44b0-ae77-e109b3792aeb",
     "outputs": [
      {
       "data": {
        "text/plain": "9478686570769398934"
       },
       "execution_count": 87,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "92a5366a-574a-449f-8014-2b931a91ee77",
     "text": "hash_murmur(\"QSSSPEF\")"
    },
    "executionTime": "2019-10-11T19:55:10.685Z"
   },
   {
    "cell": {
     "executionCount": 88,
     "executionEventId": "55fe9c6e-a3f8-4e8e-98eb-0fc5caa8f1cf",
     "hasError": false,
     "id": "050e786a-84bf-42d1-af10-33099b13a4ad",
     "outputs": [
      {
       "data": {
        "text/plain": "0"
       },
       "execution_count": 88,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "bd39f49d-b475-49c9-81f0-ef32c88fa033",
     "text": "peptide_graph.get(9478686570769398934)"
    },
    "executionTime": "2019-10-11T19:55:22.947Z"
   },
   {
    "cell": {
     "executionCount": 89,
     "executionEventId": "2657e731-8090-475a-b5d8-c40c6259b25f",
     "hasError": true,
     "id": "da005993-16dc-4b87-9a8d-00dc853e6c0a",
     "outputs": [
      {
       "ename": "NameError",
       "evalue": "name 'tqdm' is not defined",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
        "\u001b[0;32m<ipython-input-89-fe03097526ab>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0;32mfor\u001b[0m \u001b[0mrecord\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mtqdm\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mscreed\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mopen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'Homo_sapiens.GRCh38.pep.all.fa.gz'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      2\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0;34m'QSSSPEFRVQSFSERTNARKKNNH'\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mrecord\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'sequence'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      3\u001b[0m         \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrecord\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0;34m\"TOPORS\"\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mrecord\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'name'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m         \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrecord\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;31mNameError\u001b[0m: name 'tqdm' is not defined"
       ]
      }
     ],
     "persistentId": "39826558-68ec-4086-b31f-de7f8d208545",
     "text": "for record in tqdm(screed.open('Homo_sapiens.GRCh38.pep.all.fa.gz')):\n    if 'QSSSPEFRVQSFSERTNARKKNNH' in record['sequence']:\n        print(record)\n    if \"TOPORS\" in record['name']:\n        print(record)"
    },
    "executionTime": "2019-10-11T19:56:42.096Z"
   },
   {
    "cell": {
     "executionCount": 90,
     "executionEventId": "10465d2d-5dac-478d-b637-38b533a3c2a1",
     "hasError": false,
     "id": "81fe1d1b-9150-4e3a-a0e2-f2470c0bb85c",
     "outputs": [],
     "persistentId": "56e48bd4-fcc6-4d93-b4e3-0984be1fc1d5",
     "text": "import gzip\nfrom io import StringIO\nfrom pprint import pprint\n\nfrom khmer import Nodegraph\nimport matplotlib.pyplot as plt\nimport screed\nfrom sourmash._minhash import hash_murmur\nfrom khmer.khmer_args import calculate_graphsize\nimport pandas as pd\nfrom sourmash.logging import notify\nimport seaborn as sns\nfrom sklearn import metrics\nfrom tqdm import tqdm\nfrom Bio.Seq import Seq\nfrom Bio import SeqIO\n\n# make all warnings error to get a stack trace\nimport warnings\nwarnings.filterwarnings(\"default\")\n\npd.options.display.max_colwidth = 100\n\nfrom khtools.compare_kmer_content import kmerize, hpize, dayhoffize\n"
    },
    "executionTime": "2019-10-11T19:57:03.167Z"
   },
   {
    "cell": {
     "executionCount": 91,
     "executionEventId": "37da8ed2-e5f8-4396-a34a-981cf99a34e5",
     "hasError": false,
     "id": "da005993-16dc-4b87-9a8d-00dc853e6c0a",
     "outputs": [
      {
       "name": "stderr",
       "output_type": "stream",
       "text": "/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/screed/openscreed.py:39: ResourceWarning: unclosed file <_io.BufferedReader name='Homo_sapiens.GRCh38.pep.all.fa.gz'>\n  self.iter_fn = self.open_reader(filename, *args, **kwargs)\n45723it [00:01, 35755.41it/s]"
      },
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "{'name': 'ENSP00000353735.2 pep chromosome:GRCh38:9:32540544:32552586:-1 gene:ENSG00000197579.8 transcript:ENST00000360538.7 gene_biotype:protein_coding transcript_biotype:protein_coding gene_symbol:TOPORS description:TOP1 binding arginine/serine rich protein [Source:HGNC Symbol;Acc:HGNC:21653]', 'sequence': 'MGSQPPLGSPLSREEGEAPPPAPASEGRRRSRRVRLRGSCRHRPSFLGCRELAASAPARPAPASSEIMASAAKEFKMDNFSPKAGTSKLQQTVPADASPDSKCPICLDRFDNVSYLDRCLHKFCFRCVQEWSKNKAECPLCKQPFDSIFHSVRAEDDFKEYVLRPSYNGSFVTPDRRFRYRTTLTRERNASVYSPSGPVNRRTTTPPDSGVLFEGLGISTRPRDVEIPQFMRQIAVRRPTTADERSLRKIQEQDIINFRRTLYRAGARVRNIEDGGRYRDISAEFFRRNPACLHRLVPWLKRELTVLFGAHGSLVNIVQHIIMSNVTRYDLESQAFVSDLRPFLLNRTEHFIHEFISFARSPFNMAAFDQHANYDCPAPSYEEGSHSDSSVITISPDEAETQELDINVATVSQAPWDDETPGPSYSSSEQVHVTMSSLLNTSDSSDEELVTGGATSQIQGVQTNDDLNNDSDDSSDNCVIVGFVKPLAERTPELVELSSDSEDLGSYEKMETVKTQEQEQSYSSGDSDVSRCSSPHSVLGKDEQINKGHCDSSTRIKSKKEEKRSTSLSSPRNLNSSVRGDRVYSPYNHRHRKRGRSRSSDSRSQSRSGHDQKNHRKHHGKKRMKSKRSRSRESSRPRGRRDKKRSRTRDSSWSRRSQTLSLSSESTSRSRSRSSDHGKRRSRSRNRDRYYLRNNYGSRYKWEYTYYSRNKDRDGYESSYRRRTLSRAHYSRQSSSPEFRVQSFSERTNARKKNNHSERKYYYYERHRSRSLSSNRSRTASTGTDRVRNEKPGGKRKYKTRHLEGTNEVAQPSREFASKAKDSHYQKSSSKLDGNYKNESDTFSDSRSSDRETKHKRRKRKTRSLSVEIVYEGKATDTTKHHKKKKKKHKKKHKKHHGDNASRSPVVITIDSDSDKDSEVKEDTECDNSGPQDPLQNEFLAPSLEPFETKDVVTIEAEFGVLDKECDIATLSNNLNNANKTVDNIPPLAASVEQTLDVREESTFVSDLENQPSNIVSLQTEPSRQLPSPRTSLMSVCLGRDCDMS', 'description': ''}\n{'name': 'ENSP00000353735.2 pep chromosome:GRCh38:9:32540544:32552586:-1 gene:ENSG00000197579.8 transcript:ENST00000360538.7 gene_biotype:protein_coding transcript_biotype:protein_coding gene_symbol:TOPORS description:TOP1 binding arginine/serine rich protein [Source:HGNC Symbol;Acc:HGNC:21653]', 'sequence': 'MGSQPPLGSPLSREEGEAPPPAPASEGRRRSRRVRLRGSCRHRPSFLGCRELAASAPARPAPASSEIMASAAKEFKMDNFSPKAGTSKLQQTVPADASPDSKCPICLDRFDNVSYLDRCLHKFCFRCVQEWSKNKAECPLCKQPFDSIFHSVRAEDDFKEYVLRPSYNGSFVTPDRRFRYRTTLTRERNASVYSPSGPVNRRTTTPPDSGVLFEGLGISTRPRDVEIPQFMRQIAVRRPTTADERSLRKIQEQDIINFRRTLYRAGARVRNIEDGGRYRDISAEFFRRNPACLHRLVPWLKRELTVLFGAHGSLVNIVQHIIMSNVTRYDLESQAFVSDLRPFLLNRTEHFIHEFISFARSPFNMAAFDQHANYDCPAPSYEEGSHSDSSVITISPDEAETQELDINVATVSQAPWDDETPGPSYSSSEQVHVTMSSLLNTSDSSDEELVTGGATSQIQGVQTNDDLNNDSDDSSDNCVIVGFVKPLAERTPELVELSSDSEDLGSYEKMETVKTQEQEQSYSSGDSDVSRCSSPHSVLGKDEQINKGHCDSSTRIKSKKEEKRSTSLSSPRNLNSSVRGDRVYSPYNHRHRKRGRSRSSDSRSQSRSGHDQKNHRKHHGKKRMKSKRSRSRESSRPRGRRDKKRSRTRDSSWSRRSQTLSLSSESTSRSRSRSSDHGKRRSRSRNRDRYYLRNNYGSRYKWEYTYYSRNKDRDGYESSYRRRTLSRAHYSRQSSSPEFRVQSFSERTNARKKNNHSERKYYYYERHRSRSLSSNRSRTASTGTDRVRNEKPGGKRKYKTRHLEGTNEVAQPSREFASKAKDSHYQKSSSKLDGNYKNESDTFSDSRSSDRETKHKRRKRKTRSLSVEIVYEGKATDTTKHHKKKKKKHKKKHKKHHGDNASRSPVVITIDSDSDKDSEVKEDTECDNSGPQDPLQNEFLAPSLEPFETKDVVTIEAEFGVLDKECDIATLSNNLNNANKTVDNIPPLAASVEQTLDVREESTFVSDLENQPSNIVSLQTEPSRQLPSPRTSLMSVCLGRDCDMS', 'description': ''}\n{'name': 'ENSP00000369187.1 pep chromosome:GRCh38:9:32540826:32552553:-1 gene:ENSG00000197579.8 transcript:ENST00000379858.1 gene_biotype:protein_coding transcript_biotype:protein_coding gene_symbol:TOPORS description:TOP1 binding arginine/serine rich protein [Source:HGNC Symbol;Acc:HGNC:21653]', 'sequence': 'MIMASAAKEFKMDNFSPKAGTSKLQQTVPADASPDSKCPICLDRFDNVSYLDRCLHKFCFRCVQEWSKNKAECPLCKQPFDSIFHSVRAEDDFKEYVLRPSYNGSFVTPDRRFRYRTTLTRERNASVYSPSGPVNRRTTTPPDSGVLFEGLGISTRPRDVEIPQFMRQIAVRRPTTADERSLRKIQEQDIINFRRTLYRAGARVRNIEDGGRYRDISAEFFRRNPACLHRLVPWLKRELTVLFGAHGSLVNIVQHIIMSNVTRYDLESQAFVSDLRPFLLNRTEHFIHEFISFARSPFNMAAFDQHANYDCPAPSYEEGSHSDSSVITISPDEAETQELDINVATVSQAPWDDETPGPSYSSSEQVHVTMSSLLNTSDSSDEELVTGGATSQIQGVQTNDDLNNDSDDSSDNCVIVGFVKPLAERTPELVELSSDSEDLGSYEKMETVKTQEQEQSYSSGDSDVSRCSSPHSVLGKDEQINKGHCDSSTRIKSKKEEKRSTSLSSPRNLNSSVRGDRVYSPYNHRHRKRGRSRSSDSRSQSRSGHDQKNHRKHHGKKRMKSKRSRSRESSRPRGRRDKKRSRTRDSSWSRRSQTLSLSSESTSRSRSRSSDHGKRRSRSRNRDRYYLRNNYGSRYKWEYTYYSRNKDRDGYESSYRRRTLSRAHYSRQSSSPEFRVQSFSERTNARKKNNHSERKYYYYERHRSRSLSSNRSRTASTGTDRVRNEKPGGKRKYKTRHLEGTNEVAQPSREFASKAKDSHYQKSSSKLDGNYKNESDTFSDSRSSDRETKHKRRKRKTRSLSVEIVYEGKATDTTKHHKKKKKKHKKKHKKHHGDNASRSPVVITIDSDSDKDSEVKEDTECDNSGPQDPLQNEFLAPSLEPFETKDVVTIEAEFGVLDKECDIATLSNNLNNANKTVDNIPPLAASVEQTLDVREESTFVSDLENQPSNIVSLQTEPSRQLPSPRTSLMSVCLGRDCDMS', 'description': ''}\n{'name': 'ENSP00000369187.1 pep chromosome:GRCh38:9:32540826:32552553:-1 gene:ENSG00000197579.8 transcript:ENST00000379858.1 gene_biotype:protein_coding transcript_biotype:protein_coding gene_symbol:TOPORS description:TOP1 binding arginine/serine rich protein [Source:HGNC Symbol;Acc:HGNC:21653]', 'sequence': 'MIMASAAKEFKMDNFSPKAGTSKLQQTVPADASPDSKCPICLDRFDNVSYLDRCLHKFCFRCVQEWSKNKAECPLCKQPFDSIFHSVRAEDDFKEYVLRPSYNGSFVTPDRRFRYRTTLTRERNASVYSPSGPVNRRTTTPPDSGVLFEGLGISTRPRDVEIPQFMRQIAVRRPTTADERSLRKIQEQDIINFRRTLYRAGARVRNIEDGGRYRDISAEFFRRNPACLHRLVPWLKRELTVLFGAHGSLVNIVQHIIMSNVTRYDLESQAFVSDLRPFLLNRTEHFIHEFISFARSPFNMAAFDQHANYDCPAPSYEEGSHSDSSVITISPDEAETQELDINVATVSQAPWDDETPGPSYSSSEQVHVTMSSLLNTSDSSDEELVTGGATSQIQGVQTNDDLNNDSDDSSDNCVIVGFVKPLAERTPELVELSSDSEDLGSYEKMETVKTQEQEQSYSSGDSDVSRCSSPHSVLGKDEQINKGHCDSSTRIKSKKEEKRSTSLSSPRNLNSSVRGDRVYSPYNHRHRKRGRSRSSDSRSQSRSGHDQKNHRKHHGKKRMKSKRSRSRESSRPRGRRDKKRSRTRDSSWSRRSQTLSLSSESTSRSRSRSSDHGKRRSRSRNRDRYYLRNNYGSRYKWEYTYYSRNKDRDGYESSYRRRTLSRAHYSRQSSSPEFRVQSFSERTNARKKNNHSERKYYYYERHRSRSLSSNRSRTASTGTDRVRNEKPGGKRKYKTRHLEGTNEVAQPSREFASKAKDSHYQKSSSKLDGNYKNESDTFSDSRSSDRETKHKRRKRKTRSLSVEIVYEGKATDTTKHHKKKKKKHKKKHKKHHGDNASRSPVVITIDSDSDKDSEVKEDTECDNSGPQDPLQNEFLAPSLEPFETKDVVTIEAEFGVLDKECDIATLSNNLNNANKTVDNIPPLAASVEQTLDVREESTFVSDLENQPSNIVSLQTEPSRQLPSPRTSLMSVCLGRDCDMS', 'description': ''}\n"
      },
      {
       "name": "stderr",
       "output_type": "stream",
       "text": "110048it [00:03, 35579.69it/s]\n"
      }
     ],
     "persistentId": "39826558-68ec-4086-b31f-de7f8d208545",
     "text": "for record in tqdm(screed.open('Homo_sapiens.GRCh38.pep.all.fa.gz')):\n    if 'QSSSPEFRVQSFSERTNARKKNNH' in record['sequence']:\n        print(record)\n    if \"TOPORS\" in record['name']:\n        print(record)"
    },
    "executionTime": "2019-10-11T19:57:09.914Z"
   },
   {
    "cell": {
     "executionCount": 92,
     "executionEventId": "6e534b29-89d6-45fd-aa4d-c58390afa0fc",
     "hasError": false,
     "id": "da005993-16dc-4b87-9a8d-00dc853e6c0a",
     "outputs": [
      {
       "name": "stderr",
       "output_type": "stream",
       "text": "/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/screed/openscreed.py:39: ResourceWarning: unclosed file <_io.BufferedReader name='Homo_sapiens.GRCh38.pep.all.fa.gz'>\n  self.iter_fn = self.open_reader(filename, *args, **kwargs)\n46151it [00:01, 36395.48it/s]"
      },
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "---- QSSSPEFRVQSFSERTNARKKNNH ----\n{'name': 'ENSP00000353735.2 pep chromosome:GRCh38:9:32540544:32552586:-1 gene:ENSG00000197579.8 transcript:ENST00000360538.7 gene_biotype:protein_coding transcript_biotype:protein_coding gene_symbol:TOPORS description:TOP1 binding arginine/serine rich protein [Source:HGNC Symbol;Acc:HGNC:21653]', 'sequence': 'MGSQPPLGSPLSREEGEAPPPAPASEGRRRSRRVRLRGSCRHRPSFLGCRELAASAPARPAPASSEIMASAAKEFKMDNFSPKAGTSKLQQTVPADASPDSKCPICLDRFDNVSYLDRCLHKFCFRCVQEWSKNKAECPLCKQPFDSIFHSVRAEDDFKEYVLRPSYNGSFVTPDRRFRYRTTLTRERNASVYSPSGPVNRRTTTPPDSGVLFEGLGISTRPRDVEIPQFMRQIAVRRPTTADERSLRKIQEQDIINFRRTLYRAGARVRNIEDGGRYRDISAEFFRRNPACLHRLVPWLKRELTVLFGAHGSLVNIVQHIIMSNVTRYDLESQAFVSDLRPFLLNRTEHFIHEFISFARSPFNMAAFDQHANYDCPAPSYEEGSHSDSSVITISPDEAETQELDINVATVSQAPWDDETPGPSYSSSEQVHVTMSSLLNTSDSSDEELVTGGATSQIQGVQTNDDLNNDSDDSSDNCVIVGFVKPLAERTPELVELSSDSEDLGSYEKMETVKTQEQEQSYSSGDSDVSRCSSPHSVLGKDEQINKGHCDSSTRIKSKKEEKRSTSLSSPRNLNSSVRGDRVYSPYNHRHRKRGRSRSSDSRSQSRSGHDQKNHRKHHGKKRMKSKRSRSRESSRPRGRRDKKRSRTRDSSWSRRSQTLSLSSESTSRSRSRSSDHGKRRSRSRNRDRYYLRNNYGSRYKWEYTYYSRNKDRDGYESSYRRRTLSRAHYSRQSSSPEFRVQSFSERTNARKKNNHSERKYYYYERHRSRSLSSNRSRTASTGTDRVRNEKPGGKRKYKTRHLEGTNEVAQPSREFASKAKDSHYQKSSSKLDGNYKNESDTFSDSRSSDRETKHKRRKRKTRSLSVEIVYEGKATDTTKHHKKKKKKHKKKHKKHHGDNASRSPVVITIDSDSDKDSEVKEDTECDNSGPQDPLQNEFLAPSLEPFETKDVVTIEAEFGVLDKECDIATLSNNLNNANKTVDNIPPLAASVEQTLDVREESTFVSDLENQPSNIVSLQTEPSRQLPSPRTSLMSVCLGRDCDMS', 'description': ''}\n---- TOPORS ----\n{'name': 'ENSP00000353735.2 pep chromosome:GRCh38:9:32540544:32552586:-1 gene:ENSG00000197579.8 transcript:ENST00000360538.7 gene_biotype:protein_coding transcript_biotype:protein_coding gene_symbol:TOPORS description:TOP1 binding arginine/serine rich protein [Source:HGNC Symbol;Acc:HGNC:21653]', 'sequence': 'MGSQPPLGSPLSREEGEAPPPAPASEGRRRSRRVRLRGSCRHRPSFLGCRELAASAPARPAPASSEIMASAAKEFKMDNFSPKAGTSKLQQTVPADASPDSKCPICLDRFDNVSYLDRCLHKFCFRCVQEWSKNKAECPLCKQPFDSIFHSVRAEDDFKEYVLRPSYNGSFVTPDRRFRYRTTLTRERNASVYSPSGPVNRRTTTPPDSGVLFEGLGISTRPRDVEIPQFMRQIAVRRPTTADERSLRKIQEQDIINFRRTLYRAGARVRNIEDGGRYRDISAEFFRRNPACLHRLVPWLKRELTVLFGAHGSLVNIVQHIIMSNVTRYDLESQAFVSDLRPFLLNRTEHFIHEFISFARSPFNMAAFDQHANYDCPAPSYEEGSHSDSSVITISPDEAETQELDINVATVSQAPWDDETPGPSYSSSEQVHVTMSSLLNTSDSSDEELVTGGATSQIQGVQTNDDLNNDSDDSSDNCVIVGFVKPLAERTPELVELSSDSEDLGSYEKMETVKTQEQEQSYSSGDSDVSRCSSPHSVLGKDEQINKGHCDSSTRIKSKKEEKRSTSLSSPRNLNSSVRGDRVYSPYNHRHRKRGRSRSSDSRSQSRSGHDQKNHRKHHGKKRMKSKRSRSRESSRPRGRRDKKRSRTRDSSWSRRSQTLSLSSESTSRSRSRSSDHGKRRSRSRNRDRYYLRNNYGSRYKWEYTYYSRNKDRDGYESSYRRRTLSRAHYSRQSSSPEFRVQSFSERTNARKKNNHSERKYYYYERHRSRSLSSNRSRTASTGTDRVRNEKPGGKRKYKTRHLEGTNEVAQPSREFASKAKDSHYQKSSSKLDGNYKNESDTFSDSRSSDRETKHKRRKRKTRSLSVEIVYEGKATDTTKHHKKKKKKHKKKHKKHHGDNASRSPVVITIDSDSDKDSEVKEDTECDNSGPQDPLQNEFLAPSLEPFETKDVVTIEAEFGVLDKECDIATLSNNLNNANKTVDNIPPLAASVEQTLDVREESTFVSDLENQPSNIVSLQTEPSRQLPSPRTSLMSVCLGRDCDMS', 'description': ''}\n---- QSSSPEFRVQSFSERTNARKKNNH ----\n{'name': 'ENSP00000369187.1 pep chromosome:GRCh38:9:32540826:32552553:-1 gene:ENSG00000197579.8 transcript:ENST00000379858.1 gene_biotype:protein_coding transcript_biotype:protein_coding gene_symbol:TOPORS description:TOP1 binding arginine/serine rich protein [Source:HGNC Symbol;Acc:HGNC:21653]', 'sequence': 'MIMASAAKEFKMDNFSPKAGTSKLQQTVPADASPDSKCPICLDRFDNVSYLDRCLHKFCFRCVQEWSKNKAECPLCKQPFDSIFHSVRAEDDFKEYVLRPSYNGSFVTPDRRFRYRTTLTRERNASVYSPSGPVNRRTTTPPDSGVLFEGLGISTRPRDVEIPQFMRQIAVRRPTTADERSLRKIQEQDIINFRRTLYRAGARVRNIEDGGRYRDISAEFFRRNPACLHRLVPWLKRELTVLFGAHGSLVNIVQHIIMSNVTRYDLESQAFVSDLRPFLLNRTEHFIHEFISFARSPFNMAAFDQHANYDCPAPSYEEGSHSDSSVITISPDEAETQELDINVATVSQAPWDDETPGPSYSSSEQVHVTMSSLLNTSDSSDEELVTGGATSQIQGVQTNDDLNNDSDDSSDNCVIVGFVKPLAERTPELVELSSDSEDLGSYEKMETVKTQEQEQSYSSGDSDVSRCSSPHSVLGKDEQINKGHCDSSTRIKSKKEEKRSTSLSSPRNLNSSVRGDRVYSPYNHRHRKRGRSRSSDSRSQSRSGHDQKNHRKHHGKKRMKSKRSRSRESSRPRGRRDKKRSRTRDSSWSRRSQTLSLSSESTSRSRSRSSDHGKRRSRSRNRDRYYLRNNYGSRYKWEYTYYSRNKDRDGYESSYRRRTLSRAHYSRQSSSPEFRVQSFSERTNARKKNNHSERKYYYYERHRSRSLSSNRSRTASTGTDRVRNEKPGGKRKYKTRHLEGTNEVAQPSREFASKAKDSHYQKSSSKLDGNYKNESDTFSDSRSSDRETKHKRRKRKTRSLSVEIVYEGKATDTTKHHKKKKKKHKKKHKKHHGDNASRSPVVITIDSDSDKDSEVKEDTECDNSGPQDPLQNEFLAPSLEPFETKDVVTIEAEFGVLDKECDIATLSNNLNNANKTVDNIPPLAASVEQTLDVREESTFVSDLENQPSNIVSLQTEPSRQLPSPRTSLMSVCLGRDCDMS', 'description': ''}\n---- TOPORS ----\n{'name': 'ENSP00000369187.1 pep chromosome:GRCh38:9:32540826:32552553:-1 gene:ENSG00000197579.8 transcript:ENST00000379858.1 gene_biotype:protein_coding transcript_biotype:protein_coding gene_symbol:TOPORS description:TOP1 binding arginine/serine rich protein [Source:HGNC Symbol;Acc:HGNC:21653]', 'sequence': 'MIMASAAKEFKMDNFSPKAGTSKLQQTVPADASPDSKCPICLDRFDNVSYLDRCLHKFCFRCVQEWSKNKAECPLCKQPFDSIFHSVRAEDDFKEYVLRPSYNGSFVTPDRRFRYRTTLTRERNASVYSPSGPVNRRTTTPPDSGVLFEGLGISTRPRDVEIPQFMRQIAVRRPTTADERSLRKIQEQDIINFRRTLYRAGARVRNIEDGGRYRDISAEFFRRNPACLHRLVPWLKRELTVLFGAHGSLVNIVQHIIMSNVTRYDLESQAFVSDLRPFLLNRTEHFIHEFISFARSPFNMAAFDQHANYDCPAPSYEEGSHSDSSVITISPDEAETQELDINVATVSQAPWDDETPGPSYSSSEQVHVTMSSLLNTSDSSDEELVTGGATSQIQGVQTNDDLNNDSDDSSDNCVIVGFVKPLAERTPELVELSSDSEDLGSYEKMETVKTQEQEQSYSSGDSDVSRCSSPHSVLGKDEQINKGHCDSSTRIKSKKEEKRSTSLSSPRNLNSSVRGDRVYSPYNHRHRKRGRSRSSDSRSQSRSGHDQKNHRKHHGKKRMKSKRSRSRESSRPRGRRDKKRSRTRDSSWSRRSQTLSLSSESTSRSRSRSSDHGKRRSRSRNRDRYYLRNNYGSRYKWEYTYYSRNKDRDGYESSYRRRTLSRAHYSRQSSSPEFRVQSFSERTNARKKNNHSERKYYYYERHRSRSLSSNRSRTASTGTDRVRNEKPGGKRKYKTRHLEGTNEVAQPSREFASKAKDSHYQKSSSKLDGNYKNESDTFSDSRSSDRETKHKRRKRKTRSLSVEIVYEGKATDTTKHHKKKKKKHKKKHKKHHGDNASRSPVVITIDSDSDKDSEVKEDTECDNSGPQDPLQNEFLAPSLEPFETKDVVTIEAEFGVLDKECDIATLSNNLNNANKTVDNIPPLAASVEQTLDVREESTFVSDLENQPSNIVSLQTEPSRQLPSPRTSLMSVCLGRDCDMS', 'description': ''}\n"
      },
      {
       "name": "stderr",
       "output_type": "stream",
       "text": "110048it [00:03, 36127.01it/s]\n"
      }
     ],
     "persistentId": "39826558-68ec-4086-b31f-de7f8d208545",
     "text": "for record in tqdm(screed.open('Homo_sapiens.GRCh38.pep.all.fa.gz')):\n    if 'QSSSPEFRVQSFSERTNARKKNNH' in record['sequence']:\n        print(\"---- QSSSPEFRVQSFSERTNARKKNNH ----\")\n        print(record)\n    if \"TOPORS\" in record['name']:\n        print(\"---- TOPORS ----\")\n        print(record)"
    },
    "executionTime": "2019-10-11T19:57:31.143Z"
   },
   {
    "cell": {
     "executionCount": 94,
     "executionEventId": "2aeabda8-8fa4-4253-b64e-9a6bdeb0d017",
     "hasError": false,
     "id": "b242d89f-8d25-4127-b875-8a85fa69d914",
     "outputs": [
      {
       "name": "stderr",
       "output_type": "stream",
       "text": "/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/screed/openscreed.py:39: ResourceWarning: unclosed file <_io.BufferedReader name='SRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.fq.gz'>\n  self.iter_fn = self.open_reader(filename, *args, **kwargs)\n1000it [00:00, 2601.49it/s]"
      },
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "CPU times: user 388 ms, sys: 12 ms, total: 400 ms\nWall time: 395 ms\n"
      },
      {
       "name": "stderr",
       "output_type": "stream",
       "text": "\n"
      }
     ],
     "persistentId": "be06ded1-d7d4-4f21-9961-3aed5ff30cbf",
     "text": "%%time\n\nreads = 'SRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz'\nsubsampled_reads = 'SRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.fq.gz'\n\ncoding_scores = score_reads(subsampled_reads, peptide_graph, peptide_ksize)"
    },
    "executionTime": "2019-10-11T20:04:30.073Z"
   },
   {
    "cell": {
     "executionCount": 95,
     "executionEventId": "d7d24951-3cae-443a-9e95-a1708685810f",
     "hasError": false,
     "id": "cfea0c3d-642a-421a-a4ad-07fa69bea946",
     "outputs": [
      {
       "data": {
        "text/plain": "(1000, 4)"
       },
       "execution_count": 95,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "09393220-d1f9-4ddd-9c7c-bc15160f3680",
     "text": "coding_scores.shape"
    },
    "executionTime": "2019-10-11T20:04:30.245Z"
   },
   {
    "cell": {
     "executionCount": 96,
     "executionEventId": "f522ea74-9c77-4353-b959-7e2dcf9ae7a2",
     "hasError": false,
     "id": "3281c496-16a4-496e-81d4-fa57e8d1a556",
     "outputs": [
      {
       "data": {
        "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>read_id</th>\n      <th>jaccard_in_peptide_db</th>\n      <th>n_kmers</th>\n      <th>classification</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>SRR306838.10559374 Ibis_Run100924_C3PO:6:51:17601:17119/1</td>\n      <td>1.000000</td>\n      <td>16</td>\n      <td>coding</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>SRR306838.6196593 Ibis_Run100924_C3PO:6:29:16733:12435/1</td>\n      <td>0.000000</td>\n      <td>17</td>\n      <td>non-coding</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>SRR306838.20767303 Ibis_Run100924_C3PO:6:104:6864:5062/1</td>\n      <td>0.187500</td>\n      <td>16</td>\n      <td>non-coding</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>SRR306838.12582274 Ibis_Run100924_C3PO:6:62:11779:17975/1</td>\n      <td>0.066667</td>\n      <td>15</td>\n      <td>non-coding</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>SRR306838.13334230 Ibis_Run100924_C3PO:6:66:16579:20350/1</td>\n      <td>0.000000</td>\n      <td>12</td>\n      <td>non-coding</td>\n    </tr>\n  </tbody>\n</table>\n</div>",
        "text/plain": "                                                     read_id  \\\n0  SRR306838.10559374 Ibis_Run100924_C3PO:6:51:17601:17119/1   \n1   SRR306838.6196593 Ibis_Run100924_C3PO:6:29:16733:12435/1   \n2   SRR306838.20767303 Ibis_Run100924_C3PO:6:104:6864:5062/1   \n3  SRR306838.12582274 Ibis_Run100924_C3PO:6:62:11779:17975/1   \n4  SRR306838.13334230 Ibis_Run100924_C3PO:6:66:16579:20350/1   \n\n   jaccard_in_peptide_db  n_kmers classification  \n0               1.000000       16         coding  \n1               0.000000       17     non-coding  \n2               0.187500       16     non-coding  \n3               0.066667       15     non-coding  \n4               0.000000       12     non-coding  "
       },
       "execution_count": 96,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "ede6e1ba-f524-4ceb-b35b-6c20f91d437c",
     "text": "coding_scores.head()"
    },
    "executionTime": "2019-10-11T20:04:30.264Z"
   },
   {
    "cell": {
     "executionCount": 98,
     "executionEventId": "17c7fa8c-22eb-41da-b788-c94ab3ef7738",
     "hasError": false,
     "id": "4c484434-ea07-4b72-8073-567bf902bedb",
     "outputs": [
      {
       "data": {
        "text/plain": "150289"
       },
       "execution_count": 98,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "75c01782-427c-4c63-bc29-ed308ae4a142",
     "text": "true_coding_read_ids = set(coding_read_ids.values)\nlen(true_coding_read_ids)"
    },
    "executionTime": "2019-10-11T20:04:30.904Z"
   },
   {
    "cell": {
     "executionCount": 99,
     "executionEventId": "a0cd6b3b-c965-49f8-9d44-f28f6cb5b5b2",
     "hasError": false,
     "id": "9babb580-ed7e-4a80-bcb3-3d080fca4a29",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "(807698,)\n"
      }
     ],
     "persistentId": "317bb484-c93f-4e1c-b831-50734c9e6614",
     "text": "unspliced_read_ids_series = pd.read_csv(f\"{folder}/SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.read_ids.txt\", \n                                   header=None, squeeze=True)\nprint(unspliced_read_ids_series.shape)\nunspliced_read_ids_series.head()\n\nunspliced_reads = set(unspliced_read_ids_series.values)"
    },
    "executionTime": "2019-10-11T20:04:31.613Z"
   },
   {
    "cell": {
     "executionCount": 101,
     "executionEventId": "56105ae4-a7f5-463e-a4d8-8e280bb0aa43",
     "hasError": false,
     "id": "efba9e80-c61d-4adb-9d6c-af7db00a9356",
     "outputs": [
      {
       "data": {
        "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>read_id</th>\n      <th>jaccard_in_peptide_db</th>\n      <th>n_kmers</th>\n      <th>classification</th>\n      <th>short_read_id</th>\n      <th>in_cds_no_stop_codon</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>52</th>\n      <td>SRR306838.17561853 Ibis_Run100924_C3PO:6:88:9412:18824/1</td>\n      <td>0.000000</td>\n      <td>15</td>\n      <td>non-coding</td>\n      <td>SRR306838.17561853</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>55</th>\n      <td>SRR306838.733972 Ibis_Run100924_C3PO:6:4:4934:4948/1</td>\n      <td>0.100000</td>\n      <td>10</td>\n      <td>non-coding</td>\n      <td>SRR306838.733972</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>61</th>\n      <td>SRR306838.3607263 Ibis_Run100924_C3PO:6:17:17893:5364/1</td>\n      <td>0.294118</td>\n      <td>17</td>\n      <td>non-coding</td>\n      <td>SRR306838.3607263</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>81</th>\n      <td>SRR306838.15455056 Ibis_Run100924_C3PO:6:78:4010:4797/1</td>\n      <td>0.312500</td>\n      <td>16</td>\n      <td>non-coding</td>\n      <td>SRR306838.15455056</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>89</th>\n      <td>SRR306838.14450220 Ibis_Run100924_C3PO:6:72:5878:19983/1</td>\n      <td>0.062500</td>\n      <td>16</td>\n      <td>non-coding</td>\n      <td>SRR306838.14450220</td>\n      <td>True</td>\n    </tr>\n  </tbody>\n</table>\n</div>",
        "text/plain": "                                                     read_id  \\\n52  SRR306838.17561853 Ibis_Run100924_C3PO:6:88:9412:18824/1   \n55      SRR306838.733972 Ibis_Run100924_C3PO:6:4:4934:4948/1   \n61   SRR306838.3607263 Ibis_Run100924_C3PO:6:17:17893:5364/1   \n81   SRR306838.15455056 Ibis_Run100924_C3PO:6:78:4010:4797/1   \n89  SRR306838.14450220 Ibis_Run100924_C3PO:6:72:5878:19983/1   \n\n    jaccard_in_peptide_db  n_kmers classification       short_read_id  \\\n52               0.000000       15     non-coding  SRR306838.17561853   \n55               0.100000       10     non-coding    SRR306838.733972   \n61               0.294118       17     non-coding   SRR306838.3607263   \n81               0.312500       16     non-coding  SRR306838.15455056   \n89               0.062500       16     non-coding  SRR306838.14450220   \n\n    in_cds_no_stop_codon  \n52                  True  \n55                  True  \n61                  True  \n81                  True  \n89                  True  "
       },
       "execution_count": 101,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "1254868e-73a0-4c0a-82c6-bb1b85e78cb0",
     "text": "coding_scores_unspliced.query(\"jaccard_in_peptide_db < 0.5 & in_cds_no_stop_codon == True\").head()"
    },
    "executionTime": "2019-10-11T20:04:32.290Z"
   },
   {
    "cell": {
     "executionCount": 105,
     "executionEventId": "42ffdc5d-7479-45b2-9d39-728e67713a1c",
     "hasError": true,
     "id": "f6cb1260-f3d3-4c84-b1e7-229169dcdd93",
     "outputs": [
      {
       "ename": "NameError",
       "evalue": "name 'score_df_unspliced' is not defined",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
        "\u001b[0;32m<ipython-input-105-aa9fb6c8eb6b>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0msns\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mviolinplot\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mx\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'in_cds_no_stop_codon'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0my\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'coding_score'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdata\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mscore_df_unspliced\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mbw\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m0.1\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcut\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
        "\u001b[0;31mNameError\u001b[0m: name 'score_df_unspliced' is not defined"
       ]
      }
     ],
     "persistentId": "5bb1b19d-97af-43fc-a51d-420946c4c00f",
     "text": "sns.violinplot(x='in_cds_no_stop_codon', y='coding_score', data=score_df_unspliced, bw=0.1, cut=True)"
    },
    "executionTime": "2019-10-11T20:07:08.176Z"
   },
   {
    "cell": {
     "executionCount": 106,
     "executionEventId": "3352d241-8258-4ba1-91f7-3b49e8d1ba00",
     "hasError": true,
     "id": "f6cb1260-f3d3-4c84-b1e7-229169dcdd93",
     "outputs": [
      {
       "ename": "ValueError",
       "evalue": "Could not interpret input 'coding_score'",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mValueError\u001b[0m                                Traceback (most recent call last)",
        "\u001b[0;32m<ipython-input-106-1f8379b9fe89>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0msns\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mviolinplot\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mx\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'in_cds_no_stop_codon'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0my\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'coding_score'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdata\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mcoding_scores_unspliced\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mbw\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m0.1\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcut\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/seaborn/categorical.py\u001b[0m in \u001b[0;36mviolinplot\u001b[0;34m(x, y, hue, data, order, hue_order, bw, cut, scale, scale_hue, gridsize, width, inner, split, dodge, orient, linewidth, color, palette, saturation, ax, **kwargs)\u001b[0m\n\u001b[1;32m   2385\u001b[0m                              \u001b[0mbw\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcut\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mscale\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mscale_hue\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mgridsize\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2386\u001b[0m                              \u001b[0mwidth\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0minner\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msplit\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdodge\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0morient\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlinewidth\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2387\u001b[0;31m                              color, palette, saturation)\n\u001b[0m\u001b[1;32m   2388\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2389\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0max\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/seaborn/categorical.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, x, y, hue, data, order, hue_order, bw, cut, scale, scale_hue, gridsize, width, inner, split, dodge, orient, linewidth, color, palette, saturation)\u001b[0m\n\u001b[1;32m    560\u001b[0m                  color, palette, saturation):\n\u001b[1;32m    561\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 562\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mestablish_variables\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mx\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0my\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mhue\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdata\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0morient\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0morder\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mhue_order\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    563\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mestablish_colors\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcolor\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpalette\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msaturation\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    564\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mestimate_densities\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mbw\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcut\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mscale\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mscale_hue\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mgridsize\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/seaborn/categorical.py\u001b[0m in \u001b[0;36mestablish_variables\u001b[0;34m(self, x, y, hue, data, orient, order, hue_order, units)\u001b[0m\n\u001b[1;32m    153\u001b[0m                 \u001b[0;32mif\u001b[0m \u001b[0misinstance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0minput\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstring_types\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    154\u001b[0m                     \u001b[0merr\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m\"Could not interpret input '{}'\"\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mformat\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0minput\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 155\u001b[0;31m                     \u001b[0;32mraise\u001b[0m \u001b[0mValueError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0merr\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    156\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    157\u001b[0m             \u001b[0;31m# Figure out the plotting orientation\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;31mValueError\u001b[0m: Could not interpret input 'coding_score'"
       ]
      }
     ],
     "persistentId": "5bb1b19d-97af-43fc-a51d-420946c4c00f",
     "text": "sns.violinplot(x='in_cds_no_stop_codon', y='coding_score', data=coding_scores_unspliced, bw=0.1, cut=True)"
    },
    "executionTime": "2019-10-11T20:07:12.308Z"
   },
   {
    "cell": {
     "executionCount": 107,
     "executionEventId": "22e21986-a30f-49c4-8a5f-cccf52702516",
     "hasError": false,
     "id": "f6cb1260-f3d3-4c84-b1e7-229169dcdd93",
     "outputs": [
      {
       "data": {
        "text/plain": "<matplotlib.axes._subplots.AxesSubplot at 0x7f22f8050128>"
       },
       "execution_count": 107,
       "metadata": {},
       "output_type": "execute_result"
      },
      {
       "data": {
        "image/png": "iVBORw0KGgoAAAANSUhEUgAAAY4AAAELCAYAAADOeWEXAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDMuMC4zLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvnQurowAAIABJREFUeJzt3Xl8W2ed7/HPz7It73aczXFSx87SpElTkjalge4MdChkyj5QBi7tHW6BaQemzL0sd2Zoh8JrgFJmSlnaDlNKF8pw2wLdWyhNS+lCk5Bm6ZI9qbPY2ezY8Srpd//QcWIniiPVliVb3/frdV7SOefROT8rin56zvOc5zF3R0REJFl5mQ5ARERGFyUOERFJiRKHiIikRIlDRERSosQhIiIpUeIQEZGUKHGIiEhKlDhERCQlShwiIpKS/EwHkA4TJkzw+vr6TIchIjKqrFixYq+7TzxRuTGZOOrr61m+fHmmwxARGVXMbFsy5XSpSkREUqLEISIiKVHiEBGRlChxiIhISpQ4REQkJUocIiKSEiUOERFJiRKHiIikJOM3AJrZbcBSoNndT02w34AbgfcAHcBl7r5yZKMUyU3RaJR9+/axe/dumpubaWpqOrLs3knLgZZMh5h5ZlSPG8fkKbVMnjz5mKW6upq8vLH1Gz3jiQO4HfgBcMdx9l8MzA6Ws4AfB48yjHp6erj//vtZu3Zt0q9paGjg0ksvpaSkJI2RSbr19PSwa9cuduzYQWNj4+FlZ+MbNO/dRywWG1C+rNCYUBRlfLiXhjLHLEOBZwl3aGnfy461W1i9PMShXh+wPz8UYtLECUw9qY5p06YxdepUpk2bxrRp06ipqSE/Pxu+hlOT8Yjd/Rkzqx+kyPuAO9zdgRfMrMrMprj7rhEJMAe8+OKL3Hjj99m5cwcUV0Fe6MQvcufZZ5/loYcf4aor/453vOMdWK5/g4wCkUiETZs2sWbNGtasWcPrr75C8569xPzIl11poTG5KEJ9cYQzT4oyoSjG+GCZUBQlnMTHI93uWl/C6y0FdESMknxnTlUvnzi5I9NhAdAZMfZ15bG3Ky94DLGnq4Pm13ay9s8hOiNHyubl5TFl8iTmzpvPggULWLBgAfX19YRCWfAmDyLjiSMJU4E3+q03BtuUOIbBN7/5TX77299CcSUdJ19EtHJa0q/Na2/Gt7/AddddxyOPPMq3v/2tUfnrKRdEo1Gu/853WLbsKbq6ewCYWAIzy7pZMj3K5JIoNcUxJpdEKSvwExwt815vKaApUsbSpUt56KGHoKU90yEdVpzvTCuLMq0sesw+d2jrNZo6Q+zuCNHUkceujk5W/nE3Tz75JAClJcVc/J73cuWVV2btj7HR8L880Tt3zCfbzK4ArgCoq6tLd0xjxiuvvooXltA+7/3J1TT6iZVNov2UpZS88iCvvvYq7tn/hZOrHn30UR57/HHGh6P8zdxOFlT3Ul0UO/ELs1RHxFi6dClXXXUV7s4zD/8y0yElxQwqCp2KwgizK49UPdzb2duVx6p9hfxmS4x7772XRYsWcfbZZ2cw2uMbDYmjETip3/o0YOfRhdz9VuBWgMWLF+sbLEkfv/RSrr/+egqb1tIz+dTUkofHyN+3iVDHPv76sssoKChIX6AyJFVVVVRVVrCv9SB3bChjXlUPdWURppZGqS2NMqUkOy5BJask33nooYdwdx5++GEm54++//KdEdjVEWLHoXx2HgqxrS2fV1sLiMZg4oRqSktLMx3icY2GxPEAcJWZ/YJ4o3ir2jeGz0UXXcTvnnySP69cQXjP63TVLKB3wsmDJxCPkb9/C8W7XobOFmbOmsWHPvShkQtaUnbOOefwtre9jTVr1vD000+zYvmfWPPGrsMN3wZMKIHa4h4mF0eZUBxjYlG8fWNCUYzSLLt8VZzvdLV3cd9998XXq7IrPohflmqPGHs74+0ce7vy2NOVR1NHiJ2dhezrPFI2PxTipJOm8eGLlnDeeedxyimnZHVPrIwnDjO7B7gAmGBmjcA1QAGAu98MPEK8K+5G4t1xL89MpGNTQUEB37vhBlasWMFtt/2UV155nqLda+ic9lYi1fXHlA8d3EXJ9uehs4X6hhn8z8u/yDnnnJPVH3KJC4VCLFy4kIULFwLQ29tLY2Mj27ZtY9u2bWzdupVtWzbz+q7ddHV3D3htaUG8J9WEcITxRf0bzOPPywtyr3dVzOFgj7GvK8S+7oGN4Xu789nbGaIzMjChlRQXUVtby6LTZzB9+nTq6+uZPn06tbW1o6p9MOORuvulJ9jvwJUjFE5OMjMWL17MGWecwYoVK7j55lvYuPH39LbOpqtuCYQKIBYlvGMlhbvXUFs7lc9+VQljtCsoKKChoYGGhoYB292d1tZWdu/efcyya+cO1jU1HW5g71MYgvFFzvjCXsYXxZgY1FgmFceYWBwdlYnFHVp7jD1dIZo789jTGWJPkBz29RSyvxN6j2omKikuoqamhro5U3lrTQ01Ry1lZWVZ2+CdiownDskefQnk5psXcvvtt3P33XdT0N5E++yLKN28DDu0l0suuYTPfe5zFBcXZzpcSRMzi7eJVFUxd+7cY/a7O21tbQNvBjy87Gb17l0c2HVwwGvC+cbEoigTiyJMKo5SUxw93L6SyaTiDi09xs5DIXZ25LO7I4/mzhB7ugvY22n0HNUxqnpcFTU1U5hXU5PwZr+ysrLM/CEjTIlDjpGfn8+nP/1pzjzzTK6++mpK196PeYxrrrmGCy+8MNPhSYaZGRUVFVRUVDB79uyEZTo7O+M1lF27Biw7dzTyyq5ddPersZQVGrUlvdSWRDipLMrMil7qyqLkD3NltjcGW9vy2dSaT+OhEDs68tnVUUBHvxv2SoqLmDJlCjOnTuOcKVOYMmUKtbW1h2sM4XB4eIMapZQ45Lje8pa3cP755/PUU08xc+YsJQ1JWnFxccLLYBCvsezZs+dwu8r27dvZtnUrK7duYdnO+P0YhSFoKI8wq6KXBdU9zB0XIS/FWkkkBusOFLB2fwEbDxawrS2fSHBpaVxVBdMbGlhQ30BdXd3h9obq6uoxcSkp3ZQ4ZFCLFi3iqaeeYvbsWZkORcYIM2PSpElMmjSJM888c8C+5uZm1q1bF1/WruHxDRt5eHuUijCcOaGLC2q7mF5+7I11/W1ozefpnWFW7C3iUC8UFhQwZ84cPnzqqcyfP5958+Yxfvz4dP6JY54Shwyq79eXGsFlJPQllL7abVdXFy+++CK///3vefb553hyRxFn13Qfrjn0t6czj59vKGXF3kJKios4+4JzufDCC1m8eDGFhYUj/JeMbUocMihV2yWTioqKOP/88zn//PNpb2/n7rvv5t57/x9EeweU644aX185jh4L87d/+wk+8pGPUFRUlKGoxz4lDhEZFcrKyvjMZz7Dueeey1VXXUn/kYd2HApRWlHJzTd+n+nTp2cuyByh6w8yKI0/Jdlm3rx5TJlSO2BbT8z4h6u/qKQxQpQ4ZFC6VCXZqLq6esB6KBTinHPOyVA0uUeJQwbV0xPvb6+ah2SToztrlJaWZv0cFmOJEocM6uDB+B3AR88CJ5JN1BA+spQ4ZFBdXV1AfOY4kWyl7rYjS4lDBtXb2zvgUSQbjaaRZccCJQ4REUmJEoeIjHrR6ODDkMjwUuKQQfX1qup7FMlG+nyOLCUOGVR7e3y00oNtbRmOROT4uo+asVDSS4lDBnXgQMuAR5Fs1Nf7T0aGEocMat/+fQAcOHAgw5GIHHH0fUU93d26SXUEKXHIoHp74/dvRCPqjivZ4+j7imLuqnWMICUOGVTf0A55eRrOQbJHopEMlDhGjhKHDKqivByAktLSDEciItlCiUMGNWFCfIrN8eOrT1BSZOQkmpGyuLg4A5HkJiUOGVRFRQUA1ePGZTgSkSMSjYQbDoczEEluUuKQQZUHl6r6HkWywdE1jrw809wxI0iJQwbVN1y1fs2JSB8lDhmUJseRbHT02FSxmOs+jhGkxCGD0nDVko0SzQ/T2tqagUhykxKHDCpR7xWRTEs0qGFTU1MGIslN+laQQRUUFGQ6BJFjJEocO3bsyEAkuUmJQwalxCHZKNFouI2NjRmIJDcpccigdKlKstHRiSPfVOMYSfpWEJFRp6urc8B6YSjG1i2bMxRN7lHiEJFRpbW19fCozX3CIWfLlq0Je1vJ8FPiEJFRZe3atcdsKw45Pb29rF+/PgMR5R4lDhEZVV566SWOHlykNN+xYJ+kX1KJw8zGm9lNZrbSzFaY2Y1mNj7dwYmI9OfuPPfHZynNHzgfRygPZlZGee65P2YostySbI3jF0Az8CHgw8Ae4L/TFZSISCJbtmyhec9eygqOHV5k4fhuXn99Pfv27ctAZLkl2cRR7e7XufuWYPkGUJXOwEREjrZy5UoASguOnQHw1Or4TYGrVq0a0ZhyUbKJ4ykz+5iZ5QXLXwMPD0cAZvZuM3vdzDaa2VcS7L/MzPaY2apg+fRwnFdERp9NmzZRGTYKEnxz1ZVFCeXFy0h6DTqCnZm1AQ4Y8EXgzmBXCGgHrhnKyc0sBPwQeBfQCLxkZg+4+ytHFf1vd79qKOcSkdGvpaWFqnA04b78PKgoNFpaWkY4qtwzaOJw93TP3vNWYKO7bwYws18A7wOOThySIUcPXy2SSaFQiGjMjnutJOqaCmAknKjGcfpg+9195RDPPxV4o996I3BWgnIfMrPzgPXA1e7+RoIykga6oUqyyaRJk3ipyygtO3ZfZwQOdjuTJ08e+cByzIkmW7gheCwCFgMvE79sdRrwInDOEM+faK7Ho7tLPAjc4+7dZvZZ4GfAO445kNkVwBUAdXV1QwxL+vT29mY6BJHDZs2aRXcEehJUhLe2xb/OZs6cOcJR5Z5BG8fd/UJ3vxDYBpzu7ovd/QxgEbBxGM7fCJzUb30asPOoGPa5e9+IZv8JnHGcWG8N4ls8ceLEYQhNQDUOyS7z588HoDNy7FfXhtb4SM7z5s0b0ZhyUbK9qua6+5q+FXdfCywchvO/BMw2swYzKwQ+BjzQv4CZTem3egnw6jCcV5IUix3b7VEkU+rq6qgoL6MzeuzFig2t+dSdNI3KysoMRJZbkk0cr5rZT8zsAjM738z+k2H4Anf3CHAV8HhwvF+6+zoz+7qZXRIU+7yZrTOzl4HPA5cN9bwiMjqZGSfPmUtXgsSx/VAhc+aekoGock+yE0pfDnwO+EKw/gzw4+EIwN0fAR45atvX+j3/KvDV4TiXpE41Dsk2J510EitXLB+wLeZwoCu+T9IvqcTh7l3AvwfLMczsPnf/0HAGJtmhrzuuuuVKthg3bhyxo7rQRIP16urqkQ8oBw3X6Lgzhuk4kmXa2toAaG9vz3AkInFFRUXHbPMgcRQWFo5wNLlpuBLHsSOOyZhw4MABAPbv35/hSETi+qYzLg7FKA7FmFvVS01p/JKqbv4bGcm2cUiO2rV7d/DYlOFIROIKCuLdbqeWRsnPg/97+kG2t4d4bneY/Hx9pY2E4apxJLqRT8aAXbviiaPlwH66u7tPUFok/YqLiwEGtHN0RuJfQaWlpZkIKecknTjMrNjM5hxn95eHKR7JIpFIhLaDrcQK4/8Z+y5biWTShAkTAOiNHfm9ur8r/lU2frzmlxsJyc4A+FfAKuCxYH2hmR2+Uc/dn0hPeJJJfaOMxorjPVWUOCQb9A0p1N0vcTQeCpGXl0dtbW2mwsopydY4riU+km0LgLuvAurTE5Jki0OHDgEcrnH0rYtk0vjx46keV0lX5Eji2HKwgIb66YTD4QxGljuSTRwRd29NaySSdTo7OwHwguIB6yKZZGbMm7/g8LAjMYfN7YXMm39qhiPLHckmjrVm9nEgZGazzewm4Lk0xiXZxIarD4XI8DjllFPojRlRh6bOPDp6nVNO0XAjIyXZb4S/B+YD3cA9wEHgH9IVlGQHs+BSgOuucckuM2bE7znuiRo7DuUP2Cbpl+yQIx3APwWL5Ii+UUbzOuON5FVVVZkMR+Swvkbwnhjs6Yz//p06dWomQ8opJ5oB8EEGuSvc3S853j4Z/caNGwdAXueBAesimdY3504kZhzoziMcLqSsLMG0gJIWJ6pxfDd4/CBQA9wVrF8KbE1TTJIlwuEwBQUFeHd8nKry8nRPQS+SnOLiYszibRwHe/IYV1l55NKqpN2gicPdnwYws+vc/bx+ux40s2fSGplkhYLCML298cShAeQkW5gZ+fn5RGJRWnqMcZN1499ISrZxfKKZHW55MrMGQPOzjnG9vb10HGo/fB9H3w2BItmgsKCA3hgc6Clg4qRJmQ4npySbOK4GlpnZMjNbBjyFelWNec3NzQBEy+L/KZuaNNChZI/CcJjeWB57O40pU6ac+AUybJLtVfWYmc0G5gabXnN3jXg3xm3duhWAyLh6CvZvYevWrSxcOBxTzYsMXTgc5oADDtOmTct0ODnlRL2q3uHuvzezDx61a6aZ4e73pzE2ybCNGzcCEKmcihWE2bBhQ4YjEjmi//AiShwj60Q1jvOB3wN/lWCfA0ocY9iql1/GS8dDqJDekkmsenl1pkMSOax/Z42ampoMRpJ7TtSr6prg6dfdfUv/fUEDuYxRzc3NrFm9ht7xJwMQqahhxxsvsWHDBmbPnp3h6EQYMGmT5hofWck2jt+XYNu9wxmIZJfbbruNSCxGz+R5APROOBkrKOLmW27JcGQicX1TyAIaFXeEnaiNYy7xMaoqj2rnqACOnTFexoQVK1bw2OOP0zNpPh4ObvrLD9NZ8xZWLH+RJ554gosuuiizQUrOi8Vih593d3creYygE9U45gBLgSri7Rx9y+nA/0pvaJIJ+/bt4+vXXYcXVdI9ddGAfb2TTyFWXsMNN3yPbdu2ZShCkbje3t7Dz/u6jsvIGDRxuPtv3P1yYKm7X95v+by7a1j1Mcbd+c53rudgWzsdMy+EUMHAApZHx4wL6IkZ133jG0QikYzEKQIDJxZ79dVXMxhJ7km2jaPJzB40sz1m1mxmv+l/J7mMDU888QQvvvgCnbVnECtOPKChF5bQUbeEjRs28Itf/GKEIxQ5orW1lcI8pyoMf/iDRkAaSckmjp8DvwSmALXA/yM+L4eMERs2bOB73/t3YuWT6Z18ZEKc8PYXCG9/YUDZSHUDvePque22n7J8+fKRDlWE9evX09HRQWVhjLdP7uSPf3yO3bt3ZzqsnJFs4jB3v9PdI8FyF4MMty6jS3NzM1/68lfoJp+OGRcOmPEvr2M/eR37j3lNV/05RIsq+Zd/+RqbN28eyXBF+Oltt5FnUBWO8a5pXeQR44477sh0WDkj2cTxlJl9xczqzWy6mX0JeNjMqs1MHahHsU2bNvG5v7uSloNtHJr1TrywJLkX5hdyaNY76YoZf//5L/Dyyy+nN1CRwPLly3n+hRcYH44SMhhfFOMvpnby6KOPsH79+kyHlxOSTRwfBT5DfHDDZcDngP8JrAB0rWKUev7557nyqqvY39ZB+8kXEytJ7TeAh8toO/liDkVDfPGLX+TRRx/FXRVRSZ/u7m7+/Xs3MLnEqQ4f6Y77/vpOygvhP/79ewO66Up6JJU43L1hkEWN5KNMc3Mz1157LV/96lfpzCuhbe5SYqVvbj4DL6qgbe576SmdxLe//W3+9//5P2zfvn2YIxaJu+eee9ixcxf/Y3Yb/edtKi1wLp3RxiuvvsaDDz6YuQBzRFKJw8xKzOyfzezWYH22mS1Nb2gynNydHTt2cOedd/KJT36Sp595lu7aRbTPfS8ezLfxpuWH6Tj5L+mqO4uVq1Zz+eWXc8stt7B161bVQGTYbN++nbvvupMlk7pZML73mP1vr+lh3rgIt95yM/v27ctAhLkjqWHVgZ8Svyz19mC9kXjPqofSEZQMnbuza9cuVq1axapVq1ixYiX79u0FIFJVR9ecs47cFT4cLI/eyfOJVM8g/MZL3HPPPdxzzz1UVFZx+qKFLFq0iIULF1JXV6cpPiVlsViMG777XQosysdnH0pYxgw+NaeNf/5TPjfddBPXXnvtyAaZQ5JNHDPd/aNmdimAu3ea/vdnle7ubrZt28aGDRtYvXo1K1auZO+ePQBYYTE9pZOJTp9NtHwKseKqtMXhBcV0zTiP7qmLCB3cRW/bLp5+/iWWLVsGQGVVFacHSWT27Nk0NDRQXFyctnhkbHjooYd4efVqLp/TTlX4+LXYKSUx3lffwb3LlvGHP/yBc889dwSjzB3JJo4eMysm6IJrZjMBTeSUIfv27WPTpk2Hl/UbNtD4xhuHGwWPJIpZRMtriBVVwZvI8+HtLxDqiFf5i197hFhJNd11S5J6rYfLiUwsJzLxZLrcse428tt209u2i2XP/YmnnnoqHqsZU2prOXn2bGbOnMnMmTOZNWsWEydOVM1EAGhsbOTHP/oh86t7uaD2xF8776nr5KU9Rdxw/XeYN28e48drPvLhlmziuAZ4DDjJzO4GzgYuS1dQckRbWxsrVqzgtddeY8OGjWzYuIGDra2H91tRGb3hcUQnLyBWUk20pBoPV7ypRHG0vI79WDR+LTm/bTdveoARM7yogt6iCnr7EklPO6HgHpFth/az84WVh2slAKVl5cyaOZNZs2YyZ84cFi9erKGzc1BHRwfXfO1fCMW6+fTc9qQ+1vl58JlTDnLtijy+/q//yg3f+96AIdhl6JKdOva3ZrYSWAIY8AV335vWyHJULBZj48aNvPjii7zwwou88sq6eANzXggvriJSPInoSXMPJwnyR+GIoGbxGkm4HMZNB6ALINpDqOMAeR376Oncz5837WD12rV4NJ6yZs+ezZIlSzjrrLM45ZRTCIVCmfsbJO0ikQjfuO46Nm/Zwj+edpDxRcl3s51WFuXyOW3csno13/3ud/nyl7+sGuwwSiUNnw+cQ/xyVQHwq7RElGN6enrYtm0bmzdvZuXKlbzw4ou0trQAECudQG/NaUQqpxErnQh5yd52M0qFComWTyZaPhkIroV6jLyO/eS3NvL67h1suOsu7rzzTkpLy3jrW8/kjDPOYNasWdTX11NUpJH+x4poNMq//du/8dzzz/M/Tm7ntAS9qE7k7Joemjo6+PVjj1FaWspVV12l5DFMkkocZvYjYBZHxqf6jJm9092vTFtkY4y709TUxObNmw8vGzZuZEdj45G2iYIwPeW1RBoWEK2chheo0RjLI1Y6gZ7SCfTULoRIN/kHd9DbuoNlf3xxQFtJzZRaZs+ayYwZM5g5cyYNDQ3U1tYOmPBHsl9f0njyySf5yIxDvHPam29O/UBDJ51R47774nPRKXkMj2RrHOcDp3rQKd/MfgasGY4AzOzdwI1ACPiJu3/rqP1h4A7gDGAf8FF33zoc5x5u7k5bWxtNTU3s2bOH3bt3s23bNjZu3MTmzZvp7Ow4XNaKyuktqiI6+VRixdXESsYRK6ocME6UJJAfJlI9g0j1jKDR/WD88lbnfrZ3HmDXSy/zzDNHRkoNh4uob6hn1syZ1NfXU1NTw6RJk5g8eTKVlZX6Esky7s7111/P7373Oz4y4xB/Vd81pOOZwcdnxf/f3XfffYTDYa644orhCDWnJZs4XgfqgL7Ze04CVg/15GYWAn4IvIv4vSEvmdkD7v5Kv2J/Cxxw91lm9jHg28SHQBlx3d3d7Nmzh6amJpqbmw8vTU1N7NrdxN49e+juHvhBt/wwkeIqouV1xCZVEyseR7R4HOQXZuJPGFvM8KJKIkWVQD3Q11bSS15nC6HOA/R07ueVHQdYv+lJvHfgv01BQSETJk5gSr9kMmnSpAGLugqPrN/85jc89thjvK++Y8hJo09f8uiJGj//+c85+eSTueCCC4bl2Lkq2cQxHnjVzP4UrJ8JPG9mDwC4+yVv8vxvBTa6+2YAM/sF8D6gf+J4H3Bt8Pxe4AdmZp7GW5JjsRiPPvoomzdvpqmpiaamJnY3NdN2sPWYshYuIZpfSrSwFK+aSaywFC8sCx5L45eb9Kt2ZIUKiJVNJFY28cg2dyzShfUcIq/nENbTTk/PITo6D9G4vpH8da/j3cfeWFZaVsbkyZOpCZJKXV0dS5cupbBQiX+4RSIRbr/tvzhlXC8fbOgc1mObwSdPPsTGtkJu+6+fKHEMUbKJ42tpOv9U4I1+643AWccr4+4RM2slnsjS1qurs7OTH/zgB3R2xj+80ZJqoqU1+NRjEwN5Y7hnT7SHoqIili5dykMPPUR7tCfTEb15ZnhBMV5QTKx0QuIysRjW20FeT3uQYNrp6T7Ewd372bxpEwChUIglS5ZQW1s7gsHnhu7ubloOtvHOGb1p+a2VnwenVHazbHfT8B88xyTbHffpwfab2fPu/rY3cf5EH4+jaxLJlMHMrgCuAKirq3sToRxRWlrK7bffzl133cXDDz+CdbUSLZuE5xfFE4XHsGgP9BqeH4a8gjFZq7BID0svWcpVV12Fu/PLBx/PdEjp4Q6xCBbpxqLd4DGwPDy/CIt0k9/dgptx4QUXctlln1LSSJOSkhJmzmjgyZ1bWDK5m0nFg3e/rStL7c6iN9pDPNtUzGmnLRhKmEJ8gqahH8Tsz+6+6E287m3Ate7+l8H6VwHc/d/6lXk8KPO8meUDu4GJg12qWrx4sQ/XzHS7du3ijjvu4PHHHz/+cM1mWEERHgoTCxUSCxXi+WE8FI4/5ofxvm3BQiiM5xdmdWN48bpfUxbr4L3vfS8PP/ww7XkldM5/f6bDOj6PQbQ3ngCCJHDkec+R58G+UKwHi/TE2z488b+tmXHuuedx2WWfYsYMDQSdbuvXr+cfr74a7z3E5Se3sXhiz5B/k8Ucnt0V5q6N5ZSWV3LjTT9g2rRpwxPwGGNmK9x98QnLDVPiWOnup7+J1+UD64G/AHYALwEfd/d1/cpcCSxw988GjeMfdPe/Huy4w5k4+vT09NDW1sbBgwcTPvZ/3tLaevh5Z0fHoMe1/ELIDxMLhYmGCvslm0I8vyi+FBQNeE7eyNwFG97+Anltu+NfrvmFxMprkh5yZMhi0XibRKQL6z3q8XBC6CEvSABEuvHewbttFhUVU1ZeTmVFBZWVFVRUVFBeXk55efnh54kew+FReJPlKNbY2Mg1X/vHDnxJAAAQPUlEQVQXNm3ewmnje7l01iGmlkbf1LE2Hwzx841lrG/JZ8Gp8/naNdcyceLEE78wR42KxBG89j3AfxDvjnubu3/TzL4OLHf3B8ysCLgTWATsBz7W15h+POlIHG9WJBKhvb2d9vb2EyadgwcP0tIa33aove24NRwLFUBhMdG8MLEgmcQSJpni+OW1UBYMtxCLHPniT5QMIl3kRboIRePbPJK4PcXMKC0ro/xwAqgc8EV/vC//srIyDTsxikQiEX71q1/x09v+i86uLhaO7yEcSu276lCvsWZ/IZUV5Vzxmc9y8cUX656eExjpxPGmLlWlSzYljjfL3Wlvb6e1tZWWlhYOHDhw+Hn/Zf+BAxw40MLBg61EI4mv+VooHwqKiYbCAxJMrKiSaNmkNz0IYr9g4/dTtDeT19lCXpAQ8qJdhKLd0NOJRxPf+ZsXClFRUUlVVSXV48ZRVVXFuHHjqKyspKqq6pilrKxM//lzSEtLC3feeScvvvB8ynO7hPLyOO+CC7n00kspLR3inDM5YqQTx6nuvnbIBxomYyFxpMrd6ejoSJhgWltbOXDgQJBoWmhpiZeJ9Ma/zC0/TG/JBKJlk4JlIoQG6W4ajRA6tIfQoWZC7c0UHNqL98Z7oIVCISoqKxlXNY7q6ngi6EsCiRJCWVmZbsITyRLJJo5khxz5IPEb7yYR7+VkgLt7BfEnWZM0cpWZUVpaSmlpaVINf+5OY2Mj69atY926daxZs5Zt21Yd/lXnJdX0lk6kp2YBXlSB9XZSuGs1+e1N5HXsP9yYPHXaNBYsOZ9TTz2VefPmMX36dA0+KDLGJVXjMLONwF+5+6vpD2nocrHGMRza29t57bXXWLt2LevWrePll1fTkxemfc7FlGz8PQXdBzh1/qksWHAq8+fPZ968eVRWVmY6bBEZJsN6qcrM/ujuZw9LZCNAiWN4rFu3ji984R+IROKXtL75zW9y9tmj5mMgIilKNnEk28q43Mz+28wuNbMP9i1DjFGy3Pz58/nkJz8BwFlnnaWkISJA8kOOVAAdwEX9tjlw/7BHJFmlb9pNTb8pIn2SHXLk8nQHIiIio8OgicPMvuTu3zGzm0gwPpS7fz5tkUlWUZdZEelzohpHXy8qtTTnuDSOYi8io8ygicPdHwwefzZYOTO7yd3/fjgDExGR7DRcYzeou42ISI7QoD8iIpISJQ4REUnJcCUOdbkREckRw5U4bhym44iISJY70X0cD5Lg/o0+7n5J8Hj78IYlIiLZ6kT3cXw3ePwgUAPcFaxfCmxNU0wiIpLFTnQfx9MAZnadu5/Xb9eDZvZMWiMTEZGslGwbx0Qzm9G3YmYNgGZ8FxHJQcmOjns1sMzMNgfr9cAVaYlIRESy2gkTh5nlAQeB2cDcYPNr7t6dzsBERCQ7nTBxuHvMzG5w97cBL49ATCIiksWSbeN4wsw+ZBpbW0Qk5yXbxvFFoBSImFkX8TvF3d0r0haZiIhkpWRnACxPdyAiIjI6JFvjwMzGEW8gL+rb5u66l0NEJMcklTjM7NPAF4BpwCpgCfA88I70hSYiItko2cbxLwBnAtvc/UJgEbAnbVGJiEjWSjZxdLl7F4CZhd39NWBO+sISEZFslWwbR6OZVQG/Bn5rZgeAnekLS0REslWyvao+EDy91syeAiqBx9IWlYiIZK2kLlWZ2RIzK4fDI+Y+RbydQ0REckyybRw/Btr7rR8KtomISI5JNnGYux+eCdDdY6RwD4iIiIwdySaOzWb2eTMrCJYvAJtP+CoRERlzkk0cnwXeDuwAGoGz0HwcIiI5KdleVc3Ax9Ici4iIjALJ9qr6WXAfR9/6ODO7LX1hiYhItkr2UtVp7t7St+LuB1B3XBGRnJRs4sgLRscFwMyqGWKvKjOrNrPfmtmG4HHcccpFzWxVsDwwlHOKiMjQJfvlfwPwnJndG6x/BPjmEM/9FeBJd/+WmX0lWP9ygnKd7r5wiOcSEZFhklSNw93vAD4MNAHNwAfd/c4hnvt9wM+C5z8D3j/E44mIyAhI+nKTu68zsz0EEzmZWZ27bx/CuSe7+67g2LvMbNJxyhWZ2XIgAnzL3X89hHOKiMgQJTuR0yXEL1fVEq9xTAdeBeaf4HW/A2oS7PqnFGKsc/edZjYD+L2ZrXH3TQnOdQXBvSV1dXUpHF5ERFKRbI3jOuKz/v3O3ReZ2YXApSd6kbu/83j7zKzJzKYEtY0pxBNSomPsDB43m9ky4r25jkkc7n4rcCvA4sWL/ej9IiIyPJLtVdXr7vuI967Kc/engKE2WD8AfCp4/ingN0cXCO4XCQfPJwBnA68M8bwiIjIEydY4WsysDPgDcLeZNRNvcxiKbwG/NLO/BbYT76mFmS0GPuvunwZOAW4xsxjxJPctd1fiEBHJoGQTxyVAF/G5xz8BVAD/OpQTBzWYv0iwfTnw6eD5c8CCoZxHRESG16CJw8yedfdziHfD7Ws3sODxG2a2H7je3X+UxhhFRCSLDJo4gqSBu5cn2m9m44HnACUOEZEckWzjeELB5aYLhicUEREZDYaUOCB+895wBCIiIqPDkBOHiIjkFiUOERFJiRKHiIikRIlDRERSosQhIiIpUeIQEZGUKHGIiEhKlDhERCQlShwiIpISJQ4REUmJEoeIiKREiUNERFKixCEiIilR4hARkZQocYiISEqUOEREJCVKHCIikhIlDhERSYkSh4iIpESJQ0REUqLEISIiKVHiEBGRlChxiIhISpQ4REQkJUocIiKSEiUOERFJiRKHiIikRIlDRERSosQhIiIpUeIQEZGUKHGIiEhKlDhERCQlShwiIpISJQ4REUmJEoeIiKQkY4nDzD5iZuvMLGZmiwcp924ze93MNprZV0YyRhEROVYmaxxrgQ8CzxyvgJmFgB8CFwPzgEvNbN7IhCciIonkZ+rE7v4qgJkNVuytwEZ33xyU/QXwPuCVtAcoIiIJZXsbx1TgjX7rjcG2Y5jZFWa23MyW79mzZ0SCExHJRWmtcZjZ74CaBLv+yd1/k8whEmzzRAXd/VbgVoDFixcnLCMiIkOX1sTh7u8c4iEagZP6rU8Ddg7xmCIiMgTZfqnqJWC2mTWYWSHwMeCBDMeUU6qrqwFoaGjIcCQiki0y2R33A2bWCLwNeNjMHg+215rZIwDuHgGuAh4HXgV+6e7rMhVzLlqyZAm33HILH/jABzIdiohkCXMfe80Bixcv9uXLl2c6DBGRUcXMVrj7ce+r65Ptl6pERCTLKHGIiEhKlDhERCQlShwiIpISJQ4REUmJEoeIiKREiUNERFIyJu/jMLM9wLZMxzGGTAD2ZjoIkQT02Rxe09194okKjcnEIcPLzJYnc1OQyEjTZzMzdKlKRERSosQhIiIpUeKQZNya6QBEjkOfzQxQG4eIiKRENQ4REUlJWmcAlOxkZlFgTb9N73f3rccpWw885O6npj8yETCz8cCTwWoNEAX2BOtvdfeejAQmhylx5KZOd1+Y6SBEEnH3fcBCADO7Fmh39+/2L2NmRvxSe2zkIxRdqhIgXrMwsz+Y2cpgeXuCMvPN7E9mtsrMVpvZ7GD7J/ptv8XMQiP/F8hYZ2azzGytmd0MrAROMrOWfvs/ZmY/CZ5PNrP7zWx58Nlckqm4xyIljtxUHHzJrzKzXwXbmoF3ufvpwEeB7yd43WeBG4PaymKg0cxOCcqfHWyPAn+T/j9BctQ84L/cfRGwY5By3we+E9wc+NfAT0YiuFyhS1W5KdGlqgLgB2bW9+V/coLXPQ/8k5lNA+539w1m9hfAGcBL8asHFBNPQiLpsMndX0qi3DuBOcFnEmCcmRW7e2f6QssdShzS52qgCXgL8Zpo19EF3P3nZvYi8F7gcTP7NGDAz9z9qyMZrOSsQ/2ex4h//voU9XtuqCE9bXSpSvpUAruCxsZPAse0U5jZDGCzu38feAA4jXjvlw+b2aSgTLWZTR+5sCVXBZ/VA2Y228zygA/02/074Mq+laAmLcNEiUP6/Aj4lJm9QPwy1aEEZT4KrDWzVcBc4A53fwX4Z+AJM1sN/BaYMkIxi3wZeIz4D5jGftuvBM4OOnG8AvyvTAQ3VunOcRERSYlqHCIikhIlDhERSYkSh4iIpESJQ0REUqLEISIiKVHiEBGRlChxSNYys+fSeOxlZrY4XcdPhZlVmdnfZTqO/sxsq5lNyHQckp2UOCRrufsxI/SOUVVAViUOkcEocUjWMrP24PGCoIZwr5m9ZmZ3W7/R6xK87kwze87MXg6G1C43s2Iz+0VwJ/F/Ex+METMLmdntwXDda8zs6kGOu8zMvh0cc72ZnRtsLzKznwav/7OZXTjIMRINTf8tYGaw7XqLu75fTB/t9z48Y2a/MrNXzOzmYKiN453r3cEQ+S+b2ZPBtmoz+3Vw7hfM7LRg+3gzeyKI/xb6jQFlZl8MYllrZv8QbKs3s1fN7D/NbF3w2uLjxSJjjLtr0ZKVC/EJfAAuAFqBacR/7DwPnHOc1xQCm4Ezg/UK4oN5fhG4Ldh2GhAhPjT8GcBv+72+apB4lgE3BM/fA/wueP6PwE+D53OB7UDRcY5xE/A3/WItBuqBtf3KfIj40C0hYHJwvCnB+9AFzAj2/Rb48HHOMxF4A2gI1qv7nf+a4Pk7gFXB8+8DXwuevxdwYELw/qwBSoEyYB2wKIg5AiwMXvNL4BOZ/sxoGZlFNQ4ZLf7k7o0eH9huFfEvrkTmEB+s8SUAdz/o7hHgPOCuYNtqYHVQfjMww8xuMrN3AwdPEMf9weOKfjGcA9wZHPs1YBuJh6WHeNL7v2b2ZWC6Jx7m+xzgHnePunsT8DRwZrDvT+6+2d2jwD1B2USWAM+4+5Ygrv0JYv09MN7MKhn4/jwMHOhX/lfufsjd24O//9xg3xZ3X5Xg/ZAxTolDRovufs+jHH9KACP+azmRY7a7+wHiQ8kvIz4w3okm/OmLo38Mx71sluB8PwcuATqJD03/jgTFBjve0X/D8f7W470PiY7tRz0mG0uy/yYyxihxyFjzGlBrZmcCBO0b+cAzBDMTmtmpxC9XEfQcynP3+4B/AU5/E+fsf+yTgTrg9UQFjzM0fRtQftTxPhq0v0wkXhv4U7DvrWbWELRtfBR49jgxPQ+cb2YNwXmrE8R6AbDX3Q8etf1iYFy/8u83sxIzKyU+dPkfknlTZOzSLwQZU9y9J2hMvilorO0kPhvcj4GfBkO/r+LIF/HUYHvfj6g3MyHVj4CbzWwN8ev+l7l793HKfhT4hJn1AruBr7v7fjP7o5mtBR4FvgS8DXiZeC3gS+6+28zmEk8I3wIWEP9S/1Wik7j7HjO7Arg/+NuagXcB1/Z7HzqATwUv+VfgHjNbSfzS2PbgOCvN7HaOvF8/cfc/m1l9iu+RjCEaVl1klAhqCP/b3ZdmOhbJbbpUJSIiKVGNQ0YtM/sV0HDU5i+7++NDPO4PgbOP2nyju/80hWP8JfDtozZvcfcPJCo/FBafBz581OZPuvua4T6XCChxiIhIinSpSkREUqLEISIiKVHiEBGRlChxiIhISpQ4REQkJf8f3irSfpN/SnIAAAAASUVORK5CYII=\n",
        "text/plain": "<Figure size 432x288 with 1 Axes>"
       },
       "metadata": {
        "needs_background": "light"
       },
       "output_type": "display_data"
      }
     ],
     "persistentId": "5bb1b19d-97af-43fc-a51d-420946c4c00f",
     "text": "sns.violinplot(x='in_cds_no_stop_codon', y='jaccard_in_peptide_db', data=coding_scores_unspliced, \n               bw=0.1, cut=True)"
    },
    "executionTime": "2019-10-11T20:07:20.866Z"
   },
   {
    "cell": {
     "executionCount": 108,
     "executionEventId": "7888778c-e2a5-4430-978e-1ed02af5cb6f",
     "hasError": true,
     "id": "f6cb1260-f3d3-4c84-b1e7-229169dcdd93",
     "outputs": [
      {
       "ename": "NameError",
       "evalue": "name 'coding_scores_unspliced_not_low_complexity' is not defined",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
        "\u001b[0;32m<ipython-input-108-ff1f36122650>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m sns.violinplot(x='in_cds_no_stop_codon', y='jaccard_in_peptide_db', data=coding_scores_unspliced_not_low_complexity, \n\u001b[0m\u001b[1;32m      2\u001b[0m                bw=0.1, cut=True)\n",
        "\u001b[0;31mNameError\u001b[0m: name 'coding_scores_unspliced_not_low_complexity' is not defined"
       ]
      }
     ],
     "persistentId": "5bb1b19d-97af-43fc-a51d-420946c4c00f",
     "text": "sns.violinplot(x='in_cds_no_stop_codon', y='jaccard_in_peptide_db', data=coding_scores_unspliced_not_low_complexity, \n               bw=0.1, cut=True)"
    },
    "executionTime": "2019-10-11T20:07:53.023Z"
   },
   {
    "cell": {
     "executionCount": 110,
     "executionEventId": "0824197f-623b-460a-b47c-91984fac8039",
     "hasError": false,
     "id": "3f61b436-51d7-4dae-a329-c238ba62d3d5",
     "outputs": [
      {
       "data": {
        "text/plain": "[Text(0, 0.5, 'True Positive Rate'), Text(0.5, 0, 'False Positive Rate')]"
       },
       "execution_count": 110,
       "metadata": {},
       "output_type": "execute_result"
      },
      {
       "data": {
        "image/png": "iVBORw0KGgoAAAANSUhEUgAAAYUAAAEKCAYAAAD9xUlFAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDMuMC4zLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvnQurowAAIABJREFUeJzt3Xl4FFX28PHvyUJC2PcdAggooiBEVFRcWEQRcJRRFEUUwQ11HHHQ1wX15zi474gIjLviMKhBQWQURBEQBBEBkRAQguyELfty3j+qE5oYkg6kUunu83mePF1VXV11KpA6Xadu3SuqijHGGAMQ4XUAxhhjKg9LCsYYYwpZUjDGGFPIkoIxxphClhSMMcYUsqRgjDGmkCUFY4wxhSwpGGOMKWRJwRhjTKEorwMoq/r162t8fLzXYRhjTFD58ccfd6tqg9LWC7qkEB8fz7Jly7wOwxhjgoqI/B7IelY+MsYYU8iSgjHGmEKWFIwxxhSypGCMMaaQJQVjjDGFXEsKIjJVRHaKyC9HeV9E5CURSRKRn0Wkq1uxGGOMCYybVwpvAv1KeP9ioJ3vZxTwmouxGGOMCYBrzymo6gIRiS9hlUHA2+qMB7pYRGqLSBNV3eZWTMYYUyHysiFrP2Tt873uh+z9h6eLm8/LOurmfkiqSWx0Pqee2x16POJq6F4+vNYM2OI3n+Jb9qekICKjcK4maNmyZYUEZ4wJU4Un9KIn7gBP8Nn7ITezXEJRhbGf9+HZb7pxapMd/NDhV6LLZctH52VSkGKWaXErquokYBJAQkJCsesYYwwAeTmQsQvSd0Lm3qOfyAtO8m6c0COioEotiPH7KW0+MhbkyNOiAGxIggVb6DvoDPJOPz2kk0IK0MJvvjnwh0exGGMqK1XIPuCc5NN3QvoOv+lilmXuPb79HcsJvUotiKl9eD6q6p9O8IHaty+T5ORUunZtAsCjz53FkJG7C+fd5mVSSARGi8iHwBnAfrufYEyYKPg2n7YDMko4wRf8lFBv/xOJgKoNIK4hVK1XhhN87eM+oR+vTz/9lVtv/ZyICGH16tuoVSuWqlWjKywhgItJQUQ+AM4H6otICjAOnCsfVZ0IzAIuAZKAdOAGt2IxxrhM1Sm/FJzEC070aTv+PJ+xEzJTy7b96OrOST6uke+14dHnq9Z1EkMQ2bkzjTvvnM20aasBOPPM5uzbl0mtWrEVHoubrY+uLuV9BW53a//GmOOUlw3puw6f1NN2HHnSLzqflx34tiUS4gq+zftO6NUaHZ4umI9r6Hzrj45z7zg9pKq8994q7rrrC/buzSAuLponnriQ0aO7ExnpTWILuq6zjTHHKD/PubmasbvkmnzBfNa+sm2/So3DJ/lqft/gqxY9yTcMym/zbrj11s95/fUfAejduw2TJl1K69Z1PI3JkoIxwSgvBzL3QMYe5yTvP52xxzdfZDozlaM08Cte4bf5IiWaoif9gmXRVV073FB12WUnMm3aap59ti833NAF8ehehj9LCsZ4LSej+JN4cSf8gunsA8e2r9g6EFuv9Lp8XENnXfs2X67Wr9/DV19t5JZbEgDo1+8ENm26y5N7B0djScGY8pZ9qEhJZoffDdddfz7552aUfR8S4Zzcq9bzvdZ3pqvWP7z8T9N1nOaWpsLl5ubz3HOLGDduPllZuXTp0pgzz2wOUKkSAlhSMKZkO5bD3l+PXKb5TinG/6RfMJ22A3LTy7aPiOgAT+p+J/+YWvYtPkisXLmdESMS+fFHp8X9sGGdadeursdRHZ0lBWOKys2EtG2w5F+w6o2yfz4yxinBFNbd/V8b/PkEH13ds3bxxj1ZWbk8/vgCxo9fSG5uPi1b1uL11y+lX78TvA6tRJYUTHhRdVrVHPgdDmz2vf4OB/3m03ccXj8iGtoOhMgqR24nps6RLWr8T/xVathJ3nD//V/x/POLAbj99tP51796UaNGjMdRlc6SggltORmQ8g1snO287k+G7IMlfyYiynm6tX4nuPBl59WYMvrHP85m0aIUnnqqN+ee28rrcAJmScGEFlXYl+QkgY2zIWX+nzs4i64GNVsd/qnR8sj5ak0gItKT8E3wmjt3AxMn/si0aYOJioqgcePqfP/9jZWimWlZWFIwwS8nDbbMP5wI9icf+X7DrtD6YojvC/VOhti6Vt4x5SY1NYMxY75k6tSfAPj3v1cwcmQ3gKBLCGBJwVQmmu/8BCJ1PWz6wnc1sODIDtNi60Krvr5EcJFT9zfGBR9/vJbbbpvF9u2HiImJZNy48xg+vIvXYR0XSwqm4uVmOs0896yBPat9r2ucsk+gSeEIAo27Q3w/JxE0Pt3KP8ZV27cf4o47ZjN9+hoAevRowZQpAznxxPoeR3b8LCmEo01zYemTkJ9bwTtWOPSHU9452slfAjyZV63nuxro57zGNSi/MI0pxaef/sr06WuoVi2a8eN7c9ttpxMREXylouJYUghHK16GzV95t3+JhDodoF5HqH8y1O3oTNftAFGV6+lOYwpkZuYSG+ucMkeO7EZyciq33no68fG1PY6sfFlSCAd71x3Z42XGbuc1YQy06V+xscTWgzrtIaryt9c2BiA/X5kwYSn//Oe3LF48glatahMRITz5ZB+vQ3OFJYVQ992DsOSfxb/X8kJocX6FhmNMMFm3bjcjRiSycOEWAD744Bfuu+8cj6NylyWFULbhMychSCQ06opvGHBH9abQLLT/cxtzrHJy8njmme959NFvyMrKo1GjakyY0J/LLz/J69BcZ0khVB34Hb4Y5kyf80/oPtbbeIwJEr/8spNhwz5mxYrtANxwQxeefbYvdeqEx3gRlhRCUV42fHaV05Nnm/5w+r1eR2RM0MjPV1at2kmrVrWYNGkAffu29TqkCmVJIRQtGAvblkCNFtDvLeti2ZhSrF69k44dGyAinHpqIz79dAg9e7aievUqpX84xNjZItTsWA7LX3A6dbv0I6c9vzGmWAcPZjF69Cw6dXqN//53beHySy5pF5YJAexKIfSk/ua8tr4Emp7pbSzGVGJz5iQxatRnbN68n6ioCDZt2lf6h8KAJYVQkpsJix93puue6G0sxlRSe/dmcPfdc3j77ZUAdO3ahClTBtKlS2OPI6scLCmEkj++d/oSqhkPZz7kdTTGVDo//bSdfv3eZceONGJiInn00fO5554eREVZJb2AJYVQUvCkcr2ToEp1b2MxphJq374e1atXoX37ekyePJD27e2eW1GWFELJuo+c12Y9vY3DmEpCVXn//VUMGNCBmjVjiIuLZv784TRtWiNkOrArb3bNFCoy9sCGRKf56cnDvI7GGM9t2rSPiy56l2uv/Zj77vtf4fLmzWtaQiiBXSmEij1rID/HGUugelOvozHGM3l5+UyYsJT77/+KtLQc6tatSo8eLbwOK2hYUgg1kdb7qAlfa9fuYsSIRBYtSgHgyitP5uWXL6Zhw2oeRxY8LCkYY0LCxo2pdOnyOtnZeTRpUp0JE/pz2WXWNLusXE0KItIPeBGIBCar6vgi77cE3gJq+9a5T1VnuRmTMSY0tW5dh7/+tSOxsVE880xfate2AZuOhWtJQUQigVeBPkAKsFREElV1jd9qDwIfqeprItIRmAXEuxWTMSZ0ZGTk8Nhj3/CXv5xE9+7NAHjrrcuIjLT2M8fDzd9edyBJVZNVNRv4EBhUZB0FavqmawF/uBiPMSZEfPvt73Tp8jrjxy9k1KiZ5OcrgCWEcuBm+agZsMVvPgU4o8g6jwBfisgdQDWgt4vxhLYt853XqPDo892EpwMHsrj//v8xYcIyADp2bMDEiZdaE9Ny5GZaLe5fSYvMXw28qarNgUuAd0T+3M+ziIwSkWUismzXrl0uhBrkDm2DRY84093u9jQUY9wya9Z6OnWawIQJy4iKiuDhh3uyfPkozjyzudehhRQ3rxRSAP/Gwc35c3loBNAPQFUXiUgsUB/Y6b+Sqk4CJgEkJCQUTSwmbRtoPjToDK0v9joaY8rd/v2ZDB06g337MklIaMqUKQM59dRGXocVktxMCkuBdiLSGtgKDAGuKbLOZqAX8KaInATEAnYpUBaq8P3DzrQNpmNCiKqiChERQq1asbz0Uj927Ejjb3870zqwc5FrSUFVc0VkNDAHp7npVFVdLSKPActUNRG4B3hDRO7GKS0NV1W7EiiL/Rsh+XNnupp1/WtCwx9/HOS22z7n3HNbcs89PQC47rrOHkcVHlx9TsH3zMGsIsse9pteA5ztZgwhLyv18PTF73gXhzHlQFWZOnUF99zzJfv3Z7F4cQq33XY6VatGex1a2LAnmoNZxl54N8GZrtPBht40QS05OZWRI2fy9dcbAejfvx0TJ15qCaGCWVIIZr++f3j61FHexWHMccjLy+ell5bwwANfk5GRS/36cbz0Uj+GDOmEiDU1rWiWFILRuv/A71/Cpi+d+f4fwIlDvI3JmOMwffpaMjJyufrqTrz4Yj8aNLAO7LxiSSFYpG2HlAWwLxm+u//w8th6cMJl3sVlzDHIzs7j4MEs6tWLIzIygilTBrJ+/R4GDOjgdWhhz5JCsJj5V9j63eH50+6A+qdA0x4QZR1/meCxdOlWRoxIpHnzmnz++TWICCeeWJ8TT6zvdWgGSwrBI32H89r6YmjVF7reBVZvNUEkPT2HcePm8dxzi8nPV9LTc9i5M41GjWw88crEkkKwOf8FqNve6yiMKZP58zcxcuRMkpL2EhEhjBlzFo8+egFxcdayqLKxpGCMcY2qcueds3nllaUAnHJKQ6ZMGcjppzfzODJzNJYUgkF+HmTscaajbLhNEzxEhJo1Y4iOjuDBB3ty333nUKVKpNdhmRJYUggGKQsgcy/Ubgs1WnodjTEl2r07nQ0b9nLGGU7vpQ89dB5Dh55Kx44NPI7MBCKgXqVEpIqInOB2MOYofpvuvLb/q91cNpWWqvLhh79w0kmvctll00hNzQAgNjbKEkIQKTUpiEh/YBUw1zffRUQ+djsw45OfB+v/60y3/6u3sRhzFCkpBxg06EOuvvq/7N6dTseODUhPz/E6LHMMAikfPYYzYto8AFX9ya4aKtDW75zmqLXaQMPTvI7GmCPk5yuTJy/n3nvncuBAFjVrxvDss30ZMeI066IiSAWSFHJUdV+Rf2Dr3rqi/PYf59VKR6YSGjEikTff/AmAgQM7MGHCJTRrVrOUT5nKLJB7CmtF5EogQkRai8gLwGKX4zJQpHQ02NtYjCnGtdeeQsOG1fjwwyv45JOrLCGEgECSwmigG5APzAAygbvcDMr4/PG90+dRzXho1M3raIzhl1928uKLh78T9urVhuTkO7nqKuvRNFQEUj66SFXHAmMLFojI5TgJwrjJSkemksjKyuVf//qOJ574lpycfBISmnL22U7z6GrVqngcnSlPgVwpPFjMsgfKOxBThOYfLh11sFZHxjtLlqTQrdskHn30G3Jy8rn11gROOaWR12EZlxz1SkFELgL6Ac1E5Dm/t2rilJKMm7Z+D4f+gJqtoFGC19GYMJSWls1DD83jhRcWowrt2tVl8uSB9OzZyuvQjItKKh/tBH7BuYew2m/5QeA+N4MyHC4dtRtspSPjiQce+JoXX1xCRIRw771n8cgj59vQmGHgqElBVVcAK0TkPVXNrMCYjJWOTCXwwAPnsmrVTp58sjcJCU29DsdUkEDuKTQTkQ9F5GcR+a3gx/XIwtkfi+HQVqefo8bdvY7GhInExHVccsl75OTkAdCgQTW++mqYJYQwE0hSeBP4NyDAxcBHwIcuxmQKWx1Z6ci4b+fONIYMmc6gQR8ye3YSb7210uuQjIcCSQpxqjoHQFU3qOqDwAXuhhXGNP/IDvCMcYmq8u67P3PSSa8ybdpq4uKiefHFftxwQxevQzMeCuQ5hSxxnkrZICK3AFuBhu6GFca2LYFDKVC9OTSx0pFxx+bN+7nlls+YPTsJgN692zBp0qW0bl3H48iM1wJJCncD1YE7gX8CtYAb3QwqrBVeJQwGCahnc2PK7MsvNzB7dhK1a8fy3HN9GT68iz2RbIAAkoKqLvFNHgSuAxCR5m4GFbZUrXRkXJOWll349PGIEaexdesBRo3qRpMmNTyOzFQmJX4VFZHTReQyEanvmz9ZRN7GOsRzx/Yf4OBmqN4Mmp7pdTQmROTm5vPUUwtp1eoFkpNTAWeYzHHjzreEYP7kqElBRP4FvAcMBb4QkQdwxlRYCbSvmPDCzDr/VkdWOjLHb+XK7ZxxxmTGjv0fe/Zk8Mknv3odkqnkSiofDQI6q2qGiNQF/vDNr6uY0MKMKqy30pEpH1lZuTz++ALGj19Ibm4+LVvWYtKkS7noIhsfy5SspKSQqaoZAKq6V0R+tYTgou1L4cDvUL0pND3L62hMEFuxYhtDh85g7drdiMDo0afzxBO9qFEjxuvQTBAoKSm0EZGC7rEFiPebR1UvL23jItIPeBGIBCar6vhi1rkSeARnNLeVqnpN4OGHkIIbzO2usNKROS4xMVFs2JBKhw71mDx5IOec09LrkEwQKSkpXFFk/pWybFhEIoFXgT5ACrBURBJVdY3fOu2A+4GzVTVVRMLz+QfVI8dOMKaMli/fxmmnNUZE6NixAbNnD6VHjxbExgbS6tyYw0rqEO+r49x2dyBJVZMBRORDnPsUa/zWGQm8qqqpvn3uPM59BqcdP8KBTVCtCTQ72+toTBBJTc1gzJgvmTr1Jz744AqGDOkEwIUXtvY4MhOs3Pwa0QzY4jefApxRZJ32ACKyEKfE9IiqflF0QyIyChgF0LJlCF4KF3aTbaUjE7iPP17LbbfNYvv2Q8TERLJnT7rXIZkQ4GZSKO7xSC1m/+2A84HmwLci0klV9x3xIdVJwCSAhISEotsIbkeUjgZ7G4sJCtu3H+KOO2Yzfbpz0X322S2YPHkgJ55Y3+PITCgIOCmISIyqZpVh2ylAC7/55jjNWouus1hVc4CNIrIOJ0ksLcN+gtvOFbB/I8Q1gmbneB2NqeR+/PEP+vR5h9TUTKpVi2b8+N7cdtvpRERYFxWmfJRaqxCR7iKyCljvm+8sIi8HsO2lQDsRaS0iVYAhQGKRdT7B1+Oq76np9kByGeIPfv6lo4hIb2MxlV7Hjg1o0KAaF13UltWrb2P06O6WEEy5CqSA/RJwKbAHQFVXEkDX2aqaC4wG5gBrgY9UdbWIPCYiA32rzQH2iMganKel71XVPWU/jCDlXzqyEdZMMfLzlUmTfmTfPmfww6pVo1mwYDizZw+lVavaHkdnQlEg5aMIVf29SA+KeYFsXFVnAbOKLHvYb1qBv/t+ws/On2DfBohrCM3O9ToaU8msW7ebm26ayXffbWbp0q288YbzXapRo+oeR2ZCWSBJYYuIdAfU9+zBHYANx1keCktHl1vpyBTKycnj2WcX8cgj88nKyqNx4+pcfHE7r8MyYSKQpHArTgmpJbAD+J9vmTke9sCaKcaKFdsYMSKRFSu2A3DDDV149tm+1KlT1ePITLgIJCnkquoQ1yMJN7t+hn1JULUBNO/pdTSmEtiwYS/du08mNzef+PjaTJp0KX36tPU6LBNmAkkKS31NRacBM1T1oMsxhYcjSkfWFYGBtm3rct11p1KjRhX++c9eVK9exeuQTBgqtfWRqrYFHge6AatE5BMRsSuH42GlIwMcOpTNnXfOZtGiww/+T5kykBdfvNgSgvFMQH0qqOr3qnon0BU4gDP4jjlWu1dB6m9QtT60OM/raIwH5sxJ4uSTJ/Dyyz9wyy2f4zTEw8ZJNp4rtW4hItVxOrIbApwEfAr0cDmu0FbQTfYJf7HSUZjZuzeDu++ew9tvrwSgW7cmTJky0JKBqTQCOSP9AswEnlLVb12OJ/RZ6ShsTZ++httvn8XOnWnExkbx6KPn8/e/n0VUlHWCaCqPQJJCG1XNdz2ScLFnNez9FWLrQctSHww3IWLfvkxGjZpJamomPXu24o03BtC+fT2vwzLmT46aFETkWVW9B/iviPypZ9JARl4zxVhX0OrISkehTlXJz1ciIyOoXTuWCRP6k5qawc03J1h/RabSKumsNM33WqYR10wprJvssLBp0z5GjZrJhRe25r77nN5vCwbAMaYyO2oxU1V/8E2epKpf+f/g3HA2ZbVnDexdC7F1ocWFXkdjXJCXl89LLy2hU6cJzJ2bzCuv/EBmZq7XYRkTsEDucN1YzLIR5R1IWCgoHZ1wGURGexuLKXdr1+6iZ883ueuuL0hLy2HIkE4sX36zjZNsgkpJ9xSuwmmG2lpEZvi9VQPYV/ynTIms1VFIys3N58knv+OxxxaQnZ1H06Y1eO21/gwc2MHr0Iwps5K+wvyAM4ZCc+BVv+UHgRVuBhWS9qx1Wh7F1oGWvbyOxpSjiAjhyy+Tyc7OY+TIrjz1VB9q1471OixjjslRk4KqbgQ24vSKao5XwVVC20FWOgoBGRk5HDyYTcOG1YiIECZPHsCWLQe48MLWXodmzHE56j0FEfnG95oqInv9flJFZG/FhRgiCp5ittJR0Fuw4Hc6d57ItdfOKOyeol27epYQTEgoqXxU8GRV/YoIJKTtXef0dxRTG1r19joac4wOHMji/vv/x4QJywCIjo5k9+50GjSo5nFkxpSfkpqkFjzF3AKIVNU84CzgZsD+CsqioHR0wiCItN4vg9Hs2evp1GkCEyYsIyoqgnHjzmP58lGWEEzICaSt3CfA6SLSFngb+Bx4H7jUzcBCirU6ClqqysiRM5kyxWlbkZDQlKlTB3LKKY08jswYdwTynEK+quYAlwMvqOodQDN3wwohe39zRlmrUhNaWuko2IgIzZvXJDY2imee6cOiRSMsIZiQFtBwnCLyV+A64DLfMms+E6j1Bd1kD4KoGG9jMQH544+DbNiwl3PPbQXA//t/53LddafStm1djyMzxn2BPtF8AU7X2cki0hr4wN2wQsg6Kx0FC1VlypTldOz4Kldc8RF79qQDUKVKpCUEEzZKvVJQ1V9E5E7gBBE5EUhS1X+6H1oISE2CXT85paNWfb2OxpQgOTmVkSNn8vXXGwG49NL25ORYj/Em/AQy8tq5wDvAVkCAxiJynaoudDu4oFf4wNpAKx1VUgUd2D344DzS03OoXz+Ol17qx5AhnWw0NBOWArmn8DxwiaquARCRk3CSRIKbgYUEa3VU6Q0b9gnvv78KgGuuOYUXXrjImpmasBbIPYUqBQkBQFXXAtbYvjT7NsDOFVClBsRb6aiyGjmyK82b1yQxcQjvvXe5JQQT9gK5UlguIq/jXB0ADMU6xCtdQbcWbQZAlHWOVlksXbqVr7/eyNixzsA3558fT1LSHcTEWPfWxkBgSeEW4E7gHzj3FBYAL7sZVEiw0lGlkp6ew7hx83juucXk5ys9erQobHJqCcGYw0r8axCRU4C2wMeq+lTFhBQC9iXDjh8hujrEX+R1NGFv/vxN3HRTIhs2pBIRIYwZcxbdujX1OixjKqWSekn9fzhdXAwF5opIcSOwlUhE+onIOhFJEpH7SlhvsIioiITGzeuC0lHbARBd1dtYwtj+/ZncfPNMLrjgLTZsSOWUUxqyePEInn66L3Fx9vylMcUp6UphKHCqqqaJSANgFjA10A2LSCTO4Dx9gBRgqYgk+t+09q1XA6c8taSswVdahaWjwd7GEeYeemgekyYtJzo6goce6snYsedQpUqk12EZU6mVlBSyVDUNQFV3iUggLZX8dcd50C0ZQEQ+BAYBa4qs93/AU8CYMm6/ctq/CXYsg+hqEH+x19GEHVUtfL7g4YfPY+PGfYwf34uTT27ocWTGBIeSTvRtRGSG7+djoK3f/IwSPlegGbDFbz6FIh3pichpQAtV/azMkVdWha2OLrXSUQVSVd5/fxUXXvg22dl5ANSvH8fMmVdbQjCmDEq6UriiyPwrZdx2cY+DauGbzpXH88DwUjckMgoYBdCyZcsyhlHBrNVRhUtJOcCtt37OZ5/9BsB77/3MDTec5nFUxgSnksZo/uo4t52CM0BPgebAH37zNYBOwHzf5X5jIFFEBqrqsiKxTAImASQkJCiV1YHfYfsPEBUHra105Lb8fOWNN37k3nvncvBgNrVqxfDss30ZPryL16EZE7TcbKC9FGjn61V1KzAEuKbgTVXdj99QnyIyHxhTNCEElcLSUX+IjvM2lhCXlLSXkSNnMn/+JgAGDerAhAn9adq0hreBGRPkXEsKqporIqOBOUAkMFVVV4vIY8AyVU10a9+eKUgKVjpy3bff/s78+Zto2LAar7xyMYMHd7QO7IwpBwEnBRGJUdWssmxcVWfhNGX1X/bwUdY9vyzbrnQObIFtiyGqKrS5xOtoQtK+fZnUru10GTJ8eBd27UpnxIjTqFfPrsqMKS+lNjMVke4isgpY75vvLCLWzUVR6/1LR9apWnnKyspl3Lh5tGr1AuvX7wGcYTL/8Y+zLSEYU84CuVJ4CbgU5+lmVHWliFzgalTByEZYc8XixSmMGJHImjW7AJgzZwPt2tXzOCpjQlcgSSFCVX8vUq/Ncyme4HRgC2xb5PSG2tpKR+UhLS2bhx6axwsvLEYV2rWry5QpAws7sTPGuCOQpLBFRLoD6uu64g7gN3fDCjJJvmf5Wl8CVap7G0sIWLIkhWuumUFyciqRkcKYMT0YN+48qla1/oqMcVsgSeFWnBJSS2AH8D/fMlPASkflqnbtWLZuPUDnzo2YMmWg9WhqTAUqNSmo6k6cZwxMcQ5uhT8WOqWjNpd6HU3Q+u67zZx9dgtEhA4d6vP119dz+ulNiY62DuyMqUilJgUReQO/7ikKqOooVyIKNuv/67zGX2ylo2Owc2cad945m2nTVvPWW5cxbFhnAHr0aFHKJ40xbgikfPQ/v+lY4C8c2dFdeLNuso+JqvLee6u4664v2Ls3g7i46MKO7Iwx3gmkfDTNf15E3gHmuhZRMDn0B2xdCJExzoA6JiCbN+/nlls+Y/bsJAD69GnDpEkDiI+v7XFkxphj6eaiNWDtAgHWzwAU4vtBFetzJxBLlqTQu/c7HDqUTe3asTz//EVcf31n66LCmEoikHsKqRy+pxAB7AWOOrRmWCkoHXWwVkeB6tKlMS1a1OTN3a/iAAAULUlEQVTEE+vz6quX0KSJJVNjKpMSk4I4X9864/RyCpCvqpW36+qKdGgbpHwLkVWgjZWOjiY3N59XXvmBYcM6U7duVWJioli48Ebq1LEBiIypjErs+8iXAD5W1TzfjyWEAgWlo1YXQUxNr6OplFau3M4ZZ0zm7rvn8Pe/zylcbgnBmMorkHGXfxCRrq5HEmwKOsCz0tGfZGbm8uCDX5OQ8AbLl2+jZctaXH11J6/DMsYE4KjlIxGJUtVc4BxgpIhsANJwhtlUVQ3fRJG2A1IWOKWjtgO9jqZS+f77LYwYkcivv+5GBEaPPp0nnuhFjRoxXodmjAlASfcUfgC6ApdVUCzBY/0M0Hxo1RdiankdTaWRlLSXc8/9N/n5SocO9ZgyZSBnn13Jx9Q2xhyhpKQgAKq6oYJiCR6/WV9HxTnhhLqMGtWVunWr8tBD5xEb6+Zor8YYN5T0V9tARP5+tDdV9TkX4qn80ndCyjcQER32paPU1AzuuedLbrihS2GX1hMm9LdnDowJYiUlhUigOr4rBuOz/mOndBTfD2LD9wncGTPWcvvts9i+/RA//riNn366GRGxhGBMkCspKWxT1ccqLJJgEealo+3bDzF69Cz++9+1AJxzTksmTx5gycCYEFHqPQXjJ30XbJnnlI5OGOR1NBVKVXn77ZXcffccUlMzqV69Ck8+2ZtbbkkgIsL+qxgTKkpKCr0qLIpgkVRQOroIYut4HU2F2rcvk3vu+ZLU1Ez69TuBiRP706pV+JbPjAlVR00Kqrq3IgMJCmE2wlp+vpKfr0RFRVCnTlVef/1S0tNzuPbaU61cZEyICuSJZgOQvttXOoqCtqFfOvr119307Plvxo//rnDZFVd05LrrrEdTY0KZJYVAJX0Cmgcte0HVul5H45qcnDyeeOJbOneeyMKFW5gyZQWZmbleh2WMqSD2dFGgwqDV0YoV27jxxkR++mk7ACNGnMbTT/exh9CMCSP21x6IjD2w+SuQSDgh9Hr9yMnJY9y4+Tz11ELy8pT4+Nq88cYAevdu43VoxpgKZkkhEAWlo1Z9oWo9r6Mpd1FRESxZspX8fOWuu87g8ccvpHr1Kl6HZYzxgCWFQBSWjgZ7G0c5Ongwi4MHs2natAYiwuTJA9i+/RBnndXC69CMMR6yG82lydjrVzr6i9fRlIs5c5Lo1Ok1hg6dQcG4Sa1b17GEYIyxpFCqDZ9Cfi60uADi6nsdzXHZsyed66//hH793mPz5v0cPJjFnj0ZXodljKlEXE0KItJPRNaJSJKI3FfM+38XkTUi8rOIfCUirdyM55gUlI6CeIQ1VWX69DV07DiBt99eSWxsFE891ZvFi2+ifv04r8MzxlQirt1TEJFI4FWgD5ACLBWRRFVd47faCiBBVdNF5FbgKeAqt2Iqs8xU+P1/IBFBWzpSVYYOncEHH/wCQM+erXjjjQG0bx96N8yNMcfPzSuF7kCSqiarajbwIXDEo8CqOk9V032zi4HmLsZTdkmfQn4OtDgf4hp4Hc0xERE6dmxAjRpVeO21/sybd70lBGPMUbnZ+qgZsMVvPgU4o4T1RwCzi3tDREYBowBatqzA4R3XT3deg+yBtY0bU0lOTqVXL+c5g7Fjz2b48C40b17T48iMMZWdm1cKxXWQo8WuKHItkAA8Xdz7qjpJVRNUNaFBgwr6xp65DzZ96ZSO2l1eMfs8Tnl5+bz44mI6dXqNq66azs6daQBER0daQjDGBMTNK4UUwL+NY3Pgj6IriUhv4AHgPFXNcjGestmQ6CsdXQBxDb2OplRr1uzippsSWbQoBYCBAzvYOAfGmDJzMyksBdqJSGtgKzAEuMZ/BRE5DXgd6KeqO12MpeyCpK+jnJw8nnxyIf/3fwvIzs6jadMavPZafwYO7OB1aMaYIORaUlDVXBEZDczBGe95qqquFpHHgGWqmohTLqoO/MfXHfNmVR3oVkwBy9oPv38JCLSr3K2OrrlmBtOnOw26Ro7sytNP96FWrViPozLGBCtXu7lQ1VnArCLLHvab7u3m/o/ZhpmQlw3Nz4Nqjb2OpkR33XUGP/20nddfv5QLL2ztdTjGmCBnTzQXpxKXjr75ZhOPPjq/cP6cc1qydu3tlhCMMeXCOsQrKusAbJoDCLS/wutoCh04kMXYsXOZOPFHAC64oDU9ezoPgEdFWW43xpQPSwpFJc+EvCxo3rPSlI5mzVrPzTd/RkrKAaKjI3jggXM588zK9ZyfMSY0WFIoap2vdNTO+26yd+9O529/+4L33lsFQPfuzZgyZSCdOlX+JrLGmOBkScFf9kHY9AWVpXT02GPf8N57q6haNYrHH7+Qu+46g8hIKxUZY9xjScHfhs+c0lGzc6B6U09CUFV8zXN59NHz2bEjjSeeuJC2bet6Eo8xJrzY105/HrY6UlXeeONHevSYSmZmLgB16lRl2rTBlhCMMRXGkkKB7EOwydcfX7uKLR1t2LCXXr3eZtSoz1i8OIWPPlpdofs3xpgCVj4qkPwZ5GZC0x5Qo1mF7NLpwG4JDz74NRkZuTRoEMfLL1/MlVeeXCH7N8aYoiwpFPitYrvJXr16JzfemMgPP2wFYOjQU3jhhX42EpoxxlOWFABy0mCjrzeO9hXTFHXFiu388MNWmjWrweuvX0r//u0rZL/GGFMSSwoAyZ9DbgY0OQtquPdQ2K5daTRoUA1wrgz27cvkuutOtQ7sjDGVht1ohsOtjjq4UzpKT89hzJgviY9/kbVrdwHOMJmjR3e3hGCMqVTsSiEnzblSAFdaHc2bt5GRI2eyYUMqERHCggW/c9JJwTneszEm9FlS2DjbVzo6A2qW3/jP+/dn8o9/zGXSpOUAnHJKQ6ZOHURCgjcPxRljTCAsKawr/wfWvvtuM0OGTGfr1oNER0fw0EM9GTv2HKpUiSy3fRhjjBvCOynkpDvPJ0C5tjpq3Lg6e/ZkcOaZzZk8eQAnn2wd2BljgkN4J4WNsyE3HRp3h5qtjnkzqsrcucn06dMGEeGEE+ry3Xc30KVLY+vAzhgTVML7jFUOfR1t2bKfAQM+4KKL3uXf//6pcHm3bk0tIRhjgk74XinkZPiVjsre6ig/3+nA7t5753LwYDa1asUQE2P3DIwxwS18k8KmL5zmqI0SoFbZxjdev34PI0fO5JtvfgfgsstO5NVXL6Fp0xpuRGqMMRUmfJPCMZaOvv9+C716vU1mZi4NG1bjlVcuZvDgjoVjIBhjTDALz6SQkwEbZjrTZXyKOSGhKe3a1eW005rw3HN9qVfPOrAzxoSO8EwKm+ZAziFo1K3U0lFWVi7PPPM9N9+cQP36cVSpEsnChTdSo0ZMBQVrjDEVJzyTQkHpqF3JzyYsXpzCiBGJrFmzi7Vrd/Puu5cDWEIwxoSs8EsKuZmQXHLpKC0tmwcf/JoXX1yCKrRvX4+bb+5WgUEaY4w3wi8pbPoSsg9Cw9Ogdts/vf3VV8mMHDmTjRv3ERkp3HtvD8aNO5/Y2PD7VRljwk/4nelKaHX022976NPnHVShS5fGTJkykK5dm1RwgMYY453wSgq5WbAh0Zkupq+j9u3rcdddZ9CgQTXuvbcH0dH2MJoxJryEV1L4/UvIPgANOkOdduzYcYg77/yCW27pxgUXOK2Qnn++n8dBGmOMd1ztnEdE+onIOhFJEpH7ink/RkSm+d5fIiLxbsbDb9MB0HaDeeedlXTsOIGPPlrN3XfPQVVd3bUxxgQD15KCiEQCrwIXAx2Bq0WkY5HVRgCpqnoC8DzwpFvxOKWjT9mcWov+Dzdk2LBP2Ls3g7592/LJJ0PsiWRjjMHd8lF3IElVkwFE5ENgELDGb51BwCO+6enAKyIi6sLX9vxNc5n4dXvGzu7Locxt1KkTy/PPX8SwYZ0tIRhjjI+bSaEZsMVvPgU442jrqGquiOwH6gG7yzuY/Stm8Ojc8ziUGc0VV5zEK69cQuPG1ct7N8YYE9TcTArFff0uegUQyDqIyChgFEDLlsc2jnKdatlMvmoW2ee8wBXD+xzTNowxJtS5mRRSgBZ+882BP46yToqIRAG1gL1FN6Sqk4BJAAkJCcdWWrrkXQb0OgAxNY/p48YYEw7cbH20FGgnIq1FpAowBEgssk4icL1vejDwtRv3EwpZQjDGmBK5dqXgu0cwGpgDRAJTVXW1iDwGLFPVRGAK8I6IJOFcIQxxKx5jjDGlc/XhNVWdBcwqsuxhv+lM4NgHSDbGGFOubGR5Y4wxhSwpGGOMKWRJwRhjTCFLCsYYYwpZUjDGGFNIgq13UBHZBfx+jB+vjwtdaFRydszhwY45PBzPMbdS1QalrRR0SeF4iMgyVU3wOo6KZMccHuyYw0NFHLOVj4wxxhSypGCMMaZQuCWFSV4H4AE75vBgxxweXD/msLqnYIwxpmThdqVgjDGmBCGZFESkn4isE5EkEbmvmPdjRGSa7/0lIhJf8VGWrwCO+e8iskZEfhaRr0SklRdxlqfSjtlvvcEioiIS9C1VAjlmEbnS92+9WkTer+gYy1sA/7dbisg8EVnh+/99iRdxlhcRmSoiO0Xkl6O8LyLyku/38bOIdC3XAFQ1pH5wuuneALQBqgArgY5F1rkNmOibHgJM8zruCjjmC4A43/St4XDMvvVqAAuAxUCC13FXwL9zO2AFUMc339DruCvgmCcBt/qmOwKbvI77OI+5J9AV+OUo718CzMYZufJMYEl57j8UrxS6A0mqmqyq2cCHwKAi6wwC3vJNTwd6iUhxQ4MGi1KPWVXnqWq6b3Yxzkh4wSyQf2eA/wOeAjIrMjiXBHLMI4FXVTUVQFV3VnCM5S2QY1agYAStWvx5hMegoqoLKGYESj+DgLfVsRioLSJNymv/oZgUmgFb/OZTfMuKXUdVc4H9QL0Kic4dgRyzvxE43zSCWanHLCKnAS1U9bOKDMxFgfw7twfai8hCEVksIv0qLDp3BHLMjwDXikgKzvgtd1RMaJ4p6997mbg6yI5HivvGX7SJVSDrBJOAj0dErgUSgPNcjch9JR6ziEQAzwPDKyqgChDIv3MUTgnpfJyrwW9FpJOq7nM5NrcEcsxXA2+q6rMichbOaI6dVDXf/fA84er5KxSvFFKAFn7zzfnz5WThOiIShXPJWdLlWmUXyDEjIr2BB4CBqppVQbG5pbRjrgF0AuaLyCac2mtikN9sDvT/9qeqmqOqG4F1OEkiWAVyzCOAjwBUdREQi9NHUKgK6O/9WIViUlgKtBOR1iJSBedGcmKRdRKB633Tg4Gv1XcHJ0iVesy+UsrrOAkh2OvMUMoxq+p+Va2vqvGqGo9zH2Wgqi7zJtxyEcj/7U9wGhUgIvVxyknJFRpl+QrkmDcDvQBE5CScpLCrQqOsWInAMF8rpDOB/aq6rbw2HnLlI1XNFZHRwByclgtTVXW1iDwGLFPVRGAKziVmEs4VwhDvIj5+AR7z00B14D++e+qbVXWgZ0EfpwCPOaQEeMxzgL4isgbIA+5V1T3eRX18Ajzme4A3RORunDLK8GD+kiciH+CU/+r77pOMA6IBVHUizn2TS4AkIB24oVz3H8S/O2OMMeUsFMtHxhhjjpElBWOMMYUsKRhjjClkScEYY0whSwrGGGMKWVIwlY6I5InIT34/8SWsG3+03iTLuM/5vp44V/q6iOhwDNu4RUSG+aaHi0hTv/cmi0jHco5zqYh0CeAzfxORuOPdtwkPlhRMZZShql38fjZV0H6HqmpnnM4Sny7rh1V1oqq+7ZsdDjT1e+8mVV1TLlEejnMCgcX5N8CSggmIJQUTFHxXBN+KyHLfT49i1jlZRH7wXV38LCLtfMuv9Vv+uohElrK7BcAJvs/28vXTv8rXz32Mb/l4OTw+xTO+ZY+IyBgRGYzTv9R7vn1W9X3DTxCRW0XkKb+Yh4vIy8cY5yL8OkITkddEZJk44yg86lt2J05ymici83zL+orIIt/v8T8iUr2U/ZgwYknBVEZV/UpHH/uW7QT6qGpX4CrgpWI+dwvwoqp2wTkpp/i6PbgKONu3PA8YWsr+BwCrRCQWeBO4SlVPwekB4FYRqQv8BThZVU8FHvf/sKpOB5bhfKPvoqoZfm9PBy73m78KmHaMcfbD6daiwAOqmgCcCpwnIqeq6ks4/eJcoKoX+Lq+eBDo7ftdLgP+Xsp+TBgJuW4uTEjI8J0Y/UUDr/hq6Hk4ffoUtQh4QESaAzNUdb2I9AK6AUt93XtUxUkwxXlPRDKATTjdL3cANqrqb7733wJuB17BGZ9hsoh8DgTcNbeq7hKRZF+fNet9+1jo225Z4qyG0+2D/6hbV4rIKJy/6yY4A878XOSzZ/qWL/TtpwrO780YwJKCCR53AzuAzjhXuH8aNEdV3xeRJUB/YI6I3ITTzfBbqnp/APsY6t9hnogUO8aGrz+e7jidsA0BRgMXluFYpgFXAr8CH6uqinOGDjhOnBHIxgOvApeLSGtgDHC6qqaKyJs4HcMVJcBcVb26DPGaMGLlIxMsagHbfH3kX4fzLfkIItIGSPaVTBJxyihfAYNFpKFvnboS+PjUvwLxInKCb/464BtfDb6Wqs7CuYlbXAuggzjddxdnBnAZzjgA03zLyhSnqubglIHO9JWeagJpwH4RaQRcfJRYFgNnFxyTiMSJSHFXXSZMWVIwwWICcL2ILMYpHaUVs85VwC8i8hNwIs6QhWtwTp5fisjPwFyc0kqpVDUTpwfK/4jIKiAfmIhzgv3Mt71vcK5iinoTmFhwo7nIdlOBNUArVf3Bt6zMcfruVTwLjFHVlThjM68GpuKUpApMAmaLyDxV3YXTMuoD334W4/yujAGsl1RjjDF+7ErBGGNMIUsKxhhjCllSMMYYU8iSgjHGmEKWFIwxxhSypGCMMaaQJQVjjDGFLCkYY4wp9P8BoBBQVzFoMrUAAAAASUVORK5CYII=\n",
        "text/plain": "<Figure size 432x288 with 1 Axes>"
       },
       "metadata": {
        "needs_background": "light"
       },
       "output_type": "display_data"
      }
     ],
     "persistentId": "8f75a628-0448-474b-8eba-ccfae9ffc601",
     "text": "fpr, tpr, thresholds = metrics.roc_curve(coding_scores_unspliced_not_low_complexity['in_cds_no_stop_codon'], \n                                         coding_scores_unspliced_not_low_complexity['jaccard_in_peptide_db'])\n\nfig, ax = plt.subplots()\nlw = 2\nax.plot(fpr, tpr, lw=2, color='darkorange')\nax.plot([0, 1], [0, 1], color='navy', lw=lw, linestyle='--')\nax.set(xlabel='False Positive Rate', ylabel='True Positive Rate')"
    },
    "executionTime": "2019-10-11T20:08:32.132Z"
   },
   {
    "cell": {
     "executionCount": 111,
     "executionEventId": "6fc2b997-bdbf-4712-8c88-b162d2af1b45",
     "hasError": true,
     "id": "3f61b436-51d7-4dae-a329-c238ba62d3d5",
     "outputs": [
      {
       "ename": "NameError",
       "evalue": "name 'y_true' is not defined",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
        "\u001b[0;32m<ipython-input-111-88df248f8c7c>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mfpr\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtpr\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mthresholds\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmetrics\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mroc_curve\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0my_true\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0my_pred\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      2\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      3\u001b[0m \u001b[0mfig\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0max\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mplt\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msubplots\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[0mlw\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;36m2\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m \u001b[0max\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mplot\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfpr\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtpr\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlw\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m2\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcolor\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'darkorange'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;31mNameError\u001b[0m: name 'y_true' is not defined"
       ]
      }
     ],
     "persistentId": "8f75a628-0448-474b-8eba-ccfae9ffc601",
     "text": "fpr, tpr, thresholds = metrics.roc_curve(y_true, y_pred)\n\nfig, ax = plt.subplots()\nlw = 2\nax.plot(fpr, tpr, lw=2, color='darkorange')\nax.plot([0, 1], [0, 1], color='navy', lw=lw, linestyle='--')\nax.set(xlabel='False Positive Rate', ylabel='True Positive Rate')"
    },
    "executionTime": "2019-10-11T20:08:58.330Z"
   },
   {
    "cell": {
     "executionCount": 112,
     "executionEventId": "aa812828-4fb9-4e1d-a4cc-b8e7ea0afabf",
     "hasError": false,
     "id": "a18fd62e-900f-4a7d-8ada-1b3b45c4d677",
     "outputs": [],
     "persistentId": "27025afb-bac7-4e3d-822e-f3238fba3a50",
     "text": "y_true = coding_scores_unspliced_not_low_complexity['in_cds_no_stop_codon']\ny_pred = coding_scores_unspliced_not_low_complexity['jaccard_in_peptide_db']"
    },
    "executionTime": "2019-10-11T20:08:59.878Z"
   },
   {
    "cell": {
     "executionCount": 113,
     "executionEventId": "200aa321-e7ad-4cee-be34-a93bb24cbb19",
     "hasError": false,
     "id": "3f61b436-51d7-4dae-a329-c238ba62d3d5",
     "outputs": [
      {
       "data": {
        "text/plain": "[Text(0, 0.5, 'True Positive Rate'), Text(0.5, 0, 'False Positive Rate')]"
       },
       "execution_count": 113,
       "metadata": {},
       "output_type": "execute_result"
      },
      {
       "data": {
        "image/png": "iVBORw0KGgoAAAANSUhEUgAAAYUAAAEKCAYAAAD9xUlFAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDMuMC4zLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvnQurowAAIABJREFUeJzt3Xl4FFX28PHvyUJC2PcdAggooiBEVFRcWEQRcJRRFEUUwQ11HHHQ1wX15zi474gIjLviMKhBQWQURBEQBBEBkRAQguyELfty3j+qE5oYkg6kUunu83mePF1VXV11KpA6Xadu3SuqijHGGAMQ4XUAxhhjKg9LCsYYYwpZUjDGGFPIkoIxxphClhSMMcYUsqRgjDGmkCUFY4wxhSwpGGOMKWRJwRhjTKEorwMoq/r162t8fLzXYRhjTFD58ccfd6tqg9LWC7qkEB8fz7Jly7wOwxhjgoqI/B7IelY+MsYYU8iSgjHGmEKWFIwxxhSypGCMMaaQJQVjjDGFXEsKIjJVRHaKyC9HeV9E5CURSRKRn0Wkq1uxGGOMCYybVwpvAv1KeP9ioJ3vZxTwmouxGGOMCYBrzymo6gIRiS9hlUHA2+qMB7pYRGqLSBNV3eZWTMYYUyHysiFrP2Tt873uh+z9h6eLm8/LOurmfkiqSWx0Pqee2x16POJq6F4+vNYM2OI3n+Jb9qekICKjcK4maNmyZYUEZ4wJU4Un9KIn7gBP8Nn7ITezXEJRhbGf9+HZb7pxapMd/NDhV6LLZctH52VSkGKWaXErquokYBJAQkJCsesYYwwAeTmQsQvSd0Lm3qOfyAtO8m6c0COioEotiPH7KW0+MhbkyNOiAGxIggVb6DvoDPJOPz2kk0IK0MJvvjnwh0exGGMqK1XIPuCc5NN3QvoOv+lilmXuPb79HcsJvUotiKl9eD6q6p9O8IHaty+T5ORUunZtAsCjz53FkJG7C+fd5mVSSARGi8iHwBnAfrufYEyYKPg2n7YDMko4wRf8lFBv/xOJgKoNIK4hVK1XhhN87eM+oR+vTz/9lVtv/ZyICGH16tuoVSuWqlWjKywhgItJQUQ+AM4H6otICjAOnCsfVZ0IzAIuAZKAdOAGt2IxxrhM1Sm/FJzEC070aTv+PJ+xEzJTy7b96OrOST6uke+14dHnq9Z1EkMQ2bkzjTvvnM20aasBOPPM5uzbl0mtWrEVHoubrY+uLuV9BW53a//GmOOUlw3puw6f1NN2HHnSLzqflx34tiUS4gq+zftO6NUaHZ4umI9r6Hzrj45z7zg9pKq8994q7rrrC/buzSAuLponnriQ0aO7ExnpTWILuq6zjTHHKD/PubmasbvkmnzBfNa+sm2/So3DJ/lqft/gqxY9yTcMym/zbrj11s95/fUfAejduw2TJl1K69Z1PI3JkoIxwSgvBzL3QMYe5yTvP52xxzdfZDozlaM08Cte4bf5IiWaoif9gmXRVV073FB12WUnMm3aap59ti833NAF8ehehj9LCsZ4LSej+JN4cSf8gunsA8e2r9g6EFuv9Lp8XENnXfs2X67Wr9/DV19t5JZbEgDo1+8ENm26y5N7B0djScGY8pZ9qEhJZoffDdddfz7552aUfR8S4Zzcq9bzvdZ3pqvWP7z8T9N1nOaWpsLl5ubz3HOLGDduPllZuXTp0pgzz2wOUKkSAlhSMKZkO5bD3l+PXKb5TinG/6RfMJ22A3LTy7aPiOgAT+p+J/+YWvYtPkisXLmdESMS+fFHp8X9sGGdadeursdRHZ0lBWOKys2EtG2w5F+w6o2yfz4yxinBFNbd/V8b/PkEH13ds3bxxj1ZWbk8/vgCxo9fSG5uPi1b1uL11y+lX78TvA6tRJYUTHhRdVrVHPgdDmz2vf4OB/3m03ccXj8iGtoOhMgqR24nps6RLWr8T/xVathJ3nD//V/x/POLAbj99tP51796UaNGjMdRlc6SggltORmQ8g1snO287k+G7IMlfyYiynm6tX4nuPBl59WYMvrHP85m0aIUnnqqN+ee28rrcAJmScGEFlXYl+QkgY2zIWX+nzs4i64GNVsd/qnR8sj5ak0gItKT8E3wmjt3AxMn/si0aYOJioqgcePqfP/9jZWimWlZWFIwwS8nDbbMP5wI9icf+X7DrtD6YojvC/VOhti6Vt4x5SY1NYMxY75k6tSfAPj3v1cwcmQ3gKBLCGBJwVQmmu/8BCJ1PWz6wnc1sODIDtNi60Krvr5EcJFT9zfGBR9/vJbbbpvF9u2HiImJZNy48xg+vIvXYR0XSwqm4uVmOs0896yBPat9r2ucsk+gSeEIAo27Q3w/JxE0Pt3KP8ZV27cf4o47ZjN9+hoAevRowZQpAznxxPoeR3b8LCmEo01zYemTkJ9bwTtWOPSHU9452slfAjyZV63nuxro57zGNSi/MI0pxaef/sr06WuoVi2a8eN7c9ttpxMREXylouJYUghHK16GzV95t3+JhDodoF5HqH8y1O3oTNftAFGV6+lOYwpkZuYSG+ucMkeO7EZyciq33no68fG1PY6sfFlSCAd71x3Z42XGbuc1YQy06V+xscTWgzrtIaryt9c2BiA/X5kwYSn//Oe3LF48glatahMRITz5ZB+vQ3OFJYVQ992DsOSfxb/X8kJocX6FhmNMMFm3bjcjRiSycOEWAD744Bfuu+8cj6NylyWFULbhMychSCQ06opvGHBH9abQLLT/cxtzrHJy8njmme959NFvyMrKo1GjakyY0J/LLz/J69BcZ0khVB34Hb4Y5kyf80/oPtbbeIwJEr/8spNhwz5mxYrtANxwQxeefbYvdeqEx3gRlhRCUV42fHaV05Nnm/5w+r1eR2RM0MjPV1at2kmrVrWYNGkAffu29TqkCmVJIRQtGAvblkCNFtDvLeti2ZhSrF69k44dGyAinHpqIz79dAg9e7aievUqpX84xNjZItTsWA7LX3A6dbv0I6c9vzGmWAcPZjF69Cw6dXqN//53beHySy5pF5YJAexKIfSk/ua8tr4Emp7pbSzGVGJz5iQxatRnbN68n6ioCDZt2lf6h8KAJYVQkpsJix93puue6G0sxlRSe/dmcPfdc3j77ZUAdO3ahClTBtKlS2OPI6scLCmEkj++d/oSqhkPZz7kdTTGVDo//bSdfv3eZceONGJiInn00fO5554eREVZJb2AJYVQUvCkcr2ToEp1b2MxphJq374e1atXoX37ekyePJD27e2eW1GWFELJuo+c12Y9vY3DmEpCVXn//VUMGNCBmjVjiIuLZv784TRtWiNkOrArb3bNFCoy9sCGRKf56cnDvI7GGM9t2rSPiy56l2uv/Zj77vtf4fLmzWtaQiiBXSmEij1rID/HGUugelOvozHGM3l5+UyYsJT77/+KtLQc6tatSo8eLbwOK2hYUgg1kdb7qAlfa9fuYsSIRBYtSgHgyitP5uWXL6Zhw2oeRxY8LCkYY0LCxo2pdOnyOtnZeTRpUp0JE/pz2WXWNLusXE0KItIPeBGIBCar6vgi77cE3gJq+9a5T1VnuRmTMSY0tW5dh7/+tSOxsVE880xfate2AZuOhWtJQUQigVeBPkAKsFREElV1jd9qDwIfqeprItIRmAXEuxWTMSZ0ZGTk8Nhj3/CXv5xE9+7NAHjrrcuIjLT2M8fDzd9edyBJVZNVNRv4EBhUZB0FavqmawF/uBiPMSZEfPvt73Tp8jrjxy9k1KiZ5OcrgCWEcuBm+agZsMVvPgU4o8g6jwBfisgdQDWgt4vxhLYt853XqPDo892EpwMHsrj//v8xYcIyADp2bMDEiZdaE9Ny5GZaLe5fSYvMXw28qarNgUuAd0T+3M+ziIwSkWUismzXrl0uhBrkDm2DRY84093u9jQUY9wya9Z6OnWawIQJy4iKiuDhh3uyfPkozjyzudehhRQ3rxRSAP/Gwc35c3loBNAPQFUXiUgsUB/Y6b+Sqk4CJgEkJCQUTSwmbRtoPjToDK0v9joaY8rd/v2ZDB06g337MklIaMqUKQM59dRGXocVktxMCkuBdiLSGtgKDAGuKbLOZqAX8KaInATEAnYpUBaq8P3DzrQNpmNCiKqiChERQq1asbz0Uj927Ejjb3870zqwc5FrSUFVc0VkNDAHp7npVFVdLSKPActUNRG4B3hDRO7GKS0NV1W7EiiL/Rsh+XNnupp1/WtCwx9/HOS22z7n3HNbcs89PQC47rrOHkcVHlx9TsH3zMGsIsse9pteA5ztZgwhLyv18PTF73gXhzHlQFWZOnUF99zzJfv3Z7F4cQq33XY6VatGex1a2LAnmoNZxl54N8GZrtPBht40QS05OZWRI2fy9dcbAejfvx0TJ15qCaGCWVIIZr++f3j61FHexWHMccjLy+ell5bwwANfk5GRS/36cbz0Uj+GDOmEiDU1rWiWFILRuv/A71/Cpi+d+f4fwIlDvI3JmOMwffpaMjJyufrqTrz4Yj8aNLAO7LxiSSFYpG2HlAWwLxm+u//w8th6cMJl3sVlzDHIzs7j4MEs6tWLIzIygilTBrJ+/R4GDOjgdWhhz5JCsJj5V9j63eH50+6A+qdA0x4QZR1/meCxdOlWRoxIpHnzmnz++TWICCeeWJ8TT6zvdWgGSwrBI32H89r6YmjVF7reBVZvNUEkPT2HcePm8dxzi8nPV9LTc9i5M41GjWw88crEkkKwOf8FqNve6yiMKZP58zcxcuRMkpL2EhEhjBlzFo8+egFxcdayqLKxpGCMcY2qcueds3nllaUAnHJKQ6ZMGcjppzfzODJzNJYUgkF+HmTscaajbLhNEzxEhJo1Y4iOjuDBB3ty333nUKVKpNdhmRJYUggGKQsgcy/Ubgs1WnodjTEl2r07nQ0b9nLGGU7vpQ89dB5Dh55Kx44NPI7MBCKgXqVEpIqInOB2MOYofpvuvLb/q91cNpWWqvLhh79w0kmvctll00hNzQAgNjbKEkIQKTUpiEh/YBUw1zffRUQ+djsw45OfB+v/60y3/6u3sRhzFCkpBxg06EOuvvq/7N6dTseODUhPz/E6LHMMAikfPYYzYto8AFX9ya4aKtDW75zmqLXaQMPTvI7GmCPk5yuTJy/n3nvncuBAFjVrxvDss30ZMeI066IiSAWSFHJUdV+Rf2Dr3rqi/PYf59VKR6YSGjEikTff/AmAgQM7MGHCJTRrVrOUT5nKLJB7CmtF5EogQkRai8gLwGKX4zJQpHQ02NtYjCnGtdeeQsOG1fjwwyv45JOrLCGEgECSwmigG5APzAAygbvcDMr4/PG90+dRzXho1M3raIzhl1928uKLh78T9urVhuTkO7nqKuvRNFQEUj66SFXHAmMLFojI5TgJwrjJSkemksjKyuVf//qOJ574lpycfBISmnL22U7z6GrVqngcnSlPgVwpPFjMsgfKOxBThOYfLh11sFZHxjtLlqTQrdskHn30G3Jy8rn11gROOaWR12EZlxz1SkFELgL6Ac1E5Dm/t2rilJKMm7Z+D4f+gJqtoFGC19GYMJSWls1DD83jhRcWowrt2tVl8uSB9OzZyuvQjItKKh/tBH7BuYew2m/5QeA+N4MyHC4dtRtspSPjiQce+JoXX1xCRIRw771n8cgj59vQmGHgqElBVVcAK0TkPVXNrMCYjJWOTCXwwAPnsmrVTp58sjcJCU29DsdUkEDuKTQTkQ9F5GcR+a3gx/XIwtkfi+HQVqefo8bdvY7GhInExHVccsl75OTkAdCgQTW++mqYJYQwE0hSeBP4NyDAxcBHwIcuxmQKWx1Z6ci4b+fONIYMmc6gQR8ye3YSb7210uuQjIcCSQpxqjoHQFU3qOqDwAXuhhXGNP/IDvCMcYmq8u67P3PSSa8ybdpq4uKiefHFftxwQxevQzMeCuQ5hSxxnkrZICK3AFuBhu6GFca2LYFDKVC9OTSx0pFxx+bN+7nlls+YPTsJgN692zBp0qW0bl3H48iM1wJJCncD1YE7gX8CtYAb3QwqrBVeJQwGCahnc2PK7MsvNzB7dhK1a8fy3HN9GT68iz2RbIAAkoKqLvFNHgSuAxCR5m4GFbZUrXRkXJOWll349PGIEaexdesBRo3qRpMmNTyOzFQmJX4VFZHTReQyEanvmz9ZRN7GOsRzx/Yf4OBmqN4Mmp7pdTQmROTm5vPUUwtp1eoFkpNTAWeYzHHjzreEYP7kqElBRP4FvAcMBb4QkQdwxlRYCbSvmPDCzDr/VkdWOjLHb+XK7ZxxxmTGjv0fe/Zk8Mknv3odkqnkSiofDQI6q2qGiNQF/vDNr6uY0MKMKqy30pEpH1lZuTz++ALGj19Ibm4+LVvWYtKkS7noIhsfy5SspKSQqaoZAKq6V0R+tYTgou1L4cDvUL0pND3L62hMEFuxYhtDh85g7drdiMDo0afzxBO9qFEjxuvQTBAoKSm0EZGC7rEFiPebR1UvL23jItIPeBGIBCar6vhi1rkSeARnNLeVqnpN4OGHkIIbzO2usNKROS4xMVFs2JBKhw71mDx5IOec09LrkEwQKSkpXFFk/pWybFhEIoFXgT5ACrBURBJVdY3fOu2A+4GzVTVVRMLz+QfVI8dOMKaMli/fxmmnNUZE6NixAbNnD6VHjxbExgbS6tyYw0rqEO+r49x2dyBJVZMBRORDnPsUa/zWGQm8qqqpvn3uPM59BqcdP8KBTVCtCTQ72+toTBBJTc1gzJgvmTr1Jz744AqGDOkEwIUXtvY4MhOs3Pwa0QzY4jefApxRZJ32ACKyEKfE9IiqflF0QyIyChgF0LJlCF4KF3aTbaUjE7iPP17LbbfNYvv2Q8TERLJnT7rXIZkQ4GZSKO7xSC1m/+2A84HmwLci0klV9x3xIdVJwCSAhISEotsIbkeUjgZ7G4sJCtu3H+KOO2Yzfbpz0X322S2YPHkgJ55Y3+PITCgIOCmISIyqZpVh2ylAC7/55jjNWouus1hVc4CNIrIOJ0ksLcN+gtvOFbB/I8Q1gmbneB2NqeR+/PEP+vR5h9TUTKpVi2b8+N7cdtvpRERYFxWmfJRaqxCR7iKyCljvm+8sIi8HsO2lQDsRaS0iVYAhQGKRdT7B1+Oq76np9kByGeIPfv6lo4hIb2MxlV7Hjg1o0KAaF13UltWrb2P06O6WEEy5CqSA/RJwKbAHQFVXEkDX2aqaC4wG5gBrgY9UdbWIPCYiA32rzQH2iMganKel71XVPWU/jCDlXzqyEdZMMfLzlUmTfmTfPmfww6pVo1mwYDizZw+lVavaHkdnQlEg5aMIVf29SA+KeYFsXFVnAbOKLHvYb1qBv/t+ws/On2DfBohrCM3O9ToaU8msW7ebm26ayXffbWbp0q288YbzXapRo+oeR2ZCWSBJYYuIdAfU9+zBHYANx1keCktHl1vpyBTKycnj2WcX8cgj88nKyqNx4+pcfHE7r8MyYSKQpHArTgmpJbAD+J9vmTke9sCaKcaKFdsYMSKRFSu2A3DDDV149tm+1KlT1ePITLgIJCnkquoQ1yMJN7t+hn1JULUBNO/pdTSmEtiwYS/du08mNzef+PjaTJp0KX36tPU6LBNmAkkKS31NRacBM1T1oMsxhYcjSkfWFYGBtm3rct11p1KjRhX++c9eVK9exeuQTBgqtfWRqrYFHge6AatE5BMRsSuH42GlIwMcOpTNnXfOZtGiww/+T5kykBdfvNgSgvFMQH0qqOr3qnon0BU4gDP4jjlWu1dB6m9QtT60OM/raIwH5sxJ4uSTJ/Dyyz9wyy2f4zTEw8ZJNp4rtW4hItVxOrIbApwEfAr0cDmu0FbQTfYJf7HSUZjZuzeDu++ew9tvrwSgW7cmTJky0JKBqTQCOSP9AswEnlLVb12OJ/RZ6ShsTZ++httvn8XOnWnExkbx6KPn8/e/n0VUlHWCaCqPQJJCG1XNdz2ScLFnNez9FWLrQctSHww3IWLfvkxGjZpJamomPXu24o03BtC+fT2vwzLmT46aFETkWVW9B/iviPypZ9JARl4zxVhX0OrISkehTlXJz1ciIyOoXTuWCRP6k5qawc03J1h/RabSKumsNM33WqYR10wprJvssLBp0z5GjZrJhRe25r77nN5vCwbAMaYyO2oxU1V/8E2epKpf+f/g3HA2ZbVnDexdC7F1ocWFXkdjXJCXl89LLy2hU6cJzJ2bzCuv/EBmZq7XYRkTsEDucN1YzLIR5R1IWCgoHZ1wGURGexuLKXdr1+6iZ883ueuuL0hLy2HIkE4sX36zjZNsgkpJ9xSuwmmG2lpEZvi9VQPYV/ynTIms1VFIys3N58knv+OxxxaQnZ1H06Y1eO21/gwc2MHr0Iwps5K+wvyAM4ZCc+BVv+UHgRVuBhWS9qx1Wh7F1oGWvbyOxpSjiAjhyy+Tyc7OY+TIrjz1VB9q1471OixjjslRk4KqbgQ24vSKao5XwVVC20FWOgoBGRk5HDyYTcOG1YiIECZPHsCWLQe48MLWXodmzHE56j0FEfnG95oqInv9flJFZG/FhRgiCp5ittJR0Fuw4Hc6d57ItdfOKOyeol27epYQTEgoqXxU8GRV/YoIJKTtXef0dxRTG1r19joac4wOHMji/vv/x4QJywCIjo5k9+50GjSo5nFkxpSfkpqkFjzF3AKIVNU84CzgZsD+CsqioHR0wiCItN4vg9Hs2evp1GkCEyYsIyoqgnHjzmP58lGWEEzICaSt3CfA6SLSFngb+Bx4H7jUzcBCirU6ClqqysiRM5kyxWlbkZDQlKlTB3LKKY08jswYdwTynEK+quYAlwMvqOodQDN3wwohe39zRlmrUhNaWuko2IgIzZvXJDY2imee6cOiRSMsIZiQFtBwnCLyV+A64DLfMms+E6j1Bd1kD4KoGG9jMQH544+DbNiwl3PPbQXA//t/53LddafStm1djyMzxn2BPtF8AU7X2cki0hr4wN2wQsg6Kx0FC1VlypTldOz4Kldc8RF79qQDUKVKpCUEEzZKvVJQ1V9E5E7gBBE5EUhS1X+6H1oISE2CXT85paNWfb2OxpQgOTmVkSNn8vXXGwG49NL25ORYj/Em/AQy8tq5wDvAVkCAxiJynaoudDu4oFf4wNpAKx1VUgUd2D344DzS03OoXz+Ol17qx5AhnWw0NBOWArmn8DxwiaquARCRk3CSRIKbgYUEa3VU6Q0b9gnvv78KgGuuOYUXXrjImpmasBbIPYUqBQkBQFXXAtbYvjT7NsDOFVClBsRb6aiyGjmyK82b1yQxcQjvvXe5JQQT9gK5UlguIq/jXB0ADMU6xCtdQbcWbQZAlHWOVlksXbqVr7/eyNixzsA3558fT1LSHcTEWPfWxkBgSeEW4E7gHzj3FBYAL7sZVEiw0lGlkp6ew7hx83juucXk5ys9erQobHJqCcGYw0r8axCRU4C2wMeq+lTFhBQC9iXDjh8hujrEX+R1NGFv/vxN3HRTIhs2pBIRIYwZcxbdujX1OixjKqWSekn9fzhdXAwF5opIcSOwlUhE+onIOhFJEpH7SlhvsIioiITGzeuC0lHbARBd1dtYwtj+/ZncfPNMLrjgLTZsSOWUUxqyePEInn66L3Fx9vylMcUp6UphKHCqqqaJSANgFjA10A2LSCTO4Dx9gBRgqYgk+t+09q1XA6c8taSswVdahaWjwd7GEeYeemgekyYtJzo6goce6snYsedQpUqk12EZU6mVlBSyVDUNQFV3iUggLZX8dcd50C0ZQEQ+BAYBa4qs93/AU8CYMm6/ctq/CXYsg+hqEH+x19GEHVUtfL7g4YfPY+PGfYwf34uTT27ocWTGBIeSTvRtRGSG7+djoK3f/IwSPlegGbDFbz6FIh3pichpQAtV/azMkVdWha2OLrXSUQVSVd5/fxUXXvg22dl5ANSvH8fMmVdbQjCmDEq6UriiyPwrZdx2cY+DauGbzpXH88DwUjckMgoYBdCyZcsyhlHBrNVRhUtJOcCtt37OZ5/9BsB77/3MDTec5nFUxgSnksZo/uo4t52CM0BPgebAH37zNYBOwHzf5X5jIFFEBqrqsiKxTAImASQkJCiV1YHfYfsPEBUHra105Lb8fOWNN37k3nvncvBgNrVqxfDss30ZPryL16EZE7TcbKC9FGjn61V1KzAEuKbgTVXdj99QnyIyHxhTNCEElcLSUX+IjvM2lhCXlLSXkSNnMn/+JgAGDerAhAn9adq0hreBGRPkXEsKqporIqOBOUAkMFVVV4vIY8AyVU10a9+eKUgKVjpy3bff/s78+Zto2LAar7xyMYMHd7QO7IwpBwEnBRGJUdWssmxcVWfhNGX1X/bwUdY9vyzbrnQObIFtiyGqKrS5xOtoQtK+fZnUru10GTJ8eBd27UpnxIjTqFfPrsqMKS+lNjMVke4isgpY75vvLCLWzUVR6/1LR9apWnnKyspl3Lh5tGr1AuvX7wGcYTL/8Y+zLSEYU84CuVJ4CbgU5+lmVHWliFzgalTByEZYc8XixSmMGJHImjW7AJgzZwPt2tXzOCpjQlcgSSFCVX8vUq/Ncyme4HRgC2xb5PSG2tpKR+UhLS2bhx6axwsvLEYV2rWry5QpAws7sTPGuCOQpLBFRLoD6uu64g7gN3fDCjJJvmf5Wl8CVap7G0sIWLIkhWuumUFyciqRkcKYMT0YN+48qla1/oqMcVsgSeFWnBJSS2AH8D/fMlPASkflqnbtWLZuPUDnzo2YMmWg9WhqTAUqNSmo6k6cZwxMcQ5uhT8WOqWjNpd6HU3Q+u67zZx9dgtEhA4d6vP119dz+ulNiY62DuyMqUilJgUReQO/7ikKqOooVyIKNuv/67zGX2ylo2Owc2cad945m2nTVvPWW5cxbFhnAHr0aFHKJ40xbgikfPQ/v+lY4C8c2dFdeLNuso+JqvLee6u4664v2Ls3g7i46MKO7Iwx3gmkfDTNf15E3gHmuhZRMDn0B2xdCJExzoA6JiCbN+/nlls+Y/bsJAD69GnDpEkDiI+v7XFkxphj6eaiNWDtAgHWzwAU4vtBFetzJxBLlqTQu/c7HDqUTe3asTz//EVcf31n66LCmEoikHsKqRy+pxAB7AWOOrRmWCkoHXWwVkeB6tKlMS1a1OTN3a/iAAAULUlEQVTEE+vz6quX0KSJJVNjKpMSk4I4X9864/RyCpCvqpW36+qKdGgbpHwLkVWgjZWOjiY3N59XXvmBYcM6U7duVWJioli48Ebq1LEBiIypjErs+8iXAD5W1TzfjyWEAgWlo1YXQUxNr6OplFau3M4ZZ0zm7rvn8Pe/zylcbgnBmMorkHGXfxCRrq5HEmwKOsCz0tGfZGbm8uCDX5OQ8AbLl2+jZctaXH11J6/DMsYE4KjlIxGJUtVc4BxgpIhsANJwhtlUVQ3fRJG2A1IWOKWjtgO9jqZS+f77LYwYkcivv+5GBEaPPp0nnuhFjRoxXodmjAlASfcUfgC6ApdVUCzBY/0M0Hxo1RdiankdTaWRlLSXc8/9N/n5SocO9ZgyZSBnn13Jx9Q2xhyhpKQgAKq6oYJiCR6/WV9HxTnhhLqMGtWVunWr8tBD5xEb6+Zor8YYN5T0V9tARP5+tDdV9TkX4qn80ndCyjcQER32paPU1AzuuedLbrihS2GX1hMm9LdnDowJYiUlhUigOr4rBuOz/mOndBTfD2LD9wncGTPWcvvts9i+/RA//riNn366GRGxhGBMkCspKWxT1ccqLJJgEealo+3bDzF69Cz++9+1AJxzTksmTx5gycCYEFHqPQXjJ30XbJnnlI5OGOR1NBVKVXn77ZXcffccUlMzqV69Ck8+2ZtbbkkgIsL+qxgTKkpKCr0qLIpgkVRQOroIYut4HU2F2rcvk3vu+ZLU1Ez69TuBiRP706pV+JbPjAlVR00Kqrq3IgMJCmE2wlp+vpKfr0RFRVCnTlVef/1S0tNzuPbaU61cZEyICuSJZgOQvttXOoqCtqFfOvr119307Plvxo//rnDZFVd05LrrrEdTY0KZJYVAJX0Cmgcte0HVul5H45qcnDyeeOJbOneeyMKFW5gyZQWZmbleh2WMqSD2dFGgwqDV0YoV27jxxkR++mk7ACNGnMbTT/exh9CMCSP21x6IjD2w+SuQSDgh9Hr9yMnJY9y4+Tz11ELy8pT4+Nq88cYAevdu43VoxpgKZkkhEAWlo1Z9oWo9r6Mpd1FRESxZspX8fOWuu87g8ccvpHr1Kl6HZYzxgCWFQBSWjgZ7G0c5Ongwi4MHs2natAYiwuTJA9i+/RBnndXC69CMMR6yG82lydjrVzr6i9fRlIs5c5Lo1Ok1hg6dQcG4Sa1b17GEYIyxpFCqDZ9Cfi60uADi6nsdzXHZsyed66//hH793mPz5v0cPJjFnj0ZXodljKlEXE0KItJPRNaJSJKI3FfM+38XkTUi8rOIfCUirdyM55gUlI6CeIQ1VWX69DV07DiBt99eSWxsFE891ZvFi2+ifv04r8MzxlQirt1TEJFI4FWgD5ACLBWRRFVd47faCiBBVdNF5FbgKeAqt2Iqs8xU+P1/IBFBWzpSVYYOncEHH/wCQM+erXjjjQG0bx96N8yNMcfPzSuF7kCSqiarajbwIXDEo8CqOk9V032zi4HmLsZTdkmfQn4OtDgf4hp4Hc0xERE6dmxAjRpVeO21/sybd70lBGPMUbnZ+qgZsMVvPgU4o4T1RwCzi3tDREYBowBatqzA4R3XT3deg+yBtY0bU0lOTqVXL+c5g7Fjz2b48C40b17T48iMMZWdm1cKxXWQo8WuKHItkAA8Xdz7qjpJVRNUNaFBgwr6xp65DzZ96ZSO2l1eMfs8Tnl5+bz44mI6dXqNq66azs6daQBER0daQjDGBMTNK4UUwL+NY3Pgj6IriUhv4AHgPFXNcjGestmQ6CsdXQBxDb2OplRr1uzippsSWbQoBYCBAzvYOAfGmDJzMyksBdqJSGtgKzAEuMZ/BRE5DXgd6KeqO12MpeyCpK+jnJw8nnxyIf/3fwvIzs6jadMavPZafwYO7OB1aMaYIORaUlDVXBEZDczBGe95qqquFpHHgGWqmohTLqoO/MfXHfNmVR3oVkwBy9oPv38JCLSr3K2OrrlmBtOnOw26Ro7sytNP96FWrViPozLGBCtXu7lQ1VnArCLLHvab7u3m/o/ZhpmQlw3Nz4Nqjb2OpkR33XUGP/20nddfv5QLL2ztdTjGmCBnTzQXpxKXjr75ZhOPPjq/cP6cc1qydu3tlhCMMeXCOsQrKusAbJoDCLS/wutoCh04kMXYsXOZOPFHAC64oDU9ezoPgEdFWW43xpQPSwpFJc+EvCxo3rPSlI5mzVrPzTd/RkrKAaKjI3jggXM588zK9ZyfMSY0WFIoap2vdNTO+26yd+9O529/+4L33lsFQPfuzZgyZSCdOlX+JrLGmOBkScFf9kHY9AWVpXT02GPf8N57q6haNYrHH7+Qu+46g8hIKxUZY9xjScHfhs+c0lGzc6B6U09CUFV8zXN59NHz2bEjjSeeuJC2bet6Eo8xJrzY105/HrY6UlXeeONHevSYSmZmLgB16lRl2rTBlhCMMRXGkkKB7EOwydcfX7uKLR1t2LCXXr3eZtSoz1i8OIWPPlpdofs3xpgCVj4qkPwZ5GZC0x5Qo1mF7NLpwG4JDz74NRkZuTRoEMfLL1/MlVeeXCH7N8aYoiwpFPitYrvJXr16JzfemMgPP2wFYOjQU3jhhX42EpoxxlOWFABy0mCjrzeO9hXTFHXFiu388MNWmjWrweuvX0r//u0rZL/GGFMSSwoAyZ9DbgY0OQtquPdQ2K5daTRoUA1wrgz27cvkuutOtQ7sjDGVht1ohsOtjjq4UzpKT89hzJgviY9/kbVrdwHOMJmjR3e3hGCMqVTsSiEnzblSAFdaHc2bt5GRI2eyYUMqERHCggW/c9JJwTneszEm9FlS2DjbVzo6A2qW3/jP+/dn8o9/zGXSpOUAnHJKQ6ZOHURCgjcPxRljTCAsKawr/wfWvvtuM0OGTGfr1oNER0fw0EM9GTv2HKpUiSy3fRhjjBvCOynkpDvPJ0C5tjpq3Lg6e/ZkcOaZzZk8eQAnn2wd2BljgkN4J4WNsyE3HRp3h5qtjnkzqsrcucn06dMGEeGEE+ry3Xc30KVLY+vAzhgTVML7jFUOfR1t2bKfAQM+4KKL3uXf//6pcHm3bk0tIRhjgk74XinkZPiVjsre6ig/3+nA7t5753LwYDa1asUQE2P3DIwxwS18k8KmL5zmqI0SoFbZxjdev34PI0fO5JtvfgfgsstO5NVXL6Fp0xpuRGqMMRUmfJPCMZaOvv9+C716vU1mZi4NG1bjlVcuZvDgjoVjIBhjTDALz6SQkwEbZjrTZXyKOSGhKe3a1eW005rw3HN9qVfPOrAzxoSO8EwKm+ZAziFo1K3U0lFWVi7PPPM9N9+cQP36cVSpEsnChTdSo0ZMBQVrjDEVJzyTQkHpqF3JzyYsXpzCiBGJrFmzi7Vrd/Puu5cDWEIwxoSs8EsKuZmQXHLpKC0tmwcf/JoXX1yCKrRvX4+bb+5WgUEaY4w3wi8pbPoSsg9Cw9Ogdts/vf3VV8mMHDmTjRv3ERkp3HtvD8aNO5/Y2PD7VRljwk/4nelKaHX022976NPnHVShS5fGTJkykK5dm1RwgMYY453wSgq5WbAh0Zkupq+j9u3rcdddZ9CgQTXuvbcH0dH2MJoxJryEV1L4/UvIPgANOkOdduzYcYg77/yCW27pxgUXOK2Qnn++n8dBGmOMd1ztnEdE+onIOhFJEpH7ink/RkSm+d5fIiLxbsbDb9MB0HaDeeedlXTsOIGPPlrN3XfPQVVd3bUxxgQD15KCiEQCrwIXAx2Bq0WkY5HVRgCpqnoC8DzwpFvxOKWjT9mcWov+Dzdk2LBP2Ls3g7592/LJJ0PsiWRjjMHd8lF3IElVkwFE5ENgELDGb51BwCO+6enAKyIi6sLX9vxNc5n4dXvGzu7Locxt1KkTy/PPX8SwYZ0tIRhjjI+bSaEZsMVvPgU442jrqGquiOwH6gG7yzuY/Stm8Ojc8ziUGc0VV5zEK69cQuPG1ct7N8YYE9TcTArFff0uegUQyDqIyChgFEDLlsc2jnKdatlMvmoW2ee8wBXD+xzTNowxJtS5mRRSgBZ+882BP46yToqIRAG1gL1FN6Sqk4BJAAkJCcdWWrrkXQb0OgAxNY/p48YYEw7cbH20FGgnIq1FpAowBEgssk4icL1vejDwtRv3EwpZQjDGmBK5dqXgu0cwGpgDRAJTVXW1iDwGLFPVRGAK8I6IJOFcIQxxKx5jjDGlc/XhNVWdBcwqsuxhv+lM4NgHSDbGGFOubGR5Y4wxhSwpGGOMKWRJwRhjTCFLCsYYYwpZUjDGGFNIgq13UBHZBfx+jB+vjwtdaFRydszhwY45PBzPMbdS1QalrRR0SeF4iMgyVU3wOo6KZMccHuyYw0NFHLOVj4wxxhSypGCMMaZQuCWFSV4H4AE75vBgxxweXD/msLqnYIwxpmThdqVgjDGmBCGZFESkn4isE5EkEbmvmPdjRGSa7/0lIhJf8VGWrwCO+e8iskZEfhaRr0SklRdxlqfSjtlvvcEioiIS9C1VAjlmEbnS92+9WkTer+gYy1sA/7dbisg8EVnh+/99iRdxlhcRmSoiO0Xkl6O8LyLyku/38bOIdC3XAFQ1pH5wuuneALQBqgArgY5F1rkNmOibHgJM8zruCjjmC4A43/St4XDMvvVqAAuAxUCC13FXwL9zO2AFUMc339DruCvgmCcBt/qmOwKbvI77OI+5J9AV+OUo718CzMYZufJMYEl57j8UrxS6A0mqmqyq2cCHwKAi6wwC3vJNTwd6iUhxQ4MGi1KPWVXnqWq6b3Yxzkh4wSyQf2eA/wOeAjIrMjiXBHLMI4FXVTUVQFV3VnCM5S2QY1agYAStWvx5hMegoqoLKGYESj+DgLfVsRioLSJNymv/oZgUmgFb/OZTfMuKXUdVc4H9QL0Kic4dgRyzvxE43zSCWanHLCKnAS1U9bOKDMxFgfw7twfai8hCEVksIv0qLDp3BHLMjwDXikgKzvgtd1RMaJ4p6997mbg6yI5HivvGX7SJVSDrBJOAj0dErgUSgPNcjch9JR6ziEQAzwPDKyqgChDIv3MUTgnpfJyrwW9FpJOq7nM5NrcEcsxXA2+q6rMichbOaI6dVDXf/fA84er5KxSvFFKAFn7zzfnz5WThOiIShXPJWdLlWmUXyDEjIr2BB4CBqppVQbG5pbRjrgF0AuaLyCac2mtikN9sDvT/9qeqmqOqG4F1OEkiWAVyzCOAjwBUdREQi9NHUKgK6O/9WIViUlgKtBOR1iJSBedGcmKRdRKB633Tg4Gv1XcHJ0iVesy+UsrrOAkh2OvMUMoxq+p+Va2vqvGqGo9zH2Wgqi7zJtxyEcj/7U9wGhUgIvVxyknJFRpl+QrkmDcDvQBE5CScpLCrQqOsWInAMF8rpDOB/aq6rbw2HnLlI1XNFZHRwByclgtTVXW1iDwGLFPVRGAKziVmEs4VwhDvIj5+AR7z00B14D++e+qbVXWgZ0EfpwCPOaQEeMxzgL4isgbIA+5V1T3eRX18Ajzme4A3RORunDLK8GD+kiciH+CU/+r77pOMA6IBVHUizn2TS4AkIB24oVz3H8S/O2OMMeUsFMtHxhhjjpElBWOMMYUsKRhjjClkScEYY0whSwrGGGMKWVIwlY6I5InIT34/8SWsG3+03iTLuM/5vp44V/q6iOhwDNu4RUSG+aaHi0hTv/cmi0jHco5zqYh0CeAzfxORuOPdtwkPlhRMZZShql38fjZV0H6HqmpnnM4Sny7rh1V1oqq+7ZsdDjT1e+8mVV1TLlEejnMCgcX5N8CSggmIJQUTFHxXBN+KyHLfT49i1jlZRH7wXV38LCLtfMuv9Vv+uohElrK7BcAJvs/28vXTv8rXz32Mb/l4OTw+xTO+ZY+IyBgRGYzTv9R7vn1W9X3DTxCRW0XkKb+Yh4vIy8cY5yL8OkITkddEZJk44yg86lt2J05ymici83zL+orIIt/v8T8iUr2U/ZgwYknBVEZV/UpHH/uW7QT6qGpX4CrgpWI+dwvwoqp2wTkpp/i6PbgKONu3PA8YWsr+BwCrRCQWeBO4SlVPwekB4FYRqQv8BThZVU8FHvf/sKpOB5bhfKPvoqoZfm9PBy73m78KmHaMcfbD6daiwAOqmgCcCpwnIqeq6ks4/eJcoKoX+Lq+eBDo7ftdLgP+Xsp+TBgJuW4uTEjI8J0Y/UUDr/hq6Hk4ffoUtQh4QESaAzNUdb2I9AK6AUt93XtUxUkwxXlPRDKATTjdL3cANqrqb7733wJuB17BGZ9hsoh8DgTcNbeq7hKRZF+fNet9+1jo225Z4qyG0+2D/6hbV4rIKJy/6yY4A878XOSzZ/qWL/TtpwrO780YwJKCCR53AzuAzjhXuH8aNEdV3xeRJUB/YI6I3ITTzfBbqnp/APsY6t9hnogUO8aGrz+e7jidsA0BRgMXluFYpgFXAr8CH6uqinOGDjhOnBHIxgOvApeLSGtgDHC6qqaKyJs4HcMVJcBcVb26DPGaMGLlIxMsagHbfH3kX4fzLfkIItIGSPaVTBJxyihfAYNFpKFvnboS+PjUvwLxInKCb/464BtfDb6Wqs7CuYlbXAuggzjddxdnBnAZzjgA03zLyhSnqubglIHO9JWeagJpwH4RaQRcfJRYFgNnFxyTiMSJSHFXXSZMWVIwwWICcL2ILMYpHaUVs85VwC8i8hNwIs6QhWtwTp5fisjPwFyc0kqpVDUTpwfK/4jIKiAfmIhzgv3Mt71vcK5iinoTmFhwo7nIdlOBNUArVf3Bt6zMcfruVTwLjFHVlThjM68GpuKUpApMAmaLyDxV3YXTMuoD334W4/yujAGsl1RjjDF+7ErBGGNMIUsKxhhjCllSMMYYU8iSgjHGmEKWFIwxxhSypGCMMaaQJQVjjDGFLCkYY4wp9P8BoBBQVzFoMrUAAAAASUVORK5CYII=\n",
        "text/plain": "<Figure size 432x288 with 1 Axes>"
       },
       "metadata": {
        "needs_background": "light"
       },
       "output_type": "display_data"
      }
     ],
     "persistentId": "8f75a628-0448-474b-8eba-ccfae9ffc601",
     "text": "fpr, tpr, thresholds = metrics.roc_curve(y_true, y_pred)\n\nfig, ax = plt.subplots()\nlw = 2\nax.plot(fpr, tpr, lw=2, color='darkorange')\nax.plot([0, 1], [0, 1], color='navy', lw=lw, linestyle='--')\nax.set(xlabel='False Positive Rate', ylabel='True Positive Rate')"
    },
    "executionTime": "2019-10-11T20:09:00.074Z"
   },
   {
    "cell": {
     "executionCount": 114,
     "executionEventId": "4f695dce-568d-4362-b5e1-d15b5f694b05",
     "hasError": true,
     "id": "1e39d7f6-23b4-4c86-b1a9-894c2d9a8a53",
     "outputs": [
      {
       "ename": "ValueError",
       "evalue": "Classification metrics can't handle a mix of binary and continuous targets",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mValueError\u001b[0m                                Traceback (most recent call last)",
        "\u001b[0;32m<ipython-input-114-50dd0de8d61e>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mmetrics\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mclassification_report\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0my_true\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0my_pred\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/sklearn/metrics/classification.py\u001b[0m in \u001b[0;36mclassification_report\u001b[0;34m(y_true, y_pred, labels, target_names, sample_weight, digits, output_dict)\u001b[0m\n\u001b[1;32m   1522\u001b[0m     \"\"\"\n\u001b[1;32m   1523\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1524\u001b[0;31m     \u001b[0my_type\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0my_true\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0my_pred\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0m_check_targets\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0my_true\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0my_pred\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1525\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1526\u001b[0m     \u001b[0mlabels_given\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mTrue\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/sklearn/metrics/classification.py\u001b[0m in \u001b[0;36m_check_targets\u001b[0;34m(y_true, y_pred)\u001b[0m\n\u001b[1;32m     79\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0my_type\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m>\u001b[0m \u001b[0;36m1\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     80\u001b[0m         raise ValueError(\"Classification metrics can't handle a mix of {0} \"\n\u001b[0;32m---> 81\u001b[0;31m                          \"and {1} targets\".format(type_true, type_pred))\n\u001b[0m\u001b[1;32m     82\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     83\u001b[0m     \u001b[0;31m# We can't have more than one value on y_type => The set is no more needed\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;31mValueError\u001b[0m: Classification metrics can't handle a mix of binary and continuous targets"
       ]
      }
     ],
     "persistentId": "e0cd2f37-754e-4062-92c6-e5a3223e4c7c",
     "text": "metrics.classification_report(y_true, y_pred)"
    },
    "executionTime": "2019-10-11T20:09:13.585Z"
   },
   {
    "cell": {
     "executionCount": 115,
     "executionEventId": "0e39fc43-095a-4072-a18a-5fd0a264a2a8",
     "hasError": false,
     "id": "1e39d7f6-23b4-4c86-b1a9-894c2d9a8a53",
     "outputs": [
      {
       "data": {
        "text/plain": "'              precision    recall  f1-score   support\\n\\n       False       0.87      0.84      0.85       318\\n        True       0.70      0.74      0.72       156\\n\\n   micro avg       0.81      0.81      0.81       474\\n   macro avg       0.78      0.79      0.79       474\\nweighted avg       0.81      0.81      0.81       474\\n'"
       },
       "execution_count": 115,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "e0cd2f37-754e-4062-92c6-e5a3223e4c7c",
     "text": "metrics.classification_report(y_true, y_pred > 0.9)"
    },
    "executionTime": "2019-10-11T20:09:17.860Z"
   },
   {
    "cell": {
     "executionCount": 116,
     "executionEventId": "bd22f846-0a5e-4c00-9db0-80e704447081",
     "hasError": false,
     "id": "1e39d7f6-23b4-4c86-b1a9-894c2d9a8a53",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "              precision    recall  f1-score   support\n\n       False       0.87      0.84      0.85       318\n        True       0.70      0.74      0.72       156\n\n   micro avg       0.81      0.81      0.81       474\n   macro avg       0.78      0.79      0.79       474\nweighted avg       0.81      0.81      0.81       474\n\n"
      }
     ],
     "persistentId": "e0cd2f37-754e-4062-92c6-e5a3223e4c7c",
     "text": "print(metrics.classification_report(y_true, y_pred > 0.9))"
    },
    "executionTime": "2019-10-11T20:09:21.867Z"
   },
   {
    "cell": {
     "executionCount": 117,
     "executionEventId": "baaf5d37-eedf-4643-8b09-9bcf06fbca44",
     "hasError": false,
     "id": "c644f8a4-dc14-489f-80fb-21d9a39c084f",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "0.8080168776371308\n"
      }
     ],
     "persistentId": "91fecc99-2e19-42e2-9fb8-a694ea075d8f",
     "text": "print(metrics.accuracy_score(y_true, y_pred > 0.9))"
    },
    "executionTime": "2019-10-11T20:09:30.871Z"
   },
   {
    "cell": {
     "executionCount": 1,
     "executionEventId": "bf3f376d-47ee-418f-8e28-fdaf3a35df71",
     "hasError": true,
     "id": "efba9e80-c61d-4adb-9d6c-af7db00a9356",
     "outputs": [
      {
       "ename": "NameError",
       "evalue": "name 'coding_scores_unspliced' is not defined",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
        "\u001b[0;32m<ipython-input-1-2479b90a0a3b>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mcoding_scores_unspliced\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mquery\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"jaccard_in_peptide_db < 0.5 & in_cds_no_stop_codon == True\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mhead\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
        "\u001b[0;31mNameError\u001b[0m: name 'coding_scores_unspliced' is not defined"
       ]
      }
     ],
     "persistentId": "1254868e-73a0-4c0a-82c6-bb1b85e78cb0",
     "text": "coding_scores_unspliced.query(\"jaccard_in_peptide_db < 0.5 & in_cds_no_stop_codon == True\").head()"
    },
    "executionTime": "2019-10-11T20:50:43.776Z"
   },
   {
    "cell": {
     "executionCount": 2,
     "executionEventId": "68fb4196-243c-4b7f-87a2-8850e676b973",
     "hasError": false,
     "id": "0f1ea26a-ecfa-4359-8ba4-c0f0d6feb077",
     "outputs": [],
     "persistentId": "71299be4-a00f-4638-92a2-ea174ddf208d",
     "text": "%load_ext autoreload\n%autoreload 2"
    },
    "executionTime": "2019-10-11T20:50:48.541Z"
   },
   {
    "cell": {
     "executionCount": 3,
     "executionEventId": "568c566f-4543-4557-b40f-4b7162dda990",
     "hasError": false,
     "id": "5e35ad66-c7bb-43f2-86d3-6da92b22e4a6",
     "outputs": [],
     "persistentId": "ecc17310-b85a-49bd-8d60-8de454040c04",
     "text": "folder = '/home/olga/data_sm/kmer-hashing/classify_coding_vs_noncoding/'"
    },
    "executionTime": "2019-10-11T20:50:48.785Z"
   },
   {
    "cell": {
     "executionCount": 4,
     "executionEventId": "82527097-753b-4a0e-8e25-44698293bf38",
     "hasError": false,
     "id": "31c3fe01-1037-44d7-b566-b186d79dff80",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "/mnt/data_sm/olga/kmer-hashing/classify_coding_vs_noncoding\n"
      }
     ],
     "persistentId": "1d547cf2-254f-4fe2-9bf0-5b7e39be4376",
     "text": "cd $folder"
    },
    "executionTime": "2019-10-11T20:50:48.990Z"
   },
   {
    "cell": {
     "executionCount": 5,
     "executionEventId": "e0ddfed4-6b82-4e03-87f2-9ecac3190352",
     "hasError": false,
     "id": "342df663-948c-4de9-b395-be7b619e90de",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "\u001b[0m\u001b[01;31mHomo_sapiens.GRCh38.cds.all.fa.gz\u001b[0m\n\u001b[01;31mHomo_sapiens.GRCh38.ncrna.fa.gz\u001b[0m\n\u001b[01;31mHomo_sapiens.GRCh38.pep.all.fa.gz\u001b[0m\nhuman_cds_k15.nodegraph\nhuman_cds_k15.nodegraph.info\nhuman_cds_k17.nodegraph\nhuman_cds_k17.nodegraph.info\nhuman_cds_k19.nodegraph\nhuman_cds_k19.nodegraph.info\nhuman_cds_k21.nodegraph\nhuman_cds_k21.nodegraph.info\nhuman_cds_k23.nodegraph\nhuman_cds_k23.nodegraph.info\nhuman_cds_k25.nodegraph\nhuman_cds_k25.nodegraph.info\nhuman_cds_k27.nodegraph\nhuman_cds_k27.nodegraph.info\nhuman_cds_k31.nodegraph\nhuman_cds_k31.nodegraph.info\nhuman_ncrna_k15.nodegraph\nhuman_ncrna_k15.nodegraph.info\nhuman_ncrna_k15.nodegraph.tagset\nhuman_ncrna_k17.nodegraph\nhuman_ncrna_k17.nodegraph.info\nhuman_ncrna_k19.nodegraph\nhuman_ncrna_k19.nodegraph.info\nhuman_ncrna_k21.nodegraph\nhuman_ncrna_k21.nodegraph.info\nhuman_ncrna_k23.nodegraph\nhuman_ncrna_k23.nodegraph.info\nhuman_ncrna_k25.nodegraph\nhuman_ncrna_k25.nodegraph.info\nhuman_ncrna_k27.nodegraph\nhuman_ncrna_k27.nodegraph.info\nhuman_ncrna_k31.nodegraph\nhuman_ncrna_k31.nodegraph.info\n\u001b[01;31mSRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz\u001b[0m\n\u001b[01;31mSRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.fq.gz\u001b[0m\n"
      }
     ],
     "persistentId": "a30b0ea1-cbfb-4de5-bce5-329cfeb8e50c",
     "text": "ls"
    },
    "executionTime": "2019-10-11T20:50:49.195Z"
   },
   {
    "cell": {
     "executionCount": 6,
     "executionEventId": "bfe909b0-55bf-4948-80c6-efd4d87dd29c",
     "hasError": false,
     "id": "81fe1d1b-9150-4e3a-a0e2-f2470c0bb85c",
     "outputs": [],
     "persistentId": "56e48bd4-fcc6-4d93-b4e3-0984be1fc1d5",
     "text": "import gzip\nfrom io import StringIO\nfrom pprint import pprint\n\nfrom khmer import Nodegraph\nimport matplotlib.pyplot as plt\nimport screed\nfrom sourmash._minhash import hash_murmur\nfrom khmer.khmer_args import calculate_graphsize\nimport pandas as pd\nfrom sourmash.logging import notify\nimport seaborn as sns\nfrom sklearn import metrics\nfrom tqdm import tqdm\nfrom Bio.Seq import Seq\nfrom Bio import SeqIO\n\n# make all warnings error to get a stack trace\nimport warnings\nwarnings.filterwarnings(\"default\")\n\npd.options.display.max_colwidth = 100\n\nfrom khtools.compare_kmer_content import kmerize, hpize, dayhoffize\n"
    },
    "executionTime": "2019-10-11T20:50:50.906Z"
   },
   {
    "cell": {
     "executionCount": 7,
     "executionEventId": "fe250513-36d6-4a7c-b136-fe7bbd399be7",
     "hasError": false,
     "id": "e50808b6-6666-4543-9299-8075c789791c",
     "outputs": [],
     "persistentId": "1e54230e-ceec-42fc-afa6-7b32e219378b",
     "text": "from khtools.partition_reads import score_reads, make_peptide_bloom_filter, six_frame_translation_no_stops, is_low_complexity"
    },
    "executionTime": "2019-10-11T20:50:50.936Z"
   },
   {
    "cell": {
     "executionCount": 9,
     "executionEventId": "83e641b5-525d-49b7-afae-cc1c9644b695",
     "hasError": false,
     "id": "b242d89f-8d25-4127-b875-8a85fa69d914",
     "outputs": [
      {
       "name": "stderr",
       "output_type": "stream",
       "text": "/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/screed/openscreed.py:39: ResourceWarning: unclosed file <_io.BufferedReader name='SRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.fq.gz'>\n  self.iter_fn = self.open_reader(filename, *args, **kwargs)\n1000it [00:00, 2526.88it/s]"
      },
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "CPU times: user 408 ms, sys: 0 ns, total: 408 ms\nWall time: 405 ms\n"
      },
      {
       "name": "stderr",
       "output_type": "stream",
       "text": "\n"
      }
     ],
     "persistentId": "be06ded1-d7d4-4f21-9961-3aed5ff30cbf",
     "text": "%%time\n\nreads = 'SRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz'\nsubsampled_reads = 'SRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.fq.gz'\n\ncoding_scores = score_reads(subsampled_reads, peptide_graph, peptide_ksize)"
    },
    "executionTime": "2019-10-11T20:51:30.047Z"
   },
   {
    "cell": {
     "executionCount": 10,
     "executionEventId": "08bad214-00e6-478c-9748-b2ca09d645c2",
     "hasError": false,
     "id": "cfea0c3d-642a-421a-a4ad-07fa69bea946",
     "outputs": [
      {
       "data": {
        "text/plain": "(1000, 4)"
       },
       "execution_count": 10,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "09393220-d1f9-4ddd-9c7c-bc15160f3680",
     "text": "coding_scores.shape"
    },
    "executionTime": "2019-10-11T20:51:30.086Z"
   },
   {
    "cell": {
     "executionCount": 11,
     "executionEventId": "d33016c8-5b6d-4d33-946d-694cdb63b6b6",
     "hasError": false,
     "id": "3281c496-16a4-496e-81d4-fa57e8d1a556",
     "outputs": [
      {
       "data": {
        "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>read_id</th>\n      <th>jaccard_in_peptide_db</th>\n      <th>n_kmers</th>\n      <th>classification</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>SRR306838.10559374 Ibis_Run100924_C3PO:6:51:17601:17119/1</td>\n      <td>1.000000</td>\n      <td>16</td>\n      <td>coding</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>SRR306838.6196593 Ibis_Run100924_C3PO:6:29:16733:12435/1</td>\n      <td>0.000000</td>\n      <td>17</td>\n      <td>non-coding</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>SRR306838.20767303 Ibis_Run100924_C3PO:6:104:6864:5062/1</td>\n      <td>0.187500</td>\n      <td>16</td>\n      <td>non-coding</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>SRR306838.12582274 Ibis_Run100924_C3PO:6:62:11779:17975/1</td>\n      <td>0.066667</td>\n      <td>15</td>\n      <td>non-coding</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>SRR306838.13334230 Ibis_Run100924_C3PO:6:66:16579:20350/1</td>\n      <td>0.000000</td>\n      <td>12</td>\n      <td>non-coding</td>\n    </tr>\n  </tbody>\n</table>\n</div>",
        "text/plain": "                                                     read_id  \\\n0  SRR306838.10559374 Ibis_Run100924_C3PO:6:51:17601:17119/1   \n1   SRR306838.6196593 Ibis_Run100924_C3PO:6:29:16733:12435/1   \n2   SRR306838.20767303 Ibis_Run100924_C3PO:6:104:6864:5062/1   \n3  SRR306838.12582274 Ibis_Run100924_C3PO:6:62:11779:17975/1   \n4  SRR306838.13334230 Ibis_Run100924_C3PO:6:66:16579:20350/1   \n\n   jaccard_in_peptide_db  n_kmers classification  \n0               1.000000       16         coding  \n1               0.000000       17     non-coding  \n2               0.187500       16     non-coding  \n3               0.066667       15     non-coding  \n4               0.000000       12     non-coding  "
       },
       "execution_count": 11,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "ede6e1ba-f524-4ceb-b35b-6c20f91d437c",
     "text": "coding_scores.head()"
    },
    "executionTime": "2019-10-11T20:51:30.160Z"
   },
   {
    "cell": {
     "executionCount": 13,
     "executionEventId": "56c5732a-8bdb-4dc5-8c77-0a6f49d74922",
     "hasError": false,
     "id": "4c484434-ea07-4b72-8073-567bf902bedb",
     "outputs": [
      {
       "data": {
        "text/plain": "150289"
       },
       "execution_count": 13,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "75c01782-427c-4c63-bc29-ed308ae4a142",
     "text": "true_coding_read_ids = set(coding_read_ids.values)\nlen(true_coding_read_ids)"
    },
    "executionTime": "2019-10-11T20:51:30.758Z"
   },
   {
    "cell": {
     "executionCount": 14,
     "executionEventId": "828315c6-6810-4581-8303-2e22e9e37ac9",
     "hasError": false,
     "id": "9babb580-ed7e-4a80-bcb3-3d080fca4a29",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "(807698,)\n"
      }
     ],
     "persistentId": "317bb484-c93f-4e1c-b831-50734c9e6614",
     "text": "unspliced_read_ids_series = pd.read_csv(f\"{folder}/SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.read_ids.txt\", \n                                   header=None, squeeze=True)\nprint(unspliced_read_ids_series.shape)\nunspliced_read_ids_series.head()\n\nunspliced_reads = set(unspliced_read_ids_series.values)"
    },
    "executionTime": "2019-10-11T20:51:31.457Z"
   },
   {
    "cell": {
     "executionCount": 17,
     "executionEventId": "5874cb37-c7d0-4cbe-92d9-aa3deeec1264",
     "hasError": false,
     "id": "a18fd62e-900f-4a7d-8ada-1b3b45c4d677",
     "outputs": [],
     "persistentId": "27025afb-bac7-4e3d-822e-f3238fba3a50",
     "text": "y_true = coding_scores_unspliced_not_low_complexity['in_cds_no_stop_codon']\ny_pred = coding_scores_unspliced_not_low_complexity['jaccard_in_peptide_db']"
    },
    "executionTime": "2019-10-11T20:51:32.031Z"
   },
   {
    "cell": {
     "executionCount": 18,
     "executionEventId": "2ec6e57a-4a4e-4bb3-a2c2-4a32b3b4ffd3",
     "hasError": false,
     "id": "3f61b436-51d7-4dae-a329-c238ba62d3d5",
     "outputs": [
      {
       "data": {
        "text/plain": "[Text(0, 0.5, 'True Positive Rate'), Text(0.5, 0, 'False Positive Rate')]"
       },
       "execution_count": 18,
       "metadata": {},
       "output_type": "execute_result"
      },
      {
       "data": {
        "image/png": "iVBORw0KGgoAAAANSUhEUgAAAYUAAAEKCAYAAAD9xUlFAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDMuMC4zLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvnQurowAAIABJREFUeJzt3Xl4FFX28PHvyUJC2PcdAggooiBEVFRcWEQRcJRRFEUUwQ11HHHQ1wX15zi474gIjLviMKhBQWQURBEQBBEBkRAQguyELfty3j+qE5oYkg6kUunu83mePF1VXV11KpA6Xadu3SuqijHGGAMQ4XUAxhhjKg9LCsYYYwpZUjDGGFPIkoIxxphClhSMMcYUsqRgjDGmkCUFY4wxhSwpGGOMKWRJwRhjTKEorwMoq/r162t8fLzXYRhjTFD58ccfd6tqg9LWC7qkEB8fz7Jly7wOwxhjgoqI/B7IelY+MsYYU8iSgjHGmEKWFIwxxhSypGCMMaaQJQVjjDGFXEsKIjJVRHaKyC9HeV9E5CURSRKRn0Wkq1uxGGOMCYybVwpvAv1KeP9ioJ3vZxTwmouxGGOMCYBrzymo6gIRiS9hlUHA2+qMB7pYRGqLSBNV3eZWTMYYUyHysiFrP2Tt873uh+z9h6eLm8/LOurmfkiqSWx0Pqee2x16POJq6F4+vNYM2OI3n+Jb9qekICKjcK4maNmyZYUEZ4wJU4Un9KIn7gBP8Nn7ITezXEJRhbGf9+HZb7pxapMd/NDhV6LLZctH52VSkGKWaXErquokYBJAQkJCsesYYwwAeTmQsQvSd0Lm3qOfyAtO8m6c0COioEotiPH7KW0+MhbkyNOiAGxIggVb6DvoDPJOPz2kk0IK0MJvvjnwh0exGGMqK1XIPuCc5NN3QvoOv+lilmXuPb79HcsJvUotiKl9eD6q6p9O8IHaty+T5ORUunZtAsCjz53FkJG7C+fd5mVSSARGi8iHwBnAfrufYEyYKPg2n7YDMko4wRf8lFBv/xOJgKoNIK4hVK1XhhN87eM+oR+vTz/9lVtv/ZyICGH16tuoVSuWqlWjKywhgItJQUQ+AM4H6otICjAOnCsfVZ0IzAIuAZKAdOAGt2IxxrhM1Sm/FJzEC070aTv+PJ+xEzJTy7b96OrOST6uke+14dHnq9Z1EkMQ2bkzjTvvnM20aasBOPPM5uzbl0mtWrEVHoubrY+uLuV9BW53a//GmOOUlw3puw6f1NN2HHnSLzqflx34tiUS4gq+zftO6NUaHZ4umI9r6Hzrj45z7zg9pKq8994q7rrrC/buzSAuLponnriQ0aO7ExnpTWILuq6zjTHHKD/PubmasbvkmnzBfNa+sm2/So3DJ/lqft/gqxY9yTcMym/zbrj11s95/fUfAejduw2TJl1K69Z1PI3JkoIxwSgvBzL3QMYe5yTvP52xxzdfZDozlaM08Cte4bf5IiWaoif9gmXRVV073FB12WUnMm3aap59ti833NAF8ehehj9LCsZ4LSej+JN4cSf8gunsA8e2r9g6EFuv9Lp8XENnXfs2X67Wr9/DV19t5JZbEgDo1+8ENm26y5N7B0djScGY8pZ9qEhJZoffDdddfz7552aUfR8S4Zzcq9bzvdZ3pqvWP7z8T9N1nOaWpsLl5ubz3HOLGDduPllZuXTp0pgzz2wOUKkSAlhSMKZkO5bD3l+PXKb5TinG/6RfMJ22A3LTy7aPiOgAT+p+J/+YWvYtPkisXLmdESMS+fFHp8X9sGGdadeursdRHZ0lBWOKys2EtG2w5F+w6o2yfz4yxinBFNbd/V8b/PkEH13ds3bxxj1ZWbk8/vgCxo9fSG5uPi1b1uL11y+lX78TvA6tRJYUTHhRdVrVHPgdDmz2vf4OB/3m03ccXj8iGtoOhMgqR24nps6RLWr8T/xVathJ3nD//V/x/POLAbj99tP51796UaNGjMdRlc6SggltORmQ8g1snO287k+G7IMlfyYiynm6tX4nuPBl59WYMvrHP85m0aIUnnqqN+ee28rrcAJmScGEFlXYl+QkgY2zIWX+nzs4i64GNVsd/qnR8sj5ak0gItKT8E3wmjt3AxMn/si0aYOJioqgcePqfP/9jZWimWlZWFIwwS8nDbbMP5wI9icf+X7DrtD6YojvC/VOhti6Vt4x5SY1NYMxY75k6tSfAPj3v1cwcmQ3gKBLCGBJwVQmmu/8BCJ1PWz6wnc1sODIDtNi60Krvr5EcJFT9zfGBR9/vJbbbpvF9u2HiImJZNy48xg+vIvXYR0XSwqm4uVmOs0896yBPat9r2ucsk+gSeEIAo27Q3w/JxE0Pt3KP8ZV27cf4o47ZjN9+hoAevRowZQpAznxxPoeR3b8LCmEo01zYemTkJ9bwTtWOPSHU9452slfAjyZV63nuxro57zGNSi/MI0pxaef/sr06WuoVi2a8eN7c9ttpxMREXylouJYUghHK16GzV95t3+JhDodoF5HqH8y1O3oTNftAFGV6+lOYwpkZuYSG+ucMkeO7EZyciq33no68fG1PY6sfFlSCAd71x3Z42XGbuc1YQy06V+xscTWgzrtIaryt9c2BiA/X5kwYSn//Oe3LF48glatahMRITz5ZB+vQ3OFJYVQ992DsOSfxb/X8kJocX6FhmNMMFm3bjcjRiSycOEWAD744Bfuu+8cj6NylyWFULbhMychSCQ06opvGHBH9abQLLT/cxtzrHJy8njmme959NFvyMrKo1GjakyY0J/LLz/J69BcZ0khVB34Hb4Y5kyf80/oPtbbeIwJEr/8spNhwz5mxYrtANxwQxeefbYvdeqEx3gRlhRCUV42fHaV05Nnm/5w+r1eR2RM0MjPV1at2kmrVrWYNGkAffu29TqkCmVJIRQtGAvblkCNFtDvLeti2ZhSrF69k44dGyAinHpqIz79dAg9e7aievUqpX84xNjZItTsWA7LX3A6dbv0I6c9vzGmWAcPZjF69Cw6dXqN//53beHySy5pF5YJAexKIfSk/ua8tr4Emp7pbSzGVGJz5iQxatRnbN68n6ioCDZt2lf6h8KAJYVQkpsJix93puue6G0sxlRSe/dmcPfdc3j77ZUAdO3ahClTBtKlS2OPI6scLCmEkj++d/oSqhkPZz7kdTTGVDo//bSdfv3eZceONGJiInn00fO5554eREVZJb2AJYVQUvCkcr2ToEp1b2MxphJq374e1atXoX37ekyePJD27e2eW1GWFELJuo+c12Y9vY3DmEpCVXn//VUMGNCBmjVjiIuLZv784TRtWiNkOrArb3bNFCoy9sCGRKf56cnDvI7GGM9t2rSPiy56l2uv/Zj77vtf4fLmzWtaQiiBXSmEij1rID/HGUugelOvozHGM3l5+UyYsJT77/+KtLQc6tatSo8eLbwOK2hYUgg1kdb7qAlfa9fuYsSIRBYtSgHgyitP5uWXL6Zhw2oeRxY8LCkYY0LCxo2pdOnyOtnZeTRpUp0JE/pz2WXWNLusXE0KItIPeBGIBCar6vgi77cE3gJq+9a5T1VnuRmTMSY0tW5dh7/+tSOxsVE880xfate2AZuOhWtJQUQigVeBPkAKsFREElV1jd9qDwIfqeprItIRmAXEuxWTMSZ0ZGTk8Nhj3/CXv5xE9+7NAHjrrcuIjLT2M8fDzd9edyBJVZNVNRv4EBhUZB0FavqmawF/uBiPMSZEfPvt73Tp8jrjxy9k1KiZ5OcrgCWEcuBm+agZsMVvPgU4o8g6jwBfisgdQDWgt4vxhLYt853XqPDo892EpwMHsrj//v8xYcIyADp2bMDEiZdaE9Ny5GZaLe5fSYvMXw28qarNgUuAd0T+3M+ziIwSkWUismzXrl0uhBrkDm2DRY84093u9jQUY9wya9Z6OnWawIQJy4iKiuDhh3uyfPkozjyzudehhRQ3rxRSAP/Gwc35c3loBNAPQFUXiUgsUB/Y6b+Sqk4CJgEkJCQUTSwmbRtoPjToDK0v9joaY8rd/v2ZDB06g337MklIaMqUKQM59dRGXocVktxMCkuBdiLSGtgKDAGuKbLOZqAX8KaInATEAnYpUBaq8P3DzrQNpmNCiKqiChERQq1asbz0Uj927Ejjb3870zqwc5FrSUFVc0VkNDAHp7npVFVdLSKPActUNRG4B3hDRO7GKS0NV1W7EiiL/Rsh+XNnupp1/WtCwx9/HOS22z7n3HNbcs89PQC47rrOHkcVHlx9TsH3zMGsIsse9pteA5ztZgwhLyv18PTF73gXhzHlQFWZOnUF99zzJfv3Z7F4cQq33XY6VatGex1a2LAnmoNZxl54N8GZrtPBht40QS05OZWRI2fy9dcbAejfvx0TJ15qCaGCWVIIZr++f3j61FHexWHMccjLy+ell5bwwANfk5GRS/36cbz0Uj+GDOmEiDU1rWiWFILRuv/A71/Cpi+d+f4fwIlDvI3JmOMwffpaMjJyufrqTrz4Yj8aNLAO7LxiSSFYpG2HlAWwLxm+u//w8th6cMJl3sVlzDHIzs7j4MEs6tWLIzIygilTBrJ+/R4GDOjgdWhhz5JCsJj5V9j63eH50+6A+qdA0x4QZR1/meCxdOlWRoxIpHnzmnz++TWICCeeWJ8TT6zvdWgGSwrBI32H89r6YmjVF7reBVZvNUEkPT2HcePm8dxzi8nPV9LTc9i5M41GjWw88crEkkKwOf8FqNve6yiMKZP58zcxcuRMkpL2EhEhjBlzFo8+egFxcdayqLKxpGCMcY2qcueds3nllaUAnHJKQ6ZMGcjppzfzODJzNJYUgkF+HmTscaajbLhNEzxEhJo1Y4iOjuDBB3ty333nUKVKpNdhmRJYUggGKQsgcy/Ubgs1WnodjTEl2r07nQ0b9nLGGU7vpQ89dB5Dh55Kx44NPI7MBCKgXqVEpIqInOB2MOYofpvuvLb/q91cNpWWqvLhh79w0kmvctll00hNzQAgNjbKEkIQKTUpiEh/YBUw1zffRUQ+djsw45OfB+v/60y3/6u3sRhzFCkpBxg06EOuvvq/7N6dTseODUhPz/E6LHMMAikfPYYzYto8AFX9ya4aKtDW75zmqLXaQMPTvI7GmCPk5yuTJy/n3nvncuBAFjVrxvDss30ZMeI066IiSAWSFHJUdV+Rf2Dr3rqi/PYf59VKR6YSGjEikTff/AmAgQM7MGHCJTRrVrOUT5nKLJB7CmtF5EogQkRai8gLwGKX4zJQpHQ02NtYjCnGtdeeQsOG1fjwwyv45JOrLCGEgECSwmigG5APzAAygbvcDMr4/PG90+dRzXho1M3raIzhl1928uKLh78T9urVhuTkO7nqKuvRNFQEUj66SFXHAmMLFojI5TgJwrjJSkemksjKyuVf//qOJ574lpycfBISmnL22U7z6GrVqngcnSlPgVwpPFjMsgfKOxBThOYfLh11sFZHxjtLlqTQrdskHn30G3Jy8rn11gROOaWR12EZlxz1SkFELgL6Ac1E5Dm/t2rilJKMm7Z+D4f+gJqtoFGC19GYMJSWls1DD83jhRcWowrt2tVl8uSB9OzZyuvQjItKKh/tBH7BuYew2m/5QeA+N4MyHC4dtRtspSPjiQce+JoXX1xCRIRw771n8cgj59vQmGHgqElBVVcAK0TkPVXNrMCYjJWOTCXwwAPnsmrVTp58sjcJCU29DsdUkEDuKTQTkQ9F5GcR+a3gx/XIwtkfi+HQVqefo8bdvY7GhInExHVccsl75OTkAdCgQTW++mqYJYQwE0hSeBP4NyDAxcBHwIcuxmQKWx1Z6ci4b+fONIYMmc6gQR8ye3YSb7210uuQjIcCSQpxqjoHQFU3qOqDwAXuhhXGNP/IDvCMcYmq8u67P3PSSa8ybdpq4uKiefHFftxwQxevQzMeCuQ5hSxxnkrZICK3AFuBhu6GFca2LYFDKVC9OTSx0pFxx+bN+7nlls+YPTsJgN692zBp0qW0bl3H48iM1wJJCncD1YE7gX8CtYAb3QwqrBVeJQwGCahnc2PK7MsvNzB7dhK1a8fy3HN9GT68iz2RbIAAkoKqLvFNHgSuAxCR5m4GFbZUrXRkXJOWll349PGIEaexdesBRo3qRpMmNTyOzFQmJX4VFZHTReQyEanvmz9ZRN7GOsRzx/Yf4OBmqN4Mmp7pdTQmROTm5vPUUwtp1eoFkpNTAWeYzHHjzreEYP7kqElBRP4FvAcMBb4QkQdwxlRYCbSvmPDCzDr/VkdWOjLHb+XK7ZxxxmTGjv0fe/Zk8Mknv3odkqnkSiofDQI6q2qGiNQF/vDNr6uY0MKMKqy30pEpH1lZuTz++ALGj19Ibm4+LVvWYtKkS7noIhsfy5SspKSQqaoZAKq6V0R+tYTgou1L4cDvUL0pND3L62hMEFuxYhtDh85g7drdiMDo0afzxBO9qFEjxuvQTBAoKSm0EZGC7rEFiPebR1UvL23jItIPeBGIBCar6vhi1rkSeARnNLeVqnpN4OGHkIIbzO2usNKROS4xMVFs2JBKhw71mDx5IOec09LrkEwQKSkpXFFk/pWybFhEIoFXgT5ACrBURBJVdY3fOu2A+4GzVTVVRMLz+QfVI8dOMKaMli/fxmmnNUZE6NixAbNnD6VHjxbExgbS6tyYw0rqEO+r49x2dyBJVZMBRORDnPsUa/zWGQm8qqqpvn3uPM59BqcdP8KBTVCtCTQ72+toTBBJTc1gzJgvmTr1Jz744AqGDOkEwIUXtvY4MhOs3Pwa0QzY4jefApxRZJ32ACKyEKfE9IiqflF0QyIyChgF0LJlCF4KF3aTbaUjE7iPP17LbbfNYvv2Q8TERLJnT7rXIZkQ4GZSKO7xSC1m/+2A84HmwLci0klV9x3xIdVJwCSAhISEotsIbkeUjgZ7G4sJCtu3H+KOO2Yzfbpz0X322S2YPHkgJ55Y3+PITCgIOCmISIyqZpVh2ylAC7/55jjNWouus1hVc4CNIrIOJ0ksLcN+gtvOFbB/I8Q1gmbneB2NqeR+/PEP+vR5h9TUTKpVi2b8+N7cdtvpRERYFxWmfJRaqxCR7iKyCljvm+8sIi8HsO2lQDsRaS0iVYAhQGKRdT7B1+Oq76np9kByGeIPfv6lo4hIb2MxlV7Hjg1o0KAaF13UltWrb2P06O6WEEy5CqSA/RJwKbAHQFVXEkDX2aqaC4wG5gBrgY9UdbWIPCYiA32rzQH2iMganKel71XVPWU/jCDlXzqyEdZMMfLzlUmTfmTfPmfww6pVo1mwYDizZw+lVavaHkdnQlEg5aMIVf29SA+KeYFsXFVnAbOKLHvYb1qBv/t+ws/On2DfBohrCM3O9ToaU8msW7ebm26ayXffbWbp0q288YbzXapRo+oeR2ZCWSBJYYuIdAfU9+zBHYANx1keCktHl1vpyBTKycnj2WcX8cgj88nKyqNx4+pcfHE7r8MyYSKQpHArTgmpJbAD+J9vmTke9sCaKcaKFdsYMSKRFSu2A3DDDV149tm+1KlT1ePITLgIJCnkquoQ1yMJN7t+hn1JULUBNO/pdTSmEtiwYS/du08mNzef+PjaTJp0KX36tPU6LBNmAkkKS31NRacBM1T1oMsxhYcjSkfWFYGBtm3rct11p1KjRhX++c9eVK9exeuQTBgqtfWRqrYFHge6AatE5BMRsSuH42GlIwMcOpTNnXfOZtGiww/+T5kykBdfvNgSgvFMQH0qqOr3qnon0BU4gDP4jjlWu1dB6m9QtT60OM/raIwH5sxJ4uSTJ/Dyyz9wyy2f4zTEw8ZJNp4rtW4hItVxOrIbApwEfAr0cDmu0FbQTfYJf7HSUZjZuzeDu++ew9tvrwSgW7cmTJky0JKBqTQCOSP9AswEnlLVb12OJ/RZ6ShsTZ++httvn8XOnWnExkbx6KPn8/e/n0VUlHWCaCqPQJJCG1XNdz2ScLFnNez9FWLrQctSHww3IWLfvkxGjZpJamomPXu24o03BtC+fT2vwzLmT46aFETkWVW9B/iviPypZ9JARl4zxVhX0OrISkehTlXJz1ciIyOoXTuWCRP6k5qawc03J1h/RabSKumsNM33WqYR10wprJvssLBp0z5GjZrJhRe25r77nN5vCwbAMaYyO2oxU1V/8E2epKpf+f/g3HA2ZbVnDexdC7F1ocWFXkdjXJCXl89LLy2hU6cJzJ2bzCuv/EBmZq7XYRkTsEDucN1YzLIR5R1IWCgoHZ1wGURGexuLKXdr1+6iZ883ueuuL0hLy2HIkE4sX36zjZNsgkpJ9xSuwmmG2lpEZvi9VQPYV/ynTIms1VFIys3N58knv+OxxxaQnZ1H06Y1eO21/gwc2MHr0Iwps5K+wvyAM4ZCc+BVv+UHgRVuBhWS9qx1Wh7F1oGWvbyOxpSjiAjhyy+Tyc7OY+TIrjz1VB9q1471OixjjslRk4KqbgQ24vSKao5XwVVC20FWOgoBGRk5HDyYTcOG1YiIECZPHsCWLQe48MLWXodmzHE56j0FEfnG95oqInv9flJFZG/FhRgiCp5ittJR0Fuw4Hc6d57ItdfOKOyeol27epYQTEgoqXxU8GRV/YoIJKTtXef0dxRTG1r19joac4wOHMji/vv/x4QJywCIjo5k9+50GjSo5nFkxpSfkpqkFjzF3AKIVNU84CzgZsD+CsqioHR0wiCItN4vg9Hs2evp1GkCEyYsIyoqgnHjzmP58lGWEEzICaSt3CfA6SLSFngb+Bx4H7jUzcBCirU6ClqqysiRM5kyxWlbkZDQlKlTB3LKKY08jswYdwTynEK+quYAlwMvqOodQDN3wwohe39zRlmrUhNaWuko2IgIzZvXJDY2imee6cOiRSMsIZiQFtBwnCLyV+A64DLfMms+E6j1Bd1kD4KoGG9jMQH544+DbNiwl3PPbQXA//t/53LddafStm1djyMzxn2BPtF8AU7X2cki0hr4wN2wQsg6Kx0FC1VlypTldOz4Kldc8RF79qQDUKVKpCUEEzZKvVJQ1V9E5E7gBBE5EUhS1X+6H1oISE2CXT85paNWfb2OxpQgOTmVkSNn8vXXGwG49NL25ORYj/Em/AQy8tq5wDvAVkCAxiJynaoudDu4oFf4wNpAKx1VUgUd2D344DzS03OoXz+Ol17qx5AhnWw0NBOWArmn8DxwiaquARCRk3CSRIKbgYUEa3VU6Q0b9gnvv78KgGuuOYUXXrjImpmasBbIPYUqBQkBQFXXAtbYvjT7NsDOFVClBsRb6aiyGjmyK82b1yQxcQjvvXe5JQQT9gK5UlguIq/jXB0ADMU6xCtdQbcWbQZAlHWOVlksXbqVr7/eyNixzsA3558fT1LSHcTEWPfWxkBgSeEW4E7gHzj3FBYAL7sZVEiw0lGlkp6ew7hx83juucXk5ys9erQobHJqCcGYw0r8axCRU4C2wMeq+lTFhBQC9iXDjh8hujrEX+R1NGFv/vxN3HRTIhs2pBIRIYwZcxbdujX1OixjKqWSekn9fzhdXAwF5opIcSOwlUhE+onIOhFJEpH7SlhvsIioiITGzeuC0lHbARBd1dtYwtj+/ZncfPNMLrjgLTZsSOWUUxqyePEInn66L3Fx9vylMcUp6UphKHCqqqaJSANgFjA10A2LSCTO4Dx9gBRgqYgk+t+09q1XA6c8taSswVdahaWjwd7GEeYeemgekyYtJzo6goce6snYsedQpUqk12EZU6mVlBSyVDUNQFV3iUggLZX8dcd50C0ZQEQ+BAYBa4qs93/AU8CYMm6/ctq/CXYsg+hqEH+x19GEHVUtfL7g4YfPY+PGfYwf34uTT27ocWTGBIeSTvRtRGSG7+djoK3f/IwSPlegGbDFbz6FIh3pichpQAtV/azMkVdWha2OLrXSUQVSVd5/fxUXXvg22dl5ANSvH8fMmVdbQjCmDEq6UriiyPwrZdx2cY+DauGbzpXH88DwUjckMgoYBdCyZcsyhlHBrNVRhUtJOcCtt37OZ5/9BsB77/3MDTec5nFUxgSnksZo/uo4t52CM0BPgebAH37zNYBOwHzf5X5jIFFEBqrqsiKxTAImASQkJCiV1YHfYfsPEBUHra105Lb8fOWNN37k3nvncvBgNrVqxfDss30ZPryL16EZE7TcbKC9FGjn61V1KzAEuKbgTVXdj99QnyIyHxhTNCEElcLSUX+IjvM2lhCXlLSXkSNnMn/+JgAGDerAhAn9adq0hreBGRPkXEsKqporIqOBOUAkMFVVV4vIY8AyVU10a9+eKUgKVjpy3bff/s78+Zto2LAar7xyMYMHd7QO7IwpBwEnBRGJUdWssmxcVWfhNGX1X/bwUdY9vyzbrnQObIFtiyGqKrS5xOtoQtK+fZnUru10GTJ8eBd27UpnxIjTqFfPrsqMKS+lNjMVke4isgpY75vvLCLWzUVR6/1LR9apWnnKyspl3Lh5tGr1AuvX7wGcYTL/8Y+zLSEYU84CuVJ4CbgU5+lmVHWliFzgalTByEZYc8XixSmMGJHImjW7AJgzZwPt2tXzOCpjQlcgSSFCVX8vUq/Ncyme4HRgC2xb5PSG2tpKR+UhLS2bhx6axwsvLEYV2rWry5QpAws7sTPGuCOQpLBFRLoD6uu64g7gN3fDCjJJvmf5Wl8CVap7G0sIWLIkhWuumUFyciqRkcKYMT0YN+48qla1/oqMcVsgSeFWnBJSS2AH8D/fMlPASkflqnbtWLZuPUDnzo2YMmWg9WhqTAUqNSmo6k6cZwxMcQ5uhT8WOqWjNpd6HU3Q+u67zZx9dgtEhA4d6vP119dz+ulNiY62DuyMqUilJgUReQO/7ikKqOooVyIKNuv/67zGX2ylo2Owc2cad945m2nTVvPWW5cxbFhnAHr0aFHKJ40xbgikfPQ/v+lY4C8c2dFdeLNuso+JqvLee6u4664v2Ls3g7i46MKO7Iwx3gmkfDTNf15E3gHmuhZRMDn0B2xdCJExzoA6JiCbN+/nlls+Y/bsJAD69GnDpEkDiI+v7XFkxphj6eaiNWDtAgHWzwAU4vtBFetzJxBLlqTQu/c7HDqUTe3asTz//EVcf31n66LCmEoikHsKqRy+pxAB7AWOOrRmWCkoHXWwVkeB6tKlMS1a1OTN3a/iAAAULUlEQVTEE+vz6quX0KSJJVNjKpMSk4I4X9864/RyCpCvqpW36+qKdGgbpHwLkVWgjZWOjiY3N59XXvmBYcM6U7duVWJioli48Ebq1LEBiIypjErs+8iXAD5W1TzfjyWEAgWlo1YXQUxNr6OplFau3M4ZZ0zm7rvn8Pe/zylcbgnBmMorkHGXfxCRrq5HEmwKOsCz0tGfZGbm8uCDX5OQ8AbLl2+jZctaXH11J6/DMsYE4KjlIxGJUtVc4BxgpIhsANJwhtlUVQ3fRJG2A1IWOKWjtgO9jqZS+f77LYwYkcivv+5GBEaPPp0nnuhFjRoxXodmjAlASfcUfgC6ApdVUCzBY/0M0Hxo1RdiankdTaWRlLSXc8/9N/n5SocO9ZgyZSBnn13Jx9Q2xhyhpKQgAKq6oYJiCR6/WV9HxTnhhLqMGtWVunWr8tBD5xEb6+Zor8YYN5T0V9tARP5+tDdV9TkX4qn80ndCyjcQER32paPU1AzuuedLbrihS2GX1hMm9LdnDowJYiUlhUigOr4rBuOz/mOndBTfD2LD9wncGTPWcvvts9i+/RA//riNn366GRGxhGBMkCspKWxT1ccqLJJgEealo+3bDzF69Cz++9+1AJxzTksmTx5gycCYEFHqPQXjJ30XbJnnlI5OGOR1NBVKVXn77ZXcffccUlMzqV69Ck8+2ZtbbkkgIsL+qxgTKkpKCr0qLIpgkVRQOroIYut4HU2F2rcvk3vu+ZLU1Ez69TuBiRP706pV+JbPjAlVR00Kqrq3IgMJCmE2wlp+vpKfr0RFRVCnTlVef/1S0tNzuPbaU61cZEyICuSJZgOQvttXOoqCtqFfOvr119307Plvxo//rnDZFVd05LrrrEdTY0KZJYVAJX0Cmgcte0HVul5H45qcnDyeeOJbOneeyMKFW5gyZQWZmbleh2WMqSD2dFGgwqDV0YoV27jxxkR++mk7ACNGnMbTT/exh9CMCSP21x6IjD2w+SuQSDgh9Hr9yMnJY9y4+Tz11ELy8pT4+Nq88cYAevdu43VoxpgKZkkhEAWlo1Z9oWo9r6Mpd1FRESxZspX8fOWuu87g8ccvpHr1Kl6HZYzxgCWFQBSWjgZ7G0c5Ongwi4MHs2natAYiwuTJA9i+/RBnndXC69CMMR6yG82lydjrVzr6i9fRlIs5c5Lo1Ok1hg6dQcG4Sa1b17GEYIyxpFCqDZ9Cfi60uADi6nsdzXHZsyed66//hH793mPz5v0cPJjFnj0ZXodljKlEXE0KItJPRNaJSJKI3FfM+38XkTUi8rOIfCUirdyM55gUlI6CeIQ1VWX69DV07DiBt99eSWxsFE891ZvFi2+ifv04r8MzxlQirt1TEJFI4FWgD5ACLBWRRFVd47faCiBBVdNF5FbgKeAqt2Iqs8xU+P1/IBFBWzpSVYYOncEHH/wCQM+erXjjjQG0bx96N8yNMcfPzSuF7kCSqiarajbwIXDEo8CqOk9V032zi4HmLsZTdkmfQn4OtDgf4hp4Hc0xERE6dmxAjRpVeO21/sybd70lBGPMUbnZ+qgZsMVvPgU4o4T1RwCzi3tDREYBowBatqzA4R3XT3deg+yBtY0bU0lOTqVXL+c5g7Fjz2b48C40b17T48iMMZWdm1cKxXWQo8WuKHItkAA8Xdz7qjpJVRNUNaFBgwr6xp65DzZ96ZSO2l1eMfs8Tnl5+bz44mI6dXqNq66azs6daQBER0daQjDGBMTNK4UUwL+NY3Pgj6IriUhv4AHgPFXNcjGestmQ6CsdXQBxDb2OplRr1uzippsSWbQoBYCBAzvYOAfGmDJzMyksBdqJSGtgKzAEuMZ/BRE5DXgd6KeqO12MpeyCpK+jnJw8nnxyIf/3fwvIzs6jadMavPZafwYO7OB1aMaYIORaUlDVXBEZDczBGe95qqquFpHHgGWqmohTLqoO/MfXHfNmVR3oVkwBy9oPv38JCLSr3K2OrrlmBtOnOw26Ro7sytNP96FWrViPozLGBCtXu7lQ1VnArCLLHvab7u3m/o/ZhpmQlw3Nz4Nqjb2OpkR33XUGP/20nddfv5QLL2ztdTjGmCBnTzQXpxKXjr75ZhOPPjq/cP6cc1qydu3tlhCMMeXCOsQrKusAbJoDCLS/wutoCh04kMXYsXOZOPFHAC64oDU9ezoPgEdFWW43xpQPSwpFJc+EvCxo3rPSlI5mzVrPzTd/RkrKAaKjI3jggXM588zK9ZyfMSY0WFIoap2vdNTO+26yd+9O529/+4L33lsFQPfuzZgyZSCdOlX+JrLGmOBkScFf9kHY9AWVpXT02GPf8N57q6haNYrHH7+Qu+46g8hIKxUZY9xjScHfhs+c0lGzc6B6U09CUFV8zXN59NHz2bEjjSeeuJC2bet6Eo8xJrzY105/HrY6UlXeeONHevSYSmZmLgB16lRl2rTBlhCMMRXGkkKB7EOwydcfX7uKLR1t2LCXXr3eZtSoz1i8OIWPPlpdofs3xpgCVj4qkPwZ5GZC0x5Qo1mF7NLpwG4JDz74NRkZuTRoEMfLL1/MlVeeXCH7N8aYoiwpFPitYrvJXr16JzfemMgPP2wFYOjQU3jhhX42EpoxxlOWFABy0mCjrzeO9hXTFHXFiu388MNWmjWrweuvX0r//u0rZL/GGFMSSwoAyZ9DbgY0OQtquPdQ2K5daTRoUA1wrgz27cvkuutOtQ7sjDGVht1ohsOtjjq4UzpKT89hzJgviY9/kbVrdwHOMJmjR3e3hGCMqVTsSiEnzblSAFdaHc2bt5GRI2eyYUMqERHCggW/c9JJwTneszEm9FlS2DjbVzo6A2qW3/jP+/dn8o9/zGXSpOUAnHJKQ6ZOHURCgjcPxRljTCAsKawr/wfWvvtuM0OGTGfr1oNER0fw0EM9GTv2HKpUiSy3fRhjjBvCOynkpDvPJ0C5tjpq3Lg6e/ZkcOaZzZk8eQAnn2wd2BljgkN4J4WNsyE3HRp3h5qtjnkzqsrcucn06dMGEeGEE+ry3Xc30KVLY+vAzhgTVML7jFUOfR1t2bKfAQM+4KKL3uXf//6pcHm3bk0tIRhjgk74XinkZPiVjsre6ig/3+nA7t5753LwYDa1asUQE2P3DIwxwS18k8KmL5zmqI0SoFbZxjdev34PI0fO5JtvfgfgsstO5NVXL6Fp0xpuRGqMMRUmfJPCMZaOvv9+C716vU1mZi4NG1bjlVcuZvDgjoVjIBhjTDALz6SQkwEbZjrTZXyKOSGhKe3a1eW005rw3HN9qVfPOrAzxoSO8EwKm+ZAziFo1K3U0lFWVi7PPPM9N9+cQP36cVSpEsnChTdSo0ZMBQVrjDEVJzyTQkHpqF3JzyYsXpzCiBGJrFmzi7Vrd/Puu5cDWEIwxoSs8EsKuZmQXHLpKC0tmwcf/JoXX1yCKrRvX4+bb+5WgUEaY4w3wi8pbPoSsg9Cw9Ogdts/vf3VV8mMHDmTjRv3ERkp3HtvD8aNO5/Y2PD7VRljwk/4nelKaHX022976NPnHVShS5fGTJkykK5dm1RwgMYY453wSgq5WbAh0Zkupq+j9u3rcdddZ9CgQTXuvbcH0dH2MJoxJryEV1L4/UvIPgANOkOdduzYcYg77/yCW27pxgUXOK2Qnn++n8dBGmOMd1ztnEdE+onIOhFJEpH7ink/RkSm+d5fIiLxbsbDb9MB0HaDeeedlXTsOIGPPlrN3XfPQVVd3bUxxgQD15KCiEQCrwIXAx2Bq0WkY5HVRgCpqnoC8DzwpFvxOKWjT9mcWov+Dzdk2LBP2Ls3g7592/LJJ0PsiWRjjMHd8lF3IElVkwFE5ENgELDGb51BwCO+6enAKyIi6sLX9vxNc5n4dXvGzu7Locxt1KkTy/PPX8SwYZ0tIRhjjI+bSaEZsMVvPgU442jrqGquiOwH6gG7yzuY/Stm8Ojc8ziUGc0VV5zEK69cQuPG1ct7N8YYE9TcTArFff0uegUQyDqIyChgFEDLlsc2jnKdatlMvmoW2ee8wBXD+xzTNowxJtS5mRRSgBZ+882BP46yToqIRAG1gL1FN6Sqk4BJAAkJCcdWWrrkXQb0OgAxNY/p48YYEw7cbH20FGgnIq1FpAowBEgssk4icL1vejDwtRv3EwpZQjDGmBK5dqXgu0cwGpgDRAJTVXW1iDwGLFPVRGAK8I6IJOFcIQxxKx5jjDGlc/XhNVWdBcwqsuxhv+lM4NgHSDbGGFOubGR5Y4wxhSwpGGOMKWRJwRhjTCFLCsYYYwpZUjDGGFNIgq13UBHZBfx+jB+vjwtdaFRydszhwY45PBzPMbdS1QalrRR0SeF4iMgyVU3wOo6KZMccHuyYw0NFHLOVj4wxxhSypGCMMaZQuCWFSV4H4AE75vBgxxweXD/msLqnYIwxpmThdqVgjDGmBCGZFESkn4isE5EkEbmvmPdjRGSa7/0lIhJf8VGWrwCO+e8iskZEfhaRr0SklRdxlqfSjtlvvcEioiIS9C1VAjlmEbnS92+9WkTer+gYy1sA/7dbisg8EVnh+/99iRdxlhcRmSoiO0Xkl6O8LyLyku/38bOIdC3XAFQ1pH5wuuneALQBqgArgY5F1rkNmOibHgJM8zruCjjmC4A43/St4XDMvvVqAAuAxUCC13FXwL9zO2AFUMc339DruCvgmCcBt/qmOwKbvI77OI+5J9AV+OUo718CzMYZufJMYEl57j8UrxS6A0mqmqyq2cCHwKAi6wwC3vJNTwd6iUhxQ4MGi1KPWVXnqWq6b3Yxzkh4wSyQf2eA/wOeAjIrMjiXBHLMI4FXVTUVQFV3VnCM5S2QY1agYAStWvx5hMegoqoLKGYESj+DgLfVsRioLSJNymv/oZgUmgFb/OZTfMuKXUdVc4H9QL0Kic4dgRyzvxE43zSCWanHLCKnAS1U9bOKDMxFgfw7twfai8hCEVksIv0qLDp3BHLMjwDXikgKzvgtd1RMaJ4p6997mbg6yI5HivvGX7SJVSDrBJOAj0dErgUSgPNcjch9JR6ziEQAzwPDKyqgChDIv3MUTgnpfJyrwW9FpJOq7nM5NrcEcsxXA2+q6rMichbOaI6dVDXf/fA84er5KxSvFFKAFn7zzfnz5WThOiIShXPJWdLlWmUXyDEjIr2BB4CBqppVQbG5pbRjrgF0AuaLyCac2mtikN9sDvT/9qeqmqOqG4F1OEkiWAVyzCOAjwBUdREQi9NHUKgK6O/9WIViUlgKtBOR1iJSBedGcmKRdRKB633Tg4Gv1XcHJ0iVesy+UsrrOAkh2OvMUMoxq+p+Va2vqvGqGo9zH2Wgqi7zJtxyEcj/7U9wGhUgIvVxyknJFRpl+QrkmDcDvQBE5CScpLCrQqOsWInAMF8rpDOB/aq6rbw2HnLlI1XNFZHRwByclgtTVXW1iDwGLFPVRGAKziVmEs4VwhDvIj5+AR7z00B14D++e+qbVXWgZ0EfpwCPOaQEeMxzgL4isgbIA+5V1T3eRX18Ajzme4A3RORunDLK8GD+kiciH+CU/+r77pOMA6IBVHUizn2TS4AkIB24oVz3H8S/O2OMMeUsFMtHxhhjjpElBWOMMYUsKRhjjClkScEYY0whSwrGGGMKWVIwlY6I5InIT34/8SWsG3+03iTLuM/5vp44V/q6iOhwDNu4RUSG+aaHi0hTv/cmi0jHco5zqYh0CeAzfxORuOPdtwkPlhRMZZShql38fjZV0H6HqmpnnM4Sny7rh1V1oqq+7ZsdDjT1e+8mVV1TLlEejnMCgcX5N8CSggmIJQUTFHxXBN+KyHLfT49i1jlZRH7wXV38LCLtfMuv9Vv+uohElrK7BcAJvs/28vXTv8rXz32Mb/l4OTw+xTO+ZY+IyBgRGYzTv9R7vn1W9X3DTxCRW0XkKb+Yh4vIy8cY5yL8OkITkddEZJk44yg86lt2J05ymici83zL+orIIt/v8T8iUr2U/ZgwYknBVEZV/UpHH/uW7QT6qGpX4CrgpWI+dwvwoqp2wTkpp/i6PbgKONu3PA8YWsr+BwCrRCQWeBO4SlVPwekB4FYRqQv8BThZVU8FHvf/sKpOB5bhfKPvoqoZfm9PBy73m78KmHaMcfbD6daiwAOqmgCcCpwnIqeq6ks4/eJcoKoX+Lq+eBDo7ftdLgP+Xsp+TBgJuW4uTEjI8J0Y/UUDr/hq6Hk4ffoUtQh4QESaAzNUdb2I9AK6AUt93XtUxUkwxXlPRDKATTjdL3cANqrqb7733wJuB17BGZ9hsoh8DgTcNbeq7hKRZF+fNet9+1jo225Z4qyG0+2D/6hbV4rIKJy/6yY4A878XOSzZ/qWL/TtpwrO780YwJKCCR53AzuAzjhXuH8aNEdV3xeRJUB/YI6I3ITTzfBbqnp/APsY6t9hnogUO8aGrz+e7jidsA0BRgMXluFYpgFXAr8CH6uqinOGDjhOnBHIxgOvApeLSGtgDHC6qqaKyJs4HcMVJcBcVb26DPGaMGLlIxMsagHbfH3kX4fzLfkIItIGSPaVTBJxyihfAYNFpKFvnboS+PjUvwLxInKCb/464BtfDb6Wqs7CuYlbXAuggzjddxdnBnAZzjgA03zLyhSnqubglIHO9JWeagJpwH4RaQRcfJRYFgNnFxyTiMSJSHFXXSZMWVIwwWICcL2ILMYpHaUVs85VwC8i8hNwIs6QhWtwTp5fisjPwFyc0kqpVDUTpwfK/4jIKiAfmIhzgv3Mt71vcK5iinoTmFhwo7nIdlOBNUArVf3Bt6zMcfruVTwLjFHVlThjM68GpuKUpApMAmaLyDxV3YXTMuoD334W4/yujAGsl1RjjDF+7ErBGGNMIUsKxhhjCllSMMYYU8iSgjHGmEKWFIwxxhSypGCMMaaQJQVjjDGFLCkYY4wp9P8BoBBQVzFoMrUAAAAASUVORK5CYII=\n",
        "text/plain": "<Figure size 432x288 with 1 Axes>"
       },
       "metadata": {
        "needs_background": "light"
       },
       "output_type": "display_data"
      }
     ],
     "persistentId": "8f75a628-0448-474b-8eba-ccfae9ffc601",
     "text": "fpr, tpr, thresholds = metrics.roc_curve(y_true, y_pred)\n\nfig, ax = plt.subplots()\nlw = 2\nax.plot(fpr, tpr, lw=2, color='darkorange')\nax.plot([0, 1], [0, 1], color='navy', lw=lw, linestyle='--')\nax.set(xlabel='False Positive Rate', ylabel='True Positive Rate')"
    },
    "executionTime": "2019-10-11T20:51:32.262Z"
   },
   {
    "cell": {
     "executionCount": 19,
     "executionEventId": "d4f46011-7a28-4b70-8d2b-0ee5419c7dbc",
     "hasError": false,
     "id": "c644f8a4-dc14-489f-80fb-21d9a39c084f",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "0.8080168776371308\n"
      }
     ],
     "persistentId": "91fecc99-2e19-42e2-9fb8-a694ea075d8f",
     "text": "print(metrics.accuracy_score(y_true, y_pred > 0.9))"
    },
    "executionTime": "2019-10-11T20:51:32.292Z"
   },
   {
    "cell": {
     "executionCount": 20,
     "executionEventId": "2a9d947f-2577-4b7f-b375-f55eb7d0932c",
     "hasError": false,
     "id": "1e39d7f6-23b4-4c86-b1a9-894c2d9a8a53",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "              precision    recall  f1-score   support\n\n       False       0.87      0.84      0.85       318\n        True       0.70      0.74      0.72       156\n\n   micro avg       0.81      0.81      0.81       474\n   macro avg       0.78      0.79      0.79       474\nweighted avg       0.81      0.81      0.81       474\n\n"
      }
     ],
     "persistentId": "e0cd2f37-754e-4062-92c6-e5a3223e4c7c",
     "text": "print(metrics.classification_report(y_true, y_pred > 0.9))"
    },
    "executionTime": "2019-10-11T20:51:32.314Z"
   },
   {
    "cell": {
     "executionCount": 21,
     "executionEventId": "2b2663de-52fa-41d1-a27c-453752af2449",
     "hasError": false,
     "id": "efba9e80-c61d-4adb-9d6c-af7db00a9356",
     "outputs": [
      {
       "data": {
        "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>read_id</th>\n      <th>jaccard_in_peptide_db</th>\n      <th>n_kmers</th>\n      <th>classification</th>\n      <th>short_read_id</th>\n      <th>in_cds_no_stop_codon</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>52</th>\n      <td>SRR306838.17561853 Ibis_Run100924_C3PO:6:88:9412:18824/1</td>\n      <td>0.000000</td>\n      <td>15</td>\n      <td>non-coding</td>\n      <td>SRR306838.17561853</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>55</th>\n      <td>SRR306838.733972 Ibis_Run100924_C3PO:6:4:4934:4948/1</td>\n      <td>0.100000</td>\n      <td>10</td>\n      <td>non-coding</td>\n      <td>SRR306838.733972</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>61</th>\n      <td>SRR306838.3607263 Ibis_Run100924_C3PO:6:17:17893:5364/1</td>\n      <td>0.294118</td>\n      <td>17</td>\n      <td>non-coding</td>\n      <td>SRR306838.3607263</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>81</th>\n      <td>SRR306838.15455056 Ibis_Run100924_C3PO:6:78:4010:4797/1</td>\n      <td>0.312500</td>\n      <td>16</td>\n      <td>non-coding</td>\n      <td>SRR306838.15455056</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>89</th>\n      <td>SRR306838.14450220 Ibis_Run100924_C3PO:6:72:5878:19983/1</td>\n      <td>0.062500</td>\n      <td>16</td>\n      <td>non-coding</td>\n      <td>SRR306838.14450220</td>\n      <td>True</td>\n    </tr>\n  </tbody>\n</table>\n</div>",
        "text/plain": "                                                     read_id  \\\n52  SRR306838.17561853 Ibis_Run100924_C3PO:6:88:9412:18824/1   \n55      SRR306838.733972 Ibis_Run100924_C3PO:6:4:4934:4948/1   \n61   SRR306838.3607263 Ibis_Run100924_C3PO:6:17:17893:5364/1   \n81   SRR306838.15455056 Ibis_Run100924_C3PO:6:78:4010:4797/1   \n89  SRR306838.14450220 Ibis_Run100924_C3PO:6:72:5878:19983/1   \n\n    jaccard_in_peptide_db  n_kmers classification       short_read_id  \\\n52               0.000000       15     non-coding  SRR306838.17561853   \n55               0.100000       10     non-coding    SRR306838.733972   \n61               0.294118       17     non-coding   SRR306838.3607263   \n81               0.312500       16     non-coding  SRR306838.15455056   \n89               0.062500       16     non-coding  SRR306838.14450220   \n\n    in_cds_no_stop_codon  \n52                  True  \n55                  True  \n61                  True  \n81                  True  \n89                  True  "
       },
       "execution_count": 21,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "1254868e-73a0-4c0a-82c6-bb1b85e78cb0",
     "text": "coding_scores_unspliced.query(\"jaccard_in_peptide_db < 0.5 & in_cds_no_stop_codon == True\").head()"
    },
    "executionTime": "2019-10-11T20:51:32.371Z"
   },
   {
    "cell": {
     "executionCount": 29,
     "executionEventId": "2cc089f4-800c-4ff1-abde-c461430e5369",
     "hasError": false,
     "id": "4c484434-ea07-4b72-8073-567bf902bedb",
     "outputs": [
      {
       "data": {
        "text/plain": "150289"
       },
       "execution_count": 29,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "75c01782-427c-4c63-bc29-ed308ae4a142",
     "text": "true_coding_read_ids = set(coding_read_ids.values)\nlen(true_coding_read_ids)"
    },
    "executionTime": "2019-10-12T00:53:28.111Z"
   },
   {
    "cell": {
     "executionCount": 30,
     "executionEventId": "a778edc2-b4f4-48de-9339-eeed0bb25fcc",
     "hasError": false,
     "id": "9babb580-ed7e-4a80-bcb3-3d080fca4a29",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "(807698,)\n"
      }
     ],
     "persistentId": "317bb484-c93f-4e1c-b831-50734c9e6614",
     "text": "unspliced_read_ids_series = pd.read_csv(f\"{folder}/SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.read_ids.txt\", \n                                   header=None, squeeze=True)\nprint(unspliced_read_ids_series.shape)\nunspliced_read_ids_series.head()\n\nunspliced_reads = set(unspliced_read_ids_series.values)"
    },
    "executionTime": "2019-10-12T00:53:28.948Z"
   },
   {
    "cell": {
     "executionCount": 33,
     "executionEventId": "a3f70e1c-821d-4586-a8b8-80d2fedbfc7b",
     "hasError": false,
     "id": "a18fd62e-900f-4a7d-8ada-1b3b45c4d677",
     "outputs": [],
     "persistentId": "27025afb-bac7-4e3d-822e-f3238fba3a50",
     "text": "y_true = coding_scores_unspliced_not_low_complexity['in_cds_no_stop_codon']\ny_pred = coding_scores_unspliced_not_low_complexity['jaccard_in_peptide_db']"
    },
    "executionTime": "2019-10-12T00:53:35.323Z"
   },
   {
    "cell": {
     "executionCount": 34,
     "executionEventId": "a5c828f9-934b-452c-bfb9-9abd06d5c909",
     "hasError": false,
     "id": "3f61b436-51d7-4dae-a329-c238ba62d3d5",
     "outputs": [
      {
       "data": {
        "text/plain": "[Text(0, 0.5, 'True Positive Rate'), Text(0.5, 0, 'False Positive Rate')]"
       },
       "execution_count": 34,
       "metadata": {},
       "output_type": "execute_result"
      },
      {
       "data": {
        "image/png": "iVBORw0KGgoAAAANSUhEUgAAAYUAAAEKCAYAAAD9xUlFAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDMuMC4zLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvnQurowAAIABJREFUeJzt3Xl4FFX28PHvyUJC2PcdAggooiBEVFRcWEQRcJRRFEUUwQ11HHHQ1wX15zi474gIjLviMKhBQWQURBEQBBEBkRAQguyELfty3j+qE5oYkg6kUunu83mePF1VXV11KpA6Xadu3SuqijHGGAMQ4XUAxhhjKg9LCsYYYwpZUjDGGFPIkoIxxphClhSMMcYUsqRgjDGmkCUFY4wxhSwpGGOMKWRJwRhjTKEorwMoq/r162t8fLzXYRhjTFD58ccfd6tqg9LWC7qkEB8fz7Jly7wOwxhjgoqI/B7IelY+MsYYU8iSgjHGmEKWFIwxxhSypGCMMaaQJQVjjDGFXEsKIjJVRHaKyC9HeV9E5CURSRKRn0Wkq1uxGGOMCYybVwpvAv1KeP9ioJ3vZxTwmouxGGOMCYBrzymo6gIRiS9hlUHA2+qMB7pYRGqLSBNV3eZWTMYYUyHysiFrP2Tt873uh+z9h6eLm8/LOurmfkiqSWx0Pqee2x16POJq6F4+vNYM2OI3n+Jb9qekICKjcK4maNmyZYUEZ4wJU4Un9KIn7gBP8Nn7ITezXEJRhbGf9+HZb7pxapMd/NDhV6LLZctH52VSkGKWaXErquokYBJAQkJCsesYYwwAeTmQsQvSd0Lm3qOfyAtO8m6c0COioEotiPH7KW0+MhbkyNOiAGxIggVb6DvoDPJOPz2kk0IK0MJvvjnwh0exGGMqK1XIPuCc5NN3QvoOv+lilmXuPb79HcsJvUotiKl9eD6q6p9O8IHaty+T5ORUunZtAsCjz53FkJG7C+fd5mVSSARGi8iHwBnAfrufYEyYKPg2n7YDMko4wRf8lFBv/xOJgKoNIK4hVK1XhhN87eM+oR+vTz/9lVtv/ZyICGH16tuoVSuWqlWjKywhgItJQUQ+AM4H6otICjAOnCsfVZ0IzAIuAZKAdOAGt2IxxrhM1Sm/FJzEC070aTv+PJ+xEzJTy7b96OrOST6uke+14dHnq9Z1EkMQ2bkzjTvvnM20aasBOPPM5uzbl0mtWrEVHoubrY+uLuV9BW53a//GmOOUlw3puw6f1NN2HHnSLzqflx34tiUS4gq+zftO6NUaHZ4umI9r6Hzrj45z7zg9pKq8994q7rrrC/buzSAuLponnriQ0aO7ExnpTWILuq6zjTHHKD/PubmasbvkmnzBfNa+sm2/So3DJ/lqft/gqxY9yTcMym/zbrj11s95/fUfAejduw2TJl1K69Z1PI3JkoIxwSgvBzL3QMYe5yTvP52xxzdfZDozlaM08Cte4bf5IiWaoif9gmXRVV073FB12WUnMm3aap59ti833NAF8ehehj9LCsZ4LSej+JN4cSf8gunsA8e2r9g6EFuv9Lp8XENnXfs2X67Wr9/DV19t5JZbEgDo1+8ENm26y5N7B0djScGY8pZ9qEhJZoffDdddfz7552aUfR8S4Zzcq9bzvdZ3pqvWP7z8T9N1nOaWpsLl5ubz3HOLGDduPllZuXTp0pgzz2wOUKkSAlhSMKZkO5bD3l+PXKb5TinG/6RfMJ22A3LTy7aPiOgAT+p+J/+YWvYtPkisXLmdESMS+fFHp8X9sGGdadeursdRHZ0lBWOKys2EtG2w5F+w6o2yfz4yxinBFNbd/V8b/PkEH13ds3bxxj1ZWbk8/vgCxo9fSG5uPi1b1uL11y+lX78TvA6tRJYUTHhRdVrVHPgdDmz2vf4OB/3m03ccXj8iGtoOhMgqR24nps6RLWr8T/xVathJ3nD//V/x/POLAbj99tP51796UaNGjMdRlc6SggltORmQ8g1snO287k+G7IMlfyYiynm6tX4nuPBl59WYMvrHP85m0aIUnnqqN+ee28rrcAJmScGEFlXYl+QkgY2zIWX+nzs4i64GNVsd/qnR8sj5ak0gItKT8E3wmjt3AxMn/si0aYOJioqgcePqfP/9jZWimWlZWFIwwS8nDbbMP5wI9icf+X7DrtD6YojvC/VOhti6Vt4x5SY1NYMxY75k6tSfAPj3v1cwcmQ3gKBLCGBJwVQmmu/8BCJ1PWz6wnc1sODIDtNi60Krvr5EcJFT9zfGBR9/vJbbbpvF9u2HiImJZNy48xg+vIvXYR0XSwqm4uVmOs0896yBPat9r2ucsk+gSeEIAo27Q3w/JxE0Pt3KP8ZV27cf4o47ZjN9+hoAevRowZQpAznxxPoeR3b8LCmEo01zYemTkJ9bwTtWOPSHU9452slfAjyZV63nuxro57zGNSi/MI0pxaef/sr06WuoVi2a8eN7c9ttpxMREXylouJYUghHK16GzV95t3+JhDodoF5HqH8y1O3oTNftAFGV6+lOYwpkZuYSG+ucMkeO7EZyciq33no68fG1PY6sfFlSCAd71x3Z42XGbuc1YQy06V+xscTWgzrtIaryt9c2BiA/X5kwYSn//Oe3LF48glatahMRITz5ZB+vQ3OFJYVQ992DsOSfxb/X8kJocX6FhmNMMFm3bjcjRiSycOEWAD744Bfuu+8cj6NylyWFULbhMychSCQ06opvGHBH9abQLLT/cxtzrHJy8njmme959NFvyMrKo1GjakyY0J/LLz/J69BcZ0khVB34Hb4Y5kyf80/oPtbbeIwJEr/8spNhwz5mxYrtANxwQxeefbYvdeqEx3gRlhRCUV42fHaV05Nnm/5w+r1eR2RM0MjPV1at2kmrVrWYNGkAffu29TqkCmVJIRQtGAvblkCNFtDvLeti2ZhSrF69k44dGyAinHpqIz79dAg9e7aievUqpX84xNjZItTsWA7LX3A6dbv0I6c9vzGmWAcPZjF69Cw6dXqN//53beHySy5pF5YJAexKIfSk/ua8tr4Emp7pbSzGVGJz5iQxatRnbN68n6ioCDZt2lf6h8KAJYVQkpsJix93puue6G0sxlRSe/dmcPfdc3j77ZUAdO3ahClTBtKlS2OPI6scLCmEkj++d/oSqhkPZz7kdTTGVDo//bSdfv3eZceONGJiInn00fO5554eREVZJb2AJYVQUvCkcr2ToEp1b2MxphJq374e1atXoX37ekyePJD27e2eW1GWFELJuo+c12Y9vY3DmEpCVXn//VUMGNCBmjVjiIuLZv784TRtWiNkOrArb3bNFCoy9sCGRKf56cnDvI7GGM9t2rSPiy56l2uv/Zj77vtf4fLmzWtaQiiBXSmEij1rID/HGUugelOvozHGM3l5+UyYsJT77/+KtLQc6tatSo8eLbwOK2hYUgg1kdb7qAlfa9fuYsSIRBYtSgHgyitP5uWXL6Zhw2oeRxY8LCkYY0LCxo2pdOnyOtnZeTRpUp0JE/pz2WXWNLusXE0KItIPeBGIBCar6vgi77cE3gJq+9a5T1VnuRmTMSY0tW5dh7/+tSOxsVE880xfate2AZuOhWtJQUQigVeBPkAKsFREElV1jd9qDwIfqeprItIRmAXEuxWTMSZ0ZGTk8Nhj3/CXv5xE9+7NAHjrrcuIjLT2M8fDzd9edyBJVZNVNRv4EBhUZB0FavqmawF/uBiPMSZEfPvt73Tp8jrjxy9k1KiZ5OcrgCWEcuBm+agZsMVvPgU4o8g6jwBfisgdQDWgt4vxhLYt853XqPDo892EpwMHsrj//v8xYcIyADp2bMDEiZdaE9Ny5GZaLe5fSYvMXw28qarNgUuAd0T+3M+ziIwSkWUismzXrl0uhBrkDm2DRY84093u9jQUY9wya9Z6OnWawIQJy4iKiuDhh3uyfPkozjyzudehhRQ3rxRSAP/Gwc35c3loBNAPQFUXiUgsUB/Y6b+Sqk4CJgEkJCQUTSwmbRtoPjToDK0v9joaY8rd/v2ZDB06g337MklIaMqUKQM59dRGXocVktxMCkuBdiLSGtgKDAGuKbLOZqAX8KaInATEAnYpUBaq8P3DzrQNpmNCiKqiChERQq1asbz0Uj927Ejjb3870zqwc5FrSUFVc0VkNDAHp7npVFVdLSKPActUNRG4B3hDRO7GKS0NV1W7EiiL/Rsh+XNnupp1/WtCwx9/HOS22z7n3HNbcs89PQC47rrOHkcVHlx9TsH3zMGsIsse9pteA5ztZgwhLyv18PTF73gXhzHlQFWZOnUF99zzJfv3Z7F4cQq33XY6VatGex1a2LAnmoNZxl54N8GZrtPBht40QS05OZWRI2fy9dcbAejfvx0TJ15qCaGCWVIIZr++f3j61FHexWHMccjLy+ell5bwwANfk5GRS/36cbz0Uj+GDOmEiDU1rWiWFILRuv/A71/Cpi+d+f4fwIlDvI3JmOMwffpaMjJyufrqTrz4Yj8aNLAO7LxiSSFYpG2HlAWwLxm+u//w8th6cMJl3sVlzDHIzs7j4MEs6tWLIzIygilTBrJ+/R4GDOjgdWhhz5JCsJj5V9j63eH50+6A+qdA0x4QZR1/meCxdOlWRoxIpHnzmnz++TWICCeeWJ8TT6zvdWgGSwrBI32H89r6YmjVF7reBVZvNUEkPT2HcePm8dxzi8nPV9LTc9i5M41GjWw88crEkkKwOf8FqNve6yiMKZP58zcxcuRMkpL2EhEhjBlzFo8+egFxcdayqLKxpGCMcY2qcueds3nllaUAnHJKQ6ZMGcjppzfzODJzNJYUgkF+HmTscaajbLhNEzxEhJo1Y4iOjuDBB3ty333nUKVKpNdhmRJYUggGKQsgcy/Ubgs1WnodjTEl2r07nQ0b9nLGGU7vpQ89dB5Dh55Kx44NPI7MBCKgXqVEpIqInOB2MOYofpvuvLb/q91cNpWWqvLhh79w0kmvctll00hNzQAgNjbKEkIQKTUpiEh/YBUw1zffRUQ+djsw45OfB+v/60y3/6u3sRhzFCkpBxg06EOuvvq/7N6dTseODUhPz/E6LHMMAikfPYYzYto8AFX9ya4aKtDW75zmqLXaQMPTvI7GmCPk5yuTJy/n3nvncuBAFjVrxvDss30ZMeI066IiSAWSFHJUdV+Rf2Dr3rqi/PYf59VKR6YSGjEikTff/AmAgQM7MGHCJTRrVrOUT5nKLJB7CmtF5EogQkRai8gLwGKX4zJQpHQ02NtYjCnGtdeeQsOG1fjwwyv45JOrLCGEgECSwmigG5APzAAygbvcDMr4/PG90+dRzXho1M3raIzhl1928uKLh78T9urVhuTkO7nqKuvRNFQEUj66SFXHAmMLFojI5TgJwrjJSkemksjKyuVf//qOJ574lpycfBISmnL22U7z6GrVqngcnSlPgVwpPFjMsgfKOxBThOYfLh11sFZHxjtLlqTQrdskHn30G3Jy8rn11gROOaWR12EZlxz1SkFELgL6Ac1E5Dm/t2rilJKMm7Z+D4f+gJqtoFGC19GYMJSWls1DD83jhRcWowrt2tVl8uSB9OzZyuvQjItKKh/tBH7BuYew2m/5QeA+N4MyHC4dtRtspSPjiQce+JoXX1xCRIRw771n8cgj59vQmGHgqElBVVcAK0TkPVXNrMCYjJWOTCXwwAPnsmrVTp58sjcJCU29DsdUkEDuKTQTkQ9F5GcR+a3gx/XIwtkfi+HQVqefo8bdvY7GhInExHVccsl75OTkAdCgQTW++mqYJYQwE0hSeBP4NyDAxcBHwIcuxmQKWx1Z6ci4b+fONIYMmc6gQR8ye3YSb7210uuQjIcCSQpxqjoHQFU3qOqDwAXuhhXGNP/IDvCMcYmq8u67P3PSSa8ybdpq4uKiefHFftxwQxevQzMeCuQ5hSxxnkrZICK3AFuBhu6GFca2LYFDKVC9OTSx0pFxx+bN+7nlls+YPTsJgN692zBp0qW0bl3H48iM1wJJCncD1YE7gX8CtYAb3QwqrBVeJQwGCahnc2PK7MsvNzB7dhK1a8fy3HN9GT68iz2RbIAAkoKqLvFNHgSuAxCR5m4GFbZUrXRkXJOWll349PGIEaexdesBRo3qRpMmNTyOzFQmJX4VFZHTReQyEanvmz9ZRN7GOsRzx/Yf4OBmqN4Mmp7pdTQmROTm5vPUUwtp1eoFkpNTAWeYzHHjzreEYP7kqElBRP4FvAcMBb4QkQdwxlRYCbSvmPDCzDr/VkdWOjLHb+XK7ZxxxmTGjv0fe/Zk8Mknv3odkqnkSiofDQI6q2qGiNQF/vDNr6uY0MKMKqy30pEpH1lZuTz++ALGj19Ibm4+LVvWYtKkS7noIhsfy5SspKSQqaoZAKq6V0R+tYTgou1L4cDvUL0pND3L62hMEFuxYhtDh85g7drdiMDo0afzxBO9qFEjxuvQTBAoKSm0EZGC7rEFiPebR1UvL23jItIPeBGIBCar6vhi1rkSeARnNLeVqnpN4OGHkIIbzO2usNKROS4xMVFs2JBKhw71mDx5IOec09LrkEwQKSkpXFFk/pWybFhEIoFXgT5ACrBURBJVdY3fOu2A+4GzVTVVRMLz+QfVI8dOMKaMli/fxmmnNUZE6NixAbNnD6VHjxbExgbS6tyYw0rqEO+r49x2dyBJVZMBRORDnPsUa/zWGQm8qqqpvn3uPM59BqcdP8KBTVCtCTQ72+toTBBJTc1gzJgvmTr1Jz744AqGDOkEwIUXtvY4MhOs3Pwa0QzY4jefApxRZJ32ACKyEKfE9IiqflF0QyIyChgF0LJlCF4KF3aTbaUjE7iPP17LbbfNYvv2Q8TERLJnT7rXIZkQ4GZSKO7xSC1m/+2A84HmwLci0klV9x3xIdVJwCSAhISEotsIbkeUjgZ7G4sJCtu3H+KOO2Yzfbpz0X322S2YPHkgJ55Y3+PITCgIOCmISIyqZpVh2ylAC7/55jjNWouus1hVc4CNIrIOJ0ksLcN+gtvOFbB/I8Q1gmbneB2NqeR+/PEP+vR5h9TUTKpVi2b8+N7cdtvpRERYFxWmfJRaqxCR7iKyCljvm+8sIi8HsO2lQDsRaS0iVYAhQGKRdT7B1+Oq76np9kByGeIPfv6lo4hIb2MxlV7Hjg1o0KAaF13UltWrb2P06O6WEEy5CqSA/RJwKbAHQFVXEkDX2aqaC4wG5gBrgY9UdbWIPCYiA32rzQH2iMganKel71XVPWU/jCDlXzqyEdZMMfLzlUmTfmTfPmfww6pVo1mwYDizZw+lVavaHkdnQlEg5aMIVf29SA+KeYFsXFVnAbOKLHvYb1qBv/t+ws/On2DfBohrCM3O9ToaU8msW7ebm26ayXffbWbp0q288YbzXapRo+oeR2ZCWSBJYYuIdAfU9+zBHYANx1keCktHl1vpyBTKycnj2WcX8cgj88nKyqNx4+pcfHE7r8MyYSKQpHArTgmpJbAD+J9vmTke9sCaKcaKFdsYMSKRFSu2A3DDDV149tm+1KlT1ePITLgIJCnkquoQ1yMJN7t+hn1JULUBNO/pdTSmEtiwYS/du08mNzef+PjaTJp0KX36tPU6LBNmAkkKS31NRacBM1T1oMsxhYcjSkfWFYGBtm3rct11p1KjRhX++c9eVK9exeuQTBgqtfWRqrYFHge6AatE5BMRsSuH42GlIwMcOpTNnXfOZtGiww/+T5kykBdfvNgSgvFMQH0qqOr3qnon0BU4gDP4jjlWu1dB6m9QtT60OM/raIwH5sxJ4uSTJ/Dyyz9wyy2f4zTEw8ZJNp4rtW4hItVxOrIbApwEfAr0cDmu0FbQTfYJf7HSUZjZuzeDu++ew9tvrwSgW7cmTJky0JKBqTQCOSP9AswEnlLVb12OJ/RZ6ShsTZ++httvn8XOnWnExkbx6KPn8/e/n0VUlHWCaCqPQJJCG1XNdz2ScLFnNez9FWLrQctSHww3IWLfvkxGjZpJamomPXu24o03BtC+fT2vwzLmT46aFETkWVW9B/iviPypZ9JARl4zxVhX0OrISkehTlXJz1ciIyOoXTuWCRP6k5qawc03J1h/RabSKumsNM33WqYR10wprJvssLBp0z5GjZrJhRe25r77nN5vCwbAMaYyO2oxU1V/8E2epKpf+f/g3HA2ZbVnDexdC7F1ocWFXkdjXJCXl89LLy2hU6cJzJ2bzCuv/EBmZq7XYRkTsEDucN1YzLIR5R1IWCgoHZ1wGURGexuLKXdr1+6iZ883ueuuL0hLy2HIkE4sX36zjZNsgkpJ9xSuwmmG2lpEZvi9VQPYV/ynTIms1VFIys3N58knv+OxxxaQnZ1H06Y1eO21/gwc2MHr0Iwps5K+wvyAM4ZCc+BVv+UHgRVuBhWS9qx1Wh7F1oGWvbyOxpSjiAjhyy+Tyc7OY+TIrjz1VB9q1471OixjjslRk4KqbgQ24vSKao5XwVVC20FWOgoBGRk5HDyYTcOG1YiIECZPHsCWLQe48MLWXodmzHE56j0FEfnG95oqInv9flJFZG/FhRgiCp5ittJR0Fuw4Hc6d57ItdfOKOyeol27epYQTEgoqXxU8GRV/YoIJKTtXef0dxRTG1r19joac4wOHMji/vv/x4QJywCIjo5k9+50GjSo5nFkxpSfkpqkFjzF3AKIVNU84CzgZsD+CsqioHR0wiCItN4vg9Hs2evp1GkCEyYsIyoqgnHjzmP58lGWEEzICaSt3CfA6SLSFngb+Bx4H7jUzcBCirU6ClqqysiRM5kyxWlbkZDQlKlTB3LKKY08jswYdwTynEK+quYAlwMvqOodQDN3wwohe39zRlmrUhNaWuko2IgIzZvXJDY2imee6cOiRSMsIZiQFtBwnCLyV+A64DLfMms+E6j1Bd1kD4KoGG9jMQH544+DbNiwl3PPbQXA//t/53LddafStm1djyMzxn2BPtF8AU7X2cki0hr4wN2wQsg6Kx0FC1VlypTldOz4Kldc8RF79qQDUKVKpCUEEzZKvVJQ1V9E5E7gBBE5EUhS1X+6H1oISE2CXT85paNWfb2OxpQgOTmVkSNn8vXXGwG49NL25ORYj/Em/AQy8tq5wDvAVkCAxiJynaoudDu4oFf4wNpAKx1VUgUd2D344DzS03OoXz+Ol17qx5AhnWw0NBOWArmn8DxwiaquARCRk3CSRIKbgYUEa3VU6Q0b9gnvv78KgGuuOYUXXrjImpmasBbIPYUqBQkBQFXXAtbYvjT7NsDOFVClBsRb6aiyGjmyK82b1yQxcQjvvXe5JQQT9gK5UlguIq/jXB0ADMU6xCtdQbcWbQZAlHWOVlksXbqVr7/eyNixzsA3558fT1LSHcTEWPfWxkBgSeEW4E7gHzj3FBYAL7sZVEiw0lGlkp6ew7hx83juucXk5ys9erQobHJqCcGYw0r8axCRU4C2wMeq+lTFhBQC9iXDjh8hujrEX+R1NGFv/vxN3HRTIhs2pBIRIYwZcxbdujX1OixjKqWSekn9fzhdXAwF5opIcSOwlUhE+onIOhFJEpH7SlhvsIioiITGzeuC0lHbARBd1dtYwtj+/ZncfPNMLrjgLTZsSOWUUxqyePEInn66L3Fx9vylMcUp6UphKHCqqqaJSANgFjA10A2LSCTO4Dx9gBRgqYgk+t+09q1XA6c8taSswVdahaWjwd7GEeYeemgekyYtJzo6goce6snYsedQpUqk12EZU6mVlBSyVDUNQFV3iUggLZX8dcd50C0ZQEQ+BAYBa4qs93/AU8CYMm6/ctq/CXYsg+hqEH+x19GEHVUtfL7g4YfPY+PGfYwf34uTT27ocWTGBIeSTvRtRGSG7+djoK3f/IwSPlegGbDFbz6FIh3pichpQAtV/azMkVdWha2OLrXSUQVSVd5/fxUXXvg22dl5ANSvH8fMmVdbQjCmDEq6UriiyPwrZdx2cY+DauGbzpXH88DwUjckMgoYBdCyZcsyhlHBrNVRhUtJOcCtt37OZ5/9BsB77/3MDTec5nFUxgSnksZo/uo4t52CM0BPgebAH37zNYBOwHzf5X5jIFFEBqrqsiKxTAImASQkJCiV1YHfYfsPEBUHra105Lb8fOWNN37k3nvncvBgNrVqxfDss30ZPryL16EZE7TcbKC9FGjn61V1KzAEuKbgTVXdj99QnyIyHxhTNCEElcLSUX+IjvM2lhCXlLSXkSNnMn/+JgAGDerAhAn9adq0hreBGRPkXEsKqporIqOBOUAkMFVVV4vIY8AyVU10a9+eKUgKVjpy3bff/s78+Zto2LAar7xyMYMHd7QO7IwpBwEnBRGJUdWssmxcVWfhNGX1X/bwUdY9vyzbrnQObIFtiyGqKrS5xOtoQtK+fZnUru10GTJ8eBd27UpnxIjTqFfPrsqMKS+lNjMVke4isgpY75vvLCLWzUVR6/1LR9apWnnKyspl3Lh5tGr1AuvX7wGcYTL/8Y+zLSEYU84CuVJ4CbgU5+lmVHWliFzgalTByEZYc8XixSmMGJHImjW7AJgzZwPt2tXzOCpjQlcgSSFCVX8vUq/Ncyme4HRgC2xb5PSG2tpKR+UhLS2bhx6axwsvLEYV2rWry5QpAws7sTPGuCOQpLBFRLoD6uu64g7gN3fDCjJJvmf5Wl8CVap7G0sIWLIkhWuumUFyciqRkcKYMT0YN+48qla1/oqMcVsgSeFWnBJSS2AH8D/fMlPASkflqnbtWLZuPUDnzo2YMmWg9WhqTAUqNSmo6k6cZwxMcQ5uhT8WOqWjNpd6HU3Q+u67zZx9dgtEhA4d6vP119dz+ulNiY62DuyMqUilJgUReQO/7ikKqOooVyIKNuv/67zGX2ylo2Owc2cad945m2nTVvPWW5cxbFhnAHr0aFHKJ40xbgikfPQ/v+lY4C8c2dFdeLNuso+JqvLee6u4664v2Ls3g7i46MKO7Iwx3gmkfDTNf15E3gHmuhZRMDn0B2xdCJExzoA6JiCbN+/nlls+Y/bsJAD69GnDpEkDiI+v7XFkxphj6eaiNWDtAgHWzwAU4vtBFetzJxBLlqTQu/c7HDqUTe3asTz//EVcf31n66LCmEoikHsKqRy+pxAB7AWOOrRmWCkoHXWwVkeB6tKlMS1a1OTN3a/iAAAULUlEQVTEE+vz6quX0KSJJVNjKpMSk4I4X9864/RyCpCvqpW36+qKdGgbpHwLkVWgjZWOjiY3N59XXvmBYcM6U7duVWJioli48Ebq1LEBiIypjErs+8iXAD5W1TzfjyWEAgWlo1YXQUxNr6OplFau3M4ZZ0zm7rvn8Pe/zylcbgnBmMorkHGXfxCRrq5HEmwKOsCz0tGfZGbm8uCDX5OQ8AbLl2+jZctaXH11J6/DMsYE4KjlIxGJUtVc4BxgpIhsANJwhtlUVQ3fRJG2A1IWOKWjtgO9jqZS+f77LYwYkcivv+5GBEaPPp0nnuhFjRoxXodmjAlASfcUfgC6ApdVUCzBY/0M0Hxo1RdiankdTaWRlLSXc8/9N/n5SocO9ZgyZSBnn13Jx9Q2xhyhpKQgAKq6oYJiCR6/WV9HxTnhhLqMGtWVunWr8tBD5xEb6+Zor8YYN5T0V9tARP5+tDdV9TkX4qn80ndCyjcQER32paPU1AzuuedLbrihS2GX1hMm9LdnDowJYiUlhUigOr4rBuOz/mOndBTfD2LD9wncGTPWcvvts9i+/RA//riNn366GRGxhGBMkCspKWxT1ccqLJJgEealo+3bDzF69Cz++9+1AJxzTksmTx5gycCYEFHqPQXjJ30XbJnnlI5OGOR1NBVKVXn77ZXcffccUlMzqV69Ck8+2ZtbbkkgIsL+qxgTKkpKCr0qLIpgkVRQOroIYut4HU2F2rcvk3vu+ZLU1Ez69TuBiRP706pV+JbPjAlVR00Kqrq3IgMJCmE2wlp+vpKfr0RFRVCnTlVef/1S0tNzuPbaU61cZEyICuSJZgOQvttXOoqCtqFfOvr119307Plvxo//rnDZFVd05LrrrEdTY0KZJYVAJX0Cmgcte0HVul5H45qcnDyeeOJbOneeyMKFW5gyZQWZmbleh2WMqSD2dFGgwqDV0YoV27jxxkR++mk7ACNGnMbTT/exh9CMCSP21x6IjD2w+SuQSDgh9Hr9yMnJY9y4+Tz11ELy8pT4+Nq88cYAevdu43VoxpgKZkkhEAWlo1Z9oWo9r6Mpd1FRESxZspX8fOWuu87g8ccvpHr1Kl6HZYzxgCWFQBSWjgZ7G0c5Ongwi4MHs2natAYiwuTJA9i+/RBnndXC69CMMR6yG82lydjrVzr6i9fRlIs5c5Lo1Ok1hg6dQcG4Sa1b17GEYIyxpFCqDZ9Cfi60uADi6nsdzXHZsyed66//hH793mPz5v0cPJjFnj0ZXodljKlEXE0KItJPRNaJSJKI3FfM+38XkTUi8rOIfCUirdyM55gUlI6CeIQ1VWX69DV07DiBt99eSWxsFE891ZvFi2+ifv04r8MzxlQirt1TEJFI4FWgD5ACLBWRRFVd47faCiBBVdNF5FbgKeAqt2Iqs8xU+P1/IBFBWzpSVYYOncEHH/wCQM+erXjjjQG0bx96N8yNMcfPzSuF7kCSqiarajbwIXDEo8CqOk9V032zi4HmLsZTdkmfQn4OtDgf4hp4Hc0xERE6dmxAjRpVeO21/sybd70lBGPMUbnZ+qgZsMVvPgU4o4T1RwCzi3tDREYBowBatqzA4R3XT3deg+yBtY0bU0lOTqVXL+c5g7Fjz2b48C40b17T48iMMZWdm1cKxXWQo8WuKHItkAA8Xdz7qjpJVRNUNaFBgwr6xp65DzZ96ZSO2l1eMfs8Tnl5+bz44mI6dXqNq66azs6daQBER0daQjDGBMTNK4UUwL+NY3Pgj6IriUhv4AHgPFXNcjGestmQ6CsdXQBxDb2OplRr1uzippsSWbQoBYCBAzvYOAfGmDJzMyksBdqJSGtgKzAEuMZ/BRE5DXgd6KeqO12MpeyCpK+jnJw8nnxyIf/3fwvIzs6jadMavPZafwYO7OB1aMaYIORaUlDVXBEZDczBGe95qqquFpHHgGWqmohTLqoO/MfXHfNmVR3oVkwBy9oPv38JCLSr3K2OrrlmBtOnOw26Ro7sytNP96FWrViPozLGBCtXu7lQ1VnArCLLHvab7u3m/o/ZhpmQlw3Nz4Nqjb2OpkR33XUGP/20nddfv5QLL2ztdTjGmCBnTzQXpxKXjr75ZhOPPjq/cP6cc1qydu3tlhCMMeXCOsQrKusAbJoDCLS/wutoCh04kMXYsXOZOPFHAC64oDU9ezoPgEdFWW43xpQPSwpFJc+EvCxo3rPSlI5mzVrPzTd/RkrKAaKjI3jggXM588zK9ZyfMSY0WFIoap2vdNTO+26yd+9O529/+4L33lsFQPfuzZgyZSCdOlX+JrLGmOBkScFf9kHY9AWVpXT02GPf8N57q6haNYrHH7+Qu+46g8hIKxUZY9xjScHfhs+c0lGzc6B6U09CUFV8zXN59NHz2bEjjSeeuJC2bet6Eo8xJrzY105/HrY6UlXeeONHevSYSmZmLgB16lRl2rTBlhCMMRXGkkKB7EOwydcfX7uKLR1t2LCXXr3eZtSoz1i8OIWPPlpdofs3xpgCVj4qkPwZ5GZC0x5Qo1mF7NLpwG4JDz74NRkZuTRoEMfLL1/MlVeeXCH7N8aYoiwpFPitYrvJXr16JzfemMgPP2wFYOjQU3jhhX42EpoxxlOWFABy0mCjrzeO9hXTFHXFiu388MNWmjWrweuvX0r//u0rZL/GGFMSSwoAyZ9DbgY0OQtquPdQ2K5daTRoUA1wrgz27cvkuutOtQ7sjDGVht1ohsOtjjq4UzpKT89hzJgviY9/kbVrdwHOMJmjR3e3hGCMqVTsSiEnzblSAFdaHc2bt5GRI2eyYUMqERHCggW/c9JJwTneszEm9FlS2DjbVzo6A2qW3/jP+/dn8o9/zGXSpOUAnHJKQ6ZOHURCgjcPxRljTCAsKawr/wfWvvtuM0OGTGfr1oNER0fw0EM9GTv2HKpUiSy3fRhjjBvCOynkpDvPJ0C5tjpq3Lg6e/ZkcOaZzZk8eQAnn2wd2BljgkN4J4WNsyE3HRp3h5qtjnkzqsrcucn06dMGEeGEE+ry3Xc30KVLY+vAzhgTVML7jFUOfR1t2bKfAQM+4KKL3uXf//6pcHm3bk0tIRhjgk74XinkZPiVjsre6ig/3+nA7t5753LwYDa1asUQE2P3DIwxwS18k8KmL5zmqI0SoFbZxjdev34PI0fO5JtvfgfgsstO5NVXL6Fp0xpuRGqMMRUmfJPCMZaOvv9+C716vU1mZi4NG1bjlVcuZvDgjoVjIBhjTDALz6SQkwEbZjrTZXyKOSGhKe3a1eW005rw3HN9qVfPOrAzxoSO8EwKm+ZAziFo1K3U0lFWVi7PPPM9N9+cQP36cVSpEsnChTdSo0ZMBQVrjDEVJzyTQkHpqF3JzyYsXpzCiBGJrFmzi7Vrd/Puu5cDWEIwxoSs8EsKuZmQXHLpKC0tmwcf/JoXX1yCKrRvX4+bb+5WgUEaY4w3wi8pbPoSsg9Cw9Ogdts/vf3VV8mMHDmTjRv3ERkp3HtvD8aNO5/Y2PD7VRljwk/4nelKaHX022976NPnHVShS5fGTJkykK5dm1RwgMYY453wSgq5WbAh0Zkupq+j9u3rcdddZ9CgQTXuvbcH0dH2MJoxJryEV1L4/UvIPgANOkOdduzYcYg77/yCW27pxgUXOK2Qnn++n8dBGmOMd1ztnEdE+onIOhFJEpH7ink/RkSm+d5fIiLxbsbDb9MB0HaDeeedlXTsOIGPPlrN3XfPQVVd3bUxxgQD15KCiEQCrwIXAx2Bq0WkY5HVRgCpqnoC8DzwpFvxOKWjT9mcWov+Dzdk2LBP2Ls3g7592/LJJ0PsiWRjjMHd8lF3IElVkwFE5ENgELDGb51BwCO+6enAKyIi6sLX9vxNc5n4dXvGzu7Locxt1KkTy/PPX8SwYZ0tIRhjjI+bSaEZsMVvPgU442jrqGquiOwH6gG7yzuY/Stm8Ojc8ziUGc0VV5zEK69cQuPG1ct7N8YYE9TcTArFff0uegUQyDqIyChgFEDLlsc2jnKdatlMvmoW2ee8wBXD+xzTNowxJtS5mRRSgBZ+882BP46yToqIRAG1gL1FN6Sqk4BJAAkJCcdWWrrkXQb0OgAxNY/p48YYEw7cbH20FGgnIq1FpAowBEgssk4icL1vejDwtRv3EwpZQjDGmBK5dqXgu0cwGpgDRAJTVXW1iDwGLFPVRGAK8I6IJOFcIQxxKx5jjDGlc/XhNVWdBcwqsuxhv+lM4NgHSDbGGFOubGR5Y4wxhSwpGGOMKWRJwRhjTCFLCsYYYwpZUjDGGFNIgq13UBHZBfx+jB+vjwtdaFRydszhwY45PBzPMbdS1QalrRR0SeF4iMgyVU3wOo6KZMccHuyYw0NFHLOVj4wxxhSypGCMMaZQuCWFSV4H4AE75vBgxxweXD/msLqnYIwxpmThdqVgjDGmBCGZFESkn4isE5EkEbmvmPdjRGSa7/0lIhJf8VGWrwCO+e8iskZEfhaRr0SklRdxlqfSjtlvvcEioiIS9C1VAjlmEbnS92+9WkTer+gYy1sA/7dbisg8EVnh+/99iRdxlhcRmSoiO0Xkl6O8LyLyku/38bOIdC3XAFQ1pH5wuuneALQBqgArgY5F1rkNmOibHgJM8zruCjjmC4A43/St4XDMvvVqAAuAxUCC13FXwL9zO2AFUMc339DruCvgmCcBt/qmOwKbvI77OI+5J9AV+OUo718CzMYZufJMYEl57j8UrxS6A0mqmqyq2cCHwKAi6wwC3vJNTwd6iUhxQ4MGi1KPWVXnqWq6b3Yxzkh4wSyQf2eA/wOeAjIrMjiXBHLMI4FXVTUVQFV3VnCM5S2QY1agYAStWvx5hMegoqoLKGYESj+DgLfVsRioLSJNymv/oZgUmgFb/OZTfMuKXUdVc4H9QL0Kic4dgRyzvxE43zSCWanHLCKnAS1U9bOKDMxFgfw7twfai8hCEVksIv0qLDp3BHLMjwDXikgKzvgtd1RMaJ4p6997mbg6yI5HivvGX7SJVSDrBJOAj0dErgUSgPNcjch9JR6ziEQAzwPDKyqgChDIv3MUTgnpfJyrwW9FpJOq7nM5NrcEcsxXA2+q6rMichbOaI6dVDXf/fA84er5KxSvFFKAFn7zzfnz5WThOiIShXPJWdLlWmUXyDEjIr2BB4CBqppVQbG5pbRjrgF0AuaLyCac2mtikN9sDvT/9qeqmqOqG4F1OEkiWAVyzCOAjwBUdREQi9NHUKgK6O/9WIViUlgKtBOR1iJSBedGcmKRdRKB633Tg4Gv1XcHJ0iVesy+UsrrOAkh2OvMUMoxq+p+Va2vqvGqGo9zH2Wgqi7zJtxyEcj/7U9wGhUgIvVxyknJFRpl+QrkmDcDvQBE5CScpLCrQqOsWInAMF8rpDOB/aq6rbw2HnLlI1XNFZHRwByclgtTVXW1iDwGLFPVRGAKziVmEs4VwhDvIj5+AR7z00B14D++e+qbVXWgZ0EfpwCPOaQEeMxzgL4isgbIA+5V1T3eRX18Ajzme4A3RORunDLK8GD+kiciH+CU/+r77pOMA6IBVHUizn2TS4AkIB24oVz3H8S/O2OMMeUsFMtHxhhjjpElBWOMMYUsKRhjjClkScEYY0whSwrGGGMKWVIwlY6I5InIT34/8SWsG3+03iTLuM/5vp44V/q6iOhwDNu4RUSG+aaHi0hTv/cmi0jHco5zqYh0CeAzfxORuOPdtwkPlhRMZZShql38fjZV0H6HqmpnnM4Sny7rh1V1oqq+7ZsdDjT1e+8mVV1TLlEejnMCgcX5N8CSggmIJQUTFHxXBN+KyHLfT49i1jlZRH7wXV38LCLtfMuv9Vv+uohElrK7BcAJvs/28vXTv8rXz32Mb/l4OTw+xTO+ZY+IyBgRGYzTv9R7vn1W9X3DTxCRW0XkKb+Yh4vIy8cY5yL8OkITkddEZJk44yg86lt2J05ymici83zL+orIIt/v8T8iUr2U/ZgwYknBVEZV/UpHH/uW7QT6qGpX4CrgpWI+dwvwoqp2wTkpp/i6PbgKONu3PA8YWsr+BwCrRCQWeBO4SlVPwekB4FYRqQv8BThZVU8FHvf/sKpOB5bhfKPvoqoZfm9PBy73m78KmHaMcfbD6daiwAOqmgCcCpwnIqeq6ks4/eJcoKoX+Lq+eBDo7ftdLgP+Xsp+TBgJuW4uTEjI8J0Y/UUDr/hq6Hk4ffoUtQh4QESaAzNUdb2I9AK6AUt93XtUxUkwxXlPRDKATTjdL3cANqrqb7733wJuB17BGZ9hsoh8DgTcNbeq7hKRZF+fNet9+1jo225Z4qyG0+2D/6hbV4rIKJy/6yY4A878XOSzZ/qWL/TtpwrO780YwJKCCR53AzuAzjhXuH8aNEdV3xeRJUB/YI6I3ITTzfBbqnp/APsY6t9hnogUO8aGrz+e7jidsA0BRgMXluFYpgFXAr8CH6uqinOGDjhOnBHIxgOvApeLSGtgDHC6qqaKyJs4HcMVJcBcVb26DPGaMGLlIxMsagHbfH3kX4fzLfkIItIGSPaVTBJxyihfAYNFpKFvnboS+PjUvwLxInKCb/464BtfDb6Wqs7CuYlbXAuggzjddxdnBnAZzjgA03zLyhSnqubglIHO9JWeagJpwH4RaQRcfJRYFgNnFxyTiMSJSHFXXSZMWVIwwWICcL2ILMYpHaUVs85VwC8i8hNwIs6QhWtwTp5fisjPwFyc0kqpVDUTpwfK/4jIKiAfmIhzgv3Mt71vcK5iinoTmFhwo7nIdlOBNUArVf3Bt6zMcfruVTwLjFHVlThjM68GpuKUpApMAmaLyDxV3YXTMuoD334W4/yujAGsl1RjjDF+7ErBGGNMIUsKxhhjCllSMMYYU8iSgjHGmEKWFIwxxhSypGCMMaaQJQVjjDGFLCkYY4wp9P8BoBBQVzFoMrUAAAAASUVORK5CYII=\n",
        "text/plain": "<Figure size 432x288 with 1 Axes>"
       },
       "metadata": {
        "needs_background": "light"
       },
       "output_type": "display_data"
      }
     ],
     "persistentId": "8f75a628-0448-474b-8eba-ccfae9ffc601",
     "text": "fpr, tpr, thresholds = metrics.roc_curve(y_true, y_pred)\n\nfig, ax = plt.subplots()\nlw = 2\nax.plot(fpr, tpr, lw=2, color='darkorange')\nax.plot([0, 1], [0, 1], color='navy', lw=lw, linestyle='--')\nax.set(xlabel='False Positive Rate', ylabel='True Positive Rate')"
    },
    "executionTime": "2019-10-12T00:53:35.894Z"
   },
   {
    "cell": {
     "executionCount": 35,
     "executionEventId": "e262d09b-7737-4c4d-8613-87aa1b22e290",
     "hasError": false,
     "id": "c644f8a4-dc14-489f-80fb-21d9a39c084f",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "0.8080168776371308\n"
      }
     ],
     "persistentId": "91fecc99-2e19-42e2-9fb8-a694ea075d8f",
     "text": "print(metrics.accuracy_score(y_true, y_pred > 0.9))"
    },
    "executionTime": "2019-10-12T00:53:37.074Z"
   },
   {
    "cell": {
     "executionCount": 36,
     "executionEventId": "5587602d-80b5-4d89-805d-e8355f1cb63d",
     "hasError": false,
     "id": "1e39d7f6-23b4-4c86-b1a9-894c2d9a8a53",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "              precision    recall  f1-score   support\n\n       False       0.87      0.84      0.85       318\n        True       0.70      0.74      0.72       156\n\n   micro avg       0.81      0.81      0.81       474\n   macro avg       0.78      0.79      0.79       474\nweighted avg       0.81      0.81      0.81       474\n\n"
      }
     ],
     "persistentId": "e0cd2f37-754e-4062-92c6-e5a3223e4c7c",
     "text": "print(metrics.classification_report(y_true, y_pred > 0.9))"
    },
    "executionTime": "2019-10-12T00:53:38.863Z"
   },
   {
    "cell": {
     "executionCount": 38,
     "executionEventId": "8d6d8690-163b-4024-98ba-27eefaa2b5cd",
     "hasError": false,
     "id": "4c484434-ea07-4b72-8073-567bf902bedb",
     "outputs": [
      {
       "data": {
        "text/plain": "132241"
       },
       "execution_count": 38,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "75c01782-427c-4c63-bc29-ed308ae4a142",
     "text": "true_coding_read_ids = set(coding_read_ids.values)\nlen(true_coding_read_ids)"
    },
    "executionTime": "2019-10-12T01:01:25.756Z"
   },
   {
    "cell": {
     "executionCount": 39,
     "executionEventId": "a1813ecc-e64f-44e9-a7d7-85b57bd444d4",
     "hasError": false,
     "id": "9babb580-ed7e-4a80-bcb3-3d080fca4a29",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "(807698,)\n"
      }
     ],
     "persistentId": "317bb484-c93f-4e1c-b831-50734c9e6614",
     "text": "unspliced_read_ids_series = pd.read_csv(f\"{folder}/SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.read_ids.txt\", \n                                   header=None, squeeze=True)\nprint(unspliced_read_ids_series.shape)\nunspliced_read_ids_series.head()\n\nunspliced_reads = set(unspliced_read_ids_series.values)"
    },
    "executionTime": "2019-10-12T01:01:27.280Z"
   },
   {
    "cell": {
     "executionCount": 41,
     "executionEventId": "97873c1c-e768-4368-a56e-b507c34a9c22",
     "hasError": false,
     "id": "96e03c57-735e-4d8a-b8d0-e29cd2b603de",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "0.8080168776371308\n              precision    recall  f1-score   support\n\n       False       0.87      0.84      0.85       318\n        True       0.70      0.74      0.72       156\n\n   micro avg       0.81      0.81      0.81       474\n   macro avg       0.78      0.79      0.79       474\nweighted avg       0.81      0.81      0.81       474\n\n"
      },
      {
       "data": {
        "image/png": "iVBORw0KGgoAAAANSUhEUgAAAYUAAAEKCAYAAAD9xUlFAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDMuMC4zLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvnQurowAAIABJREFUeJzt3Xl4FFX28PHvyUJC2PcdAggooiBEVFRcWEQRcJRRFEUUwQ11HHHQ1wX15zi474gIjLviMKhBQWQURBEQBBEBkRAQguyELfty3j+qE5oYkg6kUunu83mePF1VXV11KpA6Xadu3SuqijHGGAMQ4XUAxhhjKg9LCsYYYwpZUjDGGFPIkoIxxphClhSMMcYUsqRgjDGmkCUFY4wxhSwpGGOMKWRJwRhjTKEorwMoq/r162t8fLzXYRhjTFD58ccfd6tqg9LWC7qkEB8fz7Jly7wOwxhjgoqI/B7IelY+MsYYU8iSgjHGmEKWFIwxxhSypGCMMaaQJQVjjDGFXEsKIjJVRHaKyC9HeV9E5CURSRKRn0Wkq1uxGGOMCYybVwpvAv1KeP9ioJ3vZxTwmouxGGOMCYBrzymo6gIRiS9hlUHA2+qMB7pYRGqLSBNV3eZWTMYYUyHysiFrP2Tt873uh+z9h6eLm8/LOurmfkiqSWx0Pqee2x16POJq6F4+vNYM2OI3n+Jb9qekICKjcK4maNmyZYUEZ4wJU4Un9KIn7gBP8Nn7ITezXEJRhbGf9+HZb7pxapMd/NDhV6LLZctH52VSkGKWaXErquokYBJAQkJCsesYYwwAeTmQsQvSd0Lm3qOfyAtO8m6c0COioEotiPH7KW0+MhbkyNOiAGxIggVb6DvoDPJOPz2kk0IK0MJvvjnwh0exGGMqK1XIPuCc5NN3QvoOv+lilmXuPb79HcsJvUotiKl9eD6q6p9O8IHaty+T5ORUunZtAsCjz53FkJG7C+fd5mVSSARGi8iHwBnAfrufYEyYKPg2n7YDMko4wRf8lFBv/xOJgKoNIK4hVK1XhhN87eM+oR+vTz/9lVtv/ZyICGH16tuoVSuWqlWjKywhgItJQUQ+AM4H6otICjAOnCsfVZ0IzAIuAZKAdOAGt2IxxrhM1Sm/FJzEC070aTv+PJ+xEzJTy7b96OrOST6uke+14dHnq9Z1EkMQ2bkzjTvvnM20aasBOPPM5uzbl0mtWrEVHoubrY+uLuV9BW53a//GmOOUlw3puw6f1NN2HHnSLzqflx34tiUS4gq+zftO6NUaHZ4umI9r6Hzrj45z7zg9pKq8994q7rrrC/buzSAuLponnriQ0aO7ExnpTWILuq6zjTHHKD/PubmasbvkmnzBfNa+sm2/So3DJ/lqft/gqxY9yTcMym/zbrj11s95/fUfAejduw2TJl1K69Z1PI3JkoIxwSgvBzL3QMYe5yTvP52xxzdfZDozlaM08Cte4bf5IiWaoif9gmXRVV073FB12WUnMm3aap59ti833NAF8ehehj9LCsZ4LSej+JN4cSf8gunsA8e2r9g6EFuv9Lp8XENnXfs2X67Wr9/DV19t5JZbEgDo1+8ENm26y5N7B0djScGY8pZ9qEhJZoffDdddfz7552aUfR8S4Zzcq9bzvdZ3pqvWP7z8T9N1nOaWpsLl5ubz3HOLGDduPllZuXTp0pgzz2wOUKkSAlhSMKZkO5bD3l+PXKb5TinG/6RfMJ22A3LTy7aPiOgAT+p+J/+YWvYtPkisXLmdESMS+fFHp8X9sGGdadeursdRHZ0lBWOKys2EtG2w5F+w6o2yfz4yxinBFNbd/V8b/PkEH13ds3bxxj1ZWbk8/vgCxo9fSG5uPi1b1uL11y+lX78TvA6tRJYUTHhRdVrVHPgdDmz2vf4OB/3m03ccXj8iGtoOhMgqR24nps6RLWr8T/xVathJ3nD//V/x/POLAbj99tP51796UaNGjMdRlc6SggltORmQ8g1snO287k+G7IMlfyYiynm6tX4nuPBl59WYMvrHP85m0aIUnnqqN+ee28rrcAJmScGEFlXYl+QkgY2zIWX+nzs4i64GNVsd/qnR8sj5ak0gItKT8E3wmjt3AxMn/si0aYOJioqgcePqfP/9jZWimWlZWFIwwS8nDbbMP5wI9icf+X7DrtD6YojvC/VOhti6Vt4x5SY1NYMxY75k6tSfAPj3v1cwcmQ3gKBLCGBJwVQmmu/8BCJ1PWz6wnc1sODIDtNi60Krvr5EcJFT9zfGBR9/vJbbbpvF9u2HiImJZNy48xg+vIvXYR0XSwqm4uVmOs0896yBPat9r2ucsk+gSeEIAo27Q3w/JxE0Pt3KP8ZV27cf4o47ZjN9+hoAevRowZQpAznxxPoeR3b8LCmEo01zYemTkJ9bwTtWOPSHU9452slfAjyZV63nuxro57zGNSi/MI0pxaef/sr06WuoVi2a8eN7c9ttpxMREXylouJYUghHK16GzV95t3+JhDodoF5HqH8y1O3oTNftAFGV6+lOYwpkZuYSG+ucMkeO7EZyciq33no68fG1PY6sfFlSCAd71x3Z42XGbuc1YQy06V+xscTWgzrtIaryt9c2BiA/X5kwYSn//Oe3LF48glatahMRITz5ZB+vQ3OFJYVQ992DsOSfxb/X8kJocX6FhmNMMFm3bjcjRiSycOEWAD744Bfuu+8cj6NylyWFULbhMychSCQ06opvGHBH9abQLLT/cxtzrHJy8njmme959NFvyMrKo1GjakyY0J/LLz/J69BcZ0khVB34Hb4Y5kyf80/oPtbbeIwJEr/8spNhwz5mxYrtANxwQxeefbYvdeqEx3gRlhRCUV42fHaV05Nnm/5w+r1eR2RM0MjPV1at2kmrVrWYNGkAffu29TqkCmVJIRQtGAvblkCNFtDvLeti2ZhSrF69k44dGyAinHpqIz79dAg9e7aievUqpX84xNjZItTsWA7LX3A6dbv0I6c9vzGmWAcPZjF69Cw6dXqN//53beHySy5pF5YJAexKIfSk/ua8tr4Emp7pbSzGVGJz5iQxatRnbN68n6ioCDZt2lf6h8KAJYVQkpsJix93puue6G0sxlRSe/dmcPfdc3j77ZUAdO3ahClTBtKlS2OPI6scLCmEkj++d/oSqhkPZz7kdTTGVDo//bSdfv3eZceONGJiInn00fO5554eREVZJb2AJYVQUvCkcr2ToEp1b2MxphJq374e1atXoX37ekyePJD27e2eW1GWFELJuo+c12Y9vY3DmEpCVXn//VUMGNCBmjVjiIuLZv784TRtWiNkOrArb3bNFCoy9sCGRKf56cnDvI7GGM9t2rSPiy56l2uv/Zj77vtf4fLmzWtaQiiBXSmEij1rID/HGUugelOvozHGM3l5+UyYsJT77/+KtLQc6tatSo8eLbwOK2hYUgg1kdb7qAlfa9fuYsSIRBYtSgHgyitP5uWXL6Zhw2oeRxY8LCkYY0LCxo2pdOnyOtnZeTRpUp0JE/pz2WXWNLusXE0KItIPeBGIBCar6vgi77cE3gJq+9a5T1VnuRmTMSY0tW5dh7/+tSOxsVE880xfate2AZuOhWtJQUQigVeBPkAKsFREElV1jd9qDwIfqeprItIRmAXEuxWTMSZ0ZGTk8Nhj3/CXv5xE9+7NAHjrrcuIjLT2M8fDzd9edyBJVZNVNRv4EBhUZB0FavqmawF/uBiPMSZEfPvt73Tp8jrjxy9k1KiZ5OcrgCWEcuBm+agZsMVvPgU4o8g6jwBfisgdQDWgt4vxhLYt853XqPDo892EpwMHsrj//v8xYcIyADp2bMDEiZdaE9Ny5GZaLe5fSYvMXw28qarNgUuAd0T+3M+ziIwSkWUismzXrl0uhBrkDm2DRY84093u9jQUY9wya9Z6OnWawIQJy4iKiuDhh3uyfPkozjyzudehhRQ3rxRSAP/Gwc35c3loBNAPQFUXiUgsUB/Y6b+Sqk4CJgEkJCQUTSwmbRtoPjToDK0v9joaY8rd/v2ZDB06g337MklIaMqUKQM59dRGXocVktxMCkuBdiLSGtgKDAGuKbLOZqAX8KaInATEAnYpUBaq8P3DzrQNpmNCiKqiChERQq1asbz0Uj927Ejjb3870zqwc5FrSUFVc0VkNDAHp7npVFVdLSKPActUNRG4B3hDRO7GKS0NV1W7EiiL/Rsh+XNnupp1/WtCwx9/HOS22z7n3HNbcs89PQC47rrOHkcVHlx9TsH3zMGsIsse9pteA5ztZgwhLyv18PTF73gXhzHlQFWZOnUF99zzJfv3Z7F4cQq33XY6VatGex1a2LAnmoNZxl54N8GZrtPBht40QS05OZWRI2fy9dcbAejfvx0TJ15qCaGCWVIIZr++f3j61FHexWHMccjLy+ell5bwwANfk5GRS/36cbz0Uj+GDOmEiDU1rWiWFILRuv/A71/Cpi+d+f4fwIlDvI3JmOMwffpaMjJyufrqTrz4Yj8aNLAO7LxiSSFYpG2HlAWwLxm+u//w8th6cMJl3sVlzDHIzs7j4MEs6tWLIzIygilTBrJ+/R4GDOjgdWhhz5JCsJj5V9j63eH50+6A+qdA0x4QZR1/meCxdOlWRoxIpHnzmnz++TWICCeeWJ8TT6zvdWgGSwrBI32H89r6YmjVF7reBVZvNUEkPT2HcePm8dxzi8nPV9LTc9i5M41GjWw88crEkkKwOf8FqNve6yiMKZP58zcxcuRMkpL2EhEhjBlzFo8+egFxcdayqLKxpGCMcY2qcueds3nllaUAnHJKQ6ZMGcjppzfzODJzNJYUgkF+HmTscaajbLhNEzxEhJo1Y4iOjuDBB3ty333nUKVKpNdhmRJYUggGKQsgcy/Ubgs1WnodjTEl2r07nQ0b9nLGGU7vpQ89dB5Dh55Kx44NPI7MBCKgXqVEpIqInOB2MOYofpvuvLb/q91cNpWWqvLhh79w0kmvctll00hNzQAgNjbKEkIQKTUpiEh/YBUw1zffRUQ+djsw45OfB+v/60y3/6u3sRhzFCkpBxg06EOuvvq/7N6dTseODUhPz/E6LHMMAikfPYYzYto8AFX9ya4aKtDW75zmqLXaQMPTvI7GmCPk5yuTJy/n3nvncuBAFjVrxvDss30ZMeI066IiSAWSFHJUdV+Rf2Dr3rqi/PYf59VKR6YSGjEikTff/AmAgQM7MGHCJTRrVrOUT5nKLJB7CmtF5EogQkRai8gLwGKX4zJQpHQ02NtYjCnGtdeeQsOG1fjwwyv45JOrLCGEgECSwmigG5APzAAygbvcDMr4/PG90+dRzXho1M3raIzhl1928uKLh78T9urVhuTkO7nqKuvRNFQEUj66SFXHAmMLFojI5TgJwrjJSkemksjKyuVf//qOJ574lpycfBISmnL22U7z6GrVqngcnSlPgVwpPFjMsgfKOxBThOYfLh11sFZHxjtLlqTQrdskHn30G3Jy8rn11gROOaWR12EZlxz1SkFELgL6Ac1E5Dm/t2rilJKMm7Z+D4f+gJqtoFGC19GYMJSWls1DD83jhRcWowrt2tVl8uSB9OzZyuvQjItKKh/tBH7BuYew2m/5QeA+N4MyHC4dtRtspSPjiQce+JoXX1xCRIRw771n8cgj59vQmGHgqElBVVcAK0TkPVXNrMCYjJWOTCXwwAPnsmrVTp58sjcJCU29DsdUkEDuKTQTkQ9F5GcR+a3gx/XIwtkfi+HQVqefo8bdvY7GhInExHVccsl75OTkAdCgQTW++mqYJYQwE0hSeBP4NyDAxcBHwIcuxmQKWx1Z6ci4b+fONIYMmc6gQR8ye3YSb7210uuQjIcCSQpxqjoHQFU3qOqDwAXuhhXGNP/IDvCMcYmq8u67P3PSSa8ybdpq4uKiefHFftxwQxevQzMeCuQ5hSxxnkrZICK3AFuBhu6GFca2LYFDKVC9OTSx0pFxx+bN+7nlls+YPTsJgN692zBp0qW0bl3H48iM1wJJCncD1YE7gX8CtYAb3QwqrBVeJQwGCahnc2PK7MsvNzB7dhK1a8fy3HN9GT68iz2RbIAAkoKqLvFNHgSuAxCR5m4GFbZUrXRkXJOWll349PGIEaexdesBRo3qRpMmNTyOzFQmJX4VFZHTReQyEanvmz9ZRN7GOsRzx/Yf4OBmqN4Mmp7pdTQmROTm5vPUUwtp1eoFkpNTAWeYzHHjzreEYP7kqElBRP4FvAcMBb4QkQdwxlRYCbSvmPDCzDr/VkdWOjLHb+XK7ZxxxmTGjv0fe/Zk8Mknv3odkqnkSiofDQI6q2qGiNQF/vDNr6uY0MKMKqy30pEpH1lZuTz++ALGj19Ibm4+LVvWYtKkS7noIhsfy5SspKSQqaoZAKq6V0R+tYTgou1L4cDvUL0pND3L62hMEFuxYhtDh85g7drdiMDo0afzxBO9qFEjxuvQTBAoKSm0EZGC7rEFiPebR1UvL23jItIPeBGIBCar6vhi1rkSeARnNLeVqnpN4OGHkIIbzO2usNKROS4xMVFs2JBKhw71mDx5IOec09LrkEwQKSkpXFFk/pWybFhEIoFXgT5ACrBURBJVdY3fOu2A+4GzVTVVRMLz+QfVI8dOMKaMli/fxmmnNUZE6NixAbNnD6VHjxbExgbS6tyYw0rqEO+r49x2dyBJVZMBRORDnPsUa/zWGQm8qqqpvn3uPM59BqcdP8KBTVCtCTQ72+toTBBJTc1gzJgvmTr1Jz744AqGDOkEwIUXtvY4MhOs3Pwa0QzY4jefApxRZJ32ACKyEKfE9IiqflF0QyIyChgF0LJlCF4KF3aTbaUjE7iPP17LbbfNYvv2Q8TERLJnT7rXIZkQ4GZSKO7xSC1m/+2A84HmwLci0klV9x3xIdVJwCSAhISEotsIbkeUjgZ7G4sJCtu3H+KOO2Yzfbpz0X322S2YPHkgJ55Y3+PITCgIOCmISIyqZpVh2ylAC7/55jjNWouus1hVc4CNIrIOJ0ksLcN+gtvOFbB/I8Q1gmbneB2NqeR+/PEP+vR5h9TUTKpVi2b8+N7cdtvpRERYFxWmfJRaqxCR7iKyCljvm+8sIi8HsO2lQDsRaS0iVYAhQGKRdT7B1+Oq76np9kByGeIPfv6lo4hIb2MxlV7Hjg1o0KAaF13UltWrb2P06O6WEEy5CqSA/RJwKbAHQFVXEkDX2aqaC4wG5gBrgY9UdbWIPCYiA32rzQH2iMganKel71XVPWU/jCDlXzqyEdZMMfLzlUmTfmTfPmfww6pVo1mwYDizZw+lVavaHkdnQlEg5aMIVf29SA+KeYFsXFVnAbOKLHvYb1qBv/t+ws/On2DfBohrCM3O9ToaU8msW7ebm26ayXffbWbp0q288YbzXapRo+oeR2ZCWSBJYYuIdAfU9+zBHYANx1keCktHl1vpyBTKycnj2WcX8cgj88nKyqNx4+pcfHE7r8MyYSKQpHArTgmpJbAD+J9vmTke9sCaKcaKFdsYMSKRFSu2A3DDDV149tm+1KlT1ePITLgIJCnkquoQ1yMJN7t+hn1JULUBNO/pdTSmEtiwYS/du08mNzef+PjaTJp0KX36tPU6LBNmAkkKS31NRacBM1T1oMsxhYcjSkfWFYGBtm3rct11p1KjRhX++c9eVK9exeuQTBgqtfWRqrYFHge6AatE5BMRsSuH42GlIwMcOpTNnXfOZtGiww/+T5kykBdfvNgSgvFMQH0qqOr3qnon0BU4gDP4jjlWu1dB6m9QtT60OM/raIwH5sxJ4uSTJ/Dyyz9wyy2f4zTEw8ZJNp4rtW4hItVxOrIbApwEfAr0cDmu0FbQTfYJf7HSUZjZuzeDu++ew9tvrwSgW7cmTJky0JKBqTQCOSP9AswEnlLVb12OJ/RZ6ShsTZ++httvn8XOnWnExkbx6KPn8/e/n0VUlHWCaCqPQJJCG1XNdz2ScLFnNez9FWLrQctSHww3IWLfvkxGjZpJamomPXu24o03BtC+fT2vwzLmT46aFETkWVW9B/iviPypZ9JARl4zxVhX0OrISkehTlXJz1ciIyOoXTuWCRP6k5qawc03J1h/RabSKumsNM33WqYR10wprJvssLBp0z5GjZrJhRe25r77nN5vCwbAMaYyO2oxU1V/8E2epKpf+f/g3HA2ZbVnDexdC7F1ocWFXkdjXJCXl89LLy2hU6cJzJ2bzCuv/EBmZq7XYRkTsEDucN1YzLIR5R1IWCgoHZ1wGURGexuLKXdr1+6iZ883ueuuL0hLy2HIkE4sX36zjZNsgkpJ9xSuwmmG2lpEZvi9VQPYV/ynTIms1VFIys3N58knv+OxxxaQnZ1H06Y1eO21/gwc2MHr0Iwps5K+wvyAM4ZCc+BVv+UHgRVuBhWS9qx1Wh7F1oGWvbyOxpSjiAjhyy+Tyc7OY+TIrjz1VB9q1471OixjjslRk4KqbgQ24vSKao5XwVVC20FWOgoBGRk5HDyYTcOG1YiIECZPHsCWLQe48MLWXodmzHE56j0FEfnG95oqInv9flJFZG/FhRgiCp5ittJR0Fuw4Hc6d57ItdfOKOyeol27epYQTEgoqXxU8GRV/YoIJKTtXef0dxRTG1r19joac4wOHMji/vv/x4QJywCIjo5k9+50GjSo5nFkxpSfkpqkFjzF3AKIVNU84CzgZsD+CsqioHR0wiCItN4vg9Hs2evp1GkCEyYsIyoqgnHjzmP58lGWEEzICaSt3CfA6SLSFngb+Bx4H7jUzcBCirU6ClqqysiRM5kyxWlbkZDQlKlTB3LKKY08jswYdwTynEK+quYAlwMvqOodQDN3wwohe39zRlmrUhNaWuko2IgIzZvXJDY2imee6cOiRSMsIZiQFtBwnCLyV+A64DLfMms+E6j1Bd1kD4KoGG9jMQH544+DbNiwl3PPbQXA//t/53LddafStm1djyMzxn2BPtF8AU7X2cki0hr4wN2wQsg6Kx0FC1VlypTldOz4Kldc8RF79qQDUKVKpCUEEzZKvVJQ1V9E5E7gBBE5EUhS1X+6H1oISE2CXT85paNWfb2OxpQgOTmVkSNn8vXXGwG49NL25ORYj/Em/AQy8tq5wDvAVkCAxiJynaoudDu4oFf4wNpAKx1VUgUd2D344DzS03OoXz+Ol17qx5AhnWw0NBOWArmn8DxwiaquARCRk3CSRIKbgYUEa3VU6Q0b9gnvv78KgGuuOYUXXrjImpmasBbIPYUqBQkBQFXXAtbYvjT7NsDOFVClBsRb6aiyGjmyK82b1yQxcQjvvXe5JQQT9gK5UlguIq/jXB0ADMU6xCtdQbcWbQZAlHWOVlksXbqVr7/eyNixzsA3558fT1LSHcTEWPfWxkBgSeEW4E7gHzj3FBYAL7sZVEiw0lGlkp6ew7hx83juucXk5ys9erQobHJqCcGYw0r8axCRU4C2wMeq+lTFhBQC9iXDjh8hujrEX+R1NGFv/vxN3HRTIhs2pBIRIYwZcxbdujX1OixjKqWSekn9fzhdXAwF5opIcSOwlUhE+onIOhFJEpH7SlhvsIioiITGzeuC0lHbARBd1dtYwtj+/ZncfPNMLrjgLTZsSOWUUxqyePEInn66L3Fx9vylMcUp6UphKHCqqqaJSANgFjA10A2LSCTO4Dx9gBRgqYgk+t+09q1XA6c8taSswVdahaWjwd7GEeYeemgekyYtJzo6goce6snYsedQpUqk12EZU6mVlBSyVDUNQFV3iUggLZX8dcd50C0ZQEQ+BAYBa4qs93/AU8CYMm6/ctq/CXYsg+hqEH+x19GEHVUtfL7g4YfPY+PGfYwf34uTT27ocWTGBIeSTvRtRGSG7+djoK3f/IwSPlegGbDFbz6FIh3pichpQAtV/azMkVdWha2OLrXSUQVSVd5/fxUXXvg22dl5ANSvH8fMmVdbQjCmDEq6UriiyPwrZdx2cY+DauGbzpXH88DwUjckMgoYBdCyZcsyhlHBrNVRhUtJOcCtt37OZ5/9BsB77/3MDTec5nFUxgSnksZo/uo4t52CM0BPgebAH37zNYBOwHzf5X5jIFFEBqrqsiKxTAImASQkJCiV1YHfYfsPEBUHra105Lb8fOWNN37k3nvncvBgNrVqxfDss30ZPryL16EZE7TcbKC9FGjn61V1KzAEuKbgTVXdj99QnyIyHxhTNCEElcLSUX+IjvM2lhCXlLSXkSNnMn/+JgAGDerAhAn9adq0hreBGRPkXEsKqporIqOBOUAkMFVVV4vIY8AyVU10a9+eKUgKVjpy3bff/s78+Zto2LAar7xyMYMHd7QO7IwpBwEnBRGJUdWssmxcVWfhNGX1X/bwUdY9vyzbrnQObIFtiyGqKrS5xOtoQtK+fZnUru10GTJ8eBd27UpnxIjTqFfPrsqMKS+lNjMVke4isgpY75vvLCLWzUVR6/1LR9apWnnKyspl3Lh5tGr1AuvX7wGcYTL/8Y+zLSEYU84CuVJ4CbgU5+lmVHWliFzgalTByEZYc8XixSmMGJHImjW7AJgzZwPt2tXzOCpjQlcgSSFCVX8vUq/Ncyme4HRgC2xb5PSG2tpKR+UhLS2bhx6axwsvLEYV2rWry5QpAws7sTPGuCOQpLBFRLoD6uu64g7gN3fDCjJJvmf5Wl8CVap7G0sIWLIkhWuumUFyciqRkcKYMT0YN+48qla1/oqMcVsgSeFWnBJSS2AH8D/fMlPASkflqnbtWLZuPUDnzo2YMmWg9WhqTAUqNSmo6k6cZwxMcQ5uhT8WOqWjNpd6HU3Q+u67zZx9dgtEhA4d6vP119dz+ulNiY62DuyMqUilJgUReQO/7ikKqOooVyIKNuv/67zGX2ylo2Owc2cad945m2nTVvPWW5cxbFhnAHr0aFHKJ40xbgikfPQ/v+lY4C8c2dFdeLNuso+JqvLee6u4664v2Ls3g7i46MKO7Iwx3gmkfDTNf15E3gHmuhZRMDn0B2xdCJExzoA6JiCbN+/nlls+Y/bsJAD69GnDpEkDiI+v7XFkxphj6eaiNWDtAgHWzwAU4vtBFetzJxBLlqTQu/c7HDqUTe3asTz//EVcf31n66LCmEoikHsKqRy+pxAB7AWOOrRmWCkoHXWwVkeB6tKlMS1a1OTN3a/iAAAULUlEQVTEE+vz6quX0KSJJVNjKpMSk4I4X9864/RyCpCvqpW36+qKdGgbpHwLkVWgjZWOjiY3N59XXvmBYcM6U7duVWJioli48Ebq1LEBiIypjErs+8iXAD5W1TzfjyWEAgWlo1YXQUxNr6OplFau3M4ZZ0zm7rvn8Pe/zylcbgnBmMorkHGXfxCRrq5HEmwKOsCz0tGfZGbm8uCDX5OQ8AbLl2+jZctaXH11J6/DMsYE4KjlIxGJUtVc4BxgpIhsANJwhtlUVQ3fRJG2A1IWOKWjtgO9jqZS+f77LYwYkcivv+5GBEaPPp0nnuhFjRoxXodmjAlASfcUfgC6ApdVUCzBY/0M0Hxo1RdiankdTaWRlLSXc8/9N/n5SocO9ZgyZSBnn13Jx9Q2xhyhpKQgAKq6oYJiCR6/WV9HxTnhhLqMGtWVunWr8tBD5xEb6+Zor8YYN5T0V9tARP5+tDdV9TkX4qn80ndCyjcQER32paPU1AzuuedLbrihS2GX1hMm9LdnDowJYiUlhUigOr4rBuOz/mOndBTfD2LD9wncGTPWcvvts9i+/RA//riNn366GRGxhGBMkCspKWxT1ccqLJJgEealo+3bDzF69Cz++9+1AJxzTksmTx5gycCYEFHqPQXjJ30XbJnnlI5OGOR1NBVKVXn77ZXcffccUlMzqV69Ck8+2ZtbbkkgIsL+qxgTKkpKCr0qLIpgkVRQOroIYut4HU2F2rcvk3vu+ZLU1Ez69TuBiRP706pV+JbPjAlVR00Kqrq3IgMJCmE2wlp+vpKfr0RFRVCnTlVef/1S0tNzuPbaU61cZEyICuSJZgOQvttXOoqCtqFfOvr119307Plvxo//rnDZFVd05LrrrEdTY0KZJYVAJX0Cmgcte0HVul5H45qcnDyeeOJbOneeyMKFW5gyZQWZmbleh2WMqSD2dFGgwqDV0YoV27jxxkR++mk7ACNGnMbTT/exh9CMCSP21x6IjD2w+SuQSDgh9Hr9yMnJY9y4+Tz11ELy8pT4+Nq88cYAevdu43VoxpgKZkkhEAWlo1Z9oWo9r6Mpd1FRESxZspX8fOWuu87g8ccvpHr1Kl6HZYzxgCWFQBSWjgZ7G0c5Ongwi4MHs2natAYiwuTJA9i+/RBnndXC69CMMR6yG82lydjrVzr6i9fRlIs5c5Lo1Ok1hg6dQcG4Sa1b17GEYIyxpFCqDZ9Cfi60uADi6nsdzXHZsyed66//hH793mPz5v0cPJjFnj0ZXodljKlEXE0KItJPRNaJSJKI3FfM+38XkTUi8rOIfCUirdyM55gUlI6CeIQ1VWX69DV07DiBt99eSWxsFE891ZvFi2+ifv04r8MzxlQirt1TEJFI4FWgD5ACLBWRRFVd47faCiBBVdNF5FbgKeAqt2Iqs8xU+P1/IBFBWzpSVYYOncEHH/wCQM+erXjjjQG0bx96N8yNMcfPzSuF7kCSqiarajbwIXDEo8CqOk9V032zi4HmLsZTdkmfQn4OtDgf4hp4Hc0xERE6dmxAjRpVeO21/sybd70lBGPMUbnZ+qgZsMVvPgU4o4T1RwCzi3tDREYBowBatqzA4R3XT3deg+yBtY0bU0lOTqVXL+c5g7Fjz2b48C40b17T48iMMZWdm1cKxXWQo8WuKHItkAA8Xdz7qjpJVRNUNaFBgwr6xp65DzZ96ZSO2l1eMfs8Tnl5+bz44mI6dXqNq66azs6daQBER0daQjDGBMTNK4UUwL+NY3Pgj6IriUhv4AHgPFXNcjGestmQ6CsdXQBxDb2OplRr1uzippsSWbQoBYCBAzvYOAfGmDJzMyksBdqJSGtgKzAEuMZ/BRE5DXgd6KeqO12MpeyCpK+jnJw8nnxyIf/3fwvIzs6jadMavPZafwYO7OB1aMaYIORaUlDVXBEZDczBGe95qqquFpHHgGWqmohTLqoO/MfXHfNmVR3oVkwBy9oPv38JCLSr3K2OrrlmBtOnOw26Ro7sytNP96FWrViPozLGBCtXu7lQ1VnArCLLHvab7u3m/o/ZhpmQlw3Nz4Nqjb2OpkR33XUGP/20nddfv5QLL2ztdTjGmCBnTzQXpxKXjr75ZhOPPjq/cP6cc1qydu3tlhCMMeXCOsQrKusAbJoDCLS/wutoCh04kMXYsXOZOPFHAC64oDU9ezoPgEdFWW43xpQPSwpFJc+EvCxo3rPSlI5mzVrPzTd/RkrKAaKjI3jggXM588zK9ZyfMSY0WFIoap2vdNTO+26yd+9O529/+4L33lsFQPfuzZgyZSCdOlX+JrLGmOBkScFf9kHY9AWVpXT02GPf8N57q6haNYrHH7+Qu+46g8hIKxUZY9xjScHfhs+c0lGzc6B6U09CUFV8zXN59NHz2bEjjSeeuJC2bet6Eo8xJrzY105/HrY6UlXeeONHevSYSmZmLgB16lRl2rTBlhCMMRXGkkKB7EOwydcfX7uKLR1t2LCXXr3eZtSoz1i8OIWPPlpdofs3xpgCVj4qkPwZ5GZC0x5Qo1mF7NLpwG4JDz74NRkZuTRoEMfLL1/MlVeeXCH7N8aYoiwpFPitYrvJXr16JzfemMgPP2wFYOjQU3jhhX42EpoxxlOWFABy0mCjrzeO9hXTFHXFiu388MNWmjWrweuvX0r//u0rZL/GGFMSSwoAyZ9DbgY0OQtquPdQ2K5daTRoUA1wrgz27cvkuutOtQ7sjDGVht1ohsOtjjq4UzpKT89hzJgviY9/kbVrdwHOMJmjR3e3hGCMqVTsSiEnzblSAFdaHc2bt5GRI2eyYUMqERHCggW/c9JJwTneszEm9FlS2DjbVzo6A2qW3/jP+/dn8o9/zGXSpOUAnHJKQ6ZOHURCgjcPxRljTCAsKawr/wfWvvtuM0OGTGfr1oNER0fw0EM9GTv2HKpUiSy3fRhjjBvCOynkpDvPJ0C5tjpq3Lg6e/ZkcOaZzZk8eQAnn2wd2BljgkN4J4WNsyE3HRp3h5qtjnkzqsrcucn06dMGEeGEE+ry3Xc30KVLY+vAzhgTVML7jFUOfR1t2bKfAQM+4KKL3uXf//6pcHm3bk0tIRhjgk74XinkZPiVjsre6ig/3+nA7t5753LwYDa1asUQE2P3DIwxwS18k8KmL5zmqI0SoFbZxjdev34PI0fO5JtvfgfgsstO5NVXL6Fp0xpuRGqMMRUmfJPCMZaOvv9+C716vU1mZi4NG1bjlVcuZvDgjoVjIBhjTDALz6SQkwEbZjrTZXyKOSGhKe3a1eW005rw3HN9qVfPOrAzxoSO8EwKm+ZAziFo1K3U0lFWVi7PPPM9N9+cQP36cVSpEsnChTdSo0ZMBQVrjDEVJzyTQkHpqF3JzyYsXpzCiBGJrFmzi7Vrd/Puu5cDWEIwxoSs8EsKuZmQXHLpKC0tmwcf/JoXX1yCKrRvX4+bb+5WgUEaY4w3wi8pbPoSsg9Cw9Ogdts/vf3VV8mMHDmTjRv3ERkp3HtvD8aNO5/Y2PD7VRljwk/4nelKaHX022976NPnHVShS5fGTJkykK5dm1RwgMYY453wSgq5WbAh0Zkupq+j9u3rcdddZ9CgQTXuvbcH0dH2MJoxJryEV1L4/UvIPgANOkOdduzYcYg77/yCW27pxgUXOK2Qnn++n8dBGmOMd1ztnEdE+onIOhFJEpH7ink/RkSm+d5fIiLxbsbDb9MB0HaDeeedlXTsOIGPPlrN3XfPQVVd3bUxxgQD15KCiEQCrwIXAx2Bq0WkY5HVRgCpqnoC8DzwpFvxOKWjT9mcWov+Dzdk2LBP2Ls3g7592/LJJ0PsiWRjjMHd8lF3IElVkwFE5ENgELDGb51BwCO+6enAKyIi6sLX9vxNc5n4dXvGzu7Locxt1KkTy/PPX8SwYZ0tIRhjjI+bSaEZsMVvPgU442jrqGquiOwH6gG7yzuY/Stm8Ojc8ziUGc0VV5zEK69cQuPG1ct7N8YYE9TcTArFff0uegUQyDqIyChgFEDLlsc2jnKdatlMvmoW2ee8wBXD+xzTNowxJtS5mRRSgBZ+882BP46yToqIRAG1gL1FN6Sqk4BJAAkJCcdWWrrkXQb0OgAxNY/p48YYEw7cbH20FGgnIq1FpAowBEgssk4icL1vejDwtRv3EwpZQjDGmBK5dqXgu0cwGpgDRAJTVXW1iDwGLFPVRGAK8I6IJOFcIQxxKx5jjDGlc/XhNVWdBcwqsuxhv+lM4NgHSDbGGFOubGR5Y4wxhSwpGGOMKWRJwRhjTCFLCsYYYwpZUjDGGFNIgq13UBHZBfx+jB+vjwtdaFRydszhwY45PBzPMbdS1QalrRR0SeF4iMgyVU3wOo6KZMccHuyYw0NFHLOVj4wxxhSypGCMMaZQuCWFSV4H4AE75vBgxxweXD/msLqnYIwxpmThdqVgjDGmBCGZFESkn4isE5EkEbmvmPdjRGSa7/0lIhJf8VGWrwCO+e8iskZEfhaRr0SklRdxlqfSjtlvvcEioiIS9C1VAjlmEbnS92+9WkTer+gYy1sA/7dbisg8EVnh+/99iRdxlhcRmSoiO0Xkl6O8LyLyku/38bOIdC3XAFQ1pH5wuuneALQBqgArgY5F1rkNmOibHgJM8zruCjjmC4A43/St4XDMvvVqAAuAxUCC13FXwL9zO2AFUMc339DruCvgmCcBt/qmOwKbvI77OI+5J9AV+OUo718CzMYZufJMYEl57j8UrxS6A0mqmqyq2cCHwKAi6wwC3vJNTwd6iUhxQ4MGi1KPWVXnqWq6b3Yxzkh4wSyQf2eA/wOeAjIrMjiXBHLMI4FXVTUVQFV3VnCM5S2QY1agYAStWvx5hMegoqoLKGYESj+DgLfVsRioLSJNymv/oZgUmgFb/OZTfMuKXUdVc4H9QL0Kic4dgRyzvxE43zSCWanHLCKnAS1U9bOKDMxFgfw7twfai8hCEVksIv0qLDp3BHLMjwDXikgKzvgtd1RMaJ4p6997mbg6yI5HivvGX7SJVSDrBJOAj0dErgUSgPNcjch9JR6ziEQAzwPDKyqgChDIv3MUTgnpfJyrwW9FpJOq7nM5NrcEcsxXA2+q6rMichbOaI6dVDXf/fA84er5KxSvFFKAFn7zzfnz5WThOiIShXPJWdLlWmUXyDEjIr2BB4CBqppVQbG5pbRjrgF0AuaLyCac2mtikN9sDvT/9qeqmqOqG4F1OEkiWAVyzCOAjwBUdREQi9NHUKgK6O/9WIViUlgKtBOR1iJSBedGcmKRdRKB633Tg4Gv1XcHJ0iVesy+UsrrOAkh2OvMUMoxq+p+Va2vqvGqGo9zH2Wgqi7zJtxyEcj/7U9wGhUgIvVxyknJFRpl+QrkmDcDvQBE5CScpLCrQqOsWInAMF8rpDOB/aq6rbw2HnLlI1XNFZHRwByclgtTVXW1iDwGLFPVRGAKziVmEs4VwhDvIj5+AR7z00B14D++e+qbVXWgZ0EfpwCPOaQEeMxzgL4isgbIA+5V1T3eRX18Ajzme4A3RORunDLK8GD+kiciH+CU/+r77pOMA6IBVHUizn2TS4AkIB24oVz3H8S/O2OMMeUsFMtHxhhjjpElBWOMMYUsKRhjjClkScEYY0whSwrGGGMKWVIwlY6I5InIT34/8SWsG3+03iTLuM/5vp44V/q6iOhwDNu4RUSG+aaHi0hTv/cmi0jHco5zqYh0CeAzfxORuOPdtwkPlhRMZZShql38fjZV0H6HqmpnnM4Sny7rh1V1oqq+7ZsdDjT1e+8mVV1TLlEejnMCgcX5N8CSggmIJQUTFHxXBN+KyHLfT49i1jlZRH7wXV38LCLtfMuv9Vv+uohElrK7BcAJvs/28vXTv8rXz32Mb/l4OTw+xTO+ZY+IyBgRGYzTv9R7vn1W9X3DTxCRW0XkKb+Yh4vIy8cY5yL8OkITkddEZJk44yg86lt2J05ymici83zL+orIIt/v8T8iUr2U/ZgwYknBVEZV/UpHH/uW7QT6qGpX4CrgpWI+dwvwoqp2wTkpp/i6PbgKONu3PA8YWsr+BwCrRCQWeBO4SlVPwekB4FYRqQv8BThZVU8FHvf/sKpOB5bhfKPvoqoZfm9PBy73m78KmHaMcfbD6daiwAOqmgCcCpwnIqeq6ks4/eJcoKoX+Lq+eBDo7ftdLgP+Xsp+TBgJuW4uTEjI8J0Y/UUDr/hq6Hk4ffoUtQh4QESaAzNUdb2I9AK6AUt93XtUxUkwxXlPRDKATTjdL3cANqrqb7733wJuB17BGZ9hsoh8DgTcNbeq7hKRZF+fNet9+1jo225Z4qyG0+2D/6hbV4rIKJy/6yY4A878XOSzZ/qWL/TtpwrO780YwJKCCR53AzuAzjhXuH8aNEdV3xeRJUB/YI6I3ITTzfBbqnp/APsY6t9hnogUO8aGrz+e7jidsA0BRgMXluFYpgFXAr8CH6uqinOGDjhOnBHIxgOvApeLSGtgDHC6qqaKyJs4HcMVJcBcVb26DPGaMGLlIxMsagHbfH3kX4fzLfkIItIGSPaVTBJxyihfAYNFpKFvnboS+PjUvwLxInKCb/464BtfDb6Wqs7CuYlbXAuggzjddxdnBnAZzjgA03zLyhSnqubglIHO9JWeagJpwH4RaQRcfJRYFgNnFxyTiMSJSHFXXSZMWVIwwWICcL2ILMYpHaUVs85VwC8i8hNwIs6QhWtwTp5fisjPwFyc0kqpVDUTpwfK/4jIKiAfmIhzgv3Mt71vcK5iinoTmFhwo7nIdlOBNUArVf3Bt6zMcfruVTwLjFHVlThjM68GpuKUpApMAmaLyDxV3YXTMuoD334W4/yujAGsl1RjjDF+7ErBGGNMIUsKxhhjCllSMMYYU8iSgjHGmEKWFIwxxhSypGCMMaaQJQVjjDGFLCkYY4wp9P8BoBBQVzFoMrUAAAAASUVORK5CYII=\n",
        "text/plain": "<Figure size 432x288 with 1 Axes>"
       },
       "metadata": {
        "needs_background": "light"
       },
       "output_type": "display_data"
      }
     ],
     "persistentId": "2c810e70-a822-4d9a-a9b0-2622dfa247bf",
     "text": "y_true = coding_scores_unspliced_not_low_complexity['in_cds_no_stop_codon']\ny_pred = coding_scores_unspliced_not_low_complexity['jaccard_in_peptide_db']\n\nfpr, tpr, thresholds = metrics.roc_curve(y_true, y_pred)\n\nfig, ax = plt.subplots()\nlw = 2\nax.plot(fpr, tpr, lw=2, color='darkorange')\nax.plot([0, 1], [0, 1], color='navy', lw=lw, linestyle='--')\nax.set(xlabel='False Positive Rate', ylabel='True Positive Rate')\n\nprint(metrics.accuracy_score(y_true, y_pred > 0.9))\n\nprint(metrics.classification_report(y_true, y_pred > 0.9))"
    },
    "executionTime": "2019-10-12T01:01:28.673Z"
   },
   {
    "cell": {
     "executionCount": 43,
     "executionEventId": "17cc953b-dd38-4956-8bed-47be346e8e07",
     "hasError": false,
     "id": "efba9e80-c61d-4adb-9d6c-af7db00a9356",
     "outputs": [
      {
       "data": {
        "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>read_id</th>\n      <th>jaccard_in_peptide_db</th>\n      <th>n_kmers</th>\n      <th>classification</th>\n      <th>short_read_id</th>\n      <th>in_cds_no_stop_codon</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>52</th>\n      <td>SRR306838.17561853 Ibis_Run100924_C3PO:6:88:9412:18824/1</td>\n      <td>0.000000</td>\n      <td>15</td>\n      <td>non-coding</td>\n      <td>SRR306838.17561853</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>61</th>\n      <td>SRR306838.3607263 Ibis_Run100924_C3PO:6:17:17893:5364/1</td>\n      <td>0.294118</td>\n      <td>17</td>\n      <td>non-coding</td>\n      <td>SRR306838.3607263</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>112</th>\n      <td>SRR306838.23727244 Ibis_Run100924_C3PO:6:117:4784:12719/1</td>\n      <td>0.066667</td>\n      <td>15</td>\n      <td>non-coding</td>\n      <td>SRR306838.23727244</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>144</th>\n      <td>SRR306838.3497462 Ibis_Run100924_C3PO:6:16:17526:15365/1</td>\n      <td>0.461538</td>\n      <td>13</td>\n      <td>non-coding</td>\n      <td>SRR306838.3497462</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>166</th>\n      <td>SRR306838.12840151 Ibis_Run100924_C3PO:6:64:3090:6141/1</td>\n      <td>0.000000</td>\n      <td>17</td>\n      <td>non-coding</td>\n      <td>SRR306838.12840151</td>\n      <td>True</td>\n    </tr>\n  </tbody>\n</table>\n</div>",
        "text/plain": "                                                       read_id  \\\n52    SRR306838.17561853 Ibis_Run100924_C3PO:6:88:9412:18824/1   \n61     SRR306838.3607263 Ibis_Run100924_C3PO:6:17:17893:5364/1   \n112  SRR306838.23727244 Ibis_Run100924_C3PO:6:117:4784:12719/1   \n144   SRR306838.3497462 Ibis_Run100924_C3PO:6:16:17526:15365/1   \n166    SRR306838.12840151 Ibis_Run100924_C3PO:6:64:3090:6141/1   \n\n     jaccard_in_peptide_db  n_kmers classification       short_read_id  \\\n52                0.000000       15     non-coding  SRR306838.17561853   \n61                0.294118       17     non-coding   SRR306838.3607263   \n112               0.066667       15     non-coding  SRR306838.23727244   \n144               0.461538       13     non-coding   SRR306838.3497462   \n166               0.000000       17     non-coding  SRR306838.12840151   \n\n     in_cds_no_stop_codon  \n52                   True  \n61                   True  \n112                  True  \n144                  True  \n166                  True  "
       },
       "execution_count": 43,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "1254868e-73a0-4c0a-82c6-bb1b85e78cb0",
     "text": "coding_scores_unspliced.query(\"jaccard_in_peptide_db < 0.5 & in_cds_no_stop_codon == True\").head()"
    },
    "executionTime": "2019-10-12T01:01:35.024Z"
   },
   {
    "cell": {
     "executionCount": 45,
     "executionEventId": "b425a531-29c6-47ed-ab32-dd1ce9130736",
     "hasError": false,
     "id": "527d8a46-a8fa-41cb-8b24-650decb3111c",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "SRR306838.17561853\t16\tchrM\t3811\t255\t67M\t*\t0\t0\tCAAGAACACCTCTGATTACTCCTGCCATCATGACCCTTGGCCATAATATGATTTATCTCCACACTAG\t78677564566847334362785755:66:5525674587556457767724344164757626035\tNH:i:1\tHI:i:1\tAS:i:65\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\n"
      }
     ],
     "persistentId": "8b2488c1-3c99-4997-b2e9-eb0caae3979c",
     "text": "unspliced_bam = 'SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.bam'\n\n# This reads aligns to the CBLN2 \nnoncoding_read_id = 'SRR306838.17561853\t'\n! samtools view $folder/$unspliced_bam | grep $noncoding_read_id"
    },
    "executionTime": "2019-10-12T01:01:50.657Z"
   },
   {
    "cell": {
     "executionCount": 46,
     "executionEventId": "9d335cd6-f39c-4da2-a7b6-f18a13907be6",
     "hasError": false,
     "id": "1ab74d6a-ae3e-4008-964f-0ef58aba5cdf",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "SRR306838.3607263\t16\tchrM\t8401\t255\t12S60M\t*\t0\t0\tGTCATGCAAAAGAATTACCCCCATACTCCTTACACTATTCCTTATCACCCAACTAAAAACATTAAACACAAA\t557:977:86<;8744366867844:78935686846668:434655896;58479533962386971613.\tNH:i:1\tHI:i:1\tAS:i:55\tnM:i:2\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\n"
      }
     ],
     "persistentId": "e75ae01f-e817-4e22-93d9-62740bd49be3",
     "text": "unspliced_bam = 'SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.bam'\n\n# This reads aligns to the CBLN2 \nnoncoding_read_id = 'SRR306838.3607263'\n! samtools view $folder/$unspliced_bam | grep $noncoding_read_id"
    },
    "executionTime": "2019-10-12T01:02:05.284Z"
   },
   {
    "cell": {
     "executionCount": 47,
     "executionEventId": "f3fe66ae-bf1a-435b-a340-580f3b4a5956",
     "hasError": false,
     "id": "9c086623-f690-4aa0-82af-768c99ebb20f",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "SRR306838.23727244\t16\tchrM\t14157\t255\t66M\t*\t0\t0\tCCCGAGCAATCTCAATTACAATATATACACCAACAAACAATGTTCAACCAGTAACTACTACTAATC\t6678867778855785236875525246355726854665351366456341337226254200.1\tNH:i:1\tHI:i:1\tAS:i:64\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\n"
      }
     ],
     "persistentId": "beff2498-4747-40c1-acfd-e30bc468e81a",
     "text": "unspliced_bam = 'SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.bam'\n\n# This reads aligns to the CBLN2 \nnoncoding_read_id = 'SRR306838.23727244'\n! samtools view $folder/$unspliced_bam | grep $noncoding_read_id"
    },
    "executionTime": "2019-10-12T01:02:31.360Z"
   },
   {
    "cell": {
     "executionCount": 48,
     "executionEventId": "6b125949-7b79-4ec3-b4b2-f16f4ee6b53f",
     "hasError": false,
     "id": "39a38c2d-a442-4bef-ae38-e2ebe781aa74",
     "outputs": [
      {
       "data": {
        "text/plain": "(29, 6)"
       },
       "execution_count": 48,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "23edfe41-6d93-44d6-9324-ba4ac19d25e0",
     "text": "coding_scores_unspliced.query(\"jaccard_in_peptide_db < 0.5 & in_cds_no_stop_codon == True\").shape"
    },
    "executionTime": "2019-10-12T01:08:09.723Z"
   },
   {
    "cell": {
     "executionCount": 49,
     "executionEventId": "67ddabce-8f94-4b0b-819d-cfbbabc58699",
     "hasError": false,
     "id": "1da7b738-0e56-411d-aaed-632c1ff4e825",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "SRR306838.17561853\t16\tchrM\t3811\t255\t67M\t*\t0\t0\tCAAGAACACCTCTGATTACTCCTGCCATCATGACCCTTGGCCATAATATGATTTATCTCCACACTAG\t78677564566847334362785755:66:5525674587556457767724344164757626035\tNH:i:1\tHI:i:1\tAS:i:65\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\nSRR306838.3607263\t16\tchrM\t8401\t255\t12S60M\t*\t0\t0\tGTCATGCAAAAGAATTACCCCCATACTCCTTACACTATTCCTTATCACCCAACTAAAAACATTAAACACAAA\t557:977:86<;8744366867844:78935686846668:434655896;58479533962386971613.\tNH:i:1\tHI:i:1\tAS:i:55\tnM:i:2\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\nSRR306838.23727244\t16\tchrM\t14157\t255\t66M\t*\t0\t0\tCCCGAGCAATCTCAATTACAATATATACACCAACAAACAATGTTCAACCAGTAACTACTACTAATC\t6678867778855785236875525246355726854665351366456341337226254200.1\tNH:i:1\tHI:i:1\tAS:i:64\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\nSRR306838.3497462\t16\tchrM\t8428\t255\t58M\t*\t0\t0\tCCTCATCACCCAACTAAAAACATTAAACACAAACTACCACCTACCTCCCTCACCAAAG\t7:78:573468:571665736743343457634611672683/472883267672035\tNH:i:1\tHI:i:1\tAS:i:55\tnM:i:1\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\nSRR306838.12840151\t16\tchrM\t4955\t255\t69M\t*\t0\t0\tCATAGCAGGCAGTTGAGGTGGATTAAACCAAACCCAGCTACGCAAAATCTTAGCATACTCCTCAATTAC\t67363788668536976554975565546:5266576643655645669336554327137354222/1\tNH:i:1\tHI:i:1\tAS:i:67\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\nSRR306838.11563592\t0\tchrM\t6434\t255\t67M\t*\t0\t0\tCCAATACCAAACGCCCCTCTTCGTCTGATCCGTCCTAATCACAGCAGTCCTACTTCTCCTATCTCTC\t514325394725555764655664754525673664554744458682956265487765559673:\tNH:i:1\tHI:i:1\tAS:i:65\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\nSRR306838.6391414\t0\tchr5\t53885588\t255\t64M\t*\t0\t0\tTGAATATATAGATGATTCTGTCTTTGTTATCTTTGTCAAATCCATGCCATCTTAAACCCTAACC\t-422347255616624476547566736456748639455766898775797565464574487\tNH:i:1\tHI:i:1\tAS:i:62\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\nSRR306838.639141\t16\tchrM\t6586\t255\t4S63M\t*\t0\t0\tGGAGCCATTCTATACCAACACCTATTCTGATTTTTCGGTCACCCTGAAGTTTATATTCTTATCCTAC\t6:65766659345367746368553685873244677547269967565366343232236477202\tNH:i:1\tHI:i:1\tAS:i:62\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\nSRR306838.13119715\t0\tchrM\t5349\t255\t67M\t*\t0\t0\tCTACGCCTAATCTACTCCACCTCAATCACACTACTCCCCATATCTAACAACGTAAAAATAAAATGAC\t301.31/213143/21465774755474666546547457666:55372375155437546675557\tNH:i:1\tHI:i:1\tAS:i:65\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\nSRR306838.9230309\t16\tchr1\t629343\t3\t65M\t*\t0\t0\tATGAAAAAACTTCCTACCACTCACCCTAGCATTACTTATATGATATGTCTCCATACCCATTACAA\t657656655846975366575724685658:65463355578767645634882.276;44/65-\tNH:i:2\tHI:i:1\tAS:i:63\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\nSRR306838.9230309\t272\tchrM\t4173\t3\t65M\t*\t0\t0\tATGAAAAAACTTCCTACCACTCACCCTAGCATTACTTATATGATATGTCTCCATACCCATTACAA\t657656655846975366575724685658:65463355578767645634882.276;44/65-\tNH:i:2\tHI:i:2\tAS:i:63\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\nSRR306838.6978270\t16\tchrM\t4931\t255\t61M\t*\t0\t0\tCCTCACTCTCTCAATCTTATCCATCATAGCAGGCAGTTGAGGTGGATTAAACCAAACCCAG\t7457474635546648315677757652676765554685765563355404668078641\tNH:i:1\tHI:i:1\tAS:i:60\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\nSRR306838.15709562\t16\tchrM\t8389\t255\t68M\t*\t0\t0\tATGGCCCACCATAATTACCCCCATACTCCTTACACTATTCCTCATCACCCAACTAAAAACATTAAACA\t5785766436:677544655578536669444647575468268364766837257544682531.4.\tNH:i:1\tHI:i:1\tAS:i:64\tnM:i:1\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\nSRR306838.16880644\t16\tchrM\t4938\t255\t63M\t*\t0\t0\tCTCTCAATCTTATCCATCATAGCAGGCAGTTGAGGTGGATTAAACCAAACCCAGCTACGCAAA\t96958957823655674794647843746367466466635663577315563362520702,\tNH:i:1\tHI:i:1\tAS:i:62\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\nSRR306838.14545842\t16\tchrM\t3738\t255\t10S56M\t*\t0\t0\tAAAAAATCAACACCCTAGCCATCATTCTACTATCAACATTACTAATAAGTGGCTCCTTTAACCTCT\t6684753788615853644775::6275486466;576414626446464633277422605604)\tNH:i:1\tHI:i:1\tAS:i:55\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\nSRR306838.11004483\t16\tchrM\t12894\t255\t69M\t*\t0\t0\tAGCATGATTTATCCTACACTCCAACTCATGAGACCCACAACAAATAGCCCTTCTAAACGCTAATCCAAG\t9576485553576:44778376:6756:4776565686:588655677775675545673458578243\tNH:i:1\tHI:i:1\tAS:i:67\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\nSRR306838.2963056\t0\tchrM\t4975\t255\t67M\t*\t0\t0\tGATTAAACCAAACCCAGCTACGCAAAATCTTAGCATACTCCTCAATTACCCACATAGGATGAATAAT\t30114636884264667645696655246544:374337877;7576285738775:6976975666\tNH:i:1\tHI:i:1\tAS:i:65\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\nSRR306838.13753456\t16\tchr1\t629331\t3\t9S63M\t*\t0\t0\tAAGAACGGCCATACACCTCCTATGAAAAAACTTCCTACCACTCACCCTAGCATTACTTATATGATATGTCTC\t647735748:754665865847494877345357:41988648265937877444633843476376304/2\tNH:i:2\tHI:i:1\tAS:i:62\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\nSRR306838.13753456\t272\tchrM\t4161\t3\t9S63M\t*\t0\t0\tAAGAACGGCCATACACCTCCTATGAAAAAACTTCCTACCACTCACCCTAGCATTACTTATATGATATGTCTC\t647735748:754665865847494877345357:41988648265937877444633843476376304/2\tNH:i:2\tHI:i:2\tAS:i:62\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\nSRR306838.12460240\t16\tchrM\t5395\t255\t13S54M\t*\t0\t0\tCAGACACTCAGACACAACGTAAAAATAAAATGACAGTTTGAACATACAAAACCCACCCCATTCCTCC\t766674779:83766:454275837767564434852344356744753515261563750367/52\tNH:i:1\tHI:i:1\tAS:i:53\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\nSRR306838.15383614\t256\tchr1\t631921\t3\t72M\t*\t0\t0\tCTAGGGTTTATCGTGTGAGCACACCATATATTTACAGTAGGAATAGACGTAGACACACGAGCATATTTCACC\t631875778516655478875766887845976577856978577866645658595678979447359555\tNH:i:2\tHI:i:2\tAS:i:70\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\nSRR306838.15383614\t0\tchrM\t6750\t3\t72M\t*\t0\t0\tCTAGGGTTTATCGTGTGAGCACACCATATATTTACAGTAGGAATAGACGTAGACACACGAGCATATTTCACC\t631875778516655478875766887845976577856978577866645658595678979447359555\tNH:i:2\tHI:i:1\tAS:i:70\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\nSRR306838.2587809\t16\tchrM\t14800\t255\t47M\t*\t0\t0\tCATCGACCTCCCCACCCCATCCAACATCTCCGCATGATGAAACTTCG\t76366368365654765863577577563773464726658161145\tNH:i:1\tHI:i:1\tAS:i:46\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\nSRR306838.16843581\t16\tchrM\t5367\t255\t7S65M\t*\t0\t0\tATTCTTAACCTCAATCACACTACTCCCCATATCTAACAACGTAAAAATAAAATGACAGTTTGAACATACAAA\t85:964847>98:777586:356667658475=6657937533435347676653654333:546:41724.\tNH:i:1\tHI:i:1\tAS:i:63\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\nSRR306838.21182939\t256\tchr1\t632905\t3\t5S62M\t*\t0\t0\tTTTATTAACAAATACAAACATCTCAGACGCTCAGGAAATAGAAACCGTCTGAACTATCCTGCCCGCC\t,403341368324478644666774824544788643455886453636676486654468772675\tNH:i:2\tHI:i:2\tAS:i:57\tnM:i:2\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\nSRR306838.21182939\t0\tchrM\t7734\t3\t5S62M\t*\t0\t0\tTTTATTAACAAATACAAACATCTCAGACGCTCAGGAAATAGAAACCGTCTGAACTATCCTGCCCGCC\t,403341368324478644666774824544788643455886453636676486654468772675\tNH:i:2\tHI:i:1\tAS:i:57\tnM:i:2\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\nSRR306838.1759208\t16\tchrM\t15817\t255\t66M\t*\t0\t0\tATACTTCACAACAATCCTAATCCTAATACCAACTATCTCCCTAATTGAAAACAAAATACTCAAATG\t955<47656638<44865858982:753679378668386<18866:56627464334917412.3\tNH:i:1\tHI:i:1\tAS:i:64\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\nSRR306838.11331605\t16\tchrM\t8425\t255\t41M\t*\t0\t0\tATTCCTCATCACCCAACTAAAAACATTAAACACAAACTACC\t84567467674566747367653764255466668170,63\tNH:i:1\tHI:i:1\tAS:i:38\tnM:i:1\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\nSRR306838.23942767\t0\tchr1\t629659\t3\t70M\t*\t0\t0\tCCGTCATCTACTCTACCATCTTTGCAGGCACACTCATCACAGCGCTAAGCTCGCACTGATTTTTTACCTG\t4/3/267854836536563756467565756473:776654763734284667647987445454465:5\tNH:i:2\tHI:i:1\tAS:i:68\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\nSRR306838.23942767\t256\tchrM\t4489\t3\t70M\t*\t0\t0\tCCGTCATCTACTCTACCATCTTTGCAGGCACACTCATCACAGCGCTAAGCTCGCACTGATTTTTTACCTG\t4/3/267854836536563756467565756473:776654763734284667647987445454465:5\tNH:i:2\tHI:i:2\tAS:i:68\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\nSRR306838.15264190\t256\tchr1\t629315\t3\t70M\t*\t0\t0\tTCCGCTACGACCAACTCATACACCTCCTATGAAAAAACTTCCTACCACTCACCCTAGCATTACTTATATG\t,034012654852332564573654564475665432476757475465666876594755467573665\tNH:i:2\tHI:i:2\tAS:i:68\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\nSRR306838.15264190\t0\tchrM\t4145\t3\t70M\t*\t0\t0\tTCCGCTACGACCAACTCATACACCTCCTATGAAAAAACTTCCTACCACTCACCCTAGCATTACTTATATG\t,034012654852332564573654564475665432476757475465666876594755467573665\tNH:i:2\tHI:i:1\tAS:i:68\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\nSRR306838.4040046\t0\tchrM\t4887\t255\t17S45M1S\t*\t0\t0\tCACACATCCACCTCGGGTCCATCATATACCAAATCTCTCCCTCACTAAACGTAAGCCTTCTCA\t511214665784471441667566455465466485956665765444226554965568597\tNH:i:1\tHI:i:1\tAS:i:42\tnM:i:1\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\nSRR306838.9874975\t16\tchrM\t8479\t255\t6S64M\t*\t0\t0\tAATACTACCAAAGCCCATAAAAATAAAAAATTATAACAAACCCTGAGAACCAAAATGAACGAAAATCTGT\t8742834777672557;677565439566544644357514572436546565530742446351/71/(\tNH:i:1\tHI:i:1\tAS:i:62\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\nSRR306838.1128147\t16\tchr1\t631932\t3\t48M\t*\t0\t0\tCGTGTGAGCACACCATATATTTACAGTAGGAATAGACGTAGACACACG\t555668657555546566334447:63674566295565588626034\tNH:i:2\tHI:i:1\tAS:i:47\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\nSRR306838.1128147\t272\tchrM\t6761\t3\t48M\t*\t0\t0\tCGTGTGAGCACACCATATATTTACAGTAGGAATAGACGTAGACACACG\t555668657555546566334447:63674566295565588626034\tNH:i:2\tHI:i:2\tAS:i:47\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\nSRR306838.22091604\t0\tchrM\t6588\t255\t70M\t*\t0\t0\tATTCTATACCAACACCTATTCTGATTTTTCGGTCACCCTGAAGTTTATATTCTTATCCTACCAGGCTTCG\t)30335737552662643369:745554335554356597336373576667665484738768675377\tNH:i:1\tHI:i:1\tAS:i:68\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\n"
      }
     ],
     "persistentId": "eea06bec-d676-41c3-a79f-56bbc89f9e98",
     "text": "df = coding_scores_unspliced.query(\"jaccard_in_peptide_db < 0.5 & in_cds_no_stop_codon == True\")\nfor read_id in df.short_read_id:\n    ! samtools view $folder/$unspliced_bam | grep $read_id"
    },
    "executionTime": "2019-10-12T01:10:33.239Z"
   },
   {
    "cell": {
     "executionCount": 50,
     "executionEventId": "4b5c82a2-d200-4998-aad3-22cb0017383b",
     "hasError": false,
     "id": "ddb69d5f-d893-4639-aef2-21456dca0e78",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "(999, 5)\n"
      },
      {
       "data": {
        "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>read_id</th>\n      <th>jaccard_in_peptide_db</th>\n      <th>n_kmers</th>\n      <th>classification</th>\n      <th>short_read_id</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>SRR306838.10559374 Ibis_Run100924_C3PO:6:51:17601:17119/1</td>\n      <td>1.000000</td>\n      <td>16</td>\n      <td>coding</td>\n      <td>SRR306838.10559374</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>SRR306838.6196593 Ibis_Run100924_C3PO:6:29:16733:12435/1</td>\n      <td>0.000000</td>\n      <td>17</td>\n      <td>non-coding</td>\n      <td>SRR306838.6196593</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>SRR306838.20767303 Ibis_Run100924_C3PO:6:104:6864:5062/1</td>\n      <td>0.187500</td>\n      <td>16</td>\n      <td>non-coding</td>\n      <td>SRR306838.20767303</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>SRR306838.12582274 Ibis_Run100924_C3PO:6:62:11779:17975/1</td>\n      <td>0.066667</td>\n      <td>15</td>\n      <td>non-coding</td>\n      <td>SRR306838.12582274</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>SRR306838.13334230 Ibis_Run100924_C3PO:6:66:16579:20350/1</td>\n      <td>0.000000</td>\n      <td>12</td>\n      <td>non-coding</td>\n      <td>SRR306838.13334230</td>\n    </tr>\n  </tbody>\n</table>\n</div>",
        "text/plain": "                                                     read_id  \\\n0  SRR306838.10559374 Ibis_Run100924_C3PO:6:51:17601:17119/1   \n1   SRR306838.6196593 Ibis_Run100924_C3PO:6:29:16733:12435/1   \n2   SRR306838.20767303 Ibis_Run100924_C3PO:6:104:6864:5062/1   \n3  SRR306838.12582274 Ibis_Run100924_C3PO:6:62:11779:17975/1   \n4  SRR306838.13334230 Ibis_Run100924_C3PO:6:66:16579:20350/1   \n\n   jaccard_in_peptide_db  n_kmers classification       short_read_id  \n0               1.000000       16         coding  SRR306838.10559374  \n1               0.000000       17     non-coding   SRR306838.6196593  \n2               0.187500       16     non-coding  SRR306838.20767303  \n3               0.066667       15     non-coding  SRR306838.12582274  \n4               0.000000       12     non-coding  SRR306838.13334230  "
       },
       "execution_count": 50,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "73b49c34-8379-4aa5-835e-57c54aced1fa",
     "text": "coding_scores_not_low_complexity = coding_scores.query('jaccard_in_peptide_db >= 0')\nprint(coding_scores_not_low_complexity.shape)\ncoding_scores_not_low_complexity.head()"
    },
    "executionTime": "2019-10-12T03:39:08.726Z"
   },
   {
    "cell": {
     "executionCount": 51,
     "executionEventId": "caec358e-a3ea-4685-a699-2011ee77e3a6",
     "hasError": false,
     "id": "9d6e4a68-d19f-4111-b815-f4e2c0f40d25",
     "outputs": [
      {
       "data": {
        "text/plain": "(1000, 5)"
       },
       "execution_count": 51,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "09393220-d1f9-4ddd-9c7c-bc15160f3680",
     "text": "coding_scores.shape"
    },
    "executionTime": "2019-10-12T03:39:09.571Z"
   },
   {
    "cell": {
     "executionCount": 52,
     "executionEventId": "f1018f7d-f3c5-4093-924c-ed3422327565",
     "hasError": false,
     "id": "adbb7805-c870-4664-a83c-3c6e2db0bece",
     "outputs": [
      {
       "data": {
        "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>read_id</th>\n      <th>jaccard_in_peptide_db</th>\n      <th>n_kmers</th>\n      <th>classification</th>\n      <th>short_read_id</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>SRR306838.10559374 Ibis_Run100924_C3PO:6:51:17601:17119/1</td>\n      <td>1.000000</td>\n      <td>16</td>\n      <td>coding</td>\n      <td>SRR306838.10559374</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>SRR306838.6196593 Ibis_Run100924_C3PO:6:29:16733:12435/1</td>\n      <td>0.000000</td>\n      <td>17</td>\n      <td>non-coding</td>\n      <td>SRR306838.6196593</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>SRR306838.20767303 Ibis_Run100924_C3PO:6:104:6864:5062/1</td>\n      <td>0.187500</td>\n      <td>16</td>\n      <td>non-coding</td>\n      <td>SRR306838.20767303</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>SRR306838.12582274 Ibis_Run100924_C3PO:6:62:11779:17975/1</td>\n      <td>0.066667</td>\n      <td>15</td>\n      <td>non-coding</td>\n      <td>SRR306838.12582274</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>SRR306838.13334230 Ibis_Run100924_C3PO:6:66:16579:20350/1</td>\n      <td>0.000000</td>\n      <td>12</td>\n      <td>non-coding</td>\n      <td>SRR306838.13334230</td>\n    </tr>\n  </tbody>\n</table>\n</div>",
        "text/plain": "                                                     read_id  \\\n0  SRR306838.10559374 Ibis_Run100924_C3PO:6:51:17601:17119/1   \n1   SRR306838.6196593 Ibis_Run100924_C3PO:6:29:16733:12435/1   \n2   SRR306838.20767303 Ibis_Run100924_C3PO:6:104:6864:5062/1   \n3  SRR306838.12582274 Ibis_Run100924_C3PO:6:62:11779:17975/1   \n4  SRR306838.13334230 Ibis_Run100924_C3PO:6:66:16579:20350/1   \n\n   jaccard_in_peptide_db  n_kmers classification       short_read_id  \n0               1.000000       16         coding  SRR306838.10559374  \n1               0.000000       17     non-coding   SRR306838.6196593  \n2               0.187500       16     non-coding  SRR306838.20767303  \n3               0.066667       15     non-coding  SRR306838.12582274  \n4               0.000000       12     non-coding  SRR306838.13334230  "
       },
       "execution_count": 52,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "ede6e1ba-f524-4ceb-b35b-6c20f91d437c",
     "text": "coding_scores.head()"
    },
    "executionTime": "2019-10-12T03:39:10.102Z"
   },
   {
    "cell": {
     "executionCount": 53,
     "executionEventId": "9906f15c-69c2-4f13-acfc-c706dae5a10a",
     "hasError": false,
     "id": "9c593919-2f4d-44c4-aaa8-0ae119337efd",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "total 353M\ndrwxrwxr-x 2 olga olga 4.0K Sep 18 10:15 .\ndrwxrwxr-x 3 olga olga 4.0K Sep 10 15:07 ..\n-rw-rw-r-- 1 olga olga  52M Oct 11 18:07 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.bam\n-rw-rw-r-- 1 olga olga 2.9M Oct 11 18:08 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.bam.bai\n-rw-rw-r-- 1 olga olga 6.1M Oct 11 18:08 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.out_CDS_without_stop_codon_or_utr.bam\n-rw-rw-r-- 1 olga olga 1.8M Oct 11 18:08 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.out_CDS_without_stop_codon_or_utr.bam.bai\n-rw-rw-r-- 1 olga olga 1.7M Oct 11 18:08 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.out_CDS_without_stop_codon_or_utr.read_ids.txt\n-rw-rw-r-- 1 olga olga  15M Oct 11 18:08 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.read_ids.txt\n-rw-rw-r-- 1 olga olga  15M Sep 10 15:55 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out_CDS_without_stop_codon.bam\n-rw-rw-r-- 1 olga olga 1.7M Oct 11 18:08 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out_CDS_without_stop_codon.bam.bai\n-rw-rw-r-- 1 olga olga  15M Sep 18 07:15 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out_CDS_without_stop_codon_or_utr.bam\n-rw-rw-r-- 1 olga olga 1.7M Oct 11 18:08 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out_CDS_without_stop_codon_or_utr.bam.bai\n-rw-rw-r-- 1 olga olga 2.8M Sep 18 10:14 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out_CDS_without_stop_codon_or_utr.read_ids.txt\n-rw-rw-r-- 1 olga olga 4.1M Sep 10 15:56 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out_CDS_without_stop_codon.read_ids.txt\n-rw-r--r-- 1 olga olga 235M Sep 10 15:10 SRR306838_GSM752691_hsa_br_F_1_trimmed_k15.tsv\n(91741,)\n"
      },
      {
       "data": {
        "text/plain": "0    SRR306838.13598607\n1    SRR306838.23303318\n2    SRR306838.19917713\n3    SRR306838.19937719\n4    SRR306838.20862490\nName: 0, dtype: object"
       },
       "execution_count": 53,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "75c01782-427c-4c63-bc29-ed308ae4a142",
     "text": "folder = '/home/olga/data_sm/kmer-hashing/brawand2011/human/'\n! ls -lha $folder\n\ncoding_read_ids = pd.read_csv(f\"{folder}/SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.out_CDS_without_stop_codon_or_utr.read_ids.txt\", \n                                   header=None, squeeze=True)\nprint(coding_read_ids.shape)\ncoding_read_ids.head()"
    },
    "executionTime": "2019-10-12T03:39:11.260Z"
   },
   {
    "cell": {
     "executionCount": 54,
     "executionEventId": "9fefc512-3ed4-447f-ae0d-035bde710bbe",
     "hasError": false,
     "id": "1b979a86-c27b-4b61-9141-e248088bbda9",
     "outputs": [
      {
       "data": {
        "text/plain": "90307"
       },
       "execution_count": 54,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "75c01782-427c-4c63-bc29-ed308ae4a142",
     "text": "true_coding_read_ids = set(coding_read_ids.values)\nlen(true_coding_read_ids)"
    },
    "executionTime": "2019-10-12T03:39:11.595Z"
   },
   {
    "cell": {
     "executionCount": 55,
     "executionEventId": "2ede64fe-35d7-4a5b-b52a-59061ca41cef",
     "hasError": false,
     "id": "af6c4741-a37b-408d-bac2-800c2c899126",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "(807698,)\n"
      }
     ],
     "persistentId": "317bb484-c93f-4e1c-b831-50734c9e6614",
     "text": "unspliced_read_ids_series = pd.read_csv(f\"{folder}/SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.read_ids.txt\", \n                                   header=None, squeeze=True)\nprint(unspliced_read_ids_series.shape)\nunspliced_read_ids_series.head()\n\nunspliced_reads = set(unspliced_read_ids_series.values)"
    },
    "executionTime": "2019-10-12T03:39:13.940Z"
   },
   {
    "cell": {
     "executionCount": 56,
     "executionEventId": "3240a420-df20-4a30-881c-2a884573ccbc",
     "hasError": false,
     "id": "0514d136-42a3-445d-8a45-5c5a92f885ae",
     "outputs": [
      {
       "name": "stderr",
       "output_type": "stream",
       "text": "/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: \nA value is trying to be set on a copy of a slice from a DataFrame.\nTry using .loc[row_indexer,col_indexer] = value instead\n\nSee the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n  \"\"\"Entry point for launching an IPython kernel.\n"
      },
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "(867, 6)\n"
      },
      {
       "name": "stderr",
       "output_type": "stream",
       "text": "/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \nA value is trying to be set on a copy of a slice from a DataFrame.\nTry using .loc[row_indexer,col_indexer] = value instead\n\nSee the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n  This is separate from the ipykernel package so we can avoid doing imports until\n"
      },
      {
       "data": {
        "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>read_id</th>\n      <th>jaccard_in_peptide_db</th>\n      <th>n_kmers</th>\n      <th>classification</th>\n      <th>short_read_id</th>\n      <th>in_cds_no_stop_codon</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>1</th>\n      <td>SRR306838.6196593 Ibis_Run100924_C3PO:6:29:16733:12435/1</td>\n      <td>0.000000</td>\n      <td>17</td>\n      <td>non-coding</td>\n      <td>SRR306838.6196593</td>\n      <td>False</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>SRR306838.20767303 Ibis_Run100924_C3PO:6:104:6864:5062/1</td>\n      <td>0.187500</td>\n      <td>16</td>\n      <td>non-coding</td>\n      <td>SRR306838.20767303</td>\n      <td>False</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>SRR306838.12582274 Ibis_Run100924_C3PO:6:62:11779:17975/1</td>\n      <td>0.066667</td>\n      <td>15</td>\n      <td>non-coding</td>\n      <td>SRR306838.12582274</td>\n      <td>False</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>SRR306838.13334230 Ibis_Run100924_C3PO:6:66:16579:20350/1</td>\n      <td>0.000000</td>\n      <td>12</td>\n      <td>non-coding</td>\n      <td>SRR306838.13334230</td>\n      <td>False</td>\n    </tr>\n    <tr>\n      <th>5</th>\n      <td>SRR306838.2740879 Ibis_Run100924_C3PO:6:13:11155:5248/1</td>\n      <td>1.000000</td>\n      <td>18</td>\n      <td>coding</td>\n      <td>SRR306838.2740879</td>\n      <td>True</td>\n    </tr>\n  </tbody>\n</table>\n</div>",
        "text/plain": "                                                     read_id  \\\n1   SRR306838.6196593 Ibis_Run100924_C3PO:6:29:16733:12435/1   \n2   SRR306838.20767303 Ibis_Run100924_C3PO:6:104:6864:5062/1   \n3  SRR306838.12582274 Ibis_Run100924_C3PO:6:62:11779:17975/1   \n4  SRR306838.13334230 Ibis_Run100924_C3PO:6:66:16579:20350/1   \n5    SRR306838.2740879 Ibis_Run100924_C3PO:6:13:11155:5248/1   \n\n   jaccard_in_peptide_db  n_kmers classification       short_read_id  \\\n1               0.000000       17     non-coding   SRR306838.6196593   \n2               0.187500       16     non-coding  SRR306838.20767303   \n3               0.066667       15     non-coding  SRR306838.12582274   \n4               0.000000       12     non-coding  SRR306838.13334230   \n5               1.000000       18         coding   SRR306838.2740879   \n\n   in_cds_no_stop_codon  \n1                 False  \n2                 False  \n3                 False  \n4                 False  \n5                  True  "
       },
       "execution_count": 56,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "b321c9fc-54af-40bc-96a9-40c59ea87994",
     "text": "coding_scores_not_low_complexity['short_read_id'] = coding_scores_not_low_complexity['read_id'].str.split().str[0]\ncoding_scores_not_low_complexity_unspliced = coding_scores_not_low_complexity.query('short_read_id in @unspliced_reads')\ncoding_scores_not_low_complexity_unspliced['in_cds_no_stop_codon'] = coding_scores_not_low_complexity_unspliced.short_read_id.isin(true_coding_read_ids)\nprint(coding_scores_not_low_complexity_unspliced.shape)\ncoding_scores_not_low_complexity_unspliced.head()"
    },
    "executionTime": "2019-10-12T03:39:15.255Z"
   },
   {
    "cell": {
     "executionCount": 57,
     "executionEventId": "eef57f95-0322-4871-a1d3-ac4ac68180a2",
     "hasError": false,
     "id": "c4e90627-a46a-4fa2-b552-6402975ee8c7",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "0.9204152249134948\n              precision    recall  f1-score   support\n\n       False       1.00      0.91      0.95       767\n        True       0.59      0.98      0.74       100\n\n   micro avg       0.92      0.92      0.92       867\n   macro avg       0.80      0.95      0.85       867\nweighted avg       0.95      0.92      0.93       867\n\n"
      },
      {
       "data": {
        "image/png": "iVBORw0KGgoAAAANSUhEUgAAAYUAAAEKCAYAAAD9xUlFAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDMuMC4zLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvnQurowAAIABJREFUeJzt3Xl8FFW2wPHfyQJh35GdoIKKKIgBFRUXFlE2Rx1FEUQRFEUdt1GfC+LzObgLImIkjMug4jCo0YFBRsEFAUEREdARAkJQZEsCJGQ/749qYk8MSQe6uno538+nP3RVV7pOJSSn7711zxVVxRhjjAGI8zoAY4wx4cOSgjHGmDKWFIwxxpSxpGCMMaaMJQVjjDFlLCkYY4wpY0nBGGNMGUsKxhhjylhSMMYYUybB6wCqq2nTppqcnOx1GMYYE1G++uqrXararKrjIi4pJCcns3LlSq/DMMaYiCIiPwVynHUfGWOMKWNJwRhjTBlLCsYYY8pYUjDGGFPGkoIxxpgyriUFEZkpIjtE5LtDvC4iMkVENojItyLS3a1YjDHGBMbNlsIrwIBKXr8Q6Oh7jAVedDEWY4wxAXBtnoKqfioiyZUcMhR4TZ31QJeJSEMRaamqv7gVU9jY+xP8+yYo3Od1JMaYCPDlhvokJZZy8tk9odfDrp7Ly8lrrYGtftuZvn2/SwoiMhanNUG7du1CEpxrVGHhDbB5gdeRGGPCnCrc889+PP3JqZzc8le+PO57El0+p5dJQSrYpxUdqKqpQCpASkpKhcdEhB/+Dp/cBfu2QM0GMOjvkFDT66iMMWFKADZugE+30n/oaZT06BHVSSETaOu33Qb42aNYQmPVFCchAJw9CZL7eRuPMSbsZGfnk5GRRffuLQGY+MwZDBuzq2zbbV4mhXRgvIi8BZwG5ETFeMKOb+DjW6Aot4LXVjn/9p0OXW8IbVzGmLD33nvfM27cP4mLE9auvYkGDZKoVSsxZAkBXEwKIvImcC7QVEQygQngtHxUdTowD7gI2ADkAde6FUtIfTcTtn1+6NfjEuDogaGLxxgT9nbsyOXWW+cze/ZaAE4/vQ3Z2fk0aJAU8ljcvPvoyipeV+Bmt87vmeyNzr/nPgNtzvn963VbQZ0WoY3JGBOWVJVZs9Zw223/Ys+eA9Sunchjj53P+PE9iY/3Zm5xxJXODns5Gc6/bc+H5l29jcUYE9bGjfsnL730FQB9+x5NauogOnRo5GlMVuYimLQUcjY5zxse7W0sxpiwd/HFx9OwYRJpaUP48MOrPU8IYC2F4Nr/M5QUQO3mUKOe19EYY8LMjz/u5qOPNnHjjSkADBhwLJs33+bJ2MGhWFIIpoPjCQ2slWCM+U1xcSnPPLOUCRMWU1BQTLduLTj99DYAYZUQwJJCcB1MCg2P8TYOY0zYWL16O6NHp/PVV84d9yNHdqVjx8YeR3VolhSC6eAgs7UUjIl5BQXFPProp0yatITi4lLatWvASy8NYsCAY70OrVKWFILJWgrGGJ/77vuIZ59dBsDNN/fgL3/pQ7164V/WxpJCMJW1FCwpGBPr/vznM1m6NJMnnujL2We39zqcgNktqcFU1lKw7iNjYs3ChRu59NK3KS4uBaBFi7p88cV1EZUQwJJC8BTkQP5uSKgFdUJXp8QY462srAOMHv0e/fv/jblz1/PXv64qe02komLQ4c26j4Il22+QOQL/Ixhjqu+dd9Zz003z2L59PzVrxjNhwjmMGtXN67COiCWFYMmxOQrGxIrt2/dzyy3zmTNnHQC9erUlLW0Ixx/f1OPIjpwlhWA52FKwO4+MiXrvvfc9c+aso06dRCZN6stNN/UgLi46eggsKQSLtRSMiWr5+cUkJTl/MseMOZWMjCzGjetBcnJDjyMLLhtoDhabo2BMVCotVaZO/ZIOHSbz00/ZAMTFCY8/3i/qEgJYUgieHOs+Miba/PDDLnr3/iu33DKf7dv38+ab33kdkuus+ygYSopg7xZAoH6y19EYY45QUVEJTz31BRMnfkJBQQlHHVWHadMGcsklJ3gdmussKQTD3p9AS6BeW0gI/2nsxphD++67HYwc+Q6rVm0H4Npru/H00/1p1KiWx5GFhiWFYLBCeMZEjdJSZc2aHbRv34DU1MH07x9bXcKWFILBBpmNiWhr1+6gc+dmiAgnn3wU7703jN6921O3bg2vQws5G2gOBhtkNiYi7dtXwPjx8+jS5UX+8Y/1ZfsvuqhjTCYEsJZCcNiKa8ZEnAULNjB27Ads2ZJDQkIcmzdnex1SWLCkEAw51n1kTKTYs+cAt9++gNdeWw1A9+4tSUsbQrduLTyOLDxYUjhSqv9dDM8YE7a++WY7Awb8jV9/zaVmzXgmTjyXO+/sRUKC9aQfZEnhSB3YCUX7oWYDSArfdVeNMdCpUxPq1q1Bp05NmDFjCJ06NfE6pLBjSeFIZfuttmYls40JK6rKG2+sYfDg46hfvya1ayeyePEoWrWqFzUF7ILN2kxHKsdWWzMmHG3enM0FF/yNq69+h3vv/XfZ/jZt6ltCqIS1FI5U2Z1HNshsTDgoKSll2rQV3HffR+TmFtG4cS169WrrdVgRw5LCkbI5CsaEjfXrdzJ6dDpLl2YCcPnlJ/L88xfSvHkdjyOLHJYUjpTNUTAmLGzalEW3bi9RWFhCy5Z1mTZtIBdffLzXYUUcV5OCiAwAJgPxwAxVnVTu9XbAq0BD3zH3quo8N2MKOmspGBMWOnRoxB//2JmkpASeeqo/DRsmeR1SRHItKYhIPPAC0A/IBFaISLqqrvM77AHgbVV9UUQ6A/OAZLdiCrqiA7D/Z4hLgHptvI7GmJhy4EARjzzyCX/4wwn07NkagFdfvZj4eLt/5ki4+d3rCWxQ1QxVLQTeAoaWO0aB+r7nDYCfXYwn+A62EuonO4nBGBMSn332E926vcSkSUsYO/Z9SksVwBJCELj5l6w1sNVvOxM4rdwxDwMfisgtQB2gr4vxBJ91HRkTUnv3FnDfff9m2rSVAHTu3Izp0wfZLaZB5GZareinpOW2rwReUdU2wEXA6yLyu5hEZKyIrBSRlTt37nQh1MNkg8zGhMy8eT/Spcs0pk1bSUJCHA891Juvvx7L6adb120wudlSyAT8bw5uw++7h0YDAwBUdamIJAFNgR3+B6lqKpAKkJKSUj6xeMfWUTAmJHJy8hk+fC7Z2fmkpLQiLW0IJ598lNdhRSU3k8IKoKOIdAC2AcOAq8odswXoA7wiIicASUAYNQWqYCuuGeMaVUUV4uKEBg2SmDJlAL/+msuf/nS6FbBzkWtJQVWLRWQ8sADndtOZqrpWRB4BVqpqOnAn8LKI3I7TtTRKVcOnJVAVaykY44qff97HTTf9k7PPbsedd/YCYMSIrh5HFRtcvWXGN+dgXrl9D/k9Xwec6WYMrtFS2LvJeW4tBWOCQlWZOXMVd975ITk5BSxblslNN/WgVq1Er0OLGXYf5eHatw1KCqF2c6hR1+tojIl4GRlZjBnzPh9/7HzYGjiwI9OnD7KEEGKWFA5XjhXCMyYYSkpKmTJlOfff/zEHDhTTtGltpkwZwLBhXRArRx9ylhQOV7bNUTAmWObMWc+BA8VceWUXJk8eQLNmVsDOK5YUDleOzVEw5nAVFpawb18BTZrUJj4+jrS0Ifz4424GDz7O69Bint3XdbispWDMYVmxYhspKamMGPEOB282PP74ppYQwoS1FA6XtRSMqZa8vCImTFjEM88so7RUycsrYseOXI46ym7UCCeWFA6XzVEwJmCLF29mzJj32bBhD3Fxwl13ncHEiedRu7bdWRRuLCkcjvxsyN8DCbWgTguvozEmbKkqt946n6lTVwBw0knNSUsbQo8erT2OzByKJYXD4V/ewm6ZM+aQRIT69WuSmBjHAw/05t57z6JGjXivwzKVsKRwOKxktjGHtGtXHhs37uG005zqpQ8+eA7Dh59M587NPI7MBCKgu49EpIaIHOt2MBHDSmYb8zuqyltvfccJJ7zAxRfPJivrAABJSQmWECJIlUlBRAYCa4CFvu1uIvKO24GFNRtkNua/ZGbuZejQt7jyyn+wa1cenTs3Iy+vyOuwzGEIpPvoEZwV0xYBqOo3Md9qsO4jYwAoLVVmzPiau+9eyN69BdSvX5Onn+7P6NGnWImKCBVIUihS1exyP+DIKW/tBus+MgaA0aPTeeWVbwAYMuQ4pk27iNat61fxVSacBTKmsF5ELgfiRKSDiDwHLHM5rvBVUgj7tgAC9ZO9jsYYT1199Uk0b16Ht966lHffvcISQhQIJCmMB04FSoG5QD5wm5tBhbW9W5y1FOq1hYSaXkdjTEh9990OJk/+7TNhnz5Hk5FxK1dcYRVNo0Ug3UcXqOo9wD0Hd4jIJTgJIvYcLG/R0LqOTOwoKCjmL3/5nMce+4yiolJSUlpx5pntAKhTp4bH0ZlgCqSl8EAF++4PdiAR42AhPFtHwcSI5cszOfXUVCZO/ISiolLGjUvhpJOO8jos45JDthRE5AJgANBaRJ7xe6k+TldSbMq2loKJDbm5hTz44CKee24ZqtCxY2NmzBhC797tvQ7NuKiy7qMdwHc4Ywhr/fbvA+51M6iwZiuumRhx//0fM3nycuLihLvvPoOHHz7XlsaMAYdMCqq6ClglIrNUNT+EMYU3m6NgYsT995/NmjU7ePzxvqSktPI6HBMigYwptBaRt0TkWxH5z8GH65GFI1Wbo2CiVnr6D1x00SyKikoAaNasDh99NNISQowJJCm8AvwVEOBC4G3gLRdjCl8HdkJRLtRsCLUaex2NMUGxY0cuw4bNYejQt5g/fwOvvrra65CMhwJJCrVVdQGAqm5U1QeA89wNK0xZK8FEEVXlb3/7lhNOeIHZs9dSu3YikycP4Npru3kdmvFQIPMUCsSZlbJRRG4EtgHN3Q0rTFkhPBMltmzJ4cYbP2D+/A0A9O17NKmpg+jQoZHHkRmvBZIUbgfqArcC/wc0AK5zM6iwZYPMJkp8+OFG5s/fQMOGSTzzTH9GjepmM5INEEBSUNXlvqf7gBEAItLGzaDClnUfmQiWm1tYNvt49OhT2LZtL2PHnkrLlvU8jsyEk0rHFESkh4hcLCJNfdsnishrxGpBPGspmAhUXFzKE08soX3758jIyAKcZTInTDjXEoL5nUMmBRH5CzALGA78S0Tux1lTYTXQKTThhRkbUzARZvXq7Zx22gzuueff7N59gHff/d7rkEyYq6z7aCjQVVUPiEhj4Gff9g+hCS3MFOVB7i8Qlwh1Y7P3zESOgoJiHn30UyZNWkJxcSnt2jUgNXUQF1wQ2+tjmapVlhTyVfUAgKruEZHvYzYhAORscv5tkAxx8Z6GYkxlVq36heHD57J+/S5EYPz4Hjz2WB/q1bNS76ZqlSWFo0XkYHlsAZL9tlHVS6p6cxEZAEwG4oEZqjqpgmMuBx7GWc1ttapeFXj4IWSDzCZC1KyZwMaNWRx3XBNmzBjCWWe18zokE0EqSwqXltueWp03FpF44AWgH5AJrBCRdFVd53dMR+A+4ExVzRKR8J3/kGMls034+vrrXzjllBaICJ07N2P+/OH06tWWpKRA7jo35jeVFcT76AjfuyewQVUzAETkLZxxinV+x4wBXlDVLN85dxzhOd1jg8wmDGVlHeCuuz5k5sxvePPNSxk2rAsA55/fwePITKRy82NEa2Cr33YmcFq5YzoBiMgSnC6mh1X1X+XfSETGAmMB2rXzqCmcY91HJry88856brppHtu376dmzXh2787zOiQTBdxMChVNj9QKzt8ROBdoA3wmIl1UNfu/vkg1FUgFSElJKf8eoZFtcxRMeNi+fT+33DKfOXOcRveZZ7ZlxowhHH98U48jM9Eg4KQgIjVVtaAa750JtPXbboNzW2v5Y5apahGwSUR+wEkSK6pxHveVlsDeg3cfWbPceOerr36mX7/XycrKp06dRCZN6stNN/UgLs5KVJjgqLJKqoj0FJE1wI++7a4i8nwA770C6CgiHUSkBjAMSC93zLv4Kq76Zk13AjKqEX9o7N8GJYVQ+yioUdfraEwM69y5Gc2a1eGCC45h7dqbGD++pyUEE1SBlM6eAgwCdgOo6moCKJ2tqsXAeGABsB54W1XXisgjIjLEd9gCYLeIrMOZLX23qu6u/mW4zMpbGI+UliqpqV+Rne0sflirViKffjqK+fOH0759Q4+jM9EokO6jOFX9qVwFxZJA3lxV5wHzyu17yO+5Anf4HuHL5igYD/zwwy6uv/59Pv98CytWbOPll53PUkcdZa1V455AksJWEekJqG/uwS1AbC3HaS0FE0JFRSU8/fRSHn54MQUFJbRoUZcLL+zodVgmRgSSFMbhdCG1A34F/u3bFztsjoIJkVWrfmH06HRWrdoOwLXXduPpp/vTqFEtjyMzsSKQpFCsqsNcjyScWfeRCYGNG/fQs+cMiotLSU5uSGrqIPr1sw8iJrQCSQorfLeKzgbmquo+l2MKP9Z9ZELgmGMaM2LEydSrV4P/+78+1K1bw+uQTAyq8u4jVT0GeBQ4FVgjIu+KSOy0HPKzIX8PJNR2bkk1Jkj27y/k1lvns3TpbxP/09KGMHnyhZYQjGcCuSUVVf1CVW8FugN7cRbfiQ1lrYSjwdawNUGyYMEGTjxxGs8//yU33vhPnBvxsHWSjeeq7D4Skbo4heyGAScA7wG9XI4rfJSNJ1jXkTlye/Yc4PbbF/Daa6sBOPXUlqSlDbFkYMJGIGMK3wHvA0+o6mcuxxN+yu48skFmc2TmzFnHzTfPY8eOXJKSEpg48VzuuOMMEhICarAbExKBJIWjVbXU9UjCla2jYIIgOzufsWPfJysrn9692/Pyy4Pp1KmJ12EZ8zuHTAoi8rSq3gn8Q0R+V5k0kJXXokKOtRTM4VFVSkuV+Pg4GjZMYtq0gWRlHeCGG1KsXpEJW5W1FGb7/q3WimtRJ9taCqb6Nm/OZuzY9zn//A7ce+9ZAGUL4BgTzg7ZmamqX/qenqCqH/k/cAaco19JIezbAgg0SPY6GhMBSkpKmTJlOV26TGPhwgymTv2S/Pxir8MyJmCBjHBdV8G+0cEOJCzt/Qm0FOq1hXi7b9xUbv36nfTu/Qq33fYvcnOLGDasC19/fYOtk2wiSmVjClfg3IbaQUTm+r1UD8iu+KuijM1kNgEoLi7l8cc/55FHPqWwsIRWrerx4osDGTLkOK9DM6baKvsI8yXOGgptgBf89u8DVrkZVNiwQngmAHFxwocfZlBYWMKYMd154ol+NGyY5HVYxhyWQyYFVd0EbMKpihqbrBCeOYQDB4rYt6+Q5s3rEBcnzJgxmK1b93L++bZcq4lshxxTEJFPfP9micgev0eWiOwJXYgesu4jU4FPP/2Jrl2nc/XVc8vKU3Ts2MQSgokKlXUfHVxys2koAglL1lIwfvbuLeC++/7NtGkrAUhMjGfXrjyaNavjcWTGBE9lt6QenMXcFohX1RLgDOAGIPp/C1StpWDKzJ//I126TGPatJUkJMQxYcI5fP31WEsIJuoEcq/cu0APETkGeA34J/AGMMjNwDyXtwOKciGpkfMwMUlVGTPmfdLSnHsrUlJaMXPmEE46ycqom+gUyDyFUlUtAi4BnlPVW4DW7oYVBqzryOCUsm7Tpj5JSQk89VQ/li4dbQnBRLWAluMUkT8CI4CLffsS3QspTFghvJj188/72LhxD2ef3R6A//mfsxkx4mSOOaaxx5EZ475AZzSfh1M6O0NEOgBvuhtWGLCS2TFHVUlL+5rOnV/g0kvfZvfuPABq1Ii3hGBiRpUtBVX9TkRuBY4VkeOBDar6f+6H5jFrKcSUjIwsxox5n48/3gTAoEGdKCqK3YrxJnYFsvLa2cDrwDZAgBYiMkJVl7gdnKdsNnNMOFjA7oEHFpGXV0TTprWZMmUAw4Z1sdXQTEwKZEzhWeAiVV0HICIn4CSJFDcD85ytoxATRo58lzfeWAPAVVedxHPPXWC3mZqYFsiYQo2DCQFAVdcD0V0ytCgPcrdDXCLUbeN1NMZFY8Z0p02b+qSnD2PWrEssIZiYF0hL4WsReQmndQAwnGgviFc2ntAB4uK9jcUE1YoV2/j4403cc4+z8M255yazYcMt1Kxp5a2NgcCSwo3ArcCfccYUPgWedzMoz5WttmZdR9EiL6+ICRMW8cwzyygtVXr1alt2y6klBGN+U+lvg4icBBwDvKOqT4QmpDCQY4PM0WTx4s1cf306GzdmERcn3HXXGZx6aiuvwzImLFVWJfV/cEpcDAcWikhFK7BVSkQGiMgPIrJBRO6t5LjLRERFJDwGr202c1TIycnnhhve57zzXmXjxixOOqk5y5aN5skn+1O7dvTPvzTmcFTWUhgOnKyquSLSDJgHzAz0jUUkHmdxnn5AJrBCRNL9B619x9XD6Z5aXt3gXWOF8KLCgw8uIjX1axIT43jwwd7cc89Z1KhhY0TGVKaypFCgqrkAqrpTRAK5U8lfT5yJbhkAIvIWMBRYV+64/wWeAO6q5vu7x+YoRCxVLZtf8NBD57BpUzaTJvXhxBObexyZMZGhsj/0R4vIXN/jHeAYv+25lXzdQa2BrX7bmZQrpCcipwBtVfWDakfultISyHFmtdLAFk2JFKrKG2+s4fzzX6OwsASApk1r8/77V1pCMKYaKmspXFpue2o137ui6aBa9qLT8ngWGFXlG4mMBcYCtGvXrpphVNP+bVBaBHVaQKLdsx4JMjP3Mm7cP/ngg/8AMGvWt1x77SkeR2VMZKpsjeaPjvC9M3EW6DmoDfCz33Y9oAuw2NfcbwGki8gQVV1ZLpZUIBUgJSVFcZMNMkeM0lLl5Ze/4u67F7JvXyENGtTk6af7M2pUN69DMyZiuXmD9gqgo6+q6jZgGHDVwRdVNQe/pT5FZDFwV/mEEHI2yBwRNmzYw5gx77N48WYAhg49jmnTBtKqVT1vAzMmwrmWFFS1WETGAwuAeGCmqq4VkUeAlaqa7ta5j0hZS8GSQjj77LOfWLx4M82b12Hq1Au57LLOVsDOmCAIOCmISE1VLajOm6vqPJxbWf33PXSIY8+tznu7xtZRCFvZ2fk0bJgEwKhR3di5M4/Ro0+hSZPaHkdmTPSo8jZTEekpImuAH33bXUUkestc2DoKYaegoJgJExbRvv1z/PjjbsBZJvPPfz7TEoIxQRZIS2EKMAhndjOqulpEznM1Ki9ZiYuwsmxZJqNHp7Nu3U4AFizYSMeOTTyOypjoFUhSiFPVn8r115a4FI+38rOcR2IdqG33tnspN7eQBx9cxHPPLUMVOnZsTFrakLIidsYYdwSSFLaKSE9AfaUrbgH+425YHsnxq45qg5aeWb48k6uumktGRhbx8cJdd/ViwoRzqFXL6hUZ47ZAksI4nC6kdsCvwL99+6KPzVEICw0bJrFt2166dj2KtLQhVtHUmBCqMimo6g6cOQbRL9vmKHjl88+3cOaZbRERjjuuKR9/fA09erQiMdEK2BkTSlUmBRF5Gb/yFAep6lhXIvKSDTKH3I4dudx663xmz17Lq69ezMiRXQHo1attFV9pjHFDIN1H//Z7ngT8gf8udBc9cmzFtVBRVWbNWsNtt/2LPXsOULt2YlkhO2OMdwLpPprtvy0irwMLXYvIS1YyOyS2bMnhxhs/YP78DQD063c0qamDSU5u6HFkxpjDKXPRAYi++wJLCmHfVpA4qB99lxculi/PpG/f19m/v5CGDZN49tkLuOaarlaiwpgwEciYQha/jSnEAXuAQy6tGbH2/gRa6iSE+BpeRxO1unVrQdu29Tn++Ka88MJFtGxpBeyMCSeVJgVxPr51xalyClCqqu6WrvaKdR25ori4lKlTv2TkyK40blyLmjUTWLLkOho1quV1aMaYClRa+8iXAN5R1RLfIzoTAtgcBResXr2d006bwe23L+COOxaU7beEYEz4CmTd5S9FpLvrkXjNCuEFTX5+MQ888DEpKS/z9de/0K5dA668sovXYRljAnDI7iMRSVDVYuAsYIyIbARycZbZVFWNrkRh3UdB8cUXWxk9Op3vv9+FCIwf34PHHutDvXo1vQ7NGBOAysYUvgS6AxeHKBZvla24Zt1Hh2vDhj2cffZfKS1VjjuuCWlpQzjzTJfX1DbGBFVlSUEAVHVjiGLxjqp1HwXBscc2ZuzY7jRuXIsHHzyHpCQ3V3s1xrihst/aZiJyx6FeVNVnXIjHG3m/QlEuJDWCJJtAFaisrAPceeeHXHttt7KS1tOmDbQ5B8ZEsMqSQjxQF1+LIaplWyuhuubOXc/NN89j+/b9fPXVL3zzzQ2IiCUEYyJcZUnhF1V9JGSReMkK4QVs+/b9jB8/j3/8Yz0AZ53VjhkzBlsyMCZKVDmmEBOyrRBeVVSV115bze23LyArK5+6dWvw+ON9ufHGFOLiYue/ijHRrrKk0CdkUXjNWgpVys7O5847PyQrK58BA45l+vSBtG9v4y/GRJtDJgVV3RPKQDxls5krVFqqlJYqCQlxNGpUi5deGkReXhFXX32ydRcZE6UCmdEc/XJsxbXyvv9+F717/5VJkz4v23fppZ0ZMcIqmhoTzSwpFOVC7nanMmrd1l5H47miohIee+wzunadzpIlW0lLW0V+frHXYRljQsRmF+Vscv6tnwxxsb0e8KpVv3Dddel88812AEaPPoUnn+xnk9CMiSH22241jygqKmHChMU88cQSSkqU5OSGvPzyYPr2tTEWY2KNJYWyQebYTQoJCXEsX76N0lLltttO49FHz6duXVtoyJhYZEkhRgvh7dtXwL59hbRqVQ8RYcaMwWzfvp8zzmjrdWjGGA/ZQHMMthQWLNhAly4vMnz4XA6um9ShQyNLCMYYSwqx1FLYvTuPa655lwEDZrFlSw779hWwe/cBr8MyxoQRV5OCiAwQkR9EZIOI3FvB63eIyDoR+VZEPhKR9m7G8zulJb/dfRTFE9dUlTlz1tG58zRee201SUkJPPFEX5Ytu56mTWt7HZ4xJoy4NqYgIvHAC0A/IBNYISLpqrrO77BVQIqq5onIOOAJ4Aq3Yvqd/ZlQWgR1WkJidP5xVFWGD5/Lm29+B0DRZJGNAAAPYElEQVTv3u15+eXBdOrUxOPIjDHhyM2WQk9gg6pmqGoh8BYw1P8AVV2kqnm+zWVAGxfj+b0YKIQnInTu3Ix69Wrw4osDWbToGksIxphDcvPuo9bAVr/tTOC0So4fDcyv6AURGQuMBWjXLojLO0bpHIVNm7LIyMiiTx8n2d1zz5mMGtWNNm3qexyZMSbcudlSqKhAjlZ4oMjVQArwZEWvq2qqqqaoakqzZs2CF2FOdBXCKykpZfLkZXTp8iJXXDGHHTtyAUhMjLeEYIwJiJsthUzA/x7HNsDP5Q8Skb7A/cA5qlrgYjy/lx09hfDWrdvJ9dens3RpJgBDhhxn6xwYY6rNzaSwAugoIh2AbcAw4Cr/A0TkFOAlYICq7nAxlopFwToKRUUlPP74Ev73fz+lsLCEVq3q8eKLAxky5DivQzPGRCDXkoKqFovIeGABznrPM1V1rYg8AqxU1XSc7qK6wN995Zi3qOoQt2L6nZzIH2i+6qq5zJnj3NA1Zkx3nnyyHw0aJHkclTEmUrla5kJV5wHzyu17yO95XzfPX6n8LOeRWAdqN/csjCN1222n8c0323nppUGcf34Hr8MxxkS42J3R7L/aWgQtGvPJJ5uZOHFx2fZZZ7Vj/fqbLSEYY4IidgviRdhqa3v3FnDPPQuZPv0rAM47rwO9ezsTwBMSYje3G2OCK3aTQgQVwps370duuOEDMjP3kpgYx/33n83pp4d2np8xJjbEblKIgEJ4u3bl8ac//YtZs9YA0LNna9LShtClS+SOgRhjwlvsJoUImM38yCOfMGvWGmrVSuDRR8/ntttOIz7euoqMMe6xpBBm3Ueqiu/2XCZOPJdff83lscfO55hjGnscmTEmFsTmx86SQti3FSQO6gexltIRUFVefvkrevWaSX5+MQCNGtVi9uzLLCEYY0ImNpNCzmZAoV47iPd+LeKNG/fQp89rjB37AcuWZfL222u9DskYE6Nis/soTAaZnQJ2y3nggY85cKCYZs1q8/zzF3L55Sd6GpcxJnbFZlIIg/GEtWt3cN116Xz55TYAhg8/ieeeG2AroRljPBWbSSEMCuGtWrWdL7/cRuvW9XjppUEMHNjJs1iMMeag2EwKHq24tnNnLs2a1QGclkF2dj4jRpxsBeyMMWEjRgeaQ9tSyMsr4q67PiQ5eTLr1+8EnGUyx4/vaQnBGBNWYq+loBrSlsKiRZsYM+Z9Nm7MIi5O+PTTnzjhhCCuHmeMMUEUe0kh71cozoOkxpDU0LXT5OTk8+c/LyQ19WsATjqpOTNnDiUlpZVr5zTGmCMVe0khBOUtPv98C8OGzWHbtn0kJsbx4IO9ueees6hRI961cxpjTDDEXlIIwWprLVrUZffuA5x+ehtmzBjMiSdaATtjTGSIvaTgQktBVVm4MIN+/Y5GRDj22MZ8/vm1dOvWwgrYGWMiSuz9xQryxLWtW3MYPPhNLrjgb/z1r9+U7T/11FaWEIwxESf2WgpBKnFRWuoUsLv77oXs21dIgwY1qVnTxgyMMZEt9pJCEFoKP/64mzFj3ueTT34C4OKLj+eFFy6iVat6wYjQGGM8E1tJoSjXuSU1vgbUPbxbQ7/4Yit9+rxGfn4xzZvXYerUC7nsss5layAYY0wki62kcHDSWv0OEHd4XT0pKa3o2LExp5zSkmee6U+TJlbAzhgTPWIsKVT/zqOCgmKeeuoLbrghhaZNa1OjRjxLllxHvXo1XQrSGGO8E1tJoZpzFJYty2T06HTWrdvJ+vW7+NvfLgGwhGCMiVqxlRQCbCnk5hbywAMfM3nyclShU6cm3HDDqSEI0BhjvBVbSSGAlsJHH2UwZsz7bNqUTXy8cPfdvZgw4VySkmLrW2WMiU2x9ZeuipLZ//nPbvr1ex1V6NatBWlpQ+jevWUIAzTGGG/FTlIoLYGczc7zBh0qPKRTpybcdttpNGtWh7vv7kViok1GM8bElthJCvszobQI6rSEROc20l9/3c+tt/6LG288lfPOcxLFs88O8DJKY4zxlKvFeURkgIj8ICIbROTeCl6vKSKzfa8vF5Fk14LxG2RWVV5/fTWdO0/j7bfXcvvtC1BV105tjDGRwrWkICLxwAvAhUBn4EoR6VzusNFAlqoeCzwLPO5WPAeTwpbCTgwc+AYjR77Lnj0H6N//GN59d5jNSDbGGNztPuoJbFDVDAAReQsYCqzzO2Yo8LDv+RxgqoiIuvCxvTQrg+lLenDPgnbsz9tAo0ZJPPvsBYwc2dUSgjHG+LiZFFoDW/22M4HTDnWMqhaLSA7QBNgV7GByft7ExIXnsD8PLr30BKZOvYgWLeoG+zTGGBPR3EwKFX38Lt8CCOQYRGQsMBagXbt2hxVMo3rCjCs/pLDHRC4dc/FhvYcxxkQ7N5NCJtDWb7sN8PMhjskUkQSgAbCn/BupaiqQCpCSknJ4XUuD3mTwQKWCnGOMMcbHzbuPVgAdRaSDiNQAhgHp5Y5JB67xPb8M+NiN8YQyIiC2GpoxxhyKay0F3xjBeGABEA/MVNW1IvIIsFJV04E04HUR2YDTQhjmVjzGGGOq5urkNVWdB8wrt+8hv+f5wB/djMEYY0zgrC/FGGNMGUsKxhhjylhSMMYYU8aSgjHGmDKWFIwxxpSRSKsOKiI7gZ8O88ub4kIJjTBn1xwb7Jpjw5Fcc3tVbVbVQRGXFI6EiKxU1RSv4wglu+bYYNccG0JxzdZ9ZIwxpowlBWOMMWViLSmkeh2AB+yaY4Ndc2xw/ZpjakzBGGNM5WKtpWCMMaYSUZkURGSAiPwgIhtE5N4KXq8pIrN9ry8XkeTQRxlcAVzzHSKyTkS+FZGPRKS9F3EGU1XX7HfcZSKiIhLxd6oEcs0icrnvZ71WRN4IdYzBFsD/7XYiskhEVvn+f1/kRZzBIiIzRWSHiHx3iNdFRKb4vh/fikj3oAagqlH1wCnTvRE4GqgBrAY6lzvmJmC67/kwYLbXcYfgms8Davuej4uFa/YdVw/4FFgGpHgddwh+zh2BVUAj33Zzr+MOwTWnAuN8zzsDm72O+wivuTfQHfjuEK9fBMzHWbnydGB5MM8fjS2FnsAGVc1Q1ULgLWBouWOGAq/6ns8B+ohIRUuDRooqr1lVF6lqnm9zGc5KeJEskJ8zwP8CTwD5oQzOJYFc8xjgBVXNAlDVHSGOMdgCuWYF6vueN+D3KzxGFFX9lApWoPQzFHhNHcuAhiLSMljnj8ak0BrY6red6dtX4TGqWgzkAE1CEp07Arlmf6NxPmlEsiqvWUROAdqq6gehDMxFgfycOwGdRGSJiCwTkQEhi84dgVzzw8DVIpKJs37LLaEJzTPV/X2vFlcX2fFIRZ/4y99iFcgxkSTg6xGRq4EU4BxXI3JfpdcsInHAs8CoUAUUAoH8nBNwupDOxWkNfiYiXVQ12+XY3BLINV8JvKKqT4vIGTirOXZR1VL3w/OEq3+/orGlkAm09dtuw++bk2XHiEgCTpOzsuZauAvkmhGRvsD9wBBVLQhRbG6p6prrAV2AxSKyGafvNT3CB5sD/b/9nqoWqeom4AecJBGpArnm0cDbAKq6FEjCqREUrQL6fT9c0ZgUVgAdRaSDiNTAGUhOL3dMOnCN7/llwMfqG8GJUFVes68r5SWchBDp/cxQxTWrao6qNlXVZFVNxhlHGaKqK70JNygC+b/9Ls5NBYhIU5zupIyQRhlcgVzzFqAPgIicgJMUdoY0ytBKB0b67kI6HchR1V+C9eZR132kqsUiMh5YgHPnwkxVXSsijwArVTUdSMNpYm7AaSEM8y7iIxfgNT8J1AX+7htT36KqQzwL+ggFeM1RJcBrXgD0F5F1QAlwt6ru9i7qIxPgNd8JvCwit+N0o4yK5A95IvImTvdfU984yQQgEUBVp+OMm1wEbADygGuDev4I/t4ZY4wJsmjsPjLGGHOYLCkYY4wpY0nBGGNMGUsKxhhjylhSMMYYU8aSggk7IlIiIt/4PZIrOTb5UNUkq3nOxb5KnKt9JSKOO4z3uFFERvqejxKRVn6vzRCRzkGOc4WIdAvga/4kIrWP9NwmNlhSMOHogKp283tsDtF5h6tqV5xiiU9W94tVdbqqvubbHAW08nvtelVdF5Qof4tzGoHF+SfAkoIJiCUFExF8LYLPRORr36NXBcecKCJf+loX34pIR9/+q/32vyQi8VWc7lPgWN/X9vHV6V/jq3Nf07d/kvy2PsVTvn0Pi8hdInIZTn2pWb5z1vJ9wk8RkXEi8oRfzKNE5PnDjHMpfoXQRORFEVkpzjoKE337bsVJTotEZJFvX38RWer7Pv5dROpWcR4TQywpmHBUy6/r6B3fvh1AP1XtDlwBTKng624EJqtqN5w/ypm+sgdXAGf69pcAw6s4/2BgjYgkAa8AV6jqSTgVAMaJSGPgD8CJqnoy8Kj/F6vqHGAlzif6bqp6wO/lOcAlfttXALMPM84BOGUtDrpfVVOAk4FzRORkVZ2CUxfnPFU9z1f64gGgr+97uRK4o4rzmBgSdWUuTFQ44PvD6C8RmOrrQy/BqelT3lLgfhFpA8xV1R9FpA9wKrDCV96jFk6CqcgsETkAbMYpv3wcsElV/+N7/VXgZmAqzvoMM0Tkn0DApblVdaeIZPhq1vzoO8cS3/tWJ846OGUf/FfdulxExuL8XrfEWXDm23Jfe7pv/xLfeWrgfN+MASwpmMhxO/Ar0BWnhfu7RXNU9Q0RWQ4MBBaIyPU4ZYZfVdX7AjjHcP+CeSJS4Robvno8PXGKsA0DxgPnV+NaZgOXA98D76iqivMXOuA4cVYgmwS8AFwiIh2Au4AeqpolIq/gFIYrT4CFqnplNeI1McS6j0ykaAD84quRPwLnU/J/EZGjgQxfl0k6TjfKR8BlItLcd0xjCXx96u+BZBE51rc9AvjE1wffQFXn4QziVnQH0D6c8t0VmQtcjLMOwGzfvmrFqapFON1Ap/u6nuoDuUCOiBwFXHiIWJYBZx68JhGpLSIVtbpMjLKkYCLFNOAaEVmG03WUW8ExVwDficg3wPE4Sxauw/nj+aGIfAssxOlaqZKq5uNUoPy7iKwBSoHpOH9gP/C93yc4rZjyXgGmHxxoLve+WcA6oL2qfunbV+04fWMVTwN3qepqnLWZ1wIzcbqkDkoF5ovIIlXdiXNn1Ju+8yzD+V4ZA1iVVGOMMX6spWCMMaaMJQVjjDFlLCkYY4wpY0nBGGNMGUsKxhhjylhSMMYYU8aSgjHGmDKWFIwxxpT5f6ECaWxsd/bDAAAAAElFTkSuQmCC\n",
        "text/plain": "<Figure size 432x288 with 1 Axes>"
       },
       "metadata": {
        "needs_background": "light"
       },
       "output_type": "display_data"
      }
     ],
     "persistentId": "2c810e70-a822-4d9a-a9b0-2622dfa247bf",
     "text": "y_true = coding_scores_not_low_complexity_unspliced['in_cds_no_stop_codon']\ny_pred = coding_scores_not_low_complexity_unspliced['jaccard_in_peptide_db']\n\nfpr, tpr, thresholds = metrics.roc_curve(y_true, y_pred)\n\nfig, ax = plt.subplots()\nlw = 2\nax.plot(fpr, tpr, lw=2, color='darkorange')\nax.plot([0, 1], [0, 1], color='navy', lw=lw, linestyle='--')\nax.set(xlabel='False Positive Rate', ylabel='True Positive Rate')\n\nprint(metrics.accuracy_score(y_true, y_pred > 0.9))\n\nprint(metrics.classification_report(y_true, y_pred > 0.9))"
    },
    "executionTime": "2019-10-12T03:39:16.309Z"
   },
   {
    "cell": {
     "executionCount": 58,
     "executionEventId": "7383379a-bde3-441f-af2d-377de26e9f8d",
     "hasError": false,
     "id": "f5b034a4-325d-41af-8295-7b22ef456d06",
     "outputs": [
      {
       "data": {
        "text/plain": "<matplotlib.axes._subplots.AxesSubplot at 0x7f3dbad23828>"
       },
       "execution_count": 58,
       "metadata": {},
       "output_type": "execute_result"
      },
      {
       "data": {
        "image/png": "iVBORw0KGgoAAAANSUhEUgAAAYYAAAELCAYAAADdriHjAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDMuMC4zLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvnQurowAAIABJREFUeJzt3Xuc3HV97/HXe2Z3s7tJCCRZCCSEBIiGix7AgFJsRcVzQBBapQKtPtqettSWVIX2VHvqUQ72nIfXRx9oUcuxXvCGaFUCoVwqIMg9YEggQImBwEIMm5D7Xmfnc/6Y34SZzezuBHbmN7vzfj4e85j5/X7f328+u9nMe76/y/eniMDMzKwok3YBZmbWWBwMZmZWxsFgZmZlHAxmZlbGwWBmZmUcDGZmVsbBYGZmZRwMZmZWxsFgZmZlWtIu4NWYO3duLFq0KO0yzMwmlYcffnhLRHSN125SBsOiRYtYtWpV2mWYmU0qkjZW0867kszMrIyDwczMyjgYzMysjIPBzMzKOBjMzKyMg8HMzMo4GMzMrIyDwczMyjgYmtjTTz/N+87/fd5z7nlcc801aZdjZg3CwdDE1qxZw9YtPezoG+KX99yTdjlm1iBqGgySviHpJUmPjbJckr4kab2kNZJOqmU9Vu7FF19E2VZyBy7kxRdfTLscM2sQte4xfAs4c4zlZwFLksfFwFdrXI+V2LJlCzFtOtHWye5duxgYGEi7JDNrADUdRC8i7pK0aIwm5wHXREQA90s6UNKhEbGplnVZwdatLzOcbSdaOwDYtm0b8+bNS7kqq9bw8DADAwMMDg4yMDBAf3//3tfFR6Xp/v5+BgYGyOfzaf8IY9q6dSs9PT10dXUxZ86ctMsZUyaToa2tjWnTpu19FKfb29tHXVZ8tLQ01nimaVczH3i+ZLo7medgqIO+vj4i00pkWgHo7+9PuSIbHh7m0ksvZc2aNWXzW1tamDFjOhAMDAwyODhEbnj4Vb9PWxayGb3GamtnOB9k2zo4++yzWblyJcODfQ1dbz5gcDiIeHXrZ7MZprW10dbWRjabZfeePQwMDJa1Oeigg7jmmmuYOXPmBFQ8trSDodK/dMVfraSLKexuYuHChbWsqWkM5oYIZUCFPYq5XC7liiwi9gkFgKFcjm3bd4y5bns2z/SWYHprML0l6GwJOlvzdLa8Mj29NU9HNmjLQouC1iy0KmjNBC0ZaM0ErclzSwbS+iy+4dl29Kb38VeXLEcEPHIN71mUzheXCBjKw1Be5AKGhsVQJNPJ/MID+nOiNyf25DLsyYneoVeme3NiT8l0qeHhPL19/fT2jf4zbtu2jf7+/qYIhm7g8JLpBUDFo6ARcTVwNcCyZcteZS5bqWltbSj6IArfPNva2lKuyFpaWrjzzjsB6O3tpaenhy1btuzzvHPnDgYHBxkaGGBwaLDwemiI/sEhdg3kGNw9NCH1ZDPQmlEhQJLQaMvkacvkmZbJMy0L07JBezaYlg2mZZLnZH75g0K7zCvzWjOgCuGz9KAcX7ppJRFw67+v5MNLK39piYBcwMCwGBgW/cNiYBgGk+mBvOjPFZ4Hhl9pV/6AgXyG/nyGwXzmlQ/7YRjKB7kJ2uPW2tpCa0sLbdPb6GptpS3pIbS1tdHa1sb06TPo6upi7ty5Zc9dXV3MmDEDVfpF1UjawbACWC7pWuDNwA4fX6ifmTNmoE070HChy9rZ2ZlyRVaqs7OTI444giOOOGK/140Icrnc3sAYHBwsexTnjXweubxSm/7+fvr7++nr62VHby/9fX309/bTPzBAf/8A+f3YnyJBZ4s4uCPHIe055nUOc0jnMPM681x89Es8+8g1/OWSHG2Z4IHNbWzuy7KpN8PmvhZe6m9l92CQ34+viZKY1tZKe3s77e3tdMzooKOjk5kdHRzc0bF3n39r8sHdOvIDvML88dq0trbS2tpKJjN5rg6oaTBI+gFwOjBXUjfwKaAVICK+BtwEvBtYD/QCf1LLeqzc7NmzaRn+NbmhPiRx4IEHpl2STRBJez+Q6ikiGBwcpK+vb2+AFEKkr+K8/v5+du3axQsvvMDG5zfy4MaevcEiwdJZQ1y/cTqDJYdT5s45iMMXH8Exhx/OrFmz9n7It7e309HRQUdHx6jzpk2bVtdv3pNVrc9Kumic5QFcUssabHQHH3ww0b+bzMBuDjzooIY7M8ImH0l7v3W/GoODg2zatInu7m5+9KMfsXr1alpaWrj88k+wYMEC5s+fT0dHxwRXbSP5k6CJzZs3DyJPdvdmDluy/7srzCZaW1vb3t1n/f39rF69mtNOO43TTz897dKayuTZ6WUTbv78+QBkBnYx/7DDUq7GrFxxl493/dSfg6GJlV7Mduihh6ZYidm+HAjpcTA0sa6urr2vfcWzmRU5GJpY6QHCRh9ywMzqx8FgAD5V1cz2cjAYQF0uszezycHBYICvejazVzgYDID29va0SzCzBuFgMABf9WxmezkYDIBsNpt2CWbWIBwMZmZWxsFgZmZlHAxmZlbGwWBmZmUcDGZmVsbBYGZmZRwMZmZWxsFgZmZlHAxmZlbGwWBmZmUcDGZmVsbBYGZmZRwMZmZWxsFgZmZlHAxmZlbGwWBmZmUcDGZmVqbmwSDpTElPSVov6eMVli+UdIekX0laI+ndta7JzMxGV9NgkJQFrgLOAo4FLpJ07IhmnwCui4gTgQuBr9SyJjMzG1utewynAOsjYkNEDALXAueNaBPAAcnrWcCLNa7JzMzG0FLj7c8Hni+Z7gbePKLN5cCtkv4amA6cUeOazMxsDLXuMajCvBgxfRHwrYhYALwb+I6kfeqSdLGkVZJW9fT01KBUMzOD2gdDN3B4yfQC9t1V9KfAdQARcR/QDswduaGIuDoilkXEsq6urhqVa2ZmVQWDpDmSvizpEUkPS7pS0pwqVn0IWCJpsaQ2CgeXV4xo8xzwzuR9jqEQDO4SmJmlpNoew7XAS8D7gPMpfHD/cLyVIiIHLAduAZ6gcPbR45KukHRu0uxvgD+X9CjwA+CPI2Lk7iarsXw+n3YJZtYgqj34PDsiPl0y/Y+SfreaFSPiJuCmEfM+WfJ6HXBalXVYjeTzeTIZX+9oZtX3GO6QdKGkTPJ4P7CyloVZfbnHYGZFY/YYJO2icBaRgMuA7ySLssBu4FM1rc7qxnvvzKxozGCIiJn1KsTS5R6DmRWN12M4aazlEfHIxJZjaXGPwcyKxjv4/MXkuR1YBjxKYbfSG4EHgLfWrjSrJweDmRWNefA5It4eEW8HNgInJReYvQk4EVhfjwLNzKy+qj0raWlErC1ORMRjwAm1KcnMzNJU7XUMT0j6OvBdCmcpfYDCBWs2RUiVhrUys2ZUbTD8CfCXwEeS6buAr9akIjMzS1VVwRAR/cA/JY99SPq3iHjfRBZm9eWrns2saKI+DY6coO1YSrwrycyKJioYfK7jJOceg5kV+dPAAPcYzOwVExUM/lSZ5NxjMLOiqj8NJHVIev0oiz82QfVYStxjMLOiau/g9h5gNXBzMn2CpL13YouIW2tTntWLg8HMiqrtMVwOnAJsB4iI1cCi2pRkZmZpqjYYchGxo6aVmJlZQ6j2yufHJP0BkJW0BPgwcG/tyjIzs7RU22P4a+A4YAD4AbAT+GitijIzs/RUOyRGL/APycPMzKaw8e7gdgNjXNUcEedOeEVmZpaq8XoMX0ie3wvMozDsNsBFwLM1qslSEBE+ZdXMgHGCISJ+ASDp0xHxOyWLbpB0V00rMzOzVFR78LlL0t4RVCUtBrpqU5Klwfd8NrOiak9XvRS4U9KGZHoR8Bc1qchS4d1IZlZU7VlJNyfXLyxNZj0ZEQO1K8vMzNIy3llJ74iI2yW9d8SioyQRET+pYW1WR+4xmFnReD2GtwG3A++psCwAB4OZ2RQz3llJn0peXhERz5QuSw5Aj0vSmcCVQBb4ekR8pkKb91MYqC+ARyPiD6rZtpmZTbxqz0r6twrzfjzeSpKywFXAWcCxwEWSjh3RZgnw98BpEXEcHmrDzPCZcmka7xjDUgpjJM0acZzhAKC9iu2fAqyPiA3J9q4FzgPWlbT5c+CqiNgGEBEvVV++mU1VxWDw8a/6G+8Yw+uBc4ADKT/OsIvCB/p45gPPl0x3A28e0eZ1AJLuobC76fKIuLmKbZvZFFYMhuHh4ZQraT7jHWO4Hrhe0qkRcd+r2H6lqB/ZP2wBlgCnAwuAuyUdHxHbyzYkXQxcDLBw4cJXUYqZTSabN28GoLe3N+VKmk+1xxg2S7pBUo+klyRdX3ol9Bi6gcNLphcAL1Zoc31EDCUHuJ+iEBRlIuLqiFgWEcu6unzRtdlU9+ADDwCw/j+fSrmS5lNtMHwfuA44FDgM+BGF+zKM5yFgiaTFktqAC4EVI9r8DHg7gKS5FHYtbcDMmtqe3j0A9Pb1p1xJ86k2GBQR34mIXPL4LmMMx10UETlgOXAL8ARwXUQ8LukKScUhu28BtkpaB9wB/I+I2Lr/P4q9Fj4DxBrNcccdD8DRRx+dciXNp9qxku6Q9HHgWgqBcAGwUtJsgIh4ebQVI+Im4KYR8z5Z8jqAy5KHpcTDblujOeaYY7j++us5aPbstEtpOtUGwwXJ88iB8/47haCo5niDNbB8Pk8mU20H0qz2Wltby56tfqodRK+qq5xt8vKuJGs07sGmp6qviJI6JX1C0tXJ9BJJ59S2NKunfD6fdglmZRwM6al238E3gUHgt5LpbuAfa1KRpcI9Bms0/ptMT7XBcFREfA4YAoiIPipfvGaTlHsM1miKweCAqL9qg2FQUgfJKaqSjgJ8o54pxP/5rNHs3r0bgIEBf9TUW7VnJX0KuBk4XNL3gNOAP65VUVZ/7jFYo3nwwQcBeP65Z9MtpAlVe1bSbZIeAd5CYRfSRyJiS00rM7Om9uIL3QBs+s1Lvs6mzvbnxPW3Ae+kMHzFb9emHDOzgsVHHgXAkYsXORTqrKoeg6SvAEfzyvhIfyHpjIi4pGaVmVlTu+SSSzj11FM55phj0i6l6VR7jOFtwPHJ8BVI+jawtmZVmVnTmzNnDu9617vSLqMpVbsr6Smg9CYIhwNrJr4cMzNLW7U9hjnAE5IeTKZPBu6TtAIgIs4ddU0zM5tUqg2GT47fxCYzH9wzs6JqT1f9xVjLJd0XEadOTElmZpamiRpnuX2CtmMpcY/BzIomKhg8nsIk52AwsyLfmcXMzMpMVDD46+Yk5x6DmRVNVDB8cIK2YylxMJhZUbV3cHuvpKcl7ZC0U9IuSTuLyyPisdqVaGZm9VTtdQyfA94TEU/UshgzM0tftbuSNjsUzMyaQ7U9hlWSfgj8jJI7t0XET2pSldWd7+BmZkXVBsMBQC/wX0vmBeBgmCIcDGZWVO2QGH9S60LMzKwxjBkMkv4uIj4n6ctUuLo5Ij5cs8qsrnzPZzMrGq/HUDzgvKrWhVi6vCvJzIrGDIaIuCF5/vZY7SR9OSL+epRlZwJXAlng6xHxmVHanQ/8CDg5IhxEdTY8PJx2CWbWICbqyufTKs2UlAWuAs4CjgUuknRshXYzgQ8DD0xQPWZm9irVehC9U4D1EbEhIgaBa4HzKrT7NIWL6PprXI+Nwj0GMyuqdTDMB54vme5O5u0l6UTg8Ii4sca1mJlZFWo9umql+XuPckrKAP8E/M24byBdLGmVpFU9PT2vrkoblc9KMrOiiQqGK0eZ3w0cXjK9AHixZHomcDxwp6RngbcAKyQtG7mhiLg6IpZFxLKurq6Jqdr28llJZlY03nUMNzDG3dki4tzk+VujNHkIWCJpMfACcCHwByXr7wDmlrzfncDf+qwkM7P0jHcdwxeS5/cC84DvJtMXAc+Ot/GIyElaDtxC4XTVb0TE45KuAFZFxIpXVbWZmdXMeNcx/AJA0qcj4ndKFt0g6a5q3iAibgJuGjHvk6O0Pb2abZqZWe1Ue4yhS9KRxYlk15B39E8hvoObmRVVO7rqpRQOEG9IphcBF9ekIkuFg8HMisYNhuSU0p3AEmBpMvvJiBgYfS2bbBwMZlY0bjBERF7SFyPiVODROtRkKXAwmFlRtccYbpX0PvnTY8ryP62ZFVV7jOEyYDqQk9RP4YrmiIgDalaZ1ZWDwcyKqr2D28xaF2JmZo2h2h4Dkg6icAC6vTgvIqq6lsEan3sMZlZUVTBI+jPgIxTGOlpNYUyj+4B31K40qycHg5kVVXvw+SPAycDGiHg7cCLgIU6nEAeDmRVVGwz9EdEPIGlaRDwJvL52ZZmZWVqqPcbQLelA4GfAbZK2UT58tk1y7jGYWVG1ZyX9XvLyckl3ALOAm2tWlZmZpaaqXUmS3iJpJuwdcfUOCscZzMxsiqn2GMNXgd0l03uSeWZmNsVUGwyKkns/RkSe/bgGwszMJo9qg2GDpA9Lak0eHwE2jLuWTRq+57OZFVUbDB8CfovCfZu7gTfj+zGYmU1J1Z6V9BJwYY1rsRS5x2BmRdWelfTt5DqG4vRBkr5Ru7Ks3vL5fNolmFmDqHZX0hsjYntxIiK24dNVzcympGqDIZOMrgqApNn4rKQpZXh4OO0SzKxBVPvh/kXgXkk/TqZ/H/g/tSnJ0uBdSWZWVO3B52skPQy8ncLd294bEetqWpnVlYPBzIqq3h0UEY9L6iG5UY+khRHxXM0qs7ryriQzK6r2rKRzJT0NPAP8AngW+Pca1mV1lsvl0i7BzBpEtQefP03hrm3/GRGLgXcC99SsKqs7B4OZFVUbDEMRsZXC2UmZiLgDOKGGdVmdDQ4Opl2CmTWIaoNhu6QZwN3A9yRdCVT1FVPSmZKekrRe0scrLL9M0jpJayT9XNIR1ZdvE8XBYGZF1QbDuUAvhXs/3wysB84ZbyVJWeAq4CzgWOAiSceOaPYrYFlEvBH4MfC5KmuyCdTf3592CWbWIMYMBkm/TF5uBrYD24B/Bv4v8IykZyT91RibOAVYHxEbImIQuBY4r7RBRNwREb3J5P3Agv3/Mey1GhgYSLsEM2sQY56uGhFvTZ5nVlouaQ5wL/CVUTYxH3i+ZLo4Muto/hSf7VQ3pdcu9PX1pViJmTWS1zSsRURslXT6GE0q3WG+4jCekj4ALAPeNsryi0mG+l64cOH+FWoVlfYSvCvJzIqqPcYwqojYNMbibuDwkukFwIsjG0k6A/gH4NyIqLhPIyKujohlEbGsq6vrtZRsidIwcI/BzIpeczCM4yFgiaTFktoo3NNhRWkDSScC/0IhFF6qcT1WojQM3GMws6KaBkNE5IDlwC3AE8B1ydAaV0g6N2n2eWAG8CNJqyWtGGVzNsHcYzCzSmo+dHZE3ATcNGLeJ0ten1HrGqyyPXv27H3d29s7Rkszaya13pVkDaw0DBwMZlbkYGhixR5DUN57MLPm5mBoYnvDQFl2796dbjFm1jAcDE1s165dhReZLDt37kq3GDNrGA6GJrZz504AQll2JK/NzBwMTWzHjh2gDCjDrp07iKh4UbqZNRkHQxPbvn07oQxkMgwPD/sAtJkBDoamtvXll0GZQjgA27ZtS7kiM2sEDoYmtmXL1kIoJMGwZcuWlCsys0bgYGhSuVyObS9vhUwWlAWgp6cn5arMrBE4GJpUT09P4WCzskSmEAy/+c1vUq7KzBqBg6FJPfvsswBoqJds38ugLLfddlu6RZlZQ6j5IHrWmDZs2ACA8rm98156yaOem5l7DE1r7dq1heMLJQYGBvZe9GZmzcvB0IRyuRyPrllDPtO2z7JHH300hYrMrJE4GJrQunXr6OvtJVqmjVgiHnzwwVRqMrPG4WBoQvfff3/hwraW8h5DPtvGvffe56ExzJqcg6EJ/fKeexieecjeC9uKomUaW7duYf369SlVZmaNwMHQZDZt2sRzGzcyNGvhvgtb2gG477776lyVmTUSB0OTKR5DyM1asM+yyGTIT5/r4wxmTc7B0GRWrVoF02YQ7QdUXD50wGGsW7fOI62aNTEHQxPJ5/P8avVqhmYeClLFNsMHHEY+n+exxx6rc3Vm1igcDE3k17/+Nbt37SI3c96obYanHwyZLA8//HAdKzOzRuJgaCLFYwfDB8wfvVG2heEZh3Df/Q/UqSozazQOhiZy9913E9PnEG2dY7YbmrWA55/bSHd3d50qM7NG4mBoEhs3buTJJ59k4KAjx22bm70YJG6++eY6VGZmjcbB0CRuvPFGUIbc3KPHbRtt08kdMJ+VN93E0NBQHaozs0biYGgCu3btYsUNNzA0ezHR2lHVOoOHHMu2l1/m5z//eY2rM7NG42BoAitXrmSgv5/BecdXvc7wAfOJztn88LrrPHaSWZOpeTBIOlPSU5LWS/p4heXTJP0wWf6ApEW1rqmZRATXr7iB/MxDyHfOqX5FiYGupTyzYQNPPvlk7Qo0s4ZT02CQlAWuAs4CjgUuknTsiGZ/CmyLiKOBfwI+W8uams3TTz/NphdfYGDu6/Z73aE5R6JMC7fffnsNKjOzRlXrHsMpwPqI2BARg8C1wHkj2pwHfDt5/WPgndIol+Xaftu4cSMA+elz93/lbBv5jlk899xzE1yVmTWyWgfDfOD5kunuZF7FNhGRA3YA+7HPw8aydetWAPKVDjoPD9Le3s75559Pe3s7DA/u26Slg82bfS9os2ZS62Co9M1/5JHMatog6WJJqySt6unpmZDimsGRRxauW8ju2brPMuUGOeecc1i+fDlnn302yo0Ihgha+15myZLxT3E1s6mjpcbb7wYOL5leALw4SptuSS3ALODlkRuKiKuBqwGWLVvm02SqdPzxx5PNZmnd+muGZ5V31qKljRtvvJGIYOXKlURL+RXR2V2biMFeTjjhhHqWbGYpq3WP4SFgiaTFktqAC4EVI9qsAP4oeX0+cHv4/MgJ09nZyfvf/35at64nu2tz2bL8zHnsznRy3Q23sDvTSb50cL38MJ3P3c8h8+Zxxhln1LlqM0tTTXsMEZGTtBy4BcgC34iIxyVdAayKiBXAvwLfkbSeQk/hwlrW1Iw++MEPcutt/wHP3s2uY94DLdMAGFj4llHXmfb8Q9C3ncsu/yzTpk2rV6lm1gA0Gb+cL1u2LFatWpV2GZPK2rVr+ehHP8rg9EMYOGzsXUOZ3T20dz/E+eefz/Lly+tUoZnVmqSHI2LZeO1qfYzBGsQb3vAGLrvsMj7/+c/TuXPkYZ59nXTSSXzoQx+qQ2Vm1mgcDE3k7LPPZunSpWzfvn3MdplMhuOOO46WFv95mDUj/89vMkcddVTaJZhZg/MgemZmVsbBYGZmZRwMZmZWxsFgZmZlHAxmZlbGwWBmZmUcDGZmVmZSDokhqQfYmHYdU8hcYEvaRZhV4L/NiXVERHSN12hSBoNNLEmrqhk/xaze/LeZDu9KMjOzMg4GMzMr42AwSO6MZ9aA/LeZAh9jMDOzMu4xmJlZGQ+7PQVJGgbWlsz63Yh4dpS2i4AbI+L42ldmBpLmAD9PJucBw0BPMn1KRAymUpjt5WCYmvoiYuz7d5qlJCK2AicASLoc2B0RXyhtI0kUdnXn61+heVdSk5C0SNLdkh5JHr9Voc1xkh6UtFrSGklLkvkfKJn/L5Ky9f8JbKqTdLSkxyR9DXgEOFzS9pLlF0r6evL6EEk/kbQq+dt8S1p1T0UOhqmpI/kQXy3pp8m8l4B3RcRJwAXAlyqs9yHgyqS3sQzolnRM0v60ZP4w8Ie1/xGsSR0L/GtEnAi8MEa7LwGfSy5+ez/w9XoU1yy8K2lqqrQrqRX4Z0nFD/fXVVjvPuAfJC0AfhIRT0t6J/Am4KFC754OCiFjVgu/joiHqmh3BvD65G8S4CBJHRHRV7vSmoeDoXlcCmwG/guFnmL/yAYR8X1JDwBnA7dI+jNAwLcj4u/rWaw1rT0lr/MU/v6K2kteCx+orhnvSmoes4BNycG8DwL7HCeQdCSwISK+BKwA3kjh7JHzJR2ctJkt6Yj6lW3NKvlb3SZpiaQM8Hsli/8DuKQ4kfSEbYI4GJrHV4A/knQ/hd1Ieyq0uQB4TNJqYClwTUSsAz4B3CppDXAbcGidajb7GHAzhS8o3SXzLwFOS06SWAf8eRrFTVW+8tnMzMq4x2BmZmUcDGZmVsbBYGZmZRwMZmZWxsFgZmZlHAxmZlbGwWCpkHRvDbd9p6SGuIG8pAMl/VXadZSS9KykuWnXYY3LwWCpiIh9Rnedog4EGioYzMbjYLBUSNqdPJ+efMP/saQnJX1PJSOjVVjvZEn3Sno0GW55pqQOSdcmV8H+kMJAf0jKSvpWMpTzWkmXjrHdOyV9Ntnmf0r67WR+u6RvJuv/StLbx9hGpWHLPwMclcz7vAo+X1LTBSW/h7sk/VTSOklfS4aBGO29zkyGT39U0s+TebMl/Sx57/slvTGZP0fSrUn9/0LJ+EOSLktqeUzSR5N5iyQ9Ien/SXo8WbdjtFpsCooIP/yo+4PCzVkATgd2AAsofFG5D3jrKOu0ARuAk5PpAygMBHkZ8I1k3huBHIVhw98E3Fay/oFj1HMn8MXk9buB/0he/w3wzeT1UuA5oH2UbXwZ+MOSWjuARcBjJW3eR2FYkSxwSLK9Q5PfQz9wZLLsNuD8Ud6nC3geWJxMzy55/08lr98BrE5efwn4ZPL6bCCAucnvZy0wHZgBPA6cmNScA05I1rkO+EDafzN+1O/hHoM1ggcjojsKg6atpvDBVMnrKQwE+BBAROyMiBzwO8B3k3lrgDVJ+w3AkZK+LOlMYOc4dfwkeX64pIa3At9Jtv0ksJHKQ5ZDIdT+p6SPAUdE5SGg3wr8ICKGI2Iz8Avg5GTZgxGxISKGgR8kbSt5C3BXRDyT1PVyhVpvB+ZImkX572clsK2k/U8jYk9E7E5+/t9Olj0TEasr/D6sCTgYrBEMlLweZvTh4EXh224l+8yPiG0Uhhm/k8Kga+PdzKVYR2kNo+7WqvB+3wfOBfooDFv+jgrNxtreyJ9htJ91tN9DpW3HiOdqa6n238SmIAeDTSZPAodJOhkgOb7QAtxFclc5ScdT2J1EcuZNJiL+DfhfwEmv4j1Lt/06YCHwVKWGowxbvguYOWJ7FyTHP7oofJt/MFl2iqTFybGFC4BfjlLTfcDbJC1O3nd2hVqFa2nrAAABOElEQVRPB7ZExM4R888CDipp/7uSOiVNpzCs9d3V/FJsavO3AJs0ImIwOVj75eRgaB+FO3l9FfhmMiz4al75oJ2fzC9+AXo1Nxv6CvA1SWsp7Hf/44gYGKXtBcAHJA0BvwGuiIiXJd0j6THg34G/A04FHqXwLf7vIuI3kpZS+MD/DPAGCh/aP630JhHRI+li4CfJz/YS8C7g8pLfQy/wR8kq/xv4gaRHKOy6ei7ZziOSvsUrv6+vR8SvJC3az9+RTTEedtusASTf8P82Is5JuxYz70oyM7My7jFYQ5L0U2DxiNkfi4hbXuN2rwJOGzH7yoj45n5s478Bnx0x+5mI+L1K7V8LFe7BPW3E7A9GxNqJfi+zIgeDmZmV8a4kMzMr42AwM7MyDgYzMyvjYDAzszIOBjMzK/P/AdrpdGw9z724AAAAAElFTkSuQmCC\n",
        "text/plain": "<Figure size 432x288 with 1 Axes>"
       },
       "metadata": {
        "needs_background": "light"
       },
       "output_type": "display_data"
      }
     ],
     "persistentId": "72511a16-18da-4358-93cf-c36a17c23f8e",
     "text": "\n\nsns.violinplot(x='in_cds_no_stop_codon', y='jaccard_in_peptide_db', data=coding_scores_not_low_complexity_unspliced, \n               bw=0.1, cut=True)"
    },
    "executionTime": "2019-10-12T03:39:17.314Z"
   },
   {
    "cell": {
     "executionCount": 59,
     "executionEventId": "504c5e07-4144-450a-97a7-54f511b9df71",
     "hasError": false,
     "id": "804034e2-612a-4733-8cfa-8769707fbb89",
     "outputs": [
      {
       "data": {
        "text/plain": "(29, 6)"
       },
       "execution_count": 59,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "23edfe41-6d93-44d6-9324-ba4ac19d25e0",
     "text": "coding_scores_unspliced.query(\"jaccard_in_peptide_db < 0.5 & in_cds_no_stop_codon == True\").shape"
    },
    "executionTime": "2019-10-12T03:39:52.474Z"
   },
   {
    "cell": {
     "executionCount": 60,
     "executionEventId": "a867423b-9b16-4ab2-90d4-a73692ceb0b8",
     "hasError": false,
     "id": "4d9dc7ba-ece3-49ca-892c-d947efb6ac0a",
     "outputs": [
      {
       "data": {
        "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>read_id</th>\n      <th>jaccard_in_peptide_db</th>\n      <th>n_kmers</th>\n      <th>classification</th>\n      <th>short_read_id</th>\n      <th>in_cds_no_stop_codon</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>52</th>\n      <td>SRR306838.17561853 Ibis_Run100924_C3PO:6:88:9412:18824/1</td>\n      <td>0.000000</td>\n      <td>15</td>\n      <td>non-coding</td>\n      <td>SRR306838.17561853</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>61</th>\n      <td>SRR306838.3607263 Ibis_Run100924_C3PO:6:17:17893:5364/1</td>\n      <td>0.294118</td>\n      <td>17</td>\n      <td>non-coding</td>\n      <td>SRR306838.3607263</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>112</th>\n      <td>SRR306838.23727244 Ibis_Run100924_C3PO:6:117:4784:12719/1</td>\n      <td>0.066667</td>\n      <td>15</td>\n      <td>non-coding</td>\n      <td>SRR306838.23727244</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>144</th>\n      <td>SRR306838.3497462 Ibis_Run100924_C3PO:6:16:17526:15365/1</td>\n      <td>0.461538</td>\n      <td>13</td>\n      <td>non-coding</td>\n      <td>SRR306838.3497462</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>166</th>\n      <td>SRR306838.12840151 Ibis_Run100924_C3PO:6:64:3090:6141/1</td>\n      <td>0.000000</td>\n      <td>17</td>\n      <td>non-coding</td>\n      <td>SRR306838.12840151</td>\n      <td>True</td>\n    </tr>\n  </tbody>\n</table>\n</div>",
        "text/plain": "                                                       read_id  \\\n52    SRR306838.17561853 Ibis_Run100924_C3PO:6:88:9412:18824/1   \n61     SRR306838.3607263 Ibis_Run100924_C3PO:6:17:17893:5364/1   \n112  SRR306838.23727244 Ibis_Run100924_C3PO:6:117:4784:12719/1   \n144   SRR306838.3497462 Ibis_Run100924_C3PO:6:16:17526:15365/1   \n166    SRR306838.12840151 Ibis_Run100924_C3PO:6:64:3090:6141/1   \n\n     jaccard_in_peptide_db  n_kmers classification       short_read_id  \\\n52                0.000000       15     non-coding  SRR306838.17561853   \n61                0.294118       17     non-coding   SRR306838.3607263   \n112               0.066667       15     non-coding  SRR306838.23727244   \n144               0.461538       13     non-coding   SRR306838.3497462   \n166               0.000000       17     non-coding  SRR306838.12840151   \n\n     in_cds_no_stop_codon  \n52                   True  \n61                   True  \n112                  True  \n144                  True  \n166                  True  "
       },
       "execution_count": 60,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "1254868e-73a0-4c0a-82c6-bb1b85e78cb0",
     "text": "coding_scores_unspliced.query(\"jaccard_in_peptide_db < 0.5 & in_cds_no_stop_codon == True\").head()"
    },
    "executionTime": "2019-10-12T03:39:53.340Z"
   },
   {
    "cell": {
     "executionCount": 61,
     "executionEventId": "73b80050-01d6-4cbb-99e1-93a82bb7a45b",
     "hasError": false,
     "id": "4d9dc7ba-ece3-49ca-892c-d947efb6ac0a",
     "outputs": [
      {
       "data": {
        "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>read_id</th>\n      <th>jaccard_in_peptide_db</th>\n      <th>n_kmers</th>\n      <th>classification</th>\n      <th>short_read_id</th>\n      <th>in_cds_no_stop_codon</th>\n    </tr>\n  </thead>\n  <tbody>\n  </tbody>\n</table>\n</div>",
        "text/plain": "Empty DataFrame\nColumns: [read_id, jaccard_in_peptide_db, n_kmers, classification, short_read_id, in_cds_no_stop_codon]\nIndex: []"
       },
       "execution_count": 61,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "1254868e-73a0-4c0a-82c6-bb1b85e78cb0",
     "text": "coding_scores_not_low_complexity_unspliced.query(\"jaccard_in_peptide_db < 0.5 & in_cds_no_stop_codon == True\").head()"
    },
    "executionTime": "2019-10-12T03:40:15.004Z"
   },
   {
    "cell": {
     "executionCount": 62,
     "executionEventId": "68e14e18-845d-48f1-9811-b7d2137c8a12",
     "hasError": false,
     "id": "804034e2-612a-4733-8cfa-8769707fbb89",
     "outputs": [
      {
       "data": {
        "text/plain": "(0, 6)"
       },
       "execution_count": 62,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "23edfe41-6d93-44d6-9324-ba4ac19d25e0",
     "text": "coding_scores_not_low_complexity_unspliced.query(\"jaccard_in_peptide_db < 0.5 & in_cds_no_stop_codon == True\").shape"
    },
    "executionTime": "2019-10-12T03:40:16.248Z"
   },
   {
    "cell": {
     "executionCount": 63,
     "executionEventId": "ea3fee6a-b080-455d-ade2-beb64755e25b",
     "hasError": false,
     "id": "4d9dc7ba-ece3-49ca-892c-d947efb6ac0a",
     "outputs": [
      {
       "data": {
        "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>read_id</th>\n      <th>jaccard_in_peptide_db</th>\n      <th>n_kmers</th>\n      <th>classification</th>\n      <th>short_read_id</th>\n      <th>in_cds_no_stop_codon</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>289</th>\n      <td>SRR306838.8536423 Ibis_Run100924_C3PO:6:41:4477:5833/1</td>\n      <td>0.588235</td>\n      <td>17</td>\n      <td>non-coding</td>\n      <td>SRR306838.8536423</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>740</th>\n      <td>SRR306838.4158712 Ibis_Run100924_C3PO:6:19:4365:17337/1</td>\n      <td>0.533333</td>\n      <td>15</td>\n      <td>non-coding</td>\n      <td>SRR306838.4158712</td>\n      <td>True</td>\n    </tr>\n  </tbody>\n</table>\n</div>",
        "text/plain": "                                                     read_id  \\\n289   SRR306838.8536423 Ibis_Run100924_C3PO:6:41:4477:5833/1   \n740  SRR306838.4158712 Ibis_Run100924_C3PO:6:19:4365:17337/1   \n\n     jaccard_in_peptide_db  n_kmers classification      short_read_id  \\\n289               0.588235       17     non-coding  SRR306838.8536423   \n740               0.533333       15     non-coding  SRR306838.4158712   \n\n     in_cds_no_stop_codon  \n289                  True  \n740                  True  "
       },
       "execution_count": 63,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "1254868e-73a0-4c0a-82c6-bb1b85e78cb0",
     "text": "coding_scores_not_low_complexity_unspliced.query(\"jaccard_in_peptide_db < 0.9 & in_cds_no_stop_codon == True\").head()"
    },
    "executionTime": "2019-10-12T03:40:20.150Z"
   },
   {
    "cell": {
     "executionCount": 64,
     "executionEventId": "a383af3b-9046-4926-896c-fe97c4abd1b6",
     "hasError": false,
     "id": "ff464c9a-3f55-4acc-b6bc-f887aa171402",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "SRR306838.8536423\t0\tchr1\t39448668\t255\t70M\t*\t0\t0\tAAAGACTTTGCTTCCCGAAGATACTCAGAAACTTGACAATTTCCTAGGAGAAGTCAGAGACAAATGGGAT\t-0454575:38569357656746757387639475375567787758567758596:6957665665585\tNH:i:1\tHI:i:1\tAS:i:66\tnM:i:1\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\n"
      }
     ],
     "persistentId": "783e2fb0-f110-45d7-8d85-507d5ff6a343",
     "text": "unspliced_bam = 'SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.bam'\n\n# This reads aligns to the CBLN2 \nnoncoding_read_id = 'SRR306838.8536423'\n! samtools view $folder/$unspliced_bam | grep $noncoding_read_id"
    },
    "executionTime": "2019-10-12T03:40:44.151Z"
   },
   {
    "cell": {
     "executionCount": 65,
     "executionEventId": "988e5f5c-c090-4f9d-8005-4a0b9ba06fff",
     "hasError": false,
     "id": "c4e90627-a46a-4fa2-b552-6402975ee8c7",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "0.8973471741637832\n              precision    recall  f1-score   support\n\n       False       1.00      0.88      0.94       767\n        True       0.53      1.00      0.69       100\n\n   micro avg       0.90      0.90      0.90       867\n   macro avg       0.76      0.94      0.82       867\nweighted avg       0.95      0.90      0.91       867\n\n"
      },
      {
       "data": {
        "image/png": "iVBORw0KGgoAAAANSUhEUgAAAYUAAAEKCAYAAAD9xUlFAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDMuMC4zLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvnQurowAAIABJREFUeJzt3Xl8FFW2wPHfyQJh35GdoIKKKIgBFRUXFlE2Rx1FEUQRFEUdt1GfC+LzObgLImIkjMug4jCo0YFBRsEFAUEREdARAkJQZEsCJGQ/749qYk8MSQe6uno538+nP3RVV7pOJSSn7711zxVVxRhjjAGI8zoAY4wx4cOSgjHGmDKWFIwxxpSxpGCMMaaMJQVjjDFlLCkYY4wpY0nBGGNMGUsKxhhjylhSMMYYUybB6wCqq2nTppqcnOx1GMYYE1G++uqrXararKrjIi4pJCcns3LlSq/DMMaYiCIiPwVynHUfGWOMKWNJwRhjTBlLCsYYY8pYUjDGGFPGkoIxxpgyriUFEZkpIjtE5LtDvC4iMkVENojItyLS3a1YjDHGBMbNlsIrwIBKXr8Q6Oh7jAVedDEWY4wxAXBtnoKqfioiyZUcMhR4TZ31QJeJSEMRaamqv7gVU9jY+xP8+yYo3Od1JMaYCPDlhvokJZZy8tk9odfDrp7Ly8lrrYGtftuZvn2/SwoiMhanNUG7du1CEpxrVGHhDbB5gdeRGGPCnCrc889+PP3JqZzc8le+PO57El0+p5dJQSrYpxUdqKqpQCpASkpKhcdEhB/+Dp/cBfu2QM0GMOjvkFDT66iMMWFKADZugE+30n/oaZT06BHVSSETaOu33Qb42aNYQmPVFCchAJw9CZL7eRuPMSbsZGfnk5GRRffuLQGY+MwZDBuzq2zbbV4mhXRgvIi8BZwG5ETFeMKOb+DjW6Aot4LXVjn/9p0OXW8IbVzGmLD33nvfM27cP4mLE9auvYkGDZKoVSsxZAkBXEwKIvImcC7QVEQygQngtHxUdTowD7gI2ADkAde6FUtIfTcTtn1+6NfjEuDogaGLxxgT9nbsyOXWW+cze/ZaAE4/vQ3Z2fk0aJAU8ljcvPvoyipeV+Bmt87vmeyNzr/nPgNtzvn963VbQZ0WoY3JGBOWVJVZs9Zw223/Ys+eA9Sunchjj53P+PE9iY/3Zm5xxJXODns5Gc6/bc+H5l29jcUYE9bGjfsnL730FQB9+x5NauogOnRo5GlMVuYimLQUcjY5zxse7W0sxpiwd/HFx9OwYRJpaUP48MOrPU8IYC2F4Nr/M5QUQO3mUKOe19EYY8LMjz/u5qOPNnHjjSkADBhwLJs33+bJ2MGhWFIIpoPjCQ2slWCM+U1xcSnPPLOUCRMWU1BQTLduLTj99DYAYZUQwJJCcB1MCg2P8TYOY0zYWL16O6NHp/PVV84d9yNHdqVjx8YeR3VolhSC6eAgs7UUjIl5BQXFPProp0yatITi4lLatWvASy8NYsCAY70OrVKWFILJWgrGGJ/77vuIZ59dBsDNN/fgL3/pQ7164V/WxpJCMJW1FCwpGBPr/vznM1m6NJMnnujL2We39zqcgNktqcFU1lKw7iNjYs3ChRu59NK3KS4uBaBFi7p88cV1EZUQwJJC8BTkQP5uSKgFdUJXp8QY462srAOMHv0e/fv/jblz1/PXv64qe02komLQ4c26j4Il22+QOQL/Ixhjqu+dd9Zz003z2L59PzVrxjNhwjmMGtXN67COiCWFYMmxOQrGxIrt2/dzyy3zmTNnHQC9erUlLW0Ixx/f1OPIjpwlhWA52FKwO4+MiXrvvfc9c+aso06dRCZN6stNN/UgLi46eggsKQSLtRSMiWr5+cUkJTl/MseMOZWMjCzGjetBcnJDjyMLLhtoDhabo2BMVCotVaZO/ZIOHSbz00/ZAMTFCY8/3i/qEgJYUgieHOs+Miba/PDDLnr3/iu33DKf7dv38+ab33kdkuus+ygYSopg7xZAoH6y19EYY45QUVEJTz31BRMnfkJBQQlHHVWHadMGcsklJ3gdmussKQTD3p9AS6BeW0gI/2nsxphD++67HYwc+Q6rVm0H4Npru/H00/1p1KiWx5GFhiWFYLBCeMZEjdJSZc2aHbRv34DU1MH07x9bXcKWFILBBpmNiWhr1+6gc+dmiAgnn3wU7703jN6921O3bg2vQws5G2gOBhtkNiYi7dtXwPjx8+jS5UX+8Y/1ZfsvuqhjTCYEsJZCcNiKa8ZEnAULNjB27Ads2ZJDQkIcmzdnex1SWLCkEAw51n1kTKTYs+cAt9++gNdeWw1A9+4tSUsbQrduLTyOLDxYUjhSqv9dDM8YE7a++WY7Awb8jV9/zaVmzXgmTjyXO+/sRUKC9aQfZEnhSB3YCUX7oWYDSArfdVeNMdCpUxPq1q1Bp05NmDFjCJ06NfE6pLBjSeFIZfuttmYls40JK6rKG2+sYfDg46hfvya1ayeyePEoWrWqFzUF7ILN2kxHKsdWWzMmHG3enM0FF/yNq69+h3vv/XfZ/jZt6ltCqIS1FI5U2Z1HNshsTDgoKSll2rQV3HffR+TmFtG4cS169WrrdVgRw5LCkbI5CsaEjfXrdzJ6dDpLl2YCcPnlJ/L88xfSvHkdjyOLHJYUjpTNUTAmLGzalEW3bi9RWFhCy5Z1mTZtIBdffLzXYUUcV5OCiAwAJgPxwAxVnVTu9XbAq0BD3zH3quo8N2MKOmspGBMWOnRoxB//2JmkpASeeqo/DRsmeR1SRHItKYhIPPAC0A/IBFaISLqqrvM77AHgbVV9UUQ6A/OAZLdiCrqiA7D/Z4hLgHptvI7GmJhy4EARjzzyCX/4wwn07NkagFdfvZj4eLt/5ki4+d3rCWxQ1QxVLQTeAoaWO0aB+r7nDYCfXYwn+A62EuonO4nBGBMSn332E926vcSkSUsYO/Z9SksVwBJCELj5l6w1sNVvOxM4rdwxDwMfisgtQB2gr4vxBJ91HRkTUnv3FnDfff9m2rSVAHTu3Izp0wfZLaZB5GZareinpOW2rwReUdU2wEXA6yLyu5hEZKyIrBSRlTt37nQh1MNkg8zGhMy8eT/Spcs0pk1bSUJCHA891Juvvx7L6adb120wudlSyAT8bw5uw++7h0YDAwBUdamIJAFNgR3+B6lqKpAKkJKSUj6xeMfWUTAmJHJy8hk+fC7Z2fmkpLQiLW0IJ598lNdhRSU3k8IKoKOIdAC2AcOAq8odswXoA7wiIicASUAYNQWqYCuuGeMaVUUV4uKEBg2SmDJlAL/+msuf/nS6FbBzkWtJQVWLRWQ8sADndtOZqrpWRB4BVqpqOnAn8LKI3I7TtTRKVcOnJVAVaykY44qff97HTTf9k7PPbsedd/YCYMSIrh5HFRtcvWXGN+dgXrl9D/k9Xwec6WYMrtFS2LvJeW4tBWOCQlWZOXMVd975ITk5BSxblslNN/WgVq1Er0OLGXYf5eHatw1KCqF2c6hR1+tojIl4GRlZjBnzPh9/7HzYGjiwI9OnD7KEEGKWFA5XjhXCMyYYSkpKmTJlOfff/zEHDhTTtGltpkwZwLBhXRArRx9ylhQOV7bNUTAmWObMWc+BA8VceWUXJk8eQLNmVsDOK5YUDleOzVEw5nAVFpawb18BTZrUJj4+jrS0Ifz4424GDz7O69Bint3XdbispWDMYVmxYhspKamMGPEOB282PP74ppYQwoS1FA6XtRSMqZa8vCImTFjEM88so7RUycsrYseOXI46ym7UCCeWFA6XzVEwJmCLF29mzJj32bBhD3Fxwl13ncHEiedRu7bdWRRuLCkcjvxsyN8DCbWgTguvozEmbKkqt946n6lTVwBw0knNSUsbQo8erT2OzByKJYXD4V/ewm6ZM+aQRIT69WuSmBjHAw/05t57z6JGjXivwzKVsKRwOKxktjGHtGtXHhs37uG005zqpQ8+eA7Dh59M587NPI7MBCKgu49EpIaIHOt2MBHDSmYb8zuqyltvfccJJ7zAxRfPJivrAABJSQmWECJIlUlBRAYCa4CFvu1uIvKO24GFNRtkNua/ZGbuZejQt7jyyn+wa1cenTs3Iy+vyOuwzGEIpPvoEZwV0xYBqOo3Md9qsO4jYwAoLVVmzPiau+9eyN69BdSvX5Onn+7P6NGnWImKCBVIUihS1exyP+DIKW/tBus+MgaA0aPTeeWVbwAYMuQ4pk27iNat61fxVSacBTKmsF5ELgfiRKSDiDwHLHM5rvBVUgj7tgAC9ZO9jsYYT1199Uk0b16Ht966lHffvcISQhQIJCmMB04FSoG5QD5wm5tBhbW9W5y1FOq1hYSaXkdjTEh9990OJk/+7TNhnz5Hk5FxK1dcYRVNo0Ug3UcXqOo9wD0Hd4jIJTgJIvYcLG/R0LqOTOwoKCjmL3/5nMce+4yiolJSUlpx5pntAKhTp4bH0ZlgCqSl8EAF++4PdiAR42AhPFtHwcSI5cszOfXUVCZO/ISiolLGjUvhpJOO8jos45JDthRE5AJgANBaRJ7xe6k+TldSbMq2loKJDbm5hTz44CKee24ZqtCxY2NmzBhC797tvQ7NuKiy7qMdwHc4Ywhr/fbvA+51M6iwZiuumRhx//0fM3nycuLihLvvPoOHHz7XlsaMAYdMCqq6ClglIrNUNT+EMYU3m6NgYsT995/NmjU7ePzxvqSktPI6HBMigYwptBaRt0TkWxH5z8GH65GFI1Wbo2CiVnr6D1x00SyKikoAaNasDh99NNISQowJJCm8AvwVEOBC4G3gLRdjCl8HdkJRLtRsCLUaex2NMUGxY0cuw4bNYejQt5g/fwOvvrra65CMhwJJCrVVdQGAqm5U1QeA89wNK0xZK8FEEVXlb3/7lhNOeIHZs9dSu3YikycP4Npru3kdmvFQIPMUCsSZlbJRRG4EtgHN3Q0rTFkhPBMltmzJ4cYbP2D+/A0A9O17NKmpg+jQoZHHkRmvBZIUbgfqArcC/wc0AK5zM6iwZYPMJkp8+OFG5s/fQMOGSTzzTH9GjepmM5INEEBSUNXlvqf7gBEAItLGzaDClnUfmQiWm1tYNvt49OhT2LZtL2PHnkrLlvU8jsyEk0rHFESkh4hcLCJNfdsnishrxGpBPGspmAhUXFzKE08soX3758jIyAKcZTInTDjXEoL5nUMmBRH5CzALGA78S0Tux1lTYTXQKTThhRkbUzARZvXq7Zx22gzuueff7N59gHff/d7rkEyYq6z7aCjQVVUPiEhj4Gff9g+hCS3MFOVB7i8Qlwh1Y7P3zESOgoJiHn30UyZNWkJxcSnt2jUgNXUQF1wQ2+tjmapVlhTyVfUAgKruEZHvYzYhAORscv5tkAxx8Z6GYkxlVq36heHD57J+/S5EYPz4Hjz2WB/q1bNS76ZqlSWFo0XkYHlsAZL9tlHVS6p6cxEZAEwG4oEZqjqpgmMuBx7GWc1ttapeFXj4IWSDzCZC1KyZwMaNWRx3XBNmzBjCWWe18zokE0EqSwqXltueWp03FpF44AWgH5AJrBCRdFVd53dMR+A+4ExVzRKR8J3/kGMls034+vrrXzjllBaICJ07N2P+/OH06tWWpKRA7jo35jeVFcT76AjfuyewQVUzAETkLZxxinV+x4wBXlDVLN85dxzhOd1jg8wmDGVlHeCuuz5k5sxvePPNSxk2rAsA55/fwePITKRy82NEa2Cr33YmcFq5YzoBiMgSnC6mh1X1X+XfSETGAmMB2rXzqCmcY91HJry88856brppHtu376dmzXh2787zOiQTBdxMChVNj9QKzt8ROBdoA3wmIl1UNfu/vkg1FUgFSElJKf8eoZFtcxRMeNi+fT+33DKfOXOcRveZZ7ZlxowhHH98U48jM9Eg4KQgIjVVtaAa750JtPXbboNzW2v5Y5apahGwSUR+wEkSK6pxHveVlsDeg3cfWbPceOerr36mX7/XycrKp06dRCZN6stNN/UgLs5KVJjgqLJKqoj0FJE1wI++7a4i8nwA770C6CgiHUSkBjAMSC93zLv4Kq76Zk13AjKqEX9o7N8GJYVQ+yioUdfraEwM69y5Gc2a1eGCC45h7dqbGD++pyUEE1SBlM6eAgwCdgOo6moCKJ2tqsXAeGABsB54W1XXisgjIjLEd9gCYLeIrMOZLX23qu6u/mW4zMpbGI+UliqpqV+Rne0sflirViKffjqK+fOH0759Q4+jM9EokO6jOFX9qVwFxZJA3lxV5wHzyu17yO+5Anf4HuHL5igYD/zwwy6uv/59Pv98CytWbOPll53PUkcdZa1V455AksJWEekJqG/uwS1AbC3HaS0FE0JFRSU8/fRSHn54MQUFJbRoUZcLL+zodVgmRgSSFMbhdCG1A34F/u3bFztsjoIJkVWrfmH06HRWrdoOwLXXduPpp/vTqFEtjyMzsSKQpFCsqsNcjyScWfeRCYGNG/fQs+cMiotLSU5uSGrqIPr1sw8iJrQCSQorfLeKzgbmquo+l2MKP9Z9ZELgmGMaM2LEydSrV4P/+78+1K1bw+uQTAyq8u4jVT0GeBQ4FVgjIu+KSOy0HPKzIX8PJNR2bkk1Jkj27y/k1lvns3TpbxP/09KGMHnyhZYQjGcCuSUVVf1CVW8FugN7cRbfiQ1lrYSjwdawNUGyYMEGTjxxGs8//yU33vhPnBvxsHWSjeeq7D4Skbo4heyGAScA7wG9XI4rfJSNJ1jXkTlye/Yc4PbbF/Daa6sBOPXUlqSlDbFkYMJGIGMK3wHvA0+o6mcuxxN+yu48skFmc2TmzFnHzTfPY8eOXJKSEpg48VzuuOMMEhICarAbExKBJIWjVbXU9UjCla2jYIIgOzufsWPfJysrn9692/Pyy4Pp1KmJ12EZ8zuHTAoi8rSq3gn8Q0R+V5k0kJXXokKOtRTM4VFVSkuV+Pg4GjZMYtq0gWRlHeCGG1KsXpEJW5W1FGb7/q3WimtRJ9taCqb6Nm/OZuzY9zn//A7ce+9ZAGUL4BgTzg7ZmamqX/qenqCqH/k/cAaco19JIezbAgg0SPY6GhMBSkpKmTJlOV26TGPhwgymTv2S/Pxir8MyJmCBjHBdV8G+0cEOJCzt/Qm0FOq1hXi7b9xUbv36nfTu/Qq33fYvcnOLGDasC19/fYOtk2wiSmVjClfg3IbaQUTm+r1UD8iu+KuijM1kNgEoLi7l8cc/55FHPqWwsIRWrerx4osDGTLkOK9DM6baKvsI8yXOGgptgBf89u8DVrkZVNiwQngmAHFxwocfZlBYWMKYMd154ol+NGyY5HVYxhyWQyYFVd0EbMKpihqbrBCeOYQDB4rYt6+Q5s3rEBcnzJgxmK1b93L++bZcq4lshxxTEJFPfP9micgev0eWiOwJXYgesu4jU4FPP/2Jrl2nc/XVc8vKU3Ts2MQSgokKlXUfHVxys2koAglL1lIwfvbuLeC++/7NtGkrAUhMjGfXrjyaNavjcWTGBE9lt6QenMXcFohX1RLgDOAGIPp/C1StpWDKzJ//I126TGPatJUkJMQxYcI5fP31WEsIJuoEcq/cu0APETkGeA34J/AGMMjNwDyXtwOKciGpkfMwMUlVGTPmfdLSnHsrUlJaMXPmEE46ycqom+gUyDyFUlUtAi4BnlPVW4DW7oYVBqzryOCUsm7Tpj5JSQk89VQ/li4dbQnBRLWAluMUkT8CI4CLffsS3QspTFghvJj188/72LhxD2ef3R6A//mfsxkx4mSOOaaxx5EZ475AZzSfh1M6O0NEOgBvuhtWGLCS2TFHVUlL+5rOnV/g0kvfZvfuPABq1Ii3hGBiRpUtBVX9TkRuBY4VkeOBDar6f+6H5jFrKcSUjIwsxox5n48/3gTAoEGdKCqK3YrxJnYFsvLa2cDrwDZAgBYiMkJVl7gdnKdsNnNMOFjA7oEHFpGXV0TTprWZMmUAw4Z1sdXQTEwKZEzhWeAiVV0HICIn4CSJFDcD85ytoxATRo58lzfeWAPAVVedxHPPXWC3mZqYFsiYQo2DCQFAVdcD0V0ytCgPcrdDXCLUbeN1NMZFY8Z0p02b+qSnD2PWrEssIZiYF0hL4WsReQmndQAwnGgviFc2ntAB4uK9jcUE1YoV2/j4403cc4+z8M255yazYcMt1Kxp5a2NgcCSwo3ArcCfccYUPgWedzMoz5WttmZdR9EiL6+ICRMW8cwzyygtVXr1alt2y6klBGN+U+lvg4icBBwDvKOqT4QmpDCQY4PM0WTx4s1cf306GzdmERcn3HXXGZx6aiuvwzImLFVWJfV/cEpcDAcWikhFK7BVSkQGiMgPIrJBRO6t5LjLRERFJDwGr202c1TIycnnhhve57zzXmXjxixOOqk5y5aN5skn+1O7dvTPvzTmcFTWUhgOnKyquSLSDJgHzAz0jUUkHmdxnn5AJrBCRNL9B619x9XD6Z5aXt3gXWOF8KLCgw8uIjX1axIT43jwwd7cc89Z1KhhY0TGVKaypFCgqrkAqrpTRAK5U8lfT5yJbhkAIvIWMBRYV+64/wWeAO6q5vu7x+YoRCxVLZtf8NBD57BpUzaTJvXhxBObexyZMZGhsj/0R4vIXN/jHeAYv+25lXzdQa2BrX7bmZQrpCcipwBtVfWDakfultISyHFmtdLAFk2JFKrKG2+s4fzzX6OwsASApk1r8/77V1pCMKYaKmspXFpue2o137ui6aBa9qLT8ngWGFXlG4mMBcYCtGvXrpphVNP+bVBaBHVaQKLdsx4JMjP3Mm7cP/ngg/8AMGvWt1x77SkeR2VMZKpsjeaPjvC9M3EW6DmoDfCz33Y9oAuw2NfcbwGki8gQVV1ZLpZUIBUgJSVFcZMNMkeM0lLl5Ze/4u67F7JvXyENGtTk6af7M2pUN69DMyZiuXmD9gqgo6+q6jZgGHDVwRdVNQe/pT5FZDFwV/mEEHI2yBwRNmzYw5gx77N48WYAhg49jmnTBtKqVT1vAzMmwrmWFFS1WETGAwuAeGCmqq4VkUeAlaqa7ta5j0hZS8GSQjj77LOfWLx4M82b12Hq1Au57LLOVsDOmCAIOCmISE1VLajOm6vqPJxbWf33PXSIY8+tznu7xtZRCFvZ2fk0bJgEwKhR3di5M4/Ro0+hSZPaHkdmTPSo8jZTEekpImuAH33bXUUkestc2DoKYaegoJgJExbRvv1z/PjjbsBZJvPPfz7TEoIxQRZIS2EKMAhndjOqulpEznM1Ki9ZiYuwsmxZJqNHp7Nu3U4AFizYSMeOTTyOypjoFUhSiFPVn8r115a4FI+38rOcR2IdqG33tnspN7eQBx9cxHPPLUMVOnZsTFrakLIidsYYdwSSFLaKSE9AfaUrbgH+425YHsnxq45qg5aeWb48k6uumktGRhbx8cJdd/ViwoRzqFXL6hUZ47ZAksI4nC6kdsCvwL99+6KPzVEICw0bJrFt2166dj2KtLQhVtHUmBCqMimo6g6cOQbRL9vmKHjl88+3cOaZbRERjjuuKR9/fA09erQiMdEK2BkTSlUmBRF5Gb/yFAep6lhXIvKSDTKH3I4dudx663xmz17Lq69ezMiRXQHo1attFV9pjHFDIN1H//Z7ngT8gf8udBc9cmzFtVBRVWbNWsNtt/2LPXsOULt2YlkhO2OMdwLpPprtvy0irwMLXYvIS1YyOyS2bMnhxhs/YP78DQD063c0qamDSU5u6HFkxpjDKXPRAYi++wJLCmHfVpA4qB99lxculi/PpG/f19m/v5CGDZN49tkLuOaarlaiwpgwEciYQha/jSnEAXuAQy6tGbH2/gRa6iSE+BpeRxO1unVrQdu29Tn++Ka88MJFtGxpBeyMCSeVJgVxPr51xalyClCqqu6WrvaKdR25ori4lKlTv2TkyK40blyLmjUTWLLkOho1quV1aMaYClRa+8iXAN5R1RLfIzoTAtgcBResXr2d006bwe23L+COOxaU7beEYEz4CmTd5S9FpLvrkXjNCuEFTX5+MQ888DEpKS/z9de/0K5dA668sovXYRljAnDI7iMRSVDVYuAsYIyIbARycZbZVFWNrkRh3UdB8cUXWxk9Op3vv9+FCIwf34PHHutDvXo1vQ7NGBOAysYUvgS6AxeHKBZvla24Zt1Hh2vDhj2cffZfKS1VjjuuCWlpQzjzTJfX1DbGBFVlSUEAVHVjiGLxjqp1HwXBscc2ZuzY7jRuXIsHHzyHpCQ3V3s1xrihst/aZiJyx6FeVNVnXIjHG3m/QlEuJDWCJJtAFaisrAPceeeHXHttt7KS1tOmDbQ5B8ZEsMqSQjxQF1+LIaplWyuhuubOXc/NN89j+/b9fPXVL3zzzQ2IiCUEYyJcZUnhF1V9JGSReMkK4QVs+/b9jB8/j3/8Yz0AZ53VjhkzBlsyMCZKVDmmEBOyrRBeVVSV115bze23LyArK5+6dWvw+ON9ufHGFOLiYue/ijHRrrKk0CdkUXjNWgpVys7O5847PyQrK58BA45l+vSBtG9v4y/GRJtDJgVV3RPKQDxls5krVFqqlJYqCQlxNGpUi5deGkReXhFXX32ydRcZE6UCmdEc/XJsxbXyvv9+F717/5VJkz4v23fppZ0ZMcIqmhoTzSwpFOVC7nanMmrd1l5H47miohIee+wzunadzpIlW0lLW0V+frHXYRljQsRmF+Vscv6tnwxxsb0e8KpVv3Dddel88812AEaPPoUnn+xnk9CMiSH22241jygqKmHChMU88cQSSkqU5OSGvPzyYPr2tTEWY2KNJYWyQebYTQoJCXEsX76N0lLltttO49FHz6duXVtoyJhYZEkhRgvh7dtXwL59hbRqVQ8RYcaMwWzfvp8zzmjrdWjGGA/ZQHMMthQWLNhAly4vMnz4XA6um9ShQyNLCMYYSwqx1FLYvTuPa655lwEDZrFlSw779hWwe/cBr8MyxoQRV5OCiAwQkR9EZIOI3FvB63eIyDoR+VZEPhKR9m7G8zulJb/dfRTFE9dUlTlz1tG58zRee201SUkJPPFEX5Ytu56mTWt7HZ4xJoy4NqYgIvHAC0A/IBNYISLpqrrO77BVQIqq5onIOOAJ4Aq3Yvqd/ZlQWgR1WkJidP5xVFWGD5/Lm29+B0DRZJGNAAAPYElEQVTv3u15+eXBdOrUxOPIjDHhyM2WQk9gg6pmqGoh8BYw1P8AVV2kqnm+zWVAGxfj+b0YKIQnInTu3Ix69Wrw4osDWbToGksIxphDcvPuo9bAVr/tTOC0So4fDcyv6AURGQuMBWjXLojLO0bpHIVNm7LIyMiiTx8n2d1zz5mMGtWNNm3qexyZMSbcudlSqKhAjlZ4oMjVQArwZEWvq2qqqqaoakqzZs2CF2FOdBXCKykpZfLkZXTp8iJXXDGHHTtyAUhMjLeEYIwJiJsthUzA/x7HNsDP5Q8Skb7A/cA5qlrgYjy/lx09hfDWrdvJ9dens3RpJgBDhhxn6xwYY6rNzaSwAugoIh2AbcAw4Cr/A0TkFOAlYICq7nAxlopFwToKRUUlPP74Ev73fz+lsLCEVq3q8eKLAxky5DivQzPGRCDXkoKqFovIeGABznrPM1V1rYg8AqxU1XSc7qK6wN995Zi3qOoQt2L6nZzIH2i+6qq5zJnj3NA1Zkx3nnyyHw0aJHkclTEmUrla5kJV5wHzyu17yO95XzfPX6n8LOeRWAdqN/csjCN1222n8c0323nppUGcf34Hr8MxxkS42J3R7L/aWgQtGvPJJ5uZOHFx2fZZZ7Vj/fqbLSEYY4IidgviRdhqa3v3FnDPPQuZPv0rAM47rwO9ezsTwBMSYje3G2OCK3aTQgQVwps370duuOEDMjP3kpgYx/33n83pp4d2np8xJjbEblKIgEJ4u3bl8ac//YtZs9YA0LNna9LShtClS+SOgRhjwlvsJoUImM38yCOfMGvWGmrVSuDRR8/ntttOIz7euoqMMe6xpBBm3Ueqiu/2XCZOPJdff83lscfO55hjGnscmTEmFsTmx86SQti3FSQO6gexltIRUFVefvkrevWaSX5+MQCNGtVi9uzLLCEYY0ImNpNCzmZAoV47iPd+LeKNG/fQp89rjB37AcuWZfL222u9DskYE6Nis/soTAaZnQJ2y3nggY85cKCYZs1q8/zzF3L55Sd6GpcxJnbFZlIIg/GEtWt3cN116Xz55TYAhg8/ieeeG2AroRljPBWbSSEMCuGtWrWdL7/cRuvW9XjppUEMHNjJs1iMMeag2EwKHq24tnNnLs2a1QGclkF2dj4jRpxsBeyMMWEjRgeaQ9tSyMsr4q67PiQ5eTLr1+8EnGUyx4/vaQnBGBNWYq+loBrSlsKiRZsYM+Z9Nm7MIi5O+PTTnzjhhCCuHmeMMUEUe0kh71cozoOkxpDU0LXT5OTk8+c/LyQ19WsATjqpOTNnDiUlpZVr5zTGmCMVe0khBOUtPv98C8OGzWHbtn0kJsbx4IO9ueees6hRI961cxpjTDDEXlIIwWprLVrUZffuA5x+ehtmzBjMiSdaATtjTGSIvaTgQktBVVm4MIN+/Y5GRDj22MZ8/vm1dOvWwgrYGWMiSuz9xQryxLWtW3MYPPhNLrjgb/z1r9+U7T/11FaWEIwxESf2WgpBKnFRWuoUsLv77oXs21dIgwY1qVnTxgyMMZEt9pJCEFoKP/64mzFj3ueTT34C4OKLj+eFFy6iVat6wYjQGGM8E1tJoSjXuSU1vgbUPbxbQ7/4Yit9+rxGfn4xzZvXYerUC7nsss5layAYY0wki62kcHDSWv0OEHd4XT0pKa3o2LExp5zSkmee6U+TJlbAzhgTPWIsKVT/zqOCgmKeeuoLbrghhaZNa1OjRjxLllxHvXo1XQrSGGO8E1tJoZpzFJYty2T06HTWrdvJ+vW7+NvfLgGwhGCMiVqxlRQCbCnk5hbywAMfM3nyclShU6cm3HDDqSEI0BhjvBVbSSGAlsJHH2UwZsz7bNqUTXy8cPfdvZgw4VySkmLrW2WMiU2x9ZeuipLZ//nPbvr1ex1V6NatBWlpQ+jevWUIAzTGGG/FTlIoLYGczc7zBh0qPKRTpybcdttpNGtWh7vv7kViok1GM8bElthJCvszobQI6rSEROc20l9/3c+tt/6LG288lfPOcxLFs88O8DJKY4zxlKvFeURkgIj8ICIbROTeCl6vKSKzfa8vF5Fk14LxG2RWVV5/fTWdO0/j7bfXcvvtC1BV105tjDGRwrWkICLxwAvAhUBn4EoR6VzusNFAlqoeCzwLPO5WPAeTwpbCTgwc+AYjR77Lnj0H6N//GN59d5jNSDbGGNztPuoJbFDVDAAReQsYCqzzO2Yo8LDv+RxgqoiIuvCxvTQrg+lLenDPgnbsz9tAo0ZJPPvsBYwc2dUSgjHG+LiZFFoDW/22M4HTDnWMqhaLSA7QBNgV7GByft7ExIXnsD8PLr30BKZOvYgWLeoG+zTGGBPR3EwKFX38Lt8CCOQYRGQsMBagXbt2hxVMo3rCjCs/pLDHRC4dc/FhvYcxxkQ7N5NCJtDWb7sN8PMhjskUkQSgAbCn/BupaiqQCpCSknJ4XUuD3mTwQKWCnGOMMcbHzbuPVgAdRaSDiNQAhgHp5Y5JB67xPb8M+NiN8YQyIiC2GpoxxhyKay0F3xjBeGABEA/MVNW1IvIIsFJV04E04HUR2YDTQhjmVjzGGGOq5urkNVWdB8wrt+8hv+f5wB/djMEYY0zgrC/FGGNMGUsKxhhjylhSMMYYU8aSgjHGmDKWFIwxxpSRSKsOKiI7gZ8O88ub4kIJjTBn1xwb7Jpjw5Fcc3tVbVbVQRGXFI6EiKxU1RSv4wglu+bYYNccG0JxzdZ9ZIwxpowlBWOMMWViLSmkeh2AB+yaY4Ndc2xw/ZpjakzBGGNM5WKtpWCMMaYSUZkURGSAiPwgIhtE5N4KXq8pIrN9ry8XkeTQRxlcAVzzHSKyTkS+FZGPRKS9F3EGU1XX7HfcZSKiIhLxd6oEcs0icrnvZ71WRN4IdYzBFsD/7XYiskhEVvn+f1/kRZzBIiIzRWSHiHx3iNdFRKb4vh/fikj3oAagqlH1wCnTvRE4GqgBrAY6lzvmJmC67/kwYLbXcYfgms8Davuej4uFa/YdVw/4FFgGpHgddwh+zh2BVUAj33Zzr+MOwTWnAuN8zzsDm72O+wivuTfQHfjuEK9fBMzHWbnydGB5MM8fjS2FnsAGVc1Q1ULgLWBouWOGAq/6ns8B+ohIRUuDRooqr1lVF6lqnm9zGc5KeJEskJ8zwP8CTwD5oQzOJYFc8xjgBVXNAlDVHSGOMdgCuWYF6vueN+D3KzxGFFX9lApWoPQzFHhNHcuAhiLSMljnj8ak0BrY6red6dtX4TGqWgzkAE1CEp07Arlmf6NxPmlEsiqvWUROAdqq6gehDMxFgfycOwGdRGSJiCwTkQEhi84dgVzzw8DVIpKJs37LLaEJzTPV/X2vFlcX2fFIRZ/4y99iFcgxkSTg6xGRq4EU4BxXI3JfpdcsInHAs8CoUAUUAoH8nBNwupDOxWkNfiYiXVQ12+XY3BLINV8JvKKqT4vIGTirOXZR1VL3w/OEq3+/orGlkAm09dtuw++bk2XHiEgCTpOzsuZauAvkmhGRvsD9wBBVLQhRbG6p6prrAV2AxSKyGafvNT3CB5sD/b/9nqoWqeom4AecJBGpArnm0cDbAKq6FEjCqREUrQL6fT9c0ZgUVgAdRaSDiNTAGUhOL3dMOnCN7/llwMfqG8GJUFVes68r5SWchBDp/cxQxTWrao6qNlXVZFVNxhlHGaKqK70JNygC+b/9Ls5NBYhIU5zupIyQRhlcgVzzFqAPgIicgJMUdoY0ytBKB0b67kI6HchR1V+C9eZR132kqsUiMh5YgHPnwkxVXSsijwArVTUdSMNpYm7AaSEM8y7iIxfgNT8J1AX+7htT36KqQzwL+ggFeM1RJcBrXgD0F5F1QAlwt6ru9i7qIxPgNd8JvCwit+N0o4yK5A95IvImTvdfU984yQQgEUBVp+OMm1wEbADygGuDev4I/t4ZY4wJsmjsPjLGGHOYLCkYY4wpY0nBGGNMGUsKxhhjylhSMMYYU8aSggk7IlIiIt/4PZIrOTb5UNUkq3nOxb5KnKt9JSKOO4z3uFFERvqejxKRVn6vzRCRzkGOc4WIdAvga/4kIrWP9NwmNlhSMOHogKp283tsDtF5h6tqV5xiiU9W94tVdbqqvubbHAW08nvtelVdF5Qof4tzGoHF+SfAkoIJiCUFExF8LYLPRORr36NXBcecKCJf+loX34pIR9/+q/32vyQi8VWc7lPgWN/X9vHV6V/jq3Nf07d/kvy2PsVTvn0Pi8hdInIZTn2pWb5z1vJ9wk8RkXEi8oRfzKNE5PnDjHMpfoXQRORFEVkpzjoKE337bsVJTotEZJFvX38RWer7Pv5dROpWcR4TQywpmHBUy6/r6B3fvh1AP1XtDlwBTKng624EJqtqN5w/ypm+sgdXAGf69pcAw6s4/2BgjYgkAa8AV6jqSTgVAMaJSGPgD8CJqnoy8Kj/F6vqHGAlzif6bqp6wO/lOcAlfttXALMPM84BOGUtDrpfVVOAk4FzRORkVZ2CUxfnPFU9z1f64gGgr+97uRK4o4rzmBgSdWUuTFQ44PvD6C8RmOrrQy/BqelT3lLgfhFpA8xV1R9FpA9wKrDCV96jFk6CqcgsETkAbMYpv3wcsElV/+N7/VXgZmAqzvoMM0Tkn0DApblVdaeIZPhq1vzoO8cS3/tWJ846OGUf/FfdulxExuL8XrfEWXDm23Jfe7pv/xLfeWrgfN+MASwpmMhxO/Ar0BWnhfu7RXNU9Q0RWQ4MBBaIyPU4ZYZfVdX7AjjHcP+CeSJS4Robvno8PXGKsA0DxgPnV+NaZgOXA98D76iqivMXOuA4cVYgmwS8AFwiIh2Au4AeqpolIq/gFIYrT4CFqnplNeI1McS6j0ykaAD84quRPwLnU/J/EZGjgQxfl0k6TjfKR8BlItLcd0xjCXx96u+BZBE51rc9AvjE1wffQFXn4QziVnQH0D6c8t0VmQtcjLMOwGzfvmrFqapFON1Ap/u6nuoDuUCOiBwFXHiIWJYBZx68JhGpLSIVtbpMjLKkYCLFNOAaEVmG03WUW8ExVwDficg3wPE4Sxauw/nj+aGIfAssxOlaqZKq5uNUoPy7iKwBSoHpOH9gP/C93yc4rZjyXgGmHxxoLve+WcA6oL2qfunbV+04fWMVTwN3qepqnLWZ1wIzcbqkDkoF5ovIIlXdiXNn1Ju+8yzD+V4ZA1iVVGOMMX6spWCMMaaMJQVjjDFlLCkYY4wpY0nBGGNMGUsKxhhjylhSMMYYU8aSgjHGmDKWFIwxxpT5f6ECaWxsd/bDAAAAAElFTkSuQmCC\n",
        "text/plain": "<Figure size 432x288 with 1 Axes>"
       },
       "metadata": {
        "needs_background": "light"
       },
       "output_type": "display_data"
      }
     ],
     "persistentId": "2c810e70-a822-4d9a-a9b0-2622dfa247bf",
     "text": "y_true = coding_scores_not_low_complexity_unspliced['in_cds_no_stop_codon']\ny_pred = coding_scores_not_low_complexity_unspliced['jaccard_in_peptide_db']\n\nfpr, tpr, thresholds = metrics.roc_curve(y_true, y_pred)\n\nfig, ax = plt.subplots()\nlw = 2\nax.plot(fpr, tpr, lw=2, color='darkorange')\nax.plot([0, 1], [0, 1], color='navy', lw=lw, linestyle='--')\nax.set(xlabel='False Positive Rate', ylabel='True Positive Rate')\n\nprint(metrics.accuracy_score(y_true, y_pred > 0.5))\n\nprint(metrics.classification_report(y_true, y_pred > 0.5))"
    },
    "executionTime": "2019-10-12T03:42:10.801Z"
   },
   {
    "cell": {
     "executionCount": 66,
     "executionEventId": "6e071a64-a3c4-41b1-a7b1-21910c9a82c3",
     "hasError": false,
     "id": "c4e90627-a46a-4fa2-b552-6402975ee8c7",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "0.9204152249134948\n              precision    recall  f1-score   support\n\n       False       1.00      0.91      0.95       767\n        True       0.59      0.98      0.74       100\n\n   micro avg       0.92      0.92      0.92       867\n   macro avg       0.80      0.95      0.85       867\nweighted avg       0.95      0.92      0.93       867\n\n"
      },
      {
       "data": {
        "image/png": "iVBORw0KGgoAAAANSUhEUgAAAYUAAAEKCAYAAAD9xUlFAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDMuMC4zLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvnQurowAAIABJREFUeJzt3Xl8FFW2wPHfyQJh35GdoIKKKIgBFRUXFlE2Rx1FEUQRFEUdt1GfC+LzObgLImIkjMug4jCo0YFBRsEFAUEREdARAkJQZEsCJGQ/749qYk8MSQe6uno538+nP3RVV7pOJSSn7711zxVVxRhjjAGI8zoAY4wx4cOSgjHGmDKWFIwxxpSxpGCMMaaMJQVjjDFlLCkYY4wpY0nBGGNMGUsKxhhjylhSMMYYUybB6wCqq2nTppqcnOx1GMYYE1G++uqrXararKrjIi4pJCcns3LlSq/DMMaYiCIiPwVynHUfGWOMKWNJwRhjTBlLCsYYY8pYUjDGGFPGkoIxxpgyriUFEZkpIjtE5LtDvC4iMkVENojItyLS3a1YjDHGBMbNlsIrwIBKXr8Q6Oh7jAVedDEWY4wxAXBtnoKqfioiyZUcMhR4TZ31QJeJSEMRaamqv7gVU9jY+xP8+yYo3Od1JMaYCPDlhvokJZZy8tk9odfDrp7Ly8lrrYGtftuZvn2/SwoiMhanNUG7du1CEpxrVGHhDbB5gdeRGGPCnCrc889+PP3JqZzc8le+PO57El0+p5dJQSrYpxUdqKqpQCpASkpKhcdEhB/+Dp/cBfu2QM0GMOjvkFDT66iMMWFKADZugE+30n/oaZT06BHVSSETaOu33Qb42aNYQmPVFCchAJw9CZL7eRuPMSbsZGfnk5GRRffuLQGY+MwZDBuzq2zbbV4mhXRgvIi8BZwG5ETFeMKOb+DjW6Aot4LXVjn/9p0OXW8IbVzGmLD33nvfM27cP4mLE9auvYkGDZKoVSsxZAkBXEwKIvImcC7QVEQygQngtHxUdTowD7gI2ADkAde6FUtIfTcTtn1+6NfjEuDogaGLxxgT9nbsyOXWW+cze/ZaAE4/vQ3Z2fk0aJAU8ljcvPvoyipeV+Bmt87vmeyNzr/nPgNtzvn963VbQZ0WoY3JGBOWVJVZs9Zw223/Ys+eA9Sunchjj53P+PE9iY/3Zm5xxJXODns5Gc6/bc+H5l29jcUYE9bGjfsnL730FQB9+x5NauogOnRo5GlMVuYimLQUcjY5zxse7W0sxpiwd/HFx9OwYRJpaUP48MOrPU8IYC2F4Nr/M5QUQO3mUKOe19EYY8LMjz/u5qOPNnHjjSkADBhwLJs33+bJ2MGhWFIIpoPjCQ2slWCM+U1xcSnPPLOUCRMWU1BQTLduLTj99DYAYZUQwJJCcB1MCg2P8TYOY0zYWL16O6NHp/PVV84d9yNHdqVjx8YeR3VolhSC6eAgs7UUjIl5BQXFPProp0yatITi4lLatWvASy8NYsCAY70OrVKWFILJWgrGGJ/77vuIZ59dBsDNN/fgL3/pQ7164V/WxpJCMJW1FCwpGBPr/vznM1m6NJMnnujL2We39zqcgNktqcFU1lKw7iNjYs3ChRu59NK3KS4uBaBFi7p88cV1EZUQwJJC8BTkQP5uSKgFdUJXp8QY462srAOMHv0e/fv/jblz1/PXv64qe02komLQ4c26j4Il22+QOQL/Ixhjqu+dd9Zz003z2L59PzVrxjNhwjmMGtXN67COiCWFYMmxOQrGxIrt2/dzyy3zmTNnHQC9erUlLW0Ixx/f1OPIjpwlhWA52FKwO4+MiXrvvfc9c+aso06dRCZN6stNN/UgLi46eggsKQSLtRSMiWr5+cUkJTl/MseMOZWMjCzGjetBcnJDjyMLLhtoDhabo2BMVCotVaZO/ZIOHSbz00/ZAMTFCY8/3i/qEgJYUgieHOs+Miba/PDDLnr3/iu33DKf7dv38+ab33kdkuus+ygYSopg7xZAoH6y19EYY45QUVEJTz31BRMnfkJBQQlHHVWHadMGcsklJ3gdmussKQTD3p9AS6BeW0gI/2nsxphD++67HYwc+Q6rVm0H4Npru/H00/1p1KiWx5GFhiWFYLBCeMZEjdJSZc2aHbRv34DU1MH07x9bXcKWFILBBpmNiWhr1+6gc+dmiAgnn3wU7703jN6921O3bg2vQws5G2gOBhtkNiYi7dtXwPjx8+jS5UX+8Y/1ZfsvuqhjTCYEsJZCcNiKa8ZEnAULNjB27Ads2ZJDQkIcmzdnex1SWLCkEAw51n1kTKTYs+cAt9++gNdeWw1A9+4tSUsbQrduLTyOLDxYUjhSqv9dDM8YE7a++WY7Awb8jV9/zaVmzXgmTjyXO+/sRUKC9aQfZEnhSB3YCUX7oWYDSArfdVeNMdCpUxPq1q1Bp05NmDFjCJ06NfE6pLBjSeFIZfuttmYls40JK6rKG2+sYfDg46hfvya1ayeyePEoWrWqFzUF7ILN2kxHKsdWWzMmHG3enM0FF/yNq69+h3vv/XfZ/jZt6ltCqIS1FI5U2Z1HNshsTDgoKSll2rQV3HffR+TmFtG4cS169WrrdVgRw5LCkbI5CsaEjfXrdzJ6dDpLl2YCcPnlJ/L88xfSvHkdjyOLHJYUjpTNUTAmLGzalEW3bi9RWFhCy5Z1mTZtIBdffLzXYUUcV5OCiAwAJgPxwAxVnVTu9XbAq0BD3zH3quo8N2MKOmspGBMWOnRoxB//2JmkpASeeqo/DRsmeR1SRHItKYhIPPAC0A/IBFaISLqqrvM77AHgbVV9UUQ6A/OAZLdiCrqiA7D/Z4hLgHptvI7GmJhy4EARjzzyCX/4wwn07NkagFdfvZj4eLt/5ki4+d3rCWxQ1QxVLQTeAoaWO0aB+r7nDYCfXYwn+A62EuonO4nBGBMSn332E926vcSkSUsYO/Z9SksVwBJCELj5l6w1sNVvOxM4rdwxDwMfisgtQB2gr4vxBJ91HRkTUnv3FnDfff9m2rSVAHTu3Izp0wfZLaZB5GZareinpOW2rwReUdU2wEXA6yLyu5hEZKyIrBSRlTt37nQh1MNkg8zGhMy8eT/Spcs0pk1bSUJCHA891Juvvx7L6adb120wudlSyAT8bw5uw++7h0YDAwBUdamIJAFNgR3+B6lqKpAKkJKSUj6xeMfWUTAmJHJy8hk+fC7Z2fmkpLQiLW0IJ598lNdhRSU3k8IKoKOIdAC2AcOAq8odswXoA7wiIicASUAYNQWqYCuuGeMaVUUV4uKEBg2SmDJlAL/+msuf/nS6FbBzkWtJQVWLRWQ8sADndtOZqrpWRB4BVqpqOnAn8LKI3I7TtTRKVcOnJVAVaykY44qff97HTTf9k7PPbsedd/YCYMSIrh5HFRtcvWXGN+dgXrl9D/k9Xwec6WYMrtFS2LvJeW4tBWOCQlWZOXMVd975ITk5BSxblslNN/WgVq1Er0OLGXYf5eHatw1KCqF2c6hR1+tojIl4GRlZjBnzPh9/7HzYGjiwI9OnD7KEEGKWFA5XjhXCMyYYSkpKmTJlOfff/zEHDhTTtGltpkwZwLBhXRArRx9ylhQOV7bNUTAmWObMWc+BA8VceWUXJk8eQLNmVsDOK5YUDleOzVEw5nAVFpawb18BTZrUJj4+jrS0Ifz4424GDz7O69Bint3XdbispWDMYVmxYhspKamMGPEOB282PP74ppYQwoS1FA6XtRSMqZa8vCImTFjEM88so7RUycsrYseOXI46ym7UCCeWFA6XzVEwJmCLF29mzJj32bBhD3Fxwl13ncHEiedRu7bdWRRuLCkcjvxsyN8DCbWgTguvozEmbKkqt946n6lTVwBw0knNSUsbQo8erT2OzByKJYXD4V/ewm6ZM+aQRIT69WuSmBjHAw/05t57z6JGjXivwzKVsKRwOKxktjGHtGtXHhs37uG005zqpQ8+eA7Dh59M587NPI7MBCKgu49EpIaIHOt2MBHDSmYb8zuqyltvfccJJ7zAxRfPJivrAABJSQmWECJIlUlBRAYCa4CFvu1uIvKO24GFNRtkNua/ZGbuZejQt7jyyn+wa1cenTs3Iy+vyOuwzGEIpPvoEZwV0xYBqOo3Md9qsO4jYwAoLVVmzPiau+9eyN69BdSvX5Onn+7P6NGnWImKCBVIUihS1exyP+DIKW/tBus+MgaA0aPTeeWVbwAYMuQ4pk27iNat61fxVSacBTKmsF5ELgfiRKSDiDwHLHM5rvBVUgj7tgAC9ZO9jsYYT1199Uk0b16Ht966lHffvcISQhQIJCmMB04FSoG5QD5wm5tBhbW9W5y1FOq1hYSaXkdjTEh9990OJk/+7TNhnz5Hk5FxK1dcYRVNo0Ug3UcXqOo9wD0Hd4jIJTgJIvYcLG/R0LqOTOwoKCjmL3/5nMce+4yiolJSUlpx5pntAKhTp4bH0ZlgCqSl8EAF++4PdiAR42AhPFtHwcSI5cszOfXUVCZO/ISiolLGjUvhpJOO8jos45JDthRE5AJgANBaRJ7xe6k+TldSbMq2loKJDbm5hTz44CKee24ZqtCxY2NmzBhC797tvQ7NuKiy7qMdwHc4Ywhr/fbvA+51M6iwZiuumRhx//0fM3nycuLihLvvPoOHHz7XlsaMAYdMCqq6ClglIrNUNT+EMYU3m6NgYsT995/NmjU7ePzxvqSktPI6HBMigYwptBaRt0TkWxH5z8GH65GFI1Wbo2CiVnr6D1x00SyKikoAaNasDh99NNISQowJJCm8AvwVEOBC4G3gLRdjCl8HdkJRLtRsCLUaex2NMUGxY0cuw4bNYejQt5g/fwOvvrra65CMhwJJCrVVdQGAqm5U1QeA89wNK0xZK8FEEVXlb3/7lhNOeIHZs9dSu3YikycP4Npru3kdmvFQIPMUCsSZlbJRRG4EtgHN3Q0rTFkhPBMltmzJ4cYbP2D+/A0A9O17NKmpg+jQoZHHkRmvBZIUbgfqArcC/wc0AK5zM6iwZYPMJkp8+OFG5s/fQMOGSTzzTH9GjepmM5INEEBSUNXlvqf7gBEAItLGzaDClnUfmQiWm1tYNvt49OhT2LZtL2PHnkrLlvU8jsyEk0rHFESkh4hcLCJNfdsnishrxGpBPGspmAhUXFzKE08soX3758jIyAKcZTInTDjXEoL5nUMmBRH5CzALGA78S0Tux1lTYTXQKTThhRkbUzARZvXq7Zx22gzuueff7N59gHff/d7rkEyYq6z7aCjQVVUPiEhj4Gff9g+hCS3MFOVB7i8Qlwh1Y7P3zESOgoJiHn30UyZNWkJxcSnt2jUgNXUQF1wQ2+tjmapVlhTyVfUAgKruEZHvYzYhAORscv5tkAxx8Z6GYkxlVq36heHD57J+/S5EYPz4Hjz2WB/q1bNS76ZqlSWFo0XkYHlsAZL9tlHVS6p6cxEZAEwG4oEZqjqpgmMuBx7GWc1ttapeFXj4IWSDzCZC1KyZwMaNWRx3XBNmzBjCWWe18zokE0EqSwqXltueWp03FpF44AWgH5AJrBCRdFVd53dMR+A+4ExVzRKR8J3/kGMls034+vrrXzjllBaICJ07N2P+/OH06tWWpKRA7jo35jeVFcT76AjfuyewQVUzAETkLZxxinV+x4wBXlDVLN85dxzhOd1jg8wmDGVlHeCuuz5k5sxvePPNSxk2rAsA55/fwePITKRy82NEa2Cr33YmcFq5YzoBiMgSnC6mh1X1X+XfSETGAmMB2rXzqCmcY91HJry88856brppHtu376dmzXh2787zOiQTBdxMChVNj9QKzt8ROBdoA3wmIl1UNfu/vkg1FUgFSElJKf8eoZFtcxRMeNi+fT+33DKfOXOcRveZZ7ZlxowhHH98U48jM9Eg4KQgIjVVtaAa750JtPXbboNzW2v5Y5apahGwSUR+wEkSK6pxHveVlsDeg3cfWbPceOerr36mX7/XycrKp06dRCZN6stNN/UgLs5KVJjgqLJKqoj0FJE1wI++7a4i8nwA770C6CgiHUSkBjAMSC93zLv4Kq76Zk13AjKqEX9o7N8GJYVQ+yioUdfraEwM69y5Gc2a1eGCC45h7dqbGD++pyUEE1SBlM6eAgwCdgOo6moCKJ2tqsXAeGABsB54W1XXisgjIjLEd9gCYLeIrMOZLX23qu6u/mW4zMpbGI+UliqpqV+Rne0sflirViKffjqK+fOH0759Q4+jM9EokO6jOFX9qVwFxZJA3lxV5wHzyu17yO+5Anf4HuHL5igYD/zwwy6uv/59Pv98CytWbOPll53PUkcdZa1V455AksJWEekJqG/uwS1AbC3HaS0FE0JFRSU8/fRSHn54MQUFJbRoUZcLL+zodVgmRgSSFMbhdCG1A34F/u3bFztsjoIJkVWrfmH06HRWrdoOwLXXduPpp/vTqFEtjyMzsSKQpFCsqsNcjyScWfeRCYGNG/fQs+cMiotLSU5uSGrqIPr1sw8iJrQCSQorfLeKzgbmquo+l2MKP9Z9ZELgmGMaM2LEydSrV4P/+78+1K1bw+uQTAyq8u4jVT0GeBQ4FVgjIu+KSOy0HPKzIX8PJNR2bkk1Jkj27y/k1lvns3TpbxP/09KGMHnyhZYQjGcCuSUVVf1CVW8FugN7cRbfiQ1lrYSjwdawNUGyYMEGTjxxGs8//yU33vhPnBvxsHWSjeeq7D4Skbo4heyGAScA7wG9XI4rfJSNJ1jXkTlye/Yc4PbbF/Daa6sBOPXUlqSlDbFkYMJGIGMK3wHvA0+o6mcuxxN+yu48skFmc2TmzFnHzTfPY8eOXJKSEpg48VzuuOMMEhICarAbExKBJIWjVbXU9UjCla2jYIIgOzufsWPfJysrn9692/Pyy4Pp1KmJ12EZ8zuHTAoi8rSq3gn8Q0R+V5k0kJXXokKOtRTM4VFVSkuV+Pg4GjZMYtq0gWRlHeCGG1KsXpEJW5W1FGb7/q3WimtRJ9taCqb6Nm/OZuzY9zn//A7ce+9ZAGUL4BgTzg7ZmamqX/qenqCqH/k/cAaco19JIezbAgg0SPY6GhMBSkpKmTJlOV26TGPhwgymTv2S/Pxir8MyJmCBjHBdV8G+0cEOJCzt/Qm0FOq1hXi7b9xUbv36nfTu/Qq33fYvcnOLGDasC19/fYOtk2wiSmVjClfg3IbaQUTm+r1UD8iu+KuijM1kNgEoLi7l8cc/55FHPqWwsIRWrerx4osDGTLkOK9DM6baKvsI8yXOGgptgBf89u8DVrkZVNiwQngmAHFxwocfZlBYWMKYMd154ol+NGyY5HVYxhyWQyYFVd0EbMKpihqbrBCeOYQDB4rYt6+Q5s3rEBcnzJgxmK1b93L++bZcq4lshxxTEJFPfP9micgev0eWiOwJXYgesu4jU4FPP/2Jrl2nc/XVc8vKU3Ts2MQSgokKlXUfHVxys2koAglL1lIwfvbuLeC++/7NtGkrAUhMjGfXrjyaNavjcWTGBE9lt6QenMXcFohX1RLgDOAGIPp/C1StpWDKzJ//I126TGPatJUkJMQxYcI5fP31WEsIJuoEcq/cu0APETkGeA34J/AGMMjNwDyXtwOKciGpkfMwMUlVGTPmfdLSnHsrUlJaMXPmEE46ycqom+gUyDyFUlUtAi4BnlPVW4DW7oYVBqzryOCUsm7Tpj5JSQk89VQ/li4dbQnBRLWAluMUkT8CI4CLffsS3QspTFghvJj188/72LhxD2ef3R6A//mfsxkx4mSOOaaxx5EZ475AZzSfh1M6O0NEOgBvuhtWGLCS2TFHVUlL+5rOnV/g0kvfZvfuPABq1Ii3hGBiRpUtBVX9TkRuBY4VkeOBDar6f+6H5jFrKcSUjIwsxox5n48/3gTAoEGdKCqK3YrxJnYFsvLa2cDrwDZAgBYiMkJVl7gdnKdsNnNMOFjA7oEHFpGXV0TTprWZMmUAw4Z1sdXQTEwKZEzhWeAiVV0HICIn4CSJFDcD85ytoxATRo58lzfeWAPAVVedxHPPXWC3mZqYFsiYQo2DCQFAVdcD0V0ytCgPcrdDXCLUbeN1NMZFY8Z0p02b+qSnD2PWrEssIZiYF0hL4WsReQmndQAwnGgviFc2ntAB4uK9jcUE1YoV2/j4403cc4+z8M255yazYcMt1Kxp5a2NgcCSwo3ArcCfccYUPgWedzMoz5WttmZdR9EiL6+ICRMW8cwzyygtVXr1alt2y6klBGN+U+lvg4icBBwDvKOqT4QmpDCQY4PM0WTx4s1cf306GzdmERcn3HXXGZx6aiuvwzImLFVWJfV/cEpcDAcWikhFK7BVSkQGiMgPIrJBRO6t5LjLRERFJDwGr202c1TIycnnhhve57zzXmXjxixOOqk5y5aN5skn+1O7dvTPvzTmcFTWUhgOnKyquSLSDJgHzAz0jUUkHmdxnn5AJrBCRNL9B619x9XD6Z5aXt3gXWOF8KLCgw8uIjX1axIT43jwwd7cc89Z1KhhY0TGVKaypFCgqrkAqrpTRAK5U8lfT5yJbhkAIvIWMBRYV+64/wWeAO6q5vu7x+YoRCxVLZtf8NBD57BpUzaTJvXhxBObexyZMZGhsj/0R4vIXN/jHeAYv+25lXzdQa2BrX7bmZQrpCcipwBtVfWDakfultISyHFmtdLAFk2JFKrKG2+s4fzzX6OwsASApk1r8/77V1pCMKYaKmspXFpue2o137ui6aBa9qLT8ngWGFXlG4mMBcYCtGvXrpphVNP+bVBaBHVaQKLdsx4JMjP3Mm7cP/ngg/8AMGvWt1x77SkeR2VMZKpsjeaPjvC9M3EW6DmoDfCz33Y9oAuw2NfcbwGki8gQVV1ZLpZUIBUgJSVFcZMNMkeM0lLl5Ze/4u67F7JvXyENGtTk6af7M2pUN69DMyZiuXmD9gqgo6+q6jZgGHDVwRdVNQe/pT5FZDFwV/mEEHI2yBwRNmzYw5gx77N48WYAhg49jmnTBtKqVT1vAzMmwrmWFFS1WETGAwuAeGCmqq4VkUeAlaqa7ta5j0hZS8GSQjj77LOfWLx4M82b12Hq1Au57LLOVsDOmCAIOCmISE1VLajOm6vqPJxbWf33PXSIY8+tznu7xtZRCFvZ2fk0bJgEwKhR3di5M4/Ro0+hSZPaHkdmTPSo8jZTEekpImuAH33bXUUkestc2DoKYaegoJgJExbRvv1z/PjjbsBZJvPPfz7TEoIxQRZIS2EKMAhndjOqulpEznM1Ki9ZiYuwsmxZJqNHp7Nu3U4AFizYSMeOTTyOypjoFUhSiFPVn8r115a4FI+38rOcR2IdqG33tnspN7eQBx9cxHPPLUMVOnZsTFrakLIidsYYdwSSFLaKSE9AfaUrbgH+425YHsnxq45qg5aeWb48k6uumktGRhbx8cJdd/ViwoRzqFXL6hUZ47ZAksI4nC6kdsCvwL99+6KPzVEICw0bJrFt2166dj2KtLQhVtHUmBCqMimo6g6cOQbRL9vmKHjl88+3cOaZbRERjjuuKR9/fA09erQiMdEK2BkTSlUmBRF5Gb/yFAep6lhXIvKSDTKH3I4dudx663xmz17Lq69ezMiRXQHo1attFV9pjHFDIN1H//Z7ngT8gf8udBc9cmzFtVBRVWbNWsNtt/2LPXsOULt2YlkhO2OMdwLpPprtvy0irwMLXYvIS1YyOyS2bMnhxhs/YP78DQD063c0qamDSU5u6HFkxpjDKXPRAYi++wJLCmHfVpA4qB99lxculi/PpG/f19m/v5CGDZN49tkLuOaarlaiwpgwEciYQha/jSnEAXuAQy6tGbH2/gRa6iSE+BpeRxO1unVrQdu29Tn++Ka88MJFtGxpBeyMCSeVJgVxPr51xalyClCqqu6WrvaKdR25ori4lKlTv2TkyK40blyLmjUTWLLkOho1quV1aMaYClRa+8iXAN5R1RLfIzoTAtgcBResXr2d006bwe23L+COOxaU7beEYEz4CmTd5S9FpLvrkXjNCuEFTX5+MQ888DEpKS/z9de/0K5dA668sovXYRljAnDI7iMRSVDVYuAsYIyIbARycZbZVFWNrkRh3UdB8cUXWxk9Op3vv9+FCIwf34PHHutDvXo1vQ7NGBOAysYUvgS6AxeHKBZvla24Zt1Hh2vDhj2cffZfKS1VjjuuCWlpQzjzTJfX1DbGBFVlSUEAVHVjiGLxjqp1HwXBscc2ZuzY7jRuXIsHHzyHpCQ3V3s1xrihst/aZiJyx6FeVNVnXIjHG3m/QlEuJDWCJJtAFaisrAPceeeHXHttt7KS1tOmDbQ5B8ZEsMqSQjxQF1+LIaplWyuhuubOXc/NN89j+/b9fPXVL3zzzQ2IiCUEYyJcZUnhF1V9JGSReMkK4QVs+/b9jB8/j3/8Yz0AZ53VjhkzBlsyMCZKVDmmEBOyrRBeVVSV115bze23LyArK5+6dWvw+ON9ufHGFOLiYue/ijHRrrKk0CdkUXjNWgpVys7O5847PyQrK58BA45l+vSBtG9v4y/GRJtDJgVV3RPKQDxls5krVFqqlJYqCQlxNGpUi5deGkReXhFXX32ydRcZE6UCmdEc/XJsxbXyvv9+F717/5VJkz4v23fppZ0ZMcIqmhoTzSwpFOVC7nanMmrd1l5H47miohIee+wzunadzpIlW0lLW0V+frHXYRljQsRmF+Vscv6tnwxxsb0e8KpVv3Dddel88812AEaPPoUnn+xnk9CMiSH22241jygqKmHChMU88cQSSkqU5OSGvPzyYPr2tTEWY2KNJYWyQebYTQoJCXEsX76N0lLltttO49FHz6duXVtoyJhYZEkhRgvh7dtXwL59hbRqVQ8RYcaMwWzfvp8zzmjrdWjGGA/ZQHMMthQWLNhAly4vMnz4XA6um9ShQyNLCMYYSwqx1FLYvTuPa655lwEDZrFlSw779hWwe/cBr8MyxoQRV5OCiAwQkR9EZIOI3FvB63eIyDoR+VZEPhKR9m7G8zulJb/dfRTFE9dUlTlz1tG58zRee201SUkJPPFEX5Ytu56mTWt7HZ4xJoy4NqYgIvHAC0A/IBNYISLpqrrO77BVQIqq5onIOOAJ4Aq3Yvqd/ZlQWgR1WkJidP5xVFWGD5/Lm29+B0DRZJGNAAAPYElEQVTv3u15+eXBdOrUxOPIjDHhyM2WQk9gg6pmqGoh8BYw1P8AVV2kqnm+zWVAGxfj+b0YKIQnInTu3Ix69Wrw4osDWbToGksIxphDcvPuo9bAVr/tTOC0So4fDcyv6AURGQuMBWjXLojLO0bpHIVNm7LIyMiiTx8n2d1zz5mMGtWNNm3qexyZMSbcudlSqKhAjlZ4oMjVQArwZEWvq2qqqqaoakqzZs2CF2FOdBXCKykpZfLkZXTp8iJXXDGHHTtyAUhMjLeEYIwJiJsthUzA/x7HNsDP5Q8Skb7A/cA5qlrgYjy/lx09hfDWrdvJ9dens3RpJgBDhhxn6xwYY6rNzaSwAugoIh2AbcAw4Cr/A0TkFOAlYICq7nAxlopFwToKRUUlPP74Ev73fz+lsLCEVq3q8eKLAxky5DivQzPGRCDXkoKqFovIeGABznrPM1V1rYg8AqxU1XSc7qK6wN995Zi3qOoQt2L6nZzIH2i+6qq5zJnj3NA1Zkx3nnyyHw0aJHkclTEmUrla5kJV5wHzyu17yO95XzfPX6n8LOeRWAdqN/csjCN1222n8c0323nppUGcf34Hr8MxxkS42J3R7L/aWgQtGvPJJ5uZOHFx2fZZZ7Vj/fqbLSEYY4IidgviRdhqa3v3FnDPPQuZPv0rAM47rwO9ezsTwBMSYje3G2OCK3aTQgQVwps370duuOEDMjP3kpgYx/33n83pp4d2np8xJjbEblKIgEJ4u3bl8ac//YtZs9YA0LNna9LShtClS+SOgRhjwlvsJoUImM38yCOfMGvWGmrVSuDRR8/ntttOIz7euoqMMe6xpBBm3Ueqiu/2XCZOPJdff83lscfO55hjGnscmTEmFsTmx86SQti3FSQO6gexltIRUFVefvkrevWaSX5+MQCNGtVi9uzLLCEYY0ImNpNCzmZAoV47iPd+LeKNG/fQp89rjB37AcuWZfL222u9DskYE6Nis/soTAaZnQJ2y3nggY85cKCYZs1q8/zzF3L55Sd6GpcxJnbFZlIIg/GEtWt3cN116Xz55TYAhg8/ieeeG2AroRljPBWbSSEMCuGtWrWdL7/cRuvW9XjppUEMHNjJs1iMMeag2EwKHq24tnNnLs2a1QGclkF2dj4jRpxsBeyMMWEjRgeaQ9tSyMsr4q67PiQ5eTLr1+8EnGUyx4/vaQnBGBNWYq+loBrSlsKiRZsYM+Z9Nm7MIi5O+PTTnzjhhCCuHmeMMUEUe0kh71cozoOkxpDU0LXT5OTk8+c/LyQ19WsATjqpOTNnDiUlpZVr5zTGmCMVe0khBOUtPv98C8OGzWHbtn0kJsbx4IO9ueees6hRI961cxpjTDDEXlIIwWprLVrUZffuA5x+ehtmzBjMiSdaATtjTGSIvaTgQktBVVm4MIN+/Y5GRDj22MZ8/vm1dOvWwgrYGWMiSuz9xQryxLWtW3MYPPhNLrjgb/z1r9+U7T/11FaWEIwxESf2WgpBKnFRWuoUsLv77oXs21dIgwY1qVnTxgyMMZEt9pJCEFoKP/64mzFj3ueTT34C4OKLj+eFFy6iVat6wYjQGGM8E1tJoSjXuSU1vgbUPbxbQ7/4Yit9+rxGfn4xzZvXYerUC7nsss5layAYY0wki62kcHDSWv0OEHd4XT0pKa3o2LExp5zSkmee6U+TJlbAzhgTPWIsKVT/zqOCgmKeeuoLbrghhaZNa1OjRjxLllxHvXo1XQrSGGO8E1tJoZpzFJYty2T06HTWrdvJ+vW7+NvfLgGwhGCMiVqxlRQCbCnk5hbywAMfM3nyclShU6cm3HDDqSEI0BhjvBVbSSGAlsJHH2UwZsz7bNqUTXy8cPfdvZgw4VySkmLrW2WMiU2x9ZeuipLZ//nPbvr1ex1V6NatBWlpQ+jevWUIAzTGGG/FTlIoLYGczc7zBh0qPKRTpybcdttpNGtWh7vv7kViok1GM8bElthJCvszobQI6rSEROc20l9/3c+tt/6LG288lfPOcxLFs88O8DJKY4zxlKvFeURkgIj8ICIbROTeCl6vKSKzfa8vF5Fk14LxG2RWVV5/fTWdO0/j7bfXcvvtC1BV105tjDGRwrWkICLxwAvAhUBn4EoR6VzusNFAlqoeCzwLPO5WPAeTwpbCTgwc+AYjR77Lnj0H6N//GN59d5jNSDbGGNztPuoJbFDVDAAReQsYCqzzO2Yo8LDv+RxgqoiIuvCxvTQrg+lLenDPgnbsz9tAo0ZJPPvsBYwc2dUSgjHG+LiZFFoDW/22M4HTDnWMqhaLSA7QBNgV7GByft7ExIXnsD8PLr30BKZOvYgWLeoG+zTGGBPR3EwKFX38Lt8CCOQYRGQsMBagXbt2hxVMo3rCjCs/pLDHRC4dc/FhvYcxxkQ7N5NCJtDWb7sN8PMhjskUkQSgAbCn/BupaiqQCpCSknJ4XUuD3mTwQKWCnGOMMcbHzbuPVgAdRaSDiNQAhgHp5Y5JB67xPb8M+NiN8YQyIiC2GpoxxhyKay0F3xjBeGABEA/MVNW1IvIIsFJV04E04HUR2YDTQhjmVjzGGGOq5urkNVWdB8wrt+8hv+f5wB/djMEYY0zgrC/FGGNMGUsKxhhjylhSMMYYU8aSgjHGmDKWFIwxxpSRSKsOKiI7gZ8O88ub4kIJjTBn1xwb7Jpjw5Fcc3tVbVbVQRGXFI6EiKxU1RSv4wglu+bYYNccG0JxzdZ9ZIwxpowlBWOMMWViLSmkeh2AB+yaY4Ndc2xw/ZpjakzBGGNM5WKtpWCMMaYSUZkURGSAiPwgIhtE5N4KXq8pIrN9ry8XkeTQRxlcAVzzHSKyTkS+FZGPRKS9F3EGU1XX7HfcZSKiIhLxd6oEcs0icrnvZ71WRN4IdYzBFsD/7XYiskhEVvn+f1/kRZzBIiIzRWSHiHx3iNdFRKb4vh/fikj3oAagqlH1wCnTvRE4GqgBrAY6lzvmJmC67/kwYLbXcYfgms8Davuej4uFa/YdVw/4FFgGpHgddwh+zh2BVUAj33Zzr+MOwTWnAuN8zzsDm72O+wivuTfQHfjuEK9fBMzHWbnydGB5MM8fjS2FnsAGVc1Q1ULgLWBouWOGAq/6ns8B+ohIRUuDRooqr1lVF6lqnm9zGc5KeJEskJ8zwP8CTwD5oQzOJYFc8xjgBVXNAlDVHSGOMdgCuWYF6vueN+D3KzxGFFX9lApWoPQzFHhNHcuAhiLSMljnj8ak0BrY6red6dtX4TGqWgzkAE1CEp07Arlmf6NxPmlEsiqvWUROAdqq6gehDMxFgfycOwGdRGSJiCwTkQEhi84dgVzzw8DVIpKJs37LLaEJzTPV/X2vFlcX2fFIRZ/4y99iFcgxkSTg6xGRq4EU4BxXI3JfpdcsInHAs8CoUAUUAoH8nBNwupDOxWkNfiYiXVQ12+XY3BLINV8JvKKqT4vIGTirOXZR1VL3w/OEq3+/orGlkAm09dtuw++bk2XHiEgCTpOzsuZauAvkmhGRvsD9wBBVLQhRbG6p6prrAV2AxSKyGafvNT3CB5sD/b/9nqoWqeom4AecJBGpArnm0cDbAKq6FEjCqREUrQL6fT9c0ZgUVgAdRaSDiNTAGUhOL3dMOnCN7/llwMfqG8GJUFVes68r5SWchBDp/cxQxTWrao6qNlXVZFVNxhlHGaKqK70JNygC+b/9Ls5NBYhIU5zupIyQRhlcgVzzFqAPgIicgJMUdoY0ytBKB0b67kI6HchR1V+C9eZR132kqsUiMh5YgHPnwkxVXSsijwArVTUdSMNpYm7AaSEM8y7iIxfgNT8J1AX+7htT36KqQzwL+ggFeM1RJcBrXgD0F5F1QAlwt6ru9i7qIxPgNd8JvCwit+N0o4yK5A95IvImTvdfU984yQQgEUBVp+OMm1wEbADygGuDev4I/t4ZY4wJsmjsPjLGGHOYLCkYY4wpY0nBGGNMGUsKxhhjylhSMMYYU8aSggk7IlIiIt/4PZIrOTb5UNUkq3nOxb5KnKt9JSKOO4z3uFFERvqejxKRVn6vzRCRzkGOc4WIdAvga/4kIrWP9NwmNlhSMOHogKp283tsDtF5h6tqV5xiiU9W94tVdbqqvubbHAW08nvtelVdF5Qof4tzGoHF+SfAkoIJiCUFExF8LYLPRORr36NXBcecKCJf+loX34pIR9/+q/32vyQi8VWc7lPgWN/X9vHV6V/jq3Nf07d/kvy2PsVTvn0Pi8hdInIZTn2pWb5z1vJ9wk8RkXEi8oRfzKNE5PnDjHMpfoXQRORFEVkpzjoKE337bsVJTotEZJFvX38RWer7Pv5dROpWcR4TQywpmHBUy6/r6B3fvh1AP1XtDlwBTKng624EJqtqN5w/ypm+sgdXAGf69pcAw6s4/2BgjYgkAa8AV6jqSTgVAMaJSGPgD8CJqnoy8Kj/F6vqHGAlzif6bqp6wO/lOcAlfttXALMPM84BOGUtDrpfVVOAk4FzRORkVZ2CUxfnPFU9z1f64gGgr+97uRK4o4rzmBgSdWUuTFQ44PvD6C8RmOrrQy/BqelT3lLgfhFpA8xV1R9FpA9wKrDCV96jFk6CqcgsETkAbMYpv3wcsElV/+N7/VXgZmAqzvoMM0Tkn0DApblVdaeIZPhq1vzoO8cS3/tWJ846OGUf/FfdulxExuL8XrfEWXDm23Jfe7pv/xLfeWrgfN+MASwpmMhxO/Ar0BWnhfu7RXNU9Q0RWQ4MBBaIyPU4ZYZfVdX7AjjHcP+CeSJS4Robvno8PXGKsA0DxgPnV+NaZgOXA98D76iqivMXOuA4cVYgmwS8AFwiIh2Au4AeqpolIq/gFIYrT4CFqnplNeI1McS6j0ykaAD84quRPwLnU/J/EZGjgQxfl0k6TjfKR8BlItLcd0xjCXx96u+BZBE51rc9AvjE1wffQFXn4QziVnQH0D6c8t0VmQtcjLMOwGzfvmrFqapFON1Ap/u6nuoDuUCOiBwFXHiIWJYBZx68JhGpLSIVtbpMjLKkYCLFNOAaEVmG03WUW8ExVwDficg3wPE4Sxauw/nj+aGIfAssxOlaqZKq5uNUoPy7iKwBSoHpOH9gP/C93yc4rZjyXgGmHxxoLve+WcA6oL2qfunbV+04fWMVTwN3qepqnLWZ1wIzcbqkDkoF5ovIIlXdiXNn1Ju+8yzD+V4ZA1iVVGOMMX6spWCMMaaMJQVjjDFlLCkYY4wpY0nBGGNMGUsKxhhjylhSMMYYU8aSgjHGmDKWFIwxxpT5f6ECaWxsd/bDAAAAAElFTkSuQmCC\n",
        "text/plain": "<Figure size 432x288 with 1 Axes>"
       },
       "metadata": {
        "needs_background": "light"
       },
       "output_type": "display_data"
      }
     ],
     "persistentId": "2c810e70-a822-4d9a-a9b0-2622dfa247bf",
     "text": "y_true = coding_scores_not_low_complexity_unspliced['in_cds_no_stop_codon']\ny_pred = coding_scores_not_low_complexity_unspliced['jaccard_in_peptide_db']\n\nfpr, tpr, thresholds = metrics.roc_curve(y_true, y_pred)\n\nfig, ax = plt.subplots()\nlw = 2\nax.plot(fpr, tpr, lw=2, color='darkorange')\nax.plot([0, 1], [0, 1], color='navy', lw=lw, linestyle='--')\nax.set(xlabel='False Positive Rate', ylabel='True Positive Rate')\n\nprint(metrics.accuracy_score(y_true, y_pred > 0.9))\n\nprint(metrics.classification_report(y_true, y_pred > 0.9))"
    },
    "executionTime": "2019-10-12T03:42:32.285Z"
   },
   {
    "cell": {
     "executionCount": 67,
     "executionEventId": "4f18a794-df05-42b7-aedb-ff74cd427f62",
     "hasError": false,
     "id": "c4e90627-a46a-4fa2-b552-6402975ee8c7",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "0.9204152249134948\n              precision    recall  f1-score   support\n\n       False       1.00      0.91      0.95       767\n        True       0.59      0.98      0.74       100\n\n   micro avg       0.92      0.92      0.92       867\n   macro avg       0.80      0.95      0.85       867\nweighted avg       0.95      0.92      0.93       867\n\n"
      },
      {
       "data": {
        "image/png": "iVBORw0KGgoAAAANSUhEUgAAAYUAAAEKCAYAAAD9xUlFAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDMuMC4zLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvnQurowAAIABJREFUeJzt3Xl8FFW2wPHfyQJh35GdoIKKKIgBFRUXFlE2Rx1FEUQRFEUdt1GfC+LzObgLImIkjMug4jCo0YFBRsEFAUEREdARAkJQZEsCJGQ/749qYk8MSQe6uno538+nP3RVV7pOJSSn7711zxVVxRhjjAGI8zoAY4wx4cOSgjHGmDKWFIwxxpSxpGCMMaaMJQVjjDFlLCkYY4wpY0nBGGNMGUsKxhhjylhSMMYYUybB6wCqq2nTppqcnOx1GMYYE1G++uqrXararKrjIi4pJCcns3LlSq/DMMaYiCIiPwVynHUfGWOMKWNJwRhjTBlLCsYYY8pYUjDGGFPGkoIxxpgyriUFEZkpIjtE5LtDvC4iMkVENojItyLS3a1YjDHGBMbNlsIrwIBKXr8Q6Oh7jAVedDEWY4wxAXBtnoKqfioiyZUcMhR4TZ31QJeJSEMRaamqv7gVU9jY+xP8+yYo3Od1JMaYCPDlhvokJZZy8tk9odfDrp7Ly8lrrYGtftuZvn2/SwoiMhanNUG7du1CEpxrVGHhDbB5gdeRGGPCnCrc889+PP3JqZzc8le+PO57El0+p5dJQSrYpxUdqKqpQCpASkpKhcdEhB/+Dp/cBfu2QM0GMOjvkFDT66iMMWFKADZugE+30n/oaZT06BHVSSETaOu33Qb42aNYQmPVFCchAJw9CZL7eRuPMSbsZGfnk5GRRffuLQGY+MwZDBuzq2zbbV4mhXRgvIi8BZwG5ETFeMKOb+DjW6Aot4LXVjn/9p0OXW8IbVzGmLD33nvfM27cP4mLE9auvYkGDZKoVSsxZAkBXEwKIvImcC7QVEQygQngtHxUdTowD7gI2ADkAde6FUtIfTcTtn1+6NfjEuDogaGLxxgT9nbsyOXWW+cze/ZaAE4/vQ3Z2fk0aJAU8ljcvPvoyipeV+Bmt87vmeyNzr/nPgNtzvn963VbQZ0WoY3JGBOWVJVZs9Zw223/Ys+eA9Sunchjj53P+PE9iY/3Zm5xxJXODns5Gc6/bc+H5l29jcUYE9bGjfsnL730FQB9+x5NauogOnRo5GlMVuYimLQUcjY5zxse7W0sxpiwd/HFx9OwYRJpaUP48MOrPU8IYC2F4Nr/M5QUQO3mUKOe19EYY8LMjz/u5qOPNnHjjSkADBhwLJs33+bJ2MGhWFIIpoPjCQ2slWCM+U1xcSnPPLOUCRMWU1BQTLduLTj99DYAYZUQwJJCcB1MCg2P8TYOY0zYWL16O6NHp/PVV84d9yNHdqVjx8YeR3VolhSC6eAgs7UUjIl5BQXFPProp0yatITi4lLatWvASy8NYsCAY70OrVKWFILJWgrGGJ/77vuIZ59dBsDNN/fgL3/pQ7164V/WxpJCMJW1FCwpGBPr/vznM1m6NJMnnujL2We39zqcgNktqcFU1lKw7iNjYs3ChRu59NK3KS4uBaBFi7p88cV1EZUQwJJC8BTkQP5uSKgFdUJXp8QY462srAOMHv0e/fv/jblz1/PXv64qe02komLQ4c26j4Il22+QOQL/Ixhjqu+dd9Zz003z2L59PzVrxjNhwjmMGtXN67COiCWFYMmxOQrGxIrt2/dzyy3zmTNnHQC9erUlLW0Ixx/f1OPIjpwlhWA52FKwO4+MiXrvvfc9c+aso06dRCZN6stNN/UgLi46eggsKQSLtRSMiWr5+cUkJTl/MseMOZWMjCzGjetBcnJDjyMLLhtoDhabo2BMVCotVaZO/ZIOHSbz00/ZAMTFCY8/3i/qEgJYUgieHOs+Miba/PDDLnr3/iu33DKf7dv38+ab33kdkuus+ygYSopg7xZAoH6y19EYY45QUVEJTz31BRMnfkJBQQlHHVWHadMGcsklJ3gdmussKQTD3p9AS6BeW0gI/2nsxphD++67HYwc+Q6rVm0H4Npru/H00/1p1KiWx5GFhiWFYLBCeMZEjdJSZc2aHbRv34DU1MH07x9bXcKWFILBBpmNiWhr1+6gc+dmiAgnn3wU7703jN6921O3bg2vQws5G2gOBhtkNiYi7dtXwPjx8+jS5UX+8Y/1ZfsvuqhjTCYEsJZCcNiKa8ZEnAULNjB27Ads2ZJDQkIcmzdnex1SWLCkEAw51n1kTKTYs+cAt9++gNdeWw1A9+4tSUsbQrduLTyOLDxYUjhSqv9dDM8YE7a++WY7Awb8jV9/zaVmzXgmTjyXO+/sRUKC9aQfZEnhSB3YCUX7oWYDSArfdVeNMdCpUxPq1q1Bp05NmDFjCJ06NfE6pLBjSeFIZfuttmYls40JK6rKG2+sYfDg46hfvya1ayeyePEoWrWqFzUF7ILN2kxHKsdWWzMmHG3enM0FF/yNq69+h3vv/XfZ/jZt6ltCqIS1FI5U2Z1HNshsTDgoKSll2rQV3HffR+TmFtG4cS169WrrdVgRw5LCkbI5CsaEjfXrdzJ6dDpLl2YCcPnlJ/L88xfSvHkdjyOLHJYUjpTNUTAmLGzalEW3bi9RWFhCy5Z1mTZtIBdffLzXYUUcV5OCiAwAJgPxwAxVnVTu9XbAq0BD3zH3quo8N2MKOmspGBMWOnRoxB//2JmkpASeeqo/DRsmeR1SRHItKYhIPPAC0A/IBFaISLqqrvM77AHgbVV9UUQ6A/OAZLdiCrqiA7D/Z4hLgHptvI7GmJhy4EARjzzyCX/4wwn07NkagFdfvZj4eLt/5ki4+d3rCWxQ1QxVLQTeAoaWO0aB+r7nDYCfXYwn+A62EuonO4nBGBMSn332E926vcSkSUsYO/Z9SksVwBJCELj5l6w1sNVvOxM4rdwxDwMfisgtQB2gr4vxBJ91HRkTUnv3FnDfff9m2rSVAHTu3Izp0wfZLaZB5GZareinpOW2rwReUdU2wEXA6yLyu5hEZKyIrBSRlTt37nQh1MNkg8zGhMy8eT/Spcs0pk1bSUJCHA891Juvvx7L6adb120wudlSyAT8bw5uw++7h0YDAwBUdamIJAFNgR3+B6lqKpAKkJKSUj6xeMfWUTAmJHJy8hk+fC7Z2fmkpLQiLW0IJ598lNdhRSU3k8IKoKOIdAC2AcOAq8odswXoA7wiIicASUAYNQWqYCuuGeMaVUUV4uKEBg2SmDJlAL/+msuf/nS6FbBzkWtJQVWLRWQ8sADndtOZqrpWRB4BVqpqOnAn8LKI3I7TtTRKVcOnJVAVaykY44qff97HTTf9k7PPbsedd/YCYMSIrh5HFRtcvWXGN+dgXrl9D/k9Xwec6WYMrtFS2LvJeW4tBWOCQlWZOXMVd975ITk5BSxblslNN/WgVq1Er0OLGXYf5eHatw1KCqF2c6hR1+tojIl4GRlZjBnzPh9/7HzYGjiwI9OnD7KEEGKWFA5XjhXCMyYYSkpKmTJlOfff/zEHDhTTtGltpkwZwLBhXRArRx9ylhQOV7bNUTAmWObMWc+BA8VceWUXJk8eQLNmVsDOK5YUDleOzVEw5nAVFpawb18BTZrUJj4+jrS0Ifz4424GDz7O69Bint3XdbispWDMYVmxYhspKamMGPEOB282PP74ppYQwoS1FA6XtRSMqZa8vCImTFjEM88so7RUycsrYseOXI46ym7UCCeWFA6XzVEwJmCLF29mzJj32bBhD3Fxwl13ncHEiedRu7bdWRRuLCkcjvxsyN8DCbWgTguvozEmbKkqt946n6lTVwBw0knNSUsbQo8erT2OzByKJYXD4V/ewm6ZM+aQRIT69WuSmBjHAw/05t57z6JGjXivwzKVsKRwOKxktjGHtGtXHhs37uG005zqpQ8+eA7Dh59M587NPI7MBCKgu49EpIaIHOt2MBHDSmYb8zuqyltvfccJJ7zAxRfPJivrAABJSQmWECJIlUlBRAYCa4CFvu1uIvKO24GFNRtkNua/ZGbuZejQt7jyyn+wa1cenTs3Iy+vyOuwzGEIpPvoEZwV0xYBqOo3Md9qsO4jYwAoLVVmzPiau+9eyN69BdSvX5Onn+7P6NGnWImKCBVIUihS1exyP+DIKW/tBus+MgaA0aPTeeWVbwAYMuQ4pk27iNat61fxVSacBTKmsF5ELgfiRKSDiDwHLHM5rvBVUgj7tgAC9ZO9jsYYT1199Uk0b16Ht966lHffvcISQhQIJCmMB04FSoG5QD5wm5tBhbW9W5y1FOq1hYSaXkdjTEh9990OJk/+7TNhnz5Hk5FxK1dcYRVNo0Ug3UcXqOo9wD0Hd4jIJTgJIvYcLG/R0LqOTOwoKCjmL3/5nMce+4yiolJSUlpx5pntAKhTp4bH0ZlgCqSl8EAF++4PdiAR42AhPFtHwcSI5cszOfXUVCZO/ISiolLGjUvhpJOO8jos45JDthRE5AJgANBaRJ7xe6k+TldSbMq2loKJDbm5hTz44CKee24ZqtCxY2NmzBhC797tvQ7NuKiy7qMdwHc4Ywhr/fbvA+51M6iwZiuumRhx//0fM3nycuLihLvvPoOHHz7XlsaMAYdMCqq6ClglIrNUNT+EMYU3m6NgYsT995/NmjU7ePzxvqSktPI6HBMigYwptBaRt0TkWxH5z8GH65GFI1Wbo2CiVnr6D1x00SyKikoAaNasDh99NNISQowJJCm8AvwVEOBC4G3gLRdjCl8HdkJRLtRsCLUaex2NMUGxY0cuw4bNYejQt5g/fwOvvrra65CMhwJJCrVVdQGAqm5U1QeA89wNK0xZK8FEEVXlb3/7lhNOeIHZs9dSu3YikycP4Npru3kdmvFQIPMUCsSZlbJRRG4EtgHN3Q0rTFkhPBMltmzJ4cYbP2D+/A0A9O17NKmpg+jQoZHHkRmvBZIUbgfqArcC/wc0AK5zM6iwZYPMJkp8+OFG5s/fQMOGSTzzTH9GjepmM5INEEBSUNXlvqf7gBEAItLGzaDClnUfmQiWm1tYNvt49OhT2LZtL2PHnkrLlvU8jsyEk0rHFESkh4hcLCJNfdsnishrxGpBPGspmAhUXFzKE08soX3758jIyAKcZTInTDjXEoL5nUMmBRH5CzALGA78S0Tux1lTYTXQKTThhRkbUzARZvXq7Zx22gzuueff7N59gHff/d7rkEyYq6z7aCjQVVUPiEhj4Gff9g+hCS3MFOVB7i8Qlwh1Y7P3zESOgoJiHn30UyZNWkJxcSnt2jUgNXUQF1wQ2+tjmapVlhTyVfUAgKruEZHvYzYhAORscv5tkAxx8Z6GYkxlVq36heHD57J+/S5EYPz4Hjz2WB/q1bNS76ZqlSWFo0XkYHlsAZL9tlHVS6p6cxEZAEwG4oEZqjqpgmMuBx7GWc1ttapeFXj4IWSDzCZC1KyZwMaNWRx3XBNmzBjCWWe18zokE0EqSwqXltueWp03FpF44AWgH5AJrBCRdFVd53dMR+A+4ExVzRKR8J3/kGMls034+vrrXzjllBaICJ07N2P+/OH06tWWpKRA7jo35jeVFcT76AjfuyewQVUzAETkLZxxinV+x4wBXlDVLN85dxzhOd1jg8wmDGVlHeCuuz5k5sxvePPNSxk2rAsA55/fwePITKRy82NEa2Cr33YmcFq5YzoBiMgSnC6mh1X1X+XfSETGAmMB2rXzqCmcY91HJry88856brppHtu376dmzXh2787zOiQTBdxMChVNj9QKzt8ROBdoA3wmIl1UNfu/vkg1FUgFSElJKf8eoZFtcxRMeNi+fT+33DKfOXOcRveZZ7ZlxowhHH98U48jM9Eg4KQgIjVVtaAa750JtPXbboNzW2v5Y5apahGwSUR+wEkSK6pxHveVlsDeg3cfWbPceOerr36mX7/XycrKp06dRCZN6stNN/UgLs5KVJjgqLJKqoj0FJE1wI++7a4i8nwA770C6CgiHUSkBjAMSC93zLv4Kq76Zk13AjKqEX9o7N8GJYVQ+yioUdfraEwM69y5Gc2a1eGCC45h7dqbGD++pyUEE1SBlM6eAgwCdgOo6moCKJ2tqsXAeGABsB54W1XXisgjIjLEd9gCYLeIrMOZLX23qu6u/mW4zMpbGI+UliqpqV+Rne0sflirViKffjqK+fOH0759Q4+jM9EokO6jOFX9qVwFxZJA3lxV5wHzyu17yO+5Anf4HuHL5igYD/zwwy6uv/59Pv98CytWbOPll53PUkcdZa1V455AksJWEekJqG/uwS1AbC3HaS0FE0JFRSU8/fRSHn54MQUFJbRoUZcLL+zodVgmRgSSFMbhdCG1A34F/u3bFztsjoIJkVWrfmH06HRWrdoOwLXXduPpp/vTqFEtjyMzsSKQpFCsqsNcjyScWfeRCYGNG/fQs+cMiotLSU5uSGrqIPr1sw8iJrQCSQorfLeKzgbmquo+l2MKP9Z9ZELgmGMaM2LEydSrV4P/+78+1K1bw+uQTAyq8u4jVT0GeBQ4FVgjIu+KSOy0HPKzIX8PJNR2bkk1Jkj27y/k1lvns3TpbxP/09KGMHnyhZYQjGcCuSUVVf1CVW8FugN7cRbfiQ1lrYSjwdawNUGyYMEGTjxxGs8//yU33vhPnBvxsHWSjeeq7D4Skbo4heyGAScA7wG9XI4rfJSNJ1jXkTlye/Yc4PbbF/Daa6sBOPXUlqSlDbFkYMJGIGMK3wHvA0+o6mcuxxN+yu48skFmc2TmzFnHzTfPY8eOXJKSEpg48VzuuOMMEhICarAbExKBJIWjVbXU9UjCla2jYIIgOzufsWPfJysrn9692/Pyy4Pp1KmJ12EZ8zuHTAoi8rSq3gn8Q0R+V5k0kJXXokKOtRTM4VFVSkuV+Pg4GjZMYtq0gWRlHeCGG1KsXpEJW5W1FGb7/q3WimtRJ9taCqb6Nm/OZuzY9zn//A7ce+9ZAGUL4BgTzg7ZmamqX/qenqCqH/k/cAaco19JIezbAgg0SPY6GhMBSkpKmTJlOV26TGPhwgymTv2S/Pxir8MyJmCBjHBdV8G+0cEOJCzt/Qm0FOq1hXi7b9xUbv36nfTu/Qq33fYvcnOLGDasC19/fYOtk2wiSmVjClfg3IbaQUTm+r1UD8iu+KuijM1kNgEoLi7l8cc/55FHPqWwsIRWrerx4osDGTLkOK9DM6baKvsI8yXOGgptgBf89u8DVrkZVNiwQngmAHFxwocfZlBYWMKYMd154ol+NGyY5HVYxhyWQyYFVd0EbMKpihqbrBCeOYQDB4rYt6+Q5s3rEBcnzJgxmK1b93L++bZcq4lshxxTEJFPfP9micgev0eWiOwJXYgesu4jU4FPP/2Jrl2nc/XVc8vKU3Ts2MQSgokKlXUfHVxys2koAglL1lIwfvbuLeC++/7NtGkrAUhMjGfXrjyaNavjcWTGBE9lt6QenMXcFohX1RLgDOAGIPp/C1StpWDKzJ//I126TGPatJUkJMQxYcI5fP31WEsIJuoEcq/cu0APETkGeA34J/AGMMjNwDyXtwOKciGpkfMwMUlVGTPmfdLSnHsrUlJaMXPmEE46ycqom+gUyDyFUlUtAi4BnlPVW4DW7oYVBqzryOCUsm7Tpj5JSQk89VQ/li4dbQnBRLWAluMUkT8CI4CLffsS3QspTFghvJj188/72LhxD2ef3R6A//mfsxkx4mSOOaaxx5EZ475AZzSfh1M6O0NEOgBvuhtWGLCS2TFHVUlL+5rOnV/g0kvfZvfuPABq1Ii3hGBiRpUtBVX9TkRuBY4VkeOBDar6f+6H5jFrKcSUjIwsxox5n48/3gTAoEGdKCqK3YrxJnYFsvLa2cDrwDZAgBYiMkJVl7gdnKdsNnNMOFjA7oEHFpGXV0TTprWZMmUAw4Z1sdXQTEwKZEzhWeAiVV0HICIn4CSJFDcD85ytoxATRo58lzfeWAPAVVedxHPPXWC3mZqYFsiYQo2DCQFAVdcD0V0ytCgPcrdDXCLUbeN1NMZFY8Z0p02b+qSnD2PWrEssIZiYF0hL4WsReQmndQAwnGgviFc2ntAB4uK9jcUE1YoV2/j4403cc4+z8M255yazYcMt1Kxp5a2NgcCSwo3ArcCfccYUPgWedzMoz5WttmZdR9EiL6+ICRMW8cwzyygtVXr1alt2y6klBGN+U+lvg4icBBwDvKOqT4QmpDCQY4PM0WTx4s1cf306GzdmERcn3HXXGZx6aiuvwzImLFVWJfV/cEpcDAcWikhFK7BVSkQGiMgPIrJBRO6t5LjLRERFJDwGr202c1TIycnnhhve57zzXmXjxixOOqk5y5aN5skn+1O7dvTPvzTmcFTWUhgOnKyquSLSDJgHzAz0jUUkHmdxnn5AJrBCRNL9B619x9XD6Z5aXt3gXWOF8KLCgw8uIjX1axIT43jwwd7cc89Z1KhhY0TGVKaypFCgqrkAqrpTRAK5U8lfT5yJbhkAIvIWMBRYV+64/wWeAO6q5vu7x+YoRCxVLZtf8NBD57BpUzaTJvXhxBObexyZMZGhsj/0R4vIXN/jHeAYv+25lXzdQa2BrX7bmZQrpCcipwBtVfWDakfultISyHFmtdLAFk2JFKrKG2+s4fzzX6OwsASApk1r8/77V1pCMKYaKmspXFpue2o137ui6aBa9qLT8ngWGFXlG4mMBcYCtGvXrpphVNP+bVBaBHVaQKLdsx4JMjP3Mm7cP/ngg/8AMGvWt1x77SkeR2VMZKpsjeaPjvC9M3EW6DmoDfCz33Y9oAuw2NfcbwGki8gQVV1ZLpZUIBUgJSVFcZMNMkeM0lLl5Ze/4u67F7JvXyENGtTk6af7M2pUN69DMyZiuXmD9gqgo6+q6jZgGHDVwRdVNQe/pT5FZDFwV/mEEHI2yBwRNmzYw5gx77N48WYAhg49jmnTBtKqVT1vAzMmwrmWFFS1WETGAwuAeGCmqq4VkUeAlaqa7ta5j0hZS8GSQjj77LOfWLx4M82b12Hq1Au57LLOVsDOmCAIOCmISE1VLajOm6vqPJxbWf33PXSIY8+tznu7xtZRCFvZ2fk0bJgEwKhR3di5M4/Ro0+hSZPaHkdmTPSo8jZTEekpImuAH33bXUUkestc2DoKYaegoJgJExbRvv1z/PjjbsBZJvPPfz7TEoIxQRZIS2EKMAhndjOqulpEznM1Ki9ZiYuwsmxZJqNHp7Nu3U4AFizYSMeOTTyOypjoFUhSiFPVn8r115a4FI+38rOcR2IdqG33tnspN7eQBx9cxHPPLUMVOnZsTFrakLIidsYYdwSSFLaKSE9AfaUrbgH+425YHsnxq45qg5aeWb48k6uumktGRhbx8cJdd/ViwoRzqFXL6hUZ47ZAksI4nC6kdsCvwL99+6KPzVEICw0bJrFt2166dj2KtLQhVtHUmBCqMimo6g6cOQbRL9vmKHjl88+3cOaZbRERjjuuKR9/fA09erQiMdEK2BkTSlUmBRF5Gb/yFAep6lhXIvKSDTKH3I4dudx663xmz17Lq69ezMiRXQHo1attFV9pjHFDIN1H//Z7ngT8gf8udBc9cmzFtVBRVWbNWsNtt/2LPXsOULt2YlkhO2OMdwLpPprtvy0irwMLXYvIS1YyOyS2bMnhxhs/YP78DQD063c0qamDSU5u6HFkxpjDKXPRAYi++wJLCmHfVpA4qB99lxculi/PpG/f19m/v5CGDZN49tkLuOaarlaiwpgwEciYQha/jSnEAXuAQy6tGbH2/gRa6iSE+BpeRxO1unVrQdu29Tn++Ka88MJFtGxpBeyMCSeVJgVxPr51xalyClCqqu6WrvaKdR25ori4lKlTv2TkyK40blyLmjUTWLLkOho1quV1aMaYClRa+8iXAN5R1RLfIzoTAtgcBResXr2d006bwe23L+COOxaU7beEYEz4CmTd5S9FpLvrkXjNCuEFTX5+MQ888DEpKS/z9de/0K5dA668sovXYRljAnDI7iMRSVDVYuAsYIyIbARycZbZVFWNrkRh3UdB8cUXWxk9Op3vv9+FCIwf34PHHutDvXo1vQ7NGBOAysYUvgS6AxeHKBZvla24Zt1Hh2vDhj2cffZfKS1VjjuuCWlpQzjzTJfX1DbGBFVlSUEAVHVjiGLxjqp1HwXBscc2ZuzY7jRuXIsHHzyHpCQ3V3s1xrihst/aZiJyx6FeVNVnXIjHG3m/QlEuJDWCJJtAFaisrAPceeeHXHttt7KS1tOmDbQ5B8ZEsMqSQjxQF1+LIaplWyuhuubOXc/NN89j+/b9fPXVL3zzzQ2IiCUEYyJcZUnhF1V9JGSReMkK4QVs+/b9jB8/j3/8Yz0AZ53VjhkzBlsyMCZKVDmmEBOyrRBeVVSV115bze23LyArK5+6dWvw+ON9ufHGFOLiYue/ijHRrrKk0CdkUXjNWgpVys7O5847PyQrK58BA45l+vSBtG9v4y/GRJtDJgVV3RPKQDxls5krVFqqlJYqCQlxNGpUi5deGkReXhFXX32ydRcZE6UCmdEc/XJsxbXyvv9+F717/5VJkz4v23fppZ0ZMcIqmhoTzSwpFOVC7nanMmrd1l5H47miohIee+wzunadzpIlW0lLW0V+frHXYRljQsRmF+Vscv6tnwxxsb0e8KpVv3Dddel88812AEaPPoUnn+xnk9CMiSH22241jygqKmHChMU88cQSSkqU5OSGvPzyYPr2tTEWY2KNJYWyQebYTQoJCXEsX76N0lLltttO49FHz6duXVtoyJhYZEkhRgvh7dtXwL59hbRqVQ8RYcaMwWzfvp8zzmjrdWjGGA/ZQHMMthQWLNhAly4vMnz4XA6um9ShQyNLCMYYSwqx1FLYvTuPa655lwEDZrFlSw779hWwe/cBr8MyxoQRV5OCiAwQkR9EZIOI3FvB63eIyDoR+VZEPhKR9m7G8zulJb/dfRTFE9dUlTlz1tG58zRee201SUkJPPFEX5Ytu56mTWt7HZ4xJoy4NqYgIvHAC0A/IBNYISLpqrrO77BVQIqq5onIOOAJ4Aq3Yvqd/ZlQWgR1WkJidP5xVFWGD5/Lm29+B0DRZJGNAAAPYElEQVTv3u15+eXBdOrUxOPIjDHhyM2WQk9gg6pmqGoh8BYw1P8AVV2kqnm+zWVAGxfj+b0YKIQnInTu3Ix69Wrw4osDWbToGksIxphDcvPuo9bAVr/tTOC0So4fDcyv6AURGQuMBWjXLojLO0bpHIVNm7LIyMiiTx8n2d1zz5mMGtWNNm3qexyZMSbcudlSqKhAjlZ4oMjVQArwZEWvq2qqqqaoakqzZs2CF2FOdBXCKykpZfLkZXTp8iJXXDGHHTtyAUhMjLeEYIwJiJsthUzA/x7HNsDP5Q8Skb7A/cA5qlrgYjy/lx09hfDWrdvJ9dens3RpJgBDhhxn6xwYY6rNzaSwAugoIh2AbcAw4Cr/A0TkFOAlYICq7nAxlopFwToKRUUlPP74Ev73fz+lsLCEVq3q8eKLAxky5DivQzPGRCDXkoKqFovIeGABznrPM1V1rYg8AqxU1XSc7qK6wN995Zi3qOoQt2L6nZzIH2i+6qq5zJnj3NA1Zkx3nnyyHw0aJHkclTEmUrla5kJV5wHzyu17yO95XzfPX6n8LOeRWAdqN/csjCN1222n8c0323nppUGcf34Hr8MxxkS42J3R7L/aWgQtGvPJJ5uZOHFx2fZZZ7Vj/fqbLSEYY4IidgviRdhqa3v3FnDPPQuZPv0rAM47rwO9ezsTwBMSYje3G2OCK3aTQgQVwps370duuOEDMjP3kpgYx/33n83pp4d2np8xJjbEblKIgEJ4u3bl8ac//YtZs9YA0LNna9LShtClS+SOgRhjwlvsJoUImM38yCOfMGvWGmrVSuDRR8/ntttOIz7euoqMMe6xpBBm3Ueqiu/2XCZOPJdff83lscfO55hjGnscmTEmFsTmx86SQti3FSQO6gexltIRUFVefvkrevWaSX5+MQCNGtVi9uzLLCEYY0ImNpNCzmZAoV47iPd+LeKNG/fQp89rjB37AcuWZfL222u9DskYE6Nis/soTAaZnQJ2y3nggY85cKCYZs1q8/zzF3L55Sd6GpcxJnbFZlIIg/GEtWt3cN116Xz55TYAhg8/ieeeG2AroRljPBWbSSEMCuGtWrWdL7/cRuvW9XjppUEMHNjJs1iMMeag2EwKHq24tnNnLs2a1QGclkF2dj4jRpxsBeyMMWEjRgeaQ9tSyMsr4q67PiQ5eTLr1+8EnGUyx4/vaQnBGBNWYq+loBrSlsKiRZsYM+Z9Nm7MIi5O+PTTnzjhhCCuHmeMMUEUe0kh71cozoOkxpDU0LXT5OTk8+c/LyQ19WsATjqpOTNnDiUlpZVr5zTGmCMVe0khBOUtPv98C8OGzWHbtn0kJsbx4IO9ueees6hRI961cxpjTDDEXlIIwWprLVrUZffuA5x+ehtmzBjMiSdaATtjTGSIvaTgQktBVVm4MIN+/Y5GRDj22MZ8/vm1dOvWwgrYGWMiSuz9xQryxLWtW3MYPPhNLrjgb/z1r9+U7T/11FaWEIwxESf2WgpBKnFRWuoUsLv77oXs21dIgwY1qVnTxgyMMZEt9pJCEFoKP/64mzFj3ueTT34C4OKLj+eFFy6iVat6wYjQGGM8E1tJoSjXuSU1vgbUPbxbQ7/4Yit9+rxGfn4xzZvXYerUC7nsss5layAYY0wki62kcHDSWv0OEHd4XT0pKa3o2LExp5zSkmee6U+TJlbAzhgTPWIsKVT/zqOCgmKeeuoLbrghhaZNa1OjRjxLllxHvXo1XQrSGGO8E1tJoZpzFJYty2T06HTWrdvJ+vW7+NvfLgGwhGCMiVqxlRQCbCnk5hbywAMfM3nyclShU6cm3HDDqSEI0BhjvBVbSSGAlsJHH2UwZsz7bNqUTXy8cPfdvZgw4VySkmLrW2WMiU2x9ZeuipLZ//nPbvr1ex1V6NatBWlpQ+jevWUIAzTGGG/FTlIoLYGczc7zBh0qPKRTpybcdttpNGtWh7vv7kViok1GM8bElthJCvszobQI6rSEROc20l9/3c+tt/6LG288lfPOcxLFs88O8DJKY4zxlKvFeURkgIj8ICIbROTeCl6vKSKzfa8vF5Fk14LxG2RWVV5/fTWdO0/j7bfXcvvtC1BV105tjDGRwrWkICLxwAvAhUBn4EoR6VzusNFAlqoeCzwLPO5WPAeTwpbCTgwc+AYjR77Lnj0H6N//GN59d5jNSDbGGNztPuoJbFDVDAAReQsYCqzzO2Yo8LDv+RxgqoiIuvCxvTQrg+lLenDPgnbsz9tAo0ZJPPvsBYwc2dUSgjHG+LiZFFoDW/22M4HTDnWMqhaLSA7QBNgV7GByft7ExIXnsD8PLr30BKZOvYgWLeoG+zTGGBPR3EwKFX38Lt8CCOQYRGQsMBagXbt2hxVMo3rCjCs/pLDHRC4dc/FhvYcxxkQ7N5NCJtDWb7sN8PMhjskUkQSgAbCn/BupaiqQCpCSknJ4XUuD3mTwQKWCnGOMMcbHzbuPVgAdRaSDiNQAhgHp5Y5JB67xPb8M+NiN8YQyIiC2GpoxxhyKay0F3xjBeGABEA/MVNW1IvIIsFJV04E04HUR2YDTQhjmVjzGGGOq5urkNVWdB8wrt+8hv+f5wB/djMEYY0zgrC/FGGNMGUsKxhhjylhSMMYYU8aSgjHGmDKWFIwxxpSRSKsOKiI7gZ8O88ub4kIJjTBn1xwb7Jpjw5Fcc3tVbVbVQRGXFI6EiKxU1RSv4wglu+bYYNccG0JxzdZ9ZIwxpowlBWOMMWViLSmkeh2AB+yaY4Ndc2xw/ZpjakzBGGNM5WKtpWCMMaYSUZkURGSAiPwgIhtE5N4KXq8pIrN9ry8XkeTQRxlcAVzzHSKyTkS+FZGPRKS9F3EGU1XX7HfcZSKiIhLxd6oEcs0icrnvZ71WRN4IdYzBFsD/7XYiskhEVvn+f1/kRZzBIiIzRWSHiHx3iNdFRKb4vh/fikj3oAagqlH1wCnTvRE4GqgBrAY6lzvmJmC67/kwYLbXcYfgms8Davuej4uFa/YdVw/4FFgGpHgddwh+zh2BVUAj33Zzr+MOwTWnAuN8zzsDm72O+wivuTfQHfjuEK9fBMzHWbnydGB5MM8fjS2FnsAGVc1Q1ULgLWBouWOGAq/6ns8B+ohIRUuDRooqr1lVF6lqnm9zGc5KeJEskJ8zwP8CTwD5oQzOJYFc8xjgBVXNAlDVHSGOMdgCuWYF6vueN+D3KzxGFFX9lApWoPQzFHhNHcuAhiLSMljnj8ak0BrY6red6dtX4TGqWgzkAE1CEp07Arlmf6NxPmlEsiqvWUROAdqq6gehDMxFgfycOwGdRGSJiCwTkQEhi84dgVzzw8DVIpKJs37LLaEJzTPV/X2vFlcX2fFIRZ/4y99iFcgxkSTg6xGRq4EU4BxXI3JfpdcsInHAs8CoUAUUAoH8nBNwupDOxWkNfiYiXVQ12+XY3BLINV8JvKKqT4vIGTirOXZR1VL3w/OEq3+/orGlkAm09dtuw++bk2XHiEgCTpOzsuZauAvkmhGRvsD9wBBVLQhRbG6p6prrAV2AxSKyGafvNT3CB5sD/b/9nqoWqeom4AecJBGpArnm0cDbAKq6FEjCqREUrQL6fT9c0ZgUVgAdRaSDiNTAGUhOL3dMOnCN7/llwMfqG8GJUFVes68r5SWchBDp/cxQxTWrao6qNlXVZFVNxhlHGaKqK70JNygC+b/9Ls5NBYhIU5zupIyQRhlcgVzzFqAPgIicgJMUdoY0ytBKB0b67kI6HchR1V+C9eZR132kqsUiMh5YgHPnwkxVXSsijwArVTUdSMNpYm7AaSEM8y7iIxfgNT8J1AX+7htT36KqQzwL+ggFeM1RJcBrXgD0F5F1QAlwt6ru9i7qIxPgNd8JvCwit+N0o4yK5A95IvImTvdfU984yQQgEUBVp+OMm1wEbADygGuDev4I/t4ZY4wJsmjsPjLGGHOYLCkYY4wpY0nBGGNMGUsKxhhjylhSMMYYU8aSggk7IlIiIt/4PZIrOTb5UNUkq3nOxb5KnKt9JSKOO4z3uFFERvqejxKRVn6vzRCRzkGOc4WIdAvga/4kIrWP9NwmNlhSMOHogKp283tsDtF5h6tqV5xiiU9W94tVdbqqvubbHAW08nvtelVdF5Qof4tzGoHF+SfAkoIJiCUFExF8LYLPRORr36NXBcecKCJf+loX34pIR9/+q/32vyQi8VWc7lPgWN/X9vHV6V/jq3Nf07d/kvy2PsVTvn0Pi8hdInIZTn2pWb5z1vJ9wk8RkXEi8oRfzKNE5PnDjHMpfoXQRORFEVkpzjoKE337bsVJTotEZJFvX38RWer7Pv5dROpWcR4TQywpmHBUy6/r6B3fvh1AP1XtDlwBTKng624EJqtqN5w/ypm+sgdXAGf69pcAw6s4/2BgjYgkAa8AV6jqSTgVAMaJSGPgD8CJqnoy8Kj/F6vqHGAlzif6bqp6wO/lOcAlfttXALMPM84BOGUtDrpfVVOAk4FzRORkVZ2CUxfnPFU9z1f64gGgr+97uRK4o4rzmBgSdWUuTFQ44PvD6C8RmOrrQy/BqelT3lLgfhFpA8xV1R9FpA9wKrDCV96jFk6CqcgsETkAbMYpv3wcsElV/+N7/VXgZmAqzvoMM0Tkn0DApblVdaeIZPhq1vzoO8cS3/tWJ846OGUf/FfdulxExuL8XrfEWXDm23Jfe7pv/xLfeWrgfN+MASwpmMhxO/Ar0BWnhfu7RXNU9Q0RWQ4MBBaIyPU4ZYZfVdX7AjjHcP+CeSJS4Robvno8PXGKsA0DxgPnV+NaZgOXA98D76iqivMXOuA4cVYgmwS8AFwiIh2Au4AeqpolIq/gFIYrT4CFqnplNeI1McS6j0ykaAD84quRPwLnU/J/EZGjgQxfl0k6TjfKR8BlItLcd0xjCXx96u+BZBE51rc9AvjE1wffQFXn4QziVnQH0D6c8t0VmQtcjLMOwGzfvmrFqapFON1Ap/u6nuoDuUCOiBwFXHiIWJYBZx68JhGpLSIVtbpMjLKkYCLFNOAaEVmG03WUW8ExVwDficg3wPE4Sxauw/nj+aGIfAssxOlaqZKq5uNUoPy7iKwBSoHpOH9gP/C93yc4rZjyXgGmHxxoLve+WcA6oL2qfunbV+04fWMVTwN3qepqnLWZ1wIzcbqkDkoF5ovIIlXdiXNn1Ju+8yzD+V4ZA1iVVGOMMX6spWCMMaaMJQVjjDFlLCkYY4wpY0nBGGNMGUsKxhhjylhSMMYYU8aSgjHGmDKWFIwxxpT5f6ECaWxsd/bDAAAAAElFTkSuQmCC\n",
        "text/plain": "<Figure size 432x288 with 1 Axes>"
       },
       "metadata": {
        "needs_background": "light"
       },
       "output_type": "display_data"
      }
     ],
     "persistentId": "2c810e70-a822-4d9a-a9b0-2622dfa247bf",
     "text": "y_true = coding_scores_not_low_complexity_unspliced['in_cds_no_stop_codon']\ny_score = coding_scores_not_low_complexity_unspliced['jaccard_in_peptide_db']\ny_pred = y_score > 0.9\n\nfpr, tpr, thresholds = metrics.roc_curve(y_true, y_score)\n\nfig, ax = plt.subplots()\nlw = 2\nax.plot(fpr, tpr, lw=2, color='darkorange')\nax.plot([0, 1], [0, 1], color='navy', lw=lw, linestyle='--')\nax.set(xlabel='False Positive Rate', ylabel='True Positive Rate')\n\nprint(metrics.accuracy_score(y_true, y_pred))\n\nprint(metrics.classification_report(y_true, y_pred))"
    },
    "executionTime": "2019-10-12T03:42:57.649Z"
   },
   {
    "cell": {
     "executionCount": 68,
     "executionEventId": "866b829f-038c-484c-97c1-3c4ecf36dd20",
     "hasError": false,
     "id": "91b1f651-6b20-4721-8c15-61a5574e1724",
     "outputs": [
      {
       "data": {
        "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>read_id</th>\n      <th>jaccard_in_peptide_db</th>\n      <th>n_kmers</th>\n      <th>classification</th>\n      <th>short_read_id</th>\n      <th>in_cds_no_stop_codon</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>20</th>\n      <td>SRR306838.7944356 Ibis_Run100924_C3PO:6:38:5591:5659/1</td>\n      <td>1.0</td>\n      <td>7</td>\n      <td>coding</td>\n      <td>SRR306838.7944356</td>\n      <td>False</td>\n    </tr>\n    <tr>\n      <th>64</th>\n      <td>SRR306838.6939297 Ibis_Run100924_C3PO:6:33:5394:5316/1</td>\n      <td>1.0</td>\n      <td>15</td>\n      <td>coding</td>\n      <td>SRR306838.6939297</td>\n      <td>False</td>\n    </tr>\n    <tr>\n      <th>66</th>\n      <td>SRR306838.23317726 Ibis_Run100924_C3PO:6:115:10861:16547/1</td>\n      <td>1.0</td>\n      <td>17</td>\n      <td>coding</td>\n      <td>SRR306838.23317726</td>\n      <td>False</td>\n    </tr>\n    <tr>\n      <th>82</th>\n      <td>SRR306838.18085420 Ibis_Run100924_C3PO:6:91:3302:10193/1</td>\n      <td>1.0</td>\n      <td>16</td>\n      <td>coding</td>\n      <td>SRR306838.18085420</td>\n      <td>False</td>\n    </tr>\n    <tr>\n      <th>106</th>\n      <td>SRR306838.16938253 Ibis_Run100924_C3PO:6:85:7026:15703/1</td>\n      <td>1.0</td>\n      <td>17</td>\n      <td>coding</td>\n      <td>SRR306838.16938253</td>\n      <td>False</td>\n    </tr>\n  </tbody>\n</table>\n</div>",
        "text/plain": "                                                        read_id  \\\n20       SRR306838.7944356 Ibis_Run100924_C3PO:6:38:5591:5659/1   \n64       SRR306838.6939297 Ibis_Run100924_C3PO:6:33:5394:5316/1   \n66   SRR306838.23317726 Ibis_Run100924_C3PO:6:115:10861:16547/1   \n82     SRR306838.18085420 Ibis_Run100924_C3PO:6:91:3302:10193/1   \n106    SRR306838.16938253 Ibis_Run100924_C3PO:6:85:7026:15703/1   \n\n     jaccard_in_peptide_db  n_kmers classification       short_read_id  \\\n20                     1.0        7         coding   SRR306838.7944356   \n64                     1.0       15         coding   SRR306838.6939297   \n66                     1.0       17         coding  SRR306838.23317726   \n82                     1.0       16         coding  SRR306838.18085420   \n106                    1.0       17         coding  SRR306838.16938253   \n\n     in_cds_no_stop_codon  \n20                  False  \n64                  False  \n66                  False  \n82                  False  \n106                 False  "
       },
       "execution_count": 68,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "14804ddd-b8c7-43ae-96da-72d23523c538",
     "text": "coding_scores_not_low_complexity_unspliced.query(\"jaccard_in_peptide_db > 0.9 & in_cds_no_stop_codon == False\").head()"
    },
    "executionTime": "2019-10-12T03:45:53.018Z"
   },
   {
    "cell": {
     "executionCount": 70,
     "executionEventId": "6eff2520-3253-42fb-98f5-fffa2b234027",
     "hasError": false,
     "id": "8aa6e1fa-8df6-4d7c-8ec2-497200eead99",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "SRR306838.7944356\t16\tchr19\t6501568\t255\t42M\t*\t0\t0\tCCATGGTATGTGCCTGTGGGGTCGATGCCATGTTCGTCACTG\t569626336635575648568542566376495452670712\tNH:i:1\tHI:i:1\tAS:i:41\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\n"
      }
     ],
     "persistentId": "3e3b6f8c-ae7e-45c3-820d-f15dbfe313d3",
     "text": "\nnoncoding_read_id = 'SRR306838.7944356\t'\n! samtools view $folder/$unspliced_bam | grep $noncoding_read_id"
    },
    "executionTime": "2019-10-12T03:46:44.495Z"
   },
   {
    "cell": {
     "executionCount": 71,
     "executionEventId": "fc37d30e-1539-4262-99e3-1edc5a8cfd77",
     "hasError": false,
     "id": "00b8ca70-dcb1-416b-9142-a2e7736ca3f4",
     "outputs": [],
     "persistentId": "4acd19e5-3727-407f-adb5-b0676c493cc6",
     "text": "df = coding_scores_not_low_complexity_unspliced.query(\"jaccard_in_peptide_db < 0.5 & in_cds_no_stop_codon == True\")\nfor read_id in df.short_read_id:\n    ! samtools view $folder/$unspliced_bam | grep $read_id"
    },
    "executionTime": "2019-10-12T03:50:27.181Z"
   },
   {
    "cell": {
     "executionCount": 72,
     "executionEventId": "e546927f-255e-44b2-8e71-ca92fe231c8d",
     "hasError": false,
     "id": "00b8ca70-dcb1-416b-9142-a2e7736ca3f4",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "SRR306838.8536423\t0\tchr1\t39448668\t255\t70M\t*\t0\t0\tAAAGACTTTGCTTCCCGAAGATACTCAGAAACTTGACAATTTCCTAGGAGAAGTCAGAGACAAATGGGAT\t-0454575:38569357656746757387639475375567787758567758596:6957665665585\tNH:i:1\tHI:i:1\tAS:i:66\tnM:i:1\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\nSRR306838.4158712\t16\tchrX\t129814671\t255\t49M14S\t*\t0\t0\tGACGATGTTGAAGGCGAAGACATAGATTGTGAGGAGGGAGAAAGAAAGGGAAGTAGAAGTAGG\t836857646958738766836846574664767626764766584546386474576222000\tNH:i:1\tHI:i:1\tAS:i:46\tnM:i:1\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\n"
      }
     ],
     "persistentId": "4acd19e5-3727-407f-adb5-b0676c493cc6",
     "text": "df = coding_scores_not_low_complexity_unspliced.query(\"jaccard_in_peptide_db < 0.9 & in_cds_no_stop_codon == True\")\nfor read_id in df.short_read_id:\n    ! samtools view $folder/$unspliced_bam | grep $read_id"
    },
    "executionTime": "2019-10-12T03:50:40.025Z"
   },
   {
    "cell": {
     "executionCount": 73,
     "executionEventId": "7fa0ea33-5018-4927-8472-45b95614cccf",
     "hasError": false,
     "id": "6f0c37a6-1c69-4167-abf7-7074c7cd6d3e",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "SRR306838.6939297\t16\tchr11\t63758272\t255\t65M\t*\t0\t0\tATGAAGGAAAATACTCAGTGTCAGCTTGAGCCTGCATTCCAAGCTTTTTTTTTAATTTGGTGTTT\t57:6669965;82667:64748:4:588668966785767898958667558635755376530.\tNH:i:1\tHI:i:1\tAS:i:63\tnM:i:0\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\n"
      }
     ],
     "persistentId": "fc1bcb0f-a943-4ecd-bf4f-7889239e3c6a",
     "text": "\nnoncoding_read_id = 'SRR306838.6939297\t'\n! samtools view $folder/$unspliced_bam | grep $noncoding_read_id"
    },
    "executionTime": "2019-10-12T03:54:52.440Z"
   },
   {
    "cell": {
     "executionCount": 74,
     "executionEventId": "938b1f28-31f0-4087-a7b0-22d1e4bc894c",
     "hasError": false,
     "id": "c4e90627-a46a-4fa2-b552-6402975ee8c7",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "accuracy score: 0.9204152249134948\n              precision    recall  f1-score   support\n\n       False       1.00      0.91      0.95       767\n        True       0.59      0.98      0.74       100\n\n   micro avg       0.92      0.92      0.92       867\n   macro avg       0.80      0.95      0.85       867\nweighted avg       0.95      0.92      0.93       867\n\n"
      },
      {
       "data": {
        "image/png": "iVBORw0KGgoAAAANSUhEUgAAAYUAAAEKCAYAAAD9xUlFAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDMuMC4zLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvnQurowAAIABJREFUeJzt3Xl8FFW2wPHfyQJh35GdoIKKKIgBFRUXFlE2Rx1FEUQRFEUdt1GfC+LzObgLImIkjMug4jCo0YFBRsEFAUEREdARAkJQZEsCJGQ/749qYk8MSQe6uno538+nP3RVV7pOJSSn7711zxVVxRhjjAGI8zoAY4wx4cOSgjHGmDKWFIwxxpSxpGCMMaaMJQVjjDFlLCkYY4wpY0nBGGNMGUsKxhhjylhSMMYYUybB6wCqq2nTppqcnOx1GMYYE1G++uqrXararKrjIi4pJCcns3LlSq/DMMaYiCIiPwVynHUfGWOMKWNJwRhjTBlLCsYYY8pYUjDGGFPGkoIxxpgyriUFEZkpIjtE5LtDvC4iMkVENojItyLS3a1YjDHGBMbNlsIrwIBKXr8Q6Oh7jAVedDEWY4wxAXBtnoKqfioiyZUcMhR4TZ31QJeJSEMRaamqv7gVU9jY+xP8+yYo3Od1JMaYCPDlhvokJZZy8tk9odfDrp7Ly8lrrYGtftuZvn2/SwoiMhanNUG7du1CEpxrVGHhDbB5gdeRGGPCnCrc889+PP3JqZzc8le+PO57El0+p5dJQSrYpxUdqKqpQCpASkpKhcdEhB/+Dp/cBfu2QM0GMOjvkFDT66iMMWFKADZugE+30n/oaZT06BHVSSETaOu33Qb42aNYQmPVFCchAJw9CZL7eRuPMSbsZGfnk5GRRffuLQGY+MwZDBuzq2zbbV4mhXRgvIi8BZwG5ETFeMKOb+DjW6Aot4LXVjn/9p0OXW8IbVzGmLD33nvfM27cP4mLE9auvYkGDZKoVSsxZAkBXEwKIvImcC7QVEQygQngtHxUdTowD7gI2ADkAde6FUtIfTcTtn1+6NfjEuDogaGLxxgT9nbsyOXWW+cze/ZaAE4/vQ3Z2fk0aJAU8ljcvPvoyipeV+Bmt87vmeyNzr/nPgNtzvn963VbQZ0WoY3JGBOWVJVZs9Zw223/Ys+eA9Sunchjj53P+PE9iY/3Zm5xxJXODns5Gc6/bc+H5l29jcUYE9bGjfsnL730FQB9+x5NauogOnRo5GlMVuYimLQUcjY5zxse7W0sxpiwd/HFx9OwYRJpaUP48MOrPU8IYC2F4Nr/M5QUQO3mUKOe19EYY8LMjz/u5qOPNnHjjSkADBhwLJs33+bJ2MGhWFIIpoPjCQ2slWCM+U1xcSnPPLOUCRMWU1BQTLduLTj99DYAYZUQwJJCcB1MCg2P8TYOY0zYWL16O6NHp/PVV84d9yNHdqVjx8YeR3VolhSC6eAgs7UUjIl5BQXFPProp0yatITi4lLatWvASy8NYsCAY70OrVKWFILJWgrGGJ/77vuIZ59dBsDNN/fgL3/pQ7164V/WxpJCMJW1FCwpGBPr/vznM1m6NJMnnujL2We39zqcgNktqcFU1lKw7iNjYs3ChRu59NK3KS4uBaBFi7p88cV1EZUQwJJC8BTkQP5uSKgFdUJXp8QY462srAOMHv0e/fv/jblz1/PXv64qe02komLQ4c26j4Il22+QOQL/Ixhjqu+dd9Zz003z2L59PzVrxjNhwjmMGtXN67COiCWFYMmxOQrGxIrt2/dzyy3zmTNnHQC9erUlLW0Ixx/f1OPIjpwlhWA52FKwO4+MiXrvvfc9c+aso06dRCZN6stNN/UgLi46eggsKQSLtRSMiWr5+cUkJTl/MseMOZWMjCzGjetBcnJDjyMLLhtoDhabo2BMVCotVaZO/ZIOHSbz00/ZAMTFCY8/3i/qEgJYUgieHOs+Miba/PDDLnr3/iu33DKf7dv38+ab33kdkuus+ygYSopg7xZAoH6y19EYY45QUVEJTz31BRMnfkJBQQlHHVWHadMGcsklJ3gdmussKQTD3p9AS6BeW0gI/2nsxphD++67HYwc+Q6rVm0H4Npru/H00/1p1KiWx5GFhiWFYLBCeMZEjdJSZc2aHbRv34DU1MH07x9bXcKWFILBBpmNiWhr1+6gc+dmiAgnn3wU7703jN6921O3bg2vQws5G2gOBhtkNiYi7dtXwPjx8+jS5UX+8Y/1ZfsvuqhjTCYEsJZCcNiKa8ZEnAULNjB27Ads2ZJDQkIcmzdnex1SWLCkEAw51n1kTKTYs+cAt9++gNdeWw1A9+4tSUsbQrduLTyOLDxYUjhSqv9dDM8YE7a++WY7Awb8jV9/zaVmzXgmTjyXO+/sRUKC9aQfZEnhSB3YCUX7oWYDSArfdVeNMdCpUxPq1q1Bp05NmDFjCJ06NfE6pLBjSeFIZfuttmYls40JK6rKG2+sYfDg46hfvya1ayeyePEoWrWqFzUF7ILN2kxHKsdWWzMmHG3enM0FF/yNq69+h3vv/XfZ/jZt6ltCqIS1FI5U2Z1HNshsTDgoKSll2rQV3HffR+TmFtG4cS169WrrdVgRw5LCkbI5CsaEjfXrdzJ6dDpLl2YCcPnlJ/L88xfSvHkdjyOLHJYUjpTNUTAmLGzalEW3bi9RWFhCy5Z1mTZtIBdffLzXYUUcV5OCiAwAJgPxwAxVnVTu9XbAq0BD3zH3quo8N2MKOmspGBMWOnRoxB//2JmkpASeeqo/DRsmeR1SRHItKYhIPPAC0A/IBFaISLqqrvM77AHgbVV9UUQ6A/OAZLdiCrqiA7D/Z4hLgHptvI7GmJhy4EARjzzyCX/4wwn07NkagFdfvZj4eLt/5ki4+d3rCWxQ1QxVLQTeAoaWO0aB+r7nDYCfXYwn+A62EuonO4nBGBMSn332E926vcSkSUsYO/Z9SksVwBJCELj5l6w1sNVvOxM4rdwxDwMfisgtQB2gr4vxBJ91HRkTUnv3FnDfff9m2rSVAHTu3Izp0wfZLaZB5GZareinpOW2rwReUdU2wEXA6yLyu5hEZKyIrBSRlTt37nQh1MNkg8zGhMy8eT/Spcs0pk1bSUJCHA891Juvvx7L6adb120wudlSyAT8bw5uw++7h0YDAwBUdamIJAFNgR3+B6lqKpAKkJKSUj6xeMfWUTAmJHJy8hk+fC7Z2fmkpLQiLW0IJ598lNdhRSU3k8IKoKOIdAC2AcOAq8odswXoA7wiIicASUAYNQWqYCuuGeMaVUUV4uKEBg2SmDJlAL/+msuf/nS6FbBzkWtJQVWLRWQ8sADndtOZqrpWRB4BVqpqOnAn8LKI3I7TtTRKVcOnJVAVaykY44qff97HTTf9k7PPbsedd/YCYMSIrh5HFRtcvWXGN+dgXrl9D/k9Xwec6WYMrtFS2LvJeW4tBWOCQlWZOXMVd975ITk5BSxblslNN/WgVq1Er0OLGXYf5eHatw1KCqF2c6hR1+tojIl4GRlZjBnzPh9/7HzYGjiwI9OnD7KEEGKWFA5XjhXCMyYYSkpKmTJlOfff/zEHDhTTtGltpkwZwLBhXRArRx9ylhQOV7bNUTAmWObMWc+BA8VceWUXJk8eQLNmVsDOK5YUDleOzVEw5nAVFpawb18BTZrUJj4+jrS0Ifz4424GDz7O69Bint3XdbispWDMYVmxYhspKamMGPEOB282PP74ppYQwoS1FA6XtRSMqZa8vCImTFjEM88so7RUycsrYseOXI46ym7UCCeWFA6XzVEwJmCLF29mzJj32bBhD3Fxwl13ncHEiedRu7bdWRRuLCkcjvxsyN8DCbWgTguvozEmbKkqt946n6lTVwBw0knNSUsbQo8erT2OzByKJYXD4V/ewm6ZM+aQRIT69WuSmBjHAw/05t57z6JGjXivwzKVsKRwOKxktjGHtGtXHhs37uG005zqpQ8+eA7Dh59M587NPI7MBCKgu49EpIaIHOt2MBHDSmYb8zuqyltvfccJJ7zAxRfPJivrAABJSQmWECJIlUlBRAYCa4CFvu1uIvKO24GFNRtkNua/ZGbuZejQt7jyyn+wa1cenTs3Iy+vyOuwzGEIpPvoEZwV0xYBqOo3Md9qsO4jYwAoLVVmzPiau+9eyN69BdSvX5Onn+7P6NGnWImKCBVIUihS1exyP+DIKW/tBus+MgaA0aPTeeWVbwAYMuQ4pk27iNat61fxVSacBTKmsF5ELgfiRKSDiDwHLHM5rvBVUgj7tgAC9ZO9jsYYT1199Uk0b16Ht966lHffvcISQhQIJCmMB04FSoG5QD5wm5tBhbW9W5y1FOq1hYSaXkdjTEh9990OJk/+7TNhnz5Hk5FxK1dcYRVNo0Ug3UcXqOo9wD0Hd4jIJTgJIvYcLG/R0LqOTOwoKCjmL3/5nMce+4yiolJSUlpx5pntAKhTp4bH0ZlgCqSl8EAF++4PdiAR42AhPFtHwcSI5cszOfXUVCZO/ISiolLGjUvhpJOO8jos45JDthRE5AJgANBaRJ7xe6k+TldSbMq2loKJDbm5hTz44CKee24ZqtCxY2NmzBhC797tvQ7NuKiy7qMdwHc4Ywhr/fbvA+51M6iwZiuumRhx//0fM3nycuLihLvvPoOHHz7XlsaMAYdMCqq6ClglIrNUNT+EMYU3m6NgYsT995/NmjU7ePzxvqSktPI6HBMigYwptBaRt0TkWxH5z8GH65GFI1Wbo2CiVnr6D1x00SyKikoAaNasDh99NNISQowJJCm8AvwVEOBC4G3gLRdjCl8HdkJRLtRsCLUaex2NMUGxY0cuw4bNYejQt5g/fwOvvrra65CMhwJJCrVVdQGAqm5U1QeA89wNK0xZK8FEEVXlb3/7lhNOeIHZs9dSu3YikycP4Npru3kdmvFQIPMUCsSZlbJRRG4EtgHN3Q0rTFkhPBMltmzJ4cYbP2D+/A0A9O17NKmpg+jQoZHHkRmvBZIUbgfqArcC/wc0AK5zM6iwZYPMJkp8+OFG5s/fQMOGSTzzTH9GjepmM5INEEBSUNXlvqf7gBEAItLGzaDClnUfmQiWm1tYNvt49OhT2LZtL2PHnkrLlvU8jsyEk0rHFESkh4hcLCJNfdsnishrxGpBPGspmAhUXFzKE08soX3758jIyAKcZTInTDjXEoL5nUMmBRH5CzALGA78S0Tux1lTYTXQKTThhRkbUzARZvXq7Zx22gzuueff7N59gHff/d7rkEyYq6z7aCjQVVUPiEhj4Gff9g+hCS3MFOVB7i8Qlwh1Y7P3zESOgoJiHn30UyZNWkJxcSnt2jUgNXUQF1wQ2+tjmapVlhTyVfUAgKruEZHvYzYhAORscv5tkAxx8Z6GYkxlVq36heHD57J+/S5EYPz4Hjz2WB/q1bNS76ZqlSWFo0XkYHlsAZL9tlHVS6p6cxEZAEwG4oEZqjqpgmMuBx7GWc1ttapeFXj4IWSDzCZC1KyZwMaNWRx3XBNmzBjCWWe18zokE0EqSwqXltueWp03FpF44AWgH5AJrBCRdFVd53dMR+A+4ExVzRKR8J3/kGMls034+vrrXzjllBaICJ07N2P+/OH06tWWpKRA7jo35jeVFcT76AjfuyewQVUzAETkLZxxinV+x4wBXlDVLN85dxzhOd1jg8wmDGVlHeCuuz5k5sxvePPNSxk2rAsA55/fwePITKRy82NEa2Cr33YmcFq5YzoBiMgSnC6mh1X1X+XfSETGAmMB2rXzqCmcY91HJry88856brppHtu376dmzXh2787zOiQTBdxMChVNj9QKzt8ROBdoA3wmIl1UNfu/vkg1FUgFSElJKf8eoZFtcxRMeNi+fT+33DKfOXOcRveZZ7ZlxowhHH98U48jM9Eg4KQgIjVVtaAa750JtPXbboNzW2v5Y5apahGwSUR+wEkSK6pxHveVlsDeg3cfWbPceOerr36mX7/XycrKp06dRCZN6stNN/UgLs5KVJjgqLJKqoj0FJE1wI++7a4i8nwA770C6CgiHUSkBjAMSC93zLv4Kq76Zk13AjKqEX9o7N8GJYVQ+yioUdfraEwM69y5Gc2a1eGCC45h7dqbGD++pyUEE1SBlM6eAgwCdgOo6moCKJ2tqsXAeGABsB54W1XXisgjIjLEd9gCYLeIrMOZLX23qu6u/mW4zMpbGI+UliqpqV+Rne0sflirViKffjqK+fOH0759Q4+jM9EokO6jOFX9qVwFxZJA3lxV5wHzyu17yO+5Anf4HuHL5igYD/zwwy6uv/59Pv98CytWbOPll53PUkcdZa1V455AksJWEekJqG/uwS1AbC3HaS0FE0JFRSU8/fRSHn54MQUFJbRoUZcLL+zodVgmRgSSFMbhdCG1A34F/u3bFztsjoIJkVWrfmH06HRWrdoOwLXXduPpp/vTqFEtjyMzsSKQpFCsqsNcjyScWfeRCYGNG/fQs+cMiotLSU5uSGrqIPr1sw8iJrQCSQorfLeKzgbmquo+l2MKP9Z9ZELgmGMaM2LEydSrV4P/+78+1K1bw+uQTAyq8u4jVT0GeBQ4FVgjIu+KSOy0HPKzIX8PJNR2bkk1Jkj27y/k1lvns3TpbxP/09KGMHnyhZYQjGcCuSUVVf1CVW8FugN7cRbfiQ1lrYSjwdawNUGyYMEGTjxxGs8//yU33vhPnBvxsHWSjeeq7D4Skbo4heyGAScA7wG9XI4rfJSNJ1jXkTlye/Yc4PbbF/Daa6sBOPXUlqSlDbFkYMJGIGMK3wHvA0+o6mcuxxN+yu48skFmc2TmzFnHzTfPY8eOXJKSEpg48VzuuOMMEhICarAbExKBJIWjVbXU9UjCla2jYIIgOzufsWPfJysrn9692/Pyy4Pp1KmJ12EZ8zuHTAoi8rSq3gn8Q0R+V5k0kJXXokKOtRTM4VFVSkuV+Pg4GjZMYtq0gWRlHeCGG1KsXpEJW5W1FGb7/q3WimtRJ9taCqb6Nm/OZuzY9zn//A7ce+9ZAGUL4BgTzg7ZmamqX/qenqCqH/k/cAaco19JIezbAgg0SPY6GhMBSkpKmTJlOV26TGPhwgymTv2S/Pxir8MyJmCBjHBdV8G+0cEOJCzt/Qm0FOq1hXi7b9xUbv36nfTu/Qq33fYvcnOLGDasC19/fYOtk2wiSmVjClfg3IbaQUTm+r1UD8iu+KuijM1kNgEoLi7l8cc/55FHPqWwsIRWrerx4osDGTLkOK9DM6baKvsI8yXOGgptgBf89u8DVrkZVNiwQngmAHFxwocfZlBYWMKYMd154ol+NGyY5HVYxhyWQyYFVd0EbMKpihqbrBCeOYQDB4rYt6+Q5s3rEBcnzJgxmK1b93L++bZcq4lshxxTEJFPfP9micgev0eWiOwJXYgesu4jU4FPP/2Jrl2nc/XVc8vKU3Ts2MQSgokKlXUfHVxys2koAglL1lIwfvbuLeC++/7NtGkrAUhMjGfXrjyaNavjcWTGBE9lt6QenMXcFohX1RLgDOAGIPp/C1StpWDKzJ//I126TGPatJUkJMQxYcI5fP31WEsIJuoEcq/cu0APETkGeA34J/AGMMjNwDyXtwOKciGpkfMwMUlVGTPmfdLSnHsrUlJaMXPmEE46ycqom+gUyDyFUlUtAi4BnlPVW4DW7oYVBqzryOCUsm7Tpj5JSQk89VQ/li4dbQnBRLWAluMUkT8CI4CLffsS3QspTFghvJj188/72LhxD2ef3R6A//mfsxkx4mSOOaaxx5EZ475AZzSfh1M6O0NEOgBvuhtWGLCS2TFHVUlL+5rOnV/g0kvfZvfuPABq1Ii3hGBiRpUtBVX9TkRuBY4VkeOBDar6f+6H5jFrKcSUjIwsxox5n48/3gTAoEGdKCqK3YrxJnYFsvLa2cDrwDZAgBYiMkJVl7gdnKdsNnNMOFjA7oEHFpGXV0TTprWZMmUAw4Z1sdXQTEwKZEzhWeAiVV0HICIn4CSJFDcD85ytoxATRo58lzfeWAPAVVedxHPPXWC3mZqYFsiYQo2DCQFAVdcD0V0ytCgPcrdDXCLUbeN1NMZFY8Z0p02b+qSnD2PWrEssIZiYF0hL4WsReQmndQAwnGgviFc2ntAB4uK9jcUE1YoV2/j4403cc4+z8M255yazYcMt1Kxp5a2NgcCSwo3ArcCfccYUPgWedzMoz5WttmZdR9EiL6+ICRMW8cwzyygtVXr1alt2y6klBGN+U+lvg4icBBwDvKOqT4QmpDCQY4PM0WTx4s1cf306GzdmERcn3HXXGZx6aiuvwzImLFVWJfV/cEpcDAcWikhFK7BVSkQGiMgPIrJBRO6t5LjLRERFJDwGr202c1TIycnnhhve57zzXmXjxixOOqk5y5aN5skn+1O7dvTPvzTmcFTWUhgOnKyquSLSDJgHzAz0jUUkHmdxnn5AJrBCRNL9B619x9XD6Z5aXt3gXWOF8KLCgw8uIjX1axIT43jwwd7cc89Z1KhhY0TGVKaypFCgqrkAqrpTRAK5U8lfT5yJbhkAIvIWMBRYV+64/wWeAO6q5vu7x+YoRCxVLZtf8NBD57BpUzaTJvXhxBObexyZMZGhsj/0R4vIXN/jHeAYv+25lXzdQa2BrX7bmZQrpCcipwBtVfWDakfultISyHFmtdLAFk2JFKrKG2+s4fzzX6OwsASApk1r8/77V1pCMKYaKmspXFpue2o137ui6aBa9qLT8ngWGFXlG4mMBcYCtGvXrpphVNP+bVBaBHVaQKLdsx4JMjP3Mm7cP/ngg/8AMGvWt1x77SkeR2VMZKpsjeaPjvC9M3EW6DmoDfCz33Y9oAuw2NfcbwGki8gQVV1ZLpZUIBUgJSVFcZMNMkeM0lLl5Ze/4u67F7JvXyENGtTk6af7M2pUN69DMyZiuXmD9gqgo6+q6jZgGHDVwRdVNQe/pT5FZDFwV/mEEHI2yBwRNmzYw5gx77N48WYAhg49jmnTBtKqVT1vAzMmwrmWFFS1WETGAwuAeGCmqq4VkUeAlaqa7ta5j0hZS8GSQjj77LOfWLx4M82b12Hq1Au57LLOVsDOmCAIOCmISE1VLajOm6vqPJxbWf33PXSIY8+tznu7xtZRCFvZ2fk0bJgEwKhR3di5M4/Ro0+hSZPaHkdmTPSo8jZTEekpImuAH33bXUUkestc2DoKYaegoJgJExbRvv1z/PjjbsBZJvPPfz7TEoIxQRZIS2EKMAhndjOqulpEznM1Ki9ZiYuwsmxZJqNHp7Nu3U4AFizYSMeOTTyOypjoFUhSiFPVn8r115a4FI+38rOcR2IdqG33tnspN7eQBx9cxHPPLUMVOnZsTFrakLIidsYYdwSSFLaKSE9AfaUrbgH+425YHsnxq45qg5aeWb48k6uumktGRhbx8cJdd/ViwoRzqFXL6hUZ47ZAksI4nC6kdsCvwL99+6KPzVEICw0bJrFt2166dj2KtLQhVtHUmBCqMimo6g6cOQbRL9vmKHjl88+3cOaZbRERjjuuKR9/fA09erQiMdEK2BkTSlUmBRF5Gb/yFAep6lhXIvKSDTKH3I4dudx663xmz17Lq69ezMiRXQHo1attFV9pjHFDIN1H//Z7ngT8gf8udBc9cmzFtVBRVWbNWsNtt/2LPXsOULt2YlkhO2OMdwLpPprtvy0irwMLXYvIS1YyOyS2bMnhxhs/YP78DQD063c0qamDSU5u6HFkxpjDKXPRAYi++wJLCmHfVpA4qB99lxculi/PpG/f19m/v5CGDZN49tkLuOaarlaiwpgwEciYQha/jSnEAXuAQy6tGbH2/gRa6iSE+BpeRxO1unVrQdu29Tn++Ka88MJFtGxpBeyMCSeVJgVxPr51xalyClCqqu6WrvaKdR25ori4lKlTv2TkyK40blyLmjUTWLLkOho1quV1aMaYClRa+8iXAN5R1RLfIzoTAtgcBResXr2d006bwe23L+COOxaU7beEYEz4CmTd5S9FpLvrkXjNCuEFTX5+MQ888DEpKS/z9de/0K5dA668sovXYRljAnDI7iMRSVDVYuAsYIyIbARycZbZVFWNrkRh3UdB8cUXWxk9Op3vv9+FCIwf34PHHutDvXo1vQ7NGBOAysYUvgS6AxeHKBZvla24Zt1Hh2vDhj2cffZfKS1VjjuuCWlpQzjzTJfX1DbGBFVlSUEAVHVjiGLxjqp1HwXBscc2ZuzY7jRuXIsHHzyHpCQ3V3s1xrihst/aZiJyx6FeVNVnXIjHG3m/QlEuJDWCJJtAFaisrAPceeeHXHttt7KS1tOmDbQ5B8ZEsMqSQjxQF1+LIaplWyuhuubOXc/NN89j+/b9fPXVL3zzzQ2IiCUEYyJcZUnhF1V9JGSReMkK4QVs+/b9jB8/j3/8Yz0AZ53VjhkzBlsyMCZKVDmmEBOyrRBeVVSV115bze23LyArK5+6dWvw+ON9ufHGFOLiYue/ijHRrrKk0CdkUXjNWgpVys7O5847PyQrK58BA45l+vSBtG9v4y/GRJtDJgVV3RPKQDxls5krVFqqlJYqCQlxNGpUi5deGkReXhFXX32ydRcZE6UCmdEc/XJsxbXyvv9+F717/5VJkz4v23fppZ0ZMcIqmhoTzSwpFOVC7nanMmrd1l5H47miohIee+wzunadzpIlW0lLW0V+frHXYRljQsRmF+Vscv6tnwxxsb0e8KpVv3Dddel88812AEaPPoUnn+xnk9CMiSH22241jygqKmHChMU88cQSSkqU5OSGvPzyYPr2tTEWY2KNJYWyQebYTQoJCXEsX76N0lLltttO49FHz6duXVtoyJhYZEkhRgvh7dtXwL59hbRqVQ8RYcaMwWzfvp8zzmjrdWjGGA/ZQHMMthQWLNhAly4vMnz4XA6um9ShQyNLCMYYSwqx1FLYvTuPa655lwEDZrFlSw779hWwe/cBr8MyxoQRV5OCiAwQkR9EZIOI3FvB63eIyDoR+VZEPhKR9m7G8zulJb/dfRTFE9dUlTlz1tG58zRee201SUkJPPFEX5Ytu56mTWt7HZ4xJoy4NqYgIvHAC0A/IBNYISLpqrrO77BVQIqq5onIOOAJ4Aq3Yvqd/ZlQWgR1WkJidP5xVFWGD5/Lm29+B0DRZJGNAAAPYElEQVTv3u15+eXBdOrUxOPIjDHhyM2WQk9gg6pmqGoh8BYw1P8AVV2kqnm+zWVAGxfj+b0YKIQnInTu3Ix69Wrw4osDWbToGksIxphDcvPuo9bAVr/tTOC0So4fDcyv6AURGQuMBWjXLojLO0bpHIVNm7LIyMiiTx8n2d1zz5mMGtWNNm3qexyZMSbcudlSqKhAjlZ4oMjVQArwZEWvq2qqqqaoakqzZs2CF2FOdBXCKykpZfLkZXTp8iJXXDGHHTtyAUhMjLeEYIwJiJsthUzA/x7HNsDP5Q8Skb7A/cA5qlrgYjy/lx09hfDWrdvJ9dens3RpJgBDhhxn6xwYY6rNzaSwAugoIh2AbcAw4Cr/A0TkFOAlYICq7nAxlopFwToKRUUlPP74Ev73fz+lsLCEVq3q8eKLAxky5DivQzPGRCDXkoKqFovIeGABznrPM1V1rYg8AqxU1XSc7qK6wN995Zi3qOoQt2L6nZzIH2i+6qq5zJnj3NA1Zkx3nnyyHw0aJHkclTEmUrla5kJV5wHzyu17yO95XzfPX6n8LOeRWAdqN/csjCN1222n8c0323nppUGcf34Hr8MxxkS42J3R7L/aWgQtGvPJJ5uZOHFx2fZZZ7Vj/fqbLSEYY4IidgviRdhqa3v3FnDPPQuZPv0rAM47rwO9ezsTwBMSYje3G2OCK3aTQgQVwps370duuOEDMjP3kpgYx/33n83pp4d2np8xJjbEblKIgEJ4u3bl8ac//YtZs9YA0LNna9LShtClS+SOgRhjwlvsJoUImM38yCOfMGvWGmrVSuDRR8/ntttOIz7euoqMMe6xpBBm3Ueqiu/2XCZOPJdff83lscfO55hjGnscmTEmFsTmx86SQti3FSQO6gexltIRUFVefvkrevWaSX5+MQCNGtVi9uzLLCEYY0ImNpNCzmZAoV47iPd+LeKNG/fQp89rjB37AcuWZfL222u9DskYE6Nis/soTAaZnQJ2y3nggY85cKCYZs1q8/zzF3L55Sd6GpcxJnbFZlIIg/GEtWt3cN116Xz55TYAhg8/ieeeG2AroRljPBWbSSEMCuGtWrWdL7/cRuvW9XjppUEMHNjJs1iMMeag2EwKHq24tnNnLs2a1QGclkF2dj4jRpxsBeyMMWEjRgeaQ9tSyMsr4q67PiQ5eTLr1+8EnGUyx4/vaQnBGBNWYq+loBrSlsKiRZsYM+Z9Nm7MIi5O+PTTnzjhhCCuHmeMMUEUe0kh71cozoOkxpDU0LXT5OTk8+c/LyQ19WsATjqpOTNnDiUlpZVr5zTGmCMVe0khBOUtPv98C8OGzWHbtn0kJsbx4IO9ueees6hRI961cxpjTDDEXlIIwWprLVrUZffuA5x+ehtmzBjMiSdaATtjTGSIvaTgQktBVVm4MIN+/Y5GRDj22MZ8/vm1dOvWwgrYGWMiSuz9xQryxLWtW3MYPPhNLrjgb/z1r9+U7T/11FaWEIwxESf2WgpBKnFRWuoUsLv77oXs21dIgwY1qVnTxgyMMZEt9pJCEFoKP/64mzFj3ueTT34C4OKLj+eFFy6iVat6wYjQGGM8E1tJoSjXuSU1vgbUPbxbQ7/4Yit9+rxGfn4xzZvXYerUC7nsss5layAYY0wki62kcHDSWv0OEHd4XT0pKa3o2LExp5zSkmee6U+TJlbAzhgTPWIsKVT/zqOCgmKeeuoLbrghhaZNa1OjRjxLllxHvXo1XQrSGGO8E1tJoZpzFJYty2T06HTWrdvJ+vW7+NvfLgGwhGCMiVqxlRQCbCnk5hbywAMfM3nyclShU6cm3HDDqSEI0BhjvBVbSSGAlsJHH2UwZsz7bNqUTXy8cPfdvZgw4VySkmLrW2WMiU2x9ZeuipLZ//nPbvr1ex1V6NatBWlpQ+jevWUIAzTGGG/FTlIoLYGczc7zBh0qPKRTpybcdttpNGtWh7vv7kViok1GM8bElthJCvszobQI6rSEROc20l9/3c+tt/6LG288lfPOcxLFs88O8DJKY4zxlKvFeURkgIj8ICIbROTeCl6vKSKzfa8vF5Fk14LxG2RWVV5/fTWdO0/j7bfXcvvtC1BV105tjDGRwrWkICLxwAvAhUBn4EoR6VzusNFAlqoeCzwLPO5WPAeTwpbCTgwc+AYjR77Lnj0H6N//GN59d5jNSDbGGNztPuoJbFDVDAAReQsYCqzzO2Yo8LDv+RxgqoiIuvCxvTQrg+lLenDPgnbsz9tAo0ZJPPvsBYwc2dUSgjHG+LiZFFoDW/22M4HTDnWMqhaLSA7QBNgV7GByft7ExIXnsD8PLr30BKZOvYgWLeoG+zTGGBPR3EwKFX38Lt8CCOQYRGQsMBagXbt2hxVMo3rCjCs/pLDHRC4dc/FhvYcxxkQ7N5NCJtDWb7sN8PMhjskUkQSgAbCn/BupaiqQCpCSknJ4XUuD3mTwQKWCnGOMMcbHzbuPVgAdRaSDiNQAhgHp5Y5JB67xPb8M+NiN8YQyIiC2GpoxxhyKay0F3xjBeGABEA/MVNW1IvIIsFJV04E04HUR2YDTQhjmVjzGGGOq5urkNVWdB8wrt+8hv+f5wB/djMEYY0zgrC/FGGNMGUsKxhhjylhSMMYYU8aSgjHGmDKWFIwxxpSRSKsOKiI7gZ8O88ub4kIJjTBn1xwb7Jpjw5Fcc3tVbVbVQRGXFI6EiKxU1RSv4wglu+bYYNccG0JxzdZ9ZIwxpowlBWOMMWViLSmkeh2AB+yaY4Ndc2xw/ZpjakzBGGNM5WKtpWCMMaYSUZkURGSAiPwgIhtE5N4KXq8pIrN9ry8XkeTQRxlcAVzzHSKyTkS+FZGPRKS9F3EGU1XX7HfcZSKiIhLxd6oEcs0icrnvZ71WRN4IdYzBFsD/7XYiskhEVvn+f1/kRZzBIiIzRWSHiHx3iNdFRKb4vh/fikj3oAagqlH1wCnTvRE4GqgBrAY6lzvmJmC67/kwYLbXcYfgms8Davuej4uFa/YdVw/4FFgGpHgddwh+zh2BVUAj33Zzr+MOwTWnAuN8zzsDm72O+wivuTfQHfjuEK9fBMzHWbnydGB5MM8fjS2FnsAGVc1Q1ULgLWBouWOGAq/6ns8B+ohIRUuDRooqr1lVF6lqnm9zGc5KeJEskJ8zwP8CTwD5oQzOJYFc8xjgBVXNAlDVHSGOMdgCuWYF6vueN+D3KzxGFFX9lApWoPQzFHhNHcuAhiLSMljnj8ak0BrY6red6dtX4TGqWgzkAE1CEp07Arlmf6NxPmlEsiqvWUROAdqq6gehDMxFgfycOwGdRGSJiCwTkQEhi84dgVzzw8DVIpKJs37LLaEJzTPV/X2vFlcX2fFIRZ/4y99iFcgxkSTg6xGRq4EU4BxXI3JfpdcsInHAs8CoUAUUAoH8nBNwupDOxWkNfiYiXVQ12+XY3BLINV8JvKKqT4vIGTirOXZR1VL3w/OEq3+/orGlkAm09dtuw++bk2XHiEgCTpOzsuZauAvkmhGRvsD9wBBVLQhRbG6p6prrAV2AxSKyGafvNT3CB5sD/b/9nqoWqeom4AecJBGpArnm0cDbAKq6FEjCqREUrQL6fT9c0ZgUVgAdRaSDiNTAGUhOL3dMOnCN7/llwMfqG8GJUFVes68r5SWchBDp/cxQxTWrao6qNlXVZFVNxhlHGaKqK70JNygC+b/9Ls5NBYhIU5zupIyQRhlcgVzzFqAPgIicgJMUdoY0ytBKB0b67kI6HchR1V+C9eZR132kqsUiMh5YgHPnwkxVXSsijwArVTUdSMNpYm7AaSEM8y7iIxfgNT8J1AX+7htT36KqQzwL+ggFeM1RJcBrXgD0F5F1QAlwt6ru9i7qIxPgNd8JvCwit+N0o4yK5A95IvImTvdfU984yQQgEUBVp+OMm1wEbADygGuDev4I/t4ZY4wJsmjsPjLGGHOYLCkYY4wpY0nBGGNMGUsKxhhjylhSMMYYU8aSggk7IlIiIt/4PZIrOTb5UNUkq3nOxb5KnKt9JSKOO4z3uFFERvqejxKRVn6vzRCRzkGOc4WIdAvga/4kIrWP9NwmNlhSMOHogKp283tsDtF5h6tqV5xiiU9W94tVdbqqvubbHAW08nvtelVdF5Qof4tzGoHF+SfAkoIJiCUFExF8LYLPRORr36NXBcecKCJf+loX34pIR9/+q/32vyQi8VWc7lPgWN/X9vHV6V/jq3Nf07d/kvy2PsVTvn0Pi8hdInIZTn2pWb5z1vJ9wk8RkXEi8oRfzKNE5PnDjHMpfoXQRORFEVkpzjoKE337bsVJTotEZJFvX38RWer7Pv5dROpWcR4TQywpmHBUy6/r6B3fvh1AP1XtDlwBTKng624EJqtqN5w/ypm+sgdXAGf69pcAw6s4/2BgjYgkAa8AV6jqSTgVAMaJSGPgD8CJqnoy8Kj/F6vqHGAlzif6bqp6wO/lOcAlfttXALMPM84BOGUtDrpfVVOAk4FzRORkVZ2CUxfnPFU9z1f64gGgr+97uRK4o4rzmBgSdWUuTFQ44PvD6C8RmOrrQy/BqelT3lLgfhFpA8xV1R9FpA9wKrDCV96jFk6CqcgsETkAbMYpv3wcsElV/+N7/VXgZmAqzvoMM0Tkn0DApblVdaeIZPhq1vzoO8cS3/tWJ846OGUf/FfdulxExuL8XrfEWXDm23Jfe7pv/xLfeWrgfN+MASwpmMhxO/Ar0BWnhfu7RXNU9Q0RWQ4MBBaIyPU4ZYZfVdX7AjjHcP+CeSJS4Robvno8PXGKsA0DxgPnV+NaZgOXA98D76iqivMXOuA4cVYgmwS8AFwiIh2Au4AeqpolIq/gFIYrT4CFqnplNeI1McS6j0ykaAD84quRPwLnU/J/EZGjgQxfl0k6TjfKR8BlItLcd0xjCXx96u+BZBE51rc9AvjE1wffQFXn4QziVnQH0D6c8t0VmQtcjLMOwGzfvmrFqapFON1Ap/u6nuoDuUCOiBwFXHiIWJYBZx68JhGpLSIVtbpMjLKkYCLFNOAaEVmG03WUW8ExVwDficg3wPE4Sxauw/nj+aGIfAssxOlaqZKq5uNUoPy7iKwBSoHpOH9gP/C93yc4rZjyXgGmHxxoLve+WcA6oL2qfunbV+04fWMVTwN3qepqnLWZ1wIzcbqkDkoF5ovIIlXdiXNn1Ju+8yzD+V4ZA1iVVGOMMX6spWCMMaaMJQVjjDFlLCkYY4wpY0nBGGNMGUsKxhhjylhSMMYYU8aSgjHGmDKWFIwxxpT5f6ECaWxsd/bDAAAAAElFTkSuQmCC\n",
        "text/plain": "<Figure size 432x288 with 1 Axes>"
       },
       "metadata": {
        "needs_background": "light"
       },
       "output_type": "display_data"
      }
     ],
     "persistentId": "2c810e70-a822-4d9a-a9b0-2622dfa247bf",
     "text": "y_true = coding_scores_not_low_complexity_unspliced['in_cds_no_stop_codon']\ny_score = coding_scores_not_low_complexity_unspliced['jaccard_in_peptide_db']\ny_pred = y_score > 0.9\n\nfpr, tpr, thresholds = metrics.roc_curve(y_true, y_score)\n\nfig, ax = plt.subplots()\nlw = 2\nax.plot(fpr, tpr, lw=2, color='darkorange')\nax.plot([0, 1], [0, 1], color='navy', lw=lw, linestyle='--')\nax.set(xlabel='False Positive Rate', ylabel='True Positive Rate')\n\nprint(\"accuracy score:\", metrics.accuracy_score(y_true, y_pred))\n\nprint(metrics.classification_report(y_true, y_pred))"
    },
    "executionTime": "2019-10-12T04:00:35.550Z"
   },
   {
    "cell": {
     "executionCount": 75,
     "executionEventId": "44af50b7-8c03-415f-bc80-0ffd38dc33f7",
     "hasError": true,
     "id": "9fbf9090-d739-4873-9dc0-4088a824c9b5",
     "outputs": [
      {
       "ename": "TypeError",
       "evalue": "score_single_sequence() missing 2 required positional arguments: 'peptide_graph' and 'peptide_ksize'",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
        "\u001b[0;32m<ipython-input-75-e1bfbbb1d94a>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mscore_single_sequence\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"AAAGACTTTGCTTCCCGAAGATACTCAGAAACTTGACAATTTCCTAGGAGAAGTCAGAGACAAATGGGAT\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
        "\u001b[0;31mTypeError\u001b[0m: score_single_sequence() missing 2 required positional arguments: 'peptide_graph' and 'peptide_ksize'"
       ]
      }
     ],
     "persistentId": "50faf94d-9309-4931-8fdf-21f88ecfeff9",
     "text": "score_single_sequence(\"AAAGACTTTGCTTCCCGAAGATACTCAGAAACTTGACAATTTCCTAGGAGAAGTCAGAGACAAATGGGAT\")"
    },
    "executionTime": "2019-10-12T04:02:49.122Z"
   },
   {
    "cell": {
     "executionCount": 76,
     "executionEventId": "9e155fe4-76eb-49ac-8269-069f24ff1458",
     "hasError": false,
     "id": "9fbf9090-d739-4873-9dc0-4088a824c9b5",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "\ttranslation: \tKTLLPEDTQKLDNFLGEVRDKWD\n\tkmers: FLGEVRD DTQKLDN LPEDTQK KLDNFLG LLPEDTQ EVRDKWD GEVRDKW PEDTQKL TQKLDNF LDNFLGE EDTQKLD NFLGEVR LGEVRDK KTLLPED TLLPEDT DNFLGEV QKLDNFL\n\tK-mers in peptide database:\n{('DNFLGEV', 7517166875925614125): 1,\n ('DTQKLDN', 5951274799996309732): 0,\n ('EDTQKLD', 14887061883264050655): 0,\n ('EVRDKWD', 829818351693786468): 1,\n ('FLGEVRD', 9516214906786795876): 1,\n ('GEVRDKW', 8385754174638241851): 1,\n ('KLDNFLG', 3011079111732742999): 1,\n ('KTLLPED', 14836733754548320237): 1,\n ('LDNFLGE', 14534966961722312621): 1,\n ('LGEVRDK', 79391250229811418): 1,\n ('LLPEDTQ', 14973099275059575838): 0,\n ('LPEDTQK', 7311319343166724544): 0,\n ('NFLGEVR', 966627286140107258): 1,\n ('PEDTQKL', 908198253004630471): 0,\n ('QKLDNFL', 4722373045142000694): 1,\n ('TLLPEDT', 7583591577639096127): 0,\n ('TQKLDNF', 1242088564466415818): 0}\n"
      },
      {
       "data": {
        "text/plain": "(0.5882352941176471, 17)"
       },
       "execution_count": 76,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "50faf94d-9309-4931-8fdf-21f88ecfeff9",
     "text": "score_single_sequence(\"AAAGACTTTGCTTCCCGAAGATACTCAGAAACTTGACAATTTCCTAGGAGAAGTCAGAGACAAATGGGAT\", peptide_graph, peptide_ksize)"
    },
    "executionTime": "2019-10-12T04:02:58.071Z"
   },
   {
    "cell": {
     "executionCount": 77,
     "executionEventId": "070586b6-4423-4e89-97bc-38c9a5cf2d8a",
     "hasError": true,
     "id": "c4e90627-a46a-4fa2-b552-6402975ee8c7",
     "outputs": [
      {
       "ename": "SyntaxError",
       "evalue": "invalid syntax (<ipython-input-77-747e5db5f746>, line 3)",
       "output_type": "error",
       "traceback": [
        "\u001b[0;36m  File \u001b[0;32m\"<ipython-input-77-747e5db5f746>\"\u001b[0;36m, line \u001b[0;32m3\u001b[0m\n\u001b[0;31m    y_pred = y_score > 0.0.5\u001b[0m\n\u001b[0m                           ^\u001b[0m\n\u001b[0;31mSyntaxError\u001b[0m\u001b[0;31m:\u001b[0m invalid syntax\n"
       ]
      }
     ],
     "persistentId": "2c810e70-a822-4d9a-a9b0-2622dfa247bf",
     "text": "y_true = coding_scores_not_low_complexity_unspliced['in_cds_no_stop_codon']\ny_score = coding_scores_not_low_complexity_unspliced['jaccard_in_peptide_db']\ny_pred = y_score > 0.0.5\n\nfpr, tpr, thresholds = metrics.roc_curve(y_true, y_score)\n\nfig, ax = plt.subplots()\nlw = 2\nax.plot(fpr, tpr, lw=2, color='darkorange')\nax.plot([0, 1], [0, 1], color='navy', lw=lw, linestyle='--')\nax.set(xlabel='False Positive Rate', ylabel='True Positive Rate')\n\nprint(\"accuracy score:\", metrics.accuracy_score(y_true, y_pred))\n\nprint(metrics.classification_report(y_true, y_pred))"
    },
    "executionTime": "2019-10-12T04:05:11.183Z"
   },
   {
    "cell": {
     "executionCount": 78,
     "executionEventId": "88577923-1b57-4bc3-9474-1d7673c2f9ca",
     "hasError": false,
     "id": "c4e90627-a46a-4fa2-b552-6402975ee8c7",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "accuracy score: 0.8973471741637832\n              precision    recall  f1-score   support\n\n       False       1.00      0.88      0.94       767\n        True       0.53      1.00      0.69       100\n\n   micro avg       0.90      0.90      0.90       867\n   macro avg       0.76      0.94      0.82       867\nweighted avg       0.95      0.90      0.91       867\n\n"
      },
      {
       "data": {
        "image/png": "iVBORw0KGgoAAAANSUhEUgAAAYUAAAEKCAYAAAD9xUlFAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDMuMC4zLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvnQurowAAIABJREFUeJzt3Xl8FFW2wPHfyQJh35GdoIKKKIgBFRUXFlE2Rx1FEUQRFEUdt1GfC+LzObgLImIkjMug4jCo0YFBRsEFAUEREdARAkJQZEsCJGQ/749qYk8MSQe6uno538+nP3RVV7pOJSSn7711zxVVxRhjjAGI8zoAY4wx4cOSgjHGmDKWFIwxxpSxpGCMMaaMJQVjjDFlLCkYY4wpY0nBGGNMGUsKxhhjylhSMMYYUybB6wCqq2nTppqcnOx1GMYYE1G++uqrXararKrjIi4pJCcns3LlSq/DMMaYiCIiPwVynHUfGWOMKWNJwRhjTBlLCsYYY8pYUjDGGFPGkoIxxpgyriUFEZkpIjtE5LtDvC4iMkVENojItyLS3a1YjDHGBMbNlsIrwIBKXr8Q6Oh7jAVedDEWY4wxAXBtnoKqfioiyZUcMhR4TZ31QJeJSEMRaamqv7gVU9jY+xP8+yYo3Od1JMaYCPDlhvokJZZy8tk9odfDrp7Ly8lrrYGtftuZvn2/SwoiMhanNUG7du1CEpxrVGHhDbB5gdeRGGPCnCrc889+PP3JqZzc8le+PO57El0+p5dJQSrYpxUdqKqpQCpASkpKhcdEhB/+Dp/cBfu2QM0GMOjvkFDT66iMMWFKADZugE+30n/oaZT06BHVSSETaOu33Qb42aNYQmPVFCchAJw9CZL7eRuPMSbsZGfnk5GRRffuLQGY+MwZDBuzq2zbbV4mhXRgvIi8BZwG5ETFeMKOb+DjW6Aot4LXVjn/9p0OXW8IbVzGmLD33nvfM27cP4mLE9auvYkGDZKoVSsxZAkBXEwKIvImcC7QVEQygQngtHxUdTowD7gI2ADkAde6FUtIfTcTtn1+6NfjEuDogaGLxxgT9nbsyOXWW+cze/ZaAE4/vQ3Z2fk0aJAU8ljcvPvoyipeV+Bmt87vmeyNzr/nPgNtzvn963VbQZ0WoY3JGBOWVJVZs9Zw223/Ys+eA9Sunchjj53P+PE9iY/3Zm5xxJXODns5Gc6/bc+H5l29jcUYE9bGjfsnL730FQB9+x5NauogOnRo5GlMVuYimLQUcjY5zxse7W0sxpiwd/HFx9OwYRJpaUP48MOrPU8IYC2F4Nr/M5QUQO3mUKOe19EYY8LMjz/u5qOPNnHjjSkADBhwLJs33+bJ2MGhWFIIpoPjCQ2slWCM+U1xcSnPPLOUCRMWU1BQTLduLTj99DYAYZUQwJJCcB1MCg2P8TYOY0zYWL16O6NHp/PVV84d9yNHdqVjx8YeR3VolhSC6eAgs7UUjIl5BQXFPProp0yatITi4lLatWvASy8NYsCAY70OrVKWFILJWgrGGJ/77vuIZ59dBsDNN/fgL3/pQ7164V/WxpJCMJW1FCwpGBPr/vznM1m6NJMnnujL2We39zqcgNktqcFU1lKw7iNjYs3ChRu59NK3KS4uBaBFi7p88cV1EZUQwJJC8BTkQP5uSKgFdUJXp8QY462srAOMHv0e/fv/jblz1/PXv64qe02komLQ4c26j4Il22+QOQL/Ixhjqu+dd9Zz003z2L59PzVrxjNhwjmMGtXN67COiCWFYMmxOQrGxIrt2/dzyy3zmTNnHQC9erUlLW0Ixx/f1OPIjpwlhWA52FKwO4+MiXrvvfc9c+aso06dRCZN6stNN/UgLi46eggsKQSLtRSMiWr5+cUkJTl/MseMOZWMjCzGjetBcnJDjyMLLhtoDhabo2BMVCotVaZO/ZIOHSbz00/ZAMTFCY8/3i/qEgJYUgieHOs+Miba/PDDLnr3/iu33DKf7dv38+ab33kdkuus+ygYSopg7xZAoH6y19EYY45QUVEJTz31BRMnfkJBQQlHHVWHadMGcsklJ3gdmussKQTD3p9AS6BeW0gI/2nsxphD++67HYwc+Q6rVm0H4Npru/H00/1p1KiWx5GFhiWFYLBCeMZEjdJSZc2aHbRv34DU1MH07x9bXcKWFILBBpmNiWhr1+6gc+dmiAgnn3wU7703jN6921O3bg2vQws5G2gOBhtkNiYi7dtXwPjx8+jS5UX+8Y/1ZfsvuqhjTCYEsJZCcNiKa8ZEnAULNjB27Ads2ZJDQkIcmzdnex1SWLCkEAw51n1kTKTYs+cAt9++gNdeWw1A9+4tSUsbQrduLTyOLDxYUjhSqv9dDM8YE7a++WY7Awb8jV9/zaVmzXgmTjyXO+/sRUKC9aQfZEnhSB3YCUX7oWYDSArfdVeNMdCpUxPq1q1Bp05NmDFjCJ06NfE6pLBjSeFIZfuttmYls40JK6rKG2+sYfDg46hfvya1ayeyePEoWrWqFzUF7ILN2kxHKsdWWzMmHG3enM0FF/yNq69+h3vv/XfZ/jZt6ltCqIS1FI5U2Z1HNshsTDgoKSll2rQV3HffR+TmFtG4cS169WrrdVgRw5LCkbI5CsaEjfXrdzJ6dDpLl2YCcPnlJ/L88xfSvHkdjyOLHJYUjpTNUTAmLGzalEW3bi9RWFhCy5Z1mTZtIBdffLzXYUUcV5OCiAwAJgPxwAxVnVTu9XbAq0BD3zH3quo8N2MKOmspGBMWOnRoxB//2JmkpASeeqo/DRsmeR1SRHItKYhIPPAC0A/IBFaISLqqrvM77AHgbVV9UUQ6A/OAZLdiCrqiA7D/Z4hLgHptvI7GmJhy4EARjzzyCX/4wwn07NkagFdfvZj4eLt/5ki4+d3rCWxQ1QxVLQTeAoaWO0aB+r7nDYCfXYwn+A62EuonO4nBGBMSn332E926vcSkSUsYO/Z9SksVwBJCELj5l6w1sNVvOxM4rdwxDwMfisgtQB2gr4vxBJ91HRkTUnv3FnDfff9m2rSVAHTu3Izp0wfZLaZB5GZareinpOW2rwReUdU2wEXA6yLyu5hEZKyIrBSRlTt37nQh1MNkg8zGhMy8eT/Spcs0pk1bSUJCHA891Juvvx7L6adb120wudlSyAT8bw5uw++7h0YDAwBUdamIJAFNgR3+B6lqKpAKkJKSUj6xeMfWUTAmJHJy8hk+fC7Z2fmkpLQiLW0IJ598lNdhRSU3k8IKoKOIdAC2AcOAq8odswXoA7wiIicASUAYNQWqYCuuGeMaVUUV4uKEBg2SmDJlAL/+msuf/nS6FbBzkWtJQVWLRWQ8sADndtOZqrpWRB4BVqpqOnAn8LKI3I7TtTRKVcOnJVAVaykY44qff97HTTf9k7PPbsedd/YCYMSIrh5HFRtcvWXGN+dgXrl9D/k9Xwec6WYMrtFS2LvJeW4tBWOCQlWZOXMVd975ITk5BSxblslNN/WgVq1Er0OLGXYf5eHatw1KCqF2c6hR1+tojIl4GRlZjBnzPh9/7HzYGjiwI9OnD7KEEGKWFA5XjhXCMyYYSkpKmTJlOfff/zEHDhTTtGltpkwZwLBhXRArRx9ylhQOV7bNUTAmWObMWc+BA8VceWUXJk8eQLNmVsDOK5YUDleOzVEw5nAVFpawb18BTZrUJj4+jrS0Ifz4424GDz7O69Bint3XdbispWDMYVmxYhspKamMGPEOB282PP74ppYQwoS1FA6XtRSMqZa8vCImTFjEM88so7RUycsrYseOXI46ym7UCCeWFA6XzVEwJmCLF29mzJj32bBhD3Fxwl13ncHEiedRu7bdWRRuLCkcjvxsyN8DCbWgTguvozEmbKkqt946n6lTVwBw0knNSUsbQo8erT2OzByKJYXD4V/ewm6ZM+aQRIT69WuSmBjHAw/05t57z6JGjXivwzKVsKRwOKxktjGHtGtXHhs37uG005zqpQ8+eA7Dh59M587NPI7MBCKgu49EpIaIHOt2MBHDSmYb8zuqyltvfccJJ7zAxRfPJivrAABJSQmWECJIlUlBRAYCa4CFvu1uIvKO24GFNRtkNua/ZGbuZejQt7jyyn+wa1cenTs3Iy+vyOuwzGEIpPvoEZwV0xYBqOo3Md9qsO4jYwAoLVVmzPiau+9eyN69BdSvX5Onn+7P6NGnWImKCBVIUihS1exyP+DIKW/tBus+MgaA0aPTeeWVbwAYMuQ4pk27iNat61fxVSacBTKmsF5ELgfiRKSDiDwHLHM5rvBVUgj7tgAC9ZO9jsYYT1199Uk0b16Ht966lHffvcISQhQIJCmMB04FSoG5QD5wm5tBhbW9W5y1FOq1hYSaXkdjTEh9990OJk/+7TNhnz5Hk5FxK1dcYRVNo0Ug3UcXqOo9wD0Hd4jIJTgJIvYcLG/R0LqOTOwoKCjmL3/5nMce+4yiolJSUlpx5pntAKhTp4bH0ZlgCqSl8EAF++4PdiAR42AhPFtHwcSI5cszOfXUVCZO/ISiolLGjUvhpJOO8jos45JDthRE5AJgANBaRJ7xe6k+TldSbMq2loKJDbm5hTz44CKee24ZqtCxY2NmzBhC797tvQ7NuKiy7qMdwHc4Ywhr/fbvA+51M6iwZiuumRhx//0fM3nycuLihLvvPoOHHz7XlsaMAYdMCqq6ClglIrNUNT+EMYU3m6NgYsT995/NmjU7ePzxvqSktPI6HBMigYwptBaRt0TkWxH5z8GH65GFI1Wbo2CiVnr6D1x00SyKikoAaNasDh99NNISQowJJCm8AvwVEOBC4G3gLRdjCl8HdkJRLtRsCLUaex2NMUGxY0cuw4bNYejQt5g/fwOvvrra65CMhwJJCrVVdQGAqm5U1QeA89wNK0xZK8FEEVXlb3/7lhNOeIHZs9dSu3YikycP4Npru3kdmvFQIPMUCsSZlbJRRG4EtgHN3Q0rTFkhPBMltmzJ4cYbP2D+/A0A9O17NKmpg+jQoZHHkRmvBZIUbgfqArcC/wc0AK5zM6iwZYPMJkp8+OFG5s/fQMOGSTzzTH9GjepmM5INEEBSUNXlvqf7gBEAItLGzaDClnUfmQiWm1tYNvt49OhT2LZtL2PHnkrLlvU8jsyEk0rHFESkh4hcLCJNfdsnishrxGpBPGspmAhUXFzKE08soX3758jIyAKcZTInTDjXEoL5nUMmBRH5CzALGA78S0Tux1lTYTXQKTThhRkbUzARZvXq7Zx22gzuueff7N59gHff/d7rkEyYq6z7aCjQVVUPiEhj4Gff9g+hCS3MFOVB7i8Qlwh1Y7P3zESOgoJiHn30UyZNWkJxcSnt2jUgNXUQF1wQ2+tjmapVlhTyVfUAgKruEZHvYzYhAORscv5tkAxx8Z6GYkxlVq36heHD57J+/S5EYPz4Hjz2WB/q1bNS76ZqlSWFo0XkYHlsAZL9tlHVS6p6cxEZAEwG4oEZqjqpgmMuBx7GWc1ttapeFXj4IWSDzCZC1KyZwMaNWRx3XBNmzBjCWWe18zokE0EqSwqXltueWp03FpF44AWgH5AJrBCRdFVd53dMR+A+4ExVzRKR8J3/kGMls034+vrrXzjllBaICJ07N2P+/OH06tWWpKRA7jo35jeVFcT76AjfuyewQVUzAETkLZxxinV+x4wBXlDVLN85dxzhOd1jg8wmDGVlHeCuuz5k5sxvePPNSxk2rAsA55/fwePITKRy82NEa2Cr33YmcFq5YzoBiMgSnC6mh1X1X+XfSETGAmMB2rXzqCmcY91HJry88856brppHtu376dmzXh2787zOiQTBdxMChVNj9QKzt8ROBdoA3wmIl1UNfu/vkg1FUgFSElJKf8eoZFtcxRMeNi+fT+33DKfOXOcRveZZ7ZlxowhHH98U48jM9Eg4KQgIjVVtaAa750JtPXbboNzW2v5Y5apahGwSUR+wEkSK6pxHveVlsDeg3cfWbPceOerr36mX7/XycrKp06dRCZN6stNN/UgLs5KVJjgqLJKqoj0FJE1wI++7a4i8nwA770C6CgiHUSkBjAMSC93zLv4Kq76Zk13AjKqEX9o7N8GJYVQ+yioUdfraEwM69y5Gc2a1eGCC45h7dqbGD++pyUEE1SBlM6eAgwCdgOo6moCKJ2tqsXAeGABsB54W1XXisgjIjLEd9gCYLeIrMOZLX23qu6u/mW4zMpbGI+UliqpqV+Rne0sflirViKffjqK+fOH0759Q4+jM9EokO6jOFX9qVwFxZJA3lxV5wHzyu17yO+5Anf4HuHL5igYD/zwwy6uv/59Pv98CytWbOPll53PUkcdZa1V455AksJWEekJqG/uwS1AbC3HaS0FE0JFRSU8/fRSHn54MQUFJbRoUZcLL+zodVgmRgSSFMbhdCG1A34F/u3bFztsjoIJkVWrfmH06HRWrdoOwLXXduPpp/vTqFEtjyMzsSKQpFCsqsNcjyScWfeRCYGNG/fQs+cMiotLSU5uSGrqIPr1sw8iJrQCSQorfLeKzgbmquo+l2MKP9Z9ZELgmGMaM2LEydSrV4P/+78+1K1bw+uQTAyq8u4jVT0GeBQ4FVgjIu+KSOy0HPKzIX8PJNR2bkk1Jkj27y/k1lvns3TpbxP/09KGMHnyhZYQjGcCuSUVVf1CVW8FugN7cRbfiQ1lrYSjwdawNUGyYMEGTjxxGs8//yU33vhPnBvxsHWSjeeq7D4Skbo4heyGAScA7wG9XI4rfJSNJ1jXkTlye/Yc4PbbF/Daa6sBOPXUlqSlDbFkYMJGIGMK3wHvA0+o6mcuxxN+yu48skFmc2TmzFnHzTfPY8eOXJKSEpg48VzuuOMMEhICarAbExKBJIWjVbXU9UjCla2jYIIgOzufsWPfJysrn9692/Pyy4Pp1KmJ12EZ8zuHTAoi8rSq3gn8Q0R+V5k0kJXXokKOtRTM4VFVSkuV+Pg4GjZMYtq0gWRlHeCGG1KsXpEJW5W1FGb7/q3WimtRJ9taCqb6Nm/OZuzY9zn//A7ce+9ZAGUL4BgTzg7ZmamqX/qenqCqH/k/cAaco19JIezbAgg0SPY6GhMBSkpKmTJlOV26TGPhwgymTv2S/Pxir8MyJmCBjHBdV8G+0cEOJCzt/Qm0FOq1hXi7b9xUbv36nfTu/Qq33fYvcnOLGDasC19/fYOtk2wiSmVjClfg3IbaQUTm+r1UD8iu+KuijM1kNgEoLi7l8cc/55FHPqWwsIRWrerx4osDGTLkOK9DM6baKvsI8yXOGgptgBf89u8DVrkZVNiwQngmAHFxwocfZlBYWMKYMd154ol+NGyY5HVYxhyWQyYFVd0EbMKpihqbrBCeOYQDB4rYt6+Q5s3rEBcnzJgxmK1b93L++bZcq4lshxxTEJFPfP9micgev0eWiOwJXYgesu4jU4FPP/2Jrl2nc/XVc8vKU3Ts2MQSgokKlXUfHVxys2koAglL1lIwfvbuLeC++/7NtGkrAUhMjGfXrjyaNavjcWTGBE9lt6QenMXcFohX1RLgDOAGIPp/C1StpWDKzJ//I126TGPatJUkJMQxYcI5fP31WEsIJuoEcq/cu0APETkGeA34J/AGMMjNwDyXtwOKciGpkfMwMUlVGTPmfdLSnHsrUlJaMXPmEE46ycqom+gUyDyFUlUtAi4BnlPVW4DW7oYVBqzryOCUsm7Tpj5JSQk89VQ/li4dbQnBRLWAluMUkT8CI4CLffsS3QspTFghvJj188/72LhxD2ef3R6A//mfsxkx4mSOOaaxx5EZ475AZzSfh1M6O0NEOgBvuhtWGLCS2TFHVUlL+5rOnV/g0kvfZvfuPABq1Ii3hGBiRpUtBVX9TkRuBY4VkeOBDar6f+6H5jFrKcSUjIwsxox5n48/3gTAoEGdKCqK3YrxJnYFsvLa2cDrwDZAgBYiMkJVl7gdnKdsNnNMOFjA7oEHFpGXV0TTprWZMmUAw4Z1sdXQTEwKZEzhWeAiVV0HICIn4CSJFDcD85ytoxATRo58lzfeWAPAVVedxHPPXWC3mZqYFsiYQo2DCQFAVdcD0V0ytCgPcrdDXCLUbeN1NMZFY8Z0p02b+qSnD2PWrEssIZiYF0hL4WsReQmndQAwnGgviFc2ntAB4uK9jcUE1YoV2/j4403cc4+z8M255yazYcMt1Kxp5a2NgcCSwo3ArcCfccYUPgWedzMoz5WttmZdR9EiL6+ICRMW8cwzyygtVXr1alt2y6klBGN+U+lvg4icBBwDvKOqT4QmpDCQY4PM0WTx4s1cf306GzdmERcn3HXXGZx6aiuvwzImLFVWJfV/cEpcDAcWikhFK7BVSkQGiMgPIrJBRO6t5LjLRERFJDwGr202c1TIycnnhhve57zzXmXjxixOOqk5y5aN5skn+1O7dvTPvzTmcFTWUhgOnKyquSLSDJgHzAz0jUUkHmdxnn5AJrBCRNL9B619x9XD6Z5aXt3gXWOF8KLCgw8uIjX1axIT43jwwd7cc89Z1KhhY0TGVKaypFCgqrkAqrpTRAK5U8lfT5yJbhkAIvIWMBRYV+64/wWeAO6q5vu7x+YoRCxVLZtf8NBD57BpUzaTJvXhxBObexyZMZGhsj/0R4vIXN/jHeAYv+25lXzdQa2BrX7bmZQrpCcipwBtVfWDakfultISyHFmtdLAFk2JFKrKG2+s4fzzX6OwsASApk1r8/77V1pCMKYaKmspXFpue2o137ui6aBa9qLT8ngWGFXlG4mMBcYCtGvXrpphVNP+bVBaBHVaQKLdsx4JMjP3Mm7cP/ngg/8AMGvWt1x77SkeR2VMZKpsjeaPjvC9M3EW6DmoDfCz33Y9oAuw2NfcbwGki8gQVV1ZLpZUIBUgJSVFcZMNMkeM0lLl5Ze/4u67F7JvXyENGtTk6af7M2pUN69DMyZiuXmD9gqgo6+q6jZgGHDVwRdVNQe/pT5FZDFwV/mEEHI2yBwRNmzYw5gx77N48WYAhg49jmnTBtKqVT1vAzMmwrmWFFS1WETGAwuAeGCmqq4VkUeAlaqa7ta5j0hZS8GSQjj77LOfWLx4M82b12Hq1Au57LLOVsDOmCAIOCmISE1VLajOm6vqPJxbWf33PXSIY8+tznu7xtZRCFvZ2fk0bJgEwKhR3di5M4/Ro0+hSZPaHkdmTPSo8jZTEekpImuAH33bXUUkestc2DoKYaegoJgJExbRvv1z/PjjbsBZJvPPfz7TEoIxQRZIS2EKMAhndjOqulpEznM1Ki9ZiYuwsmxZJqNHp7Nu3U4AFizYSMeOTTyOypjoFUhSiFPVn8r115a4FI+38rOcR2IdqG33tnspN7eQBx9cxHPPLUMVOnZsTFrakLIidsYYdwSSFLaKSE9AfaUrbgH+425YHsnxq45qg5aeWb48k6uumktGRhbx8cJdd/ViwoRzqFXL6hUZ47ZAksI4nC6kdsCvwL99+6KPzVEICw0bJrFt2166dj2KtLQhVtHUmBCqMimo6g6cOQbRL9vmKHjl88+3cOaZbRERjjuuKR9/fA09erQiMdEK2BkTSlUmBRF5Gb/yFAep6lhXIvKSDTKH3I4dudx663xmz17Lq69ezMiRXQHo1attFV9pjHFDIN1H//Z7ngT8gf8udBc9cmzFtVBRVWbNWsNtt/2LPXsOULt2YlkhO2OMdwLpPprtvy0irwMLXYvIS1YyOyS2bMnhxhs/YP78DQD063c0qamDSU5u6HFkxpjDKXPRAYi++wJLCmHfVpA4qB99lxculi/PpG/f19m/v5CGDZN49tkLuOaarlaiwpgwEciYQha/jSnEAXuAQy6tGbH2/gRa6iSE+BpeRxO1unVrQdu29Tn++Ka88MJFtGxpBeyMCSeVJgVxPr51xalyClCqqu6WrvaKdR25ori4lKlTv2TkyK40blyLmjUTWLLkOho1quV1aMaYClRa+8iXAN5R1RLfIzoTAtgcBResXr2d006bwe23L+COOxaU7beEYEz4CmTd5S9FpLvrkXjNCuEFTX5+MQ888DEpKS/z9de/0K5dA668sovXYRljAnDI7iMRSVDVYuAsYIyIbARycZbZVFWNrkRh3UdB8cUXWxk9Op3vv9+FCIwf34PHHutDvXo1vQ7NGBOAysYUvgS6AxeHKBZvla24Zt1Hh2vDhj2cffZfKS1VjjuuCWlpQzjzTJfX1DbGBFVlSUEAVHVjiGLxjqp1HwXBscc2ZuzY7jRuXIsHHzyHpCQ3V3s1xrihst/aZiJyx6FeVNVnXIjHG3m/QlEuJDWCJJtAFaisrAPceeeHXHttt7KS1tOmDbQ5B8ZEsMqSQjxQF1+LIaplWyuhuubOXc/NN89j+/b9fPXVL3zzzQ2IiCUEYyJcZUnhF1V9JGSReMkK4QVs+/b9jB8/j3/8Yz0AZ53VjhkzBlsyMCZKVDmmEBOyrRBeVVSV115bze23LyArK5+6dWvw+ON9ufHGFOLiYue/ijHRrrKk0CdkUXjNWgpVys7O5847PyQrK58BA45l+vSBtG9v4y/GRJtDJgVV3RPKQDxls5krVFqqlJYqCQlxNGpUi5deGkReXhFXX32ydRcZE6UCmdEc/XJsxbXyvv9+F717/5VJkz4v23fppZ0ZMcIqmhoTzSwpFOVC7nanMmrd1l5H47miohIee+wzunadzpIlW0lLW0V+frHXYRljQsRmF+Vscv6tnwxxsb0e8KpVv3Dddel88812AEaPPoUnn+xnk9CMiSH22241jygqKmHChMU88cQSSkqU5OSGvPzyYPr2tTEWY2KNJYWyQebYTQoJCXEsX76N0lLltttO49FHz6duXVtoyJhYZEkhRgvh7dtXwL59hbRqVQ8RYcaMwWzfvp8zzmjrdWjGGA/ZQHMMthQWLNhAly4vMnz4XA6um9ShQyNLCMYYSwqx1FLYvTuPa655lwEDZrFlSw779hWwe/cBr8MyxoQRV5OCiAwQkR9EZIOI3FvB63eIyDoR+VZEPhKR9m7G8zulJb/dfRTFE9dUlTlz1tG58zRee201SUkJPPFEX5Ytu56mTWt7HZ4xJoy4NqYgIvHAC0A/IBNYISLpqrrO77BVQIqq5onIOOAJ4Aq3Yvqd/ZlQWgR1WkJidP5xVFWGD5/Lm29+B0DRZJGNAAAPYElEQVTv3u15+eXBdOrUxOPIjDHhyM2WQk9gg6pmqGoh8BYw1P8AVV2kqnm+zWVAGxfj+b0YKIQnInTu3Ix69Wrw4osDWbToGksIxphDcvPuo9bAVr/tTOC0So4fDcyv6AURGQuMBWjXLojLO0bpHIVNm7LIyMiiTx8n2d1zz5mMGtWNNm3qexyZMSbcudlSqKhAjlZ4oMjVQArwZEWvq2qqqqaoakqzZs2CF2FOdBXCKykpZfLkZXTp8iJXXDGHHTtyAUhMjLeEYIwJiJsthUzA/x7HNsDP5Q8Skb7A/cA5qlrgYjy/lx09hfDWrdvJ9dens3RpJgBDhhxn6xwYY6rNzaSwAugoIh2AbcAw4Cr/A0TkFOAlYICq7nAxlopFwToKRUUlPP74Ev73fz+lsLCEVq3q8eKLAxky5DivQzPGRCDXkoKqFovIeGABznrPM1V1rYg8AqxU1XSc7qK6wN995Zi3qOoQt2L6nZzIH2i+6qq5zJnj3NA1Zkx3nnyyHw0aJHkclTEmUrla5kJV5wHzyu17yO95XzfPX6n8LOeRWAdqN/csjCN1222n8c0323nppUGcf34Hr8MxxkS42J3R7L/aWgQtGvPJJ5uZOHFx2fZZZ7Vj/fqbLSEYY4IidgviRdhqa3v3FnDPPQuZPv0rAM47rwO9ezsTwBMSYje3G2OCK3aTQgQVwps370duuOEDMjP3kpgYx/33n83pp4d2np8xJjbEblKIgEJ4u3bl8ac//YtZs9YA0LNna9LShtClS+SOgRhjwlvsJoUImM38yCOfMGvWGmrVSuDRR8/ntttOIz7euoqMMe6xpBBm3Ueqiu/2XCZOPJdff83lscfO55hjGnscmTEmFsTmx86SQti3FSQO6gexltIRUFVefvkrevWaSX5+MQCNGtVi9uzLLCEYY0ImNpNCzmZAoV47iPd+LeKNG/fQp89rjB37AcuWZfL222u9DskYE6Nis/soTAaZnQJ2y3nggY85cKCYZs1q8/zzF3L55Sd6GpcxJnbFZlIIg/GEtWt3cN116Xz55TYAhg8/ieeeG2AroRljPBWbSSEMCuGtWrWdL7/cRuvW9XjppUEMHNjJs1iMMeag2EwKHq24tnNnLs2a1QGclkF2dj4jRpxsBeyMMWEjRgeaQ9tSyMsr4q67PiQ5eTLr1+8EnGUyx4/vaQnBGBNWYq+loBrSlsKiRZsYM+Z9Nm7MIi5O+PTTnzjhhCCuHmeMMUEUe0kh71cozoOkxpDU0LXT5OTk8+c/LyQ19WsATjqpOTNnDiUlpZVr5zTGmCMVe0khBOUtPv98C8OGzWHbtn0kJsbx4IO9ueees6hRI961cxpjTDDEXlIIwWprLVrUZffuA5x+ehtmzBjMiSdaATtjTGSIvaTgQktBVVm4MIN+/Y5GRDj22MZ8/vm1dOvWwgrYGWMiSuz9xQryxLWtW3MYPPhNLrjgb/z1r9+U7T/11FaWEIwxESf2WgpBKnFRWuoUsLv77oXs21dIgwY1qVnTxgyMMZEt9pJCEFoKP/64mzFj3ueTT34C4OKLj+eFFy6iVat6wYjQGGM8E1tJoSjXuSU1vgbUPbxbQ7/4Yit9+rxGfn4xzZvXYerUC7nsss5layAYY0wki62kcHDSWv0OEHd4XT0pKa3o2LExp5zSkmee6U+TJlbAzhgTPWIsKVT/zqOCgmKeeuoLbrghhaZNa1OjRjxLllxHvXo1XQrSGGO8E1tJoZpzFJYty2T06HTWrdvJ+vW7+NvfLgGwhGCMiVqxlRQCbCnk5hbywAMfM3nyclShU6cm3HDDqSEI0BhjvBVbSSGAlsJHH2UwZsz7bNqUTXy8cPfdvZgw4VySkmLrW2WMiU2x9ZeuipLZ//nPbvr1ex1V6NatBWlpQ+jevWUIAzTGGG/FTlIoLYGczc7zBh0qPKRTpybcdttpNGtWh7vv7kViok1GM8bElthJCvszobQI6rSEROc20l9/3c+tt/6LG288lfPOcxLFs88O8DJKY4zxlKvFeURkgIj8ICIbROTeCl6vKSKzfa8vF5Fk14LxG2RWVV5/fTWdO0/j7bfXcvvtC1BV105tjDGRwrWkICLxwAvAhUBn4EoR6VzusNFAlqoeCzwLPO5WPAeTwpbCTgwc+AYjR77Lnj0H6N//GN59d5jNSDbGGNztPuoJbFDVDAAReQsYCqzzO2Yo8LDv+RxgqoiIuvCxvTQrg+lLenDPgnbsz9tAo0ZJPPvsBYwc2dUSgjHG+LiZFFoDW/22M4HTDnWMqhaLSA7QBNgV7GByft7ExIXnsD8PLr30BKZOvYgWLeoG+zTGGBPR3EwKFX38Lt8CCOQYRGQsMBagXbt2hxVMo3rCjCs/pLDHRC4dc/FhvYcxxkQ7N5NCJtDWb7sN8PMhjskUkQSgAbCn/BupaiqQCpCSknJ4XUuD3mTwQKWCnGOMMcbHzbuPVgAdRaSDiNQAhgHp5Y5JB67xPb8M+NiN8YQyIiC2GpoxxhyKay0F3xjBeGABEA/MVNW1IvIIsFJV04E04HUR2YDTQhjmVjzGGGOq5urkNVWdB8wrt+8hv+f5wB/djMEYY0zgrC/FGGNMGUsKxhhjylhSMMYYU8aSgjHGmDKWFIwxxpSRSKsOKiI7gZ8O88ub4kIJjTBn1xwb7Jpjw5Fcc3tVbVbVQRGXFI6EiKxU1RSv4wglu+bYYNccG0JxzdZ9ZIwxpowlBWOMMWViLSmkeh2AB+yaY4Ndc2xw/ZpjakzBGGNM5WKtpWCMMaYSUZkURGSAiPwgIhtE5N4KXq8pIrN9ry8XkeTQRxlcAVzzHSKyTkS+FZGPRKS9F3EGU1XX7HfcZSKiIhLxd6oEcs0icrnvZ71WRN4IdYzBFsD/7XYiskhEVvn+f1/kRZzBIiIzRWSHiHx3iNdFRKb4vh/fikj3oAagqlH1wCnTvRE4GqgBrAY6lzvmJmC67/kwYLbXcYfgms8Davuej4uFa/YdVw/4FFgGpHgddwh+zh2BVUAj33Zzr+MOwTWnAuN8zzsDm72O+wivuTfQHfjuEK9fBMzHWbnydGB5MM8fjS2FnsAGVc1Q1ULgLWBouWOGAq/6ns8B+ohIRUuDRooqr1lVF6lqnm9zGc5KeJEskJ8zwP8CTwD5oQzOJYFc8xjgBVXNAlDVHSGOMdgCuWYF6vueN+D3KzxGFFX9lApWoPQzFHhNHcuAhiLSMljnj8ak0BrY6red6dtX4TGqWgzkAE1CEp07Arlmf6NxPmlEsiqvWUROAdqq6gehDMxFgfycOwGdRGSJiCwTkQEhi84dgVzzw8DVIpKJs37LLaEJzTPV/X2vFlcX2fFIRZ/4y99iFcgxkSTg6xGRq4EU4BxXI3JfpdcsInHAs8CoUAUUAoH8nBNwupDOxWkNfiYiXVQ12+XY3BLINV8JvKKqT4vIGTirOXZR1VL3w/OEq3+/orGlkAm09dtuw++bk2XHiEgCTpOzsuZauAvkmhGRvsD9wBBVLQhRbG6p6prrAV2AxSKyGafvNT3CB5sD/b/9nqoWqeom4AecJBGpArnm0cDbAKq6FEjCqREUrQL6fT9c0ZgUVgAdRaSDiNTAGUhOL3dMOnCN7/llwMfqG8GJUFVes68r5SWchBDp/cxQxTWrao6qNlXVZFVNxhlHGaKqK70JNygC+b/9Ls5NBYhIU5zupIyQRhlcgVzzFqAPgIicgJMUdoY0ytBKB0b67kI6HchR1V+C9eZR132kqsUiMh5YgHPnwkxVXSsijwArVTUdSMNpYm7AaSEM8y7iIxfgNT8J1AX+7htT36KqQzwL+ggFeM1RJcBrXgD0F5F1QAlwt6ru9i7qIxPgNd8JvCwit+N0o4yK5A95IvImTvdfU984yQQgEUBVp+OMm1wEbADygGuDev4I/t4ZY4wJsmjsPjLGGHOYLCkYY4wpY0nBGGNMGUsKxhhjylhSMMYYU8aSggk7IlIiIt/4PZIrOTb5UNUkq3nOxb5KnKt9JSKOO4z3uFFERvqejxKRVn6vzRCRzkGOc4WIdAvga/4kIrWP9NwmNlhSMOHogKp283tsDtF5h6tqV5xiiU9W94tVdbqqvubbHAW08nvtelVdF5Qof4tzGoHF+SfAkoIJiCUFExF8LYLPRORr36NXBcecKCJf+loX34pIR9/+q/32vyQi8VWc7lPgWN/X9vHV6V/jq3Nf07d/kvy2PsVTvn0Pi8hdInIZTn2pWb5z1vJ9wk8RkXEi8oRfzKNE5PnDjHMpfoXQRORFEVkpzjoKE337bsVJTotEZJFvX38RWer7Pv5dROpWcR4TQywpmHBUy6/r6B3fvh1AP1XtDlwBTKng624EJqtqN5w/ypm+sgdXAGf69pcAw6s4/2BgjYgkAa8AV6jqSTgVAMaJSGPgD8CJqnoy8Kj/F6vqHGAlzif6bqp6wO/lOcAlfttXALMPM84BOGUtDrpfVVOAk4FzRORkVZ2CUxfnPFU9z1f64gGgr+97uRK4o4rzmBgSdWUuTFQ44PvD6C8RmOrrQy/BqelT3lLgfhFpA8xV1R9FpA9wKrDCV96jFk6CqcgsETkAbMYpv3wcsElV/+N7/VXgZmAqzvoMM0Tkn0DApblVdaeIZPhq1vzoO8cS3/tWJ846OGUf/FfdulxExuL8XrfEWXDm23Jfe7pv/xLfeWrgfN+MASwpmMhxO/Ar0BWnhfu7RXNU9Q0RWQ4MBBaIyPU4ZYZfVdX7AjjHcP+CeSJS4Robvno8PXGKsA0DxgPnV+NaZgOXA98D76iqivMXOuA4cVYgmwS8AFwiIh2Au4AeqpolIq/gFIYrT4CFqnplNeI1McS6j0ykaAD84quRPwLnU/J/EZGjgQxfl0k6TjfKR8BlItLcd0xjCXx96u+BZBE51rc9AvjE1wffQFXn4QziVnQH0D6c8t0VmQtcjLMOwGzfvmrFqapFON1Ap/u6nuoDuUCOiBwFXHiIWJYBZx68JhGpLSIVtbpMjLKkYCLFNOAaEVmG03WUW8ExVwDficg3wPE4Sxauw/nj+aGIfAssxOlaqZKq5uNUoPy7iKwBSoHpOH9gP/C93yc4rZjyXgGmHxxoLve+WcA6oL2qfunbV+04fWMVTwN3qepqnLWZ1wIzcbqkDkoF5ovIIlXdiXNn1Ju+8yzD+V4ZA1iVVGOMMX6spWCMMaaMJQVjjDFlLCkYY4wpY0nBGGNMGUsKxhhjylhSMMYYU8aSgjHGmDKWFIwxxpT5f6ECaWxsd/bDAAAAAElFTkSuQmCC\n",
        "text/plain": "<Figure size 432x288 with 1 Axes>"
       },
       "metadata": {
        "needs_background": "light"
       },
       "output_type": "display_data"
      }
     ],
     "persistentId": "2c810e70-a822-4d9a-a9b0-2622dfa247bf",
     "text": "y_true = coding_scores_not_low_complexity_unspliced['in_cds_no_stop_codon']\ny_score = coding_scores_not_low_complexity_unspliced['jaccard_in_peptide_db']\ny_pred = y_score > 0.5\n\nfpr, tpr, thresholds = metrics.roc_curve(y_true, y_score)\n\nfig, ax = plt.subplots()\nlw = 2\nax.plot(fpr, tpr, lw=2, color='darkorange')\nax.plot([0, 1], [0, 1], color='navy', lw=lw, linestyle='--')\nax.set(xlabel='False Positive Rate', ylabel='True Positive Rate')\n\nprint(\"accuracy score:\", metrics.accuracy_score(y_true, y_pred))\n\nprint(metrics.classification_report(y_true, y_pred))"
    },
    "executionTime": "2019-10-12T04:05:13.744Z"
   },
   {
    "cell": {
     "executionCount": 79,
     "executionEventId": "ea1934af-1168-4f39-bf27-9cb729981731",
     "hasError": true,
     "id": "9c593919-2f4d-44c4-aaa8-0ae119337efd",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "total 360M\ndrwxrwxr-x 2 olga olga 4.0K Oct 11 20:58 .\ndrwxrwxr-x 3 olga olga 4.0K Sep 10 15:07 ..\n-rw-rw-r-- 1 olga olga  52M Oct 11 18:07 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.bam\n-rw-rw-r-- 1 olga olga 2.9M Oct 11 18:08 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.bam.bai\n-rw-rw-r-- 1 olga olga 7.0M Oct 11 20:58 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.out_CDS_without_stop_codon.bam\n-rw-rw-r-- 1 olga olga 6.1M Oct 11 18:08 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.out_CDS_without_stop_codon_or_utr.bam\n-rw-rw-r-- 1 olga olga 1.8M Oct 11 18:08 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.out_CDS_without_stop_codon_or_utr.bam.bai\n-rw-rw-r-- 1 olga olga 1.7M Oct 11 18:08 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.out_CDS_without_stop_codon_or_utr.read_ids.txt\n-rw-rw-r-- 1 olga olga  15M Oct 11 18:08 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.read_ids.txt\n-rw-rw-r-- 1 olga olga  15M Sep 10 15:55 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out_CDS_without_stop_codon.bam\n-rw-rw-r-- 1 olga olga 1.7M Oct 11 18:08 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out_CDS_without_stop_codon.bam.bai\n-rw-rw-r-- 1 olga olga  15M Sep 18 07:15 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out_CDS_without_stop_codon_or_utr.bam\n-rw-rw-r-- 1 olga olga 1.7M Oct 11 18:08 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out_CDS_without_stop_codon_or_utr.bam.bai\n-rw-rw-r-- 1 olga olga 2.8M Sep 18 10:14 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out_CDS_without_stop_codon_or_utr.read_ids.txt\n-rw-rw-r-- 1 olga olga 4.1M Sep 10 15:56 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out_CDS_without_stop_codon.read_ids.txt\n-rw-r--r-- 1 olga olga 235M Sep 10 15:10 SRR306838_GSM752691_hsa_br_F_1_trimmed_k15.tsv\n"
      },
      {
       "ename": "FileNotFoundError",
       "evalue": "[Errno 2] File b'/home/olga/data_sm/kmer-hashing/brawand2011/human//SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.out_CDS_without_stop_codon.read_ids.txt' does not exist: b'/home/olga/data_sm/kmer-hashing/brawand2011/human//SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.out_CDS_without_stop_codon.read_ids.txt'",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
        "\u001b[0;32m<ipython-input-79-ed85261acd20>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      5\u001b[0m \u001b[0;31m#                                    header=None, squeeze=True)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      6\u001b[0m coding_read_ids = pd.read_csv(f\"{folder}/SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.out_CDS_without_stop_codon.read_ids.txt\", \n\u001b[0;32m----> 7\u001b[0;31m                                    header=None, squeeze=True)\n\u001b[0m\u001b[1;32m      8\u001b[0m \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcoding_read_ids\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshape\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      9\u001b[0m \u001b[0mcoding_read_ids\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mhead\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/io/parsers.py\u001b[0m in \u001b[0;36mparser_f\u001b[0;34m(filepath_or_buffer, sep, delimiter, header, names, index_col, usecols, squeeze, prefix, mangle_dupe_cols, dtype, engine, converters, true_values, false_values, skipinitialspace, skiprows, skipfooter, nrows, na_values, keep_default_na, na_filter, verbose, skip_blank_lines, parse_dates, infer_datetime_format, keep_date_col, date_parser, dayfirst, iterator, chunksize, compression, thousands, decimal, lineterminator, quotechar, quoting, doublequote, escapechar, comment, encoding, dialect, tupleize_cols, error_bad_lines, warn_bad_lines, delim_whitespace, low_memory, memory_map, float_precision)\u001b[0m\n\u001b[1;32m    700\u001b[0m                     skip_blank_lines=skip_blank_lines)\n\u001b[1;32m    701\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 702\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0m_read\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfilepath_or_buffer\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkwds\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    703\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    704\u001b[0m     \u001b[0mparser_f\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__name__\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mname\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/io/parsers.py\u001b[0m in \u001b[0;36m_read\u001b[0;34m(filepath_or_buffer, kwds)\u001b[0m\n\u001b[1;32m    427\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    428\u001b[0m     \u001b[0;31m# Create the parser.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 429\u001b[0;31m     \u001b[0mparser\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mTextFileReader\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfilepath_or_buffer\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwds\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    430\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    431\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0mchunksize\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0miterator\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/io/parsers.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, f, engine, **kwds)\u001b[0m\n\u001b[1;32m    893\u001b[0m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0moptions\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'has_index_names'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mkwds\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'has_index_names'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    894\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 895\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_make_engine\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mengine\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    896\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    897\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mclose\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/io/parsers.py\u001b[0m in \u001b[0;36m_make_engine\u001b[0;34m(self, engine)\u001b[0m\n\u001b[1;32m   1120\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_make_engine\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mengine\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'c'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1121\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mengine\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m'c'\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1122\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_engine\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mCParserWrapper\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mf\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0moptions\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1123\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1124\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mengine\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m'python'\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/io/parsers.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, src, **kwds)\u001b[0m\n\u001b[1;32m   1851\u001b[0m         \u001b[0mkwds\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'usecols'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0musecols\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1852\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1853\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_reader\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mparsers\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mTextReader\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msrc\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwds\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1854\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0munnamed_cols\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_reader\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0munnamed_cols\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1855\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32mpandas/_libs/parsers.pyx\u001b[0m in \u001b[0;36mpandas._libs.parsers.TextReader.__cinit__\u001b[0;34m()\u001b[0m\n",
        "\u001b[0;32mpandas/_libs/parsers.pyx\u001b[0m in \u001b[0;36mpandas._libs.parsers.TextReader._setup_parser_source\u001b[0;34m()\u001b[0m\n",
        "\u001b[0;31mFileNotFoundError\u001b[0m: [Errno 2] File b'/home/olga/data_sm/kmer-hashing/brawand2011/human//SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.out_CDS_without_stop_codon.read_ids.txt' does not exist: b'/home/olga/data_sm/kmer-hashing/brawand2011/human//SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.out_CDS_without_stop_codon.read_ids.txt'"
       ]
      }
     ],
     "persistentId": "75c01782-427c-4c63-bc29-ed308ae4a142",
     "text": "folder = '/home/olga/data_sm/kmer-hashing/brawand2011/human/'\n! ls -lha $folder\n\n# coding_read_ids = pd.read_csv(f\"{folder}/SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.out_CDS_without_stop_codon_or_utr.read_ids.txt\", \n#                                    header=None, squeeze=True)\ncoding_read_ids = pd.read_csv(f\"{folder}/SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.out_CDS_without_stop_codon.read_ids.txt\", \n                                   header=None, squeeze=True)\nprint(coding_read_ids.shape)\ncoding_read_ids.head()"
    },
    "executionTime": "2019-10-12T04:10:48.849Z"
   },
   {
    "cell": {
     "executionCount": 80,
     "executionEventId": "66f46cd8-ecad-4a98-87c8-0d3c5da695da",
     "hasError": false,
     "id": "9c593919-2f4d-44c4-aaa8-0ae119337efd",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "total 362M\ndrwxrwxr-x 2 olga olga 4.0K Oct 11 21:11 .\ndrwxrwxr-x 3 olga olga 4.0K Sep 10 15:07 ..\n-rw-rw-r-- 1 olga olga  52M Oct 11 18:07 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.bam\n-rw-rw-r-- 1 olga olga 2.9M Oct 11 18:08 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.bam.bai\n-rw-rw-r-- 1 olga olga 1.9M Oct 11 21:11 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.CDS_without_stop_codon.read_ids.txt\n-rw-rw-r-- 1 olga olga 7.0M Oct 11 20:58 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.out_CDS_without_stop_codon.bam\n-rw-rw-r-- 1 olga olga 6.1M Oct 11 18:08 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.out_CDS_without_stop_codon_or_utr.bam\n-rw-rw-r-- 1 olga olga 1.8M Oct 11 18:08 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.out_CDS_without_stop_codon_or_utr.bam.bai\n-rw-rw-r-- 1 olga olga 1.7M Oct 11 18:08 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.out_CDS_without_stop_codon_or_utr.read_ids.txt\n-rw-rw-r-- 1 olga olga  15M Oct 11 18:08 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.read_ids.txt\n-rw-rw-r-- 1 olga olga  15M Sep 10 15:55 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out_CDS_without_stop_codon.bam\n-rw-rw-r-- 1 olga olga 1.7M Oct 11 18:08 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out_CDS_without_stop_codon.bam.bai\n-rw-rw-r-- 1 olga olga  15M Sep 18 07:15 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out_CDS_without_stop_codon_or_utr.bam\n-rw-rw-r-- 1 olga olga 1.7M Oct 11 18:08 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out_CDS_without_stop_codon_or_utr.bam.bai\n-rw-rw-r-- 1 olga olga 2.8M Sep 18 10:14 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out_CDS_without_stop_codon_or_utr.read_ids.txt\n-rw-rw-r-- 1 olga olga 4.1M Sep 10 15:56 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out_CDS_without_stop_codon.read_ids.txt\n-rw-r--r-- 1 olga olga 235M Sep 10 15:10 SRR306838_GSM752691_hsa_br_F_1_trimmed_k15.tsv\n(105304,)\n"
      },
      {
       "data": {
        "text/plain": "0    SRR306838.13598607\n1    SRR306838.23303318\n2    SRR306838.19917713\n3    SRR306838.19937719\n4    SRR306838.20862490\nName: 0, dtype: object"
       },
       "execution_count": 80,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "75c01782-427c-4c63-bc29-ed308ae4a142",
     "text": "folder = '/home/olga/data_sm/kmer-hashing/brawand2011/human/'\n! ls -lha $folder\n\n# coding_read_ids = pd.read_csv(f\"{folder}/SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.out_CDS_without_stop_codon_or_utr.read_ids.txt\", \n#                                    header=None, squeeze=True)\ncoding_read_ids = pd.read_csv(f\"{folder}/SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.CDS_without_stop_codon.read_ids.txt\", \n                                   header=None, squeeze=True)\nprint(coding_read_ids.shape)\ncoding_read_ids.head()"
    },
    "executionTime": "2019-10-12T04:12:19.239Z"
   },
   {
    "cell": {
     "executionCount": 81,
     "executionEventId": "51b29717-f185-46c5-a48c-e5e4a89e2763",
     "hasError": false,
     "id": "1b979a86-c27b-4b61-9141-e248088bbda9",
     "outputs": [
      {
       "data": {
        "text/plain": "103944"
       },
       "execution_count": 81,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "75c01782-427c-4c63-bc29-ed308ae4a142",
     "text": "true_coding_read_ids = set(coding_read_ids.values)\nlen(true_coding_read_ids)"
    },
    "executionTime": "2019-10-12T04:12:19.746Z"
   },
   {
    "cell": {
     "executionCount": 82,
     "executionEventId": "a0442b4e-1671-4b6f-90a8-87d57f8197ba",
     "hasError": false,
     "id": "af6c4741-a37b-408d-bac2-800c2c899126",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "(807698,)\n"
      }
     ],
     "persistentId": "317bb484-c93f-4e1c-b831-50734c9e6614",
     "text": "unspliced_read_ids_series = pd.read_csv(f\"{folder}/SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.read_ids.txt\", \n                                   header=None, squeeze=True)\nprint(unspliced_read_ids_series.shape)\nunspliced_read_ids_series.head()\n\nunspliced_reads = set(unspliced_read_ids_series.values)"
    },
    "executionTime": "2019-10-12T04:12:21.123Z"
   },
   {
    "cell": {
     "executionCount": 83,
     "executionEventId": "141686f2-9788-4fae-96c6-2cf0947b14d4",
     "hasError": false,
     "id": "0514d136-42a3-445d-8a45-5c5a92f885ae",
     "outputs": [
      {
       "name": "stderr",
       "output_type": "stream",
       "text": "/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: \nA value is trying to be set on a copy of a slice from a DataFrame.\nTry using .loc[row_indexer,col_indexer] = value instead\n\nSee the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n  \"\"\"Entry point for launching an IPython kernel.\n"
      },
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "(867, 6)\n"
      },
      {
       "name": "stderr",
       "output_type": "stream",
       "text": "/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: \nA value is trying to be set on a copy of a slice from a DataFrame.\nTry using .loc[row_indexer,col_indexer] = value instead\n\nSee the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n  This is separate from the ipykernel package so we can avoid doing imports until\n"
      },
      {
       "data": {
        "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>read_id</th>\n      <th>jaccard_in_peptide_db</th>\n      <th>n_kmers</th>\n      <th>classification</th>\n      <th>short_read_id</th>\n      <th>in_cds_no_stop_codon</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>1</th>\n      <td>SRR306838.6196593 Ibis_Run100924_C3PO:6:29:16733:12435/1</td>\n      <td>0.000000</td>\n      <td>17</td>\n      <td>non-coding</td>\n      <td>SRR306838.6196593</td>\n      <td>False</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>SRR306838.20767303 Ibis_Run100924_C3PO:6:104:6864:5062/1</td>\n      <td>0.187500</td>\n      <td>16</td>\n      <td>non-coding</td>\n      <td>SRR306838.20767303</td>\n      <td>False</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>SRR306838.12582274 Ibis_Run100924_C3PO:6:62:11779:17975/1</td>\n      <td>0.066667</td>\n      <td>15</td>\n      <td>non-coding</td>\n      <td>SRR306838.12582274</td>\n      <td>False</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>SRR306838.13334230 Ibis_Run100924_C3PO:6:66:16579:20350/1</td>\n      <td>0.000000</td>\n      <td>12</td>\n      <td>non-coding</td>\n      <td>SRR306838.13334230</td>\n      <td>False</td>\n    </tr>\n    <tr>\n      <th>5</th>\n      <td>SRR306838.2740879 Ibis_Run100924_C3PO:6:13:11155:5248/1</td>\n      <td>1.000000</td>\n      <td>18</td>\n      <td>coding</td>\n      <td>SRR306838.2740879</td>\n      <td>False</td>\n    </tr>\n  </tbody>\n</table>\n</div>",
        "text/plain": "                                                     read_id  \\\n1   SRR306838.6196593 Ibis_Run100924_C3PO:6:29:16733:12435/1   \n2   SRR306838.20767303 Ibis_Run100924_C3PO:6:104:6864:5062/1   \n3  SRR306838.12582274 Ibis_Run100924_C3PO:6:62:11779:17975/1   \n4  SRR306838.13334230 Ibis_Run100924_C3PO:6:66:16579:20350/1   \n5    SRR306838.2740879 Ibis_Run100924_C3PO:6:13:11155:5248/1   \n\n   jaccard_in_peptide_db  n_kmers classification       short_read_id  \\\n1               0.000000       17     non-coding   SRR306838.6196593   \n2               0.187500       16     non-coding  SRR306838.20767303   \n3               0.066667       15     non-coding  SRR306838.12582274   \n4               0.000000       12     non-coding  SRR306838.13334230   \n5               1.000000       18         coding   SRR306838.2740879   \n\n   in_cds_no_stop_codon  \n1                 False  \n2                 False  \n3                 False  \n4                 False  \n5                 False  "
       },
       "execution_count": 83,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "b321c9fc-54af-40bc-96a9-40c59ea87994",
     "text": "coding_scores_not_low_complexity['short_read_id'] = coding_scores_not_low_complexity['read_id'].str.split().str[0]\ncoding_scores_not_low_complexity_unspliced = coding_scores_not_low_complexity.query('short_read_id in @unspliced_reads')\ncoding_scores_not_low_complexity_unspliced['in_cds_no_stop_codon'] = coding_scores_not_low_complexity_unspliced.short_read_id.isin(true_coding_read_ids)\nprint(coding_scores_not_low_complexity_unspliced.shape)\ncoding_scores_not_low_complexity_unspliced.head()"
    },
    "executionTime": "2019-10-12T04:12:21.732Z"
   },
   {
    "cell": {
     "executionCount": 84,
     "executionEventId": "99589b68-5d73-43d8-ae82-74a13733ba1f",
     "hasError": false,
     "id": "c4e90627-a46a-4fa2-b552-6402975ee8c7",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "accuracy score: 0.9077277970011534\n              precision    recall  f1-score   support\n\n       False       1.00      0.90      0.94       756\n        True       0.58      0.99      0.73       111\n\n   micro avg       0.91      0.91      0.91       867\n   macro avg       0.79      0.94      0.84       867\nweighted avg       0.95      0.91      0.92       867\n\n"
      },
      {
       "data": {
        "image/png": "iVBORw0KGgoAAAANSUhEUgAAAYUAAAEKCAYAAAD9xUlFAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDMuMC4zLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvnQurowAAIABJREFUeJzt3Xd4FWX2wPHvSSGhhA4iIEUFFVEQIygqFooozVVXUQRRBEVR17bqz4K4rotdFBEiYEXFZVGjwiIqWJCqiAjoGooQFGkBQhop5/fHXOI1huSmzJ1bzud57pM7cyczZ1Luued9Z95XVBVjjDEGIMbrAIwxxoQOSwrGGGOKWVIwxhhTzJKCMcaYYpYUjDHGFLOkYIwxppglBWOMMcUsKRhjjClmScEYY0yxOK8DqKjGjRtrmzZtvA7DGGPCytdff71TVZuUt13YJYU2bdqwYsUKr8MwxpiwIiI/B7KdNR8ZY4wpZknBGGNMMUsKxhhjillSMMYYU8ySgjHGmGKuJQURmS4i20Xk+0O8LiLyrIikich3ItLFrViMMcYExs1K4WWgbxmvnw+08z1GAS+4GIsxxpgAuHafgqp+LiJtythkEPCqOvOBLhGR+iJyuKr+6lZMxpgwowpaCEWFzlctBC3643Kpz4vKeO3g86Jy9lENxyq5r3K3Kz2mZf+rRWJcASee2Q26P+jqj9zLm9daAFv8ltN96/6UFERkFE41QatWrYISnDHlUvX7Jy469JtBqL1RhXRMJY5FdM8hrwp3fdibJz/rwomH/8ay9uuId/mYXiYFKWVdqX8BqpoCpAAkJyeH31+JFkHqJbDhA68jMdXiYDIo8jqQ6CAxILEQE+t8ldjS18WU+Hpwm7K2I6b010rbT5nbVeFYZZyXxMRC2hb4fCt9BnWjsGvXiE4K6cARfsstgV88isU9memw7FFIe8frSIwrpHrfPA7uJxhvVGFxrBiQ0j4/Rq49e3LZsCGDLl0OB2Dc0/kMHrWzeNltXiaFVGCMiLwFdAP2Rlx/wvZV8NpJFBdAA2fDkf08DclUk+I3weh6wzLueu+9Hxg9+kNiYoQ1a26gXr1EataMD1pCABeTgoi8CZwNNBaRdGAsOJWPqk4G5gAXAGlANnC1W7G47uePIeN/f17/0zsUJ4TTxkK7vwQ1LGNMeNi+PYubb57LzJlrADj11Jbs2ZNLvXqJQY/FzauPLi/ndQVudOv4QbN3E8zqXfY2x17u+hUDxpjwo6rMmLGaW275L7t351CrVjyPPHIuY8Z0JTbWm3uLw27o7JCT8aPzNalV6U1DMfHQ6brgxmSMCQujR3/IlClfA9Cr15GkpPSnbdsGnsZkSaGq9m12vrbqCb0meRuLMSasXHjhscycuYYnn+zD1Vd3RkKgj8qSQlVl+pJCXbt/whhTtp9+2sUnn2zk+uuTAejb92g2bbrFk76DQ7GkUFUHK4UkSwrGmNIVFBTx1FOLGTt2IXl5BXTu3IxTT20JEFIJASwpVJ1VCsaYMqxatY0RI1L5+mvnivthwzrRrl1Dj6M6NEsKVWWVgjGmFHl5BTz88OeMH7+IgoIiWrWqx5Qp/enb92ivQyuTJYWq0CLI3OI8Tzqi7G2NMVHlnns+4emnlwBw442n8K9/9SQpKcHjqMpnSaEqsn6Donyo2QTia3odjTEmhPz976ezeHE6jz3WizPPbO11OAGzmdeqYt/PzlfrTzAm6s2fv56LL36bggJnoMRmzerw1VfXhFVCAEsKVZNp/QnGRLuMjBxGjHiPPn1eZ/bsdbz00sri10LhvoOKsuajqthnVx4ZE83eeWcdN9wwh23b9pOQEMvYsWcxfHhnr8OqEksKVWGVgjFRadu2/dx001xmzVoLQPfuRzBt2kCOPbaxx5FVnSWFqrBKwZio9N57PzBr1lpq145n/Phe3HDDKcTEhF9TUWksKVSFVQrGRI3c3AISE523zJEjT2bDhgxGjz6FNm3qexxZ9bKO5qqwSsGYiFdUpEycuIy2bSfw8897AIiJER59tHfEJQSwpFB5+VmQuwtia0Ctpl5HY4xxwY8/7qRHj5e46aa5bNu2nzff/N7rkFxnzUeVtc/vTmax3GpMJMnPL+SJJ75i3LjPyMsr5LDDajNpUj8uuug4r0NznSWFyrL+BGMi0vffb2fYsHdYuXIbAFdf3Zknn+xDgwbRMWqBJYXKsv4EYyJSUZGyevV2WreuR0rKAPr0OcrrkILKkkJlWaVgTMRYs2Y7HTo0QUQ48cTDeO+9wfTo0Zo6dWp4HVrQWWN4ZVlSMCbsZWbmMWbMHDp2fIH//Gdd8foLLmgXlQkBrFKoPGs+MiaszZuXxqhRH7B5817i4mLYtGmP1yGFBEsKlWWVgjFhaffuHG69dR6vvroKgC5dDmfatIF07tzM48hCgyWFyvCfXKeuTa5jTLj49ttt9O37Or/9lkVCQizjxp3N7bd3Jy7OWtIPsqRQGdnbofAAJDaC+NpeR2OMCVD79o2oU6cG7ds3YurUgbRv38jrkEKOJYXKsP4EY8KCqvLGG6sZMOAY6tZNoFateBYuHE7z5kkRM4BddbOaqTKsP8GYkLdp0x7OO+91rrzyHe6+++Pi9S1b1rWEUAarFCrDKgVjQlZhYRGTJi3nnns+ISsrn4YNa9K9u/X9BcqSQmVYpWBMSFq3bgcjRqSyeHE6AJdeejzPPXc+TZta31+gLClUhlUKxoScjRsz6Nx5CgcOFHL44XWYNKkfF154rNdhhR1Xk4KI9AUmALHAVFUdX+L1VsArQH3fNner6hw3Y6oWVikYE3Latm3AX//agcTEOJ54og/16yd6HVJYci0piEgs8DzQG0gHlotIqqqu9dvsPuBtVX1BRDoAc4A2bsVUbaxSMMZzOTn5PPTQZ/zlL8fRtWsLAF555UJiY+36mapw86fXFUhT1Q2qegB4CxhUYhsF6vqe1wN+cTGe6pGfAzk7ICYeatsdkMZ44YsvfqZz5ymMH7+IUaPep6hIASwhVAM3m49aAFv8ltOBbiW2eRD4SERuAmoDvVyMp3ocvJM5qaVNrmNMkO3bl8c993zMpEkrAOjQoQmTJ/e3S0yrkZvvaqX9lrTE8uXAy6raErgAeE3kz++0IjJKRFaIyIodO3a4EGoFWH+CMZ6YM+cnOnacxKRJK4iLi+GBB3rwzTejOPXUll6HFlHcrBTSAf+Lg1vy5+ahEUBfAFVdLCKJQGNgu/9GqpoCpAAkJyeXTCzBZf0JxgTd3r25DBkymz17cklObs60aQM58cTDvA4rIrmZFJYD7USkLbAVGAxcUWKbzUBP4GUROQ5IBDwuBcphlYIxQaGqqEJMjFCvXiLPPtuX337L4m9/O9UGsHORa0lBVQtEZAwwD+dy0+mqukZEHgJWqGoqcDvwoojcitO0NFxVva0EymOVgjGu++WXTG644UPOPLMVt9/eHYChQzt5HFV0cPU+Bd89B3NKrHvA7/la4HQ3Y6h2VikY4xpVZfr0ldx++0fs3ZvHkiXp3HDDKdSsGe91aFHD7miuqEyrFIxxw4YNGYwc+T6ffroRgH792jF5cn9LCEFmSaEiVH9vPkqyAbaMqQ6FhUU8++xS7r33U3JyCmjcuBbPPtuXwYM7ImKXmgabJYWKyNkBhXmQ2ABqJHkdjTERY9asdeTkFHD55R2ZMKEvTZrYAHZesaRQEfusP8GY6nDgQCGZmXk0alSL2NgYpk0byE8/7WLAgGO8Di3q2XVdFWGdzMZU2fLlW0lOTmHo0Hc4eLHhscc2toQQIqxSqAi7HNWYSsvOzmfs2AU89dQSioqU7Ox8tm/P4rDD6ngdmvFjSaEirFIwplIWLtzEyJHvk5a2m5gY4Y47TmPcuHOoVcuuLAo1lhQqwioFYypEVbn55rlMnLgcgBNOaMq0aQM55ZQWHkdmDsWSQkVYpWBMhYgIdesmEB8fw3339eDuu8+gRo1Yr8MyZbCkUBFWKRhTrp07s1m/fjfdujmjl95//1kMGXIiHTo08TgyE4iArj4SkRoicrTbwYS0glzI/g0kFmof7nU0xoQcVeWtt77nuOOe58ILZ5KRkQNAYmKcJYQwUm5SEJF+wGpgvm+5s4i843ZgIScz3fma1BJirPw1xl96+j4GDXqLyy//Dzt3ZtOhQxOys/O9DstUQiDNRw/hzJi2AEBVv43KqsH6E4z5k6IiZerUb7jzzvns25dH3boJPPlkH0aMOMmGqAhTgSSFfFXdU+IXHNrDW7vB+hOM+ZMRI1J5+eVvARg48BgmTbqAFi3qlvNdJpQF0qewTkQuBWJEpK2IPAMscTmu0GOVgjF/cuWVJ9C0aW3eeuti3n33MksIESCQpDAGOBkoAmYDucAtbgYVkqxSMIbvv9/OhAm/fybs2fNINmy4mcsusxFNI0UgzUfnqepdwF0HV4jIRTgJInpYpWCiWF5eAf/615c88sgX5OcXkZzcnNNPd/4Xateu4XF0pjoFUincV8q6e6s7kJBnlYKJUkuXpnPyySmMG/cZ+flFjB6dzAknHOZ1WMYlh6wUROQ8oC/QQkSe8nupLk5TUvRQtUrBRJ2srAPcf/8CnnlmCarQrl1Dpk4dSI8erb0OzbiorOaj7cD3OH0Ia/zWZwJ3uxlUyMnZBQU5kFAPEqwjzUSHe+/9lAkTlhITI9x552k8+ODZNjVmFDhkUlDVlcBKEZmhqrlBjCn0WJVgotC9957J6tXbefTRXiQnN/c6HBMkgfQptBCRt0TkOxH538GH65GFEutPMFEgNfVHLrhgBvn5hQA0aVKbTz4ZZgkhygSSFF4GXgIEOB94G3jLxZhCj1UKJoJt357F4MGzGDToLebOTeOVV1Z5HZLxUCBJoZaqzgNQ1fWqeh9wjrthhRibm9lEIFXl9de/47jjnmfmzDXUqhXPhAl9ufrqzl6HZjwUyH0KeeLclbJeRK4HtgJN3Q0rxGRa85GJLJs37+X66z9g7tw0AHr1OpKUlP60bdvA48iM1wJJCrcCdYCbgX8C9YBr3Awq5FjzkYkwH320nrlz06hfP5GnnurD8OGd7Y5kAwSQFFR1qe9pJjAUQERauhlUyLGOZhMBsrIOFN99PGLESWzduo9Ro07m8MOTPI7MhJIy+xRE5BQRuVBEGvuWjxeRV4mmAfEK8iDrV5AYqGNXYZjwU1BQxGOPLaJ162fYsCEDcKbJHDv2bEsI5k8OmRRE5F/ADGAI8F8RuRdnToVVQPvghBcC9m91vtZpATE2e6kJL6tWbaNbt6ncddfH7NqVw7vv/uB1SCbElfUuNwjopKo5ItIQ+MW3/GNwQgsR1p9gwlBeXgEPP/w548cvoqCgiFat6pGS0p/zzou++bFMxZSVFHJVNQdAVXeLyA9RlxDA+hNM2Fm58leGDJnNunU7EYExY07hkUd6kpSU4HVoJgyUlRSOFJGDw2ML0MZvGVW9qLydi0hfYAIQC0xV1fGlbHMp8CDObG6rVPWKwMMPAqsUTJhJSIhj/foMjjmmEVOnDuSMM+xv1wSurKRwcYnliRXZsYjEAs8DvYF0YLmIpKrqWr9t2gH3AKeraoaIhN79D1YpmDDwzTe/ctJJzRAROnRowty5Q+je/QgSE60fzFRMWQPifVLFfXcF0lR1A4CIvIXTT7HWb5uRwPOqmuE75vYqHrP6WaVgQlhGRg533PER06d/y5tvXszgwR0BOPfcth5HZsKVmx8jWgBb/JbTgW4ltmkPICKLcJqYHlTV/5bckYiMAkYBtGoV5DdnqxRMiHrnnXXccMMctm3bT0JCLLt2ZXsdkokAbiaF0m6P1FKO3w44G2gJfCEiHVV1zx++STUFSAFITk4uuQ/32OQ6JgRt27afm26ay6xZTtF9+ulHMHXqQI49trHHkZlIEHBSEJEEVc2rwL7TgSP8llviXNZacpslqpoPbBSRH3GSxPIKHMc9uRmQnwU1kpwJdozx2Ndf/0Lv3q+RkZFL7drxjB/fixtuOIWYGBuiwlSPckdJFZGuIrIa+Mm33ElEngtg38uBdiLSVkRqAIOB1BLbvItvxFXfXdPtgQ0ViN9d/lWCjQtjQkCHDk1o0qQ25513FGvW3MCYMV0tIZhqFcjQ2c8C/YFdAKq6igCGzlbVAmAMMA9YB7ytqmtE5CERGejbbB6wS0TW4twtfaeq7qr4abjE+hOMx4qKlJSUr9mzx5n8sGbNeD7/fDhz5w6hdev6HkdnIlEgzUcxqvpziREUCwPZuarOAeaUWPeA33MFbvM9Qo/1JxgP/fjjTq699n2+/HIzy5dv5cUXnc9Shx1Wx+PITCQLJClsEZGugPruPbgJiI7pOK1SMB7Izy/kyScX8+CDC8nLK6RZszqcf347r8MyUSKQpDAapwmpFfAb8LFvXeSzSsEE2cqVvzJiRCorV24D4OqrO/Pkk31o0KCmx5GZaBFIUihQ1cGuRxKKrFIwQbR+/W66dp1KQUERbdrUJyWlP717H+V1WCbKBJIUlvsuFZ0JzFbVTJdjCh1WKZggOuqohgwdeiJJSTX45z97UqdODa9DMlGo3KuPVPUo4GHgZGC1iLwrIpFfORTmw/5fAHHmUjCmmu3ff4Cbb57L4sW/3/g/bdpAJkw43xKC8Uwgl6Siql+p6s1AF2AfzuQ7kW3/VkCd2dZi472OxkSYefPSOP74STz33DKuv/5DnAvxsHmSjefKbT4SkTo4A9kNBo4D3gO6uxyX96zpyLhg9+4cbr11Hq++ugqAk08+nGnTBloyMCEjkD6F74H3gcdU9QuX4wkd1slsqtmsWWu58cY5bN+eRWJiHOPGnc1tt51GXFxABbsxQRFIUjhSVYtcjyTUWKVgqtGePbmMGvU+GRm59OjRmhdfHED79o28DsuYPzlkUhCRJ1X1duA/IvKnkUkDmXktrFmlYKpIVSkqUmJjY6hfP5FJk/qRkZHDddcl23hFJmSVVSnM9H2t0IxrEcMqBVMFmzbtYdSo9zn33LbcffcZAMUT4BgTyg7ZmKmqy3xPj1PVT/wfOB3Okc0qBVMJhYVFPPvsUjp2nMT8+RuYOHEZubkFXodlTMAC6eG6ppR1I6o7kJCiCvt+dp5bpWACtG7dDnr0eJlbbvkvWVn5DB7ckW++uc7mSTZhpaw+hctwLkNtKyKz/V5KAvaU/l0RIm8v5O+H+NqQ2MDraEyIKygo4tFHv+Shhz7nwIFCmjdP4oUX+jFw4DFeh2ZMhZX1EWYZzhwKLYHn/dZnAivdDMpzNrmOqYCYGOGjjzZw4EAhI0d24bHHelO/fqLXYRlTKYdMCqq6EdiIMypqdLH+BFOOnJx8MjMP0LRpbWJihKlTB7Blyz7OPbet16EZUyWH7FMQkc98XzNEZLffI0NEdgcvRA/YlUemDJ9//jOdOk3myitnFw9P0a5dI0sIJiKU1Xx0cMrNxsEIJKRYpWBKsW9fHvfc8zGTJq0AID4+lp07s2nSpLbHkRlTfcq6JPXgXcxHALGqWgicBlwHRPZ/gVUKpoS5c3+iY8dJTJq0gri4GMaOPYtvvhllCcFEnECulXsXOEVEjgJeBT4E3gD6uxmYp6xSMD6qysiR7zNtmnNtRXJyc6ZPH8gJJxzmcWTGuCOQ+xSKVDUfuAh4RlVvAiJ7ggGrFIyPiNCyZV0SE+N44oneLF48whKCiWgBTccpIn8FhgIX+tZF7gQDRQW+uRRscp1o9csvmaxfv5szz2wNwP/935kMHXoiRx3V0OPIjHFfoHc0n4MzdPYGEWkLvOluWB7a/wtoEdRuBnEJXkdjgkhVmTbtGzp0eJ6LL36bXbuyAahRI9YSgoka5VYKqvq9iNwMHC0ixwJpqvpP90PziPUnRKUNGzIYOfJ9Pv10IwD9+7cnPz/6Row3JpCZ184EXgN8bSo0E5GhqrrI7eA8Yf0JUeXgAHb33beA7Ox8GjeuxbPP9mXw4I42G5qJSoH0KTwNXKCqawFE5DicJJHsZmCe2WdJIZoMG/Yub7yxGoArrjiBZ545zy4zNVEtkD6FGgcTAoCqrgNquBeSxzKt+SiajBzZhZYt65KaOpgZMy6yhGCiXiCVwjciMgWnOgAYQiQPiGfNRxFt+fKtfPrpRu66y5n45uyz25CWdhMJCTa8tTEQWFK4HrgZ+DtOn8LnwHNuBuUp62iOSNnZ+Ywdu4CnnlpCUZHSvfsRxZecWkIw5ndl/jeIyAnAUcA7qvpYcELymFUKEWfhwk1ce20q69dnEBMj3HHHaZx8cnOvwzImJJU1Sur/4QxxMQSYLyKlzcBWJhHpKyI/ikiaiNxdxnaXiIiKiLed13l7nUdcTajZyNNQTNXt3ZvLdde9zznnvML69RmccEJTliwZweOP96FWrci9/9KYqiirUhgCnKiqWSLSBJgDTA90xyISizM5T28gHVguIqn+nda+7ZJwmqeWVjT4ape5xflqk+tEhPvvX0BKyjfEx8dw//09uOuuM6hRI9brsIwJaWUlhTxVzQJQ1R0iEsiVSv664tzotgFARN4CBgFrS2z3D+Ax4I4K7r/6WX9C2FPV4vsLHnjgLDZu3MP48T05/vimHkdmTHgo643+SBGZ7Xu8Axzltzy7jO87qAWwxW85nRID6YnIScARqvpBhSN3g/UnhC1V5Y03VnPuua9y4EAhAI0b1+L99y+3hGBMBZRVKVxcYnliBfddWvuLFr/oVB5PA8PL3ZHIKGAUQKtWLr5hW6UQltLT9zF69Id88MH/AJgx4zuuvvokj6MyJjyVNUfzJ1XcdzrOBD0HtQR+8VtOAjoCC33lfjMgVUQGquqKErGkACkAycnJilusUggrRUXKiy9+zZ13zicz8wD16iXw5JN9GD68s9ehGRO23LxAeznQzjeq6lZgMHDFwRdVdS9+U32KyELgjpIJIaisUggbaWm7GTnyfRYu3ATAoEHHMGlSP5o3T/I2MGPCnGtJQVULRGQMMA+IBaar6hoReQhYoaqpbh270qxSCBtffPEzCxduomnT2kyceD6XXNLBBrAzphoEnBREJEFV8yqyc1Wdg3Mpq/+6Bw6x7dkV2Xe1KyqEzHTneVJLT0MxpduzJ5f69RMBGD68Mzt2ZDNixEk0alTL48iMiRzlXmYqIl1FZDXwk2+5k4hE3jAXWb+CFkKtwyAu0etojJ+8vALGjl1A69bP8NNPuwBnmsy///10SwjGVLNAKoVngf44dzejqqtE5BxXo/KC9SeEpCVL0hkxIpW1a3cAMG/eetq1s7vNjXFLIEkhRlV/LtFeW+hSPN6x/oSQkpV1gPvvX8AzzyxBFdq1a8i0aQOLB7EzxrgjkKSwRUS6AuobuuIm4H/uhuUBqxRCxtKl6VxxxWw2bMggNla4447ujB17FjVr2nhFxrgtkKQwGqcJqRXwG/Cxb11ksUohZNSvn8jWrfvo1Okwpk0baCOaGhNE5SYFVd2Oc49BZLNKwVNffrmZ008/AhHhmGMa8+mnV3HKKc2Jj7cB7IwJpnKTgoi8iN/wFAep6ihXIvKKVQqe2L49i5tvnsvMmWt45ZULGTasEwDdux9RzncaY9wQSPPRx37PE4G/8MeB7iKDzc0cVKrKjBmrueWW/7J7dw61asUXD2RnjPFOIM1HM/2XReQ1YL5rEXnhQCbkZkBsAtRs4nU0EW/z5r1cf/0HzJ2bBkDv3keSkjKANm3qexyZMaYyw1y0BSLrusCDk+vUtcl13LZ0aTq9er3G/v0HqF8/kaefPo+rrupkQ1QYEyIC6VPI4Pc+hRhgN3DIqTXD0j7rTwiWzp2bccQRdTn22MY8//wFHH64DWBnTCgpMymI8/GtE84opwBFqure0NVesU5m1xQUFDFx4jKGDetEw4Y1SUiIY9Gia2jQoKbXoRljSlHm2Ee+BPCOqhb6HpGXEMAuR3XJqlXb6NZtKrfeOo/bbptXvN4SgjGhK5B5l5eJSBfXI/GSVQrVKje3gPvu+5Tk5Bf55ptfadWqHpdf3tHrsIwxAThk85GIxKlqAXAGMFJE1gNZONNsqqpGTqKwSqHafPXVFkaMSOWHH3YiAmPGnMIjj/QkKSnB69CMMQEoq09hGdAFuDBIsXjHKoVqkZa2mzPPfImiIuWYYxoxbdpATj/dfqbGhJOykoIAqOr6IMXijT9MrmN30VbF0Uc3ZNSoLjRsWJP77z+LxEQ3Z3s1xrihrP/aJiJy26FeVNWnXIgn+LJ/g6J856a1eOsArYiMjBxuv/0jrr66c/GQ1pMm9bN7DowJY2UlhVigDr6KIWJZf0KlzJ69jhtvnMO2bfv5+utf+fbb6xARSwjGhLmyksKvqvpQ0CLxivUnVMi2bfsZM2YO//nPOgDOOKMVU6cOsGRgTIQot08h4lmlEBBV5dVXV3HrrfPIyMilTp0aPPpoL66/PpmYmOj4UzEmGpSVFHoGLQovWaUQkD17crn99o/IyMilb9+jmTy5H61b2wB2xkSaQyYFVd0dzEA8Y5XCIRUVKUVFSlxcDA0a1GTKlP5kZ+dz5ZUnWnORMREqkDuaI5tVCqX64Yed9OjxEuPHf1m87uKLOzB0qI1oakwks6RglcIf5OcX8sgjX9Cp02QWLdrCtGkryc0t8DosY0yQRPfdRflZkLsLYmtAraZeR+O5lSt/5ZprUvn2220AjBhxEo8/3ttuQjMmikT3f/s+3+Q6SUeARG/RlJ9fyNixC3nssUUUFipt2tTnxRcH0KvXkV6HZowJsuhOCtafAEBcXAxLl26lqEi55ZZuPPzwudSpU8PrsIwxHojupBDF/QmZmXlkZh6gefMkRISpUwewbdt+TjvNxn8yJppFb5sJRG2lMG9eGh07vsCQIbM5OG9S27YNLCEYYywpAFC3tbdxBMmuXdlcddW79O07g82b95KZmceuXTleh2WMCSGuJgUR6SsiP4pImojcXcrrt4nIWhH5TkQ+EZHgvjvvi45KQVWZNWstHTpM4tVXV5GYGMdjj/ViyZJrady4ltfhGWNCiGt9CiISCzwP9AbSgeUikqqqa/02Wwkkq2q2iIwGHgMucyumP8mM/D4FVWXIkNm8+eb3APTo0ZoXXxxA+/aNPI7MGBOK3KwUugJpqrpBVQ8y0uVLAAAPNklEQVQAbwGD/DdQ1QWqmu1bXAK0dDGeP9IiyPS7JDVCiQgdOjQhKakGL7zQjwULrrKEYIw5JDevPmoBbPFbTge6lbH9CGBuaS+IyChgFECrVtX0qT57OxQegJqNIT6ymlA2bsxgw4YMevZ07jO4667TGT68My1b1vU4MmNMqHOzUihtgBwtdUORK4Fk4PHSXlfVFFVNVtXkJk2aVE90EdifUFhYxIQJS+jY8QUuu2wW27dnARAfH2sJwRgTEDcrhXTAv12mJfBLyY1EpBdwL3CWqua5GM8fRVh/wtq1O7j22lQWL3bmmx448Bib58AYU2FuJoXlQDsRaQtsBQYDV/hvICInAVOAvqq63cVY/ixCKoX8/EIefXQR//jH5xw4UEjz5km88EI/Bg48xuvQjDFhyLWkoKoFIjIGmIcz3/N0VV0jIg8BK1Q1Fae5qA7wb99wzJtVdaBbMf1BhFQKV1wxm1mznAu6Ro7swuOP96ZevUSPozLGhCtXh7lQ1TnAnBLrHvB73svN45cpQiqFW27pxrffbmPKlP6ce25br8MxxoS56L2jOUwrhc8+28S4cQuLl884oxXr1t1oCcEYUy2id0C8MKsU9u3L46675jN58tcAnHNOW3r0cG4Aj4uL3txujKle0ZkU8rMhZwfExEPtw7yOplxz5vzEddd9QHr6PuLjY7j33jM59dTg3ednjIke0ZkUMsNjcp2dO7P529/+y4wZqwHo2rUF06YNpGNHmyXOGOOO6EwKYTKPwkMPfcaMGaupWTOOhx8+l1tu6UZsbOgmMWNM+IvOpBDC8yioKr7Lcxk37mx++y2LRx45l6OOauhxZMaYaBCdHztDsFJQVV588Wu6d59Obm4BAA0a1GTmzEssIRhjgiY6k0KIVQrr1++mZ89XGTXqA5YsSeftt9d4HZIxJkpFd/ORx5WCM4DdUu6771Nycgpo0qQWzz13PpdeeryncRljold0JoUQuEdhzZrtXHNNKsuWbQVgyJATeOaZvjYTmjHGU9GXFEJkcp2VK7exbNlWWrRIYsqU/vTr196zWIwx5qDoSwrZO6AwDxIbQo06QT30jh1ZNGlSG3Aqgz17chk69EQbwM4YEzKir6PZg07m7Ox87rjjI9q0mcC6dTsAZ5rMMWO6WkIwxoSU6KsUgnw56oIFGxk58n3Wr88gJkb4/POfOe64apo9zhhjqln0JYUgVQp79+by97/PJyXlGwBOOKEp06cPIjm5uavHNcaYqoi+pBCESuHLLzczePAstm7NJD4+hvvv78Fdd51BjRqxrh3TGGOqQ/QlhSBUCs2a1WHXrhxOPbUlU6cO4PjjbQA7Y0x4iL6k4EKloKrMn7+B3r2PREQ4+uiGfPnl1XTu3MwGsDPGhJXoe8eq5kphy5a9DBjwJued9zovvfRt8fqTT25uCcEYE3aiq1LIz4Hs7RATB7WbVWlXRUXOAHZ33jmfzMwD1KuXQEKC9RkYY8JbdCWF/enO1zotIabyb+A//bSLkSPf57PPfgbgwguP5fnnL6B586TqiNIYYzwTXUmhGvoTvvpqCz17vkpubgFNm9Zm4sTzueSSDsVzIBhjTDiLrqRQDf0JycnNadeuISeddDhPPdWHRo1sADtjTOSIrqRQiUohL6+AJ574iuuuS6Zx41rUqBHLokXXkJSU4FKQxhjjnehKChWsFJYsSWfEiFTWrt3BunU7ef31i5xvt4RgjIlQ0ZUUAqwUsrIOcN99nzJhwlJUoX37Rlx33clBCNAYY7wVXUkhgErhk082MHLk+2zcuIfYWOHOO7szduzZJCZG14/KGBOdouedTtUvKZQ+uc7//reL3r1fQxU6d27GtGkD6dLl8CAGaYwx3oqepJCzEwpyIaE+JNQtdZP27Rtxyy3daNKkNnfe2Z34eLsZzRgTXaInKWT+uT/ht9/2c/PN/+X660/mnHPaAvD00329iM4YY0KCq4PziEhfEflRRNJE5O5SXk8QkZm+15eKSBvXgtn3e3+CqvLaa6vo0GESb7+9hltvnYequnZoY4wJF64lBRGJBZ4Hzgc6AJeLSIcSm40AMlT1aOBp4FG34jlYKWzOaUO/fm8wbNi77N6dQ58+R/Huu4PtjmRjjMHd5qOuQJqqbgAQkbeAQcBav20GAQ/6ns8CJoqIqAsf24v2bGbyolO4a15j9men0aBBIk8/fR7DhnWyhGCMMT5uJoUWwBa/5XSg26G2UdUCEdkLNAJ2Vncwe39LZ9z8s9ifDRdffBwTJ15As2Z1qvswxhgT1txMCqV9/C5ZAQSyDSIyChgF0KpV5cYtapAUw9TLP+JA8lguHnVRpfZhjDGRzs2kkA743xDQEvjlENuki0gcUA/YXXJHqpoCpAAkJydXrmmp/5sM6GedycYYUxY3rz5aDrQTkbYiUgMYDKSW2CYVuMr3/BLgUzf6E4qJOA9jjDGlcq1S8PURjAHmAbHAdFVdIyIPAStUNRWYBrwmImk4FcJgt+IxxhhTPldvXlPVOcCcEuse8HueC/zVzRiMMcYEzmaWN8YYU8ySgjHGmGKWFIwxxhSzpGCMMaaYJQVjjDHFJNxGBxWRHcDPlfz2xrgwhEaIs3OODnbO0aEq59xaVZuUt1HYJYWqEJEVqprsdRzBZOccHeyco0Mwztmaj4wxxhSzpGCMMaZYtCWFFK8D8ICdc3Swc44Orp9zVPUpGGOMKVu0VQrGGGPKEJFJQUT6isiPIpImIneX8nqCiMz0vb5URNoEP8rqFcA53yYia0XkOxH5RERaexFndSrvnP22u0REVETC/kqVQM5ZRC71/a7XiMgbwY6xugXwt91KRBaIyErf3/cFXsRZXURkuohsF5HvD/G6iMizvp/HdyLSpVoDUNWIeuAM070eOBKoAawCOpTY5gZgsu/5YGCm13EH4ZzPAWr5no+OhnP2bZcEfA4sAZK9jjsIv+d2wEqggW+5qddxB+GcU4DRvucdgE1ex13Fc+4BdAG+P8TrFwBzcWauPBVYWp3Hj8RKoSuQpqobVPUA8BYwqMQ2g4BXfM9nAT1Fwnr2nXLPWVUXqGq2b3EJzkx44SyQ3zPAP4DHgNxgBueSQM55JPC8qmYAqOr2IMdY3QI5ZwXq+p7X488zPIYVVf2cUmag9DMIeFUdS4D6InJ4dR0/EpNCC2CL33K6b12p26hqAbAXaBSU6NwRyDn7G4HzSSOclXvOInIScISqfhDMwFwUyO+5PdBeRBaJyBIR6Ru06NwRyDk/CFwpIuk487fcFJzQPFPR//cKcXWSHY+U9om/5CVWgWwTTgI+HxG5EkgGznI1IveVec4iEgM8DQwPVkBBEMjvOQ6nCelsnGrwCxHpqKp7XI7NLYGc8+XAy6r6pIichjObY0dVLXI/PE+4+v4ViZVCOnCE33JL/lxOFm8jInE4JWdZ5VqoC+ScEZFewL3AQFXNC1JsbinvnJOAjsBCEdmE0/aaGuadzYH+bb+nqvmquhH4ESdJhKtAznkE8DaAqi4GEnHGCIpUAf2/V1YkJoXlQDsRaSsiNXA6klNLbJMKXOV7fgnwqfp6cMJUuefsa0qZgpMQwr2dGco5Z1Xdq6qNVbWNqrbB6UcZqKorvAm3WgTyt/0uzkUFiEhjnOakDUGNsnoFcs6bgZ4AInIcTlLYEdQogysVGOa7CulUYK+q/lpdO4+45iNVLRCRMcA8nCsXpqvqGhF5CFihqqnANJwSMw2nQhjsXcRVF+A5Pw7UAf7t61PfrKoDPQu6igI854gS4DnPA/qIyFqgELhTVXd5F3XVBHjOtwMvisitOM0ow8P5Q56IvInT/NfY108yFogHUNXJOP0mFwBpQDZwdbUeP4x/dsYYY6pZJDYfGWOMqSRLCsYYY4pZUjDGGFPMkoIxxphilhSMMcYUs6RgQo6IFIrIt36PNmVs2+ZQo0lW8JgLfSNxrvINEXFMJfZxvYgM8z0fLiLN/V6bKiIdqjnO5SLSOYDv+ZuI1KrqsU10sKRgQlGOqnb2e2wK0nGHqGonnMESH6/oN6vqZFV91bc4HGju99q1qrq2WqL8Pc5JBBbn3wBLCiYglhRMWPBVBF+IyDe+R/dStjleRJb5qovvRKSdb/2VfuuniEhsOYf7HDja9709feP0r/aNc5/gWz9efp+f4gnfugdF5A4RuQRnfKkZvmPW9H3CTxaR0SLymF/Mw0XkuUrGuRi/gdBE5AURWSHOPArjfOtuxklOC0RkgW9dHxFZ7Ps5/ltE6pRzHBNFLCmYUFTTr+noHd+67UBvVe0CXAY8W8r3XQ9MUNXOOG/K6b5hDy4DTvetLwSGlHP8AcBqEUkEXgYuU9UTcEYAGC0iDYG/AMer6onAw/7frKqzgBU4n+g7q2qO38uzgIv8li8DZlYyzr44w1ocdK+qJgMnAmeJyImq+izOuDjnqOo5vqEv7gN6+X6WK4DbyjmOiSIRN8yFiQg5vjdGf/HARF8beiHOmD4lLQbuFZGWwGxV/UlEegInA8t9w3vUxEkwpZkhIjnAJpzhl48BNqrq/3yvvwLcCEzEmZ9hqoh8CAQ8NLeq7hCRDb4xa37yHWORb78VibM2zrAP/rNuXSoio3D+rw/HmXDmuxLfe6pv/SLfcWrg/NyMASwpmPBxK/Ab0Amnwv3TpDmq+oaILAX6AfNE5FqcYYZfUdV7AjjGEP8B80Sk1Dk2fOPxdMUZhG0wMAY4twLnMhO4FPgBeEdVVZx36IDjxJmBbDzwPHCRiLQF7gBOUdUMEXkZZ2C4kgSYr6qXVyBeE0Ws+ciEi3rAr74x8ofifEr+AxE5EtjgazJJxWlG+QS4RESa+rZpKIHPT/0D0EZEjvYtDwU+87XB11PVOTiduKVdAZSJM3x3aWYDF+LMAzDTt65CcapqPk4z0Km+pqe6QBawV0QOA84/RCxLgNMPnpOI1BKR0qouE6UsKZhwMQm4SkSW4DQdZZWyzWXA9yLyLXAszpSFa3HePD8Ske+A+ThNK+VS1VycESj/LSKrgSJgMs4b7Ae+/X2GU8WU9DIw+WBHc4n9ZgBrgdaqusy3rsJx+voqngTuUNVVOHMzrwGm4zRJHZQCzBWRBaq6A+fKqDd9x1mC87MyBrBRUo0xxvixSsEYY0wxSwrGGGOKWVIwxhhTzJKCMcaYYpYUjDHGFLOkYIwxppglBWOMMcUsKRhjjCn2/0pAkGJ0oExIAAAAAElFTkSuQmCC\n",
        "text/plain": "<Figure size 432x288 with 1 Axes>"
       },
       "metadata": {
        "needs_background": "light"
       },
       "output_type": "display_data"
      }
     ],
     "persistentId": "2c810e70-a822-4d9a-a9b0-2622dfa247bf",
     "text": "y_true = coding_scores_not_low_complexity_unspliced['in_cds_no_stop_codon']\ny_score = coding_scores_not_low_complexity_unspliced['jaccard_in_peptide_db']\ny_pred = y_score > 0.5\n\nfpr, tpr, thresholds = metrics.roc_curve(y_true, y_score)\n\nfig, ax = plt.subplots()\nlw = 2\nax.plot(fpr, tpr, lw=2, color='darkorange')\nax.plot([0, 1], [0, 1], color='navy', lw=lw, linestyle='--')\nax.set(xlabel='False Positive Rate', ylabel='True Positive Rate')\n\nprint(\"accuracy score:\", metrics.accuracy_score(y_true, y_pred))\n\nprint(metrics.classification_report(y_true, y_pred))"
    },
    "executionTime": "2019-10-12T04:12:21.963Z"
   },
   {
    "cell": {
     "executionCount": 85,
     "executionEventId": "602b6002-0f7e-43e0-a08a-76f508d17ca7",
     "hasError": false,
     "id": "f5b034a4-325d-41af-8295-7b22ef456d06",
     "outputs": [
      {
       "data": {
        "text/plain": "<matplotlib.axes._subplots.AxesSubplot at 0x7f3e49e7b438>"
       },
       "execution_count": 85,
       "metadata": {},
       "output_type": "execute_result"
      },
      {
       "data": {
        "image/png": "iVBORw0KGgoAAAANSUhEUgAAAYYAAAELCAYAAADdriHjAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDMuMC4zLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvnQurowAAIABJREFUeJzt3Xl4XPV97/H3d0aStcubvGDjFcfGoVwIhkChBZI0hUChTbgFmvRpetvSJYQEaLMXCLT3ydKkFwgk4eayZGlYUkwMpkDSgNnBS8xuwLFlWxCwLNuytVmame/9Y87IM/JIOrY1c0aaz+t55plzzpxz5qvx+Hzmd5bfMXdHREQkIxZ1ASIiUloUDCIikkPBICIiORQMIiKSQ8EgIiI5FAwiIpJDwSAiIjkUDCIikkPBICIiOSqiLuBQTJ061efNmxd1GSIiY8ratWt3uHvzSPONyWCYN28ea9asiboMEZExxcy2hJlPu5JERCSHgkFERHIoGEREJIeCQUREcigYREQkh4JBRERyKBhERCSHgkFERHKMyQvcZHR8+9vf5tHHVgHQ2NjAzTfdRFNTU8RViUjU1GIoU4lEgocffoTdiQp2VkzhrdZW1q9fH3VZIlICChoMZnarmW03s5eHeN3M7AYz22hmL5rZ+wpZj+zX0tLCvn299M04ht75vwexOK+++mrUZYlICSj0rqTbge8APxzi9bOBRcHj/cB3g2cpsC1b0l2mpGonQyyO10ykpaUl2qJkVLg73d3d7Nmzh46ODvbs2cPevXsHhrMfiUQi6nLHvHg8Tn19PU1NTTQ2Ng48mpqaaGhoGBiuq6sjFhsbO2kKGgzu/riZzRtmlvOBH7q7A8+a2UQzm+nuvy1kXQKtra0ApCY0AJCoamDL1m1RliR59Pf352zg8z3Sr3WwJ3jeu7eLRDI55DprK436SqeuIkmlpYr41xycnoTRkzBqKpyaCo+6nCElMbYlKujsN7r6nKEqjZnRUF9HY1MjjY0TaQyCZHCgDA6XCRMmFPXvgegPPs8CsrdGrcE0BUOBdXZ2YhWVEEt/BbxiAt3duyKuShKJBF/+0pfYvPk37N3bSU/vviHnrYxDQxXUVaSojyeZVpliQa1T35QKNvxOfWV6OPNcV+HEx8CP1jc7Krhhw3T+8LxzePjBlXzyqHdZ1FT6rZuUQ1fC6Oo3OvtjdPZb8IjRmTA6+3vo7Gun87fG29sqeCORnmdfYujgm1BVSUN9PdNnzuTaa69jypQpBf87og4GyzMt7ydkZpcAlwDMmTOnkDWVhUQiAZa1hbAYicTQvzKleKZNn05nZycxi5FM7qavv/+AeczSodBUmUw/qlI0TUgxsSpFU5UzcUIqPa0qxYR4BH/EYdqwq4I//Mg5/MOnPo07bFj3wzERDDGDhkqnodKBFP0p6NgXo6Mvxu4+o6Mvtv+xL0FFLEbMKtiZgmSexltFPE5jYyOTp0xh2rTpVFQUZ5MddTC0Akdmjc8G3s43o7vfAtwCsGzZstJtV44RjY2NeP8+SKUgFsMS+2ior4+6rLJXUVHBlVdeOTDu7nR1dbFz586hH+072Nrezu539pDyA/9r1FQYDVVOTUWK2liS2opgONhFk/Mc9wNer4qlg6iYlkxKcMODK3GHR/5rJZctiSYU+pLQkzS6g91a3YlY8Gz7n5ODx+P0JOPsDXYt5dPUUM/kKVOYPGUqCyZPZvKgx5QpU5g8eTINDQ1YsT98og+GFcClZnYn6YPOHTq+UBzNzembOFl/Fz6hgVhfFzOOnBFxVTKYmVFfX099ff2ILeVkMklHR8dAYLS3tw8Md3R00NnZSVdXJ7v27mVbVydde7rp7unF84RJtrhBTaXtD4xYMtjvvz886kY5XBY1JbhsybtsWPdDLluSOKTWwuFs1LsTRne/kwhxCKamegJ1tbXU1ddTX99Ac309dXV1NDU15WzkM49JkyYV7Zf/oSpodWb2U+AMYKqZtQJXA5UA7v494EHgI8BGoBv4y0LWI/vNnz8fgHj3ThJV9VT07mL+/FMirkoORzweH9j4hJVKpejp6QlCo2vgOfPIP54Ol9aDDJeJ1TCzpo9ZdUmOqE1yRF2SWXVJ6ivzL7uoaeRA6E3A291x3uqq4O2ueHq4u4qdvRz8Rr2hgWn1DdTV1VFXV0d9sIEfbry2tpZ4fAzuqxtBoc9KuniE1x34VCFrkPwWLlxILBYj1tVGrLoJT/SxePHiqMuSIovFYgMbukOVCZfsIMkOlMxwW1sbLZs389jWLezr23/cpGmCcURNXxAUiYHAaKz0gVZGV78FAZB+vN0V5+2eKtp79tdREY8ze/Yslh6zgBkzZuRsyMtpoz4aSrs9IwVTXV3N/AULeGN7G6kJjQAcffTREVclY1F2uEybNm3E+VOpFO+++y5btmyhpaWFrVu30rJ5E8+0tND9Vu/AfO+fto+/f28nyzfX8POW2oHpE6oqOXLOHI6fN5958+Yxd+5c5s6dyxFHHFHyu2jGCn2KZex3jjmGTfevJFnVQG1dnc72kqKIxWLMnDmTmTNncvLJJw9Md3d27NjBli1beO6557jnnnvYvLeC7T1xzjzzTD784Q8zb948pk+fPmYuFBurFAxlbMmSJdx3331U7trMkuOO1X82iZSZ0dzcTHNzMyeccAI7duzg0UcfpbGxkS996UtUVlZGXWLZ0JagjB111FEAWLKfoxYujLgakf3MjNNOOw2AE044QaFQZAqGMjZ79uyBYe1GklI10hlPMvoUDGWsurp6YHj69OkRViJyoMyFXVFc4FXuFAwCcFDnvosUk1oMxadgEAAaGhqiLkEkh1oK0VEwCAA1NTVRlyAiJULBIABUVVVFXYKIlAgFgwCoawARGaBgEABd3CYiA7Q1EEAH+kRkPwWDiIjkUDAIoHPFRWQ/BYMACgYR2U/BIICCQUT2UzCIiEgOBYMAajGIyH4KBgEUDCKyn4JBRERyKBhERCSHgkFERHIoGEREJIeCQUREcigYREQkh4JBAPWuKiL7KRhERCRHwYPBzM4ys9fNbKOZfSHP63PM7FEz+7WZvWhmHyl0TSIiMrSCBoOZxYGbgLOBpcDFZrZ00GxfAe529+OBi4CbC1mTiIgMr9AthpOAje6+yd37gDuB8wfN40BjMNwEvF3gmkREZBgVBV7/LGBb1ngr8P5B81wDPGJmnwbqgA8VuCbJQ30liUhGoVsM+U51GbwFuhi43d1nAx8BfmRmB9RlZpeY2RozW9PW1laAUsubgkFEMgodDK3AkVnjszlwV9FfAXcDuPszQDUwdfCK3P0Wd1/m7suam5sLVG75SqVSUZcgIiUiVDCY2RQzu9HM1pnZWjO73symhFh0NbDIzOabWRXpg8srBs2zFfhg8D5Hkw4GNQmKTC0GEckI22K4E9gOfAy4gPSG+66RFnL3BHAp8DDwGumzj14xs2vN7LxgtiuBvzGzF4CfAp90baWKLplMRl2CiJSIsAefJ7v7dVnj/2JmfxxmQXd/EHhw0LSrsoZfBU4NWYcUiIJBRDLCthgeNbOLzCwWPP4UWFnIwqS4EolE1CWISIkYtsVgZntJn0VkwBXAj4KX4kAncHVBq5OiUYtBRDKGDQZ3byhWIRIttRhEJGOkFsP7hnvd3deNbjkSFZ2uKiIZIx18/lbwXA0sA14gvVvpWOA54LTClSbFpGAQkYxhDz67+5nufiawBXhfcIHZCcDxwMZiFCgiIsUV9qykJe7+UmbE3V8GjitMSSIiEqWw1zG8ZmY/AH5M+iylT5C+YE3GCd3BTUQywgbDXwJ/D3wmGH8c+G5BKpJIxGK6mZ+IpIUKBnfvBf49eBzAzP7T3T82moVJccXj8ahLEJESMVo/ExeM0nokIgoGEckYrWBQp3djnIJBRDK0Y1kABYOI7DdawaBTWsY4BYOIZIQOBjOrMbPFQ7z8+VGqRyKis5JEJCPsHdz+CFgPPBSMH2dmA3dic/dHClOeFIuuYxCRjLA/E68BTgJ2A7j7emBeYUqSKCgYRCQjbDAk3L2joJVIpBQMIpIR9srnl83sz4C4mS0CLgOeLlxZUmy6zbaIZIRtMXwaeC+wD/gpsAf4bKGKkuJTMIhIRtguMbqBLwcPGYcUDCKSMdId3O5nmKua3f28Ua9IIqFgEJGMkVoM/xY8fxSYQbrbbYCLgZYC1SQRUDCISMawweDuqwDM7Dp3//2sl+43s8cLWpmIiEQi7MHnZjMb6EHVzOYDzYUpSaKgFoOIZIQ9XfVy4DEz2xSMzwP+tiAVSSR0HYOIZIQ9K+mh4PqFJcGkDe6+r3BliYhIVEY6K+kD7v4rM/vooJcWmhnufm8Ba5MiUotBSo12b0ZnpBbD6cCvgD/K85oDCoZxQsEgpSYTDPpuFt9IZyVdHQxe6+6bs18LDkCPyMzOAq4H4sAP3P1reeb5U9Id9Tnwgrv/WZh1y+jRfz4RyQh7VtJ/5pn2s5EWMrM4cBNwNrAUuNjMlg6aZxHwReBUd38v6mojEgoGEckY6RjDEtJ9JDUNOs7QCFSHWP9JwEZ33xSs707gfODVrHn+BrjJ3XcBuPv28OXLaNGNeqTUpFIpAJLJZMSVlJ+RjjEsBs4FJpJ7nGEv6Q36SGYB27LGW4H3D5rnPQBm9hTp3U3XuPtDg1dkZpcAlwDMmTMnxFuLyFi2fXv6N6IOQhffSMcYfg783MxOcfdnDmH9+fZPDP5XrgAWAWcAs4EnzOwYd989qJZbgFsAli1bpm+KyDj31ltvAbCzvT3iSspP2P0H75rZ/WbWZmbbzezn2VdCD6MVODJrfDbwdp55fu7u/cEB7tdJB4UUkX6VSalpa2sLnt+NuJLyEzYY/gO4G5gJHAHcQ/q+DCNZDSwys/lmVgVcBKwYNM99wJkAZjaV9K6lTUhRZfbnipSK3t4eAPb16lraYgsbDObuP3L3RPD4McN0x53h7gngUuBh4DXgbnd/xcyuNbNMl90PA+1m9irwKPBP7q62Y5EpGKTUTJs2HYDm6dMjrqT8hO0r6VEz+wJwJ+lAuBBYaWaTAdx951ALuvuDwIODpl2VNezAFcFDIqJgkFIzc+ZMYH9ASPGEDYYLg+fBHef9L9JBEeZ4g5SwRCIRdQkiOWbMmAFARUXYzZSMlrCd6IW6ylnGLrUYpNTU1NQAusYmCqE+cTOrNbOvmNktwfgiMzu3sKVJManFIKVKV+UXX9govg3oA343GG8F/qUgFUkk1GKQUpM5hVpXPhdf2GBY6O7fAPoB3L2H/BevyRil/3xSajo6OqIuoWyFDYY+M6shOEXVzBYCOrl4HFGLQUpN5srnrq6uiCspP2EP918NPAQcaWY/AU4FPlmooqT41GKQUrNtW7qbtW1bt0RcSfkJe1bSL8xsHXAy6V1In3H3HQWtTETKWuaEiER/f8SVlJ+DOUH4dOA00ruTKoHlBalIIqG+kqTU1NXVAVAbPEvxhD1d9Wbg74CXgJeBvzWzmwpZmIiUt+lBVxjTZ8yMuJLyE7bFcDpwTNB9BWZ2B+mQEBEpiLlz5wJQW1sbcSXlJ+xZSa8D2XfHORJ4cfTLkajoIiIpNZldSfF4POJKyk/YFsMU4DUzez4YPxF4xsxWALj7eUMuKSIiY0rYYLhq5FlkLFN/NFKq1JotvrCnq64a7nUze8bdTxmdkiQK+s8npUpnzBXfaP1MrB6l9UhEFAxSajLfSX03i2+0gkGRPsbpP5+UKrUYik87lgVQMEjp0XcyOqMVDPoXHOP0n1BEMkYrGP58lNYjEdFZSSKSEbZLjI+a2Ztm1mFme8xsr5ntybzu7i8XrkQRESmmsNcxfAP4I3d/rZDFSHS0K0lEMsLuP3hXoSAiUh7CthjWmNldwH1k3bnN3e8tSFVSdDolUEQywgZDI9ANfDhrmgMKhnFCwSAiGWG7xPjLQhci0dI9n0UkY9hgMLPPufs3zOxG8lzd7O6XFawyKSrd81lEMkZqMWQOOK8pdCESLQWDiGQMGwzufn/wfMdw85nZje7+6SFeOwu4HogDP3D3rw0x3wXAPcCJ7q4gKjIFg4hkjNblrqfmm2hmceAm4GxgKXCxmS3NM18DcBnw3CjVIwdJwSAiGYXuB+EkYKO7b3L3PuBO4Pw8811H+iK63gLXI0NIJBJRlyAiJaLQwTAL2JY13hpMG2BmxwNHuvsDBa5FhqGzkkQko9C9q+abPnB2k5nFgH8HrhzxDcwuMbM1Zramra3t0KqUIWlXkohkjFYwXD/E9FbgyKzx2cDbWeMNwDHAY2bWApwMrDCzZYNX5O63uPsyd1/W3Nw8OlXLAF3gJiIZI13HcD/D3J3N3c8Lnm8fYpbVwCIzmw+8BVwE/FnW8h3A1Kz3ewz4R52VVHwKBhHJGOk6hn8Lnj8KzAB+HIxfDLSMtHJ3T5jZpcDDpE9XvdXdXzGza4E17r7ikKqWUaf7MYhIxkjXMawCMLPr3P33s16638weD/MG7v4g8OCgaVcNMe8ZYdYpIiKFE/ZnYrOZLciMBLuGtKN/HFGLQUQywvauejnpA8SbgvF5wCUFqUgioRv1iEjGiMEQnFK6B1gELAkmb3D3fUMvJWONWgwikjFiMLh7ysy+5e6nAC8UoSaJgFoMIpIR9mfiI2b2MdPWY9xSi0FEMsIeY7gCqAMSZtZL+opmd/fGglUmRaXMF5GMsHdwayh0IRIttRhEJCNsiwEzm0T6AHR1Zpq7h7qWQUqfWgwikhEqGMzsr4HPkO7raD3pPo2eAT5QuNKkmNRiEJGMsFuDzwAnAlvc/UzgeEBdnI4jajGISEbYYOh1914AM5vg7huAxYUrS4pNLQYRyQh7jKHVzCYC9wG/MLNd5HafLWOcWgwikhH2rKQ/CQavMbNHgSbgoYJVJUWnYBCRjFD7D8zsZDNrgIEeVx8lfZxBRETGmbA7lr8LdGaNdwXTRERknAkbDOZZt/hy9xQHcQ2ElD7dwU1EMsIGwyYzu8zMKoPHZ4BNIy4lIiJjTthg+Dvgd0nft7kVeD+6H8O4kkwmoy5BREpE2LOStgMXFbgWiZB2JYlIRtizku4IrmPIjE8ys1sLV5YUWyqViroEESkRYXclHevuuzMj7r4Lna46rmhXkohkhA2GWNC7KgBmNhmdlTSuqMUgIhlhN+7fAp42s58F4/8T+NfClCRRUItBRDLCHnz+oZmtBc4kffe2j7r7qwWtTIoqkUhEXYKIlIjQu4Pc/RUzayO4UY+ZzXH3rQWrTIqqv78/6hJEpESEPSvpPDN7E9gMrAJagP8qYF1SZH19fVGXICIlIuzB5+tI37XtDXefD3wQeKpgVUlRZF+70NvbG2ElIlJKwgZDv7u3kz47KebujwLHFbAuKYLsMOju7o6wEhEpJWGPMew2s3rgCeAnZrYd0NHKMa6zszPvsIiUt7AthvOAbtL3fn4I2AicG2ZBMzvLzF43s41m9oU8r19hZq+a2Ytm9t9mNjds8XJ49u7dm3dYRMrbsMFgZk8Gg+8Cu4FdwHeA/w1sNrPNZvYPwywfB24CzgaWAheb2dJBs/0aWObuxwI/A75xKH+IHLw9e/bkHRaR8jZsMLj7acFzg7s3Bo+G4NEELCPdihjKScBGd9/k7n3AncD5g97jUXfP7OB+Fph9qH+MHJyOjo6B4d27dw8zp4iUk7C7kvIKDkifMcwss4BtWeOtwbSh/BVDnAZrZpeY2RozW9PW1nawpUoemTBIVjfR0aEWg4ikHVYwALj7b4d5Od8d5vP272xmnyDdAvnmEO9zi7svc/dlzc3NB1+oHCDTYkhVN7Fz166IqxGRUlHojvBagSOzxmcDbw+eycw+BHwZON3d9xW4Jgl0dHSAxYjt28uepE4yE5G0w24xjGA1sMjM5ptZFemb/azInsHMjge+D5wX3BBIiiQdDIYl++jq7FRHeiICFDgY3D0BXAo8DLwG3B30uXStmZ0XzPZNoB64x8zWm9mKIVYno2z37t04MbAY7q4zk0QEKMI9Fdz9QeDBQdOuyhr+UKFrkPzad+5Mh4Klfx/s2rWLSZMmjbCUiIx3hd6VJCVsx452PBYDiwPQ3t4ecUUiUgoUDGWqr6+PvXs60qEQS38NduzYEXFVIlIKFAxlavv29HF+j8XxoMXwzjvvRFmSiJQIBUOZGggBi6fPTJpQz29/O9wlKSJSLhQMZaq1tTU9EEu3FhJVDfuniUhZUzCUqdbWVixeMbAbKVndwNZtCgYRUTCUrW3btpGqbhrotCRV3UTn3j26lkFEFAzlatPmFhITGgfGU9UTAdiyZUtUJYlIiVAwlKG9e/fStv1dUjWTB6alatIXtv3mN7+JqiwRKREKhjL0xhtvAJCs3R8MXlWHVVazYcOGqMoSkRKhYChDa9euBYuRbJi+f6IZffUzWL16De55e0YXkTKhYCgz7s4TTz5Fsn4axKtyXks2zaK9fQdvvvlmRNWJSClQMJSZl156iW1bt9A/5agDXuufNA+LV3D//fdHUJmIlAoFQ5m58847sYoJ9E9ecOCLFRPYN2k+Dz/8iDrUEyljCoYy8tJLL/H000/TO/29EM/f43rfzGPp6+/njjvuKHJ1IlIqFAxlIpVKcdPNN2NVtfRNfy8AE7Y+S7y7nXh3OzUbHmTC1mfx6ib6mhfzwAMP0NLSEm3RIhIJBUOZ+OUvf8mG116je9YJEK8EINa9E0v2Y8l+Kva+Q6x7JwB9RxyHxyq58Tvf0RlKImVIwVAG9u3bx/e+/31S9c0k8hx0Hswra+iZeRxr16zh+eefL0KFIlJKFAxlYOXKlexsb6d31glgFmqZ/mlLoLqBW2+9Ta0GkTKjYBjnEokEP/7JT0g2zCDZMDP8grE4vTOO5fXXN6QviBORsqFgGOeeeeYZdra3pw84h2wtZPRPWYhVVrNixYoCVScipUjBMM4tv+8+mFBHYuKRB79wrIJ9U47iySefHLgVqIiMfwqGcaylpYV1a9eyb+oSsEP7p+5rXkIqlVKrQaSMKBjGsdtuuw2LVdDf/J5DXodXN5KYNId7ly/X1dAiZULBME499dRTrFq1it6Zx+KVNYe1rt5ZJ9LT08sNN9wwStWJjExnw0VHwTAOtba28rWvfx2vnUzfjGMPe31e00TvzONYtWoVy5cvH4UKRcKzgzxpQg6fgmGc2blzJ1dc+Y909vTTtfBMiI3OP3HfzN8hMXEO199wA6tWrRqVdYqEoZZD8SkYxpHu7m4+97nP07ajnc6jPoRXN43eyi1Gz4IzSNU1c911/8ILL7wweusWyUOBEJ2CB4OZnWVmr5vZRjP7Qp7XJ5jZXcHrz5nZvELXNB719/fzlX/+Zzb+ZiNdC84gVd88+m8Sr6Br0YdIVNbxhS9+kU2bNo3+e4gEMsGQSqUirqT8FDQYzCwO3AScDSwFLjazpYNm+ytgl7sfBfw78PVC1jRe3XPPPaxbu5aeuaeSPJRrFsKqqKZz0YfpTcC1116nX3VSMDt27Ii6hLJV6BbDScBGd9/k7n3AncD5g+Y5H8h0/v8z4IOmo00Hpa+vj7vv+RnJxiNIHMapqWH5hHq6Z51AS8tmVq9eXfD3k/K0bds2AHbt3BlxJeWn0MEwC9iWNd4aTMs7j7sngA5gSoHrGlc2bNjA7l076Zu66OAWTPZRXV3NBRdcQHV1NST7Qi+amLwAYnGeeOKJg6xWJJzMdTM7duiq+2IrdDDk++U/eN9DmHkws0vMbI2ZrWlraxuV4saLhQsXUlFRSbzz4D4XS/Rx7rnncumll3LOOedgifDBEOveCakkS5cO3jMoMjo6O/cGz10RV1J+8t/fcfS0Atk7vGcDbw8xT6uZVQBNwAFtR3e/BbgFYNmyZdqxnaWuro5TTjmZJ59+lkTTrNDHGLyiigceeAB3Z+XKlXhFbajlrL+Hmq3PUFFZyWmnnXY4pYsMqbl5GvAKzc0FOJFChlXoFsNqYJGZzTezKuAiYHCnOyuAvwiGLwB+5TqiedA++9nPsmDeXGo3/pLKttdDLZNqmEFnrJa773+YzlgtqYYZIy5jvR3Ub1jJhP49XPvVr9LQ0HC4pYvktXjx4vTzkqMjrqT8WKG3wWb2EeD/AHHgVnf/VzO7Fljj7ivMrBr4EXA86ZbCRe4+7HmQy5Yt8zVr1hS07rGou7ubq66+mjWrV5OYNI/eOSfjVeFaASPyFJXvvkLN27+mvq6Ob3z9axx9tP7DSuG4Ox0dHTQ2NhIbpQs1y52ZrXX3ZSPONxZ/nCsYhpZIJLjrrru4/fbbSbjRc8TxpGoP81h+sp+at9dhXe2cfPLJXH755UyfPn10ChaRogkbDIU+xiBFVlFRwcc//nFOP/10vv3tb7Nu3XOjst6Jkybz2X+6htNPP11914iMc2oxjGPuzmuvvUZPT89hrcfMWLx4MXV1daNUmYhEQS0Gwcx0OqmIHDQd0RERkRwKBhERyaFgEBGRHAoGERHJoWAQEZEcCgYREcmhYBARkRxj8gI3M2sDtkRdxzgyFdDtsqQU6bs5uua6+4jd1Y7JYJDRZWZrwlwNKVJs+m5GQ7uSREQkh4JBRERyKBgEgjvjiZQgfTcjoGMMIiKSQy0GERHJoW63xyEzSwIvZU36Y3dvGWLeecAD7n5M4SsTATObAvx3MDoDSAJtwfhJ7t4XSWEyQMEwPvW4+3FRFyGSj7u3A8cBmNk1QKe7/1v2PJa+TaC5e6r4FYp2JZUJM5tnZk+Y2brg8bt55nmvmT1vZuvN7EUzWxRM/0TW9O+bWbz4f4GMd2Z2lJm9bGbfA9YBR5rZ7qzXLzKzHwTD083sXjNbE3w3T46q7vFIwTA+1QQb8fVmtjyYth34A3d/H3AhcEOe5f4OuD5obSwDWs3s6GD+U4PpSeDjhf8TpEwtBf6fux8PvDXMfDcA3wgufvtT4AfFKK5caFfS+JRvV1Il8B0zy2zc35NnuWeAL5vZbOBed3/TzD4InACsTrfuqSEdMiKF8Bt3Xx1ivg8Bi4PvJMAkM6tx98O7wbkACoZycjnwLvA/SLcUewfP4O7/YWbPAecAD5vZXwMG3OHuXyxmsVK2urKGU6S/fxnVWcOGDlQXjHYllY8m4Lf256AxAAAExUlEQVTBwbw/Bw44TmBmC4BN7n4DsAI4lvTZIxeY2bRgnslmNrd4ZUu5Cr6ru8xskZnFgD/JevmXwKcyI0FLWEaJgqF83Az8hZk9S3o3UleeeS4EXjaz9cAS4Ifu/irwFeARM3sR+AUws0g1i3weeIj0D5TWrOmfAk4NTpJ4FfibKIobr3Tls4iI5FCLQUREcigYREQkh4JBRERyKBhERCSHgkFERHIoGEREJIeCQSJhZk8XcN2PmVlJ3EDezCaa2T9EXUc2M2sxs6lR1yGlS8EgkXD3A3p3HacmAiUVDCIjUTBIJMysM3g+I/iF/zMz22BmP7GsntHyLHeimT1tZi8E3S03mFmNmd0ZXAV7F+mO/jCzuJndHnTl/JKZXT7Meh8zs68H63zDzH4vmF5tZrcFy//azM4cZh35ui3/GrAwmPZNS/tmVk0XZn0Oj5vZcjN71cy+F3QDMdR7nRV0n/6Cmf13MG2ymd0XvPezZnZsMH2KmT0S1P99svofMrMrglpeNrPPBtPmmdlrZvZ/zeyVYNmaoWqRccjd9dCj6A/SN2cBOAPoAGaT/qHyDHDaEMtUAZuAE4PxRtIdQV4B3BpMOxZIkO42/ATgF1nLTxymnseAbwXDHwF+GQxfCdwWDC8BtgLVQ6zjRuDjWbXWAPOAl7Pm+RjpbkXiwPRgfTODz6EXWBC89gvggiHepxnYBswPxidnvf/VwfAHgPXB8A3AVcHwOYADU4PP5yWgDqgHXgGOD2pOAMcFy9wNfCLq74wexXuoxSCl4Hl3b/V0p2nrSW+Y8llMuiPA1QDuvsfdE8DvAz8Opr0IvBjMvwlYYGY3mtlZwJ4R6rg3eF6bVcNpwI+CdW8AtpC/y3JIh9qXzOzzwFzP3wX0acBP3T3p7u8Cq4ATg9eed/dN7p4EfhrMm8/JwOPuvjmoa2eeWn8FTDGzJnI/n5XArqz5l7t7l7t3Bn//7wWvbXb39Xk+DykDCgYpBfuyhpMM3R28kf61m88B0919F+luxh8j3enaSDdzydSRXcOQu7XyvN9/AOcBPaS7Lf9AntmGW9/gv2Gov3WozyHfun3Qc9hawv6byDikYJCxZANwhJmdCBAcX6gAHie4q5yZHUN6dxLBmTcxd/9P4J+B9x3Ce2av+z3AHOD1fDMO0W35XqBh0PouDI5/NJP+Nf988NpJZjY/OLZwIfDkEDU9A5xuZvOD952cp9YzgB3uvmfQ9LOBSVnz/7GZ1ZpZHelurZ8I86HI+KZfATJmuHtfcLD2xuBgaA/pO3l9F7gt6BZ8Pfs3tLOC6ZkfQIdys6Gbge+Z2Uuk97t/0t33DTHvhcAnzKwfeAe41t13mtlTZvYy8F/A54BTgBdI/4r/nLu/Y2ZLSG/wvwb8DumN9vJ8b+LubWZ2CXBv8LdtB/4AuCbrc+gG/iJY5KvAT81sHeldV1uD9awzs9vZ/3n9wN1/bWbzDvIzknFG3W6LlIDgF/4/uvu5Udciol1JIiKSQy0GKUlmthyYP2jy59394cNc703AqYMmX+/utx3EOv4Q+PqgyZvd/U/yzX84LH0P7gmDJv+5u7802u8lkqFgEBGRHNqVJCIiORQMIiKSQ8EgIiI5FAwiIpJDwSAiIjn+P0FVBUEXKCEVAAAAAElFTkSuQmCC\n",
        "text/plain": "<Figure size 432x288 with 1 Axes>"
       },
       "metadata": {
        "needs_background": "light"
       },
       "output_type": "display_data"
      }
     ],
     "persistentId": "72511a16-18da-4358-93cf-c36a17c23f8e",
     "text": "\n\nsns.violinplot(x='in_cds_no_stop_codon', y='jaccard_in_peptide_db', data=coding_scores_not_low_complexity_unspliced, \n               bw=0.1, cut=True)"
    },
    "executionTime": "2019-10-12T04:12:24.390Z"
   },
   {
    "cell": {
     "executionCount": 86,
     "executionEventId": "bb2b3899-e398-4ea2-aba5-8ba5573bf9a9",
     "hasError": false,
     "id": "c4e90627-a46a-4fa2-b552-6402975ee8c7",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "accuracy score: 0.9077277970011534\n              precision    recall  f1-score   support\n\n       False       1.00      0.90      0.94       756\n        True       0.58      0.99      0.73       111\n\n   micro avg       0.91      0.91      0.91       867\n   macro avg       0.79      0.94      0.84       867\nweighted avg       0.95      0.91      0.92       867\n\n"
      },
      {
       "data": {
        "image/png": "iVBORw0KGgoAAAANSUhEUgAAAYUAAAEKCAYAAAD9xUlFAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDMuMC4zLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvnQurowAAIABJREFUeJzt3Xd4FWX2wPHvSSGhhA4iIEUFFVEQIygqFooozVVXUQRRBEVR17bqz4K4rotdFBEiYEXFZVGjwiIqWJCqiAjoGooQFGkBQhop5/fHXOI1huSmzJ1bzud57pM7cyczZ1Luued9Z95XVBVjjDEGIMbrAIwxxoQOSwrGGGOKWVIwxhhTzJKCMcaYYpYUjDHGFLOkYIwxppglBWOMMcUsKRhjjClmScEYY0yxOK8DqKjGjRtrmzZtvA7DGGPCytdff71TVZuUt13YJYU2bdqwYsUKr8MwxpiwIiI/B7KdNR8ZY4wpZknBGGNMMUsKxhhjillSMMYYU8ySgjHGmGKuJQURmS4i20Xk+0O8LiLyrIikich3ItLFrViMMcYExs1K4WWgbxmvnw+08z1GAS+4GIsxxpgAuHafgqp+LiJtythkEPCqOvOBLhGR+iJyuKr+6lZMxpgwowpaCEWFzlctBC3643Kpz4vKeO3g86Jy9lENxyq5r3K3Kz2mZf+rRWJcASee2Q26P+jqj9zLm9daAFv8ltN96/6UFERkFE41QatWrYISnDHlUvX7Jy469JtBqL1RhXRMJY5FdM8hrwp3fdibJz/rwomH/8ay9uuId/mYXiYFKWVdqX8BqpoCpAAkJyeH31+JFkHqJbDhA68jMdXiYDIo8jqQ6CAxILEQE+t8ldjS18WU+Hpwm7K2I6b010rbT5nbVeFYZZyXxMRC2hb4fCt9BnWjsGvXiE4K6cARfsstgV88isU9memw7FFIe8frSIwrpHrfPA7uJxhvVGFxrBiQ0j4/Rq49e3LZsCGDLl0OB2Dc0/kMHrWzeNltXiaFVGCMiLwFdAP2Rlx/wvZV8NpJFBdAA2fDkf08DclUk+I3weh6wzLueu+9Hxg9+kNiYoQ1a26gXr1EataMD1pCABeTgoi8CZwNNBaRdGAsOJWPqk4G5gAXAGlANnC1W7G47uePIeN/f17/0zsUJ4TTxkK7vwQ1LGNMeNi+PYubb57LzJlrADj11Jbs2ZNLvXqJQY/FzauPLi/ndQVudOv4QbN3E8zqXfY2x17u+hUDxpjwo6rMmLGaW275L7t351CrVjyPPHIuY8Z0JTbWm3uLw27o7JCT8aPzNalV6U1DMfHQ6brgxmSMCQujR3/IlClfA9Cr15GkpPSnbdsGnsZkSaGq9m12vrbqCb0meRuLMSasXHjhscycuYYnn+zD1Vd3RkKgj8qSQlVl+pJCXbt/whhTtp9+2sUnn2zk+uuTAejb92g2bbrFk76DQ7GkUFUHK4UkSwrGmNIVFBTx1FOLGTt2IXl5BXTu3IxTT20JEFIJASwpVJ1VCsaYMqxatY0RI1L5+mvnivthwzrRrl1Dj6M6NEsKVWWVgjGmFHl5BTz88OeMH7+IgoIiWrWqx5Qp/enb92ivQyuTJYWq0CLI3OI8Tzqi7G2NMVHlnns+4emnlwBw442n8K9/9SQpKcHjqMpnSaEqsn6Donyo2QTia3odjTEmhPz976ezeHE6jz3WizPPbO11OAGzmdeqYt/PzlfrTzAm6s2fv56LL36bggJnoMRmzerw1VfXhFVCAEsKVZNp/QnGRLuMjBxGjHiPPn1eZ/bsdbz00sri10LhvoOKsuajqthnVx4ZE83eeWcdN9wwh23b9pOQEMvYsWcxfHhnr8OqEksKVWGVgjFRadu2/dx001xmzVoLQPfuRzBt2kCOPbaxx5FVnSWFqrBKwZio9N57PzBr1lpq145n/Phe3HDDKcTEhF9TUWksKVSFVQrGRI3c3AISE523zJEjT2bDhgxGjz6FNm3qexxZ9bKO5qqwSsGYiFdUpEycuIy2bSfw8897AIiJER59tHfEJQSwpFB5+VmQuwtia0Ctpl5HY4xxwY8/7qRHj5e46aa5bNu2nzff/N7rkFxnzUeVtc/vTmax3GpMJMnPL+SJJ75i3LjPyMsr5LDDajNpUj8uuug4r0NznSWFyrL+BGMi0vffb2fYsHdYuXIbAFdf3Zknn+xDgwbRMWqBJYXKsv4EYyJSUZGyevV2WreuR0rKAPr0OcrrkILKkkJlWaVgTMRYs2Y7HTo0QUQ48cTDeO+9wfTo0Zo6dWp4HVrQWWN4ZVlSMCbsZWbmMWbMHDp2fIH//Gdd8foLLmgXlQkBrFKoPGs+MiaszZuXxqhRH7B5817i4mLYtGmP1yGFBEsKlWWVgjFhaffuHG69dR6vvroKgC5dDmfatIF07tzM48hCgyWFyvCfXKeuTa5jTLj49ttt9O37Or/9lkVCQizjxp3N7bd3Jy7OWtIPsqRQGdnbofAAJDaC+NpeR2OMCVD79o2oU6cG7ds3YurUgbRv38jrkEKOJYXKsP4EY8KCqvLGG6sZMOAY6tZNoFateBYuHE7z5kkRM4BddbOaqTKsP8GYkLdp0x7OO+91rrzyHe6+++Pi9S1b1rWEUAarFCrDKgVjQlZhYRGTJi3nnns+ISsrn4YNa9K9u/X9BcqSQmVYpWBMSFq3bgcjRqSyeHE6AJdeejzPPXc+TZta31+gLClUhlUKxoScjRsz6Nx5CgcOFHL44XWYNKkfF154rNdhhR1Xk4KI9AUmALHAVFUdX+L1VsArQH3fNner6hw3Y6oWVikYE3Latm3AX//agcTEOJ54og/16yd6HVJYci0piEgs8DzQG0gHlotIqqqu9dvsPuBtVX1BRDoAc4A2bsVUbaxSMMZzOTn5PPTQZ/zlL8fRtWsLAF555UJiY+36mapw86fXFUhT1Q2qegB4CxhUYhsF6vqe1wN+cTGe6pGfAzk7ICYeatsdkMZ44YsvfqZz5ymMH7+IUaPep6hIASwhVAM3m49aAFv8ltOBbiW2eRD4SERuAmoDvVyMp3ocvJM5qaVNrmNMkO3bl8c993zMpEkrAOjQoQmTJ/e3S0yrkZvvaqX9lrTE8uXAy6raErgAeE3kz++0IjJKRFaIyIodO3a4EGoFWH+CMZ6YM+cnOnacxKRJK4iLi+GBB3rwzTejOPXUll6HFlHcrBTSAf+Lg1vy5+ahEUBfAFVdLCKJQGNgu/9GqpoCpAAkJyeXTCzBZf0JxgTd3r25DBkymz17cklObs60aQM58cTDvA4rIrmZFJYD7USkLbAVGAxcUWKbzUBP4GUROQ5IBDwuBcphlYIxQaGqqEJMjFCvXiLPPtuX337L4m9/O9UGsHORa0lBVQtEZAwwD+dy0+mqukZEHgJWqGoqcDvwoojcitO0NFxVva0EymOVgjGu++WXTG644UPOPLMVt9/eHYChQzt5HFV0cPU+Bd89B3NKrHvA7/la4HQ3Y6h2VikY4xpVZfr0ldx++0fs3ZvHkiXp3HDDKdSsGe91aFHD7miuqEyrFIxxw4YNGYwc+T6ffroRgH792jF5cn9LCEFmSaEiVH9vPkqyAbaMqQ6FhUU8++xS7r33U3JyCmjcuBbPPtuXwYM7ImKXmgabJYWKyNkBhXmQ2ABqJHkdjTERY9asdeTkFHD55R2ZMKEvTZrYAHZesaRQEfusP8GY6nDgQCGZmXk0alSL2NgYpk0byE8/7WLAgGO8Di3q2XVdFWGdzMZU2fLlW0lOTmHo0Hc4eLHhscc2toQQIqxSqAi7HNWYSsvOzmfs2AU89dQSioqU7Ox8tm/P4rDD6ngdmvFjSaEirFIwplIWLtzEyJHvk5a2m5gY4Y47TmPcuHOoVcuuLAo1lhQqwioFYypEVbn55rlMnLgcgBNOaMq0aQM55ZQWHkdmDsWSQkVYpWBMhYgIdesmEB8fw3339eDuu8+gRo1Yr8MyZbCkUBFWKRhTrp07s1m/fjfdujmjl95//1kMGXIiHTo08TgyE4iArj4SkRoicrTbwYS0glzI/g0kFmof7nU0xoQcVeWtt77nuOOe58ILZ5KRkQNAYmKcJYQwUm5SEJF+wGpgvm+5s4i843ZgIScz3fma1BJirPw1xl96+j4GDXqLyy//Dzt3ZtOhQxOys/O9DstUQiDNRw/hzJi2AEBVv43KqsH6E4z5k6IiZerUb7jzzvns25dH3boJPPlkH0aMOMmGqAhTgSSFfFXdU+IXHNrDW7vB+hOM+ZMRI1J5+eVvARg48BgmTbqAFi3qlvNdJpQF0qewTkQuBWJEpK2IPAMscTmu0GOVgjF/cuWVJ9C0aW3eeuti3n33MksIESCQpDAGOBkoAmYDucAtbgYVkqxSMIbvv9/OhAm/fybs2fNINmy4mcsusxFNI0UgzUfnqepdwF0HV4jIRTgJInpYpWCiWF5eAf/615c88sgX5OcXkZzcnNNPd/4Xateu4XF0pjoFUincV8q6e6s7kJBnlYKJUkuXpnPyySmMG/cZ+flFjB6dzAknHOZ1WMYlh6wUROQ8oC/QQkSe8nupLk5TUvRQtUrBRJ2srAPcf/8CnnlmCarQrl1Dpk4dSI8erb0OzbiorOaj7cD3OH0Ia/zWZwJ3uxlUyMnZBQU5kFAPEqwjzUSHe+/9lAkTlhITI9x552k8+ODZNjVmFDhkUlDVlcBKEZmhqrlBjCn0WJVgotC9957J6tXbefTRXiQnN/c6HBMkgfQptBCRt0TkOxH538GH65GFEutPMFEgNfVHLrhgBvn5hQA0aVKbTz4ZZgkhygSSFF4GXgIEOB94G3jLxZhCj1UKJoJt357F4MGzGDToLebOTeOVV1Z5HZLxUCBJoZaqzgNQ1fWqeh9wjrthhRibm9lEIFXl9de/47jjnmfmzDXUqhXPhAl9ufrqzl6HZjwUyH0KeeLclbJeRK4HtgJN3Q0rxGRa85GJLJs37+X66z9g7tw0AHr1OpKUlP60bdvA48iM1wJJCrcCdYCbgX8C9YBr3Awq5FjzkYkwH320nrlz06hfP5GnnurD8OGd7Y5kAwSQFFR1qe9pJjAUQERauhlUyLGOZhMBsrIOFN99PGLESWzduo9Ro07m8MOTPI7MhJIy+xRE5BQRuVBEGvuWjxeRV4mmAfEK8iDrV5AYqGNXYZjwU1BQxGOPLaJ162fYsCEDcKbJHDv2bEsI5k8OmRRE5F/ADGAI8F8RuRdnToVVQPvghBcC9m91vtZpATE2e6kJL6tWbaNbt6ncddfH7NqVw7vv/uB1SCbElfUuNwjopKo5ItIQ+MW3/GNwQgsR1p9gwlBeXgEPP/w548cvoqCgiFat6pGS0p/zzou++bFMxZSVFHJVNQdAVXeLyA9RlxDA+hNM2Fm58leGDJnNunU7EYExY07hkUd6kpSU4HVoJgyUlRSOFJGDw2ML0MZvGVW9qLydi0hfYAIQC0xV1fGlbHMp8CDObG6rVPWKwMMPAqsUTJhJSIhj/foMjjmmEVOnDuSMM+xv1wSurKRwcYnliRXZsYjEAs8DvYF0YLmIpKrqWr9t2gH3AKeraoaIhN79D1YpmDDwzTe/ctJJzRAROnRowty5Q+je/QgSE60fzFRMWQPifVLFfXcF0lR1A4CIvIXTT7HWb5uRwPOqmuE75vYqHrP6WaVgQlhGRg533PER06d/y5tvXszgwR0BOPfcth5HZsKVmx8jWgBb/JbTgW4ltmkPICKLcJqYHlTV/5bckYiMAkYBtGoV5DdnqxRMiHrnnXXccMMctm3bT0JCLLt2ZXsdkokAbiaF0m6P1FKO3w44G2gJfCEiHVV1zx++STUFSAFITk4uuQ/32OQ6JgRt27afm26ay6xZTtF9+ulHMHXqQI49trHHkZlIEHBSEJEEVc2rwL7TgSP8llviXNZacpslqpoPbBSRH3GSxPIKHMc9uRmQnwU1kpwJdozx2Ndf/0Lv3q+RkZFL7drxjB/fixtuOIWYGBuiwlSPckdJFZGuIrIa+Mm33ElEngtg38uBdiLSVkRqAIOB1BLbvItvxFXfXdPtgQ0ViN9d/lWCjQtjQkCHDk1o0qQ25513FGvW3MCYMV0tIZhqFcjQ2c8C/YFdAKq6igCGzlbVAmAMMA9YB7ytqmtE5CERGejbbB6wS0TW4twtfaeq7qr4abjE+hOMx4qKlJSUr9mzx5n8sGbNeD7/fDhz5w6hdev6HkdnIlEgzUcxqvpziREUCwPZuarOAeaUWPeA33MFbvM9Qo/1JxgP/fjjTq699n2+/HIzy5dv5cUXnc9Shx1Wx+PITCQLJClsEZGugPruPbgJiI7pOK1SMB7Izy/kyScX8+CDC8nLK6RZszqcf347r8MyUSKQpDAapwmpFfAb8LFvXeSzSsEE2cqVvzJiRCorV24D4OqrO/Pkk31o0KCmx5GZaBFIUihQ1cGuRxKKrFIwQbR+/W66dp1KQUERbdrUJyWlP717H+V1WCbKBJIUlvsuFZ0JzFbVTJdjCh1WKZggOuqohgwdeiJJSTX45z97UqdODa9DMlGo3KuPVPUo4GHgZGC1iLwrIpFfORTmw/5fAHHmUjCmmu3ff4Cbb57L4sW/3/g/bdpAJkw43xKC8Uwgl6Siql+p6s1AF2AfzuQ7kW3/VkCd2dZi472OxkSYefPSOP74STz33DKuv/5DnAvxsHmSjefKbT4SkTo4A9kNBo4D3gO6uxyX96zpyLhg9+4cbr11Hq++ugqAk08+nGnTBloyMCEjkD6F74H3gcdU9QuX4wkd1slsqtmsWWu58cY5bN+eRWJiHOPGnc1tt51GXFxABbsxQRFIUjhSVYtcjyTUWKVgqtGePbmMGvU+GRm59OjRmhdfHED79o28DsuYPzlkUhCRJ1X1duA/IvKnkUkDmXktrFmlYKpIVSkqUmJjY6hfP5FJk/qRkZHDddcl23hFJmSVVSnM9H2t0IxrEcMqBVMFmzbtYdSo9zn33LbcffcZAMUT4BgTyg7ZmKmqy3xPj1PVT/wfOB3Okc0qBVMJhYVFPPvsUjp2nMT8+RuYOHEZubkFXodlTMAC6eG6ppR1I6o7kJCiCvt+dp5bpWACtG7dDnr0eJlbbvkvWVn5DB7ckW++uc7mSTZhpaw+hctwLkNtKyKz/V5KAvaU/l0RIm8v5O+H+NqQ2MDraEyIKygo4tFHv+Shhz7nwIFCmjdP4oUX+jFw4DFeh2ZMhZX1EWYZzhwKLYHn/dZnAivdDMpzNrmOqYCYGOGjjzZw4EAhI0d24bHHelO/fqLXYRlTKYdMCqq6EdiIMypqdLH+BFOOnJx8MjMP0LRpbWJihKlTB7Blyz7OPbet16EZUyWH7FMQkc98XzNEZLffI0NEdgcvRA/YlUemDJ9//jOdOk3myitnFw9P0a5dI0sIJiKU1Xx0cMrNxsEIJKRYpWBKsW9fHvfc8zGTJq0AID4+lp07s2nSpLbHkRlTfcq6JPXgXcxHALGqWgicBlwHRPZ/gVUKpoS5c3+iY8dJTJq0gri4GMaOPYtvvhllCcFEnECulXsXOEVEjgJeBT4E3gD6uxmYp6xSMD6qysiR7zNtmnNtRXJyc6ZPH8gJJxzmcWTGuCOQ+xSKVDUfuAh4RlVvAiJ7ggGrFIyPiNCyZV0SE+N44oneLF48whKCiWgBTccpIn8FhgIX+tZF7gQDRQW+uRRscp1o9csvmaxfv5szz2wNwP/935kMHXoiRx3V0OPIjHFfoHc0n4MzdPYGEWkLvOluWB7a/wtoEdRuBnEJXkdjgkhVmTbtGzp0eJ6LL36bXbuyAahRI9YSgoka5VYKqvq9iNwMHC0ixwJpqvpP90PziPUnRKUNGzIYOfJ9Pv10IwD9+7cnPz/6Row3JpCZ184EXgN8bSo0E5GhqrrI7eA8Yf0JUeXgAHb33beA7Ox8GjeuxbPP9mXw4I42G5qJSoH0KTwNXKCqawFE5DicJJHsZmCe2WdJIZoMG/Yub7yxGoArrjiBZ545zy4zNVEtkD6FGgcTAoCqrgNquBeSxzKt+SiajBzZhZYt65KaOpgZMy6yhGCiXiCVwjciMgWnOgAYQiQPiGfNRxFt+fKtfPrpRu66y5n45uyz25CWdhMJCTa8tTEQWFK4HrgZ+DtOn8LnwHNuBuUp62iOSNnZ+Ywdu4CnnlpCUZHSvfsRxZecWkIw5ndl/jeIyAnAUcA7qvpYcELymFUKEWfhwk1ce20q69dnEBMj3HHHaZx8cnOvwzImJJU1Sur/4QxxMQSYLyKlzcBWJhHpKyI/ikiaiNxdxnaXiIiKiLed13l7nUdcTajZyNNQTNXt3ZvLdde9zznnvML69RmccEJTliwZweOP96FWrci9/9KYqiirUhgCnKiqWSLSBJgDTA90xyISizM5T28gHVguIqn+nda+7ZJwmqeWVjT4ape5xflqk+tEhPvvX0BKyjfEx8dw//09uOuuM6hRI9brsIwJaWUlhTxVzQJQ1R0iEsiVSv664tzotgFARN4CBgFrS2z3D+Ax4I4K7r/6WX9C2FPV4vsLHnjgLDZu3MP48T05/vimHkdmTHgo643+SBGZ7Xu8Axzltzy7jO87qAWwxW85nRID6YnIScARqvpBhSN3g/UnhC1V5Y03VnPuua9y4EAhAI0b1+L99y+3hGBMBZRVKVxcYnliBfddWvuLFr/oVB5PA8PL3ZHIKGAUQKtWLr5hW6UQltLT9zF69Id88MH/AJgx4zuuvvokj6MyJjyVNUfzJ1XcdzrOBD0HtQR+8VtOAjoCC33lfjMgVUQGquqKErGkACkAycnJilusUggrRUXKiy9+zZ13zicz8wD16iXw5JN9GD68s9ehGRO23LxAeznQzjeq6lZgMHDFwRdVdS9+U32KyELgjpIJIaisUggbaWm7GTnyfRYu3ATAoEHHMGlSP5o3T/I2MGPCnGtJQVULRGQMMA+IBaar6hoReQhYoaqpbh270qxSCBtffPEzCxduomnT2kyceD6XXNLBBrAzphoEnBREJEFV8yqyc1Wdg3Mpq/+6Bw6x7dkV2Xe1KyqEzHTneVJLT0MxpduzJ5f69RMBGD68Mzt2ZDNixEk0alTL48iMiRzlXmYqIl1FZDXwk2+5k4hE3jAXWb+CFkKtwyAu0etojJ+8vALGjl1A69bP8NNPuwBnmsy///10SwjGVLNAKoVngf44dzejqqtE5BxXo/KC9SeEpCVL0hkxIpW1a3cAMG/eetq1s7vNjXFLIEkhRlV/LtFeW+hSPN6x/oSQkpV1gPvvX8AzzyxBFdq1a8i0aQOLB7EzxrgjkKSwRUS6AuobuuIm4H/uhuUBqxRCxtKl6VxxxWw2bMggNla4447ujB17FjVr2nhFxrgtkKQwGqcJqRXwG/Cxb11ksUohZNSvn8jWrfvo1Okwpk0baCOaGhNE5SYFVd2Oc49BZLNKwVNffrmZ008/AhHhmGMa8+mnV3HKKc2Jj7cB7IwJpnKTgoi8iN/wFAep6ihXIvKKVQqe2L49i5tvnsvMmWt45ZULGTasEwDdux9RzncaY9wQSPPRx37PE4G/8MeB7iKDzc0cVKrKjBmrueWW/7J7dw61asUXD2RnjPFOIM1HM/2XReQ1YL5rEXnhQCbkZkBsAtRs4nU0EW/z5r1cf/0HzJ2bBkDv3keSkjKANm3qexyZMaYyw1y0BSLrusCDk+vUtcl13LZ0aTq9er3G/v0HqF8/kaefPo+rrupkQ1QYEyIC6VPI4Pc+hRhgN3DIqTXD0j7rTwiWzp2bccQRdTn22MY8//wFHH64DWBnTCgpMymI8/GtE84opwBFqure0NVesU5m1xQUFDFx4jKGDetEw4Y1SUiIY9Gia2jQoKbXoRljSlHm2Ee+BPCOqhb6HpGXEMAuR3XJqlXb6NZtKrfeOo/bbptXvN4SgjGhK5B5l5eJSBfXI/GSVQrVKje3gPvu+5Tk5Bf55ptfadWqHpdf3tHrsIwxAThk85GIxKlqAXAGMFJE1gNZONNsqqpGTqKwSqHafPXVFkaMSOWHH3YiAmPGnMIjj/QkKSnB69CMMQEoq09hGdAFuDBIsXjHKoVqkZa2mzPPfImiIuWYYxoxbdpATj/dfqbGhJOykoIAqOr6IMXijT9MrmN30VbF0Uc3ZNSoLjRsWJP77z+LxEQ3Z3s1xrihrP/aJiJy26FeVNWnXIgn+LJ/g6J856a1eOsArYiMjBxuv/0jrr66c/GQ1pMm9bN7DowJY2UlhVigDr6KIWJZf0KlzJ69jhtvnMO2bfv5+utf+fbb6xARSwjGhLmyksKvqvpQ0CLxivUnVMi2bfsZM2YO//nPOgDOOKMVU6cOsGRgTIQot08h4lmlEBBV5dVXV3HrrfPIyMilTp0aPPpoL66/PpmYmOj4UzEmGpSVFHoGLQovWaUQkD17crn99o/IyMilb9+jmTy5H61b2wB2xkSaQyYFVd0dzEA8Y5XCIRUVKUVFSlxcDA0a1GTKlP5kZ+dz5ZUnWnORMREqkDuaI5tVCqX64Yed9OjxEuPHf1m87uKLOzB0qI1oakwks6RglcIf5OcX8sgjX9Cp02QWLdrCtGkryc0t8DosY0yQRPfdRflZkLsLYmtAraZeR+O5lSt/5ZprUvn2220AjBhxEo8/3ttuQjMmikT3f/s+3+Q6SUeARG/RlJ9fyNixC3nssUUUFipt2tTnxRcH0KvXkV6HZowJsuhOCtafAEBcXAxLl26lqEi55ZZuPPzwudSpU8PrsIwxHojupBDF/QmZmXlkZh6gefMkRISpUwewbdt+TjvNxn8yJppFb5sJRG2lMG9eGh07vsCQIbM5OG9S27YNLCEYYywpAFC3tbdxBMmuXdlcddW79O07g82b95KZmceuXTleh2WMCSGuJgUR6SsiP4pImojcXcrrt4nIWhH5TkQ+EZHgvjvvi45KQVWZNWstHTpM4tVXV5GYGMdjj/ViyZJrady4ltfhGWNCiGt9CiISCzwP9AbSgeUikqqqa/02Wwkkq2q2iIwGHgMucyumP8mM/D4FVWXIkNm8+eb3APTo0ZoXXxxA+/aNPI7MGBOK3KwUugJpqrpBVQ8y0uVLAAAPNklEQVQAbwGD/DdQ1QWqmu1bXAK0dDGeP9IiyPS7JDVCiQgdOjQhKakGL7zQjwULrrKEYIw5JDevPmoBbPFbTge6lbH9CGBuaS+IyChgFECrVtX0qT57OxQegJqNIT6ymlA2bsxgw4YMevZ07jO4667TGT68My1b1vU4MmNMqHOzUihtgBwtdUORK4Fk4PHSXlfVFFVNVtXkJk2aVE90EdifUFhYxIQJS+jY8QUuu2wW27dnARAfH2sJwRgTEDcrhXTAv12mJfBLyY1EpBdwL3CWqua5GM8fRVh/wtq1O7j22lQWL3bmmx448Bib58AYU2FuJoXlQDsRaQtsBQYDV/hvICInAVOAvqq63cVY/ixCKoX8/EIefXQR//jH5xw4UEjz5km88EI/Bg48xuvQjDFhyLWkoKoFIjIGmIcz3/N0VV0jIg8BK1Q1Fae5qA7wb99wzJtVdaBbMf1BhFQKV1wxm1mznAu6Ro7swuOP96ZevUSPozLGhCtXh7lQ1TnAnBLrHvB73svN45cpQiqFW27pxrffbmPKlP6ce25br8MxxoS56L2jOUwrhc8+28S4cQuLl884oxXr1t1oCcEYUy2id0C8MKsU9u3L46675jN58tcAnHNOW3r0cG4Aj4uL3txujKle0ZkU8rMhZwfExEPtw7yOplxz5vzEddd9QHr6PuLjY7j33jM59dTg3ednjIke0ZkUMsNjcp2dO7P529/+y4wZqwHo2rUF06YNpGNHmyXOGOOO6EwKYTKPwkMPfcaMGaupWTOOhx8+l1tu6UZsbOgmMWNM+IvOpBDC8yioKr7Lcxk37mx++y2LRx45l6OOauhxZMaYaBCdHztDsFJQVV588Wu6d59Obm4BAA0a1GTmzEssIRhjgiY6k0KIVQrr1++mZ89XGTXqA5YsSeftt9d4HZIxJkpFd/ORx5WCM4DdUu6771Nycgpo0qQWzz13PpdeeryncRljold0JoUQuEdhzZrtXHNNKsuWbQVgyJATeOaZvjYTmjHGU9GXFEJkcp2VK7exbNlWWrRIYsqU/vTr196zWIwx5qDoSwrZO6AwDxIbQo06QT30jh1ZNGlSG3Aqgz17chk69EQbwM4YEzKir6PZg07m7Ox87rjjI9q0mcC6dTsAZ5rMMWO6WkIwxoSU6KsUgnw56oIFGxk58n3Wr88gJkb4/POfOe64apo9zhhjqln0JYUgVQp79+by97/PJyXlGwBOOKEp06cPIjm5uavHNcaYqoi+pBCESuHLLzczePAstm7NJD4+hvvv78Fdd51BjRqxrh3TGGOqQ/QlhSBUCs2a1WHXrhxOPbUlU6cO4PjjbQA7Y0x4iL6k4EKloKrMn7+B3r2PREQ4+uiGfPnl1XTu3MwGsDPGhJXoe8eq5kphy5a9DBjwJued9zovvfRt8fqTT25uCcEYE3aiq1LIz4Hs7RATB7WbVWlXRUXOAHZ33jmfzMwD1KuXQEKC9RkYY8JbdCWF/enO1zotIabyb+A//bSLkSPf57PPfgbgwguP5fnnL6B586TqiNIYYzwTXUmhGvoTvvpqCz17vkpubgFNm9Zm4sTzueSSDsVzIBhjTDiLrqRQDf0JycnNadeuISeddDhPPdWHRo1sADtjTOSIrqRQiUohL6+AJ574iuuuS6Zx41rUqBHLokXXkJSU4FKQxhjjnehKChWsFJYsSWfEiFTWrt3BunU7ef31i5xvt4RgjIlQ0ZUUAqwUsrIOcN99nzJhwlJUoX37Rlx33clBCNAYY7wVXUkhgErhk082MHLk+2zcuIfYWOHOO7szduzZJCZG14/KGBOdouedTtUvKZQ+uc7//reL3r1fQxU6d27GtGkD6dLl8CAGaYwx3oqepJCzEwpyIaE+JNQtdZP27Rtxyy3daNKkNnfe2Z34eLsZzRgTXaInKWT+uT/ht9/2c/PN/+X660/mnHPaAvD00329iM4YY0KCq4PziEhfEflRRNJE5O5SXk8QkZm+15eKSBvXgtn3e3+CqvLaa6vo0GESb7+9hltvnYequnZoY4wJF64lBRGJBZ4Hzgc6AJeLSIcSm40AMlT1aOBp4FG34jlYKWzOaUO/fm8wbNi77N6dQ58+R/Huu4PtjmRjjMHd5qOuQJqqbgAQkbeAQcBav20GAQ/6ns8CJoqIqAsf24v2bGbyolO4a15j9men0aBBIk8/fR7DhnWyhGCMMT5uJoUWwBa/5XSg26G2UdUCEdkLNAJ2Vncwe39LZ9z8s9ifDRdffBwTJ15As2Z1qvswxhgT1txMCqV9/C5ZAQSyDSIyChgF0KpV5cYtapAUw9TLP+JA8lguHnVRpfZhjDGRzs2kkA743xDQEvjlENuki0gcUA/YXXJHqpoCpAAkJydXrmmp/5sM6GedycYYUxY3rz5aDrQTkbYiUgMYDKSW2CYVuMr3/BLgUzf6E4qJOA9jjDGlcq1S8PURjAHmAbHAdFVdIyIPAStUNRWYBrwmImk4FcJgt+IxxhhTPldvXlPVOcCcEuse8HueC/zVzRiMMcYEzmaWN8YYU8ySgjHGmGKWFIwxxhSzpGCMMaaYJQVjjDHFJNxGBxWRHcDPlfz2xrgwhEaIs3OODnbO0aEq59xaVZuUt1HYJYWqEJEVqprsdRzBZOccHeyco0Mwztmaj4wxxhSzpGCMMaZYtCWFFK8D8ICdc3Swc44Orp9zVPUpGGOMKVu0VQrGGGPKEJFJQUT6isiPIpImIneX8nqCiMz0vb5URNoEP8rqFcA53yYia0XkOxH5RERaexFndSrvnP22u0REVETC/kqVQM5ZRC71/a7XiMgbwY6xugXwt91KRBaIyErf3/cFXsRZXURkuohsF5HvD/G6iMizvp/HdyLSpVoDUNWIeuAM070eOBKoAawCOpTY5gZgsu/5YGCm13EH4ZzPAWr5no+OhnP2bZcEfA4sAZK9jjsIv+d2wEqggW+5qddxB+GcU4DRvucdgE1ex13Fc+4BdAG+P8TrFwBzcWauPBVYWp3Hj8RKoSuQpqobVPUA8BYwqMQ2g4BXfM9nAT1Fwnr2nXLPWVUXqGq2b3EJzkx44SyQ3zPAP4DHgNxgBueSQM55JPC8qmYAqOr2IMdY3QI5ZwXq+p7X488zPIYVVf2cUmag9DMIeFUdS4D6InJ4dR0/EpNCC2CL33K6b12p26hqAbAXaBSU6NwRyDn7G4HzSSOclXvOInIScISqfhDMwFwUyO+5PdBeRBaJyBIR6Ru06NwRyDk/CFwpIuk487fcFJzQPFPR//cKcXWSHY+U9om/5CVWgWwTTgI+HxG5EkgGznI1IveVec4iEgM8DQwPVkBBEMjvOQ6nCelsnGrwCxHpqKp7XI7NLYGc8+XAy6r6pIichjObY0dVLXI/PE+4+v4ViZVCOnCE33JL/lxOFm8jInE4JWdZ5VqoC+ScEZFewL3AQFXNC1JsbinvnJOAjsBCEdmE0/aaGuadzYH+bb+nqvmquhH4ESdJhKtAznkE8DaAqi4GEnHGCIpUAf2/V1YkJoXlQDsRaSsiNXA6klNLbJMKXOV7fgnwqfp6cMJUuefsa0qZgpMQwr2dGco5Z1Xdq6qNVbWNqrbB6UcZqKorvAm3WgTyt/0uzkUFiEhjnOakDUGNsnoFcs6bgZ4AInIcTlLYEdQogysVGOa7CulUYK+q/lpdO4+45iNVLRCRMcA8nCsXpqvqGhF5CFihqqnANJwSMw2nQhjsXcRVF+A5Pw7UAf7t61PfrKoDPQu6igI854gS4DnPA/qIyFqgELhTVXd5F3XVBHjOtwMvisitOM0ow8P5Q56IvInT/NfY108yFogHUNXJOP0mFwBpQDZwdbUeP4x/dsYYY6pZJDYfGWOMqSRLCsYYY4pZUjDGGFPMkoIxxphilhSMMcYUs6RgQo6IFIrIt36PNmVs2+ZQo0lW8JgLfSNxrvINEXFMJfZxvYgM8z0fLiLN/V6bKiIdqjnO5SLSOYDv+ZuI1KrqsU10sKRgQlGOqnb2e2wK0nGHqGonnMESH6/oN6vqZFV91bc4HGju99q1qrq2WqL8Pc5JBBbn3wBLCiYglhRMWPBVBF+IyDe+R/dStjleRJb5qovvRKSdb/2VfuuniEhsOYf7HDja9709feP0r/aNc5/gWz9efp+f4gnfugdF5A4RuQRnfKkZvmPW9H3CTxaR0SLymF/Mw0XkuUrGuRi/gdBE5AURWSHOPArjfOtuxklOC0RkgW9dHxFZ7Ps5/ltE6pRzHBNFLCmYUFTTr+noHd+67UBvVe0CXAY8W8r3XQ9MUNXOOG/K6b5hDy4DTvetLwSGlHP8AcBqEUkEXgYuU9UTcEYAGC0iDYG/AMer6onAw/7frKqzgBU4n+g7q2qO38uzgIv8li8DZlYyzr44w1ocdK+qJgMnAmeJyImq+izOuDjnqOo5vqEv7gN6+X6WK4DbyjmOiSIRN8yFiQg5vjdGf/HARF8beiHOmD4lLQbuFZGWwGxV/UlEegInA8t9w3vUxEkwpZkhIjnAJpzhl48BNqrq/3yvvwLcCEzEmZ9hqoh8CAQ8NLeq7hCRDb4xa37yHWORb78VibM2zrAP/rNuXSoio3D+rw/HmXDmuxLfe6pv/SLfcWrg/NyMASwpmPBxK/Ab0Amnwv3TpDmq+oaILAX6AfNE5FqcYYZfUdV7AjjGEP8B80Sk1Dk2fOPxdMUZhG0wMAY4twLnMhO4FPgBeEdVVZx36IDjxJmBbDzwPHCRiLQF7gBOUdUMEXkZZ2C4kgSYr6qXVyBeE0Ws+ciEi3rAr74x8ofifEr+AxE5EtjgazJJxWlG+QS4RESa+rZpKIHPT/0D0EZEjvYtDwU+87XB11PVOTiduKVdAZSJM3x3aWYDF+LMAzDTt65CcapqPk4z0Km+pqe6QBawV0QOA84/RCxLgNMPnpOI1BKR0qouE6UsKZhwMQm4SkSW4DQdZZWyzWXA9yLyLXAszpSFa3HePD8Ske+A+ThNK+VS1VycESj/LSKrgSJgMs4b7Ae+/X2GU8WU9DIw+WBHc4n9ZgBrgdaqusy3rsJx+voqngTuUNVVOHMzrwGm4zRJHZQCzBWRBaq6A+fKqDd9x1mC87MyBrBRUo0xxvixSsEYY0wxSwrGGGOKWVIwxhhTzJKCMcaYYpYUjDHGFLOkYIwxppglBWOMMcUsKRhjjCn2/0pAkGJ0oExIAAAAAElFTkSuQmCC\n",
        "text/plain": "<Figure size 432x288 with 1 Axes>"
       },
       "metadata": {
        "needs_background": "light"
       },
       "output_type": "display_data"
      }
     ],
     "persistentId": "2c810e70-a822-4d9a-a9b0-2622dfa247bf",
     "text": "y_true = coding_scores_not_low_complexity_unspliced['in_cds_no_stop_codon']\ny_score = coding_scores_not_low_complexity_unspliced['jaccard_in_peptide_db']\ny_pred = y_score > 0.5\n\nfpr, tpr, thresholds = metrics.roc_curve(y_true, y_score)\n\nfig, ax = plt.subplots()\nlw = 2\nax.plot(fpr, tpr, lw=2, color='darkorange')\nax.plot([0, 1], [0, 1], color='navy', lw=lw, linestyle='--')\nax.set(xlabel='False Positive Rate', ylabel='True Positive Rate')\n\nprint(\"accuracy score:\", metrics.accuracy_score(y_true, y_pred))\n\nprint(metrics.classification_report(y_true, y_pred))"
    },
    "executionTime": "2019-10-12T04:12:37.662Z"
   },
   {
    "cell": {
     "executionCount": 87,
     "executionEventId": "4110a9f9-c104-4005-b3fd-5ecb5e61d217",
     "hasError": false,
     "id": "f5b034a4-325d-41af-8295-7b22ef456d06",
     "outputs": [
      {
       "data": {
        "text/plain": "<matplotlib.axes._subplots.AxesSubplot at 0x7f3e49d8c0b8>"
       },
       "execution_count": 87,
       "metadata": {},
       "output_type": "execute_result"
      },
      {
       "data": {
        "image/png": "iVBORw0KGgoAAAANSUhEUgAAAYYAAAELCAYAAADdriHjAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDMuMC4zLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvnQurowAAIABJREFUeJzt3Xl4XPV97/H3d0aStcubvGDjFcfGoVwIhkChBZI0hUChTbgFmvRpetvSJYQEaLMXCLT3ydKkFwgk4eayZGlYUkwMpkDSgNnBS8xuwLFlWxCwLNuytVmame/9Y87IM/JIOrY1c0aaz+t55plzzpxz5qvx+Hzmd5bfMXdHREQkIxZ1ASIiUloUDCIikkPBICIiORQMIiKSQ8EgIiI5FAwiIpJDwSAiIjkUDCIikkPBICIiOSqiLuBQTJ061efNmxd1GSIiY8ratWt3uHvzSPONyWCYN28ea9asiboMEZExxcy2hJlPu5JERCSHgkFERHIoGEREJIeCQUREcigYREQkh4JBRERyKBhERCSHgkFERHKMyQvcZHR8+9vf5tHHVgHQ2NjAzTfdRFNTU8RViUjU1GIoU4lEgocffoTdiQp2VkzhrdZW1q9fH3VZIlICChoMZnarmW03s5eHeN3M7AYz22hmL5rZ+wpZj+zX0tLCvn299M04ht75vwexOK+++mrUZYlICSj0rqTbge8APxzi9bOBRcHj/cB3g2cpsC1b0l2mpGonQyyO10ykpaUl2qJkVLg73d3d7Nmzh46ODvbs2cPevXsHhrMfiUQi6nLHvHg8Tn19PU1NTTQ2Ng48mpqaaGhoGBiuq6sjFhsbO2kKGgzu/riZzRtmlvOBH7q7A8+a2UQzm+nuvy1kXQKtra0ApCY0AJCoamDL1m1RliR59Pf352zg8z3Sr3WwJ3jeu7eLRDI55DprK436SqeuIkmlpYr41xycnoTRkzBqKpyaCo+6nCElMbYlKujsN7r6nKEqjZnRUF9HY1MjjY0TaQyCZHCgDA6XCRMmFPXvgegPPs8CsrdGrcE0BUOBdXZ2YhWVEEt/BbxiAt3duyKuShKJBF/+0pfYvPk37N3bSU/vviHnrYxDQxXUVaSojyeZVpliQa1T35QKNvxOfWV6OPNcV+HEx8CP1jc7Krhhw3T+8LxzePjBlXzyqHdZ1FT6rZuUQ1fC6Oo3OvtjdPZb8IjRmTA6+3vo7Gun87fG29sqeCORnmdfYujgm1BVSUN9PdNnzuTaa69jypQpBf87og4GyzMt7ydkZpcAlwDMmTOnkDWVhUQiAZa1hbAYicTQvzKleKZNn05nZycxi5FM7qavv/+AeczSodBUmUw/qlI0TUgxsSpFU5UzcUIqPa0qxYR4BH/EYdqwq4I//Mg5/MOnPo07bFj3wzERDDGDhkqnodKBFP0p6NgXo6Mvxu4+o6Mvtv+xL0FFLEbMKtiZgmSexltFPE5jYyOTp0xh2rTpVFQUZ5MddTC0Akdmjc8G3s43o7vfAtwCsGzZstJtV44RjY2NeP8+SKUgFsMS+2ior4+6rLJXUVHBlVdeOTDu7nR1dbFz586hH+072Nrezu539pDyA/9r1FQYDVVOTUWK2liS2opgONhFk/Mc9wNer4qlg6iYlkxKcMODK3GHR/5rJZctiSYU+pLQkzS6g91a3YlY8Gz7n5ODx+P0JOPsDXYt5dPUUM/kKVOYPGUqCyZPZvKgx5QpU5g8eTINDQ1YsT98og+GFcClZnYn6YPOHTq+UBzNzembOFl/Fz6hgVhfFzOOnBFxVTKYmVFfX099ff2ILeVkMklHR8dAYLS3tw8Md3R00NnZSVdXJ7v27mVbVydde7rp7unF84RJtrhBTaXtD4xYMtjvvz886kY5XBY1JbhsybtsWPdDLluSOKTWwuFs1LsTRne/kwhxCKamegJ1tbXU1ddTX99Ac309dXV1NDU15WzkM49JkyYV7Zf/oSpodWb2U+AMYKqZtQJXA5UA7v494EHgI8BGoBv4y0LWI/vNnz8fgHj3ThJV9VT07mL+/FMirkoORzweH9j4hJVKpejp6QlCo2vgOfPIP54Ol9aDDJeJ1TCzpo9ZdUmOqE1yRF2SWXVJ6ivzL7uoaeRA6E3A291x3uqq4O2ueHq4u4qdvRz8Rr2hgWn1DdTV1VFXV0d9sIEfbry2tpZ4fAzuqxtBoc9KuniE1x34VCFrkPwWLlxILBYj1tVGrLoJT/SxePHiqMuSIovFYgMbukOVCZfsIMkOlMxwW1sbLZs389jWLezr23/cpGmCcURNXxAUiYHAaKz0gVZGV78FAZB+vN0V5+2eKtp79tdREY8ze/Yslh6zgBkzZuRsyMtpoz4aSrs9IwVTXV3N/AULeGN7G6kJjQAcffTREVclY1F2uEybNm3E+VOpFO+++y5btmyhpaWFrVu30rJ5E8+0tND9Vu/AfO+fto+/f28nyzfX8POW2oHpE6oqOXLOHI6fN5958+Yxd+5c5s6dyxFHHFHyu2jGCn2KZex3jjmGTfevJFnVQG1dnc72kqKIxWLMnDmTmTNncvLJJw9Md3d27NjBli1beO6557jnnnvYvLeC7T1xzjzzTD784Q8zb948pk+fPmYuFBurFAxlbMmSJdx3331U7trMkuOO1X82iZSZ0dzcTHNzMyeccAI7duzg0UcfpbGxkS996UtUVlZGXWLZ0JagjB111FEAWLKfoxYujLgakf3MjNNOOw2AE044QaFQZAqGMjZ79uyBYe1GklI10hlPMvoUDGWsurp6YHj69OkRViJyoMyFXVFc4FXuFAwCcFDnvosUk1oMxadgEAAaGhqiLkEkh1oK0VEwCAA1NTVRlyAiJULBIABUVVVFXYKIlAgFgwCoawARGaBgEABd3CYiA7Q1EEAH+kRkPwWDiIjkUDAIoHPFRWQ/BYMACgYR2U/BIICCQUT2UzCIiEgOBYMAajGIyH4KBgEUDCKyn4JBRERyKBhERCSHgkFERHIoGEREJIeCQUREcigYREQkh4JBAPWuKiL7KRhERCRHwYPBzM4ys9fNbKOZfSHP63PM7FEz+7WZvWhmHyl0TSIiMrSCBoOZxYGbgLOBpcDFZrZ00GxfAe529+OBi4CbC1mTiIgMr9AthpOAje6+yd37gDuB8wfN40BjMNwEvF3gmkREZBgVBV7/LGBb1ngr8P5B81wDPGJmnwbqgA8VuCbJQ30liUhGoVsM+U51GbwFuhi43d1nAx8BfmRmB9RlZpeY2RozW9PW1laAUsubgkFEMgodDK3AkVnjszlwV9FfAXcDuPszQDUwdfCK3P0Wd1/m7suam5sLVG75SqVSUZcgIiUiVDCY2RQzu9HM1pnZWjO73symhFh0NbDIzOabWRXpg8srBs2zFfhg8D5Hkw4GNQmKTC0GEckI22K4E9gOfAy4gPSG+66RFnL3BHAp8DDwGumzj14xs2vN7LxgtiuBvzGzF4CfAp90baWKLplMRl2CiJSIsAefJ7v7dVnj/2JmfxxmQXd/EHhw0LSrsoZfBU4NWYcUiIJBRDLCthgeNbOLzCwWPP4UWFnIwqS4EolE1CWISIkYtsVgZntJn0VkwBXAj4KX4kAncHVBq5OiUYtBRDKGDQZ3byhWIRIttRhEJGOkFsP7hnvd3deNbjkSFZ2uKiIZIx18/lbwXA0sA14gvVvpWOA54LTClSbFpGAQkYxhDz67+5nufiawBXhfcIHZCcDxwMZiFCgiIsUV9qykJe7+UmbE3V8GjitMSSIiEqWw1zG8ZmY/AH5M+iylT5C+YE3GCd3BTUQywgbDXwJ/D3wmGH8c+G5BKpJIxGK6mZ+IpIUKBnfvBf49eBzAzP7T3T82moVJccXj8ahLEJESMVo/ExeM0nokIgoGEckYrWBQp3djnIJBRDK0Y1kABYOI7DdawaBTWsY4BYOIZIQOBjOrMbPFQ7z8+VGqRyKis5JEJCPsHdz+CFgPPBSMH2dmA3dic/dHClOeFIuuYxCRjLA/E68BTgJ2A7j7emBeYUqSKCgYRCQjbDAk3L2joJVIpBQMIpIR9srnl83sz4C4mS0CLgOeLlxZUmy6zbaIZIRtMXwaeC+wD/gpsAf4bKGKkuJTMIhIRtguMbqBLwcPGYcUDCKSMdId3O5nmKua3f28Ua9IIqFgEJGMkVoM/xY8fxSYQbrbbYCLgZYC1SQRUDCISMawweDuqwDM7Dp3//2sl+43s8cLWpmIiEQi7MHnZjMb6EHVzOYDzYUpSaKgFoOIZIQ9XfVy4DEz2xSMzwP+tiAVSSR0HYOIZIQ9K+mh4PqFJcGkDe6+r3BliYhIVEY6K+kD7v4rM/vooJcWmhnufm8Ba5MiUotBSo12b0ZnpBbD6cCvgD/K85oDCoZxQsEgpSYTDPpuFt9IZyVdHQxe6+6bs18LDkCPyMzOAq4H4sAP3P1reeb5U9Id9Tnwgrv/WZh1y+jRfz4RyQh7VtJ/5pn2s5EWMrM4cBNwNrAUuNjMlg6aZxHwReBUd38v6mojEgoGEckY6RjDEtJ9JDUNOs7QCFSHWP9JwEZ33xSs707gfODVrHn+BrjJ3XcBuPv28OXLaNGNeqTUpFIpAJLJZMSVlJ+RjjEsBs4FJpJ7nGEv6Q36SGYB27LGW4H3D5rnPQBm9hTp3U3XuPtDg1dkZpcAlwDMmTMnxFuLyFi2fXv6N6IOQhffSMcYfg783MxOcfdnDmH9+fZPDP5XrgAWAWcAs4EnzOwYd989qJZbgFsAli1bpm+KyDj31ltvAbCzvT3iSspP2P0H75rZ/WbWZmbbzezn2VdCD6MVODJrfDbwdp55fu7u/cEB7tdJB4UUkX6VSalpa2sLnt+NuJLyEzYY/gO4G5gJHAHcQ/q+DCNZDSwys/lmVgVcBKwYNM99wJkAZjaV9K6lTUhRZfbnipSK3t4eAPb16lraYgsbDObuP3L3RPD4McN0x53h7gngUuBh4DXgbnd/xcyuNbNMl90PA+1m9irwKPBP7q62Y5EpGKTUTJs2HYDm6dMjrqT8hO0r6VEz+wJwJ+lAuBBYaWaTAdx951ALuvuDwIODpl2VNezAFcFDIqJgkFIzc+ZMYH9ASPGEDYYLg+fBHef9L9JBEeZ4g5SwRCIRdQkiOWbMmAFARUXYzZSMlrCd6IW6ylnGLrUYpNTU1NQAusYmCqE+cTOrNbOvmNktwfgiMzu3sKVJManFIKVKV+UXX9govg3oA343GG8F/qUgFUkk1GKQUpM5hVpXPhdf2GBY6O7fAPoB3L2H/BevyRil/3xSajo6OqIuoWyFDYY+M6shOEXVzBYCOrl4HFGLQUpN5srnrq6uiCspP2EP918NPAQcaWY/AU4FPlmooqT41GKQUrNtW7qbtW1bt0RcSfkJe1bSL8xsHXAy6V1In3H3HQWtTETKWuaEiER/f8SVlJ+DOUH4dOA00ruTKoHlBalIIqG+kqTU1NXVAVAbPEvxhD1d9Wbg74CXgJeBvzWzmwpZmIiUt+lBVxjTZ8yMuJLyE7bFcDpwTNB9BWZ2B+mQEBEpiLlz5wJQW1sbcSXlJ+xZSa8D2XfHORJ4cfTLkajoIiIpNZldSfF4POJKyk/YFsMU4DUzez4YPxF4xsxWALj7eUMuKSIiY0rYYLhq5FlkLFN/NFKq1JotvrCnq64a7nUze8bdTxmdkiQK+s8npUpnzBXfaP1MrB6l9UhEFAxSajLfSX03i2+0gkGRPsbpP5+UKrUYik87lgVQMEjp0XcyOqMVDPoXHOP0n1BEMkYrGP58lNYjEdFZSSKSEbZLjI+a2Ztm1mFme8xsr5ntybzu7i8XrkQRESmmsNcxfAP4I3d/rZDFSHS0K0lEMsLuP3hXoSAiUh7CthjWmNldwH1k3bnN3e8tSFVSdDolUEQywgZDI9ANfDhrmgMKhnFCwSAiGWG7xPjLQhci0dI9n0UkY9hgMLPPufs3zOxG8lzd7O6XFawyKSrd81lEMkZqMWQOOK8pdCESLQWDiGQMGwzufn/wfMdw85nZje7+6SFeOwu4HogDP3D3rw0x3wXAPcCJ7q4gKjIFg4hkjNblrqfmm2hmceAm4GxgKXCxmS3NM18DcBnw3CjVIwdJwSAiGYXuB+EkYKO7b3L3PuBO4Pw8811H+iK63gLXI0NIJBJRlyAiJaLQwTAL2JY13hpMG2BmxwNHuvsDBa5FhqGzkkQko9C9q+abPnB2k5nFgH8HrhzxDcwuMbM1Zramra3t0KqUIWlXkohkjFYwXD/E9FbgyKzx2cDbWeMNwDHAY2bWApwMrDCzZYNX5O63uPsyd1/W3Nw8OlXLAF3gJiIZI13HcD/D3J3N3c8Lnm8fYpbVwCIzmw+8BVwE/FnW8h3A1Kz3ewz4R52VVHwKBhHJGOk6hn8Lnj8KzAB+HIxfDLSMtHJ3T5jZpcDDpE9XvdXdXzGza4E17r7ikKqWUaf7MYhIxkjXMawCMLPr3P33s16638weD/MG7v4g8OCgaVcNMe8ZYdYpIiKFE/ZnYrOZLciMBLuGtKN/HFGLQUQywvauejnpA8SbgvF5wCUFqUgioRv1iEjGiMEQnFK6B1gELAkmb3D3fUMvJWONWgwikjFiMLh7ysy+5e6nAC8UoSaJgFoMIpIR9mfiI2b2MdPWY9xSi0FEMsIeY7gCqAMSZtZL+opmd/fGglUmRaXMF5GMsHdwayh0IRIttRhEJCNsiwEzm0T6AHR1Zpq7h7qWQUqfWgwikhEqGMzsr4HPkO7raD3pPo2eAT5QuNKkmNRiEJGMsFuDzwAnAlvc/UzgeEBdnI4jajGISEbYYOh1914AM5vg7huAxYUrS4pNLQYRyQh7jKHVzCYC9wG/MLNd5HafLWOcWgwikhH2rKQ/CQavMbNHgSbgoYJVJUWnYBCRjFD7D8zsZDNrgIEeVx8lfZxBRETGmbA7lr8LdGaNdwXTRERknAkbDOZZt/hy9xQHcQ2ElD7dwU1EMsIGwyYzu8zMKoPHZ4BNIy4lIiJjTthg+Dvgd0nft7kVeD+6H8O4kkwmoy5BREpE2LOStgMXFbgWiZB2JYlIRtizku4IrmPIjE8ys1sLV5YUWyqViroEESkRYXclHevuuzMj7r4Lna46rmhXkohkhA2GWNC7KgBmNhmdlTSuqMUgIhlhN+7fAp42s58F4/8T+NfClCRRUItBRDLCHnz+oZmtBc4kffe2j7r7qwWtTIoqkUhEXYKIlIjQu4Pc/RUzayO4UY+ZzXH3rQWrTIqqv78/6hJEpESEPSvpPDN7E9gMrAJagP8qYF1SZH19fVGXICIlIuzB5+tI37XtDXefD3wQeKpgVUlRZF+70NvbG2ElIlJKwgZDv7u3kz47KebujwLHFbAuKYLsMOju7o6wEhEpJWGPMew2s3rgCeAnZrYd0NHKMa6zszPvsIiUt7AthvOAbtL3fn4I2AicG2ZBMzvLzF43s41m9oU8r19hZq+a2Ytm9t9mNjds8XJ49u7dm3dYRMrbsMFgZk8Gg+8Cu4FdwHeA/w1sNrPNZvYPwywfB24CzgaWAheb2dJBs/0aWObuxwI/A75xKH+IHLw9e/bkHRaR8jZsMLj7acFzg7s3Bo+G4NEELCPdihjKScBGd9/k7n3AncD5g97jUXfP7OB+Fph9qH+MHJyOjo6B4d27dw8zp4iUk7C7kvIKDkifMcwss4BtWeOtwbSh/BVDnAZrZpeY2RozW9PW1nawpUoemTBIVjfR0aEWg4ikHVYwALj7b4d5Od8d5vP272xmnyDdAvnmEO9zi7svc/dlzc3NB1+oHCDTYkhVN7Fz166IqxGRUlHojvBagSOzxmcDbw+eycw+BHwZON3d9xW4Jgl0dHSAxYjt28uepE4yE5G0w24xjGA1sMjM5ptZFemb/azInsHMjge+D5wX3BBIiiQdDIYl++jq7FRHeiICFDgY3D0BXAo8DLwG3B30uXStmZ0XzPZNoB64x8zWm9mKIVYno2z37t04MbAY7q4zk0QEKMI9Fdz9QeDBQdOuyhr+UKFrkPzad+5Mh4Klfx/s2rWLSZMmjbCUiIx3hd6VJCVsx452PBYDiwPQ3t4ecUUiUgoUDGWqr6+PvXs60qEQS38NduzYEXFVIlIKFAxlavv29HF+j8XxoMXwzjvvRFmSiJQIBUOZGggBi6fPTJpQz29/O9wlKSJSLhQMZaq1tTU9EEu3FhJVDfuniUhZUzCUqdbWVixeMbAbKVndwNZtCgYRUTCUrW3btpGqbhrotCRV3UTn3j26lkFEFAzlatPmFhITGgfGU9UTAdiyZUtUJYlIiVAwlKG9e/fStv1dUjWTB6alatIXtv3mN7+JqiwRKREKhjL0xhtvAJCs3R8MXlWHVVazYcOGqMoSkRKhYChDa9euBYuRbJi+f6IZffUzWL16De55e0YXkTKhYCgz7s4TTz5Fsn4axKtyXks2zaK9fQdvvvlmRNWJSClQMJSZl156iW1bt9A/5agDXuufNA+LV3D//fdHUJmIlAoFQ5m58847sYoJ9E9ecOCLFRPYN2k+Dz/8iDrUEyljCoYy8tJLL/H000/TO/29EM/f43rfzGPp6+/njjvuKHJ1IlIqFAxlIpVKcdPNN2NVtfRNfy8AE7Y+S7y7nXh3OzUbHmTC1mfx6ib6mhfzwAMP0NLSEm3RIhIJBUOZ+OUvf8mG116je9YJEK8EINa9E0v2Y8l+Kva+Q6x7JwB9RxyHxyq58Tvf0RlKImVIwVAG9u3bx/e+/31S9c0k8hx0Hswra+iZeRxr16zh+eefL0KFIlJKFAxlYOXKlexsb6d31glgFmqZ/mlLoLqBW2+9Ta0GkTKjYBjnEokEP/7JT0g2zCDZMDP8grE4vTOO5fXXN6QviBORsqFgGOeeeeYZdra3pw84h2wtZPRPWYhVVrNixYoCVScipUjBMM4tv+8+mFBHYuKRB79wrIJ9U47iySefHLgVqIiMfwqGcaylpYV1a9eyb+oSsEP7p+5rXkIqlVKrQaSMKBjGsdtuuw2LVdDf/J5DXodXN5KYNId7ly/X1dAiZULBME499dRTrFq1it6Zx+KVNYe1rt5ZJ9LT08sNN9wwStWJjExnw0VHwTAOtba28rWvfx2vnUzfjGMPe31e00TvzONYtWoVy5cvH4UKRcKzgzxpQg6fgmGc2blzJ1dc+Y909vTTtfBMiI3OP3HfzN8hMXEO199wA6tWrRqVdYqEoZZD8SkYxpHu7m4+97nP07ajnc6jPoRXN43eyi1Gz4IzSNU1c911/8ILL7wweusWyUOBEJ2CB4OZnWVmr5vZRjP7Qp7XJ5jZXcHrz5nZvELXNB719/fzlX/+Zzb+ZiNdC84gVd88+m8Sr6Br0YdIVNbxhS9+kU2bNo3+e4gEMsGQSqUirqT8FDQYzCwO3AScDSwFLjazpYNm+ytgl7sfBfw78PVC1jRe3XPPPaxbu5aeuaeSPJRrFsKqqKZz0YfpTcC1116nX3VSMDt27Ii6hLJV6BbDScBGd9/k7n3AncD5g+Y5H8h0/v8z4IOmo00Hpa+vj7vv+RnJxiNIHMapqWH5hHq6Z51AS8tmVq9eXfD3k/K0bds2AHbt3BlxJeWn0MEwC9iWNd4aTMs7j7sngA5gSoHrGlc2bNjA7l076Zu66OAWTPZRXV3NBRdcQHV1NST7Qi+amLwAYnGeeOKJg6xWJJzMdTM7duiq+2IrdDDk++U/eN9DmHkws0vMbI2ZrWlraxuV4saLhQsXUlFRSbzz4D4XS/Rx7rnncumll3LOOedgifDBEOveCakkS5cO3jMoMjo6O/cGz10RV1J+8t/fcfS0Atk7vGcDbw8xT6uZVQBNwAFtR3e/BbgFYNmyZdqxnaWuro5TTjmZJ59+lkTTrNDHGLyiigceeAB3Z+XKlXhFbajlrL+Hmq3PUFFZyWmnnXY4pYsMqbl5GvAKzc0FOJFChlXoFsNqYJGZzTezKuAiYHCnOyuAvwiGLwB+5TqiedA++9nPsmDeXGo3/pLKttdDLZNqmEFnrJa773+YzlgtqYYZIy5jvR3Ub1jJhP49XPvVr9LQ0HC4pYvktXjx4vTzkqMjrqT8WKG3wWb2EeD/AHHgVnf/VzO7Fljj7ivMrBr4EXA86ZbCRe4+7HmQy5Yt8zVr1hS07rGou7ubq66+mjWrV5OYNI/eOSfjVeFaASPyFJXvvkLN27+mvq6Ob3z9axx9tP7DSuG4Ox0dHTQ2NhIbpQs1y52ZrXX3ZSPONxZ/nCsYhpZIJLjrrru4/fbbSbjRc8TxpGoP81h+sp+at9dhXe2cfPLJXH755UyfPn10ChaRogkbDIU+xiBFVlFRwcc//nFOP/10vv3tb7Nu3XOjst6Jkybz2X+6htNPP11914iMc2oxjGPuzmuvvUZPT89hrcfMWLx4MXV1daNUmYhEQS0Gwcx0OqmIHDQd0RERkRwKBhERyaFgEBGRHAoGERHJoWAQEZEcCgYREcmhYBARkRxj8gI3M2sDtkRdxzgyFdDtsqQU6bs5uua6+4jd1Y7JYJDRZWZrwlwNKVJs+m5GQ7uSREQkh4JBRERyKBgEgjvjiZQgfTcjoGMMIiKSQy0GERHJoW63xyEzSwIvZU36Y3dvGWLeecAD7n5M4SsTATObAvx3MDoDSAJtwfhJ7t4XSWEyQMEwPvW4+3FRFyGSj7u3A8cBmNk1QKe7/1v2PJa+TaC5e6r4FYp2JZUJM5tnZk+Y2brg8bt55nmvmT1vZuvN7EUzWxRM/0TW9O+bWbz4f4GMd2Z2lJm9bGbfA9YBR5rZ7qzXLzKzHwTD083sXjNbE3w3T46q7vFIwTA+1QQb8fVmtjyYth34A3d/H3AhcEOe5f4OuD5obSwDWs3s6GD+U4PpSeDjhf8TpEwtBf6fux8PvDXMfDcA3wgufvtT4AfFKK5caFfS+JRvV1Il8B0zy2zc35NnuWeAL5vZbOBed3/TzD4InACsTrfuqSEdMiKF8Bt3Xx1ivg8Bi4PvJMAkM6tx98O7wbkACoZycjnwLvA/SLcUewfP4O7/YWbPAecAD5vZXwMG3OHuXyxmsVK2urKGU6S/fxnVWcOGDlQXjHYllY8m4Lf256AxAAAExUlEQVTBwbw/Bw44TmBmC4BN7n4DsAI4lvTZIxeY2bRgnslmNrd4ZUu5Cr6ru8xskZnFgD/JevmXwKcyI0FLWEaJgqF83Az8hZk9S3o3UleeeS4EXjaz9cAS4Ifu/irwFeARM3sR+AUws0g1i3weeIj0D5TWrOmfAk4NTpJ4FfibKIobr3Tls4iI5FCLQUREcigYREQkh4JBRERyKBhERCSHgkFERHIoGEREJIeCQSJhZk8XcN2PmVlJ3EDezCaa2T9EXUc2M2sxs6lR1yGlS8EgkXD3A3p3HacmAiUVDCIjUTBIJMysM3g+I/iF/zMz22BmP7GsntHyLHeimT1tZi8E3S03mFmNmd0ZXAV7F+mO/jCzuJndHnTl/JKZXT7Meh8zs68H63zDzH4vmF5tZrcFy//azM4cZh35ui3/GrAwmPZNS/tmVk0XZn0Oj5vZcjN71cy+F3QDMdR7nRV0n/6Cmf13MG2ymd0XvPezZnZsMH2KmT0S1P99svofMrMrglpeNrPPBtPmmdlrZvZ/zeyVYNmaoWqRccjd9dCj6A/SN2cBOAPoAGaT/qHyDHDaEMtUAZuAE4PxRtIdQV4B3BpMOxZIkO42/ATgF1nLTxymnseAbwXDHwF+GQxfCdwWDC8BtgLVQ6zjRuDjWbXWAPOAl7Pm+RjpbkXiwPRgfTODz6EXWBC89gvggiHepxnYBswPxidnvf/VwfAHgPXB8A3AVcHwOYADU4PP5yWgDqgHXgGOD2pOAMcFy9wNfCLq74wexXuoxSCl4Hl3b/V0p2nrSW+Y8llMuiPA1QDuvsfdE8DvAz8Opr0IvBjMvwlYYGY3mtlZwJ4R6rg3eF6bVcNpwI+CdW8AtpC/y3JIh9qXzOzzwFzP3wX0acBP3T3p7u8Cq4ATg9eed/dN7p4EfhrMm8/JwOPuvjmoa2eeWn8FTDGzJnI/n5XArqz5l7t7l7t3Bn//7wWvbXb39Xk+DykDCgYpBfuyhpMM3R28kf61m88B0919F+luxh8j3enaSDdzydSRXcOQu7XyvN9/AOcBPaS7Lf9AntmGW9/gv2Gov3WozyHfun3Qc9hawv6byDikYJCxZANwhJmdCBAcX6gAHie4q5yZHUN6dxLBmTcxd/9P4J+B9x3Ce2av+z3AHOD1fDMO0W35XqBh0PouDI5/NJP+Nf988NpJZjY/OLZwIfDkEDU9A5xuZvOD952cp9YzgB3uvmfQ9LOBSVnz/7GZ1ZpZHelurZ8I86HI+KZfATJmuHtfcLD2xuBgaA/pO3l9F7gt6BZ8Pfs3tLOC6ZkfQIdys6Gbge+Z2Uuk97t/0t33DTHvhcAnzKwfeAe41t13mtlTZvYy8F/A54BTgBdI/4r/nLu/Y2ZLSG/wvwb8DumN9vJ8b+LubWZ2CXBv8LdtB/4AuCbrc+gG/iJY5KvAT81sHeldV1uD9awzs9vZ/3n9wN1/bWbzDvIzknFG3W6LlIDgF/4/uvu5Udciol1JIiKSQy0GKUlmthyYP2jy59394cNc703AqYMmX+/utx3EOv4Q+PqgyZvd/U/yzX84LH0P7gmDJv+5u7802u8lkqFgEBGRHNqVJCIiORQMIiKSQ8EgIiI5FAwiIpJDwSAiIjn+P0FVBUEXKCEVAAAAAElFTkSuQmCC\n",
        "text/plain": "<Figure size 432x288 with 1 Axes>"
       },
       "metadata": {
        "needs_background": "light"
       },
       "output_type": "display_data"
      }
     ],
     "persistentId": "72511a16-18da-4358-93cf-c36a17c23f8e",
     "text": "\n\nsns.violinplot(x='in_cds_no_stop_codon', y='jaccard_in_peptide_db', data=coding_scores_not_low_complexity_unspliced, \n               bw=0.1, cut=True)"
    },
    "executionTime": "2019-10-12T04:12:40.488Z"
   },
   {
    "cell": {
     "executionCount": 88,
     "executionEventId": "d60ab4d4-c86f-41a4-8bb2-7d9698b44f78",
     "hasError": false,
     "id": "804034e2-612a-4733-8cfa-8769707fbb89",
     "outputs": [
      {
       "data": {
        "text/plain": "(1, 6)"
       },
       "execution_count": 88,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "23edfe41-6d93-44d6-9324-ba4ac19d25e0",
     "text": "coding_scores_not_low_complexity_unspliced.query(\"jaccard_in_peptide_db < 0.5 & in_cds_no_stop_codon == True\").shape"
    },
    "executionTime": "2019-10-12T04:12:54.076Z"
   },
   {
    "cell": {
     "executionCount": 89,
     "executionEventId": "3f4f1c9d-cf40-45f8-8c76-e00bb7d57491",
     "hasError": false,
     "id": "4d9dc7ba-ece3-49ca-892c-d947efb6ac0a",
     "outputs": [
      {
       "data": {
        "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>read_id</th>\n      <th>jaccard_in_peptide_db</th>\n      <th>n_kmers</th>\n      <th>classification</th>\n      <th>short_read_id</th>\n      <th>in_cds_no_stop_codon</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>289</th>\n      <td>SRR306838.8536423 Ibis_Run100924_C3PO:6:41:4477:5833/1</td>\n      <td>0.588235</td>\n      <td>17</td>\n      <td>non-coding</td>\n      <td>SRR306838.8536423</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>740</th>\n      <td>SRR306838.4158712 Ibis_Run100924_C3PO:6:19:4365:17337/1</td>\n      <td>0.533333</td>\n      <td>15</td>\n      <td>non-coding</td>\n      <td>SRR306838.4158712</td>\n      <td>True</td>\n    </tr>\n    <tr>\n      <th>991</th>\n      <td>SRR306838.20339512 Ibis_Run100924_C3PO:6:102:15178:5329/1</td>\n      <td>0.000000</td>\n      <td>16</td>\n      <td>non-coding</td>\n      <td>SRR306838.20339512</td>\n      <td>True</td>\n    </tr>\n  </tbody>\n</table>\n</div>",
        "text/plain": "                                                       read_id  \\\n289     SRR306838.8536423 Ibis_Run100924_C3PO:6:41:4477:5833/1   \n740    SRR306838.4158712 Ibis_Run100924_C3PO:6:19:4365:17337/1   \n991  SRR306838.20339512 Ibis_Run100924_C3PO:6:102:15178:5329/1   \n\n     jaccard_in_peptide_db  n_kmers classification       short_read_id  \\\n289               0.588235       17     non-coding   SRR306838.8536423   \n740               0.533333       15     non-coding   SRR306838.4158712   \n991               0.000000       16     non-coding  SRR306838.20339512   \n\n     in_cds_no_stop_codon  \n289                  True  \n740                  True  \n991                  True  "
       },
       "execution_count": 89,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "1254868e-73a0-4c0a-82c6-bb1b85e78cb0",
     "text": "coding_scores_not_low_complexity_unspliced.query(\"jaccard_in_peptide_db < 0.9 & in_cds_no_stop_codon == True\").head()"
    },
    "executionTime": "2019-10-12T04:12:54.671Z"
   },
   {
    "cell": {
     "executionCount": 90,
     "executionEventId": "d5caa457-ae0c-48fa-b4dc-3f8801d0f2da",
     "hasError": false,
     "id": "804034e2-612a-4733-8cfa-8769707fbb89",
     "outputs": [
      {
       "data": {
        "text/plain": "(3, 6)"
       },
       "execution_count": 90,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "23edfe41-6d93-44d6-9324-ba4ac19d25e0",
     "text": "coding_scores_not_low_complexity_unspliced.query(\"jaccard_in_peptide_db < 0.9 & in_cds_no_stop_codon == True\").shape"
    },
    "executionTime": "2019-10-12T04:13:01.913Z"
   },
   {
    "cell": {
     "executionCount": 91,
     "executionEventId": "74205ae1-615d-4c09-953f-a0881183b10e",
     "hasError": false,
     "id": "00b8ca70-dcb1-416b-9142-a2e7736ca3f4",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "SRR306838.8536423\t0\tchr1\t39448668\t255\t70M\t*\t0\t0\tAAAGACTTTGCTTCCCGAAGATACTCAGAAACTTGACAATTTCCTAGGAGAAGTCAGAGACAAATGGGAT\t-0454575:38569357656746757387639475375567787758567758596:6957665665585\tNH:i:1\tHI:i:1\tAS:i:66\tnM:i:1\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\nSRR306838.4158712\t16\tchrX\t129814671\t255\t49M14S\t*\t0\t0\tGACGATGTTGAAGGCGAAGACATAGATTGTGAGGAGGGAGAAAGAAAGGGAAGTAGAAGTAGG\t836857646958738766836846574664767626764766584546386474576222000\tNH:i:1\tHI:i:1\tAS:i:46\tnM:i:1\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\nSRR306838.20339512\t0\tchr2\t26192330\t255\t54M16S\t*\t0\t0\tCTTCGGACTCGCTAAAATACTATCCATGTCAGAATTCAAATCCTTCCTCTTCACTTATCATCTTCACAAT\t31/3331654544412543575556554166666648555667768766468567466675:68555765\tNH:i:1\tHI:i:1\tAS:i:51\tnM:i:1\tRG:Z:SRR306838_GSM752691_hsa_br_F_1\n"
      }
     ],
     "persistentId": "4acd19e5-3727-407f-adb5-b0676c493cc6",
     "text": "df = coding_scores_not_low_complexity_unspliced.query(\"jaccard_in_peptide_db < 0.9 & in_cds_no_stop_codon == True\")\nfor read_id in df.short_read_id:\n    ! samtools view $folder/$unspliced_bam | grep $read_id"
    },
    "executionTime": "2019-10-12T04:13:18.346Z"
   },
   {
    "cell": {
     "executionCount": 92,
     "executionEventId": "80e14e68-227c-4b78-9d7f-9852ff61c0bc",
     "hasError": true,
     "id": "f4935c7f-21c5-4eec-9288-85c6a74097ad",
     "outputs": [
      {
       "name": "stderr",
       "output_type": "stream",
       "text": "/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/screed/openscreed.py:39: ResourceWarning: unclosed file <_io.BufferedReader name='SRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz'>\n  self.iter_fn = self.open_reader(filename, *args, **kwargs)\n0it [00:00, ?it/s]\n"
      },
      {
       "ename": "TypeError",
       "evalue": "'int' object is not iterable",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
        "\u001b[0;32m<timed exec>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n",
        "\u001b[0;32m~/code/kmer-hashing/kh-tools/khtools/partition_reads.py\u001b[0m in \u001b[0;36mscore_reads\u001b[0;34m(reads, peptide_graph, peptide_ksize, jaccard_threshold, molecule, verbose)\u001b[0m\n\u001b[1;32m    173\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    174\u001b[0m         jaccard, n_kmers = score_single_sequence(sequence, peptide_graph,\n\u001b[0;32m--> 175\u001b[0;31m                                                  peptide_ksize, molecule, verbose)\n\u001b[0m\u001b[1;32m    176\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    177\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mjaccard\u001b[0m \u001b[0;34m>\u001b[0m \u001b[0mjaccard_threshold\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/code/kmer-hashing/kh-tools/khtools/partition_reads.py\u001b[0m in \u001b[0;36mscore_single_sequence\u001b[0;34m(sequence, peptide_graph, peptide_ksize, molecule, verbose)\u001b[0m\n\u001b[1;32m    142\u001b[0m         fraction_in_peptide_db, n_kmers = score_single_translation(\n\u001b[1;32m    143\u001b[0m             \u001b[0mtranslation\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpeptide_graph\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpeptide_ksize\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmolecule\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mmolecule\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 144\u001b[0;31m             verbose=verbose)\n\u001b[0m\u001b[1;32m    145\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    146\u001b[0m         \u001b[0;31m# Save the highest jaccard\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;31mTypeError\u001b[0m: 'int' object is not iterable"
       ]
      }
     ],
     "persistentId": "d1dc9a8f-8c57-42e1-bb81-50290bf5316c",
     "text": "%%time\n\nreads = 'SRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz'\n\ncoding_scores = score_reads(reads, peptide_graph, peptide_ksize)"
    },
    "executionTime": "2019-10-12T04:16:04.589Z"
   },
   {
    "cell": {
     "executionCount": 1,
     "executionEventId": "a24705f1-8d18-4182-a580-b0d041b0bed3",
     "hasError": false,
     "id": "624c183c-e9c9-469f-9cd4-571fed8511ac",
     "outputs": [],
     "persistentId": "71299be4-a00f-4638-92a2-ea174ddf208d",
     "text": "%load_ext autoreload\n%autoreload 2"
    },
    "executionTime": "2019-10-12T04:18:08.674Z"
   },
   {
    "cell": {
     "executionCount": 2,
     "executionEventId": "26e5ce42-45dc-4a87-9ba8-ce4273cfdad1",
     "hasError": false,
     "id": "83c92d0d-bd81-4ae7-9f88-242ac299cfcd",
     "outputs": [],
     "persistentId": "ecc17310-b85a-49bd-8d60-8de454040c04",
     "text": "folder = '/home/olga/data_sm/kmer-hashing/classify_coding_vs_noncoding/'"
    },
    "executionTime": "2019-10-12T04:18:09.435Z"
   },
   {
    "cell": {
     "executionCount": 3,
     "executionEventId": "d96008ce-23a7-4b17-8c60-5105aeddd95c",
     "hasError": false,
     "id": "802d541f-5fd3-48b7-99a1-9883a78e612f",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "/mnt/data_sm/olga/kmer-hashing/classify_coding_vs_noncoding\n"
      }
     ],
     "persistentId": "1d547cf2-254f-4fe2-9bf0-5b7e39be4376",
     "text": "cd $folder"
    },
    "executionTime": "2019-10-12T04:18:09.480Z"
   },
   {
    "cell": {
     "executionCount": 4,
     "executionEventId": "e52572dd-77d9-4826-8d8a-160012775441",
     "hasError": false,
     "id": "bb96f994-0b34-4694-980d-db02f28e5e9b",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "\u001b[0m\u001b[01;31mHomo_sapiens.GRCh38.cds.all.fa.gz\u001b[0m\n\u001b[01;31mHomo_sapiens.GRCh38.ncrna.fa.gz\u001b[0m\n\u001b[01;31mHomo_sapiens.GRCh38.pep.all.fa.gz\u001b[0m\nhuman_cds_k15.nodegraph\nhuman_cds_k15.nodegraph.info\nhuman_cds_k17.nodegraph\nhuman_cds_k17.nodegraph.info\nhuman_cds_k19.nodegraph\nhuman_cds_k19.nodegraph.info\nhuman_cds_k21.nodegraph\nhuman_cds_k21.nodegraph.info\nhuman_cds_k23.nodegraph\nhuman_cds_k23.nodegraph.info\nhuman_cds_k25.nodegraph\nhuman_cds_k25.nodegraph.info\nhuman_cds_k27.nodegraph\nhuman_cds_k27.nodegraph.info\nhuman_cds_k31.nodegraph\nhuman_cds_k31.nodegraph.info\nhuman_ncrna_k15.nodegraph\nhuman_ncrna_k15.nodegraph.info\nhuman_ncrna_k15.nodegraph.tagset\nhuman_ncrna_k17.nodegraph\nhuman_ncrna_k17.nodegraph.info\nhuman_ncrna_k19.nodegraph\nhuman_ncrna_k19.nodegraph.info\nhuman_ncrna_k21.nodegraph\nhuman_ncrna_k21.nodegraph.info\nhuman_ncrna_k23.nodegraph\nhuman_ncrna_k23.nodegraph.info\nhuman_ncrna_k25.nodegraph\nhuman_ncrna_k25.nodegraph.info\nhuman_ncrna_k27.nodegraph\nhuman_ncrna_k27.nodegraph.info\nhuman_ncrna_k31.nodegraph\nhuman_ncrna_k31.nodegraph.info\n\u001b[01;31mSRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz\u001b[0m\n\u001b[01;31mSRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.fq.gz\u001b[0m\n"
      }
     ],
     "persistentId": "a30b0ea1-cbfb-4de5-bce5-329cfeb8e50c",
     "text": "ls"
    },
    "executionTime": "2019-10-12T04:18:10.117Z"
   },
   {
    "cell": {
     "executionCount": 5,
     "executionEventId": "43941527-23bb-4406-8f01-9ecc4bdf5a7a",
     "hasError": false,
     "id": "3286464d-a5c9-45c3-8540-f4b967804d47",
     "outputs": [],
     "persistentId": "56e48bd4-fcc6-4d93-b4e3-0984be1fc1d5",
     "text": "import gzip\nfrom io import StringIO\nfrom pprint import pprint\n\nfrom khmer import Nodegraph\nimport matplotlib.pyplot as plt\nimport screed\nfrom sourmash._minhash import hash_murmur\nfrom khmer.khmer_args import calculate_graphsize\nimport pandas as pd\nfrom sourmash.logging import notify\nimport seaborn as sns\nfrom sklearn import metrics\nfrom tqdm import tqdm\nfrom Bio.Seq import Seq\nfrom Bio import SeqIO\n\n# make all warnings error to get a stack trace\nimport warnings\nwarnings.filterwarnings(\"default\")\n\npd.options.display.max_colwidth = 100\n\nfrom khtools.compare_kmer_content import kmerize, hpize, dayhoffize\n"
    },
    "executionTime": "2019-10-12T04:18:12.804Z"
   },
   {
    "cell": {
     "executionCount": 6,
     "executionEventId": "6890972c-63f2-4bd2-9608-175c138ee0d9",
     "hasError": false,
     "id": "a2b2503b-d248-407a-b936-191c45fc5e1f",
     "outputs": [],
     "persistentId": "1e54230e-ceec-42fc-afa6-7b32e219378b",
     "text": "from khtools.partition_reads import score_reads, make_peptide_bloom_filter, six_frame_translation_no_stops, is_low_complexity"
    },
    "executionTime": "2019-10-12T04:18:12.832Z"
   },
   {
    "cell": {
     "executionCount": 7,
     "executionEventId": "40440627-0e4c-4c3a-8663-e611dcfecba4",
     "hasError": false,
     "id": "a9bb9cbc-0224-43b1-99b0-a861bab8cf85",
     "outputs": [
      {
       "name": "stderr",
       "output_type": "stream",
       "text": "/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/screed/openscreed.py:39: ResourceWarning: unclosed file <_io.BufferedReader name='Homo_sapiens.GRCh38.pep.all.fa.gz'>\n  self.iter_fn = self.open_reader(filename, *args, **kwargs)\n"
      },
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "CPU times: user 39.2 s, sys: 2.38 s, total: 41.6 s\nWall time: 41.6 s\n"
      }
     ],
     "persistentId": "be06ded1-d7d4-4f21-9961-3aed5ff30cbf",
     "text": "%%time\npeptide_ksize = 7\n\npeptide_graph = make_peptide_bloom_filter(\"Homo_sapiens.GRCh38.pep.all.fa.gz\", peptide_ksize)"
    },
    "executionTime": "2019-10-12T04:18:54.529Z"
   },
   {
    "cell": {
     "executionCount": 8,
     "executionEventId": "a926b735-e025-411d-ac19-adaf9a235149",
     "hasError": true,
     "id": "38925b79-49df-403d-b4f7-57b005296192",
     "outputs": [
      {
       "ename": "NameError",
       "evalue": "name 'subsampled_reads' is not defined",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
        "\u001b[0;32m<timed exec>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n",
        "\u001b[0;31mNameError\u001b[0m: name 'subsampled_reads' is not defined"
       ]
      }
     ],
     "persistentId": "be06ded1-d7d4-4f21-9961-3aed5ff30cbf",
     "text": "%%time\n\nreads = 'SRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz'\n\ncoding_scores = score_reads(subsampled_reads, peptide_graph, peptide_ksize)"
    },
    "executionTime": "2019-10-12T04:18:54.739Z"
   },
   {
    "cell": {
     "executionCount": 9,
     "executionEventId": "3f05f79c-618e-4354-ba18-955168d16389",
     "hasError": true,
     "id": "38925b79-49df-403d-b4f7-57b005296192",
     "outputs": [
      {
       "name": "stderr",
       "output_type": "stream",
       "text": "[autoreload of khtools.partition_reads failed: Traceback (most recent call last):\n  File \"/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/IPython/extensions/autoreload.py\", line 245, in check\n    superreload(m, reload, self.old_objects)\n  File \"/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/IPython/extensions/autoreload.py\", line 368, in superreload\n    module = reload(module)\n  File \"/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/imp.py\", line 315, in reload\n    return importlib.reload(module)\n  File \"/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/importlib/__init__.py\", line 166, in reload\n    _bootstrap._exec(spec, module)\n  File \"<frozen importlib._bootstrap>\", line 618, in _exec\n  File \"<frozen importlib._bootstrap_external>\", line 674, in exec_module\n  File \"<frozen importlib._bootstrap_external>\", line 781, in get_code\n  File \"<frozen importlib._bootstrap_external>\", line 741, in source_to_code\n  File \"<frozen importlib._bootstrap>\", line 219, in _call_with_frames_removed\n  File \"/home/olga/code/kmer-hashing/kh-tools/khtools/partition_reads.py\", line 111\n    fraction_in_peptide_db = n_kmers_in_peptide_db / n_kmers\n    ^\nIndentationError: expected an indented block\n]\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/screed/openscreed.py:39: ResourceWarning: unclosed file <_io.BufferedReader name='SRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz'>\n  self.iter_fn = self.open_reader(filename, *args, **kwargs)\n0it [00:00, ?it/s]\n"
      },
      {
       "ename": "TypeError",
       "evalue": "'int' object is not iterable",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
        "\u001b[0;32m<timed exec>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n",
        "\u001b[0;32m~/code/kmer-hashing/kh-tools/khtools/partition_reads.py\u001b[0m in \u001b[0;36mscore_reads\u001b[0;34m(reads, peptide_graph, peptide_ksize, jaccard_threshold, molecule, verbose)\u001b[0m\n\u001b[1;32m    173\u001b[0m             scoring_lines.append(\n\u001b[1;32m    174\u001b[0m                 [description, -1, n_kmers, 'low complexity'])\n\u001b[0;32m--> 175\u001b[0;31m             \u001b[0;32mcontinue\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    176\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    177\u001b[0m         jaccard, n_kmers = score_single_sequence(sequence, peptide_graph,\n",
        "\u001b[0;32m~/code/kmer-hashing/kh-tools/khtools/partition_reads.py\u001b[0m in \u001b[0;36mscore_single_sequence\u001b[0;34m(sequence, peptide_graph, peptide_ksize, molecule, verbose)\u001b[0m\n\u001b[1;32m    142\u001b[0m     \u001b[0mmax_fraction_in_peptide_db\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;36m0\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    143\u001b[0m     \u001b[0;32mfor\u001b[0m \u001b[0mtranslation\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mtranslations\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 144\u001b[0;31m         \u001b[0mtranslation\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mencode_peptide\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtranslation\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmolecule\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    145\u001b[0m         fraction_in_peptide_db, n_kmers = score_single_translation(\n\u001b[1;32m    146\u001b[0m             \u001b[0mtranslation\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpeptide_graph\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpeptide_ksize\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmolecule\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mmolecule\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;31mTypeError\u001b[0m: 'int' object is not iterable"
       ]
      }
     ],
     "persistentId": "be06ded1-d7d4-4f21-9961-3aed5ff30cbf",
     "text": "%%time\n\nreads = 'SRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz'\n\ncoding_scores = score_reads(reads, peptide_graph, peptide_ksize)"
    },
    "executionTime": "2019-10-12T04:21:12.159Z"
   },
   {
    "cell": {
     "executionCount": 10,
     "executionEventId": "61e3490a-2dd6-4482-9efc-0afacd80fbdc",
     "hasError": true,
     "id": "38925b79-49df-403d-b4f7-57b005296192",
     "outputs": [
      {
       "name": "stderr",
       "output_type": "stream",
       "text": "/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/screed/openscreed.py:39: ResourceWarning: unclosed file <_io.BufferedReader name='SRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz'>\n  self.iter_fn = self.open_reader(filename, *args, **kwargs)\n0it [00:00, ?it/s]\n"
      },
      {
       "ename": "TypeError",
       "evalue": "'int' object is not iterable",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
        "\u001b[0;32m<timed exec>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n",
        "\u001b[0;32m~/code/kmer-hashing/kh-tools/khtools/partition_reads.py\u001b[0m in \u001b[0;36mscore_reads\u001b[0;34m(reads, peptide_graph, peptide_ksize, jaccard_threshold, molecule, verbose)\u001b[0m\n\u001b[1;32m    177\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    178\u001b[0m         jaccard, n_kmers = score_single_sequence(sequence, peptide_graph,\n\u001b[0;32m--> 179\u001b[0;31m                                                  peptide_ksize, molecule, verbose)\n\u001b[0m\u001b[1;32m    180\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    181\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mjaccard\u001b[0m \u001b[0;34m>\u001b[0m \u001b[0mjaccard_threshold\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/code/kmer-hashing/kh-tools/khtools/partition_reads.py\u001b[0m in \u001b[0;36mscore_single_sequence\u001b[0;34m(sequence, peptide_graph, peptide_ksize, molecule, verbose)\u001b[0m\n\u001b[1;32m    146\u001b[0m         fraction_in_peptide_db, n_kmers = score_single_translation(\n\u001b[1;32m    147\u001b[0m             \u001b[0mtranslation\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpeptide_graph\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpeptide_ksize\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmolecule\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mmolecule\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 148\u001b[0;31m             verbose=verbose)\n\u001b[0m\u001b[1;32m    149\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    150\u001b[0m         \u001b[0;31m# Save the highest jaccard\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;31mTypeError\u001b[0m: 'int' object is not iterable"
       ]
      }
     ],
     "persistentId": "be06ded1-d7d4-4f21-9961-3aed5ff30cbf",
     "text": "%%time\n\nreads = 'SRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz'\n\ncoding_scores = score_reads(reads, peptide_graph, peptide_ksize)"
    },
    "executionTime": "2019-10-12T04:22:09.413Z"
   },
   {
    "cell": {
     "executionCount": 11,
     "executionEventId": "e651d150-c4f4-443f-bd23-111c3881cbdd",
     "hasError": false,
     "id": "74edbd37-99c3-4e60-80c4-3494df741e79",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "> \u001b[0;32m/home/olga/code/kmer-hashing/kh-tools/khtools/partition_reads.py\u001b[0m(148)\u001b[0;36mscore_single_sequence\u001b[0;34m()\u001b[0m\n\u001b[0;32m    146 \u001b[0;31m        fraction_in_peptide_db, n_kmers = score_single_translation(\n\u001b[0m\u001b[0;32m    147 \u001b[0;31m            \u001b[0mtranslation\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpeptide_graph\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpeptide_ksize\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmolecule\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mmolecule\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m--> 148 \u001b[0;31m            verbose=verbose)\n\u001b[0m\u001b[0;32m    149 \u001b[0;31m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    150 \u001b[0;31m        \u001b[0;31m# Save the highest jaccard\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\n"
      },
      {
       "name": "stdin",
       "output_type": "stream",
       "text": "ipdb>  dir()\n"
      },
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "['fraction_in_peptide_db', 'low_complexity', 'max_fraction_in_peptide_db', 'max_n_kmers', 'molecule', 'n_kmers', 'nucleotide_ksize', 'peptide_graph', 'peptide_ksize', 'seq', 'sequence', 'translation', 'translations', 'verbose']\n"
      },
      {
       "name": "stdin",
       "output_type": "stream",
       "text": "ipdb>  a\n"
      },
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "sequence = 'CCCCCCCCCACCACCACCCCCCCCACCCCCCCCCCCCCCCCCCCCCCCCCCACCCCCCCACACACCCCCAACACCC'\npeptide_graph = <khmer._oxli.graphs.Nodegraph object at 0x7fc65f529930>\npeptide_ksize = 7\nmolecule = 'protein'\nverbose = False\n"
      },
      {
       "name": "stdin",
       "output_type": "stream",
       "text": "ipdb>  low_complexity\n"
      },
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "False\n"
      },
      {
       "name": "stdin",
       "output_type": "stream",
       "text": "ipdb>  n_kmers\n"
      },
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "18\n"
      },
      {
       "name": "stdin",
       "output_type": "stream",
       "text": "ipdb>  len(sequence)\n"
      },
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "76\n"
      },
      {
       "name": "stdin",
       "output_type": "stream",
       "text": "ipdb>  q\n"
      }
     ],
     "persistentId": "fefceffb-109b-4796-a998-8ad02d563488",
     "text": "%debug"
    },
    "executionTime": "2019-10-12T04:23:48.244Z"
   },
   {
    "cell": {
     "executionCount": 2,
     "executionEventId": "0e11c5a7-75ac-4318-838f-3708f4264b0b",
     "hasError": false,
     "id": "802d541f-5fd3-48b7-99a1-9883a78e612f",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "/mnt/data_sm/olga/kmer-hashing/classify_coding_vs_noncoding\n"
      }
     ],
     "persistentId": "1d547cf2-254f-4fe2-9bf0-5b7e39be4376",
     "text": "cd $folder"
    },
    "executionTime": "2019-10-12T04:23:54.348Z"
   },
   {
    "cell": {
     "executionCount": 3,
     "executionEventId": "0d2ecf92-7a45-4d44-becb-cc52fe6ef079",
     "hasError": false,
     "id": "bb96f994-0b34-4694-980d-db02f28e5e9b",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "\u001b[0m\u001b[01;31mHomo_sapiens.GRCh38.cds.all.fa.gz\u001b[0m\n\u001b[01;31mHomo_sapiens.GRCh38.ncrna.fa.gz\u001b[0m\n\u001b[01;31mHomo_sapiens.GRCh38.pep.all.fa.gz\u001b[0m\nhuman_cds_k15.nodegraph\nhuman_cds_k15.nodegraph.info\nhuman_cds_k17.nodegraph\nhuman_cds_k17.nodegraph.info\nhuman_cds_k19.nodegraph\nhuman_cds_k19.nodegraph.info\nhuman_cds_k21.nodegraph\nhuman_cds_k21.nodegraph.info\nhuman_cds_k23.nodegraph\nhuman_cds_k23.nodegraph.info\nhuman_cds_k25.nodegraph\nhuman_cds_k25.nodegraph.info\nhuman_cds_k27.nodegraph\nhuman_cds_k27.nodegraph.info\nhuman_cds_k31.nodegraph\nhuman_cds_k31.nodegraph.info\nhuman_ncrna_k15.nodegraph\nhuman_ncrna_k15.nodegraph.info\nhuman_ncrna_k15.nodegraph.tagset\nhuman_ncrna_k17.nodegraph\nhuman_ncrna_k17.nodegraph.info\nhuman_ncrna_k19.nodegraph\nhuman_ncrna_k19.nodegraph.info\nhuman_ncrna_k21.nodegraph\nhuman_ncrna_k21.nodegraph.info\nhuman_ncrna_k23.nodegraph\nhuman_ncrna_k23.nodegraph.info\nhuman_ncrna_k25.nodegraph\nhuman_ncrna_k25.nodegraph.info\nhuman_ncrna_k27.nodegraph\nhuman_ncrna_k27.nodegraph.info\nhuman_ncrna_k31.nodegraph\nhuman_ncrna_k31.nodegraph.info\n\u001b[01;31mSRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz\u001b[0m\n\u001b[01;31mSRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.fq.gz\u001b[0m\n"
      }
     ],
     "persistentId": "a30b0ea1-cbfb-4de5-bce5-329cfeb8e50c",
     "text": "ls"
    },
    "executionTime": "2019-10-12T04:23:54.983Z"
   },
   {
    "cell": {
     "executionCount": 4,
     "executionEventId": "b19c5688-5b20-49ea-925b-5189698704c7",
     "hasError": false,
     "id": "3286464d-a5c9-45c3-8540-f4b967804d47",
     "outputs": [],
     "persistentId": "56e48bd4-fcc6-4d93-b4e3-0984be1fc1d5",
     "text": "import gzip\nfrom io import StringIO\nfrom pprint import pprint\n\nfrom khmer import Nodegraph\nimport matplotlib.pyplot as plt\nimport screed\nfrom sourmash._minhash import hash_murmur\nfrom khmer.khmer_args import calculate_graphsize\nimport pandas as pd\nfrom sourmash.logging import notify\nimport seaborn as sns\nfrom sklearn import metrics\nfrom tqdm import tqdm\nfrom Bio.Seq import Seq\nfrom Bio import SeqIO\n\n# make all warnings error to get a stack trace\nimport warnings\nwarnings.filterwarnings(\"default\")\n\npd.options.display.max_colwidth = 100\n\nfrom khtools.compare_kmer_content import kmerize, hpize, dayhoffize\n"
    },
    "executionTime": "2019-10-12T04:23:56.969Z"
   },
   {
    "cell": {
     "executionCount": 5,
     "executionEventId": "ef447b08-607f-4462-bf72-d5af1c24c304",
     "hasError": false,
     "id": "a2b2503b-d248-407a-b936-191c45fc5e1f",
     "outputs": [],
     "persistentId": "1e54230e-ceec-42fc-afa6-7b32e219378b",
     "text": "from khtools.partition_reads import score_reads, make_peptide_bloom_filter, six_frame_translation_no_stops, is_low_complexity"
    },
    "executionTime": "2019-10-12T04:23:57.005Z"
   },
   {
    "cell": {
     "executionCount": 6,
     "executionEventId": "1b866682-a70a-47e8-ab65-de2f907aae6e",
     "hasError": false,
     "id": "a9bb9cbc-0224-43b1-99b0-a861bab8cf85",
     "outputs": [
      {
       "name": "stderr",
       "output_type": "stream",
       "text": "/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/screed/openscreed.py:39: ResourceWarning: unclosed file <_io.BufferedReader name='Homo_sapiens.GRCh38.pep.all.fa.gz'>\n  self.iter_fn = self.open_reader(filename, *args, **kwargs)\n"
      },
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "CPU times: user 39.3 s, sys: 1.99 s, total: 41.3 s\nWall time: 41.3 s\n"
      }
     ],
     "persistentId": "be06ded1-d7d4-4f21-9961-3aed5ff30cbf",
     "text": "%%time\npeptide_ksize = 7\n\npeptide_graph = make_peptide_bloom_filter(\"Homo_sapiens.GRCh38.pep.all.fa.gz\", peptide_ksize)"
    },
    "executionTime": "2019-10-12T04:24:38.362Z"
   },
   {
    "cell": {
     "executionCount": 7,
     "executionEventId": "daf9b319-a91b-4d1b-aee8-77c45e4cbb3e",
     "hasError": true,
     "id": "38925b79-49df-403d-b4f7-57b005296192",
     "outputs": [
      {
       "name": "stderr",
       "output_type": "stream",
       "text": "/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/screed/openscreed.py:39: ResourceWarning: unclosed file <_io.BufferedReader name='SRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz'>\n  self.iter_fn = self.open_reader(filename, *args, **kwargs)\n0it [00:00, ?it/s]\n"
      },
      {
       "ename": "TypeError",
       "evalue": "'int' object is not iterable",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
        "\u001b[0;32m<timed exec>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n",
        "\u001b[0;32m~/code/kmer-hashing/kh-tools/khtools/partition_reads.py\u001b[0m in \u001b[0;36mscore_reads\u001b[0;34m(reads, peptide_graph, peptide_ksize, jaccard_threshold, molecule, verbose)\u001b[0m\n\u001b[1;32m    177\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    178\u001b[0m         jaccard, n_kmers = score_single_sequence(sequence, peptide_graph,\n\u001b[0;32m--> 179\u001b[0;31m                                                  peptide_ksize, molecule, verbose)\n\u001b[0m\u001b[1;32m    180\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    181\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mjaccard\u001b[0m \u001b[0;34m>\u001b[0m \u001b[0mjaccard_threshold\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/code/kmer-hashing/kh-tools/khtools/partition_reads.py\u001b[0m in \u001b[0;36mscore_single_sequence\u001b[0;34m(sequence, peptide_graph, peptide_ksize, molecule, verbose)\u001b[0m\n\u001b[1;32m    146\u001b[0m         fraction_in_peptide_db, n_kmers = score_single_translation(\n\u001b[1;32m    147\u001b[0m             \u001b[0mtranslation\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpeptide_graph\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpeptide_ksize\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmolecule\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mmolecule\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 148\u001b[0;31m             verbose=verbose)\n\u001b[0m\u001b[1;32m    149\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    150\u001b[0m         \u001b[0;31m# Save the highest jaccard\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;31mTypeError\u001b[0m: 'int' object is not iterable"
       ]
      }
     ],
     "persistentId": "be06ded1-d7d4-4f21-9961-3aed5ff30cbf",
     "text": "%%time\n\nreads = 'SRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz'\n\ncoding_scores = score_reads(reads, peptide_graph, peptide_ksize)"
    },
    "executionTime": "2019-10-12T04:24:38.564Z"
   },
   {
    "cell": {
     "executionCount": 8,
     "executionEventId": "39e16950-20ab-4685-a5e9-a8e454d37eab",
     "hasError": false,
     "id": "74edbd37-99c3-4e60-80c4-3494df741e79",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "> \u001b[0;32m/home/olga/code/kmer-hashing/kh-tools/khtools/partition_reads.py\u001b[0m(148)\u001b[0;36mscore_single_sequence\u001b[0;34m()\u001b[0m\n\u001b[0;32m    146 \u001b[0;31m        fraction_in_peptide_db, n_kmers = score_single_translation(\n\u001b[0m\u001b[0;32m    147 \u001b[0;31m            \u001b[0mtranslation\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpeptide_graph\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpeptide_ksize\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmolecule\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mmolecule\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m--> 148 \u001b[0;31m            verbose=verbose)\n\u001b[0m\u001b[0;32m    149 \u001b[0;31m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    150 \u001b[0;31m        \u001b[0;31m# Save the highest jaccard\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\n"
      },
      {
       "name": "stdin",
       "output_type": "stream",
       "text": "ipdb>  dir()\n"
      },
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "['fraction_in_peptide_db', 'low_complexity', 'max_fraction_in_peptide_db', 'max_n_kmers', 'molecule', 'n_kmers', 'nucleotide_ksize', 'peptide_graph', 'peptide_ksize', 'seq', 'sequence', 'translation', 'translations', 'verbose']\n"
      },
      {
       "name": "stdin",
       "output_type": "stream",
       "text": "ipdb>  a\n"
      },
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "sequence = 'CCCCCCCCCACCACCACCCCCCCCACCCCCCCCCCCCCCCCCCCCCCCCCCACCCCCCCACACACCCCCAACACCC'\npeptide_graph = <khmer._oxli.graphs.Nodegraph object at 0x7fca21636930>\npeptide_ksize = 7\nmolecule = 'protein'\nverbose = False\n"
      },
      {
       "name": "stdin",
       "output_type": "stream",
       "text": "ipdb>  nucleotide_ksize\n"
      },
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "21\n"
      },
      {
       "name": "stdin",
       "output_type": "stream",
       "text": "ipdb>  translations\n"
      },
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "[Seq('PPPTTTPPTPPPPPPPPTPPHTPNT', ExtendedIUPACProtein()), Seq('PPPPPPPPPPPPPPPPPPPHTPPTP', ExtendedIUPACProtein()), Seq('PPHHHPPHPPPPPPPPHPPTHPQH', ExtendedIUPACProtein()), Seq('GCWGCVGGWGGGGGGGGWGGWWWGG', ExtendedIUPACProtein()), Seq('GVGGVWGGGGGGGGGGGGGGGGGGG', ExtendedIUPACProtein()), Seq('VLGVCGGVGGGGGGGGVGGVVVGG', ExtendedIUPACProtein())]\n"
      },
      {
       "name": "stdin",
       "output_type": "stream",
       "text": "ipdb>  print(low_complexity)\n"
      },
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "False\n"
      },
      {
       "name": "stdin",
       "output_type": "stream",
       "text": "ipdb>  76-21+1\n"
      },
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "56\n"
      },
      {
       "name": "stdin",
       "output_type": "stream",
       "text": "ipdb>  q\n"
      }
     ],
     "persistentId": "fefceffb-109b-4796-a998-8ad02d563488",
     "text": "%debug"
    },
    "executionTime": "2019-10-12T04:28:48.635Z"
   },
   {
    "cell": {
     "executionCount": 9,
     "executionEventId": "db481ae4-d675-467f-8440-1b0736664fac",
     "hasError": true,
     "id": "4babf021-168e-4575-be9f-a070c1adf48c",
     "outputs": [
      {
       "ename": "NameError",
       "evalue": "name 'coding_scores' is not defined",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
        "\u001b[0;32m<ipython-input-9-d293a3385843>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mcoding_scores_not_low_complexity\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcoding_scores\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mquery\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'jaccard_in_peptide_db >= 0'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      2\u001b[0m \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcoding_scores_not_low_complexity\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshape\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      3\u001b[0m \u001b[0mcoding_scores_not_low_complexity\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mhead\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;31mNameError\u001b[0m: name 'coding_scores' is not defined"
       ]
      }
     ],
     "persistentId": "73b49c34-8379-4aa5-835e-57c54aced1fa",
     "text": "coding_scores_not_low_complexity = coding_scores.query('jaccard_in_peptide_db >= 0')\nprint(coding_scores_not_low_complexity.shape)\ncoding_scores_not_low_complexity.head()"
    },
    "executionTime": "2019-10-12T04:28:48.739Z"
   },
   {
    "cell": {
     "executionCount": 10,
     "executionEventId": "58f466f5-2581-415c-8b08-3e9b9a09dc1b",
     "hasError": true,
     "id": "38925b79-49df-403d-b4f7-57b005296192",
     "outputs": [
      {
       "name": "stderr",
       "output_type": "stream",
       "text": "/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/screed/openscreed.py:39: ResourceWarning: unclosed file <_io.BufferedReader name='SRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz'>\n  self.iter_fn = self.open_reader(filename, *args, **kwargs)\n0it [00:00, ?it/s]\n"
      },
      {
       "ename": "TypeError",
       "evalue": "'int' object is not iterable",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
        "\u001b[0;32m<timed exec>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n",
        "\u001b[0;32m~/code/kmer-hashing/kh-tools/khtools/partition_reads.py\u001b[0m in \u001b[0;36mscore_reads\u001b[0;34m(reads, peptide_graph, peptide_ksize, jaccard_threshold, molecule, verbose)\u001b[0m\n\u001b[1;32m    182\u001b[0m         jaccard, n_kmers = score_single_sequence(sequence, peptide_graph,\n\u001b[1;32m    183\u001b[0m                                                  \u001b[0mpeptide_ksize\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmolecule\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 184\u001b[0;31m                                                  verbose)\n\u001b[0m\u001b[1;32m    185\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    186\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mjaccard\u001b[0m \u001b[0;34m>\u001b[0m \u001b[0mjaccard_threshold\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/code/kmer-hashing/kh-tools/khtools/partition_reads.py\u001b[0m in \u001b[0;36mscore_single_sequence\u001b[0;34m(sequence, peptide_graph, peptide_ksize, molecule, verbose)\u001b[0m\n\u001b[1;32m    148\u001b[0m         fraction_in_peptide_db, n_kmers = score_single_translation(\n\u001b[1;32m    149\u001b[0m             \u001b[0mtranslation\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpeptide_graph\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpeptide_ksize\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmolecule\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mmolecule\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 150\u001b[0;31m             verbose=verbose)\n\u001b[0m\u001b[1;32m    151\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    152\u001b[0m         \u001b[0;31m# Save the highest jaccard\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;31mTypeError\u001b[0m: 'int' object is not iterable"
       ]
      }
     ],
     "persistentId": "be06ded1-d7d4-4f21-9961-3aed5ff30cbf",
     "text": "%%time\n\nreads = 'SRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz'\n\ncoding_scores = score_reads(reads, peptide_graph, peptide_ksize)"
    },
    "executionTime": "2019-10-12T04:53:27.059Z"
   },
   {
    "cell": {
     "executionCount": 2,
     "executionEventId": "6f832ca7-7d1e-46b0-9e84-6f5368d878b7",
     "hasError": false,
     "id": "802d541f-5fd3-48b7-99a1-9883a78e612f",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "/mnt/data_sm/olga/kmer-hashing/classify_coding_vs_noncoding\n"
      }
     ],
     "persistentId": "1d547cf2-254f-4fe2-9bf0-5b7e39be4376",
     "text": "cd $folder"
    },
    "executionTime": "2019-10-12T04:53:39.813Z"
   },
   {
    "cell": {
     "executionCount": 3,
     "executionEventId": "e92ee1f2-49f0-4ab8-9afa-823fb64fd5fc",
     "hasError": false,
     "id": "bb96f994-0b34-4694-980d-db02f28e5e9b",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "\u001b[0m\u001b[01;31mHomo_sapiens.GRCh38.cds.all.fa.gz\u001b[0m\n\u001b[01;31mHomo_sapiens.GRCh38.ncrna.fa.gz\u001b[0m\n\u001b[01;31mHomo_sapiens.GRCh38.pep.all.fa.gz\u001b[0m\nhuman_cds_k15.nodegraph\nhuman_cds_k15.nodegraph.info\nhuman_cds_k17.nodegraph\nhuman_cds_k17.nodegraph.info\nhuman_cds_k19.nodegraph\nhuman_cds_k19.nodegraph.info\nhuman_cds_k21.nodegraph\nhuman_cds_k21.nodegraph.info\nhuman_cds_k23.nodegraph\nhuman_cds_k23.nodegraph.info\nhuman_cds_k25.nodegraph\nhuman_cds_k25.nodegraph.info\nhuman_cds_k27.nodegraph\nhuman_cds_k27.nodegraph.info\nhuman_cds_k31.nodegraph\nhuman_cds_k31.nodegraph.info\nhuman_ncrna_k15.nodegraph\nhuman_ncrna_k15.nodegraph.info\nhuman_ncrna_k15.nodegraph.tagset\nhuman_ncrna_k17.nodegraph\nhuman_ncrna_k17.nodegraph.info\nhuman_ncrna_k19.nodegraph\nhuman_ncrna_k19.nodegraph.info\nhuman_ncrna_k21.nodegraph\nhuman_ncrna_k21.nodegraph.info\nhuman_ncrna_k23.nodegraph\nhuman_ncrna_k23.nodegraph.info\nhuman_ncrna_k25.nodegraph\nhuman_ncrna_k25.nodegraph.info\nhuman_ncrna_k27.nodegraph\nhuman_ncrna_k27.nodegraph.info\nhuman_ncrna_k31.nodegraph\nhuman_ncrna_k31.nodegraph.info\n\u001b[01;31mSRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz\u001b[0m\n\u001b[01;31mSRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.fq.gz\u001b[0m\n"
      }
     ],
     "persistentId": "a30b0ea1-cbfb-4de5-bce5-329cfeb8e50c",
     "text": "ls"
    },
    "executionTime": "2019-10-12T04:53:40.391Z"
   },
   {
    "cell": {
     "executionCount": 4,
     "executionEventId": "a0ff5b23-3c4d-4d4a-82bd-3b21ccdea1ac",
     "hasError": false,
     "id": "3286464d-a5c9-45c3-8540-f4b967804d47",
     "outputs": [],
     "persistentId": "56e48bd4-fcc6-4d93-b4e3-0984be1fc1d5",
     "text": "import gzip\nfrom io import StringIO\nfrom pprint import pprint\n\nfrom khmer import Nodegraph\nimport matplotlib.pyplot as plt\nimport screed\nfrom sourmash._minhash import hash_murmur\nfrom khmer.khmer_args import calculate_graphsize\nimport pandas as pd\nfrom sourmash.logging import notify\nimport seaborn as sns\nfrom sklearn import metrics\nfrom tqdm import tqdm\nfrom Bio.Seq import Seq\nfrom Bio import SeqIO\n\n# make all warnings error to get a stack trace\nimport warnings\nwarnings.filterwarnings(\"default\")\n\npd.options.display.max_colwidth = 100\n\nfrom khtools.compare_kmer_content import kmerize, hpize, dayhoffize\n"
    },
    "executionTime": "2019-10-12T04:53:42.072Z"
   },
   {
    "cell": {
     "executionCount": 5,
     "executionEventId": "868e155e-0fbe-4584-806b-da9c7dcb7318",
     "hasError": false,
     "id": "a2b2503b-d248-407a-b936-191c45fc5e1f",
     "outputs": [],
     "persistentId": "1e54230e-ceec-42fc-afa6-7b32e219378b",
     "text": "from khtools.partition_reads import score_reads, make_peptide_bloom_filter, six_frame_translation_no_stops, is_low_complexity"
    },
    "executionTime": "2019-10-12T04:53:42.253Z"
   },
   {
    "cell": {
     "executionCount": 6,
     "executionEventId": "55dbc59f-a9b1-4219-8d57-6ba23e8315f4",
     "hasError": false,
     "id": "a9bb9cbc-0224-43b1-99b0-a861bab8cf85",
     "outputs": [
      {
       "name": "stderr",
       "output_type": "stream",
       "text": "/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/screed/openscreed.py:39: ResourceWarning: unclosed file <_io.BufferedReader name='Homo_sapiens.GRCh38.pep.all.fa.gz'>\n  self.iter_fn = self.open_reader(filename, *args, **kwargs)\n"
      },
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "CPU times: user 43.3 s, sys: 3.02 s, total: 46.3 s\nWall time: 46.4 s\n"
      }
     ],
     "persistentId": "be06ded1-d7d4-4f21-9961-3aed5ff30cbf",
     "text": "%%time\npeptide_ksize = 7\n\npeptide_graph = make_peptide_bloom_filter(\"Homo_sapiens.GRCh38.pep.all.fa.gz\", peptide_ksize)"
    },
    "executionTime": "2019-10-12T04:54:28.667Z"
   },
   {
    "cell": {
     "executionCount": 7,
     "executionEventId": "0096d602-7fa4-4c3f-943a-44d05b7b5a62",
     "hasError": true,
     "id": "38925b79-49df-403d-b4f7-57b005296192",
     "outputs": [
      {
       "name": "stderr",
       "output_type": "stream",
       "text": "/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/screed/openscreed.py:39: ResourceWarning: unclosed file <_io.BufferedReader name='SRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz'>\n  self.iter_fn = self.open_reader(filename, *args, **kwargs)\n0it [00:00, ?it/s]\n"
      },
      {
       "ename": "TypeError",
       "evalue": "'int' object is not iterable",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
        "\u001b[0;32m<timed exec>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n",
        "\u001b[0;32m~/code/kmer-hashing/kh-tools/khtools/partition_reads.py\u001b[0m in \u001b[0;36mscore_reads\u001b[0;34m(reads, peptide_graph, peptide_ksize, jaccard_threshold, molecule, verbose)\u001b[0m\n\u001b[1;32m    182\u001b[0m         jaccard, n_kmers = score_single_sequence(sequence, peptide_graph,\n\u001b[1;32m    183\u001b[0m                                                  \u001b[0mpeptide_ksize\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmolecule\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 184\u001b[0;31m                                                  verbose)\n\u001b[0m\u001b[1;32m    185\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    186\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mjaccard\u001b[0m \u001b[0;34m>\u001b[0m \u001b[0mjaccard_threshold\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/code/kmer-hashing/kh-tools/khtools/partition_reads.py\u001b[0m in \u001b[0;36mscore_single_sequence\u001b[0;34m(sequence, peptide_graph, peptide_ksize, molecule, verbose)\u001b[0m\n\u001b[1;32m    148\u001b[0m         fraction_in_peptide_db, n_kmers = score_single_translation(\n\u001b[1;32m    149\u001b[0m             \u001b[0mtranslation\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpeptide_graph\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpeptide_ksize\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmolecule\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mmolecule\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 150\u001b[0;31m             verbose=verbose)\n\u001b[0m\u001b[1;32m    151\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    152\u001b[0m         \u001b[0;31m# Save the highest jaccard\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;31mTypeError\u001b[0m: 'int' object is not iterable"
       ]
      }
     ],
     "persistentId": "be06ded1-d7d4-4f21-9961-3aed5ff30cbf",
     "text": "%%time\n\nreads = 'SRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz'\n\ncoding_scores = score_reads(reads, peptide_graph, peptide_ksize)"
    },
    "executionTime": "2019-10-12T04:54:28.785Z"
   },
   {
    "cell": {
     "executionCount": 8,
     "executionEventId": "112e4e70-2bce-45e5-9350-fb812e9a86cf",
     "hasError": false,
     "id": "74edbd37-99c3-4e60-80c4-3494df741e79",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "> \u001b[0;32m/home/olga/code/kmer-hashing/kh-tools/khtools/partition_reads.py\u001b[0m(150)\u001b[0;36mscore_single_sequence\u001b[0;34m()\u001b[0m\n\u001b[0;32m    148 \u001b[0;31m        fraction_in_peptide_db, n_kmers = score_single_translation(\n\u001b[0m\u001b[0;32m    149 \u001b[0;31m            \u001b[0mtranslation\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpeptide_graph\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpeptide_ksize\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmolecule\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mmolecule\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m--> 150 \u001b[0;31m            verbose=verbose)\n\u001b[0m\u001b[0;32m    151 \u001b[0;31m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    152 \u001b[0;31m        \u001b[0;31m# Save the highest jaccard\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\n"
      },
      {
       "name": "stdin",
       "output_type": "stream",
       "text": "ipdb>  dir()\n"
      },
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "['fraction_in_peptide_db', 'low_complexity', 'max_fraction_in_peptide_db', 'max_n_kmers', 'molecule', 'n_kmers', 'nucleotide_ksize', 'peptide_graph', 'peptide_ksize', 'seq', 'sequence', 'translation', 'translations', 'verbose']\n"
      },
      {
       "name": "stdin",
       "output_type": "stream",
       "text": "ipdb>  a\n"
      },
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "sequence = 'CCCCCCCCCACCACCACCCCCCCCACCCCCCCCCCCCCCCCCCCCCCCCCCACCCCCCCACACACCCCCAACACCC'\npeptide_graph = <khmer._oxli.graphs.Nodegraph object at 0x7f76dd345930>\npeptide_ksize = 7\nmolecule = 'protein'\nverbose = False\n"
      },
      {
       "name": "stdin",
       "output_type": "stream",
       "text": "ipdb>  record\n"
      },
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "*** NameError: name 'record' is not defined\n"
      },
      {
       "name": "stdin",
       "output_type": "stream",
       "text": "ipdb>  u\n"
      },
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "> \u001b[0;32m/home/olga/code/kmer-hashing/kh-tools/khtools/partition_reads.py\u001b[0m(184)\u001b[0;36mscore_reads\u001b[0;34m()\u001b[0m\n\u001b[0;32m    182 \u001b[0;31m        jaccard, n_kmers = score_single_sequence(sequence, peptide_graph,\n\u001b[0m\u001b[0;32m    183 \u001b[0;31m                                                 \u001b[0mpeptide_ksize\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmolecule\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m--> 184 \u001b[0;31m                                                 verbose)\n\u001b[0m\u001b[0;32m    185 \u001b[0;31m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    186 \u001b[0;31m        \u001b[0;32mif\u001b[0m \u001b[0mjaccard\u001b[0m \u001b[0;34m>\u001b[0m \u001b[0mjaccard_threshold\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\n"
      },
      {
       "name": "stdin",
       "output_type": "stream",
       "text": "ipdb>  record\n"
      },
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "{'name': 'SRR306838.2318 Ibis_Run100924_C3PO:6:1:15779:1141/1', 'sequence': 'CCCCCCCCCACCACCACCCCCCCCACCCCCCCCCCCCCCCCCCCCCCCCCCACCCCCCCACACACCCCCAACACCC', 'annotations': '', 'quality': '2263688B;9<<9;=;9<=><:;=:@<@<<<A<<I9S?;<<<C;>;@;S5:;;M<O:7;;C<;:F;;;;898972@'}\n"
      },
      {
       "name": "stdin",
       "output_type": "stream",
       "text": "ipdb>  is_low_complexity\n"
      },
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "<function is_low_complexity at 0x7f76dd2ca378>\n"
      },
      {
       "name": "stdin",
       "output_type": "stream",
       "text": "ipdb>  q\n"
      }
     ],
     "persistentId": "fefceffb-109b-4796-a998-8ad02d563488",
     "text": "%debug"
    },
    "executionTime": "2019-10-12T05:08:08.671Z"
   },
   {
    "cell": {
     "executionCount": 9,
     "executionEventId": "7160686b-e036-481f-b17b-973aab7d7f1d",
     "hasError": false,
     "id": "74edbd37-99c3-4e60-80c4-3494df741e79",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "> \u001b[0;32m/home/olga/code/kmer-hashing/kh-tools/khtools/partition_reads.py\u001b[0m(150)\u001b[0;36mscore_single_sequence\u001b[0;34m()\u001b[0m\n\u001b[0;32m    148 \u001b[0;31m        fraction_in_peptide_db, n_kmers = score_single_translation(\n\u001b[0m\u001b[0;32m    149 \u001b[0;31m            \u001b[0mtranslation\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpeptide_graph\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpeptide_ksize\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmolecule\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mmolecule\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m--> 150 \u001b[0;31m            verbose=verbose)\n\u001b[0m\u001b[0;32m    151 \u001b[0;31m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    152 \u001b[0;31m        \u001b[0;31m# Save the highest jaccard\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\n"
      },
      {
       "name": "stdin",
       "output_type": "stream",
       "text": "ipdb>  q\n"
      }
     ],
     "persistentId": "fefceffb-109b-4796-a998-8ad02d563488",
     "text": "%debug"
    },
    "executionTime": "2019-10-12T05:08:11.293Z"
   },
   {
    "cell": {
     "executionCount": 10,
     "executionEventId": "5847a61a-ff0a-450d-ab65-54e2a85712d4",
     "hasError": true,
     "id": "38925b79-49df-403d-b4f7-57b005296192",
     "outputs": [
      {
       "name": "stderr",
       "output_type": "stream",
       "text": "/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/screed/openscreed.py:39: ResourceWarning: unclosed file <_io.BufferedReader name='SRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz'>\n  self.iter_fn = self.open_reader(filename, *args, **kwargs)\n0it [00:00, ?it/s]\n"
      },
      {
       "ename": "TypeError",
       "evalue": "'int' object is not iterable",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
        "\u001b[0;32m<timed exec>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n",
        "\u001b[0;32m~/code/kmer-hashing/kh-tools/khtools/partition_reads.py\u001b[0m in \u001b[0;36mscore_reads\u001b[0;34m(reads, peptide_graph, peptide_ksize, jaccard_threshold, molecule, verbose)\u001b[0m\n\u001b[1;32m    182\u001b[0m         jaccard, n_kmers = score_single_sequence(sequence, peptide_graph,\n\u001b[1;32m    183\u001b[0m                                                  \u001b[0mpeptide_ksize\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmolecule\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 184\u001b[0;31m                                                  verbose)\n\u001b[0m\u001b[1;32m    185\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    186\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mjaccard\u001b[0m \u001b[0;34m>\u001b[0m \u001b[0mjaccard_threshold\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/code/kmer-hashing/kh-tools/khtools/partition_reads.py\u001b[0m in \u001b[0;36mscore_single_sequence\u001b[0;34m(sequence, peptide_graph, peptide_ksize, molecule, verbose)\u001b[0m\n\u001b[1;32m    148\u001b[0m         fraction_in_peptide_db, n_kmers = score_single_translation(\n\u001b[1;32m    149\u001b[0m             \u001b[0mtranslation\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpeptide_graph\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpeptide_ksize\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmolecule\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mmolecule\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 150\u001b[0;31m             verbose=verbose)\n\u001b[0m\u001b[1;32m    151\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    152\u001b[0m         \u001b[0;31m# Save the highest jaccard\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;31mTypeError\u001b[0m: 'int' object is not iterable"
       ]
      }
     ],
     "persistentId": "be06ded1-d7d4-4f21-9961-3aed5ff30cbf",
     "text": "%%time\n\nreads = 'SRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz'\n\ncoding_scores = score_reads(reads, peptide_graph, peptide_ksize)"
    },
    "executionTime": "2019-10-12T05:08:12.982Z"
   },
   {
    "cell": {
     "executionCount": 11,
     "executionEventId": "e6f5b1fc-d929-4e5e-86d8-e8cac746b72f",
     "hasError": false,
     "id": "38925b79-49df-403d-b4f7-57b005296192",
     "outputs": [
      {
       "name": "stderr",
       "output_type": "stream",
       "text": "/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/screed/openscreed.py:39: ResourceWarning: unclosed file <_io.BufferedReader name='SRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz'>\n  self.iter_fn = self.open_reader(filename, *args, **kwargs)\n0it [00:00, ?it/s]/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n187it [00:00, 1862.34it/s]/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n326it [00:00, 1688.53it/s]/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n471it [00:00, 1607.88it/s]/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n635it [00:00, 1616.64it/s]/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n803it [00:00, 1635.07it/s]/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n949it [00:00, 1576.77it/s]/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n1134it [00:00, 1649.21it/s]/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n1317it [00:00, 1698.28it/s]/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n1503it [00:00, 1742.24it/s]/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n1688it [00:01, 1772.97it/s]/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n1862it [00:01, 1688.83it/s]/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n2034it [00:01, 1697.55it/s]/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n2203it [00:01, 1419.30it/s]/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n2360it [00:01, 1460.40it/s]/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n2542it [00:01, 1549.22it/s]/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n2725it [00:01, 1623.94it/s]/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n2908it [00:01, 1678.44it/s]/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n3081it [00:01, 1692.31it/s]/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n3253it [00:02, 1447.19it/s]/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n3407it [00:02, 1473.20it/s]/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n3562it [00:02, 1493.78it/s]/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n3716it [00:02, 1496.82it/s]/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n3883it [00:02, 1544.66it/s]/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n4051it [00:02, 1582.83it/s]/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n4214it [00:02, 1593.54it/s]/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n4375it [00:02, 1458.58it/s]/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py:175: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n  \"the new string hashing behaviour.\", BiopythonWarning)\n\n\nKeyboardInterrupt\n\n"
      }
     ],
     "persistentId": "be06ded1-d7d4-4f21-9961-3aed5ff30cbf",
     "text": "%%time\n\nreads = 'SRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz'\n\ncoding_scores = score_reads(reads, peptide_graph, peptide_ksize)"
    },
    "executionTime": "2019-10-12T05:09:33.093Z"
   },
   {
    "cell": {
     "executionCount": 12,
     "executionEventId": "9f06c90d-e500-4d93-97d9-9f964cced7bf",
     "hasError": true,
     "id": "38925b79-49df-403d-b4f7-57b005296192",
     "outputs": [
      {
       "name": "stderr",
       "output_type": "stream",
       "text": "Exception ignored in: <_io.FileIO name='SRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz' mode='rb' closefd=True>\nResourceWarning: unclosed file <_io.BufferedReader name='SRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz'>\n0it [00:00, ?it/s]\n"
      },
      {
       "ename": "BiopythonWarning",
       "evalue": "Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mBiopythonWarning\u001b[0m                          Traceback (most recent call last)",
        "\u001b[0;32m<timed exec>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n",
        "\u001b[0;32m~/code/kmer-hashing/kh-tools/khtools/partition_reads.py\u001b[0m in \u001b[0;36mscore_reads\u001b[0;34m(reads, peptide_graph, peptide_ksize, jaccard_threshold, molecule, verbose)\u001b[0m\n\u001b[1;32m    173\u001b[0m         jaccard, n_kmers = score_single_sequence(sequence, peptide_graph,\n\u001b[1;32m    174\u001b[0m                                                  \u001b[0mpeptide_ksize\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmolecule\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 175\u001b[0;31m                                                  verbose)\n\u001b[0m\u001b[1;32m    176\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    177\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mjaccard\u001b[0m \u001b[0;34m>\u001b[0m \u001b[0mjaccard_threshold\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/code/kmer-hashing/kh-tools/khtools/partition_reads.py\u001b[0m in \u001b[0;36mscore_single_sequence\u001b[0;34m(sequence, peptide_graph, peptide_ksize, molecule, verbose)\u001b[0m\n\u001b[1;32m    133\u001b[0m         \u001b[0mtranslation\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mencode_peptide\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtranslation\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmolecule\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    134\u001b[0m         low_complexity, n_kmers = compute_low_complexity(translation,\n\u001b[0;32m--> 135\u001b[0;31m                                                          peptide_ksize)\n\u001b[0m\u001b[1;32m    136\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mlow_complexity\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    137\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0;34m-\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mn_kmers\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/code/kmer-hashing/kh-tools/khtools/partition_reads.py\u001b[0m in \u001b[0;36mcompute_low_complexity\u001b[0;34m(sequence, ksize)\u001b[0m\n\u001b[1;32m    110\u001b[0m \u001b[0;32mdef\u001b[0m \u001b[0mcompute_low_complexity\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msequence\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mksize\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    111\u001b[0m     \u001b[0;34m\"\"\"CHeck if seqauence is low complexity, i.e. low entropy, mostly repetitive\"\"\"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 112\u001b[0;31m     \u001b[0mkmers\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mkmerize\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msequence\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mksize\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    113\u001b[0m     \u001b[0mn_kmers\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkmers\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    114\u001b[0m     \u001b[0mn_possible_kmers_on_sequence\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msequence\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m-\u001b[0m \u001b[0mksize\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0;36m1\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/code/kmer-hashing/kh-tools/khtools/compare_kmer_content.py\u001b[0m in \u001b[0;36mkmerize\u001b[0;34m(seq, ksize)\u001b[0m\n\u001b[1;32m     75\u001b[0m \u001b[0;32mdef\u001b[0m \u001b[0mkmerize\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mseq\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mksize\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     76\u001b[0m     \u001b[0;34m\"\"\"Return the set of unique k-mers from the sequence\"\"\"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 77\u001b[0;31m     \u001b[0;32mreturn\u001b[0m \u001b[0mset\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mseq\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0mksize\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mi\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mrange\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mseq\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m-\u001b[0m\u001b[0mksize\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     78\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     79\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py\u001b[0m in \u001b[0;36m__hash__\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    173\u001b[0m                       \u001b[0;34m\"or my_dict[id(my_seq)] if you want the old behaviour, \"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    174\u001b[0m                       \u001b[0;34m\"or use hash(str(my_seq)) or my_dict[str(my_seq)] for \"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 175\u001b[0;31m                       \"the new string hashing behaviour.\", BiopythonWarning)\n\u001b[0m\u001b[1;32m    176\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mhash\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    177\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;31mBiopythonWarning\u001b[0m: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour."
       ]
      }
     ],
     "persistentId": "be06ded1-d7d4-4f21-9961-3aed5ff30cbf",
     "text": "%%time\n\nwarnings.filterwarnings(\"error\")\n\nreads = 'SRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz'\n\ncoding_scores = score_reads(reads, peptide_graph, peptide_ksize)"
    },
    "executionTime": "2019-10-12T05:09:42.843Z"
   },
   {
    "cell": {
     "executionCount": 13,
     "executionEventId": "90db3f7d-24db-4372-8851-5bbe3b1976f6",
     "hasError": true,
     "id": "38925b79-49df-403d-b4f7-57b005296192",
     "outputs": [
      {
       "name": "stderr",
       "output_type": "stream",
       "text": "Exception ignored in: <_io.FileIO name='SRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz' mode='rb' closefd=True>\nResourceWarning: unclosed file <_io.BufferedReader name='SRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz'>\n0it [00:00, ?it/s]\n"
      },
      {
       "ename": "BiopythonWarning",
       "evalue": "Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mBiopythonWarning\u001b[0m                          Traceback (most recent call last)",
        "\u001b[0;32m<timed exec>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n",
        "\u001b[0;32m~/code/kmer-hashing/kh-tools/khtools/partition_reads.py\u001b[0m in \u001b[0;36mscore_reads\u001b[0;34m(reads, peptide_graph, peptide_ksize, jaccard_threshold, molecule, verbose)\u001b[0m\n\u001b[1;32m    173\u001b[0m         jaccard, n_kmers = score_single_sequence(sequence, peptide_graph,\n\u001b[1;32m    174\u001b[0m                                                  \u001b[0mpeptide_ksize\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmolecule\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 175\u001b[0;31m                                                  verbose)\n\u001b[0m\u001b[1;32m    176\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    177\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mjaccard\u001b[0m \u001b[0;34m>\u001b[0m \u001b[0mjaccard_threshold\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/code/kmer-hashing/kh-tools/khtools/partition_reads.py\u001b[0m in \u001b[0;36mscore_single_sequence\u001b[0;34m(sequence, peptide_graph, peptide_ksize, molecule, verbose)\u001b[0m\n\u001b[1;32m    133\u001b[0m         \u001b[0mtranslation\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mencode_peptide\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtranslation\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmolecule\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    134\u001b[0m         low_complexity, n_kmers = compute_low_complexity(translation,\n\u001b[0;32m--> 135\u001b[0;31m                                                          peptide_ksize)\n\u001b[0m\u001b[1;32m    136\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mlow_complexity\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    137\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0;34m-\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mn_kmers\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/code/kmer-hashing/kh-tools/khtools/partition_reads.py\u001b[0m in \u001b[0;36mcompute_low_complexity\u001b[0;34m(sequence, ksize)\u001b[0m\n\u001b[1;32m    110\u001b[0m \u001b[0;32mdef\u001b[0m \u001b[0mcompute_low_complexity\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msequence\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mksize\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    111\u001b[0m     \u001b[0;34m\"\"\"CHeck if seqauence is low complexity, i.e. low entropy, mostly repetitive\"\"\"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 112\u001b[0;31m     \u001b[0mkmers\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mkmerize\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msequence\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mksize\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    113\u001b[0m     \u001b[0mn_kmers\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkmers\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    114\u001b[0m     \u001b[0mn_possible_kmers_on_sequence\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msequence\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m-\u001b[0m \u001b[0mksize\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0;36m1\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/code/kmer-hashing/kh-tools/khtools/compare_kmer_content.py\u001b[0m in \u001b[0;36mkmerize\u001b[0;34m(seq, ksize)\u001b[0m\n\u001b[1;32m     75\u001b[0m \u001b[0;32mdef\u001b[0m \u001b[0mkmerize\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mseq\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mksize\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     76\u001b[0m     \u001b[0;34m\"\"\"Return the set of unique k-mers from the sequence\"\"\"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 77\u001b[0;31m     \u001b[0;32mreturn\u001b[0m \u001b[0mset\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mseq\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0mksize\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mi\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mrange\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mseq\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m-\u001b[0m\u001b[0mksize\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     78\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     79\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/Bio/Seq.py\u001b[0m in \u001b[0;36m__hash__\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    173\u001b[0m                       \u001b[0;34m\"or my_dict[id(my_seq)] if you want the old behaviour, \"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    174\u001b[0m                       \u001b[0;34m\"or use hash(str(my_seq)) or my_dict[str(my_seq)] for \"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 175\u001b[0;31m                       \"the new string hashing behaviour.\", BiopythonWarning)\n\u001b[0m\u001b[1;32m    176\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mhash\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    177\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;31mBiopythonWarning\u001b[0m: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour."
       ]
      }
     ],
     "persistentId": "be06ded1-d7d4-4f21-9961-3aed5ff30cbf",
     "text": "%%time\n\nwarnings.filterwarnings(\"error\")\n\nreads = 'SRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz'\n\ncoding_scores = score_reads(reads, peptide_graph, peptide_ksize)"
    },
    "executionTime": "2019-10-12T05:10:58.056Z"
   },
   {
    "cell": {
     "executionCount": 14,
     "executionEventId": "c23013f0-8e6c-4d0f-9cf5-f098d31a310c",
     "hasError": true,
     "id": "38925b79-49df-403d-b4f7-57b005296192",
     "outputs": [
      {
       "name": "stderr",
       "output_type": "stream",
       "text": "Exception ignored in: <_io.FileIO name='SRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz' mode='rb' closefd=True>\nResourceWarning: unclosed file <_io.BufferedReader name='SRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz'>\n0it [00:00, ?it/s]\n"
      },
      {
       "ename": "AttributeError",
       "evalue": "__enter__",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
        "\u001b[0;32m<timed exec>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n",
        "\u001b[0;32m~/code/kmer-hashing/kh-tools/khtools/partition_reads.py\u001b[0m in \u001b[0;36mscore_reads\u001b[0;34m(reads, peptide_graph, peptide_ksize, jaccard_threshold, molecule, verbose)\u001b[0m\n\u001b[1;32m    176\u001b[0m         jaccard, n_kmers = score_single_sequence(sequence, peptide_graph,\n\u001b[1;32m    177\u001b[0m                                                  \u001b[0mpeptide_ksize\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmolecule\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 178\u001b[0;31m                                                  verbose)\n\u001b[0m\u001b[1;32m    179\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    180\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mjaccard\u001b[0m \u001b[0;34m>\u001b[0m \u001b[0mjaccard_threshold\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/code/kmer-hashing/kh-tools/khtools/partition_reads.py\u001b[0m in \u001b[0;36mscore_single_sequence\u001b[0;34m(sequence, peptide_graph, peptide_ksize, molecule, verbose)\u001b[0m\n\u001b[1;32m    133\u001b[0m         \u001b[0mtranslation\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mencode_peptide\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtranslation\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmolecule\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    134\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 135\u001b[0;31m         \u001b[0;32mwith\u001b[0m \u001b[0mwarnings\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfilterwarnings\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'ignore'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    136\u001b[0m             \u001b[0;31m# Ignore Biopython warning of seq objects being strings now\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    137\u001b[0m             low_complexity, n_kmers = compute_low_complexity(translation,\n",
        "\u001b[0;31mAttributeError\u001b[0m: __enter__"
       ]
      }
     ],
     "persistentId": "be06ded1-d7d4-4f21-9961-3aed5ff30cbf",
     "text": "%%time\n\nwarnings.filterwarnings(\"error\")\n\nreads = 'SRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz'\n\ncoding_scores = score_reads(reads, peptide_graph, peptide_ksize)"
    },
    "executionTime": "2019-10-12T05:11:03.989Z"
   },
   {
    "cell": {
     "executionCount": 15,
     "executionEventId": "b956639b-10ca-46f9-b9c3-8b5c71fbc992",
     "hasError": false,
     "id": "38925b79-49df-403d-b4f7-57b005296192",
     "outputs": [
      {
       "name": "stderr",
       "output_type": "stream",
       "text": "Exception ignored in: <_io.FileIO name='SRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz' mode='rb' closefd=True>\nResourceWarning: unclosed file <_io.BufferedReader name='SRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz'>\n795517it [06:57, 1904.05it/s]\n"
      },
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "CPU times: user 6min 44s, sys: 17.3 s, total: 7min 1s\nWall time: 6min 58s\n"
      }
     ],
     "persistentId": "be06ded1-d7d4-4f21-9961-3aed5ff30cbf",
     "text": "%%time\n\nwarnings.filterwarnings(\"error\")\n\nreads = 'SRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz'\n\ncoding_scores = score_reads(reads, peptide_graph, peptide_ksize)"
    },
    "executionTime": "2019-10-12T05:19:26.281Z"
   },
   {
    "cell": {
     "executionCount": 16,
     "executionEventId": "813eb9f3-09eb-4cfc-b2e0-b9164856aa12",
     "hasError": false,
     "id": "4babf021-168e-4575-be9f-a070c1adf48c",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "(793969, 4)\n"
      },
      {
       "data": {
        "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>read_id</th>\n      <th>jaccard_in_peptide_db</th>\n      <th>n_kmers</th>\n      <th>classification</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>1</th>\n      <td>SRR306838.1617 Ibis_Run100924_C3PO:6:1:9036:1074/1</td>\n      <td>0.000000</td>\n      <td>0</td>\n      <td>non-coding</td>\n    </tr>\n    <tr>\n      <th>7</th>\n      <td>SRR306838.2723 Ibis_Run100924_C3PO:6:1:15754:1168/1</td>\n      <td>0.769231</td>\n      <td>13</td>\n      <td>non-coding</td>\n    </tr>\n    <tr>\n      <th>9</th>\n      <td>SRR306838.3707 Ibis_Run100924_C3PO:6:1:5241:1269/1</td>\n      <td>1.000000</td>\n      <td>16</td>\n      <td>coding</td>\n    </tr>\n    <tr>\n      <th>10</th>\n      <td>SRR306838.4251 Ibis_Run100924_C3PO:6:1:5141:1314/1</td>\n      <td>0.000000</td>\n      <td>15</td>\n      <td>non-coding</td>\n    </tr>\n    <tr>\n      <th>11</th>\n      <td>SRR306838.4936 Ibis_Run100924_C3PO:6:1:5473:1375/1</td>\n      <td>0.000000</td>\n      <td>14</td>\n      <td>non-coding</td>\n    </tr>\n  </tbody>\n</table>\n</div>",
        "text/plain": "                                                read_id  \\\n1    SRR306838.1617 Ibis_Run100924_C3PO:6:1:9036:1074/1   \n7   SRR306838.2723 Ibis_Run100924_C3PO:6:1:15754:1168/1   \n9    SRR306838.3707 Ibis_Run100924_C3PO:6:1:5241:1269/1   \n10   SRR306838.4251 Ibis_Run100924_C3PO:6:1:5141:1314/1   \n11   SRR306838.4936 Ibis_Run100924_C3PO:6:1:5473:1375/1   \n\n    jaccard_in_peptide_db  n_kmers classification  \n1                0.000000        0     non-coding  \n7                0.769231       13     non-coding  \n9                1.000000       16         coding  \n10               0.000000       15     non-coding  \n11               0.000000       14     non-coding  "
       },
       "execution_count": 16,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "73b49c34-8379-4aa5-835e-57c54aced1fa",
     "text": "coding_scores_not_low_complexity = coding_scores.query('jaccard_in_peptide_db >= 0')\nprint(coding_scores_not_low_complexity.shape)\ncoding_scores_not_low_complexity.head()"
    },
    "executionTime": "2019-10-12T05:19:40.373Z"
   },
   {
    "cell": {
     "executionCount": 17,
     "executionEventId": "6d76d8b2-e3ba-473e-ac1c-46cb966b176c",
     "hasError": false,
     "id": "b6acb14d-c636-4b97-ba0f-c750e3362d2f",
     "outputs": [
      {
       "data": {
        "text/plain": "(795517, 4)"
       },
       "execution_count": 17,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "c381cc92-ba84-43f1-95f7-d19b70d547f5",
     "text": "coding_scores.shape"
    },
    "executionTime": "2019-10-12T05:19:43.321Z"
   },
   {
    "cell": {
     "executionCount": 18,
     "executionEventId": "80f39e2c-473e-460f-8876-02f3c44f7ad4",
     "hasError": false,
     "id": "dc3a6fb5-3fdc-4e6b-8290-67727cc314b2",
     "outputs": [
      {
       "data": {
        "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>read_id</th>\n      <th>jaccard_in_peptide_db</th>\n      <th>n_kmers</th>\n      <th>classification</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>SRR306838.1531 Ibis_Run100924_C3PO:6:1:15718:1062/1</td>\n      <td>-1.0</td>\n      <td>2</td>\n      <td>low complexity</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>SRR306838.1617 Ibis_Run100924_C3PO:6:1:9036:1074/1</td>\n      <td>0.0</td>\n      <td>0</td>\n      <td>non-coding</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>SRR306838.1672 Ibis_Run100924_C3PO:6:1:15751:1077/1</td>\n      <td>-1.0</td>\n      <td>1</td>\n      <td>low complexity</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>SRR306838.1938 Ibis_Run100924_C3PO:6:1:15775:1103/1</td>\n      <td>-1.0</td>\n      <td>1</td>\n      <td>low complexity</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>SRR306838.2318 Ibis_Run100924_C3PO:6:1:15779:1141/1</td>\n      <td>-1.0</td>\n      <td>7</td>\n      <td>non-coding</td>\n    </tr>\n  </tbody>\n</table>\n</div>",
        "text/plain": "                                               read_id  jaccard_in_peptide_db  \\\n0  SRR306838.1531 Ibis_Run100924_C3PO:6:1:15718:1062/1                   -1.0   \n1   SRR306838.1617 Ibis_Run100924_C3PO:6:1:9036:1074/1                    0.0   \n2  SRR306838.1672 Ibis_Run100924_C3PO:6:1:15751:1077/1                   -1.0   \n3  SRR306838.1938 Ibis_Run100924_C3PO:6:1:15775:1103/1                   -1.0   \n4  SRR306838.2318 Ibis_Run100924_C3PO:6:1:15779:1141/1                   -1.0   \n\n   n_kmers  classification  \n0        2  low complexity  \n1        0      non-coding  \n2        1  low complexity  \n3        1  low complexity  \n4        7      non-coding  "
       },
       "execution_count": 18,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "b44eec0d-17d9-459a-8fac-49e78b25a50c",
     "text": "coding_scores.head()"
    },
    "executionTime": "2019-10-12T05:19:51.262Z"
   },
   {
    "cell": {
     "executionCount": 19,
     "executionEventId": "e7718148-6aa4-4d7e-8ae3-c5d971196c73",
     "hasError": false,
     "id": "55bff9df-c043-4f6b-8af4-37aa528edb16",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "total 362M\ndrwxrwxr-x 2 olga olga 4.0K Oct 11 21:11 .\ndrwxrwxr-x 3 olga olga 4.0K Sep 10 15:07 ..\n-rw-rw-r-- 1 olga olga  52M Oct 11 18:07 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.bam\n-rw-rw-r-- 1 olga olga 2.9M Oct 11 18:08 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.bam.bai\n-rw-rw-r-- 1 olga olga 1.9M Oct 11 21:11 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.CDS_without_stop_codon.read_ids.txt\n-rw-rw-r-- 1 olga olga 7.0M Oct 11 20:58 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.out_CDS_without_stop_codon.bam\n-rw-rw-r-- 1 olga olga 6.1M Oct 11 18:08 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.out_CDS_without_stop_codon_or_utr.bam\n-rw-rw-r-- 1 olga olga 1.8M Oct 11 18:08 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.out_CDS_without_stop_codon_or_utr.bam.bai\n-rw-rw-r-- 1 olga olga 1.7M Oct 11 18:08 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.out_CDS_without_stop_codon_or_utr.read_ids.txt\n-rw-rw-r-- 1 olga olga  15M Oct 11 18:08 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.read_ids.txt\n-rw-rw-r-- 1 olga olga  15M Sep 10 15:55 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out_CDS_without_stop_codon.bam\n-rw-rw-r-- 1 olga olga 1.7M Oct 11 18:08 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out_CDS_without_stop_codon.bam.bai\n-rw-rw-r-- 1 olga olga  15M Sep 18 07:15 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out_CDS_without_stop_codon_or_utr.bam\n-rw-rw-r-- 1 olga olga 1.7M Oct 11 18:08 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out_CDS_without_stop_codon_or_utr.bam.bai\n-rw-rw-r-- 1 olga olga 2.8M Sep 18 10:14 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out_CDS_without_stop_codon_or_utr.read_ids.txt\n-rw-rw-r-- 1 olga olga 4.1M Sep 10 15:56 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out_CDS_without_stop_codon.read_ids.txt\n-rw-r--r-- 1 olga olga 235M Sep 10 15:10 SRR306838_GSM752691_hsa_br_F_1_trimmed_k15.tsv\n(91741,)\n"
      },
      {
       "data": {
        "text/plain": "0    SRR306838.13598607\n1    SRR306838.23303318\n2    SRR306838.19917713\n3    SRR306838.19937719\n4    SRR306838.20862490\nName: 0, dtype: object"
       },
       "execution_count": 19,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "75c01782-427c-4c63-bc29-ed308ae4a142",
     "text": "folder = '/home/olga/data_sm/kmer-hashing/brawand2011/human/'\n! ls -lha $folder\n\ncoding_read_ids = pd.read_csv(f\"{folder}/SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.out_CDS_without_stop_codon_or_utr.read_ids.txt\", \n                                   header=None, squeeze=True)\n\n# Try just CDS minus stop codons .... but classification goes down from 92% to 90%\n# coding_read_ids = pd.read_csv(f\"{folder}/SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.CDS_without_stop_codon.read_ids.txt\", \n#                                    header=None, squeeze=True)\nprint(coding_read_ids.shape)\ncoding_read_ids.head()"
    },
    "executionTime": "2019-10-12T05:19:53.106Z"
   },
   {
    "cell": {
     "executionCount": 20,
     "executionEventId": "7739dc9f-d046-4cca-a0e4-acf6db9cbdd4",
     "hasError": false,
     "id": "db871d31-0058-412c-8354-87b029f75f87",
     "outputs": [
      {
       "data": {
        "text/plain": "90307"
       },
       "execution_count": 20,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "75c01782-427c-4c63-bc29-ed308ae4a142",
     "text": "true_coding_read_ids = set(coding_read_ids.values)\nlen(true_coding_read_ids)"
    },
    "executionTime": "2019-10-12T05:19:53.141Z"
   },
   {
    "cell": {
     "executionCount": 21,
     "executionEventId": "ef2cc9f3-d4ca-46ee-9486-efd10b726f78",
     "hasError": false,
     "id": "a58b463e-1564-4862-9560-26bfac03f506",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "(807698,)\n"
      }
     ],
     "persistentId": "317bb484-c93f-4e1c-b831-50734c9e6614",
     "text": "unspliced_read_ids_series = pd.read_csv(f\"{folder}/SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.read_ids.txt\", \n                                   header=None, squeeze=True)\nprint(unspliced_read_ids_series.shape)\nunspliced_read_ids_series.head()\n\nunspliced_reads = set(unspliced_read_ids_series.values)"
    },
    "executionTime": "2019-10-12T05:19:55.563Z"
   },
   {
    "cell": {
     "executionCount": 22,
     "executionEventId": "58a35222-6d3d-4619-9f5b-e7fca524f52b",
     "hasError": true,
     "id": "ee39825a-fed4-4b9b-9278-e5d0efadd7b5",
     "outputs": [
      {
       "ename": "SettingWithCopyWarning",
       "evalue": "\nA value is trying to be set on a copy of a slice from a DataFrame.\nTry using .loc[row_indexer,col_indexer] = value instead\n\nSee the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mSettingWithCopyWarning\u001b[0m                    Traceback (most recent call last)",
        "\u001b[0;32m<ipython-input-22-05a0cae6293e>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mcoding_scores_not_low_complexity\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'short_read_id'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcoding_scores_not_low_complexity\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'read_id'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msplit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      2\u001b[0m \u001b[0mcoding_scores_not_low_complexity_unspliced\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcoding_scores_not_low_complexity\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mquery\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'short_read_id in @unspliced_reads'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      3\u001b[0m \u001b[0mcoding_scores_not_low_complexity_unspliced\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'in_cds_no_stop_codon'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcoding_scores_not_low_complexity_unspliced\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshort_read_id\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0misin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtrue_coding_read_ids\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcoding_scores_not_low_complexity_unspliced\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshape\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m \u001b[0mcoding_scores_not_low_complexity_unspliced\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mhead\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m__setitem__\u001b[0;34m(self, key, value)\u001b[0m\n\u001b[1;32m   3368\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3369\u001b[0m             \u001b[0;31m# set column\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3370\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_set_item\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3371\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3372\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_setitem_slice\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkey\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m_set_item\u001b[0;34m(self, key, value)\u001b[0m\n\u001b[1;32m   3450\u001b[0m         \u001b[0;31m# value exception to occur first\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3451\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3452\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_check_setitem_copy\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3453\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3454\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0minsert\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mloc\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcolumn\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mallow_duplicates\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/generic.py\u001b[0m in \u001b[0;36m_check_setitem_copy\u001b[0;34m(self, stacklevel, t, force)\u001b[0m\n\u001b[1;32m   3285\u001b[0m             \u001b[0;32melif\u001b[0m \u001b[0mvalue\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m'warn'\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3286\u001b[0m                 warnings.warn(t, com.SettingWithCopyWarning,\n\u001b[0;32m-> 3287\u001b[0;31m                               stacklevel=stacklevel)\n\u001b[0m\u001b[1;32m   3288\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3289\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m__delitem__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;31mSettingWithCopyWarning\u001b[0m: \nA value is trying to be set on a copy of a slice from a DataFrame.\nTry using .loc[row_indexer,col_indexer] = value instead\n\nSee the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy"
       ]
      }
     ],
     "persistentId": "b321c9fc-54af-40bc-96a9-40c59ea87994",
     "text": "coding_scores_not_low_complexity['short_read_id'] = coding_scores_not_low_complexity['read_id'].str.split().str[0]\ncoding_scores_not_low_complexity_unspliced = coding_scores_not_low_complexity.query('short_read_id in @unspliced_reads')\ncoding_scores_not_low_complexity_unspliced['in_cds_no_stop_codon'] = coding_scores_not_low_complexity_unspliced.short_read_id.isin(true_coding_read_ids)\nprint(coding_scores_not_low_complexity_unspliced.shape)\ncoding_scores_not_low_complexity_unspliced.head()"
    },
    "executionTime": "2019-10-12T05:19:58.029Z"
   },
   {
    "cell": {
     "executionCount": 23,
     "executionEventId": "f680f9ae-ff2b-42cc-acb9-7129b233e188",
     "hasError": true,
     "id": "ee39825a-fed4-4b9b-9278-e5d0efadd7b5",
     "outputs": [
      {
       "ename": "SettingWithCopyWarning",
       "evalue": "\nA value is trying to be set on a copy of a slice from a DataFrame.\nTry using .loc[row_indexer,col_indexer] = value instead\n\nSee the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mSettingWithCopyWarning\u001b[0m                    Traceback (most recent call last)",
        "\u001b[0;32m<ipython-input-23-d0c42e6d7428>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mcoding_scores_not_low_complexity\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mloc\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'short_read_id'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcoding_scores_not_low_complexity\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'read_id'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msplit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      2\u001b[0m \u001b[0mcoding_scores_not_low_complexity_unspliced\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcoding_scores_not_low_complexity\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mquery\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'short_read_id in @unspliced_reads'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      3\u001b[0m \u001b[0mcoding_scores_not_low_complexity_unspliced\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'in_cds_no_stop_codon'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcoding_scores_not_low_complexity_unspliced\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshort_read_id\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0misin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtrue_coding_read_ids\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcoding_scores_not_low_complexity_unspliced\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshape\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m \u001b[0mcoding_scores_not_low_complexity_unspliced\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mhead\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/indexing.py\u001b[0m in \u001b[0;36m__setitem__\u001b[0;34m(self, key, value)\u001b[0m\n\u001b[1;32m    188\u001b[0m             \u001b[0mkey\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcom\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mapply_if_callable\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mobj\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    189\u001b[0m         \u001b[0mindexer\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_get_setitem_indexer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 190\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_setitem_with_indexer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mindexer\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    191\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    192\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_validate_key\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkey\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/indexing.py\u001b[0m in \u001b[0;36m_setitem_with_indexer\u001b[0;34m(self, indexer, value)\u001b[0m\n\u001b[1;32m    603\u001b[0m                 \u001b[0;31m# we have an equal len list/ndarray\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    604\u001b[0m                 \u001b[0;32melif\u001b[0m \u001b[0mcan_do_equal_len\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 605\u001b[0;31m                     \u001b[0msetter\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlabels\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    606\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    607\u001b[0m                 \u001b[0;31m# per label values\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/indexing.py\u001b[0m in \u001b[0;36msetter\u001b[0;34m(item, v)\u001b[0m\n\u001b[1;32m    541\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    542\u001b[0m                 \u001b[0;31m# reset the sliced object if unique\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 543\u001b[0;31m                 \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mobj\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mitem\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0ms\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    544\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    545\u001b[0m             \u001b[0;32mdef\u001b[0m \u001b[0mcan_do_equal_len\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m__setitem__\u001b[0;34m(self, key, value)\u001b[0m\n\u001b[1;32m   3368\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3369\u001b[0m             \u001b[0;31m# set column\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3370\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_set_item\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3371\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3372\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_setitem_slice\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkey\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m_set_item\u001b[0;34m(self, key, value)\u001b[0m\n\u001b[1;32m   3450\u001b[0m         \u001b[0;31m# value exception to occur first\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3451\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3452\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_check_setitem_copy\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3453\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3454\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0minsert\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mloc\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcolumn\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mallow_duplicates\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/generic.py\u001b[0m in \u001b[0;36m_check_setitem_copy\u001b[0;34m(self, stacklevel, t, force)\u001b[0m\n\u001b[1;32m   3285\u001b[0m             \u001b[0;32melif\u001b[0m \u001b[0mvalue\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m'warn'\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3286\u001b[0m                 warnings.warn(t, com.SettingWithCopyWarning,\n\u001b[0;32m-> 3287\u001b[0;31m                               stacklevel=stacklevel)\n\u001b[0m\u001b[1;32m   3288\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3289\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m__delitem__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;31mSettingWithCopyWarning\u001b[0m: \nA value is trying to be set on a copy of a slice from a DataFrame.\nTry using .loc[row_indexer,col_indexer] = value instead\n\nSee the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy"
       ]
      }
     ],
     "persistentId": "b321c9fc-54af-40bc-96a9-40c59ea87994",
     "text": "coding_scores_not_low_complexity.loc[:, 'short_read_id'] = coding_scores_not_low_complexity['read_id'].str.split().str[0]\ncoding_scores_not_low_complexity_unspliced = coding_scores_not_low_complexity.query('short_read_id in @unspliced_reads')\ncoding_scores_not_low_complexity_unspliced['in_cds_no_stop_codon'] = coding_scores_not_low_complexity_unspliced.short_read_id.isin(true_coding_read_ids)\nprint(coding_scores_not_low_complexity_unspliced.shape)\ncoding_scores_not_low_complexity_unspliced.head()"
    },
    "executionTime": "2019-10-12T05:20:45.339Z"
   },
   {
    "cell": {
     "executionCount": 24,
     "executionEventId": "f2785797-37db-4a56-8bbb-03ca20e05432",
     "hasError": false,
     "id": "ed784edf-68ff-42c6-8bf2-8ebb5d463c19",
     "outputs": [
      {
       "data": {
        "text/plain": "1             SRR306838.1617\n7             SRR306838.2723\n9             SRR306838.3707\n10            SRR306838.4251\n11            SRR306838.4936\n12            SRR306838.5359\n13            SRR306838.6299\n14            SRR306838.8107\n15           SRR306838.10022\n17           SRR306838.10673\n18           SRR306838.10841\n19           SRR306838.11357\n24           SRR306838.13390\n25           SRR306838.13837\n26           SRR306838.13968\n28           SRR306838.14545\n31           SRR306838.14924\n34           SRR306838.15166\n35           SRR306838.15290\n36           SRR306838.15435\n37           SRR306838.15721\n39           SRR306838.15853\n41           SRR306838.16067\n42           SRR306838.16073\n43           SRR306838.16099\n46           SRR306838.16735\n47           SRR306838.17165\n48           SRR306838.17812\n49           SRR306838.18429\n50           SRR306838.18431\n                 ...        \n795484    SRR306838.24496707\n795485    SRR306838.24497783\n795486    SRR306838.24498630\n795487    SRR306838.24498923\n795488    SRR306838.24500615\n795489    SRR306838.24500944\n795490    SRR306838.24501224\n795491    SRR306838.24501482\n795492    SRR306838.24501695\n795493    SRR306838.24502139\n795494    SRR306838.24502320\n795495    SRR306838.24502629\n795496    SRR306838.24503808\n795497    SRR306838.24504359\n795498    SRR306838.24504516\n795499    SRR306838.24504807\n795500    SRR306838.24504881\n795501    SRR306838.24505219\n795502    SRR306838.24506841\n795503    SRR306838.24507188\n795504    SRR306838.24507378\n795507    SRR306838.24507766\n795508    SRR306838.24508567\n795509    SRR306838.24508607\n795510    SRR306838.24508944\n795511    SRR306838.24509694\n795512    SRR306838.24509978\n795513    SRR306838.24510192\n795514    SRR306838.24510500\n795515    SRR306838.24510625\nName: read_id, Length: 793969, dtype: object"
       },
       "execution_count": 24,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "49d36e97-c1f4-400a-a81a-a2605e537be5",
     "text": "coding_scores_not_low_complexity['read_id'].str.split().str[0]\n"
    },
    "executionTime": "2019-10-12T05:20:56.701Z"
   },
   {
    "cell": {
     "executionCount": 25,
     "executionEventId": "7c38c696-e4e9-437b-84ba-ec0662fefa57",
     "hasError": true,
     "id": "ed784edf-68ff-42c6-8bf2-8ebb5d463c19",
     "outputs": [
      {
       "ename": "SettingWithCopyWarning",
       "evalue": "\nA value is trying to be set on a copy of a slice from a DataFrame.\nTry using .loc[row_indexer,col_indexer] = value instead\n\nSee the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mSettingWithCopyWarning\u001b[0m                    Traceback (most recent call last)",
        "\u001b[0;32m<ipython-input-25-6b23b8052e02>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mcoding_scores_not_low_complexity\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'short_read_id'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcoding_scores_not_low_complexity\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'read_id'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msplit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m__setitem__\u001b[0;34m(self, key, value)\u001b[0m\n\u001b[1;32m   3368\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3369\u001b[0m             \u001b[0;31m# set column\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3370\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_set_item\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3371\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3372\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_setitem_slice\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkey\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m_set_item\u001b[0;34m(self, key, value)\u001b[0m\n\u001b[1;32m   3450\u001b[0m         \u001b[0;31m# value exception to occur first\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3451\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3452\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_check_setitem_copy\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3453\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3454\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0minsert\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mloc\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcolumn\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mallow_duplicates\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/generic.py\u001b[0m in \u001b[0;36m_check_setitem_copy\u001b[0;34m(self, stacklevel, t, force)\u001b[0m\n\u001b[1;32m   3285\u001b[0m             \u001b[0;32melif\u001b[0m \u001b[0mvalue\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m'warn'\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3286\u001b[0m                 warnings.warn(t, com.SettingWithCopyWarning,\n\u001b[0;32m-> 3287\u001b[0;31m                               stacklevel=stacklevel)\n\u001b[0m\u001b[1;32m   3288\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3289\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m__delitem__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;31mSettingWithCopyWarning\u001b[0m: \nA value is trying to be set on a copy of a slice from a DataFrame.\nTry using .loc[row_indexer,col_indexer] = value instead\n\nSee the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy"
       ]
      }
     ],
     "persistentId": "49d36e97-c1f4-400a-a81a-a2605e537be5",
     "text": "coding_scores_not_low_complexity['short_read_id'] = coding_scores_not_low_complexity['read_id'].str.split().str[0]\n"
    },
    "executionTime": "2019-10-12T05:26:44.430Z"
   },
   {
    "cell": {
     "executionCount": 26,
     "executionEventId": "fb9ffc2f-2b0a-4928-a47e-3fb3afa194c6",
     "hasError": false,
     "id": "57b12f67-735b-46fb-b9cc-507e630145c3",
     "outputs": [
      {
       "data": {
        "text/plain": "(793969,)"
       },
       "execution_count": 26,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "ef56b800-879e-4fbf-bbf0-e35d64e50fea",
     "text": "coding_scores_not_low_complexity['read_id'].str.split().str[0].shape"
    },
    "executionTime": "2019-10-12T05:26:56.208Z"
   },
   {
    "cell": {
     "executionCount": 27,
     "executionEventId": "258907cd-2e07-40d1-b2ff-d66de3e6faaa",
     "hasError": false,
     "id": "a7ce9f14-6f4c-4a7e-bf8a-737f6f08187a",
     "outputs": [
      {
       "data": {
        "text/plain": "(793969, 5)"
       },
       "execution_count": 27,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "4e286241-67b7-4cf6-b40a-0e841e2e4649",
     "text": "coding_scores_not_low_complexity.shape"
    },
    "executionTime": "2019-10-12T05:26:59.565Z"
   },
   {
    "cell": {
     "executionCount": 28,
     "executionEventId": "f515d0f4-fed4-4d93-8be0-2e4aab81fc29",
     "hasError": true,
     "id": "ed784edf-68ff-42c6-8bf2-8ebb5d463c19",
     "outputs": [
      {
       "name": "stderr",
       "output_type": "stream",
       "text": "[autoreload of khtools.partition_reads failed: Traceback (most recent call last):\n  File \"/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/IPython/extensions/autoreload.py\", line 245, in check\n    superreload(m, reload, self.old_objects)\n  File \"/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/IPython/extensions/autoreload.py\", line 368, in superreload\n    module = reload(module)\n  File \"/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/imp.py\", line 315, in reload\n    return importlib.reload(module)\n  File \"/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/importlib/__init__.py\", line 166, in reload\n    _bootstrap._exec(spec, module)\n  File \"<frozen importlib._bootstrap>\", line 618, in _exec\n  File \"<frozen importlib._bootstrap_external>\", line 678, in exec_module\n  File \"<frozen importlib._bootstrap>\", line 219, in _call_with_frames_removed\n  File \"/home/olga/code/kmer-hashing/kh-tools/khtools/partition_reads.py\", line 269, in <module>\n    @click.options('--csv', default=False,\nAttributeError: module 'click' has no attribute 'options'\n]\n"
      },
      {
       "ename": "SettingWithCopyWarning",
       "evalue": "\nA value is trying to be set on a copy of a slice from a DataFrame.\nTry using .loc[row_indexer,col_indexer] = value instead\n\nSee the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mSettingWithCopyWarning\u001b[0m                    Traceback (most recent call last)",
        "\u001b[0;32m<ipython-input-28-33d2a67a7173>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mcoding_scores_not_low_complexity\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'short_read_id'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcoding_scores_not_low_complexity\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'read_id'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msplit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mvalues\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m__setitem__\u001b[0;34m(self, key, value)\u001b[0m\n\u001b[1;32m   3368\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3369\u001b[0m             \u001b[0;31m# set column\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3370\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_set_item\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3371\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3372\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_setitem_slice\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkey\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m_set_item\u001b[0;34m(self, key, value)\u001b[0m\n\u001b[1;32m   3450\u001b[0m         \u001b[0;31m# value exception to occur first\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3451\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3452\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_check_setitem_copy\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3453\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3454\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0minsert\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mloc\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcolumn\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mallow_duplicates\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/generic.py\u001b[0m in \u001b[0;36m_check_setitem_copy\u001b[0;34m(self, stacklevel, t, force)\u001b[0m\n\u001b[1;32m   3285\u001b[0m             \u001b[0;32melif\u001b[0m \u001b[0mvalue\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m'warn'\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3286\u001b[0m                 warnings.warn(t, com.SettingWithCopyWarning,\n\u001b[0;32m-> 3287\u001b[0;31m                               stacklevel=stacklevel)\n\u001b[0m\u001b[1;32m   3288\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3289\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m__delitem__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;31mSettingWithCopyWarning\u001b[0m: \nA value is trying to be set on a copy of a slice from a DataFrame.\nTry using .loc[row_indexer,col_indexer] = value instead\n\nSee the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy"
       ]
      }
     ],
     "persistentId": "49d36e97-c1f4-400a-a81a-a2605e537be5",
     "text": "coding_scores_not_low_complexity['short_read_id'] = coding_scores_not_low_complexity['read_id'].str.split().str[0].values\n"
    },
    "executionTime": "2019-10-12T05:54:27.191Z"
   },
   {
    "cell": {
     "executionCount": 30,
     "executionEventId": "b1579715-2a16-413c-936d-1e755a8fc23a",
     "hasError": true,
     "id": "8456fb98-7cab-4692-96df-85549ff65435",
     "outputs": [
      {
       "ename": "SyntaxError",
       "evalue": "invalid syntax (<ipython-input-30-dcc92b39cc41>, line 1)",
       "output_type": "error",
       "traceback": [
        "\u001b[0;36m  File \u001b[0;32m\"<ipython-input-30-dcc92b39cc41>\"\u001b[0;36m, line \u001b[0;32m1\u001b[0m\n\u001b[0;31m    peptide_graph.\u001b[0m\n\u001b[0m                  ^\u001b[0m\n\u001b[0;31mSyntaxError\u001b[0m\u001b[0;31m:\u001b[0m invalid syntax\n"
       ]
      }
     ],
     "persistentId": "d28dc029-70e7-4a3f-9455-6ed6882da2c9",
     "text": "peptide_graph."
    },
    "executionTime": "2019-10-13T23:28:22.180Z"
   },
   {
    "cell": {
     "executionCount": 31,
     "executionEventId": "1a88a426-0f9b-4529-b962-92a2b16d8ea0",
     "hasError": false,
     "id": "8456fb98-7cab-4692-96df-85549ff65435",
     "outputs": [
      {
       "data": {
        "text/plain": "<khmer._oxli.graphs.Nodegraph at 0x7f76dd345930>"
       },
       "execution_count": 31,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "d28dc029-70e7-4a3f-9455-6ed6882da2c9",
     "text": "peptide_graph"
    },
    "executionTime": "2019-10-13T23:28:24.082Z"
   },
   {
    "cell": {
     "executionCount": 32,
     "executionEventId": "7bcf71eb-6a45-460c-99a5-e7c46b8a03d0",
     "hasError": false,
     "id": "8456fb98-7cab-4692-96df-85549ff65435",
     "outputs": [
      {
       "data": {
        "text/plain": "['__class__',\n '__delattr__',\n '__dir__',\n '__doc__',\n '__eq__',\n '__format__',\n '__ge__',\n '__getattribute__',\n '__gt__',\n '__hash__',\n '__init__',\n '__init_subclass__',\n '__le__',\n '__lt__',\n '__ne__',\n '__new__',\n '__pyx_vtable__',\n '__reduce__',\n '__reduce_ex__',\n '__repr__',\n '__setattr__',\n '__setstate__',\n '__sizeof__',\n '__str__',\n '__subclasshook__',\n '_kmer_type_error',\n '_validate_partitionmap',\n 'abundance_distribution',\n 'add',\n 'add_stop_tag',\n 'add_tag',\n 'assign_partition_id',\n 'calc_connected_graph_size',\n 'consume',\n 'consume_and_tag',\n 'consume_partitioned_fasta',\n 'consume_seqfile',\n 'consume_seqfile_and_tag',\n 'consume_seqfile_banding',\n 'consume_seqfile_banding_with_mask',\n 'consume_seqfile_with_mask',\n 'count',\n 'count_kmers_within_radius',\n 'count_partitions',\n 'divide_tags_into_subsets',\n 'do_subset_partition',\n 'extract_unique_paths',\n 'find_all_tags',\n 'find_all_tags_list',\n 'find_high_degree_nodes',\n 'find_spectral_error_positions',\n 'get',\n 'get_kmer_counts',\n 'get_kmer_hashes',\n 'get_kmer_hashes_as_hashset',\n 'get_kmers',\n 'get_max_count',\n 'get_median_count',\n 'get_min_count',\n 'get_partition_id',\n 'get_raw_tables',\n 'get_stop_tags',\n 'get_tags_and_positions',\n 'get_tags_for_sequence',\n 'get_tagset',\n 'get_use_bigcount',\n 'hash',\n 'hashsizes',\n 'iter_stop_tags',\n 'join_partitions',\n 'kmer_degree',\n 'ksize',\n 'load',\n 'load_partitionmap',\n 'load_stop_tags',\n 'load_tagset',\n 'median_at_least',\n 'merge_subset',\n 'merge_subset_from_disk',\n 'n_occupied',\n 'n_tables',\n 'n_tags',\n 'n_unique_kmers',\n 'neighbors',\n 'output_partitions',\n 'partition',\n 'print_stop_tags',\n 'print_tagset',\n 'repartition_largest_partition',\n 'reverse_hash',\n 'sanitize_kmer',\n 'sanitize_seq_kmer',\n 'save',\n 'save_partitionmap',\n 'save_stop_tags',\n 'save_tagset',\n 'set_partition_id',\n 'set_use_bigcount',\n 'tag_density',\n 'tags',\n 'traverse_linear_path',\n 'trim_below_abundance',\n 'trim_on_abundance',\n 'trim_on_stoptags',\n 'update']"
       },
       "execution_count": 32,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "d28dc029-70e7-4a3f-9455-6ed6882da2c9",
     "text": "dir(peptide_graph)"
    },
    "executionTime": "2019-10-13T23:28:27.971Z"
   },
   {
    "cell": {
     "executionCount": 33,
     "executionEventId": "699bc314-d0ab-4adb-b4a6-febaf1b6edb5",
     "hasError": false,
     "id": "197acbd3-f63f-4e82-92f8-49f1719f128f",
     "outputs": [
      {
       "data": {
        "text/plain": "<function Nodegraph.tags>"
       },
       "execution_count": 33,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "b32b1264-231a-4b32-9229-c49de4f17595",
     "text": "peptide_graph.tags"
    },
    "executionTime": "2019-10-13T23:29:02.977Z"
   },
   {
    "cell": {
     "executionCount": 34,
     "executionEventId": "a5bad843-8b47-4422-a06a-7527c0161f4c",
     "hasError": false,
     "id": "197acbd3-f63f-4e82-92f8-49f1719f128f",
     "outputs": [
      {
       "data": {
        "text/plain": "<generator at 0x7f76dd1c6168>"
       },
       "execution_count": 34,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "b32b1264-231a-4b32-9229-c49de4f17595",
     "text": "peptide_graph.tags()"
    },
    "executionTime": "2019-10-13T23:29:05.309Z"
   },
   {
    "cell": {
     "executionCount": 35,
     "executionEventId": "d88e4858-808c-411f-9a53-6d6d38623da7",
     "hasError": false,
     "id": "197acbd3-f63f-4e82-92f8-49f1719f128f",
     "outputs": [
      {
       "data": {
        "text/plain": "[]"
       },
       "execution_count": 35,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "b32b1264-231a-4b32-9229-c49de4f17595",
     "text": "list(peptide_graph.tags())"
    },
    "executionTime": "2019-10-13T23:29:08.059Z"
   },
   {
    "cell": {
     "executionCount": 36,
     "executionEventId": "b413f210-fa99-425d-8c46-ae3b19d6a707",
     "hasError": false,
     "id": "32eb0977-d285-4fbc-ac7b-36417b01f17c",
     "outputs": [
      {
       "data": {
        "text/plain": "\u001b[0;31mDocstring:\u001b[0m\nHashgraph.add_tag(self, kmer)\nAdd a k-mer to the tagset.\n\u001b[0;31mType:\u001b[0m      builtin_function_or_method\n"
       },
       "metadata": {},
       "output_type": "display_data"
      }
     ],
     "persistentId": "5e420857-1689-4657-983e-7e72891cf0ef",
     "text": "peptide_graph.add_tag?"
    },
    "executionTime": "2019-10-13T23:29:14.754Z"
   },
   {
    "cell": {
     "executionCount": 37,
     "executionEventId": "8ea3869e-4c89-4dda-956a-ea7b61fff588",
     "hasError": false,
     "id": "4184e698-9412-4b17-9ba9-0759603d891c",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "total 250109537\n-rw-rw-r-- 1 olga    21628188 Sep 14 06:46 \u001b[0m\u001b[01;31mHomo_sapiens.GRCh38.cds.all.fa.gz\u001b[0m\n-rw-rw-r-- 1 olga    16335856 Sep 14 06:46 \u001b[01;31mHomo_sapiens.GRCh38.ncrna.fa.gz\u001b[0m\n-rw-rw-r-- 1 olga    14155330 Sep 14 06:46 \u001b[01;31mHomo_sapiens.GRCh38.pep.all.fa.gz\u001b[0m\n-rw-rw-r-- 1 olga 16000000036 Sep 14 07:11 human_cds_k15.nodegraph\n-rw-rw-r-- 1 olga          65 Sep 14 07:11 human_cds_k15.nodegraph.info\n-rw-rw-r-- 1 olga 16000000036 Sep 14 07:22 human_cds_k17.nodegraph\n-rw-rw-r-- 1 olga          65 Sep 14 07:22 human_cds_k17.nodegraph.info\n-rw-rw-r-- 1 olga 16000000036 Sep 14 07:30 human_cds_k19.nodegraph\n-rw-rw-r-- 1 olga          65 Sep 14 07:30 human_cds_k19.nodegraph.info\n-rw-rw-r-- 1 olga 16000000036 Sep 14 07:39 human_cds_k21.nodegraph\n-rw-rw-r-- 1 olga          65 Sep 14 07:39 human_cds_k21.nodegraph.info\n-rw-rw-r-- 1 olga 16000000036 Sep 14 07:47 human_cds_k23.nodegraph\n-rw-rw-r-- 1 olga          65 Sep 14 07:47 human_cds_k23.nodegraph.info\n-rw-rw-r-- 1 olga 16000000036 Sep 14 07:55 human_cds_k25.nodegraph\n-rw-rw-r-- 1 olga          65 Sep 14 07:55 human_cds_k25.nodegraph.info\n-rw-rw-r-- 1 olga 16000000036 Sep 14 08:03 human_cds_k27.nodegraph\n-rw-rw-r-- 1 olga          65 Sep 14 08:03 human_cds_k27.nodegraph.info\n-rw-rw-r-- 1 olga 16000000036 Sep 14 08:12 human_cds_k31.nodegraph\n-rw-rw-r-- 1 olga          65 Sep 14 08:12 human_cds_k31.nodegraph.info\n-rw-rw-r-- 1 olga 16000000036 Sep 14 07:18 human_ncrna_k15.nodegraph\n-rw-rw-r-- 1 olga          65 Sep 14 07:18 human_ncrna_k15.nodegraph.info\n-rw-rw-r-- 1 olga     5962862 Sep 14 07:15 human_ncrna_k15.nodegraph.tagset\n-rw-rw-r-- 1 olga 16000000036 Sep 14 07:26 human_ncrna_k17.nodegraph\n-rw-rw-r-- 1 olga          65 Sep 14 07:26 human_ncrna_k17.nodegraph.info\n-rw-rw-r-- 1 olga 16000000036 Sep 14 07:34 human_ncrna_k19.nodegraph\n-rw-rw-r-- 1 olga          65 Sep 14 07:34 human_ncrna_k19.nodegraph.info\n-rw-rw-r-- 1 olga 16000000036 Sep 14 07:42 human_ncrna_k21.nodegraph\n-rw-rw-r-- 1 olga          65 Sep 14 07:43 human_ncrna_k21.nodegraph.info\n-rw-rw-r-- 1 olga 16000000036 Sep 14 07:51 human_ncrna_k23.nodegraph\n-rw-rw-r-- 1 olga          65 Sep 14 07:51 human_ncrna_k23.nodegraph.info\n-rw-rw-r-- 1 olga 16000000036 Sep 14 07:59 human_ncrna_k25.nodegraph\n-rw-rw-r-- 1 olga          65 Sep 14 07:59 human_ncrna_k25.nodegraph.info\n-rw-rw-r-- 1 olga 16000000036 Sep 14 08:07 human_ncrna_k27.nodegraph\n-rw-rw-r-- 1 olga          65 Sep 14 08:07 human_ncrna_k27.nodegraph.info\n-rw-rw-r-- 1 olga 16000000036 Sep 14 08:16 human_ncrna_k31.nodegraph\n-rw-rw-r-- 1 olga          65 Sep 14 08:16 human_ncrna_k31.nodegraph.info\n-rw-r--r-- 1 olga    53721122 Sep 18 05:33 \u001b[01;31mSRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz\u001b[0m\n-rw-rw-r-- 1 olga       26029 Oct 11 23:03 SRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.coding_peptides.fasta\n-rw-r--r-- 1 olga       73009 Sep 18 05:33 \u001b[01;31mSRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.fq.gz\u001b[0m\n-rw-rw-r-- 1 olga         132 Oct 11 23:03 SRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.low_complexity_nucleotides.fasta\n-rw-rw-r-- 1 olga           0 Oct 11 22:41 SRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.low_complexity_peptides.fasta\n-rw-rw-r-- 1 olga       92415 Oct 11 23:03 SRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.noncoding_nucleotides.fasta\n"
      }
     ],
     "persistentId": "9c72c9dd-4e3a-4661-ab4f-a9b7bf1d822a",
     "text": "ll "
    },
    "executionTime": "2019-10-13T23:29:51.583Z"
   },
   {
    "cell": {
     "executionCount": 38,
     "executionEventId": "ffc45be7-f7e1-4854-b9f4-55b86a6da507",
     "hasError": false,
     "id": "4184e698-9412-4b17-9ba9-0759603d891c",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "total 239G\ndrwxrwxr-x 2 olga olga 4.0K Oct 11 22:41 \u001b[0m\u001b[01;34m.\u001b[0m/\ndrwxrwxr-x 5 olga olga 4.0K Sep 14 08:39 \u001b[01;34m..\u001b[0m/\n-rw-rw-r-- 1 olga olga  21M Sep 14 06:46 \u001b[01;31mHomo_sapiens.GRCh38.cds.all.fa.gz\u001b[0m\n-rw-rw-r-- 1 olga olga  16M Sep 14 06:46 \u001b[01;31mHomo_sapiens.GRCh38.ncrna.fa.gz\u001b[0m\n-rw-rw-r-- 1 olga olga  14M Sep 14 06:46 \u001b[01;31mHomo_sapiens.GRCh38.pep.all.fa.gz\u001b[0m\n-rw-rw-r-- 1 olga olga  15G Sep 14 07:11 human_cds_k15.nodegraph\n-rw-rw-r-- 1 olga olga   65 Sep 14 07:11 human_cds_k15.nodegraph.info\n-rw-rw-r-- 1 olga olga  15G Sep 14 07:22 human_cds_k17.nodegraph\n-rw-rw-r-- 1 olga olga   65 Sep 14 07:22 human_cds_k17.nodegraph.info\n-rw-rw-r-- 1 olga olga  15G Sep 14 07:30 human_cds_k19.nodegraph\n-rw-rw-r-- 1 olga olga   65 Sep 14 07:30 human_cds_k19.nodegraph.info\n-rw-rw-r-- 1 olga olga  15G Sep 14 07:39 human_cds_k21.nodegraph\n-rw-rw-r-- 1 olga olga   65 Sep 14 07:39 human_cds_k21.nodegraph.info\n-rw-rw-r-- 1 olga olga  15G Sep 14 07:47 human_cds_k23.nodegraph\n-rw-rw-r-- 1 olga olga   65 Sep 14 07:47 human_cds_k23.nodegraph.info\n-rw-rw-r-- 1 olga olga  15G Sep 14 07:55 human_cds_k25.nodegraph\n-rw-rw-r-- 1 olga olga   65 Sep 14 07:55 human_cds_k25.nodegraph.info\n-rw-rw-r-- 1 olga olga  15G Sep 14 08:03 human_cds_k27.nodegraph\n-rw-rw-r-- 1 olga olga   65 Sep 14 08:03 human_cds_k27.nodegraph.info\n-rw-rw-r-- 1 olga olga  15G Sep 14 08:12 human_cds_k31.nodegraph\n-rw-rw-r-- 1 olga olga   65 Sep 14 08:12 human_cds_k31.nodegraph.info\n-rw-rw-r-- 1 olga olga  15G Sep 14 07:18 human_ncrna_k15.nodegraph\n-rw-rw-r-- 1 olga olga   65 Sep 14 07:18 human_ncrna_k15.nodegraph.info\n-rw-rw-r-- 1 olga olga 5.7M Sep 14 07:15 human_ncrna_k15.nodegraph.tagset\n-rw-rw-r-- 1 olga olga  15G Sep 14 07:26 human_ncrna_k17.nodegraph\n-rw-rw-r-- 1 olga olga   65 Sep 14 07:26 human_ncrna_k17.nodegraph.info\n-rw-rw-r-- 1 olga olga  15G Sep 14 07:34 human_ncrna_k19.nodegraph\n-rw-rw-r-- 1 olga olga   65 Sep 14 07:34 human_ncrna_k19.nodegraph.info\n-rw-rw-r-- 1 olga olga  15G Sep 14 07:42 human_ncrna_k21.nodegraph\n-rw-rw-r-- 1 olga olga   65 Sep 14 07:43 human_ncrna_k21.nodegraph.info\n-rw-rw-r-- 1 olga olga  15G Sep 14 07:51 human_ncrna_k23.nodegraph\n-rw-rw-r-- 1 olga olga   65 Sep 14 07:51 human_ncrna_k23.nodegraph.info\n-rw-rw-r-- 1 olga olga  15G Sep 14 07:59 human_ncrna_k25.nodegraph\n-rw-rw-r-- 1 olga olga   65 Sep 14 07:59 human_ncrna_k25.nodegraph.info\n-rw-rw-r-- 1 olga olga  15G Sep 14 08:07 human_ncrna_k27.nodegraph\n-rw-rw-r-- 1 olga olga   65 Sep 14 08:07 human_ncrna_k27.nodegraph.info\n-rw-rw-r-- 1 olga olga  15G Sep 14 08:16 human_ncrna_k31.nodegraph\n-rw-rw-r-- 1 olga olga   65 Sep 14 08:16 human_ncrna_k31.nodegraph.info\n-rw-r--r-- 1 olga olga  52M Sep 18 05:33 \u001b[01;31mSRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz\u001b[0m\n-rw-rw-r-- 1 olga olga  26K Oct 11 23:03 SRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.coding_peptides.fasta\n-rw-r--r-- 1 olga olga  72K Sep 18 05:33 \u001b[01;31mSRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.fq.gz\u001b[0m\n-rw-rw-r-- 1 olga olga  132 Oct 11 23:03 SRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.low_complexity_nucleotides.fasta\n-rw-rw-r-- 1 olga olga    0 Oct 11 22:41 SRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.low_complexity_peptides.fasta\n-rw-rw-r-- 1 olga olga  91K Oct 11 23:03 SRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.noncoding_nucleotides.fasta\n"
      }
     ],
     "persistentId": "9c72c9dd-4e3a-4661-ab4f-a9b7bf1d822a",
     "text": "ls -lha"
    },
    "executionTime": "2019-10-13T23:29:56.447Z"
   },
   {
    "cell": {
     "executionCount": 39,
     "executionEventId": "34709df2-ed33-4e4a-915f-0d419bf72872",
     "hasError": true,
     "id": "440861d3-a5b5-4232-b1fa-e201a5c84a7c",
     "outputs": [
      {
       "ename": "SyntaxError",
       "evalue": "invalid syntax (<ipython-input-39-04e9edd31817>, line 2)",
       "output_type": "error",
       "traceback": [
        "\u001b[0;36m  File \u001b[0;32m\"<ipython-input-39-04e9edd31817>\"\u001b[0;36m, line \u001b[0;32m2\u001b[0m\n\u001b[0;31m    cat human_ncrna_k17.nodegraph.info\u001b[0m\n\u001b[0m                      ^\u001b[0m\n\u001b[0;31mSyntaxError\u001b[0m\u001b[0;31m:\u001b[0m invalid syntax\n"
       ]
      }
     ],
     "persistentId": "39d48808-d467-4d42-a862-97485096088c",
     "text": "\ncat human_ncrna_k17.nodegraph.info"
    },
    "executionTime": "2019-10-13T23:30:00.090Z"
   },
   {
    "cell": {
     "executionCount": 40,
     "executionEventId": "b13abf17-2226-4909-96b6-f97c56508ffd",
     "hasError": true,
     "id": "440861d3-a5b5-4232-b1fa-e201a5c84a7c",
     "outputs": [
      {
       "ename": "SyntaxError",
       "evalue": "invalid syntax (<ipython-input-40-1b20fd05adce>, line 2)",
       "output_type": "error",
       "traceback": [
        "\u001b[0;36m  File \u001b[0;32m\"<ipython-input-40-1b20fd05adce>\"\u001b[0;36m, line \u001b[0;32m2\u001b[0m\n\u001b[0;31m    @ cat human_ncrna_k17.nodegraph.info\u001b[0m\n\u001b[0m                        ^\u001b[0m\n\u001b[0;31mSyntaxError\u001b[0m\u001b[0;31m:\u001b[0m invalid syntax\n"
       ]
      }
     ],
     "persistentId": "39d48808-d467-4d42-a862-97485096088c",
     "text": "\n@ cat human_ncrna_k17.nodegraph.info"
    },
    "executionTime": "2019-10-13T23:30:02.818Z"
   },
   {
    "cell": {
     "executionCount": 41,
     "executionEventId": "a2c02d79-7947-42f7-bff3-b89aea6016fb",
     "hasError": false,
     "id": "440861d3-a5b5-4232-b1fa-e201a5c84a7c",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "30178100 unique k-mers\nfalse positive rate estimated to be 0.000\n"
      }
     ],
     "persistentId": "39d48808-d467-4d42-a862-97485096088c",
     "text": "cat human_ncrna_k17.nodegraph.info"
    },
    "executionTime": "2019-10-13T23:30:06.170Z"
   },
   {
    "cell": {
     "executionCount": 42,
     "executionEventId": "0cc358a4-dd61-4237-9722-2c1d0eeca2b2",
     "hasError": false,
     "id": "a2d7efae-7961-4059-ab6d-ba8338bac514",
     "outputs": [
      {
       "data": {
        "text/plain": "10841698"
       },
       "execution_count": 42,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "f6bb84e5-c5f5-4d4b-90fb-724792d5d78d",
     "text": "peptide_graph.n_unique_kmers()"
    },
    "executionTime": "2019-10-14T00:00:55.763Z"
   },
   {
    "cell": {
     "executionCount": 43,
     "executionEventId": "f78f1fda-4749-4c84-a8a5-43eaef739a6d",
     "hasError": true,
     "id": "e148f523-73b6-4fac-a054-ba32c4add1f9",
     "outputs": [
      {
       "ename": "SettingWithCopyWarning",
       "evalue": "\nA value is trying to be set on a copy of a slice from a DataFrame.\nTry using .loc[row_indexer,col_indexer] = value instead\n\nSee the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mSettingWithCopyWarning\u001b[0m                    Traceback (most recent call last)",
        "\u001b[0;32m<ipython-input-43-75293f21b612>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0mcoding_scores_not_low_complexity_unspliced\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcoding_scores_not_low_complexity\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mquery\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'short_read_id in @unspliced_reads'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m \u001b[0mcoding_scores_not_low_complexity_unspliced\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'in_cds_no_stop_codon'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcoding_scores_not_low_complexity_unspliced\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshort_read_id\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0misin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtrue_coding_read_ids\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      3\u001b[0m \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcoding_scores_not_low_complexity_unspliced\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshape\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[0mcoding_scores_not_low_complexity_unspliced\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mhead\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m__setitem__\u001b[0;34m(self, key, value)\u001b[0m\n\u001b[1;32m   3368\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3369\u001b[0m             \u001b[0;31m# set column\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3370\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_set_item\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3371\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3372\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_setitem_slice\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkey\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m_set_item\u001b[0;34m(self, key, value)\u001b[0m\n\u001b[1;32m   3450\u001b[0m         \u001b[0;31m# value exception to occur first\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3451\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3452\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_check_setitem_copy\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3453\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3454\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0minsert\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mloc\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcolumn\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mallow_duplicates\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/generic.py\u001b[0m in \u001b[0;36m_check_setitem_copy\u001b[0;34m(self, stacklevel, t, force)\u001b[0m\n\u001b[1;32m   3285\u001b[0m             \u001b[0;32melif\u001b[0m \u001b[0mvalue\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m'warn'\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3286\u001b[0m                 warnings.warn(t, com.SettingWithCopyWarning,\n\u001b[0;32m-> 3287\u001b[0;31m                               stacklevel=stacklevel)\n\u001b[0m\u001b[1;32m   3288\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3289\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m__delitem__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;31mSettingWithCopyWarning\u001b[0m: \nA value is trying to be set on a copy of a slice from a DataFrame.\nTry using .loc[row_indexer,col_indexer] = value instead\n\nSee the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy"
       ]
      }
     ],
     "persistentId": "b321c9fc-54af-40bc-96a9-40c59ea87994",
     "text": "coding_scores_not_low_complexity_unspliced = coding_scores_not_low_complexity.query('short_read_id in @unspliced_reads')\ncoding_scores_not_low_complexity_unspliced['in_cds_no_stop_codon'] = coding_scores_not_low_complexity_unspliced.short_read_id.isin(true_coding_read_ids)\nprint(coding_scores_not_low_complexity_unspliced.shape)\ncoding_scores_not_low_complexity_unspliced.head()"
    },
    "executionTime": "2019-10-17T07:33:28.803Z"
   },
   {
    "cell": {
     "executionCount": 1,
     "executionEventId": "c062db36-c8b1-4c6b-8abc-0154a2423d7a",
     "hasError": false,
     "id": "0b3e2eda-6e8f-4e38-b0b2-e7128c9a8734",
     "outputs": [],
     "persistentId": "71299be4-a00f-4638-92a2-ea174ddf208d",
     "text": "%load_ext autoreload\n%autoreload 2\n\nfolder = '/home/olga/data_sm/kmer-hashing/classify_coding_vs_noncoding/'"
    },
    "executionTime": "2019-10-17T07:33:54.361Z"
   },
   {
    "cell": {
     "executionCount": 2,
     "executionEventId": "69499170-c142-47ce-b657-51928aa9ab6d",
     "hasError": false,
     "id": "7086a937-f522-4f1d-b965-f47fa2ae1681",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "/mnt/data_sm/olga/kmer-hashing/classify_coding_vs_noncoding\n"
      }
     ],
     "persistentId": "1d547cf2-254f-4fe2-9bf0-5b7e39be4376",
     "text": "cd $folder"
    },
    "executionTime": "2019-10-17T07:33:54.702Z"
   },
   {
    "cell": {
     "executionCount": 3,
     "executionEventId": "d5f5d31e-86f0-4784-9173-0a728e9cfb8b",
     "hasError": false,
     "id": "de3ef66e-071b-46eb-9afe-ba96402c03c8",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "\u001b[0m\u001b[01;31mHomo_sapiens.GRCh38.cds.all.fa.gz\u001b[0m\n\u001b[01;31mHomo_sapiens.GRCh38.ncrna.fa.gz\u001b[0m\n\u001b[01;31mHomo_sapiens.GRCh38.pep.all.fa.gz\u001b[0m\nHomo_sapiens.GRCh38.pep.subset.fa\n\u001b[01;31mHomo_sapiens.GRCh38.pep.subset.fa.gz\u001b[0m\nhuman_cds_k15.nodegraph\nhuman_cds_k15.nodegraph.info\nhuman_cds_k17.nodegraph\nhuman_cds_k17.nodegraph.info\nhuman_cds_k19.nodegraph\nhuman_cds_k19.nodegraph.info\nhuman_cds_k21.nodegraph\nhuman_cds_k21.nodegraph.info\nhuman_cds_k23.nodegraph\nhuman_cds_k23.nodegraph.info\nhuman_cds_k25.nodegraph\nhuman_cds_k25.nodegraph.info\nhuman_cds_k27.nodegraph\nhuman_cds_k27.nodegraph.info\nhuman_cds_k31.nodegraph\nhuman_cds_k31.nodegraph.info\nhuman_ncrna_k15.nodegraph\nhuman_ncrna_k15.nodegraph.info\nhuman_ncrna_k15.nodegraph.tagset\nhuman_ncrna_k17.nodegraph\nhuman_ncrna_k17.nodegraph.info\nhuman_ncrna_k19.nodegraph\nhuman_ncrna_k19.nodegraph.info\nhuman_ncrna_k21.nodegraph\nhuman_ncrna_k21.nodegraph.info\nhuman_ncrna_k23.nodegraph\nhuman_ncrna_k23.nodegraph.info\nhuman_ncrna_k25.nodegraph\nhuman_ncrna_k25.nodegraph.info\nhuman_ncrna_k27.nodegraph\nhuman_ncrna_k27.nodegraph.info\nhuman_ncrna_k31.nodegraph\nhuman_ncrna_k31.nodegraph.info\n\u001b[01;31mSRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz\u001b[0m\nSRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.coding_peptides.fasta\n\u001b[01;31mSRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.fq.gz\u001b[0m\nSRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.low_complexity_nucleotides.fasta\nSRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.low_complexity_peptides.fasta\nSRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.noncoding_nucleotides.fasta\n"
      }
     ],
     "persistentId": "a30b0ea1-cbfb-4de5-bce5-329cfeb8e50c",
     "text": "ls"
    },
    "executionTime": "2019-10-17T07:33:55.106Z"
   },
   {
    "cell": {
     "executionCount": 4,
     "executionEventId": "08771cb9-218b-4722-be2c-e2b8c0fe62d4",
     "hasError": false,
     "id": "65eb435c-87c1-432b-9aff-484154616041",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "total 239G\ndrwxrwxr-x 2 olga olga 4.0K Oct 13 16:57 \u001b[0m\u001b[01;34m.\u001b[0m/\ndrwxrwxr-x 5 olga olga 4.0K Sep 14 08:39 \u001b[01;34m..\u001b[0m/\n-rw-rw-r-- 1 olga olga  21M Sep 14 06:46 \u001b[01;31mHomo_sapiens.GRCh38.cds.all.fa.gz\u001b[0m\n-rw-rw-r-- 1 olga olga  16M Sep 14 06:46 \u001b[01;31mHomo_sapiens.GRCh38.ncrna.fa.gz\u001b[0m\n-rw-rw-r-- 1 olga olga  14M Sep 14 06:46 \u001b[01;31mHomo_sapiens.GRCh38.pep.all.fa.gz\u001b[0m\n-rw-rw-r-- 1 olga olga 2.6M Oct 13 17:26 Homo_sapiens.GRCh38.pep.subset.fa\n-rw-rw-r-- 1 olga olga 489K Oct 13 17:26 \u001b[01;31mHomo_sapiens.GRCh38.pep.subset.fa.gz\u001b[0m\n-rw-rw-r-- 1 olga olga  15G Sep 14 07:11 human_cds_k15.nodegraph\n-rw-rw-r-- 1 olga olga   65 Sep 14 07:11 human_cds_k15.nodegraph.info\n-rw-rw-r-- 1 olga olga  15G Sep 14 07:22 human_cds_k17.nodegraph\n-rw-rw-r-- 1 olga olga   65 Sep 14 07:22 human_cds_k17.nodegraph.info\n-rw-rw-r-- 1 olga olga  15G Sep 14 07:30 human_cds_k19.nodegraph\n-rw-rw-r-- 1 olga olga   65 Sep 14 07:30 human_cds_k19.nodegraph.info\n-rw-rw-r-- 1 olga olga  15G Sep 14 07:39 human_cds_k21.nodegraph\n-rw-rw-r-- 1 olga olga   65 Sep 14 07:39 human_cds_k21.nodegraph.info\n-rw-rw-r-- 1 olga olga  15G Sep 14 07:47 human_cds_k23.nodegraph\n-rw-rw-r-- 1 olga olga   65 Sep 14 07:47 human_cds_k23.nodegraph.info\n-rw-rw-r-- 1 olga olga  15G Sep 14 07:55 human_cds_k25.nodegraph\n-rw-rw-r-- 1 olga olga   65 Sep 14 07:55 human_cds_k25.nodegraph.info\n-rw-rw-r-- 1 olga olga  15G Sep 14 08:03 human_cds_k27.nodegraph\n-rw-rw-r-- 1 olga olga   65 Sep 14 08:03 human_cds_k27.nodegraph.info\n-rw-rw-r-- 1 olga olga  15G Sep 14 08:12 human_cds_k31.nodegraph\n-rw-rw-r-- 1 olga olga   65 Sep 14 08:12 human_cds_k31.nodegraph.info\n-rw-rw-r-- 1 olga olga  15G Sep 14 07:18 human_ncrna_k15.nodegraph\n-rw-rw-r-- 1 olga olga   65 Sep 14 07:18 human_ncrna_k15.nodegraph.info\n-rw-rw-r-- 1 olga olga 5.7M Sep 14 07:15 human_ncrna_k15.nodegraph.tagset\n-rw-rw-r-- 1 olga olga  15G Sep 14 07:26 human_ncrna_k17.nodegraph\n-rw-rw-r-- 1 olga olga   65 Sep 14 07:26 human_ncrna_k17.nodegraph.info\n-rw-rw-r-- 1 olga olga  15G Sep 14 07:34 human_ncrna_k19.nodegraph\n-rw-rw-r-- 1 olga olga   65 Sep 14 07:34 human_ncrna_k19.nodegraph.info\n-rw-rw-r-- 1 olga olga  15G Sep 14 07:42 human_ncrna_k21.nodegraph\n-rw-rw-r-- 1 olga olga   65 Sep 14 07:43 human_ncrna_k21.nodegraph.info\n-rw-rw-r-- 1 olga olga  15G Sep 14 07:51 human_ncrna_k23.nodegraph\n-rw-rw-r-- 1 olga olga   65 Sep 14 07:51 human_ncrna_k23.nodegraph.info\n-rw-rw-r-- 1 olga olga  15G Sep 14 07:59 human_ncrna_k25.nodegraph\n-rw-rw-r-- 1 olga olga   65 Sep 14 07:59 human_ncrna_k25.nodegraph.info\n-rw-rw-r-- 1 olga olga  15G Sep 14 08:07 human_ncrna_k27.nodegraph\n-rw-rw-r-- 1 olga olga   65 Sep 14 08:07 human_ncrna_k27.nodegraph.info\n-rw-rw-r-- 1 olga olga  15G Sep 14 08:16 human_ncrna_k31.nodegraph\n-rw-rw-r-- 1 olga olga   65 Sep 14 08:16 human_ncrna_k31.nodegraph.info\n-rw-r--r-- 1 olga olga  52M Sep 18 05:33 \u001b[01;31mSRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz\u001b[0m\n-rw-rw-r-- 1 olga olga  26K Oct 11 23:03 SRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.coding_peptides.fasta\n-rw-r--r-- 1 olga olga  72K Sep 18 05:33 \u001b[01;31mSRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.fq.gz\u001b[0m\n-rw-rw-r-- 1 olga olga  132 Oct 11 23:03 SRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.low_complexity_nucleotides.fasta\n-rw-rw-r-- 1 olga olga    0 Oct 11 22:41 SRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.low_complexity_peptides.fasta\n-rw-rw-r-- 1 olga olga  91K Oct 11 23:03 SRR306838_GSM752691_hsa_br_F_1_trimmed_subsampled.noncoding_nucleotides.fasta\n"
      }
     ],
     "persistentId": "9c72c9dd-4e3a-4661-ab4f-a9b7bf1d822a",
     "text": "ls -lha"
    },
    "executionTime": "2019-10-17T07:33:55.302Z"
   },
   {
    "cell": {
     "executionCount": 5,
     "executionEventId": "f37b8d30-6d34-4007-bedd-9324c2d0ee5e",
     "hasError": false,
     "id": "6d994037-bf4f-4a12-8581-3875fbc7861e",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "30178100 unique k-mers\nfalse positive rate estimated to be 0.000\n"
      }
     ],
     "persistentId": "39d48808-d467-4d42-a862-97485096088c",
     "text": "cat human_ncrna_k17.nodegraph.info"
    },
    "executionTime": "2019-10-17T07:33:55.519Z"
   },
   {
    "cell": {
     "executionCount": 6,
     "executionEventId": "2577571b-fe1d-4b01-acbf-927a9a571f47",
     "hasError": false,
     "id": "2521f0be-563b-4a60-82cd-353dce343e0d",
     "outputs": [],
     "persistentId": "56e48bd4-fcc6-4d93-b4e3-0984be1fc1d5",
     "text": "import gzip\nfrom io import StringIO\nfrom pprint import pprint\n\nfrom khmer import Nodegraph\nimport matplotlib.pyplot as plt\nimport screed\nfrom sourmash._minhash import hash_murmur\nfrom khmer.khmer_args import calculate_graphsize\nimport pandas as pd\nfrom sourmash.logging import notify\nimport seaborn as sns\nfrom sklearn import metrics\nfrom tqdm import tqdm\nfrom Bio.Seq import Seq\nfrom Bio import SeqIO\n\n# make all warnings error to get a stack trace\nimport warnings\nwarnings.filterwarnings(\"default\")\n\npd.options.display.max_colwidth = 100\n\nfrom khtools.compare_kmer_content import kmerize, hpize, dayhoffize\n"
    },
    "executionTime": "2019-10-17T07:33:57.093Z"
   },
   {
    "cell": {
     "executionCount": 7,
     "executionEventId": "4ff111ef-8ccd-4ad2-9ebd-07cb565373ab",
     "hasError": true,
     "id": "bd29a121-86d7-4b60-9a1d-20971a6bcf21",
     "outputs": [
      {
       "ename": "ImportError",
       "evalue": "cannot import name 'make_peptide_bloom_filter'",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mImportError\u001b[0m                               Traceback (most recent call last)",
        "\u001b[0;32m<ipython-input-7-4d434ef47aee>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0;32mfrom\u001b[0m \u001b[0mkhtools\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpartition_reads\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0mscore_reads\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmake_peptide_bloom_filter\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msix_frame_translation_no_stops\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mis_low_complexity\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
        "\u001b[0;31mImportError\u001b[0m: cannot import name 'make_peptide_bloom_filter'"
       ]
      }
     ],
     "persistentId": "1e54230e-ceec-42fc-afa6-7b32e219378b",
     "text": "from khtools.partition_reads import score_reads, make_peptide_bloom_filter, six_frame_translation_no_stops, is_low_complexity"
    },
    "executionTime": "2019-10-17T07:33:57.488Z"
   },
   {
    "cell": {
     "executionCount": 8,
     "executionEventId": "592990e9-854b-4160-b45b-a3a4b6c57594",
     "hasError": false,
     "id": "a45589ec-d77f-4e20-b501-ded4a73f542a",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "total 362M\ndrwxrwxr-x 2 olga olga 4.0K Oct 11 21:11 .\ndrwxrwxr-x 3 olga olga 4.0K Sep 10 15:07 ..\n-rw-rw-r-- 1 olga olga  52M Oct 11 18:07 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.bam\n-rw-rw-r-- 1 olga olga 2.9M Oct 11 18:08 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.bam.bai\n-rw-rw-r-- 1 olga olga 1.9M Oct 11 21:11 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.CDS_without_stop_codon.read_ids.txt\n-rw-rw-r-- 1 olga olga 7.0M Oct 11 20:58 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.out_CDS_without_stop_codon.bam\n-rw-rw-r-- 1 olga olga 6.1M Oct 11 18:08 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.out_CDS_without_stop_codon_or_utr.bam\n-rw-rw-r-- 1 olga olga 1.8M Oct 11 18:08 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.out_CDS_without_stop_codon_or_utr.bam.bai\n-rw-rw-r-- 1 olga olga 1.7M Oct 11 18:08 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.out_CDS_without_stop_codon_or_utr.read_ids.txt\n-rw-rw-r-- 1 olga olga  15M Oct 11 18:08 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.read_ids.txt\n-rw-rw-r-- 1 olga olga  15M Sep 10 15:55 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out_CDS_without_stop_codon.bam\n-rw-rw-r-- 1 olga olga 1.7M Oct 11 18:08 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out_CDS_without_stop_codon.bam.bai\n-rw-rw-r-- 1 olga olga  15M Sep 18 07:15 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out_CDS_without_stop_codon_or_utr.bam\n-rw-rw-r-- 1 olga olga 1.7M Oct 11 18:08 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out_CDS_without_stop_codon_or_utr.bam.bai\n-rw-rw-r-- 1 olga olga 2.8M Sep 18 10:14 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out_CDS_without_stop_codon_or_utr.read_ids.txt\n-rw-rw-r-- 1 olga olga 4.1M Sep 10 15:56 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out_CDS_without_stop_codon.read_ids.txt\n-rw-r--r-- 1 olga olga 235M Sep 10 15:10 SRR306838_GSM752691_hsa_br_F_1_trimmed_k15.tsv\n(91741,)\n"
      },
      {
       "data": {
        "text/plain": "0    SRR306838.13598607\n1    SRR306838.23303318\n2    SRR306838.19917713\n3    SRR306838.19937719\n4    SRR306838.20862490\nName: 0, dtype: object"
       },
       "execution_count": 8,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "75c01782-427c-4c63-bc29-ed308ae4a142",
     "text": "folder = '/home/olga/data_sm/kmer-hashing/brawand2011/human/'\n! ls -lha $folder\n\ncoding_read_ids = pd.read_csv(f\"{folder}/SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.out_CDS_without_stop_codon_or_utr.read_ids.txt\", \n                                   header=None, squeeze=True)\n\n# Try just CDS minus stop codons .... but classification goes down from 92% to 90%\n# coding_read_ids = pd.read_csv(f\"{folder}/SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.CDS_without_stop_codon.read_ids.txt\", \n#                                    header=None, squeeze=True)\nprint(coding_read_ids.shape)\ncoding_read_ids.head()"
    },
    "executionTime": "2019-10-17T07:33:57.951Z"
   },
   {
    "cell": {
     "executionCount": 9,
     "executionEventId": "51fe2f88-f2a5-4ac9-bafd-dd179166139d",
     "hasError": false,
     "id": "6e060ff0-135a-4e24-90f0-9dec04bbdc00",
     "outputs": [
      {
       "data": {
        "text/plain": "90307"
       },
       "execution_count": 9,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "75c01782-427c-4c63-bc29-ed308ae4a142",
     "text": "true_coding_read_ids = set(coding_read_ids.values)\nlen(true_coding_read_ids)"
    },
    "executionTime": "2019-10-17T07:33:58.002Z"
   },
   {
    "cell": {
     "executionCount": 10,
     "executionEventId": "fadfea89-db83-4b3c-ad43-a98b5b99bc3b",
     "hasError": false,
     "id": "d3b53b1d-4ded-4afa-a3d2-bce833550ed3",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "(807698,)\n"
      }
     ],
     "persistentId": "317bb484-c93f-4e1c-b831-50734c9e6614",
     "text": "unspliced_read_ids_series = pd.read_csv(f\"{folder}/SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.read_ids.txt\", \n                                   header=None, squeeze=True)\nprint(unspliced_read_ids_series.shape)\nunspliced_read_ids_series.head()\n\nunspliced_reads = set(unspliced_read_ids_series.values)"
    },
    "executionTime": "2019-10-17T07:33:58.859Z"
   },
   {
    "cell": {
     "executionCount": 11,
     "executionEventId": "846b8b14-9f28-46f8-a1ca-48535083ba07",
     "hasError": true,
     "id": "c1e8b6ff-8105-4c1d-a146-1267973d4cf1",
     "outputs": [
      {
       "ename": "NameError",
       "evalue": "name 'coding_scores_not_low_complexity' is not defined",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
        "\u001b[0;32m<ipython-input-11-09d3cccfac48>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mcoding_scores_not_low_complexity\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'read_id'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msplit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshape\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
        "\u001b[0;31mNameError\u001b[0m: name 'coding_scores_not_low_complexity' is not defined"
       ]
      }
     ],
     "persistentId": "ef56b800-879e-4fbf-bbf0-e35d64e50fea",
     "text": "coding_scores_not_low_complexity['read_id'].str.split().str[0].shape"
    },
    "executionTime": "2019-10-17T07:33:58.899Z"
   },
   {
    "cell": {
     "executionCount": 12,
     "executionEventId": "6f74b6f5-8509-4cc5-89fc-cffb79959ffa",
     "hasError": true,
     "id": "201f061a-50d9-4b35-aac8-1c6d936c1eb7",
     "outputs": [
      {
       "ename": "NameError",
       "evalue": "name 'coding_scores_not_low_complexity' is not defined",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
        "\u001b[0;32m<ipython-input-12-a95d9739adcc>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mcoding_scores_not_low_complexity\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshape\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
        "\u001b[0;31mNameError\u001b[0m: name 'coding_scores_not_low_complexity' is not defined"
       ]
      }
     ],
     "persistentId": "4e286241-67b7-4cf6-b40a-0e841e2e4649",
     "text": "coding_scores_not_low_complexity.shape"
    },
    "executionTime": "2019-10-17T07:33:59.241Z"
   },
   {
    "cell": {
     "executionCount": 13,
     "executionEventId": "c2ac3da3-b3cd-4789-9f93-ea5690705833",
     "hasError": true,
     "id": "0ba3a826-0490-4f3b-bea5-5e202ae298f8",
     "outputs": [
      {
       "ename": "NameError",
       "evalue": "name 'coding_scores_not_low_complexity' is not defined",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
        "\u001b[0;32m<ipython-input-13-1a21d10bdb6d>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mcoding_scores_not_low_complexity\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'short_read_id'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mlist\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcoding_scores_not_low_complexity\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'read_id'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msplit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mvalues\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
        "\u001b[0;31mNameError\u001b[0m: name 'coding_scores_not_low_complexity' is not defined"
       ]
      }
     ],
     "persistentId": "49d36e97-c1f4-400a-a81a-a2605e537be5",
     "text": "coding_scores_not_low_complexity['short_read_id'] = list(coding_scores_not_low_complexity['read_id'].str.split().str[0].values)\n"
    },
    "executionTime": "2019-10-17T07:34:26.403Z"
   },
   {
    "cell": {
     "executionCount": 14,
     "executionEventId": "8f98df01-1972-494c-ab99-75496f388391",
     "hasError": true,
     "id": "70d37b43-2fe3-4f70-9d3f-12a0aca87052",
     "outputs": [
      {
       "ename": "NameError",
       "evalue": "name 'make_peptide_bloom_filter' is not defined",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
        "\u001b[0;32m<timed exec>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n",
        "\u001b[0;31mNameError\u001b[0m: name 'make_peptide_bloom_filter' is not defined"
       ]
      }
     ],
     "persistentId": "be06ded1-d7d4-4f21-9961-3aed5ff30cbf",
     "text": "%%time\npeptide_ksize = 7\n\npeptide_graph = make_peptide_bloom_filter(\"Homo_sapiens.GRCh38.pep.all.fa.gz\", peptide_ksize)"
    },
    "executionTime": "2019-10-17T07:34:34.894Z"
   },
   {
    "cell": {
     "executionCount": 15,
     "executionEventId": "d9e10ac8-a9c1-44d1-9d0f-b6b3dc1e1546",
     "hasError": true,
     "id": "547297c6-4cc5-4cc0-b614-3c3bdcbd64b8",
     "outputs": [
      {
       "ename": "NameError",
       "evalue": "name 'peptide_graph' is not defined",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
        "\u001b[0;32m<ipython-input-15-9c36917c4a40>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mpeptide_graph\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mn_unique_kmers\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
        "\u001b[0;31mNameError\u001b[0m: name 'peptide_graph' is not defined"
       ]
      }
     ],
     "persistentId": "f6bb84e5-c5f5-4d4b-90fb-724792d5d78d",
     "text": "peptide_graph.n_unique_kmers()"
    },
    "executionTime": "2019-10-17T07:34:35.312Z"
   },
   {
    "cell": {
     "executionCount": 16,
     "executionEventId": "f00d0a6a-e3c4-468e-8d2d-983250df57fd",
     "hasError": true,
     "id": "bd29a121-86d7-4b60-9a1d-20971a6bcf21",
     "outputs": [
      {
       "ename": "ImportError",
       "evalue": "cannot import name 'make_peptide_bloom_filter'",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mImportError\u001b[0m                               Traceback (most recent call last)",
        "\u001b[0;32m<ipython-input-16-4d434ef47aee>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0;32mfrom\u001b[0m \u001b[0mkhtools\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpartition_reads\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0mscore_reads\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmake_peptide_bloom_filter\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msix_frame_translation_no_stops\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mis_low_complexity\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
        "\u001b[0;31mImportError\u001b[0m: cannot import name 'make_peptide_bloom_filter'"
       ]
      }
     ],
     "persistentId": "1e54230e-ceec-42fc-afa6-7b32e219378b",
     "text": "from khtools.partition_reads import score_reads, make_peptide_bloom_filter, six_frame_translation_no_stops, is_low_complexity"
    },
    "executionTime": "2019-10-17T07:34:40.555Z"
   },
   {
    "cell": {
     "executionCount": 17,
     "executionEventId": "fed3324d-be89-43f1-8107-b2856f90d333",
     "hasError": true,
     "id": "bd29a121-86d7-4b60-9a1d-20971a6bcf21",
     "outputs": [
      {
       "ename": "ImportError",
       "evalue": "cannot import name 'is_low_complexity'",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mImportError\u001b[0m                               Traceback (most recent call last)",
        "\u001b[0;32m<ipython-input-17-851073fc4693>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0;32mfrom\u001b[0m \u001b[0mkhtools\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpartition_reads\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0mscore_reads\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msix_frame_translation_no_stops\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mis_low_complexity\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      2\u001b[0m \u001b[0;32mfrom\u001b[0m \u001b[0mkhtools\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mbloom_filter\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0mmake_peptide_bloom_filter\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;31mImportError\u001b[0m: cannot import name 'is_low_complexity'"
       ]
      }
     ],
     "persistentId": "1e54230e-ceec-42fc-afa6-7b32e219378b",
     "text": "from khtools.partition_reads import score_reads, six_frame_translation_no_stops, is_low_complexity\nfrom khtools.bloom_filter import make_peptide_bloom_filter"
    },
    "executionTime": "2019-10-17T07:34:51.596Z"
   },
   {
    "cell": {
     "executionCount": 18,
     "executionEventId": "593de35b-b769-4a97-8e84-bdf33d45db5f",
     "hasError": false,
     "id": "bd29a121-86d7-4b60-9a1d-20971a6bcf21",
     "outputs": [],
     "persistentId": "1e54230e-ceec-42fc-afa6-7b32e219378b",
     "text": "from khtools.partition_reads import score_reads, six_frame_translation_no_stops, compute_low_complexity\nfrom khtools.bloom_filter import make_peptide_bloom_filter"
    },
    "executionTime": "2019-10-17T07:35:03.802Z"
   },
   {
    "cell": {
     "executionCount": 19,
     "executionEventId": "e2d214d9-529c-4f41-b87b-689aee4bb54f",
     "hasError": false,
     "id": "70d37b43-2fe3-4f70-9d3f-12a0aca87052",
     "outputs": [
      {
       "name": "stderr",
       "output_type": "stream",
       "text": "/home/olga/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/screed/openscreed.py:39: ResourceWarning: unclosed file <_io.BufferedReader name='Homo_sapiens.GRCh38.pep.all.fa.gz'>\n  self.iter_fn = self.open_reader(filename, *args, **kwargs)\n"
      },
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "CPU times: user 37.9 s, sys: 2.71 s, total: 40.7 s\nWall time: 40.7 s\n"
      }
     ],
     "persistentId": "be06ded1-d7d4-4f21-9961-3aed5ff30cbf",
     "text": "%%time\npeptide_ksize = 7\n\npeptide_graph = make_peptide_bloom_filter(\"Homo_sapiens.GRCh38.pep.all.fa.gz\", peptide_ksize)"
    },
    "executionTime": "2019-10-17T07:35:45.917Z"
   },
   {
    "cell": {
     "executionCount": 20,
     "executionEventId": "44ba989a-c413-41fd-90ad-ac2e5840ef40",
     "hasError": false,
     "id": "547297c6-4cc5-4cc0-b614-3c3bdcbd64b8",
     "outputs": [
      {
       "data": {
        "text/plain": "10841698"
       },
       "execution_count": 20,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "f6bb84e5-c5f5-4d4b-90fb-724792d5d78d",
     "text": "peptide_graph.n_unique_kmers()"
    },
    "executionTime": "2019-10-17T07:35:45.925Z"
   },
   {
    "cell": {
     "executionCount": 21,
     "executionEventId": "eae6ceba-3755-4689-ae6d-d2db9817c29f",
     "hasError": false,
     "id": "9e9dadc6-e3c4-485d-b441-dc0c5ebd0b03",
     "outputs": [
      {
       "data": {
        "text/plain": "[]"
       },
       "execution_count": 21,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "b32b1264-231a-4b32-9229-c49de4f17595",
     "text": "list(peptide_graph.tags())"
    },
    "executionTime": "2019-10-17T07:35:45.932Z"
   },
   {
    "cell": {
     "executionCount": 22,
     "executionEventId": "23d8bbd0-edf6-4a8d-81e0-166573ce7973",
     "hasError": false,
     "id": "86427dcf-37fc-40f0-999b-cc3a1b28ebcb",
     "outputs": [
      {
       "data": {
        "text/plain": "\u001b[0;31mDocstring:\u001b[0m\nHashgraph.add_tag(self, kmer)\nAdd a k-mer to the tagset.\n\u001b[0;31mType:\u001b[0m      builtin_function_or_method\n"
       },
       "metadata": {},
       "output_type": "display_data"
      }
     ],
     "persistentId": "5e420857-1689-4657-983e-7e72891cf0ef",
     "text": "peptide_graph.add_tag?"
    },
    "executionTime": "2019-10-17T07:35:46.095Z"
   },
   {
    "cell": {
     "executionCount": 23,
     "executionEventId": "149575ae-ec74-4e52-8425-63edd24dd85d",
     "hasError": false,
     "id": "be26ea07-f680-434c-b56c-9ac77d40e7b5",
     "outputs": [
      {
       "data": {
        "text/plain": "['__class__',\n '__delattr__',\n '__dir__',\n '__doc__',\n '__eq__',\n '__format__',\n '__ge__',\n '__getattribute__',\n '__gt__',\n '__hash__',\n '__init__',\n '__init_subclass__',\n '__le__',\n '__lt__',\n '__ne__',\n '__new__',\n '__pyx_vtable__',\n '__reduce__',\n '__reduce_ex__',\n '__repr__',\n '__setattr__',\n '__setstate__',\n '__sizeof__',\n '__str__',\n '__subclasshook__',\n '_kmer_type_error',\n '_validate_partitionmap',\n 'abundance_distribution',\n 'add',\n 'add_stop_tag',\n 'add_tag',\n 'assign_partition_id',\n 'calc_connected_graph_size',\n 'consume',\n 'consume_and_tag',\n 'consume_partitioned_fasta',\n 'consume_seqfile',\n 'consume_seqfile_and_tag',\n 'consume_seqfile_banding',\n 'consume_seqfile_banding_with_mask',\n 'consume_seqfile_with_mask',\n 'count',\n 'count_kmers_within_radius',\n 'count_partitions',\n 'divide_tags_into_subsets',\n 'do_subset_partition',\n 'extract_unique_paths',\n 'find_all_tags',\n 'find_all_tags_list',\n 'find_high_degree_nodes',\n 'find_spectral_error_positions',\n 'get',\n 'get_kmer_counts',\n 'get_kmer_hashes',\n 'get_kmer_hashes_as_hashset',\n 'get_kmers',\n 'get_max_count',\n 'get_median_count',\n 'get_min_count',\n 'get_partition_id',\n 'get_raw_tables',\n 'get_stop_tags',\n 'get_tags_and_positions',\n 'get_tags_for_sequence',\n 'get_tagset',\n 'get_use_bigcount',\n 'hash',\n 'hashsizes',\n 'iter_stop_tags',\n 'join_partitions',\n 'kmer_degree',\n 'ksize',\n 'load',\n 'load_partitionmap',\n 'load_stop_tags',\n 'load_tagset',\n 'median_at_least',\n 'merge_subset',\n 'merge_subset_from_disk',\n 'n_occupied',\n 'n_tables',\n 'n_tags',\n 'n_unique_kmers',\n 'neighbors',\n 'output_partitions',\n 'partition',\n 'print_stop_tags',\n 'print_tagset',\n 'repartition_largest_partition',\n 'reverse_hash',\n 'sanitize_kmer',\n 'sanitize_seq_kmer',\n 'save',\n 'save_partitionmap',\n 'save_stop_tags',\n 'save_tagset',\n 'set_partition_id',\n 'set_use_bigcount',\n 'tag_density',\n 'tags',\n 'traverse_linear_path',\n 'trim_below_abundance',\n 'trim_on_abundance',\n 'trim_on_stoptags',\n 'update']"
       },
       "execution_count": 23,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "d28dc029-70e7-4a3f-9455-6ed6882da2c9",
     "text": "dir(peptide_graph)"
    },
    "executionTime": "2019-10-17T07:35:46.103Z"
   },
   {
    "cell": {
     "executionCount": 24,
     "executionEventId": "15a19f9b-3c70-4a46-8603-836193e9faa0",
     "hasError": false,
     "id": "980d827e-4e57-4f71-bfcc-9dbc561e7491",
     "outputs": [
      {
       "name": "stderr",
       "output_type": "stream",
       "text": "Exception ignored in: <_io.FileIO name='SRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz' mode='rb' closefd=True>\nResourceWarning: unclosed file <_io.BufferedReader name='SRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz'>\n795517it [07:23, 1794.13it/s]\n"
      },
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "CPU times: user 7min 16s, sys: 10.9 s, total: 7min 26s\nWall time: 7min 23s\n"
      }
     ],
     "persistentId": "be06ded1-d7d4-4f21-9961-3aed5ff30cbf",
     "text": "%%time\n\nwarnings.filterwarnings(\"error\")\n\nreads = 'SRR306838_GSM752691_hsa_br_F_1_trimmed.fq.gz'\n\ncoding_scores = score_reads(reads, peptide_graph, peptide_ksize)"
    },
    "executionTime": "2019-10-17T07:43:09.470Z"
   },
   {
    "cell": {
     "executionCount": 25,
     "executionEventId": "04e328a1-1898-4ffd-9a91-993e7d4e53fb",
     "hasError": false,
     "id": "9bea540d-d5b3-4450-973e-b2d73040c25e",
     "outputs": [
      {
       "data": {
        "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>read_id</th>\n      <th>jaccard_in_peptide_db</th>\n      <th>n_kmers</th>\n      <th>classification</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>SRR306838.1531 Ibis_Run100924_C3PO:6:1:15718:1062/1</td>\n      <td>-1.0</td>\n      <td>2</td>\n      <td>low complexity</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>SRR306838.1617 Ibis_Run100924_C3PO:6:1:9036:1074/1</td>\n      <td>0.0</td>\n      <td>0</td>\n      <td>non-coding</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>SRR306838.1672 Ibis_Run100924_C3PO:6:1:15751:1077/1</td>\n      <td>-1.0</td>\n      <td>1</td>\n      <td>low complexity</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>SRR306838.1938 Ibis_Run100924_C3PO:6:1:15775:1103/1</td>\n      <td>-1.0</td>\n      <td>1</td>\n      <td>low complexity</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>SRR306838.2318 Ibis_Run100924_C3PO:6:1:15779:1141/1</td>\n      <td>-1.0</td>\n      <td>7</td>\n      <td>non-coding</td>\n    </tr>\n  </tbody>\n</table>\n</div>",
        "text/plain": "                                               read_id  jaccard_in_peptide_db  \\\n0  SRR306838.1531 Ibis_Run100924_C3PO:6:1:15718:1062/1                   -1.0   \n1   SRR306838.1617 Ibis_Run100924_C3PO:6:1:9036:1074/1                    0.0   \n2  SRR306838.1672 Ibis_Run100924_C3PO:6:1:15751:1077/1                   -1.0   \n3  SRR306838.1938 Ibis_Run100924_C3PO:6:1:15775:1103/1                   -1.0   \n4  SRR306838.2318 Ibis_Run100924_C3PO:6:1:15779:1141/1                   -1.0   \n\n   n_kmers  classification  \n0        2  low complexity  \n1        0      non-coding  \n2        1  low complexity  \n3        1  low complexity  \n4        7      non-coding  "
       },
       "execution_count": 25,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "b44eec0d-17d9-459a-8fac-49e78b25a50c",
     "text": "coding_scores.head()"
    },
    "executionTime": "2019-10-17T07:43:09.663Z"
   },
   {
    "cell": {
     "executionCount": 26,
     "executionEventId": "923fc40f-808d-4f44-9d13-faefc5a4bec6",
     "hasError": false,
     "id": "47ca21c4-3da0-401e-b25a-536a4fc77a30",
     "outputs": [
      {
       "data": {
        "text/plain": "(795517, 4)"
       },
       "execution_count": 26,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "c381cc92-ba84-43f1-95f7-d19b70d547f5",
     "text": "coding_scores.shape"
    },
    "executionTime": "2019-10-17T07:43:09.693Z"
   },
   {
    "cell": {
     "executionCount": 27,
     "executionEventId": "7accd32b-900f-47af-b32a-0c4c30b49fe7",
     "hasError": false,
     "id": "86909708-30e2-4533-9c04-1ee39bb3c07e",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "(793969, 4)\n"
      },
      {
       "data": {
        "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>read_id</th>\n      <th>jaccard_in_peptide_db</th>\n      <th>n_kmers</th>\n      <th>classification</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>1</th>\n      <td>SRR306838.1617 Ibis_Run100924_C3PO:6:1:9036:1074/1</td>\n      <td>0.000000</td>\n      <td>0</td>\n      <td>non-coding</td>\n    </tr>\n    <tr>\n      <th>7</th>\n      <td>SRR306838.2723 Ibis_Run100924_C3PO:6:1:15754:1168/1</td>\n      <td>0.769231</td>\n      <td>13</td>\n      <td>non-coding</td>\n    </tr>\n    <tr>\n      <th>9</th>\n      <td>SRR306838.3707 Ibis_Run100924_C3PO:6:1:5241:1269/1</td>\n      <td>1.000000</td>\n      <td>16</td>\n      <td>coding</td>\n    </tr>\n    <tr>\n      <th>10</th>\n      <td>SRR306838.4251 Ibis_Run100924_C3PO:6:1:5141:1314/1</td>\n      <td>0.000000</td>\n      <td>15</td>\n      <td>non-coding</td>\n    </tr>\n    <tr>\n      <th>11</th>\n      <td>SRR306838.4936 Ibis_Run100924_C3PO:6:1:5473:1375/1</td>\n      <td>0.000000</td>\n      <td>14</td>\n      <td>non-coding</td>\n    </tr>\n  </tbody>\n</table>\n</div>",
        "text/plain": "                                                read_id  \\\n1    SRR306838.1617 Ibis_Run100924_C3PO:6:1:9036:1074/1   \n7   SRR306838.2723 Ibis_Run100924_C3PO:6:1:15754:1168/1   \n9    SRR306838.3707 Ibis_Run100924_C3PO:6:1:5241:1269/1   \n10   SRR306838.4251 Ibis_Run100924_C3PO:6:1:5141:1314/1   \n11   SRR306838.4936 Ibis_Run100924_C3PO:6:1:5473:1375/1   \n\n    jaccard_in_peptide_db  n_kmers classification  \n1                0.000000        0     non-coding  \n7                0.769231       13     non-coding  \n9                1.000000       16         coding  \n10               0.000000       15     non-coding  \n11               0.000000       14     non-coding  "
       },
       "execution_count": 27,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "73b49c34-8379-4aa5-835e-57c54aced1fa",
     "text": "coding_scores_not_low_complexity = coding_scores.query('jaccard_in_peptide_db >= 0')\nprint(coding_scores_not_low_complexity.shape)\ncoding_scores_not_low_complexity.head()"
    },
    "executionTime": "2019-10-17T07:43:09.705Z"
   },
   {
    "cell": {
     "executionCount": 28,
     "executionEventId": "838337ba-9b4d-4ac8-9641-9b0a7f474408",
     "hasError": false,
     "id": "a45589ec-d77f-4e20-b501-ded4a73f542a",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "total 362M\ndrwxrwxr-x 2 olga olga 4.0K Oct 11 21:11 .\ndrwxrwxr-x 3 olga olga 4.0K Sep 10 15:07 ..\n-rw-rw-r-- 1 olga olga  52M Oct 11 18:07 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.bam\n-rw-rw-r-- 1 olga olga 2.9M Oct 11 18:08 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.bam.bai\n-rw-rw-r-- 1 olga olga 1.9M Oct 11 21:11 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.CDS_without_stop_codon.read_ids.txt\n-rw-rw-r-- 1 olga olga 7.0M Oct 11 20:58 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.out_CDS_without_stop_codon.bam\n-rw-rw-r-- 1 olga olga 6.1M Oct 11 18:08 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.out_CDS_without_stop_codon_or_utr.bam\n-rw-rw-r-- 1 olga olga 1.8M Oct 11 18:08 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.out_CDS_without_stop_codon_or_utr.bam.bai\n-rw-rw-r-- 1 olga olga 1.7M Oct 11 18:08 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.out_CDS_without_stop_codon_or_utr.read_ids.txt\n-rw-rw-r-- 1 olga olga  15M Oct 11 18:08 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.read_ids.txt\n-rw-rw-r-- 1 olga olga  15M Sep 10 15:55 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out_CDS_without_stop_codon.bam\n-rw-rw-r-- 1 olga olga 1.7M Oct 11 18:08 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out_CDS_without_stop_codon.bam.bai\n-rw-rw-r-- 1 olga olga  15M Sep 18 07:15 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out_CDS_without_stop_codon_or_utr.bam\n-rw-rw-r-- 1 olga olga 1.7M Oct 11 18:08 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out_CDS_without_stop_codon_or_utr.bam.bai\n-rw-rw-r-- 1 olga olga 2.8M Sep 18 10:14 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out_CDS_without_stop_codon_or_utr.read_ids.txt\n-rw-rw-r-- 1 olga olga 4.1M Sep 10 15:56 SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out_CDS_without_stop_codon.read_ids.txt\n-rw-r--r-- 1 olga olga 235M Sep 10 15:10 SRR306838_GSM752691_hsa_br_F_1_trimmed_k15.tsv\n(91741,)\n"
      },
      {
       "data": {
        "text/plain": "0    SRR306838.13598607\n1    SRR306838.23303318\n2    SRR306838.19917713\n3    SRR306838.19937719\n4    SRR306838.20862490\nName: 0, dtype: object"
       },
       "execution_count": 28,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "75c01782-427c-4c63-bc29-ed308ae4a142",
     "text": "folder = '/home/olga/data_sm/kmer-hashing/brawand2011/human/'\n! ls -lha $folder\n\ncoding_read_ids = pd.read_csv(f\"{folder}/SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.out_CDS_without_stop_codon_or_utr.read_ids.txt\", \n                                   header=None, squeeze=True)\n\n# Try just CDS minus stop codons .... but classification goes down from 92% to 90%\n# coding_read_ids = pd.read_csv(f\"{folder}/SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.CDS_without_stop_codon.read_ids.txt\", \n#                                    header=None, squeeze=True)\nprint(coding_read_ids.shape)\ncoding_read_ids.head()"
    },
    "executionTime": "2019-10-17T07:43:10.170Z"
   },
   {
    "cell": {
     "executionCount": 29,
     "executionEventId": "86031fee-645a-47a2-8947-1baf2c347724",
     "hasError": false,
     "id": "6e060ff0-135a-4e24-90f0-9dec04bbdc00",
     "outputs": [
      {
       "data": {
        "text/plain": "90307"
       },
       "execution_count": 29,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "75c01782-427c-4c63-bc29-ed308ae4a142",
     "text": "true_coding_read_ids = set(coding_read_ids.values)\nlen(true_coding_read_ids)"
    },
    "executionTime": "2019-10-17T07:43:10.232Z"
   },
   {
    "cell": {
     "executionCount": 30,
     "executionEventId": "02e1482f-4fd8-4c9f-8982-d8c73a935885",
     "hasError": false,
     "id": "d3b53b1d-4ded-4afa-a3d2-bce833550ed3",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "(807698,)\n"
      }
     ],
     "persistentId": "317bb484-c93f-4e1c-b831-50734c9e6614",
     "text": "unspliced_read_ids_series = pd.read_csv(f\"{folder}/SRR306838_GSM752691_hsa_br_F_1Aligned.sortedByCoord.out.bam.unspliced.read_ids.txt\", \n                                   header=None, squeeze=True)\nprint(unspliced_read_ids_series.shape)\nunspliced_read_ids_series.head()\n\nunspliced_reads = set(unspliced_read_ids_series.values)"
    },
    "executionTime": "2019-10-17T07:43:10.989Z"
   },
   {
    "cell": {
     "executionCount": 32,
     "executionEventId": "dd6ba692-46d1-42fe-a1c0-cb8fe8747aa2",
     "hasError": true,
     "id": "0ba3a826-0490-4f3b-bea5-5e202ae298f8",
     "outputs": [
      {
       "ename": "SettingWithCopyWarning",
       "evalue": "\nA value is trying to be set on a copy of a slice from a DataFrame.\nTry using .loc[row_indexer,col_indexer] = value instead\n\nSee the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mSettingWithCopyWarning\u001b[0m                    Traceback (most recent call last)",
        "\u001b[0;32m<ipython-input-32-436bda3e9523>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mcoding_scores_not_low_complexity\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'short_read_id'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mlist\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcoding_scores_not_low_complexity\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'read_id'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msplit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mvalues\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      2\u001b[0m \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcoding_scores_not_low_complexity\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshape\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      3\u001b[0m \u001b[0mcoding_scores_not_low_complexity\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mhead\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m__setitem__\u001b[0;34m(self, key, value)\u001b[0m\n\u001b[1;32m   3368\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3369\u001b[0m             \u001b[0;31m# set column\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3370\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_set_item\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3371\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3372\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_setitem_slice\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkey\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m_set_item\u001b[0;34m(self, key, value)\u001b[0m\n\u001b[1;32m   3450\u001b[0m         \u001b[0;31m# value exception to occur first\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3451\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3452\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_check_setitem_copy\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3453\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3454\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0minsert\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mloc\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcolumn\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mallow_duplicates\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/generic.py\u001b[0m in \u001b[0;36m_check_setitem_copy\u001b[0;34m(self, stacklevel, t, force)\u001b[0m\n\u001b[1;32m   3285\u001b[0m             \u001b[0;32melif\u001b[0m \u001b[0mvalue\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m'warn'\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3286\u001b[0m                 warnings.warn(t, com.SettingWithCopyWarning,\n\u001b[0;32m-> 3287\u001b[0;31m                               stacklevel=stacklevel)\n\u001b[0m\u001b[1;32m   3288\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3289\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m__delitem__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;31mSettingWithCopyWarning\u001b[0m: \nA value is trying to be set on a copy of a slice from a DataFrame.\nTry using .loc[row_indexer,col_indexer] = value instead\n\nSee the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy"
       ]
      }
     ],
     "persistentId": "49d36e97-c1f4-400a-a81a-a2605e537be5",
     "text": "coding_scores_not_low_complexity['short_read_id'] = list(coding_scores_not_low_complexity['read_id'].str.split().str[0].values)\nprint(coding_scores_not_low_complexity.shape)\ncoding_scores_not_low_complexity.head()"
    },
    "executionTime": "2019-10-17T07:54:59.141Z"
   },
   {
    "cell": {
     "executionCount": 33,
     "executionEventId": "2ddd6faa-9431-47d5-b79f-8ab225f813a0",
     "hasError": true,
     "id": "0ba3a826-0490-4f3b-bea5-5e202ae298f8",
     "outputs": [
      {
       "ename": "SettingWithCopyWarning",
       "evalue": "\nA value is trying to be set on a copy of a slice from a DataFrame.\nTry using .loc[row_indexer,col_indexer] = value instead\n\nSee the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mSettingWithCopyWarning\u001b[0m                    Traceback (most recent call last)",
        "\u001b[0;32m<ipython-input-33-a146c472ba1f>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mcoding_scores_not_low_complexity\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'short_read_id'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcoding_scores_not_low_complexity\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'read_id'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msplit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mvalues\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      2\u001b[0m \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcoding_scores_not_low_complexity\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshape\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      3\u001b[0m \u001b[0mcoding_scores_not_low_complexity\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mhead\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m__setitem__\u001b[0;34m(self, key, value)\u001b[0m\n\u001b[1;32m   3368\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3369\u001b[0m             \u001b[0;31m# set column\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3370\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_set_item\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3371\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3372\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_setitem_slice\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkey\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m_set_item\u001b[0;34m(self, key, value)\u001b[0m\n\u001b[1;32m   3450\u001b[0m         \u001b[0;31m# value exception to occur first\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3451\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3452\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_check_setitem_copy\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3453\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3454\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0minsert\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mloc\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcolumn\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mallow_duplicates\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/generic.py\u001b[0m in \u001b[0;36m_check_setitem_copy\u001b[0;34m(self, stacklevel, t, force)\u001b[0m\n\u001b[1;32m   3285\u001b[0m             \u001b[0;32melif\u001b[0m \u001b[0mvalue\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m'warn'\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3286\u001b[0m                 warnings.warn(t, com.SettingWithCopyWarning,\n\u001b[0;32m-> 3287\u001b[0;31m                               stacklevel=stacklevel)\n\u001b[0m\u001b[1;32m   3288\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3289\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m__delitem__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;31mSettingWithCopyWarning\u001b[0m: \nA value is trying to be set on a copy of a slice from a DataFrame.\nTry using .loc[row_indexer,col_indexer] = value instead\n\nSee the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy"
       ]
      }
     ],
     "persistentId": "49d36e97-c1f4-400a-a81a-a2605e537be5",
     "text": "coding_scores_not_low_complexity['short_read_id'] = coding_scores_not_low_complexity['read_id'].str.split().str[0].values\nprint(coding_scores_not_low_complexity.shape)\ncoding_scores_not_low_complexity.head()"
    },
    "executionTime": "2019-10-17T07:55:10.540Z"
   },
   {
    "cell": {
     "executionCount": 34,
     "executionEventId": "8d3307e6-7fe0-4f30-920d-1ce8a2c4c856",
     "hasError": false,
     "id": "0a18a780-18df-4a0e-8e42-93c4f2a4b6a1",
     "outputs": [
      {
       "data": {
        "text/plain": "array(['SRR306838.1617', 'SRR306838.2723', 'SRR306838.3707', ...,\n       'SRR306838.24510192', 'SRR306838.24510500', 'SRR306838.24510625'],\n      dtype=object)"
       },
       "execution_count": 34,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "5ef9f333-5da4-493e-9f95-93efd87f9316",
     "text": "coding_scores_not_low_complexity['read_id'].str.split().str[0].values"
    },
    "executionTime": "2019-10-17T07:55:17.490Z"
   },
   {
    "cell": {
     "executionCount": 35,
     "executionEventId": "666c43fa-a49f-4b19-a8da-66469fb0ad19",
     "hasError": true,
     "id": "0ba3a826-0490-4f3b-bea5-5e202ae298f8",
     "outputs": [
      {
       "ename": "SettingWithCopyWarning",
       "evalue": "\nA value is trying to be set on a copy of a slice from a DataFrame.\nTry using .loc[row_indexer,col_indexer] = value instead\n\nSee the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mSettingWithCopyWarning\u001b[0m                    Traceback (most recent call last)",
        "\u001b[0;32m<ipython-input-35-a8efc3d0be59>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mcoding_scores_not_low_complexity\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'short_read_id'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcoding_scores_not_low_complexity\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'read_id'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msplit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      2\u001b[0m \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcoding_scores_not_low_complexity\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshape\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      3\u001b[0m \u001b[0mcoding_scores_not_low_complexity\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mhead\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m__setitem__\u001b[0;34m(self, key, value)\u001b[0m\n\u001b[1;32m   3368\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3369\u001b[0m             \u001b[0;31m# set column\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3370\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_set_item\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3371\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3372\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_setitem_slice\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkey\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m_set_item\u001b[0;34m(self, key, value)\u001b[0m\n\u001b[1;32m   3450\u001b[0m         \u001b[0;31m# value exception to occur first\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3451\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3452\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_check_setitem_copy\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3453\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3454\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0minsert\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mloc\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcolumn\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mallow_duplicates\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/generic.py\u001b[0m in \u001b[0;36m_check_setitem_copy\u001b[0;34m(self, stacklevel, t, force)\u001b[0m\n\u001b[1;32m   3285\u001b[0m             \u001b[0;32melif\u001b[0m \u001b[0mvalue\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m'warn'\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3286\u001b[0m                 warnings.warn(t, com.SettingWithCopyWarning,\n\u001b[0;32m-> 3287\u001b[0;31m                               stacklevel=stacklevel)\n\u001b[0m\u001b[1;32m   3288\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3289\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m__delitem__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;31mSettingWithCopyWarning\u001b[0m: \nA value is trying to be set on a copy of a slice from a DataFrame.\nTry using .loc[row_indexer,col_indexer] = value instead\n\nSee the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy"
       ]
      }
     ],
     "persistentId": "49d36e97-c1f4-400a-a81a-a2605e537be5",
     "text": "coding_scores_not_low_complexity['short_read_id'] = coding_scores_not_low_complexity['read_id'].str.split().str[0]\nprint(coding_scores_not_low_complexity.shape)\ncoding_scores_not_low_complexity.head()"
    },
    "executionTime": "2019-10-17T07:55:27.236Z"
   },
   {
    "cell": {
     "executionCount": 36,
     "executionEventId": "81696e7f-2fcb-43a7-a585-4df59d8d52e8",
     "hasError": false,
     "id": "0a18a780-18df-4a0e-8e42-93c4f2a4b6a1",
     "outputs": [
      {
       "data": {
        "text/plain": "1             SRR306838.1617\n7             SRR306838.2723\n9             SRR306838.3707\n10            SRR306838.4251\n11            SRR306838.4936\n12            SRR306838.5359\n13            SRR306838.6299\n14            SRR306838.8107\n15           SRR306838.10022\n17           SRR306838.10673\n18           SRR306838.10841\n19           SRR306838.11357\n24           SRR306838.13390\n25           SRR306838.13837\n26           SRR306838.13968\n28           SRR306838.14545\n31           SRR306838.14924\n34           SRR306838.15166\n35           SRR306838.15290\n36           SRR306838.15435\n37           SRR306838.15721\n39           SRR306838.15853\n41           SRR306838.16067\n42           SRR306838.16073\n43           SRR306838.16099\n46           SRR306838.16735\n47           SRR306838.17165\n48           SRR306838.17812\n49           SRR306838.18429\n50           SRR306838.18431\n                 ...        \n795484    SRR306838.24496707\n795485    SRR306838.24497783\n795486    SRR306838.24498630\n795487    SRR306838.24498923\n795488    SRR306838.24500615\n795489    SRR306838.24500944\n795490    SRR306838.24501224\n795491    SRR306838.24501482\n795492    SRR306838.24501695\n795493    SRR306838.24502139\n795494    SRR306838.24502320\n795495    SRR306838.24502629\n795496    SRR306838.24503808\n795497    SRR306838.24504359\n795498    SRR306838.24504516\n795499    SRR306838.24504807\n795500    SRR306838.24504881\n795501    SRR306838.24505219\n795502    SRR306838.24506841\n795503    SRR306838.24507188\n795504    SRR306838.24507378\n795507    SRR306838.24507766\n795508    SRR306838.24508567\n795509    SRR306838.24508607\n795510    SRR306838.24508944\n795511    SRR306838.24509694\n795512    SRR306838.24509978\n795513    SRR306838.24510192\n795514    SRR306838.24510500\n795515    SRR306838.24510625\nName: read_id, Length: 793969, dtype: object"
       },
       "execution_count": 36,
       "metadata": {},
       "output_type": "execute_result"
      }
     ],
     "persistentId": "5ef9f333-5da4-493e-9f95-93efd87f9316",
     "text": "coding_scores_not_low_complexity['read_id'].str.split().str[0]"
    },
    "executionTime": "2019-10-17T07:55:28.683Z"
   },
   {
    "cell": {
     "executionCount": 37,
     "executionEventId": "3d635fcc-aaf1-4f37-bea4-3bb537343e27",
     "hasError": false,
     "id": "0a18a780-18df-4a0e-8e42-93c4f2a4b6a1",
     "outputs": [],
     "persistentId": "5ef9f333-5da4-493e-9f95-93efd87f9316",
     "text": "# coding_scores_not_low_complexity['read_id'].str.split().str[0]"
    },
    "executionTime": "2019-10-17T07:55:31.726Z"
   },
   {
    "cell": {
     "executionCount": 38,
     "executionEventId": "e530e624-170a-42c3-bf95-2b873f1ff402",
     "hasError": true,
     "id": "0ba3a826-0490-4f3b-bea5-5e202ae298f8",
     "outputs": [
      {
       "ename": "SettingWithCopyWarning",
       "evalue": "\nA value is trying to be set on a copy of a slice from a DataFrame\n\nSee the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mSettingWithCopyWarning\u001b[0m                    Traceback (most recent call last)",
        "\u001b[0;32m<ipython-input-38-a88882f1fe13>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mcoding_scores_not_low_complexity\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mloc\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'short_read_id'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m:\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcoding_scores_not_low_complexity\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'read_id'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msplit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      2\u001b[0m \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcoding_scores_not_low_complexity\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshape\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      3\u001b[0m \u001b[0mcoding_scores_not_low_complexity\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mhead\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/indexing.py\u001b[0m in \u001b[0;36m__setitem__\u001b[0;34m(self, key, value)\u001b[0m\n\u001b[1;32m    188\u001b[0m             \u001b[0mkey\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcom\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mapply_if_callable\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mobj\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    189\u001b[0m         \u001b[0mindexer\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_get_setitem_indexer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 190\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_setitem_with_indexer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mindexer\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    191\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    192\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_validate_key\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkey\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/indexing.py\u001b[0m in \u001b[0;36m_setitem_with_indexer\u001b[0;34m(self, indexer, value)\u001b[0m\n\u001b[1;32m    375\u001b[0m                     \u001b[0mlabels\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mindex\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0minsert\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mindex\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    376\u001b[0m                     \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mobj\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_data\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mobj\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreindex\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlabels\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_data\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 377\u001b[0;31m                     \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mobj\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_maybe_update_cacher\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mclear\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    378\u001b[0m                     \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mobj\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_is_copy\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    379\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/generic.py\u001b[0m in \u001b[0;36m_maybe_update_cacher\u001b[0;34m(self, clear, verify_is_copy)\u001b[0m\n\u001b[1;32m   3142\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3143\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mverify_is_copy\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3144\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_check_setitem_copy\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mstacklevel\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m5\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mt\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'referant'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3145\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3146\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mclear\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/generic.py\u001b[0m in \u001b[0;36m_check_setitem_copy\u001b[0;34m(self, stacklevel, t, force)\u001b[0m\n\u001b[1;32m   3285\u001b[0m             \u001b[0;32melif\u001b[0m \u001b[0mvalue\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m'warn'\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3286\u001b[0m                 warnings.warn(t, com.SettingWithCopyWarning,\n\u001b[0;32m-> 3287\u001b[0;31m                               stacklevel=stacklevel)\n\u001b[0m\u001b[1;32m   3288\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3289\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m__delitem__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;31mSettingWithCopyWarning\u001b[0m: \nA value is trying to be set on a copy of a slice from a DataFrame\n\nSee the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy"
       ]
      }
     ],
     "persistentId": "49d36e97-c1f4-400a-a81a-a2605e537be5",
     "text": "coding_scores_not_low_complexity.loc['short_read_id', :] = coding_scores_not_low_complexity['read_id'].str.split().str[0]\nprint(coding_scores_not_low_complexity.shape)\ncoding_scores_not_low_complexity.head()"
    },
    "executionTime": "2019-10-17T07:55:42.260Z"
   },
   {
    "cell": {
     "executionCount": 39,
     "executionEventId": "780e8e35-6843-4d81-a036-f002c4b93605",
     "hasError": true,
     "id": "e148f523-73b6-4fac-a054-ba32c4add1f9",
     "outputs": [
      {
       "ename": "SettingWithCopyWarning",
       "evalue": "\nA value is trying to be set on a copy of a slice from a DataFrame.\nTry using .loc[row_indexer,col_indexer] = value instead\n\nSee the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy",
       "output_type": "error",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
        "\u001b[0;31mSettingWithCopyWarning\u001b[0m                    Traceback (most recent call last)",
        "\u001b[0;32m<ipython-input-39-75293f21b612>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0mcoding_scores_not_low_complexity_unspliced\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcoding_scores_not_low_complexity\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mquery\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'short_read_id in @unspliced_reads'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m \u001b[0mcoding_scores_not_low_complexity_unspliced\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'in_cds_no_stop_codon'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcoding_scores_not_low_complexity_unspliced\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshort_read_id\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0misin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtrue_coding_read_ids\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      3\u001b[0m \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcoding_scores_not_low_complexity_unspliced\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshape\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[0mcoding_scores_not_low_complexity_unspliced\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mhead\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m__setitem__\u001b[0;34m(self, key, value)\u001b[0m\n\u001b[1;32m   3368\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3369\u001b[0m             \u001b[0;31m# set column\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3370\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_set_item\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3371\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3372\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_setitem_slice\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkey\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m_set_item\u001b[0;34m(self, key, value)\u001b[0m\n\u001b[1;32m   3450\u001b[0m         \u001b[0;31m# value exception to occur first\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3451\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3452\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_check_setitem_copy\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3453\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3454\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0minsert\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mloc\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcolumn\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mallow_duplicates\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m~/miniconda3/envs/kmer-hashing/lib/python3.6/site-packages/pandas/core/generic.py\u001b[0m in \u001b[0;36m_check_setitem_copy\u001b[0;34m(self, stacklevel, t, force)\u001b[0m\n\u001b[1;32m   3285\u001b[0m             \u001b[0;32melif\u001b[0m \u001b[0mvalue\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m'warn'\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3286\u001b[0m                 warnings.warn(t, com.SettingWithCopyWarning,\n\u001b[0;32m-> 3287\u001b[0;31m                               stacklevel=stacklevel)\n\u001b[0m\u001b[1;32m   3288\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3289\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m__delitem__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;31mSettingWithCopyWarning\u001b[0m: \nA value is trying to be set on a copy of a slice from a DataFrame.\nTry using .loc[row_indexer,col_indexer] = value instead\n\nSee the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy"
       ]
      }
     ],
     "persistentId": "b321c9fc-54af-40bc-96a9-40c59ea87994",
     "text": "coding_scores_not_low_complexity_unspliced = coding_scores_not_low_complexity.query('short_read_id in @unspliced_reads')\ncoding_scores_not_low_complexity_unspliced['in_cds_no_stop_codon'] = coding_scores_not_low_complexity_unspliced.short_read_id.isin(true_coding_read_ids)\nprint(coding_scores_not_low_complexity_unspliced.shape)\ncoding_scores_not_low_complexity_unspliced.head()"
    },
    "executionTime": "2019-10-17T07:57:15.606Z"
   },
   {
    "cell": {
     "executionCount": 40,
     "executionEventId": "b74bec83-680b-426c-a651-e05712127b39",
     "hasError": false,
     "id": "ddfff1e6-fd17-4da4-aee8-e95d812c866f",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "accuracy score: 0.8768244717880631\n              precision    recall  f1-score   support\n\n       False       1.00      0.86      0.92    590637\n        True       0.52      0.97      0.68     90245\n\n   micro avg       0.88      0.88      0.88    680882\n   macro avg       0.76      0.92      0.80    680882\nweighted avg       0.93      0.88      0.89    680882\n\n"
      },
      {
       "data": {
        "image/png": "iVBORw0KGgoAAAANSUhEUgAAAYUAAAEKCAYAAAD9xUlFAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDMuMC4zLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvnQurowAAIABJREFUeJzt3Xl8FFW2wPHfyQJh3xHZkUVFkIgBFBUXFlEkOOooiiCKICjKuI36XBCfz0FcUUSMwLgMKg6DGhwYZBRcEBAEEQGVsAhBkS1BhCRkOe+P6sQ2ZulAqquX8/188qGrutJ9KiF1+t576l5RVYwxxhiAGK8DMMYYEzosKRhjjCliScEYY0wRSwrGGGOKWFIwxhhTxJKCMcaYIpYUjDHGFLGkYIwxpoglBWOMMUXivA6goho2bKitW7f2OgxjjAkrX3755V5VbVTecWGXFFq3bs2qVau8DsMYY8KKiPwQyHHWfWSMMaaIJQVjjDFFLCkYY4wpYknBGGNMEUsKxhhjiriWFERkpojsFpFvSnleROQ5EUkTka9FpKtbsRhjjAmMmy2FV4D+ZTx/EdDe9zUKeNHFWIwxxgTAtfsUVPUTEWldxiGDgNfUWQ90uYjUFZHjVfUnt2IyxpgKU4WCXMg/8tu//o8LStlf+LiglP2F35vvd0wpz33xXRUSYo9w6tk9oOd4V0/Xy5vXmgE7/LbTffv+kBREZBROa4KWLVsGJThjjAu0wHehK+WiGMiFsyIX5jK/N9DXzfPux6Vwz7/78tTH3Tj1+J/5ov0G4l1+Ty+TgpSwT0s6UFVTgBSApKSkEo8xJuoUXWAD+7QZ/Iuu/zGFn5q9u8Aek5g4iKkCsVUgJt751/9xTBWIjS/lmLKeK2N/bBUkpgp8vws++Zl+yd3J794jopNCOtDCb7s58KNHsZhQoupc8Py/KHyspewvfnxJr6G/vUZJjwN6vnB/fukXvmBddDXfq9/QsSm6kAZyQa2Ei27hMSXtD+DCTEw8SEmfYd2RmZnNli0ZdO16PAATJucyePTeom23eZkUUoGxIvIW0AM4YOMJflQhOwMO74KsvZCdCfnZkJcFeX7/5mf/9gmsIM+5UBTkO/s0v9i+PN9+vwtqQb7fhbO0xyVt+18k8397D82nxAtySRfr3x3nd+E1gQvkwhesi25AF+a4oF5gw817733LmDH/JiZGWL/+ZurUSaBatfigJQRwMSmIyJvAeUBDEUkHxoPT8lHVacB84GIgDTgMXO9WLGEjPxd2rYRv34RNc+DQLq8j8pBATCxIjPNFjHMxKdwu2lfCF8WOE/E7Vvye93tc6r5ijwufi4nz/qJrF9iIsXv3IW67bQGzZ68H4IwzmpOZmU2dOglBj8XN6qOry3legVvcev+wcnAnrJwE377htAoKxdeEmk2hWiNIqAuxCRBfHWKrQlw1Zzuumu9iEQ8S67uQxvpdNIrviwWJ+/1FMyaWootm4YWYwv3+F2f/i3ThxZLfv4f/Mf7HFr5WaRfq372uXehMdFBVZs1ax7hx/2H//iyqV4/nsccuYOzY7sTGenNvcdhNnR1Rcg/Dskdg9bOQn+Psq9sO2lwMp1wHjRN/u/AaYyLOmDH/5qWXvgSgT58TSEm5hDZt6nkakyUFL6g6XUSf3gcHtzv72v0Jut8LTbrZJ2VjosSll57E7Nnreeqpflx/fSISAn/7lhSCTRU+vdfpLgJolAh9XoSmZ3gblzHGdZs27ePDD7cyenQSAP37t2PbtnGejB2UxpJCMGkBfDQOvpri9L33fgE63+jrbzfGRKq8vAKefnoZ48cvIScnj8TEJpxxRnOAkEoIYEkheAryIPUK2PyeMwA88F/QLtnrqIwxLlu7dhcjRqTy5ZdOxf2wYV1o376+x1GVzpJCMKjCopuchFC1LgycA616ex2VMcZFOTl5PProJ0ycuJS8vAJatqzDSy9dQv/+7bwOrUyWFIJh8zz4ZqZTPnrZfGh6ptcRGWNcdt99H/LMM8sBuOWWbvztb72pVauqx1GVz5KC21Thvzc5j8961BKCMVHir389i2XL0pk0qQ/nnNPK63ACZkXwbtu33rkzObYqnDbW62iMMS5ZtGgzl1/+Nnl5BQA0aVKTzz+/IawSAlhScN+er51/G3VxpicwxkSUjIwsRox4j379/sHcuRv5+9/XFD0XCvcdVJR1H7lt3wbn39b9vI3DGFPp3nlnIzffPJ9du36latVYxo8/l+HDE70O65hYUnDbPmeCK+p39DYOY0yl2bXrV269dQFz5jgf+nr2bMGMGcmcdFJDjyM7dpYU3FaYFBqe4m0cxphK89573zJnzgZq1Ihn4sQ+3HxzN2Jiwq+rqCSWFNyUlw2Zm51J7eqd6HU0xphjkJ2dR0KCc8kcOfJ0tmzJYMyYbrRuXdfjyCqXDTS7af93ztQWddtBXOjXJxtj/qigQJky5QvatJnMDz9kAhATIzz+eN+ISwhgScFdhV1HDazryJhw9N13e+nV6+/ceusCdu36lTff/MbrkFxn3UduKqw8svEEY8JKbm4+Tz75ORMmfExOTj7HHVeDqVMHcNllJ3sdmussKbjJKo+MCTvffLObYcPeYc0aZznc669P5Kmn+lGvXjWPIwsOSwpuspaCMWGnoEBZt243rVrVISVlIP36tfU6pKCypOCWvGzITPNVHnXwOhpjTBnWr99Nx46NEBFOPfU43ntvML16taJmzeibhcAGmt3yu8qj0FpEwxjjOHgwh7Fj59Op04v8618bi/ZffHH7qEwIYC0F9xR2HVnlkTEhaeHCNEaNep/t2w8QFxfDtm2ZXocUEiwpuKWoHNUGmY0JJfv3Z3H77Qt57bW1AHTtejwzZiSTmNjE48hCgyUFt9g9CsaEnK++2kX//v/g558PUbVqLBMmnMedd/YkLs560gtZUnBLUfeRtRSMCRUdOjSgZs0qdOjQgOnTk+nQoYHXIYUcSwpu8K88qm9zHhnjFVXljTfWMXDgidSuXZXq1eNZsmQ4TZvWipgJ7CqbtZnckPG9VR4Z47Ft2zK58MJ/cO2173Dvvf8t2t+8eW1LCGWwloIb9togszFeyc8vYOrUldx334ccOpRL/frV6NmzhddhhQ1LCm6wQWZjPLFx4x5GjEhl2bJ0AK688hSef/4iGjeu4XFk4cOSghvsHgVjgm7r1gwSE1/iyJF8jj++JlOnDuDSS0/yOqyw42pSEJH+wGQgFpiuqhOLPd8SeBWo6zvmXlWd72ZMQWH3KBgTdG3a1OPPf+5IQkIcTz7Zj7p1bTzvaLiWFEQkFngB6AukAytFJFVVN/gd9gDwtqq+KCIdgflAa7diCgqrPDImKLKycnnkkY/5059Opnv3ZgC8+uqlxMZa/cyxcPOn1x1IU9UtqnoEeAsYVOwYBWr7HtcBfnQxnuAoqjxqa5VHxrjk009/IDHxJSZOXMqoUfMoKFAASwiVwM3uo2bADr/tdKBHsWMeBj4QkVuBGkAfF+MJjr02yGyMW375JYf77vsvU6euAqBjx0ZMm3aJlZhWIjfTakm/JS22fTXwiqo2By4GXheRP8QkIqNEZJWIrNqzZ48LoVai/TbIbIwb5s/fRKdOU5k6dRVxcTE89FAvVq8exRlnNPc6tIjiZkshHfAvDm7OH7uHRgD9AVR1mYgkAA2B3f4HqWoKkAKQlJRUPLGEFrtHwZhKd+BANkOGzCUzM5ukpKbMmJHMqace53VYEcnNpLASaC8ibYCdwGDgmmLHbAd6A6+IyMlAAhDiTYFy2D0KxlQKVUUVYmKEOnUSeO65/vz88yH+8pczbAI7F7mWFFQ1T0TGAgtxyk1nqup6EXkEWKWqqcCdwMsicjtO19JwVQ3tlkBZ8nKs8siYSvDjjwe5+eZ/c845Lbnzzp4ADB3axeOoooOr9yn47jmYX2zfQ36PNwBnuRlDUGX4Vlur194qj4w5CqrKzJlruPPODzhwIIfly9O5+eZuVKsW73VoUcPuaK5MVnlkzFHbsiWDkSPn8dFHWwEYMKA906ZdYgkhyCwpVKb9toaCMRWVn1/Ac8+t4P77PyIrK4+GDavz3HP9GTy4EyJWahpslhQqk7UUjDkqc+ZsJCsrj6uv7sTkyf1p1MgmsPOKJYXKZBPhGROQI0fyOXgwhwYNqhMbG8OMGcls2rSPgQOtQMNrVtdVWazyyJiArFy5k6SkFIYOfYfCYsOTTmpoCSFEWEuhsmR8B5pvlUfGlOLw4VzGj1/M008vp6BAOXw4l927D3HccTW9Ds34saRQWazryJhSLVmyjZEj55GWtp+YGOGuu85kwoTzqV7dKotCjSWFymJrKBjzB6rKbbctYMqUlQB07tyYGTOS6datmceRmdJYUqgsVnlkzB+ICLVrVyU+PoYHHujFvfeeTZUqsV6HZcpgSaGy7LN7FIwB2Lv3MJs376dHD2f20gcfPJchQ06lY8dGHkdmAhFQ9ZGIVBGRdm4HE7Z+V3lka8Ka6KSqvPXWN5x88gtceulsMjKyAEhIiLOEEEbKTQoiMgBYByzybSeKyDtuBxZWCiuPbLU1E6XS039h0KC3uPrqf7F372E6dmzE4cO5XodljkIg3UeP4KyYthhAVb+yVkMxhV1H9a3ryESXggJl+vTV3H33In75JYfatavy1FP9GDHiNJuiIkwFkhRyVTWz2C84fKe3dkNh5VFDG2Q20WXEiFReeeUrAJKTT2Tq1Itp1qx2Od9lQlkgYwobReRKIEZE2ojIs8Byl+MKL3aPgolS117bmcaNa/DWW5fz7rtXWUKIAIEkhbHA6UABMBfIBsa5GVTYsSU4TZT45pvdTJ7822fC3r1PYMuW27jqKpvRNFIE0n10oareA9xTuENELsNJEMa/8qiezd1iIlNOTh5/+9tnPPbYp+TmFpCU1JSzzmoJQI0aVTyOzlSmQFoKD5Sw7/7KDiRsZXzvVB7VOQHiq3kdjTGVbsWKdE4/PYUJEz4mN7eAMWOS6Nz5OK/DMi4ptaUgIhcC/YFmIvK031O1cbqSDPhNb2HjCSayHDp0hAcfXMyzzy5HFdq3r8/06cn06tXK69CMi8rqPtoNfIMzhrDeb/9B4F43gworVnlkItT993/E5MkriIkR7r77TB5++DxbGjMKlJoUVHUNsEZEZqlqdhBjCi82vYWJUPfffw7r1u3m8cf7kJTU1OtwTJAEMqbQTETeEpGvReT7wi/XIwsXNhGeiRCpqd9x8cWzyM3NB6BRoxp8+OEwSwhRJpCk8Arwd0CAi4C3gbdcjCl8WOWRiQC7dx9i8OA5DBr0FgsWpPHqq2u9Dsl4KJCkUF1VFwKo6mZVfQA4392wwoRVHpkwpqr84x9fc/LJLzB79nqqV49n8uT+XH99otehGQ8Fcp9Cjjh3pWwWkdHATqCxu2GFCas8MmFq+/YDjB79PgsWpAHQp88JpKRcQps29TyOzHgtkKRwO1ATuA34P6AOcIObQYUNG2Q2YeqDDzazYEEadesm8PTT/Rg+PNHuSDZAAElBVVf4Hh4EhgKISHM3gwobVo5qwsihQ0eK7j4eMeI0du78hVGjTuf442t5HJkJJWWOKYhINxG5VEQa+rZPEZHXsAnxHFZ5ZMJAXl4BkyYtpVWrZ9myJQNwlskcP/48SwjmD0pNCiLyN2AWMAT4j4jcj7OmwlqgQ3DCC2FWeWTCwNq1u+jRYzr33PNf9u3L4t13v/U6JBPiyuo+GgR0UdUsEakP/Ojb/i44oYW4wsqjuu2s8siEnJycPB599BMmTlxKXl4BLVvWISXlEi680NbHMmUrKylkq2oWgKruF5FvLSH4sUFmE6LWrPmJIUPmsnHjXkRg7NhuPPZYb2rVqup1aCYMlJUUThCRwumxBWjtt42qXlbei4tIf2AyEAtMV9WJJRxzJfAwzmpua1X1msDD95CVo5oQVbVqHJs3Z3DiiQ2YPj2Zs89u6XVIJoyUlRQuL7Y9pSIvLCKxwAtAXyAdWCkiqaq6we+Y9sB9wFmqmiEi4XP/g1UemRCyevVPnHZaE0SEjh0bsWDBEHr2bEFCQiBV58b8pqwJ8T48xtfuDqSp6hYAEXkLZ5xig98xI4EXVDXD9567j/E9g6ew+6i+dR8Z72RkZHHXXR8wc+ZXvPnm5Qwe3AmACy5o43FkJly5+TGiGbDDbzsd6FHsmA4AIrIUp4vpYVX9T/EXEpFRwCiAli1DoCmclwMZm5zKo/oneR2NiVLvvLORm2+ez65dv1K1aiz79h32OiQTAdxMCiXdHqklvH974DygOfCpiHRS1czffZNqCpACkJSUVPw1gs8qj4yHdu36lVtvXcCcOU5r9ayzWjB9ejInndTQ48hMJAg4KYhIVVXNqcBrpwMt/Lab45S1Fj9muarmAltF5DucJLGyAu8TfFZ5ZDzy5Zc/0rfv62RkZFOjRjwTJ/bh5pu7ERNjU1SYylHuLKki0l1E1gGbfNtdROT5AF57JdBeRNqISBVgMJBa7Jh38c246rtrugOwpQLxe8Mqj4xHOnZsRKNGNbjwwrasX38zY8d2t4RgKlUgU2c/B1wC7ANQ1bUEMHW2quYBY4GFwEbgbVVdLyKPiEiy77CFwD4R2YBzt/Tdqrqv4qcRZNZSMEFSUKCkpHxJZqaz+GG1avF88slwFiwYQqtWdT2OzkSiQLqPYlT1h2IzKOYH8uKqOh+YX2zfQ36PFbjD9xU+rKVgguC77/Zy443z+Oyz7axcuZOXX3Y+Sx13XE2PIzORLJCksENEugPqu/fgViB6l+MsrDxCrPLIuCI3N5+nnlrGww8vIScnnyZNanLRRe29DstEiUCSwhicLqSWwM/Af337olPmJl/lUVurPDKVbs2anxgxIpU1a3YBcP31iTz1VD/q1bP/ayY4AkkKeao62PVIwoVNl21csnnzfrp3n05eXgGtW9clJeUS+vZt63VYJsoEkhRW+kpFZwNzVfWgyzGFNhtPMC5p27Y+Q4eeSq1aVfi//+tNzZpVvA7JRKFyq49UtS3wKHA6sE5E3hWR6G05WOWRqSS//nqE225bwLJlv934P2NGMpMnX2QJwXgmkJJUVPVzVb0N6Ar8grP4TnSyloKpBAsXpnHKKVN5/vkvGD363ziFeNg6ycZz5XYfiUhNnInsBgMnA+8BPV2OKzTlH/GrPLLV1kzF7d+fxe23L+S119YCcPrpxzNjRrIlAxMyAhlT+AaYB0xS1U9djie0Fc151Bbiq3sdjQkzc+Zs4JZb5rN79yESEuKYMOE87rjjTOLiAmqwGxMUgSSFE1S1wPVIwoFVHpmjlJmZzahR88jIyKZXr1a8/PJAOnRo4HVYxvxBqUlBRJ5S1TuBf4nIH2YmDWTltYhjg8ymAlSVggIlNjaGunUTmDp1ABkZWdx0U5LNV2RCVlkthdm+fyu04lpEs0FmE6Bt2zIZNWoeF1zQhnvvPRugaAEcY0JZqZ2ZqvqF7+HJqvqh/xfOgHP0saRgypGfX8Bzz62gU6epLFq0hSlTviA7O8/rsIwJWCAjXDeUsG9EZQcS8qzyyJRj48Y99Or1CuPG/YdDh3IZPLgTq1ffZOskm7BS1pjCVThlqG1EZK7fU7WAzJK/K4JZ5ZEpRV5eAY8//hmPPPIJR47k07RpLV58cQDJyfbhwYSfsj7CfIGzhkJz4AW//QeBNW4GFZIKK4/q2yCz+b2YGOGDD7Zw5Eg+I0d2ZdKkvtStm+B1WMYclVKTgqpuBbbizIpqCiuPGtp4goGsrFwOHjxC48Y1iIkRpk8fyI4dv3DBBW28Ds2YY1LqmIKIfOz7N0NE9vt9ZYjI/uCFGCJskNn4fPLJD3TpMo1rr51bND1F+/YNLCGYiFBW91HhkpsNgxFIyLN7FKLeL7/kcN99/2Xq1FUAxMfHsnfvYRo1quFxZMZUnrJKUgvvYm4BxKpqPnAmcBMQXX8F+UecxXVstbWotWDBJjp1msrUqauIi4th/PhzWb16lCUEE3ECqZV7F+gmIm2B14B/A28Al7gZWEjJ+B4K8qzyKAqpKiNHzmPGDKe2IimpKTNnJtO583EeR2aMOwK5T6FAVXOBy4BnVfVWoJm7YYWYwq4jqzyKOiJC8+a1SUiI48kn+7Js2QhLCCaiBbQcp4j8GRgKXOrbF+9eSCGosBzVKo+iwo8/HmTz5v2cc04rAP7nf85h6NBTadu2vseRGeO+QO9oPh9n6uwtItIGeNPdsEJMUeWRtRQimaoyY8ZqOnZ8gcsvf5t9+w4DUKVKrCUEEzXKbSmo6jcichvQTkROAtJU9f/cDy2EFFUeWUshUm3ZksHIkfP46KOtAFxySQdyc23GeBN9All57RzgdWAnIEATERmqqkvdDi4kWOVRRCucwO6BBxZz+HAuDRtW57nn+jN4cCdbDc1EpUDGFJ4BLlbVDQAicjJOkkhyM7CQkbHJqTyqc4JVHkWgYcPe5Y031gFwzTWdefbZC63M1ES1QMYUqhQmBABV3QhUcS+kEGN3Mke0kSO70rx5bVJTBzNr1mWWEEzUC6SlsFpEXsJpHQAMIZomxLPKo4iycuVOPvpoK/fc4yx8c955rUlLu5WqVW16a2MgsKQwGrgN+CvOmMInwPNuBhVS9tv0FpHg8OFcxo9fzNNPL6egQOnZs0VRyaklBGN+U+Zfg4h0BtoC76jqpOCEFGL2WvdRuFuyZBs33pjK5s0ZxMQId911Jqef3tTrsIwJSWXNkvo/OFNcDAEWiUhJK7CVSUT6i8h3IpImIveWcdwVIqIiElqD11Z5FNYOHMjmppvmcf75r7J5cwadOzdm+fIRPPFEP6pXj677L40JVFkthSHAqap6SEQaAfOBmYG+sIjE4izO0xdIB1aKSKr/oLXvuFo43VMrKhq866zyKKw9+OBiUlJWEx8fw4MP9uKee86mSpVYr8MyJqSVlRRyVPUQgKruEZFAKpX8dce50W0LgIi8BQwCNhQ77n+BScBdFXx991nlUdhR1aL7Cx566Fy2bs1k4sTenHJKY48jMyY8lHWhP0FE5vq+3gHa+m3PLeP7CjUDdvhtp1NsIj0ROQ1ooarvVzjyYLA1FMKGqvLGG+u44ILXOHIkH4CGDaszb97VlhCMqYCyWgqXF9ueUsHXLul2UC160ml5PAMML/eFREYBowBatmxZwTCOwT4rRw0H6em/MGbMv3n//e8BmDXra66//jSPozImPJW1RvOHx/ja6TgL9BRqDvzot10L6AQs8TX3mwCpIpKsqquKxZICpAAkJSUpwWKVRyGtoEB5+eUvufvuRRw8eIQ6dary1FP9GD480evQjAlbbhZorwTa+2ZV3QkMBq4pfFJVD+C31KeILAHuKp4QPGOVRyEtLW0/I0fOY8mSbQAMGnQiU6cOoGnTWt4GZkyYcy0pqGqeiIwFFgKxwExVXS8ijwCrVDXVrfeuFFZ5FNI+/fQHlizZRuPGNZgy5SKuuKKjTWBnTCUIOCmISFVVzanIi6vqfJxSVv99D5Vy7HkVeW3X2RoKISczM5u6dRMAGD48kT17DjNixGk0aGBJ25jKUm6ZqYh0F5F1wCbfdhcRifxpLmwNhZCRk5PH+PGLadXqWTZt2gc4y2T+9a9nWUIwppIF0lJ4DrgE5+5mVHWtiJzvalShwCqPQsLy5emMGJHKhg17AFi4cDPt2zfwOCpjIlcgSSFGVX8o1l+b71I8ocPuUfDUoUNHePDBxTz77HJUoX37+syYkVw0iZ0xxh2BJIUdItIdUN/UFbcC37sblsfyj0DG9ziVRyd7HU3UWbEinWuumcuWLRnExgp33dWT8ePPpVo1m6/IGLcFkhTG4HQhtQR+Bv7r2xe5iiqP2ljlkQfq1k1g585f6NLlOGbMSLYZTY0JonKTgqruxrnHIHrYIHPQffbZds46qwUiwoknNuSjj66jW7emxMfbBHbGBFO5SUFEXsZveopCqjrKlYhCgU2EFzS7dx/ittsWMHv2el599VKGDesCQM+eLcr5TmOMGwLpPvqv3+ME4E/8fqK7yGP3KLhOVZk1ax3jxv2H/fuzqF49vmgiO2OMdwLpPprtvy0irwOLXIsoFBR2H1k5qiu2bz/A6NHvs2BBGgB9+55ASspAWreu63FkxpijmeaiDRC5dYFWeeSqFSvS6dPndX799Qh16ybwzDMXct11XWyKCmNCRCBjChn8NqYQA+wHSl1aM+xlplnlkYsSE5vQokVtTjqpIS+8cDHHH28T2BkTSspMCuJ8fOuCM8spQIGqBm/qai/YdNmVKi+vgClTvmDYsC7Ur1+NqlXjWLr0BurVq+Z1aMaYEpQ595EvAbyjqvm+r8hOCGCDzJVo7dpd9OgxndtvX8gddyws2m8JwZjQFci6y1+ISFfXIwkVdo/CMcvOzuOBBz4iKellVq/+iZYt63D11Z28DssYE4BSu49EJE5V84CzgZEishk4hLPMpqpqZCYKmwjvmHz++Q5GjEjl22/3IgJjx3bjscd6U6tWVa9DM8YEoKwxhS+ArsClQYrFe0WVR9hqa0chLW0/55zzdwoKlBNPbMCMGcmcdVYQ19Q2xhyzspKCAKjq5iDF4r3fVR7V8DqasNOuXX1GjepK/frVePDBc0lIcHO1V2OMG8r6q20kIneU9qSqPu1CPN6yyqMKycjI4s47P+D66xOLprSeOnWA3XNgTBgrKynEAjXxtRiigq2hELC5czdyyy3z2bXrV7788ie++uomRMQSgjFhrqyk8JOqPhK0SEKBTYRXrl27fmXs2Pn8618bATj77JZMnz7QkoExEaLcMYWoYvcolEpVee21tdx++0IyMrKpWbMKjz/eh9Gjk4iJib7/KsZEqrKSQu+gRREK8nN/qzxqYHMeFZeZmc2dd35ARkY2/fu3Y9q0AbRqZRPYGRNpSk0Kqro/mIF4LnOTVR4VU1CgFBQocXEx1KtXjZdeuoTDh3O59tpTrbvImAgVyB3N0WGvdR35+/bbvfTq9XcmTvysaN/ll3dk6FCb0dSYSGZJoZBNbwFAbm4+jz32KV26TGPp0h3MmLGG7Ow8r8MyxgSJ3V1UyCqPWLPmJ264IZWvvtoFwIgRp/HEE33tJjRjooj9tRfb6oM5AAAQbUlEQVSK4sqj3Nx8xo9fwqRJS8nPV1q3rsvLLw+kT58TvA7NGBNklhQg6iuP4uJiWLFiJwUFyrhxPXj00QuoWbOK12EZYzxgSQF+qzyq3TpqKo8OHszh4MEjNG1aCxFh+vSB7Nr1K2ee2cLr0IwxHrKBZvhtkDlKpsteuDCNTp1eZMiQuRSum9SmTT1LCMYYSwpA1EyEt2/fYa677l3695/F9u0HOHgwh337srwOyxgTQlxNCiLSX0S+E5E0Ebm3hOfvEJENIvK1iHwoIq3cjKdUET7IrKrMmbOBjh2n8tpra0lIiGPSpD4sX34jDRtW9zo8Y0wIcW1MQURigReAvkA6sFJEUlV1g99ha4AkVT0sImOAScBVbsVUqgi+R0FVGTJkLm+++Q0AvXq14uWXB9KhQwOPIzPGhCI3WwrdgTRV3aKqR4C3gEH+B6jqYlU97NtcDjR3MZ6SRXjlkYjQsWMjatWqwosvDmDx4ussIRhjSuVm9VEzYIffdjrQo4zjRwALSnpCREYBowBatqzk5R0zN0FBbkRVHm3dmsGWLRn07u3cZ3DPPWcxfHgizZvX9jgyY0yoc7OlUNIEOVrigSLXAknAEyU9r6opqpqkqkmNGjWqxBCJqMqj/PwCJk9eTqdOL3LVVXPYvfsQAPHxsZYQjDEBcbOlkA741zg2B34sfpCI9AHuB85V1RwX4ylZYeVR/fAeZN6wYQ833pjKsmXpACQnn2jrHBhjKszNpLASaC8ibYCdwGDgGv8DROQ04CWgv6rudjGW0oV5SyE3N5/HH1/K//7vJxw5kk/TprV48cUBJCef6HVoxpgw5FpSUNU8ERkLLMRZ73mmqq4XkUeAVaqaitNdVBP4p2865u2qmuxWTCUK84nwrrlmLnPmOIlt5MiuPPFEX+rUSfA4KmNMuHJ1mgtVnQ/ML7bvIb/Hfdx8/3L5Vx7VP8nTUI7WuHE9+OqrXbz00iVccEEbr8MxxoS56L6jOTPtt8qjKjW9jiYgH3+8jQkTlhRtn312SzZuvMUSgjGmUkT3hHiFXUdhMJ7wyy853HPPIqZN+xKA889vQ69ezg3gcXHRnduNMZUnupNCmFQezZ+/iZtuep/09F+Ij4/h/vvP4Ywzgn+fnzEm8kV3UgjxyqO9ew/zl7/8h1mz1gHQvXszZsxIplOnxh5HZoyJVFGeFEJ7IrxHHvmYWbPWUa1aHI8+egHjxvUgNta6iowx7onepPC7yqPQmfNIVfGV5zJhwnn8/PMhHnvsAtq2re9xZMaYaBC9HztDrPJIVXn55S/p2XMm2dl5ANSrV43Zs6+whGCMCZroTQoh1HW0efN+evd+jVGj3mf58nTefnu91yEZY6JU9HYfhcAaCs4Edit44IGPyMrKo1Gj6jz//EVceWVoDnwbYyJf9CaFvd62FNav380NN6TyxRc7ARgypDPPPtvfVkIzxngqepOCxzeurVmziy++2EmzZrV46aVLGDCggydxGGOMv+hMCh5VHu3Zc4hGjZyFfIYM6UxmZjZDh55qE9gZY0JGdA40F1UetQpK5dHhw7ncddcHtG49mY0b9wDOMpljx3a3hGCMCSnR2VII4iDz4sVbGTlyHps3ZxATI3zyyQ+cfHIlrx5njDGVJEqTgvtrKBw4kM1f/7qIlJTVAHTu3JiZMweRlNTUtfc0xphjFZ1JweXKo88+287gwXPYufMg8fExPPhgL+6552yqVIl15f2MMaayRGdS2O/uRHhNmtRk374szjijOdOnD+SUU2wCO2NMeIi+pJCfC/u/cx5XUuWRqrJo0Rb69j0BEaFdu/p89tn1JCY2sQnsjDFhJfquWJVcebRjxwEGDnyTCy/8B3//+1dF+08/vaklBGNM2Im+lkIlVR4VFDgT2N199yIOHjxCnTpVqVrVxgyMMeEtCpPCsQ8yb9q0j5Ej5/Hxxz8AcOmlJ/HCCxfTtGmtyojQGGM8E4VJ4dhaCp9/voPevV8jOzuPxo1rMGXKRVxxRceiNRCMMSacRWFSOLY5j5KSmtK+fX1OO+14nn66Hw0a2AR2xpjIEV1J4Sgqj3Jy8njyyc+56aYkGjasTpUqsSxdegO1alV1MVBjjPFGdCWFzM0VqjxavjydESNS2bBhDxs37uUf/7gMwBKCMSZiRVdSCHCQ+dChIzzwwEdMnrwCVejQoQE33XR6EAI0xhhvRWlSKH084cMPtzBy5Dy2bs0kNla4++6ejB9/HgkJ0fWjMsZEp+i60pVTefT99/vo2/d1VCExsQkzZiTTtevxQQzQGGO8FWVJoezuow4dGjBuXA8aNarB3Xf3JD7ebkYzxkSX6EkK/pVHvqTw88+/cttt/2H06NM5//w2ADzzTH+vIjTGGM+5OjmPiPQXke9EJE1E7i3h+aoiMtv3/AoRae1aMH6VRxpfg9dfX0vHjlN5++313H77QlTVtbc2xphw4VpSEJFY4AXgIqAjcLWIFO+3GQFkqGo74BngcbfiKew62l6QyIABbzBs2Lvs359Fv35teffdwXZHsjHG4G73UXcgTVW3AIjIW8AgYIPfMYOAh32P5wBTRETUhY/tBXvWM21pN+75TyK/ZqVRr14CzzxzIcOGdbGEYIwxPm4mhWbADr/tdKBHaceoap6IHAAaAHsrO5gDO75lwqJz+TVLuPzyk5ky5WKaNDn2qbONMSaSuJkUSvr4XbwFEMgxiMgoYBRAy5YtjyqYejWV6YMXcqTHo1x+Y/JRvYYxxkQ6N5NCOtDCb7s58GMpx6SLSBxQB9hf/IVUNQVIAUhKSjq6rqVL3mTggIKj+lZjjIkWblYfrQTai0gbEakCDAZSix2TClzne3wF8JEb4wlFJMb5MsYYUyLXWgq+MYKxwEIgFpipqutF5BFglaqmAjOA10UkDaeFMNiteIwxxpTP1ZvXVHU+ML/Yvof8HmcDf3YzBmOMMYGzvhRjjDFFLCkYY4wpYknBGGNMEUsKxhhjilhSMMYYU0TCbXZQEdkD/HCU394QF6bQCHF2ztHBzjk6HMs5t1LVRuUdFHZJ4ViIyCpVTfI6jmCyc44Ods7RIRjnbN1HxhhjilhSMMYYUyTakkKK1wF4wM45Otg5RwfXzzmqxhSMMcaULdpaCsYYY8oQkUlBRPqLyHcikiYi95bwfFURme17foWItA5+lJUrgHO+Q0Q2iMjXIvKhiLTyIs7KVN45+x13hYioiIR9pUog5ywiV/p+1+tF5I1gx1jZAvi/3VJEFovIGt//74u9iLOyiMhMEdktIt+U8ryIyHO+n8fXItK1UgNQ1Yj6wpmmezNwAlAFWAt0LHbMzcA03+PBwGyv4w7COZ8PVPc9HhMN5+w7rhbwCbAcSPI67iD8ntsDa4B6vu3GXscdhHNOAcb4HncEtnkd9zGecy+gK/BNKc9fDCzAWbnyDGBFZb5/JLYUugNpqrpFVY8AbwGDih0zCHjV93gO0FtESloaNFyUe86qulhVD/s2l+OshBfOAvk9A/wvMAnIDmZwLgnknEcCL6hqBoCq7g5yjJUtkHNWoLbvcR3+uMJjWFHVTyhhBUo/g4DX1LEcqCsix1fW+0diUmgG7PDbTvftK/EYVc0DDgANghKdOwI5Z38jcD5phLNyz1lETgNaqOr7wQzMRYH8njsAHURkqYgsF5H+QYvOHYGc88PAtSKSjrN+y63BCc0zFf17rxBXF9nxSEmf+IuXWAVyTDgJ+HxE5FogCTjX1YjcV+Y5i0gM8AwwPFgBBUEgv+c4nC6k83Bag5+KSCdVzXQ5NrcEcs5XA6+o6lMicibOao6dVDVSF2V39foViS2FdKCF33Zz/ticLDpGROJwmpxlNddCXSDnjIj0Ae4HklU1J0ixuaW8c64FdAKWiMg2nL7X1DAfbA70//Z7qpqrqluB73CSRLgK5JxHAG8DqOoyIAFnjqBIFdDf+9GKxKSwEmgvIm1EpArOQHJqsWNSget8j68APlLfCE6YKvecfV0pL+EkhHDvZ4ZyzllVD6hqQ1VtraqtccZRklV1lTfhVopA/m+/i1NUgIg0xOlO2hLUKCtXIOe8HegNICIn4ySFPUGNMrhSgWG+KqQzgAOq+lNlvXjEdR+pap6IjAUW4lQuzFTV9SLyCLBKVVOBGThNzDScFsJg7yI+dgGe8xNATeCfvjH17aqa7FnQxyjAc44oAZ7zQqCfiGwA8oG7VXWfd1EfmwDP+U7gZRG5HacbZXg4f8gTkTdxuv8a+sZJxgPxAKo6DWfc5GIgDTgMXF+p7x/GPztjjDGVLBK7j4wxxhwlSwrGGGOKWFIwxhhTxJKCMcaYIpYUjDHGFLGkYEKOiOSLyFd+X63LOLZ1abNJVvA9l/hm4lzrmyLixKN4jdEiMsz3eLiINPV7brqIdKzkOFeKSGIA3/MXEal+rO9tooMlBROKslQ10e9rW5Ded4iqdsGZLPGJin6zqk5T1dd8m8OBpn7P3aiqGyolyt/inEpgcf4FsKRgAmJJwYQFX4vgUxFZ7fvqWcIxp4jIF77Wxdci0t63/1q//S+JSGw5b/cJ0M73vb198/Sv881zX9W3f6L8tj7Fk759D4vIXSJyBc78UrN871nN9wk/SUTGiMgkv5iHi8jzRxnnMvwmQhORF0VklTjrKEzw7bsNJzktFpHFvn39RGSZ7+f4TxGpWc77mChiScGEomp+XUfv+PbtBvqqalfgKuC5Er5vNDBZVRNxLsrpvmkPrgLO8u3PB4aU8/4DgXUikgC8Alylqp1xZgAYIyL1gT8Bp6jqqcCj/t+sqnOAVTif6BNVNcvv6TnAZX7bVwGzjzLO/jjTWhS6X1WTgFOBc0XkVFV9DmdenPNV9Xzf1BcPAH18P8tVwB3lvI+JIhE3zYWJCFm+C6O/eGCKrw89H2dOn+KWAfeLSHNgrqpuEpHewOnASt/0HtVwEkxJZolIFrANZ/rlE4Gtqvq97/lXgVuAKTjrM0wXkX8DAU/Nrap7RGSLb86aTb73WOp73YrEWQNn2gf/VbeuFJFROH/Xx+MsOPN1se89w7d/qe99quD83IwBLCmY8HE78DPQBaeF+4dFc1T1DRFZAQwAForIjTjTDL+qqvcF8B5D/CfME5ES19jwzcfTHWcStsHAWOCCCpzLbOBK4FvgHVVVca7QAceJswLZROAF4DIRaQPcBXRT1QwReQVnYrjiBFikqldXIF4TRaz7yISLOsBPvjnyh+J8Sv4dETkB2OLrMknF6Ub5ELhCRBr7jqkvga9P/S3QWkTa+baHAh/7+uDrqOp8nEHckiqADuJM312SucClOOsAzPbtq1CcqpqL0w10hq/rqTZwCDggIscBF5USy3LgrMJzEpHqIlJSq8tEKUsKJlxMBa4TkeU4XUeHSjjmKuAbEfkKOAlnycINOBfPD0Tka2ARTtdKuVQ1G2cGyn+KyDqgAJiGc4F93/d6H+O0Yop7BZhWONBc7HUzgA1AK1X9wrevwnH6xiqeAu5S1bU4azOvB2bidEkVSgEWiMhiVd2DUxn1pu99luP8rIwBbJZUY4wxfqylYIwxpoglBWOMMUUsKRhjjCliScEYY0wRSwrGGGOKWFIwxhhTxJKCMcaYIpYUjDHGFPl/zB2bZgYa6voAAAAASUVORK5CYII=\n",
        "text/plain": "<Figure size 432x288 with 1 Axes>"
       },
       "metadata": {
        "needs_background": "light"
       },
       "output_type": "display_data"
      }
     ],
     "persistentId": "2c810e70-a822-4d9a-a9b0-2622dfa247bf",
     "text": "y_true = coding_scores_not_low_complexity_unspliced['in_cds_no_stop_codon']\ny_score = coding_scores_not_low_complexity_unspliced['jaccard_in_peptide_db']\ny_pred = y_score > 0.5\n\nfpr, tpr, thresholds = metrics.roc_curve(y_true, y_score)\n\nfig, ax = plt.subplots()\nlw = 2\nax.plot(fpr, tpr, lw=2, color='darkorange')\nax.plot([0, 1], [0, 1], color='navy', lw=lw, linestyle='--')\nax.set(xlabel='False Positive Rate', ylabel='True Positive Rate')\n\nprint(\"accuracy score:\", metrics.accuracy_score(y_true, y_pred))\n\nprint(metrics.classification_report(y_true, y_pred))"
    },
    "executionTime": "2019-10-17T07:57:20.822Z"
   },
   {
    "cell": {
     "executionCount": 41,
     "executionEventId": "43550586-465e-4837-95b2-714fda553819",
     "hasError": false,
     "id": "ddfff1e6-fd17-4da4-aee8-e95d812c866f",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "AUC: 0.9332425339782375\naccuracy score: 0.8768244717880631\n              precision    recall  f1-score   support\n\n       False       1.00      0.86      0.92    590637\n        True       0.52      0.97      0.68     90245\n\n   micro avg       0.88      0.88      0.88    680882\n   macro avg       0.76      0.92      0.80    680882\nweighted avg       0.93      0.88      0.89    680882\n\n"
      },
      {
       "data": {
        "image/png": "iVBORw0KGgoAAAANSUhEUgAAAYUAAAEKCAYAAAD9xUlFAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDMuMC4zLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvnQurowAAIABJREFUeJzt3Xl8FFW2wPHfyQJh3xHZkUVFkIgBFBUXFlEkOOooiiCKICjKuI36XBCfz0FcUUSMwLgMKg6DGhwYZBRcEBAEEQGVsAhBkS1BhCRkOe+P6sQ2ZulAqquX8/188qGrutJ9KiF1+t576l5RVYwxxhiAGK8DMMYYEzosKRhjjCliScEYY0wRSwrGGGOKWFIwxhhTxJKCMcaYIpYUjDHGFLGkYIwxpoglBWOMMUXivA6goho2bKitW7f2OgxjjAkrX3755V5VbVTecWGXFFq3bs2qVau8DsMYY8KKiPwQyHHWfWSMMaaIJQVjjDFFLCkYY4wpYknBGGNMEUsKxhhjiriWFERkpojsFpFvSnleROQ5EUkTka9FpKtbsRhjjAmMmy2FV4D+ZTx/EdDe9zUKeNHFWIwxxgTAtfsUVPUTEWldxiGDgNfUWQ90uYjUFZHjVfUnt2IyxpgKU4WCXMg/8tu//o8LStlf+LiglP2F35vvd0wpz33xXRUSYo9w6tk9oOd4V0/Xy5vXmgE7/LbTffv+kBREZBROa4KWLVsGJThjjAu0wHehK+WiGMiFsyIX5jK/N9DXzfPux6Vwz7/78tTH3Tj1+J/5ov0G4l1+Ty+TgpSwT0s6UFVTgBSApKSkEo8xJuoUXWAD+7QZ/Iuu/zGFn5q9u8Aek5g4iKkCsVUgJt751/9xTBWIjS/lmLKeK2N/bBUkpgp8vws++Zl+yd3J794jopNCOtDCb7s58KNHsZhQoupc8Py/KHyspewvfnxJr6G/vUZJjwN6vnB/fukXvmBddDXfq9/QsSm6kAZyQa2Ei27hMSXtD+DCTEw8SEmfYd2RmZnNli0ZdO16PAATJucyePTeom23eZkUUoGxIvIW0AM4YOMJflQhOwMO74KsvZCdCfnZkJcFeX7/5mf/9gmsIM+5UBTkO/s0v9i+PN9+vwtqQb7fhbO0xyVt+18k8397D82nxAtySRfr3x3nd+E1gQvkwhesi25AF+a4oF5gw817733LmDH/JiZGWL/+ZurUSaBatfigJQRwMSmIyJvAeUBDEUkHxoPT8lHVacB84GIgDTgMXO9WLGEjPxd2rYRv34RNc+DQLq8j8pBATCxIjPNFjHMxKdwu2lfCF8WOE/E7Vvye93tc6r5ijwufi4nz/qJrF9iIsXv3IW67bQGzZ68H4IwzmpOZmU2dOglBj8XN6qOry3legVvcev+wcnAnrJwE377htAoKxdeEmk2hWiNIqAuxCRBfHWKrQlw1Zzuumu9iEQ8S67uQxvpdNIrviwWJ+/1FMyaWootm4YWYwv3+F2f/i3ThxZLfv4f/Mf7HFr5WaRfq372uXehMdFBVZs1ax7hx/2H//iyqV4/nsccuYOzY7sTGenNvcdhNnR1Rcg/Dskdg9bOQn+Psq9sO2lwMp1wHjRN/u/AaYyLOmDH/5qWXvgSgT58TSEm5hDZt6nkakyUFL6g6XUSf3gcHtzv72v0Jut8LTbrZJ2VjosSll57E7Nnreeqpflx/fSISAn/7lhSCTRU+vdfpLgJolAh9XoSmZ3gblzHGdZs27ePDD7cyenQSAP37t2PbtnGejB2UxpJCMGkBfDQOvpri9L33fgE63+jrbzfGRKq8vAKefnoZ48cvIScnj8TEJpxxRnOAkEoIYEkheAryIPUK2PyeMwA88F/QLtnrqIwxLlu7dhcjRqTy5ZdOxf2wYV1o376+x1GVzpJCMKjCopuchFC1LgycA616ex2VMcZFOTl5PProJ0ycuJS8vAJatqzDSy9dQv/+7bwOrUyWFIJh8zz4ZqZTPnrZfGh6ptcRGWNcdt99H/LMM8sBuOWWbvztb72pVauqx1GVz5KC21Thvzc5j8961BKCMVHir389i2XL0pk0qQ/nnNPK63ACZkXwbtu33rkzObYqnDbW62iMMS5ZtGgzl1/+Nnl5BQA0aVKTzz+/IawSAlhScN+er51/G3VxpicwxkSUjIwsRox4j379/sHcuRv5+9/XFD0XCvcdVJR1H7lt3wbn39b9vI3DGFPp3nlnIzffPJ9du36latVYxo8/l+HDE70O65hYUnDbPmeCK+p39DYOY0yl2bXrV269dQFz5jgf+nr2bMGMGcmcdFJDjyM7dpYU3FaYFBqe4m0cxphK89573zJnzgZq1Ihn4sQ+3HxzN2Jiwq+rqCSWFNyUlw2Zm51J7eqd6HU0xphjkJ2dR0KCc8kcOfJ0tmzJYMyYbrRuXdfjyCqXDTS7af93ztQWddtBXOjXJxtj/qigQJky5QvatJnMDz9kAhATIzz+eN+ISwhgScFdhV1HDazryJhw9N13e+nV6+/ceusCdu36lTff/MbrkFxn3UduKqw8svEEY8JKbm4+Tz75ORMmfExOTj7HHVeDqVMHcNllJ3sdmussKbjJKo+MCTvffLObYcPeYc0aZznc669P5Kmn+lGvXjWPIwsOSwpuspaCMWGnoEBZt243rVrVISVlIP36tfU6pKCypOCWvGzITPNVHnXwOhpjTBnWr99Nx46NEBFOPfU43ntvML16taJmzeibhcAGmt3yu8qj0FpEwxjjOHgwh7Fj59Op04v8618bi/ZffHH7qEwIYC0F9xR2HVnlkTEhaeHCNEaNep/t2w8QFxfDtm2ZXocUEiwpuKWoHNUGmY0JJfv3Z3H77Qt57bW1AHTtejwzZiSTmNjE48hCgyUFt9g9CsaEnK++2kX//v/g558PUbVqLBMmnMedd/YkLs560gtZUnBLUfeRtRSMCRUdOjSgZs0qdOjQgOnTk+nQoYHXIYUcSwpu8K88qm9zHhnjFVXljTfWMXDgidSuXZXq1eNZsmQ4TZvWipgJ7CqbtZnckPG9VR4Z47Ft2zK58MJ/cO2173Dvvf8t2t+8eW1LCGWwloIb9togszFeyc8vYOrUldx334ccOpRL/frV6NmzhddhhQ1LCm6wQWZjPLFx4x5GjEhl2bJ0AK688hSef/4iGjeu4XFk4cOSghvsHgVjgm7r1gwSE1/iyJF8jj++JlOnDuDSS0/yOqyw42pSEJH+wGQgFpiuqhOLPd8SeBWo6zvmXlWd72ZMQWH3KBgTdG3a1OPPf+5IQkIcTz7Zj7p1bTzvaLiWFEQkFngB6AukAytFJFVVN/gd9gDwtqq+KCIdgflAa7diCgqrPDImKLKycnnkkY/5059Opnv3ZgC8+uqlxMZa/cyxcPOn1x1IU9UtqnoEeAsYVOwYBWr7HtcBfnQxnuAoqjxqa5VHxrjk009/IDHxJSZOXMqoUfMoKFAASwiVwM3uo2bADr/tdKBHsWMeBj4QkVuBGkAfF+MJjr02yGyMW375JYf77vsvU6euAqBjx0ZMm3aJlZhWIjfTakm/JS22fTXwiqo2By4GXheRP8QkIqNEZJWIrNqzZ48LoVai/TbIbIwb5s/fRKdOU5k6dRVxcTE89FAvVq8exRlnNPc6tIjiZkshHfAvDm7OH7uHRgD9AVR1mYgkAA2B3f4HqWoKkAKQlJRUPLGEFrtHwZhKd+BANkOGzCUzM5ukpKbMmJHMqace53VYEcnNpLASaC8ibYCdwGDgmmLHbAd6A6+IyMlAAhDiTYFy2D0KxlQKVUUVYmKEOnUSeO65/vz88yH+8pczbAI7F7mWFFQ1T0TGAgtxyk1nqup6EXkEWKWqqcCdwMsicjtO19JwVQ3tlkBZ8nKs8siYSvDjjwe5+eZ/c845Lbnzzp4ADB3axeOoooOr9yn47jmYX2zfQ36PNwBnuRlDUGX4Vlur194qj4w5CqrKzJlruPPODzhwIIfly9O5+eZuVKsW73VoUcPuaK5MVnlkzFHbsiWDkSPn8dFHWwEYMKA906ZdYgkhyCwpVKb9toaCMRWVn1/Ac8+t4P77PyIrK4+GDavz3HP9GTy4EyJWahpslhQqk7UUjDkqc+ZsJCsrj6uv7sTkyf1p1MgmsPOKJYXKZBPhGROQI0fyOXgwhwYNqhMbG8OMGcls2rSPgQOtQMNrVtdVWazyyJiArFy5k6SkFIYOfYfCYsOTTmpoCSFEWEuhsmR8B5pvlUfGlOLw4VzGj1/M008vp6BAOXw4l927D3HccTW9Ds34saRQWazryJhSLVmyjZEj55GWtp+YGOGuu85kwoTzqV7dKotCjSWFymJrKBjzB6rKbbctYMqUlQB07tyYGTOS6datmceRmdJYUqgsVnlkzB+ICLVrVyU+PoYHHujFvfeeTZUqsV6HZcpgSaGy7LN7FIwB2Lv3MJs376dHD2f20gcfPJchQ06lY8dGHkdmAhFQ9ZGIVBGRdm4HE7Z+V3lka8Ka6KSqvPXWN5x88gtceulsMjKyAEhIiLOEEEbKTQoiMgBYByzybSeKyDtuBxZWCiuPbLU1E6XS039h0KC3uPrqf7F372E6dmzE4cO5XodljkIg3UeP4KyYthhAVb+yVkMxhV1H9a3ryESXggJl+vTV3H33In75JYfatavy1FP9GDHiNJuiIkwFkhRyVTWz2C84fKe3dkNh5VFDG2Q20WXEiFReeeUrAJKTT2Tq1Itp1qx2Od9lQlkgYwobReRKIEZE2ojIs8Byl+MKL3aPgolS117bmcaNa/DWW5fz7rtXWUKIAIEkhbHA6UABMBfIBsa5GVTYsSU4TZT45pvdTJ7822fC3r1PYMuW27jqKpvRNFIE0n10oareA9xTuENELsNJEMa/8qiezd1iIlNOTh5/+9tnPPbYp+TmFpCU1JSzzmoJQI0aVTyOzlSmQFoKD5Sw7/7KDiRsZXzvVB7VOQHiq3kdjTGVbsWKdE4/PYUJEz4mN7eAMWOS6Nz5OK/DMi4ptaUgIhcC/YFmIvK031O1cbqSDPhNb2HjCSayHDp0hAcfXMyzzy5HFdq3r8/06cn06tXK69CMi8rqPtoNfIMzhrDeb/9B4F43gworVnlkItT993/E5MkriIkR7r77TB5++DxbGjMKlJoUVHUNsEZEZqlqdhBjCi82vYWJUPfffw7r1u3m8cf7kJTU1OtwTJAEMqbQTETeEpGvReT7wi/XIwsXNhGeiRCpqd9x8cWzyM3NB6BRoxp8+OEwSwhRJpCk8Arwd0CAi4C3gbdcjCl8WOWRiQC7dx9i8OA5DBr0FgsWpPHqq2u9Dsl4KJCkUF1VFwKo6mZVfQA4392wwoRVHpkwpqr84x9fc/LJLzB79nqqV49n8uT+XH99otehGQ8Fcp9Cjjh3pWwWkdHATqCxu2GFCas8MmFq+/YDjB79PgsWpAHQp88JpKRcQps29TyOzHgtkKRwO1ATuA34P6AOcIObQYUNG2Q2YeqDDzazYEEadesm8PTT/Rg+PNHuSDZAAElBVVf4Hh4EhgKISHM3gwobVo5qwsihQ0eK7j4eMeI0du78hVGjTuf442t5HJkJJWWOKYhINxG5VEQa+rZPEZHXsAnxHFZ5ZMJAXl4BkyYtpVWrZ9myJQNwlskcP/48SwjmD0pNCiLyN2AWMAT4j4jcj7OmwlqgQ3DCC2FWeWTCwNq1u+jRYzr33PNf9u3L4t13v/U6JBPiyuo+GgR0UdUsEakP/Ojb/i44oYW4wsqjuu2s8siEnJycPB599BMmTlxKXl4BLVvWISXlEi680NbHMmUrKylkq2oWgKruF5FvLSH4sUFmE6LWrPmJIUPmsnHjXkRg7NhuPPZYb2rVqup1aCYMlJUUThCRwumxBWjtt42qXlbei4tIf2AyEAtMV9WJJRxzJfAwzmpua1X1msDD95CVo5oQVbVqHJs3Z3DiiQ2YPj2Zs89u6XVIJoyUlRQuL7Y9pSIvLCKxwAtAXyAdWCkiqaq6we+Y9sB9wFmqmiEi4XP/g1UemRCyevVPnHZaE0SEjh0bsWDBEHr2bEFCQiBV58b8pqwJ8T48xtfuDqSp6hYAEXkLZ5xig98xI4EXVDXD9567j/E9g6ew+6i+dR8Z72RkZHHXXR8wc+ZXvPnm5Qwe3AmACy5o43FkJly5+TGiGbDDbzsd6FHsmA4AIrIUp4vpYVX9T/EXEpFRwCiAli1DoCmclwMZm5zKo/oneR2NiVLvvLORm2+ez65dv1K1aiz79h32OiQTAdxMCiXdHqklvH974DygOfCpiHRS1czffZNqCpACkJSUVPw1gs8qj4yHdu36lVtvXcCcOU5r9ayzWjB9ejInndTQ48hMJAg4KYhIVVXNqcBrpwMt/Lab45S1Fj9muarmAltF5DucJLGyAu8TfFZ5ZDzy5Zc/0rfv62RkZFOjRjwTJ/bh5pu7ERNjU1SYylHuLKki0l1E1gGbfNtdROT5AF57JdBeRNqISBVgMJBa7Jh38c246rtrugOwpQLxe8Mqj4xHOnZsRKNGNbjwwrasX38zY8d2t4RgKlUgU2c/B1wC7ANQ1bUEMHW2quYBY4GFwEbgbVVdLyKPiEiy77CFwD4R2YBzt/Tdqrqv4qcRZNZSMEFSUKCkpHxJZqaz+GG1avF88slwFiwYQqtWdT2OzkSiQLqPYlT1h2IzKOYH8uKqOh+YX2zfQ36PFbjD9xU+rKVgguC77/Zy443z+Oyz7axcuZOXX3Y+Sx13XE2PIzORLJCksENEugPqu/fgViB6l+MsrDxCrPLIuCI3N5+nnlrGww8vIScnnyZNanLRRe29DstEiUCSwhicLqSWwM/Af337olPmJl/lUVurPDKVbs2anxgxIpU1a3YBcP31iTz1VD/q1bP/ayY4AkkKeao62PVIwoVNl21csnnzfrp3n05eXgGtW9clJeUS+vZt63VYJsoEkhRW+kpFZwNzVfWgyzGFNhtPMC5p27Y+Q4eeSq1aVfi//+tNzZpVvA7JRKFyq49UtS3wKHA6sE5E3hWR6G05WOWRqSS//nqE225bwLJlv934P2NGMpMnX2QJwXgmkJJUVPVzVb0N6Ar8grP4TnSyloKpBAsXpnHKKVN5/vkvGD363ziFeNg6ycZz5XYfiUhNnInsBgMnA+8BPV2OKzTlH/GrPLLV1kzF7d+fxe23L+S119YCcPrpxzNjRrIlAxMyAhlT+AaYB0xS1U9djie0Fc151Bbiq3sdjQkzc+Zs4JZb5rN79yESEuKYMOE87rjjTOLiAmqwGxMUgSSFE1S1wPVIwoFVHpmjlJmZzahR88jIyKZXr1a8/PJAOnRo4HVYxvxBqUlBRJ5S1TuBf4nIH2YmDWTltYhjg8ymAlSVggIlNjaGunUTmDp1ABkZWdx0U5LNV2RCVlkthdm+fyu04lpEs0FmE6Bt2zIZNWoeF1zQhnvvPRugaAEcY0JZqZ2ZqvqF7+HJqvqh/xfOgHP0saRgypGfX8Bzz62gU6epLFq0hSlTviA7O8/rsIwJWCAjXDeUsG9EZQcS8qzyyJRj48Y99Or1CuPG/YdDh3IZPLgTq1ffZOskm7BS1pjCVThlqG1EZK7fU7WAzJK/K4JZ5ZEpRV5eAY8//hmPPPIJR47k07RpLV58cQDJyfbhwYSfsj7CfIGzhkJz4AW//QeBNW4GFZIKK4/q2yCz+b2YGOGDD7Zw5Eg+I0d2ZdKkvtStm+B1WMYclVKTgqpuBbbizIpqCiuPGtp4goGsrFwOHjxC48Y1iIkRpk8fyI4dv3DBBW28Ds2YY1LqmIKIfOz7N0NE9vt9ZYjI/uCFGCJskNn4fPLJD3TpMo1rr51bND1F+/YNLCGYiFBW91HhkpsNgxFIyLN7FKLeL7/kcN99/2Xq1FUAxMfHsnfvYRo1quFxZMZUnrJKUgvvYm4BxKpqPnAmcBMQXX8F+UecxXVstbWotWDBJjp1msrUqauIi4th/PhzWb16lCUEE3ECqZV7F+gmIm2B14B/A28Al7gZWEjJ+B4K8qzyKAqpKiNHzmPGDKe2IimpKTNnJtO583EeR2aMOwK5T6FAVXOBy4BnVfVWoJm7YYWYwq4jqzyKOiJC8+a1SUiI48kn+7Js2QhLCCaiBbQcp4j8GRgKXOrbF+9eSCGosBzVKo+iwo8/HmTz5v2cc04rAP7nf85h6NBTadu2vseRGeO+QO9oPh9n6uwtItIGeNPdsEJMUeWRtRQimaoyY8ZqOnZ8gcsvf5t9+w4DUKVKrCUEEzXKbSmo6jcichvQTkROAtJU9f/cDy2EFFUeWUshUm3ZksHIkfP46KOtAFxySQdyc23GeBN9All57RzgdWAnIEATERmqqkvdDi4kWOVRRCucwO6BBxZz+HAuDRtW57nn+jN4cCdbDc1EpUDGFJ4BLlbVDQAicjJOkkhyM7CQkbHJqTyqc4JVHkWgYcPe5Y031gFwzTWdefbZC63M1ES1QMYUqhQmBABV3QhUcS+kEGN3Mke0kSO70rx5bVJTBzNr1mWWEEzUC6SlsFpEXsJpHQAMIZomxLPKo4iycuVOPvpoK/fc4yx8c955rUlLu5WqVW16a2MgsKQwGrgN+CvOmMInwPNuBhVS9tv0FpHg8OFcxo9fzNNPL6egQOnZs0VRyaklBGN+U+Zfg4h0BtoC76jqpOCEFGL2WvdRuFuyZBs33pjK5s0ZxMQId911Jqef3tTrsIwJSWXNkvo/OFNcDAEWiUhJK7CVSUT6i8h3IpImIveWcdwVIqIiElqD11Z5FNYOHMjmppvmcf75r7J5cwadOzdm+fIRPPFEP6pXj677L40JVFkthSHAqap6SEQaAfOBmYG+sIjE4izO0xdIB1aKSKr/oLXvuFo43VMrKhq866zyKKw9+OBiUlJWEx8fw4MP9uKee86mSpVYr8MyJqSVlRRyVPUQgKruEZFAKpX8dce50W0LgIi8BQwCNhQ77n+BScBdFXx991nlUdhR1aL7Cx566Fy2bs1k4sTenHJKY48jMyY8lHWhP0FE5vq+3gHa+m3PLeP7CjUDdvhtp1NsIj0ROQ1ooarvVzjyYLA1FMKGqvLGG+u44ILXOHIkH4CGDaszb97VlhCMqYCyWgqXF9ueUsHXLul2UC160ml5PAMML/eFREYBowBatmxZwTCOwT4rRw0H6em/MGbMv3n//e8BmDXra66//jSPozImPJW1RvOHx/ja6TgL9BRqDvzot10L6AQs8TX3mwCpIpKsqquKxZICpAAkJSUpwWKVRyGtoEB5+eUvufvuRRw8eIQ6dary1FP9GD480evQjAlbbhZorwTa+2ZV3QkMBq4pfFJVD+C31KeILAHuKp4QPGOVRyEtLW0/I0fOY8mSbQAMGnQiU6cOoGnTWt4GZkyYcy0pqGqeiIwFFgKxwExVXS8ijwCrVDXVrfeuFFZ5FNI+/fQHlizZRuPGNZgy5SKuuKKjTWBnTCUIOCmISFVVzanIi6vqfJxSVv99D5Vy7HkVeW3X2RoKISczM5u6dRMAGD48kT17DjNixGk0aGBJ25jKUm6ZqYh0F5F1wCbfdhcRifxpLmwNhZCRk5PH+PGLadXqWTZt2gc4y2T+9a9nWUIwppIF0lJ4DrgE5+5mVHWtiJzvalShwCqPQsLy5emMGJHKhg17AFi4cDPt2zfwOCpjIlcgSSFGVX8o1l+b71I8ocPuUfDUoUNHePDBxTz77HJUoX37+syYkVw0iZ0xxh2BJIUdItIdUN/UFbcC37sblsfyj0DG9ziVRyd7HU3UWbEinWuumcuWLRnExgp33dWT8ePPpVo1m6/IGLcFkhTG4HQhtQR+Bv7r2xe5iiqP2ljlkQfq1k1g585f6NLlOGbMSLYZTY0JonKTgqruxrnHIHrYIHPQffbZds46qwUiwoknNuSjj66jW7emxMfbBHbGBFO5SUFEXsZveopCqjrKlYhCgU2EFzS7dx/ittsWMHv2el599VKGDesCQM+eLcr5TmOMGwLpPvqv3+ME4E/8fqK7yGP3KLhOVZk1ax3jxv2H/fuzqF49vmgiO2OMdwLpPprtvy0irwOLXIsoFBR2H1k5qiu2bz/A6NHvs2BBGgB9+55ASspAWreu63FkxpijmeaiDRC5dYFWeeSqFSvS6dPndX799Qh16ybwzDMXct11XWyKCmNCRCBjChn8NqYQA+wHSl1aM+xlplnlkYsSE5vQokVtTjqpIS+8cDHHH28T2BkTSspMCuJ8fOuCM8spQIGqBm/qai/YdNmVKi+vgClTvmDYsC7Ur1+NqlXjWLr0BurVq+Z1aMaYEpQ595EvAbyjqvm+r8hOCGCDzJVo7dpd9OgxndtvX8gddyws2m8JwZjQFci6y1+ISFfXIwkVdo/CMcvOzuOBBz4iKellVq/+iZYt63D11Z28DssYE4BSu49EJE5V84CzgZEishk4hLPMpqpqZCYKmwjvmHz++Q5GjEjl22/3IgJjx3bjscd6U6tWVa9DM8YEoKwxhS+ArsClQYrFe0WVR9hqa0chLW0/55zzdwoKlBNPbMCMGcmcdVYQ19Q2xhyzspKCAKjq5iDF4r3fVR7V8DqasNOuXX1GjepK/frVePDBc0lIcHO1V2OMG8r6q20kIneU9qSqPu1CPN6yyqMKycjI4s47P+D66xOLprSeOnWA3XNgTBgrKynEAjXxtRiigq2hELC5czdyyy3z2bXrV7788ie++uomRMQSgjFhrqyk8JOqPhK0SEKBTYRXrl27fmXs2Pn8618bATj77JZMnz7QkoExEaLcMYWoYvcolEpVee21tdx++0IyMrKpWbMKjz/eh9Gjk4iJib7/KsZEqrKSQu+gRREK8nN/qzxqYHMeFZeZmc2dd35ARkY2/fu3Y9q0AbRqZRPYGRNpSk0Kqro/mIF4LnOTVR4VU1CgFBQocXEx1KtXjZdeuoTDh3O59tpTrbvImAgVyB3N0WGvdR35+/bbvfTq9XcmTvysaN/ll3dk6FCb0dSYSGZJoZBNbwFAbm4+jz32KV26TGPp0h3MmLGG7Ow8r8MyxgSJ3V1UyCqPWLPmJ264IZWvvtoFwIgRp/HEE33tJjRjooj9tRfb6oM5AAAQbUlEQVSK4sqj3Nx8xo9fwqRJS8nPV1q3rsvLLw+kT58TvA7NGBNklhQg6iuP4uJiWLFiJwUFyrhxPXj00QuoWbOK12EZYzxgSQF+qzyq3TpqKo8OHszh4MEjNG1aCxFh+vSB7Nr1K2ee2cLr0IwxHrKBZvhtkDlKpsteuDCNTp1eZMiQuRSum9SmTT1LCMYYSwpA1EyEt2/fYa677l3695/F9u0HOHgwh337srwOyxgTQlxNCiLSX0S+E5E0Ebm3hOfvEJENIvK1iHwoIq3cjKdUET7IrKrMmbOBjh2n8tpra0lIiGPSpD4sX34jDRtW9zo8Y0wIcW1MQURigReAvkA6sFJEUlV1g99ha4AkVT0sImOAScBVbsVUqgi+R0FVGTJkLm+++Q0AvXq14uWXB9KhQwOPIzPGhCI3WwrdgTRV3aKqR4C3gEH+B6jqYlU97NtcDjR3MZ6SRXjlkYjQsWMjatWqwosvDmDx4ussIRhjSuVm9VEzYIffdjrQo4zjRwALSnpCREYBowBatqzk5R0zN0FBbkRVHm3dmsGWLRn07u3cZ3DPPWcxfHgizZvX9jgyY0yoc7OlUNIEOVrigSLXAknAEyU9r6opqpqkqkmNGjWqxBCJqMqj/PwCJk9eTqdOL3LVVXPYvfsQAPHxsZYQjDEBcbOlkA741zg2B34sfpCI9AHuB85V1RwX4ylZYeVR/fAeZN6wYQ833pjKsmXpACQnn2jrHBhjKszNpLASaC8ibYCdwGDgGv8DROQ04CWgv6rudjGW0oV5SyE3N5/HH1/K//7vJxw5kk/TprV48cUBJCef6HVoxpgw5FpSUNU8ERkLLMRZ73mmqq4XkUeAVaqaitNdVBP4p2865u2qmuxWTCUK84nwrrlmLnPmOIlt5MiuPPFEX+rUSfA4KmNMuHJ1mgtVnQ/ML7bvIb/Hfdx8/3L5Vx7VP8nTUI7WuHE9+OqrXbz00iVccEEbr8MxxoS56L6jOTPtt8qjKjW9jiYgH3+8jQkTlhRtn312SzZuvMUSgjGmUkT3hHiFXUdhMJ7wyy853HPPIqZN+xKA889vQ69ezg3gcXHRnduNMZUnupNCmFQezZ+/iZtuep/09F+Ij4/h/vvP4Ywzgn+fnzEm8kV3UgjxyqO9ew/zl7/8h1mz1gHQvXszZsxIplOnxh5HZoyJVFGeFEJ7IrxHHvmYWbPWUa1aHI8+egHjxvUgNta6iowx7onepPC7yqPQmfNIVfGV5zJhwnn8/PMhHnvsAtq2re9xZMaYaBC9HztDrPJIVXn55S/p2XMm2dl5ANSrV43Zs6+whGCMCZroTQoh1HW0efN+evd+jVGj3mf58nTefnu91yEZY6JU9HYfhcAaCs4Edit44IGPyMrKo1Gj6jz//EVceWVoDnwbYyJf9CaFvd62FNav380NN6TyxRc7ARgypDPPPtvfVkIzxngqepOCxzeurVmziy++2EmzZrV46aVLGDCggydxGGOMv+hMCh5VHu3Zc4hGjZyFfIYM6UxmZjZDh55qE9gZY0JGdA40F1UetQpK5dHhw7ncddcHtG49mY0b9wDOMpljx3a3hGCMCSnR2VII4iDz4sVbGTlyHps3ZxATI3zyyQ+cfHIlrx5njDGVJEqTgvtrKBw4kM1f/7qIlJTVAHTu3JiZMweRlNTUtfc0xphjFZ1JweXKo88+287gwXPYufMg8fExPPhgL+6552yqVIl15f2MMaayRGdS2O/uRHhNmtRk374szjijOdOnD+SUU2wCO2NMeIi+pJCfC/u/cx5XUuWRqrJo0Rb69j0BEaFdu/p89tn1JCY2sQnsjDFhJfquWJVcebRjxwEGDnyTCy/8B3//+1dF+08/vaklBGNM2Im+lkIlVR4VFDgT2N199yIOHjxCnTpVqVrVxgyMMeEtCpPCsQ8yb9q0j5Ej5/Hxxz8AcOmlJ/HCCxfTtGmtyojQGGM8E4VJ4dhaCp9/voPevV8jOzuPxo1rMGXKRVxxRceiNRCMMSacRWFSOLY5j5KSmtK+fX1OO+14nn66Hw0a2AR2xpjIEV1J4Sgqj3Jy8njyyc+56aYkGjasTpUqsSxdegO1alV1MVBjjPFGdCWFzM0VqjxavjydESNS2bBhDxs37uUf/7gMwBKCMSZiRVdSCHCQ+dChIzzwwEdMnrwCVejQoQE33XR6EAI0xhhvRWlSKH084cMPtzBy5Dy2bs0kNla4++6ejB9/HgkJ0fWjMsZEp+i60pVTefT99/vo2/d1VCExsQkzZiTTtevxQQzQGGO8FWVJoezuow4dGjBuXA8aNarB3Xf3JD7ebkYzxkSX6EkK/pVHvqTw88+/cttt/2H06NM5//w2ADzzTH+vIjTGGM+5OjmPiPQXke9EJE1E7i3h+aoiMtv3/AoRae1aMH6VRxpfg9dfX0vHjlN5++313H77QlTVtbc2xphw4VpSEJFY4AXgIqAjcLWIFO+3GQFkqGo74BngcbfiKew62l6QyIABbzBs2Lvs359Fv35teffdwXZHsjHG4G73UXcgTVW3AIjIW8AgYIPfMYOAh32P5wBTRETUhY/tBXvWM21pN+75TyK/ZqVRr14CzzxzIcOGdbGEYIwxPm4mhWbADr/tdKBHaceoap6IHAAaAHsrO5gDO75lwqJz+TVLuPzyk5ky5WKaNDn2qbONMSaSuJkUSvr4XbwFEMgxiMgoYBRAy5YtjyqYejWV6YMXcqTHo1x+Y/JRvYYxxkQ6N5NCOtDCb7s58GMpx6SLSBxQB9hf/IVUNQVIAUhKSjq6rqVL3mTggIKj+lZjjIkWblYfrQTai0gbEakCDAZSix2TClzne3wF8JEb4wlFJMb5MsYYUyLXWgq+MYKxwEIgFpipqutF5BFglaqmAjOA10UkDaeFMNiteIwxxpTP1ZvXVHU+ML/Yvof8HmcDf3YzBmOMMYGzvhRjjDFFLCkYY4wpYknBGGNMEUsKxhhjilhSMMYYU0TCbXZQEdkD/HCU394QF6bQCHF2ztHBzjk6HMs5t1LVRuUdFHZJ4ViIyCpVTfI6jmCyc44Ods7RIRjnbN1HxhhjilhSMMYYUyTakkKK1wF4wM45Otg5RwfXzzmqxhSMMcaULdpaCsYYY8oQkUlBRPqLyHcikiYi95bwfFURme17foWItA5+lJUrgHO+Q0Q2iMjXIvKhiLTyIs7KVN45+x13hYioiIR9pUog5ywiV/p+1+tF5I1gx1jZAvi/3VJEFovIGt//74u9iLOyiMhMEdktIt+U8ryIyHO+n8fXItK1UgNQ1Yj6wpmmezNwAlAFWAt0LHbMzcA03+PBwGyv4w7COZ8PVPc9HhMN5+w7rhbwCbAcSPI67iD8ntsDa4B6vu3GXscdhHNOAcb4HncEtnkd9zGecy+gK/BNKc9fDCzAWbnyDGBFZb5/JLYUugNpqrpFVY8AbwGDih0zCHjV93gO0FtESloaNFyUe86qulhVD/s2l+OshBfOAvk9A/wvMAnIDmZwLgnknEcCL6hqBoCq7g5yjJUtkHNWoLbvcR3+uMJjWFHVTyhhBUo/g4DX1LEcqCsix1fW+0diUmgG7PDbTvftK/EYVc0DDgANghKdOwI5Z38jcD5phLNyz1lETgNaqOr7wQzMRYH8njsAHURkqYgsF5H+QYvOHYGc88PAtSKSjrN+y63BCc0zFf17rxBXF9nxSEmf+IuXWAVyTDgJ+HxE5FogCTjX1YjcV+Y5i0gM8AwwPFgBBUEgv+c4nC6k83Bag5+KSCdVzXQ5NrcEcs5XA6+o6lMicibOao6dVDVSF2V39foViS2FdKCF33Zz/ticLDpGROJwmpxlNddCXSDnjIj0Ae4HklU1J0ixuaW8c64FdAKWiMg2nL7X1DAfbA70//Z7qpqrqluB73CSRLgK5JxHAG8DqOoyIAFnjqBIFdDf+9GKxKSwEmgvIm1EpArOQHJqsWNSget8j68APlLfCE6YKvecfV0pL+EkhHDvZ4ZyzllVD6hqQ1VtraqtccZRklV1lTfhVopA/m+/i1NUgIg0xOlO2hLUKCtXIOe8HegNICIn4ySFPUGNMrhSgWG+KqQzgAOq+lNlvXjEdR+pap6IjAUW4lQuzFTV9SLyCLBKVVOBGThNzDScFsJg7yI+dgGe8xNATeCfvjH17aqa7FnQxyjAc44oAZ7zQqCfiGwA8oG7VXWfd1EfmwDP+U7gZRG5HacbZXg4f8gTkTdxuv8a+sZJxgPxAKo6DWfc5GIgDTgMXF+p7x/GPztjjDGVLBK7j4wxxhwlSwrGGGOKWFIwxhhTxJKCMcaYIpYUjDHGFLGkYEKOiOSLyFd+X63LOLZ1abNJVvA9l/hm4lzrmyLixKN4jdEiMsz3eLiINPV7brqIdKzkOFeKSGIA3/MXEal+rO9tooMlBROKslQ10e9rW5Ded4iqdsGZLPGJin6zqk5T1dd8m8OBpn7P3aiqGyolyt/inEpgcf4FsKRgAmJJwYQFX4vgUxFZ7fvqWcIxp4jIF77Wxdci0t63/1q//S+JSGw5b/cJ0M73vb198/Sv881zX9W3f6L8tj7Fk759D4vIXSJyBc78UrN871nN9wk/SUTGiMgkv5iHi8jzRxnnMvwmQhORF0VklTjrKEzw7bsNJzktFpHFvn39RGSZ7+f4TxGpWc77mChiScGEomp+XUfv+PbtBvqqalfgKuC5Er5vNDBZVRNxLsrpvmkPrgLO8u3PB4aU8/4DgXUikgC8Alylqp1xZgAYIyL1gT8Bp6jqqcCj/t+sqnOAVTif6BNVNcvv6TnAZX7bVwGzjzLO/jjTWhS6X1WTgFOBc0XkVFV9DmdenPNV9Xzf1BcPAH18P8tVwB3lvI+JIhE3zYWJCFm+C6O/eGCKrw89H2dOn+KWAfeLSHNgrqpuEpHewOnASt/0HtVwEkxJZolIFrANZ/rlE4Gtqvq97/lXgVuAKTjrM0wXkX8DAU/Nrap7RGSLb86aTb73WOp73YrEWQNn2gf/VbeuFJFROH/Xx+MsOPN1se89w7d/qe99quD83IwBLCmY8HE78DPQBaeF+4dFc1T1DRFZAQwAForIjTjTDL+qqvcF8B5D/CfME5ES19jwzcfTHWcStsHAWOCCCpzLbOBK4FvgHVVVca7QAceJswLZROAF4DIRaQPcBXRT1QwReQVnYrjiBFikqldXIF4TRaz7yISLOsBPvjnyh+J8Sv4dETkB2OLrMknF6Ub5ELhCRBr7jqkvga9P/S3QWkTa+baHAh/7+uDrqOp8nEHckiqADuJM312SucClOOsAzPbtq1CcqpqL0w10hq/rqTZwCDggIscBF5USy3LgrMJzEpHqIlJSq8tEKUsKJlxMBa4TkeU4XUeHSjjmKuAbEfkKOAlnycINOBfPD0Tka2ARTtdKuVQ1G2cGyn+KyDqgAJiGc4F93/d6H+O0Yop7BZhWONBc7HUzgA1AK1X9wrevwnH6xiqeAu5S1bU4azOvB2bidEkVSgEWiMhiVd2DUxn1pu99luP8rIwBbJZUY4wxfqylYIwxpoglBWOMMUUsKRhjjCliScEYY0wRSwrGGGOKWFIwxhhTxJKCMcaYIpYUjDHGFPl/zB2bZgYa6voAAAAASUVORK5CYII=\n",
        "text/plain": "<Figure size 432x288 with 1 Axes>"
       },
       "metadata": {
        "needs_background": "light"
       },
       "output_type": "display_data"
      }
     ],
     "persistentId": "2c810e70-a822-4d9a-a9b0-2622dfa247bf",
     "text": "y_true = coding_scores_not_low_complexity_unspliced['in_cds_no_stop_codon']\ny_score = coding_scores_not_low_complexity_unspliced['jaccard_in_peptide_db']\ny_pred = y_score > 0.5\n\nfpr, tpr, thresholds = metrics.roc_curve(y_true, y_score)\nauc = metrics.auc(fpr, tpr)\nprint(f\"AUC: {auc}\")\n\nfig, ax = plt.subplots()\nlw = 2\nax.plot(fpr, tpr, lw=2, color='darkorange')\nax.plot([0, 1], [0, 1], color='navy', lw=lw, linestyle='--')\nax.set(xlabel='False Positive Rate', ylabel='True Positive Rate')\n\nprint(\"accuracy score:\", metrics.accuracy_score(y_true, y_pred))\n\nprint(metrics.classification_report(y_true, y_pred))"
    },
    "executionTime": "2019-10-17T07:59:20.144Z"
   },
   {
    "cell": {
     "executionCount": 42,
     "executionEventId": "d10407ad-db67-4ae8-9121-2cf587fa2ee3",
     "hasError": false,
     "id": "ddfff1e6-fd17-4da4-aee8-e95d812c866f",
     "outputs": [
      {
       "name": "stdout",
       "output_type": "stream",
       "text": "AUC: 0.9332425339782375\naccuracy score: 0.9019756727303703\n              precision    recall  f1-score   support\n\n       False       0.99      0.90      0.94    590637\n        True       0.58      0.94      0.72     90245\n\n   micro avg       0.90      0.90      0.90    680882\n   macro avg       0.79      0.92      0.83    680882\nweighted avg       0.94      0.90      0.91    680882\n\n"
      },
      {
       "data": {
        "image/png": "iVBORw0KGgoAAAANSUhEUgAAAYUAAAEKCAYAAAD9xUlFAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDMuMC4zLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvnQurowAAIABJREFUeJzt3Xl8FFW2wPHfyQJh3xHZkUVFkIgBFBUXFlEkOOooiiCKICjKuI36XBCfz0FcUUSMwLgMKg6DGhwYZBRcEBAEEQGVsAhBkS1BhCRkOe+P6sQ2ZulAqquX8/188qGrutJ9KiF1+t576l5RVYwxxhiAGK8DMMYYEzosKRhjjCliScEYY0wRSwrGGGOKWFIwxhhTxJKCMcaYIpYUjDHGFLGkYIwxpoglBWOMMUXivA6goho2bKitW7f2OgxjjAkrX3755V5VbVTecWGXFFq3bs2qVau8DsMYY8KKiPwQyHHWfWSMMaaIJQVjjDFFLCkYY4wpYknBGGNMEUsKxhhjiriWFERkpojsFpFvSnleROQ5EUkTka9FpKtbsRhjjAmMmy2FV4D+ZTx/EdDe9zUKeNHFWIwxxgTAtfsUVPUTEWldxiGDgNfUWQ90uYjUFZHjVfUnt2IyxpgKU4WCXMg/8tu//o8LStlf+LiglP2F35vvd0wpz33xXRUSYo9w6tk9oOd4V0/Xy5vXmgE7/LbTffv+kBREZBROa4KWLVsGJThjjAu0wHehK+WiGMiFsyIX5jK/N9DXzfPux6Vwz7/78tTH3Tj1+J/5ov0G4l1+Ty+TgpSwT0s6UFVTgBSApKSkEo8xJuoUXWAD+7QZ/Iuu/zGFn5q9u8Aek5g4iKkCsVUgJt751/9xTBWIjS/lmLKeK2N/bBUkpgp8vws++Zl+yd3J794jopNCOtDCb7s58KNHsZhQoupc8Py/KHyspewvfnxJr6G/vUZJjwN6vnB/fukXvmBddDXfq9/QsSm6kAZyQa2Ei27hMSXtD+DCTEw8SEmfYd2RmZnNli0ZdO16PAATJucyePTeom23eZkUUoGxIvIW0AM4YOMJflQhOwMO74KsvZCdCfnZkJcFeX7/5mf/9gmsIM+5UBTkO/s0v9i+PN9+vwtqQb7fhbO0xyVt+18k8397D82nxAtySRfr3x3nd+E1gQvkwhesi25AF+a4oF5gw817733LmDH/JiZGWL/+ZurUSaBatfigJQRwMSmIyJvAeUBDEUkHxoPT8lHVacB84GIgDTgMXO9WLGEjPxd2rYRv34RNc+DQLq8j8pBATCxIjPNFjHMxKdwu2lfCF8WOE/E7Vvye93tc6r5ijwufi4nz/qJrF9iIsXv3IW67bQGzZ68H4IwzmpOZmU2dOglBj8XN6qOry3legVvcev+wcnAnrJwE377htAoKxdeEmk2hWiNIqAuxCRBfHWKrQlw1Zzuumu9iEQ8S67uQxvpdNIrviwWJ+/1FMyaWootm4YWYwv3+F2f/i3ThxZLfv4f/Mf7HFr5WaRfq372uXehMdFBVZs1ax7hx/2H//iyqV4/nsccuYOzY7sTGenNvcdhNnR1Rcg/Dskdg9bOQn+Psq9sO2lwMp1wHjRN/u/AaYyLOmDH/5qWXvgSgT58TSEm5hDZt6nkakyUFL6g6XUSf3gcHtzv72v0Jut8LTbrZJ2VjosSll57E7Nnreeqpflx/fSISAn/7lhSCTRU+vdfpLgJolAh9XoSmZ3gblzHGdZs27ePDD7cyenQSAP37t2PbtnGejB2UxpJCMGkBfDQOvpri9L33fgE63+jrbzfGRKq8vAKefnoZ48cvIScnj8TEJpxxRnOAkEoIYEkheAryIPUK2PyeMwA88F/QLtnrqIwxLlu7dhcjRqTy5ZdOxf2wYV1o376+x1GVzpJCMKjCopuchFC1LgycA616ex2VMcZFOTl5PProJ0ycuJS8vAJatqzDSy9dQv/+7bwOrUyWFIJh8zz4ZqZTPnrZfGh6ptcRGWNcdt99H/LMM8sBuOWWbvztb72pVauqx1GVz5KC21Thvzc5j8961BKCMVHir389i2XL0pk0qQ/nnNPK63ACZkXwbtu33rkzObYqnDbW62iMMS5ZtGgzl1/+Nnl5BQA0aVKTzz+/IawSAlhScN+er51/G3VxpicwxkSUjIwsRox4j379/sHcuRv5+9/XFD0XCvcdVJR1H7lt3wbn39b9vI3DGFPp3nlnIzffPJ9du36latVYxo8/l+HDE70O65hYUnDbPmeCK+p39DYOY0yl2bXrV269dQFz5jgf+nr2bMGMGcmcdFJDjyM7dpYU3FaYFBqe4m0cxphK89573zJnzgZq1Ihn4sQ+3HxzN2Jiwq+rqCSWFNyUlw2Zm51J7eqd6HU0xphjkJ2dR0KCc8kcOfJ0tmzJYMyYbrRuXdfjyCqXDTS7af93ztQWddtBXOjXJxtj/qigQJky5QvatJnMDz9kAhATIzz+eN+ISwhgScFdhV1HDazryJhw9N13e+nV6+/ceusCdu36lTff/MbrkFxn3UduKqw8svEEY8JKbm4+Tz75ORMmfExOTj7HHVeDqVMHcNllJ3sdmussKbjJKo+MCTvffLObYcPeYc0aZznc669P5Kmn+lGvXjWPIwsOSwpuspaCMWGnoEBZt243rVrVISVlIP36tfU6pKCypOCWvGzITPNVHnXwOhpjTBnWr99Nx46NEBFOPfU43ntvML16taJmzeibhcAGmt3yu8qj0FpEwxjjOHgwh7Fj59Op04v8618bi/ZffHH7qEwIYC0F9xR2HVnlkTEhaeHCNEaNep/t2w8QFxfDtm2ZXocUEiwpuKWoHNUGmY0JJfv3Z3H77Qt57bW1AHTtejwzZiSTmNjE48hCgyUFt9g9CsaEnK++2kX//v/g558PUbVqLBMmnMedd/YkLs560gtZUnBLUfeRtRSMCRUdOjSgZs0qdOjQgOnTk+nQoYHXIYUcSwpu8K88qm9zHhnjFVXljTfWMXDgidSuXZXq1eNZsmQ4TZvWipgJ7CqbtZnckPG9VR4Z47Ft2zK58MJ/cO2173Dvvf8t2t+8eW1LCGWwloIb9togszFeyc8vYOrUldx334ccOpRL/frV6NmzhddhhQ1LCm6wQWZjPLFx4x5GjEhl2bJ0AK688hSef/4iGjeu4XFk4cOSghvsHgVjgm7r1gwSE1/iyJF8jj++JlOnDuDSS0/yOqyw42pSEJH+wGQgFpiuqhOLPd8SeBWo6zvmXlWd72ZMQWH3KBgTdG3a1OPPf+5IQkIcTz7Zj7p1bTzvaLiWFEQkFngB6AukAytFJFVVN/gd9gDwtqq+KCIdgflAa7diCgqrPDImKLKycnnkkY/5059Opnv3ZgC8+uqlxMZa/cyxcPOn1x1IU9UtqnoEeAsYVOwYBWr7HtcBfnQxnuAoqjxqa5VHxrjk009/IDHxJSZOXMqoUfMoKFAASwiVwM3uo2bADr/tdKBHsWMeBj4QkVuBGkAfF+MJjr02yGyMW375JYf77vsvU6euAqBjx0ZMm3aJlZhWIjfTakm/JS22fTXwiqo2By4GXheRP8QkIqNEZJWIrNqzZ48LoVai/TbIbIwb5s/fRKdOU5k6dRVxcTE89FAvVq8exRlnNPc6tIjiZkshHfAvDm7OH7uHRgD9AVR1mYgkAA2B3f4HqWoKkAKQlJRUPLGEFrtHwZhKd+BANkOGzCUzM5ukpKbMmJHMqace53VYEcnNpLASaC8ibYCdwGDgmmLHbAd6A6+IyMlAAhDiTYFy2D0KxlQKVUUVYmKEOnUSeO65/vz88yH+8pczbAI7F7mWFFQ1T0TGAgtxyk1nqup6EXkEWKWqqcCdwMsicjtO19JwVQ3tlkBZ8nKs8siYSvDjjwe5+eZ/c845Lbnzzp4ADB3axeOoooOr9yn47jmYX2zfQ36PNwBnuRlDUGX4Vlur194qj4w5CqrKzJlruPPODzhwIIfly9O5+eZuVKsW73VoUcPuaK5MVnlkzFHbsiWDkSPn8dFHWwEYMKA906ZdYgkhyCwpVKb9toaCMRWVn1/Ac8+t4P77PyIrK4+GDavz3HP9GTy4EyJWahpslhQqk7UUjDkqc+ZsJCsrj6uv7sTkyf1p1MgmsPOKJYXKZBPhGROQI0fyOXgwhwYNqhMbG8OMGcls2rSPgQOtQMNrVtdVWazyyJiArFy5k6SkFIYOfYfCYsOTTmpoCSFEWEuhsmR8B5pvlUfGlOLw4VzGj1/M008vp6BAOXw4l927D3HccTW9Ds34saRQWazryJhSLVmyjZEj55GWtp+YGOGuu85kwoTzqV7dKotCjSWFymJrKBjzB6rKbbctYMqUlQB07tyYGTOS6datmceRmdJYUqgsVnlkzB+ICLVrVyU+PoYHHujFvfeeTZUqsV6HZcpgSaGy7LN7FIwB2Lv3MJs376dHD2f20gcfPJchQ06lY8dGHkdmAhFQ9ZGIVBGRdm4HE7Z+V3lka8Ka6KSqvPXWN5x88gtceulsMjKyAEhIiLOEEEbKTQoiMgBYByzybSeKyDtuBxZWCiuPbLU1E6XS039h0KC3uPrqf7F372E6dmzE4cO5XodljkIg3UeP4KyYthhAVb+yVkMxhV1H9a3ryESXggJl+vTV3H33In75JYfatavy1FP9GDHiNJuiIkwFkhRyVTWz2C84fKe3dkNh5VFDG2Q20WXEiFReeeUrAJKTT2Tq1Itp1qx2Od9lQlkgYwobReRKIEZE2ojIs8Byl+MKL3aPgolS117bmcaNa/DWW5fz7rtXWUKIAIEkhbHA6UABMBfIBsa5GVTYsSU4TZT45pvdTJ7822fC3r1PYMuW27jqKpvRNFIE0n10oareA9xTuENELsNJEMa/8qiezd1iIlNOTh5/+9tnPPbYp+TmFpCU1JSzzmoJQI0aVTyOzlSmQFoKD5Sw7/7KDiRsZXzvVB7VOQHiq3kdjTGVbsWKdE4/PYUJEz4mN7eAMWOS6Nz5OK/DMi4ptaUgIhcC/YFmIvK031O1cbqSDPhNb2HjCSayHDp0hAcfXMyzzy5HFdq3r8/06cn06tXK69CMi8rqPtoNfIMzhrDeb/9B4F43gworVnlkItT993/E5MkriIkR7r77TB5++DxbGjMKlJoUVHUNsEZEZqlqdhBjCi82vYWJUPfffw7r1u3m8cf7kJTU1OtwTJAEMqbQTETeEpGvReT7wi/XIwsXNhGeiRCpqd9x8cWzyM3NB6BRoxp8+OEwSwhRJpCk8Arwd0CAi4C3gbdcjCl8WOWRiQC7dx9i8OA5DBr0FgsWpPHqq2u9Dsl4KJCkUF1VFwKo6mZVfQA4392wwoRVHpkwpqr84x9fc/LJLzB79nqqV49n8uT+XH99otehGQ8Fcp9Cjjh3pWwWkdHATqCxu2GFCas8MmFq+/YDjB79PgsWpAHQp88JpKRcQps29TyOzHgtkKRwO1ATuA34P6AOcIObQYUNG2Q2YeqDDzazYEEadesm8PTT/Rg+PNHuSDZAAElBVVf4Hh4EhgKISHM3gwobVo5qwsihQ0eK7j4eMeI0du78hVGjTuf442t5HJkJJWWOKYhINxG5VEQa+rZPEZHXsAnxHFZ5ZMJAXl4BkyYtpVWrZ9myJQNwlskcP/48SwjmD0pNCiLyN2AWMAT4j4jcj7OmwlqgQ3DCC2FWeWTCwNq1u+jRYzr33PNf9u3L4t13v/U6JBPiyuo+GgR0UdUsEakP/Ojb/i44oYW4wsqjuu2s8siEnJycPB599BMmTlxKXl4BLVvWISXlEi680NbHMmUrKylkq2oWgKruF5FvLSH4sUFmE6LWrPmJIUPmsnHjXkRg7NhuPPZYb2rVqup1aCYMlJUUThCRwumxBWjtt42qXlbei4tIf2AyEAtMV9WJJRxzJfAwzmpua1X1msDD95CVo5oQVbVqHJs3Z3DiiQ2YPj2Zs89u6XVIJoyUlRQuL7Y9pSIvLCKxwAtAXyAdWCkiqaq6we+Y9sB9wFmqmiEi4XP/g1UemRCyevVPnHZaE0SEjh0bsWDBEHr2bEFCQiBV58b8pqwJ8T48xtfuDqSp6hYAEXkLZ5xig98xI4EXVDXD9567j/E9g6ew+6i+dR8Z72RkZHHXXR8wc+ZXvPnm5Qwe3AmACy5o43FkJly5+TGiGbDDbzsd6FHsmA4AIrIUp4vpYVX9T/EXEpFRwCiAli1DoCmclwMZm5zKo/oneR2NiVLvvLORm2+ez65dv1K1aiz79h32OiQTAdxMCiXdHqklvH974DygOfCpiHRS1czffZNqCpACkJSUVPw1gs8qj4yHdu36lVtvXcCcOU5r9ayzWjB9ejInndTQ48hMJAg4KYhIVVXNqcBrpwMt/Lab45S1Fj9muarmAltF5DucJLGyAu8TfFZ5ZDzy5Zc/0rfv62RkZFOjRjwTJ/bh5pu7ERNjU1SYylHuLKki0l1E1gGbfNtdROT5AF57JdBeRNqISBVgMJBa7Jh38c246rtrugOwpQLxe8Mqj4xHOnZsRKNGNbjwwrasX38zY8d2t4RgKlUgU2c/B1wC7ANQ1bUEMHW2quYBY4GFwEbgbVVdLyKPiEiy77CFwD4R2YBzt/Tdqrqv4qcRZNZSMEFSUKCkpHxJZqaz+GG1avF88slwFiwYQqtWdT2OzkSiQLqPYlT1h2IzKOYH8uKqOh+YX2zfQ36PFbjD9xU+rKVgguC77/Zy443z+Oyz7axcuZOXX3Y+Sx13XE2PIzORLJCksENEugPqu/fgViB6l+MsrDxCrPLIuCI3N5+nnlrGww8vIScnnyZNanLRRe29DstEiUCSwhicLqSWwM/Af337olPmJl/lUVurPDKVbs2anxgxIpU1a3YBcP31iTz1VD/q1bP/ayY4AkkKeao62PVIwoVNl21csnnzfrp3n05eXgGtW9clJeUS+vZt63VYJsoEkhRW+kpFZwNzVfWgyzGFNhtPMC5p27Y+Q4eeSq1aVfi//+tNzZpVvA7JRKFyq49UtS3wKHA6sE5E3hWR6G05WOWRqSS//nqE225bwLJlv934P2NGMpMnX2QJwXgmkJJUVPVzVb0N6Ar8grP4TnSyloKpBAsXpnHKKVN5/vkvGD363ziFeNg6ycZz5XYfiUhNnInsBgMnA+8BPV2OKzTlH/GrPLLV1kzF7d+fxe23L+S119YCcPrpxzNjRrIlAxMyAhlT+AaYB0xS1U9djie0Fc151Bbiq3sdjQkzc+Zs4JZb5rN79yESEuKYMOE87rjjTOLiAmqwGxMUgSSFE1S1wPVIwoFVHpmjlJmZzahR88jIyKZXr1a8/PJAOnRo4HVYxvxBqUlBRJ5S1TuBf4nIH2YmDWTltYhjg8ymAlSVggIlNjaGunUTmDp1ABkZWdx0U5LNV2RCVlkthdm+fyu04lpEs0FmE6Bt2zIZNWoeF1zQhnvvPRugaAEcY0JZqZ2ZqvqF7+HJqvqh/xfOgHP0saRgypGfX8Bzz62gU6epLFq0hSlTviA7O8/rsIwJWCAjXDeUsG9EZQcS8qzyyJRj48Y99Or1CuPG/YdDh3IZPLgTq1ffZOskm7BS1pjCVThlqG1EZK7fU7WAzJK/K4JZ5ZEpRV5eAY8//hmPPPIJR47k07RpLV58cQDJyfbhwYSfsj7CfIGzhkJz4AW//QeBNW4GFZIKK4/q2yCz+b2YGOGDD7Zw5Eg+I0d2ZdKkvtStm+B1WMYclVKTgqpuBbbizIpqCiuPGtp4goGsrFwOHjxC48Y1iIkRpk8fyI4dv3DBBW28Ds2YY1LqmIKIfOz7N0NE9vt9ZYjI/uCFGCJskNn4fPLJD3TpMo1rr51bND1F+/YNLCGYiFBW91HhkpsNgxFIyLN7FKLeL7/kcN99/2Xq1FUAxMfHsnfvYRo1quFxZMZUnrJKUgvvYm4BxKpqPnAmcBMQXX8F+UecxXVstbWotWDBJjp1msrUqauIi4th/PhzWb16lCUEE3ECqZV7F+gmIm2B14B/A28Al7gZWEjJ+B4K8qzyKAqpKiNHzmPGDKe2IimpKTNnJtO583EeR2aMOwK5T6FAVXOBy4BnVfVWoJm7YYWYwq4jqzyKOiJC8+a1SUiI48kn+7Js2QhLCCaiBbQcp4j8GRgKXOrbF+9eSCGosBzVKo+iwo8/HmTz5v2cc04rAP7nf85h6NBTadu2vseRGeO+QO9oPh9n6uwtItIGeNPdsEJMUeWRtRQimaoyY8ZqOnZ8gcsvf5t9+w4DUKVKrCUEEzXKbSmo6jcichvQTkROAtJU9f/cDy2EFFUeWUshUm3ZksHIkfP46KOtAFxySQdyc23GeBN9All57RzgdWAnIEATERmqqkvdDi4kWOVRRCucwO6BBxZz+HAuDRtW57nn+jN4cCdbDc1EpUDGFJ4BLlbVDQAicjJOkkhyM7CQkbHJqTyqc4JVHkWgYcPe5Y031gFwzTWdefbZC63M1ES1QMYUqhQmBABV3QhUcS+kEGN3Mke0kSO70rx5bVJTBzNr1mWWEEzUC6SlsFpEXsJpHQAMIZomxLPKo4iycuVOPvpoK/fc4yx8c955rUlLu5WqVW16a2MgsKQwGrgN+CvOmMInwPNuBhVS9tv0FpHg8OFcxo9fzNNPL6egQOnZs0VRyaklBGN+U+Zfg4h0BtoC76jqpOCEFGL2WvdRuFuyZBs33pjK5s0ZxMQId911Jqef3tTrsIwJSWXNkvo/OFNcDAEWiUhJK7CVSUT6i8h3IpImIveWcdwVIqIiElqD11Z5FNYOHMjmppvmcf75r7J5cwadOzdm+fIRPPFEP6pXj677L40JVFkthSHAqap6SEQaAfOBmYG+sIjE4izO0xdIB1aKSKr/oLXvuFo43VMrKhq866zyKKw9+OBiUlJWEx8fw4MP9uKee86mSpVYr8MyJqSVlRRyVPUQgKruEZFAKpX8dce50W0LgIi8BQwCNhQ77n+BScBdFXx991nlUdhR1aL7Cx566Fy2bs1k4sTenHJKY48jMyY8lHWhP0FE5vq+3gHa+m3PLeP7CjUDdvhtp1NsIj0ROQ1ooarvVzjyYLA1FMKGqvLGG+u44ILXOHIkH4CGDaszb97VlhCMqYCyWgqXF9ueUsHXLul2UC160ml5PAMML/eFREYBowBatmxZwTCOwT4rRw0H6em/MGbMv3n//e8BmDXra66//jSPozImPJW1RvOHx/ja6TgL9BRqDvzot10L6AQs8TX3mwCpIpKsqquKxZICpAAkJSUpwWKVRyGtoEB5+eUvufvuRRw8eIQ6dary1FP9GD480evQjAlbbhZorwTa+2ZV3QkMBq4pfFJVD+C31KeILAHuKp4QPGOVRyEtLW0/I0fOY8mSbQAMGnQiU6cOoGnTWt4GZkyYcy0pqGqeiIwFFgKxwExVXS8ijwCrVDXVrfeuFFZ5FNI+/fQHlizZRuPGNZgy5SKuuKKjTWBnTCUIOCmISFVVzanIi6vqfJxSVv99D5Vy7HkVeW3X2RoKISczM5u6dRMAGD48kT17DjNixGk0aGBJ25jKUm6ZqYh0F5F1wCbfdhcRifxpLmwNhZCRk5PH+PGLadXqWTZt2gc4y2T+9a9nWUIwppIF0lJ4DrgE5+5mVHWtiJzvalShwCqPQsLy5emMGJHKhg17AFi4cDPt2zfwOCpjIlcgSSFGVX8o1l+b71I8ocPuUfDUoUNHePDBxTz77HJUoX37+syYkVw0iZ0xxh2BJIUdItIdUN/UFbcC37sblsfyj0DG9ziVRyd7HU3UWbEinWuumcuWLRnExgp33dWT8ePPpVo1m6/IGLcFkhTG4HQhtQR+Bv7r2xe5iiqP2ljlkQfq1k1g585f6NLlOGbMSLYZTY0JonKTgqruxrnHIHrYIHPQffbZds46qwUiwoknNuSjj66jW7emxMfbBHbGBFO5SUFEXsZveopCqjrKlYhCgU2EFzS7dx/ittsWMHv2el599VKGDesCQM+eLcr5TmOMGwLpPvqv3+ME4E/8fqK7yGP3KLhOVZk1ax3jxv2H/fuzqF49vmgiO2OMdwLpPprtvy0irwOLXIsoFBR2H1k5qiu2bz/A6NHvs2BBGgB9+55ASspAWreu63FkxpijmeaiDRC5dYFWeeSqFSvS6dPndX799Qh16ybwzDMXct11XWyKCmNCRCBjChn8NqYQA+wHSl1aM+xlplnlkYsSE5vQokVtTjqpIS+8cDHHH28T2BkTSspMCuJ8fOuCM8spQIGqBm/qai/YdNmVKi+vgClTvmDYsC7Ur1+NqlXjWLr0BurVq+Z1aMaYEpQ595EvAbyjqvm+r8hOCGCDzJVo7dpd9OgxndtvX8gddyws2m8JwZjQFci6y1+ISFfXIwkVdo/CMcvOzuOBBz4iKellVq/+iZYt63D11Z28DssYE4BSu49EJE5V84CzgZEishk4hLPMpqpqZCYKmwjvmHz++Q5GjEjl22/3IgJjx3bjscd6U6tWVa9DM8YEoKwxhS+ArsClQYrFe0WVR9hqa0chLW0/55zzdwoKlBNPbMCMGcmcdVYQ19Q2xhyzspKCAKjq5iDF4r3fVR7V8DqasNOuXX1GjepK/frVePDBc0lIcHO1V2OMG8r6q20kIneU9qSqPu1CPN6yyqMKycjI4s47P+D66xOLprSeOnWA3XNgTBgrKynEAjXxtRiigq2hELC5czdyyy3z2bXrV7788ie++uomRMQSgjFhrqyk8JOqPhK0SEKBTYRXrl27fmXs2Pn8618bATj77JZMnz7QkoExEaLcMYWoYvcolEpVee21tdx++0IyMrKpWbMKjz/eh9Gjk4iJib7/KsZEqrKSQu+gRREK8nN/qzxqYHMeFZeZmc2dd35ARkY2/fu3Y9q0AbRqZRPYGRNpSk0Kqro/mIF4LnOTVR4VU1CgFBQocXEx1KtXjZdeuoTDh3O59tpTrbvImAgVyB3N0WGvdR35+/bbvfTq9XcmTvysaN/ll3dk6FCb0dSYSGZJoZBNbwFAbm4+jz32KV26TGPp0h3MmLGG7Ow8r8MyxgSJ3V1UyCqPWLPmJ264IZWvvtoFwIgRp/HEE33tJjRjooj9tRfb6oM5AAAQbUlEQVSK4sqj3Nx8xo9fwqRJS8nPV1q3rsvLLw+kT58TvA7NGBNklhQg6iuP4uJiWLFiJwUFyrhxPXj00QuoWbOK12EZYzxgSQF+qzyq3TpqKo8OHszh4MEjNG1aCxFh+vSB7Nr1K2ee2cLr0IwxHrKBZvhtkDlKpsteuDCNTp1eZMiQuRSum9SmTT1LCMYYSwpA1EyEt2/fYa677l3695/F9u0HOHgwh337srwOyxgTQlxNCiLSX0S+E5E0Ebm3hOfvEJENIvK1iHwoIq3cjKdUET7IrKrMmbOBjh2n8tpra0lIiGPSpD4sX34jDRtW9zo8Y0wIcW1MQURigReAvkA6sFJEUlV1g99ha4AkVT0sImOAScBVbsVUqgi+R0FVGTJkLm+++Q0AvXq14uWXB9KhQwOPIzPGhCI3WwrdgTRV3aKqR4C3gEH+B6jqYlU97NtcDjR3MZ6SRXjlkYjQsWMjatWqwosvDmDx4ussIRhjSuVm9VEzYIffdjrQo4zjRwALSnpCREYBowBatqzk5R0zN0FBbkRVHm3dmsGWLRn07u3cZ3DPPWcxfHgizZvX9jgyY0yoc7OlUNIEOVrigSLXAknAEyU9r6opqpqkqkmNGjWqxBCJqMqj/PwCJk9eTqdOL3LVVXPYvfsQAPHxsZYQjDEBcbOlkA741zg2B34sfpCI9AHuB85V1RwX4ylZYeVR/fAeZN6wYQ833pjKsmXpACQnn2jrHBhjKszNpLASaC8ibYCdwGDgGv8DROQ04CWgv6rudjGW0oV5SyE3N5/HH1/K//7vJxw5kk/TprV48cUBJCef6HVoxpgw5FpSUNU8ERkLLMRZ73mmqq4XkUeAVaqaitNdVBP4p2865u2qmuxWTCUK84nwrrlmLnPmOIlt5MiuPPFEX+rUSfA4KmNMuHJ1mgtVnQ/ML7bvIb/Hfdx8/3L5Vx7VP8nTUI7WuHE9+OqrXbz00iVccEEbr8MxxoS56L6jOTPtt8qjKjW9jiYgH3+8jQkTlhRtn312SzZuvMUSgjGmUkT3hHiFXUdhMJ7wyy853HPPIqZN+xKA889vQ69ezg3gcXHRnduNMZUnupNCmFQezZ+/iZtuep/09F+Ij4/h/vvP4Ywzgn+fnzEm8kV3UgjxyqO9ew/zl7/8h1mz1gHQvXszZsxIplOnxh5HZoyJVFGeFEJ7IrxHHvmYWbPWUa1aHI8+egHjxvUgNta6iowx7onepPC7yqPQmfNIVfGV5zJhwnn8/PMhHnvsAtq2re9xZMaYaBC9HztDrPJIVXn55S/p2XMm2dl5ANSrV43Zs6+whGCMCZroTQoh1HW0efN+evd+jVGj3mf58nTefnu91yEZY6JU9HYfhcAaCs4Edit44IGPyMrKo1Gj6jz//EVceWVoDnwbYyJf9CaFvd62FNav380NN6TyxRc7ARgypDPPPtvfVkIzxngqepOCxzeurVmziy++2EmzZrV46aVLGDCggydxGGOMv+hMCh5VHu3Zc4hGjZyFfIYM6UxmZjZDh55qE9gZY0JGdA40F1UetQpK5dHhw7ncddcHtG49mY0b9wDOMpljx3a3hGCMCSnR2VII4iDz4sVbGTlyHps3ZxATI3zyyQ+cfHIlrx5njDGVJEqTgvtrKBw4kM1f/7qIlJTVAHTu3JiZMweRlNTUtfc0xphjFZ1JweXKo88+287gwXPYufMg8fExPPhgL+6552yqVIl15f2MMaayRGdS2O/uRHhNmtRk374szjijOdOnD+SUU2wCO2NMeIi+pJCfC/u/cx5XUuWRqrJo0Rb69j0BEaFdu/p89tn1JCY2sQnsjDFhJfquWJVcebRjxwEGDnyTCy/8B3//+1dF+08/vaklBGNM2Im+lkIlVR4VFDgT2N199yIOHjxCnTpVqVrVxgyMMeEtCpPCsQ8yb9q0j5Ej5/Hxxz8AcOmlJ/HCCxfTtGmtyojQGGM8E4VJ4dhaCp9/voPevV8jOzuPxo1rMGXKRVxxRceiNRCMMSacRWFSOLY5j5KSmtK+fX1OO+14nn66Hw0a2AR2xpjIEV1J4Sgqj3Jy8njyyc+56aYkGjasTpUqsSxdegO1alV1MVBjjPFGdCWFzM0VqjxavjydESNS2bBhDxs37uUf/7gMwBKCMSZiRVdSCHCQ+dChIzzwwEdMnrwCVejQoQE33XR6EAI0xhhvRWlSKH084cMPtzBy5Dy2bs0kNla4++6ejB9/HgkJ0fWjMsZEp+i60pVTefT99/vo2/d1VCExsQkzZiTTtevxQQzQGGO8FWVJoezuow4dGjBuXA8aNarB3Xf3JD7ebkYzxkSX6EkK/pVHvqTw88+/cttt/2H06NM5//w2ADzzTH+vIjTGGM+5OjmPiPQXke9EJE1E7i3h+aoiMtv3/AoRae1aMH6VRxpfg9dfX0vHjlN5++313H77QlTVtbc2xphw4VpSEJFY4AXgIqAjcLWIFO+3GQFkqGo74BngcbfiKew62l6QyIABbzBs2Lvs359Fv35teffdwXZHsjHG4G73UXcgTVW3AIjIW8AgYIPfMYOAh32P5wBTRETUhY/tBXvWM21pN+75TyK/ZqVRr14CzzxzIcOGdbGEYIwxPm4mhWbADr/tdKBHaceoap6IHAAaAHsrO5gDO75lwqJz+TVLuPzyk5ky5WKaNDn2qbONMSaSuJkUSvr4XbwFEMgxiMgoYBRAy5YtjyqYejWV6YMXcqTHo1x+Y/JRvYYxxkQ6N5NCOtDCb7s58GMpx6SLSBxQB9hf/IVUNQVIAUhKSjq6rqVL3mTggIKj+lZjjIkWblYfrQTai0gbEakCDAZSix2TClzne3wF8JEb4wlFJMb5MsYYUyLXWgq+MYKxwEIgFpipqutF5BFglaqmAjOA10UkDaeFMNiteIwxxpTP1ZvXVHU+ML/Yvof8HmcDf3YzBmOMMYGzvhRjjDFFLCkYY4wpYknBGGNMEUsKxhhjilhSMMYYU0TCbXZQEdkD/HCU394QF6bQCHF2ztHBzjk6HMs5t1LVRuUdFHZJ4ViIyCpVTfI6jmCyc44Ods7RIRjnbN1HxhhjilhSMMYYUyTakkKK1wF4wM45Otg5RwfXzzmqxhSMMcaULdpaCsYYY8oQkUlBRPqLyHcikiYi95bwfFURme17foWItA5+lJUrgHO+Q0Q2iMjXIvKhiLTyIs7KVN45+x13hYioiIR9pUog5ywiV/p+1+tF5I1gx1jZAvi/3VJEFovIGt//74u9iLOyiMhMEdktIt+U8ryIyHO+n8fXItK1UgNQ1Yj6wpmmezNwAlAFWAt0LHbMzcA03+PBwGyv4w7COZ8PVPc9HhMN5+w7rhbwCbAcSPI67iD8ntsDa4B6vu3GXscdhHNOAcb4HncEtnkd9zGecy+gK/BNKc9fDCzAWbnyDGBFZb5/JLYUugNpqrpFVY8AbwGDih0zCHjV93gO0FtESloaNFyUe86qulhVD/s2l+OshBfOAvk9A/wvMAnIDmZwLgnknEcCL6hqBoCq7g5yjJUtkHNWoLbvcR3+uMJjWFHVTyhhBUo/g4DX1LEcqCsix1fW+0diUmgG7PDbTvftK/EYVc0DDgANghKdOwI5Z38jcD5phLNyz1lETgNaqOr7wQzMRYH8njsAHURkqYgsF5H+QYvOHYGc88PAtSKSjrN+y63BCc0zFf17rxBXF9nxSEmf+IuXWAVyTDgJ+HxE5FogCTjX1YjcV+Y5i0gM8AwwPFgBBUEgv+c4nC6k83Bag5+KSCdVzXQ5NrcEcs5XA6+o6lMicibOao6dVDVSF2V39foViS2FdKCF33Zz/ticLDpGROJwmpxlNddCXSDnjIj0Ae4HklU1J0ixuaW8c64FdAKWiMg2nL7X1DAfbA70//Z7qpqrqluB73CSRLgK5JxHAG8DqOoyIAFnjqBIFdDf+9GKxKSwEmgvIm1EpArOQHJqsWNSget8j68APlLfCE6YKvecfV0pL+EkhHDvZ4ZyzllVD6hqQ1VtraqtccZRklV1lTfhVopA/m+/i1NUgIg0xOlO2hLUKCtXIOe8HegNICIn4ySFPUGNMrhSgWG+KqQzgAOq+lNlvXjEdR+pap6IjAUW4lQuzFTV9SLyCLBKVVOBGThNzDScFsJg7yI+dgGe8xNATeCfvjH17aqa7FnQxyjAc44oAZ7zQqCfiGwA8oG7VXWfd1EfmwDP+U7gZRG5HacbZXg4f8gTkTdxuv8a+sZJxgPxAKo6DWfc5GIgDTgMXF+p7x/GPztjjDGVLBK7j4wxxhwlSwrGGGOKWFIwxhhTxJKCMcaYIpYUjDHGFLGkYEKOiOSLyFd+X63LOLZ1abNJVvA9l/hm4lzrmyLixKN4jdEiMsz3eLiINPV7brqIdKzkOFeKSGIA3/MXEal+rO9tooMlBROKslQ10e9rW5Ded4iqdsGZLPGJin6zqk5T1dd8m8OBpn7P3aiqGyolyt/inEpgcf4FsKRgAmJJwYQFX4vgUxFZ7fvqWcIxp4jIF77Wxdci0t63/1q//S+JSGw5b/cJ0M73vb198/Sv881zX9W3f6L8tj7Fk759D4vIXSJyBc78UrN871nN9wk/SUTGiMgkv5iHi8jzRxnnMvwmQhORF0VklTjrKEzw7bsNJzktFpHFvn39RGSZ7+f4TxGpWc77mChiScGEomp+XUfv+PbtBvqqalfgKuC5Er5vNDBZVRNxLsrpvmkPrgLO8u3PB4aU8/4DgXUikgC8Alylqp1xZgAYIyL1gT8Bp6jqqcCj/t+sqnOAVTif6BNVNcvv6TnAZX7bVwGzjzLO/jjTWhS6X1WTgFOBc0XkVFV9DmdenPNV9Xzf1BcPAH18P8tVwB3lvI+JIhE3zYWJCFm+C6O/eGCKrw89H2dOn+KWAfeLSHNgrqpuEpHewOnASt/0HtVwEkxJZolIFrANZ/rlE4Gtqvq97/lXgVuAKTjrM0wXkX8DAU/Nrap7RGSLb86aTb73WOp73YrEWQNn2gf/VbeuFJFROH/Xx+MsOPN1se89w7d/qe99quD83IwBLCmY8HE78DPQBaeF+4dFc1T1DRFZAQwAForIjTjTDL+qqvcF8B5D/CfME5ES19jwzcfTHWcStsHAWOCCCpzLbOBK4FvgHVVVca7QAceJswLZROAF4DIRaQPcBXRT1QwReQVnYrjiBFikqldXIF4TRaz7yISLOsBPvjnyh+J8Sv4dETkB2OLrMknF6Ub5ELhCRBr7jqkvga9P/S3QWkTa+baHAh/7+uDrqOp8nEHckiqADuJM312SucClOOsAzPbtq1CcqpqL0w10hq/rqTZwCDggIscBF5USy3LgrMJzEpHqIlJSq8tEKUsKJlxMBa4TkeU4XUeHSjjmKuAbEfkKOAlnycINOBfPD0Tka2ARTtdKuVQ1G2cGyn+KyDqgAJiGc4F93/d6H+O0Yop7BZhWONBc7HUzgA1AK1X9wrevwnH6xiqeAu5S1bU4azOvB2bidEkVSgEWiMhiVd2DUxn1pu99luP8rIwBbJZUY4wxfqylYIwxpoglBWOMMUUsKRhjjCliScEYY0wRSwrGGGOKWFIwxhhTxJKCMcaYIpYUjDHGFPl/zB2bZgYa6voAAAAASUVORK5CYII=\n",
        "text/plain": "<Figure size 432x288 with 1 Axes>"
       },
       "metadata": {
        "needs_background": "light"
       },
       "output_type": "display_data"
      }
     ],
     "persistentId": "2c810e70-a822-4d9a-a9b0-2622dfa247bf",
     "text": "y_true = coding_scores_not_low_complexity_unspliced['in_cds_no_stop_codon']\ny_score = coding_scores_not_low_complexity_unspliced['jaccard_in_peptide_db']\ny_pred = y_score > 0.9\n\nfpr, tpr, thresholds = metrics.roc_curve(y_true, y_score)\nauc = metrics.auc(fpr, tpr)\nprint(f\"AUC: {auc}\")\n\nfig, ax = plt.subplots()\nlw = 2\nax.plot(fpr, tpr, lw=2, color='darkorange')\nax.plot([0, 1], [0, 1], color='navy', lw=lw, linestyle='--')\nax.set(xlabel='False Positive Rate', ylabel='True Positive Rate')\n\nprint(\"accuracy score:\", metrics.accuracy_score(y_true, y_pred))\n\nprint(metrics.classification_report(y_true, y_pred))"
    },
    "executionTime": "2019-10-17T08:00:44.299Z"
   }
  ],
  "kernelspec": {
   "display_name": "Python [conda env:kmer-hashing]",
   "language": "python",
   "name": "conda-env-kmer-hashing-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  },
  "uuid": "f3318337-1fb5-47f2-8605-3c42d2cec7f3"
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
