{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/mnt/pureScratch/olga/bingwu-test-data/nextflow_results_starsolo_mm10\n"
     ]
    }
   ],
   "source": [
    "cd ~/pureScratch/bingwu-test-data/nextflow_results_starsolo_mm10/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 17G\n",
      "drwxrwxr-x 1 olga olga    0 Aug  8 06:05 \u001b[0m\u001b[01;34m.\u001b[0m/\n",
      "drwxrwxr-x 1 olga olga    0 Aug  7 14:38 \u001b[01;34m..\u001b[0m/\n",
      "-rw-rw-r-- 1 olga olga 1.9M Aug  8 06:01 actb.bam\n",
      "-rw-rw-r-- 1 olga olga  71K Aug  8 06:01 actb.bam.bai\n",
      "-rw-rw-r-- 1 olga olga    0 Aug  8 06:02 actb_R1.fastq\n",
      "-rw-rw-r-- 1 olga olga    0 Aug  8 06:05 \u001b[01;31mactb_R1.fastq.gz\u001b[0m\n",
      "-rw-rw-r-- 1 olga olga    0 Aug  8 06:02 actb_R2.fastq\n",
      "-rw-rw-r-- 1 olga olga 2.8M Aug  8 06:02 actb_read_ids.txt\n",
      "-rw-rw-r-- 1 olga olga 6.0G Jul 24 17:49 S10_L001_R1_001Aligned.sortedByCoord.out.bam\n",
      "-rw-rw-r-- 1 olga olga 4.1M Jul 24 17:49 S10_L001_R1_001Aligned.sortedByCoord.out.bam.bai\n",
      "-rw-rw-r-- 1 olga olga 3.1G Jun 12 14:23 \u001b[01;31mS10_L001_R1_001.fastq.gz\u001b[0m\n",
      "drwxrwxr-x 1 olga olga    0 Jul 26 08:31 \u001b[01;34mS10_L001_R1_001Solo.out\u001b[0m/\n",
      "-rw-rw-r-- 1 olga olga 7.5G Jun 12 14:23 \u001b[01;31mS10_L001_R2_001.fastq.gz\u001b[0m\n",
      "drwxrwxr-x 1 olga olga    0 Jul 26 08:31 \u001b[01;34mS11_L001_R1_001Solo.out\u001b[0m/\n"
     ]
    }
   ],
   "source": [
    "ls -lha"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# conda environments:\n",
      "#\n",
      "base                  *  /home/olga/miniconda3\n",
      "DeNovoPipeline           /home/olga/miniconda3/envs/DeNovoPipeline\n",
      "FRC_align                /home/olga/miniconda3/envs/FRC_align\n",
      "MaSuRCA                  /home/olga/miniconda3/envs/MaSuRCA\n",
      "abyss                    /home/olga/miniconda3/envs/abyss\n",
      "aguamenti                /home/olga/miniconda3/envs/aguamenti\n",
      "allpaths                 /home/olga/miniconda3/envs/allpaths\n",
      "busco                    /home/olga/miniconda3/envs/busco\n",
      "bwa                      /home/olga/miniconda3/envs/bwa\n",
      "eggnog                   /home/olga/miniconda3/envs/eggnog\n",
      "fastqc                   /home/olga/miniconda3/envs/fastqc\n",
      "kmer-hashing             /home/olga/miniconda3/envs/kmer-hashing\n",
      "kmergenie                /home/olga/miniconda3/envs/kmergenie\n",
      "maca                     /home/olga/miniconda3/envs/maca\n",
      "nfcore-rnaseq-dev        /home/olga/miniconda3/envs/nfcore-rnaseq-dev\n",
      "picard                   /home/olga/miniconda3/envs/picard\n",
      "pranathi-parallelize-compare     /home/olga/miniconda3/envs/pranathi-parallelize-compare\n",
      "qaTools                  /home/olga/miniconda3/envs/qaTools\n",
      "samtools                 /home/olga/miniconda3/envs/samtools\n",
      "soapdenovo2              /home/olga/miniconda3/envs/soapdenovo2\n",
      "sourmash-dayhoff         /home/olga/miniconda3/envs/sourmash-dayhoff\n",
      "sourmash-ukhs            /home/olga/miniconda3/envs/sourmash-ukhs\n",
      "spades                   /home/olga/miniconda3/envs/spades\n",
      "tabula-muris-env         /home/olga/miniconda3/envs/tabula-muris-env\n",
      "tabula-muris-env-aaron     /home/olga/miniconda3/envs/tabula-muris-env-aaron\n",
      "tick-analysis            /home/olga/miniconda3/envs/tick-analysis\n",
      "trimmomatic              /home/olga/miniconda3/envs/trimmomatic\n",
      "velocyto                 /home/olga/miniconda3/envs/velocyto\n",
      "\n"
     ]
    }
   ],
   "source": [
    "! conda env list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Usage: samtools view [options] <in.bam>|<in.sam>|<in.cram> [region ...]\n",
      "\n",
      "Options:\n",
      "  -b       output BAM\n",
      "  -C       output CRAM (requires -T)\n",
      "  -1       use fast BAM compression (implies -b)\n",
      "  -u       uncompressed BAM output (implies -b)\n",
      "  -h       include header in SAM output\n",
      "  -H       print SAM header only (no alignments)\n",
      "  -c       print only the count of matching records\n",
      "  -o FILE  output file name [stdout]\n",
      "  -U FILE  output reads not selected by filters to FILE [null]\n",
      "  -t FILE  FILE listing reference names and lengths (see long help) [null]\n",
      "  -L FILE  only include reads overlapping this BED FILE [null]\n",
      "  -r STR   only include reads in read group STR [null]\n",
      "  -R FILE  only include reads with read group listed in FILE [null]\n",
      "  -q INT   only include reads with mapping quality >= INT [0]\n",
      "  -l STR   only include reads in library STR [null]\n",
      "  -m INT   only include reads with number of CIGAR operations consuming\n",
      "           query sequence >= INT [0]\n",
      "  -f INT   only include reads with all  of the FLAGs in INT present [0]\n",
      "  -F INT   only include reads with none of the FLAGS in INT present [0]\n",
      "  -G INT   only EXCLUDE reads with all  of the FLAGs in INT present [0]\n",
      "  -s FLOAT subsample reads (given INT.FRAC option value, 0.FRAC is the\n",
      "           fraction of templates/read pairs to keep; INT part sets seed)\n",
      "  -M       use the multi-region iterator (increases the speed, removes\n",
      "           duplicates and outputs the reads as they are ordered in the file)\n",
      "  -x STR   read tag to strip (repeatable) [null]\n",
      "  -B       collapse the backward CIGAR operation\n",
      "  -?       print long help, including note about region specification\n",
      "  -S       ignored (input format is auto-detected)\n",
      "      --input-fmt-option OPT[=VAL]\n",
      "               Specify a single input file format option in the form\n",
      "               of OPTION or OPTION=VALUE\n",
      "  -O, --output-fmt FORMAT[,OPT[=VAL]]...\n",
      "               Specify output format (SAM, BAM, CRAM)\n",
      "      --output-fmt-option OPT[=VAL]\n",
      "               Specify a single output file format option in the form\n",
      "               of OPTION or OPTION=VALUE\n",
      "  -T, --reference FILE\n",
      "               Reference sequence FASTA FILE [null]\n",
      "  -@, --threads INT\n",
      "               Number of additional threads to use [0]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "! samtools view"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- `chr5:142,901,115-142,908,754` is Actb locus in mm10 genome\n",
    "- `-s 0.1` - subsample to 10% of reads"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "samtools view -b -s 0.1 S10_L001_R1_001Aligned.sortedByCoord.out.bam \\\n",
    "    chr5:142,901,115-142,908,754 \\\n",
    "    > actb.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "! samtools index actb.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-rw-rw-r-- 1 olga olga 1.9M Aug  8 06:01 actb.bam\n"
     ]
    }
   ],
   "source": [
    "ls -lha actb.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "A00111:245:HGKNTDMXX:1:2257:5629:23390\t16\tchr5\t142545502\t3\t13M357918N77M\t*\t0\t0\tTGGCTTTTTGGTGGTTGAGGGACTTCCTGTAACCAATTATTTCATGGATACTTGGAATGACTATTAAAAAAAGACAAAACCAAAAAAAAA\t:,F,F:F:,FF,FF,F,FFF:F,:FF:FFFFFFFF,:F,FFF::F,,FFF,FFFFFFFFF,F,FFF:FFFFFFFFFFFFFFFFFFF:FFF\tNH:i:2\tHI:i:1\tAS:i:81\tnM:i:2\tRG:Z:S10_L001_R1_001\n",
      "A00111:245:HGKNTDMXX:1:1476:10818:34695\t16\tchr5\t142645502\t255\t25M257596N65M\t*\t0\t0\tTTTTTTTTTTTATTTTTTTTTTTTTTAAGGTAAGGTGTGCACTTTTATTGGTCTCAAGTCAGTGTACAGGCCAGCCCTGGCTGCCTCAAC\tFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF:FFFFFFFFFFFFFFFFFFFFF:FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\tNH:i:1\tHI:i:1\tAS:i:78\tnM:i:0\tRG:Z:S10_L001_R1_001\n",
      "A00111:245:HGKNTDMXX:1:1417:28411:10175\t16\tchr5\t142645505\t255\t16M257602N74M\t*\t0\t0\tTTTTTTTTATTTTTTTTAAGGTAAGGTGTGCACTTTTATTGGTCTCAAGTCAGTGTACAGGCCAGCCCTGGCTGCCTCAACACCTCAACC\tFFFFFFFF,F:FFF::FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\tNH:i:1\tHI:i:1\tAS:i:78\tnM:i:0\tRG:Z:S10_L001_R1_001\n",
      "A00111:245:HGKNTDMXX:1:2227:31891:33395\t16\tchr5\t142645505\t255\t17M257601N73M\t*\t0\t0\tTTTTTTTTATTTTTTTTTAAGGTAAGGTGTGCACTTTTATTGGTCTCAAGTCAGTGTACAGGCCAGCCCTGGCTGCCTCAACACCTCAAC\tFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF:FFFFF,FFF:FF:FFFFFFFFFFFFFFFFFFFFF:FFFFFFFFFFFFFFFFFFFF\tNH:i:1\tHI:i:1\tAS:i:78\tnM:i:0\tRG:Z:S10_L001_R1_001\n",
      "A00111:245:HGKNTDMXX:1:2301:9733:36448\t272\tchr5\t142670923\t1\t9M232521N71M10S\t*\t0\t0\tACCTTTAACCCATTATTTCATGTATACTTTTATTTACTATTAAAAAAAGACAAAACCAAAAAAAAAAAAAAAAAAAACAAAACAATGTAC\t,:F:,:FF:,,,,,,F:,:,:,:,:F,F:,,F,F,F,,FF,FF:FFFF,F,FFF,,,FFFFFFFFFFF:FFFFFFFFFF:FFFFFFFFFF\tNH:i:3\tHI:i:2\tAS:i:60\tnM:i:8\tRG:Z:S10_L001_R1_001\n",
      "A00111:245:HGKNTDMXX:1:1215:22761:24721\t0\tchr5\t142677292\t255\t21M256948N69M\t*\t0\t0\tAAAGAAAGAAAGAGAGAGAGAAGGGGCTGGTGAGATGGCTCAGTGGTTAAGAGCGCCGACTGCTCTTCCAAAGGTCCTGAGTTCAAATCC\tFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\tNH:i:1\tHI:i:1\tAS:i:88\tnM:i:0\tRG:Z:S10_L001_R1_001\n",
      "A00111:245:HGKNTDMXX:1:1474:10411:10473\t0\tchr5\t142677295\t255\t18M256948N72M\t*\t0\t0\tGAAAGAAAGAGAGAGAGAAGGGGCTGGTGAGATGGCTCAGTGGTTAAGAGCGCCGACTGCTCTTCCAAAGGTCCTGAGTTCAAATCCCAG\tFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\tNH:i:1\tHI:i:1\tAS:i:88\tnM:i:0\tRG:Z:S10_L001_R1_001\n",
      "A00111:245:HGKNTDMXX:1:2381:15040:11600\t16\tchr5\t142694062\t255\t33M209026N53M90N4M\t*\t0\t0\tTGTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTGTAAGGTAAGGTGTGCACTTTTATTGGTCTCAAGTCAGTGTACAGGCCAGCCCTGG\t,:,FFFFFF:FF,FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\tNH:i:1\tHI:i:1\tAS:i:80\tnM:i:0\tRG:Z:S10_L001_R1_001\n",
      "A00111:245:HGKNTDMXX:1:1350:26621:13964\t272\tchr5\t142694062\t3\t29M209032N61M\t*\t0\t0\tTGTTTTTTTTTTTTTTTTTTTTTTTTTTTTAAGGTAAGGTGTGCACTTTTATTGGTCTCAAGTCAGTGTACAGGCCAGCCCTGGCTGCCT\tF,FFFF:FFFFFFFFFFFFF::FFFFFFFFFFFFFFFFFFFFFFFFFFFFF,FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\tNH:i:2\tHI:i:2\tAS:i:78\tnM:i:0\tRG:Z:S10_L001_R1_001\n",
      "A00111:245:HGKNTDMXX:1:2286:28809:32722\t16\tchr5\t142694073\t255\t29M209016N61M\t*\t0\t0\tTTTTTTTTTTTTTTTTTTTTTTTTTTTGCTTGTGTAAGGTAAGGTGTGCACTTTTATTGGTCTCAAGTCAGTGTACAGGCCAGCCCTGGC\tFFF,:FFFFFFF:FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\tNH:i:1\tHI:i:1\tAS:i:78\tnM:i:0\tRG:Z:S10_L001_R1_001\n",
      "samtools view: writing to standard output failed: Broken pipe\n",
      "samtools view: error closing standard output: -1\n"
     ]
    }
   ],
   "source": [
    "! samtools view actb.bam | head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "73163 + 0 in total (QC-passed reads + QC-failed reads)\n",
      "541 + 0 secondary\n",
      "0 + 0 supplementary\n",
      "0 + 0 duplicates\n",
      "73163 + 0 mapped (100.00% : N/A)\n",
      "0 + 0 paired in sequencing\n",
      "0 + 0 read1\n",
      "0 + 0 read2\n",
      "0 + 0 properly paired (N/A : N/A)\n",
      "0 + 0 with itself and mate mapped\n",
      "0 + 0 singletons (N/A : N/A)\n",
      "0 + 0 with mate mapped to a different chr\n",
      "0 + 0 with mate mapped to a different chr (mapQ>=5)\n"
     ]
    }
   ],
   "source": [
    "! samtools flagstat actb.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# This file was produced by samtools stats (1.9+htslib-1.9) and can be plotted using plot-bamstats\n",
      "# This file contains statistics for all reads.\n",
      "# The command line was:  stats actb.bam\n",
      "# CHK, Checksum\t[2]Read Names\t[3]Sequences\t[4]Qualities\n",
      "# CHK, CRC32 of reads which passed filtering followed by addition (32bit overflow)\n",
      "CHK\t437b5e4c\tbd26c602\te3f279f4\n",
      "# Summary Numbers. Use `grep ^SN | cut -f 2-` to extract this part.\n",
      "SN\traw total sequences:\t72622\n",
      "SN\tfiltered sequences:\t0\n",
      "SN\tsequences:\t72622\n",
      "SN\tis sorted:\t1\n",
      "SN\t1st fragments:\t72622\n",
      "SN\tlast fragments:\t0\n",
      "SN\treads mapped:\t72622\n",
      "SN\treads mapped and paired:\t0\t# paired-end technology bit set + both mates mapped\n",
      "SN\treads unmapped:\t0\n",
      "SN\treads properly paired:\t0\t# proper-pair bit set\n",
      "SN\treads paired:\t0\t# paired-end technology bit set\n",
      "SN\treads duplicated:\t0\t# PCR or optical duplicate bit set\n",
      "SN\treads MQ0:\t3\t# mapped and MQ=0\n",
      "SN\treads QC failed:\t0\n",
      "SN\tnon-primary alignments:\t541\n",
      "SN\ttotal length:\t6535980\t# ignores clipping\n",
      "SN\ttotal first fragment length:\t6535980\t# ignores clipping\n",
      "SN\ttotal last fragment length:\t0\t# ignores clipping\n",
      "SN\tbases mapped:\t6535980\t# ignores clipping\n",
      "SN\tbases mapped (cigar):\t6440257\t# more accurate\n",
      "SN\tbases trimmed:\t0\n",
      "SN\tbases duplicated:\t0\n",
      "SN\tmismatches:\t0\t# from NM fields\n",
      "SN\terror rate:\t0.000000e+00\t# mismatches / bases mapped (cigar)\n",
      "SN\taverage length:\t90\n",
      "SN\taverage first fragment length:\t90\n",
      "SN\taverage last fragment length:\t0\n",
      "SN\tmaximum length:\t90\n",
      "SN\tmaximum first fragment length:\t0\n",
      "SN\tmaximum last fragment length:\t0\n",
      "SN\taverage quality:\t35.8\n",
      "SN\tinsert size average:\t0.0\n",
      "SN\tinsert size standard deviation:\t0.0\n",
      "SN\tinward oriented pairs:\t0\n",
      "SN\toutward oriented pairs:\t0\n",
      "SN\tpairs with other orientation:\t0\n",
      "SN\tpairs on different chromosomes:\t0\n",
      "SN\tpercentage of properly paired reads (%):\t0.0\n",
      "# First Fragment Qualities. Use `grep ^FFQ | cut -f 2-` to extract this part.\n",
      "# Columns correspond to qualities and rows to cycles. First column is the cycle number.\n",
      "# Last Fragment Qualities. Use `grep ^LFQ | cut -f 2-` to extract this part.\n",
      "# Columns correspond to qualities and rows to cycles. First column is the cycle number.\n",
      "# GC Content of first fragments. Use `grep ^GCF | cut -f 2-` to extract this part.\n",
      "GCF\t4.27\t0\n",
      "GCF\t9.05\t2\n",
      "GCF\t10.80\t1\n",
      "GCF\t12.56\t0\n",
      "GCF\t13.57\t1\n",
      "GCF\t14.57\t3\n",
      "GCF\t15.83\t11\n",
      "GCF\t17.09\t12\n",
      "GCF\t18.09\t26\n",
      "GCF\t19.10\t54\n",
      "GCF\t20.35\t85\n",
      "GCF\t21.61\t168\n",
      "GCF\t22.61\t262\n",
      "GCF\t23.62\t397\n",
      "GCF\t24.62\t691\n",
      "GCF\t25.88\t1145\n",
      "GCF\t27.14\t2002\n",
      "GCF\t28.14\t3320\n",
      "GCF\t29.15\t2742\n",
      "GCF\t30.15\t1818\n",
      "GCF\t31.41\t1509\n",
      "GCF\t32.66\t1689\n",
      "GCF\t33.67\t2532\n",
      "GCF\t34.67\t2446\n",
      "GCF\t35.68\t1729\n",
      "GCF\t36.93\t1535\n",
      "GCF\t38.19\t2535\n",
      "GCF\t39.20\t1147\n",
      "GCF\t40.20\t1662\n",
      "GCF\t41.46\t2537\n",
      "GCF\t42.71\t2340\n",
      "GCF\t43.72\t2027\n",
      "GCF\t44.72\t1847\n",
      "GCF\t45.73\t3494\n",
      "GCF\t46.98\t1559\n",
      "GCF\t48.24\t1192\n",
      "GCF\t49.25\t1564\n",
      "GCF\t50.25\t1410\n",
      "GCF\t51.26\t1778\n",
      "GCF\t52.51\t2635\n",
      "GCF\t53.77\t1341\n",
      "GCF\t54.77\t2347\n",
      "GCF\t55.78\t4910\n",
      "GCF\t56.78\t3307\n",
      "GCF\t58.04\t1626\n",
      "GCF\t59.30\t1121\n",
      "GCF\t60.30\t671\n",
      "GCF\t61.31\t1101\n",
      "GCF\t62.56\t1731\n",
      "GCF\t63.82\t2336\n",
      "GCF\t64.82\t171\n",
      "GCF\t65.83\t22\n",
      "GCF\t66.83\t6\n",
      "GCF\t68.09\t8\n",
      "GCF\t69.35\t6\n",
      "GCF\t70.35\t4\n",
      "GCF\t71.36\t1\n",
      "GCF\t72.36\t0\n",
      "GCF\t73.62\t3\n",
      "GCF\t76.38\t0\n",
      "GCF\t79.15\t1\n",
      "GCF\t81.41\t0\n",
      "GCF\t83.67\t1\n",
      "# GC Content of last fragments. Use `grep ^GCL | cut -f 2-` to extract this part.\n",
      "# ACGT content per cycle. Use `grep ^GCC | cut -f 2-` to extract this part. The columns are: cycle; A,C,G,T base counts as a percentage of all A/C/G/T bases [%]; and N and O counts as a percentage of all A/C/G/T bases [%]\n",
      "GCC\t1\t18.25\t30.98\t34.99\t15.79\t0.13\t0.00\n",
      "GCC\t2\t32.58\t17.56\t25.78\t24.08\t0.00\t0.00\n",
      "GCC\t3\t30.35\t18.48\t26.02\t25.15\t0.00\t0.00\n",
      "GCC\t4\t28.67\t21.04\t28.17\t22.13\t0.00\t0.00\n",
      "GCC\t5\t27.45\t19.65\t25.98\t26.91\t0.00\t0.00\n",
      "GCC\t6\t28.76\t24.09\t24.64\t22.51\t0.00\t0.00\n",
      "GCC\t7\t33.74\t16.99\t23.47\t25.80\t0.00\t0.00\n",
      "GCC\t8\t26.03\t23.19\t28.53\t22.25\t0.00\t0.00\n",
      "GCC\t9\t29.75\t21.47\t27.59\t21.18\t0.00\t0.00\n",
      "GCC\t10\t32.22\t17.49\t22.43\t27.87\t0.00\t0.00\n",
      "GCC\t11\t29.60\t22.40\t27.44\t20.57\t0.00\t0.00\n",
      "GCC\t12\t27.92\t22.32\t27.77\t21.99\t0.00\t0.00\n",
      "GCC\t13\t31.17\t18.56\t29.95\t20.32\t0.00\t0.00\n",
      "GCC\t14\t30.46\t19.79\t26.38\t23.36\t0.00\t0.00\n",
      "GCC\t15\t29.28\t18.85\t26.42\t25.45\t0.01\t0.00\n",
      "GCC\t16\t31.36\t19.82\t26.50\t22.32\t0.00\t0.00\n",
      "GCC\t17\t28.89\t22.34\t26.14\t22.63\t0.00\t0.00\n",
      "GCC\t18\t27.52\t21.80\t27.89\t22.80\t0.00\t0.00\n",
      "GCC\t19\t29.78\t21.90\t24.30\t24.03\t0.00\t0.00\n",
      "GCC\t20\t31.33\t19.91\t25.76\t22.99\t0.00\t0.00\n",
      "GCC\t21\t32.13\t20.32\t26.38\t21.17\t0.00\t0.00\n",
      "GCC\t22\t29.28\t19.15\t27.70\t23.87\t0.00\t0.00\n",
      "GCC\t23\t30.71\t19.82\t22.33\t27.14\t0.00\t0.00\n",
      "GCC\t24\t31.90\t22.74\t22.98\t22.39\t0.00\t0.00\n",
      "GCC\t25\t30.03\t21.75\t23.80\t24.42\t0.00\t0.00\n",
      "GCC\t26\t32.42\t19.91\t23.89\t23.77\t0.00\t0.00\n",
      "GCC\t27\t30.62\t19.51\t24.97\t24.91\t0.00\t0.00\n",
      "GCC\t28\t31.39\t22.43\t23.20\t22.98\t0.00\t0.00\n",
      "GCC\t29\t29.93\t23.16\t23.98\t22.93\t0.01\t0.00\n",
      "GCC\t30\t30.22\t22.31\t24.91\t22.57\t0.00\t0.00\n",
      "GCC\t31\t30.17\t21.29\t24.57\t23.98\t0.00\t0.00\n",
      "GCC\t32\t32.93\t18.71\t22.38\t25.98\t0.00\t0.00\n",
      "GCC\t33\t31.47\t19.26\t25.27\t24.00\t0.00\t0.00\n",
      "GCC\t34\t32.52\t23.59\t23.25\t20.65\t0.00\t0.00\n",
      "GCC\t35\t32.71\t22.02\t22.00\t23.27\t0.00\t0.00\n",
      "GCC\t36\t34.25\t19.50\t24.32\t21.93\t0.01\t0.00\n",
      "GCC\t37\t33.12\t19.42\t23.90\t23.56\t0.00\t0.00\n",
      "GCC\t38\t33.84\t17.37\t22.28\t26.51\t0.01\t0.00\n",
      "GCC\t39\t31.95\t19.32\t24.85\t23.88\t0.00\t0.00\n",
      "GCC\t40\t31.41\t19.82\t21.61\t27.16\t0.00\t0.00\n",
      "GCC\t41\t35.74\t21.05\t22.51\t20.70\t0.01\t0.00\n",
      "GCC\t42\t34.92\t19.36\t22.74\t22.98\t0.00\t0.00\n",
      "GCC\t43\t30.72\t20.59\t22.03\t26.67\t0.00\t0.00\n",
      "GCC\t44\t34.22\t18.10\t23.05\t24.62\t0.00\t0.00\n",
      "GCC\t45\t35.84\t19.24\t23.28\t21.64\t0.00\t0.00\n",
      "GCC\t46\t33.04\t21.39\t23.90\t21.67\t0.00\t0.00\n",
      "GCC\t47\t32.25\t19.52\t24.10\t24.13\t0.00\t0.00\n",
      "GCC\t48\t33.56\t17.55\t25.01\t23.87\t0.00\t0.00\n",
      "GCC\t49\t36.74\t19.82\t22.82\t20.62\t0.00\t0.00\n",
      "GCC\t50\t31.63\t21.48\t22.22\t24.67\t0.01\t0.00\n",
      "GCC\t51\t34.17\t17.79\t23.69\t24.34\t0.00\t0.00\n",
      "GCC\t52\t34.17\t21.07\t23.37\t21.39\t0.00\t0.00\n",
      "GCC\t53\t33.47\t22.28\t22.93\t21.31\t0.02\t0.00\n",
      "GCC\t54\t37.27\t19.32\t21.55\t21.86\t0.00\t0.00\n",
      "GCC\t55\t35.77\t19.64\t22.88\t21.71\t0.11\t0.00\n",
      "GCC\t56\t33.80\t18.74\t24.42\t23.04\t0.02\t0.00\n",
      "GCC\t57\t32.83\t21.14\t24.30\t21.73\t0.00\t0.00\n",
      "GCC\t58\t37.42\t20.00\t22.06\t20.53\t0.00\t0.00\n",
      "GCC\t59\t35.87\t20.58\t22.07\t21.48\t0.00\t0.00\n",
      "GCC\t60\t34.70\t21.64\t22.18\t21.49\t0.00\t0.00\n",
      "GCC\t61\t33.39\t19.59\t23.38\t23.63\t0.00\t0.00\n",
      "GCC\t62\t35.49\t19.82\t23.34\t21.34\t0.00\t0.00\n",
      "GCC\t63\t34.80\t22.62\t20.42\t22.15\t0.00\t0.00\n",
      "GCC\t64\t35.19\t18.44\t24.14\t22.23\t0.00\t0.00\n",
      "GCC\t65\t37.17\t17.88\t21.39\t23.56\t0.00\t0.00\n",
      "GCC\t66\t36.12\t19.31\t23.04\t21.53\t0.00\t0.00\n",
      "GCC\t67\t40.47\t19.36\t19.97\t20.21\t0.00\t0.00\n",
      "GCC\t68\t38.26\t21.79\t19.49\t20.46\t0.00\t0.00\n",
      "GCC\t69\t38.90\t18.44\t22.68\t19.97\t0.00\t0.00\n",
      "GCC\t70\t39.23\t19.82\t20.28\t20.67\t0.00\t0.00\n",
      "GCC\t71\t35.72\t19.88\t22.61\t21.79\t0.00\t0.00\n",
      "GCC\t72\t36.00\t17.90\t23.19\t22.91\t0.00\t0.00\n",
      "GCC\t73\t36.57\t20.74\t21.11\t21.57\t0.01\t0.00\n",
      "GCC\t74\t36.60\t19.62\t23.30\t20.48\t0.01\t0.00\n",
      "GCC\t75\t39.09\t16.49\t22.36\t22.06\t0.00\t0.00\n",
      "GCC\t76\t39.43\t19.25\t20.74\t20.58\t0.00\t0.00\n",
      "GCC\t77\t39.13\t20.27\t21.24\t19.37\t0.00\t0.00\n",
      "GCC\t78\t36.78\t20.51\t22.99\t19.72\t0.00\t0.00\n",
      "GCC\t79\t37.53\t18.42\t21.95\t22.11\t0.00\t0.00\n",
      "GCC\t80\t41.37\t19.43\t20.22\t18.98\t0.00\t0.00\n",
      "GCC\t81\t38.61\t20.46\t21.21\t19.73\t0.00\t0.00\n",
      "GCC\t82\t37.81\t20.12\t20.02\t22.05\t0.00\t0.00\n",
      "GCC\t83\t36.51\t19.87\t20.03\t23.59\t0.00\t0.00\n",
      "GCC\t84\t40.98\t17.63\t17.80\t23.58\t0.00\t0.00\n",
      "GCC\t85\t37.97\t20.99\t21.16\t19.88\t0.00\t0.00\n",
      "GCC\t86\t39.34\t18.61\t20.34\t21.72\t0.00\t0.00\n",
      "GCC\t87\t38.59\t21.56\t19.09\t20.76\t0.00\t0.00\n",
      "GCC\t88\t38.79\t19.61\t19.88\t21.72\t0.00\t0.00\n",
      "GCC\t89\t35.47\t19.14\t22.96\t22.43\t0.00\t0.00\n",
      "GCC\t90\t35.86\t19.99\t23.38\t20.78\t0.01\t0.00\n",
      "# ACGT content per cycle for first fragments. Use `grep ^FBC | cut -f 2-` to extract this part. The columns are: cycle; A,C,G,T base counts as a percentage of all A/C/G/T bases [%]; and N and O counts as a percentage of all A/C/G/T bases [%]\n",
      "# ACGT content per cycle for last fragments. Use `grep ^LBC | cut -f 2-` to extract this part. The columns are: cycle; A,C,G,T base counts as a percentage of all A/C/G/T bases [%]; and N and O counts as a percentage of all A/C/G/T bases [%]\n",
      "LBC\t1\t18.25\t30.98\t34.99\t15.79\t0.13\t0.00\n",
      "LBC\t2\t32.58\t17.56\t25.78\t24.08\t0.00\t0.00\n",
      "LBC\t3\t30.35\t18.48\t26.02\t25.15\t0.00\t0.00\n",
      "LBC\t4\t28.67\t21.04\t28.17\t22.13\t0.00\t0.00\n",
      "LBC\t5\t27.45\t19.65\t25.98\t26.91\t0.00\t0.00\n",
      "LBC\t6\t28.76\t24.09\t24.64\t22.51\t0.00\t0.00\n",
      "LBC\t7\t33.74\t16.99\t23.47\t25.80\t0.00\t0.00\n",
      "LBC\t8\t26.03\t23.19\t28.53\t22.25\t0.00\t0.00\n",
      "LBC\t9\t29.75\t21.47\t27.59\t21.18\t0.00\t0.00\n",
      "LBC\t10\t32.22\t17.49\t22.43\t27.87\t0.00\t0.00\n",
      "LBC\t11\t29.60\t22.40\t27.44\t20.57\t0.00\t0.00\n",
      "LBC\t12\t27.92\t22.32\t27.77\t21.99\t0.00\t0.00\n",
      "LBC\t13\t31.17\t18.56\t29.95\t20.32\t0.00\t0.00\n",
      "LBC\t14\t30.46\t19.79\t26.38\t23.36\t0.00\t0.00\n",
      "LBC\t15\t29.28\t18.85\t26.42\t25.45\t0.01\t0.00\n",
      "LBC\t16\t31.36\t19.82\t26.50\t22.32\t0.00\t0.00\n",
      "LBC\t17\t28.89\t22.34\t26.14\t22.63\t0.00\t0.00\n",
      "LBC\t18\t27.52\t21.80\t27.89\t22.80\t0.00\t0.00\n",
      "LBC\t19\t29.78\t21.90\t24.30\t24.03\t0.00\t0.00\n",
      "LBC\t20\t31.33\t19.91\t25.76\t22.99\t0.00\t0.00\n",
      "LBC\t21\t32.13\t20.32\t26.38\t21.17\t0.00\t0.00\n",
      "LBC\t22\t29.28\t19.15\t27.70\t23.87\t0.00\t0.00\n",
      "LBC\t23\t30.71\t19.82\t22.33\t27.14\t0.00\t0.00\n",
      "LBC\t24\t31.90\t22.74\t22.98\t22.39\t0.00\t0.00\n",
      "LBC\t25\t30.03\t21.75\t23.80\t24.42\t0.00\t0.00\n",
      "LBC\t26\t32.42\t19.91\t23.89\t23.77\t0.00\t0.00\n",
      "LBC\t27\t30.62\t19.51\t24.97\t24.91\t0.00\t0.00\n",
      "LBC\t28\t31.39\t22.43\t23.20\t22.98\t0.00\t0.00\n",
      "LBC\t29\t29.93\t23.16\t23.98\t22.93\t0.01\t0.00\n",
      "LBC\t30\t30.22\t22.31\t24.91\t22.57\t0.00\t0.00\n",
      "LBC\t31\t30.17\t21.29\t24.57\t23.98\t0.00\t0.00\n",
      "LBC\t32\t32.93\t18.71\t22.38\t25.98\t0.00\t0.00\n",
      "LBC\t33\t31.47\t19.26\t25.27\t24.00\t0.00\t0.00\n",
      "LBC\t34\t32.52\t23.59\t23.25\t20.65\t0.00\t0.00\n",
      "LBC\t35\t32.71\t22.02\t22.00\t23.27\t0.00\t0.00\n",
      "LBC\t36\t34.25\t19.50\t24.32\t21.93\t0.01\t0.00\n",
      "LBC\t37\t33.12\t19.42\t23.90\t23.56\t0.00\t0.00\n",
      "LBC\t38\t33.84\t17.37\t22.28\t26.51\t0.01\t0.00\n",
      "LBC\t39\t31.95\t19.32\t24.85\t23.88\t0.00\t0.00\n",
      "LBC\t40\t31.41\t19.82\t21.61\t27.16\t0.00\t0.00\n",
      "LBC\t41\t35.74\t21.05\t22.51\t20.70\t0.01\t0.00\n",
      "LBC\t42\t34.92\t19.36\t22.74\t22.98\t0.00\t0.00\n",
      "LBC\t43\t30.72\t20.59\t22.03\t26.67\t0.00\t0.00\n",
      "LBC\t44\t34.22\t18.10\t23.05\t24.62\t0.00\t0.00\n",
      "LBC\t45\t35.84\t19.24\t23.28\t21.64\t0.00\t0.00\n",
      "LBC\t46\t33.04\t21.39\t23.90\t21.67\t0.00\t0.00\n",
      "LBC\t47\t32.25\t19.52\t24.10\t24.13\t0.00\t0.00\n",
      "LBC\t48\t33.56\t17.55\t25.01\t23.87\t0.00\t0.00\n",
      "LBC\t49\t36.74\t19.82\t22.82\t20.62\t0.00\t0.00\n",
      "LBC\t50\t31.63\t21.48\t22.22\t24.67\t0.01\t0.00\n",
      "LBC\t51\t34.17\t17.79\t23.69\t24.34\t0.00\t0.00\n",
      "LBC\t52\t34.17\t21.07\t23.37\t21.39\t0.00\t0.00\n",
      "LBC\t53\t33.47\t22.28\t22.93\t21.31\t0.02\t0.00\n",
      "LBC\t54\t37.27\t19.32\t21.55\t21.86\t0.00\t0.00\n",
      "LBC\t55\t35.77\t19.64\t22.88\t21.71\t0.11\t0.00\n",
      "LBC\t56\t33.80\t18.74\t24.42\t23.04\t0.02\t0.00\n",
      "LBC\t57\t32.83\t21.14\t24.30\t21.73\t0.00\t0.00\n",
      "LBC\t58\t37.42\t20.00\t22.06\t20.53\t0.00\t0.00\n",
      "LBC\t59\t35.87\t20.58\t22.07\t21.48\t0.00\t0.00\n",
      "LBC\t60\t34.70\t21.64\t22.18\t21.49\t0.00\t0.00\n",
      "LBC\t61\t33.39\t19.59\t23.38\t23.63\t0.00\t0.00\n",
      "LBC\t62\t35.49\t19.82\t23.34\t21.34\t0.00\t0.00\n",
      "LBC\t63\t34.80\t22.62\t20.42\t22.15\t0.00\t0.00\n",
      "LBC\t64\t35.19\t18.44\t24.14\t22.23\t0.00\t0.00\n",
      "LBC\t65\t37.17\t17.88\t21.39\t23.56\t0.00\t0.00\n",
      "LBC\t66\t36.12\t19.31\t23.04\t21.53\t0.00\t0.00\n",
      "LBC\t67\t40.47\t19.36\t19.97\t20.21\t0.00\t0.00\n",
      "LBC\t68\t38.26\t21.79\t19.49\t20.46\t0.00\t0.00\n",
      "LBC\t69\t38.90\t18.44\t22.68\t19.97\t0.00\t0.00\n",
      "LBC\t70\t39.23\t19.82\t20.28\t20.67\t0.00\t0.00\n",
      "LBC\t71\t35.72\t19.88\t22.61\t21.79\t0.00\t0.00\n",
      "LBC\t72\t36.00\t17.90\t23.19\t22.91\t0.00\t0.00\n",
      "LBC\t73\t36.57\t20.74\t21.11\t21.57\t0.01\t0.00\n",
      "LBC\t74\t36.60\t19.62\t23.30\t20.48\t0.01\t0.00\n",
      "LBC\t75\t39.09\t16.49\t22.36\t22.06\t0.00\t0.00\n",
      "LBC\t76\t39.43\t19.25\t20.74\t20.58\t0.00\t0.00\n",
      "LBC\t77\t39.13\t20.27\t21.24\t19.37\t0.00\t0.00\n",
      "LBC\t78\t36.78\t20.51\t22.99\t19.72\t0.00\t0.00\n",
      "LBC\t79\t37.53\t18.42\t21.95\t22.11\t0.00\t0.00\n",
      "LBC\t80\t41.37\t19.43\t20.22\t18.98\t0.00\t0.00\n",
      "LBC\t81\t38.61\t20.46\t21.21\t19.73\t0.00\t0.00\n",
      "LBC\t82\t37.81\t20.12\t20.02\t22.05\t0.00\t0.00\n",
      "LBC\t83\t36.51\t19.87\t20.03\t23.59\t0.00\t0.00\n",
      "LBC\t84\t40.98\t17.63\t17.80\t23.58\t0.00\t0.00\n",
      "LBC\t85\t37.97\t20.99\t21.16\t19.88\t0.00\t0.00\n",
      "LBC\t86\t39.34\t18.61\t20.34\t21.72\t0.00\t0.00\n",
      "LBC\t87\t38.59\t21.56\t19.09\t20.76\t0.00\t0.00\n",
      "LBC\t88\t38.79\t19.61\t19.88\t21.72\t0.00\t0.00\n",
      "LBC\t89\t35.47\t19.14\t22.96\t22.43\t0.00\t0.00\n",
      "LBC\t90\t35.86\t19.99\t23.38\t20.78\t0.01\t0.00\n",
      "# Insert sizes. Use `grep ^IS | cut -f 2-` to extract this part. The columns are: insert size, pairs total, inward oriented pairs, outward oriented pairs, other pairs\n",
      "# Read lengths. Use `grep ^RL | cut -f 2-` to extract this part. The columns are: read length, count\n",
      "RL\t90\t72622\n",
      "# Read lengths - first fragments. Use `grep ^FRL | cut -f 2-` to extract this part. The columns are: read length, count\n",
      "# Read lengths - last fragments. Use `grep ^LRL | cut -f 2-` to extract this part. The columns are: read length, count\n",
      "# Indel distribution. Use `grep ^ID | cut -f 2-` to extract this part. The columns are: length, number of insertions, number of deletions\n",
      "ID\t1\t2096\t1478\n",
      "ID\t2\t780\t296\n",
      "ID\t3\t275\t73\n",
      "ID\t4\t121\t20\n",
      "ID\t5\t34\t3\n",
      "ID\t6\t4\t0\n",
      "ID\t7\t2\t0\n",
      "ID\t9\t0\t1\n",
      "# Indels per cycle. Use `grep ^IC | cut -f 2-` to extract this part. The columns are: cycle, number of insertions (fwd), .. (rev) , number of deletions (fwd), .. (rev)\n",
      "IC\t7\t0\t0\t0\t2\n",
      "IC\t9\t0\t0\t0\t10\n",
      "IC\t10\t0\t1\t0\t1\n",
      "IC\t11\t0\t11\t0\t11\n",
      "IC\t12\t0\t12\t0\t11\n",
      "IC\t13\t0\t19\t0\t19\n",
      "IC\t14\t0\t25\t0\t18\n",
      "IC\t15\t0\t42\t0\t21\n",
      "IC\t16\t0\t43\t0\t40\n",
      "IC\t17\t0\t64\t0\t20\n",
      "IC\t18\t0\t56\t0\t10\n",
      "IC\t19\t0\t30\t0\t36\n",
      "IC\t20\t0\t32\t0\t13\n",
      "IC\t21\t0\t29\t0\t19\n",
      "IC\t22\t0\t24\t0\t19\n",
      "IC\t23\t0\t21\t0\t32\n",
      "IC\t24\t0\t18\t0\t20\n",
      "IC\t25\t0\t18\t0\t23\n",
      "IC\t26\t0\t27\t0\t34\n",
      "IC\t27\t0\t16\t0\t36\n",
      "IC\t28\t0\t34\t0\t45\n",
      "IC\t29\t0\t28\t0\t15\n",
      "IC\t30\t0\t23\t0\t13\n",
      "IC\t31\t0\t11\t0\t61\n",
      "IC\t32\t0\t17\t0\t43\n",
      "IC\t33\t0\t50\t0\t17\n",
      "IC\t34\t0\t20\t0\t16\n",
      "IC\t35\t0\t130\t0\t25\n",
      "IC\t36\t0\t49\t0\t53\n",
      "IC\t37\t0\t55\t0\t18\n",
      "IC\t38\t0\t38\t0\t49\n",
      "IC\t39\t0\t95\t0\t24\n",
      "IC\t40\t0\t31\t0\t27\n",
      "IC\t41\t0\t43\t0\t22\n",
      "IC\t42\t0\t71\t0\t18\n",
      "IC\t43\t0\t57\t0\t33\n",
      "IC\t44\t0\t93\t0\t14\n",
      "IC\t45\t0\t55\t0\t18\n",
      "IC\t46\t0\t32\t0\t34\n",
      "IC\t47\t0\t139\t0\t55\n",
      "IC\t48\t0\t68\t0\t16\n",
      "IC\t49\t0\t20\t0\t19\n",
      "IC\t50\t0\t17\t0\t25\n",
      "IC\t51\t0\t49\t0\t37\n",
      "IC\t52\t0\t134\t0\t11\n",
      "IC\t53\t0\t14\t0\t49\n",
      "IC\t54\t0\t96\t0\t10\n",
      "IC\t55\t0\t41\t0\t29\n",
      "IC\t56\t0\t30\t0\t41\n",
      "IC\t57\t0\t27\t0\t10\n",
      "IC\t58\t0\t21\t0\t13\n",
      "IC\t59\t0\t43\t0\t49\n",
      "IC\t60\t0\t26\t0\t47\n",
      "IC\t61\t0\t14\t0\t35\n",
      "IC\t62\t0\t91\t0\t50\n",
      "IC\t63\t0\t105\t0\t108\n",
      "IC\t64\t0\t51\t0\t33\n",
      "IC\t65\t0\t38\t0\t54\n",
      "IC\t66\t0\t39\t0\t21\n",
      "IC\t67\t0\t69\t0\t37\n",
      "IC\t68\t0\t15\t0\t11\n",
      "IC\t69\t0\t81\t0\t15\n",
      "IC\t70\t0\t30\t0\t7\n",
      "IC\t71\t0\t59\t0\t13\n",
      "IC\t72\t0\t95\t0\t18\n",
      "IC\t73\t0\t16\t0\t38\n",
      "IC\t74\t0\t31\t0\t12\n",
      "IC\t75\t0\t107\t0\t14\n",
      "IC\t76\t0\t60\t0\t16\n",
      "IC\t77\t0\t40\t0\t10\n",
      "IC\t78\t0\t43\t0\t14\n",
      "IC\t79\t0\t143\t0\t4\n",
      "IC\t80\t0\t40\t0\t9\n",
      "IC\t81\t0\t0\t0\t1\n",
      "# Coverage distribution. Use `grep ^COV | cut -f 2-` to extract this part.\n",
      "COV\t[1-1]\t1\t1794\n",
      "COV\t[2-2]\t2\t685\n",
      "COV\t[3-3]\t3\t283\n",
      "COV\t[4-4]\t4\t156\n",
      "COV\t[5-5]\t5\t209\n",
      "COV\t[6-6]\t6\t206\n",
      "COV\t[7-7]\t7\t18\n",
      "COV\t[8-8]\t8\t12\n",
      "COV\t[9-9]\t9\t15\n",
      "COV\t[10-10]\t10\t5\n",
      "COV\t[11-11]\t11\t29\n",
      "COV\t[12-12]\t12\t35\n",
      "COV\t[13-13]\t13\t63\n",
      "COV\t[14-14]\t14\t55\n",
      "COV\t[15-15]\t15\t40\n",
      "COV\t[16-16]\t16\t46\n",
      "COV\t[17-17]\t17\t111\n",
      "COV\t[18-18]\t18\t64\n",
      "COV\t[19-19]\t19\t46\n",
      "COV\t[20-20]\t20\t43\n",
      "COV\t[21-21]\t21\t40\n",
      "COV\t[22-22]\t22\t45\n",
      "COV\t[23-23]\t23\t71\n",
      "COV\t[24-24]\t24\t51\n",
      "COV\t[25-25]\t25\t20\n",
      "COV\t[26-26]\t26\t14\n",
      "COV\t[27-27]\t27\t23\n",
      "COV\t[28-28]\t28\t17\n",
      "COV\t[29-29]\t29\t12\n",
      "COV\t[30-30]\t30\t6\n",
      "COV\t[31-31]\t31\t11\n",
      "COV\t[32-32]\t32\t14\n",
      "COV\t[33-33]\t33\t14\n",
      "COV\t[34-34]\t34\t3\n",
      "COV\t[35-35]\t35\t5\n",
      "COV\t[36-36]\t36\t8\n",
      "COV\t[37-37]\t37\t5\n",
      "COV\t[38-38]\t38\t12\n",
      "COV\t[39-39]\t39\t7\n",
      "COV\t[40-40]\t40\t7\n",
      "COV\t[41-41]\t41\t3\n",
      "COV\t[42-42]\t42\t12\n",
      "COV\t[43-43]\t43\t10\n",
      "COV\t[44-44]\t44\t5\n",
      "COV\t[45-45]\t45\t7\n",
      "COV\t[46-46]\t46\t5\n",
      "COV\t[47-47]\t47\t4\n",
      "COV\t[48-48]\t48\t6\n",
      "COV\t[49-49]\t49\t6\n",
      "COV\t[50-50]\t50\t10\n",
      "COV\t[51-51]\t51\t8\n",
      "COV\t[52-52]\t52\t9\n",
      "COV\t[53-53]\t53\t11\n",
      "COV\t[54-54]\t54\t6\n",
      "COV\t[55-55]\t55\t1\n",
      "COV\t[56-56]\t56\t11\n",
      "COV\t[57-57]\t57\t12\n",
      "COV\t[58-58]\t58\t3\n",
      "COV\t[59-59]\t59\t3\n",
      "COV\t[60-60]\t60\t6\n",
      "COV\t[61-61]\t61\t5\n",
      "COV\t[62-62]\t62\t6\n",
      "COV\t[63-63]\t63\t3\n",
      "COV\t[64-64]\t64\t8\n",
      "COV\t[65-65]\t65\t6\n",
      "COV\t[66-66]\t66\t5\n",
      "COV\t[67-67]\t67\t3\n",
      "COV\t[68-68]\t68\t2\n",
      "COV\t[69-69]\t69\t6\n",
      "COV\t[70-70]\t70\t4\n",
      "COV\t[71-71]\t71\t10\n",
      "COV\t[72-72]\t72\t5\n",
      "COV\t[73-73]\t73\t2\n",
      "COV\t[74-74]\t74\t6\n",
      "COV\t[75-75]\t75\t19\n",
      "COV\t[76-76]\t76\t23\n",
      "COV\t[77-77]\t77\t7\n",
      "COV\t[78-78]\t78\t6\n",
      "COV\t[79-79]\t79\t5\n",
      "COV\t[80-80]\t80\t2\n",
      "COV\t[81-81]\t81\t1\n",
      "COV\t[82-82]\t82\t4\n",
      "COV\t[83-83]\t83\t2\n",
      "COV\t[84-84]\t84\t2\n",
      "COV\t[85-85]\t85\t1\n",
      "COV\t[87-87]\t87\t1\n",
      "COV\t[88-88]\t88\t3\n",
      "COV\t[89-89]\t89\t5\n",
      "COV\t[90-90]\t90\t2\n",
      "COV\t[93-93]\t93\t2\n",
      "COV\t[94-94]\t94\t3\n",
      "COV\t[95-95]\t95\t2\n",
      "COV\t[96-96]\t96\t1\n",
      "COV\t[97-97]\t97\t1\n",
      "COV\t[98-98]\t98\t3\n",
      "COV\t[100-100]\t100\t3\n",
      "COV\t[103-103]\t103\t1\n",
      "COV\t[104-104]\t104\t3\n",
      "COV\t[105-105]\t105\t1\n",
      "COV\t[106-106]\t106\t2\n",
      "COV\t[107-107]\t107\t3\n",
      "COV\t[108-108]\t108\t1\n",
      "COV\t[109-109]\t109\t3\n",
      "COV\t[110-110]\t110\t2\n",
      "COV\t[111-111]\t111\t2\n",
      "COV\t[112-112]\t112\t1\n",
      "COV\t[113-113]\t113\t2\n",
      "COV\t[114-114]\t114\t1\n",
      "COV\t[116-116]\t116\t3\n",
      "COV\t[117-117]\t117\t3\n",
      "COV\t[118-118]\t118\t2\n",
      "COV\t[119-119]\t119\t3\n",
      "COV\t[120-120]\t120\t4\n",
      "COV\t[121-121]\t121\t2\n",
      "COV\t[122-122]\t122\t2\n",
      "COV\t[123-123]\t123\t2\n",
      "COV\t[124-124]\t124\t1\n",
      "COV\t[125-125]\t125\t4\n",
      "COV\t[126-126]\t126\t1\n",
      "COV\t[127-127]\t127\t1\n",
      "COV\t[128-128]\t128\t3\n",
      "COV\t[129-129]\t129\t3\n",
      "COV\t[130-130]\t130\t1\n",
      "COV\t[131-131]\t131\t2\n",
      "COV\t[132-132]\t132\t6\n",
      "COV\t[133-133]\t133\t1\n",
      "COV\t[136-136]\t136\t2\n",
      "COV\t[137-137]\t137\t3\n",
      "COV\t[138-138]\t138\t2\n",
      "COV\t[139-139]\t139\t3\n",
      "COV\t[140-140]\t140\t2\n",
      "COV\t[141-141]\t141\t3\n",
      "COV\t[142-142]\t142\t3\n",
      "COV\t[143-143]\t143\t2\n",
      "COV\t[144-144]\t144\t3\n",
      "COV\t[145-145]\t145\t2\n",
      "COV\t[149-149]\t149\t3\n",
      "COV\t[150-150]\t150\t2\n",
      "COV\t[151-151]\t151\t2\n",
      "COV\t[152-152]\t152\t5\n",
      "COV\t[153-153]\t153\t5\n",
      "COV\t[154-154]\t154\t1\n",
      "COV\t[155-155]\t155\t2\n",
      "COV\t[156-156]\t156\t1\n",
      "COV\t[158-158]\t158\t4\n",
      "COV\t[159-159]\t159\t1\n",
      "COV\t[160-160]\t160\t2\n",
      "COV\t[161-161]\t161\t2\n",
      "COV\t[162-162]\t162\t3\n",
      "COV\t[165-165]\t165\t1\n",
      "COV\t[166-166]\t166\t1\n",
      "COV\t[167-167]\t167\t1\n",
      "COV\t[168-168]\t168\t1\n",
      "COV\t[169-169]\t169\t1\n",
      "COV\t[170-170]\t170\t1\n",
      "COV\t[171-171]\t171\t2\n",
      "COV\t[173-173]\t173\t3\n",
      "COV\t[174-174]\t174\t1\n",
      "COV\t[175-175]\t175\t1\n",
      "COV\t[176-176]\t176\t3\n",
      "COV\t[177-177]\t177\t1\n",
      "COV\t[179-179]\t179\t1\n",
      "COV\t[180-180]\t180\t4\n",
      "COV\t[181-181]\t181\t2\n",
      "COV\t[182-182]\t182\t1\n",
      "COV\t[183-183]\t183\t3\n",
      "COV\t[188-188]\t188\t2\n",
      "COV\t[190-190]\t190\t2\n",
      "COV\t[191-191]\t191\t2\n",
      "COV\t[192-192]\t192\t3\n",
      "COV\t[193-193]\t193\t1\n",
      "COV\t[194-194]\t194\t1\n",
      "COV\t[195-195]\t195\t1\n",
      "COV\t[196-196]\t196\t4\n",
      "COV\t[199-199]\t199\t3\n",
      "COV\t[201-201]\t201\t2\n",
      "COV\t[202-202]\t202\t1\n",
      "COV\t[205-205]\t205\t1\n",
      "COV\t[207-207]\t207\t2\n",
      "COV\t[208-208]\t208\t1\n",
      "COV\t[210-210]\t210\t1\n",
      "COV\t[211-211]\t211\t1\n",
      "COV\t[212-212]\t212\t4\n",
      "COV\t[213-213]\t213\t1\n",
      "COV\t[214-214]\t214\t1\n",
      "COV\t[215-215]\t215\t2\n",
      "COV\t[218-218]\t218\t1\n",
      "COV\t[220-220]\t220\t1\n",
      "COV\t[222-222]\t222\t1\n",
      "COV\t[223-223]\t223\t6\n",
      "COV\t[224-224]\t224\t3\n",
      "COV\t[225-225]\t225\t1\n",
      "COV\t[226-226]\t226\t4\n",
      "COV\t[227-227]\t227\t1\n",
      "COV\t[228-228]\t228\t2\n",
      "COV\t[229-229]\t229\t2\n",
      "COV\t[230-230]\t230\t1\n",
      "COV\t[231-231]\t231\t1\n",
      "COV\t[232-232]\t232\t1\n",
      "COV\t[233-233]\t233\t1\n",
      "COV\t[234-234]\t234\t1\n",
      "COV\t[235-235]\t235\t4\n",
      "COV\t[236-236]\t236\t5\n",
      "COV\t[237-237]\t237\t4\n",
      "COV\t[238-238]\t238\t2\n",
      "COV\t[239-239]\t239\t2\n",
      "COV\t[240-240]\t240\t4\n",
      "COV\t[241-241]\t241\t4\n",
      "COV\t[242-242]\t242\t3\n",
      "COV\t[243-243]\t243\t4\n",
      "COV\t[244-244]\t244\t2\n",
      "COV\t[245-245]\t245\t3\n",
      "COV\t[246-246]\t246\t2\n",
      "COV\t[247-247]\t247\t8\n",
      "COV\t[248-248]\t248\t3\n",
      "COV\t[249-249]\t249\t3\n",
      "COV\t[250-250]\t250\t3\n",
      "COV\t[251-251]\t251\t4\n",
      "COV\t[252-252]\t252\t2\n",
      "COV\t[253-253]\t253\t2\n",
      "COV\t[254-254]\t254\t6\n",
      "COV\t[255-255]\t255\t2\n",
      "COV\t[256-256]\t256\t3\n",
      "COV\t[257-257]\t257\t3\n",
      "COV\t[258-258]\t258\t2\n",
      "COV\t[259-259]\t259\t1\n",
      "COV\t[265-265]\t265\t1\n",
      "COV\t[266-266]\t266\t1\n",
      "COV\t[267-267]\t267\t1\n",
      "COV\t[271-271]\t271\t1\n",
      "COV\t[275-275]\t275\t1\n",
      "COV\t[278-278]\t278\t1\n",
      "COV\t[280-280]\t280\t1\n",
      "COV\t[286-286]\t286\t1\n",
      "COV\t[291-291]\t291\t1\n",
      "COV\t[303-303]\t303\t1\n",
      "COV\t[304-304]\t304\t1\n",
      "COV\t[306-306]\t306\t3\n",
      "COV\t[307-307]\t307\t1\n",
      "COV\t[308-308]\t308\t1\n",
      "COV\t[315-315]\t315\t1\n",
      "COV\t[316-316]\t316\t1\n",
      "COV\t[319-319]\t319\t1\n",
      "COV\t[321-321]\t321\t2\n",
      "COV\t[324-324]\t324\t1\n",
      "COV\t[325-325]\t325\t2\n",
      "COV\t[338-338]\t338\t1\n",
      "COV\t[344-344]\t344\t1\n",
      "COV\t[346-346]\t346\t1\n",
      "COV\t[349-349]\t349\t1\n",
      "COV\t[350-350]\t350\t2\n",
      "COV\t[352-352]\t352\t1\n",
      "COV\t[357-357]\t357\t1\n",
      "COV\t[364-364]\t364\t1\n",
      "COV\t[369-369]\t369\t1\n",
      "COV\t[370-370]\t370\t1\n",
      "COV\t[376-376]\t376\t1\n",
      "COV\t[378-378]\t378\t2\n",
      "COV\t[379-379]\t379\t1\n",
      "COV\t[382-382]\t382\t1\n",
      "COV\t[390-390]\t390\t1\n",
      "COV\t[397-397]\t397\t1\n",
      "COV\t[398-398]\t398\t1\n",
      "COV\t[400-400]\t400\t1\n",
      "COV\t[411-411]\t411\t1\n",
      "COV\t[414-414]\t414\t1\n",
      "COV\t[416-416]\t416\t1\n",
      "COV\t[418-418]\t418\t1\n",
      "COV\t[422-422]\t422\t1\n",
      "COV\t[426-426]\t426\t2\n",
      "COV\t[428-428]\t428\t1\n",
      "COV\t[430-430]\t430\t1\n",
      "COV\t[432-432]\t432\t1\n",
      "COV\t[437-437]\t437\t1\n",
      "COV\t[440-440]\t440\t1\n",
      "COV\t[453-453]\t453\t1\n",
      "COV\t[454-454]\t454\t1\n",
      "COV\t[458-458]\t458\t1\n",
      "COV\t[459-459]\t459\t1\n",
      "COV\t[462-462]\t462\t1\n",
      "COV\t[463-463]\t463\t1\n",
      "COV\t[464-464]\t464\t1\n",
      "COV\t[467-467]\t467\t1\n",
      "COV\t[475-475]\t475\t1\n",
      "COV\t[478-478]\t478\t1\n",
      "COV\t[483-483]\t483\t1\n",
      "COV\t[492-492]\t492\t1\n",
      "COV\t[493-493]\t493\t2\n",
      "COV\t[494-494]\t494\t1\n",
      "COV\t[496-496]\t496\t2\n",
      "COV\t[497-497]\t497\t1\n",
      "COV\t[502-502]\t502\t1\n",
      "COV\t[508-508]\t508\t1\n",
      "COV\t[511-511]\t511\t1\n",
      "COV\t[512-512]\t512\t1\n",
      "COV\t[514-514]\t514\t2\n",
      "COV\t[516-516]\t516\t1\n",
      "COV\t[540-540]\t540\t3\n",
      "COV\t[541-541]\t541\t2\n",
      "COV\t[547-547]\t547\t1\n",
      "COV\t[550-550]\t550\t1\n",
      "COV\t[551-551]\t551\t1\n",
      "COV\t[559-559]\t559\t1\n",
      "COV\t[560-560]\t560\t1\n",
      "COV\t[562-562]\t562\t1\n",
      "COV\t[563-563]\t563\t1\n",
      "COV\t[565-565]\t565\t1\n",
      "COV\t[578-578]\t578\t2\n",
      "COV\t[583-583]\t583\t1\n",
      "COV\t[585-585]\t585\t1\n",
      "COV\t[586-586]\t586\t1\n",
      "COV\t[588-588]\t588\t1\n",
      "COV\t[603-603]\t603\t1\n",
      "COV\t[608-608]\t608\t1\n",
      "COV\t[613-613]\t613\t2\n",
      "COV\t[616-616]\t616\t1\n",
      "COV\t[617-617]\t617\t1\n",
      "COV\t[618-618]\t618\t1\n",
      "COV\t[620-620]\t620\t1\n",
      "COV\t[623-623]\t623\t1\n",
      "COV\t[634-634]\t634\t1\n",
      "COV\t[656-656]\t656\t1\n",
      "COV\t[682-682]\t682\t1\n",
      "COV\t[687-687]\t687\t1\n",
      "COV\t[690-690]\t690\t1\n",
      "COV\t[707-707]\t707\t1\n",
      "COV\t[708-708]\t708\t1\n",
      "COV\t[721-721]\t721\t1\n",
      "COV\t[726-726]\t726\t1\n",
      "COV\t[733-733]\t733\t1\n",
      "COV\t[742-742]\t742\t1\n",
      "COV\t[746-746]\t746\t1\n",
      "COV\t[749-749]\t749\t1\n",
      "COV\t[750-750]\t750\t1\n",
      "COV\t[753-753]\t753\t1\n",
      "COV\t[755-755]\t755\t1\n",
      "COV\t[767-767]\t767\t1\n",
      "COV\t[771-771]\t771\t1\n",
      "COV\t[776-776]\t776\t1\n",
      "COV\t[789-789]\t789\t1\n",
      "COV\t[791-791]\t791\t1\n",
      "COV\t[794-794]\t794\t1\n",
      "COV\t[798-798]\t798\t1\n",
      "COV\t[810-810]\t810\t1\n",
      "COV\t[826-826]\t826\t1\n",
      "COV\t[833-833]\t833\t1\n",
      "COV\t[834-834]\t834\t1\n",
      "COV\t[836-836]\t836\t2\n",
      "COV\t[838-838]\t838\t1\n",
      "COV\t[878-878]\t878\t1\n",
      "COV\t[886-886]\t886\t1\n",
      "COV\t[887-887]\t887\t1\n",
      "COV\t[893-893]\t893\t1\n",
      "COV\t[901-901]\t901\t1\n",
      "COV\t[907-907]\t907\t1\n",
      "COV\t[910-910]\t910\t1\n",
      "COV\t[912-912]\t912\t1\n",
      "COV\t[924-924]\t924\t1\n",
      "COV\t[928-928]\t928\t1\n",
      "COV\t[937-937]\t937\t1\n",
      "COV\t[942-942]\t942\t1\n",
      "COV\t[950-950]\t950\t1\n",
      "COV\t[957-957]\t957\t1\n",
      "COV\t[967-967]\t967\t2\n",
      "COV\t[972-972]\t972\t2\n",
      "COV\t[974-974]\t974\t1\n",
      "COV\t[975-975]\t975\t1\n",
      "COV\t[981-981]\t981\t1\n",
      "COV\t[984-984]\t984\t1\n",
      "COV\t[985-985]\t985\t1\n",
      "COV\t[987-987]\t987\t1\n",
      "COV\t[990-990]\t990\t1\n",
      "COV\t[995-995]\t995\t1\n",
      "COV\t[997-997]\t997\t1\n",
      "COV\t[1000<]\t1000\t706\n",
      "# GC-depth. Use `grep ^GCD | cut -f 2-` to extract this part. The columns are: GC%, unique sequence percentiles, 10th, 25th, 50th, 75th and 90th depth percentile\n",
      "GCD\t0.0\t22.222\t0.000\t0.000\t0.000\t0.000\t0.000\n",
      "GCD\t32.0\t33.333\t0.004\t0.004\t0.004\t0.004\t0.004\n",
      "GCD\t34.0\t44.444\t0.076\t0.076\t0.076\t0.076\t0.076\n",
      "GCD\t35.0\t55.556\t0.018\t0.018\t0.018\t0.018\t0.018\n",
      "GCD\t36.0\t77.778\t0.004\t0.004\t0.004\t0.027\t0.027\n",
      "GCD\t40.0\t88.889\t0.014\t0.014\t0.014\t0.014\t0.014\n",
      "GCD\t42.0\t100.000\t0.036\t0.036\t0.036\t0.036\t0.036\n"
     ]
    }
   ],
   "source": [
    "! samtools stats actb.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No input file specified.\n",
      "Usage: samtools fastq [options...] <in.bam>\n",
      "Options:\n",
      "  -0 FILE              write reads designated READ_OTHER to FILE\n",
      "  -1 FILE              write reads designated READ1 to FILE\n",
      "  -2 FILE              write reads designated READ2 to FILE\n",
      "                       note: if a singleton file is specified with -s, only\n",
      "                       paired reads will be written to the -1 and -2 files.\n",
      "  -f INT               only include reads with all  of the FLAGs in INT present [0]\n",
      "  -F INT               only include reads with none of the FLAGS in INT present [0]\n",
      "  -G INT               only EXCLUDE reads with all  of the FLAGs in INT present [0]\n",
      "  -n                   don't append /1 and /2 to the read name\n",
      "  -N                   always append /1 and /2 to the read name\n",
      "  -O                   output quality in the OQ tag if present\n",
      "  -s FILE              write singleton reads designated READ1 or READ2 to FILE\n",
      "  -t                   copy RG, BC and QT tags to the FASTQ header line\n",
      "  -T TAGLIST           copy arbitrary tags to the FASTQ header line\n",
      "  -v INT               default quality score if not given in file [1]\n",
      "  -i                   add Illumina Casava 1.8 format entry to header (eg 1:N:0:ATCACG)\n",
      "  -c                   compression level [0..9] to use when creating gz or bgzf fastq files\n",
      "  --i1 FILE            write first index reads to FILE\n",
      "  --i2 FILE            write second index reads to FILE\n",
      "  --barcode-tag TAG    Barcode tag [default: BC]\n",
      "  --quality-tag TAG    Quality tag [default: QT]\n",
      "  --index-format STR   How to parse barcode and quality tags\n",
      "\n",
      "      --input-fmt-option OPT[=VAL]\n",
      "               Specify a single input file format option in the form\n",
      "               of OPTION or OPTION=VALUE\n",
      "      --reference FILE\n",
      "               Reference sequence FASTA FILE [null]\n",
      "  -@, --threads INT\n",
      "               Number of additional threads to use [0]\n",
      "\n",
      "Reads are designated READ1 if FLAG READ1 is set and READ2 is not set.\n",
      "Reads are designated READ2 if FLAG READ1 is not set and READ2 is set.\n",
      "Reads are designated READ_OTHER if FLAGs READ1 and READ2 are either both set\n",
      "or both unset.\n",
      "Run 'samtools flags' for more information on flag codes and meanings.\n",
      "\n",
      "The index-format string describes how to parse the barcode and quality tags, for example:\n",
      "   i14i8       the first 14 characters are index 1, the next 8 characters are index 2\n",
      "   n8i14       ignore the first 8 characters, and use the next 14 characters for index 1\n",
      "If the tag contains a separator, then the numeric part can be replaced with '*' to mean\n",
      "'read until the separator or end of tag', for example:\n",
      "   n*i*        ignore the left part of the tag until the separator, then use the second part\n",
      "               of the tag as index 1\n",
      "\n",
      "Examples:\n",
      " To get just the paired reads in separate files, use:\n",
      "   samtools fastq -1 paired1.fq -2 paired2.fq -0 /dev/null -s /dev/null -n -F 0x900 in.bam\n",
      "\n",
      " To get all non-supplementary/secondary reads in a single file, redirect the output:\n",
      "   samtools fastq -F 0x900 in.bam > all_reads.fq\n"
     ]
    }
   ],
   "source": [
    "! samtools fastq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Too many arguments.\n",
      "Usage: samtools fastq [options...] <in.bam>\n",
      "Options:\n",
      "  -0 FILE              write reads designated READ_OTHER to FILE\n",
      "  -1 FILE              write reads designated READ1 to FILE\n",
      "  -2 FILE              write reads designated READ2 to FILE\n",
      "                       note: if a singleton file is specified with -s, only\n",
      "                       paired reads will be written to the -1 and -2 files.\n",
      "  -f INT               only include reads with all  of the FLAGs in INT present [0]\n",
      "  -F INT               only include reads with none of the FLAGS in INT present [0]\n",
      "  -G INT               only EXCLUDE reads with all  of the FLAGs in INT present [0]\n",
      "  -n                   don't append /1 and /2 to the read name\n",
      "  -N                   always append /1 and /2 to the read name\n",
      "  -O                   output quality in the OQ tag if present\n",
      "  -s FILE              write singleton reads designated READ1 or READ2 to FILE\n",
      "  -t                   copy RG, BC and QT tags to the FASTQ header line\n",
      "  -T TAGLIST           copy arbitrary tags to the FASTQ header line\n",
      "  -v INT               default quality score if not given in file [1]\n",
      "  -i                   add Illumina Casava 1.8 format entry to header (eg 1:N:0:ATCACG)\n",
      "  -c                   compression level [0..9] to use when creating gz or bgzf fastq files\n",
      "  --i1 FILE            write first index reads to FILE\n",
      "  --i2 FILE            write second index reads to FILE\n",
      "  --barcode-tag TAG    Barcode tag [default: BC]\n",
      "  --quality-tag TAG    Quality tag [default: QT]\n",
      "  --index-format STR   How to parse barcode and quality tags\n",
      "\n",
      "      --input-fmt-option OPT[=VAL]\n",
      "               Specify a single input file format option in the form\n",
      "               of OPTION or OPTION=VALUE\n",
      "      --reference FILE\n",
      "               Reference sequence FASTA FILE [null]\n",
      "  -@, --threads INT\n",
      "               Number of additional threads to use [0]\n",
      "\n",
      "Reads are designated READ1 if FLAG READ1 is set and READ2 is not set.\n",
      "Reads are designated READ2 if FLAG READ1 is not set and READ2 is set.\n",
      "Reads are designated READ_OTHER if FLAGs READ1 and READ2 are either both set\n",
      "or both unset.\n",
      "Run 'samtools flags' for more information on flag codes and meanings.\n",
      "\n",
      "The index-format string describes how to parse the barcode and quality tags, for example:\n",
      "   i14i8       the first 14 characters are index 1, the next 8 characters are index 2\n",
      "   n8i14       ignore the first 8 characters, and use the next 14 characters for index 1\n",
      "If the tag contains a separator, then the numeric part can be replaced with '*' to mean\n",
      "'read until the separator or end of tag', for example:\n",
      "   n*i*        ignore the left part of the tag until the separator, then use the second part\n",
      "               of the tag as index 1\n",
      "\n",
      "Examples:\n",
      " To get just the paired reads in separate files, use:\n",
      "   samtools fastq -1 paired1.fq -2 paired2.fq -0 /dev/null -s /dev/null -n -F 0x900 in.bam\n",
      "\n",
      " To get all non-supplementary/secondary reads in a single file, redirect the output:\n",
      "   samtools fastq -F 0x900 in.bam > all_reads.fq\n",
      "ls: cannot access '*fastq': No such file or directory\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "samtools fastq -1 actb_R1.fastq -2 actb_R2.fastq samtools flagstat actb.bam.fastq -s singleton.fastq -n actb.bam\n",
    "ls -lha *fastq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Tool:    bedtools bamtofastq (aka bamToFastq)\n",
      "Version: v2.28.0\n",
      "Summary: Convert BAM alignments to FASTQ files. \n",
      "\n",
      "Usage:   bamToFastq [OPTIONS] -i <BAM> -fq <FQ> \n",
      "\n",
      "Options:\n",
      "\t-fq2\tFASTQ for second end.  Used if BAM contains paired-end data.\n",
      "\t\tBAM should be sorted by query name is creating paired FASTQ.\n",
      "\n",
      "\t-tags\tCreate FASTQ based on the mate info\n",
      "\t\tin the BAM R2 and Q2 tags.\n",
      "\n",
      "Tips: \n",
      "\tIf you want to create a single, interleaved FASTQ file \n",
      "\tfor paired-end data, you can just write both to /dev/stdout:\n",
      "\n",
      "\tbedtools bamtofastq -i x.bam -fq /dev/stdout -fq2 /dev/stdout > x.ilv.fq\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!bedtools bamtofastq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "! samtools view actb.bam | cut -f 1 > actb_read_ids.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-rw-rw-r-- 1 olga olga 0 Aug  8 06:02 actb_R1.fastq\n",
      "-rw-rw-r-- 1 olga olga 0 Aug  8 06:02 actb_R2.fastq\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "bedtools bamtofastq -fq actb_R1.fastq -fq2 actb_R2.fastq -i actb.bam\n",
    "ls -lha *fastq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "! aws s3 sync --exclude \"*\" --include \"S10*_R*fastq.gz\" s3://olgabot-maca/bingwu-test-data/rawdata/ ."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "A00111:245:HGKNTDMXX:1:2257:5629:23390\n",
      "A00111:245:HGKNTDMXX:1:1476:10818:34695\n",
      "A00111:245:HGKNTDMXX:1:1417:28411:10175\n",
      "A00111:245:HGKNTDMXX:1:2227:31891:33395\n",
      "A00111:245:HGKNTDMXX:1:2301:9733:36448\n",
      "A00111:245:HGKNTDMXX:1:1215:22761:24721\n",
      "A00111:245:HGKNTDMXX:1:1474:10411:10473\n",
      "A00111:245:HGKNTDMXX:1:2381:15040:11600\n",
      "A00111:245:HGKNTDMXX:1:1350:26621:13964\n",
      "A00111:245:HGKNTDMXX:1:2286:28809:32722\n"
     ]
    }
   ],
   "source": [
    "! head actb_read_ids.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "@A00111:245:HGKNTDMXX:1:1101:3206:1000 1:N:0:NGGAGATG\n",
      "CGAACATCACGCATCGCGTCACAGGT\n",
      "+\n",
      "FFFFFFFFFFFFFFFFFFFFFFFFFF\n",
      "@A00111:245:HGKNTDMXX:1:1101:3893:1000 1:N:0:NGGAGATG\n",
      "TTGCCTCTGCATATTGTTCCTGATTG\n",
      "+\n",
      "FFFFFFFFFFFFFFFFFFFFFFFFFF\n",
      "@A00111:245:HGKNTDMXX:1:1101:3929:1000 1:N:0:NGGAGATG\n",
      "GATGCTACACAGGCCTTCATTGCTTT\n",
      "\n",
      "gzip: stdout: Broken pipe\n"
     ]
    }
   ],
   "source": [
    "! zcat S10_L001_R1_001.fastq.gz |head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Usage: grep [OPTION]... PATTERN [FILE]...\n",
      "Search for PATTERN in each FILE or standard input.\n",
      "PATTERN is, by default, a basic regular expression (BRE).\n",
      "Example: grep -i 'hello world' menu.h main.c\n",
      "\n",
      "Regexp selection and interpretation:\n",
      "  -E, --extended-regexp     PATTERN is an extended regular expression (ERE)\n",
      "  -F, --fixed-strings       PATTERN is a set of newline-separated strings\n",
      "  -G, --basic-regexp        PATTERN is a basic regular expression (BRE)\n",
      "  -P, --perl-regexp         PATTERN is a Perl regular expression\n",
      "  -e, --regexp=PATTERN      use PATTERN for matching\n",
      "  -f, --file=FILE           obtain PATTERN from FILE\n",
      "  -i, --ignore-case         ignore case distinctions\n",
      "  -w, --word-regexp         force PATTERN to match only whole words\n",
      "  -x, --line-regexp         force PATTERN to match only whole lines\n",
      "  -z, --null-data           a data line ends in 0 byte, not newline\n",
      "\n",
      "Miscellaneous:\n",
      "  -s, --no-messages         suppress error messages\n",
      "  -v, --invert-match        select non-matching lines\n",
      "  -V, --version             display version information and exit\n",
      "      --help                display this help text and exit\n",
      "\n",
      "Output control:\n",
      "  -m, --max-count=NUM       stop after NUM matches\n",
      "  -b, --byte-offset         print the byte offset with output lines\n",
      "  -n, --line-number         print line number with output lines\n",
      "      --line-buffered       flush output on every line\n",
      "  -H, --with-filename       print the file name for each match\n",
      "  -h, --no-filename         suppress the file name prefix on output\n",
      "      --label=LABEL         use LABEL as the standard input file name prefix\n",
      "  -o, --only-matching       show only the part of a line matching PATTERN\n",
      "  -q, --quiet, --silent     suppress all normal output\n",
      "      --binary-files=TYPE   assume that binary files are TYPE;\n",
      "                            TYPE is 'binary', 'text', or 'without-match'\n",
      "  -a, --text                equivalent to --binary-files=text\n",
      "  -I                        equivalent to --binary-files=without-match\n",
      "  -d, --directories=ACTION  how to handle directories;\n",
      "                            ACTION is 'read', 'recurse', or 'skip'\n",
      "  -D, --devices=ACTION      how to handle devices, FIFOs and sockets;\n",
      "                            ACTION is 'read' or 'skip'\n",
      "  -r, --recursive           like --directories=recurse\n",
      "  -R, --dereference-recursive  likewise, but follow all symlinks\n",
      "      --include=FILE_PATTERN  search only files that match FILE_PATTERN\n",
      "      --exclude=FILE_PATTERN  skip files and directories matching FILE_PATTERN\n",
      "      --exclude-from=FILE   skip files matching any file pattern from FILE\n",
      "      --exclude-dir=PATTERN  directories that match PATTERN will be skipped.\n",
      "  -L, --files-without-match  print only names of FILEs containing no match\n",
      "  -l, --files-with-matches  print only names of FILEs containing matches\n",
      "  -c, --count               print only a count of matching lines per FILE\n",
      "  -T, --initial-tab         make tabs line up (if needed)\n",
      "  -Z, --null                print 0 byte after FILE name\n",
      "\n",
      "Context control:\n",
      "  -B, --before-context=NUM  print NUM lines of leading context\n",
      "  -A, --after-context=NUM   print NUM lines of trailing context\n",
      "  -C, --context=NUM         print NUM lines of output context\n",
      "  -NUM                      same as --context=NUM\n",
      "      --color[=WHEN],\n",
      "      --colour[=WHEN]       use markers to highlight the matching strings;\n",
      "                            WHEN is 'always', 'never', or 'auto'\n",
      "  -U, --binary              do not strip CR characters at EOL (MSDOS/Windows)\n",
      "  -u, --unix-byte-offsets   report offsets as if CRs were not there\n",
      "                            (MSDOS/Windows)\n",
      "\n",
      "'egrep' means 'grep -E'.  'fgrep' means 'grep -F'.\n",
      "Direct invocation as either 'egrep' or 'fgrep' is deprecated.\n",
      "When FILE is -, read standard input.  With no FILE, read . if a command-line\n",
      "-r is given, - otherwise.  If fewer than two FILEs are given, assume -h.\n",
      "Exit status is 0 if any line is selected, 1 otherwise;\n",
      "if any error occurs and -q is not given, the exit status is 2.\n",
      "\n",
      "Report bugs to: bug-grep@gnu.org\n",
      "GNU grep home page: <http://www.gnu.org/software/grep/>\n",
      "General help using GNU software: <http://www.gnu.org/gethelp/>\n"
     ]
    }
   ],
   "source": [
    "! grep --help"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "@A00111:245:HGKNTDMXX:1:1101:3893:1000 1:N:0:NGGAGATG\n",
      "TTGCCTCTGCATATTGTTCCTGATTG\n",
      "+\n",
      "FFFFFFFFFFFFFFFFFFFFFFFFFF\n"
     ]
    }
   ],
   "source": [
    "! zgrep -A 3 '@A00111:245:HGKNTDMXX:1:1101:3893:1000' S10_L001_R1_001.fastq.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "@A00111:245:HGKNTDMXX:1:1101:15962:2300 1:N:0:AGGAGATG\n",
      "CTAAGACCAATAGCAATTTTGTCGGT\n",
      "+\n",
      "FFFFFFFFFFFFFFFFFFFFFFFFF:\n",
      "--\n",
      "@A00111:245:HGKNTDMXX:1:1101:10402:3004 1:N:0:TGGAGATG\n",
      "CGTTAGATCATCTGTTGACAAAGTTG\n",
      "+\n",
      ",FF::FFFFFF,FFFFFFF,FF:FFF\n",
      "--\n",
      "/bin/grep: write error: Broken pipe\n",
      "\n",
      "gzip: stdout: Broken pipe\n"
     ]
    }
   ],
   "source": [
    "! zgrep -A 3 --file actb_read_ids.txt S10_L001_R1_001.fastq.gz | head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Usage: /bin/zgrep [OPTION]... [-e] PATTERN [FILE]...\n",
      "Look for instances of PATTERN in the input FILEs, using their\n",
      "uncompressed contents if they are compressed.\n",
      "\n",
      "OPTIONs are the same as for 'grep'.\n",
      "\n",
      "Report bugs to <bug-gzip@gnu.org>.\n"
     ]
    }
   ],
   "source": [
    "! zgrep --help"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "! zgrep -A 3 --file actb_read_ids.txt S10_L001_R1_001.fastq.gz | gzip -c - > actb_R1.fastq.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "! zgrep -A 3 --file actb_read_ids.txt S10_L001_R2_001.fastq.gz | gzip -c - > actb_R2.fastq.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "ls -lha *fastq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
